#Preparing for Data Downloads and Uploads

## Overview

The GDC Data Transfer Tool is intended to be used in conjunction with the [GDC Data Portal](https://portal.gdc.cancer.gov) and the [GDC Data Submission Portal](https://portal.gdc.cancer.gov/submission/) to transfer data to or from the GDC. First, the GDC Data Portal&#39;s interface is used to generate a manifest file or obtain UUID(s) and (for Controlled-Access Data) an authentication token. The GDC Data Transfer Tool is then used to transfer the data files listed in the manifest file or identified by UUID(s).

##Downloads
### Obtaining a Manifest File for Data Download

The GDC Data Transfer Tool supports downloading multiple files listed in a GDC manifest file. Manifest files can be generated and downloaded directly from the GDC Data Portal:

First, select the data files of interest. Click the *Cart* button in the row corresponding to the file desired. The button will turn blue to indicate that the file has been selected.

![GDC Data Portal: Selecting Files of Interest](images/06-20_Data-Portal-File-Selection.png "Selecting Files of Interest")


Once all files of interest have been selected, click on the *Cart* button in the upper right-hand corner. This will bring up the cart page, which provides an overview of all currently selected files. This list of files can be downloaded as a manifest file by clicking on the *Download Cart* button and selecting *Manifest* from the drop down.

![GDC Data Portal: Cart Page](images/06-20-v2_Data-Portal-Cart-Page.png)

### Obtaining UUIDs for Data Download

A manifest file is not required to download files from GDC. The GDC Data Transfer Tool will accept file UUID(s) instead of a manifest file for downloading individual data files. To obtain a data file's UUID from the GDC Data Portal, click the file name to find its detail page including its GDC UUID.

![GDC Data Portal: Detailed File Page](images/06-20_Data-portal-file-detail-pagev2.png)


### Obtaining an Authentication Token for Data Downloads

The GDC Data Transfer Tool requires an authentication token to download from GDC data portal to download Controlled-Access Data. Tokens can be generated and downloaded directly from the GDC Data Portal.

To generate a token, first log in to the GDC Data Portal by clicking the *Login* button in the top right corner of the page. This will redirect to the eRA Commons login page. After successful authentication, the GDC Data Portal will display the username in place of the *Login* button. Here, the user Ian Miller is logged in to the GDC Data Portal, indicated by the username IANMILLER.

Clicking the username will open a drop-down menu. Select *Download Token* from the menu to generate an authentication token.

![GDC Data Portal User Dropdown Menu](images/06-20_auth_example_download_token-2.png)
<!---**GDC Data Portal User Dropdown Menu**--->

**NOTE:** The authentication token should be kept in a secure location, as it allows access to all data accessible by the associated user.

##Uploads
### Obtaining a Manifest File for Data Uploads
Multiple data file uploads are supported by the GDC Data Transfer Tool via a manifest file. Manifest files can be generated and downloaded directly from the GDC Submission Portal.  A project's manifest file can be downloaded from the projects's dashboard.  

![GDC Submission Portal Manifest Download](images/10-10-16_manifest_upload.png)
<!---**GDC Submission Portal Manifest Download**--->

**NOTE:** To download a project's manifest file click on the _Download Manifest_ button located on the home page of the project, just below the four status charts.  A manifest will be generated for the entire project or if previous files have already been upload only the files that remain to be uploaded.  

A manifest for individual files can also be downloaded from the transaction tab and browse tab pages of the submission portal's project.  More information on the process can be found under the Submission Portal's documentation section entitled [Uploading the Submittable Data File to the GDC](https://docs.gdc.cancer.gov/Data_Submission_Portal/Users_Guide/Data_Submission_Walkthrough/#uploading-the-submittable-data-file-to-the-gdc).

### Obtaining UUIDs for Data Uploads
A UUID can be used for data submission with the Data Transfer Tool.  The UUID for submittable data uploads can be obtained from the Submission Portal or from the API GraphQL endpoint.  In the Submission Portal the UUID for a data file can be found in the Manifest YAML file located in the _id:_ row located under the file size entry.

![Submission Manifest yaml file](images/10-18_yaml_submission_UUID_example.png)


 A second location to obtain a UUID in the Submission Portal is on the Browse Tab page. Under the Submittable Data Files section a UUID can be found by opening up the file's detail page. By clicking on the Submitter ID of the upload file a new window will display a Summary of the file's details, which contains the UUID.

![Submission Portal Browse Page Details](images/submission_portal_browse_page_UUID.png)  

###GraphQL

A UUID can be obtained from the API GraphQL endpoint.  An overview of what GraphQL and its uses is located on the API documentation page section [Querying Submitted Data Using GraphQL](https://docs.gdc.cancer.gov/API/Users_Guide/Submission/#querying-submitted-data-using-graphql)

The following example will query the endpoint to produce a UUID along with submitter_id, file_name, and project_id.

=== "GraphQl_Bare"

    ```GraphQL
    {
      submitted_unaligned_reads (project_id: "GDC-INTERNAL", submitter_id: "Blood-00001-aliquot_lane1_barcode23.fastq") {
       id
       submitter_id
       file_name
       project_id
    }
    }
    ```

=== "Escaped_Json"
    
    ```json
    {\n  submitted_unaligned_reads (project_id: \"GDC-INTERNAL\", submitter_id: \"Blood-00001-aliquot_lane1_barcode23.fastq\") {\n   id\n   submitter_id\n   file_name\n   project_id\n}\n}\n
    
    ```

=== "Query_json"

    ```json
    {
            "query": "{\n \n  submitted_unaligned_reads (project_id: \"GDC-INTERNAL\", submitter_id: \"Blood-00001-aliquot_lane1_barcode23.fastq\") {\n    id\n    submitter_id\n    file_name\n    project_id\n}\n}",
            "variables": null
    }
    ```

=== "Shell_command"

    ```Shell
    export token=ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTO
    $ curl --request POST --header "X-Auth-Token: $token" 'https://api.gdc.cancer.gov/v0/submission/graphql' -d@data.json
    ```

=== "API_Response"

    ```json
    {
      "data": {
        "submitted_unaligned_reads": [
          {
            "file_name": "dummy.fastq",
            "id": "616eab2f-791a-4641-8cd6-ee195a10a201",
            "project_id": "GDC-INTERNAL",
            "submitter_id": "Blood-00001-aliquot_lane1_barcode23.fastq"
          }
        ]
      }
    ```

### Obtaining an Authentication Token for Data Uploads
While biospecimen and clinical metadata may be uploaded via the GDC Data Submission Portal, file upload must be done using the Data Transfer Tool or API.  An authentication token is required for data upload and can be generated on the GDC Data Submission Portal.

To generate a token, first log in to the GDC Data Submission Portal by clicking the *Login* button in the top right corner of the page.  This will create a popup window that will redirect to the eRA Commons login page. After successful authentication, the GDC Submission Portal will display the username in place of the *Login* button. Here, the user Ian Miller is logged in to the GDC Submission Portal, indicated by the username IANMILLER.

Clicking the username will open a drop-down menu. Select *Download Token* from the menu to generate an authentication token.

![GDC Submission Portal User Dropdown Menu](images/10-27_Submission_Portal_Auth_Download_Tab.png)
<!---**GDC Submission Portal User Dropdown Menu**--->


#Data Downloads with the Data Transfer Tool UI

##Data Transfer Tool UI: Overview
The UI version of the Data Transfer Tool was created for users who prefer a graphical interface over the command line or have limited command line experience.  The command line version is recommended for those users with more command line experience, require large data transfers of GDC data, or need to download a large numbers of data files.

###System Recommendations

The system recommendations for using the GDC Data Transfer Tool are as follows:

* OS: Linux (Ubuntu 14.x or later), OS X (10.9 Mavericks or later), or Windows (7 or later)
* CPU: At least four 64-bit cores, Intel or AMD
* RAM: At least 2 GiB
* Storage: Enterprise-class storage system capable of at least 1 Gb/s (gigabit per second) write throughput and sufficient free space for BAM files.

###Binary Distributions

Binary distributions are available on the GDC Transfer Tool page. To install the GDC Data Transfer UI download the respective binary distribution and unzip the distribution's archive to a location on the target system that is easily accessible.

###Binary Installation
Once the binary has been positioned in an appropriate location on the client's file system the application will need to run though a one-time installation process.  On first execution the binary install splash screen will appear showing the progress of the installation.  A hidden directory is created within the user's home directory labeled dtt that holds configuration and executable files.

![GDC DTT UI Installation](images/GDC_DTT_UI_INSTALLv7.png "GDC Data Transfer Tool UI Install")


###Preparing for Data Download

The GDC Data Transfer Tool UI is a stand-alone client application intended to work with data file information stored on the GDC Data Portals.  Data download information must first be gathered from the GDC Data Portal.  From there a manifest file can be [generated](https://docs.gdc.cancer.gov/Data_Transfer_Tool/Users_Guide/Preparing_for_Data_Download_and_Upload/#obtaining-a-manifest-file-for-data-download) to supply the client.  Alternatively, individual file UUIDs can be provided to the UUID entry window located on the Download tab in the client.

![GDC DTT UI Start Page](images/DTT_UI_Start_Page.png)
##Downloads with UUIDs
The Data Transfer Tool UI can download files by individual UUID.  UUIDs can be entered into the client while on the download tab.  The single entry field labeled "Enter UUID(s)" allows the user to enter UUIDs individually.     

To obtain a data file's UUID from the GDC Data Portal, click on the file name to display the file's summary page which includes vital information such as its GDC UUID.  

##Downloads with Manifest
A portal-generated manifest file can be used with the Data Transfer Tool UI.  From the Download tab home page click on the Select Manifest File button.  A file system search window will popup allowing navigation to the manifest file.  

![GDC DTT UI Manifest Button Example](images/Manifest_button_DTT_UI_Start_Window.png "GDC Data Transfer Tool UI Manifest Button")     

##Download Progress Page  
The Download Progress Page is the command console for the Data Transfer Tool UI and allows users to monitor downloads. Progress of all downloads including the ability to start, stop, and restart a download are performed on the Download Progress Page.  Once file UUIDs or a manifest has been added to the queue the download can be started by clicking on the download button located at the lower right hand side of the page.  

![GDC DTT UI Download Progress Page_Download](images/Download_Progress_Page_download.png)

Once a download has completed, information about the downloads can be viewed from the Completed tab located at the bottom of the page.  Any Stopped or Failed downloads can also be viewed from their respective labels located at the bottom of the Status page.  

![DTT_UI_Download_Completed_Tab](images/DTT_UI_download_completed.png)

##Controlled Access File Downloads

Some files in the GDC are controlled access.  If you require access to these files please review the process outlined in the documentation [Obtaining Access to Controlled Data](https://gdc.cancer.gov/access-data/obtaining-access-controlled-data).    After appropriate authorization has been granted an access token can be generated to allow the Data Transfer Tool UI application access to the requested data files.  Documentation explaining the process of generating a token is located in the [Obtaining an Authentication Token for Data Downloads](https://docs.gdc.cancer.gov/Data_Transfer_Tool/Users_Guide/Preparing_for_Data_Download_and_Upload/#obtaining-an-authentication-token-for-data-downloads).  Once a token has been downloaded to a secure location on the client's local filesystem the Data Transfer Tool UI can now access it.  

![No_Token_icon](images/No_Token_file_dtt_ui.png)  

The current status of client authorization is viewable in the upper right corner of the application.  If the image and wording on the token manager access button is in red then no valid client token file has been uploaded.  To upload a valid token file click on the token status button.  The token manager window should appear allowing either a drag and drop token file upload or a file navigation window can be opened to navigate to the file location.

![Token Manager Window](images/Token_Manager_Window.png)

The token manager will verify access and display the projects for which the user has access.  To complete the token upload process click on the save button within the Token Manager window.

![Valid Token](images/validated_token.png)

##Settings and Advanced Settings

While the default download options will work for the majority of use cases, there are a vareity of ways to customize or modify the download process within the DTT UI.  Details of each of the settings are listed below.

![DTT Settings and Advanced Settings Page](images/DTT_Settings_Page.png)   

| Settings | Details
|----------|---------|
| Server URL | Default: https://api.gdc.cancer.gov |
| Number of Client Connections: Default (3) | Number of concurrent client threads |
| Destination Folder: Default (User's Home Directory) | User selectable download file location |  
| Calculate Inbound Segment and check Md5sum on Restart: Default (On) | Verify previous partial downloaded files via segment check sum |
| Calculates check sums on previous downloaded files Default (On) | Verify downloaded files via file level check sum |
| Save Logs: - Download Navigation windows for client downloads | Export download or token log files via drop down and export log button |
| Debug Logging: Default (Off) | Enable debug level logging for file downloads |
| Block Size (Bytes): Default (1048576) | HTTP chunk size transfers |
| Save Interval (Bytes): Default (1000000) | save interval in bytes |
| Auto Retry: Default (On) | Enables auto retries of failed downloads |
| Retry(s): Default (5) | Number of retry attempts to download a file after failure |
| Seconds between Retrys: Default (5) | Number of seconds between retires |  


# Troubleshooting Guide

If you encounter issues when using the Data Transfer Tool for downloading files please reference the section below for helpful hints and recommendations.  

## Speed Performance During Download
The Data Transfer Tool has two performance tuning options that are presented during download operations.  The two options are:

* -n OR --n-processes  - The "-n" OR "--n-processes" option assists with assigning the number of threads to the download process.  The default is 4 and can not be lowered below three threads.

* --http-chunk-size - The "--http-chunk-size" setting can improve performance but we do not provide any hard settings due to the eclectic nature of client networks and their connections to the internet but instead encourage clients to experiment with changing the default setting of 1048576 bytes to larger size ranges.       

## Very Large Manifests
Some clients have needed to create very large manifest files to satisfy the scope of their work.  Using very large manifest files can from time to time lead to the end user experiencing  network time outs or dropped connections due to network topologies, internet connections, or load on client side systems.  Network time out or dropped network connect can manifest as a hung or unresponsive download session. To help mitigate these issues we recommended that clients breakup their manifest files into smaller chunks.  

## General Tips
* To avoid running into older software bugs/conflicts we recommend you always use the latest version of the client whenever possible.  
* When experiencing download problems while using an access tokens try downloading a new token first before reporting it to the GDC Help Desk.  

## Logging
For troubleshooting purposes the GDC User Services Team may request that you run the command line application with the following flags { --debug --log-file }.  These flags will run the application in debug mode and create a logfile file with the debug logs in it.  
Example Usage:

=== "Debug-Logfile"

    ```shell
    gdc-client download -m lung.manifest.txt -t token.file --debug --log-file logfile.txt
    ```

## OS Compatibility with the Data Transfer Tool
The Data transfer Tool is offered in three OS compatible versions; Mac OS, Windows, and Ubuntu Linux.  We have successfully tested the Ubuntu binary on CentOS 7.x and 8.x, RHEL 7/8 and Scientific Linux 7/8 with the client but have had problems with CentOS 6.x and RHEL 6 and SL6.  To work around this problem we have asked users to build their own client from our [github](https://github.com/NCI-GDC/gdc-client) repository with the assistance of an instruction document that we provide on request via the GDC Helpdesk.        

## Network Troubleshooting

Network problems can appear as dropped network connections or even a stalled application.  The GDC Helpdesk might request more network information to assist in diagnosing the problem.  The two tests they will request the end user to run are ping and traceroute (tracert on the windows platform) against our api servers.  Please capture the output from these tests into a text file and attach it to the reply email.   

Examples:

=== "Ping"

    ```shell
    >ping api.gdc.cancer.gov
     PING api.gdc.cancer.gov (192.170.230.246): 56 data bytes
     64 bytes from 192.170.230.246: icmp_seq=0 ttl=249 time=4.235 ms
     64 bytes from 192.170.230.246: icmp_seq=1 ttl=249 time=4.783 ms
    ```

=== "Traceroute"

    ```shell
    >traceroute api.gdc.cancer.gov
     traceroute to api.gdc.cancer.gov (192.170.230.246), 64 hops max, 52 byte packets
     1  h01-391-250-v1011.gw.uchicago.net (10.151.0.2)  4.595 ms  3.602 ms  3.322 ms
     2  h01-391-250-to-b65-ll129-300.p2p.uchicago.net (10.5.1.32)  13.285 ms  9.241 ms  5.156 ms
     3  b65-ll129-300-to-borderfw.p2p.uchicago.net (192.170.192.32)  3.218 ms  3.364 ms  3.396 ms
     4  borderfw-to-b65-ll129-500.p2p.uchicago.net (192.170.192.36)  3.605 ms  3.741 ms  3.833 ms
     5  b65-scidmz-01-to-b65-ll129-500.uchicago.net (128.135.247.182)  4.223 ms  5.428 ms  3.999 ms
     6  192.170.224.97 (192.170.224.97)  4.003 ms  3.970 ms  6.260 ms
     7  lnk-g30-scidist-01.scidmz.uchicago.net (192.170.224.66)  4.530 ms  4.649 ms  6.021 ms
     8  192.170.230.246 (192.170.230.246)  4.158 ms  4.273 ms  5.134 ms
    ```

## Common Error Codes
This is a list of the most common error codes the Data Transfer Tool generates and their meaning
<ul TYPE="square">
<li> Unable to connect to API - you might be running an out of date client so consider upgrading.</li>
<li> Error: Max Retries Exceeded - network connect timeouts </li>
<li>CryptographyDeprecationWarning - a warning that you should consider upgrading to a higher   
  version of python - please upgrade to 2.7.x or higher.</li>
<li>ERROR: An unexpected error has occurred during normal operation of the client -	This could  be a variety of problems and we ask you to contact our helpdesk.</li>
<li>ECONNRESET - network connection dropped.</li>


#Data Downloads from the UI

##Data Transfer Tool UI: Overview
The UI version of the Data Transfer Tool was created for users who prefer a graphical interface over the command line or have no command line experience.  For more with more command line experience who require large data transfers of GDC data or need to download a large numbers of data files the command line version is recommend.

###System Recommendations

The system recommendations for using the GDC Data Transfer Tool are as follows:

* OS: Linux (Ubuntu 14.x or later), OS X (10.9 Mavericks or later), or Windows (7 or later)
* CPU: At least four 64-bit cores, Intel or AMD
* RAM: At least 2 GiB
* Storage: Enterprise-class storage system capable of at least 1 Gb/s (gigabit per second) write throughput and sufficient free space for BAM files.

Binary Distributions

Binary distributions are available on the GDC Transfer Tool page. To install the GDC Data Transfer UI Software download the respective binary distribution and unzip the distribution's archive to a location on the target system that is easily accessible.

###Binary Installation
Once the binary has been positioned in an appropriate location on the client's file system the application will need to run though a one time installation process.  On first execution the binary install splash screen will appear showing the progress of the installation.  A hidden directory is created within the user home directory labled dtt that holds configuration and executable files.
![GDC DTT UI Installation](images/GDC_DTT_UI_INSTALLv7.png "GDC Data Transfer Tool UI Install")


###Preparing for Data Download

The GDC Data Transfer Tool UI is a stand alone client application intended to work with data file information stored on the GDC Data Portals.  Data download information must first be gathered from the GDC Portal.  From their a manifest file can be [generate](https://docs.gdc.cancer.gov/Data_Transfer_Tool/Users_Guide/Preparing_for_Data_Download_and_Upload/#obtaining-a-manifest-file-for-data-download) to supply the client or individual file UUIDs can be looked up and added to the UUIDs entry window located on the Download tab in the client.

![GDC DTT UI Start Page](images/DTT_UI_Start_Page.png)

##Downloads with UUIDs
The Data Transfer Tool UI can download files by individual UUID.  UUIDs can be entered into the client while on the download tab.  The single entry field labeled "Enter UUID(s)"

To obtain a data file's UUID from the GDC Data Portal, click the file name to find its detail page including its GDC UUID.


###Data Transfer Tool Configuration File  
The DTT has the ability to save and reuse configuration parameters in the format of a flat text file via a command line argument.  A simple text file needs to be created first with an extension of either txt or dtt.  The supported section headers are upload and download which can be used independently of each other or used in the same configuration file.  Each section header corresponds to the main functions of the application which are to either download data from the GDC portals or to upload data to the submission system of the GDC.  The configurable parameters are those listed in the help menus under either [download](https://docs.gdc.cancer.gov/Data_Transfer_Tool/Users_Guide/Accessing_Built-in_Help/#download-help-menu) or [upload](https://docs.gdc.cancer.gov/Data_Transfer_Tool/Users_Guide/Accessing_Built-in_Help/#upload-help-menu).            


Example usage:

    gdc-client download d45ec02b-13c3-4afa-822d-443ccd3795ca --config my-dtt-config.dtt

Example of configuration file:

    [upload]
    path = /some/upload/path
    upload_part_size = 1073741824


    [download]
    dir = /some/download/path
    http_chunk_size = 2048
    retry_amount = 6


###Display Config Parameters
This command line flag can be used with either the download or upload application feature to display what settings are active within a custom data transfer tool configuration file.  

    gdc-client settings download --config my-dtt-config.dtt
    [download]
    no_auto_retry = False
    no_file_md5sum = False
    save_interval = 1073741824
    http_chunk_size = 2048
    server = http://exmple-site.com
    n_processes = 8
    no_annotations = False
    no_related_files = False
    retry_amount = 6
    no_segment_md5sum = False
    manifest = []
    wait_time = 5.0
    no_verify = True
    dir = /some/download/path


# Getting Started

## The GDC Data Transfer Tool: An Overview

Raw sequence data, stored as BAM files, make up the bulk of data stored at the NCI Genomic Data Commons (GDC). The size of a single file can vary greatly. Most BAM files stored in the GDC are in the 50 MB - 40 GB size range, with some of the whole genome BAM files reaching sizes of 200-300 GB.

The GDC Data Transfer Tool, a command-line driven application, provides an optimized method of transferring data to and from the GDC and enables resumption of interrupted transfers.

## Downloading the GDC Data Transfer Tool

### System Recommendations

The system recommendations for using the GDC Data Transfer Tool are as follows:

* OS: Linux (Ubuntu 16.x or later), OS X (10.9 Mavericks or later), or Windows (8 or later)
* CPU: At least two 64-bit cores, Intel or AMD
* RAM: At least 8 GiB
* Storage: Enterprise-class storage system capable of at least 1 Gb/s (gigabit per second) write throughput and sufficient free space for BAM files.

### Binary Distributions

Binary distributions are available on the [GDC Transfer Tool page](https://gdc.cancer.gov/access-data/gdc-data-transfer-tool). To install the GDC Data Transfer Tool, download the respective binary distribution and unzip the distribution's archive to a location on the target system. It is recommended that the binary be copied to a located that is in the user's path so that is it accessible from any location within the terminal or command prompt.   

### Release Notes

Release Notes are available on the [GDC Data Transfer Tool Release Notes](https://docs.gdc.cancer.gov/Data_Transfer_Tool/Release_Notes/DTT_Release_Notes) Page.


#Data Transfer Tool Command Line Documentation



## Downloads

### Downloading Data Using a Manifest File

A convenient way to download multiple files from the GDC is to use a manifest file generated by the GDC Data Portal. After generating a manifest file (see [Preparing for Data Download and Upload](Preparing_for_Data_Download_and_Upload.md) for instructions), initiate the download using the GDC Data Transfer Tool by supplying the **-m** or **--manifest** option, followed by the location and name of the manifest file. OS X users can drag and drop the manifest file into Terminal to provide its location.

The following is an example of a command for downloading files from GDC using a manifest file:

	gdc-client download -m  /Users/JohnDoe/Downloads/gdc_manifest_6746fe840d924cf623b4634b5ec6c630bd4c06b5.txt

### Downloading Data Using GDC File UUIDs

The GDC Data Transfer Tool also supports downloading of one or more individual files using UUID(s) instead of a manifest file. To do this, enter the UUID(s) after the download command:

	gdc-client download 22a29915-6712-4f7a-8dba-985ae9a1f005

Multiple UUIDs can be specified, separated by a space:

	gdc-client download e5976406-473a-4fbd-8c97-e95187cdc1bd fb3e261b-92ac-4027-b4d9-eb971a92a4c3

### Resuming a Failed Download

The GDC Data Transfer Tool supports resumption of interrupted downloads. To resume an incomplete download, repeat the download of the manifest or UUID(s) in the same folder as the initial download.  Failed downloads will appear in the destination folder with a .partial extension. This feature allows users the ability to identify quickly where the download stopped.  For large downloads this feature can let the user identify where the download was interrupted  and edit the manifest accordingly.

	gdc-client download f80ec672-d00f-42d5-b5ae-c7e06bc39da1

### Download Latest Version of a File
The GDC Data Transfer Tool supports file versioning.  Our backend data storage supports multiple file versions so older and current versions can be accessible to our users.  For information about accessing file versioning information with our API and finding older UUID information from current UUIDs please check out the [the API User Guide](https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#example-of-retrieving-file-version-information) section in our API documentation.  When working with older manifests or older lists of UUIDs the latest version of a file can always be download with the --latest flag.   

=== "Shell"

    ```Shell
    gdc-client download 426de656-7e34-4a49-b87e-6e2563fa3cdd --latest -t gdc-user-token.2018.txt
    ```

=== "Output"

    ```
    Downloading LATEST versions of files
    Latest version for 426de656-7e34-4a49-b87e-6e2563fa3cdd ==> 6633bfbd-87f1-4d3a-a475-7ad1e8c2017a
    100% [#############################################################################################################################] Time: 0:01:16  14.10 MB/s
    Successfully downloaded: 1
    ```

### Downloading Controlled-Access Data

A user authentication token is required for downloading Controlled-Access Data from GDC. Tokens can be obtained from the GDC Data Portal (see instructions in [Obtaining an Authentication Token](Preparing_for_Data_Download_and_Upload.md#obtaining-an-authentication-token)). Once downloaded, the token *file* can be passed to the GDC Data Transfer Tool using the **-t** or **--token-file** option:

	gdc-client download -m gdc_manifest_e24fac38d3b19f67facb74d3efa746e08b0c82c2.txt -t gdc-user-token.2015-06-17T09-10-02-04-00.txt


### Directory structure of downloaded files

The directory in which the files are downloaded will include folders named by the file UUID. Inside these folders, along with the the data and zipped metadata or index files, will exist a logs folder. The logs folder contains state files that insure that downloads are accurate and allow for resumption of failed or prematurely stopped downloads.  While a download is in progress a file will have a .partial extension.  This will also remain if a download failed.  Once a file is finished downloading the extension will be removed.  If an identical manifest is retried another attempt will be made to download files containing a .partial extension.  

	C501.TCGA-BI-A0VR-10A-01D-A10S-08.5_gdc_realn.bam.partial  logs 	 

## Uploads

### Uploading Data Using a Manifest File

GDC Data Transfer Tool supports uploading molecular data using a manifest file to the Data Submission Portal. The manifest file for submittable data files can be retrieved from the GDC Data Submission Portal, or directly from the GDC Submission API given a submittable data file UUID. The user authentication token file needs to be specified using the **-t** or **--token-file** option.

First, generate an upload manifest, either using the GDC Data Submission Portal, or [using a call](/API/Users_Guide/Submission.md#upload-manifest) to the GDC Submission API `manifest` endpoint (as in the following example):

=== "Manifest"

    ```shell
    export token=ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTO

    curl --header "X-Auth-Token: $token" 'https://api.gdc.cancer.gov/submission/CGCI/BLGSP/manifest?ids=460ad2fe-5a7f-4797-9e18-336d33e21444' >manifest.yml
    ```

=== "Upload"

    ```shell
    gdc-client upload --manifest manifest.yml --token-file token.txt
    ```

### Uploading Data Using a GDC File UUID

The GDC Data Transfer Tool also supports uploading molecular data using a file UUID. The tool will first make a request to get the filename and project id from GDC API, and then upload the corresponding file from the current directory.

	gdc-client upload cd939bdd-b607-4dd4-87a6-fad12893932d -t token.txt

### Resuming a Failed Upload

By default, GDC Data Transfer Tool uses multipart transfer to upload files. If an upload failed but some parts were transmitted successfully, a resume file will be saved with the filename *resume\_[manifest\_filename]*. Running the upload command again will resume the transfer of only those parts of the file that failed to upload in the previous attempt.

	gdc-client upload -m manifest.yml -t token

### Deleting Previously Uploaded Data

Previously uploaded data can be replaced with new data by deleting it first using the **--delete** switch:

	gdc-client upload -m manifest.yml -t token --delete


## Troubleshooting

### Invalid Token

An error message about an &#39;invalid token&#39; means that a new authentication token needs to be obtained from the GDC Data Portal or the GDC Data Submission Portal as described in [Preparing for Data Download and Upload](Preparing_for_Data_Download_and_Upload.md).


	 403 Client Error: FORBIDDEN: {
		  "message": "Your token is invalid or expired, please get a new token from GDC Data Portal"
		  }

### dbGaP Permissions Error

Users may see the following error message when attempting to download a file from GDC:

	 403 Client Error: FORBIDDEN: {
		  "message": "You don't have access to the data: Please specify a X-Auth-Token"
		  }


This error message indicates that the user does not have dbGaP access to the project to which the file belongs. Instructions for requesting access from dbGaP can be found [here](https://gdc.cancer.gov/access-data/obtaining-access-controlled-data/).

### File Availability Error

Users may also see the following error message when attempting to download a file from GDC:

	 403 Client Error: FORBIDDEN: {
		"message": "You don't have access to the data: Requested file abd28349-92cd-48a3-863a-007a218de80f does not allow read access"
	  	}

This error message means that the file is not available for download. This may be because the file has not been uploaded or released yet or that it is not a file entity.

### GDC Upload Privileges Error

Users may see the following error message when attempting to upload a file:

	 Can't upload: {
		 "message": "You don't have access to the data: You don't have create role to do 'upload'"
	     }

This means that the user has dbGaP read access to the data, but does not have GDC upload privileges. Users can contact [The database of Genotypes and Phenotypes (dbGaP) ](https://www.ncbi.nlm.nih.gov/gap) to request upload privileges.

### File in Uploaded State Error

Re-uploading a file may return the following error:

	 Can't upload: {
		  "message": "File in uploaded state, upload not allowed"
		  }

To resolve this issue, delete the file using the **--delete** switch before re-uploading.

### Microsoft Windows Executable Error

Attempting to run gdc-client.exe by double-clicking it in the Windows Explorer will produce a window that blinks once and disappears.

This is normal, the executable must be run using the command prompt. Click 'Start', followed by 'Run' and type 'cmd' into the text bar.  Then navigate to the path containing the executable using the 'cd' command.     

## Help Menus

The GDC Data Transfer Tool comes with built-in help menus. These menus are displayed when the GDC Data Transfer Tool is run with flags -h or --help for any of the main arguments to the tool. Running the GDC Data Transfer Tool without argument or flag will present a list of available command options.



=== "Shell"

    ```Shell
    gdc-client --help
    ```

=== "Output"

    ```shell 
    usage: gdc-client [-h] [--version] {download,upload,settings} ...
    
    The Genomic Data Commons Command Line Client
    
    optional arguments:
      -h, --help            show this help message and exit
      --version             show program's version number and exit
    
    commands:
      {download,upload,settings}
                            for more information, specify -h after a command
        download            download data from the GDC
        upload              upload data to the GDC
        settings            display default settings
    ```

 The available menus are provided below.

### Root menu

The GDC Data Transfer Tool displays the following output when executed without any arguments.

=== "Shell"

    ```Shell
    gdc-client
    ```

=== "Output"

    ```shell
    usage: gdc-client [-h] [--version] {download,upload,settings} ...
    gdc-client: error: too few arguments
    ```


### Download help menu

The GDC Data Transfer Tool displays the following help menu for its download functionality.

=== "Shell"

    ```Shell
    gdc-client download --help
    ```

=== "Output"

    ```shell
    usage: gdc-client download [-h] [--debug]
                               [--log-file LOG_FILE]
                               [--color_off] [-t TOKEN_FILE]
                               [-d DIR] [-s server]
                               [--no-segment-md5sums]
                               [--no-file-md5sum]
                               [-n N_PROCESSES]
                               [--http-chunk-size HTTP_CHUNK_SIZE]
                               [--save-interval SAVE_INTERVAL]
                               [--no-verify]
                               [--no-related-files]
                               [--no-annotations]
                               [--no-auto-retry]
                               [--retry-amount RETRY_AMOUNT]
                               [--wait-time WAIT_TIME]
                               [--latest] [--config FILE] [-u]
                               [-m MANIFEST]
                               [file_id [file_id ...]]

    positional arguments:
    file_id               The GDC UUID of the file(s) to download

    optional arguments:
    -h, --help            show this help message and exit
    --debug               Enable debug logging. If a failure occurs, the program
                          will stop.
    --log-file LOG_FILE   Save logs to file. Amount logged affected by --debug
    --color_off           Disable colored output
    -t TOKEN_FILE, --token-file TOKEN_FILE
                          GDC API auth token file
    -d DIR, --dir DIR     Directory to download files to. Defaults to current
                          dir
    -s server, --server server
                          The TCP server address server[:port]
    --no-segment-md5sums  Do not calculate inbound segment md5sums and/or do not
                          verify md5sums on restart
    --no-file-md5sum      Do not verify file md5sum after download
    -n N_PROCESSES, --n-processes N_PROCESSES
                          Number of client connections.
    --http-chunk-size HTTP_CHUNK_SIZE, -c HTTP_CHUNK_SIZE
                          Size in bytes of standard HTTP block size.
    --save-interval SAVE_INTERVAL
                          The number of chunks after which to flush state file.
                          A lower save interval will result in more frequent
                          printout but lower performance.
    --no-verify           Perform insecure SSL connection and transfer
    --no-related-files    Do not download related files.
    --no-annotations      Do not download annotations.
    --no-auto-retry       Ask before retrying to download a file
    --retry-amount RETRY_AMOUNT
                          Number of times to retry a download
    --wait-time WAIT_TIME
                          Amount of seconds to wait before retrying
    --latest              Download latest version of a file if it exists
    --config FILE         Path to INI-type config file
    -u, --udt             Use the UDT protocol.
    -m MANIFEST, --manifest MANIFEST
                          GDC download manifest file
    ```

### Upload help menu

The GDC Data Transfer Tool displays the following help menu for its upload functionality.


=== "Shell"

    ```Shell
    gdc-client upload --help
    ```

=== "Output"
    ```shell
    usage: gdc-client upload [-h] [--debug]
                                                [--log-file LOG_FILE]
                                                [--color_off] [-t TOKEN_FILE]
                                                [--project-id PROJECT_ID]
                                                [--path path]
                                                [--upload-id UPLOAD_ID]
                                                [--insecure] [--server SERVER]
                                                [--part-size PART_SIZE]
                                                [--upload-part-size UPLOAD_PART_SIZE]
                                                [-n N_PROCESSES]
                                                [--disable-multipart] [--abort]
                                                [--resume] [--delete]
                                                [--manifest MANIFEST]
                                                [--config FILE]
                                                [file_id [file_id ...]]
    positional arguments:
     file_id               The GDC UUID of the file(s) to upload
    
    optional arguments:
     -h, --help            show this help message and exit
     --debug               Enable debug logging. If a failure occurs, the program
                           will stop.
     --log-file LOG_FILE   Save logs to file. Amount logged affected by --debug
     --color_off           Disable colored output
     -t TOKEN_FILE, --token-file TOKEN_FILE
                           GDC API auth token file
     --project-id PROJECT_ID, -p PROJECT_ID
                           The project ID that owns the file
     --path path, -f path  directory path to find file
     --upload-id UPLOAD_ID, -u UPLOAD_ID
                           Multipart upload id
     --insecure, -k        Allow connections to server without certs
     --server SERVER, -s SERVER
                           GDC API server address
     --part-size PART_SIZE
                           DEPRECATED in favor of [--upload-part-size]
     --upload-part-size UPLOAD_PART_SIZE, -c UPLOAD_PART_SIZE
                           Part size for multipart upload
     -n N_PROCESSES, --n-processes N_PROCESSES
                           Number of client connections
     --disable-multipart   Disable multipart upload
     --abort               Abort previous multipart upload
     --resume, -r          Resume previous multipart upload
     --delete              Delete an uploaded file
     --manifest MANIFEST, -m MANIFEST
                           Manifest which describes files to be uploaded
     --config FILE         Path to INI-type config file
    ```

##Data Transfer Tool Configuration File  
The DTT has the ability to save and reuse configuration parameters in the format of a flat text file via a command line argument.  A simple text file needs to be created first with an extension of either txt or dtt.  The supported section headers are upload and download which can be used independently of each other or used in the same configuration file.  Each section header corresponds to the main functions of the application which are to either download data from the GDC portals or to upload data to the submission system of the GDC.  The configurable parameters are those listed in the help menus under either [download](https://docs.gdc.cancer.gov/Data_Transfer_Tool/Users_Guide/Data_Download_and_Upload/#download-help-menu) or [upload](https://docs.gdc.cancer.gov/Data_Transfer_Tool/Users_Guide/Data_Download_and_Upload/#upload-help-menu) displayed under the output tabs.             


Example usage:

    gdc-client download d45ec02b-13c3-4afa-822d-443ccd3795ca --config my-dtt-config.dtt

Example of configuration file:

    [upload]
    path = /some/upload/path
    upload_part_size = 1073741824


    [download]
    dir = /some/download/path
    http_chunk_size = 2048
    retry_amount = 6


###Display Config Parameters
This command line flag can be used with either the download or upload application feature to display what settings are active within a custom data transfer tool configuration file.  

    gdc-client settings download --config my-dtt-config.dtt
    [download]
    no_auto_retry = False
    no_file_md5sum = False
    save_interval = 1073741824
    http_chunk_size = 2048
    server = http://exmple-site.com
    n_processes = 8
    no_annotations = False
    no_related_files = False
    retry_amount = 6
    no_segment_md5sum = False
    manifest = []
    wait_time = 5.0
    no_verify = True
    dir = /some/download/path




## Help Menus

The GDC Data Transfer Tool comes with built-in help menus. These menus are displayed when the GDC Data Transfer Tool is run with flags -h or --help for any of the main arguments to the tool. Running the GDC Data Transfer Tool without argument or flag will present a list of available command options.


=== "Shell"

    ```shell
    gdc-client --help
    ```

=== "Output"

    ```shell
    usage: gdc-client [-h] [--version] {download,upload,settings} ...

    The Genomic Data Commons Command Line Client

    optional arguments:
      -h, --help            show this help message and exit
      --version             show program's version number and exit

    commands:
      {download,upload,settings}
                            for more information, specify -h after a command
        download            download data from the GDC
        upload              upload data to the GDC
        settings            display default settings
    ```

 The available menus are provided below.

### Root menu

The GDC Data Transfer Tool displays the following output when executed without any arguments.

=== "Shell"

    ```shell
    gdc-client
    ```

=== "Output"

```shell
usage: gdc-client [-h] [--version] {download,upload,settings} ...
gdc-client: error: too few arguments
```


### Download help menu

The GDC Data Transfer Tool displays the following help menu for its download functionality.

=== "Shell"

    ```Shell
    gdc-client download --help
    ```

=== "Output"

    ```shell
    usage: gdc-client download [-h] [--debug]
                              [--log-file LOG_FILE]
                              [--color_off] [-t TOKEN_FILE]
                              [-d DIR] [-s server]
                              [--no-segment-md5sums]
                              [--no-file-md5sum]
                              [-n N_PROCESSES]
                              [--http-chunk-size HTTP_CHUNK_SIZE]
                              [--save-interval SAVE_INTERVAL]
                              [--no-verify]
                              [--no-related-files]
                              [--no-annotations]
                              [--no-auto-retry]
                              [--retry-amount RETRY_AMOUNT]
                              [--wait-time WAIT_TIME]
                              [--latest] [--config FILE] [-u]
                              [-m MANIFEST]
                              [file_id [file_id ...]]

    positional arguments:
    file_id               The GDC UUID of the file(s) to download

    optional arguments:
    -h, --help            show this help message and exit
    --debug               Enable debug logging. If a failure occurs, the program
                          will stop.
    --log-file LOG_FILE   Save logs to file. Amount logged affected by --debug
    --color_off           Disable colored output
    -t TOKEN_FILE, --token-file TOKEN_FILE
                          GDC API auth token file
    -d DIR, --dir DIR     Directory to download files to. Defaults to current
                          dir
    -s server, --server server
                          The TCP server address server[:port]
    --no-segment-md5sums  Do not calculate inbound segment md5sums and/or do not
                          verify md5sums on restart
    --no-file-md5sum      Do not verify file md5sum after download
    -n N_PROCESSES, --n-processes N_PROCESSES
                          Number of client connections.
    --http-chunk-size HTTP_CHUNK_SIZE, -c HTTP_CHUNK_SIZE
                          Size in bytes of standard HTTP block size.
    --save-interval SAVE_INTERVAL
                          The number of chunks after which to flush state file.
                          A lower save interval will result in more frequent
                          printout but lower performance.
    --no-verify           Perform insecure SSL connection and transfer
    --no-related-files    Do not download related files.
    --no-annotations      Do not download annotations.
    --no-auto-retry       Ask before retrying to download a file
    --retry-amount RETRY_AMOUNT
                          Number of times to retry a download
    --wait-time WAIT_TIME
                          Amount of seconds to wait before retrying
    --latest              Download latest version of a file if it exists
    --config FILE         Path to INI-type config file
    -u, --udt             Use the UDT protocol.
    -m MANIFEST, --manifest MANIFEST
                          GDC download manifest file
    ```

### Upload help menu

The GDC Data Transfer Tool displays the following help menu for its upload functionality.

=== "Shell"

    ```shell
    gdc-client upload --help
    ```

=== "Output"

    ```shell
    usage: gdc-client upload [-h] [--debug]
                                                [--log-file LOG_FILE]
                                                [--color_off] [-t TOKEN_FILE]
                                                [--project-id PROJECT_ID]
                                                [--path path]
                                                [--upload-id UPLOAD_ID]
                                                [--insecure] [--server SERVER]
                                                [--part-size PART_SIZE]
                                                [--upload-part-size UPLOAD_PART_SIZE]
                                                [-n N_PROCESSES]
                                                [--disable-multipart] [--abort]
                                                [--resume] [--delete]
                                                [--manifest MANIFEST]
                                                [--config FILE]
                                                [file_id [file_id ...]]
    positional arguments:
    file_id               The GDC UUID of the file(s) to upload

    optional arguments:
    -h, --help            show this help message and exit
    --debug               Enable debug logging. If a failure occurs, the program
                          will stop.
    --log-file LOG_FILE   Save logs to file. Amount logged affected by --debug
    --color_off           Disable colored output
    -t TOKEN_FILE, --token-file TOKEN_FILE
                          GDC API auth token file
    --project-id PROJECT_ID, -p PROJECT_ID
                          The project ID that owns the file
    --path path, -f path  directory path to find file
    --upload-id UPLOAD_ID, -u UPLOAD_ID
                          Multipart upload id
    --insecure, -k        Allow connections to server without certs
    --server SERVER, -s SERVER
                          GDC API server address
    --part-size PART_SIZE
                          DEPRECATED in favor of [--upload-part-size]
    --upload-part-size UPLOAD_PART_SIZE, -c UPLOAD_PART_SIZE
                          Part size for multipart upload
    -n N_PROCESSES, --n-processes N_PROCESSES
                          Number of client connections
    --disable-multipart   Disable multipart upload
    --abort               Abort previous multipart upload
    --resume, -r          Resume previous multipart upload
    --delete              Delete an uploaded file
    --manifest MANIFEST, -m MANIFEST
                          Manifest which describes files to be uploaded
    --config FILE         Path to INI-type config file
    ```


# Data Transfer Tool Release Notes Archive





## v0.9.00

* __GDC Product__: Data Transfer Tool
* __Release Date__: May 9, 2016


### New Features and Changes

* Optimizations to improve download performance
* Improved error message when `upload` argument is used without additional arguments

### Bugs Fixed Since Last Release

* None to report

### Known Issues and Workarounds

* On some terminals, dragging and dropping a file into the interactive client will add single quotes (' ') around the file path. This causes the interactive client to misinterpret the file path and generate an error when attempting to load a manifest file or token.
	* *Workaround:* Manually type out the file name or remove the single quotes from around the file path.

Release details are maintained in the [GDC Data Transfer Tool Change Log](https://github.com/NCI-GDC/gdc-client/blob/master/CHANGELOG.md)





## v0.8.00

* __GDC Product__: Data Transfer Tool
* __Release Date__: May 2, 2016


### New Features and Changes

* Usability improvements to command line interface
* Updated help menus
* Warning on insecure token file permissions
* Validation of upload manifest against new upload manifest YAML schema
* Default number of processes (concurrent downloads) is now 1
* User-Agent header now includes GDC client name and version

### Bugs Fixed Since Last Release

* None to report

### Known Issues and Workarounds

* On some terminals, dragging and dropping a file into the interactive client will add single quotes (' ') around the file path. This causes the interactive client to misinterpret the file path and generate an error when attempting to load a manifest file or token.
	* *Workaround:* Manually type out the file name or remove the single quotes from around the file path.

Release details are maintained in the [GDC Data Transfer Tool Change Log](https://github.com/NCI-GDC/gdc-client/blob/master/CHANGELOG.md)





## v0.7.00

* __GDC Product__: Data Transfer Tool
* __Release Date__: March 30, 2016


### New Features and Changes

* Incorporated improved exception handling
* API response errors are now displayed

### Bugs Fixed Since Last Release

* gdc-client upload --delete function has been fixed

### Known Issues and Workarounds

* On some terminals, dragging and dropping a file into the interactive client will add single quotes (' ') around the file path. This causes the interactive client to misinterpret the file path and generate an error when attempting to load a manifest file or token.
	* *Workaround:* Manually type out the file name or remove the single quotes from around the file path.
* Delete function is not working.
	* *Workaround:* Use an API call instead of the Data Transfer Tool to delete files

Release details are maintained in the [GDC Data Transfer Tool Change Log](https://github.com/NCI-GDC/gdc-client/blob/master/CHANGELOG.md)





## v0.6.00

* __GDC Product__: Data Transfer Tool
* __Release Date__: March 10, 2016


### New Features and Changes

* Data Transfer Tool version numbers are now independent of GDC API version numbers

### Bugs Fixed Since Last Release

* Parsing of command line arguments relating to authentication tokens has been fixed. Arguments `-t` and `--token-file` accept token files, whereas `-T` and `--token` accept token strings.

### Known Issues and Workarounds

* On some terminals, dragging and dropping a file into the interactive client will add single quotes (' ') around the file path. This causes the interactive client to misinterpret the file path and generate an error when attempting to load a manifest file or token.
	* *Workaround:* Manually type out the file name or remove the single quotes from around the file path.
* Delete function is not working.
	* *Workaround:* Use an API call instead of the Data Transfer Tool to delete files

Release details are maintained in the [GDC Data Transfer Tool Change Log](https://github.com/NCI-GDC/gdc-client/blob/master/CHANGELOG.md)






## v0.4.00

* __GDC Product__: Data Transfer Tool
* __Release Date__: March 3, 2016


### New Features and Changes

* Help menus and options are more consistent across download, upload, and interactive modes
* Default behavior is command line mode rather than interactive mode
* Clearer error message when file IDs or manifests are not supplied in download mode

### Bugs Fixed Since Last Release

* None to report

### Known Issues and Workarounds

* On some terminals, dragging and dropping a file into the interactive client will add single quotes (' ') around the file path. This causes the interactive client to misinterpret the file path and generate an error when attempting to load a manifest file or token.
	* *Workaround:* Manually type out the file name or remove the single quotes from around the file path.

Release details are maintained in the [GDC Data Transfer Tool Change Log](https://github.com/NCI-GDC/gdc-client/blob/master/CHANGELOG.md)






## v0.3.20

* __GDC Product__: Data Transfer Tool
* __Release Date__: January 28, 2016


### New Features and Changes

* None to report

### Bugs Fixed Since Last Release

* GDC API compatibility fixes

### Known Issues and Workarounds

* On some terminals, dragging and dropping a file into the interactive client will add single quotes (' ') around the file path. This causes the interactive client to misinterpret the file path and generate an error when attempting to load a manifest file or token.
	* *Workaround:* Manually type out the file name or remove the single quotes from around the file path.

Release details are maintained in the [GDC Data Transfer Tool Change Log](https://github.com/NCI-GDC/gdc-client/blob/master/CHANGELOG.md)






## v0.2.18

* __GDC Product__: Data Transfer Tool
* __Release Date__: December 18, 2015

### New Features and Changes

* None to report

### Bugs Fixed Since Last Release

* Fixed a bug where adding a new line to the end of an auth token would break the header for a request.

### Known Issues and Workarounds

* On some terminals, dragging and dropping a file into the interactive client will add single quotes (' ') around the file path. This causes the interactive client to misinterpret the file path and generate an error when attempting to load a manifest file or token.
	* *Workaround:* Manually type out the file name or remove the single quotes from around the file path.

Release details are maintained in the [GDC Data Transfer Tool Change Log](https://github.com/NCI-GDC/gdc-client/blob/master/CHANGELOG.md)






## v0.2.18-spr3

* __GDC Product__: Data Transfer Tool
* __Release Date__: November 18, 2015

### New Features and Changes

* Added molecular data upload via manifest files.
* Single part or multi-part upload is available dependent on file size.
* Users can resume partially uploaded files for multi-part uploads.

### Bugs Fixed Since Last Release

* None to report

### Known Issues and Workarounds

* On some terminals, dragging and dropping a file into the interactive client will add single quotes (' ') around the file path. This causes the interactive client to misinterpret the file path and generate an error when attempting to load a manifest file or token.
	* *Workaround:* Manually type out the file name or remove the single quotes from around the file path.

Release details are maintained in the [GDC Data Transfer Tool Change Log](https://github.com/NCI-GDC/gdc-client/blob/master/CHANGELOG.md)






## v0.2.15.2

* __GDC Product__: Data Transfer Tool
* __Release Date__: August 31, 2015

### New Features and Changes

* None to report

### Bugs Fixed Since Last Release

* Fixed an issue where the authorization token was not properly passed to related files downloads, preventing download of controlled access related files.
* Fixed an issue where some error messages would not return a descriptive response for the user.
* Fixed an issue where the client would not properly read URLs if a trailing '/' was not added at the end.

### Known Issues and Workarounds

* On some terminals, dragging and dropping a file into the interactive client will add single quotes (' ') around the file path. This causes the interactive client to misinterpret the file path and generate an error when attempting to load a manifest file or token.
	* *Workaround:* Manually type out the file name or remove the single quotes from around the file path.

Release details are maintained in the [GDC Data Transfer Tool Change Log](https://github.com/NCI-GDC/gdc-client/blob/master/CHANGELOG.md)






## v0.2.15.1

* __GDC Product__: Data Transfer Tool
* __Release Date__: August 7, 2015

### New Features and Changes

* Reorganized the High Performance Download Client into the GDC Client. This change was made in order to accomodate future functionality into one consolidated client.
* The GDC Client will now download associated files and annotations.

### Bugs Fixed Since Last Release

* None to report

### Known Issues and Workarounds

* On some Mac OS X systems, it is possible that a directory user permission error might be raised when attempting to download files.
  * *Workaround:* Switch the current working directory and attempt to download again

Release details are maintained in the [GDC Data Transfer Tool Change Log](https://github.com/NCI-GDC/gdc-client/blob/master/CHANGELOG.md)






## v0.2.13

* __GDC Product__: Data Transfer Tool
* __Release Date__: July 23, 2015

### New Features and Changes

* Removed debug definition from default configuration so users can specify prior to compile if they want verbose logging output

### Bugs Fixed Since Last Release

* Fixed a receive buffer space bug

### Known Issues and Workarounds

* None to report

Release details are maintained in the [GDC Data Transfer Tool Change Log](https://github.com/NCI-GDC/gdc-client/blob/master/CHANGELOG.md)






## v0.1.13

* __GDC Product__: Data Transfer Tool
* __Release Date__: June 2, 2015

### New Features and Changes

* Provides support for interactive text mode
* Adds a default server Uniform Resource Locator (URL)
* Updated command line syntax to default to TCP/HTTP in support of ease of use
* Provides support for Windows
* Packaged for OSX, Ubuntu, and Windows

### Bugs Fixed Since Last Release

* None to report

### Known Issues and Workarounds

* None to report

Release details are maintained in the [GDC Data Transfer Tool Change Log](https://github.com/NCI-GDC/gdc-client/blob/master/CHANGELOG.md)






## v0.1.10

* __GDC Product__: Data Transfer Tool
* __Release Date__: March 18, 2015

### New Features and Changes

* Allow users to retrieve large, high volume molecular data using a manifest file generated by the GDC Data Portal
* Allow users to download retrieved files

### Bugs Fixed Since Last Release

* None to report

### Known Issues and Workarounds

* Does not support the Windows platform

Release details are maintained in the [GDC Data Transfer Tool Change Log](https://github.com/NCI-GDC/gdc-client/blob/master/CHANGELOG.md)


# Data Transfer Tool UI Release Notes


| Version | Date |
|---|---|
| [v1.0.0](DTT_UI_Release_Notes.md#v100) | July 8, 2021 |
| [v0.6.0](DTT_UI_Release_Notes.md#v054) | August 12, 2020 |
| [v0.5.4](DTT_UI_Release_Notes.md#v054) | April 5, 2018 |
| [v0.5.3](DTT_UI_Release_Notes.md#v053) | December 14, 2017 |


## v1.0.0
* __GDC Product__: Data Transfer Tool UI
* __Release Date__: July 8, 2021

### New Features and Changes
* The DTT-UI now uses the latest version of the Data Transfer Tool (v1.6.1)

### Bugs Fixed Since Last Release
* None

### Known Issues and Workarounds
* Download speeds for large numbers of small files may be better handled with the Command Line version of the Data Transfer Tool
* Data Submission to the GDC is not supported in the Data Transfer Tool UI.  Instead users must use the Command Line Data Transfer Tool

## v0.6.0
* __GDC Product__: Data Transfer Tool UI
* __Release Date__: August 12, 2020

### New Features and Changes
* The server option can now be indicated within the interface. Previously the DTT-UI server defaulted to `api.gdc.cancer.gov`
* The DTT-UI now uses Data Transfer Tool v1.6, which uses Python3.  

### Bugs Fixed Since Last Release
* None

### Known Issues and Workarounds
* Download speeds for large numbers of small files may be better handled with the Command Line version of the Data Transfer Tool
* Data Submission to the GDC is not supported in the Data Transfer Tool UI.  Instead users must use the Command Line Data Transfer Tool



## v0.5.4
* __GDC Product__: Data Transfer Tool UI
* __Release Date__: April 5, 2018

### New Features and Changes
* None

### Bugs Fixed Since Last Release
* Download is now enabled for GDC reference and publication files. <!--SV-1045, DTT-100-->

### Known Issues and Workarounds
* Download speeds for large numbers of small files may be better handled with the Command Line version of the Data Transfer Tool
* Data Submission to the GDC is not supported in the Data Transfer Tool UI.  Instead users must use the Command Line Data Transfer Tool



## v0.5.3
* __GDC Product__: Data Transfer Tool UI
* __Release Date__: December 14, 2017


### New Features and Changes
* This is the first release for the Data Transfer Tool User Interface.  It allows users to download controlled access data using a simplified point and click interface.  This is a beta release and we welcome feedback on the user experience.  Important updates compared to the Command Line version include:
  * Upload and store authentication token between sessions
  * Easily view progress on a download manifest as the files are completed
  * View download history

### Bugs Fixed Since Last Release
* None

### Known Issues and Workarounds
* Download speeds for large numbers of small files may be better handled with the Command Line version of the Data Transfer Tool
* Data Submission to the GDC is not supported in the Data Transfer Tool UI.  Instead users must use the Command Line Data Transfer Tool


# Data Transfer Tool Release Notes

| Version | Date |
|---|---|
| [v2.0.0](DTT_Release_Notes.md#v200) | July 30, 2024 |
| [v1.6.1](DTT_Release_Notes.md#v161) | May 17, 2021 |
| [v1.6.0](DTT_Release_Notes.md#v160) | July 8, 2020 |
| [v1.5.0](DTT_Release_Notes.md#v150) | January 30, 2020 |
| [v1.4.0](DTT_Release_Notes.md#v140) | December 18, 2018 |
| [v1.3.0](DTT_Release_Notes.md#v130) | August 22, 2017 |
| [v1.2.0](DTT_Release_Notes.md#v120) | Oct 31, 2016 |
| [v1.1.0](DTT_Release_Notes.md#v110) | September 7, 2016 |
| [v1.0.1](DTT_Release_Notes.md#v101) | June 2, 2016 |
| [v1.0.0](DTT_Release_Notes.md#v100) | May 26, 2016 |

## V2.0.0
* __GDC Product__: Data Transfer Tool
* __Release Date__: July 30, 2024

### New Features and Changes
* None

### Bugs Fixed Since Last Release

* Fixed logging issue that was causing client to crash <!--DEV-2799-->
* Upgraded Python version <!--DEV-2537-->

### Known Issues and Workarounds
* Use of non-ASCII characters in token passed to Data Transfer Tool will produce incorrect error message "Internal server error: Auth service temporarily unavailable".
* On some terminals, dragging and dropping a file into the interactive client will add single quotes (' ') around the file path. This causes the interactive client to misinterpret the file path and generate an error when attempting to load a manifest file or token.
	* *Workaround:* Manually type out the file name or remove the single quotes from around the file path.
* When any files mentioned in the upload manifest are not present in the upload directory the submission will hang at the missing file.
	* *Workaround:* Edit the manifest to specify only the the files that are present in the upload directory for submission or copy the missing files into the upload directory.


## V1.6.1
* __GDC Product__: Data Transfer Tool
* __Release Date__: May 17, 2021

### New Features and Changes
* None

### Bugs Fixed Since Last Release
* Fixed issue with resuming large file downloads. <!--DEV-506-->
* Fixed issue with error reporting. <!--DEV-505-->
* Improved multi-part file upload. <!--DEV-401-->

### Known Issues and Workarounds
* Use of non-ASCII characters in token passed to Data Transfer Tool will produce incorrect error message "Internal server error: Auth service temporarily unavailable".
* On some terminals, dragging and dropping a file into the interactive client will add single quotes (' ') around the file path. This causes the interactive client to misinterpret the file path and generate an error when attempting to load a manifest file or token.
	* *Workaround:* Manually type out the file name or remove the single quotes from around the file path.
* When any files mentioned in the upload manifest are not present in the upload directory the submission will hang at the missing file.
	* *Workaround:* Edit the manifest to specify only the the files that are present in the upload directory for submission or copy the missing files into the upload directory.


## V1.6.0
* __GDC Product__: Data Transfer Tool
* __Release Date__: July 8, 2020

### New Features and Changes
* None

### Bugs Fixed Since Last Release
* Fixed issue with file upload requiring a manifest <!--DEV-170-->
* Restored multi-part upload feature <!--DEV-228-->
* Fixed error reporting issue <!--DEV-139-->

### Known Issues and Workarounds
* Use of non-ASCII characters in token passed to Data Transfer Tool will produce incorrect error message "Internal server error: Auth service temporarily unavailable".
* On some terminals, dragging and dropping a file into the interactive client will add single quotes (' ') around the file path. This causes the interactive client to misinterpret the file path and generate an error when attempting to load a manifest file or token.
	* *Workaround:* Manually type out the file name or remove the single quotes from around the file path.
* When any files mentioned in the upload manifest are not present in the upload directory the submission will hang at the missing file.
	* *Workaround:* Edit the manifest to specify only the the files that are present in the upload directory for submission or copy the missing files into the upload directory.

## V1.5.0
* __GDC Product__: Data Transfer Tool
* __Release Date__: January 30, 2020

### New Features and Changes
* Data transfer tool code now uses Python 3. <!--DTT-1109-->

### Bugs Fixed Since Last Release
* Problems with downloading associated annotations is fixed.  <!--SV-1641-->

### Known Issues and Workarounds
* Use of non-ASCII characters in token passed to Data Transfer Tool will produce incorrect error message "Internal server error: Auth service temporarily unavailable".
* On some terminals, dragging and dropping a file into the interactive client will add single quotes (' ') around the file path. This causes the interactive client to misinterpret the file path and generate an error when attempting to load a manifest file or token.
	* *Workaround:* Manually type out the file name or remove the single quotes from around the file path.
* When any files mentioned in the upload manifest are not present in the upload directory the submission will hang at the missing file.
	* *Workaround:* Edit the manifest to specify only the the files that are present in the upload directory for submission or copy the missing files into the upload directory.


## V1.4.0
* __GDC Product__: Data Transfer Tool
* __Release Date__: December 18, 2018

### New Features and Changes
* Enabled download latest file version feature <!--DTT-112-->     
* Removal of Interactive mode <!--DTT-30--><!--DTT-105-->
* Enabled display of all default settings <!--DTT-30--><!--DTT-105-->
* Standardized upload and download help menus <!--DTT-95-->

### Bugs Fixed Since Last Release
* Download flag --no-related-files bug preventing file downloads fixed <!--DTT-94-->
* File name handling with forward slashes bug fixed <!--DTT-99-->
* Download flag --no-segment-md5sums bug fixed. <!--DTT-11-->

### Known Issues and Workarounds
* Use of non-ASCII characters in token passed to Data Transfer Tool will produce incorrect error message "Internal server error: Auth service temporarily unavailable".
* On some terminals, dragging and dropping a file into the interactive client will add single quotes (' ') around the file path. This causes the interactive client to misinterpret the file path and generate an error when attempting to load a manifest file or token.
	* *Workaround:* Manually type out the file name or remove the single quotes from around the file path.
* When any files mentioned in the upload manifest are not present in the upload directory the submission will hang at the missing file.
	* *Workaround:* Edit the manifest to specify only the the files that are present in the upload directory for submission or copy the missing files into the upload directory.


## v1.3.0
* __GDC Product__: Data Transfer Tool
* __Release Date__: August 22, 2017


### New Features and Changes
*	Faster performance when downloading many small files <!--DTT-29, DTT-42-->
* Faster performance overall <!--DTT-29-->
*	Better handling of time outs <!--DTT-23-->
* Uses new default API URL (htts://api.gdc.cancer.gov) <!--DTT-34-->
*	Better logging <!--DTT-38, DTT-12-->

### Bugs Fixed Since Last Release
* Submission manifest **local_file_path:** will now modify path as expected <!--DTT-27-->
* Upload flags --path/-f will modify the upload path as expected <!--DTT-28-->
*	When deleting uploaded files you will no longer need a file in the current directory of the same name <!--DTT-36-->
* Can specify manifest path for upload <!--DTT-32-->

### Known Issues and Workarounds
* Use of non-ASCII characters in token passed to Data Transfer Tool will produce incorrect error message "Internal server error: Auth service temporarily unavailable".
* On some terminals, dragging and dropping a file into the interactive client will add single quotes (' ') around the file path. This causes the interactive client to misinterpret the file path and generate an error when attempting to load a manifest file or token.
	* *Workaround:* Manually type out the file name or remove the single quotes from around the file path.
* When any files mentioned in the upload manifest are not present in the upload directory the submission will hang at the missing file.
	* *Workaround:* Edit the manifest to specify only the the files that are present in the upload directory for submission or copy the missing files into the upload directory.  


## v1.2.0
* __GDC Product__: Data Transfer Tool
* __Release Date__: Oct 19th 2016


### New Features and Changes
* Better handling of connectivity interruptions

### Bugs Fixed Since Last Release
* Uploads via manifest file has been fixed.
* Legacy -i/--identifier flag removed.
* Improved error messaging when uploading without a token.

### Known Issues and Workarounds
* Use of non-ASCII characters in token passed to Data Transfer Tool will produce incorrect error message "Internal server error: Auth service temporarily unavailable".
* On some terminals, dragging and dropping a file into the interactive client will add single quotes (' ') around the file path. This causes the interactive client to misinterpret the file path and generate an error when attempting to load a manifest file or token.
	* *Workaround:* Manually type out the file name or remove the single quotes from around the file path.
* When any files mentioned in the upload manifest are not present in the upload directory the submission will hang at the missing file.
	* *Workaround:* Edit the manifest to specify only the the files that are present in the upload directory for submission or copy the missing files into the upload directory.  
* Upload flags --path/-f do not modify the upload path as expected.
	* *Workaround:* Copy the Data Transfer Tool into the the root of the submittable data directory and run from  there.  
* Submission manifest field **local_file_path:** does not modify upload path expected.
	* *Workaround:* Run Data Transfer Tool from root of the submittable data directory so that data is in the current working directory of the Data Transfer Tool.

## v1.1.0

* __GDC Product__: Data Transfer Tool
* __Release Date__: September 7, 2016


### New Features and Changes

* Partial extension added to all download files created during download. Removed after successful download.  
* Number of processes started by default changed to 8 (-n flag).

### Bugs Fixed Since Last Release

* None to report.

### Known Issues and Workarounds

* Use of non-ASCII characters in token passed to Data Transfer Tool will produce incorrect error message "Internal server error: Auth service temporarily unavailable".
* On some terminals, dragging and dropping a file into the interactive client will add single quotes (' ') around the file path. This causes the interactive client to misinterpret the file path and generate an error when attempting to load a manifest file or token.
	* *Workaround:* Manually type out the file name or remove the single quotes from around the file path.
* Use of a manifest file for uploads to the Submission Portal will produce an error message "ERROR: global name 'read_manifest' is not defined". <!--SV-457-->
	* *Workaround:* Upload files via UUID instead or use the API/Submission Portal.

## v1.0.1

* __GDC Product__: Data Transfer Tool
* __Release Date__: June 2, 2016


### New Features and Changes

* MD5 checksum verification of downloaded files.
* BAM index files (.bai) are now automatically downloaded with parent BAM.
* UDT mode included to help improve certain high-speed transfers between the GDC and distant locations.

### Bugs Fixed Since Last Release

* None to report.

### Known Issues and Workarounds

* Use of non-ASCII characters in token passed to Data Transfer Tool will produce incorrect error message "Internal server error: Auth service temporarily unavailable".
* On some terminals, dragging and dropping a file into the interactive client will add single quotes (' ') around the file path. This causes the interactive client to misinterpret the file path and generate an error when attempting to load a manifest file or token.
	* *Workaround:* Manually type out the file name or remove the single quotes from around the file path.

## v1.0.0

* __GDC Product__: Data Transfer Tool
* __Release Date__: May 26, 2016


### New Features and Changes

* Single-thread and multi-threaded download capability
* User-friendly command line interface
* Progress bars provide visual representation of transfer status
* Optional interactive (REPL) mode
* Detailed help menus for upload and download functionality
* Support for authentication using a token file
* Support for authentication using a token string
* Resumption of incomplete uploads and downloads
* Initiation of transfers using manifests
* Initiation of transfers using file UUIDs
* Advanced configuration options
* Binary distributions available for Linux (Ubuntu), OS X, and Windows

### Bugs Fixed Since Last Release

* None to report.

### Known Issues and Workarounds

* Use of non-ASCII characters in token passed to Data Transfer Tool will produce incorrect error message "Internal server error: Auth service temporarily unavailable".
* On some terminals, dragging and dropping a file into the interactive client will add single quotes (' ') around the file path. This causes the interactive client to misinterpret the file path and generate an error when attempting to load a manifest file or token.
	* *Workaround:* Manually type out the file name or remove the single quotes from around the file path.


# OncoMatrix

## Introduction to OncoMatrix

The OncoMatrix tool is a web-based tool for visualizing coding mutations such as Simple Somatic Mutations (SSM) and Copy Number Variations (CNV) from the NCI Genomic Data Commons (GDC).

## Accessing the OncoMatrix Chart

At the Analysis Center, click on the 'OncoMatrix' card to launch the app.

[![Analysis Tools with OncoMatrix Card](./images/oncomatrix/1-analysis_center.png)](./images/oncomatrix/1-analysis_center.png 'Click to see the full image.')

Users can view publicly available genes as well as login with credentials to access controlled data.

## Quick Reference Guide

There are three main panels in the OncoMatrix tool: [control panel](#control-panel), [matrix plot](#matrix-plot), and [legend panel](#legend-panel).

[![OncoMatrix Overview](images/oncomatrix/oncomatrix_overview.png)](images/oncomatrix/oncomatrix_overview.png 'Click to see the full image.')

### Control Panel

The control panel has various functionalities with which users can change or modify the appearance of the matrix. The control panel provides flexibility and a wide range of options to maximize user control.

[![OncoMatrix Control Panel](images/oncomatrix/oncomatrix_control_panel.png)](images/oncomatrix/oncomatrix_control_panel.png 'Click to see the full image.')

__Control Panel:__

* __Cases:__ Choose how to sort the cases, specify the maximum number of cases to display, group cases according to selected GDC variables, and adjust the visible characters of the case labels
* __Genes:__ Modify how cases are represented for each gene (Absolute, Percent, or None), row group and label lengths, rendering style, how genes are sorted, the maximum number of genes displayed, and the existing gene set
    * __Edit Group:__ Displays a panel of currently selected genes, which can be modified by clicking on a gene to remove it from the gene set, searching for a particular gene to add, loading top variably expressed genes, or loading a pre-defined gene set provided by the MSigDB database
    * __Create Group:__ Create a new gene set by searching for a particular gene, loading top mutated genes, or loading a pre-defined gene set provided by the MSigDB database
* __Mutation:__ Choose to show all or select mutations based on consequence
* __CNV:__ Choose to show or hide CNV data
* __Variables:__ Search and select variables to add to the bottom of the matrix
* __Cell Layout:__ Modify the format of the cells by changing colors, cell dimensions and spacing, and label formatting
* __Legend Layout:__ Alter the legend by changing the font size, dimensions and spacing, and other formatting preferences
* __Download:__ Download the matrix in svg format
* __Zoom:__ Adjust the zoom level by using the up and down arrows on the input box, entering a number, or using the sliding scale to view the case labels.
<!--
* __Undo:__ Undo changes made to the matrix
* __Redo:__ Redo changes made to the matrix
* __Restore:__ Restore the matrix to its default settings
-->

### Matrix Plot

The OncoMatrix plot displays the genes or variables along the left of the panel with each column representing an individual case.

__Matrix cells__

Each column in the matrix represents a case. Hovering over a cell will display the corresponding case submitter_id, gene name, copy number information, and mutation consequence if any are provided. Clicking on a cell also gives users the option to launch the Disco Plot.

[![Disco Plot button](images/oncomatrix/oncomatrix_disco_plot_button.png)](images/oncomatrix/oncomatrix_disco_plot_button.png 'Click to see the full image.')

The Disco Plot is a circular plot that shows all the mutations and CNVs for a given case. The Disco Plot also displays the legend for the mutation class and the CNV.

[![Disco Plot](images/oncomatrix/oncomatrix_disco_plot.png)](images/oncomatrix/oncomatrix_disco_plot.png 'Click to see the full image.')

__Automatic Zoom__

To perform an automatic zoom, users can click on and hold a case column then drag the mouse from left to right to form a zoom boundary. From the pop-up window, users can choose to zoom in to the cases, list all highlighted cases, or create a cohort of the selected cases.

[![Dragging and zooming on case columns](images/oncomatrix_zoom.png)](images/oncomatrix_zoom.png 'Click to see the full image.')

The individual case columns are now visible with a demarcated boundary. Above the cases, a slider has been provided for moving from one view to another to accommodate all cases.

[![Zoom using input control and slider](images/oncomatrix_zoom2.png)](images/oncomatrix_zoom2.png 'Click to see the full image.')

__Genes__

In the panel of genes on the left, users can hover over a gene to view the number of mutated samples, a breakdown of consequence type, and copy number gain and loss counts.

[![Hover over gene](images/oncomatrix_gene_hover.png)](images/oncomatrix_gene_hover.png 'Click to see the full image.')

Clicking on a gene opens a pop-up window where users can rename it, launch the [ProteinPaint Lollipop plot](proteinpaint_lollipop.md), display the [Gene Summary Page](mutation_frequency.md#gene-and-mutation-summary-pages), and replace or remove the gene. The lollipop plot displays all cases across the GDC affected by SSMs in the selected gene.

[![Click on a gene](images/oncomatrix_gene_click.png)](images/oncomatrix_gene_click.png 'Click to see the full image.')

__Variables__

Any variables added to the matrix appear at the bottom of the plot. Users can hover over a cell in a variable row to display the case submitter_id and their value for the given variable.

[![Hover over a variable cell](images/oncomatrix_variable_cell.png)](images/oncomatrix_variable_cell.png "Click to see the full image.")

Clicking on a variable allows users to rename it, edit it by excluding categories, replace it with a different variable, or remove it entirely.

[![Click on a variable](images/oncomatrix_variable_click.png)](images/oncomatrix_variable_click.png "Click to see the full image.")

__Drag and drop genes and variables__

By default, the genes in the matrix are sorted in descending order according to which genes have the highest number of rendered cases. Users can override this by dragging and dropping gene and variable row labels to sort the rows manually.

### Legend Panel

Below the matrix, the legend displays color coding for mutation classes, CNV, as well as each variable that is selected to appear in the plot.

[![OncoMatrix Legend](images/oncomatrix_legend.png)](images/oncomatrix_legend.png 'Click to see the full image.')

Clicking on `Consequences` offers options to show only truncating mutations, show only protein-changing mutations, or hide consequences.

[![OncoMatrix Legend - Consequences button](images/oncomatrix_legend_consequences.png)](images/oncomatrix_legend_consequences.png 'Click to see the full image.')

Clicking on `CNV` allows users to hide CNV.

[![OncoMatrix Legend - CNV button](images/oncomatrix_legend_cnv.png)](images/oncomatrix_legend_cnv.png 'Click to see the full image.')

Additionally, users can click on a variable's category to hide a specific group, only show a specific group, or show all groups for the selected variable.

[![OncoMatrix Legend - variable category](images/oncomatrix_legend_var_cat.png)](images/oncomatrix_legend_var_cat.png 'Click to see the full image.')

## Features

The following features are viewable once the matrix application is loaded.
There are three main panels as outlined in the figure below i.e., the `Control panel`, `Matrix chart`, and the `Legend panel`.

[![Oncomatrix chart overview](./images/oncomatrix/oncomatrix_overview.png)](./images/oncomatrix/oncomatrix_overview.png 'Click to see the full image.')

Each of the features and functionalities are described in detail in the following sections.

## Matrix plot

### Hovering on sample columns

Each column in the matrix represents a sample.
Hover over sample cells/columns to display information about the sample such as case id, gene name, Copy number information and mutation/mutation class (if any provided) as shown.

[![Hover over cell/columns to display sample information](./images/oncomatrix/3-sample_hovering.png)](./images/oncomatrix/3-sample_hovering.png 'Click to see the full image.')

### Drag to zoom

A user may click a row label and drag it while keeping the mouse button down, to sort the rows manually. Click and hold on a column of sample and drag the mouse from left to right to form a zoom boundary as shown in the image and leave the mouse.

[![Drag the mouse to select an area](./images/oncomatrix/4-drag2zoom.png)](./images/oncomatrix/4-drag2zoom.png 'Click to see the full image.')

This allows for an automatic zoom as shown. The individual sample columns are now visible with a well demarcated boundary. Above the samples, a slider (as shown in gray) has been provided for moving from one view to another to accommodate all cases.

[![Zoom using input control and slider](./images/oncomatrix/5-slide&zoom.png)](./images/oncomatrix/5-slide&zoom.png 'Click to see the full image.')

Additionally, to have a finer control on the zoom the user may follow the steps outlined in the section - Zooming

### Clicking on Sample columns

In the same zoomed in view as shown above, click on any sample column for TP53. This displays a clickable button `Disco plot` as shown.

[![Click on sample label](./images/oncomatrix/6-clicksamplebtn.png)](./images/oncomatrix/6-clicksamplebtn.png 'Click to see the full image.')

Click on the disco plot button to display a circular plot that shows all the mutations for a given sample as shown.

[![Disco plot](./images/oncomatrix/7-discoplot1.png)](./images/oncomatrix/7-discoplot1.png 'Click to see the full image.')

The disco plot can also be accessed by following steps outlined in the section - Disco Plot

### Clicking on gene/variable labels

Click on `TP53` gene label to display the following options.

[![Click gene label](./images/oncomatrix/8-gene_sort_icons.png)](./images/oncomatrix/8-gene_sort_icons.png 'Click to see the full image.')

The first row in the options highlighted by a red box as shown in the image above allows the user to sort rows and move rows up and down (please note that rows can also be moved by dragging and dropping as outlined in section Drag and Drop Gene Label/Variable variable). Every time a sorting icon is clicked the chart will update and reload.

Click the first arrow as shown by clicking the gene label TP53. This will sort the samples against the gene at the top left corner which is TP53 in this example.

[![Click to sort against the top left corner gene](./images/oncomatrix/9-diag_arrow.png)](./images/oncomatrix/9-diag_arrow.png 'Click to see the full image.')

Next, click on the left arrow as shown. This allows for sorting samples against the gene.

[![Move up a gene row](./images/oncomatrix/10-left_arrow.png)](./images/oncomatrix/10-left_arrow.png 'Click to see the full image.')

Now click the down arrow as shown.
The row with TP53 cases will move below ATRX.

[![Move down a gene row](./images/oncomatrix/11-down_arrow.png)](./images/oncomatrix/11-down_arrow.png 'Click to see the full image.')

Click the gene label `TP53` and click the up arrow as shown.

[![Move up a gene row](./images/oncomatrix/12-up_arrow.png)](./images/oncomatrix/12-up_arrow.png 'Click to see the full image.')

The row containing TP53 cases now moves back up in position 1 above ATRX.

Click TP53 again to showcase the edit menu.

Click on 'Replace' as shown above to replace TP53 gene variable with 'Primary site' as shown below. The chart updates with the first row as 'Primary site' thereby replacing TP53 gene variable as shown below. User may choose to sort samples by clicking the 'Primary site' label.

[![Replace a gene row](./images/oncomatrix/13-replace_gene.png)](./images/oncomatrix/13-replace_gene.png 'Click to see the full image.')

[![Replaced row](./images/oncomatrix/14-term-replaced.png)](./images/oncomatrix/14-term-replaced.png 'Click to see the full image.')

Click on the label 'Primary site' and click the option 'Remove' as shown to remove the row completely.

[![Remove a term](./images/oncomatrix/15-remove-primary.png)](./images/oncomatrix/15-remove-primary.png 'Click to see the full image.')

This updates the chart. User may choose to add back TP53 through the gene panel.

Click on `Replace` as shown above to replace TP53 gene variable with `Primary site` as shown below.

### Drag and Drop Gene Label/Variable

The genes on the matrix are sorted by default on the number of cases with the gene having the highest number of cases at the top of the matrix. A user may choose to override this by dragging a gene label and dropping it above or below any other gene in order to customize their own gene groupings.

Select 'PTEN' gene label and drag it below the gene labeled 'EGFR' as shown. When dragging a gene label, hover over EGFR such that the EGFR gene label would appear blue.

[![Drag a gene row](./images/oncomatrix/16-drag-gene-row.png)](./images/oncomatrix/16-drag-gene-row.png 'Click to see the full image.')

When the EGFR gene label appears blue, then drop the PTEN gene label row. The display updates to show PTEN below EGFR as shown below.

[![Dragged gene row](./images/oncomatrix/17-dragged-gene-row.png)](./images/oncomatrix/17-dragged-gene-row.png 'Click to see the full image.')

## Control panel

The control panel as shown has various functionalities with which users can change or modify the appearance of the matrix. The control panel provides flexibility and a wide range of options to maximize user control.

[![Control Panel](./images/oncomatrix/18-control-panel.png)](./images/oncomatrix/18-control-panel.png 'Click to see the full image.')

### Cases

Within the control panel, the first button displays the number of cases that are shown as columns of the matrix. The default view is as shown.

[![Default view](./images/oncomatrix/19-default-view.png)](./images/oncomatrix/19-default-view.png 'Click to see the full image.')

Click on the `Cases` button to display the following options as shown.

1. Sort Cases
2. Maximum #cases
3. Group cases by
4. Sort Case Groups
5. Case Group Label Max Length
6. Case Label Max Length

These sections are described below.

[![Cases menu](./images/oncomatrix/20-cases-controls.png)](./images/oncomatrix/20-cases-controls.png 'Click to see the full image.')

__Sort Case Priority__

The default sort setting sorts the cases 'by presence' which sorts samples with matching consequence to the left.

[![Sort by presence](./images/oncomatrix/sort-by-presence.png)](./images/oncomatrix/sort-by-presence.png 'Click to see the full image.')

To change to sorting by consequence, select the option 'Sort by consequence'.

[![Sort by consequence](./images/oncomatrix/sort-by-consequence.png)](./images/oncomatrix/sort-by-consequence.png 'Click to see the full image.')

Note that these sorting options correspond to SSM only. For CNV, user has to show CNV data using the CNV button for sorting options to become viewable. This step is explained in the 'CNV' section below.

__Maximum #cases__

There is a default number of samples that are shown in the matrix chart. Users can choose to increase or decrease the number of samples. This allows the chart to re-render and display the number of columns based on the user's selection. Figure below shows increased cases to 10000. Please note that any high arbitrary number can be selected but the chart will only show the maximum cases that GDC has.

[![](./images/oncomatrix/max-cases.png)](./images/oncomatrix/max-cases.png 'Click to see the full image.')

The chart will reload with new cases added.

__Group cases by__

This option allows users to group cases by different variables from the GDC dictionary. Click on the `+` icon shown in blue to display different variables such as demographics, diagnoses, Exposures etc. Users may also search for a variable from the search bar provided in the menu as shown by `Search Variables`.

[![Group cases by another variable](./images/oncomatrix/group-cases-by.png)](./images/oncomatrix/group-cases-by.png 'Click to see the full image.')

Click 'Age at diagnosis' from the options. The matrix reloads to show the following view.

[![Select a grouping variable](./images/oncomatrix/22-group-age-at-diag.png)](./images/oncomatrix/22-group-age-at-diag.png 'Click to see the full image.')

As shown above, labels for different age groups show up vertically and all cases get distributed with a clearcut separation according to the age bins.

Click on the 'Age at diagnosis' (blue pill, as shown). This opens a short menu with action items. Click on the first item 'Edit' as shown.

[![Click the variable button to edit](./images/oncomatrix/23-edit-groups.png)](./images/oncomatrix/23-edit-groups.png 'Click to see the full image.')

Drag the red lines on the density distribution to select binning or input numbers for custom binning and select 'Apply'.

[![Edit the binning of values](./images/oncomatrix/24-binning.png)](./images/oncomatrix/24-binning.png 'Click to see the full image.')

The matrix reloads with new bin groupings The labels for the groups are user controlled and hence can be modified according to user requirements.

Click on the blue pill for 'Age at diagnosis' again and click 'Replace'. Select 'Primary site' as shown.

[![Replace the grouping variable](./images/oncomatrix/25-replace-group.png)](./images/oncomatrix/25-replace-group.png 'Click to see the full image.')

The matrix reloads with the new variable distribution.

The last option on the menu is 'Remove'. Click on the '917 Cases' button, followed by 'Age at diagnosis' shown in blue to reveal the menu option. Click 'Remove' to completely get rid of any groups.

[![Remove the grouping variable](./images/oncomatrix/26-remove-group.png)](./images/oncomatrix/26-remove-group.png 'Click to see the full image.')

This will remove all and any groupings and show the default view again.

### Sort Case Groups

Add the variable 'Age at diagnosis' again using the 'Group Cases by' button as shown in the previous section. By default, groups are loaded ordered by their name. Change the selection to 'Case count' as shown below.

[![Sorting Case groups](./images/oncomatrix/27-sort-case-groups.png)](./images/oncomatrix/27-sort-case-groups.png 'Click to see the full image.')

The third selection option 'Hits' orders the groupings based on the number of gene variants for a particular case or case group for the genes in display. Click 'Hits' under 'Sort Case Groups' to change the order of groupings.

Next, hover over the first group label '<=30 years (81)' as shown below.

[![Display case group informatino](./images/oncomatrix/28-label-hover.png)](./images/oncomatrix/28-label-hover.png 'Click to see the full image.')

This shows the number of cases in parenthesis of the group label and the breakdown for the  number of variants and CNV for all the samples within that group for the genes in display.

## Mutation

By default, only protein-changing mutations are shown. This control allows hiding or showing mutations with the following options:

1. Show all mutations

Select this option to show all the mutations.

[![Show all mutations](images/oncomatrix/show-all-mutation.png)](images/oncomatrix/show-all-mutation.png 'Click to see the full image.')

2. Show truncating mutations

Select this option to show only truncating mutations

[![Show truncating mutations](images/oncomatrix/show-truncating-mutations.png)](images/oncomatrix/show-truncating-mutations.png 'Click to see the full image.')

3. To show selected mutations as per requirement, user can choose from the following options after selecting 'Show selected mutations'

[![Show selected mutations](images/oncomatrix/show-selected-mutations.png)](images/oncomatrix/show-selected-mutations.png 'Click to see the full image.')

## CNV

By default, the CNV's are hidden from view as shown by the striked label on the CNV button in the control panel.

[![CNV button striked](images/oncomatrix/CNV-striked.png)](images/oncomatrix/CNV-striked.png 'Click to see the full image.')

To show all CNV data, select the option 'Show all CNV'.

[![Show all CNV](images/oncomatrix/Show-all-CNV.png)](images/oncomatrix/Show-all-CNV.png 'Click to see the full image.')

To view/hide selected CNV (gain or loss) select 'Show selected CNV' allows user to display options for selecting/de-selecting checkboxes and click 'Apply'.

[![Show selected CNV](images/oncomatrix/show-selected-CNV.png)](images/oncomatrix/show-selected-CNV.png 'Click to see the full image.')

### Sorting CNV data

After making selections to show the CNV data, user can sort the cases 'by CNV' from the sorting options under the 'Cases' button

[![Sort by CNV](images/oncomatrix/sort-by-cnv.png)](images/oncomatrix/sort-by-cnv.png 'Click to see the full image.')

## Genes

The gene panel as shown below has several options as listed below for modifying the genes visible on the plot as well as their appearance/style.

- Display Case Counts for Gene
- Genomic Alterations Rendering
- Sort Genes
- Maximum # Genes
- Gene Set

[![Genes menu](./images/oncomatrix/29-genes-btn.png)](./images/oncomatrix/29-genes-btn.png 'Click to see the full image.')

### Display Case Counts for Gene

This option allows change in the number of cases that is represented in parentheses next to the gene variable label as shown below. By default, the number of cases for each gene is an `Absolute`.

Click on the button `50 Genes` to display the menu and select `Percent` as shown below.

[![Option to display or hide case counts for the row](./images/oncomatrix/29-genes-percent.png)](./images/oncomatrix/29-genes-percent.png 'Click to see the full image.')

This shows the case counts as a percentage of the absolute values as shown.

[![Counts as percent of cases](./images/oncomatrix/30-genes-percent-display.png)](./images/oncomatrix/30-genes-percent-display.png 'Click to see the full image.')

User has the option to hide the display of case counts. Click `Genes` button again and select `None` for `Display Case Counts for Gene` as shown below

[![Genes None](./images/oncomatrix/31-genes_none1.png)](./images/oncomatrix/31-genes_none1.png 'Click to see the full image.')

This hides all the case counts as shown.

[![Hide case counts](./images/oncomatrix/32-genes-none-display.png)](./images/oncomatrix/32-genes-none-display.png 'Click to see the full image.')

### Genomic Alterations Rendering

The style of rendering for the sample cells/columns is an Oncoprint style by default. Click on `Stacked` option via `50 Genes` button as shown below.

[![Rendering Style](./images/oncomatrix/rendering-options.png)](./images/oncomatrix/rendering-options.png 'Click to see the full image.')

The mutations and CNV are now stacked on top of each other as shown below.

[![Stacked rendering](./images/oncomatrix/stacked.png)](./images/oncomatrix/stacked.png 'Click to see the full image.')

There is also the option to show single rectangles to render the most severe variants.

[![Single rendering](./images/oncomatrix/single-rendering.png)](./images/oncomatrix/single-rendering.png 'Click to see the full image.')

### Sort Genes

The default sorting option for genes is `By Case Count`. This means the genes are sorted by the number of cases from increasing to decreasing order. Click `50 Genes` button on the control panel, and select `By Input Data Order` under the `Sort Genes` as shown below.

[![Case count](./images/oncomatrix/35-sort-genes.png)](./images/oncomatrix/35-sort-genes.png 'Click to see the full image.')

The genes will now sort according to the order that is stored in the dataset and queried. However, please note that the sorting order can be overridden by the users choice as described in the section - Drag and Drop Gene Label/Variable.

### Maximum # Genes

The number of genes to display on the matrix plot can be modified by the input option as shown below. Click `50 Genes` button and change input number for `Maximum # Genes` to 70.

[![Maximum number of genes to display](./images/oncomatrix/36-max_genes.png)](./images/oncomatrix/36-max_genes.png 'Click to see the full image.')

The chart updates and loads the extra 20 genes. User can modify the set of genes by using the `Gene set` option next.

### Editing gene set

Gene groups can be edited using the `Gene set` option as shown below. Click `50 Genes` button to display this option and then click the `Edit` button in the `Gene set` as shown.

[![Edit geneset](./images/oncomatrix/37-geneset_edit1.png)](./images/oncomatrix/37-geneset_edit1.png 'Click to see the full image.')

Select or deselect the blue checkbox to change the display to show `Cancer Gene Census` (CGC) genes only as shown below.

[![Geneset edit CGC option](./images/oncomatrix/38-geneset_edit2.png)](./images/oncomatrix/38-geneset_edit2.png 'Click to see the full image.')

More information about CGC can be found at https://cancer.sanger.ac.uk/census. Figure displays the top 50 CGC genes.
User may choose to remove single genes one at a time by clicking over the genes.To do so, hover over TP53 as shown in the image below. A red cross mark appears with a description box. Click TP53 to delete the gene as shown below.

[![Deleting a gene](./images/oncomatrix/39-geneset_edit3.png)](./images/oncomatrix/39-geneset_edit3.png 'Click to see the full image.')

User may choose to delete all genes from view by clicking the `Clear` button as shown below. However, a gene/variable selection is mandatory for the chart to load.

[![Clear genes](./images/oncomatrix/clear-genes.png)](./images/oncomatrix/clear-genes.png 'Click to see the full image.')

To restore the cleared genes, click the 'Restore' button.

### MSigDB genes

The MSigDB database (Human Molecular Signatures Database) has 33591 gene sets divided into 9 major collections and several subcollections. Users can choose to view the gene sets on the matrix plot.

Click on the `50 Genes` button. Then click on the `Gene set - Edit`. Here user can see a button with a dropdown for loading MSigDB genes.

[![msigdb dropdown](./images/oncomatrix/msigdb1.png)](./images/oncomatrix/msigdb1.png 'Click to see the full image.')

Click on this dropdown to display a tree for the different gene sets.

[![msigdb tree](./images/oncomatrix/msigdb-tree.png)](./images/oncomatrix/msigdb-tree.png 'Click to see the full image.')

Select `C2: curated gene sets` and select `NABA_COLLAGENS` as shown below.

[![NABA collagens gene set](./images/oncomatrix/naba-collagens.png)](./images/oncomatrix/naba-collagens.png 'Click to see the full image.')

This loads the following genes as shown below.

[![NABA collagens genes loaded](./images/oncomatrix/44-msigdb2.png)](./images/oncomatrix/44-msigdb2.png 'Click to see the full image.')

Click `Submit` and the matrix will update to reflect the selected MSigDB gene set as shown.

[![NABA collagens genes matrix](./images/oncomatrix/45-msigdb_collagen_matrix.png)](./images/oncomatrix/45-msigdb_collagen_matrix.png 'Click to see the full image.')

## Variables

The third button from the left called `Variables` allows user to add in additional variables in the form of rows on the matrix. Click `Variables` to display a tree of variables and select `Disease type`, `Index date` and `Primary site`. Click the button `Submit 3 variables` as shown.

[![Variables menu](./images/oncomatrix/46-variables.png)](./images/oncomatrix/46-variables.png 'Click to see the full image.')

This updates the chart to display the selected variables on the very top of the matrix as shown below. User may choose to configure these rows by following steps outlined in section Clicking on gene/variable labels.

[![Loaded variables](./images/oncomatrix/47-variable-loaded.png)](./images/oncomatrix/47-variable-loaded.png 'Click to see the full image.')

## Cell Layout

The cell layout menu enables customization of the appearance such as cell dimensions, spacing, font sizes, and borders. You may mouseover an input to see the description for that input, or try checking or editing inputs to test the effects of the control input and undo/redo as needed.

[![Cell layout](./images/oncomatrix/48-cell_layout.png)](./images/oncomatrix/48-cell_layout.png 'Click to see the full image.')

## Legend Layout

The legend layout menu enables customization of the appearance of the legend, such as dimensions, spacing, and font sizes. These customizations can help avoid or minimize the need for post-download edits when generating figures.

[![Legend layout](./images/oncomatrix/49-legend_layout.png)](./images/oncomatrix/49-legend_layout.png 'Click to see the full image.')

## Zooming

The matrix plot offers an interactive zoom panel as shown below with which a user can zoom in to view individual samples. There are two ways to use this panel. One by changing the input number and second by sliding the zoom bar to a desired zoom level as shown.

[![Zoom slider](./images/oncomatrix/zoom_slider.png)](./images/oncomatrix/zoom_slider.png 'Click to see the full image.')

Change zoom level to 10+ as shown.

[![Zoom slider plus](./images/oncomatrix/zoom-slider-plus.png)](./images/oncomatrix/zoom-slider-plus.png 'Click to see the full image.')

Scroll down to view individual samples at the bottom of the plot as shown below.

[![Zoomed in matrix view](./images/oncomatrix/52-zoomed-in.png)](./images/oncomatrix/52-zoomed-in.png 'Click to see the full image.')

The zoom action can also be implemented by following steps as outlined in section - Drag to zoom.

## Disco Plot

Click on any sample to reveal a second type of plot called as the `Disco Plot` as shown.

[![Sample label clicking to show disco plot button](./images/oncomatrix/53-disco_plot1.png)](./images/oncomatrix/53-disco_plot1.png 'Click to see the full image.')

Click on `Disco plot` as shown above in gray. This loads a new chart above the matrix plot as shown below.

[![Disco plot](./images/oncomatrix/54-disco_plot2.png)](./images/oncomatrix/54-disco_plot2.png 'Click to see the full image.')

This plot shows all the mutations and CNV associated with that sample id as shown above. The plot also displays the legend for the mutation class and the CNV.

To reset the zoom level to default, click on the `Reset` button as shown. This will reset the zoom level to a default of 1.0

[![Reset Zoom](./images/oncomatrix/zoom_slider.png)](./images/oncomatrix/zoom_slider.png 'Click to see the full image.')

<!--
## Undo/Redo/Restore

User may also choose to undo settings by clicking the undo button on the control panel as shown below.

[![Undo Zoom](./images/oncomatrix/56-undo_zoom.png)](./images/oncomatrix/56-undo_zoom.png 'Click to see the full image.')

Click `undo` to go back to the previous zoom level as shown above.

User may also choose to restore the initial state by clicking the Restore button as shown.

[![Restore](./images/oncomatrix/57-restore.png)](./images/oncomatrix/57-restore.png 'Click to see the full image.')

This restores the chart back to default settings as shown.

[![Default matrix on restore](./images/oncomatrix/58-default-view.png)](./images/oncomatrix/58-default-view.png 'Click to see the full image.')
-->

## Download

The control panel shows an option to download the plot as an svg after user has specified their customizations. Select the `Download` button as shown below to save the svg.

[![Download button](./images/oncomatrix/59-download_svg1.png)](./images/oncomatrix/59-download_svg1.png 'Click to see the full image.')

The download will get saved to the default download folder as shown at the bottom of the browser window.

[![Download](./images/oncomatrix/download.png)](./images/oncomatrix/download.png 'Click to see the full image.')

## Legend

The legend for the matrix is below the plot and shows color coding for different mutation classes as well as color codes for CNV as shown here. This legend is interactive and user may choose to hide or show features such as mutation classes or copy number changes.


Click on the legend icons to hide anything.

[![Legend](./images/oncomatrix/61-legend.png)](./images/oncomatrix/61-legend.png 'Click to see the full image.')


# Cohort Builder

The Cohort Builder is a good starting point for users looking to gather information for a specific disease, project, or group of patients. Building a cohort allows users to download files, perform analyses, and query metadata for the same group of cases in multiple sections of the GDC Data Portal. This section will cover the process of building a cohort and downstream actions will be documented in their respective sections.

> **NOTE:** Filters within the Cohort Builder are applied to the cases in your cohort. If you wish to target specific types of files for download, use the filters within the Repository.

The Cohort Builder can be accessed in one of the following ways:

* Selecting the Cohort Builder link in the GDC Data Portal header

[![Cohort Builder In Header](images/ToolLinksInHeader.png)](images/ToolLinksInHeader.png "Click to see the full image.")

* Selecting the play button on the Cohort Builder card in the Analysis Center

[![Cohort Builder Play Buttons](images/CohortBuilderInAnalysisCenter.png)](images/CohortBuilderInAnalysisCenter.png "Click to see the full image.")

## Cohort Builder Panel ##

[![Cohort Builder Panel](images/CohortBuilderPanel.png)](images/CohortBuilderPanel.png "Image of Cohort Builder Panel. Click to see the full image.")

The Cohort Builder tool will be displayed as a panel in the Analysis Center and is used to filter the current cohort to a specific set of cases. The current cohort is always displayed in the [main toolbar](getting_started.md#main-toolbar) and can be changed from the main toolbar.

At the left side of the panel are a series of broad filter categories can be selected. Each filter category contains a set of specific filters within cohort builder cards that can be used to narrow your cohort to the desired set.

## Cohort Builder Cards
Each card within the Cohort Builder can be used to apply the corresponding filters on the current cohort. As filters are applied, they will be displayed on the [Query Expressions](getting_started.md#query-expressions) section.

Additional features can be accessed at the top right of each card's header to facilitate filtering:

* **Search**: the search icon can be selected to reveal or hide a search field for entering text to search within the values of the current card. This feature is only available when the values are enums.
* **Flip Card**: cards can be flipped to reveal or hide a summary chart. This feature is only available when the values can be meaningfully displayed as bar graphs.
* **Reset Card**: this button will reset any filtering that has been applied within the card

[![Cohort Builder Card Features](images/search_flip_reset_card.png)](images/search_flip_reset_card.png "Image of Cohort Builder Card Features. Click to see the full image.")

In addition, filters in each card can be sorted, either alphabetically or by the number of cases based on current filters, by selecting one of the two headers directly underneath the card title. The default sort is alphabetical order.

[![Cohort Builder Card Filter Sort](images/alpha_desc_card_filter_order.png)](images/alpha_desc_card_filter_order.png "Image of Cohort Builder Card Filter Sort. Click to see the full image.")

The first six (or fewer) filters are shown for each card, but can be expanded to show 20 filters at once by clicking the "+" button which also indicates the number of additional filters not in view. The expanded view can be toggled off by clicking the resulting "show less" button.

[![Cohort Builder Card Filter Expand](images/card_expand.png)](images/card_expand.png "Image of how to expand cohort builder card filters. Click to see the full image.")

### Biospecimen Filters
The filters within the "Biospecimen" category allow for cases that have certain types of biospecimens. For example, filtering in the "Tissue Type" card for "normal" will ensure that cases within your cohort have a normal tissue type. These filters may be useful for studies that require only cases for which a certain type of biospecimen is available.

### Available Data Filters

Toward the bottom of the list of filter categories, "Available Data" can be selected.  These filters differ from the other default filters as they allow for cases that have certain types of associated data files. For example, filtering in the "Experimental Strategy" card for "RNA-Seq" will only display cases in the active cohort that have associated RNA-Seq files. These filters may be useful for studies that require only cases for which a certain type of analysis was performed.

> **NOTE:** Some of the "Biospecimen" and "Available Data" filters are also available in the Repository, where they may be used to specify the types of files for browsing and download. For example, selecting "WXS" in the "Experimental Strategy" card will ensure that all cases in your cohort have WXS data. This, however, does not mean that your cohort will not also have other types of data, such as RNA-Seq or WGS. Likewise, selecting "normal" in the "Tissue Type" card narrows your cohort to those cases that have normal samples, but does not exclude cases with other types of samples. If the goal is to view or download only the files of a specific type, use the filters in the Repository rather than the filters located here in the Cohort Builder.

## Custom Filters ##

If a filter cannot be found within one of the categories, use the "Add a Custom Filter" button in the "Custom Filters" category to access any filters that are not displayed. Browse through the list of additional filters, or use the search box to search for filters by name. Once a filter is selected, it is then added to the "Custom Filters" category.  A custom filter can be removed from this category by choosing the "X" at the top right of the filter card.

[![Custom Filter](images/CustomFilter.png)](images/CustomFilter.png "Image of Custom Filter search box. Click to see the full image.")

Filters that exist in the GDC but do not have any cases that have a value for the filter can also be removed from the "Custom Filters" list by selecting the "Only show properties with values" box.

[![Custom Filter Only Show Props Vals](images/only_show_props_vals.png)](images/only_show_props_vals.png "Image of toggle for properties with values for custom filters in cohort builder. Click to see the full image.")

## Cohort Builder Search

The Cohort Builder includes the ability to search across all the cards within it. This feature is located on the right of the Cohort Builder header.

[![Cohort Builder Header](images/CohortBuilderHeader.png)](images/CohortBuilderHeader.png "Image of Cohort Builder Header. Click to see the full image.")

As a search term is entered, the Cohort Builder Search feature will display a list of properties that contain matching results. When a result is moused over, additional information is displayed to its left, including a description of the property and a list of values that match the search term.

[![Cohort Builder Search Results](images/CohortBuilderSearchResults.png)](images/CohortBuilderSearchResults.png "Image of Cohort Builder Search Results. Click to see the full image.")

When a result is selected, the card corresponding to the selected result will be displayed.

## Closing the Cohort Builder

Once a custom cohort is built and filtering is complete, users can close the Cohort Builder and use the custom cohort with other tools.

To close the Cohort Builder panel and display all the tools within the Analysis Center, click on the "X" button on the left of the Cohort Builder header.

[![Cohort Builder Header](images/CohortBuilderHeader.png)](images/CohortBuilderHeader.png "Image of Cohort Builder Header. Click to see the full image.")

Alternatively, users can select the Analysis Center link or any of the other links on the GDC Data Portal header to close the Cohort Builder.

[![Header with Links to Tools](images/ToolLinksInHeader.png)](images/ToolLinksInHeader.png "Click to see the full image.")

Changes made to the cohort with the Cohort Builder will persist through the other sections of the GDC Data Portal.

Users can then perform the following actions:

* Download files associated with the cohort from the [Repository](Repository.md)
* Analyze data from the cohort in the [Analysis Center](analysis_center.md)

## Cohort Types

Depending on how they are modified or created, cohorts can have different types of filters and thus behave differently after a data release with regard to the cases they contain. The following are the types of filters cohorts can have:

* __Custom Queries__
* __Specific List of Cases__

### Custom Queries

A very common way to modify a cohort is by using the filters in the Cohort Builder. Using the filters in the Cohort Builder to build a cohort will create a cohort with custom queries (see note below for an exception). Examples of cohorts with custom queries would be:

1. All cases in the `TCGA-BRCA` project.
1. Cases with a primary site of `brain` and a gender of `male`.

Cohorts based on custom queries will change depending on the data available. For example, if a data release adds cases to the TCGA-BRCA project, the first cohort example will include the new cases automatically and increase in size. 

The query expression section will display these custom queries with information about the properties and values that were applied as filters to the cohort.

[![Query Expressions section with TCGA-BRCA cohort](images/QueryExpressionsTCGABRCA.png)](images/QueryExpressions.png "Click to see the full image.")

[![Query Expressions section with male brain cancer cohort](images/QueryExpressionsBrainMale.png)](images/QueryExpressionsBrainMale.png "Click to see the full image.")

**NOTE:** The Case ID filter in the Cohort Builder will result in a cohort based on a specific list of cases.

### Specific List of Cases

Cohorts can also be based on a list of specific cases. These cases do not necessarily share common properties and can comprise any group of released cases. Data releases that happen after these cohorts are created will not add additional cases to the cohort, but could subtract cases if there were redactions. A common way to create cohorts based on a list of specific cases is to use the Import New Cohort function in the Cohort Bar. Another common way is to create the cohorts from one of the many analysis tools available in the GDC (e.g. Clinical Data Analysis, Mutation Frequency, or Set Operations).

The query expression section will display a case UUID with the Case ID property if the filter has only 1 specific case. Otherwise, it will display the number of cases in the list.

[![Query Expressions section with 1 specific case](images/QueryExpressions1Case.png)](images/QueryExpressions1Case.png "Click to see the full image.")

[![Query Expressions section with many specific cases](images/QueryExpressionsManyCases.png)](images/QueryExpressionsManyCases.png "Click to see the full image.")

> **NOTE:** 
If an imported cohort was originally created by exporting a cohort with custom queries, it will still result in a cohort with specific cases. The export function saves a list of cases, but does not preserve the custom queries used to filter for those cases.

# Cohort Comparison

The Cohort Comparison tool displays graphs and tables that demonstrate the similarities and differences between the active cohort and a different cohort. The following features are displayed for each of the two cohorts:

* A key detailing the number of cases in each cohort and the color that represents each (blue/orange)

* A Venn diagram, which shows the number of cases shared between the cohorts

* A selectable survival plot that compares both sets with information about the percentage of represented cases

* A breakdown of each cohort by selectable clinical facets with a bar graph and table. The facets included are `Vital_Status`, `Gender`, `Race`, `Ethnicity`, and `Age_at_Diagnosis`. A p-value (if it can be calculated from the data) that demonstrates whether the statuses are proportionally represented is displayed for the `Vital_Status`, `Gender`, and `Ethnicity` facets.

* Additional cohorts can be created containing subsets of these two cohorts

[![Cohort Comparison](images/cohort_comparison_page.png)](images/cohort_comparison_page.png "Click to see the full image.")

Note that clicking the "Open Venn diagram" link will launch the [Set Operations](set_operations.md) tool with the same cohorts used in the Cohort Comparison tool.


# GDC Analysis Center


The Analysis Center is the central hub for accessing the tools that support cohort analysis. The Analysis Center can be accessed by clicking on the 'Analysis Center' icon in the GDC Data Portal header, the "Explore Our Cancer Datasets" button on the home page, or  one of the sites in the human anatomical outline or bar graph.

[![Analysis Center View](images/FullAnalysisCenter.png)](images/FullAnalysisCenter.png "Click to see the full image.")

The Analysis Center consists of a main toolbar and a query expressions section, both of which are always displayed. The main toolbar displays the active cohort and can be used to create and manage custom cohorts. The query expression section displays the filters applied to the active cohort.

Available tools are displayed under the Query Expression section of the Analysis Center. Each analysis tool is showcased within a tool 'card', which has several items related to the analysis tool such as:

* A teal 'Play' button to launch the analysis tool on the given cohort
* A 'Demo' button that launches a demonstration of the analysis tool on an example cohort
* Clicking on the name of the analysis tool in the tool card toggles a drop down description of the analysis tool
* The number of cases from the cohort that the analysis will be performed on is at the bottom of the card

## Core Tools ##

The 'Core Tools' section contains the GDC tools that constitute the main functionality of the Data Portal.

* [Projects](Projects.md) tool
* [Cohort Builder](cohort_builder.md)
* [Repository](Repository.md)

## Analysis Tools ##

The 'Analysis Tools' section contains the tools available for specific analyses available for the active cohort.

* [BAM Slicing Download](BAMslicing.md)
* [Clinical Data Analysis](clinical_data_analysis.md)
* [Cohort Comparison](cohort_comparison.md)
* [Cohort Level MAF](cohortMAF.md)
* [Gene Expression Clustering](gene_expression_clustering.md)
* [Mutation Frequency](mutation_frequency.md)
* [OncoMatrix](oncomatrix.md)
* [ProteinPaint](proteinpaint_lollipop.md)
* [Sequence Reads](proteinpaint_bam.md)
* [Set Operations](set_operations.md)

Each can be launched by clicking the Play buttons in each of the tool cards.

If there is not sufficient data in the active cohort to use a particular tool, the play button will be grayed out and will not be usable until a new cohort with sufficient data is selected.

[![Analysis Center Tools](images/AnalysisCenterTools.png)](images/AnalysisCenterTool.png "Click to see the full image.")

## Tool Panel

As each tool is selected, it is loaded in the 'Analysis Center' within a panel.

[![Analysis Center Tools Panel](images/AnalysisCenterToolPanel.png)](images/AnalysisCenterToolPanel.png "Click to see the full image.")

To close a tool and return to the default view that displays all the tool cards within the Analysis Center, click the "X" to the left of the tool's header.


# ProteinPaint Sequence Reads Tool

## Introduction to ProteinPaint Sequence Reads

Sequence Reads is a web-based tool that uses the ProteinPaint BAM track and NCI Genomic Data Commons (GDC) BAM Slicing API to allow users to visualize read alignments from a BAM file. Given a variant (i.e. Chromosome number, Position, Reference Allele and Alternative Allele), the Sequence Reads tool can classify reads supporting the reference and alternative allele into separate groups.

## Quick Reference Guide

At the Analysis Center, click on the 'Sequence Reads' card to launch the app.

[![Analysis Center](images/seq_read_vis.png)](images/seq_read_vis.png "Click to see the full image.")

This feature requires access to controlled data, which is maintained by the Database of Genotypes and Phenotypes (dbGaP) [See Obtaining Access to Controlled Data](https://gdc.cancer.gov/access-data/obtaining-access-controlled-data). In order to use this tool, users must be logged in with valid credentials. Otherwise, users will be prompted to login.

### Selecting BAM Files and Variants

Once logged in, the Sequence Reads tool will display a search bar, as well as a link to browse the first 1,000 available BAM files for the active cohort. Users can choose to select a BAM from the available list, or search for a specific BAM file by entering four types of inputs: file name, file UUID, case ID, or case UUID.

The tool will verify the query string and return all matching GDC BAM files in a table, from which the user can select one or multiple to use with the tool.

[![Search Bar](images/proteinpaint_sequence_reads_BAM_search.png)](images/proteinpaint_sequence_reads_BAM_search.png "Click to see the full image.")

If an exact match is entered (i.e. a file name or file UUID), the Sequence Reads tool will find that BAM and present brief information about the file.

[![File Info](images/file_info.png)](images/file_info.png "Click to see the full image.")

In the subsequent section, the mutation table displays somatic mutations catalogued by the GDC for this case, if available. Users can select a mutation to visualize read alignments on this variant.

[![Somatic Mutations](images/som_mut.png)](images/som_mut.png "Click to see the full image.")

Alternatively, the `Gene or position` button at the top of the mutation table allows users to enter a custom genomic region for BAM visualization.

[![Gene Search](images/gene_search.png)](images/gene_search.png "Click to see the full image.")

Once at least one BAM file is selected and a gene, position, SNP, or variant is entered, the Sequence Reads tool will display the BAM visualization.

[![ProteinPaint Genome Browser](images/pp_gen_browse.png)](images/pp_gen_browse.png "Click to see the full image.")

### Toolbar

* __Current Position in Genome:__ Displays the coordinates of the region currently displayed
* __Reference Genome Build:__ Refers to the genome build that was used for mapping the reads; the GDC uses Reference Genome Build 38 (hg38)
* __Zoom Buttons:__ Zooms in (`In`) or out (`Out x2`, `x10`, and `x50`) of the current view

### Reference Genome Sequence

The Reference Genome Sequence displays the reference genome build against which the reads have been aligned.

### Gene Models

The Genome Models row displays the gene model structure from the view range. When zoomed into a coding exon, the letters correspond to the 1-letter amino acid code for each amino acid and are placed under its corresponding 3-letter nucleotide codon under the reference genome sequence. The arrows describe the orientation of the strand of the gene model being displayed (right arrow for forward strand and left arrow for reverse strand).

Graphical representations of the reads are displayed as they are aligned on the chromosome. Sequence can be read when zoomed in.

### ProteinPaint BAM Track

#### Pileup Plot

The Pileup Plot shows the total read depth at each nucleotide position of the region being displayed.

[![Pileup plot](images/pileup_plot.png)](images/pileup_plot.png "Click to see the full image.")

#### Read Alignment Plot

This visual contains the main read alignment plot of the reads from the BAM file.

[![Read alignment plot](images/align_plot.png)](images/align_plot.png "Click to see the full image.")

When completely zoomed out, base-pair quality of each nucleotide in each read is not displayed. Users can zoom into the plot via the toolbar or by dragging on the genomic ruler (a) to zoom into the selected region (b).

[![Zoom](images/zoom.png)](images/zoom.png "Click to see the full image.")

Clicking on a read in the plot launches a window that displays the alignment between the read and reference sequence, as well as the chromosome, coordinates, read length, template length, CIGAR score, flag, and name. If the read is paired, the position of the other segment will be displayed below. This pop-up also contains two buttons, `Copy read sequence` and `Show gene model`, which copy the nucleotide sequence of the read to the computer clipboard and display the gene model, respectively.

[![Read Information Panel](images/read_info.png)](images/read_info.png "Click to see the full image.")

#### Mutation Rendering

Mutations are rendered as follows:

* __Insertion:__ The alphabet representing the nucleotides is displayed between the two reference nucleotides in cyan color, with the shade scaled by base quality
    * If more than one nucleotide is inserted, a number is printed between the two reference nucleotides indicating the number of inserted nucleotides
    * Clicking the read with multiple insertions will display the complete inserted nucleotide sequence

    [![Insertion](images/sn_insert4.png)](images/sn_insert4.png "Click to see the full image.")

* __Deletion:__ A black line represents the span of deleted bases

    [![Deletion](images/sn_deletion.png)](images/sn_deletion.png "Click to see the full image.")

* __Substitution or Mismatch:__ The substituted nucleotide is highlighted in red background, with the shade of red scaled by base quality

    [![Substitution](images/sn_sub.png)](images/sn_sub.png "Click to see the full image.")

* __Splicing:__ The different fragments of a read separated due to splicing are joined by a gray line

    [![Splicing](images/splicing.png)](images/splicing.png "Click to see the full image.")


#### Color Coding of Reads

Color codes in the background of the read describe the quality of the read alignment and its mate (in case of paired-end sequencing). These colors are assigned both on the basis of the CIGAR sequence (if it contains a softclip) and the flag value of both the read and its mate.

* __Gray:__ Suggests that both the read (at least part of it) and its mate are properly aligned and the insert size is within expected range
* __Blue:__ Indicates that part of the read is soft clipped
* __Brown:__ Indicates that the mate of the read is unmapped
* __Green:__ Indicates that the template has the wrong insert size
* __Pink:__ Indicates that the orientation of the read and its mate is not correct
* __Orange:__ Indicates the read and its mate are mapped in different chromosomes

#### BAM Track Configuration Panel

The BAM Track Configuration Panel, which can be accessed by clicking the `CONFIG` option next to the Pileup plot, provides buttons for toggling between single-end and paired-end mode.

[![BAM Track Configuration Panel](images/BAM_track_config.png)](images/BAM_track_config.png "Click to see the full image.")

In single-end display each read is displayed individually without displaying any connections with its respective mate. In case of the paired-end display the two paired reads are joined by a gray dotted-line if the coordinates of the two reads do not overlap. When the coordinates of the two read-pairs overlap, the overlapped region is highlighted by a blue line.

The BAM Track Configuration Panel also provides a check box to show/hide PCR and optical duplicated reads.

## Launch the Sequence Reads Tool
At the Analysis Center, click on the "Sequence Reads" card to launch the app.

[![Analysis Center](images/seq_read_vis.png)](images/seq_read_vis.png "Click to see the full image.")

A user needs to be logged in to use this feature. If not, the user will be prompted to log in.
Once the user logs in, a search bar and submit button will appear as below.

[![Search Bar](images/search_bar.png)](images/search_bar.png "Click to see the full image.")

## Find and Display BAM Files in GDC
To find a BAM file in GDC, a user can enter four types of inputs including file name, file UUID, case ID, or case UUID. The tool will verify the query string and return matching BAM files.

As an example, using case ID "TCGA-06-0211" will return 9 BAM files available from this case displayed in a table. One or multiple BAM files from this table must be selected to proceed.

[![Query](images/query.png)](images/query.png "Click to see the full image.")

When a file name or UUID is provided, it will display brief information about the file. A user does not need to select anything here as the file is automatically selected.

[![File Info](images/file_info.png)](images/file_info.png "Click to see the full image.")

The subsequent section displays somatic mutations catalogued by GDC for this case, if available. A user can select a mutation to visualize read alignment on this variant.

[![Somatic Mutations](images/som_mut.png)](images/som_mut.png "Click to see the full image.")

Alternatively, a user can enter a custom genomic region for BAM visualization. At the toggle button on top of the mutation table, click the "Gene or position" option to show the gene search box.

[![Gene Search](images/gene_search.png)](images/gene_search.png "Click to see the full image.")

Follow the instructions to enter gene, position, SNP, or variant. Press ENTER to validate the input.

Lastly, press the "Submit" button to view read alignment from the selected BAM file over the selected mutation or genomic region. The server will verify the user's access to the requested BAM file and query the GDC API to slice the BAM file at the selected region. This may take from 10 seconds to a minute.

An error message will appear if the user does not have access to the requested BAM file. Please follow the instructions to obtain access.

[![Access Alert](images/access_alert.png)](images/access_alert.png "Click to see the full image.")

Once the BAM visualization is successfully displayed, the search interface is hidden, and a button named "Back to input form" is shown. Clicking the button will bring the user back to the search interface so a user can change the BAM file or mutation.

[![BAM Visualization](images/BAM_vis.png)](images/BAM_vis.png "Click to see the full image.")

Click the "Download GDC BAM Slice" button to download the BAM slice file used in this visualization.

## Using ProteinPaint Genome Browser

[![ProteinPaint Genome Browser](images/pp_gen_browse.png)](images/pp_gen_browse.png "Click to see the full image.")

Various fields labeled in the above figure are described below:

## Current Position in Genome  
The Current Position in Genome text box displays the coordinates of the region currently displayed on the screen. It initially shows the coordinates specified in the URL. On pan/zoom by the user, this region displays the updated coordinates of the view region.

## Reference Genome Build
The Reference Genome Build button refers to the genome build specified by the user that was used for mapping the reads. The GDC uses Reference Genome Build 38 (hg38).

## Zoom Buttons
A user can zoom in/out of the current view by clicking the "In" (zoom in) or "Out x2" (zoom out) buttons. By clicking on the x10 and x50 button, a user can zoom out 10 and 50-fold respectively. Alternatively, a user may choose to zoom into a smaller region by dragging on the genomic ruler (a) to zoom into the selected region (b) as shown below.

[![Zoom](images/zoom.png)](images/zoom.png "Click to see the full image.")

## Reference Genome Sequence
The Reference Genome Sequence displays the reference genome build against which the reads have been aligned.

## Gene Models
This Genome Models row displays the gene model structure from the view range. When zoomed into a coding exon, the letters correspond to the 1-letter amino acid code for each amino acid and are placed under its corresponding 3-letter nucleotide codon under the reference genome sequence. The arrows describe the orientation of the strand of the gene model being displayed (right arrow for forward strand and left arrow for reverse strand).

# ProteinPaint BAM Track Features

## Pileup Plot

[![Pileup plot](images/pileup_plot.png)](images/pileup_plot.png "Click to see the full image.")

The Pileup Plot shows the total read depth at each nucleotide position of the region being displayed.
Color codes of bars representing various possibilities:

Gray - Reference allele nucleotides
Blue - Soft clipped nucleotides
Mismatches:
* nucleotide "A" - Red (color code: #ca0020)
* nucleotide "T" - Orange (color code: #f4a582)
* nucleotide "C" - Light blue (color code: #92c5de)
* nucleotide "G" - Dark blue (color code: #0571b0)

## Read Alignment Plot

[![Read alignment plot](images/align_plot.png)](images/align_plot.png "Click to see the full image.")

The Read Alignment Plot contains the main read alignment plot of the reads from the BAM file.

## Rendering of Various Mutations

### Insertion
In case of a single nucleotide insertion, the alphabet representing the nucleotide (A/T/C/G) is displayed between the two reference nucleotides in cyan color.

[![Insertion](images/sn_insert1.png)](images/sn_insert1.png "Click to see the full image.")

Darkness of the inserted nucleotide is determined by the base quality, as an example below of an inserted T with low quality.

[![Insertion](images/sn_insert2.png)](images/sn_insert2.png "Click to see the full image.")

If more than one nucleotide is inserted, a number is printed between the two reference nucleotides indicating the number of inserted nucleotides. The text color is full cyan and does not account for the quality of inserted bases. Showing below is a read with two insertions, first with 2 bases, and second with T.

[![Insertion](images/sn_insert3.png)](images/sn_insert3.png "Click to see the full image.")

On clicking this read, the read information panel is displayed where the complete inserted nucleotide sequence is shown in cyan color.

[![Insertion](images/sn_insert4.png)](images/sn_insert4.png "Click to see the full image.")

### Deletion
A black line represents the span of deleted bases.

[![Deletion](images/sn_deletion.png)](images/sn_deletion.png "Click to see the full image.")

### Substitution (or Mismatch)
In case of substitutions (or mismatches), the substituted nucleotide ("A") is highlighted in red background, with the shade of red scaled by base quality.

[![Substitution](images/sn_sub.png)](images/sn_sub.png "Click to see the full image.")

### Splicing
In case of splicing, the different fragments of a read separated due to splicing are joined by a gray line as shown below. In the example below, the reads contain spliced fragments that are separated by a 1915bp intron.

[![Splicing](images/splicing.png)](images/splicing.png "Click to see the full image.")

## Zooming the Read Alignment Plot
The rendering of the reads depends upon the zoom level (horizontal zoom) chosen by the user and the number of reads mapped at the display region (vertical zoom).

### Horizontal Zoom   
The BAM track has three levels of horizontal zoom:

### Overview Level
This is the completely zoomed out mode (shown below). [At this resolution,](https://proteinpaint.stjude.org/?genome=hg19&block=1&bamfile=TP53_del,proteinpaint_demo/hg19/bam/TP53_del.bam&position=chr17:7575308-7580395&bedjfilterbyname=NM_000546)
base-pair quality of each nucleotide in each read is not displayed as
each read occupies a very small area on the screen. Also the reference
sequence at the top is not displayed. Only reads which contain big
insertions/deletions/softclips or are discordant are represented by
their respective colors Also see the section on color codes of various reads.

[![Overview Level](images/overview.png)](images/overview.png "Click to see the full image.")

### Base-pair Quality Level
[At this level of zoom](https://proteinpaint.stjude.org/?genome=hg19&block=1&bamfile=TP53_del,proteinpaint_demo/hg19/bam/TP53_del.bam&position=chr17:7577215-7578486&bedjfilterbyname=NM_000546) (shown below), in addition to color codes of reads, the phred base pair quality score of each read is also displayed. Poor base-pair quality of nucleotides is represented by lighter shades of the respective color and darker shades represent high base-pair quality. For example, dark gray color represents a higher quality nucleotide in a properly mapped read than light gray which represents poor base-pair quality.

[![Base-pair Quality Level](images/base-pair_qual.png)](images/pair_qual.png "Click to see the full image.")

### Base-pair Resolution Level
[At this resolution](https://proteinpaint.stjude.org/?genome=hg19&block=1&bamfile=TP53_del,proteinpaint_demo/hg19/bam/TP53_del.wrongbp.bam&position=chr17:7578371-7578417&bedjfilterbyname=NM_000546), all information including the read sequence of each read is displayed along with reference genome nucleotides at the top. For simplicity (as discussed later under [vertical zoom](#Vertical zoom: examining subset of reads)), only a few reads are shown in the figure below.

[![Base-pair Resolution Level](images/base-pair_res.png)](images/pair_res.png "Click to see the full image.")

### Vertical Zoom: Examining Subset of Reads
ppBAM can display up to 7000 reads, and will downsample if the number of reads in a region is over 7000. This is especially helpful for displaying high-depth sequencing data. However, displaying nucleotides from each read for such a large number of reads is not feasible. Therefore, the pixel width of each read is reduced to accommodate all reads in the region (Panoramic view, figure below). When the user clicks on a read, that part of the alignment stack is enlarged to show the nucleotides within each read (Nucleotide view, figure below) stacked near the cursor click. Reads at the top and bottom of the stack can be viewed by scrolling up/down with the scroll-bar. The top/bottom of the green scroll-bar can be adjusted to display more reads on the screen by reducing the individual width of each read. On clicking the gray area of the scroll bar region, the panoramic view is displayed again.

[![Vertical Zoom](images/vert_zoom.png)](images/vert_zoom.png "Click to see the full image.")

# BAM Track Configuration Panel

The BAM Track Configuration Panel can be accessed by clicking the "CONFIG" option next to the pileup plot. The BAM Track Configuration Panel (shown below) provides buttons for toggling between single-end and paired-end mode. It also provides a check box to show/hide PCR and optical duplicated reads.

## BAM track configuration panel figure

[![BAM Track Configuration Panel](images/BAM_track_config.png)](images/BAM_track_config.png "Click to see the full image.")

## Single and Paired-end Read
[The configuration panel](#BAM track configuration panel) (above) provides a toggle to change view between single-end (default) and paired-end view (shown below: see [Link](https://proteinpaint.stjude.org/?genome=hg19&block=1&bamfile=crebbp_del,proteinpaint_demo/hg19/bam/crebbp.bam&position=chr16:3800245-3803429) for example). In single-end display each read is displayed individually without displaying any connections with its respective mate. In case of the paired-end display the two paired reads are joined by a gray dotted-line if the coordinates of the two reads do not overlap. When the coordinates of the two read-pairs overlap, the overlapped region is highlighted by a blue line.

The following shows reads in single-end mode.

[![Single-end Mode](images/single_end.png)](images/single_end.png "Click to see the full image.")

The same track above shows in paired mode.

[![Paired Mode](images/paired_mode.png)](images/paired_mode.png "Click to see the full image.")

## Show/Hide Read Names
Read names are available only when the variant field is
specified. There is a checkbox that displays read names on the left
side of the main BAM track as shown below. The read names are only
displayed when the main BAM track has base-pair level resolution and
is in nucleotide view (see section on vertical zoom in case of high-depth sequencing data).

[![BAM Track](images/BAM_track.png)](images/BAM_track.png "Click to see the full image.")

## Displaying PCR and Optical Duplicated Reads

[![PCR and Optical Duplicated Reads](images/PCR.png)](images/PCR.png "Click to see the full image.")

The checkbox in the configuration panel can be toggled to switch on/off the display of PCR and optical duplicates. [In the above figure](https://proteinpaint.stjude.org/?genome=hg19&block=1&bamfile=long_del,proteinpaint_demo/hg19/bam/PCRopticalduplicates.bam&position=chrX:129150008-129150048&variant=chrX.129150027.GGAAAGCGTAGGGGTCTTTGCTTGCAAGAACAA.GT&Bedjfilterbyname=NM_001184772), a total of 29 reads are shown when PCR/optical duplicates are displayed (Figure a) whereas a total of 19 reads are displayed supporting the alternative allele when PCR/optical duplicates are not displayed (default, Figure b).

## Strictness
Strictness of the on-the-fly genotyping analysis. This option is available when the BAM track is performing on-the-fly genotyping against a variant. The user can toggle between Lenient and Strict (default) mode as shown in the ppBAM configuration panel.

# Read Information Panel

For displaying the various features of individual reads, on clicking a particular read (in nucleotide view) a new panel opens displaying the information about the selected read (as shown below).

[![Read Information Panel](images/read_info.png)](images/read_info.png "Click to see the full image.")

In this panel (as shown above), the top row shows the reference sequence that is aligned to the read. The second row shows the nucleotide sequence of the read. The colors of the nucleotides of the read are based on the CIGAR sequence of the read and follow the color codes as described in the section color coding of reads. In the third row, three clickable buttons are available which have the following functions as described below. The fourth row contains the start, stop, read length, template length, CIGAR sequence, flag and name of read.

When the read sequence is not available, but the CIGAR sequence is available then the read panel displays the message "Nucleotide sequence not available for this read".

[![No Read Sequence](images/no_read_sequence.png)](images/no_read_sequence.png "Click to see the full image.")

## Copy Read Sequence
The Copy Read Sequence feature copies the nucleotide sequence of the read being displayed to the computer clipboard so that it can be pasted outside of ppBAM.

## Show Gene Models
On clicking the Show Gene Models button, the gene model (as shown below) is displayed.

[![Gene Model](images/gene_model.png)](images/gene_model.png "Click to see the full image.")

## Read Details

The fourth row contains details about the read present in the BAM file

* **CHR** -Chromosome ID of the read.

* **START** -Contains the start position of the read.

* **STOP** -Contains the stop position of the read.

* **This Read** -Contains length of the read.

* **TEMPLATE** -Contains length of the template of which the current read is part of.

* **CIGAR** -Contains CIGAR sequence of the read.

* **FLAG** -Contains the flag number (from BAM file) of the read.

* **NAME** -Contains the name of the read.

## Color Coding of Reads

(a)    Paired-end view

[![Paired-end View](images/paired_end.png)](images/paired_end.png "Click to see the full image.")

(b)    Base-pair resolution mode showing nucleotides of each of the reads

[![Base-pair Resolution Mode](images/base-pair_res2.png)](images/pair_res2.png "Click to see the full image.")

In the figure above, [structural](https://proteinpaint.stjude.org/?genome=hg19&block=1&bamfile=crebbp_del,proteinpaint_demo/hg19/bam/crebbp.bam&position=chr16:3800245-3803429) variant deletion in the CREBBP gene is shown. Reads near the vicinity of the deletion have various colors (gray, green, brown and blue) based on their features as explained below. In the paired-end view (a) an overview of the deletion is shown. In Base-pair resolution mode (b) showing nucleotides of each of the reads there are softclipped reads starting near position chr16: 3,801,439.

Color codes in the background (as shown above) of the read describe the quality of the alignment of the read and its mate (in case of paired-end sequencing). These colors are assigned both on the basis of the CIGAR sequence (if it contains a softclip) and the flag value of both the read and its mate.

### Gray
Presence of gray background nucleotides in a read suggests that both the read (at least part of it) and its mate are properly aligned and the insert size is within expected range (as shown below).

[![Gray Background Nucleotides](images/gray_bg.png)](images/gray_bg.png "Click to see the full image.")

### Blue
Presence of blue-background nucleotides in a read indicates that part of the read is soft clipped (as shown below). The last 42 nucleotides in the read below are softclipped based on CIGAR sequence (58M42S).

[![Blue Background Nucleotides](images/blue_bg.png)](images/blue_bg.png "Click to see the full image.")

### Brown
A brown colored background (in the main read alignment plot) indicates that the mate of the read is unmapped. Such reads have a flag value that contains the 0x8 bit. On clicking a read with unmapped mate in the read information panel, the current read sequence is displayed along with a button "Show unmapped mate".

[![Brown Background Nucleotides](images/brown_bg.png)](images/brown_bg.png "Click to see the full image.")

On clicking the button "Show unmapped mate", the sequence of the unmapped mate is also displayed.

[![Unmapped Mate](images/unmapped_mate.png)](images/unmapped_mate.png "Click to see the full image.")

### Green
A green background (shown below) indicates that the template has the wrong insert size. As shown in the Read Info figure below, the reads labeled green have a higher insert size than normal (gray) reads because of the structural deletion. In paired-end view, generally such read-pairs have a much longer gray-dashed line than properly aligned (Gray) read-pairs.

[![Green Background Nucleotides](images/green_bg.png)](images/green_bg.png "Click to see the full image.")

### Pink
A pink color background indicates that the orientation of the read and its mate is not correct (See Link to wrong orientation [example](https://proteinpaint.stjude.org/?genome=hg19&block=1&bamfile=test,proteinpaint_demo/hg19/bam/CBFB-MYH11.bam&position=chr16:67115893-67117379)). Several orientations are taken into consideration. The figure below displays an example of an inversion caused due to CBFB-MYH11 gene fusion found in [Acute Myeloid Leukemia (AML) patients](https://pubmed.ncbi.nlm.nih.gov/32015759/). Here the read and its mate are oriented in the reverse direction (R1R2).
* F1F2 - When both read and its mate are pointing in the forward direction (-> ->).
* R1R2 - When both read and its mate are pointing in the reverse direction (<- <-).
* F1R2 - When both read and its mate are pointing in forward and reverse direction but are pointing in opposite directions (<-  ->).

[![Pink Background Nucleotides](images/pink_bg.png)](images/pink_bg.png "Click to see the full image.")

### Orange
Orange background color indicates the read and its mate are mapped in different chromosomes ([as shown below](https://proteinpaint.stjude.org/?genome=hg38&block=1&bamfile=SJACT019118_G1%20WGS,proteinpaint_demo/hg19/bam/discordant_reads.bam&position=chr7:16464-16464&hlregion=chr7:16463-16463)). The displayed read is mapped in chr7:16363-16512 whereas its mate is mapped in chr16.

[![Orange Background Nucleotides](images/orange_bg.png)](images/orange_bg.png "Click to see the full image.")

# Variant Mode

Variant Mode provides an intuitive view of a variant specified by the user inside ppBAM. On specifying the chromosome, position, reference and alternative allele; the reads covering the variant region are displayed and classified into groups supporting the reference allele, alternative allele, none (neither reference nor alternative allele) and ambiguous groups. This mode is invoked when the "variant" field is specified containing the chromosome, position, reference and alternative allele of the variant.

## Alternative, Reference, None and Ambiguous Read Classification Groups

For a given variant (SNV or indel), reads mapping to the variant
region are [classified](https://proteinpaint.stjude.org/?genome=hg19&block=1&position=chr17:7578341-7578441&bamfile=Test,proteinpaint_demo/hg19/bam/TP53_del.wrongbp_amb.bam&variant=chr17.7578383.AGCAGCGCTCATGGTGGGG.A) into Reference, Alternative (possibly multiple
alternative alleles when multi-allele variants are queried), None
(neither reference nor alternative allele) and Ambiguous (unclassified
reads) groups by using the Smith-Waterman alignment (as shown in
figure below).

[![Variant](images/variant.png)](images/variant.png "Click to see the full image.")

Reads are classified into supporting Alternative/Reference allele on the basis of "Allele similarity" which consists of selecting the allele with which the read has the highest identity ratio (number of matched nucleotides/total alignment length) (highest identity ratio highlighted in bold). Reads which do not completely match with neither reference nor alternative alleles are classified into none group (when strictness level = 'Strict') and those that have equal allele similarity to both reference and alternative allele (or multiple alternative alleles in case of multi-allele variants) are classified into the ambiguous group. The barplot on the right displays the "Allele similarity" for each read. This barplot is especially helpful in analyzing reads classified into the none group by indicating the alternative/reference allele with which it has maximum sequence similarity.

[![Read Classification](images/read_class.png)](images/read_class.png "Click to see the full image.")

Above figure describes the methodology for classification of reads
into Reference, Alternative (possibly multiple alternative alleles),
None (neither reference nor alternative allele) and Ambiguous
groups. Classification of seven reads are described for a multi-allele
variant (G/GTCA) and (CG/ATGTCA). (a) Sequence alignment of various
reads to alternative and reference alleles (red colored nucleotides
represent alternative and reference allele nucleotides): Read1 and
Read2 completely support the alternative allele-1 and reference allele
respectively. Read3 has the highest sequence similarity to the
alternative allele-1 but has a mismatch and is therefore classified
into the none group when strictness = 'Strict' and classified into
alternative allele-1 when strictness = 'Lenient'. Read4 has equal
similarity to both reference and alternative allele-1 and is
classified into the ambiguous group. Read5 has the highest allele
similarity and a complete match to alternative allele-2. Read6 has the
highest sequence similarity to the alternative allele-2 but has a
mismatch and is therefore classified into the none group when strictness = 'Strict'
and classified into alternative allele-2 when strictness =
'Lenient'. Read7 has equal allele similarity to alternative allele-1
and alternative allele-2 and is classified into the ambiguous group
(b) A generalized flow chart for classifying reads into (possibly)
multiple alternative/reference alleles using "Allele similarity"
values and the "Strictness" setting. The "None" group contains reads
which do not support neither reference nor alternative allele(s) will
be created only when the "Strictness" setting is set to "Strict"
(default). Ambiguous group consists of reads that have equal allele
similarity to two (or more) alleles.

## Ambiguous Reads

In certain indels such as in the [TP53 example](https://proteinpaint.stjude.org/?genome=hg19&block=1&bamfile=TP53_del,proteinpaint_demo/hg19/bam/TP53_del.bam&position=chr17:7578191-7578591&variant=chr17.7578383.AGCAGCGCTCATGGTGGGG.A&bedjfilterbyname=NM_000546), certain reads in the variant region can have equal similarity towards both the reference and alternative allele. A large number of ambiguous reads are on the left-side of the indel (Figure a below) because the deletion starts with the sequence GCAGCGC which is also found in the right flanking sequence resulting in equal similarity to both reference and alternative alleles for any read ending within this part of the indel region (as shown in figure below). On viewing the read alignment for the ambiguous read (Figure b below) through the read information panel, it is observed that the read has equal similarity to both reference and alternative alleles. Nucleotides highlighted in red indicate those which are part of reference/alternative allele.

[![Ambiguous Reads](images/ambig_reads.png)](images/ambig_reads.png "Click to see the full image.")

## Fisher-strand Analysis to Check for Strand Bias in Variants
Fisher-strand (FS) analysis on ratio of forward/reverse strand reads in the alternative and reference groups can help in detecting possible [strand bias](https://gatk.broadinstitute.org/hc/en-us/articles/5358832960539-FisherStrand) that may be present in the variant of interest. The FS score is the Phred-scaled p-value from the Fisher test of the contingency table consisting of forward/reverse strand reads from both the alternative and reference alleles (as shown in figure below). To increase performance for high-depth sequencing examples, when the sequencing depth is greater than 300 the chi-square test is used (if equal or lower than this number, the Fisher's exact test is used). If FS score is [greater than 60](https://gatk.broadinstitute.org/hc/en-us/articles/360035890471), the FS score is highlighted in red (as shown below) indicating that there may be a possible strand bias in the variant.

[![Fisher-strand Analysis](images/fisher_strand.png)](images/fisher_strand.png "Click to see the full image.")

In the figure above, an example of a [complex indel](https://proteinpaint.stjude.org/?genome=hg19&block=1&bamfile=strand_bias,proteinpaint_demo/hg19/bam/strand_bias.bam&position=chr10:8100668-8100707&variant=chr10.8100686.CAC.CCCTGCCTGTTGTGAGCTGCTCTACGTGCCCTACGTGCT) is shown containing Fisher strand bias. The FS score is highlighted in red indicating this particular variant may contain strand bias.

## Strictness in On-the-fly Genotyping
A user can also optionally change the strictness of the algorithm to Lenient/Strict (default) from the ppBAM configuration panel. For strictness level = 'Lenient', reads are classified based on higher sequence similarity to reference/alternative allele. In case of strictness level = 'Strict', the exact sequence of the reference/alternative allele in the read is compared against the allele sequence given by the user. Reads that do not match either allele are classified into the none group (when strictness level = 'Strict'), the allele similarity plot shows the color of the allele to which the read has maximum sequence similarity.

[![Strictness](images/strictness.png)](images/strictness.png "Click to see the full image.")

The lenient strictness level can be helpful, when the user wants a lenient estimate of the number of reads supporting the particular indel of interest or when the user is confident that only one alternative allele exists. This can also be helpful when there are reads with low base-pair quality calls near the variant region. In contrast when the strictness level is set to 'Strict', a more conservative estimate of the read support is provided for each allele and may indicate the presence of a wrong variant call (if present) or may indicate presence of multiple alternative alleles.

In case of the TP53 deletion example, select reads with wrong base
pair calls are
[shown](https://proteinpaint.stjude.org/?genome=hg19&block=1&bamfile=TP53_del,proteinpaint_demo/hg19/bam/TP53_del.wrongbp.bam&position=chr17:7578371-7578417&variant=chr17.7578383.AGCAGCGCTCATGGTGGGG.A&bedjfilterbyname=NM_000546). For
strictness level = 'Lenient', there are two reads that support the
alternative allele. However, read
NB501822:110:HLWKJBGX5:4:22410:10829:14705 has a wrong base pair call
at position 7578401.

[![Lenient](images/lenient.png)](images/lenient.png "Click to see the full image.")

When the strictness level is changed to 'Strict', this read is
classified into the none group.

[![Strict](images/strict.png)](images/strict.png "Click to see the full image.")

Similarly, reads NB501822:113:HGCGYBGX5:3:23612:16815:9517 (wrong base-pair call at 7578401) and NB501822:113:HGCGYBGX5:2:22101:3565:18789 (wrong base-pair call at 7578391) are classified in the reference allele group when strictness level = 'Lenient' but are classified into the none group when strictness level is set to 'Strict'.  

The 'Lenient' strictness level is generally only helpful in cases where only one alternative allele is present as it assumes only the given reference and alternative allele are the only possible cases. For multi-allelic variants or when a region has a large number of reads with low Phred base-pair quality nucleotides, the 'Strict' (default) level should be used.

## Display of Read Alignment with Respect to Reference and Alternative Allele

In case of reads that are classified into the none group (when strictness level = 'Strict') it can be difficult to understand the classification into that group. For example, in case of insertions with the wrong nucleotide (with respect to the predicted alternative allele) the sequence of the inserted nucleotides is not shown in the main BAM track and can only be viewed through the read information panel. As an example, a [3bp insertion in CEBPA exon](https://proteinpaint.stjude.org/?genome=hg19&block=1&position=chr19:33792266-33792515&bamfile=CEBPA,proteinpaint_demo/hg19/bam/CEBPA.bam&variant=chr19.33792391.C.CTGC) is discussed below. In Figure (a) (shown below) most reads with 3bp insertion have been classified into the alternative allele. However, there are some reads (as highlighted in Figure a) with 3bp insertions that are classified into the none group. The diff score plot suggests that these reads have higher sequence similarity to the alternative allele (and are classified into the alternative group when strictness = 'Lenient') and they seem to support the alternative allele. However, when we click on this read (Fig. b) and click on the "Read Alignment" button (which is available only when the variant field is specified in the URL) the Smith-Waterman alignment of the read with the reference and alternative allele is displayed (Figure b). The indel nucleotides are highlighted in red. In case of the read in the none group (75085182), a mismatch is observed in the indel region between the read and the alternative allele (highlighted by '*' in the alignment row) which explains the classification into the none group. In contrast, the read shown from the alternative allele group (75085106) has a complete match with the alternative allele and is therefore classified into the alternative allele group.

Display of read alignment of the read with respect to both the reference and alternative allele helps provide an intuitive view for describing classification of a read into its respective group.

[![Read Alignment](images/read_align.png)](images/read_align.png "Click to see the full image.")


# Set Operations

Up to three cohorts, gene sets, and mutation sets can be compared and exported based on complex overlapping subsets. The features of this page include:

[![Set Operations](images/set_operations_page.png)](images/set_operations_page.png "Click to see the full image.")

* __Venn Diagram:__ Visually displays the overlapping items included within the three cohorts or sets. Subsets based on overlap can be selected by clicking one or many sections of the Venn diagram. As sections of the Venn Diagram become highlighted in blue, their corresponding row in the overlap table becomes selected.  

* __Summary Table:__ Displays the alias, entity type, and name for each set included in this analysis

* __Overlap Table:__ Displays the number of overlapping items with set operations rather than a visual diagram. Subsets can be selected by checking boxes in the "Select" column, which will highlight the corresponding section of the Venn Diagram. As rows are selected, the "Union of selected sets" row is populated. Each row has an option to create a new set or cohort from the subset, or export the subset as a TSV.

# GDC BAM Slicing Download User Guide

## Introduction to BAM slicing
The GDC BAM slicing download feature is a tool for slicing individual BAM files based on the variant, gene, position, or SNPs for individual case/entities in the NCI GDC. In addition, it also allows users to download unmapped reads from a BAM file. To begin, follow the following steps as outlined.

[![BAM slicing download UI](images/BAM_Slicing/bam_slice_download_UI.png)](images/BAM_Slicing/bam_slice_download_UI.png 'Click to see the full image.')

## Download

### Searching for a case
To search for a case, enter a string id which could be a file name, file id, case UUID or case ID. For example, enter and search for the case 'TCGA-C8-A12N' as shown. Please note that the complete id must be used. Partial ids are not allowed.

[![Searching by a string id](images/BAM_Slicing/search_by_string.png)](images/BAM_Slicing/search_by_string.png 'Click to see the full image.')

A user may also choose to browse the first thousand BAM files. Click on the label "Or, browse first 1000 BAM files out of 141461 total" to load the following view. Scroll and select the case of interest.

[![Browsing the first 1000 BAM files](images/BAM_Slicing/first_1000_bams.png)](images/BAM_Slicing/first_1000_bams.png 'Click to see the full image.')

### Selecting Variant/Gene, Position or Unmapped reads

Upon searching by a string id, the following view is displayed. User must select an Entity ID associated with the case ID as shown. This view shows the 'Experimental strategy', 'Sample Type', and 'Size' of the file associated with that entity id.

[![Selecting Entities](images/BAM_Slicing/selecting_entities.png)](images/BAM_Slicing/selecting_entities.png 'Click to see the full image.')

Choosing a BAM file directly from the thousand files will display the following view. This view has selected an entity id chosen by the user with 'miRNA-Seq' as the experimental strategy, sample type being the 'Primary Tumor' with a file size of 83.03 MB.

[![View upon Selection](images/BAM_Slicing/selecting_entities02.png)](images/BAM_Slicing/selecting_entities02.png 'Click to see the full image.')

### Selecting a Variant

From the view as shown above, user can choose from 48 variants. Select WDR44 as shown and click 'Submit' to download the BAM slice for this case and gene variant.

[![Selecting a variant](images/BAM_Slicing/selecting_variants.png)](images/BAM_Slicing/selecting_variants.png 'Click to see the full image.')

### Selecting a Gene

Click on the next tab to access the view that allows selecting BAM files for a particular gene, snp or a specific position/range in the genome. After making your selection, click the 'Submit' button at the bottom of the view to download the slices.

[![Selecting a gene](images/BAM_Slicing/selecting_genes.png)](images/BAM_Slicing/selecting_genes.png 'Click to see the full image.')

### Selecting unmapped reads

Click the tab for accessing the 'Unmapped reads' as shown below. Click 'Submit' to download the unmapped reads.

[![Selecting unmapped reads](images/BAM_Slicing/selecting_unmapped_reads.png)](images/BAM_Slicing/selecting_unmapped_reads.png 'Click to see the full image.')

## Saved Downloads

All downloads are saved in the 'Downloads' folder. 


# Gene Expression Clustering Tool

## Introduction to Gene Expression Clustering

The Gene Expression Clustering tool is a web-based tool for performing sample clustering by selecting a desired set of genes from the NCI Genomic Data Commons (GDC), and visualizing a heatmap of a z-score transformed matrix.

## Quick Reference Guide

At the Analysis Center, click the 'Gene Expression Clustering' card to launch the heatmap.

[![Analysis Center Gene Expression Clustering Card](images/analysis_center_GEC_tool.png)](images/analysis_center_GEC_tool.png "Click to see the full image.")

Users can view publicly available genes as well as login with credentials to access controlled data.

There are four main panels in the Gene Expression Clustering tool: [controls](#Controls), [heatmap](#Heatmap), [variables](#Variables), and [legend](#Legend).

[![Gene Expression Clustering Tool Features](images/GEC_tool_features.png)](images/GEC_tool_features.png "Click to see the full image.")

### Controls

The control panel can modify the displayed data or the appearance of the matrix. Their functionalities are outlined below.

[![Gene Expression Clustering Tool Controls](images/GEC_tool_controls.png)](images/GEC_tool_controls.png "Click to see the full image.")


* __Clustering:__ Modify the default clustering of the heatmap (Average or Complete), alter the column and row dendrogram dimensions, and change the z-score cap
* __Cases:__ Adjust the visible characters of the case labels
* __Genes:__ Modify how cases are represented for each gene (Absolute, Percent, or None), row group and label lengths, rendering style, and the existing gene set
    * __Edit Group:__ Displays a panel of currently selected genes, which can be modified by clicking on a gene to remove it from the gene set, searching for a particular gene to add, loading top variably expressed genes, or loading a pre-defined gene set provided by the MSigDB database
    * __Create Group:__ Create a new gene set by searching for a particular gene, loading top mutated genes, or loading a pre-defined gene set provided by the MSigDB database
* __Variables:__ Search and select variables to add to the matrix below the heatmap
* __Cell Layout:__ Modify the format of the cells by changing colors, cell dimensions, and label formatting
* __Legend Layout:__ Alter the legend by changing the font size, dimensions, and other formatting preferences
* __Download:__ Download the plot in svg format
* __Zoom:__ Adjust the zoom level by using the up and down arrows on the input box, entering a number, or using the sliding scale to view the case labels
<!--
* __Undo:__ Undo changes made to the matrix (Coming Soon)
* __Redo:__ Redo changes made to the matrix (Coming Soon)
* __Restore:__ Restore the matrix to its default settings (Coming Soon)
-->

### Heatmap

The Gene Expression Clustering heatmap displays the active cohort's cases along the top horizontally, genes along the left column, and the z-score transformed gene expression value.

[![Gene Expression Clustering Tool Heatmap](images/GEC_tool_heatmap.png)](images/GEC_tool_heatmap.png "Click to see the full image.")

Hovering over a cell in the heatmap displays the case submitter_id, gene name, and gene expression value.

[![Gene Expression Clustering Tool Heatmap Cell](images/GEC_tool_cell.png)](images/GEC_tool_cell.png "Click to see the full image.")

Clicking on a cell also gives users the option to launch the [Disco plot](oncomatrix.md#disco-plot), a circos plot displaying copy number data and consequences for that case.

#### Selecting cases on the cluster

Cases on the cluster can be selected by clicking on the dendrogram. Once part of the dendrogram is selected, users can choose to zoom in to the cases, list all highlighted cases, or create a cohort of the selected cases.

[![Gene Expression Clustering Tool Heatmap Cases Dendrogram](images/GEC_tool_heatmap_cases.png)](images/GEC_tool_heatmap_cases.png "Click to see the full image.")

Click on a case in the dendrogram to showcase the Disco plot or the GDC [Case Summary Page](getting_started.md#cohort-case-table).

[![Gene Expression Clustering Tool Heatmap Case Selection](images/GEC_tool_heatmap_case_selection.png)](images/GEC_tool_heatmap_case_selection.png "Click to see the full image.")

In the column of genes on the left, click on a gene to rename it, launch the [ProteinPaint Lollipop plot](proteinpaint_lollipop.md), display the GDC [Gene Summary Page](mutation_frequency.md#gene-and-mutation-summary-pages), or remove the gene. The lollipop plot displays all cases across the GDC affected by SSMs in the selected gene.

[![Gene Expression Clustering Tool Gene Selection](images/GEC_tool_gene.png)](images/GEC_tool_gene.png "Click to see the full image.")

### Variables

Any variables added to the matrix appear below the heatmap. Users can hover over a cell to display the case submitter_id and their value for the given variable.

[![Gene Expression Clustering Tool Variables](images/GEC_tool_variables.png)](images/GEC_tool_variables.png "Click to see the full image.")

Click on a variable to rename it, edit it by excluding categories, replace it with a different variable, or remove it entirely.

[![Gene Expression Clustering Tool Variable Selection](images/GEC_tool_variable_selection.png)](images/GEC_tool_variable_selection.png "Click to see the full image.")

### Legend

In addition to the color coding system for the gene expression values, the legend displays the number of cases from the active cohort in each category for all variables that are selected to appear in the matrix.

[![Gene Expression Clustering Tool Legend](images/GEC_tool_legend.png)](images/GEC_tool_legend.png "Click to see the full image.")

Users can click on a variable in the legend to hide a specific category, only show a specific category, or show all categories for the selected variable.

[![Gene Expression Clustering Tool Legend Selection](images/GEC_tool_legend_selection.png)](images/GEC_tool_legend_selection.png "Click to see the full image.")


## Accessing the Tool

At the analysis center, click the 'Gene Expression Clustering' card to launch the heatmap.

[![Analysis Tools with Gene Expression clustering app card](./images/geneexpclust/1-Analysis-center.png)](./images/geneexpclust/1-Analysis-center.png 'Click to see the full image.')

 View publicly available genes as well as login with credentials to access controlled data.

## Features

The following features are viewable once the default heatmap is loaded. The default heatmap shows all the glioma cases. There are four main panels as outlined in the figure i.e., the 'Controls', 'Heatmap', 'Variables' and the 'Legend'. Each of the features and functionalities are described in detail in the following sections.

[![Default view](./images/geneexpclust/2-default-view.png)](./images/geneexpclust/2-default-view.png 'Click to see the full image.')

## Controls

The control panel as shown has various functionalities with which users can change or modify the appearance of the matrix. The control panel provides flexibility and a wide range of options to maximize user control.

[![Controls](./images/geneexpclust/3-controls.png)](./images/geneexpclust/3-controls.png 'Click to see the full image.')

### Clustering

The clustering control button provides several options to modify the default clustering of the heatmap. Click on the button labeled 'Clustering' to display a menu with options as shown.

[![Clustering button](./images/geneexpclust/4-clustering-control.png)](./images/geneexpclust/4-clustering-control.png 'Click to see the full image.')

#### Clustering method

Click on the 'Complete' option as highlighted to change the method of clustering. The heatmap will render again to show the complete clustering method.

[![Complete clustering method](./images/geneexpclust/5-clustering-method.png)](./images/geneexpclust/5-clustering-method.png 'Click to see the full image.')

The maximum height of the column dendrogram is shown in the next highlighted option as shown.

[![Dendrogram height](./images/geneexpclust/6-col-dendrogram-height.png)](./images/geneexpclust/6-col-dendrogram-height.png 'Click to see the full image.')

Click or edit the number in the input box to adjust the height of the column dendrograms as shown.

[![Adjusting column dendrogram height](./images/geneexpclust/7-adj-dend-height.png)](./images/geneexpclust/7-adj-dend-height.png 'Click to see the full image.')

#### Row Dendrogram Width

Similary, row dendrogram width can also be modified as per user requirement as shown.

[![Adjusting row dendrogram width](./images/geneexpclust/8-row-dend-height.png)](./images/geneexpclust/8-row-dend-height.png 'Click to see the full image.')

[![Adjusting row dendrogram width](./images/geneexpclust/9-adj-row-dend-height.png)](./images/geneexpclust/9-adj-row-dend-height.png 'Click to see the full image.')

#### Z-score Cap

Z scores are used to compare gene expression across samples. A Z-score of zero indicates that the gene's expression level is the same as the mean expression level across all samples, while a positive Z-score indicates that the gene is expressed at a higher level than the mean, and a negative Z-score indicates that the gene is expressed at a lower level than the mean.

User can increase or decrease the Z-score Capping. Increase the Z-score cap from 5 to 10 as shown. Samples with lower gene expression gets lighter to allow highlighting of clusters with higher expression values as shown in red in the heatmap.

[![Z-score capping](./images/geneexpclust/10-zscore-cap.png)](./images/geneexpclust/10-zscore-capt.png 'Click to see the full image.')

### Cases

#### Adjusting the zoom using the zoom buttons

Adjust the zoom level by using arrows on the input box or entering a number to be able to view the sample lables as shown.

[![Adjusting the Zoom](./images/geneexpclust/11-adj-zoom.png)](./images/geneexpclust/11-adj-zoom.png 'Click to see the full image.')

The 'Cases' control has the option 'Case Label Character Limit' to adjust the visible characters of these sample labels. The default is '32'. Change that to '10' to see the new limit applied to the sample labels as shown. Note that reducing the character limit truncates the labels.

[![Adjusting the Zoom](./images/geneexpclust/case-label-char.png)](./images/geneexpclust/case-label-char.png 'Click to see the full image.')

### Genes

User can modify the existing default gene set by clicking the 'Genes' button in the controls as shown. This displays the option to edit genes as well as variables from the dropdown as shown.

[![Geneset edit](./images/geneexpclust/12-geneset-edit.png)](./images/geneexpclust/12-geneset-edit.png 'Click to see the full image.')

#### Modifying Genes

Click the 'Edit Group' button as shown in the 'Gene set' to display a panel of current selected genes.

[![Editing geneset](./images/geneexpclust/13-geneset-editing.png)](./images/geneexpclust/13-geneset-editing.png 'Click to see the full image.')

#### Add/Delete a gene

In the search box, type in any gene name for example 'Wee1' as shown and click submit.

[![Searching genes](./images/geneexpclust/14-search-wee1.png)](./images/geneexpclust/14-search-wee1.png 'Click to see the full image.')

The heatmap loads again after performing a clustering that includes 'WEE1' as shown.

[![Adding genes one by one](./images/geneexpclust/15-wee1-heatmap.png)](./images/geneexpclust/15-wee1-heatmap.png 'Click to see the full image.')

Click on the 'Edit' functionality again within the 'Gene set' menu option. To delete a gene, hover over the gene as shown. A red cross mark will appear as shown.

[![Deleting genes one by one](./images/geneexpclust/16-delete-genes.png)](./images/geneexpclust/16-delete-genes.png 'Click to see the full image.')

Click on the gene 'Wee1' to delete the gene from the gene set. Click submit to redo the clustering.

#### Load top variably expressed genes

User has the option to load the top genes that are variably expressed. To do so, click on the 'Edit Group' button under the 'Genes' controls. Click on the button that reads 'Load top variably expressed genes'. The genes will change to the top most variable genes as shown in this selected cohort.

Click submit to reload the heatmap.

[![Load top variably expressed genes](./images/geneexpclust/17-top-variably-exp-genes.png)](./images/geneexpclust/17-top-variably-exp-genes.png 'Click to see the full image.')

#### Load MSigDB gene set

The gene expression clustering tool also enables users to load a pre-defined gene set provided by the MSigDB database. The current version enabled is the latest. Click on the dropdown button 'Load MSigDB (2023.2.Hs) gene set' and choose one of the following gene sets as shown.

[![MSigDB tree](./images/geneexpclust/msigdb-tree.png)](./images/geneexpclust/msigdb-tree.png 'Click to see the full image.')

For example, select a hallmark gene set for 'Hypoxia' as shown.

[![Hallmark hypoxia gene set](./images/geneexpclust/18-msigdb-tree.png)](./images/geneexpclust/18-msigdb-tree.png 'Click to see the full image.')

Note the info icon next to the gene set that provides additional information about this gene set as well as a link to the database and the original publication PMID as shown.

[![Info icon](./images/geneexpclust/19-geneset-info-i.png)](./images/geneexpclust/19-geneset-info-i.png 'Click to see the full image.')

Upon selecting a MSigDB gene set, the genes get updated as shown.

[![Selected geneset hypoxia](./images/geneexpclust/20-geneset-hypoxia.png)](./images/geneexpclust/20-geneset-hypoxia.png 'Click to see the full image.')

Click 'Submit' to reload the heatmap with the new gene set from MSigDB.

#### Adding gene as a variable

User also has the option to add gene variant terms as variable to line up mutation consequences with clustered gene expression data.

To do so, click the button 'Genes' and click 'Edit Group'.

[![Genes as variables](./images/geneexpclust/21-gene-as-var.png)](./images/geneexpclust/21-gene-as-var.png 'Click to see the full image.')

From the dropdown, select 'Variables' as shown.

[![Choosing variables from dropdown](./images/geneexpclust/22-dropdown-var.png)](./images/geneexpclust/22-dropdown-var.png 'Click to see the full image.')

Search and select 'KRAS'.

[![Searching a gene as variable](./images/geneexpclust/23-kras-var.png)](./images/geneexpclust/23-kras-var.png 'Click to see the full image.')

Click 'Submit' to reload the heatmap with the newly added KRAS gene as a variable. This displays the consequence type for the clustered samples for which KRAS has both the mutation calls and the gene expression data as shown.

[![Heatmap with KRAS gene as a variable](./images/geneexpclust/24-kras-var-row.png)](./images/geneexpclust/24-kras-var-row.png 'Click to see the full image.')

### Variables

The button 'Variables' in the controls allows the user to search and select variables that get added below the heatmap.

Click the button 'Variables' to show the following dictionary tree.

[![Variables dictionary tree](./images/geneexpclust/25-add-var.png)](./images/geneexpclust/25-add-var.png 'Click to see the full image.')

Click the '+' button on the 'Demographic' to display all the terms under the parent term as shown. Select terms 'Ethnicity' and 'Year of birth' and click 'Submit 2 terms'.

[![Selecting and submitting variables](./images/geneexpclust/26-selecting-vars.png)](./images/geneexpclust/26-selecting-vars.png 'Click to see the full image.')

Once the variable terms are submitted, the heatmap will display the added variables as shown.

[![Variable heatmap](./images/geneexpclust/27-var-heatmap.png)](./images/geneexpclust/27-var-heatmap.png 'Click to see the full image.')

### Download

The control panel shows an option to download the plot as an svg after user has specified their customizations. Select the 'Download' button as shown below to save the svg.

[![Download button](./images/geneexpclust/28-download-btn.png)](./images/geneexpclust/28-download-btn.png 'Click to see the full image.')

The download will get saved to the default download folder as shown at the bottom of the browser window.

[![Saved download](./images/geneexpclust/29-downloaded-svg.png)](./images/geneexpclust/29-downloaded-svg.png 'Click to see the full image.')

## Heatmap

### Selecting cases on the cluster

Cases on the cluster can be selected interactively by clicking on the column dendrograms. Click on the dendrograms above the heatmap as shown. The dendrograms get highlighted in red.

[![Selecting case cluster](./images/geneexpclust/30-selecting-sample-cluster.png)](./images/geneexpclust/30-selecting-sample-cluster.png 'Click to see the full image.')

Once the dendrograms are selected, two options are displayed. A user can choose to zoom in the cases or list all the cases highlighted in the dendrograms.

### Clicking a case column

Click on a case label to display the options as shown.

[![Clicking case column](./images/geneexpclust/31-clicking-case-col.png)](./images/geneexpclust/31-clicking-case-col.png 'Click to see the full image.')

User may choose to launch:
- a circos plot by clicking 'Disco plot' button,
- a webpage containing information about the case by clicking the case id
- Gene summary page by clicking on the gene name 'PDGFRA'

### Clicking a gene label

Click on a gene row label to display the following options

[![Clicking gene label](./images/geneexpclust/32-clicking-gene-label.png)](./images/geneexpclust/32-clicking-gene-label.png 'Click to see the full image.')

User can choose to change variable name by deleting and typing in a new name in the box where 'PDGFRA' is currently applied. User may also choose to launch the lollipop plot or gene summary page or remove this row entirely.

### Hovering over/Clicking a cell

Hover over a cell of the heatmap to show information about the case. The information displayed shows the case id, the gene name (CCND1) and the z-score transformed value (4.04..)

[![Hovering over a case](./images/geneexpclust/33-hover-over-case.png)](./images/geneexpclust/33-hover-over-case.png 'Click to see the full image.')

## Variables

### Clicking a Variable

Click on a variable (for example 'Project id' here) row label to display the options as shown.

[![Clicking a variable](./images/geneexpclust/34-clicking-var.png)](./images/geneexpclust/34-clicking-var.png 'Click to see the full image.')

User can change the variable name (input box), edit the variable to exclude categories ('Edit' button), replace the variable by another one ('Replace' button) or remove the row containing the variable entirely by clicking the 'Remove' button.

### Renaming a variable

To rename a variable, edit the default name of the variable in the input box as shown.

[![Rename variable](./images/geneexpclust/35-renaming-var.png)](./images/geneexpclust/35-renaming-var.png 'Click to see the full image.')

After renaming the variable as per user preference, click 'submit'. The row now shows a new variable name.

### Editing a variable

To edit groups within the variable, click the 'Edit' button. Now, user can drag the categories from group 1 into group 2 to create two separate groups and also have the option to exclude a category. After making the choice, click 'Apply' to reload the chart.

[![Editing a variable](./images/geneexpclust/36-editing-var.png)](./images/geneexpclust/36-editing-var.png 'Click to see the full image.')

### Replacing a variable

To replace a variable, click on the row label for that variable and click 'Replace'. This shows the GDC dictionary from which a user can select a variable of choice as shown.

[![Replace variable](./images/geneexpclust/37-replacing-var.png)](./images/geneexpclust/37-replacing-var.png 'Click to see the full image.')

### Removing a variable

To remove a row containing a variable entirely, click on the row label for that variable and click 'Remove'. This removes the entire row from the heatmap.

[![Remove variable](./images/geneexpclust/38-remove-var.png)](./images/geneexpclust/38-remove-var.png 'Click to see the full image.')

## Legend

### Interacting with legend filters

Variables can be filtered upon via the legend. Click a legend item to display the following options. User may choose to 'Hide', 'Show only', or 'Show all' categories from a selected variable. This would allow the user to filter down on the category of choice.

[![Clicking legend icons](./images/geneexpclust/39-clicking-legend-icons.png)](./images/geneexpclust/39-clicking-legend-icons.png 'Click to see the full image.')


# Cohort Level MAF

## Introduction to Cohort Level MAF
The Cohort Level MAF tool is a web-based tool for searching and selecting a desired set of open-access Mutation Annotation Format (MAF) files from the NCI Genomic Data Commons (GDC), and downloading the aggregated and compressed file. 

## Downloads

### Data Query

To retrieve all open-access MAF files with the specified workflow type, select an experimental strategy by clicking either 'WXS' or 'Targeted Sequencing'. Users can then visualize all MAF files with the chosen experimental strategy. Users may choose MAF files by selecting the rows for cases of interest as shown in the table.

[![Querying Data to retrieve MAF files](./images/CohortMAF/Data_query.png)](./images/CohortMAF/Data_query.png 'Click to see the full image.')

Additionally, user can specify the column headers in the downloaded MAF files. To do so, click the highlighted label '0 of 140 columns selected, click to change'. This shows a list with checkboxes to make selections. Select all by clicking the checkbox next to 'Column Name' or make individual selections.

[![Querying Data to retrieve MAF files](./images/CohortMAF/column-selection.png)](./images/CohortMAF/column-selection.png 'Click to see the full image.')

After all selections are made, the files are ready for download.

### Data download

A button in the bottom right corner of the screen displays the total size of all selected file. To download all selected files, click the button as shown. These files are aggregated, sorted, compressed, and downloaded to the browser or the 'Downloads' folder.

[![Downloading Data to retrieve the aggregated, sorted and compressed MAF files](./images/CohortMAF/Data_download.png)](./images/CohortMAF/Data_download.png 'Click to see the full image.')






# GDC Analysis Tool Software Development Kit (SDK)

This guide will detail the process of developing applications for the GDC Data Portal 2.0. It describes the structure of
the GDC Data Portal, how to use the GDC Data Portal API, and how to develop applications for the GDC Data Portal.

The GDC Data Portal is a repository and computational platform for cancer researchers who need to understand cancer, its
clinical progression, and response to therapy. The GDC Data Portal supports the development of applications that allow
for analysis, visualization, and refinement of cohorts.

The GDC Data Portal is built on top of the [GDC API](https://docs.gdc.cancer.gov/API/Users_Guide/Getting_Started/),
which provides access to the GDC data. The GDC Data Portal provides an Analysis Tool Framework (ATF) for developing
applications that can be used to analyze, visualize, and download data from the GDC.

The GDC Data Portal is built with the [React](https://reactjs.org/) framework and 
the [Redux](https://redux.js.org/) library for state management. The GDC Data Portal uses [NextJS](https://nextjs.org/) as its application framework which 
provides server-side rendering of React components. [Mantine.dev](https://mantine.dev/) is the component library, and
styling is through [TailwindCSS](https://tailwindcss.com/). The GDC Data Portal is built on top of the GDC API, which provides access to 
the GDC data.

![This image details the architecture of the GDC Data Portal. 
It shows the interaction between the GDC Data API, the core
module and the user interface.](./images/developers_guide/V2_architecture.png "Architecture of the GDC Data Portal")

*Architecture of the GDC Data Portal*

## Overview of an Application

Applications are React higher-order components (HOC) that are rendered in
the [Analysis Center](https://portal.gdc.cancer.gov/analysis_page?app=). The GDC Data Portal's major functions such as
Projects, Repository, and ProteinPaint are all applications. Each application handles a specific task such as analysis or
visualization and can also be used to refine and build cohorts. Applications are cohort centric and can 
query the GDC API for additional information.

Local and Global filters are available to applications. Local filters are filters that are specific to the application
and are used to refine the data that is displayed in the application.For example in the Mutation Frequency application, the local filters are the gene
and mutation type filters. In the figure below the local filters are highlighted in yellow. These filters are used to
refine the input cohort allowing users to drill down to specific genes and mutation types of interest in the cohort.

![Mutation Frequency](./images/developers_guide/mutation_frequency_parts_highlighted.png)

### Local vs Global Filters

The GDC Data Portal application's input can be the current cohort or multiple user defined cohorts. The application then 
allow users to add filters refining the cohort, create new additional cohorts, or display the data in a visualization. 
Applications typically have:

* **Local filters**  Refine the data displayed in the application
* **Global filters**  Defines the cohort
* **UI Components** Display the data in the application
* **State** Stores the data displayed in the application
* **Actions** Update the state of the application

Applications can also create new cohorts. These cohorts can be used by other GDC Data Portal applications. The figure
below illustrates the application components and cohort filters.

![Local and Cohort Filters](./images/developers_guide/local_and_cohort_filters.png)

## Cohorts and Filters

From an application perspective, a cohort is an Object containing the following information:

```typescript
interface Cohort {
    id: string;        // unique id for cohort
    name: string;      // name of cohort
    filters: FilterSet; // active filters for cohort
    caseSet: CaseSetDataAndStatus; // case ids that are in the cohort
    modified?: boolean; // flag which is set to true is modified and unsaved
    modified_datetime: string; // last time cohort was modified
    saved?: boolean; // flag indicating if cohort has been saved.
    counts: CountsDataAndStatus; //case, file, etc. counts of a cohort
}
```

Likely the most important part of the cohort is the `filters` field. The `filters` field contains the active filters for
the cohort. The `filters` field is a `FilterSet` object. The `FilterSet` object contains the active filters for the
cohort. When calling either the GDC REST API or GDC GraphQL API the `FilterSet` is converted to the appropriate format
for the API. The `FilterSet` object is of the form:

```typescript
interface FilterSet {
    op: "and" | "or"; // operator for combining filters
    root: Record<string, Operation>; // map of filter name to filter operation
}
```

Operation are GDC API filters as described in
the [GDC API](https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#filters-specifying-the-query). These
are:

* Equals
* NotEquals
* LessThan
* LessThanOrEquals
* GreaterThan
* GreaterThanOrEquals
* Exists
* Missing
* Includes
* Excludes
* ExcludeIfAny
* Intersection
* Union

The `root` field is a map of filter names (as defined in the GDC API) to filter operation. The filter operation can be
either a single operation or a `FilterSet` object. The `op` field will eventually support either `and` or `or`, however
at this time only `and` is supported. The `and` operator is used to combine filters using the `and` operator. The `or`
operator is used to combine filters using the `or` operator. The `FilterSet` object is converted to the appropriate
format for the GDC API when the cohort is saved.

When using the GDC REpresentational State Transfer (REST) API, the FilterSet can be converted into the appropriate 
format using the `filterSetToOperation` function. When using the GDC GraphQL API, the FilterSet can be using the
`convertFilterSetToGraphQL` function. The API guide will provide information on what format the filters should be in for the API. Also as the code is in TypeScript,
the IDE will provide information on the format as well.

### Obtaining Cohort Information

The current active cohort can be accessed via the selector `selectCurrentCohort`. This selector returns the current
cohort, which is the cohort that is currently being displayed in the Cohort Management Bar. Accessing the current cohort
is done via the
selector:

```typescript
import {useCoreSelector, selectCurrentCohort} from '@gff/core';

const currentCohort = useSelector(selectCurrentCohort);
```

By using the selector, the component/application will be updated when the cohort changes. There are also selectors for
getting a particular field from the cohort. For example, to get the cohort name, the selector `selectCurrentCohortName`
can be used. The selectors are:

* `selectCurrentCohort`
* `selectCurrentCohortName`
* `selectCurrentCohortId`
* `selectCurrentCohortFilters`
* `selectCurrentCohortModified`
* `selectCurrentCohortModifiedDatetime`
* `selectCurrentCohortSaved`
* `selectCurrentCohortCounts`

The current active filters can be accessed via the selector `selectCurrentCohortFilters`. This selector returns the
current filters,
which are the filters that are currently being displayed in the Cohort Management Bar. Accessing the current filters is
done via the
selector:

```typescript
import {useCoreSelector, selectCurrentFilters} from '@gff/core';

const currentFilters = useSelector(selectCurrentCohortFilters);
```

By using the selector, the application will be updated when the filters change. The filters are returned as
a `FilterSet` object described above.

All the cohorts can be selected using the selector `selectAllCohorts`. This selector returns all the cohorts in the
store. Accessing all the cohorts is done via the selector:

```typescript
import {useCoreSelector, selectAllCohorts} from '@gff/core';

const allCohorts = useSelector(selectAllCohorts);
```

## Using the GDC Data Portal Application API

The GDC Data Portal provides a number of hooks for querying the GDC API. These hooks are located in the `@gff/core` package.
The hooks are designed to work in a manner similar to the RTL Query hooks. The hooks take arguments and return an
object.
The object contains the data and the status of the query. The status of the query is stored in the `isSuccess` variable.
The @gff/core package also provides a set of selectors that return values stored in the core redux store: `CoreStore`.

There are a number of hooks and selectors that are available for querying the GDC API, a subset of which are shown
below:

![hooks and selectors](./images/developers_guide/hooks_and_selectors.png)

### Case Information

The GDC Data Portal provides several hooks for querying case information. These hooks are located in the `@gff/core`
package. Cases can be queried using several different methods. The `useAllCases` hook returns all the cases in the GDC
and can be filtered by the current cohort as shown below

```typescript
import {useCoreSelector, useAllCases} from '@gff/core';

...

const [pageSize, setPageSize] = useState(10);
const [offset, setOffset] = useState(0);
const [searchTerm, setSearchTerm] = useState<string>("");
const [sortBy, setSortBy] = useState<SortBy[]>([]);
const cohortFilters = useCoreSelector((state) =>
    selectCurrentCohortFilters(state),
);


const {data, isFetching, isSuccess, isError, pagination} = useAllCases({
    fields: [
        "case_id",
        "submitter_id",
        "primary_site",
        "disease_type",
        "project.project_id",
        "project.program.name",
        "demographic.gender",
        "demographic.race",
        "demographic.ethnicity",
        "demographic.days_to_death",
        "demographic.vital_status",
        "diagnoses.primary_diagnosis",
        "diagnoses.age_at_diagnosis",
        "summary.file_count",
        "summary.data_categories.data_category",
        "summary.data_categories.file_count",
        "summary.experimental_strategies.experimental_strategy",
        "summary.experimental_strategies.file_count",
        "files.file_id",
        "files.access",
        "files.acl",
        "files.file_name",
        "files.file_size",
        "files.state",
        "files.data_type",
    ],
    size: pageSize,
    filters: cohortFilters,
    from: offset * pageSize,
    sortBy: sortBy,
    searchTerm,
});
```

The `useAllCases` hook takes a number of arguments:

* `fields` - The fields to return from the GDC API
* `size` - The number of cases to return
* `filters` - The filters to apply to the cases
* `from` - The starting index of the cases to return
* `sortBy` - The fields to sort the cases by
* `searchTerm` - The search term to use to search the cases

This call is used in the Table view tab of the Cohort Management Bar.

Information for a single case can be queried using the `useCaseSummary` hook. This call is used in the caseView page:
[portal.gdc.cancer.gov/cases/5693302a-4548-4c0b-8725-0cb7c67bc4f8](https://portal.gdc.cancer.gov/cases/5693302a-4548-4c0b-8725-0cb7c67bc4f8)

```typescript
  const {data, isFetching} = useCaseSummary({
    filters: {
        content: {
            field: "case_id",
            value: case_id,
        },
        op: "=",
    },
    fields: [
        "files.access",
        "files.acl",
        "files.data_type",
        "files.file_name",
        "files.file_size",
        "files.file_id",
        "files.data_format",
        "files.state",
        "files.created_datetime",
        "files.updated_datetime",
        "files.submitter_id",
        "files.data_category",
        "files.type",
        "files.md5sum",
        "case_id",
        "submitter_id",
        "project.name",
        "disease_type",
        "project.project_id",
        "primary_site",
        "project.program.name",
        "summary.file_count",
        "summary.data_categories.file_count",
        "summary.data_categories.data_category",
        "summary.experimental_strategies.experimental_strategy",
        "summary.experimental_strategies.file_count",
        "demographic.ethnicity",
        "demographic.demographic_id",
        "demographic.gender",
        "demographic.race",
        "demographic.submitter_id",
        "demographic.days_to_birth",
        "demographic.days_to_death",
        "demographic.vital_status",
        "diagnoses.submitter_id",
        "diagnoses.diagnosis_id",
        "diagnoses.classification_of_tumor",
        "diagnoses.age_at_diagnosis",
        "diagnoses.days_to_last_follow_up",
        "diagnoses.days_to_last_known_disease_status",
        "diagnoses.days_to_recurrence",
        "diagnoses.last_known_disease_status"]
});
```

The `useCaseSummary` hook takes a number of arguments:

* `fields` - The fields to return from the GDC API
* `filters` - The filters to apply to the cases and where the caseId is passed in

### File Information

Similar to the case information, the GDC Data Portal provides a number of hooks for querying file information. These hooks
are located in the `@gff/core` package.
To get a list of files associated with a cohort, the `useGetFilesQuery` hook can be used. This call is used in the
Repository application which is used to display the files associated with a cohort.
The `useGetFilesQuery` hook takes a number of arguments:

```typescript
import {
    useCoreDispatch,
    useCoreSelector,
    selectCurrentCohortFilters,
    buildCohortGqlOperator,
    joinFilters,
    useFilesSize,
} from "@gff/core";

...

const coreDispatch = useCoreDispatch();
const [sortBy, setSortBy] = useState<SortBy[]>([]); // states to handle table sorting and pagination
const [pageSize, setPageSize] = useState(20);
const [offset, setOffset] = useState(0);

const repositoryFilters = useAppSelector((state) => selectFilters(state)); // as this is a app get the repository filters from the app state (local filters)
const cohortFilters = useCoreSelector((state) =>    // get the cohort filters from the core state (global filters)
    selectCurrentCohortFilters(state),
);

const {data, isFetching, isError, isSuccess} = useGetFilesQuery({
    case_filters: buildCohortGqlOperator(cohortFilters),
    filters: buildCohortGqlOperator(repositoryFilters),
    expand: [
        "annotations", //annotations
        "cases.project", //project_id
        "cases",
    ],
    size: pageSize,
    from: offset * pageSize,
    sortBy: sortBy,
});
```

The `useGetFilesQuery` hook takes a number of arguments:

* `case_filters` - The filters to apply to the cases
* `filters` - The filters to apply to the files
* `expand` - The fields to expand
* `size` - The number of files to return
* `from` - The starting index of the files to return
* `sortBy` - The fields to sort the files by

Note this hook was designed to take global filters (e.g. the current cohort as `case_filters`) and local filters (the
Repository filters).

Information for a single file can be queried using the `useFileSummary` hook. This call is used in
the `File Summary View`
page [portal.gdc.cancer.gov/files/0b5a9e7e-8e2e-4b7a-9b7e-ff5d9c5b2b2b](https://portal.gdc.cancer.gov/files/0b5a9e7e-8e2e-4b7a-9b7e-ff5d9c5b2b2b)

```typescript
 const {data: {files} = {}, isFetching} = useGetFilesQuery({
    filters: {
        op: "=",
        content: {
            field: "file_id",
            value: setCurrentFile,
        },
    },
    expand: [
        "cases",
        "cases.annotations",
        "cases.project",
        "cases.samples",
        "cases.samples.portions",
        "cases.samples.portions.analytes",
        "cases.samples.portions.slides",
        "cases.samples.portions.analytes.aliquots",
        "associated_entities",
        "analysis",
        "analysis.input_files",
        "analysis.metadata.read_groups",
        "downstream_analyses",
        "downstream_analyses.output_files",
        "index_files",
    ],
});
```

The `useFileSummary` hook takes several arguments:

* `filters` - The filters to apply to the cases and where the file uuid is passed in
* `expand` - The fields to expand

### Sets: Gene, SSMS, and Case

Sets are supported by the GDC API and are used to create an entity that represents a set of items as a `set_id`. Sets
are either gene sets, SSM sets, or case sets. All of the GDC APIs support passing sets as a filter parameter.
The GDC Data Portal provides a number of hooks for creating and querying set information.

A set can be created using one of the following hooks:

* `useCreateGeneSetFromValuesMutation`
* `useCreateSsmsSetFromValuesMutation`
* `useCreateCaseSetFromValuesMutation`
* `useCreateGeneSetFromFiltersMutation`
* `useCreateSsmsSetFromFiltersMutation`
* `useCreateCaseSetFromFiltersMutation`

These functions will create a set from either a list of values or a filter set. The `useCreate...SetFromValuesMutation`
hooks take a single parameter `values` which is an array of values, while the `useCreate...SetFromFiltersMutation`
hooks take one required parameter `filters`
that is either a filter set or JSON object. Both calls return the created `set_id` if the set was successfully created.

As the above hooks are Redux Toolkit Query hooks, namely mutation hooks, they return a tuple of the form:
`[mutationHook, response]` which is a function to call the mutation and the response from the mutation. The mutation
hook can be used like:

```typescript
 const [createSet, response] = createSetHook();

const handleCreateSet = async () => {
    const {data} = await createSet({
        variables: {
            values: ["TP53", "KRAS", "EGFR"],
        },
    });
    if (response.isSuccess) {
        dispatch(
            addSet({
                setType,
                setName: form.values.name.trim(),
                setId: response.data as string,
            }),
        );
    }
    ;
}
```

Once a set is created it can be altered using the following hooks:

* `useAppendToGeneSetMutation`
* `useAppendToSsmSetMutation`
* `useRemoveFromGeneSetMutation`
* `useRemoveFromSsmSetMutation`

Sets can be managed using the following actions:

* `addSet`
* `removeSet`
* `updateSet`

The following selectors are available for getting set information:

* `selectAllSets`
* `selectSetById`
* `selectSetByName`
* `selectSetByType`

Finally, the following hooks are available for querying set size:

* `useGeneSetCountsQuery`
* `useSsmSetCountsQuery`
* `useCaseSetCountsQuery`

## Creating a Cohort

Depending on the application function, it may be beneficial to create a new cohort. Although the GDC Data Portal SDK provides a
number of functions for creating a new cohort, it is highly recommended that the application use the provided `Button` and
`SaveCohortModal` components to create a new cohort. The `Button` and `SaveCohortModal` components are located in
the `@gff/portal-proto` package.

To create a cohort using the SaveCohortModal component the following code can be used:
In summary, the above code flow is:

1. The `ProjectsCohortButton` component renders a button with the label "Save New Cohort"
2. When the button is clicked, it sets the state variable `showSaveCohort` to true, which triggers the rendering of
   the `SaveCohortModal` component.
3. The `SaveCohortModal` component passed:
   * An onClose function that sets the showSaveCohort state variable to false.
   * A `filters` prop, which is an object defining the filters for the cohort based on the selected projects.
4. The `SaveCohortModal` will use the passed filter to create, name, and save the cohort when the save button is clicked.

Additional details on the `SaveCohortModal` component can be found in the [Component Library](#component-library)
section.

## Altering a Cohort

Altering a cohort is done by dispatching actions to add, remove, or clear filters. The following actions are available
for altering the current cohort:

* `updateCohortFilter`
* `removeCohortFilter`
* `clearCohortFilters`

Note that all of these operations are applied to the current cohort. The current cohort is the cohort that is currently
being displayed in the Cohort Management Bar. The current cohort can be programmatically accessed via the `selectCurrentCohort` selector.
The current cohort's filters can be accessed via the `selectCurrentCohortFilters` selector.

### Updating, Removing, and Clearing filters

To update the current selected cohort's filter, the `updateCohortFilter` action can be used. The `updateCohortFilter`
action takes two arguments:

```typescript
interface UpdateFilterParams {
    field: string;
    operation: Operation;
}
```

where `field` is the field to update and `operation` is the operation to apply to the field. For example to update the
`cases.project.project_id` field to include the project `TCGA-ACC` the following code can be used:

```typescript
import {useCoreDispatch, updateCohortFilter} from '@gff/core';

const coreDispatch = useCoreDispatch();

coreDispatch(updateCohortFilter({
    field: "cases.project.project_id",
    operation: {
        op: "in",
        content: {
            field: "cases.project.project_id",
            value: ["TCGA-ACC"],
        },
    },
}));
```

This will update the current cohort's filter to include the project `TCGA-ACC`. The `removeCohortFilter` action can be
used to remove a filter from the current cohort. The `removeCohortFilter` action takes a single argument:

```typescript
interface RemoveFilterParams {
    field: string;
}
```

where `field` is the field to remove. For example, to remove the `cases.project.project_id` field from the current
cohort's filter, the following code can be used:

```typescript
import {useCoreDispatch, removeCohortFilter} from '@gff/core';

const coreDispatch = useCoreDispatch();

coreDispatch(removeCohortFilter({
    field: "cases.project.project_id",
}));
```

This will remove the `cases.project.project_id` field from the current cohort's filter. The `clearCohortFilters` action
can be used to clear all the filters from the current cohort. The `clearCohortFilters` action takes no arguments. For
example, to clear all the filters from the current cohort, the following code can be used:

```typescript
import {useCoreDispatch, clearCohortFilters} from '@gff/core';

const coreDispatch = useCoreDispatch();

coreDispatch(clearCohortFilters());
```

This will clear all the filters from the current cohort.

## Updating the Cohort Name

The cohort name can be updated using the `updateCohortName` action. The `updateCohortName` action takes a single
argument:

```typescript
interface UpdateCohortNameParams {
    name: string;
}
```

where `name` is the new name for the cohort. For example, to update the current cohort's name to `My Cohort`, the
following code can be used:

```typescript
import {useCoreDispatch, updateCohortName} from '@gff/core';

const coreDispatch = useCoreDispatch();

coreDispatch(updateCohortName({
    name: "My Cohort",
}));
```

This will update the current cohort's name to `My Cohort`.

## Setting the Current Cohort

The current cohort can be set using the `setCurrentCohort` action. The `setCurrentCohort` action takes a single
argument:

```typescript
interface SetCurrentCohortParams {
    cohortId: string;
}
```

where `cohortId` is the id of the cohort to set as the current cohort. For example, to set the cohort with id `1234` as
the current cohort, the following code can be used:

```typescript
import {useCoreDispatch, setCurrentCohort} from '@gff/core';

const coreDispatch = useCoreDispatch();

coreDispatch(setCurrentCohort({
    cohortId: "1234",
}));
```

This will set the cohort with ID `1234` as the current cohort.

## Total Count Information

Count information can be queried using the `useTotalCounts` hook. This hook takes a number of arguments:

```typescript
import {useTotalCounts} from "@gff/core";

const {data, isFetching, isSuccess, isError} = useTotalCounts();
```

This will return the total counts for the GDC. The data in the response is of the form:

```typescript
interface TotalCounts {
    counts: {
        caseCounts: number;
        fileCounts: number;
        genesCounts: number;
        mutationCounts: number;
        repositoryCaseCounts: number;
        projectsCounts: number;
        primarySiteCounts: number;
    },
    status: DataStatus;
}
```

where `DataStatus` is defined as:

```typescript
export type DataStatus = "uninitialized" | "pending" | "fulfilled" | "rejected";
```

## Application Card Counts
The application cards show the counts for the data required by them. The data types below are supported:

* caseCount 
* fileCount
* genesCount
* mutationCount
* ssmCaseCount
* sequenceReadCaseCount
* geneExpressionCaseCount
* mafFileCount

Each of these use a specific GraphQL query to the GDC Data API to get the count. If an application requires a 
specialized count, then the developer will need to implement and register a count function that returns the following:
```typescript
[
    {
        data: number,                // The count for the specific data type
        isFetching: boolean,         // True if the query is fetching data 
        isSuccess: boolean,          // True if query sucessfully completes
        isError: boolean             // True if the query has encountered an error
    }      
]
```
or use [RTK Query's ```useLazyQuery```](https://redux-toolkit.js.org/rtk-query/api/created-api/hooks#uselazyquery). 
For example:

```typescript
import { graphqlAPISlice } from "../../gdcapi/gdcgraphql";
import { buildCohortGqlOperator, FilterSet, joinFilters } from "../../cohort";

const graphQLQuery = `
  query ssmsCaseCountQuery($ssmCaseFilter: FiltersArgument) {
  viewer {
    repository {
      ssmsCases : cases {
        hits(case_filters: $ssmCaseFilter, first: 0) {
          total
        }
      }
    }
  }
}`;

/**
 * Injects endpoints for case counts for ssmsCounts
 */
const ssmsCaseCountSlice = graphqlAPISlice.injectEndpoints({
  endpoints: (builder) => ({
    ssmsCaseCount: builder.query<number, FilterSet>({
      query: (cohortFilters) => {
        const graphQLFilters = {
          ssmCaseFilter: buildCohortGqlOperator(
            joinFilters(cohortFilters ?? { mode: "and", root: {} }, {
              mode: "and",
              root: {
                "cases.available_variation_data": {
                  operator: "includes",
                  field: "cases.available_variation_data",
                  operands: ["ssm"],
                },
              },
            }),
          ),
        };
        return {
          graphQLFilters,
          graphQLQuery,
        };
      },
      transformResponse: (response) =>
        response?.data?.viewer?.repository?.ssmsCases?.hits?.total ?? 0,
    }),
  }),
});

export const { useLazySsmsCaseCountQuery } = ssmsCaseCountSlice;
```

This function then needs to be registered by adding the call:

```typescript
import { CountHookRegistry}  from "@gff/core";

...

CountHookRegistry.getInstance().registerHook("ssmCaseCount", useLazySsmsCaseCountQuery);
```

The count function is now registered with it name passed as the first argument to ```registerHook``` , and is used to set the value of the ```countsField``` in the 
application registration described below. An appropriate place to add the registration call is in ```_app.tsx```.


## Component Library

The GDC Data Portal provides a number of components that make it easy to develop applications
for it. These components are located in the `@gff/portal-proto`
package.
In several components, the GDC Data Portal uses the [Mantine](https://mantine.dev/) component library but base components and
encapsulates calls to the GDC API so that developers do not have to.

### Buttons

The GDC Data Portal provides a number of buttons that can be used for various purposes. These buttons are located in
the `@gff/portal-proto` package.
The buttons are:

* `DownloadButton` - A button that can be used to download data from the GDC API.
* `SaveCohortButton` - A button that can be used to save a cohort.

The `DownloadButton` component is used in the Repository application to download data from the GDC API.
The `DownloadButton` component takes a number of arguments:

```tsx
<DownloadButton
    inactiveText={`Download ${numFilesCanAccess} Authorized File${
        numFilesCanAccess !== 1 ? "s" : ""
    }`}
    activeText=""
    disabled={
        numFilesCanAccess === 0 ||
        (user.username && dbGapList.length > 0 && !checked)
    }
    endpoint="data"
    extraParams={{
        ids: (filesByCanAccess?.true || []).map((file) => file.file_id),
        annotations: true,
        related_files: true,
    }}
    method="POST"
    setActive={setActive}
/>
```

The parameters for the DownloadButton are defined in the GDC Data Portal 2.0 SDK API documentation. The `DownloadButton`
component will take care of
calling the GDC API and downloading the data. The `DownloadButton` component will also provide status that can be used
with
a progress bar or spinner to display the progress of the download.

The `SaveCohortButton` component is used in the Repository application to save a cohort.

![img.png](images/developers_guide/components/create_cohort_button.png)

The `SaveCohortButton` component takes a number of arguments:

```tsx
<CohortCreationButton
    numCases={cohort1Count}
    label={cohort1Count.toLocaleString()}
    filtersCallback={async () =>
        generateFilters(caseSetIds[0], caseSetIds[1])
    }
/>
```

The CohortCreationButton component will show the number of selected cases and will create a new saved cohort
when the button is clicked. The `filtersCallback` is a function that returns the filters for the cohort.

### Modals

Modals are used to show transitory information or obtain information from the user. The GDC Data Portal provides many
modals that can be used for various purposes. One such modal is the `SaveCohortModal` component mentioned previously.

![img.png](images/developers_guide/components/save_cohort_modal.png)

* `SaveCohortModal` - A modal that can be used to save a cohort.
* Various modals for displaying information on Sets:
  * `CaseSetModal`
  * `GeneSetModal`
  * `MutationSetModal`
* `SaveOrCreateEntityModal` - A modal that can be used to save or create a new entity.

These modals and others, are documented in the Portal 2.0 SDK API documentation.

### Charts

Basic charts are provided for use within an application, although developers are free to use any desired charting 
library compatible with React 18.
The charts provided are:

* `BarChart` - A bar chart

![Bar Chart](images/developers_guide/primary_site.png)

The `BarChart` component (based on Plotly) is passed data in the form:

```typescript
import {PlotData} from "plotly.js";


export interface BarChartData {
    datasets: Partial<PlotData>[];
    yAxisTitle?: string;
    tickvals?: number[];
    ticktext?: string[];
    label_text?: string[] | number[];
    title?: string;
    filename?: string;
}
```

Where `datasets` is an array of PlotData objects, which at the minimum contain the `x` and `y` fields.

The `yAxisTitle` is the title for the y-axis, the `tickvals` and `ticktext` are the tick values and text for
the x-axis, the `label_text` is the text for the labels, the `title` is the title for the chart,
and the `filename` is the filename to use when downloading the chart.

Note that BarChart needs to be imported as a dynamic component:

```tsx
import dynamic from "next/dynamic";

const BarChart = dynamic(() => import("@/components/charts/BarChart"), {
    ssr: false,
});
```

* `Cancer Distribution` - A cancer distribution chart
  
  ![cancer distribution](images/developers_guide/most-frequently-mutated-genes-bar-chart.png)

The `CancerDistribution` component (based on Plotly) is different as it passed the Gene Symbol
and optionally cohort and gene filters.

```typescript
interface CNVPlotProps {
    readonly gene: string;
    readonly height?: number;
    readonly genomicFilters?: FilterSet;
    readonly cohortFilters?: FilterSet;
}
```

These charts and others are documented in the GDC Data Portal 2.0 SDK API documentation.

### Facets

Facet components are provided for use in building local filters for an application. There are several types of facet
components:

* `EnumFacet` - A facet that is used to filter on an enum field
* `DateFacet` - A facet that is used to filter a date field
* `NumericRangeFacet` - A facet that is used to filter on a range field
* `PercentileFacet` - A facet that is used to filter on a percentile field
* `AgeRangeFacet` - A facet that is used to filter on an age range field
* `TextFacet` - A facet that is used to filter a text field
* `BooleanFacet` - A facet that is used to filter on a boolean field

![Enum Facet Component](images/developers_guide/components/enum_facet.png)

*Enum Facet*

![Range Facet Component](images/developers_guide/components/numeric_range_facet.png)

*Range Facet*

![Date Range Facet Component](images/developers_guide/components/date_range_facet.png)

*Date Range Facet*

![Number Range Facet Component](images/developers_guide/components/number_range.png)

*Number Range Facet*

![Percent Range Facet Component](images/developers_guide/components/percentile_facet.png)

*Percentile Facet*

![Age Range Facet Component](images/developers_guide/components/age_range_facet.png)

*Age Range Facet*

![Exact Value Facet Component](images/developers_guide/components/exact_value_facet.png)

*Exact Value Facet*

![Boolean Toggle Facet Component](images/developers_guide/components/toggle_facet.png)

*Toggle Facet*

The facet components are documented in the GDC Data Portal 2.0 SDK API documentation. As these components are passed data fetcher
and filter management hooks, they can be used for both cohort and local filters in an application.

### VerticalTable

The VerticalTable component is used to display data in a table format. The VerticalTable component is a Mantine
component using React table version 8. The VerticalTable component has a number of parameters, the most important
being data, columns, and filters. The data is the data to display in the table, the columns are the columns to display
in the table, and is where the fields are rendered.
The table has support for searching, sorting, and pagination. It can be configured to render many different types of
columns, including text, numeric, and date. The table can also be configured
to use React components for rendering columns.
The Vertical Table is used for most of the table views in the GDC Data Portal. There are a number of examples of its use and
is documented in the GDC Data Portal 2.0 SDK API documentation.

![vertical_table.png](images/developers_guide/components/vertical_table.png)
*Vertical Table*

## Style Guide

The [GDC Data Portal Style Guide](PDF/GDC_Data_Portal_Style_Guide.pdf) provides a comprehensive reference for creating consistent designs adhering to the Data Portal's visual and accessibility standards. Within the Style Guide, you will find information about our predefined styles, available library of pre-built components, as well as guidelines on best practices for accessibility, responsive design, and usability.

## Application Development

### Getting Started

The GDC Data Portal 2.0 is a monorepo that contains all the code for the GDC Data Portal. The monorepo is managed
using [lerna](https://lerna.js.org) and [npm](https://www.npmjs.com/), and contains the following packages:

* `@gff/core` - Contains the core components and hooks for the GDC Data Portal.
* `@gff/portal-proto` - Contains the UI components and application framework (using NextJS) for the GDC Data Portal.

Note that in the future, the UI components located in the `@gff/portal-proto` package will be refactored into a 
separate package , and `@gff/portal-proto` will be renamed to `@gff/portal`.

Developers can get started by cloning the repo and following the instructions in
the [README.md](https://github.com/NCI-GDC/gdc-frontend-framework/blob/develop/README.md) file.

### Application Layout

A typical application will have the following layout. The main section of the application is the area where components
like tables, graphs, and other components are displayed. Local filters are displayed on the left side and
depending on the number of facets, will scroll vertically. This is a typical layout but other layouts are possible, like
in the case of ProteinPaint. Applications are encouraged to use vertical space as much as possible, as horizontal
scrolling can be a poor user experience.

This section will describe parts of the Project application and how it is structured. The Project application is a
simple application that displays a table of projects and allows the user to filter the projects by a number of filters.
As the local filters are selected the table display is updated, but the cohort is not changed (i.e. cohort filters are
not
updated). The Project application's source code is in the `@gff/portal-proto` package in the `src/features/projectsCenter`
directory.
The user can create a new saved cohort by selecting projects and clicking the "Save New Cohort" button. This will open
a modal that will allow the user to name the cohort and save it. The Project application is a good example of how to use
the GDC Data Portal 2.0 SDK to create an application.

![projects_app_parts_outlined.png](images/developers_guide/projects_app_parts_outlined.png)
*Major Sections of an Application*

## Local State

Depending on the application, it may be necessary to maintain the local state. For example, in the Projects application,
the selected local filters, in this case, represented as Enumeration Facets, are stored in the local state. This allows
the application to remember the selected filters when the user navigates away from the page and then returns. Persisting
the state uses [Redux Toolkit] and [Redux Persist] to store the state in local storage. While the CoreState is managed
by the portal core, the local state is managed by the application. Using a separate store for the local state allows
the application to manage the state without having to worry about affecting the core state.

The GDC Data Portal core package provides a number of functions to assist in the creation and persisting of the redux store and will
create  handlers such as 'AppState', 'AppDispatch', and 'AppSelector'. The 'AppState' is the type of the local state, the AppDispatch
is the type of the dispatch function, and the 'AppSelector' is the type of the selector function.

All of these can be automatically created using the `createAppStore` function:

```typescript
import {createAppStore} from "@gff/core";
import {projectCenterFiltersReducer} from "./projectCenterFiltersSlice";

const PROJECT_APP_NAME = "ProjectCenter";

// create the store, context and selector for the Project  Center
// Note the project app has a local store and context which isolates
// the filters and other store/cache values

const reducers = combineReducers({
    projectApp: projectCenterFiltersReducer, // An application may have more that one reducer
});

export const {id, AppStore, AppContext, useAppSelector, useAppDispatch} =
    createAppStore({
        reducers: reducers,
        name: PROJECT_APP_NAME,
        version: "0.0.1",
    });

export type AppState = ReturnType<typeof reducers>;
```

Calling this function will create the local store given the reducers, its name, and version number of the application.
The name of the application is used to create the local storage key for the application. The `id` is the id of the
application and is used to create the local storage key. The `AppStore` is the local store, the `AppContext` is the
local context, the `useAppSelector` is the selector hook, and the `useAppDispatch` is the dispatch hook.

Since there is now a local store, developers can create a slice associated with the local state. This is a standard Redux Toolkit
slice and will contain the reducer, actions, and selectors for the local state.

### Persisting the Local State

If it is desirable to persist the local state, this is done with the `persistReducer` function from the
[redux-persist](https://github.com/rt2zz/redux-persist) package. Any reducer can be persisted by creating a
persisted store and passing the reducer to the `persistReducer` function. For example, the `createAppStore` function
can be modified to persist the local filter state as:

```typescript
import {combineReducers} from "redux";
import {persistReducer} from "redux-persist";
import storage from "redux-persist/lib/storage";
import {createAppStore} from "@gff/core";
import {projectCenterFiltersReducer} from "./projectCenterFiltersSlice";

const PROJECT_APP_NAME = "ProjectCenter";

const persistConfig = {
    key: PROJECT_APP_NAME,
    version: 1,
    storage,
    whitelist: ["projectApp"],
};

// create the store, context and selector for the Project Center
// Note the project app has a local store and context which isolates
// the filters and other store/cache values

const reducers = combineReducers({
    projectApp: projectCenterFiltersReducer,
});

export const {id, AppStore, AppContext, useAppSelector, useAppDispatch} =
    createAppStore({
        reducers: persistReducer(persistConfig, reducers),
        name: PROJECT_APP_NAME,
        version: "0.0.1",
    });

export type AppState = ReturnType<typeof reducers>;
```

For example, the `projectCenterFiltersSlice.ts` which handles the local filters, is defined as:

```typescript
import {createSlice, PayloadAction} from "@reduxjs/toolkit";
import {Operation, FilterSet} from "@gff/core";
import {AppState} from "./appApi";

export interface ProjectCenterFiltersState {
    readonly filters: FilterSet;
}

const initialState: ProjectCenterFiltersState = {
    filters: {mode: "and", root: {}},
};

const slice = createSlice({
    name: "projectCenter/filters",
    initialState,
    reducers: {
        updateProjectFilter: (
            state,
            action: PayloadAction<{ field: string; operation: Operation }>,
        ) => {
            return {
                ...state,
                filters: {
                    mode: "and",
                    root: {
                        ...state.filters.root,
                        [action.payload.field]: action.payload.operation,
                    },
                },
            };
        },
        removeProjectFilter: (state, action: PayloadAction<string>) => {
            // eslint-disable-next-line @typescript-eslint/no-unused-vars
            const {[action.payload]: _, ...updated} = state.filters.root;
            return {
                ...state,
                filters: {
                    mode: "and",
                    root: updated,
                },
            };
        },
        clearProjectFilters: () => {
            return {filters: {mode: "and", root: {}}};
        },
    },
    extraReducers: {},
});

export const projectCenterFiltersReducer = slice.reducer;
export const {updateProjectFilter, removeProjectFilter, clearProjectFilters} =
    slice.actions;

export const selectFilters = (state: AppState): FilterSet | undefined =>
    state.projectApp.filters;

export const selectProjectFiltersByName = (
    state: AppState,
    name: string,
): Operation | undefined => {
    return state.projectApp.filters.root[name];
};
```

The above code creates a slice for the local state. The slice contains the reducer and the actions for the local state.

The reducer is `projectCenterFiltersReducer` and the actions are `updateProjectFilter`, `removeProjectFilter`,
and `clearProjectFilters`.
The selectors are `selectFilters` and `selectProjectFiltersByName`. The `selectFilters` selector returns the filters for
the
application, while the `selectProjectFiltersByName` selector returns the filter for a given name.

## Application Hooks

The above can be used to define hooks for use in the local filter EnumFacet component. For example,
the `useProjectFiltersByName` hook is
implemented as:

```typescript
export const useUpdateProjectsFacetFilter = (): UpdateFacetFilterFunction => {
    const dispatch = useAppDispatch();
    // update the filter for this facet

    return (field: string, operation: Operation) => {
        dispatch(updateProjectFilter({field: field, operation: operation}));
    };
};
```

Clearing the local filters is done using the `clearProjectFilters` action:

```typescript
export const useClearProjectsFacetFilters = (): ClearFacetFiltersFunction => {
    const dispatch = useAppDispatch();
    // clear the filters for this facet

    return () => {
        dispatch(clearProjectFilters());
    };
};
```

Note that the dispatch function is the `useAppDispatch` hook returned by the `createAppStore` function. The
user-selected local filters can be retrieved using the `useProjectsFilters` hook created by combining the
`useAppSelector` hook and the `selectFilters selector`:

```typescript
export const useProjectsFilters = (): FilterSet => {
    return useAppSelector((state) => selectFilters(state));
};
```

## Creating a New Cohort

The Project application allows users to create a new cohort from the selected projects. The cohort is created using the
`SaveCohortModal` component. The `SaveCohortModal` component passes the current cohort filters and the local project
filters to create a new saved cohort. In the case of the Project application, the `SaveCohortModal` component is used
in a button component. The button component is passed the selected projects and the `SaveCohortModal` component is
rendered when the button is clicked. The `SaveCohortModal` component passes the current cohort filters and the local
project filters to create a new saved cohort. The `SaveCohortModal` component is used in the Project application as:

```tsx
import React, {useState} from "react";
import {Button, Tooltip} from "@mantine/core";
import {CountsIcon} from "@/components/tailwindComponents";
import SaveCohortModal from "@/components/Modals/SaveCohortModal";

const ProjectsCohortButton = ({pickedProjects,}: { pickedProjects: string[]; }): JSX.Element => {
    const [showSaveCohort, setShowSaveCohort] = useState(false);

    return (
        <>
            <Tooltip
                label="Save a new cohort of cases in selected project(s)"
                withArrow
            >
        <span>
          <Button
              data-testid="button-create-new-cohort-projects-table"
              variant="outline"
              color="primary"
              disabled={pickedProjects.length == 0}
              leftIcon={
                  pickedProjects.length ? (
                      <CountsIcon $count={pickedProjects.length}>
                          {pickedProjects.length}{" "}
                      </CountsIcon>
                  ) : null
              }
              onClick={() => setShowSaveCohort(true)}
              className="border-primary data-disabled:opacity-50 data-disabled:bg-base-max data-disabled:text-primary"
          >
            Save New Cohort
          </Button>
        </span>
            </Tooltip>
            {showSaveCohort && (
                <SaveCohortModal
                    onClose={() => setShowSaveCohort(false)}
                    filters={{
                        mode: "and",
                        root: {
                            "cases.project.project_id": {
                                operator: "includes",
                                field: "cases.project.project_id",
                                operands: pickedProjects,
                            },
                        },
                    }}
                />
            )}
        </>
    );
};

export default ProjectsCohortButton;
```

This custom button component uses the state variable `showSaveCohort` to determine if the `SaveCohortModal` component
needs to be shown.
The `SaveCohortModal` component is passed to the current list of projects selected by the user and handles the creation of
the cohort and saving it.

## Application Demo

In addition to the actual application, it can have a demo. The demo can be used to show the
application's functionality and is shown when the demo button is clicked. The demo button is shown when the
application is registered with `hasDemo: true,` as described in
the [Application Registration](#application-registration) section.

The application can determine if the demo button should be shown by using the `useHasDemo` hook. The `useHasDemo` hook
returns a boolean indicating if the demo button should be shown. The demo button can be shown using the following code:

```tsx
import {useIsDemoApp} from "@/hooks/useIsDemoApp";

const GenesAndMutationFrequencyAnalysisTool: React.FC = () => {
    const isDemoMode = useIsDemoApp();
...
```

## Application Registration

An application needs to be "registered" to be used in the GDC Data Portal. Registration is done by adding the application
using `createGdcAppWithOwnStore`function. If the app is not using its store, then the `createGdcApp` function can be
used.

```typescript
import {createGdcAppWithOwnStore} from "@gff/core";
import {AppContext, AppStore, id} from "@/features/projectsCenter/appApi";
import {ProjectsCenter} from "@/features/projectsCenter/ProjectsCenter";

export default createGdcAppWithOwnStore({
    App: ProjectsCenter,
    id: id,
    name: "Projects Center",
    version: "v1.0.0",
    requiredEntityTypes: [],
    store: AppStore,
    context: AppContext,
});

export const ProjectsCenterAppId: string = id;
```

The above code registers the application with the GDC Data Portal. The `createGdcAppWithOwnStore` function takes a number of
arguments:

* `App`: React.ComponentType - The application component
* `id`: string - The id of the application
* `name`: string - The name of the application
* `version`: string - The version of the application
* `requiredEntityTypes`: string[] - The required entity types for the application
* `store`: Store - The store for the application
* `context`: Context - The context for the application

The required entity types are the data types that the application requires to function. For example, the Mutation
Frequency application
requires the `ssms` entity type. While this value is not currently used, it will be in the future to determine if
the application has the data it needs to run.

The other registration needed for the application is in
[packages/portal-proto/src/features/user-flow/workflow/registeredApps.tsx](https://github.com/NCI-GDC/gdc-frontend-framework/blob/f9ab9710450172978f5f588558cbdaa2d2301418/packages/portal-proto/src/features/user-flow/workflow/registeredApps.tsx)
This file contains an array of registered applications. For example the entry for the Project Center is:

```typescript
import ProjectsIcon from "public/user-flow/icons/crowd-of-users.svg";

...
{
    name: "Projects", 
        icon: (<ProjectsIcon
            width={64}
            height={64}
            viewBox="0 -20 128 128"
            role="img"
            aria-label="Projects icon" />),
        tags: [], 
        hasDemo: false,
        id: "Projects",
        countsField: "caseCount",
        description: "View the Projects available within the GDC and select them for further exploration and analysis.",
}
...
```

The above code registers the Project Center application with the GDC Data Portal. The members of the object are:

* `name` The name of the application
* `icon` The icon as an SVG file, its size and position can be adjusted using the `width`, `height`, and `viewBox`
properties
* `tags` The tags for the application used for searching (which is not currently active)
* `hasDemo` A boolean indicating if the application has a demo. If so, the demo button will be shown.
* `id` The id of the application, which needs to match the id of the application registered in
the `createGdcAppWithOwnStore` function
* `countsField` The field to use for the counts in the application. This is used to determine if the application can
be used.
* `description` The description of the application
* `noDataTooltip` The tooltip to show if the application has no data

When the application is registered, it will be available in the GDC Data Portal. The application can be accessed by clicking on the
application card.
The visual elements of the card are:

![application_card.png](images/developers_guide/application_card.png)

*Application Card and Associated Elements*

## Source Code Layout

While developers have freedom in structuring application code, the following is a recommended layout for an
application's source code:

![source code layout](images/developers_guide/app_source_code_layout_fig.png)

*Application Source Code Layout*

## Appendix

### Using Selectors and Hooks

Although a complete guide to React hooks and selectors is out of the scope of this document, a brief
overview of how to use them for application development is provided. For more information on hooks and selectors please see the
[React Hooks](https://react.dev/reference/react/hooks) documentation. As the GDC uses the Redux-toolkit, calls
described in the [Redux Toolkit](https://redux-toolkit.js.org/tutorials/typescript) documentation are used as examples.

#### Selectors

Selectors are used to access the state of the GDC Data  Portal's main redux store. Using selectors is the preferred method for
accessing the state of the GDC Data Portal. Selectors are functions that take the state as an argument and return a value.

```typescript
import {useCoreSelector, selectCurrentCohort} from '@gff/core';

const currentCohort = useSelector(selectCurrentCohort);
```

The selector will return the current value of the item in the store. Consult the GDC 2.0 API documentation for a
complete list of selectors.

#### Hooks

Fetching data from the GDC API is done via hooks. Hooks are functions that take arguments and return a value. The value
returned is typically a promise that resolves to the data requested. The GDC Data Portal provides a number of hooks for
fetching data from the GDC 2.0 API. These hooks are located in the `@gff/core` package.

```typescript
import {useGeneSymbol} from '@gff/core';

const {data: geneSymbolDict, isSuccess} = useGeneSymbol(
    field === "genes.gene_id" ? facetValues.map((x) => x.toString()) : [],
);
```

GDC Data Portal hooks are designed to work similarly to the RTL Query hooks. The hooks take arguments and return an object,
which contains the data and the status of the query. The status of the query is stored in the `isSuccess` variable.
The data returned from the query is stored in the `data` variable. The object returned from a GDC hook is of the form:

```typescript
{
    data: any;
    isSuccess: boolean;
    isLoading: boolean;
    isError: boolean;
    error: Error;
}
```

where `data` is the data returned from the query, `isSuccess` is a boolean indicating if the query was
successful, `isLoading`
is a boolean indicating if the query is currently loading, `isError` is a boolean indicating if the query resulted in an
error, and `error` is the error returned from the query.

### Querying the GDC API Directly

There may be cases where there is a need to query the GDC API directly. The GDC Data Portal provides a number of functions for
querying  the GDC API. These functions are located in the `@gff/core` package and include:

* `fetchGdcProjects` - Fetches project data
* `fetchGdcAnnotations` - Fetches annotation data
* `fetchGdcSsms` - Fetches ssms data
* `fetchGdcCases` - Fetches cases data
* `fetchGdcFiles` - Fetches files data

The functions are wrappers around `fetchGdcEntities` function. The `fetchGdcEntities` function takes a number of arguments:

```typescript
export interface GdcApiRequest {
    readonly filters?: GqlOperation;
    readonly case_filters?: GqlOperation;
    readonly fields?: ReadonlyArray<string>;
    readonly expand?: ReadonlyArray<string>;
    readonly format?: "JSON" | "TSV" | "XML";
    readonly size?: number;
    readonly from?: number;
    readonly sortBy?: ReadonlyArray<SortBy>;
    readonly facets?: ReadonlyArray<string>;
}
```

There is also support for the GraphQL API. The `fetchGdcGraphQL` function takes two arguments:

```typescript
export const graphqlAPI = async <T>(
    query: string,
    variables: Record<string, unknown>,
): Promise<GraphQLApiResponse<T>> => ...
```

where `query` is the GraphQL query and `variables` are the variables for the query.

#### API Documentation

To access the Developers documentation for the GDC API, use the following commands in your terminal:

```shell
git clone git@github.com:NCI-GDC/gdc-frontend-framework.git
cd gdc-frontend-framework/
git checkout feat/with_api_docs
```

Next, within the repo, open `docs/api/index.html` in your browser.


# Repository

## Introduction ##

The Repository tool is where data files associated with each case in the current cohort can be browsed and downloaded. It also offers file filters for identifying files of interest.

> **NOTE:** Filters within the Repository are applied to the files associated with your cohort. If your goal is to filter the cases within your cohort, use the filters located in the Cohort Builder.


The Repository tool can be reached in one of these two ways:

* Choosing the Repository link in the GDC Data Portal header
* Clicking the play button on the Repository card in the Analysis Center

[![Repository In Header](images/ToolLinksInHeader.png)](images/ToolLinksInHeader.png "Click to see the full image.")

## Choosing a Cohort ##

The files displayed in the Repository will reflect the files that are associated with the active cohort. The current active cohort is displayed in the Main Toolbar.

[![Main Cohort Toolbar](images/MainCohortToolbar.png)](images/MainCohortToolbar.png "Click to see the full image.")

For users who want to browse all files that are available at the GDC, create a new cohort via the main toolbar and use it with the Repository tool.

## Filtering a Set of Files ##

As most users are searching for specific types of files, a set of commonly-used default facet cards can be used in the left panel of the Repository tool to allow users to filter the files presented in the table on the right. The facet cards are as follows:

* **Experimental Strategy**: Experimental strategies used for molecular characterization of the cancer
* **WGS Coverage**: Range of coverage for WGS aligned reads
* **Data Category**: A high-level data file category, such as "Raw Sequencing Data" or "Transcriptome Profiling"
* **Data Type**: Data file type, such as "Aligned Reads" or "Gene Expression Quantification". Data Type is more granular than Data Category.
* **Data Format**: Format of the data file
* **Workflow Type**: Bioinformatics workflow used to generate or harmonize the data file
* **Platform**: Technological platform on which experimental data was produced
* **Access**: Indicator of whether access to the data file is open or controlled
* **Tissue Type**: Type of tissue collected, such as "Normal" or "Tumor"
* **Tumor Descriptor**: Description of the disease present in the tumor specimen, such as "Primary" or "Metastatic"
* **Specimen Type**: Type of material taken. This includes particular types of cellular molecules, cells, tissues, organs, body fluids, embryos, and body excretory substances.
* **Preservation Method**: Method used to preserve the sample, such as "OCT" or "Snap Frozen" 

Values within each facet can be sorted alphabetically by choosing the "Name" header on the top left of each card. Alternatively, the "Files" header may be selected to sort the values by the number of files available.

Note that the categories displayed in the filters represent the values available for the active cohort.

[![Full Repository](images/FullRepo.png)](images/FullRepo.png "Click to see the full image.")

If a different filter needs to be used, a custom filter can be applied by choosing the "Add a Custom Filter" button at the top of the default filters. Each custom filter can then be searched and chosen within the pop-up window. Once a custom filter is selected, a new filter card will appear at the top of the default filters.  Custom filters can be removed from the Repository by choosing the X at the top right of each filter card.

[![Custom File Filter](images/CustomFileFilter.png)](images/CustomFileFilter.png "Click to see the full image.")

## Viewing Images ##

To view images associated with the active cohort, select the View Images button above the files table to launch the Slide Image Viewer.

## Files Table

The table shows the list of all the files associated with the active cohort, subject to any filtering that has been applied in the Repository. By default, the table provides the following information for each file:

* **Access**: Displays whether the file is open or controlled access. Users must login to the GDC Portal and have the appropriate credentials to access these files.
* **File Name**: Name of the file. Clicking the link will bring the user to the File Summary Page.
* **Cases**: The number of cases associated with the file
* **Project**: The Project that the file belongs to. Clicking the link will bring the user to the Project Summary Page.
* **Data Category**: Type of data
* **Data Format**: The file format
* **File Size**: The size of the file
* **Annotations**: Whether there are any annotations

Additional information such as Data Type and Experimental Strategy can be displayed using the Customize Columms button above the table. The table can be sorted by clicking on the headers, and the search bar above the table can be used to locate specific files.

The JSON / TSV buttons will download the files' details (file name, file size, data category, access type, etc.) in JSON and TSV format, respectively.

## Downloading a Set of Files ##

When filtering has been completed, files are ready to be downloaded. Depending on the number and size of files, the GDC has several options and recommendations for downloading them. While any amount of data can be downloaded using the GDC Data Transfer Tool or the API, files can be downloaded directly from the Data Portal if the size is 5 GB or less in total and the number of files does not exceed 10,000. For any downloads larger than 5 GB or 10,000 files, it's recommended that the download be performed using the [GDC Data Transfer Tool](https://gdc.cancer.gov/access-data/gdc-data-transfer-tool).

### Generating a Manifest File for the Data Transfer Tool ###

Select the Manifest button above the table to generate a manifest file required for batch download using the Data Transfer Tool. The manifest contains a list of the UUIDs corresponding to the files associated with the active cohort, subject to any filtering in the Repository.

### Adding/Removing Files to the Cart for Download ###

Downloads can also be performed using the Cart by first adding a set of files to the Cart. This can be done using the following methods:
* Clicking the cart icon on the left of each file. This will toggle between adding to and removing the file from the cart.
* Selecting the Add All Files to Cart button. This will add all the files in the current cohort to the Cart, subject to any filtering that has been applied in the Repository.

[![Add Files To Cart](images/AddFilesToCart.png)](images/AddFilesToCart.png "Click to see the full image.")

### Cart

The Cart page can then be reached by clicking the Cart icon at the top right of the portal.

At the upper-right of the page is a summary of all files currently in the cart:
* Number of files
* Number of cases associated with the files
* Total file size

The Cart page displays the file count by project and authorization level, as well as a table of all files that have been added to the Cart.  Files can be removed from the Cart using the trash icons at the left of each file in the table or by selecting the "Remove from Cart" option at the top of the Cart page, which removes either all files or the unauthorized ones.

[![Cart Page](images/CartPage.png)](images/CartPage.png "Click to see the full image.")

### Cart Items Table

The Cart Items table shows the list of all the files that were added to the Cart and has the same functionality as the table in the Repository. By default, it displays the following information for each file:

* **Access**: Displays whether the file is open or controlled access. Users must login to the GDC Portal and have the appropriate credentials to access these files.
* **File Name**: Name of the file. Clicking the link will bring the user to the File Summary Page.
* **Cases**: The number of cases associated with the file
* **Project**: The Project that the file belongs to. Clicking the link will bring the user to the Project Summary Page.
* **Data Category**: Type of data
* **Data Format**: The file format
* **File Size**: The size of the file
* **Annotations**: Whether there are any annotations


Additional information can be displayed using the Customize Columms button above the table. Sort can be applied by clicking on the table headers, and the search bar provides additional options for locating specific files. Details of the files can be downloaded using the JSON and TSV buttons above the table.


### Downloading Files from the Cart

To download files in the Cart, select the Download Cart button and choose either:

* **Manifest**: Downloads a manifest for the files that can be passed to the GDC Data Transfer Tool. A manifest file contains a list of the UUIDs that correspond to the files in the cart.
* **Cart**: Download the files directly through the browser. Users have to be cautious of the amount of data in the cart since this option will not optimize bandwidth and will not provide resume capabilities. This option can only be used if the total size of the files in the Cart does not exceed 5 GB.

### Additional Data Download

Additional data can be downloaded from the Cart page using the Download Associated Data button at the top of the page and choosing one of the available options.

[![Cart Metadata](images/CartAssociatedData.png)](images/CartAssociatedData.png "Click to see the full image.")

* __Clinical:__ TSV / Clinical: JSON - This includes all clinical information from the cases that are associated with the files (available as TSV or JSON)
* __Biospecimen:__ TSV / Biospecimen: JSON - This includes all biospecimen information from the cases that are associated with the files (available as TSV or JSON).
* __Sample Sheet:__ A TSV with commonly-used elements associated with each file, such as sample barcode and sample type.
* __Metadata:__ This includes all of the metadata associated with each and every file in the cart.  Note that this file is only available in JSON format and may take several minutes to download.

## File Summary Page ##

Clicking on a file name, in the tables that appear on both the Repository and Cart pages, launches the File Summary Page. Each File Summary Page provides information about a data file, such as size, MD5 checksum, and data format; information on the type of data included; links to the associated cases and biospecimen; and information about how the data file was generated or processed.

The page also includes buttons to download the file, add it to the file cart, or (for BAM files) utilize the BAM slicing function.

[![File Summary Page](images/FileSummaryPage.png)](images/FileSummaryPage.png "Click to see the full image.")

In the lower section of the screen, the following tables provide more details about the file and its characteristics:

* __Associated Cases/Biospecimen__: List of cases or biospecimen the file is directly attached to
* __Analysis and Reference Genome__: Information on the workflow and reference genome used for file generation
* __Read Groups__: Information on the read groups associated with the file
* __Metadata Files__: Experiment metadata, run metadata, and analysis metadata associated with the file
* __Downstream Analysis Files__: List of downstream analysis files
* __File Versions__: List of all versions of the file


# Projects

At a high level, data in the Genomic Data Commons is organized by project. Typically, a project is a specific effort to study a particular type(s) of cancer undertaken as part of a larger cancer research program. The GDC Data Portal allows users to access aggregate project-level information via the Projects tool and Project Summary Pages.

## Projects Tool

The Projects tool provides an overview of all harmonized data available in the GDC, organized by project. It also provides filtering, navigation, and advanced visualization features that allow users to identify and browse projects of interest. Users can access the Projects tool from the GDC Data Portal header.

[![Project Tool In Header](images/ToolLinksInHeader.png)](images/ToolLinksInHeader.png "Click to see the full image.")

On the left, a panel of facets allows users to apply filters to find projects of interest. When filters are applied, the table on the right is updated to display only the matching projects. When no filters are applied, all projects are displayed.

The right side of the Projects tool displays a table that contains a list of projects and specific details about each project, such as the number of cases, types of diseases and primary sites, the program involved, and the experimental strategies available. When a project contains more than one value for the disease type and primary site properties, the full list of values can be expanded by choosing the drop down icon next to the name of the property.

[![Project Page](images/ProjectsPage.png)](images/ProjectsPage.png "Click to see the full image.")

Cohorts can be created by selecting individual projects and using the Save New Cohort button above the table. The checkbox in the header allows all projects on the current page of the table to be selected at the same time.

### Facets Panel

Facets represent properties of the data that can be used for filtering. The facets panel on the left allows users to filter the projects presented in the Table.

Users can filter by the following facets:

*   __Primary Site__: Anatomical site of the cancer under investigation or review
*   __Program__: Research program that the project is part of
*   __Disease Type__: Type of cancer studied
*   __Data Category__: Type of data available in the project
*   __Experimental Strategy__: Experimental strategies used for molecular characterization of the cancer

Filters can be applied by selecting values of interest in the available facets, for example "WXS" and "RNA-Seq" in the "Experimental Strategy" facet, and "Brain" in the "Primary Site" facet. When facet filters are applied, the Table is updated to display matching projects.

## Creating Cohorts From Selected Projects

Custom cohorts consisting of specific projects can be created by selecting those projects in the table using the check boxes next to the project names and clicking the "Save New Cohort" button above the table.

[![Create Cohorts](images/ProjectsCreateCohorts.png)](images/ProjectsCreateCohorts.png "Click to see the full image.")

## Project Summary Page

Clicking the link for each project name on the table will bring users to that specific project's summary page.  This page contains basic information about the contents of a project as well as the percentages of cases within the project that contain a specific experimental strategy or data category.

[![Project Entity](images/ProjectEntity.png)](images/ProjectEntity.png "Click to see the full image.")

Four buttons on the left of the header allow the user to perform a variety of actions related to the project:

* __Save New Cohort__: Creates a new cohort consisting of all the cases in the project
* __Biospecimen__: Downloads biospecimen metadata associated with all cases in the project in either TSV or JSON format
* __Clinical__: Downloads clinical metadata about all cases in the project in either TSV or JSON format
* __Manifest__: Downloads a manifest for all data files available in the project. The manifest can be used with the [GDC Data Transfer Tool](../../Data_Transfer_Tool/Users_Guide/Getting_Started.md) to download the files.

### Primary Sites Table

Summary pages for projects with multiple primary sites also include a Primary Sites table. Each row of the table contains information relevant to a specific primary site within the project, and additional cohorts can be created using buttons located within the table.

[![Primary Sites Table](images/PrimarySitesTable.png)](images/PrimarySitesTable.png "Click to see the full image.")


# Clinical Data Analysis

The Clinical Data Analysis tool allows for a set of customizable charts to be generated for a set of clinical attributes. Users can select which clinical fields they want to display and visualize the data using various supported plot types.  The clinical analysis features include:

* Ability to select which clinical fields to display
* Examine the clinical data of each field using these visualizations:
    * Histogram
    * Survival Plot
    * Box and QQ Plots
* Create custom bins for each field and re-visualize the data with those bins
* Select specific cases from a clinical field and use them to create a new cohort, or modify/remove from an existing cohort
* Download the visualizations of each plot type for each variable in SVG or PNG
* Download the data table of each field in JSON or TSV format
* Print all clinical variable cards in the analysis with their active plot to a single PDF

## Enabling Clinical Variable Cards

* In the Analysis Center, select the *Clinical Data Analysis* tool card.

* In the Clinical Data Analysis tool, use the control panel on the left side of the analysis to display which clinical variables you want.  To enable or disable specific variables for display, click the on/off toggle controls:

[![Enable Clinical Cards](images/CDACards.png)](images/CDACards.png "Click to see the full image.")

The clinical fields are grouped into these categories:

* __Demographic:__ Data for the characterization of the patient by means of segmenting the population (e.g. characterization by age, sex, race, etc.).
* __Diagnosis:__ Data from the investigation, analysis, and recognition of the presence and nature of the disease, condition, or injury from expressed signs and symptoms; also, the scientific determination of any kind; the concise results of such an investigation.
* __Treatment:__ Records of the administration and intention of therapeutic agents provided to a patient to alter the course of a pathologic process.
* __Exposure:__ Clinically-relevant patient information not immediately resulting from genetic predispositions.

## Exploring Clinical Card Visualizations

Users can explore different visualizations for each clinical field they have enabled for display. All cards support histograms and survival plots. Additionally, continuous variables can be graphically represented as box and QQ plots. To switch between plot types, click the different plot type icons in the top-right of each card.

### Histogram

The histogram plot type supports these features:

* View the distribution of cases (# and % of cases) in the cohort for the clinical field's data categories as a histogram
* View the distribution of cases in tabular format
* Select the cases for specific data categories to create new cohorts, append to existing cohorts, or remove from existing cohorts
* Download the histogram visualization in SVG or PNG format
* Download the raw data used to generate the histogram in JSON format

[![Clinical Data Analysis Histograms](images/CDAHistograms.png)](images/CDAHistograms.png "Click to see the full image.")

Note that the histogram plot applies to, and can be displayed for, both categorical and continuous variables.

### Survival Plot

The survival analysis is used to analyze the occurrence of event data over time.  In the GDC, survival analysis is performed on the mortality of the cases. Thus, the values are retrieved from [GDC Data Dictionary](../../../Data_Dictionary) properties and a survival analysis requires the following fields:

*  Data on the time to a particular event (days to death or last follow up).
    * Fields:  __demographic.days_to_death__ or __demographic.days_to_last_follow_up__
*  Information on whether the event has occurred (alive/deceased).
    * Fields:  __demographic.vital_status__
*  Data split into different categories or groups (i.e. gender, etc.).
    * Fields:  __demographic.gender__

The survival analysis in the GDC uses a Kaplan-Meier estimator:

[![Kaplan-Meier Estimator](images/gdc-kaplan-meier-estimator2.png)](images/gdc-kaplan-meier-estimator2.png "Click to see the full image.")

Where:

 * S(t) is the estimated survival probability for any particular one of the t time periods.
 * n<sub>i</sub> is the number of subjects at risk at the beginning of time period t<sub>i</sub>.
 * and d<sub>i</sub> is the number of subjects who die during time period t<sub>i</sub>.

The table below is an example data set to calculate survival for a set of seven cases:

[![Sample Survival Analysis Table](images/gdc-sample-survival-table.png)](images/gdc-sample-survival-table.png "Click to see the full image.")

The calculated cumulated survival probability can be plotted against the interval to obtain a survival plot like the one shown below.

[![Sample Survival Analysis Plot](images/gdc-survival-plot.png)](images/gdc-survival-plot.png "Click to see the full image.")

The survival plot type supports these features:

* View the distribution of cases (# and % of cases) in the cohort for the clinical field's data categories as a table.
* Select and plot the survival analysis for the cases of specific data categories in the table:
    * By default the top 2 categories (highest # of cases) are displayed.
    * Users can manually select and plot up to 5 categories at a time.
* Download the survival plot visualization in SVG or PNG format
* Download the raw data used to generate the survival plot in JSON or TSV format

[![Clinical Data Analysis Survival Plot](images/CDASurvivalPlot.png)](images/CDASurvivalPlot.png "Click to see the full image.")

Note that the survival plot applies to, and can be displayed for, both categorical and continuous variables.

### Box and QQ Plots

The box and QQ plot types support these features:

* View the quartiles (Q1, Q2/median, and Q3) as well as the mean, minimum, and maximum values in the cohort for the clinical field as a box plot
* View the descriptive statistics in the cohort for the clinical field in tabular format
* Plot the quantiles of the clinical field's distribution with quantiles of a theoretical normal distribution as a QQ plot
* Download the box and QQ plot visualizations in SVG or PNG format
* Download the raw data used to generate the QQ plot in JSON or TSV format

[![Clinical Data Analysis Box and QQ Plots](images/CDABoxQQPlots.png)](images/CDABoxQQPlots.png "Click to see the full image.")

Note that the box and QQ plots apply to, and can be displayed for, continuous variables only.

Certain continuous variables that are measured with units of time, such as Days to Birth, include a toggle to switch between displaying the data in years or days. A standard formula is employed for converting between years and days:

* 1 year = 365.25 days

[![Year and Day Toggle](images/CDAYearsDaysToggle.png)](images/CDAYearsDaysToggle.png "Click to see the full image.")

## Creating Custom Bins

For each clinical variable, whether categorical or continuous, users can create custom bins to group the data in ways they find scientifically interesting or significant.  Once saved, the bins are applied to these visualizations and they are then re-rendered:

* Histogram and associated data table
* Survival plot and associated data table

Custom bins can be reset to their defaults at any time for each card by selecting the *__Reset to Default__* option after clicking *__Customize Bins__*.

### Categorical Binning

To create custom bins for a categorical variable, click *__Customize Bins__*, then *__Edit Bins__*.  A configuration window appears where the user can create their bins:

[![Clinical Data Analysis Categorical Binning](images/CDACatBins.png)](images/CDACatBins.png "Click to see the full image.")

The user can:

* Group existing individual values into a single group
* Give a custom name to each group
* Ungroup previously grouped values
* Completely hide values from being shown in the visualization
* Re-show previously hidden values

### Continuous Binning

To create custom bins for a continuous variable, click *__Customize Bins__*, then *__Edit Bins__*.  A configuration window appears where the user can create their bins:

[![Clinical Data Analysis Continuous Binning](images/CDAContBins.png)](images/CDAContBins.png "Click to see the full image.")

The user can choose one of these continuous binning methods:

* (1) Create equidistant bins based on a set interval:
    * User must choose the interval (e.g. equidistant bins of 1,825 days for the Age of Diagnosis field)
    * User can optionally define the starting and ending value between which the equidistant bins will be created
* (2) Create completely custom ranges:
    * User manually enters 1 or more bins with custom ranges
    * User must enter a name for each range and the start and end values
    * The ranges can be of different interval lengths


# ProteinPaint Tool

## Introduction to ProteinPaint

ProteinPaint is a web-based, dynamic visualization tool that displays a lollipop chart based on the multidimensional skewer version 3 (mds3 track). This tool utilizes variant annotations from GDC datasets. Given a particular gene, it displays variants associated with that gene as well as the occurrence, disease type, and demographic information of the associated case given a case.

## Quick Reference Guide

At the Analysis Center, click on the 'ProteinPaint' card to launch the app.

[![Analysis Center](images/lollipop1.png)](images/lollipop1.png "Click to see the full image.")

Users can view publicly available variants as well as login with credentials in order to access controlled data.

When launched, ProteinPaint will display a search box where users can enter a gene symbol, alias, or GENCODE accession. Once a gene is entered, a lollipop frame is displayed with the name of the chart in the header.

[![Lollipop Frame](images/lollipop2.png)](images/lollipop2.png "Click to see the full image.")

In addition to the search box, there are two other main panels in the ProteinPaint tool: [Lollipop chart](#Lollipop-Chart-Panel), and [legend](#Legend-panel).

[![3 Main Panels](images/lollipop3.png)](images/lollipop3.png "Click to see the full image.")

### Lollipop Chart Panel

After entering a gene, the tool will display a Lollipop chart for the GDC variants as well as a Protein View for the default isoform.

In the Lollipop chart, the circular discs for each variant are color coded per GDC mutation classes and are proportional in size to the number of occurrences. Variants in the same position are arranged in descending order of occurrences.

[![Lollipop Chart](images/lollipop10.png)](images/lollipop10.png "Click to see the full image.")

Exon variants report the amino acid change at the referenced codon. For example, G12D is a G > D substitution at the 12th codon of the protein.

The default isoform will appear directly to the right of the gene name. Clicking on the isoform number will open a display to view/select other isoforms and switch the display [track](#protein-view).

[![Display All Isoforms](images/lollipop8.png)](images/lollipop8.png "Click to see the full image.")

Clicking on the number of variants link, to the left of the plot, opens a menu where users can view annotations and manipulate the Lollipop:

* __List:__ Displays all variants, each of which can be selected to launch the annotation table which displays consequence, mutation, sample submitter_id, and other data related to the sample
    * __Mutation:__ Launches the [GDC Mutation Summary Page](mutation_frequency.md#gene-and-mutation-summary-pages)
    * __Sample:__ Launches the [GDC Case Summary Page](quick_start.md#cohort-case-table). Users also have the ability to create a new cohort or launch the [Disco plot](oncomatrix.md#disco-plot).
* __Collapse/Expand:__ Collapses or expands all skewers in the lollipop
* __Download:__ Downloads the mutations in a TXT file
* __As lollipops:__ Displays variants via circular discs proportional to the number of occurrences
* __Occurrence as Y axis:__ Sorts variants on the y-axis by number of occurrences

Clicking on the number of samples opens a window to view annotations grouped by GDC case properties such as disease type and primary site. Selecting a value adds a new Lollipop subtrack that displays only the samples with the given value. This side-by-side view allows for a comparison between the mutations in the main track versus the subtrack.

[![Adenomas and Adenocarcinomas Example: Side-by-Side View](images/lollipop24.png)](images/lollipop24.png "Click to see the full image.")

Each subtrack offers advanced filtering for users to narrow down particular features. Clicking on the value to the right of the Lollipop launches a pop-up window where users can add subsequent filters using the `+AND` or `+OR` options.

[![Advanced Filtering](images/lollipop25.png)](images/lollipop25.png "Click to see the full image.")

Detailed variant annotation is viewable by clicking on the disc next to the variant label. The sunburst chart is composed of a ring hierarchy, arranged by disease types then broken down by primary sites.

[![Variant Occurrence](images/lollipop31.png)](images/lollipop31.png "Click to see the full image.")

Hovering over the inner and outer rings displays the disease type or primary site, number of samples, and cohort size.

[![Node](images/lollipop33.png)](images/lollipop33.png "Click to see the full image.")

An aggregate table displaying all the samples associated with that variant is available by clicking the 'Info' button in the center of the sunburst.

[![Sample Annotation Table](images/lollipop_sample_table.png)](images/lollipop_sample_table.png "Click to see the full image.")

The top of the table displays consequence, mutation, and occurrence count with a link to the [GDC Mutation Summary Page](mutation_frequency.md#gene-and-mutation-summary-pages).

The sample table contains a number of columns for various associated features per sample such as Disease type, Mutations, and Tumor DNA Mutant Allele Frequency. Users can create a new cohort by selecting the checkboxes in the first column then clicking 'Create Cohort' in the bottom right corner of the table. The table also includes options to launch the [Disco plot](oncomatrix.md#disco-plot) and the [GDC Case Summary Page](quick_start.md#cohort-case-table) for each sample.

#### Protein View

The Protein View, which displays the nucleotides, codons in the exon region, introns, and protein domains, is the primary area in which a user will visualize and interact with protein coding regions.

[![Protein View](images/lollipop6.png)](images/lollipop6.png "Click to see the full image.")

To zoom into the Protein View, users can highlight a region or use the zoom buttons (`In`, `Out x2`, `x10`, and `x50`) in the toolbar. For viewing a nucleotide of interest, click and drag in the top Protein length scale. The region appears highlighted in red with the calculated protein length in center.

[![Zoom Method](images/lollipop37.gif)](images/lollipop37.gif "Click to see the full image.")

Zooming in to the protein track displays the codons and the nucleotides. Hovering over the nucleotide position displays a tooltip with the exon, amino acids position, RNA position, and protein domain.

[![Codons and Nucleotides](images/lollipop37.png)](images/lollipop37.png "Click to see the full image.")

By clicking on the isoform number in the Lollipop chart, users can switch the display track between genomic, splicing RNA, exon only, protein, and aggregate of all isoforms.

### Legend Panel

#### Protein Domains

The Protein View color codes regions by the protein domain present on the full-length protein region in the exon display.

[![Protein Domains](images/lollipop38.png)](images/lollipop38.png "Click to see the full image.")

The legend offers simple filtering for the variants shown in the lollipop. To the right of `PROTEIN`, users can click on the color to hide that particular protein domain. Clicking on the color again shows the protein domain.

Custom protein domains are added by clicking on the `+ add protein domain` button at the bottom of the list. An input box appears requiring the following information:

1. Name, text with space, no semicolon: Name of the protein domain
2. Range, two integers joined by space: Codon position - start and stop
3. Color (e.g., red, #FF0000, rgb (255,0,0)): Color to assign to the protein domain

The protein domains also include links to databases of protein families such as the [Conserved Domains Database (CDD)](https://www.ncbi.nlm.nih.gov/cdd/), [Simple Modular Architecture Research Tool (SMART)](https://smart.embl.de), and [Pfam](https://www.ebi.ac.uk/interpro/).

#### GDC Mutation Class

The GDC mutation class color coding for the lollipop discs appears below the legend for the protein domains.

[![Color Coding Legend](images/lollipop40.png)](images/lollipop40.png "Click to see the full image.")

Clicking on a mutation class opens a pop-up menu with show/hide functionalities:

* __Hide:__ Remove all of the lollipop discs for the particular mutation class
* __Show only:__ Only show the lollipop discs for the particular mutation class
* __Show all:__ Display the lollipop discs for all mutation classes

[![Mutation Pop-Up Menu](images/lollipop42.png)](images/lollipop42.png "Click to see the full image.")

### Additional Features

In the toolbar, the `More` button offers methods to download figures and data:

[![ProteinPaint More Options](images/lollipop44.png)](images/lollipop44.png "Click to see the full image.")

* __Export SVG:__ Download the Lollipop and legend as an SVG file
* __Reference DNA Sequence:__ Display the DNA sequence as plain text for easy copying and pasting
* __Highlight:__ Highlight a region in the Lollipop by selecting it in the chart or entering it in a text box

## ProteinPaint Features
When selected, ProteinPaint will display the search-box as illustrated below. Once a user enters a gene symbol, alias, or GENCODE accession, a lollipop frame is displayed with the name of the chart in the header. The example below is of the gene AKT1. All gene symbols are based on the [HGNC](https://www.genenames.org/about/guidelines/) guidelines.

[![Lollipop Frame](images/lollipop2.png)](images/lollipop2.png "Click to see the full image.")

There are 3 main panels as outlined in the figure below:

1. Search box
2. Lollipop chart panel
3. Legend panel

[![3 Main Panels](images/lollipop3.png)](images/lollipop3.png "Click to see the full image.")

### Search Box

[![Search Box](images/lollipop4.png)](images/lollipop4.png "Click to see the full image.")

The example below uses the KRAS gene. The name of the gene (e.g., 'KRAS'), GENCODE accession no. (e.g., [ENST00000311936](http://useast.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;g=ENSG00000133703;r=12:25205246-25250936;t=ENST00000311936), [ENSP00000308495](http://useast.ensembl.org/Homo_sapiens/Transcript/ProteinSummary?db=core;g=ENSG00000133703;r=12:25205246-25250936;t=ENST00000311936)) or RefSeq accession (e.g., NM_004985) can be used as the search item. In case a wrong gene is entered, the search box will display an error. For gene searches only, typing a few letters reveals a menu of possible matches. Choose from either a menu option or hit enter.

[![Search Box Example](images/lollipop5.png)](images/lollipop5.png "Click to see the full image.")

### Lollipop Chart Panel

### Protein View
After searching for KRAS, the Protein View for the default isoform appears in a new frame. The Protein View displays the nucleotides, codons in the exon region, introns, and protein domains as shown below.

[![Protein View](images/lollipop6.png)](images/lollipop6.png "Click to see the full image.")

The legend offers simple filtering for the variants showing in the lollipop. Clicking the color for a protein domain on the right of PROTEIN for example, hides that protein domain. Clicking on the color again shows the protein domain. Similar show/hide functions are available by clicking on the legend labels.

The default isoform for KRAS on hg38 genome build is NM_004985. Hovering over the isoform label will highlight it as shown below.

[![Default Isoform](images/lollipop7.png)](images/lollipop7.png "Click to see the full image.")

A user can select the isoform by clicking on the isoform number as shown in the figure above. Clicking this will open a display to view all the other isoforms as well as the option to switch the display track as shown below in the figure.

[![Display All Isoforms](images/lollipop8.png)](images/lollipop8.png "Click to see the full image.")

From **Switch Display**, a user can update to one of the following:
1. Genomic display
2. Splicing RNA
3. Exon only
4. Protein track
5. Aggregate of all isoforms

The Protein track is the primary area in which a user will visualize and interact with protein coding regions.

#### Protein Track

Under **Switch Isoform**, the available RefSeq and Ensembl isoform builds are listed. A condensed display and the protein length is shown for each isoform. The current selection appears in red text. The default KRAS isoform for example, is NM_004985 with 189 amino acids. To change the isoform, click on the appropriate line highlighted in yellow.

[![Switch Isoforms](images/lollipop9.png)](images/lollipop9.png "Click to see the full image.")

The pop-up window disappears and the lollipop track rerenders with the newly selected isoform.

# Lollipop Charts

The lollipop chart for the GDC variants appears above the Protein View. The circular disc for each variant is proportional to the number of occurrences. Variants in the same position are arranged in descending order of magnitude. There are eight types of variants found in the lollipop chart (see legend).

[![Lollipop Chart](images/lollipop10.png)](images/lollipop10.png "Click to see the full image.")

Exon variants report the amino acid change at the referenced codon. For example, G12D is a G > D substitution at the 12th codon of the protein.

Clickable links for the number of cases (e.g. 1315 samples) and number of variants (e.g. 99 out of 110 variants) appear to the left of the lollipop. Clicking on these links reveals detailed annotations about the samples and variants, described in [Viewing Variants and Case Samples](#Viewing Variants and Case Samples).

[![Sample and Variant Annotations](images/lollipop11.png)](images/lollipop11.png "Click to see the full image.")

## Viewing Variants and Case Samples

### Variant Annotations and Chart Manipulation
Click on the number of variants linked to the left of the lollipop for viewing annotations and manipulating the lollipop. For variant annotation, click on 'List'.

[![Variant Annotation](images/lollipop12.png)](images/lollipop12.png "Click to see the full image.")

A pop-up window appears with the entire list of variants, as shown below.

[![List of Variants](images/lollipop13.png)](images/lollipop13.png "Click to see the full image.")

Click on the variant of interest and a new annotation table appears. From the table, view various associated features per sample such as: Disease type, Primary site, Project id, Gender, Race, Ethnicity, and Tumor DNA Mutant Allele Frequency(MAF). In the figure below, 333 occurrences are shown for the G12D variant, which represents a missense mutation at chromosome chr12:25245350 C>T.  
[![Annotation Table](images/lollipop14.png)](images/lollipop14.png "Click to see the full image.")

The first sample that is highlighted in yellow is a male with ductal and lobular neoplasms with a tumor DNA MAF of 31/125. This indicates 31 mutant alleles were found out of 125 total alleles.

The GDC dataset includes an 'Access' column to indicate whether the data is controlled or open. Users must obtain permission from dbGaP to view controlled data [See Obtaining Access to Controlled Data](https://gdc.cancer.gov/access-data/obtaining-access-controlled-data). Click on the sample hyperlink and the GDC's case summary for the sample will appear in a new tab.

Click 'Back to list' and select another sample, as shown below.

[![Back to List](images/lollipop15.png)](images/lollipop15.png "Click to see the full image.")

After clicking on the variant menu again, select the 'Collapse' option to collapse all skewers in the lollipop.

[![Collapse](images/lollipop16.png)](images/lollipop16.png "Click to see the full image.")

To expand any previously collapsed skewers, open the variant menu, and click on 'Expand'.

[![Expand](images/lollipop17.png)](images/lollipop17.png "Click to see the full image.")

The lollipop chart includes an option to arrange variants by the range of occurrences. Open the variant menu and click on  'Occurrence as Y axis'.

[![Occurrence as Y Axis](images/lollipop18.png)](images/lollipop18.png "Click to see the full image.")

The lollipop re-renders with the variants sorted on the y-axis from lowest and highest occurrence. Hover over a variant to display the number of occurrences. In the example below, a user is hovering over G12D to display 333 occurrences of this variant.

[![Number of Occurrences](images/lollipop19.png)](images/lollipop19.png "Click to see the full image.")

Clicking on the variant loads the sample table again as shown below.

[![Sample Table](images/lollipop20.png)](images/lollipop20.png "Click to see the full image.")

### Case Filtering

Clicking on the sample hyperlink on the left of the lollipop (e.g. 1315 samples) opens a menu to list all samples. Aggregate data for all samples by attribute appears in a series of tabs. The ability for advanced filtering and creating subtracks is available from this new display.

[![Menu to List All Samples](images/lollipop21.png)](images/lollipop21.png "Click to see the full image.")

Click on 1315 samples to view annotations grouped by attributes such as: Disease type, Primary site, Project id, Gender, Race, Ethnicity, etc.. For each attribute, the number of values is represented by 'n' to the right of the group label. In the figure below, 21 values for Disease type are reported.

[![Annotations Grouped by Attributes Example: Disease type](images/lollipop22.png)](images/lollipop22.png "Click to see the full image.")

To start filtering, click on the value label or the value's sample fraction. Clicking on 'Adenomas and Adenocarcinomas' or '675/ 4866' for example, loads a new lollipop subtrack underneath the main GDC lollipop track.

[![Filtering Example: Adenomas and Adenocarcinomas](images/lollipop23.png)](images/lollipop23.png "Click to see the full image.")

This new subtrack only shows the 675 Adenomas and Adenocarcinomas samples. This side-by-side view allows for a comparison between the mutations in the main track vs the subtrack.

[![Adenomas and Adenocarcinomas Example: Side-by-Side View](images/lollipop24.png)](images/lollipop24.png "Click to see the full image.")

Each subtrack offers advanced filtering, shown below, for users to narrow down particular features.

[![Advanced Filtering](images/lollipop25.png)](images/lollipop25.png "Click to see the full image.")

Clicking on 'Filter' displays a pop-up window with the feature the user selected previously from the sample annotation menu (e.g. Disease type: Adenomas and Adenocarcinomas). Clicking on either +AND or +OR displays a new pop-up with a search bar. Search for the desired term and click on the term's button. In the image below a user selected 'gender' by clicking the '+AND'.

[![Filter Pop-Up Window](images/lollipop26.png)](images/lollipop26.png "Click to see the full image.")

By clicking on 'Gender', all available values appear with checkboxes (i.e. male and female) as shown below. In this example, male with 293 data points is selected.

[![Gender Filter Example 1](images/lollipop27.png)](images/lollipop27.png "Click to see the full image.")

Click 'Apply' and the subtrack re-renders to reflect the updated filter. In the example below, the subtrack reduces from 675 samples to the 293 male samples with adenomas and adenocarcinomas. The figure shows the difference in mutations in the two tracks. Out of the original 333 samples, 72 of 293 males report the G12D mutation.

[![Gender Filter Example 2](images/lollipop28.png)](images/lollipop28.png "Click to see the full image.")

Click on the 'Close' option to remove the subtrack from the page.

[![Remove Subtrack](images/lollipop29.png)](images/lollipop29.png "Click to see the full image.")

### Viewing in the Lollipop Display
In the lollipop chart, users can drag the protein track down by clicking the name of the gene on the left of the protein track and pulling it below the lollipop chart.

[![Protein Track](images/lollipop30.png)](images/lollipop30.png "Click to see the full image.")

Detailed variant annotation is viewable by clicking on the variant disc next to the label. For G12D highlighted in a red outline in the image above, click on the '333' disc. A sunburst chart will appear, shown below.

The center displays the occurrence of the variant (333) above the variant label. The ring

[![Variant Occurrence](images/lollipop31.png)](images/lollipop31.png "Click to see the full image.")

hierarchy is arranged by disease types then broken down by primary sites. Hovering over the inner ring displays the disease type, number of samples, and cohort size. In this example, the inner green ring displays 'Plasma Cell Tumors' with 28 samples out of a total 949 samples.

The outer ring represents the primary sites. Hovering over the primary site displays the number of samples relative to the disease type. In the figure below, for Ductal and lobular

[![Variant Occurrence](images/lollipop32.png)](images/lollipop32.png "Click to see the full image.")

neoplasms, there are 105 samples with the primary site as pancreas out of 316 total samples.  

[![Node](images/lollipop33.png)](images/lollipop33.png "Click to see the full image.")

Clicking on a node displays a sample table for the disease type or primary site. In the figure below, the user selected 'Plasma Cell Tumors'. The sample annotation table appears for all Plasma Cell Tumors.

[![Sample Annotation Table 1](images/lollipop34.png)](images/lollipop34.png "Click to see the full image.")

An aggregate sample table is available by clicking the 'Info' button in the center of the sunburst. This displays all the samples associated with that variant. In the screen recording below the aggregate sample table appears for KRAS - G12D.

[![Sample Annotation Table 2](images/lollipop35.gif)](images/lollipop35.gif "Click to see the full image.")

Clicking on the sample name hyperlink opens a new tab to the sample's  GDC Case Summary page.

Clicking on the variant label in the center removes the sunburst chart.

[![Variant Label](images/lollipop36.gif)](images/lollipop36.gif "Click to see the full image.")

### Working With the Protein Track
There are two zoom methods: highlighting a region and zoom buttons in the toolbar. For viewing a nucleotide of interest, click and drag the mouse in the top, x-axis, Protein length scale. The region appears highlighted in red with the calculated protein length in center.

[![Zoom Method](images/lollipop37.gif)](images/lollipop37.gif "Click to see the full image.")

Once the mouse is released, the lollipop re-renders as the selected region.

The zoom buttons in the toolbar is the second option to zoom in and out based on the center position of the lollipop. For zooming out, users can choose to zoom out 2x, 10x or 50x times.

[![Zoom Options](images/lollipop36.png)](images/lollipop36.png "Click to see the full image.")

Zooming in causes the protein track to display the codons and the nucleotides as shown below. Hovering over the nucleotide position displays a tooltip with the exon, amino acids position, RNA position, and protein domain. As shown in the image below, at codon 12, the second exon of the transcript, RNA position 225 bp, the reference allele is a 'G'. There is a substitution at 'G' to A, V and D in the KRAS gene for isoform NM_004985 for which the cases are as shown below.

[![Codons and Nucleotides](images/lollipop37.png)](images/lollipop37.png "Click to see the full image.")

# Legend Panel

The protein track color codes regions by the protein domain present on the full-length protein region in the exon display. For KRAS, the protein domains are shown in the red box in the image below.

[![Protein Domains](images/lollipop38.png)](images/lollipop38.png "Click to see the full image.")

### Protein Domain Legend
Clicking on the colored box next to the protein domain label removes the color from the protein track, as depicted below.

[![Remove Color from Protein Track](images/lollipop39.png)](images/lollipop39.png "Click to see the full image.")

Custom protein domains are added by clicking on the '+add protein domain' button at the bottom of the list. An input box appears requiring the following information:
1. Name, text with space, no semicolon: This is the name of the protein domain
2. Range, two integers joined by space: This is the codon position - start and stop
3. Color (e.g., red, #FF0000, rgb (255,0,0)): This is the color to assign to the protein domain.

### GDC Mutations
The lollipop discs are color coded per GDC mutation classes. The legend for the mutations appears below the protein domains with more advanced show/hide functions.

[![Color Coding Legend](images/lollipop40.png)](images/lollipop40.png "Click to see the full image.")

The classification for the type of variant is color coded as follows:

[![Color Coding Classification](images/lollipop41.png)](images/lollipop41.png "Click to see the full image.")

Clicking on a mutation prompts a pop-up menu to appear with the description of the mutation. Options to 'hide' or 'show only' are specific to the mutation. The option 'show all' includes all previously hidden mutations. Selecting 'MISSENSE' shown in the figure below by the yellow highlight displays the initial menu with the 'hide' and 'show only' buttons.

[![Mutation Pop-Up Menu](images/lollipop42.png)](images/lollipop42.png "Click to see the full image.")

Clicking 'Hide' removes all of the mutation discs from the lollipop. The mutation is reordered to the end of the list and the font is striked through and grayed out. The discs reappear when the mutation label is clicked again.

[![Hide Mutation Discs](images/lollipop43.png)](images/lollipop43.png "Click to see the full image.")

# More Options

ProteinPaint offers methods to download figures and data. Click the 'More' button in the toolbar to display various options as shown below.

[![ProteinPaint More Options](images/lollipop44.png)](images/lollipop44.png "Click to see the full image.")

### Exporting the Figure

Click "Export SVG" to download the lollipop and legend as an SVG file.

[![Export SVG](images/lollipop45.png)](images/lollipop45.png "Click to see the full image.")

The exported figure will contain following contents, reflecting a user's customization:
* Displayed datasets, including custom data
* Expand/fold states of all mutations
* Sequences in the protein if at zoom-in level
* Show/hide state of exon boundaries
* Sunburst charts
* Protein domains without the hidden ones
* All mutations without the hidden classes or origins
* Legend for protein domain, mutation class and origin

### Copying the DNA Sequence

The 'More' button also includes a 'DNA sequence' button.

[![DNA Sequence](images/lollipop46.png)](images/lollipop46.png "Click to see the full image.")

Clicking on 'DNA sequence' displays the DNA sequence as plain text for easy copying and pasting.

[![DNA Sequence Plan Text](images/lollipop47.png)](images/lollipop47.png "Click to see the full image.")

### Popup Option

The pop up option under the More button allows for popping open another window with the same lollipop display selected by the user. Below is an example.

[![Popup](images/lollipop48.png)](images/lollipop48.png "Click to see the full image.")

[![Popup Window](images/lollipop49.png)](images/lollipop49.png "Click to see the full image.")


# Getting Started with the GDC Data Portal

## Accessing the GDC Data Portal
First, go to [https://portal.gdc.cancer.gov/](https://portal.gdc.cancer.gov/).

## GDC Data Portal Header

The header of the GDC Data Portal contains frequently used links and features.

[![Header](images/Header.png)](images/Header.png "Click to see the full image.")

On the upper-left is the GDC Data Portal logo, which links to the home page of the GDC Data Portal. Below the logo are links in the following order:

* [Analysis Center](analysis_center.md): the central hub for accessing all the tools in the GDC Data Portal
* [Projects](Projects.md): allows exploration of all the projects within the GDC Data Portal
* [Cohort Builder](cohort_builder.md): the Cohort Builder tool consists of a variety of clinical and biospecimen filters for building custom cohorts for analysis
* [Repository](Repository.md): allows exploration of files associated with a cohort

On the right are the following features:

* **Video Guides**: [videos](Video_Tutorials.md) that demonstrate the various features of the Data Portal
* **Send Feedback**: send feedback to the GDC team
* **Browse Annotations**: the Annotations Browser, where the user can view and search for [annotations](../../Encyclopedia/pages/Annotations.md) that may provide additional context when analyzing GDC data
* **Manage Sets**: review gene and mutation sets that have been saved, upload new sets, and delete existing sets
* **Cart**: the data files that are ready for download
* **Login**: allows authentication for access to [controlled access datasets](../../Encyclopedia/pages/Controlled_Access.md). Once authentication is successful, the eRA Commons username will be displayed in place of the "Login" button. Clicking on the username will then allow users to see which projects they have access to and to download an [authentication token](../../Data/Data_Security/Data_Security.md#authentication-tokens) for use with the [GDC Data Transfer Tool](../../Data_Transfer_Tool/Users_Guide/Getting_Started.md) and the [API](../../API/Users_Guide/Getting_Started.md).
* **GDC Applications**: contains links to other GDC sites and applications
* **Search**: search for projects, cases, files, genes, mutations, and annotations within the GDC Data Portal


## Cohorts

The GDC Data Portal 2.0 is a cohort-centric cancer research platform. Users can create custom cohorts based on specific projects, primary sites, disease types, or any combination of clinical, biospecimen, and molecular features. Custom cohorts can then be used with various tools in the Analysis Center to perform further analysis. Files from custom cohorts can also be downloaded for further analysis with other research tools.

If the user does not already have a custom cohort when they are in the Analysis Center, a custom cohort ("Unsaved_Cohort") containing all cases in the GDC will be automatically created. This allows the user to explore the Analysis Center without first needing to create a cohort.

Additional cohorts can be created using the main toolbar in the Analysis Center. Cohorts can also be saved or deleted using the main toolbar. See the section below on the Analysis Center for more information on the main toolbar.

Unsaved cohorts are not retained once the browser tab is closed. Saved cohorts continue to be accessible as long as the same browser is used and should be available through data releases.

## Home Page

The Analysis Center can be accessed by clicking on the "Explore Our Cancer Datasets" button on the left side of the home page.

On the right side of the home page are a human anatomical outline and a bar graph. Choosing a site on the outline or graph will lead the user to the Analysis Center and automatically create a custom cohort consisting of cases corresponding to that site.

[![Home Page](images/HomePage.png)](images/HomePage.png "Click to see the full image.")

## Analysis Center

The Analysis Center can be accessed by clicking on the corresponding link in the GDC Data Portal header, on the "Explore Our Cancer Datasets" button on the home page, or on one of the sites in the human anatomical outline or bar graph.

The Analysis Center has the following sections:

* **Main Toolbar**: manage and create custom cohorts
* **Query Expressions**: displays the filters applied to the current cohort
* **Analysis Tools**: all analysis tools available are located in the Analysis Center as individual cards. When individual analysis tools are launched, they are displayed in this section of the [Analysis Center](analysis_center.md).

### Main Toolbar ##

By default, the main toolbar is always visible in the Analysis Center and Tools. Users can use the main toolbar to view information and perform a number of actions on their cohorts.

[![Main Cohort Toolbar](images/MainCohortToolbar.png)](images/MainCohortToolbar.png "Main Cohort Toolbar. Click to see the full image.")

The name of the current cohort is displayed in a field on the left. Previously created cohorts can be accessed by choosing this field and selecting their names from the dropdown menu.

The main toolbar also contains a set of buttons that are used to manage or create new cohorts. To the left of the cohort name is the "Discard Changes" button, which discards unsaved changes that have been made to the current cohort.

To the right of the cohort name are the following buttons:

* **Save Cohort:** Two options for saving cohorts are available in the dropdown menu. Select the "Save" option to save the active cohort and any changes made to it. Select the "Save As" option to save the active cohort, along with any changes made to it, as a new cohort. Cohorts with unsaved changes have a yellow exclamation mark icon displayed next to their names. Custom cohorts that are saved should persist through releases and continue to be accessible if the same browser is used. When the GDC releases new data, saved cohorts will be updated to include the newly released cases matching the filters applied to the cohort. **It is recommended that users export and securely store any cohort that cannot be easily recreated in case the browser session is cleared.**
* **Create New Unsaved Cohort:** Adds a new unsaved cohort with all the cases in the GDC and changes the active cohort to this new cohort
* **Delete Cohort:** Deletes the current active cohort.  This action cannot be undone.
* **Import New Cohort:** Imports a set of cases as a cohort. These can be imported as a plain text list of UUIDs or submitter_ids (barcodes).
* **Export Cohort:**  Exports the active cohort to a file. A cohort will be exported as a list of UUIDs. Exporting a cohort allows users to obtain a static list of the cases which are in their cohort at the time of export.

Two other buttons are located on the far right of the toolbar:

* __Expand/Collapse:__ Displays the number of cases associated with the current cohort. Displays summary charts of the current cohort, as well as a table of the cases in the current cohort. The Summary View and Table View buttons can be used to toggle between a display of the summary charts and the table.
* __Pin/Unpin Cohort Bar:__ Toggles between pinning the main toolbar to the top of the Analysis Center so that it is always in view, and unpinning it from the top of the Analysis Center.

#### Cohort Summary Charts

Cohort summary charts display graphics that show the number of cases with each value of a set of commonly used properties. The following buttons are available at the top of the summary charts:

* __Files:__  Displays the number of files associated with the active cohort and provides the ability to add these files to the cart, download a manifest, or download metadata associated with the files
* __Custom Filters:__ Allows for the cohort to be filtered by a custom set of cases, mutations, or genes
* __Biospecimen/Clinical:__ Downloads the biospecimen/clinical metadata for the cohort in JSON or TSV format

In the default view, the number of cases for the top five most common values are displayed. Other values can be searched by choosing the magnifying glass button at the top right of each card.

[![Summary Charts](images/SummaryCharts.png)](images/SummaryCharts.png "Cohort Summary Charts. Click to see the full image.")

The middle button at the top right of each card displays the selection view. The selection view displays the same values as the default view as a number instead of a graph. In addition, each value can be selected to filter the active cohort for cases with a specific set of values.

#### Cohort Case Table

The case table displays a list of cases in the active cohort along with associated metadata.  It also allows for each case to be selected with a checkbox for saving a new cohort or exporting metadata. All cases on the current page of the table can be selected at the same time by using the checkbox in the header. Note that expanding the case table, with the cohort bar pinned, may obscure the view of the analysis center or analysis tools.

[![Cohort Table](images/CohortTable.png)](images/CohortTable.png "Cohort Summary Table. Click to see the full image.")


The top right of the case table features a search function that can be used to query specific cases. The following buttons are available at the top left of the case table:

* __Save New Cohort:__  Allows for a new cohort to be created based on the selected cases from the table. The new cohort can comprise only the selected cases, the selected cases added to the active cohort, or the selected cases subtracted from the active cohort.
* __Biospecimen/Clinical:__ Downloads the biospecimen/clinical metadata for the cohort in JSON or TSV format. When no cases are selected, the metadata pertains to the entire cohort. When cases are selected, the metadata pertains to only the selected cases.
* __JSON/TSV:__ Downloads the information in the case table in JSON or TSV format

The case summary panel can be collapsed by selecting the 'Collapse' button that replaces the 'Expand' button.

### Case Summary Page

Users can launch the Case Summary Page by clicking a Case ID in the Cohort Case Table. The Case Summary Page displays case details including the project and disease information, data files that are available for that case, and the experimental strategies employed. A button in the top-left corner of the page allows the user to add all files associated with the case to the file cart.

[![Case Summary Page](images/CaseSummaryPage.png)](images/CaseSummaryPage.png "Click to see the full image.")

### Import New Cohort

The Import New Cohort button in the main toolbar allows for a set of cases to be imported. These can be entered directly into the text box as a plain text list of UUIDs or submitter_ids (barcodes) or imported as a CSV, TSV, or TXT file. Users can hover over the orange (i) to verify accepted case identifiers, delimiters, and file formats.

[![Import New Cohort](images/ImportNewCohort.png)](images/ImportNewCohort.png "Click to see the full image.")

Clicking the `Submit` button will prompt users to name and save their new cohort, after which it will be made the active cohort.


### Query Expressions
The query expressions section displays information about the filters applied to the current cohort and allows convenient operations to be performed on those filters.

[![Query Expressions](images/QueryExpressions.png)](images/QueryExpressions.png "Cohort Query Expressions. Click to see the full image.")


In the top-left corner of this section is the name of the current cohort. To its right is a "Clear All" option, which will remove all filtering applied on the current cohort.

On the top-right corner of this section are the following two buttons:

* **Collapse/Expand Selected Values**: by default, a full list of all the values that have been selected for each property is displayed. This button allows the user to switch from this default expanded view to a minimized view, which only displays the number of values selected for each property.
* **Collapse/Expand Filters Section**: by default, a maximum of three rows will be displayed at a time for the filters selected for the current cohort. This button allows the user to switch from displaying a maximum of three rows for the selected filters to displaying an unlimited number of rows. This button is only enabled if the display of the selected filters for the current cohort exceeds three rows.

The main area of the query expressions section displays the filters applied to the active cohort. Individual values can be removed by clicking on them. Properties can be removed by clicking on the "X" to the extreme right of each property group of values.

If desired, selected values can be collapsed by clicking on the left arrow on the left of the values. When collapsed, values can be expanded again by clicking on the right arrow.

[![Expanded Enums](images/ExpandedEnums.png)](images/ExpandedEnums.png "Expand values for Query Expression. Click to see the full image.")

[![Collapsed Enums](images/CollapsedEnums.png)](images/CollapsedEnums.png "Collapse values for Query Expression. Click to see the full image.")


### Analysis Center Tools

Available tools are displayed under the Query Expression section of the Analysis Center.

[![Analysis Center Tools](images/AnalysisCenterTools.png)](images/AnalysisCenterTools.png "Click to see the full image.")

Each tool is showcased within a 'card', which can be launched using the teal 'Play' button.

## Cohort Builder and Cohort Analysis ##

To build and analyze a cohort of interest using an analysis tool in the Analysis Center:

1. Choose the Cohort Builder icon on either the GDC Data Portal header, or click on the Cohort Builder card in the Analysis Center.  The [Cohort Builder](cohort_builder.md) will appear on the screen.
2. Create a custom cohort based on filters available in the Cohort Builder

[![Cohort Builder](images/CohortBuilderQuickStart.png)](images/CohortBuilderQuickStart.png "Image of the GDC Cohort Builder Tool. Click to see the full image.")

3. Either choose the Analysis Center icon on the GDC Data Portal header, or click on the "X" on the left of the Cohort Builder header. All the tools in the [Analysis Center](analysis_center.md) will be displayed on the screen.
4. Choose an analysis tool from the list of tools in the Analysis Center to perform an analysis of a cohort

## Manage Sets ##

The `Manage Sets` button at the top of the GDC Portal stores sets of genes or mutations of interest.  On this page, users can review the sets that have been saved as well as upload new sets and delete existing sets.

[![Manage Sets](images/ManageSets.png)](images/ManageSets.png "Click to see the full image.")

### Upload Sets

Clicking the `Create Set` button shows options for creating Gene or Mutation sets.  

[![Create Set](images/CreateSet.png)](images/CreateSet.png "Click to see the full image.")

Upon clicking one of the menu items, users are shown a dialog where they can enter unique identifiers (i.e. gene symbols, mutation UUIDs, etc.) that describe the set.

[![Upload Gene Set](images/UploadGeneSet.png)](images/UploadGeneSet.png "Click to see the full image.")

Clicking the `Submit` button will add the set of items to the list of sets on the Manage Sets page.

[![Manage Saved Sets](images/ManageSavedSets.png)](images/ManageSavedSets.png "Click to see the full image.")

### Export Sets

Users can export selected sets on this page by first clicking the checkboxes next to each set, then clicking the `Export selected` button at the top of the table.

[![Export Sets](images/ExportSets.png)](images/ExportSets.png "Click to see the full image.")

A text file containing the ID of each gene or UUID of each mutation is downloaded after clicking this button.

### Review Sets

There are a few buttons in the list of sets that allows a user to get further information about each one.

* __# Items__: Clicking the button under the # Items column launches a table with Gene ID and Symbol for gene sets or Mutation ID and Consequence for mutation sets

* __Delete/Download__: To the right of the # Items column are buttons that will delete the set or download the list as a TSV file


# Tutorial Videos

## Overview

The [GDC Data Portal 2.0](https://portal.gdc.cancer.gov) centers around the idea of building cohorts, or groups of cases, before analyzing or downloading data.

In this GDC 2.0 Video Tutorial, learn how to:

* Build a cohort
* Analyze a cohort using GDC analysis tools
* Download data associated with a cohort
* View projects and available data in the GDC, and filter to create custom cohorts

GDC 2.0 Videos are available in the [NCI GDC YouTube Playlist](https://www.youtube.com/playlist?list=PLaXJeOudgf60aE6tqrZcIhm9B4Ivxar4a). For additional details, please see the [GDC 2.0 User's Guide](getting_started.md).

[![GDC 2.0 Workflow Flowchart](images/v2_workflow.png)](images/v2_workflow.png "Click to see the full image.")

## Build Cohort

<table>
<tr>
<td>
Cohorts are created using the <b>Cohort Builder</b>, which allows users to specify cases with custom filters for things like disease characteristics, patient demographics, data type, and more.
<br>
<br>
In this video, learn how to use the <b>Cohort Builder</b> to build a cohort of patients with lung cancer, that are over the age of 50, and have harmonized RNA-Seq data.
</td>

<td><iframe width="400" height="300" src="https://www.youtube.com/embed/ppEYuxbarGw" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen> </iframe>

</td></tr>
</table>



## Analyze Cohort

### Analysis Center

<table>
<tr>
<td><iframe width="400" height="300" src="https://www.youtube.com/embed/N2my3pTtSiU" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen> </iframe>
</td>
<td>
Once a cohort is created, the cohort can be analyzed using an analysis tool in the <b>Analysis Center</b>.
<br>
<br>

In this video, learn how to analyze a cohort in the <b>Analysis Center</b>. Instructional videos are available for each analysis tool.

</td>
</tr>
</table>

### Analysis Tools

A cohort can be analyzed by selecting an analysis tool in the Analysis Center. The GDC provides analysis tools for performing both gene-level variant analysis and clinical data analysis. The table below identifies and describes each tool, and provides video instruction.

<table>
<tr><td><b>Tool</b></td> <td><b>Description</b></td><td><b>Video</b></td></tr>

<tr>
<td><b>Clinical Data Analysis</b></td>
<td>Use clinical variables to perform basic statistical analysis of your cohort</td>
<td><iframe width="240" height="160" src="https://www.youtube.com/embed/0DuI4NEQ6CU" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen > </iframe>
</td>
</tr>
<tr>
<td><b>Cohort Comparison</b></td>
<td>Display the survival analysis of your cohorts and compare characteristics such as gender, vital status and age at diagnosis</td>
<td><iframe width="240" height="160" src="https://www.youtube.com/embed/NT7-TB78VjA" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen > </iframe>
</td>
</tr>
<tr>
<td><b>Cohort Level MAF</b></td>
<td>Download an aggregated set of variants in MAF format based on the cases in your active cohort.</td>
<td><iframe width="240" height="160" src="https://www.youtube.com/embed/4megNwrPsbs" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen > </iframe>
</td>
</tr>
<tr>
<td><b>Gene Expression Clustering</b></td>
<td>Display gene expression visualization </td>
<td><iframe width="240" height="160" src="https://www.youtube.com/embed/f58U6hL4Wok" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen > </iframe>
</td>
</tr>
<tr>
<td><b>Mutation Frequency</b></td>
<td>Visualize most frequently mutated genes and somatic mutations</td>
<td><iframe width="240" height="160" src="https://www.youtube.com/embed/dbbuk-zhmvc" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen > </iframe>
</td>
</tr>
<tr>
<td><b>OncoMatrix</b></td>
<td>Visualize the top most mutated cases and genes affected by high impact mutations in your cohort</td>
<td><iframe width="240" height="160" src="https://www.youtube.com/embed/ZgSeJCqLmzY" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen > </iframe>
</td>
</tr>
<tr>
<td><b>ProteinPaint</b></td>
<td>Visualize mutations in protein-coding genes by consequence type and protein domain</td>
<td><iframe width="240" height="160" src="https://www.youtube.com/embed/5AXvhtFd_Es" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen > </iframe>
</td>
</tr>
<tr>
<td><b>Sequence Reads</b></td>
<td>Visualize sequencing reads for a given gene, position, SNP, or variant</td>
<td><iframe width="240" height="160" src="https://www.youtube.com/embed/PkQ6cEzRCds" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen > </iframe>
</td>
</tr>
<tr>
<td><b>Set Operations</b></td>
<td>Display a Venn diagram and compare/contrast your cohorts or sets of the same type</td>
<td><iframe width="240" height="160" src="https://www.youtube.com/embed/QHJwyL_pLak" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen > </iframe>
</td>
</tr>
<tr>
<td><b>BAM Slicing</b></td>
<td>Slice a specific region, gene, or chromosome from a GDC-harmonized BAM file.</td>
<td><iframe width="240" height="160" src="https://www.youtube.com/embed/Nd9Ex9s_prk" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen > </iframe>
</td>
</tr>
</table>

## Download Cohort Data

<table>
<tr>
<td>
Genomic, clinical, and biospecimen data associated with a cohort can be downloaded using the <b>Repository</b>.
<br>
<br>
In this video, learn how to download data associated with a cohort in the <b>Repository</b>.
</td>
<td><iframe width="400" height="300" src="https://www.youtube.com/embed/edipjoF1Tyk" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen> </iframe>
</td>
</tr>
</table>

## View Projects

<table>
<tr>
<td><iframe width="400" height="300" src="https://www.youtube.com/embed/rKn2WJuA850" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen> </iframe>
</td>
<td>

GDC data is organized by projects within a larger cancer research program. Projects are generally composed of a particular type of cancer, such as the low-grade glioma (LGG) and sarcoma (SARC) projects within the TCGA program.
<br>
In this video, learn how to use The GDC Data Portal to access project-level information via the Projects tool and Project Summary Page and navigate and filter data to create custom cohorts.
</td>
</tr>
</table>


# Mutation Frequency

The Mutation Frequency tool visualizes the most frequently mutated genes and the most frequent somatic mutations for the active cohort. To launch the Mutation Frequency tool, click on its card from the Tools section of the Analysis Center.

[![Mutation Frequency](images/MutationFrequency.png)](images/MutationFrequency.png "Click to see the full image.")

This tool includes the following visualizations:

## Mutated Genes Histogram

The most frequently mutated genes are represented with a histogram that shows the percentage of cases affected within the active cohort. The histogram can be downloaded as an image (SVG/PNG) or raw data (JSON) using the button at the top right of the graphic.

[![Mutation Table](images/MutHisto.png)](images/MutHisto.png "Click to see the full image.")

## Survival Plot for Mutated Genes and Mutations

The mutation frequency survival plot is represented with two Kaplan-Meier curves based on cases with and without a specific mutation or mutated gene. Cases for both curves can be further filtered using the various filters available in the left panel of the Mutation Frequency tool. For example, selecting "high" for the VEP impact filter will limit the cases in both curves to those whose mutations have a high VEP impact.

The Log-Rank Test p-value is also displayed here. The survival plot can be downloaded as an image (SVG/PNG) or raw data (JSON/TSV) and the view can be reset using the buttons at the top right of the graphic.

[![Mutation Table](images/MutSurvival.png)](images/MutSurvival.png "Click to see the full image.")

## Genes/Mutations Table

The genes/mutations table displays the most frequently mutated genes or the most frequent mutations in the active cohort by percent frequency in descending order. Additional columns show CNV information as well as the number of affected cases. The "Cohort" toggle can be used to filter the current cohort by a specific gene or mutation, and the "Survival" button allows the user to modify the survival plot. The red arrow button allows for the percentage of affected cases to be displayed on a project-level. The data displayed in the table can be exported as a TSV using the button at the top left of the table. Additional cohorts can be created using buttons located within the table.

[![Mutation Table](images/MutationTableWithButtons.png)](images/MutationTableWithButtons.png "Click to see the full image.")

The table can be searched using the field at the top right of the table.

[![Mutation Search](images/MutSearchWithButtons.png)](images/MutSearchWithButtons.png "Click to see the full image.")

Additionally, clicking the button in the "# Mutations" column within the genes table will automatically apply a search for the corresponding gene in the mutations table. This is a convenient way to view the specific mutations in a given gene.

### Gene and Mutation Summary Pages

Users can click on the Symbol links in the Genes Table and the DNA Change links in the Mutations Table to view the Gene and Mutation Summary Pages, respectively. These pages display information about specific genes and mutations, along with visualizations and data showcasing the relationship between themselves, the projects, and cases within the GDC. The gene and mutation data that is visualized on these pages are produced from the Open-Access MAF files available for download on the GDC Portal.

Gene Summary Pages describe each gene with mutation data and provide results related to the analyses that are performed on these genes.  

__Summary__

The summary section of the gene page contains the following information:

[![Gene Summary Page](images/GeneSummaryPage.png)](images/GeneSummaryPage.png "Click to see the full image.")

* __Symbol:__ The gene symbol
* __Name:__ Full name of the gene
* __Synonyms:__ Synonyms of the gene name or symbol, if available
* __Type:__ A broad classification of the gene
* __Location:__ The chromosome on which the gene is located and its coordinates
* __Strand:__ If the gene is located on the forward (+) or reverse (-) strand
* __Description:__ A description of gene function and downstream consequences of gene alteration
* __Annotation:__ A notation/link that states whether the gene is part of [The Cancer Gene Census](http://cancer.sanger.ac.uk/census/)

__External References__

A list with links that lead to external databases with additional information about each gene is displayed here. These external databases include: [Entrez](https://www.ncbi.nlm.nih.gov/gquery/), [Uniprot](http://www.uniprot.org/), [Hugo Gene Nomenclature Committee](http://www.genenames.org/), [Online Mendelian Inheritance in Man](https://www.omim.org/), [Ensembl](http://may2015.archive.ensembl.org/index.html), and [CIViC](https://civicdb.org/home).

__Cancer Distribution__

A table and two bar graphs (one for mutations, one for CNV events) show how many cases are affected by mutations and CNV events within the gene as a ratio and percentage. Each row/bar represents the number of cases for each project.  The final column in the table lists the number of unique mutations observed on the gene for each project.

[![Gene Summary Page Cancer Distribution](images/GeneSummaryPageCancerDistribution.png)](images/GeneSummaryPageCancerDistribution.png "Click to see the full image.")

__Most Frequent Mutations__

The 20 most frequent mutations in the gene are displayed as a bar graph that indicates the number of cases that share each mutation.  

[![Gene Summary Page Most Frequent Somatic Mutations](images/MostFreqSomaticMutations.png)](images/MostFreqSomaticMutations.png "Click to see the full image.")

A table is displayed below that lists information about each mutation including:

* __DNA Change:__ The chromosome and starting coordinates of the mutation are displayed along with the nucleotide differences between the reference and tumor allele
* __Protein Change:__ The gene and amino acid change
* __Type:__ A general classification of the mutation
* __Consequences:__ The effects the mutation has on the gene coding for a protein (i.e. synonymous, missense, non-coding transcript)
* __# Affected Cases in Gene:__ The number of affected cases, expressed as number across all mutations within the Gene
* __# Affected Cases Across GDC:__ The number of affected cases, expressed as number across all projects. Choosing the arrow next to the percentage will expand the selection with a breakdown of each affected project.
* __Impact:__ A subjective classification of the severity of the variant consequence. This is determined using [Ensembl VEP](https://useast.ensembl.org/info/docs/tools/vep/index.html), [PolyPhen](http://genetics.bwh.harvard.edu/wiki/!pph2/about), and [SIFT](http://sift.jcvi.org/). The categories are outlined [here](https://docs.gdc.cancer.gov/Data/File_Formats/MAF_Format/#impact-categories).

*Note: The Mutation UUID can be displayed in this table by selecting it from the Customize Columns button, represented by three parallel lines*

The Mutation Summary Page contains information about one somatic mutation and how it affects the associated gene. Each mutation is identified by its chromosomal position and nucleotide-level change.

__Summary__

[![Mutation Summary Page](images/MutationSummaryPage.png)](images/MutationSummaryPage.png "Click to see the full image.")

* __UUID:__ A unique identifier (UUID) for this mutation
* __DNA Change:__ Denotes the chromosome number, position, and nucleotide change of the mutation
* __Type:__ A broad categorization of the mutation
* __Reference Genome Assembly:__ The reference genome in which the chromosomal position refers to
* __Allele in the Reference Assembly:__ The nucleotide(s) that compose the site in the reference assembly
* __Functional Impact:__ A subjective classification of the severity of the variant consequence.

__External References__

  A separate panel contains links to databases that contain information about the specific mutation. These include [dbSNP](https://www.ncbi.nlm.nih.gov/projects/SNP/), [COSMIC](http://cancer.sanger.ac.uk/cosmic), and [CIViC](https://civicdb.org/home).

__Consequences__

The consequences of the mutation are displayed in a table.  The set of consequence terms, defined by the [Sequence Ontology](http://www.sequenceontology.org).

[![Mutation Summary Page Consequences](images/MutationSummaryPageConsequences.png)](images/MutationSummaryPageConsequences.png "Click to see the full image.")

The fields that describe each consequence are listed below:

  * __Gene:__ The symbol for the affected gene
  * __AA Change:__ Details on the amino acid change, including compounds and position, if applicable
  * __Consequences:__ The biological consequence of each mutation
  * __Coding DNA Change:__ The specific nucleotide change and position of the mutation within the gene
  * __Impact:__ VEP, SIFT, and/or PolyPhen Impact ratings
  * __Gene Strand:__ If the gene is located on the forward (+) or reverse (-) strand
  * __Transcript:__ The transcript(s) affected by the mutation. Each contains a link to the [Ensembl](https://www.ensembl.org) entry for the transcript

__Cancer Distribution__

A table and bar graph shows how many cases are affected by the particular mutation. Each row/bar represents the number of cases for each project.

[![Mutation Summary Page Cancer Distribution](images/MutationSummaryPageCancerDistribution.png)](images/MutationSummaryPageCancerDistribution.png "Click to see the full image.")

The table contains the following fields:

  * __Project__: The ID for a specific project
  * __Disease Type__: The disease associated with the project
  * __Primary Site__: The anatomical site affected by the disease
  * __# SSM Affected Cases__: The number of affected cases and total number of cases displayed as a fraction and percentage

## Custom Gene and Mutation Filters

[![Custom Gene/Mutation Filters](images/CustomGeneMutationFilters.png)](images/CustomGeneMutationFilters.png "Click to see the full image.")

The `+ Add Custom Gene Filters` button in the left panel of the Mutation Frequency tool allows users to filter mutation frequency by genes. Users can enter unique identifiers (i.e. gene symbols, gene IDs, etc.) directly into the text box as a plain text list or upload a list of unique identifiers as a CSV, TSV, or TXT file. Users can hover over the orange (i) to verify accepted gene identifiers, delimiters, and file formats.

[![Custom Gene Filters](images/CustomGeneFilters.png)](images/CustomGeneFilters.png "Click to see the full image.")

The `+ Add Custom Mutation Filters` button allows users to filter mutation frequency by mutations. Users can enter unique identifiers (i.e. mutation UUIDs, etc.) directly into the text box as a plain text list or upload a list of unique identifiers as a CSV, TSV, or TXT file. Users can hover over the orange (i) to verify accepted mutation identifiers, delimiters, and file formats.

[![Custom Mutation Filters](images/CustomMutationFilters.png)](images/CustomMutationFilters.png "Click to see the full image.")

## Mutation Frequency Facet Filters

A set of frequently-used properties are available to filter genes and mutations in the left panel of the Mutation Frequency tool. Using each of these filters will dynamically change the graphics and table to represent the filtered data.

* __Biotype:__  Classification of the type of gene according to Ensembl. The biotypes can be grouped into protein coding, pseudogene, long noncoding and short noncoding. Examples of biotypes in each group are as follows:
    * __Protein coding:__ IGC gene, IGD gene, IG gene, IGJ gene, IGLV gene, IGM gene, IGV gene, IGZ gene, nonsense mediated decay, nontranslating CDS, non stop decay, polymorphic pseudogene, TRC gene, TRD gene, TRJ gene, TRV gene.
    * __Pseudogene:__ Disrupted domain, IGC pseudogene, IGJ pseudogene, IG pseudogene, IGV pseudogene, processed pseudogene, transcribed processed pseudogene, transcribed unitary pseudogene, transcribed unprocessed pseudogene, translated processed pseudogene, translated unprocessed pseudogene, TRJ pseudogene, TRV pseudogene, unprocessed pseudogene.
    * __Long noncoding:__ 3 prime overlapping ncrna, ambiguous orf, antisense, antisense RNA, lincRNA, macro lincRNA, ncrna host, processed transcript, sense intronic, sense overlapping.
    * __Short noncoding:__ miRNA, miRNA pseudogene, miscRNA, miscRNA pseudogene, Mt rRNA, Mt tRNA, rRNA, scRNA, snlRNA, snoRNA, snRNA, tRNA, tRNA pseudogene, vaultRNA.
* __Is Cancer Gene Census:__ Whether or not a gene is part of [The Cancer Gene Census](http://cancer.sanger.ac.uk/census/). Note that this is switched on as a default.
* __Impact:__ A subjective classification of the severity of the variant consequence. These scores are determined using the following three tools:
    * __[VEP](http://useast.ensembl.org/info/genome/variation/prediction/index.html):__
        * __HIGH (H):__ The variant is assumed to have high (disruptive) impact in the protein, probably causing protein truncation, loss of function or triggering nonsense mediated decay
        * __MODERATE (M):__ A non-disruptive variant that might change protein effectiveness
        * __LOW (L):__ Assumed to be mostly harmless or unlikely to change protein behavior
        * __MODIFIER (MO):__ Usually non-coding variants or variants affecting non-coding genes, where predictions are difficult or there is no evidence of impact
    * __[PolyPhen](http://genetics.bwh.harvard.edu/pph/):__
        * __probably damaging (PR):__ It is with high confidence supposed to affect protein function or structure
        * __possibly damaging (PO):__ It is supposed to affect protein function or structure
        * __benign (BE):__ Most likely lacking any phenotypic effect
        * __unknown (UN):__ When in some rare cases, the lack of data does not allow PolyPhen to make a prediction
    * __[SIFT](http://sift.jcvi.org/):__
        * __tolerated:__ Not likely to have a phenotypic effect
        * __tolerated_low_confidence:__ More likely to have a phenotypic effect than 'tolerated'
        * __deleterious:__ Likely to have a phenotypic effect
        * __deleterious_low_confidence:__ Less likely to have a phenotypic effect than 'deleterious'
* __Consequence Type:__  Consequence type of this variation; [sequence ontology](http://www.sequenceontology.org/) terms
* __Type:__ A general classification of the mutation

### Saving a Gene or Mutation Set

After filtration, a set of genes or mutations can be saved by choosing the "Save/Edit Gene Set" or "Save/Edit Mutation Set" button at the top left of the table.  


# Quick Start Page

The purpose of this guide is to quickly introduce researchers to the GDC Data Portal. This is not a comprehensive overview of the Data Portal and may not contain details for your specific use-case. Please see the rest of the Data Portal documentation pages for information about specific tools.

Start at [https://portal.gdc.cancer.gov/](https://portal.gdc.cancer.gov/).

### Building and Analyzing a Cohort


__Step 1:__ Go to the Cohort Builder at the top left of the Data Portal.

??? note "Click to Expand/Collapse Animation"

    ![Animation of Browsing to the Cohort Builder](images/agifs/GotoCB-animation.gif)

__Step 2:__ Use the filters in the Cohort Builder to filter down the full set of GDC cases to a subset you are interested in. Filter categories can be selected in the left panel.

??? note "Click to Expand/Collapse Animation" 

    ![Animation of Filtering Down Cohort](images/agifs/Filter-animation.gif)

__Step 3:__ Save your cohort by clicking the "Save" icon in the cohort bar, choosing "Save", and naming your cohort when prompted.

??? note "Click to Expand/Collapse Animation"

    ![Animation of Saving a Cohort](images/agifs/CohortSave-animation.gif)

__Step 4:__ The cohort you created is now your active cohort.  Go to the Analysis Center at the top left of the Data Portal.  Choose the tool you would like to use from the list.  The analysis will apply to the data from your cohort.

??? note "Click to Expand/Collapse Animation"

    ![Animation of Browsing to the Analysis Center](images/agifs/GotoAC-animation.gif)

### Downloading Files

__Step 1:__ Select your cohort of interest, [create one](#building-and-analyzing-a-cohort) if you have not already.

??? note "Click to Expand/Collapse Animation"

    ![Animation of Selecting an Active Cohort](images/agifs/ChangeCohort-animation.gif)

---

__Step 2:__ Go to the Repository tool from the Analysis Center or the top left of the portal.

??? note "Click to Expand/Collapse Animation"

    ![Animation of Browsing to the Repository](images/agifs/GoToRepo-animation.gif)

---

__Step 3:__ The Repository tool will display files that are associated with the cases in your active cohort. Narrow down your file selection by filtering them using the panel on the left.

??? note "Click to Expand/Collapse Animation"

    ![Animation of Filtering the Repository](images/agifs/SelectFiltersRepo-animation.gif)

---

__Step 4:__ When you are done filtering, add the files to your cart.  Then go to the cart.

??? note "Click to Expand/Collapse Animation"

    ![Animation of Adding Files to and Browsing to the Cart](images/agifs/AddFilesCart-animation.gif)

---

__Step 5:__ The files in the cart can be downloaded directly from the browser or a manifest can be downloaded and passed to the [GDC Data Transfer Tool](Data_Transfer_Tool/Users_Guide/Getting_Started.md).

??? note "Click to Expand/Collapse Animation"

    ![Animation of Selecting the Cart or Manifest Download In the Cart Page](images/agifs/DownloadCart-animation.gif)

### Viewing Mutations

__Step 1:__ Select your cohort of interest or [create one](#building-and-analyzing-a-cohort) if you have not already.

??? note "Click to Expand/Collapse Animation"

    ![Animation of Selecting an Active Cohort](images/agifs/ChangeCohort-animation.gif)

---

__Step 2:__ Launch the Mutation Frequency tool from the Analysis Center.

??? note "Click to Expand/Collapse Animation"

    ![Animation of Launching the Mutation Frequency Tool](images/agifs/GoToMutFreq-animation.gif)

---

__Step 3:__ The Mutation Frequency tool visualizes the most frequently mutated genes and the most frequent somatic mutations for the active cohort. Narrow down your results by filtering them using the panel on the left.

??? note "Click to Expand/Collapse Animation"

    ![Animation of Filtering Mutation Frequency](images/agifs/SelectFiltersMF-animation.gif)

---

__Step 4:__ To view the most frequent somatic mutations, navigate to the Mutations tab.

??? note "Click to Expand/Collapse Animation"

    ![Animation of Toggling to the Mutations Table](images/agifs/ViewMutationsTable-animation.gif)

---

__Step 5:__ The mutation table can be downloaded by clicking on the TSV button at the top of the table.

??? note "Click to Expand/Collapse Animation"

    ![Animation of Downloading the Mutations Table](images/agifs/DownloadMutationsTable-animation.gif)


# Data Portal Release Notes

## Release 0.3.24-spr5

* __GDC Product__: GDC Data Portal
* __Release Date__: April 18, 2016

### New Features and Changes

*   This is a bug-fixing release, no new features were added.

### Bugs Fixed Since Last Release

*   Tables and Export
    *   Table export return correct content (XML, TSV) although file extension is incorrect (always JSON) <!-- PGDC-1986 -->
    *   Nested values are not exported properly <!-- PGDC-910 SV-13 -->
    *   Project table summary chart does not display colors with Internet Explorer 11 <!-- PGDC-1953 -->
*   Cart and Download
    *   When saving a file for download, the download pop-up indicates "from:blob:" <!-- PGDC-115 -->
    *   User with update only can also download files also role should not allow <!-- PGDC-1782 -->
    *   User can add more files to the cart than supported if adding files one by one <!-- PGDC-1957 -->
    *   In very specific situations, the cart can display inconsistent data <!-- PGDC-1028 -->   
    *   In some situations the cart will display the authentication window for authenticated users when trying to download <!-- PGDC-1959 -->  
*   Search
    *   In the data page, some pie chart titles are too long <!-- PGDC-914 -->
    *   Facets displaying histogram do not display a mouseover tooltip if the value is very low <!-- PGDC-913 -->
*   Layout, Browser specific and Accessibility
    *   No favicon is displayed on Internet Explorer <!-- PGDC-1948 -->
    *   Download cart button does not function properly with Safari 9.0.3 <!-- PGDC-1978 -->
    *   Layout issue when browser is reduced to a small window size <!-- PGDC-16 -->


### Known Issues and Workarounds

*   General
    *   If a user has previously logged into the Portal and left a session without logging out, if the user returns to the Portal after the user's sessionID expires, it looks as if the user is still authenticated. The user cannot download the token and gets an error message that would not close. The user should clear the cache to properly log out. 
*   Tables and Export
    *   Table sorting icon does not include numbers <!-- PGDC-35 -->
    *   Restore default table column arrangement does not restore to the default but it restores to the previous state <!-- PGDC-1769 -->
*   Cart and Download
    *   Make the cart limit warning message more explanatory <!-- PGDC-1952 -->   
    *   In some situations, adding filtered files to the cart might fail <!-- PGDC-1981 -->   
*   Layout, Browser specific and Accessibility
    *   When disabling CSS, footer elements are displayed out of order <!-- PGDC-1972 -->
    *   If javascript is disabled html tags are displayed in the warning message <!-- PGDC-1835 -->
    *   Layout issues when using the browser zoom in function on tables <!-- PGDC-116 -->
    *   Cart download spinner not showing at the proper place <!-- PGDC-2056 -->
    *   Not all facets are expanded by default when loading the app <!-- PGDC-2061 -->
*   Non UI-related tickets
    *   Investigations were done on this issue "Authenticated user with access to TARGET projects cannot download Target DCC files. They will get a pop-up message." was related  to ACL permissions (not UI related).
    *   Investigations were done on this issue "Missing pathology files and slide images for some TCGA datasets since they are not connected to the biospecimen chain yet" was related to Data not imported (not UI related).
    *   Associated entities is empty for some files (note: this is a data issue) <!-- PGDC-1887 -->


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).

## Release 0.3.24.2

* __GDC Product__: GDC Data Portal
* __Release Date__: April 4, 2016

### New Features and Changes

*   Implemented a homepage to provide high-level information about the GDC Data Portal
*   Created the GDC Legacy Archive to provide access to legacy data
*   Updated the GDC Data Portal to support harmonized data
*   Updated the login process to be consistent with the GDC Data Submission Portal login
*   Added the ability to export clinical and biospecimen data from Project and Case entity pages (replacing download clinical XML)
*   Improved GDC Data Portal search capabilities:
    *   Added customized facets, expanding the GDC Data Portal search capabilities to non-default facets
    *   Updated the list of fields suggested in the Advanced Search and Custom Facets
    *   Added prefix search on case submitter ID on the Facet panel
    *   Improved operator auto-suggestion based on field type in the Advanced Search
*   Improved the download experience by letting the browser handle the download (allow canceling)
*   Enhanced Cart capabilities:
    *   Added the ability to download metadata files (SRA XML and MAGE-TAB)
    *   Added the ability to download biospecimen and clinical data
*   Improved the Biospecimen tree in the Case entity page (users can search for ID and expand the tree)
*   Improved support for Section 508 standards and the Internet Explorer browser
*   Improved the table view in the Projects page (Added a "Total" row with links to the search page)
*   Improved the BAM Slicing UI (added an example and the ability to use tabulations for BED format in the text box)

### Bugs Fixed Since Last Release

*   Disease Type does not auto filter (Data Facet)
*   When exporting the annotation table to a file, the create date field is not displayed in the same format than the UI
*   This ticket is not applicable anymore: "In the Case Entity Page, the "Download Clinical XML" button is not applicable to TARGET projects. The label is incorrect but the functionality is accurate.". The Case entity page has a standard download clinical feature.


### Known Issues and Workarounds

*   General
    *   If a user has previously logged into the Portal and left a session without logging out, if the user returns to the Portal after the user's sessionID expires, it looks as if the user is still authenticated. The user cannot download the token and gets an error message that would not close. The user should clear the cache to properly log out. 
*   Tables and Export
    *   Table sorting icon does not include numbers <!-- PGDC-35 -->
    *   Table export return correct content (XML, TSV) although file extension is incorrect (always JSON) <!-- PGDC-1986 -->
    *   Nested values are not exported properly <!-- PGDC-910 SV-13 -->
    *   Project table summary chart does not display colors with Internet Explorer 11 <!-- PGDC-1953 -->
    *   Exporting a table containing a very large number of records might trigger an exception (when the server is timing out)<!-- PGDC-1962 -->
    *   Restore default table column arrangement does not restore to the default but it restores to the previous state <!-- PGDC-1769 -->
*   Cart and Download
    *   When saving a file for download, the download pop-up indicates "from:blob:" <!-- PGDC-115 -->
    *   User with update only can also download files also role should not allow <!-- PGDC-1782 -->
    *   User can add more files to the cart than supported if adding files one by one <!-- PGDC-1957 -->
    *   In very specific situations, the cart can display inconsistent data <!-- PGDC-1028 -->   
    *   Make the cart limit warning message more explanatory <!-- PGDC-1952 -->   
    *   In some situations the cart will display the authentication window for authenticated users when trying to download <!-- PGDC-1959 -->   
*   Search
    *   In the data page, some pie chart titles are too long <!-- PGDC-914 -->
    *   Facets displaying histogram do not display a mouseover tooltip if the value is very low <!-- PGDC-913 -->
*   Layout, Browser specific and Accessibility
    *   No favicon is displayed on Internet Explorer <!-- PGDC-1948 -->
    *   When disabling CSS, footer elements are displayed out of order <!-- PGDC-1972 -->
    *   Download cart button does not function properly with Safari 9.0.3 <!-- PGDC-1978 -->
    *   If javascript is disabled html tags are displayed in the warning message <!-- PGDC-1835 -->
    *   Layout issue when browser is reduced to a small window size <!-- PGDC-16 -->
    *   Layout issues when using the browser zoom in function on tables <!-- PGDC-116 -->
*   Non UI-related tickets
    *   Investigations were done on this issue "Authenticated user with access to TARGET projects cannot download Target DCC files. They will get a pop-up message." was related  to ACL permissions (not UI related).
    *   Investigations were done on this issue "Missing pathology files and slide images for some TCGA datasets since they are not connected to the biospecimen chain yet" was related to Data not imported (not UI related).
    *   Associated entities is empty for some files (note: this is a data issue) <!-- PGDC-1887 -->


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).

## Release 0.2.18.3

* __GDC Product__: GDC Data Portal
* __Release Date__: November 23, 2015

### New Features and Changes

* BAM slicing UI in the File Entity page (only available for BAM files with BAI files)

### Bugs Fixed Since Last Release

* Disease Type does not auto filter (Data Facet)

### Known Issues and Workarounds

* Missing pathology files and slide images for some TCGA datasets since they are not connected to the biospecimen chain yet
* In some specific situation (complex queries using must_not term), the advanced search might returns incorrect results
* When exporting the annotation table to a file, the create date field is not displayed in the same format than the UI
* In the advanced search, adding a trailing space to the query causes the UI to display a syntax error although the query is valid
* Checksum missing for MAGE-TAB files
* User can download large files (> 1GB) from the Portal but it fails when extracting the tar.gz file. User should use the GDC Download Transfer Tool for large downloads
* Authenticated user with access to TARGET projects cannot download Target DCC files. They will get a pop-up message.
* In Case Entity Page, the "Download Clinical XML" button is not applicable to TARGET projects. The label is incorrect but the functionality is accurate.
* The advanced search suggests more fields than expected (e.g. "cases.aliquots" should not be displayed). User should remove the field from the query and use the long field name (e.g. "cases.samples.portions.analytes.aliquots")
* If a user has previously logged into the Portal and left a session without logging out, if he comes back to the Portal after his sessionID expires, it looks like the user is still authenticated. The user cannot download the token and gets an error message that would not close. The user should clear the cache to properly log out.
* Sorting on size and removing files from cart does not work in IE 10
* Annotations page not section 508 compliant
* Facets are displayed above the results table when window is small
* Table sorting icon does not include numbers

Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).

## Release 0.2.15-oicr4

* __GDC Product__: GDC Data Portal
* __Release Date__: October 1, 2015

### New Features and Changes

*   n/a

### Bugs Fixed Since Last Release

*   Data access and subtype units are correct in the data download statistics report
*   Improved 508 compliance of the portal (including the data download statistics report)
*   Addressed an issue with the user getting an empty page in specific browsing situations
*   "My projects" filter has been re-enabled for users authenticated with eRACommons but without dbGap authorization.

### Known Issues and Workarounds

*   Missing pathology files and slide images for some TCGA datasets since they are not connected to the biospecimen chain yet
*   In some specific situation (complex queries using must_not term), the advanced search might returns incorrect results
*   When exporting the annotation table to a file, the create date field is not displayed in the same format than the UI
*   In the advanced search, adding a trailing space to the query causes the UI to display a syntax error although the query is valid
*   Checksum missing for MAGE-TAB files
*   User can download large files (> 1GB) from the Portal but it fails when extracting the tar.gz file. User should use the GDC Download Transfer Tool for large downloads
*   Authenticated user with access to TARGET projects cannot download Target DCC files. They will get a pop-up message.
*   In Case Entity Page, the "Download Clinical XML" button is not applicable to TARGET projects. The label is incorrect but the functionality is accurate.
*   The advanced search suggests more fields than expected (e.g. "cases.aliquots" should not be displayed). User should remove the field from the query and use the long field name (e.g. "cases.samples.portions.analytes.aliquots")
*   If a user has previously logged into the Portal and left a session without logging out, if he comes back to the Portal after his sessionID expires, it looks like the user is still authenticated. The user cannot download the token and gets an error message that would not close. The user should clear the cache to properly log out.
*   Sorting on size and removing files from cart does not work in IE 10
*   Annotations page not section 508 compliant
*   Facets are displayed above the results table when window is small
*   Table sorting icon does not include numbers

Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).

## Release 0.2.15-oicr2

* __GDC Product__: GDC Data Portal
* __Release Date__: August 31, 2015

### New Features and Changes

*   Authentication, Authorization & Pop-up messages:
    *   If a user tries to download a related file that is controlled from the File Entity Page, he will get a pop-up message with appropriate guidance ("Login" or "No access to the file")
    *   If a user tries to download a controlled clinical file from the Case page, he will get a pop-up message to indicate that he does not have access to the file
    *   If a user authenticates to the Portal with an eRa account without dbGap authorization, he will get a warning message. Then he will have the choice to logout or to continue browsing the Portal with his eRa account. If he continues browsing the Portal with his eRa account, "My projects" filter will be hidden (temporary solution).
*   If User downloads large files (from the Cart or from the File table), the Portal displays a spinner to indicate the download is in progress
*   Data Download Report does not show the "Data Level" section anymore.

### Bugs Fixed Since Last Release

*   The add to Cart button in File Entity Page changes its display value following a click. User can then click on "Remove from Cart"
*   Total Case value in the Cart matches with the number of cases associated with the files in the Cart
*   When User is authenticated, "My project flag" in Case table indicates that the Cases belongs to his projects
*   In Projects table, if User clicks on the count on Files, it links to the Data page - File table
*   In File Entity Page, if there are no associated cases, it will display the message "No Cases Found." instead of "No Participants Found."

### Known Issues and Workarounds

*   Missing pathology files and slide images for some TCGA datasets since they are not connected to the biospecimen chain yet
*   In some specific situation (complex queries using must_not term), the advanced search might returns incorrect results
*   When exporting the annotation table to a file, the create date field is not displayed in the same format than the UI
*   In the advanced search, adding a trailing space to the query causes the UI to display a syntax error although the query is valid
*   Checksum missing for MAGE-TAB files
*   Data download statistics report is not Section 508 compliant
*   User can download large files (> 1GB) from the Portal but it fails when extracting the tar.gz file. User should use the GDC Download Transfer Tool for large downloads
*   Authenticated user with access to TARGET projects cannot download Target DCC files. They will get a pop-up message.
*   In Case Entity Page, the "Download Clinical XML"button is not applicable to TARGET projects. The label is incorrect but the functionality is accurate.
*   The advanced search suggests more fields than expected (e.g. "cases.aliquots" should not be displayed). User should remove the field from the query and use the long field name (e.g. "cases.samples.portions.analytes.aliquots")
*   Removing a filter after paginating table results could cause no results to be displayed. User should press Clear on the filters and start again.
*   If a user has previously logged into the Portal and left a session without logging out, if he comes back to the Portal after his sessionID expires, it looks like the user is still authenticated. The user cannot download the token and gets an error message that would not close. The user should clear the cache to properly log out.

Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).

## Release 0.2.15-oicr1

* __GDC Product__: GDC Data Portal
* __Release Date__: August 12, 2015

### New Features and Changes

*   Renamed all references of High Performance Download Client (HPDC) to GDC Data Transfer Tool
*   Implemented more links between the Projects page and the Data page
*   Improved usability and visual experience:
    *   Better handling of login failure
    *   User feedback when no results are available following a search
    *   Warning for unsupported browsers (Internet Explorer 9 is not supported)

### Bugs Fixed Since Last Release

*   Optimizations for downloading large files from the browser
*   Fixed various issues related to file search
*   Fixed Section 508 compliance issues
*   Clicking on "Total case" link in the Project List Page does not return results.

### Known Issues and Workarounds

*   Missing pathology files and slide images for some TCGA datasets since they are not connected to the biospecimen chain yet
*   In some specific situation (complex queries using must_not term), the advanced search might returns incorrect results
*   The add to cart button in the files data view does change its display value following a click (file is correctly added to cart though)
*   When exporting the annotation table to a file, the create date field is not displayed in the same format than the UI
*   In the advanced search, adding a trailing space to the query causes the UI to display a syntax error although the query is valid
*   Checksum missing for MAGE-TAB files
*   Data download statistics report is not Section 508 compliant

Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).

## Release 0.2.13

* __GDC Product__: GDC Data Portal
* __Release Date__: July 23, 2015

### New Features and Changes

*   Improved data portal searching. Three search mechanisms are available to the user:
    *   **Facet search**: Starting with all content available on the GDC, allow users to filter down their search by clicking on elements on the left of the screen. This feature is available in **Projects** and **Data** page. Range support was also added to facets in this release
    *   **Advanced Search**: Starting with all content available on the GDC, allow users to build a custom and complex query using all of GDC capabilities (any field with the parameter of their choice).
    *   **Quick Search**: When looking for specific portal element, allow users to launch the **Quick Search** by clicking the "?" or "CRTL+SPACE" and find high-level informations of some entities.
*   Updated styling to align with NCI new visual identify
*   Created a pie chart widget allowing user to easily switch between a pie chart and a table view.
*   Improved Usability and visual experience:
    *   Added tooltips to various sections of the portal
    *   Added a range facet with barchart
    *   Add more charts (summary, cart)

### Bugs Fixed Since Last Release

*   Hooked-up reports to real data
*   Fixed various issues on GQL (Advanced Search)
*   Table export to export appropriate columns
*   Allow users to sort project list table

### Known Issues and Workarounds

*   Checksum missing for MAGE-TAB files

Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).

## Release 0.1.10

* __GDC Product__: GDC Data Portal
* __Release Date__: March 18, 2015

### New Features and Changes

*   Authentication and Authorization
    *   Allow users to authenticate to the portal using their to ERA Commons credentials
    *   Allow users to narrow down their search based on their own projects
    *   Allow users to download a token link (for use with the API) in the portal
    *   Allow users to download protected data (if they have permissions to do so)
*   Reports
    *   Allow users to view a data download statistics report
*   Improve usability and visual experience:
    *   Allow users to view projects data using a new type of graph ("githut" style)
    *   Implement more features into the table widget (sort per column, hide/show columns, re-arrange columns, export to JSON, XML, CSV, TSV, maintain user preferences)
    *   Display UI and API version at the bottom of the page
*   Search
    *   Allow users to select multiple terms in facets (OR operand)
    *   Improvements on advance search with auto-loading of possible fields
*   Cart
    *   Allow users to view more details through additional pie charts
    *   Allow users to download a manifest from the cart
    *   Improved the mechanism of adding data to cart in various sections of the portal.
*   Updated style/theme to match GDC Website
*   Display a NCI Warning banner to inform users about GDC policy

### Bugs Fixed Since Last Release

*   Improvements in 508 compliance

### Known Issues and Workarounds

*   TARGET data is currently not available
*   Data:
    *   Absence of "real" download statistics for the data download statistics report
    *   Missing checksum for magetab files in file entity page
*   Project list table not sortable

Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).

## Release 0.1.8

* __GDC Product__: GDC Data Portal
* __Release Date__: January 22, 2015

### New Features and Changes

*   Allow users to perform a project search and obtain a list of projects (Project Search and List Pages)
*   Allow users to retrieve project details (Project Entity Page)
*   Allow users to perform an annotations search and obtain a list of annotations (Annotations Search and List Pages)
*   Allow users to retrieve entity details (Annotation Entity Page)
*   Implement a basic search feature (Basic Search - Facets))
*   Implement an advance search feature (Advanced Search - Query Language)
*   Allow users to retrieve file details (File Entity Page)
*   Allow users to retrieve participant details (Participant Entity Page)
*   Allow users to view and add files to a cart (Cart)
*   Allow users to view reports (Reports - Data Download Statistics)
*   Allow users to export tables of search results (Export Tables)
*   Allow users to download files (Download)
*   Allow users to authenticate using eRA Commons (Authentication)

### Bugs Fixed Since Last Release

*   Initial Release - Not Applicable

### Known Issues and Workarounds

*   TARGET data is currently not available

Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).


# Data Portal Release Notes

| Version | Date |
|---|---|
| [v2.2.0](Data_Portal_Release_Notes.md#release-220) | June 26, 2024 |
| [v2.1.0](Data_Portal_Release_Notes.md#release-210) | April 30, 2024 |
| [v2.0.0](Data_Portal_Release_Notes.md#release-200) | February 8, 2024 |
| [v1.30.4](Data_Portal_Release_Notes.md#release-1304) | May 11, 2023 |
| [v1.30.0](Data_Portal_Release_Notes.md#release-1300) | July 8, 2022 |
| [v1.29.0](Data_Portal_Release_Notes.md#release-1290) | August 23, 2021 |
| [v1.28.0](Data_Portal_Release_Notes.md#release-1280) | May 17, 2021 |
| [v1.25.1](Data_Portal_Release_Notes.md#release-1251) | August 14, 2020 |
| [v1.25.0](Data_Portal_Release_Notes.md#release-1250) | July 2, 2020 |
| [v1.24.1](Data_Portal_Release_Notes.md#release-1240) | March 10, 2020 |
| [v1.23.1](Data_Portal_Release_Notes.md#release-1231) | December 10, 2019 |
| [v1.23.0](Data_Portal_Release_Notes.md#release-1230) | November 6, 2019 |
| [v1.22.0](Data_Portal_Release_Notes.md#release-1220) | July 31, 2019 |
| [v1.21.0](Data_Portal_Release_Notes.md#release-1210) | June 5, 2019 |
| [v1.20.0](Data_Portal_Release_Notes.md#release-1200) | April 17, 2019 |
| [v1.19.0](Data_Portal_Release_Notes.md#release-1190) | February 20, 2019 |
| [v1.18.0](Data_Portal_Release_Notes.md#release-1180) | December 18, 2018 |
| [v1.17.0](Data_Portal_Release_Notes.md#release-1170) | November 7, 2018 |
| [v1.16.0](Data_Portal_Release_Notes.md#release-1160) | September 27, 2018 |
| [v1.15.0](Data_Portal_Release_Notes.md#release-1150) | August 23, 2018 |
| [v1.14.0](Data_Portal_Release_Notes.md#release-1140) | June 13, 2018 |
| [v1.13.0](Data_Portal_Release_Notes.md#release-1130) | May 21, 2018 |
| [v1.12.0](Data_Portal_Release_Notes.md#release-1120) | February 15, 2018 |
| [v1.11.0](Data_Portal_Release_Notes.md#release-1110) | December 21, 2017 |
| [v1.10.0](Data_Portal_Release_Notes.md#release-1100) | November 16, 2017 |
| [v1.9.0](Data_Portal_Release_Notes.md#release-190) | October 24, 2017 |
| [v1.8.0](Data_Portal_Release_Notes.md#release-180)| August 22, 2017 |
| [v1.6.0](Data_Portal_Release_Notes.md#release-160) | June 29, 2017 |
| [v1.5.2](Data_Portal_Release_Notes.md#release-152) | May 9, 2017 |
| [v1.4.1](Data_Portal_Release_Notes.md#release-141) | October 31, 2016 |
| [v1.3.0](Data_Portal_Release_Notes.md#release-130) | September 7, 2016 |
| [v1.2.0](Data_Portal_Release_Notes.md#release-120) | August 9th, 2016 |
| [v1.1.0](Data_Portal_Release_Notes.md#release-110) | June 1st, 2016 |
| [v1.0.1](Data_Portal_Release_Notes.md#release-101) | May 18, 2016 |

---
## Release 2.2.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  June 26, 2024

### New Features and Changes

* __GDC 1.0__:
    * GDC 1.0 has been officially retired and can no longer be reached.
* __Annotations__:
    * Annotations tables have been added to the __project, case, and file summary pages__. <!--PEAR-1853/1839/1855-->
    * Links to the case summary page have been removed for annotations that concern a redaction of the case. <!--PEAR-1942-->
* __ProteinPaint__:
    * A new option to toggle lollipops pointing up or down is now available.
* __OncoMatrix__:
    * Advanced sorting options for power users have been added.
    * Implemented a prototype for adding gene expression rows.
* __Gene Expression Clustering__:
    * Allows re-sort cases by dictionary variable, gene mutation, or expression level.
* __Sequence Reads__:
    * Adds ability to visualize truncated BAM slice when the slice file size exceeds 20MB and streaming is terminated.
* __Cohorts__ created from analysis tools now consistently consist of a specific list of cases that will remain unchanged after a data release. This includes cohorts created from the gene and mutation summary pages. <!--PEAR-1911-->
* The files tables in the __Cart__ and the __Repository__ now allow searching for files based on the associated cases' submitter ID and UUID. <!--PEAR-1900-->
* In the __case summary page__, values whose units are days, e.g. Days to Death or Days to Birth, are now displayed in years and days as appropriate for the user's convenience. <!--PEAR-1467-->
* __Quick Search__ results and the headers of all __summary pages__ have been updated with new designs and icons. <!--PEAR-1301-->
* Word wrapping has been improved for __Quick Search__ results to avoid unexpected word breaks. <!--PEAR-1897-->
* The Best Overall Response card in the Treatments category of the __Cohort Builder__ has been moved to a new position in the category. <!--PEAR-1988-->
* The text referencing the deletion of custom sets in the __Manage Sets__ page has been updated. <!--PEAR-1939-->
* The Access column has been added to the Source Files and Download Analyses Files tables in the __file summary page__. <!--PEAR-1872-->
* Text within the downloaded histogram image has been updated for greater clarity in the __Clinical Data Analysis__ tool. <!--PEAR-1805-->
* Filter panels in the __Projects__, __Mutation Frequency__, and __Repository__ tools have been standardized and now consistently allow scrolling to occur independently of the tables on the right. <!--PEAR-1579-->

### Bugs Fixed Since Last Release
* __Section 508 Accessibility__:
    * Aria labels have been added to the tables in Set Operations. <!--PEAR-1936-->
    * The Venn diagrams in __Set Operations__ and __Cohort Comparison__ now have the appropriate alt text and roles. <!--PEAR-1936-->
    * The Venn diagram button in __Cohort Comparison__ has been updated with both an aria label and an informative label. <!--PEAR-1936-->
    * The statistics table and its TSV for Box and QQ plots in __Clinical Data Analysis__ now contain data for Q1 and Q3. <!--PEAR-1935-->
    * Alt text has been added to both the Box plot and the QQ plot in __Clinical Data Analysis__. <!--PEAR-1935-->
    * Responsiveness for the header, footer, home page, and also the Projects and Repository tools has been improved so that these areas are accessible at a 200% zoom level. <!--PEAR-1925/1912-->
* __Cases Table__:
    * The downloaded TSV now contains the expected tabs. <!--PEAR-1947-->
    * The correct number of annotations will now be displayed for each case. <!--PEAR-1965-->
    * The Customize Columns options are no longer cut off at the bottom. <!--PEAR-1997-->
    * Search now correctly displays results even if the same search input is removed and then reapplied quickly. <!--PEAR-1740-->
* __Cohorts__:
    * Cohorts containing FM-AD cases will now update correctly when users with dbGaP access to FM-AD (phs001179) log in or out. <!--SV-2389-->
    * Cohorts created based on CNV losses or gains will now have the correct composition when filtered by additional mutated genes. <!--PEAR-1597-->
* __Cohort Builder__:
    * The buckets for Age At Index will no longer display incorrect ranges and counts. <!--PEAR-1964-->
    * Cards now display at the correct width when either the browser window is small or the zoom level is increased. <!--PEAR-1945-->
* __Mutation Frequency__:
    * Gene/mutation sets created from the tables in the Mutation Frequency tool will now contain the expected genes/mutations even if the cohort has Available Data filters or Biospecimen filters. <!--SV-2314-->
    * Users will no longer be able to create several cohorts in quick succession from Mutation Frequency without waiting for previous actions to be completed. <!--PEAR-1922-->
* __OncoMatrix__:
    * Made OncoMatrix react to divide-by term edits from the label click menu.
    * Fixed the continuous term scale for density plots.
    * Gene expression variable may not show expression data for all applicable cases, especially with a large cohort size.
* __Gene Expression Clustering__:
    * Fixed the continuous term scale for density plots.
* __Sequence Reads__:
    * The tool now displays the correct number of available BAM files when a cohort filter is in use.
* __Cohort MAF__:
    * Added the "tumor_bam_uuid" column.
* Limited the CSS reset to avoid conflict with embedded styles, by using scoped normalize CSS rules.
* Fixed conflicting CSS that can alter portal styling.
* Fixed an issue where __Sample Sheet__ downloads can be incomplete due to missing sample type information. <!--PEAR-1972-->
* Addressed an issue where changes to default filters in the __Cohort Builder__ and the __Repository__ may not be reflected after a release. <!--PEAR-1960-->
* The __Query Expressions__ section now correctly displays a maximum of 3 rows by default. Additionally, the button to display more than 3 rows at a time is enabled only when the cohort query exceeds 3 rows. <!--PEAR-1578-->
* The loading spinner is no longer displayed above the other areas of the Analysis Center when the __Cohort Comparison__ tool is loading. <!--SV-2360-->
* Default values in the Custom Bins modal within the __Clinical Data Analysis__ tool are now properly updated when the user toggles between displaying continuous values in days and years. <!--PEAR-1981-->
* The right side of the chart on the __Home Page__ is no longer cut off at smaller browser sizes. <!--PEAR-1167-->
* Tooltips are no longer displayed when there is no description available for filter properties. <!--SV-2425/PEAR-869-->

### Known Issues and Workarounds
* __Section 508 Accessibility__:
    * There are known Section 508 accessibility issues that the GDC plans to address in subsequent releases. If a user encounters a Section 508 barrier, please contact GDC Support (support@nci-gdc.datacommons.io) for assistance. Known Section 508 issues are identified below.
        * There are keyboard focus and navigation issues in analysis tools that use popup windows/overlays for custom user selections. Impacted analysis tools include BAM Slicing, Sequence Reads, Gene Expression Clustering, OncoMatrix, and ProteinPaint.
        * Heatmaps within the Sequence Reads tool do not contain concise alternative text or equivalent alternatives. Additionally, an equivalent alternative to the body plot on the home page is not available.
        * In the Gene Expression Clustering tool and OncoMatrix, there are no headers for genes, clusters, and/or cases in the heatmap.
        * In the Gene Expression Clustering tool, color is used to convey gene expression values but there are no patterns to convey the same information as color. Color is also used in ProteinPaint and the Sequence Reads tool to convey consequence type but there are no distinguishing patterns.
        * Some text can be difficult to read on a small screen at a 200% zoom level.
* __Survival Plot__:
    * The survival plot in Cohort Comparison does not display text indicating that there is insufficient survival data to plot. <!--SV-2357-->
    * In Mutation Frequency, the downloaded image may display a survival curve when none is plotted within the portal. <!--SV-2356-->
    * When the survival plot is zoomed in and an image is downloaded, the curves within the image may extend beyond the y-axis. <!--SV-2348-->
* __Cart__:
    * Spinners on the Download Cart and Download Associated Data buttons may be displayed longer than expected. This is a visual issue and does not affect the use of these buttons. <!--SV-2343-->
    * Using multiple browser tabs with the portal when adding or removing files from the cart may result in the cart not being updated as expected. <!--SV-2412-->
* In the __files, cases, and annotations tables__, the case ID search field is case-sensitive. If the search does not return the expected results, try changing the input to uppercase as case IDs are most commonly uppercased.
* __Cohorts__ filtered by mutated genes and SSMs not in those genes will result in 0 cases since the mutations have to belong to those particular genes in order to match cases for the results. As a workaround, first filter the cohort by the mutated genes and export the cohort using the Export Cohort feature in the Cohort Bar. Then, reimport the cohort using the Import New Cohort feature before applying the SSM filters. <!--SV-2331/PEAR-1616-->
* The __Slide Image Viewer__ will display a black image temporarily if a user zooms in on a slide then switches to another slide. <!--SV-2370-->
* The annotations table in the __file summary page__ does not include the Case ID column. This column is planned to be added in a future update. <!--PEAR-1899-->
* In __ProteinPaint__, the "Gene Expression" option is non-functional when filtering samples in a sub-track.
* In __Gene Expression Clustering__, the tooltip is not displayed when clicking an expression data cell.
* The custom range inputs for the __Age at Index__ card in the __Cohort Builder__ are not behaving as expected. As a workaround, use the predefined ranges available. Alternatively, use the custom range inputs on the Days tab to query for ages in years.

## Release 2.1.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  April 30, 2024

### New Features and Changes
* __Annotations Browser__
    * The Annotations Browser and annotation summary page have been implemented. <!--PEAR-1838/1181/1182/1160-->
* __Repository__:
    * Files can now be filtered by Tissue Type, Tumor Descriptor, Specimen Type, and Preservation Method. <!--PEAR-1514-->
    * Metadata and Sample Sheet downloads have been added to the Repository. <!--PEAR-1863-->
* __Cohort Level MAF__:
    * A cohort level MAF analysis tool has been added to the Analysis Center.
* __BAM Slicing Download and Sequence Reads__:
    * In BAM Slicing Download, call GDC API directly from client without going through ProteinPaint backend. No limits are applied on slicing region size or BAM slice file size.
    * In Sequence Reads Visualization, user can slice a BAM with a range lower than 300Kb, and if the resulting BAM slice is under 100Mb. Slicing and caching a BAM slice bigger than 100Mb will abort and user will be notified to reduce region size and try again. Before creating new cache file, find out old enough ones to delete to free up storage.
    * For both apps: The table listing available cases and bam files can be filtered by assay types.
* __Gene Expression Clustering__:
    * Enable gene variant legend group filter.
    * Support creating a single-case cohort.
    * Supported more clustering and distance calculation methods.
* __OncoMatrix__:
    * Enable downloading data.
    * Hide synonymous mutations by default.
    * Improve the matrix sorting options to easily toggle sorting by cnv and/or consequence.
    * Add Mutation and CNV control buttons, and hide CNV by default.
    * Create a mutations/consequences legend group for mutations.
    * Enable selecting individual mutation classes upon clicking the Mutation/CNV button.
    * Support creating a single-case cohort.
    * Display hints about persisted matrix gene set and option to unhide CNV and mutations when there is no matrix data to render.
    * Group similar mutation class colors together when sorting matrix samples and if CNVs are displayed.
    * Add "Single" style to render consequence data, as alternative to Stacked and OncoPrint styles.
* __ProteinPaint__:
    * Allow visualizing SSM in any genomic locus, besides "protein" mode.
    * Support creating a single-case cohort.
* The performance of the __Clinical Data Analysis__ tool has been improved, especially when large cohorts are used with QQ plots. <!--PEAR-1536-->
* __Quick Search__ now returns results for the latest versions of files when searching for older versions of those files. <!--PEAR-1804-->
* The X button on the __Unexpected Error__ dialog box has been removed. <!--SV-2367-->
* Buttons for launching demos have been removed from the selection view of __Cohort Comparison__ and __Set Operations__. <!--SV-2328/2327-->
* Responsiveness improvements have been made to the __Analysis Center__ and the __Cohort Bar__. <!--PEAR-1836-->
* The UX/UI for the __Cohort Builder__ has been improved. <!--PEAR-1547-->
* The __case summary page__ has been enhanced with a table listing all the files associated with the case. Additionally, a link to the table is now available in the header of the summary page, and information has been added to the File Counts summary tables to lead users to the new files table. The clinical and biospecimen supplements tables have also been removed from the case summary page. <!--PEAR-1822/1849/1833/PEAR-1832-->
* Set names for sets of the same type are now enforced to be unique when editing names in __Manage Sets__. <!--PEAR-1359-->
* Number range cards in the __Cohort Builder__ no longer display the custom range option when there is no data. <!--PEAR-661-->
* The Cohort Buider image on the __home page__ has been updated to reflect the latest design. <!--PEAR-1593-->
* The tooltip on the __Mutation Frequency__ card in the Analysis Center has been updated. <!--PEAR-1877-->

### Bugs Fixed Since Last Release
* __Section 508 Accessibility__:
    * Small aria-label inconsistencies have been addressed. <!--PEAR-1715-->
    * Keyboard focus is now returned to the triggering element when modals are closed. <!--PEAR-1658-->
    * Screen readers will now read out the contents of toast messages. <!--PEAR-1765-->
    * Toggles in the Clinical Data Analysis tool now have the correct number of labels. <!--PEAR-1764-->
    * Table header checkboxes are now correctly labelled. <!--PEAR-1754-->
    * Modal icons now have appropriate null alt text. <!--PEAR-1753-->
    * Assistive technologies no longer behave incorrectly with some controls due to incorrect, missing, or redundant labels, attributes, or roles. <!--PEAR-1672-->
    * Aria labels have been added to Cancer Gene Census annotation icon in Mutation Frequency. <!--PEAR-740-->
    * The Survival icon is now appropriately hidden from the accessibility tree for the benefit of screen readers. <!--PEAR-739-->
* __Cohorts__:
    * Using "Save As" to replace a cohort with itself will no longer result in an error notification despite the replacement being successful. <!--SV-2363-->
    * Saving a cohort that was previously saved now displays the correct message. <!--PEAR-1651-->
    * Cohorts will now display data in Mutation Frequency, Cohort Builder, and the summary charts even when removing gene/mutation filters from a cohort temporarily results in 0 cases. <!--SV-2414-->
    * Cohorts now contain the correct cases when created from the cases table by using the "Existing Cohort With Selected Cases" and "Existing Cohort Without Selected Cases" options with a cohort containing gene or mutation filters. <!--PEAR-1043-->
    * When saving a cohort, the confirmation notification will no longer be automatically dismissed before the saving dialog has closed. <!--SV-2366-->
* __Cohort Builder__:
    * Cohort Builder cards for number ranges now display an informative message rather than a spinner when there is no data for the facet. <!--PEAR-1646-->
    * Removing a custom Cohort Builder card no longer incorrectly removes the associated filters from the current cohort. <!--PEAR-1612-->    
    * Filters related to numeric values in the Cohort Builder now correctly displays the numbers entered. <!--SV-2383-->
* __Case Summary Page__:
    * The Biospecimen tree in the case summary page is no longer hidden when the bioId provided in the URL does not exist. <!--PEAR-1202-->
    * The error that sometimes occurs when viewing the __Follow-Ups/Molecular Tests__ tab in the case summary page has been resolved. <!--SV-2431-->
* __Mutation Frequency__:
    * The survival plot in __Mutation Frequency__ no longer flickers when the cohort has 0 cases. <!--SV-2331/PEAR-1701-->
    * Attempting to download a TSV of all the mutations in the GDC no longer results in an error due to the length of time needed to generate the TSV. <!--SV-2388-->
* __All ProteinPaint-based Tools__:
    * In GDC query, do not supply empty "case_filters{content[]}" that will slow down API. Lollipop and OncoMatrix are now faster when there's no cohort.
    * Updated mutation class definitions and rank for protein_altering_variant. Affects all tools that can show mutation data.  
    * Deprecated term "sample_type" is dropped from GDC dictionary.
* __BAM Slicing Download and Sequence Reads__:
    * When downloading GDC BAM slice (no caching), do not limit request region max size.
    * Reloading page while streaming/downloading GDC BAM slice to client will not crash server.
    * App UI requires hitting Enter to search by GDC file or case, and will no longer auto search (on pressing any key) to avoid showing duplicate SSM table.
    * BAM track bug fix to handle reads with no sequence.
    * BAM track bug fix for hide/show toggling at track menu.
* __Disco Plot__:
    * Bug fix for disco plot launched from sunburst showing AAchange in sandbox header rather than undefined.
    * Pass the cohort filter to the lollipop track from the matrix and disco plot label click.
* __Gene Expression Clustering__:
    * Enable gene variant legend group filter.
    * Support creating a single-case cohort.
    * Supported more clustering and distance calculation methods.
* __OncoMatrix__:
    * Fix position errors after OncoMatrix/hierCluster zooming in/out caused by outdated imgBox.
    * Do not allow hiding all the alteration groups.
    * Disable the geneset submit button when there there is less than a minNumGenes option (3 for hier cluster, 1 for matrix).
    * Add to OncoMatrix mutation/cnv buttons all available mutation/cnv classes in all GDC instead of only within current cohort.
    * Change the definition of truncating/protein-changing mutation, change OncoMatrix mutation classes sorting order.
    * Fix the detection of sorting-related updates in the matrix app, as distinct from the Gene Expression Clustering.
    * Pass the cohort filter to the lollipop track from the matrix and disco plot label click.
* __ProteinPaint__:
    * Sample summary table will scroll if too tall.
    * Bug fix to convert "case." to "cases." in case_filters[] for GDC mds3 sunburst clicking to load sample table.
    * Do not force the sample table to be positioned relative to screen bottom after a sunburst click.
    * Prevent double-clicking on a sunburst ring so that same sample table will not appear duplicated.
    * Bug fix for Lollipop category total sample count to respond/shrink with cohort change.
* Tokens are no longer refreshed when the __User Profile__ is viewed. <!--PEAR-1818-->
* __Quick Search__ now correctly displays results even if the same search input is applied twice quickly. <!--SV-2410-->
* In __Set Operations__, saving gene and mutation sets will now be successful if the saving dialog is manually dismissed after the Save button is clicked. <!--SV-2368-->
* Users will no longer be able to download more than 5 GB of files in total at a time via the browser from the __cart__. <!--SV-2342-->
* Table buttons in __Clinical Data Analysis__ no longer overlay the survival plot on smaller screens when many survival plots are displayed at the same time. <!--PEAR-1600-->
* The correct file size total will now be displayed in the __Repository__ when filtering is applied within the tool and the active cohort contains Available Data filters. <!--SV-2376-->
* Downloading the **Clinical/Biospecimen TSV or JSON** before the cohort has fully loaded will no longer result in an error. <!--SV-2402-->

### Known Issues and Workarounds

* __Section 508 Accessibility__:
    * There are known Section 508 accessibility issues that the GDC plans to address in subsequent releases. If a user encounters a Section 508 barrier, please contact GDC Support (support@nci-gdc.datacommons.io) for assistance. Known Section 508 issues are identified below.
        * There are keyboard focus and navigation issues in analysis tools that use popup windows/overlays for custom user selections. Impacted analysis tools include BAM Slicing, Sequence Reads, Gene Expression Clustering, OncoMatrix, and ProteinPaint.
        * Heatmaps within the Sequence Reads tool do not contain concise alternative text or equivalent alternatives. Additionally, equivalent alternatives to the Box plots, QQ plots, Venn diagrams, and the body plot on the home page are not available.
        * In the Gene Expression Clustering tool and OncoMatrix, there are no headers for genes, clusters, and/or cases in the heatmap.
        * In the Gene Expression Clustering tool, color is used to convey gene expression values but there are no patterns to convey the same information as color. Color is also used in ProteinPaint and the Sequence Reads tool to convey consequence type but there are no distinguishing patterns.
        * Some text can be difficult to read on a small screen at a 200% zoom level.
* __Cohorts__:
    * Cohorts are under active development and their behavior may change in the first several months after the release of GDC Portal 2.0. As this process may result in the loss of saved cohorts on the portal, we highly recommend [exporting cohorts](/Data_Portal/Users_Guide/getting_started.md#main-toolbar) locally.
    * Cohorts created based on CNV losses or gains may not have the correct composition when filtered by additional mutated genes. As a workaround, first filter by the mutated genes before creating cohorts based on CNV losses and gains.
    * Cohorts filtered by mutated genes and SSMs not in those genes may unexpectedly result in 0 cases. <!--SV-2331/PEAR-1616-->
    * Cohorts containing FM-AD cases may not update correctly when users with dbGaP access to FM-AD (phs001179) log in or out. As a workaround, logging in before creating cohorts with FM-AD cases is recommended. <!--SV-2389-->
* __Survival Plot__:
    * The survival plot in Cohort Comparison does not display text indicating that there is insufficient survival data to plot. <!--SV-2357-->
    * In Mutation Frequency, the downloaded image may display a survival curve when none is plotted within the portal. <!--SV-2356-->
    * When the survival plot is zoomed in and an image is downloaded, the curves within the image may extend beyond the y-axis. <!--SV-2348-->
* __Cart__:
    * Spinners on the Download Cart and Download Associated Data buttons may be displayed longer than expected. This is a visual issue and does not affect the use of these buttons. <!--SV-2343-->
    * Using multiple browser tabs with the portal when adding or removing files from the cart may result in the cart not being updated as expected. <!--SV-2412-->
* The aggregated MAF generated using the __Cohort Level MAF__ tool is missing the tumor_bam_uuid column. The tumor_sample_uuid and case_id should be used for reproducibility until the tumor_bam_uuid has been added.
* In both the __Sequence Reads__ and __BAM Slicing Download__ tools, the number of available BAM files may be overcounted when a cohort filter is in use.
* Gene/mutation sets created from the tables in the __Mutation Frequency__ tool may contain 0 genes/mutations if the cohort has Available Data filters or Biospecimen filters. <!--SV-2314-->
* The TSV of the __cases table__ may not contain the expected tabs. <!--DEV-2324-->
* In the Repository and cases table, the case ID search field is case-sensitive. If the search does not return the expected results, try changing the input to uppercase as case IDs are most commonly uppercased.
* When the __Cohort Comparison__ tool is loading, the loading spinner may be displayed above the other areas of the Analysis Center. <!--SV-2360-->
* The __Slide Image Viewer__ will display a black image temporarily if a user zooms in on a slide then switches to another slide. <!--SV-2370-->

## Release 2.0.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  February 8, 2024

### New Features and Changes

GDC 2.0 is a major update to the original GDC Data Portal introduced in 2016. This latest version adopts a "cohort-centric" workflow, in which users build custom sets of cases to analyze, and introduces several new analysis tools. New features of GDC 2.0 include:

* A cohort-centric workflow in which a cohort is first built and then analyzed using tools on the Data Portal. All of these functionalities can be reached from the Analysis Center.  
    * This includes a toolbar, that can be used to view or modify an existing cohort while using any analysis tool.
* Core tools that compose the main functionalities of the GDC Data Portal:
    * __Cohort Builder:__ Build a cohort of cases using clinical and biospecimen properties
    * __Repository:__ Download files based on a specific cohort
    * __Projects:__ Browse, filter, and create cohorts based on GDC projects
* Analysis tools that analyze specific cohorts:
    * __Mutation Frequency:__ Analyze somatic mutations that were called in the WXS and Targeted Sequencing pipelines and their associated genes
    * __Clinical Data Analysis:__ Analyze and visualize clinical data associated with your cohort
    * __Cohort Comparison:__ Analyze the properties of multiple cohorts
    * __Set Operations:__ Display a Venn diagram and compare/contrast cohorts or gene/mutation sets
    * __BAM Slicing Download:__ Download a specific region of a BAM file created by the GDC
    * __ProteinPaint:__ Visualize somatic mutations on a specific linear gene or chromosomal region
    * __Gene Expression Clustering:__ Visualizes gene expression clustering for a specific cohort
    * __Sequence Reads:__ Visualize the reads within a specific BAM file
    * __OncoMatrix:__ Visualize the most commonly mutated genes across a cohort

### Bugs Fixed Since Last Release

Not applicable as this is the initial release of GDC 2.0.

### Known Issues and Workarounds

* __Section 508 Accessibility__:
    * There are known Section 508 accessibility issues that the GDC plans to address in subsequent releases. If a user encounters a Section 508 barrier, please contact GDC Support (support@nci-gdc.datacommons.io) for assistance. Known Section 508 issues are identified below.
        * There are keyboard focus and navigation issues in analysis tools that use popup windows/overlays for custom user selections. Impacted analysis tools include BAM Slicing, Sequence Reads, Gene Expression Clustering, OncoMatrix, and ProteinPaint.
        * Heatmaps within the Sequence Reads tool do not contain concise alternative text or equivalent alternatives. Additionally, equivalent alternatives to the Box plots, QQ plots, Venn diagrams, and the body plot on the home page are not available.
        * In the Gene Expression Clustering tool and OncoMatrix, there are no headers for genes, clusters, and/or cases in the heatmap.
        * In the Gene Expression Clustering tool, color is used to convey gene expression values but there are no patterns to convey the same information as color. Color is also used in ProteinPaint and the Sequence Reads tool to convey consequence type but there are no distinguishing patterns.
        * Some text can be difficult to read on a small screen at a 200% zoom level.
        * Keyboard focus is not returned to the triggering element when modals are closed.
        * Assistive technologies may not behave correctly with some controls due to incorrect, missing, or redundant labels, attributes, or roles.
* __Cohorts__:
    * Cohorts are under active development and their behavior may change in the first several months after the release of GDC Portal 2.0. As this process may result in the loss of saved cohorts on the portal, we highly recommend [exporting cohorts](/Data_Portal/Users_Guide/getting_started.md#main-toolbar) locally.
    * Cohorts created based on CNV losses or gains may not have the correct composition when filtered by additional mutated genes. As a workaround, first filter by the mutated genes before creating cohorts based on CNV losses and gains.
    * Cohorts filtered by mutated genes and SSMs not in those genes may unexpectedly result in 0 cases. <!--SV-2331/PEAR-1616-->
    * When saving a cohort, the confirmation notification may be automatically dismissed before the saving dialog has closed. <!--SV-2366-->
    * Using "Save As" to replace a cohort with itself will result in an error notification despite the replacement being successful. <!--SV-2363-->
    * Cohorts containing FM-AD cases may not update correctly when users with dbGaP access to FM-AD (phs001179) log in or out. As a workaround, logging in before creating cohorts with FM-AD cases is recommended. <!--SV-2389-->
    * If removing gene/mutation filters from a cohort temporarily results in 0 cases, cohorts may not display data in Mutation Frequency, Cohort Builder, and the summary charts. As a workaround, remove the gene and mutation filters, then add them back. <!--SV-2414-->
* __Survival Plot__:
    * The survival plot in Cohort Comparison does not display text indicating that there is insufficient survival data to plot. <!--SV-2357-->
    * The survival plot in Mutation Frequency may flicker when the cohort has 0 cases. <!--SV-2331/PEAR-1701-->
    * In Mutation Frequency, the downloaded image may display a survival curve when none is plotted within the portal. <!--SV-2356-->
    * When the survival plot is zoomed in and an image is downloaded, the curves within the image may extend beyond the y-axis. <!--SV-2348-->
* __Cart__:
    * Spinners on the Download Cart and Download Associated Data buttons may be displayed longer than expected. This is a visual issue and does not affect the use of these buttons. <!--SV-2343-->
    * More than 5 GB of files in total may be downloaded at a time via the browser if the user first attempts to download controlled access data without being logged in, then logs in via the information dialog displayed before continuing with the download. <!--SV-2342-->
    * Using multiple browser tabs with the portal when adding or removing files from the cart may result in the cart not being updated as expected. <!--SV-2412-->
* __Mutation Frequency__:
    * Gene/mutation sets created from the tables in Mutation Frequency may contain 0 genes/mutations if the cohort has Available Data filters or Biospecimen filters. <!--SV-2314-->
    * Attempting to download a TSV of all the mutations in the GDC may result in an error due to the length of time needed to generate the TSV. As a workaround, limit the number of mutations downloaded to 1.5 million. <!--SV-2388-->
* __Main Toolbar__:
    * Attempting to download the **Clinical/Biospecimen TSV or JSON** before the cohort has fully loaded may result in an error. <!--SV-2402-->
    * The TSV of the **cases table** may not contain the expected tabs. <!--DEV-2324-->
* __OncoMatrix__:
    * Manually deleting all genes will result in an error message "Error: Cannot read properties of undefined (reading 'lst')". The user can close and re-open OncoMatrix for use.
    * Dragging genes only works once. After one gene is dragged to a new position, no genes can be dragged to new positions.
    * In OncoMatrix, the cases may not get re-sorted as expected after a certain sequence of actions
* __ProteinPaint__:
    * A nested filter may be constructed for a Lollipop subtrack, e.g. sex=male AND ( primarysite=aa OR disease=bb ), but cannot be translated into GDC cohort filters. The translation code has a preliminary implementation that only works for "flat" filters without nesting.
    * Cohorts cannot be created using the Create Cohort button in ProteinPaint for a single sample
    * In ProteinPaint, the total number of samples in a category breakdown and the total number of samples in the sunburst ring are not based on a user's current cohort
    * In ProteinPaint, when clicking on a sunburst ring, the sample table is not showing up
    * In ProteinPaint, a Disco plot launched from the sunburst ring can show "undefined" in the plot header
    * A ProteinPaint plot launched from OncoMatrix and Gene Expression Clustering does not observe the current cohort and displays mutated cases for all GDC
* In the __Gene Expression Clustering__ tool, if any part of the dendrogram is selected and the current cohort is modified, then the new dendrogram will render with scattered subtrees selected.
* The "A" in the Allele Summary text is cut off in the __Sequence Reads__ tool.
* __Quick Search__ may not display results if the the same search input is applied twice quickly. As a workaround, temporarily change the input before reentering the intended search. <!--SV-2410-->
* Filters related to numeric values may display a smaller number than what the user entered within the __Cohort Builder__. This is a visual issue and does not affect the filters applied to the cohort. <!--SV-2383-->
* When the __Cohort Comparison__ tool is loading, the loading spinner may be displayed above the other areas of the Analysis Center. <!--SV-2360-->
* The __Repository__ tool may display an incorrect file size total of 0 bytes when filtering is applied within the tool and the active cohort contains Available Data filters. <!--SV-2376-->
* The __Slide Image Viewer__ will display a black image temporarily if a user zooms in on a slide then switches to another slide. <!--SV-2370-->
* In __Set Operations__, the saving of gene and mutation sets may be unsuccessful if the saving dialog is manually dismissed after the Save button is clicked. <!--SV-2368-->
* Clicking the X button on the __Unexpected Error dialog box__ does not dismiss it. The workaround is to click the OK button. <!--SV-2367-->

## Release 1.30.4

* __GDC Product__: GDC Data Portal
* __Release Date__:  May 11, 2023

### New Features and Changes

* The GDC Legacy Archive has officially been retired.
    * The Legacy Archive Portal can no longer be reached.
    * Any API call to query files from the Legacy Archive will no longer work.
    * Downloads for files from the Legacy Archive will work normally with manifests that were generated previously.


### Bugs Fixed Since Last Release

* The Clinical TSV download in the case entity page and cart now contain TSVs for pathology detail, follow up, and molecular test entities.   <!--SV-2004-->
* Fixed bug in which demographic information would not download as a TSV when diagnosis information was not available. <!--DEV-1238-->

### Known Issues and Workarounds

*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  Negative numbers may be displayed for the Missing value category in the Treatment node within a Clinical Analysis.  This occurs with projects that have multiple treatment nodes per case. All other values should be accurate. <!--SV-1604-->
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->

## Release 1.30.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  July 8, 2022

### New Features and Changes

* None

### Bugs Fixed Since Last Release

* Fixed error in which the manifest could not be downloaded directly from the GDC Data Portal in certain instances. <!--SV-2112-->

### Known Issues and Workarounds

*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  Negative numbers may be displayed for the Missing value category in the Treatment node within a Clinical Analysis.  This occurs with projects that have multiple treatment nodes per case. All other values should be accurate. <!--SV-1604-->
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    * The footer says version 1.9, but it is actually 1.13
    *	Filtering by vital_status does not function in the Legacy Archive due to updates in how this property has been indexed.  A workaround is to perform the case level filtering in the GDC Data Portal and copy the filter string for use in the Legacy Archive or the legacy API. <!--SV-1508-->
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->


## Release 1.29.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  August 23, 2021

### New Features and Changes <!--REQ-431-->

* None

### Bugs Fixed Since Last Release

* Fixed error in which the data summaries in the clinical analysis and cart pages were only partially displayed when viewed in Chrome v.91.0.4472. <!--SV-1940-->

### Known Issues and Workarounds

*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  Negative numbers may be displayed for the Missing value category in the Treatment node within a Clinical Analysis.  This occurs with projects that have multiple treatment nodes per case. All other values should be accurate. <!--SV-1604-->
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    * The footer says version 1.9, but it is actually 1.13
    *	Filtering by vital_status does not function in the Legacy Archive due to updates in how this property has been indexed.  A workaround is to perform the case level filtering in the GDC Data Portal and copy the filter string for use in the Legacy Archive or the legacy API. <!--SV-1508-->
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->

## Release 1.28.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  May 17, 2021

### New Features and Changes <!--REQ-422-->

* New columns were added to the "molecular test" table at the bottom of the case entity page to display additional molecular test fields.

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

*  When accessing the data portal with Chrome v.91.0.4472, users may experience some display errors.  This includes the data summary in the clinical analysis and cart pages being only partially displayed. <!--SV-1940-->
*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  Negative numbers may be displayed for the Missing value category in the Treatment node within a Clinical Analysis.  This occurs with projects that have multiple treatment nodes per case. All other values should be accurate. <!--SV-1604-->
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    * The footer says version 1.9, but it is actually 1.13
    *	Filtering by vital_status does not function in the Legacy Archive due to updates in how this property has been indexed.  A workaround is to perform the case level filtering in the GDC Data Portal and copy the filter string for use in the Legacy Archive or the legacy API. <!--SV-1508-->
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->

## Release 1.25.1

* __GDC Product__: GDC Data Portal
* __Release Date__:  August 14, 2020

### New Features and Changes <!--REQ-408-->

* API improvements were made to increase portal performance.

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  Negative numbers may be displayed for the Missing value category in the Treatment node within a Clinical Analysis.  This occurs with projects that have multiple treatment nodes per case. All other values should be accurate. <!--SV-1604-->
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    * The footer says version 1.9, but it is actually 1.13
    *	Filtering by vital_status does not function in the Legacy Archive due to updates in how this property has been indexed.  A workaround is to perform the case level filtering in the GDC Data Portal and copy the filter string for use in the Legacy Archive or the legacy API. <!--SV-1508-->
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->

## Release 1.25.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  July 2, 2020

### New Features and Changes <!--REQ-395-->

* Suppressed Experimental Strategy filter on the Exploration page as this currently filters for files with a particular strategy, not for cases.  This may cause confusion amongst users.  The filter will be re-instated in a future release once the logic is available to filter more appropriately for cases tied to a specific strategy. <!--PRTL-3119-->
* Updated the filter control panel styling across the Portal to have clearer titles (e.g. "Search Cases" instead of "Cases" in the quick search box). <!--PRTL-2987-->
* Made minor updates to the styling of the filter query display at the top of the Exploration page (spacing, borders). <!--PRTL-2988-->
* Added an expand/collapse control to the quick search bar of Clinical tab on the Exploration page, to be consistent with other Exploration tabs. <!--PRTL-2990-->
* Added a clear title above the counts in each filter control panel across the Portal (e.g. "# Cases", "# Genes", etc.). <!--PRTL-2991-->
* Moved various action buttons above the results table on the Repository Page to more accessible locations. <!--PRTL-2994-->
* Improved load time of the initial custom filter list on the Repository Page, when clicking "Add a Filter Filter" or "Add a Case/Biospecimen Filter". <!--PRTL-3057-->

### Bugs Fixed Since Last Release

* Fixed a bug in the Age at Diagnosis table on the Cohort Comparison page, where the # of cases in the table was not consistent with the # of cases shown when clicking the link to the Exploration page. <!--PRTL-3032-->
* Fixed minor positional accuracy issue of the lollipop data points on the Protein Viewer. <!--PRTL-2967-->
* Fixed bug on the Protein Viewer where, if clicking to switch between different lollipop data points, details of the previous lollipop was not closing. <!--PRTL-2968-->
* Fixed bug where the quick search bar on the Exploratin Page's Genes filter tab was not expanding/collapsing properly. <!--PRTL-2989-->
* Fixed bug in the pop-up warning message when adding or removing items from the Cart, where long filenames were spilling outside the border of the pop-up. <!--PRTL-3031-->
* Fixed typo in the "View Cases in Exploration" button on the Repository page. <!--PRTL-3122-->
* Fixed typo in the pop-up user consent message when downloading controlled files from the Cart. <!--PRTL-3123-->

### Known Issues and Workarounds

*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  Negative numbers may be displayed for the Missing value category in the Treatment node within a Clinical Analysis.  This occurs with projects that have multiple treatment nodes per case. All other values should be accurate. <!--SV-1604-->
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    * The footer says version 1.9, but it is actually 1.13
    *	Filtering by vital_status does not function in the Legacy Archive due to updates in how this property has been indexed.  A workaround is to perform the case level filtering in the GDC Data Portal and copy the filter string for use in the Legacy Archive or the legacy API. <!--SV-1508-->
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->

## Release 1.24.1

* __GDC Product__: GDC Data Portal
* __Release Date__:  March 10, 2020

### New Features and Changes <!--REQ-395-->

* Removed unnecessary comma and y-axis value from title of the mutation details pop-up in the Protein Viewer. <!--PRTL-2969-->
* Added Tobacco Smoking Status field to the Exposures tab on the Case entity page. <!--PRTL-2952-->
* Added a link to the Cart where users can access instructions for downloading the GDC Genome Build reference files. <!--PRTL-2929-->
* Added logic to prevent duplicate fetching of data for Clinical Analysis survival plots and optimize rendering. <!--PRTL-2910-->
* Added a button to clear searches for certain Portal search controls that were previously missing this ability. <!--PRTL-2900-->
* Reduced whitespace between Oncogrid and its control panel to optimize spacing and layout. <!--PRTL-2899-->
* Made entire Clinical Analysis results page responsive (card columns now scale & stack in response to the size of the browser window). <!--PRTL-2882-->
* Replaced Clinical Analysis function for printing clinical cards to a single PDF file, with more flexible functionality to instead download all the cards in SVG and/or PNG format. <!--PRTL-2870-->
* Added message to notify users when they try to access the Portal using Microsoft Internet Explorer, indicating which browsers are officially supported. <!--PRTL-2868-->
* Added arrow icon to sortable columns across the Portal to indicate the current sort direction. <!--PRTL-2330-->

### Bugs Fixed Since Last Release

* Fixed bug where clicking a primary site on the Human Body Image was not re-directing to the Exploration page. <!--PRTL-2979-->
* Fixed layout issue where long Annotation Notes were exceeding the border of the text box. <!--PRTL-2944-->
* Fixed layout issue where the Repository header and action buttons were scaling and wrapping incorrectly if the browser window is shrunk beyond a certain threshold. <!--PRTL-2938-->
* Fixed layout issue where the responsive Clinical Analysis Cards were clipping improperly as the browser window is shrunk beyond a certain threshold. <!--PRTL-2928-->
* Fixed bug where the Clinical Tab on the Exploration page was crashing when entering a custom range of Years for the Age at Diagnosis facet. <!--PRTL-2913-->
* Fixed various minor cosmetic and color issues in PNG, SVG downloads of the Clinical Analysis survival plots. <!--PRTL-2887-->
* Fixed bug where the x-axis in PNG, SVG downloads of histograms across the Portal was being bolded incorrectly. <!--PRTL-2885-->
* Fixed bug where the expand/collapse symbols in the UI were incorrectly being exported in the TSV download of the Projects table. <!--PRTL-2883-->
* Fixed bug where Oncogrid's modal for customizing colors could not be scrolled below the fold if it was shrunk beyond a certain threshold. <!--PRTL-2881-->
* Fixed incorrect DTT hyperlink in the GDC Apps menu. <!--PRTL-2867-->
* Fixed bug where the "dbSNP rs ID" facet could not be minimized in the Exploration page's Mutations facet tab. <!--PRTL-2840-->
* Fixed layout issue where the Portal's header incorrectly overlaps some content when a notification banner is displayed. <!--PRTL-2753-->
* Fixed some minor layout & styling issues in the Exploration page's facets panel. <!--PRTL-2614-->
* Fixed bug where the Case ID on the Exploration page's Cases facet tab was not searchable in certain scenarios. <!--PRTL-2587-->
* Fixed bug where the Expand/Collapse button state was not changing properly when being used in the Biospecimen section of the Case entity page. <!--PRTL-2575-->
* Fixed incorrect capitalization of "dbGaP" in the Summary section of the Project entity page. <!--PRTL-2436-->
* Fixed layout issue where the Advanced Search query box on the Repository page could expanded beyond the margins of the box's border. <!--PRTL-2271-->

### Known Issues and Workarounds

*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  Negative numbers may be displayed for the Missing value category in the Treatment node within a Clinical Analysis.  This occurs with projects that have multiple treatment nodes per case. All other values should be accurate. <!--SV-1604-->
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    * The footer says version 1.9, but it is actually 1.13
    *	Filtering by vital_status does not function in the Legacy Archive due to updates in how this property has been indexed.  A workaround is to perform the case level filtering in the GDC Data Portal and copy the filter string for use in the Legacy Archive or the legacy API. <!--SV-1508-->
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->

## Release 1.23.1

* __GDC Product__: GDC Data Portal
* __Release Date__:  December 10, 2019

### New Features and Changes <!--REQ-396-->

* Updated display of x-axis units on the homepage Human Body chart to more easily display increased case counts for newly-added projects <!--PRTL-2925-->

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  Negative numbers may be displayed for the Missing value category in the Treatment node within a Clinical Analysis.  This occurs with projects that have multiple treatment nodes per case. All other values should be accurate. <!--SV-1604-->
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    * The footer says version 1.9, but it is actually 1.13
    *	Filtering by vital_status does not function in the Legacy Archive due to updates in how this property has been indexed.  A workaround is to perform the case level filtering in the GDC Data Portal and copy the filter string for use in the Legacy Archive or the legacy API. <!--SV-1508-->
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->

## Release 1.23.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  November 6, 2019

### New Features and Changes <!--REQ-390-->

* Added Clinical Data Analysis feature that allows Users to: <!--FEAT-520, FEAT-521-->
    * Explore clinical data via the new Clinical Tab on the Exploration page.
    * Build custom Case sets based on that clinical data for later analysis.
    * Create an analysis to examine the clinical variables in a Case set, using various tools including histograms, survival plots, box plots, QQ plots, and custom binning.
    * Download the data (as TSV, JSON) and plots (as PNG, SVG) of each clinical variable in an anlysis.
    * Save an analysis to local storage to resume later (as long as storage is not cleared).
* Added links to CIViC annotations on the Gene and Mutation entity pages. <!--PRTL-2543-->
* Updated the default Top Mutated Genes histogram on the Exploration page to display only COSMIC Genes by default. <!--PRTL-2371-->
* Added Follow-Ups tab and nested Molecular Tests to Case entity page. <!--PRTL-2704-->
* Added text to BAM slicing modal to instruct Users how to access unmapped reads. <!--PRTL-2620-->

### Bugs Fixed Since Last Release

* Fixed font in exported PNGs, SVGs to be consistent with the Portal UI. <!--PRTL-2653-->
* Made custom Case and File filters in the Repository page case insensitive. <!--PRTL-2571-->
* Fixed bug where pfam domains in Protein Viewer could not be clicked in Firefox. <!--PRTL-2511-->
* Fixed bug where TSV download button could not be clicked in MS Edge. <!--PRTL-2357-->
* Fixed controlled access alert pop-up in the Cart so that the modal disappears correctly once the User has successfully logged in and initiated the download. <!--PRTL-2429-->

### Known Issues and Workarounds

*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  Negative numbers may be displayed for the Missing value category in the Treatment node within a Clinical Analysis.  This occurs with projects that have multiple treatment nodes per case. All other values should be accurate. <!--SV-1604-->
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    * The footer says version 1.9, but it is actually 1.13
    *	Filtering by vital_status does not function in the Legacy Archive due to updates in how this property has been indexed.  A workaround is to perform the case level filtering in the GDC Data Portal and copy the filter string for use in the Legacy Archive or the legacy API. <!--SV-1508-->
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->

## Release 1.22.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  July 31, 2019

### New Features and Changes <!--REQ-387-->

* Replaced existing Clinical, Biospecimen columns on the Projects page with 4 columns: Clinical, Clinical Supplement, Biospecimen, Biospecimen Supplement. The Clinical and Biospecimen columns now link directly to the project page, and their counts indicate the total cases in the project. The Clinical Supplement and Biospecimen Supplement columns work the same as the old Clinical and Biospecimen columns - They link to the Repository page with Files filtered based on the Project and Data Category (Clinical or Biospecimen). <!--PRTL-2528-->
* Added a new icon to the GDC Apps menu, which links to the GDC Publications website page. <!--PRTL-2547-->
* Added the Synchronous Malignancy field to the Diagnoses / Treatments tab on the Case entity page. <!--PRTL-2582-->
* Added the Pack Years Smoked field to the Exposures tab on the Case entity page. <!--PRTL-2584-->
* Increased length of x-axis labels on histograms to 10 characters so that projects with names that are typically standard 10 chars will display fully (e.g. most TCGA projects like TCGA-BRCA). <!--PRTL-2598-->

### Bugs Fixed Since Last Release

* Fixed bug where the PNG, SVG files for the Overall Survival Plot could not be downloaded. <!--PRTL-2528-->

### Known Issues and Workarounds

*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Filtering by vital_status does not function in the Legacy Archive due to updates in how this property has been indexed.  A workaround is to perform the case level filtering in the GDC Data Portal and copy the filter string for use in the Legacy Archive or the legacy API. <!--SV-1508-->
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->   

## Release 1.21.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  June 5, 2019

### New Features and Changes <!--REQ-383-->

* Changed all Survival Plots to display the Duration (x-axis) in years instead of days. <!--PRTL-2404-->
* Updated data references to clinical properties throughout the Portal to match the underlying changes in the GDC data dictionary. <!--PRTL-2459-->

### Bugs Fixed Since Last Release

* Fixed bug where X-axis labels in histograms were cut off when displayed. <!--PRTL-1896-->
* Renamed the 'Experimental Strategies' facet on the Projects page to singular form. <!--PRTL-2262-->
* Fixed bug where columns with a % value of infinity (due to division by zero) show as 'NaN%'.  Replaced instead with a label of '--'. <!--PRTL-2384-->
* Fixed bug where the download button in the cart access banner was still disabled after a user logged in from the banner.  Instead, the experience is now improved so that after login, the banner is closed and the user must explicitly click 'Download' again. <!--PRTL-2393-->
* Fixed bug where if a new user logs into the Portal and views their profile, the app crashes if the user has no projects assigned yet. <!--PRTL-2529-->
* Fixed bug where Survival Rate numbers in the Survival Plot plot y-axis did not scale properly and overlapped into the axis lines. <!--PRTL-2530-->

### Known Issues and Workarounds

*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->   

## Release 1.20.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  April 17, 2019

### New Features and Changes <!--REQ-382-->

* Upgraded the Portal to use the latest React Javascript library (version 16.8) <!--PRTL-2440-->

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    

## Release 1.19.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  February 20, 2019

### New Features and Changes <!--REQ-381-->

* Added support for viewing of controlled-access mutations in the Data Portal
* Added a new data access notification to remind logged-in users with access to controlled data that they need to follow their data use agreement.  The message is fixed at the top of the Portal.<!--PRTL-2400, PRTL-2434-->
* Added the ability to search for previous versions of files.  If the user enters the UUID of a previous version that cannot be found, the Portal returns the UUID of the latest version available. <!--PRTL-2387-->
* Renamed the Data Category for "Raw Sequencing Data" to "Sequencing Reads" throughout the portal where this appears, to be consistent with the Data Dictionary. <!--PRTL-118-->
* Added a link in the Portal footer to the GDC support page. <!--PRTL-2383-->

### Bugs Fixed Since Last Release

* Fixed bug where Survival Plot button never stops loading if plotting mutated vs. non-mutated cases for a single Gene. <!--PRTL-2398-->
* Fixed inconsistent button styling when downloading controlled Downstream Analyses Files from File Entity page. <!--PRTL-2395-->
* Removed unnecessary Survival column from Arrange Columns button on Case Entity, Gene Entity pages. <!--PRTL-2281-->
* Removed unnecessary whitespace from pie charts on Repository page. <!--PRTL-1923-->
* Added missing File Size unit to Clinical Supplement File, Biospecimen Supplement File tables on Case Entity page. <!--PRTL-2070-->
* Fixed bug where clicking on Case Counts in Projects Graph tab was going to the Repository Files tab instead of the Cases tab. <!--PRTL-2272-->
* Fixed bug where the counts shown beside customer filters on the Repository Cases tab were not updating when filtering on other facets. <!--PRTL-2412-->
* Fixed bug where clicking the # of Affected Cases denominator on the Gene page's Most Frequent Somatic Mutations table displayed an incorrect number of Cases.

### Known Issues and Workarounds

*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    

## Release 1.18.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  December 18, 2018

### New Features and Changes <!--REQ-335-->

* A new data access message has been added when downloading controlled data.  Users must agree to abide by data access control policies when downloading controlled data. <!--PRTL-2324,PRTL-2370,-->
* In the Mutation free-text search in Exploration, mutation display now includes the UUID, genomic location, and matched search term for easier mutation searching. <!--PRTL-1996, PRTL-2367-->
* The ability to sort on ranked columns has been made available. <!--PRTL-2274,PRTL-2336,PRTL-2365-->

### Bugs Fixed Since Last Release

*  In some cases, text was being cut off on the Project page visualization tab.  Text is no longer cut off. <!--PRTL-2290-->
*  HGNC link on Gene page broke as the source format url changed; The format was updated and the link is now functional <!--PRTL-2380-->
*  In the biospecimen details on the Case page, the cart icon would disappear once clicked.  It now is always visible. <!--PRTL-2282-->

### Known Issues and Workarounds

*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    

## Release 1.17.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  November 7, 2018

### New Features and Changes <!--REQ-334-->

* Copy Number Variation (CNV) data derived from GISTIC results are now available in the portal:
	* View number of CNV events on a gene in a cohort in the Explore Gene table tab <!-- PRTL-2259, PRTL-2328 -->
	* Explore CNVs associated with a gene on the Gene Entity Page <!-- PRTL-2252,PRTL-2273,PRTL-2344 -->
	* Explore CNVs concurrently with mutations on the Oncogrid with new visualization <!-- PRTL-2244,PRTL-2251,PRTL-2256,PRTL-2257,PRTL-2275,PRTL-2276, PRTL-2214, PRTL-2314, PRTL-2315, PRTL-2317, PRTL-2319, PRTL-2325, PRTL-2326, PRTL-2327,PRTL-2266  -->

### Bugs Fixed Since Last Release

*  None

### Known Issues and Workarounds

*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    

## Release 1.16.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  September 27, 2018

### New Features and Changes <!--REQ-333-->

* Updated Human Body Image to aggregate all current primary sites to available Major Primary Sites <!--PRTL-2248-->


### Bugs Fixed Since Last Release

* Fixed link on cart download error popup <!--PRTL-2284-->
* Updated Cancer Distribution table to have dropdown menus for primary_site and disease_type <!--PRTL-2286-->
* Updated Y-axis label on `Top Mutated Cancer Genes in Selected Projects Graph` <!--PRTL-2264-->
* Updated Set Operation Image to remove stray text <!--PRTL-2214-->

### Known Issues and Workarounds

*  Advanced Search
    * For advanced search and custom file facet filtering there are some properties that will appear as options that are no longer supported (e.g. file_state). <!--API-530-->
*  Custom facet filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  SIFT and PolyPhen annotations are missing from the Export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    


## Release 1.15.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  August 23, 2018

### New Features and Changes <!--REQ-329-->

* File Versions are now visible in the "File Versions" section on the File Entity Page. <!--PRTL-2015-->
* "View Files in Repository" and "View Cases in Repository" button methods were updated to work faster. <!--PRTL-2003 PRTL-2106 -->

### Bugs Fixed Since Last Release

* Fixed warning messages that prompted users to login even when already logged in.  Error warnings now correctly prompt users to reference dbGAP for data access if already signed in. <!--PRTL-1937-->
* Fixed error where you could click Go on Case ID wildcard facet before inputting any data. <!--PRTL-2083-->
* Fixed cart header to be a consistent color for the whole table. <!--PRTL-1943-->
* Fixed error where you could save a set with no name or items, which resulted in an infinite spinner. <!--PRTL-2218--> <!--PRTL-1858-->
* Fixed table width issue when FM-AD was selected as a filter. <!--PRTL-2054-->
* Updated broken help link on Advanced Query. <!--PRTL-2218-->

### Known Issues and Workarounds

*  Advanced Search
    * For advanced search and custom file facet filtering there are some properties that will appear as options that are no longer supported (e.g. file_state). <!--API-530-->
*  Custom facet filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  SIFT and PolyPhen annotations are missing from the Export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    


## Release 1.14.0

* __GDC Product__: GDC Data Portal
* __Release Date__: June 13, 2018

### New Features and Changes <!--REQ-322-->

* Added new Experimental Strategies Diagnostic Slide Image, Bisulfite-Seq, ChIP-Seq, and ATAC-Seq to Case and Project entity pages.

### Bugs Fixed Since Last Release

* Fixed download of clinical and biospecimen data from the Repository when Case table rows are selected.


### Known Issues and Workarounds

*  Custom facet filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  SIFT and PolyPhen annotations are missing from the Export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  When user is logged in and try to download a controlled file he does not have access to, he's prompted to log in. He should be promted to request access. <!-- PRTL-1937 -->
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).


## Release 1.13.0

* __GDC Product__: GDC Data Portal
* __Release Date__: May 21, 2018

### New Features and Changes <!--REQ-322-->

*  Added new image viewer functionality for viewing tissue slide images <!--PRTL-2016-->


### Bugs Fixed Since Last Release

* Updated gene reference labels on gene entity page to adhere to preferred usage <!--PRTL-1911-->
* Fixed issue with user profile displaying all projects twice <!--PRTL-2035-->


### Known Issues and Workarounds

*  Custom facet filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  SIFT and PolyPhen annotations are missing from the Export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  When user is logged in and try to download a controlled file he does not have access to, he's prompted to log in. He should be promted to request access. <!-- PRTL-1937 -->
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    



Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).

## Release 1.12.0

* __GDC Product__: GDC Data Portal
* __Release Date__: February 15, 2018

### New Features and Changes <!--REQ-322-->

*  Provided the ability to export clinical and biospecimen data in a TSV format from the Case, Project, Exploration, Repository and Cart pages.<!--PRTL-1929-->
*  Removed from the Project entity page the sections about mutated genes, somatic mutations and affected Cases and replaced with a button "Explore data" that will open the Exploration page filtered on the project. Indeed the Exploration page provides the same information.  Added a breakdown of cases per primary site for a Project entity page with multiple primary sites (e.g. FM-AD). <!--PRTL-1903-->
*  Added display of coding DNA change and impacts for all the transcripts (instead of canonical transcript only) in the Mutation entity page - Consequences section. In the mutation table (e.g. in Repository), the impacts and consequences are displayed for the canonical transcript only. <!--PRTL-1927-->

### Bugs Fixed Since Last Release

*  Replaced the suggested set name when saving a set with selected items, e.g. for case set the suggested name is now "Custom Case selection". <!--PRTL-1911-->
*  Fixed the protein viewer to indicate when there are overlapping mutations. Mousing over the dot showing multiple mutations will open a right panel with the list of all the corresponding mutations.  <!--SV-750-->
*  Fixed Mutation entity page - Consequences table: the "Coding DNA Change" column is now populated for all the transcripts. <!--SV-751-->
*  Fixed download clinical and download biospecimen actions from TCGA-BRCA project.
*  Fixed facet behavior that did not reset back to showing all options after pressing reset-arrow. <!--PRTL-1928-->
*  Fixed error when user was trying to save a set with no value in the textbox "Save top:".  <!--PRTL-1909-->
*  Removed somatic mutation section from Case entity page for cases with no open-access mutation data (e.g. FM-AD or TARGET cases). <!--PRTL-1926-->
*  Fixed error where a blank page appears after unselecting `Cancer Gene Census` mutation facet. <!--PRTL-1933-->
*  Fixed duplicated date in sample sheet name (e.g. gdc_sample_sheet_YYYY-MM-DD_HH-MM.tsv.YYYY-MM-DD_HH-MM.tsv). <!--SV-942-->
*  Fixed error when annotations were not downloaded along with the file (in File entity page and Cart).  <!--SV-401-->

### Known Issues and Workarounds

*  Custom facet filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  SIFT and PolyPhen annotations are missing from the Export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  When user is logged in and try to download a controlled file he does not have access to, he's prompted to log in. He should be promted to request access. <!-- PRTL-1937 -->
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).


## Release 1.11.0

* __GDC Product__: GDC Data Portal
* __Release Date__: December 21, 2017

### New Features and Changes <!--REQ-322-->

* Updated UI to support SIFT and Polyphen annotations <!--PRTL-1404-->
* A `Sample Sheet` can now be created which allows easy association between file names and the case and sample submitter_id <!--PRTL-1872-->
* Updated Advanced Search page to include options to `Add All Files to Cart`, `Download Manifest`, and `View X Cases in Exploration` <!--PRTL-1796-->
* Provide clear message rather than blank screen if survival plots cannot be calculated for particular cohort comparison <!--PRTL-1842-->
* Display sample_type on associated entities section on file page <!--PRTL-1890-->
* Allows for special characters in case, gene, and mutation set upload (`-, :, >, .`) <!--PRTL-1847-->


### Bugs Fixed Since Last Release

*  Fixed error when trying to download large number of files from the Legacy Archive cart <!--LGCY-75-->
*  Fixed number of annotations displayed in Legacy Archive for particular entities <!--LGCY-74-->
*  Replaced missing bars to indicate proportion of applicable files and cases on project entity page in Cases and File Counts by Data Category table <!--PRTL-1725-->
*  Fixed project page display when projects are selected that contain no mutation data in the facet panel <!--PRTL-1754-->
*  Fixed error where exporting case sets as TSV included fewer cases than the total <!--PRTL-1888-->
*  Fixed error in exploration section when adding custom facets.  Previously selecting 'Only show fields with values' did not result in the expected behavior <!--PRTL-1901-->
*  Fixed error where number of associated entities for a file was showing an incorrect number <!--PRTL-1891-->

### Known Issues and Workarounds

*  Sample sheet will download with a file name including the date duplicated (e.g. gdc_sample_sheet_YYYY-MM-DD_HH-MM.tsv.YYYY-MM-DD_HH-MM.tsv)<!--SV-942-->
*  Custom facet filters
    * Definitions are missing from the property list when adding custom facet file or case filters <!--SV-916-->
*  Visualizations
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
    *  In the protein viewer there may be overlapping mutations.  In this case mousing over a point will just show a single mutation and the other mutations at this location will not be apparent.  <!--SV-750-->
*  Entity page
    *  On the mutation entity page, in the Consequences Table, the "Coding DNA Change" column is not populated for rows that do not correspond to the canonical mutation. <!-- SV-751 -->
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only.
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).



## Release 1.10.0

* __GDC Product__: GDC Data Portal
* __Release Date__: November 16, 2017

### New Features and Changes <!--REQ-322-->

* Support for uploading Case and Mutation sets in Exploration page <!--PRTL-1452, PRTL-1453-->
* Support for saving, editing, removing Case, Gene and Mutation sets in the Exploration page <!--PRTL-1472,PRTL-1464, PRTL-1473, PRTL-1468, PRTL-1465, PRTL-1470,PRTL-1469, PRTL-1466, PRTL-1471-->
* Added a Managed Sets menu where the user can see their saved sets <!--PRTL-1597-->
* Added an Analysis menu with two analyses: Set Operation and Cohort Comparison <!--PRTL-1599, PRTL-1600-->
* Added a User Profile page that shows all the projects and permissions assigned to the user: available in the username dropdown after the user logs in <!--PRTL-1458-->

### Bugs Fixed Since Last Release

*  Project page
    *  On the project page, the Summary Case Count link should open the case tab on the Repository page - instead it opens the file page <!--PRTL-1591-->

### Known Issues and Workarounds

*  Custom facet filters
    * Definitions are missing from the property list when adding custom facet file or case filters <!--SV-916-->
    * Selecting 'Only show fields with values' will show some fields without values in the Repository section.  This works correctly under the Exploration section. <!--SV-917-->
*  Visualizations
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
    *  In the protein viewer there may be overlapping mutations.  In this case mousing over a point will just show a single mutation and the other mutations at this location will not be apparent.  <!--SV-750-->
*  Entity page
    *  On the mutation entity page, in the Consequences Table, the "Coding DNA Change" column is not populated for rows that do not correspond to the canonical mutation. <!-- SV-751 -->
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only.
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).

## Release 1.9.0

* __GDC Product__: GDC Data Portal

* __Release Date__: October 24, 2017

### New Features and Changes <!--REQ-317-->

* Support for projects with multiple primary sites per project <!--PRTL-1478, PRTL-1676, PRTL-1668, PRTL-1675,PRTL-1685,PRTL-1695-->
* Support for slides that are linked to `sample` rather than `portion` <!--PRTL-1696-->

### Bugs Fixed Since Last Release

None

### Known Issues and Workarounds
*  Visualizations
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
    *  In the protein viewer there may be overlapping mutations.  In this case mousing over a point will just show a single mutation and the other mutations at this location will not be apparent.  <!--SV-750-->
*  Project page
    *  On the project page, the Summary Case Count link should open the case tab on the Repository page - instead it opens the file page <!--PRTL-1591-->
*  Entity page
    *  On the mutation entity page, in the Consequences Table, the "Coding DNA Change" column is not populated for rows that do not correspond to the canonical mutation. <!-- SV-751 -->
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only.
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).


## Release 1.8.0

* __GDC Product__: GDC Data Portal
* __Release Date__: August 22, 2017

### New Features and Changes

Major features/changes:

* A feature that links the exploration and repository pages was added. For example:
    - In the exploration page, cases with a specific mutation could be selected. This set could then be linked to the repository page to download the data files associated with these cases.
    - In the repository menu, the user can select cases associated with specific files. The set could then be linked to exploration page to view the variants associated with this set of cases.

*  Users can now upload a custom gene list to the exploration page and leverage the GDC search and visualization features for cases and variants associated with the gene set.

*  Filters added for the gene entity page. For example:
    - Clicking on a mutated gene from the project page will display mutations associated with the gene that are present in this project (filtered protein viewer, etc.).
    - Clicking on a mutated gene from the exploration page will display the mutations associated with the gene filtered by additional search criteria, such as "primary site is Kidney and mutation impact is high".

* UUIDs are now hidden from tables and charts to simplify readability. The UUIDs can still be exported and viewed in the tables using the "arrange columns" feature. In the mutation table, UUIDs are automatically exported.

* Mutation entity page - one consequence per transcript is shown (10 rows by default) in the consequence table. The user should display all rows before exporting the table.

### Bugs Fixed Since Last Release
*  Exploration
    *  Combining "Variant Caller" mutation filter with a case filter will display incorrect counts in the mutation facet. The number of mutations in the resulting mutation table is correct. <!-- API-307 -->
    *  Mutation table: it is difficult to click on the denominator in "#Affected Cases in Cohort" column displayed to the left side of the bar. The user should click at a specific position at the top of the number to be able to go to the corresponding link. <!-- PRTL-1377 -->


### Known Issues and Workarounds
*  Visualizations
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
    *  In the protein viewer there may be overlapping mutations.  In this case mousing over a point will just show a single mutation and the other mutations at this location will not be apparent.  <!--SV-750-->
*  Project page
    *  On the project page, the Summary Case Count link should open the case tab on the Repository page - instead it opens the file page <!--PRTL-1591-->
*  Entity page
    *  On the mutation entity page, in the Consequences Table, the "Coding DNA Change" column is not populated for rows that do not correspond to the canonical mutation. <!-- SV-751 -->
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only.
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).


## Release 1.6.0

* __GDC Product__: GDC Data Portal
* __Release Date__: June 29, 2017

### New Features and Changes

There was a major new release of the GDC Data Portal focused on Data Analysis, Visualization, and Exploration (DAVE). Some important new features include the following:

*  New visual for the Homepage: a human body provides the number of Cases per Primary Site with a link to an advanced Cancer Projects search
*  The Projects menu provides the Top 20 Cancer Genes across the GDC Projects and the Case Distribution per Project
*  A new menu "Exploration" is an advanced Cancer Projects search which provides the ability to apply Case, Gene, and Mutation filters to look for:
    * List of Cases with the largest number of Somatic Mutations
    *	The most frequently mutated Genes
    *	The most frequent Variants
    *	Oncogrid view of mutation frequency
*  Visualizations are provided across the Project, Case, Gene and Mutation entity pages:
    *	List of most frequently mutated genes and most frequent variants
    *	Survival plots for patients with or without specific variants
    *	Survival plots for patients with or without variants in specific genes
    *	Lollipop plots of mutation frequency across protein domains
*  Links to external databases (COSMIC, dbSNP, Uniprot, Ensembl, OMIM, HGNC)
*  Quick Search for Gene and Mutation entity pages
*  The ability to export the current view of a table in TSV
*  Retired GDC cBioPortal

_For detailed updates please review the [Data Portal User Guide](../Users_Guide/getting_started/)._

### Bugs Fixed Since Last Release

*  BAM Slicing dialog box does not disappear automatically upon executing the BAM slicing function. The box can be closed manually. <!-- PRTL-282 -->
*  Very long URLs will produce a 400 error.  Users may encounter this after clicking on "source files" on a file page where the target file is derived from hundreds of other files such as for MAF files. <!-- SV-396 / PRTL-342-->
*  If bam slicing produces an error pop-up message it will be obscured behind the original dialog box. <!--SV-419-->
    *   Internet Explorer users are not able to use the "Only show fields with no values" when adding custom facets <!-- PGDC-2467 / PRTL-109 -->
    *   Exporting large tables in the Data Portal may produce a 500 error.  Filtering this list to include fewer cases or files should eliminate the error <!--API-223-->    

### Known Issues and Workarounds
*  New Visualizations
    *  Cannot export Data Portal graphs in PNG in Internet Explorer. Graphs can be exported to PNG or SVG from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet would not display chart legend and title when re-opening previously downloaded SVG files, recommendation is to open downloaded SVG files with another software.
    *  In the protein viewer there may be overlapping mutations.  In this case mousing over a point will just show a single mutation and the other mutations at this location will not be apparent.  <!--SV-750-->
*  Exploration
    *  Combining "Variant Caller" mutation filter with a case filter will display wrong counts in the mutation facet. The number of mutations in the result mutation table is correct. <!-- API-307 -->
    *  Mutation table: it is difficult to click on the denominator in "#Affected Cases in Cohort" column displayed to the left side of the bar. The user should click at a specific position at the top of the number to be able to go to the corresponding link. <!-- PRTL-1377 -->
*  Entity page
    *  On the mutation entity page, in the Consequences Table, the "Coding DNA Change" column is not populated for rows that do not correspond to the canonical mutation. <!-- SV-751 -->
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only.
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).



## Release 1.5.2

* __GDC Product__: GDC Data Portal
* __Release Date__: May 9, 2017

### New Features and Changes

* Removed link to Data Download Statistics Report <!--PRTL-1081-->
* Updated version numbers of API, GDC Data Portal, and Data Release

### Bugs Fixed Since Last Release

*  None

### Known Issues and Workarounds

*   General
    *   Exporting large tables in the Data Portal may produce a 500 error.  Filtering this list to include fewer cases or files should eliminate the error <!--API-223-->
    *   After successful authentication, the authentication popup does not close for Internet Explorer users running in "Compatibility View". Workaround is to uncheck "Display Intranet sites in Compatibility View" in Internet Explorer options. Alternatively, refreshing the portal will correctly display authentication status. <!-- PGDC-2403 / PRTL-133 -->
    *   BAM Slicing dialog box does not disappear automatically upon executing the BAM slicing function. The box can be closed manually. <!-- PRTL-282 -->
    *   Due to preceding issue, If bam slicing produces an error pop-up message it will be obscured behind the original dialog box. <!--SV-419-->
    *   Very long URLs will produce a 400 error.  Users may encounter this after clicking on "source files" on a file page where the target file is derived from hundreds of other files such as for MAF files.  To produce a list of source files an API call can be used with the search parameter "fields=analysis.input_files.file_name". <!-- SV-396 / PRTL-342-->
		*   Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.


Example

    https://api.gdc.cancer.gov/files/455e26f7-03f2-46f7-9e7a-9c51ac322461?pretty=true&fields=analysis.input_files.file_name




*   Cart
    *   Counts displayed in the top right of the screen, next to the Cart icon, may become inconsistent if files are removed from the server. <!-- PGDC-2403 / PRTL-133 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   Internet Explorer users are not able to use the "Only show fields with no values" when adding custom facets <!-- PGDC-2467 / PRTL-109 -->
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).





## Release 1.4.1

* __GDC Product__: GDC Data Portal
* __Release Date__: October 31, 2016

### New Features and Changes

* Added a search feature to help users select values of interest in certain facets that have many values. <!-- PRTL-21 -->
* Added support for annotation ID queries in quick search. <!-- PRTL-29 -->
* Added a warning when a value greater than 90 is entered in the "Age at Diagnosis" facet. <!-- PRTL-77 -->
* Added Sample Type column to file entity page. <!-- PRTL-42 -->
* Authentication tokens are refreshed every time they are downloaded from the GDC Data Portal. <!-- PRTL-278 -->
* Buttons are inactive when an action is in progress. <!-- PRTL-270 -->
* Improved navigation features in the overview chart on portal homepage. <!-- PRTL-2 -->
* Removed State/Status from File and Case entity pages <!-- PRTL-292 -->
* Removed the "My Projects" feature. <!-- PRTL-174 -->
* Removed "Created" and "Updated" dates from clinical and biospecimen entities. <!-- PRTL-3 -->

### Bugs Fixed Since Last Release

*  Advanced search did not accept negative values for integer fields. <!-- PRTL-283 -->
*  Moving from facet search to advanced search resulted in an incorrect advanced search query. <!-- PRTL-284 -->
*  Some facets were cut off in Internet Explorer and Firefox. <!-- PRTL-290 -->

### Known Issues and Workarounds

*   General
    *   Exporting large tables in the Data Portal may produce a 500 error.  Filtering this list to include fewer cases or files should eliminate the error <!--API-223-->
    *   After successful authentication, the authentication popup does not close for Internet Explorer users running in "Compatibility View". Workaround is to uncheck "Display Intranet sites in Compatibility View" in Internet Explorer options. Alternatively, refreshing the portal will correctly display authentication status. <!-- PGDC-2403 / PRTL-133 -->
    *   BAM Slicing dialog box does not disappear automatically upon executing the BAM slicing function. The box can be closed manually. <!-- PRTL-282 -->
    *   Due to preceding issue, If bam slicing produces an error pop-up message it will be obscured behind the original dialog box. <!--SV-419-->
    *   Very long URLs will produce a 400 error.  Users may encounter this after clicking on "source files" on a file page where the target file is derived from hundreds of other files such as for MAF files.  To produce a list of source files an API call can be used with the search parameter "fields=analysis.input_files.file_name". <!-- SV-396 / PRTL-342-->
		*   Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.


Example

    https://api.gdc.cancer.gov/files/455e26f7-03f2-46f7-9e7a-9c51ac322461?pretty=true&fields=analysis.input_files.file_name




*   Cart
    *   Counts displayed in the top right of the screen, next to the Cart icon, may become inconsistent if files are removed from the server. <!-- PGDC-2403 / PRTL-133 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   Internet Explorer users are not able to use the "Only show fields with no values" when adding custom facets <!-- PGDC-2467 / PRTL-109 -->
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).




## Release 1.3.0

* __GDC Product__: GDC Data Portal
* __Release Date__: September 7, 2016

### New Features and Changes

*   A new "Metadata" button on the cart page to download merged clinical, biospecimen, and file metadata in a single consolidated JSON file. **May require clearing browser cache** <!-- PRTL-177 -->
*   Added a banner on the Data Portal to help users find data <!-- PRTL-237 -->
*   Added support for "Enter" key on login button <!-- PRTL-136 -->
*   On the Data page, the browser will remember which facet tab was selected when hitting the "Back" button <!-- PRTL-147 -->
*   In file entity page, if there is a link to one single file, redirect to this file's entity page instead of a list page.  <!-- PRTL-75 -->


### Bugs Fixed Since Last Release

* 	Adding a mix of open and controlled files to the cart from any Case entity pages was creating authorization issues <!-- PRTL-226 -->
*  Opening multiple browser tabs and adding files in those browser tabs was not refreshing the cart in other tabs. <!-- PRTL-181 -->
*   When user logs in from the advanced search page, the login popup does not automatically close <!-- PRTL-88 -->
*   When removing a file from the cart and clicking undo, GDC loses track of permission status of the user towards this file and will ask for the user to log-in again. <!-- PGDC-2496 -->
*   Download File Metadata button produces incomplete JSON output omitting such fields as file_name and submitter_id.  The current workaround includes using the API to return file metadata. <!-- SV-359 / PRTL-177 -->    
*   Annotations notes do not wrap to the next line at the beginning or the end of a word, some words might be split in two lines <!-- PGDC-2474 / PRTL-182-->   
*   Sorting annotations by Case UUID causes error <!-- PRTL-95 -->   

### Known Issues and Workarounds

*   General
    *   When no filters are engaged in the Legacy Archive or Data Portal, clicking the Download Manifest button may produce a 500 error and the message "We are currently experiencing issues. Please try again later.".  To avoid this error the user can first filter by files or cases to reduce the number files added to the manifest.
    *   After successful authentication, the authentication popup does not close for Internet Explorer users running in "Compatibility View". Workaround is to uncheck "Display Intranet sites in Compatibility View" in Internet Explorer options. Alternatively, refreshing the portal will correctly display authentication status. <!-- PGDC-2403 / PRTL-133 -->
    *   BAM Slicing dialog box does not disappear automatically upon executing the BAM slicing function. The box can be closed manually. <!-- PRTL-282 -->
    *   Due to preceding issue, If bam slicing produces an error pop-up message it will be obscured behind the original dialog box. <!--SV-419-->
    *   Very long URLs will produce a 400 error.  Users may encounter this after clicking on "source files" on a file page where the target file is derived from hundreds of other files such as for MAF files.  To produce a list of source files an API call can be used with the search parameter "fields=analysis.input_files.file_name". <!-- SV-396 -->
    *   On the Legacy Archive, searches for "Case Submitter ID Prefix" containing special characters are not displayed correctly above the result list. The result list is correct, however. <!--SV-412 / LGCY-33-->

Example

    https://api.gdc.cancer.gov/files/455e26f7-03f2-46f7-9e7a-9c51ac322461?pretty=true&fields=analysis.input_files.file_name




*   Cart
    *   Counts displayed in the top right of the screen, next to the Cart icon, may become inconsistent if files are removed from the server. <!-- PGDC-2403 / PRTL-133 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   Internet Explorer users are not able to use the "Only show fields with no values" when adding custom facets <!-- PGDC-2467 / PRTL-109 -->
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).

## Release 1.2.0

* __GDC Product__: GDC Data Portal
* __Release Date__: August 9th, 2016

### New Features and Changes

*   Added a retry (1x) mechanism for API calls <!-- PGDC-2393 -->
*   Added support for ID fields in custom facets <!-- PGDC-1222 -->
*   Added Case Submitter ID to the Annotation entity page <!-- PGDC-747 -->
*   Added a link to Biospeciment in the Case entity page <!-- PGDC-2346 -->

### Bugs Fixed Since Last Release

*   General.
    *   Not possible to use the browser's back button after hitting a 404 page <!-- PGDC-2429 -->
    *   404 page missing from Legacy Archive Portal <!-- PGDC-2477 -->
    *   Table widget icon and export JSON icon should be different <!-- PGDC-2446 -->    
    *   Download SRA XML files from the legacy archive portal might not be possible in some context <!-- PGDC-2457 --> <!-- PGDC-2469 -->
*   Data and facets
    *   Default values for age at diagnosis is showing 0 to 89 instead of 0 to 90 <!-- PGDC-2478 -->
    *   Biospecimen search in the case entity page does not highlight (but does bold and filter) results in yellow when title case is not followed <!-- PGDC-2451 -->
    *   Table sorting icon does not include numbers <!-- PGDC-35 -->   
    *   '--' symbol is missing on empty fields (blank instead), additional missing fields identified since last release.  <!-- PGDC-2447 -->    
### Known Issues and Workarounds

*   General
    *   When no filters are engaged in the Legacy Archive or Data Portal, clicking the Download Manifest button may produce a 500 error and the message "We are currently experiencing issues. Please try again later.".  To avoid this error the user can first filter by files or cases to reduce the number files added to the manifest.
    *   After successful authentication, the authentication popup does not close for Internet Explorer users running in "Compatibility View". This only impact users at the NIH. Workaround is to uncheck "Display Intranet sites in Compatibility View" in Internet Explorer options. Alternatively, refreshing the portal will correctly display authentication status.
    *   When user login from the advanced search page, the login popup does not automatically close <!-- PRTL-88 -->
*   Cart
    *   When removing a file from the cart and clicking undo, GDC looses track of permission status of the user towards this file and will ask for the user to log-in again. <!-- PGDC-2496 -->     
    *   Counts displayed in the top right of the screen, next to the Cart icon, might get inconsistent if files are removed from the server. <!-- PGDC-2403 -->
    *   Download File Metadata button produces incomplete JSON output omitting such fields as file_name and submitter_id.  The current workaround includes using the API to return file metadata. <!-- SV-359 -->
*   Annotations
    *   Annotations notes do not wrap to the next line at the beginning or the end of a word, some words might be split in two lines <!-- PGDC-2474 -->
    *   Sorting annotations by Case UUID causes error <!-- PRTL-95 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   Internet Explorer users are not able to use the "Only show fields with no values" when adding custom facets <!-- PGDC-2467 -->
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibilty mode <!-- PGDC-2480 -->    


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).


## Release 1.1.0

* __GDC Product__: GDC Data Portal
* __Release Date__: June 1st, 2016

### New Features and Changes

*   This is a bug-fixing release, no new features were added.

### Bugs Fixed Since Last Release

*   General
    *  Fixed 508 compliance issues. <!-- PGDC-2497 --><!-- PGDC-2431 -->
    *  Disabled download manifest action on projects without files.  <!-- PGDC-2416 -->
    *  Updated the portal to indicate to the user that his session expired when he tries to download the authentication token. <!-- PGDC-2455 -->   
    *  Unselected "My project" filter after user logs-in. <!-- PGDC-2462 -->  
    *  Fixed missing padding when query includes "My Projects". <!-- PGDC-2420 -->
    *  Enforced "Add to cart" limitation to 10,000 files everywhere on the Data Portal. <!-- PGDC-2409 -->
*   Tables
    *  Improved usability of the "Sort" feature  <!-- PGDC-1771 -->
    *  Updated the "Add all files to cart" button to add all files corresponding to the current query (and not only displayed files). <!-- PGDC-2439 -->
    *  Fixed an issue where Platform would show "0" when selected platform is "Affymetrix SNP 6.0". <!-- PGDC-2419 -->
*   Data
    *  Corrected default values populated when adding a custom range facet. <!-- PGDC-2445 --> <!-- PGDC-2444 -->  <!-- PGDC-2225 -->    
    *  Fixed an issue preventing the user to sort by File Submitter ID in data tables.<!-- PGDC-2430 -->    
*   File Entity Page
    *  Improved "Associated Cases/Biospecimen" table for files associated to a lot of cases.  <!-- PGDC-1993 -->    
    *  Fixed an error when performing BAM Slicing. <!-- PGDC-2433 -->

### Known Issues and Workarounds

*   General.
    *   After successful authentication, the authentication popup does not close for Internet Explorer users running in "Compatibility View". This only impact users at the NIH. Workaround is to uncheck "Display Intranet sites in Compatibility View" in Internet Explorer options. Alternatively, refreshing the portal will correctly display authentication status.
    *   Download SRA XML files from the legacy archive portal might not be possible in some context <!-- PGDC-2457 --> <!-- PGDC-2469 -->  
    *   Not possible to use the browser's back button after hitting a 404 page <!-- PGDC-2429 -->
    *   404 page missing from Legacy Archive Portal <!-- PGDC-2477 -->
    *   Table widget icon and export JSON icon should be different <!-- PGDC-2446 -->     
*   Data and facets
    *   Default values for age at diagnosis is showing 0 to 89 instead of 0 to 90 <!-- PGDC-2478 -->
    *   Biospecimen search in the case entity page does not highlight (but does bold and filter) results in yellow when title case is not followed <!-- PGDC-2451 -->
    *   Table sorting icon does not include numbers <!-- PGDC-35 -->    
    *   '--' symbol is missing on empty fields (blank instead), additional missing fields identified since last release.  <!-- PGDC-2447 -->    
*   Cart
    *   When removing a file from the cart and clicking undo, GDC looses track of permission status of the user towards this file and will ask for the user to log-in again. <!-- PGDC-2496 -->     
    *   Counts displayed in the top right of the screen, next to the Cart icon, might get inconsistent if files are removed from the server. <!-- PGDC-2403 -->
*   Annotations
    *   Annotations notes do not wrap to the next line at the beginning or the end of a word, some words might be split in two lines <!-- PGDC-2474 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   Internet Explorer users are not able to use the "Only show fields with no values" when adding custom facets <!-- PGDC-2467 -->
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibilty mode <!-- PGDC-2480 -->    


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).


## Release 1.0.1

* __GDC Product__: GDC Data Portal
* __Release Date__: May 18, 2016

### New Features and Changes

*   This is a bug-fixing release, no new features were added.

### Bugs Fixed Since Last Release

*   Tables and Export
    *   Restore default table column arrangement does not restore to the default but it restores to the previous state <!-- PGDC-1769 -->
*   Cart and Download
    *   Make the cart limit warning message more explanatory <!-- PGDC-1952 -->   
    *   In some situations, adding filtered files to the cart might fail <!-- PGDC-1981 -->   
*   Layout, Browser specific and Accessibility
    *   When disabling CSS, footer elements are displayed out of order <!-- PGDC-1972 -->
    *   If javascript is disabled html tags are displayed in the warning message <!-- PGDC-1835 -->
    *   Layout issues when using the browser zoom in function on tables <!-- PGDC-116 -->
    *   Cart download spinner not showing at the proper place <!-- PGDC-2056 -->
    *   Not all facets are expanded by default when loading the app <!-- PGDC-2061 -->

### Known Issues and Workarounds

*   General
    *   If a user has previously logged into the Portal and left a session without logging out, if the user returns to the Portal after the user's sessionID expires, it looks as if the user is still authenticated. The user cannot download the token and gets an error message that would not close. The user should clear the cache to properly log out.
    *   '--' symbol is missing on empty fields (blank instead) <!-- PGDC-2418 -->
    *   Download manifest button is available for TARGET projects with 0 files, resulting in error if user clic on button <!-- PGDC-2416 -->
    *   After successful authentication, the authentication popup does not close for Internet Explorer users running in "Compatibility View". This only impact users at the NIH. Workaround is to uncheck "Display Intranet sites in Compatibility View" in Internet Explorer options. Alternatively, refreshing the portal will correctly display authentication status.
*   Data    
    *   When adding a custom range facet, default values are incorrectly populated <!-- PGDC-2445 --> <!-- PGDC-2444 -->  <!-- PGDC-2225 -->
    *   The portal might return incorrect match between cases and files when using field cases.samples.portions.created_datetime (custom facet or advanced search). Note: this is not a UI issue. <!-- PGDC-2440 -->
    *   Sorting File Submitter ID option on the file tab result in a Data Portal Error <!-- PGDC-2430 -->
*   Tables and Export
    *   Table sorting icon does not include numbers <!-- PGDC-35 -->    
*   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).

Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).


# Advanced Search

Only available in the Repository view, the Advanced Search page offers complex query building capabilities to identify specific set of cases and files.

[![Access to Advanced Search from Data View](images/gdc-data-portal-access-advanced-search-data-view_v2.png)](images/gdc-data-portal-access-advanced-search-data-view.png "Click to see the full image.")


## Overview: GQL

Advanced search allows for structured queries to search for files and cases. This is done via Genomic Query Language (GQL), a query language created by the [GDC](https://gdc.cancer.gov/) and [OICR](https://oicr.on.ca/).

[![Advanced Search View](images/gdc-data-portal-advanced-search.png)](images/gdc-data-portal-advanced-search.png "Click to see the full image.")

A simple query in GQL (also known as a 'clause') consists of a __field__, followed by an __operator__, followed by one or more __values__. For example, the simple query `cases.primary_site = Brain` will find all cases for projects in which the primary site is Brain:

[![Advanced Search Example](images/gdc-data-portal-advanced-search-example.png)](images/gdc-data-portal-advanced-search-example.png "Click to see the full image.")

Note that it is not possible to compare two fields (e.g. disease_type = project.name).

> __Note:__ GQL is not a database query language. For example, GQL does not have a "SELECT" statement.

### Switching between Advanced Search and Facet Filters

When accessing Advanced Search from Repository View, a query created using facet filters in Repository View will be automatically translated to an Advanced Search GQL Query.

A query created in Advanced Search is not translated back to facet filters. Clicking on "Back to Facet Search" will return the user to Data View and reset the filters.

## Using the Advanced Search

When opening the Advanced Search Page (via the Repository View), the search field will be automatically populated with facets filters already applied (if any).

This default query can be removed by pressing "Reset".

Once the query has been entered and is identified as a "Valid Query", click on "Search" to run your query.

### Auto-complete

As a query is being written, the GDC Data Portal will analyze the context and offer a list of auto-complete suggestions. Auto-complete suggests both fields and values as described below.

### Field Auto-complete

The list of auto-complete suggestions includes __all__ available fields matching the user text input. The user has to scroll down to see more fields in the dropdown:

[![Field Auto-complete](images/gdc-data-portal-advanced-search-project.png)](images/gdc-data-portal-advanced-search-project.png "Click to see the full image.")

### Value Auto-complete

The list of auto-complete suggestions includes top 100 values that match the user text input. The user has to scroll down to see more values in the dropdown.

The value auto-complete is not aware of the general context of the query, the system will display all available values in GDC for the selected field. It means the query could return 0 results depending of other filters.

[![Value Auto-complete](images/gdc-data-portal-advanced-search-value.png)](images/gdc-data-portal-advanced-search-value.png "Click to see the full image.")

> __Note:__ Quotes are automatically added to the value if it contains spaces.

## Setting Precedence of Operators

You can use parentheses in complex GQL statements to enforce the precedence of operators.

For example, if you want to find all the open files in TCGA program as well as the files in TARGET program, you can use parentheses to enforce the precedence of the Boolean operators in your query, i.e.:

	(files.access = open and cases.project.program.name = TCGA) or cases.project.program.name = TARGET

> __Note:__ Without parentheses, the statement will be evaluated left-to-right.

## Keywords

A GQL keyword is a word that joins two or more clauses together to form a complex GQL query.

**List of Keywords:**

*   __AND__
*   __OR__

> __Note:__ Parentheses can be used to control the order in which clauses are executed.

### "__AND__" Keyword

Used to combine multiple clauses, allowing you to refine your search.

Examples:

* Find all open files in breast cancer:

		cases.primary_site = Breast and files.access = open


* Find all open files in breast cancer and data type is gene expression quantification:

		cases.primary_site = Breast and files.access = open and files.data_type = "Gene Expression Quantification"


### "__OR__" Keyword

Used to combine multiple clauses, allowing you to expand your search.

> __Note:__ The __IN__ keyword can be an alternative to __OR__ and result in simplified queries.

Examples:

*   Find all files that are raw sequencing data or aligned reads:

		files.data_type = "Aligned Reads" or files.data_type = "Raw sequencing data"

*   Find all files where cases are male or vital status is alive:

		cases.demographic.gender = male or cases.diagnoses.vital_status = alive


## Operators

An operator in GQL is one or more symbols or words comparing the value of a field on its left with one or more values on its right, such that only true results are retrieved by the clause.

### List of Operators and Query format

| Operator | Description |
| --- | --- |
| = | Field EQUAL Value (String or Number) |
| != | Field NOT EQUAL Value (String or Number) |
| < | Field LOWER THAN Value (Number or Date) |
| <= | Field LOWER THAN OR EQUAL Value (Number or Date) |
| \> | Field GREATER THAN Value (Number or Date) |
| \>= | Field GREATER THAN OR EQUAL Value (Number or Date) |
| IN | Field IN [Value 1, Value 2] |
| EXCLUDE | Field EXCLUDE [Value 1, Value 2] |
| IS MISSING | Field IS MISSING |
| NOT MISSING | Field NOT MISSING |


### "__=__" Operator - __EQUAL__

The "__=__" operator is used to search for files where the value of the specified field exactly matches the specified value.

Examples:

*   Find all files that are gene expression quantification:

		files.data_type = "Gene Expression Quantification"

*   Find all cases whose gender is female:

		cases.demographic.gender = female

### "__!=__" Operator - __NOT EQUAL__

The "__!=__" operator is used to search for files where the value of the specified field does not match the specified value.

The "__!=__" operator will not match a field that has no value (i.e. a field that is empty). For example:
	
		cases.demographic.gender != male

This search will only match cases who have a gender and the gender is not male. To find cases other than male or with no gender populated, you would need to search:

		cases.demographic.gender != male or cases.demographic.gender is missing.

Example:

*   Find all files with an experimental strategy that is not genotyping array:

		files.experimental_strategy != "Genotyping array"


### "__>__" Operator - __GREATER THAN__

The "__>__" operator is used to search for files where the value of the specified field is greater than the specified value.

Example:

* Find all cases whose number of days to death is greater than 60:

		cases.diagnoses.days_to_death > 60


### "__>=__" Operator - __GREATER THAN OR EQUALS__

The "__>=__" operator is used to search for files where the value of the specified field is greater than or equal to the specified value.

Example:

* Find all cases whose number of days to death is equal or greater than 60:

		cases.diagnoses.days_to_death >= 60

### "__<__" Operator - __LESS THAN__

The "__<__" operator is used to search for files where the value of the specified field is less than the specified value.

Example:

*   Find all cases whose age at diagnosis is less than 400 days:

		cases.diagnoses.age_at_diagnosis < 400


### "__<=__" Operator - __LESS THAN OR EQUALS__

The "__<=__" operator is used to search for files where the value of the specified field is less than or equal to the specified value.

Example:

*   Find all cases with a number of days to death less than or equal to 20:

		cases.diagnoses.days_to_death <= 20


### "__IN__" Operator

The "__IN__" operator is used to search for files where the value of the specified field is one of multiple specified values. The values are specified as a comma-delimited list, surrounded by brackets [ ].

Using "__IN__" is equivalent to using multiple "__=__" (__EQUALS__) statements, but is shorter and more convenient. That is, these two following statement will retrieve the same output:

		cases.project.name IN [ProjectA, ProjectB, ProjectC] 
		cases.project.name = "ProjectA" OR cases.project.name = "ProjectB" OR cases.project.name = "ProjectC"

Examples:

*   Find all files in breast, brain, and lung cancer:

		cases.primary_site IN [Breast, Brain, Lung]

*   Find all files that are annotated somactic mutations or raw simple somatic mutations:

		files.data_type IN ["Annotated Somatic Mutation", "Raw Simple Somatic Mutation"]


### "__EXCLUDE__" Operator

The "__EXCLUDE__" operator is used to search for files where the value of the specified field is not one of multiple specified values.

Using "__EXCLUDE__" is equivalent to using multiple "__!=__" (__NOT_EQUALS__) statements, but is shorter and more convenient. That is, these two following statement will retrieve the same output:

		cases.project.name EXCLUDE [ProjectA, ProjectB, ProjectC]
		cases.project.name != "ProjectA" OR cases.project.name != "ProjectB" OR cases.project.name != "ProjectC"

The "__EXCLUDE__" operator will not match a field that has no value (i.e. a field that is empty). For example:

		files.experimental_strategy EXCLUDE ["WGS","WXS"] 

This search will only match files that have an experimental strategy **and** the experimental strategy is not "WGS" or "WXS". To find files with an experimental strategy different than "WGS" or "WXS" **or is not assigned**, you would need to type:

		files.experimental_strategy in ["WXS","WGS"] or files.experimental_strategy is missing

Examples:

*   Find all files where experimental strategy is not WXS, WGS, Genotyping array:

		files.experimental_strategy EXCLUDE [WXS, WGS, "Genotyping array"]

### "__IS MISSING__" Operator

The "__IS__" operator can only be used with "__MISSING__". That is, it is used to search for files where the specified field has no value.

Examples:

*   Find all cases where gender is missing:

		cases.demographic.gender is MISSING

### "__NOT MISSING__" Operator

The "__NOT__" operator can only be used with "__MISSING__". That is, it is used to search for files where the specified field has a value.

Examples:

*   Find all cases where race is not missing:

		cases.demographic.race NOT MISSING

## Special Cases

### Date Format

The date format should be the following: **YYYY-MM-DD** (without quotes).

Example:

		files.updated_datetime > 2015-12-31


### Using Quotes

A value must be quoted if it contains a space. Otherwise the advanced search will not be able to interpret the value.

Quotes are not necessary if the value consists of one single word.

* Example: Find all cases with primary site is brain and data type is copy number segment:

		cases.primary_site = Brain and files.data_type = "Copy Number Segment"

### Age at Diagnosis - Unit in Days

The unit for age at diagnosis is in **days**. The user has to convert the number of years to number of days.

The __conversion factor__ is 1 year = 365.25 days

*   Example: Find all cases whose age at diagnosis > 40 years old (40 * 365.25)

		cases.diagnoses.age_at_diagnosis > 14610



## Fields Reference

The full list of fields available on the GDC Data Portal can be found through the GDC API using the following endpoint:

[https://api.gdc.cancer.gov/gql/_mapping](https://api.gdc.cancer.gov/gql/_mapping)

# Authentication

## Overview

The GDC Data Portal provides granular metadata for all datasets available in the GDC. Any user can see a listing of all available data files, including controlled-access files. The GDC Data Portal also allows users to download open-access files without logging in. However, downloading of controlled-access files is restricted to authorized users and requires authentication.

## Logging into the GDC

To login to the GDC, users must click on the `Login` button on the top right of the GDC website.

![Login](images/gdc-login.png)

After clicking Login, users authenticate themselves using their eRA Commons login and password.  If authentication is successful, the eRA Commons username will be displayed in the upper right corner of the screen, in place of the "Login" button.

Upon successful authentication, GDC Data Portal users can:

- see which controlled-access files they have access to;
- download controlled-access files directly from the GDC Data Portal;
- download an authentication token for use with the GDC Data Transfer Tool or the GDC API.

Controlled-access files are identified using a "lock" icon:

[![GDC Data Portal Main Page](images/gdc-data-portal-controlled-files.png)](images/gdc-data-portal-controlled-files.png "Click to see the full image.")

The rest of this section describes controlled data access features of the GDC Data Portal available to authorized users. For more information about open and controlled-access data, and about obtaining access to controlled data, see [Data Access Processes and Tools](https://gdc.cancer.gov/access-data/data-access-processes-and-tools).

## User Profile

After logging into the GDC Portal, users can view which projects they have access to by clicking the `User Profile` section in the dropdown menu in the top corner of the screen.

[![User Profile Drop Down](images/gdc-user-profile-dropdown.png)](images/gdc-user-profile-dropdown.png "Click to see the full image.")

Clicking this button shows the list of projects.

[![User Profile](images/gdc-user-profile.png)](images/gdc-user-profile.png "Click to see the full image.")

## GDC Authentication Tokens

The GDC Data Portal provides authentication tokens for use with the GDC Data Transfer Tool or the GDC API. To download a token:

1. Log into the GDC using your eRA Commons credentials
2. Click the username in the top right corner of the screen
3. Select the "Download token" option

![Token Download Button](images/gdc-data-portal-token-download.png)

A new token is generated each time the `Download Token` button is clicked.

For more information about authentication tokens, see [Data Security](../../Data/Data_Security/Data_Security.md#authentication-tokens).

**NOTE:** The authentication token should be kept in a secure location, as it allows access to all data accessible by the associated user account.

## Logging Out

To log out of the GDC, click the username in the top right corner of the screen, and select the Logout option.

![Logout link](images/gdc-data-portal-token-download.png)


# Analysis

In addition to the [Exploration Page](Exploration.md), the GDC Data Portal also has features used to save and compare sets of cases, genes, and mutations. These sets can either be generated with existing filters (e.g. males with lung cancer) or through custom selection (e.g. a user-generated list of case IDs).

Note that saving a set only saves the type of entity included in the set. For example, a saved case set will not include filters that were applied to genes or mutations. Please be aware that your custom sets are deleted during each new GDC data release. You can export them and re-upload them in the "Manage Sets" link at the top right of the Portal.

## Generating a Cohort for Analysis

Cohort sets are completely customizable and can be generated for cases, genes, or mutations using the following methods:

__Apply Filters in Exploration:__ Sets can be assembled using the existing filters in the Exploration page.  They can be saved by choosing the "Save/Edit Case Set" button under the pie charts for case sets. This will prompt a decision to save as new case set. The same can be done for both gene and mutation filters, and can be applied and saved in the Genes and Mutations tab, respectively.

[![Exploration Set](images/GDC-ExplorationSet-Cohort_v2.png)](images/GDC-ExplorationSet-Cohort_v2.png "Click to see the full image.")

__Upload ID Set:__ This feature is available in the "Manage Sets" link at the top right of the Portal. Choose "Upload Set" and then select whether the set comprises cases, genes, or mutations. A set of IDs or UUIDs can then be uploaded in a text file or copied and pasted into the list of identifiers field along with a name identifying the set. Once the list of identifiers is uploaded, the IDs are validated and grouped according to whether or not the identifier matched an existing GDC ID.

[![Upload Set](images/GDC-UploadSet-Cohort_v2.png)](images/GDC-UploadSet-Cohort_v2.png "Click to see the full image.")

### Upload Case Set

In the `Cases` filters panel, instead of supplying cases one-by-one, users can supply a list of cases.  Clicking on the `Upload Case Set` button will launch a dialog as shown below, where users can supply a list of cases or upload a comma-separated text file of cases.

[![Upload Case Set](images/gdc-exploration-case-set.png)](images/gdc-exploration-case-set.png "Click to see the full image.")

After supplying a list of cases, a table below will appear which indicates whether the case was found.

[![Upload Case Set Validation](images/gdc-exploration-case-set-validation.png)](images/gdc-exploration-case-set-validation.png "Click to see the full image.")

Clicking on `Submit` will filter the results in the Exploration Page by those cases.

[![Upload Case Set Results](images/case-set-filter_v3.png)](images/case-set-filter_v2.png "Click to see the full image.")

### Upload Gene Set

In the `Genes` filters panel, instead of supplying genes one-by-one, users can supply a list of genes.  Clicking on the `Upload Gene Set` button will launch a dialog as shown below, where users can supply a list of genes or upload a comma-separated text file of genes.

[![Upload Gene Set](images/Exploration-Upload-Gene-Set.png)](images/Exploration-Upload-Gene-Set.png "Click to see the full image.")

After supplying a list of genes, a table below will appear which indicates whether the gene was found.

[![Upload Gene Set Validation](images/Exploration-Upload-Gene-Set-Validation.png)](images/Exploration-Upload-Gene-Set-Validation.png "Click to see the full image.")

Clicking on `Submit` will filter the results in the Exploration Page by those genes.

### Upload Mutation Set

In the `Mutations` filters panel, instead of supplying mutation id's one-by-one, users can supply a list of mutations.  Clicking on the `Upload Mutation Set` button will launch a dialog as shown below, where users can supply a list of mutations or upload a comma-separated text file of mutations.

[![Upload Case Set](images/gdc-exploration-mutation-set.png)](images/gdc-exploration-mutation-set.png "Click to see the full image.")

After supplying a list of mutations, a table below will appear which indicates whether the mutation was found.

[![Upload Case Set Validation](images/gdc-exploration-mutation-set-validation.png)](images/gdc-exploration-mutation-set-validation.png "Click to see the full image.")

Clicking on `Submit` will filter the results in the Exploration Page by those mutations.

[![Upload Case Set Results](images/mutation-set-filter_v2.png)](images/mutation-set-filter_v2.png "Click to see the full image.")

## Analysis Page
Clicking on the `Analysis` button in the top toolbar will launch the Analysis Page which displays the various options available for comparing saved sets.

[![Analysis Tab](images/GDC-Analysis-Tab_v2.png)](images/GDC-Analysis-Tab_v2.png "Click to see the full image.")

There are three tabs on this page:

* __Launch Analysis__: Where users can select either to do `Set Operations`, `Cohort Comparison` or `Clinical Data Analysis`.
* __Results__: Where users can view the results of current or previous set analyses.

## Analysis Page: Set Operations

Up to three sets of the same set type can be compared and exported based on complex overlapping subsets. The features of this page include:

[![Set Operations](images/GDC-SetOpsFull-Cohort.png)](images/GDC-SetOpsFull-Cohort.png "Click to see the full image.")

* __Venn Diagram:__ Visually displays the overlapping items included within the three sets. Subsets based on overlap can be selected by clicking one or many sections of the Venn diagram. As sections of the Venn Diagram become highlighted in blue, their corresponding row in the overlap table becomes highlighted.  

* __Summary Table:__ Displays the alias, item type, and name for each set included in this analysis.

* __Overlap Table:__ Displays the number of overlapping items with set operations rather than a visual diagram. Subsets can be selected by checking boxes in the "Select" column, which will highlight the corresponding section of the Venn Diagram. As rows are selected, the "Union of selected sets" row is populated. Each row has an option to save the subset as a new set,  export the set as a TSV, or view files in the repository. The links that correspond to the number of items in each row will open the cohort in the Exploration Page.  

## Analysis Tab: Cohort Comparison

The "Cohort Comparison" analysis displays a series of graphs and tables that demonstrate the similarities and differences between two case sets. The following features are displayed for each two sets:

* A key detailing the number of cases in each cohort and the color that represents each (blue/gold).

* A Venn diagram, which shows the overlap between the two cohorts.  The Venn diagram can be opened in a 'Set Operations' tab by choosing "Open Venn diagram in new tab".

* A selectable [survival plot](Exploration.md#survival-analysis) that compares both sets with information about the percentage of represented cases.

[![Top Cohort](images/GDC-Cohort-Comparison-Top_v2.png)](images/GDC-Cohort-Comparison-Top_v2.png "Click to see the full image.")

* A breakdown of each cohort by selectable clinical facets with a bar graph and table. Facets include `vital_status`, `gender`, `race`, `ethnicity`, and `age_at_diagnosis`.  A p-value (if it can be calculated from the data) that demonstrates whether the statuses are proportionally represented is displayed for the `vital_status`, `gender`, and `ethnicity` facets.  

[![Clinical Cohort](images/GDC-Clinical-Cohort_v2.png)](images/GDC-Clinical-Cohort_v2.png "Click to see the full image.")

## Analysis Tab: Clinical Data Analysis

The "Clinical Data Analysis" feature allows users to specifically examine the clinical data of a single case set in more detail.  Users can select which clinical fields they want to display and visualize the data using various supported plot types.  The clinical analysis features include:

* Ability to select which clinical fields to display
* Examine the clinical data of each field using these visualizations:
    * Histogram
    * Survival Plot
    * Box Plot
    * QQ Plot
* Create custom bins for each field and re-visualize the data with those bins
* Select specific cases from a clinical field and use them to create a new set, or modify/remove from an existing set
* Download the visualizations of each plot type for each variable in SVG, PNG, JSON formats
* Download the data table of each field in TSV format
* Print all clinical variable cards in the analysis with their active plot to a single PDF

### Selecting a Case Set

First, a case set must be selected to run the clinical analysis on:

[![Select Clinical Cohort](images/GDC-Select-Clinical-Cohort.png)](images/GDC-Select-Clinical-Cohort.png "Click to see the full image.")

Click __'Run'__ - The results page loads with a new tab for the new clinical analysis for the selected set:

[![Select Clinical Cohort](images/GDC-Load-Clinical-Analysis.png)](images/GDC-Load-Clinical-Analysis.png "Click to see the full image.")

### Enabling Clinical Variable Cards

Users can use the control panel on the left side of the analysis to display which clinical variables they want.  To enable or disable specific variables for display, click the on/off toggle controls:

[![Enable Clinical Cards](images/GDC-Enable-Clinical-Cards.png)](images/GDC-Enable-Clinical-Cards.png "Click to see the full image.")

The clinical fields are grouped into these categories:

* __Demographic:__ Data for the characterization of the patient by means of segmenting the population (e.g. characterization by age, sex, race, etc.).
* __Diagnoses:__ Data from the investigation, analysis, and recognition of the presence and nature of disease, condition, or injury from expressed signs and symptoms; also, the scientific determination of any kind; the concise results of such an investigation.
* __Treatments:__ Records of the administration and intention of therapeutic agents provided to a patient to alter the course of a pathologic process.
* __Exposures:__ Clinically-relevant patient information not immediately resulting from genetic predispositions.

Since the list of fields can be long, users can collapse and expand the field list for each clinical category for easier browsing, or use the search box:

[![Search Clinical Fields](images/GDC-Search-Clinical-Fields.png)](images/GDC-Search-Clinical-Fields.png "Click to see the full image.")

### Exploring Clinical Card Visualizations

Users can explore different visualizations for each clinical field they have enabled for display. Each card supports theses plot types:

* Histogram
* Survival Plot
* Box Plot & QQ Plot (these plots are visualized side-by-side)

To switch between plot types, click the different plot type icons in the top-right of each card.

#### Histogram

The histogram plot type suppports these features:

* View the distribution of cases (# and % of cases) in the cohort for the clinical field's data categories as a histogram
* View the distribution of cases in tabular format
* Select the cases for specific data categories to create new sets, append to existing sets, or remove from existing sets
* Download the histogram visualization in SVG or PNG format
* Download the raw data used to generate the histogram in JSON format

[![Clinical Analysis Histogram](images/GDC-Clinical-Analysis-Histogram.png)](images/GDC-Clinical-Analysis-Histogram.png "Click to see the full image.")

Note that the histogram plot applies to, and can be displayed for, both categorical and continuous variables.

#### Survival Plot

The survival plot type supports these features:

* View the distribution of cases (# and % of cases) in the cohort for the clinical field's data categories as a table
* Select and plot the survival analysis for the cases of specific data categories in the table:
    * By default the top 2 categories (highest # of cases) are displayed
    * Users can manually select and plot up to 5 categories at a time
* Download the survival plot visualization in SVG or PNG format
* Download the raw data used to generate the survival plot in JSON or TSV format

[![Clinical Analysis Survival Plot](images/GDC-Clinical-Analysis-Survival-Plot.png)](images/GDC-Clinical-Analysis-Survival-Plot.png "Click to see the full image.")

Note that the survival plot applies to, and can be displayed for, both categorical and continuous variables.

#### Box Plot & QQ Plot

The box plot and QQ plot are displayed side-by-side in the same visualization.  This visualization supports these features:

* View standard summary statistics for the clinical field's data across the cohort as both a box plot visualization and a data table:
    * Minimum
    * Maximum
    * Mean
    * Median
    * Standard Deviation
    * Interquartile Range (IQR)
* View a QQ plot visualization to explore whether the clinical field's data across the cohort is normally distributed, where:
    * The clinical data values are plotted as the sample quantiles on the vertical axis
    * The the quantiles of the normal distribution are plotted on the horizontal axis
* Download the box plot and QQ plot visualizations in SVG or PNG format
* Download the raw data used to generate the QQ plot in JSON or TSV format

[![Clinical Analysis Box & QQ Plots](images/GDC-Clinical-Analysis-Box-And-QQ-Plots.png)](images/GDC-Clinical-Analysis-Box-And-QQ-Plots.png "Click to see the full image.")

Note that the box plot and QQ plot only apply to continuous variables.  They cannot be displayed for categorical variables.

### Creating Custom Bins

For each clinical variable, whether categorical or continuous, users can create custom bins to group the data in ways they find scientifically interesting or significant.  Once saved, the bins are applied to these visualizations and they are then re-rendered:

* Histogram and associated data table
* Survival plot and associated data table

Custom bins can be reset to their defaults at any time for each card.  Note that custom bins are __saved per analysis__.

#### Categorical Binning

To create custom bins for a categorical variable, click *__Customize Bins__*, then *__Edit Bins__*.  A configuration window appears where the user can create their bins:

[![Clinical Analysis Categorical Bins](images/GDC-Clinical-Analysis-Categorical-Bins.png)](images/GDC-Clinical-Analysis-Categorical-Bins.png "Click to see the full image.")

The user can:

* Group existing individual values into a single group
* Give a custom name to each group
* Ungroup previously grouped values
* Completely hide values from being shown in the visualization
* Re-show previously hidden values

#### Continuous Binning

To create custom bins for a continuous variable, click *__Customize Bins__*, then *__Edit Bins__*.  A configuration window appears where the user can create their bins:

[![Clinical Analysis Continuous Bins](images/GDC-Clinical-Analysis-Continuous-Bins.png)](images/GDC-Clinical-Analysis-Continuous-Bins.png "Click to see the full image.")

The user can choose one of these continuous binning methods:

* (1) Create equi-distant bins based on a set interval:
    * User must choose the interval (e.g. equi-distant bins of 1,825 days for the Age of Diagnosis field)
    * User can optionally define the starting and ending value between which the equi-distant bins will be created
* (2) Create completely custom ranges:
    * User manually enters 1 or more bins with custom ranges
    * User must enter a name for each range and the start and end values
    * The ranges can be of different interval lengths

Before saving the bins, if there are errors in the configuration, the user will be notified to correct them and try saving again.  For example:

[![Clinical Analysis Continuous Bins Error Example 1](images/GDC-Clinical-Analysis-Continuous-Bins-Error-Example1.png)](images/GDC-Clinical-Analysis-Continuous-Bins-Error-Example1.png "Click to see the full image.")

[![Clinical Analysis Continuous Bins Error Example 2](images/GDC-Clinical-Analysis-Continuous-Bins-Error-Example2.png)](images/GDC-Clinical-Analysis-Continuous-Bins-Error-Example2.png "Click to see the full image.")

### Other Useful Functions

Clinical Analysis also provides these additional useful functions:

* Like other analysis types, all Clinical Analysis tabs are saved to the browser's local storage:
    * Each Analysis tab and its associated configurations (active cards, active plots, custom bins) is saved and is not deleted until local storage is cleared
    * The currently-enabled clinical cards and their currently-selected plot types are __saved per analysis__
    * Custom bins are __saved per analysis__
* Switch the current set that the analysis applies to - This does the following:
    * Applies all currently-enabled clinical cards and their currently-selected plot types to the data in the set being switched to
    * Re-renders all active visualizations to reflect the data in the set being switched to
* Rename your analysis with a custom name
* Copy your current analysis to a new analysis:
    * User is prompted to name the new copy
    * All currently-enabled clinical cards and their currently-selected plot types are copied to the new analysis
    * A new vertical tab appears on the left for the new copy
* Print the current analysis to PDF format:
    * All currently-enabled clinical cards and their currently-selected plot types are printed
    * 3 cards per page, with the fixed Overall Survival Plot displayed as the first card in the entire file

## Analysis Page: Results

The results of the previous analyses are displayed on this page.

[![Results](images/gdc-analysis-resultstab_v2.png)](images/gdc-analysis-resultstab_v2.png "Click to see the full image.")

Each tab at the left side of the page is labeled according to the analysis type and the date that the analysis was performed and can be reviewed as long as it is present. The "Delete All" button will remove all of the previous analyses.  


# Annotations

Annotations are notes added to individual cases, samples or files.

## Annotations View

The Annotations View provides an overview of the available annotations and allows users to browse and filter the annotations based on a number of annotation properties (facets), such as the type of entity the annotation is attached to or the annotation category.

The view presents a list of annotations in tabular format on the right, and a facet panel on the left that allows users to filter the annotations displayed in the table. If facet filters are applied, the tabs on the right will display only the matching annotations. If no filters are applied, the tabs on the right will display information about all available data.

Clicking on an annotation ID in the annotations list will take the user to the [Annotation Detail Page](#annotation-detail-page).

[![Annotations View](images/gdc-data-portal-annotations.png)](images/gdc-data-portal-annotations.png "Click to see the full image.")

### Facets Panel

The following facets are available to search for annotations:

* __Annotation ID__: Seach using annotation ID
* __Entity ID__: Seach using entity ID
* __Case UUID__: Seach using case UUID
* __Primary Site__: Anatomical site of the cancer
* __Project__: A cancer research project, typically part of a larger cancer research program
* __Entity Type__: The type of entity the annotation is associated with: Patient, Sample, Portion, Slide, Analyte, Aliquot
* __Annotation Category__: Search by annotation category.
* __Annotation Created__: Search for annotations by date of creation.
* __Annotation Classification__: Search by annotation classification.

#### Annotation Categories and Classification

For more details about categories and classifications please refer to the [TCGA Annotations page on NCI Wiki](https://wiki.nci.nih.gov/display/TCGA/Introduction+to+Annotations).

## Annotation Detail Page

The annotation entity page provides more details about a specific annotation. It is available by clicking on an annotation ID in Annotations View.

[![Annotation Entity Page](images/annotations-entity-page.png)](images/annotations-entity-page.png "Click to see the full image.")


# Repository

The Repository Page is the primary method of accessing data in the GDC Data Portal. It provides an overview of all cases and files available in the GDC and offers users a variety of filters for identifying and browsing cases and files of interest. Users can access the [Repository Page](https://portal.gdc.cancer.gov/v1/repository) from the GDC Data Portal Home Page or from the Data Portal toolbar.

## Filters / Facets
On the left, a panel of data facets allows users to filter cases and files using a variety of criteria. If facet filters are applied, the tabs on the right will display information about matching cases and files. If no filters are applied, the tabs on the right will display information about all available data.

On the right, two tabs contain information about available data:

* `Files` tab provides a list of files, select information about each file, and links to [individual file detail pages](#file-summary-page).
* `Cases` tab provides a list of cases, select information about each case, and links to [individual case summary pages](Exploration.md#case-summary-page).

The banner above the tabs on the right displays any active facet filters and provides access to advanced search.

The top of the Repository Page, in the "Files" tab, contains a few summary pie charts for Primary Sites, Projects, Data Category, Data Type, and Data Format. These reflect all available data or, if facet filters are applied, only the data that matches the filters. Clicking on a specific slice in a pie chart, or on a number in a table, applies corresponding facet filters. The scope of these pie chart will change depending on whether you have the "Files" tab or the "Cases" tab selected.

[![Data View](images/gdc-data-portal-repository-view_v2.png)](images/gdc-data-portal-repository-view_v2.png "Click to see the full image.")

### Facets Panel

Facets represent properties of the data that can be used for filtering. The facets panel on the left allows users to filter the cases and files presented in the tabs on the right.

The facets panel is divided into two tabs, with the `Files` tab containing facets pertaining to data files and experimental strategies, while the `Cases` tab containing facets pertaining to the cases and biospecimen information. Users can apply filters in both tabs simultaneously. The applied filters will be displayed in the banner above the tabs on the right, with the option to open the filter in [Advanced Search](Advanced_Search.md) to further refine the query.

[![Facet Filters Applied in Data View](images/data-view-with-facet-filters-applied_v2.png)](images/data-view-with-facet-filters-applied_v2.png "Click to see the full image.")

The default set of facets is listed below.

*Files* facets tab:

* __File__: Specify individual files using filename or UUID.
* __Data Category__: A high-level data file category, such as "Raw Sequencing Data" or "Transcriptome Profiling".
* __Data Type__: Data file type, such as "Aligned Reads" or "Gene Expression Quantification". Data Type is more granular than Data Category.
* __Experimental Strategy__: Experimental strategies used for molecular characterization of the cancer.
* __Workflow Type__: Bioinformatics workflow used to generate or harmonize the data file.
* __Data Format__: Format of the data file.
* __Platform__: Technological platform on which experimental data was produced.
* __Access Level__: Indicator of whether access to the data file is open or controlled.

*Cases* facets tab:

* __Case__: Specify individual cases using submitter ID (barcode) or UUID.
* __Case ID__: Search for cases using a part (prefix) of the submitter ID (barcode).
* __Primary Site__: Anatomical site of the cancer under investigation or review.
* __Program__: A cancer research program, typically consisting of multiple focused projects.
* __Project__: A cancer research project, typically part of a larger cancer research program.
* __Disease Type__: Type of cancer studied.
* __Gender__: Gender of the patient.
* __Age at Diagnosis__: Patient age at the time of diagnosis.
* __Vital Status__: Indicator of whether the patient was living or deceased at the date of last contact.
* __Days to Death__: Number of days from date of diagnosis to death of the patient.
* __Race__: Race of the patient.
* __Ethnicity__: Ethnicity of the patient.

### Adding Custom Facets

The Repository Page provides access to additional data facets beyond the automatically listed group filters. Facets corresponding to additional properties listed in the [GDC Data Dictionary](../../Data_Dictionary/index.md) can be added using the "Add a Filter" link available at the top of the `Cases` and `Files` facet tabs:

[![Add a Facet](images/gdc-data-portal-data-add-facet.png)](images/gdc-data-portal-data-add-facet.png "Click to see the full image.")

The link opens a search window that allows the user to find an additional facet by name or description. Not all facets have values available for filtering; checking the "Only show fields with values" checkbox will limit the search results to only those that do. When selecting a facet from the list of search results below the search box will add it to the facets panel.

[![Search for a Facet](images/gdc-data-portal-data-facet-search.png)](images/gdc-data-portal-data-facet-search.png "Click to see the full image.")

Newly added facets will show up at the top of the facets panel and can be removed individually by clicking on the "__x__" to the right of the facet name. The default set of facets can be restored by clicking "Reset".

[![Customize Facet](images/gdc-data-portal-data-facet-tumor_stage.png)](images/gdc-data-portal-data-facet-tumor_stage.png "Click to see the full image.")

## Annotations View

The Annotations View provides an overview of the available annotations and allows users to browse and filter the annotations based on a number of annotation properties (facets), such as the type of entity the annotation is attached to or the annotation category. This page can be found by clicking on the [Browse Annotations](https://portal.gdc.cancer.gov/v1/annotations) link, located at the top right of the repository page.

[![Annotations View](images/Browse_Annotations.png)](images/Browse_Annotations.png "Click to see the full image.")

The view presents a list of annotations in tabular format on the right, and a facet panel on the left that allows users to filter the annotations displayed in the table. If facet filters are applied, the tabs on the right will display only the matching annotations. If no filters are applied, the tabs on the right will display information about all available annotations.

[![Annotations View](images/gdc-data-portal-annotations.png)](images/gdc-data-portal-annotations.png "Click to see the full image.")

Clicking on an annotation ID in the annotations list will take the user to the Annotation Summary Page. The Annotation Summary Page provides more details about a specific annotation.

[![Annotation Entity Page](images/annotations-entity-page.png)](images/annotations-entity-page.png "Click to see the full image.")

## Results

### Navigation

After utilizing the Repository Page to narrow down a specific set of cases, users can choose to continue to explore the mutations and genes affected by these cases by clicking the `View Cases in Exploration` button as shown in the image below.

[![Exploration File Navigation](images/gdc-view-in-exploration_v3.png)](images/gdc-view-in-exploration_v3.png "Click to see the full image.")

Clicking this button will navigate the users to the [Exploration Page](Exploration.md), filtered by the cases within the cohort.

### Files List

The `Files` tab on the right provides a list of available files and select information about each file. If facet filters are applied, the list includes only matching files. Otherwise, the list includes all data files available in the GDC Data Portal.

[![Files Tab](images/gdc-data-portal-data-files.png)](images/gdc-data-portal-data-files.png "Click to see the full image.")

The "*File Name*" column includes links to [File Summary Pages](#file-summary-page) where the user can learn more about each file.

Users can add individual file(s) to the [cart](Cart.md) using the cart button next to each file. Alternatively, all files that match the current facet filters can be added to the cart using the menu in the top left corner of the table:

[![Files Tab](images/gdc-data-portal-data-files-add-cart.png)](images/gdc-data-portal-data-files-add-cart.png "Click to see the full image.")

## File Summary Page

The File Summary page provides information about a data file, including file properties like size, MD5 checksum, and data format; information on the type of data included; links to the associated cases and biospecimen; and information about how the data file was generated or processed.

The page also includes buttons to download the file, add it to the file [cart](Cart.md), or (for BAM files) utilize the BAM slicing function.

[![Files Detail Page](images/gdc-data-portal-files-entity-page.png)](images/gdc-data-portal-files-entity-page.png "Click to see the full image.")

In the lower section of the screen, the following tables provide more details about the file and its characteristics:

* __Associated Cases / Biospecimen__: List of cases or biospecimen the file is directly attached to.
* __Analysis and Reference Genome__: Information on the workflow and reference genome used for file generation.
* __Read Groups__: Information on the read groups associated with the file.
* __Metadata Files__: Experiment metadata, run metadata and analysis metadata associated with the file.
* __Downstream Analysis Files__: List of downstream analysis files generated by the file.
* __File Versions__: List of all versions of the file.


[![Files Entity Page](images/gdc-data-portal-files-entity-page-part2_v2.png)](images/gdc-data-portal-files-entity-page-part2_v2.png "Click to see the full image.")

### BAM Slicing

BAM file Summary Pages have a "BAM Slicing" button. This function allows the user to specify a region of a BAM file for download. Clicking on it will open the BAM Slicing window:

[![BAM Slicing Window](images/gdc-data-portal-bam-slicing_v2.png)](images/gdc-data-portal-bam-slicing_v2.png "Click to see the full image.")

During preparation of the slice, the icon on the BAM Slicing button will be spinning, and the file will be offered for download to the user as soon as it is ready.

### Cases List

The `Cases` tab on the right provides a list of available cases and select information about each case. If facet filters are applied, the list includes only matching cases. Otherwise, the list includes all cases available in the GDC Data Portal.

[![Cases Tab](images/gdc-data-portal-data-cases_v3.png)](images/gdc-data-portal-data-cases_v3.png "Click to see the full image.")

From the left side, the list starts with a shopping cart icon, allowing the user to add all files associated with a case to the [file cart](Cart.md) for downloading at a later time. The following columns in the list includes links to [Case Summary Pages](Exploration.md#case-summary-page) in the *Case UUID* column, the Submitter ID (i.e. TCGA Barcode), and counts of the available file types for each case. Clicking on a count will apply facet filters to display the corresponding files. On the last column, there are image slide icons and a number that indicate whether there are slide images available and how many.

## Image Viewer

The Image Viewer allows users to visualize tissue and diagnostic slide images.

[![Image Viewer](images/Image_viewer_browser.png)](images/Image_viewer_browser.png "Click to see the full image.")

### How to Access the Image Viewer

* __Repository Page__: From the main search on the Repository Page by clicking on the "View images" button. It will display the tissue slide images of all the cases resulting from the query.

[![Image Viewer](images/Image_Viewer_from_Repository.png)](images/Image_Viewer_from_Repository.png "Click to see the full image.")

* __Case Table in Repository Page__: Click on the image viewer icon in the Case table. It will display in the image viewer all the tissue slide images attached to the Case.

[![Cases Tab](images/gdc-data-portal-data-cases_v3.png)](images/gdc-data-portal-data-cases_v3.png "Click to see the full image.")

* __Case Summary Page:__ Selecting a Case ID in the Repository Cases table will direct the user to the [Case Summary Page](Exploration.md#case-summary-page). For cases with images, the Image Viewer icon will appear in the Case Summary section or in the Biospecimen - Slides details section. Clicking on the Image Viewer icon will display the Image Viewer for the slide images attached to the case.

 [![Image Viewer](images/Image_viewer_case_summary.png)](images/Image_viewer_case_summary.png "Click to see the full image.")
 [![Image Viewer](images/Image_viewer_case_slide_section.png)](images/Image_viewer_case_slide_section.png "Click to see the full image.")

* __The Image File Page__: You can visualize the slide image directly in the File Summary Page by selecting an image file in the Repository's files table.

[![Image Viewer](images/Repository_select_image.png)](images/Repository_select_image.png "Click to see the full image.")

[![Image Viewer](images/Image_viewer_File_entity.png)](images/Image_viewer_File_entity.png "Click to see the full image.")

### Image Viewer Features
In the image viewer, a user can:

* Zoom in and zoom out by clicking on + and - icons.
* Reset to default display by clicking on the Home icon.
* Display the image in full screen mode by clicking on the Expand icon.
* View the slide detail by clicking on "Details" button.
* Selecting the area of interest with the thumbnail at the top-right corner.

[![Image Viewer](images/Image_viewer_features.png)](images/Image_viewer_features.png "Click to see the full image.")


# Projects

At a high level, data in the Genomic Data Commons is organized by project. Typically, a project is a specific effort to look at particular type(s) of cancer undertaken as part of a larger cancer research program. The GDC Data Portal allows users to access aggregate project-level information via the Projects Page and Project Summary Pages.

## Projects Page

The Projects Page provides an overview of all harmonized data available in the Genomic Data Commons, organized by project. It also provides filtering, navigation, and advanced visualization features that allow users to identify and browse projects of interest. Users can access the [Projects Page](https://portal.gdc.cancer.gov/v1/projects) from the GDC Data Portal Home page or from the Data Portal toolbar.

On the left, a panel of facets allow users to apply filters to find projects of interest. When facet filters are applied, the table and visualizations on the right are updated to display only the matching projects. When no filters are applied, all projects are displayed.

The right side of the Projects Page displays a few visualizations of the data (Top Mutated Genes in Selected Projects and Case Distribution per Project). Below these graphs is a table that contains a list of projects and select details about each project, such as the number of cases and data files. The Graph tab provides a visual representation of this information.

[![Projects Page, Main Window (Table View)](images/gdc-data-portal-project-page_v3.png)](images/gdc-data-portal-project-page_v3.png "Click to see the full image.")

### Visualizations

[![Projects Visualizations)](images/gdc_project_visualizations3.png)](images/gdc_project_visualizations3.png "Click to see the full image.")

#### Top Mutated Cancer Genes in Selected Projects

This dynamically generated bar graph shows the 20 genes with the most mutations across all projects. The genes are filtered by those that are part of the Cancer Gene Census and that have the following types of mutations: `missense_variant`, `frameshift_variant`, `start_lost`, `stop_lost`, `initiator_codon_variant`, and `stop_gained`. The bars represent the frequency of mutations per gene and is broken down into different colored segments by project. The graphic is updated as filters are applied for projects, programs, disease types, and data categories available in the project.

> __Note:__ Due to these filters, the number of cases displayed here will be less that the total number of cases per project.

Hovering the cursor over each bar will display information about the number of cases affected by the disease type and clicking on each bar will launch the [Gene Summary Page](Exploration.md#gene-summary-page) for the gene associated with the mutation.

Users can toggle the Y-Axis of this bar graph between a percentage or raw number of cases affected.

#### Case Distribution per Project

A pie chart displays the relative number of cases for each project. Hovering the cursor over each portion of the graph will display the project with the number of associated cases. Filtering projects at the left panel will update the pie chart.

### Projects Table

The `Table` tab lists projects by Project ID and provides additional information about each project. If no facet filters have been applied, the table will display all available projects; otherwise it will display only those projects that match the selected criteria.

[![Projects Table)](images/gdc-projects-table-view_v2.png)](images/gdc-projects-table-view_v2.png "Click to see the full image.")

The table provides links to [Project Summary Pages](Projects.md#project-summary-page) in the Project ID column. Columns with file and case counts include links to open the corresponding files or cases in [Repository Page](Repository.md).

### Projects Graph

The `Graph` tab contains an interactive view of information in the Table tab. The numerical values in Case Count, File Count, and File Size columns are represented by bars of varying length according to size. These columns are sorted independently in descending order. Mousing over an element of the graph connects it to associated elements in other columns, including Project ID and major Primary Sites.

[![Graph Mouseover](images/gdc-table-graph-mouse-over.png)](images/gdc-table-graph-mouse-over.png "Click to see the full image.")

Most elements in the graph are clickable, allowing the user to open the associated cases or files in [Repository Page](Repository.md).

Like the projects table, the graph will reflect any applied facet filters.

### Facets Panel

Facets represent properties of the data that can be used for filtering. The facets panel on the left allows users to filter the projects presented in the Table and Graph tabs as well as visualizations.

[![Panel with Applied Filters](images/gdc-data-portal-project-page-facets4.png)](images/gdc-data-portal-project-page-facets4.png "Click to see the full image.")

Users can filter by the following facets:

*   __Project__: Individual project ID.
*   __Primary Site__: Anatomical site of the cancer under investigation or review.
*   __Program__: Research program that the project is part of.
*   __Disease Type__: Type of cancer studied.
*   __Data Category__: Type of data available in the project.
*   __Experimental Strategy__: Experimental strategies used for molecular characterization of the cancer.

Filters can be applied by selecting values of interest in the available facets, for example "WXS" and "RNA-Seq" in the "Experimental Strategy" facet and "Brain" in the "Primary Site" facet. When facet filters are applied, the Table and Graph tabs are updated to display matching projects, and the banner above the tabs  summarizes the applied filters. The banner allows the user to click on filter elements to remove the associated filters and includes a link to view the matching cases and files.

[![Panel with Applied Filters](images/panel-with-applied-filters.png)](images/panel-with-applied-filters.png "Click to see the full image.")

For information on how to use facet filters, see [Getting Started](Getting_Started.md#facet-filters).

## Project Summary Page

Each project has a Summary Page that provides an overview of all available cases, files, and annotations available. Clicking on the numbers in the summary table will display the corresponding data.

[![Project Summary Page](images/gdc-project-entity-page_v4.png)](images/gdc-project-entity-page_v4.png "Click to see the full image.")

Four buttons in the top right corner of the screen allow the user to explore or download the entire project dataset, along with the associated project metadata:

* __Explore Project Data__: Opens Exploration Page with summary project information.
* __Biospecimen__: Downloads biospecimen metadata associated with all cases in the project in either TSV or JSON format.
* __Clinical__: Downloads clinical metadata about all cases in the project in either TSV or JSON format.
* __Manifest__: Downloads a manifest for all data files available in the project. The manifest can be used with the GDC Data Transfer Tool to download the files.


# Gene and Mutation Summary Pages

Many parts of the GDC website contain links to Gene and Mutation summary pages. These pages display information about specific genes and mutations, along with visualizations and data showcasing the relationship between themselves, the projects, and cases within the GDC. The gene and mutation data that is visualized on these pages are produced from the Open-Access MAF files available for download on the GDC Portal.

## Gene Summary Page

Gene Summary Pages describe each gene with mutation data and provides results related to the analyses that are performed on these genes.  

### Summary

The summary section of the gene page contains the following information:

[![Gene Summary](images/GDC-Gene-Summary_v2.png)](images/GDC-Gene-Summary_v2.png "Click to see the full image.")

* __Symbol:__ The gene symbol
* __Name:__ Full name of the gene
* __Synonyms:__ Synonyms of the gene name or symbol, if available
* __Type:__ A broad classification of the gene
* __Location:__ The chromosome on which the gene is located and its coordinates
* __Strand:__ If the gene is located on the forward (+) or reverse (-) strand
* __Description:__ A description of gene function and downstream consequences of gene alteration
- __Annotation:__ A notation/link that states whether the gene is part of [The Cancer Gene Census](http://cancer.sanger.ac.uk/census/)

### External References

A list with links that lead to external databases with additional information about each gene is displayed here. These external databases include: [Entrez](https://www.ncbi.nlm.nih.gov/gquery/), [Uniprot](http://www.uniprot.org/), [Hugo Gene Nomenclature Committee](http://www.genenames.org/), [Online Mendelian Inheritance in Man](https://www.omim.org/), [Ensembl](http://may2015.archive.ensembl.org/index.html), and [CIViC](https://civicdb.org/home).

### Cancer Distribution

A table and two bar graphs (one for mutations, one for CNV events) show how many cases are affected by mutations and CNV events within the gene as a ratio and percentage. Each row/bar represents the number of cases for each project.  The final column in the table lists the number of unique mutations observed on the gene for each project.

[![Cancer Distribution](images/GDC-Gene-CancerDist_v2.png)](images/GDC-Gene-CancerDist_v2.png "Click to see the full image.")

### Protein Viewer

[![Protein Plot](images/GDC-Gene-ProteinGraph.png)](images/GDC-Gene-ProteinGraph.png "Click to see the full image.")

Mutations and their frequency across cases are mapped to a graphical visualization of protein-coding regions with a lollipop plot. Pfam domains are highlighted along the x-axis to assign functionality to specific protein-coding regions. The bottom track represents a view of the full gene length. Different transcripts can be selected by using the drop-down menu above the plot.  

The panel to the right of the plot allows the plot to be filtered by mutation consequences or impact. The plot will dynamically change as filters are applied.  Mutation consequence and impact is denoted in the plot by color.

Note: The impact filter on this panel will not display the annotations for alternate transcripts.

The plot can be viewed at different zoom levels by clicking and dragging across the x-axis, clicking and dragging across the bottom track, or double clicking the pfam domain IDs. The `Reset` button can be used to bring the zoom level back to its original position. The plot can also be exported as a PNG image, SVG image or as JSON formatted text by choosing the `Download` button above the plot.

### Most Frequent Mutations

The 20 most frequent mutations in the gene are displayed as a bar graph that indicates the number of cases that share each mutation.  

[![Gene MFM](images/GDC-Gene-MFM.png)](images/GDC-Gene-MFM.png "Click to see the full image.")

A table is displayed below that lists information about each mutation including:

* __DNA Change:__ The chromosome and starting coordinates of the mutation are displayed along with the nucleotide differences between the reference and tumor allele
* __Type:__ A general classification of the mutation
* __Consequences:__ The effects the mutation has on the gene coding for a protein (i.e. synonymous, missense, non-coding transcript)
* __# Affected Cases in Gene:__ The number of affected cases, expressed as number across all mutations within the Gene
* __# Affected Cases Across GDC:__ The number of affected cases, expressed as number across all projects. Choosing the arrow next to the percentage will expand the selection with a breakdown of each affected project
* __Impact:__ A subjective classification of the severity of the variant consequence. This determined using [Ensembl VEP](http://www.ensembl.org/info/genome/variation/predicted_data.html), [PolyPhen](http://genetics.bwh.harvard.edu/pph/), and [SIFT](http://sift.jcvi.org/). The categories are outlined [here](https://docs.gdc.cancer.gov/Data/File_Formats/MAF_Format/#impact-categories).

*Note: The Mutation UUID can be displayed in this table by selecting it from the drop-down represented by three parallel lines*

Clicking the `Open in Exploration` button will navigate the user to the Exploration page, showing the same results in the table (mutations filtered by the gene).

## Mutation Summary Page

  The Mutation Summary Page contains information about one somatic mutation and how it affects the associated gene. Each mutation is identified by its chromosomal position and nucleotide-level change.

### Summary

  [![Mutation Summary](images/GDC-Mutation-Summary_v2.png)](images/GDC-Mutation-Summary_v2.png "Click to see the full image.")

  - __ID:__ A unique identifier (UUID) for this mutation
  - __DNA Change:__ Denotes the chromosome number, position, and nucleotide change of the mutation
  - __Type:__ A broad categorization of the mutation
  - __Reference Genome Assembly:__ The reference genome in which the chromosomal position refers to
  - __Allele in the Reference Assembly:__ The nucleotide(s) that compose the site in the reference assembly
  - __Functional Impact:__ A subjective classification of the severity of the variant consequence.

#### External References

  A separate panel contains links to databases that contain information about the specific mutation. These include [dbSNP](https://www.ncbi.nlm.nih.gov/projects/SNP/), [COSMIC](http://cancer.sanger.ac.uk/cosmic), and [CIViC](https://civicdb.org/home).

### Consequences

The consequences of the mutation are displayed in a table.  The set of consequence terms, defined by the [Sequence Ontology](http://www.sequenceontology.org).

  [![Mutation Consequences](images/GDC-Mutation-Consequences.png)](images/GDC-Mutation-Consequences.png "Click to see the full image.")

The fields that describe each consequence are listed below:

  * __Gene:__ The symbol for the affected gene
  * __AA Change:__ Details on the amino acid change, including compounds and position, if applicable
  * __Consequence:__ The biological consequence of each mutation
  * __Coding DNA Change:__ The specific nucleotide change and position of the mutation within the gene
  * __Strand:__ If the gene is located on the forward (+) or reverse (-) strand
  * __Transcript(s):__ The transcript(s) affected by the mutation. Each contains a link to the [Ensembl](https://www.ensembl.org) entry for the transcript   

### Cancer Distribution

A table and bar graph shows how many cases are affected by the particular mutation. Each row/bar represents the number of cases for each project.

  [![Mutation Distribution](images/GDC-Mutation-CancerDist.png)](images/GDC-Mutation-CancerDist.png "Click to see the full image.")

The table contains the following fields:

  * __Project ID__: The ID for a specific project
  * __Disease Type__: The disease associated with the project
  * __Site__: The anatomical site affected by the disease
  * __# Affected Cases__: The number of affected cases and total number of cases displayed as a fraction and percentage

### Protein Viewer

  [![Mutation Protein Graph](images/GDC-Mutation-ProteinGraph.png)](images/GDC-Mutation-ProteinGraph.png "Click to see the full image.")

  The protein viewer displays a plot representing the position of mutations along the polypeptide chain. The y-axis represents the number of cases that exhibit each mutation, whereas the x-axis represents the polypeptide chain sequence. [Pfam domains](http://pfam.xfam.org/) that were identified along the polypeptide chain are identified with colored rectangles labeled with pfam IDs. See the Gene Summary Page for additional details about the protein viewer.

  The panel to the right of the plot allows the plot to be filtered by mutation consequences or impact.  The plot will dynamically change as filters are applied.  Mutation consequence and impact is denoted in the plot by color.

  Note: The impact filter on this panel will not display the annotations for alternate transcripts.

  The plot can be viewed at different zoom levels by clicking and dragging across the x-axis, clicking and dragging across the bottom track, or double clicking the pfam domain IDs. The `Reset` button can be used to bring the zoom level back to its original position. The plot can also be exported as a PNG image, SVG image or as JSON formatted text by choosing the `Download` button above the plot.


# Getting Started

## The GDC Data Portal v1.0: An Overview

The Genomic Data Commons (GDC) Data Portal provides users with web-based access to data from cancer genomics studies. Key GDC Data Portal features include:

*   Open, granular access to information about all datasets available in the GDC.
*   Advanced search and visualization-assisted filtering of data files.
*   Data visualization tools to support the analysis and exploration of data (including on a gene and mutation level from Open-Access MAF files).
*   Cart for collecting data files of interest.
*   Authentication using eRA Commons credentials and auathorization using dbGaP for access to controlled data files.
*   Secure data download directly from the cart or using the [GDC Data Transfer Tool](https://gdc.cancer.gov/access-data/gdc-data-transfer-tool).

For more information about available datasets, see the [GDC Website](https://gdc.cancer.gov/about-data).



## Accessing the GDC Data Portal

The GDC Data Portal is accessible using a web browser such as Chrome, Firefox, and Microsoft Edge at the following URL:

[https://portal.gdc.cancer.gov](https://portal.gdc.cancer.gov)

The front page displays a summary of all available datasets:

[![GDC Home Page](images/GDC-Home-Page.png)](images/GDC-Home-Page.png "Click to see the full image.")


## Views

The GDC Data Portal provides five navigation options (*Views*) for browsing available harmonized datasets:

[![GDC Views](images/gdc-home-view-options.png)](images/gdc-home-view-options.png "Click to see the full image.")

* __Projects__: The Projects link directs users to the [Projects Page](Projects.md), which gives an overall summary of project-level information, including the available data for each project.

* __Exploration__: The Exploration link takes users to the [Exploration Page](Exploration.md), which allows users to explore data by utilizing various case, genes and mutation filters.

* __Analysis__: The Analysis link directs users to the [Analysis Page](Custom_Set_Analysis.md).  This page has features available for users to compare different cohorts or analyze the clinical variables of a specific cohort.  These cohorts can either be generated with existing filters (e.g. males with lung cancer) or through custom selection.

* __Repository__: The Repository link directs users to the [Repository Page](Repository.md).  Here users can see the data files available for download at the GDC and apply file/case filters to narrow down their search.

* __Human Outline__: The home page displays a human anatomical outline that can be used to refine their search. Choosing an associated organ will direct the user to a listing of all projects associated with that primary site. For example, clicking on the human brain will show only cases and projects associated with brain cancer (TCGA-GBM and TCGA-LGG).  The number of cases associated with each primary site is also displayed here and separated by project.

Each view provides a distinct representation of the same underlying set of GDC data and metadata. Previously, the GDC also provided access to certain unharmonized data files generated by GDC-hosted projects, which could be found in the GDC Legacy Archive.

The Projects, Exploration, Analysis and Repository pages can be accessed from the GDC Data Portal front page and from the toolbar (see below). The annotations view is accessible from Repository view.

## Toolbar

The toolbar available at the top of all pages in the GDC Data Portal provides convenient navigation links and access to authentication and quick search.

The left portion of this toolbar provides access to the __Home Page__, __Projects Page__, __Exploration Page__, __Analysis Page__, and a link to __Repository Page__:

[![GDC Data Portal Toolbar (Left)](images/gdc-data-portal-top-menu-bar-left.png)](images/gdc-data-portal-top-menu-bar-left.png "Click to see the full image.")

The right portion of this toolbar provides access to [quick search](#quick-search), [manage sets](#manage-sets), [authentication functions](Repository.md#authentication), the [cart](Cart.md), and the GDC Apps menu:

[![GDC Data Portal Toolbar (Left)](images/gdc-data-portal-top-menu-bar-right.png)](images/gdc-data-portal-top-menu-bar-right.png "Click to see the full image.")

The GDC Apps menu provides links to all resources provided by the GDC.

[![GDC Apps](images/gdc-data-portal-gdc-apps_v2.png)](images/gdc-data-portal-gdc-apps_v2.png "Click to see the full image.")

## Tables

Tabular listings are the primary method of representing available data in the GDC Data Portal. Tables are available in all views and in the file cart. Users can customize each table by specifying columns, size, and sorting.


### Table Sort

The sort button is available in the top right corner of each table. To sort by a column, place a checkmark next to it and select the preferred sort direction. If multiple columns are selected for sorting, data is sorted column-by-column in the order that the columns appear in the sort menu: the topmost selected column becomes the primary sorting parameter; the selected column below it is used for secondary sort, etc.

[![Sorting a table](images/gdc-data-portal-table-sort.png)](images/gdc-data-portal-table-sort.png "Click to see the full image.")

### Table Arrangement

The arrange button allows users to adjust the order of columns in the table and select which columns are displayed.

![Selecting table columns](images/gdc-data-portal-table-column-selection.png)

### Table Size

Table size can be adjusted using the menu in the bottom left corner of the table. The menu sets the maximum number of rows to display. If the number of entries to be displayed exceeds the maximum number of rows, then the table will be paginated, and navigation buttons will be provided in the bottom right corner of the table to navigate between pages.

![Specifying table size](images/gdc-data-portal-table-size-and-pagination.png)

### Table Export

In the Repository, Projects, and Annotations views, tables can be exported in either a JSON or TSV format. The `JSON` button will export the entire table's contents into a JSON file.  The `TSV` button will export the current view of the table into a TSV file.

[![Table Columns Filtering](images/gdc-data-portal-table-export.png)](images/gdc-data-portal-table-export.png "Click to see the full image.")


## Filtering and Searching

The GDC Data Portal offers three different means of searching and filtering the available data: facet filters, quick search, and advanced search.

### Facet Filters

Facets on the left of each view (Projects, Exploration, and Repository) represent properties of the data that can be used for filtering. Some of the available facets are project name, disease type, patient gender and age at diagnosis, and various data formats and categories. Each facet displays the name of the data property, the available values, and numbers of matching entities for each value (files, cases, mutations, genes, annotations, or projects, depending on the context).

Below are two file facets available in the Repository view. A _Data Type_ facet filter is applied, filtering for "Aligned Reads" files.

![Facets (no filter applied)](images/gdc-data-portal-facet-selection.png)

Multiple selections within a facet are treated as an "OR" query: e.g. "Aligned Reads" OR "Annotated Somatic Mutation". Selections in different facets are treated as "AND" queries: e.g. Data Type: "Aligned Reads" AND Experimental Strategy: "RNA-Seq".

The information displayed in each facet reflects this: in the example above, marking the "Aligned Reads" checkbox does not change the numbers or the available values in the _Data Type_ facet where the checkbox is found, but it does change the values available in the _Experimental Strategy_ facet. The _Experimental Strategy_ facet now displays only values from files of _Data Type_ "Aligned Reads".

Custom facet filters can be added in the [Repository View](Repository.md) to expand the GDC Data Portal's filtering capabilities.

### Quick Search

The quick search feature allows users to find cases, files, mutations, or genes using a search query (i.e. UUID, filename, gene name, DNA Change, project name, id, disease type or primary site). Quick search is available by clicking on the magnifier in the right section of the toolbar (which appears on every page) or by using the search bar on the Home Page.

[![Quick Search](images/gdc-quick-search-home.png)](images/gdc-quick-search-home.png "Click to see the full image.")

Search results are displayed as the user is typing, with labels indicating the type of each search result in the list (project, case, or file). Users will see a brief description of the search results, which may include the UUID, submitter ID, or file name. Clicking on a selected result or pressing enter will open a detail page with additional information.

__Home Page Quick Search:__

[![Quick Search, Searching for an Entity](images/gdc-quick-search-results.png)](images/gdc-quick-search-results.png "Click to see the full image.")

__Toolbar Quick Search:__

[![Quick Search, Searching for an Entity](images/gdc-quick-search2.png)](images/gdc-quick-search2.png "Click to see the full image.")

### Advanced Search

Advanced Search is available in Repository View. It allows users to construct complex queries with a custom query language and auto-complete suggestions. See [Advanced Search](Advanced_Search.md) for details.

## Manage Sets
The `Manage Sets` button at the top of the GDC Portal stores sets of cases, genes, or mutations of interest.  On this page, users can review the sets that have been saved as well as upload new sets and delete existing sets.

[![Manage Sets](images/gdc-manage-sets.png)](images/gdc-manage-sets.png "Click to see the full image.")

### Upload Sets

Clicking the `Upload Set` button shows options for creating Case, Gene, or Mutation sets.  

[![Upload Sets](images/gdc-upload-sets.png)](images/gdc-upload-sets.png "Click to see the full image.")

Upon clicking one of the menu items, users are shown a dialog where they can enter unique identifiers (i.e. UUIDs, TCGA Barcodes, gene symbols, mutation UUIDs, etc.) that describe the set.

[![Manage Sets Gene](images/gdc-manage-set-gene.png)](images/gdc-manage-sets.png "Click to see the full image.")

Clicking the `Submit` button will add the set of items to the list of sets on the Manage Sets page.

[![New Sets Gene](images/gdc-new-set.png)](images/gdc-manage-sets.png "Click to see the full image.")

### Export Sets

Users can export selected sets on this page by first clicking the checkboxes next to each set, then clicking the `Export selected` button at the top of the table.

[![Export Sets](images/gdc-export-sets.png)](images/gdc-export-sets.png "Click to see the full image.")

A text file containing the UUID of each case, gene or mutation is downloaded after clicking this button.

### Review Sets

There are a few buttons in the list of sets that allows a user to get further information about each one.

* __# Items__: Clicking the link under the # Items column navigates the user to the Exploration page using the set as a filter.

* __Download/View__: To the right of the # Items column are buttons that will download the list as a TSV or open the cases in the Repository Page.

### Creating Sets from GDC Portal Filters
Many pages on the GDC Portal have an option called `Save Sets` that allows users to save a group of cases, mutations, or genes for further analysis.  After using the filtering options on the `Exploration` Page as an example, users can click the `Save Case/Gene/Mutation Set` button to save this set.

[![Save Sets](images/gdc-exploration-save-sets.png)](images/gdc-quick-search2.png "Click to see the full image.")


# Cart and File Download

While browsing the GDC Data Portal, files can either be downloaded individually from [File Summary Pages](Repository.md#file-summary-page) or collected in the file cart to be downloaded as a bundle.  Clicking on the shopping cart icon that is next to any item in the GDC will add the item to your cart.

## GDC Cart

[![Cart](images/cart-overview_v2.png)](images/cart-overview_v2.png "Click to see the full image.")

## Cart Summary

The Cart Summary Page shows a summary of all files currently in the cart:

* Number of files.
* Number of cases associated with the files.
* Total file size.

The Cart page also displays two tables:

* __File count by project__: Breaks down the files and cases by each project.
* __File count by authorization level__: Breaks down the files in the cart by authorization level.  A user must be logged into the GDC in order to download 'Controlled-Access files'.

The cart also directs users how to download files in the cart.  For large data files, it is recommended that the [GDC Data Transfer Tool](https://gdc.cancer.gov/access-data/gdc-data-transfer-tool) be used.

## Cart Items

[![Cart](images/gdc-cart-items_v2.png)](images/gdc-cart-items_v2.png "Click to see the full image.")

The Cart Items table shows the list of all the files that were added to the Cart.  The table gives the folowing information for each file in the cart:

* __Access__: Displays whether the file is open or controlled access.  Users must login to the GDC Portal and have the appropriate credentials to access these files.
* __File Name__: Name of the file.  Clicking the link will bring the user to the [File Summary Page](#file-summary-page).
* __Cases__: How many cases does the file contain.  Clicking the link will bring the user to the [Case Summary Page](Exploration.md#case-summary-page).
* __Project__: The Project that the file belongs to.  Clicking the link will bring the user to the [Project Summary Page](Projects.md#project-summary-page).
* __Category__: Type of data.
* __Format__: The file format.
* __Size__: The size of the file.
* __Annotations__: Whether there are any annotations.

# Download Options

[![Cart](images/gdc-download-options_v2.png)](images/gdc-download-options_v2.png "Click to see the full image.")

The following buttons on the Cart page allows users to download files that are related to the ones in the cart.  The following download options are available:

* __Biospecimen:__ Downloads biospecimen data related to files in the cart in either TSV or JSON format.
* __Clinical:__ Downloads clinical data related to files in the cart in either TSV or JSON format.
* __Sample Sheet:__ Downloads a tab-separated file which contains the associated case/sample IDs and the sample type (Tumor/Normal) for each file in the cart.
* __Metadata:__ GDC harmonized clinical, biospecimen, and file metadata associated with the files in the cart.
* __Download:__
    * __Manifest:__ Download a manifest file for use with the GDC Data Transfer Tool to download files.  A manifest file contains a list of the UUIDs that correspond to the files in the cart.
    * __Cart:__ Download the files in the Cart directly through the browser. Users have to be cautious of the amount of data in the cart since this option will not optimize bandwidth and will not provide resume capabilities.
* __Remove from Cart:__ Remove all files or unauthorized files from the cart.
* __SRA XML, MAGE-TAB:__ This option is available in the GDC Legacy Archive only. It is used to download metadata files associated with the files in the cart.

The cart allows users to download up to 5 GB of data directly through the web browser. This is not recommended for downloading large volumes of data, in particular due to the absence of a retry/resume mechanism. For downloads over 5 GB we recommend using the `Download Manifest` button and download a manifest file that can be imported into [GDC Data Transfer Tool](https://docs.gdc.cancer.gov/Data_Transfer_Tool/Users_Guide/Getting_Started/).

>__Note__: when downloading multiple files from the cart, they are automatically bundled into one single Gzipped (.tar.gz) file.

## [GDC Data Transfer Tool](https://gdc.cancer.gov/access-data/gdc-data-transfer-tool)

The `Download Manifest` button will download a manifest file that can be imported into the GDC Data Transfer Tool.   Below is an example of the contents of a manifest file used for download:

=== "Manifest"

    ```
    id	filename	md5	size	state
    4ea9c657-8f85-44d0-9a77-ad59cced8973	mdanderson.org_ESCA.MDA_RPPA_Core.mage-tab.1.1.0.tar.gz		2516051	live
    b8342cd5-330e-440b-b53a-1112341d87db	mdanderson.org_SARC.MDA_RPPA_Core.mage-tab.1.1.0.tar.gz		4523632	live
    c57673ac-998a-4a50-a12b-4cac5dc3b72e	mdanderson.org_KIRP.MDA_RPPA_Core.mage-tab.1.2.0.tar.gz		4195746	live
    3f22dd8d-59c8-43a4-89cf-3b595f2e5a06	14-3-3_beta-R-V_GBL1112940.tif	56df0e4b4fc092fc3643bd2e316ac05b	6257840	live
    7ce05059-9197-4d38-830f-04356f5f851a	14-3-3_beta-R-V_GBL11066140.tif	6abfee483974bc2e61a37b5499ae9a07	6261580	live
    8e00d22a-ca6f-4da8-a1c3-f23144cb21b7	14-3-3_beta-R-V_GBL1112940.tif	56df0e4b4fc092fc3643bd2e316ac05b	6257840	live
    96487cd7-8fa8-4bee-9863-17004a70b2e9	14-3-3_beta-R-V_GBL1112940.tif	56df0e4b4fc092fc3643bd2e316ac05b	6257840	live
    ```

The Manifest contains a list of the file UUIDs in the cart and can be used together with the GDC Data Transfer Tool to download all files.

Information on the GDC Data Transfer Tool is available in the [GDC Data Transfer Tool User's Guide](../../Data_Transfer_Tool/Users_Guide/Getting_Started.md).

# Controlled Files

If a user tries to download a cart containing controlled files and without being authenticated, a pop-up will be displayed to offer the user either to download only open access files or to login into the GDC Data Portal through eRA Commons. See [Authentication](#Authentication) for details.

Once a user is logged in, controlled files that they have access to can be downloaded. To download files from the portal, users must agree to the GDC and individual project Data Use Agreements by selecting the agreement checkbox on the Access Alert message.

[![Cart Page](images/gdc-data-portal-download-cart_v2.png)](images/gdc-data-portal-download-cart_v2.png "Click to see the full image.")

# Authentication

The GDC Data Portal provides granular metadata for all datasets available in the GDC. Any user can see a listing of all available data files, including controlled-access files. The GDC Data Portal also allows users to download open-access files without logging in. However, downloading of controlled-access files is restricted to authorized users and requires authentication.

## Logging into the GDC

To login to the GDC, users must click on the `Login` button on the top right of the GDC Website.

![Login](images/gdc-login.png)

After clicking Login, users authenticate themselves using their eRA Commons login and password.  If authentication is successful, the eRA Commons username will be displayed in the upper right corner of the screen, in place of the "Login" button.

Upon successful authentication, GDC Data Portal users can:

- See which controlled-access files they can access.
- Download controlled-access files directly from the GDC Data Portal.
- Download an authentication token for use with the GDC Data Transfer Tool or the GDC API.
- See controlled-access mutation data they can access.

Controlled-access files are identified using a "lock" icon:

[![GDC Data Portal Main Page](images/gdc-data-portal-controlled-files.png)](images/gdc-data-portal-controlled-files.png "Click to see the full image.")

The rest of this section describes controlled data access features of the GDC Data Portal available to authorized users. For more information about open and controlled-access data, and about obtaining access to controlled data, see [Data Access Processes and Tools](https://gdc.cancer.gov/access-data/data-access-processes-and-tools).

## User Profile

After logging into the GDC Portal, users can view which projects they have access to by clicking the `User Profile` section in the dropdown menu in the top corner of the screen.

[![User Profile Drop Down](images/gdc-user-profile-dropdown.png)](images/gdc-user-profile-dropdown.png "Click to see the full image.")

Clicking this button shows the list of projects.

[![User Profile](images/gdc-user-profile.png)](images/gdc-user-profile.png "Click to see the full image.")

## GDC Authentication Tokens

The GDC Data Portal provides authentication tokens for use with the GDC Data Transfer Tool or the GDC API. To download a token:

1. Log into the GDC using your eRA Commons credentials.
2. Click the username in the top right corner of the screen.
3. Select the "Download token" option.

![Token Download Button](images/gdc-data-portal-token-download.png)

A new token is generated each time the `Download Token` button is clicked.

For more information about authentication tokens, see [Data Security](../../Data/Data_Security/Data_Security.md#authentication-tokens).

>__Note:__ The authentication token should be kept in a secure location, as it allows access to all data accessible by the associated user account.

## Logging Out

To log out of the GDC, click the username in the top right corner of the screen, and select the Logout option.

![Logout link](images/gdc-data-portal-token-download.png)


# Image Viewer

The Image viewer allows users to visualize tissue slide images.

[![Image Viewer](images/Image_viewer_browser.png)](images/Image_viewer_browser.png "Click to see the full image.")

## How to access the image viewer

The image viewer is available from:
* __Case entity page__: Click on the image viewer icon in the Case summary section or in the Biospecimen section - Slides detail. It will display in the image viewer the tissue slide images attached to the Case.

 [![Image Viewer](images/Image_viewer_case_summary.png)](images/Image_viewer_case_summary.png "Click to see the full image.")
 [![Image Viewer](images/Image_viewer_case_slide_section.png)](images/Image_viewer_case_slide_section.png "Click to see the full image.")

* __Case table in Exploration and Repository pages__: Click on the image viewer icon in the Case table. It will display in the image viewer all the tissue slide images attached to the Case.

 [![Image Viewer](images/Image_viewer_Case_table.png)](images/Image_viewer_Case_table.png "Click to see the full image.")

* __Repository page - main search__: After selecting your query, click on "View images" in Repository. It will display the tissue slide images of all the cases resulting from the query.

[![Image Viewer](images/Image_Viewer_from_Repository.png)](images/Image_Viewer_from_Repository.png "Click to see the full image.")


* __Directly in the File entity page__: You can visualize the tissue slide image directly in the file entity page.

[![Image Viewer](images/Image_viewer_File_entity.png)](images/Image_viewer_File_entity.png "Click to see the full image.")


## Image viewer features
In the image viewer, you can:
* Zoom in and zoom out by clicking on + and - icons
* Reset to default display by clicking on the Home icon
* Display the image in full screen mode by clicking on the Expand icon
* View the slide detail by clicking on "Details" button
* Selecting the area of interest with the thumbnail at the top-right corner

[![Image Viewer](images/Image_viewer_features.png)](images/Image_viewer_features.png "Click to see the full image.")


## Example of navigation to the image viewer

1. Go to Repository - Case facet and click on Add a Case/Biospecimen filter
2. In the search box of the filter pop-up, look to percent_tumor_cells and add this filter to the Repository
3. Enter a percentage between 60% and 80% then click on Go!
4. Click on "View Images" button

__Result__: The images displayed on the image viewer are filtered based on your query.

[![Image Viewer](images/Image_viewer_example-1.png)](images/Image_viewer_example-1.png "Click to see the full image.")
[![Image Viewer](images/Image_viewer_example-2.png)](images/Image_viewer_example-2.png "Click to see the full image.")


# Mutation Summary Page

The Mutation Summary Page contains information about one somatic mutation and how it affects the associated gene. Each mutation is identified by its chromosomal position and nucleotide-level change.

## Summary

[![Mutation Summary](images/GDC-Mutation-Summary_v2.png)](images/GDC-Mutation-Summary_v2.png "Click to see the full image.")

- __ID:__ A unique identifier (UUID) for this mutation
- __DNA Change:__ Denotes the chromosome number, position, and nucleotide change of the mutation
- __Type:__ A broad categorization of the mutation
- __Reference Genome Assembly:__ The reference genome in which the chromosomal position refers to
- __Allele in the Reference Assembly:__ The nucleotide(s) that compose the site in the reference assembly
- __Functional Impact:__ A subjective classification of the severity of the variant consequence. The categories are:
  - __HIGH__: The variant is assumed to have high (disruptive) impact in the protein, probably causing protein truncation, loss of function or triggering nonsense mediated decay
  - __MODERATE__: A non-disruptive variant that might change protein effectiveness
  - __LOW__: Assumed to be mostly harmless or unlikely to change protein behavior

### External References

A separate panel contains links to databases that contain information about the specific mutation. These include [dbSNP](https://www.ncbi.nlm.nih.gov/projects/SNP/), [COSMIC](http://cancer.sanger.ac.uk/cosmic), and [CIViC](https://civicdb.org/home).

## Consequences

[![Mutation Consequences](images/GDC-Mutation-Consequences.png)](images/GDC-Mutation-Consequences.png "Click to see the full image.")

The consequences of the mutation are displayed in a table. The fields that detail each mutation are listed below:

* __Gene:__ The symbol for the affected gene
* __AA Change:__ Details on the amino acid change, including compounds and position, if applicable
* __Consequence:__ The biological consequence of each mutation
* __Coding DNA Change:__ The specific nucleotide change and position of the mutation within the gene
* __Strand:__ If the gene is located on the forward (+) or reverse (-) strand
* __Transcript(s):__ The transcript(s) affected by the mutation. Each contains a link to the [Ensembl](https://www.ensembl.org) entry for the transcript   

## Cancer Distribution

[![Mutation Distribution](images/GDC-Mutation-CancerDist.png)](images/GDC-Mutation-CancerDist.png "Click to see the full image.")

The Cancer Distribution table contains information about how the mutation affects each project, which can be exported as a JSON object. The table contains the following fields:

* __Project ID__: The ID for a specific project
* __Disease Type__: The disease associated with the project
* __Site__: The anatomical site affected by the disease
* __# Affected Cases__: The number of affected cases and total number of cases displayed as a fraction and percentage

## Protein Viewer

[![Mutation Protein Graph](images/GDC-Mutation-ProteinGraph.png)](images/GDC-Mutation-ProteinGraph.png "Click to see the full image.")

The protein viewer displays a plot representing the position of mutations along the polypeptide chain associated with the mutation. The y-axis represents the number of cases that exhibit each mutation, whereas the x-axis represents the polypeptide chain sequence. [Pfam domains](http://pfam.xfam.org/) that were identified along the polypeptide chain are identified with colored rectangles labeled with pfam IDs. See the Gene Summary Page for additional details about the protein viewer.


# Gene Summary Page

The Gene Summary Page describes each gene with mutation data featured at the GDC and provides results related to the analyses that are performed on these genes.  

## Summary

The summary section of the gene page contains the following information:

[![Gene Summary](images/GDC-Gene-Summary_v2.png)](images/GDC-Gene-Summary_v2.png "Click to see the full image.")

* __Symbol:__ The gene symbol
* __Name:__ Full name of the gene
* __Synonyms:__ Synonyms of the gene name or symbol, if available
* __Type:__ A broad classification of the gene
* __Location:__ The chromosome on which the gene is located and its coordinates
* __Strand:__ If the gene is located on the forward (+) or reverse (-) strand
* __Description:__ A description of gene function and downstream consequences of gene alteration
- __Annotation:__ A notation/link that states whether the gene is part of [The Cancer Gene Census](http://cancer.sanger.ac.uk/census/)

## External References

A list with links that lead to external databases with additional information about each gene is displayed here. These external databases include: [Entrez](https://www.ncbi.nlm.nih.gov/gquery/), [Uniprot](http://www.uniprot.org/), [Hugo Gene Nomenclature Committee](http://www.genenames.org/), [Online Mendelian Inheritance in Man](https://www.omim.org/), [Ensembl](http://may2015.archive.ensembl.org/index.html), and [CIViC](https://civicdb.org/home).

## Cancer Distribution

A table and bar graph shows how many cases are affected by mutations within the gene as a ratio and percentage. Each row/bar represents the number of cases for each project.  The final column in the table lists the number of unique mutations observed on the gene for each project.

[![Cancer Distribution](images/GDC-Gene-CancerDist.png)](images/GDC-Gene-CancerDist.png "Click to see the full image.")

## Protein Viewer

[![Protein Plot](images/GDC-Gene-ProteinGraph.png)](images/GDC-Gene-ProteinGraph.png "Click to see the full image.")

Mutations and their frequency across cases are mapped to a graphical visualization of protein-coding regions with a lollipop plot. Pfam domains are highlighted along the x-axis to assign functionality to specific protein-coding regions. The bottom track represents a view of the full gene length. Different transcripts can be selected by using the drop-down menu above the plot.  

The panel to the right of the plot allows the plot to be filtered by mutation consequences or impact.  The plot will dynamically change as filters are applied.  Mutation consequence and impact is denoted in the plot by color.

Note: The impact filter on this panel will not display the annotations for alternate transcripts.

The plot can be viewed at different zoom levels by clicking and dragging across the x-axis, clicking and dragging across the bottom track, or double clicking the pfam domain IDs. The `Reset` button can be used to bring the zoom level back to its original position. The plot can also be exported as a PNG image, SVG image or as JSON formatted text by choosing the `Download` button above the plot.

## Most Frequent Mutations

The most frequent somatic mutations in the gene are displayed in a tabular view at the bottom of the page.

[![Gene MFM](images/GDC-Gene-MFM.png)](images/GDC-Gene-MFM.png "Click to see the full image.")

A table is displayed below that lists information about each mutation including:

* __Mutation ID:__ A UUID Code for the mutation assigned by the GDC, when clicked will bring a user to the Mutation Summary Page
* __DNA Change:__ The chromosome and starting coordinates of the mutation are displayed along with the nucleotide differences between the reference and tumor allele
* __Type:__ A general classification of the mutation
* __Consequences:__ The effects the mutation has on the gene coding for a protein (i.e. synonymous, missense, non-coding transcript)
* __# Affected Cases in Gene:__ The number of affected cases, expressed as number and percentage across all mutations within the gene
* __# Affected Cases Across GDC:__ The number of affected cases, expressed as number across all projects. Choosing the arrow next to the percentage will expand the selection with a breakdown of each affected project
* __Impact:__ A subjective classification of the severity of the variant consequence. The categories are:
  - __HIGH__: The variant is assumed to have high (disruptive) impact in the protein, probably causing protein truncation, loss of function or triggering nonsense mediated decay
  - __MODERATE__: A non-disruptive variant that might change protein effectiveness
  - __LOW__: Assumed to be mostly harmless or unlikely to change protein behavior
  - __MODIFIER__: Usually non-coding variants or variants affecting non-coding genes, where predictions are difficult or there is no evidence of impact


# Exploration

The Exploration Page allows users to explore data in the GDC using advanced filters/facets, which includes those on a gene and mutation level. Users choose filters on specific `Cases`, `Genes`, and/or `Mutations` on the left of this page and then can visualize these results on the right.  The Gene/Mutation data for these visualizations comes from the Open-Access MAF files on the GDC Data Portal.  There is also a `Clinical` tab with filters that apply specifically to clinical data.

[![Exploration Page](images/GDC-Exploration-Page_v6.png)](images/GDC-Exploration-Page_v6.png "Click to see the full image.")

## Filters / Facets
On the left of this page, users can create advanced filters to narrow down results to create synthetic cohorts.

### Case Filters

The first tab of filters is for cases in the GDC.

[![Exploration Case Filters](images/Exploration-Cases-Filter_v2.png)](images/Exploration-Cases-Filter_v2.png "Click to see the full image.")

These criteria limit the results only to specific cases within the GDC. The default filters available are:

* __Case:__ Specify individual cases using submitter ID (barcode), UUID, or list of Cases ('Case Set').
* __Primary Site:__ Anatomical site of the cancer under investigation or review.
* __Program:__ A cancer research program, typically consisting of multiple focused projects.
* __Project:__ A cancer research project, typically part of a larger cancer research program.
* __Disease Type:__ Type of cancer studied.
* __Experimental Strategy:__ Experimental strategy used for molecular characterization of the cancer.
* __Sample Type:__ Describes the source of a biospecimen used for a laboratory test.
* __Available Variation Data:__ Indicates the types of genomic variation data that a case has been tested for.

### Clinical Filters

The second tab of filters is used to specifically explore clinical data for cases in the GDC.

[![Exploration Clinical Filters](images/Exploration-Clinical-Filter.png)](images/Exploration-Clinical-Filter.png "Click to see the full image.")

Users can filter by specific clinical variables, grouped into these categories:

* __Demographic:__ Data for the characterization of the patient by means of segmenting the population (e.g. characterization by age, sex, race, etc.).
* __Diagnoses:__ Data from the investigation, analysis, and recognition of the presence and nature of disease, condition, or injury from expressed signs and symptoms; also, the scientific determination of any kind; the concise results of such an investigation.
* __Treatments:__ Records of the administration and intention of therapeutic agents provided to a patient to alter the course of a pathologic process.
* __Exposures:__ Clinically-relevant patient information not immediately resulting from genetic predispositions.

### Gene Filters

The third tab of filters is for genes affected by mutations in the GDC.

[![Exploration Gene Filters](images/Exploration-Gene-Filter_v2.png)](images/Exploration-Gene-Filter_v2.png "Click to see the full image.")

Users can filter by:

* __Gene:__ Specify a Gene Symbol, ID, or list of Genes ('Gene Set').
* __Biotype:__  Classification of the type of gene according to Ensembl. The biotypes can be grouped into protein coding, pseudogene, long noncoding and short noncoding. Examples of biotypes in each group are as follows:
    * __Protein coding:__ IGC gene, IGD gene, IG gene, IGJ gene, IGLV gene, IGM gene, IGV gene, IGZ gene, nonsense mediated decay, nontranslating CDS, non stop decay, polymorphic pseudogene, TRC gene, TRD gene, TRJ gene, TRV gene.
    * __Pseudogene:__ disrupted domain, IGC pseudogene, IGJ pseudogene, IG pseudogene, IGV pseudogene, processed pseudogene, transcribed processed pseudogene, transcribed unitary pseudogene, transcribed unprocessed pseudogene, translated processed pseudogene, translated unprocessed pseudogene, TRJ pseudogene, TRV pseudogene, unprocessed pseudogene.
    * __Long noncoding:__ 3 prime overlapping ncrna, ambiguous orf, antisense, antisense RNA, lincRNA, macro lincRNA, ncrna host, processed transcript, sense intronic, sense overlapping.
    * __Short noncoding:__ miRNA, miRNA pseudogene, miscRNA, miscRNA pseudogene, Mt rRNA, Mt tRNA, rRNA, scRNA, snlRNA, snoRNA, snRNA, tRNA, tRNA pseudogene, vaultRNA.
* __Is Cancer Gene Census:__ Whether or not a gene is part of [The Cancer Gene Census](http://cancer.sanger.ac.uk/census/).

### Mutation Filters

The final tab of filters is for specific mutations.

[![Exploration Mutation Filters](images/Exploration-Mutations-Filter_v2.png)](images/Exploration-Mutations-Filter_v2.png "Click to see the full image.")

Users can filter by:

* __Mutation:__  Unique ID for that mutation.  Users can use the following:
    * UUID - c7c0aeaa-29ed-5a30-a9b6-395ba4133c63
    * DNA Change - 	chr12:g.121804752delC
    * COSMIC ID - COSM202522
    * List of any mutation UUIDs or DNA Change id's ('Mutation Set').
* __Impact:__ A subjective classification of the severity of the variant consequence. These scores are determined using the three following tools:
    * __[Ensembl VEP](http://useast.ensembl.org/info/genome/variation/prediction/index.html):__
        * __HIGH (H):__ The variant is assumed to have high (disruptive) impact in the protein, probably causing protein truncation, loss of function or triggering nonsense mediated decay.
        * __MODERATE (M):__ A non-disruptive variant that might change protein effectiveness.
        * __LOW (L):__ Assumed to be mostly harmless or unlikely to change protein behavior.
        * __MODIFIER (MO):__ Usually non-coding variants or variants affecting non-coding genes, where predictions are difficult or there is no evidence of impact.
    * __[PolyPhen](http://genetics.bwh.harvard.edu/pph2/):__
        * __probably damaging (PR):__ It is with high confidence supposed to affect protein function or structure.
        * __possibly damaging (PO):__ It is supposed to affect protein function or structure.
        * __benign (BE):__ Most likely lacking any phenotypic effect.
        * __unknown (UN):__ When in some rare cases, the lack of data does not allow PolyPhen to make a prediction.
    * __[SIFT](http://sift.jcvi.org/):__
        * __tolerated:__ Not likely to have a phenotypic effect.
        * __tolerated_low_confidence:__ More likely to have a phenotypic effect than 'tolerated'.
        * __deleterious:__ Likely to have a phenotypic effect.
        * __deleterious_low_confidence:__ Less likely to have a phenotypic effect than 'deleterious'.
* __Consequence Type:__  Consequence type of this variation; [sequence ontology](http://www.sequenceontology.org/) terms.
* __Type:__ A general classification of the mutation.
* __Variant Caller:__ The variant caller used to identify the mutation.
* __COSMIC ID:__ This option will filter out only mutations with a COSMIC ID.
* __dbSNP rs ID:__ This option will filter out only mutations with a SNP identifer maintained in dbSNP.

## Results

As users add filters to the data on the Exploration Page, the Results section will automatically be updated.  Results are divided into different tabs:  `Cases`, `Genes`, `Mutations`, and `OncoGrid`.  

To illustrate these tabs, Case, Gene, and Mutation filters have been chosen (Genes in the Cancer Gene Census, that have a missense variant for the TCGA-BRCA project) and a description of what each tab displays follows.

### Cases

The `Cases` tab gives an overview of all the cases/patients who correspond to the filters chosen (Cohort).

[![Exploration Case Example](images/Exploration-Case-Example_v3.png)](images/Exploration-Case-Example_v2.png "Click to see the full image.")

The top of this section contains a few pie graphs with categorical information regarding the Primary Site, Project, Disease Type, Gender, and Vital Status.

Below these pie charts is a tabular view of cases, which can be exported, sorted and saved using the buttons on the right and includes the following information:

* __Case ID (Submitter ID):__ The Case ID / submitter ID of that case/patient (i.e. TCGA Barcode).
* __Project:__ The study name for the project for which the case belongs.
* __Primary Site:__ The primary site of the cancer/project.
* __Gender:__ The gender of the case.
* __Files:__ The total number of files available for that case.
* __Available Files per Data Category:__ Seven columns displaying the number of files available in each of the seven data categories.  These link to the files for the specific case.
* __# Mutations:__ The number of SSMs (simple somatic mutations) detected in that case.
* __# Genes:__ The number of genes affected by mutations in that case.
* __Slides:__ The total number of slides available for that case. For more information about [slide images](Repository.md#image-viewer-features).

>__Note__: By default, the UUID is not displayed on summary page tables. You can display the UUID by clicking on the icon with 3 parallel lines and checking the UUID option.

### Case Summary Page

The Case Summary Page displays case details including the project and disease information, data files that are available for that case, and the experimental strategies employed. A button in the top-right corner of the page allows the user to add all files associated with the case to the file [cart](Cart.md).

[![Case Page](images/gdc-case-entity-page_v2.png)](images/gdc-case-entity-page_v2.png "Click to see the full image.")

#### Clinical and Biospecimen Information

The page also provides clinical and biospecimen information about that case. Links to export clinical and biospecimen information in JSON format are provided.

[![Case Page, Clinical and Biospecimen](images/image_clinical_and_biospecimen_information.png)](images/image_clinical_and_biospecimen_information.png "Click to see the full image.")


Some clinical records can support multiple records of the same type (Diagnoses, Family Histories, Exposures, Follow-Ups, Molecular Tests).  If only one record exists, the UUID of the record is provided at the top of the corresponding tab.

[![Case Page, Single Clinical Record](images/gdc-case-clinical-single-record.png)](images/gdc-case-clinical-single-record.png "Click to see the full image.")

If there are multiple records, they are listed as horizontal tabs.

[![Case Page, Multiple Clinical Records](images/gdc-case-clinical-multiple-records.png)](images/gdc-case-clinical-multiple-records.png "Click to see the full image.")

Some record types are further nested under another.  For example, a Diagnosis record may have multiple associated Treatment records.  Or a Follow-Up record may have multiple associated Molecular Test Records.  The associated sub-records are listed in a table on the tab.

[![Case Page, Nested Clinical Records](images/gdc-case-clinical-nested-records.png)](images/gdc-case-clinical-nested-records.png "Click to see the full image.")

#### Biospecimen Search

A search filter just below the biospecimen section can be used to find and filter biospecimen data. The wildcard search will highlight entities in the tree that match the characters typed. This will search both the case submitter ID, as well as the additional metadata for each entity. For example, searching 'Primary Tumor' will highlight samples that match that type.

[![Biospecimen Search](images/gdc_case_biospecimen_search_v3.png)](images/gdc_case_biospecimen_search_v3.png "Click to see the full image.")

#### Most Frequent Somatic Mutations for a Case

The Case Entity Page also lists the mutations found in that particular case.

[![Case Page](images/gdc-case-entity-mfm.png)](images/gdc-case-entity-mfm.png "Click to see the full image.")

For more information, please go to the [Most Frequent Somatic Mutation](#most-frequent-somatic-mutations) section.

### Genes

The `Genes` tab will give an overview of all the genes that match the criteria of the filters (Cohort).

[![Exploration Gene Example](images/Exploration-Gene-Example_v3.png)](images/Exploration-Gene-Example_v3.png "Click to see the full image.")

The top of this tab contains a bar graph of the most frequently mutated genes. Hovering over each bar in the plot will display information about the percentage of cases affected. In addition, this section contains a survival curve. The survival curve is calculated using the Kaplan-Meier estimator based on all cases with survival data within the specified Exploration Page search. For more information on how these values are determined, please go to the [Survival Analysis](#survival-analysis) section. Users may choose to download the underlying data in JSON or TSV format or an image of the graph in SVG or PNG format by clicking the `download` icon at the top of each graph.

Below these graphs is a tabular view of the genes affected, which includes the following information:

* __Symbol:__ The gene symbol, which links to the Gene Summary Page.
* __Name:__ Full name of the gene.
* __# SSM Affected Cases in Cohort:__ The number of cases affected by SSMs (simple somatic mutations) in the Cohort.
* __# SSM Affected Cases Across the GDC:__ The number of cases within all the projects in the GDC that contain a mutation on this gene.  Clicking the red arrow will display the cases broken down by project.
* __# CNV Gain:__ The number of CNV (copy number variation) events detected in that gene which resulted in an increase (gain) in the gene's copy number.
* __# CNV Loss:__ The number of CNV events detected in that gene which resulted in a decrease (loss) in the gene's copy number.
* __# Mutations:__ The number of SSMs (simple somatic mutations) detected in that gene.
* __Annotations:__ Includes a COSMIC symbol if the gene belongs to [The Cancer Gene Census](http://cancer.sanger.ac.uk/census/).
* __Survival:__ An icon that, when clicked, will plot the survival rate between cases in the project with mutated and non-mutated forms of the gene.

#### CNV Data

The CNV visualization in GDC Exploration reports only gains, losses, or neither, however, in some situations it is important to distinguish between high level amplifications and gains, and single and double copy losses. For this, you will need to download the [primary data](https://portal.gdc.cancer.gov/v1/repository?filters=%7B%22op%22%3A%22and%22%2C%22content%22%3A%5B%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22files.analysis.workflow_type%22%2C%22value%22%3A%5B%22ASCAT2%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22files.data_type%22%2C%22value%22%3A%5B%22Gene%20Level%20Copy%20Number%22%5D%7D%7D%5D%7D) and determine cutoffs that are suitable for your work. 

### Gene Summary Page

Gene Summary Pages describe each gene with mutation data and provides results related to the analyses that are performed on these genes.  

The summary section of the Gene Page contains the following information:

[![Gene Summary](images/GDC-Gene-Summary_v2.png)](images/GDC-Gene-Summary_v2.png "Click to see the full image.")

* __Symbol:__ The gene symbol.
* __Name:__ Full name of the gene.
* __Synonyms:__ Synonyms of the gene name or symbol, if available.
* __Type:__ A broad classification of the gene.
* __Location:__ The chromosome on which the gene is located and its coordinates.
* __Strand:__ If the gene is located on the forward (+) or reverse (-) strand.
* __Description:__ A description of gene function and downstream consequences of gene alteration.
* __Annotation:__ A notation/link that states whether the gene is part of [The Cancer Gene Census](http://cancer.sanger.ac.uk/census/).

#### External References

A list with links that lead to external databases with additional information about each gene is displayed here. These external databases include:

*  [Entrez](https://www.ncbi.nlm.nih.gov/gquery/)
*  [Uniprot](http://www.uniprot.org/)
*  [Hugo Gene Nomenclature Committee](http://www.genenames.org/)
*  [Online Mendelian Inheritance in Man](https://www.omim.org/)
*  [Ensembl](http://nov2020.archive.ensembl.org/index.html)
*  [CIViC](https://civicdb.org/home)

#### Cancer Distribution

A table and two bar graphs show how many cases are affected by mutations and copy number variation within the gene as a ratio and percentage. Each row/bar represents the number of cases for each project.  The final column in the table lists the number of unique mutations observed on the gene for each project.

[![Cancer Distribution](images/GDC-Gene-CancerDist.png)](images/GDC-Gene-CancerDist.png "Click to see the full image.")

#### Protein Viewer

Mutations and their frequency across cases are mapped to a graphical visualization of protein-coding regions with a lollipop plot. Pfam domains are highlighted along the x-axis to assign functionality to specific protein-coding regions. The bottom track represents a view of the full gene length. Different transcripts can be selected by using the drop-down menu above the plot.  

[![Protein Plot](images/GDC-Gene-ProteinGraph.png)](images/GDC-Gene-ProteinGraph.png "Click to see the full image.")

The panel to the right of the plot allows the plot to be filtered by mutation consequences or impact. The plot will dynamically change as filters are applied.  Mutation consequence and impact is denoted in the plot by color.

>__Note__: The impact filter on this panel will not display the annotations for alternate transcripts.

The plot can be viewed at different zoom levels by clicking and dragging across the x-axis, clicking and dragging across the bottom track, or double clicking the pfam domain IDs. The `Reset` button can be used to bring the zoom level back to its original position. The plot can also be exported as a PNG image, SVG image or as JSON formatted text by choosing the `Download` button above the plot.

#### Most Frequent Somatic Mutations

The 20 most frequent mutations in the gene are displayed as a bar graph that indicates the number of cases that share each mutation.  

[![Gene MFM](images/GDC-Gene-MFM.png)](images/GDC-Gene-MFM.png "Click to see the full image.")

A table is displayed below that lists information about each mutation including:

* __DNA Change:__ The chromosome and starting coordinates of the mutation are displayed along with the nucleotide differences between the reference and tumor allele.
* __Type:__ A general classification of the mutation.
* __Consequences:__ The effects the mutation has on the gene coding for a protein (i.e. synonymous, missense, non-coding transcript).
* __# Affected Cases in Gene:__ The number of affected cases, expressed as number across all mutations within the selected Gene.
* __# Affected Cases Across GDC:__ The number of affected cases, expressed as number across all projects. Choosing the arrow next to the percentage will expand the selection with a breakdown of each affected project.
* __Impact:__  A [subjective classification](#mutation-filters) of the severity of the variant consequence. This is determined by three different tools:
    * __[Ensembl VEP](http://useast.ensembl.org/info/genome/variation/prediction/index.html)__
    * __[PolyPhen](http://genetics.bwh.harvard.edu/pph/)__
    * __[SIFT](http://sift.jcvi.org/)__


Clicking the `Open in Exploration` button will navigate the user to the Exploration Page, showing the same results in the table (mutations filtered by the gene).

### Mutations

The `Mutations` tab will give an overview of all the mutations that match the criteria of the filters (Cohort).  

Open-access mutation data is displayed by defualt.  To access controlled access mutations, users must apply to the correct data access authority, be granted access, and login to the portal.  If a user is logged in and has been granted access to controlled-access mutations, they will be integrated with open-access mutations throughout the portal visualizations and counts.

[![Exploration Mutation Example](images/Exploration-Mutation-Example_v2.png)](images/Exploration-Mutation-Example_v2.png "Click to see the full image.")

At the top of this tab contains a survival curve. The survival curve is calculated using the Kaplan-Meier estimator based on all cases with survival data within the specified Exploration Page search. For more information on how these values are determined, please go to the [Survival Analysis](#survival-analysis) section. Users may choose to download the underlying data in JSON or TSV format or an image of the graph in SVG or PNG format by clicking the `download` icon at the top of the graph.

A table is displayed below that lists information about each mutation:

* __DNA Change:__ The chromosome and starting coordinates of the mutation are displayed along with the nucleotide differences between the reference and tumor allele.
* __Type:__ A general classification of the mutation.
* __Consequences:__ The effects the mutation has on the gene coding for a protein (i.e. synonymous, missense, non-coding transcript).  A link to the [Gene Summary Page](Exploration.md#gene-summary-page) for the gene affected by the mutation is included.
* __# Affected Cases in Cohort:__ The number of affected cases in the Cohort as a fraction and as a percentage.
* __# Affected Cases in Across all Projects:__ The number of affected cases, expressed as number across all projects. Clicking the arrow next to the percentage will display a breakdown of each affected project.
* __Impact:__  A [subjective classification](#mutation-filters) of the severity of the variant consequence. This is determined by three different tools:
    * __[Ensembl VEP](http://useast.ensembl.org/info/genome/variation/prediction/index.html)__
    * __[PolyPhen](http://genetics.bwh.harvard.edu/pph/)__
    * __[SIFT](http://sift.jcvi.org/)__
* __Survival:__ An icon that when clicked, will plot the survival rate between the gene's mutated and non-mutated cases.

### Mutation Summary Page

  The Mutation Summary Page contains information about one somatic mutation and how it affects the associated gene. Each mutation is identified by its chromosomal position and nucleotide-level change.

  [![Mutation Summary](images/GDC-Mutation-Summary_v2.png)](images/GDC-Mutation-Summary_v2.png "Click to see the full image.")

  - __UUID:__ A unique identifier (UUID) for this mutation.
  - __DNA Change:__ Denotes the chromosome number, position, and nucleotide change of the mutation.
  - __Type:__ A broad categorization of the mutation.
  - __Reference Genome Assembly:__ The reference genome in which the chromosomal position refers to.
  - __Allele in the Reference Assembly:__ The nucleotide(s) that compose the site in the reference assembly.
  - __Functional Impact:__ A subjective classification of the severity of the variant consequence.

#### External References

  A separate panel contains links to databases that contain information about the specific mutation. These include [dbSNP](https://www.ncbi.nlm.nih.gov/projects/SNP/), [COSMIC](http://cancer.sanger.ac.uk/cosmic), and [CIViC](https://civicdb.org/home).

#### Consequences

The consequences of the mutation are displayed in a table.  The set of consequence terms, defined by the [Sequence Ontology](http://www.sequenceontology.org).

  [![Mutation Consequences](images/GDC-Mutation-Consequences.png)](images/GDC-Mutation-Consequences.png "Click to see the full image.")

The fields that describe each consequence are listed below:

  * __Gene:__ The symbol for the affected gene.
  * __AA Change:__ Details on the amino acid change, including compounds and position, if applicable.
  * __Consequence:__ The biological consequence of each mutation.
  * __Coding DNA Change:__ The specific nucleotide change and position of the mutation within the gene.
  * __Impact:__  A [subjective classification](#mutation-filters) of the severity of the variant consequence. This is    determined by three different tools:
    * __[Ensembl VEP](http://useast.ensembl.org/info/genome/variation/prediction/index.html)__
    * __[PolyPhen](http://genetics.bwh.harvard.edu/pph/)__
    * __[SIFT](http://sift.jcvi.org/)__
  * __Strand:__ If the gene is located on the forward (+) or reverse (-) strand.
  * __Transcript(s):__ The transcript(s) affected by the mutation. Each contains a link to the [Ensembl](https://www.ensembl.org) entry for the transcript.

#### Cancer Distribution

A table and bar graph shows how many cases are affected by the particular mutation. Each row/bar represents the number of cases for each project.

  [![Mutation Distribution](images/GDC-Mutation-CancerDist.png)](images/GDC-Mutation-CancerDist.png "Click to see the full image.")

The table contains the following fields:

  * __Project ID__: The ID for a specific project.
  * __Disease Type__: The disease associated with the project.
  * __Site__: The anatomical site affected by the disease.
  * __# SSM Affected Cases__: The number of affected cases and total number of cases displayed as a fraction and percentage.

#### Protein Viewer

The protein viewer displays a plot representing the position of mutations along the polypeptide chain. The y-axis represents the number of cases that exhibit each mutation, whereas the x-axis represents the polypeptide chain sequence. [Pfam domains](http://pfam.xfam.org/) that were identified along the polypeptide chain are identified with colored rectangles labeled with pfam IDs. See the [Gene Summary Page](#gene-summary-page) for additional details about the [protein viewer](#protein-viewer).

  [![Mutation Protein Graph](images/GDC-Mutation-ProteinGraph.png)](images/GDC-Mutation-ProteinGraph.png "Click to see the full image.")

## OncoGrid

The Exploration Page includes an OncoGrid plot of the cases with the most mutations, for the top 50 mutated genes affected by high impact mutations. Genes displayed on the left of the grid (Y-axis) correspond to individual cases on the bottom of the grid (X-axis). Additionally, the plot also indicates in each cell any CNV events detected for these top mutated cases and genes.

[![Exploration Oncogrid Example](images/Exploration-Oncogrid-Example_v2.png)](images/Exploration-Oncogrid-Example_v2.png "Click to see the full image.")

The grid is color-coded with a legend at the top which describes what type of mutation consequence and CNV event is observed for each gene/case combination. Clinical information and the available data for each case are available at the bottom of the grid.

The right side of the grid displays additional information about the genes:

* __Gene Sets:__ Describes whether a gene is part of [The Cancer Gene Census](http://cancer.sanger.ac.uk/census/).  (The Cancer Gene Census is an ongoing effort to catalogue those genes for which mutations have been causally implicated in cancer)
* __# Cases Affected:__ Identifies all cases in the GDC affected with a mutation in this gene

### OncoGrid Options

To facilitate readability and comparisons, drag-and-drop can be used to reorder the gene rows.  Double clicking a row in the "# Cases Affected" bar at the right side of the graphic launches the respective Gene Summary Page. Hovering over a cell will display information about the mutation such as its ID, affected case, and biological consequence. Clicking on the cell will bring the user to the respective Mutation Summary Page.  

A tool bar at the top right of the graphic allows the user to export the data as a JSON object, PNG image, or SVG image.  Seven buttons are available in this toolbar:

* __Customize Colors:__ Users can customize the colors that represent mutation consequence types and CNV gains/losses.
* __Download:__ Users can choose to export the contents either to a static image file (PNG or SVG format) or the underlying data in JSON format.
* __Reload Grid:__ Sets all OncoGrid rows, columns, and zoom levels back to their initial positions.
* __Cluster Data:__ Clusters the rows and columns to place mutated genes with the same cases and cases with the same mutated genes together.
* __Toggle Heatmap:__ The view can be toggled between cells representing mutation consequences or number of mutations in each gene.
* __Toggle Gridlines:__ Turn the gridlines on and off.
* __Toggle Crosshairs:__ Turns crosshairs on, so that users can zoom into specific sections of the OncoGrid.
* __Fullscreen:__ Turns Fullscreen mode on/off.

### OncoGrid Color Picker

To customize the colors for mutation consequence types and CNV gains/losses, a user can click the color picker icon in the OncoGrid toolbar.  

* __Customize Colors:__ Opens a control where the user can pick their own colors or apply a suggested theme and save their changes.
* __Reset to Default:__ Resets all colors to the defaults initially used by OncoGrid.

[![Exploration Oncogrid Color Picker](images/Exploration-Oncogrid-Color-Picker.png)](images/Exploration-Oncogrid-Color-Picker.png "Click to see the full image.")

## File Navigation

After utilizing the Exploration Page to narrow down a specific cohort, users can find the specific files that relate to this group by clicking on the `View Files in Repository` button as shown in the image below.

[![Exploration File Navigation](images/Exploration-View-Files_v4.png)](images/Exploration-View-Files_v4.png "Click to see the full image.")

Clicking this button will navigate the users to the Repository Page, filtered by the cases within the cohort.

[![Input Set Explanation](images/gdc-input-set_v2.png)](images/gdc-input-set_v2.png "Click to see the full image.")

The filters chosen on the Exploration Page are displayed as an `input set` on the Repository Page.  Additional filters may be added on top of this `input set`, but the original set cannot be modified and instead a new `input set` must be created from original data.  

---

## Survival Analysis

The survival analysis, which is seen in both the `Gene` and `Mutation` tabs, is used to analyze the occurrence of event data over time.  In the GDC, survival analysis is performed on the mortality of the cases. Thus, the values are retrieved from [GDC Data Dictionary](../../../Data_Dictionary) properties and a survival analysis requires the following fields:

*  Data on the time to a particular event (days to death or last follow up).
    * Fields:  __demographic.days_to_death__ or __demographic.days_to_last_follow_up__
*  Information on whether the event has occurred (alive/deceased).
    * Fields:  __demographic.vital_status__
*  Data split into different categories or groups (i.e. gender, etc.).
    * Fields:  __demographic.gender__

The survival analysis in the GDC uses a Kaplan-Meier estimator:

[![Kaplan-Meier Estimator](images/gdc-kaplan-meier-estimator2.png)](images/gdc-kaplan-meier-estimator2.png "Click to see the full image.")

Where:

 * S(t) is the estimated survival probability for any particular one of the t time periods.
 * n<sub>i</sub> is the number of subjects at risk at the beginning of time period t<sub>i</sub>.
 * and d<sub>i</sub> is the number of subjects who die during time period t<sub>i</sub>.

The table below is an example data set to calculate survival for a set of seven cases:

[![Sample Survival Analysis Table](images/gdc-sample-survival-table.png)](images/gdc-sample-survival-table.png "Click to see the full image.")

The calculated cumulated survival probability can be plotted against the interval to obtain a survival plot like the one shown below.

[![Sample Survival Analysis Plot](images/gdc-survival-plot.png)](images/gdc-survival-plot.png "Click to see the full image.")


# Data Portal Release Notes

## Release 0.3.24-spr5

* __GDC Product__: GDC Data Portal
* __Release Date__: April 18, 2016

### New Features and Changes

*   This is a bug-fixing release, no new features were added.

### Bugs Fixed Since Last Release

*   Tables and Export
    *   Table export return correct content (XML, TSV) although file extension is incorrect (always JSON) <!-- PGDC-1986 -->
    *   Nested values are not exported properly <!-- PGDC-910 SV-13 -->
    *   Project table summary chart does not display colors with Internet Explorer 11 <!-- PGDC-1953 -->
*   Cart and Download
    *   When saving a file for download, the download pop-up indicates "from:blob:" <!-- PGDC-115 -->
    *   User with update only can also download files also role should not allow <!-- PGDC-1782 -->
    *   User can add more files to the cart than supported if adding files one by one <!-- PGDC-1957 -->
    *   In very specific situations, the cart can display inconsistent data <!-- PGDC-1028 -->   
    *   In some situations the cart will display the authentication window for authenticated users when trying to download <!-- PGDC-1959 -->  
*   Search
    *   In the data page, some pie chart titles are too long <!-- PGDC-914 -->
    *   Facets displaying histogram do not display a mouseover tooltip if the value is very low <!-- PGDC-913 -->
*   Layout, Browser specific and Accessibility
    *   No favicon is displayed on Internet Explorer <!-- PGDC-1948 -->
    *   Download cart button does not function properly with Safari 9.0.3 <!-- PGDC-1978 -->
    *   Layout issue when browser is reduced to a small window size <!-- PGDC-16 -->


### Known Issues and Workarounds

*   General
    *   If a user has previously logged into the Portal and left a session without logging out, if the user returns to the Portal after the user's sessionID expires, it looks as if the user is still authenticated. The user cannot download the token and gets an error message that would not close. The user should clear the cache to properly log out. 
*   Tables and Export
    *   Table sorting icon does not include numbers <!-- PGDC-35 -->
    *   Restore default table column arrangement does not restore to the default but it restores to the previous state <!-- PGDC-1769 -->
*   Cart and Download
    *   Make the cart limit warning message more explanatory <!-- PGDC-1952 -->   
    *   In some situations, adding filtered files to the cart might fail <!-- PGDC-1981 -->   
*   Layout, Browser specific and Accessibility
    *   When disabling CSS, footer elements are displayed out of order <!-- PGDC-1972 -->
    *   If javascript is disabled html tags are displayed in the warning message <!-- PGDC-1835 -->
    *   Layout issues when using the browser zoom in function on tables <!-- PGDC-116 -->
    *   Cart download spinner not showing at the proper place <!-- PGDC-2056 -->
    *   Not all facets are expanded by default when loading the app <!-- PGDC-2061 -->
*   Non UI-related tickets
    *   Investigations were done on this issue "Authenticated user with access to TARGET projects cannot download Target DCC files. They will get a pop-up message." was related  to ACL permissions (not UI related).
    *   Investigations were done on this issue "Missing pathology files and slide images for some TCGA datasets since they are not connected to the biospecimen chain yet" was related to Data not imported (not UI related).
    *   Associated entities is empty for some files (note: this is a data issue) <!-- PGDC-1887 -->


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).

## Release 0.3.24.2

* __GDC Product__: GDC Data Portal
* __Release Date__: April 4, 2016

### New Features and Changes

*   Implemented a homepage to provide high-level information about the GDC Data Portal
*   Created the GDC Legacy Archive to provide access to legacy data
*   Updated the GDC Data Portal to support harmonized data
*   Updated the login process to be consistent with the GDC Data Submission Portal login
*   Added the ability to export clinical and biospecimen data from Project and Case entity pages (replacing download clinical XML)
*   Improved GDC Data Portal search capabilities:
    *   Added customized facets, expanding the GDC Data Portal search capabilities to non-default facets
    *   Updated the list of fields suggested in the Advanced Search and Custom Facets
    *   Added prefix search on case submitter ID on the Facet panel
    *   Improved operator auto-suggestion based on field type in the Advanced Search
*   Improved the download experience by letting the browser handle the download (allow canceling)
*   Enhanced Cart capabilities:
    *   Added the ability to download metadata files (SRA XML and MAGE-TAB)
    *   Added the ability to download biospecimen and clinical data
*   Improved the Biospecimen tree in the Case entity page (users can search for ID and expand the tree)
*   Improved support for Section 508 standards and the Internet Explorer browser
*   Improved the table view in the Projects page (Added a "Total" row with links to the search page)
*   Improved the BAM Slicing UI (added an example and the ability to use tabulations for BED format in the text box)

### Bugs Fixed Since Last Release

*   Disease Type does not auto filter (Data Facet)
*   When exporting the annotation table to a file, the create date field is not displayed in the same format than the UI
*   This ticket is not applicable anymore: "In the Case Entity Page, the "Download Clinical XML" button is not applicable to TARGET projects. The label is incorrect but the functionality is accurate.". The Case entity page has a standard download clinical feature.


### Known Issues and Workarounds

*   General
    *   If a user has previously logged into the Portal and left a session without logging out, if the user returns to the Portal after the user's sessionID expires, it looks as if the user is still authenticated. The user cannot download the token and gets an error message that would not close. The user should clear the cache to properly log out. 
*   Tables and Export
    *   Table sorting icon does not include numbers <!-- PGDC-35 -->
    *   Table export return correct content (XML, TSV) although file extension is incorrect (always JSON) <!-- PGDC-1986 -->
    *   Nested values are not exported properly <!-- PGDC-910 SV-13 -->
    *   Project table summary chart does not display colors with Internet Explorer 11 <!-- PGDC-1953 -->
    *   Exporting a table containing a very large number of records might trigger an exception (when the server is timing out)<!-- PGDC-1962 -->
    *   Restore default table column arrangement does not restore to the default but it restores to the previous state <!-- PGDC-1769 -->
*   Cart and Download
    *   When saving a file for download, the download pop-up indicates "from:blob:" <!-- PGDC-115 -->
    *   User with update only can also download files also role should not allow <!-- PGDC-1782 -->
    *   User can add more files to the cart than supported if adding files one by one <!-- PGDC-1957 -->
    *   In very specific situations, the cart can display inconsistent data <!-- PGDC-1028 -->   
    *   Make the cart limit warning message more explanatory <!-- PGDC-1952 -->   
    *   In some situations the cart will display the authentication window for authenticated users when trying to download <!-- PGDC-1959 -->   
*   Search
    *   In the data page, some pie chart titles are too long <!-- PGDC-914 -->
    *   Facets displaying histogram do not display a mouseover tooltip if the value is very low <!-- PGDC-913 -->
*   Layout, Browser specific and Accessibility
    *   No favicon is displayed on Internet Explorer <!-- PGDC-1948 -->
    *   When disabling CSS, footer elements are displayed out of order <!-- PGDC-1972 -->
    *   Download cart button does not function properly with Safari 9.0.3 <!-- PGDC-1978 -->
    *   If javascript is disabled html tags are displayed in the warning message <!-- PGDC-1835 -->
    *   Layout issue when browser is reduced to a small window size <!-- PGDC-16 -->
    *   Layout issues when using the browser zoom in function on tables <!-- PGDC-116 -->
*   Non UI-related tickets
    *   Investigations were done on this issue "Authenticated user with access to TARGET projects cannot download Target DCC files. They will get a pop-up message." was related  to ACL permissions (not UI related).
    *   Investigations were done on this issue "Missing pathology files and slide images for some TCGA datasets since they are not connected to the biospecimen chain yet" was related to Data not imported (not UI related).
    *   Associated entities is empty for some files (note: this is a data issue) <!-- PGDC-1887 -->


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).

## Release 0.2.18.3

* __GDC Product__: GDC Data Portal
* __Release Date__: November 23, 2015

### New Features and Changes

* BAM slicing UI in the File Entity page (only available for BAM files with BAI files)

### Bugs Fixed Since Last Release

* Disease Type does not auto filter (Data Facet)

### Known Issues and Workarounds

* Missing pathology files and slide images for some TCGA datasets since they are not connected to the biospecimen chain yet
* In some specific situation (complex queries using must_not term), the advanced search might returns incorrect results
* When exporting the annotation table to a file, the create date field is not displayed in the same format than the UI
* In the advanced search, adding a trailing space to the query causes the UI to display a syntax error although the query is valid
* Checksum missing for MAGE-TAB files
* User can download large files (> 1GB) from the Portal but it fails when extracting the tar.gz file. User should use the GDC Download Transfer Tool for large downloads
* Authenticated user with access to TARGET projects cannot download Target DCC files. They will get a pop-up message.
* In Case Entity Page, the "Download Clinical XML" button is not applicable to TARGET projects. The label is incorrect but the functionality is accurate.
* The advanced search suggests more fields than expected (e.g. "cases.aliquots" should not be displayed). User should remove the field from the query and use the long field name (e.g. "cases.samples.portions.analytes.aliquots")
* If a user has previously logged into the Portal and left a session without logging out, if he comes back to the Portal after his sessionID expires, it looks like the user is still authenticated. The user cannot download the token and gets an error message that would not close. The user should clear the cache to properly log out.
* Sorting on size and removing files from cart does not work in IE 10
* Annotations page not section 508 compliant
* Facets are displayed above the results table when window is small
* Table sorting icon does not include numbers

Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).

## Release 0.2.15-oicr4

* __GDC Product__: GDC Data Portal
* __Release Date__: October 1, 2015

### New Features and Changes

*   n/a

### Bugs Fixed Since Last Release

*   Data access and subtype units are correct in the data download statistics report
*   Improved 508 compliance of the portal (including the data download statistics report)
*   Addressed an issue with the user getting an empty page in specific browsing situations
*   "My projects" filter has been re-enabled for users authenticated with eRACommons but without dbGap authorization.

### Known Issues and Workarounds

*   Missing pathology files and slide images for some TCGA datasets since they are not connected to the biospecimen chain yet
*   In some specific situation (complex queries using must_not term), the advanced search might returns incorrect results
*   When exporting the annotation table to a file, the create date field is not displayed in the same format than the UI
*   In the advanced search, adding a trailing space to the query causes the UI to display a syntax error although the query is valid
*   Checksum missing for MAGE-TAB files
*   User can download large files (> 1GB) from the Portal but it fails when extracting the tar.gz file. User should use the GDC Download Transfer Tool for large downloads
*   Authenticated user with access to TARGET projects cannot download Target DCC files. They will get a pop-up message.
*   In Case Entity Page, the "Download Clinical XML" button is not applicable to TARGET projects. The label is incorrect but the functionality is accurate.
*   The advanced search suggests more fields than expected (e.g. "cases.aliquots" should not be displayed). User should remove the field from the query and use the long field name (e.g. "cases.samples.portions.analytes.aliquots")
*   If a user has previously logged into the Portal and left a session without logging out, if he comes back to the Portal after his sessionID expires, it looks like the user is still authenticated. The user cannot download the token and gets an error message that would not close. The user should clear the cache to properly log out.
*   Sorting on size and removing files from cart does not work in IE 10
*   Annotations page not section 508 compliant
*   Facets are displayed above the results table when window is small
*   Table sorting icon does not include numbers

Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).

## Release 0.2.15-oicr2

* __GDC Product__: GDC Data Portal
* __Release Date__: August 31, 2015

### New Features and Changes

*   Authentication, Authorization & Pop-up messages:
    *   If a user tries to download a related file that is controlled from the File Entity Page, he will get a pop-up message with appropriate guidance ("Login" or "No access to the file")
    *   If a user tries to download a controlled clinical file from the Case page, he will get a pop-up message to indicate that he does not have access to the file
    *   If a user authenticates to the Portal with an eRa account without dbGap authorization, he will get a warning message. Then he will have the choice to logout or to continue browsing the Portal with his eRa account. If he continues browsing the Portal with his eRa account, "My projects" filter will be hidden (temporary solution).
*   If User downloads large files (from the Cart or from the File table), the Portal displays a spinner to indicate the download is in progress
*   Data Download Report does not show the "Data Level" section anymore.

### Bugs Fixed Since Last Release

*   The add to Cart button in File Entity Page changes its display value following a click. User can then click on "Remove from Cart"
*   Total Case value in the Cart matches with the number of cases associated with the files in the Cart
*   When User is authenticated, "My project flag" in Case table indicates that the Cases belongs to his projects
*   In Projects table, if User clicks on the count on Files, it links to the Data page - File table
*   In File Entity Page, if there are no associated cases, it will display the message "No Cases Found." instead of "No Participants Found."

### Known Issues and Workarounds

*   Missing pathology files and slide images for some TCGA datasets since they are not connected to the biospecimen chain yet
*   In some specific situation (complex queries using must_not term), the advanced search might returns incorrect results
*   When exporting the annotation table to a file, the create date field is not displayed in the same format than the UI
*   In the advanced search, adding a trailing space to the query causes the UI to display a syntax error although the query is valid
*   Checksum missing for MAGE-TAB files
*   Data download statistics report is not Section 508 compliant
*   User can download large files (> 1GB) from the Portal but it fails when extracting the tar.gz file. User should use the GDC Download Transfer Tool for large downloads
*   Authenticated user with access to TARGET projects cannot download Target DCC files. They will get a pop-up message.
*   In Case Entity Page, the "Download Clinical XML"button is not applicable to TARGET projects. The label is incorrect but the functionality is accurate.
*   The advanced search suggests more fields than expected (e.g. "cases.aliquots" should not be displayed). User should remove the field from the query and use the long field name (e.g. "cases.samples.portions.analytes.aliquots")
*   Removing a filter after paginating table results could cause no results to be displayed. User should press Clear on the filters and start again.
*   If a user has previously logged into the Portal and left a session without logging out, if he comes back to the Portal after his sessionID expires, it looks like the user is still authenticated. The user cannot download the token and gets an error message that would not close. The user should clear the cache to properly log out.

Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).

## Release 0.2.15-oicr1

* __GDC Product__: GDC Data Portal
* __Release Date__: August 12, 2015

### New Features and Changes

*   Renamed all references of High Performance Download Client (HPDC) to GDC Data Transfer Tool
*   Implemented more links between the Projects page and the Data page
*   Improved usability and visual experience:
    *   Better handling of login failure
    *   User feedback when no results are available following a search
    *   Warning for unsupported browsers (Internet Explorer 9 is not supported)

### Bugs Fixed Since Last Release

*   Optimizations for downloading large files from the browser
*   Fixed various issues related to file search
*   Fixed Section 508 compliance issues
*   Clicking on "Total case" link in the Project List Page does not return results.

### Known Issues and Workarounds

*   Missing pathology files and slide images for some TCGA datasets since they are not connected to the biospecimen chain yet
*   In some specific situation (complex queries using must_not term), the advanced search might returns incorrect results
*   The add to cart button in the files data view does change its display value following a click (file is correctly added to cart though)
*   When exporting the annotation table to a file, the create date field is not displayed in the same format than the UI
*   In the advanced search, adding a trailing space to the query causes the UI to display a syntax error although the query is valid
*   Checksum missing for MAGE-TAB files
*   Data download statistics report is not Section 508 compliant

Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).

## Release 0.2.13

* __GDC Product__: GDC Data Portal
* __Release Date__: July 23, 2015

### New Features and Changes

*   Improved data portal searching. Three search mechanisms are available to the user:
    *   **Facet search**: Starting with all content available on the GDC, allow users to filter down their search by clicking on elements on the left of the screen. This feature is available in **Projects** and **Data** page. Range support was also added to facets in this release
    *   **Advanced Search**: Starting with all content available on the GDC, allow users to build a custom and complex query using all of GDC capabilities (any field with the parameter of their choice).
    *   **Quick Search**: When looking for specific portal element, allow users to launch the **Quick Search** by clicking the "?" or "CRTL+SPACE" and find high-level informations of some entities.
*   Updated styling to align with NCI new visual identify
*   Created a pie chart widget allowing user to easily switch between a pie chart and a table view.
*   Improved Usability and visual experience:
    *   Added tooltips to various sections of the portal
    *   Added a range facet with barchart
    *   Add more charts (summary, cart)

### Bugs Fixed Since Last Release

*   Hooked-up reports to real data
*   Fixed various issues on GQL (Advanced Search)
*   Table export to export appropriate columns
*   Allow users to sort project list table

### Known Issues and Workarounds

*   Checksum missing for MAGE-TAB files

Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).

## Release 0.1.10

* __GDC Product__: GDC Data Portal
* __Release Date__: March 18, 2015

### New Features and Changes

*   Authentication and Authorization
    *   Allow users to authenticate to the portal using their to ERA Commons credentials
    *   Allow users to narrow down their search based on their own projects
    *   Allow users to download a token link (for use with the API) in the portal
    *   Allow users to download protected data (if they have permissions to do so)
*   Reports
    *   Allow users to view a data download statistics report
*   Improve usability and visual experience:
    *   Allow users to view projects data using a new type of graph ("githut" style)
    *   Implement more features into the table widget (sort per column, hide/show columns, re-arrange columns, export to JSON, XML, CSV, TSV, maintain user preferences)
    *   Display UI and API version at the bottom of the page
*   Search
    *   Allow users to select multiple terms in facets (OR operand)
    *   Improvements on advance search with auto-loading of possible fields
*   Cart
    *   Allow users to view more details through additional pie charts
    *   Allow users to download a manifest from the cart
    *   Improved the mechanism of adding data to cart in various sections of the portal.
*   Updated style/theme to match GDC Website
*   Display a NCI Warning banner to inform users about GDC policy

### Bugs Fixed Since Last Release

*   Improvements in 508 compliance

### Known Issues and Workarounds

*   TARGET data is currently not available
*   Data:
    *   Absence of "real" download statistics for the data download statistics report
    *   Missing checksum for magetab files in file entity page
*   Project list table not sortable

Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).

## Release 0.1.8

* __GDC Product__: GDC Data Portal
* __Release Date__: January 22, 2015

### New Features and Changes

*   Allow users to perform a project search and obtain a list of projects (Project Search and List Pages)
*   Allow users to retrieve project details (Project Entity Page)
*   Allow users to perform an annotations search and obtain a list of annotations (Annotations Search and List Pages)
*   Allow users to retrieve entity details (Annotation Entity Page)
*   Implement a basic search feature (Basic Search - Facets))
*   Implement an advance search feature (Advanced Search - Query Language)
*   Allow users to retrieve file details (File Entity Page)
*   Allow users to retrieve participant details (Participant Entity Page)
*   Allow users to view and add files to a cart (Cart)
*   Allow users to view reports (Reports - Data Download Statistics)
*   Allow users to export tables of search results (Export Tables)
*   Allow users to download files (Download)
*   Allow users to authenticate using eRA Commons (Authentication)

### Bugs Fixed Since Last Release

*   Initial Release - Not Applicable

### Known Issues and Workarounds

*   TARGET data is currently not available

Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).


# Data Portal Release Notes

| Version | Date |
|---|---|
| [v1.30.4](Data_Portal_V1_Release_Notes.md#release-1304) | May 11, 2023 |
| [v1.30.0](Data_Portal_V1_Release_Notes.md#release-1300) | July 8, 2022 |
| [v1.29.0](Data_Portal_V1_Release_Notes.md#release-1290) | August 23, 2021 |
| [v1.28.0](Data_Portal_V1_Release_Notes.md#release-1280) | May 17, 2021 |
| [v1.25.1](Data_Portal_V1_Release_Notes.md#release-1251) | August 14, 2020 |
| [v1.25.0](Data_Portal_V1_Release_Notes.md#release-1250) | July 2, 2020 |
| [v1.24.1](Data_Portal_V1_Release_Notes.md#release-1240) | March 10, 2020 |
| [v1.23.1](Data_Portal_V1_Release_Notes.md#release-1231) | December 10, 2019 |
| [v1.23.0](Data_Portal_V1_Release_Notes.md#release-1230) | November 6, 2019 |
| [v1.22.0](Data_Portal_V1_Release_Notes.md#release-1220) | July 31, 2019 |
| [v1.21.0](Data_Portal_V1_Release_Notes.md#release-1210) | June 5, 2019 |
| [v1.20.0](Data_Portal_V1_Release_Notes.md#release-1200) | April 17, 2019 |
| [v1.19.0](Data_Portal_V1_Release_Notes.md#release-1190) | February 20, 2019 |
| [v1.18.0](Data_Portal_V1_Release_Notes.md#release-1180) | December 18, 2018 |
| [v1.17.0](Data_Portal_V1_Release_Notes.md#release-1170) | November 7, 2018 |
| [v1.16.0](Data_Portal_V1_Release_Notes.md#release-1160) | September 27, 2018 |
| [v1.15.0](Data_Portal_V1_Release_Notes.md#release-1150) | August 23, 2018 |
| [v1.14.0](Data_Portal_V1_Release_Notes.md#release-1140) | June 13, 2018 |
| [v1.13.0](Data_Portal_V1_Release_Notes.md#release-1130) | May 21, 2018 |
| [v1.12.0](Data_Portal_V1_Release_Notes.md#release-1120) | February 15, 2018 |
| [v1.11.0](Data_Portal_V1_Release_Notes.md#release-1110) | December 21, 2017 |
| [v1.10.0](Data_Portal_V1_Release_Notes.md#release-1100) | November 16, 2017 |
| [v1.9.0](Data_Portal_V1_Release_Notes.md#release-190) | October 24, 2017 |
| [v1.8.0](Data_Portal_V1_Release_Notes.md#release-180)| August 22, 2017 |
| [v1.6.0](Data_Portal_V1_Release_Notes.md#release-160) | June 29, 2017 |
| [v1.5.2](Data_Portal_V1_Release_Notes.md#release-152) | May 9, 2017 |
| [v1.4.1](Data_Portal_V1_Release_Notes.md#release-141) | October 31, 2016 |
| [v1.3.0](Data_Portal_V1_Release_Notes.md#release-130) | September 7, 2016 |
| [v1.2.0](Data_Portal_V1_Release_Notes.md#release-120) | August 9th, 2016 |
| [v1.1.0](Data_Portal_V1_Release_Notes.md#release-110) | June 1st, 2016 |
| [v1.0.1](Data_Portal_V1_Release_Notes.md#release-101) | May 18, 2016 |

---
## Release 1.30.4

* __GDC Product__: GDC Data Portal
* __Release Date__:  May 11, 2023

### New Features and Changes

* The GDC Legacy Archive has officially been retired.
    * The Legacy Archive Portal can no longer be reached.
    * Any API call to query files from the Legacy Archive will no longer work.
    * Downloads for files from the Legacy Archive will work normally with manifests that were generated previously.


### Bugs Fixed Since Last Release

* The Clinical TSV download in the case entity page and cart now contain TSVs for pathology detail, follow up, and molecular test entities.   <!--SV-2004-->
* Fixed bug in which demographic information would not download as a TSV when diagnosis information was not available. <!--DEV-1238-->

### Known Issues and Workarounds

*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  Negative numbers may be displayed for the Missing value category in the Treatment node within a Clinical Analysis.  This occurs with projects that have multiple treatment nodes per case. All other values should be accurate. <!--SV-1604-->
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->

## Release 1.30.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  July 8, 2022

### New Features and Changes

* None

### Bugs Fixed Since Last Release

* Fixed error in which the manifest could not be downloaded directly from the GDC Data Portal in certain instances. <!--SV-2112-->

### Known Issues and Workarounds

*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  Negative numbers may be displayed for the Missing value category in the Treatment node within a Clinical Analysis.  This occurs with projects that have multiple treatment nodes per case. All other values should be accurate. <!--SV-1604-->
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    * The footer says version 1.9, but it is actually 1.13
    *	Filtering by vital_status does not function in the Legacy Archive due to updates in how this property has been indexed.  A workaround is to perform the case level filtering in the GDC Data Portal and copy the filter string for use in the Legacy Archive or the legacy API. <!--SV-1508-->
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->


## Release 1.29.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  August 23, 2021

### New Features and Changes <!--REQ-431-->

* None

### Bugs Fixed Since Last Release

* Fixed error in which the data summaries in the clinical analysis and cart pages were only partially displayed when viewed in Chrome v.91.0.4472. <!--SV-1940-->

### Known Issues and Workarounds

*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  Negative numbers may be displayed for the Missing value category in the Treatment node within a Clinical Analysis.  This occurs with projects that have multiple treatment nodes per case. All other values should be accurate. <!--SV-1604-->
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    * The footer says version 1.9, but it is actually 1.13
    *	Filtering by vital_status does not function in the Legacy Archive due to updates in how this property has been indexed.  A workaround is to perform the case level filtering in the GDC Data Portal and copy the filter string for use in the Legacy Archive or the legacy API. <!--SV-1508-->
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->

## Release 1.28.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  May 17, 2021

### New Features and Changes <!--REQ-422-->

* New columns were added to the "molecular test" table at the bottom of the case entity page to display additional molecular test fields.

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

*  When accessing the data portal with Chrome v.91.0.4472, users may experience some display errors.  This includes the data summary in the clinical analysis and cart pages being only partially displayed. <!--SV-1940-->
*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  Negative numbers may be displayed for the Missing value category in the Treatment node within a Clinical Analysis.  This occurs with projects that have multiple treatment nodes per case. All other values should be accurate. <!--SV-1604-->
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    * The footer says version 1.9, but it is actually 1.13
    *	Filtering by vital_status does not function in the Legacy Archive due to updates in how this property has been indexed.  A workaround is to perform the case level filtering in the GDC Data Portal and copy the filter string for use in the Legacy Archive or the legacy API. <!--SV-1508-->
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->

## Release 1.25.1

* __GDC Product__: GDC Data Portal
* __Release Date__:  August 14, 2020

### New Features and Changes <!--REQ-408-->

* API improvements were made to increase portal performance.

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  Negative numbers may be displayed for the Missing value category in the Treatment node within a Clinical Analysis.  This occurs with projects that have multiple treatment nodes per case. All other values should be accurate. <!--SV-1604-->
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    * The footer says version 1.9, but it is actually 1.13
    *	Filtering by vital_status does not function in the Legacy Archive due to updates in how this property has been indexed.  A workaround is to perform the case level filtering in the GDC Data Portal and copy the filter string for use in the Legacy Archive or the legacy API. <!--SV-1508-->
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->

## Release 1.25.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  July 2, 2020

### New Features and Changes <!--REQ-395-->

* Suppressed Experimental Strategy filter on the Exploration page as this currently filters for files with a particular strategy, not for cases.  This may cause confusion amongst users.  The filter will be re-instated in a future release once the logic is available to filter more appropriately for cases tied to a specific strategy. <!--PRTL-3119-->
* Updated the filter control panel styling across the Portal to have clearer titles (e.g. "Search Cases" instead of "Cases" in the quick search box). <!--PRTL-2987-->
* Made minor updates to the styling of the filter query display at the top of the Exploration page (spacing, borders). <!--PRTL-2988-->
* Added an expand/collapse control to the quick search bar of Clinical tab on the Exploration page, to be consistent with other Exploration tabs. <!--PRTL-2990-->
* Added a clear title above the counts in each filter control panel across the Portal (e.g. "# Cases", "# Genes", etc.). <!--PRTL-2991-->
* Moved various action buttons above the results table on the Repository Page to more accessible locations. <!--PRTL-2994-->
* Improved load time of the initial custom filter list on the Repository Page, when clicking "Add a Filter Filter" or "Add a Case/Biospecimen Filter". <!--PRTL-3057-->

### Bugs Fixed Since Last Release

* Fixed a bug in the Age at Diagnosis table on the Cohort Comparison page, where the # of cases in the table was not consistent with the # of cases shown when clicking the link to the Exploration page. <!--PRTL-3032-->
* Fixed minor positional accuracy issue of the lollipop data points on the Protein Viewer. <!--PRTL-2967-->
* Fixed bug on the Protein Viewer where, if clicking to switch between different lollipop data points, details of the previous lollipop was not closing. <!--PRTL-2968-->
* Fixed bug where the quick search bar on the Exploratin Page's Genes filter tab was not expanding/collapsing properly. <!--PRTL-2989-->
* Fixed bug in the pop-up warning message when adding or removing items from the Cart, where long filenames were spilling outside the border of the pop-up. <!--PRTL-3031-->
* Fixed typo in the "View Cases in Exploration" button on the Repository page. <!--PRTL-3122-->
* Fixed typo in the pop-up user consent message when downloading controlled files from the Cart. <!--PRTL-3123-->

### Known Issues and Workarounds

*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  Negative numbers may be displayed for the Missing value category in the Treatment node within a Clinical Analysis.  This occurs with projects that have multiple treatment nodes per case. All other values should be accurate. <!--SV-1604-->
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    * The footer says version 1.9, but it is actually 1.13
    *	Filtering by vital_status does not function in the Legacy Archive due to updates in how this property has been indexed.  A workaround is to perform the case level filtering in the GDC Data Portal and copy the filter string for use in the Legacy Archive or the legacy API. <!--SV-1508-->
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->

## Release 1.24.1

* __GDC Product__: GDC Data Portal
* __Release Date__:  March 10, 2020

### New Features and Changes <!--REQ-395-->

* Removed unnecessary comma and y-axis value from title of the mutation details pop-up in the Protein Viewer. <!--PRTL-2969-->
* Added Tobacco Smoking Status field to the Exposures tab on the Case entity page. <!--PRTL-2952-->
* Added a link to the Cart where users can access instructions for downloading the GDC Genome Build reference files. <!--PRTL-2929-->
* Added logic to prevent duplicate fetching of data for Clinical Analysis survival plots and optimize rendering. <!--PRTL-2910-->
* Added a button to clear searches for certain Portal search controls that were previously missing this ability. <!--PRTL-2900-->
* Reduced whitespace between Oncogrid and its control panel to optimize spacing and layout. <!--PRTL-2899-->
* Made entire Clinical Analysis results page responsive (card columns now scale & stack in response to the size of the browser window). <!--PRTL-2882-->
* Replaced Clinical Analysis function for printing clinical cards to a single PDF file, with more flexible functionality to instead download all the cards in SVG and/or PNG format. <!--PRTL-2870-->
* Added message to notify users when they try to access the Portal using Microsoft Internet Explorer, indicating which browsers are officially supported. <!--PRTL-2868-->
* Added arrow icon to sortable columns across the Portal to indicate the current sort direction. <!--PRTL-2330-->

### Bugs Fixed Since Last Release

* Fixed bug where clicking a primary site on the Human Body Image was not re-directing to the Exploration page. <!--PRTL-2979-->
* Fixed layout issue where long Annotation Notes were exceeding the border of the text box. <!--PRTL-2944-->
* Fixed layout issue where the Repository header and action buttons were scaling and wrapping incorrectly if the browser window is shrunk beyond a certain threshold. <!--PRTL-2938-->
* Fixed layout issue where the responsive Clinical Analysis Cards were clipping improperly as the browser window is shrunk beyond a certain threshold. <!--PRTL-2928-->
* Fixed bug where the Clinical Tab on the Exploration page was crashing when entering a custom range of Years for the Age at Diagnosis facet. <!--PRTL-2913-->
* Fixed various minor cosmetic and color issues in PNG, SVG downloads of the Clinical Analysis survival plots. <!--PRTL-2887-->
* Fixed bug where the x-axis in PNG, SVG downloads of histograms across the Portal was being bolded incorrectly. <!--PRTL-2885-->
* Fixed bug where the expand/collapse symbols in the UI were incorrectly being exported in the TSV download of the Projects table. <!--PRTL-2883-->
* Fixed bug where Oncogrid's modal for customizing colors could not be scrolled below the fold if it was shrunk beyond a certain threshold. <!--PRTL-2881-->
* Fixed incorrect DTT hyperlink in the GDC Apps menu. <!--PRTL-2867-->
* Fixed bug where the "dbSNP rs ID" facet could not be minimized in the Exploration page's Mutations facet tab. <!--PRTL-2840-->
* Fixed layout issue where the Portal's header incorrectly overlaps some content when a notification banner is displayed. <!--PRTL-2753-->
* Fixed some minor layout & styling issues in the Exploration page's facets panel. <!--PRTL-2614-->
* Fixed bug where the Case ID on the Exploration page's Cases facet tab was not searchable in certain scenarios. <!--PRTL-2587-->
* Fixed bug where the Expand/Collapse button state was not changing properly when being used in the Biospecimen section of the Case entity page. <!--PRTL-2575-->
* Fixed incorrect capitalization of "dbGaP" in the Summary section of the Project entity page. <!--PRTL-2436-->
* Fixed layout issue where the Advanced Search query box on the Repository page could expanded beyond the margins of the box's border. <!--PRTL-2271-->

### Known Issues and Workarounds

*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  Negative numbers may be displayed for the Missing value category in the Treatment node within a Clinical Analysis.  This occurs with projects that have multiple treatment nodes per case. All other values should be accurate. <!--SV-1604-->
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    * The footer says version 1.9, but it is actually 1.13
    *	Filtering by vital_status does not function in the Legacy Archive due to updates in how this property has been indexed.  A workaround is to perform the case level filtering in the GDC Data Portal and copy the filter string for use in the Legacy Archive or the legacy API. <!--SV-1508-->
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->

## Release 1.23.1

* __GDC Product__: GDC Data Portal
* __Release Date__:  December 10, 2019

### New Features and Changes <!--REQ-396-->

* Updated display of x-axis units on the homepage Human Body chart to more easily display increased case counts for newly-added projects <!--PRTL-2925-->

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  Negative numbers may be displayed for the Missing value category in the Treatment node within a Clinical Analysis.  This occurs with projects that have multiple treatment nodes per case. All other values should be accurate. <!--SV-1604-->
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    * The footer says version 1.9, but it is actually 1.13
    *	Filtering by vital_status does not function in the Legacy Archive due to updates in how this property has been indexed.  A workaround is to perform the case level filtering in the GDC Data Portal and copy the filter string for use in the Legacy Archive or the legacy API. <!--SV-1508-->
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->

## Release 1.23.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  November 6, 2019

### New Features and Changes <!--REQ-390-->

* Added Clinical Data Analysis feature that allows Users to: <!--FEAT-520, FEAT-521-->
    * Explore clinical data via the new Clinical Tab on the Exploration page.
    * Build custom Case sets based on that clinical data for later analysis.
    * Create an analysis to examine the clinical variables in a Case set, using various tools including histograms, survival plots, box plots, QQ plots, and custom binning.
    * Download the data (as TSV, JSON) and plots (as PNG, SVG) of each clinical variable in an anlysis.
    * Save an analysis to local storage to resume later (as long as storage is not cleared).
* Added links to CIViC annotations on the Gene and Mutation entity pages. <!--PRTL-2543-->
* Updated the default Top Mutated Genes histogram on the Exploration page to display only COSMIC Genes by default. <!--PRTL-2371-->
* Added Follow-Ups tab and nested Molecular Tests to Case entity page. <!--PRTL-2704-->
* Added text to BAM slicing modal to instruct Users how to access unmapped reads. <!--PRTL-2620-->

### Bugs Fixed Since Last Release

* Fixed font in exported PNGs, SVGs to be consistent with the Portal UI. <!--PRTL-2653-->
* Made custom Case and File filters in the Repository page case insensitive. <!--PRTL-2571-->
* Fixed bug where pfam domains in Protein Viewer could not be clicked in Firefox. <!--PRTL-2511-->
* Fixed bug where TSV download button could not be clicked in MS Edge. <!--PRTL-2357-->
* Fixed controlled access alert pop-up in the Cart so that the modal disappears correctly once the User has successfully logged in and initiated the download. <!--PRTL-2429-->

### Known Issues and Workarounds

*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  Negative numbers may be displayed for the Missing value category in the Treatment node within a Clinical Analysis.  This occurs with projects that have multiple treatment nodes per case. All other values should be accurate. <!--SV-1604-->
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    * The footer says version 1.9, but it is actually 1.13
    *	Filtering by vital_status does not function in the Legacy Archive due to updates in how this property has been indexed.  A workaround is to perform the case level filtering in the GDC Data Portal and copy the filter string for use in the Legacy Archive or the legacy API. <!--SV-1508-->
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->

## Release 1.22.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  July 31, 2019

### New Features and Changes <!--REQ-387-->

* Replaced existing Clinical, Biospecimen columns on the Projects page with 4 columns: Clinical, Clinical Supplement, Biospecimen, Biospecimen Supplement. The Clinical and Biospecimen columns now link directly to the project page, and their counts indicate the total cases in the project. The Clinical Supplement and Biospecimen Supplement columns work the same as the old Clinical and Biospecimen columns - They link to the Repository page with Files filtered based on the Project and Data Category (Clinical or Biospecimen). <!--PRTL-2528-->
* Added a new icon to the GDC Apps menu, which links to the GDC Publications website page. <!--PRTL-2547-->
* Added the Synchronous Malignancy field to the Diagnoses / Treatments tab on the Case entity page. <!--PRTL-2582-->
* Added the Pack Years Smoked field to the Exposures tab on the Case entity page. <!--PRTL-2584-->
* Increased length of x-axis labels on histograms to 10 characters so that projects with names that are typically standard 10 chars will display fully (e.g. most TCGA projects like TCGA-BRCA). <!--PRTL-2598-->

### Bugs Fixed Since Last Release

* Fixed bug where the PNG, SVG files for the Overall Survival Plot could not be downloaded. <!--PRTL-2528-->

### Known Issues and Workarounds

*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Filtering by vital_status does not function in the Legacy Archive due to updates in how this property has been indexed.  A workaround is to perform the case level filtering in the GDC Data Portal and copy the filter string for use in the Legacy Archive or the legacy API. <!--SV-1508-->
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->   

## Release 1.21.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  June 5, 2019

### New Features and Changes <!--REQ-383-->

* Changed all Survival Plots to display the Duration (x-axis) in years instead of days. <!--PRTL-2404-->
* Updated data references to clinical properties throughout the Portal to match the underlying changes in the GDC data dictionary. <!--PRTL-2459-->

### Bugs Fixed Since Last Release

* Fixed bug where X-axis labels in histograms were cut off when displayed. <!--PRTL-1896-->
* Renamed the 'Experimental Strategies' facet on the Projects page to singular form. <!--PRTL-2262-->
* Fixed bug where columns with a % value of infinity (due to division by zero) show as 'NaN%'.  Replaced instead with a label of '--'. <!--PRTL-2384-->
* Fixed bug where the download button in the cart access banner was still disabled after a user logged in from the banner.  Instead, the experience is now improved so that after login, the banner is closed and the user must explicitly click 'Download' again. <!--PRTL-2393-->
* Fixed bug where if a new user logs into the Portal and views their profile, the app crashes if the user has no projects assigned yet. <!--PRTL-2529-->
* Fixed bug where Survival Rate numbers in the Survival Plot plot y-axis did not scale properly and overlapped into the axis lines. <!--PRTL-2530-->

### Known Issues and Workarounds

*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->   

## Release 1.20.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  April 17, 2019

### New Features and Changes <!--REQ-382-->

* Upgraded the Portal to use the latest React Javascript library (version 16.8) <!--PRTL-2440-->

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    

## Release 1.19.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  February 20, 2019

### New Features and Changes <!--REQ-381-->

* Added support for viewing of controlled-access mutations in the Data Portal
* Added a new data access notification to remind logged-in users with access to controlled data that they need to follow their data use agreement.  The message is fixed at the top of the Portal.<!--PRTL-2400, PRTL-2434-->
* Added the ability to search for previous versions of files.  If the user enters the UUID of a previous version that cannot be found, the Portal returns the UUID of the latest version available. <!--PRTL-2387-->
* Renamed the Data Category for "Raw Sequencing Data" to "Sequencing Reads" throughout the portal where this appears, to be consistent with the Data Dictionary. <!--PRTL-118-->
* Added a link in the Portal footer to the GDC support page. <!--PRTL-2383-->

### Bugs Fixed Since Last Release

* Fixed bug where Survival Plot button never stops loading if plotting mutated vs. non-mutated cases for a single Gene. <!--PRTL-2398-->
* Fixed inconsistent button styling when downloading controlled Downstream Analyses Files from File Entity page. <!--PRTL-2395-->
* Removed unnecessary Survival column from Arrange Columns button on Case Entity, Gene Entity pages. <!--PRTL-2281-->
* Removed unnecessary whitespace from pie charts on Repository page. <!--PRTL-1923-->
* Added missing File Size unit to Clinical Supplement File, Biospecimen Supplement File tables on Case Entity page. <!--PRTL-2070-->
* Fixed bug where clicking on Case Counts in Projects Graph tab was going to the Repository Files tab instead of the Cases tab. <!--PRTL-2272-->
* Fixed bug where the counts shown beside customer filters on the Repository Cases tab were not updating when filtering on other facets. <!--PRTL-2412-->
* Fixed bug where clicking the # of Affected Cases denominator on the Gene page's Most Frequent Somatic Mutations table displayed an incorrect number of Cases.

### Known Issues and Workarounds

*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    

## Release 1.18.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  December 18, 2018

### New Features and Changes <!--REQ-335-->

* A new data access message has been added when downloading controlled data.  Users must agree to abide by data access control policies when downloading controlled data. <!--PRTL-2324,PRTL-2370,-->
* In the Mutation free-text search in Exploration, mutation display now includes the UUID, genomic location, and matched search term for easier mutation searching. <!--PRTL-1996, PRTL-2367-->
* The ability to sort on ranked columns has been made available. <!--PRTL-2274,PRTL-2336,PRTL-2365-->

### Bugs Fixed Since Last Release

*  In some cases, text was being cut off on the Project page visualization tab.  Text is no longer cut off. <!--PRTL-2290-->
*  HGNC link on Gene page broke as the source format url changed; The format was updated and the link is now functional <!--PRTL-2380-->
*  In the biospecimen details on the Case page, the cart icon would disappear once clicked.  It now is always visible. <!--PRTL-2282-->

### Known Issues and Workarounds

*  Pre-release Data Portal login is not supported on Internet Explorer or the last version of Edge (42).  Edge 41 does login successfully.
*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    

## Release 1.17.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  November 7, 2018

### New Features and Changes <!--REQ-334-->

* Copy Number Variation (CNV) data derived from GISTIC results are now available in the portal:
	* View number of CNV events on a gene in a cohort in the Explore Gene table tab <!-- PRTL-2259, PRTL-2328 -->
	* Explore CNVs associated with a gene on the Gene Entity Page <!-- PRTL-2252,PRTL-2273,PRTL-2344 -->
	* Explore CNVs concurrently with mutations on the Oncogrid with new visualization <!-- PRTL-2244,PRTL-2251,PRTL-2256,PRTL-2257,PRTL-2275,PRTL-2276, PRTL-2214, PRTL-2314, PRTL-2315, PRTL-2317, PRTL-2319, PRTL-2325, PRTL-2326, PRTL-2327,PRTL-2266  -->

### Bugs Fixed Since Last Release

*  None

### Known Issues and Workarounds

*  Custom Facet Filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  SIFT and PolyPhen annotations are missing from the export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    

## Release 1.16.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  September 27, 2018

### New Features and Changes <!--REQ-333-->

* Updated Human Body Image to aggregate all current primary sites to available Major Primary Sites <!--PRTL-2248-->


### Bugs Fixed Since Last Release

* Fixed link on cart download error popup <!--PRTL-2284-->
* Updated Cancer Distribution table to have dropdown menus for primary_site and disease_type <!--PRTL-2286-->
* Updated Y-axis label on `Top Mutated Cancer Genes in Selected Projects Graph` <!--PRTL-2264-->
* Updated Set Operation Image to remove stray text <!--PRTL-2214-->

### Known Issues and Workarounds

*  Advanced Search
    * For advanced search and custom file facet filtering there are some properties that will appear as options that are no longer supported (e.g. file_state). <!--API-530-->
*  Custom facet filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  SIFT and PolyPhen annotations are missing from the Export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    


## Release 1.15.0

* __GDC Product__: GDC Data Portal
* __Release Date__:  August 23, 2018

### New Features and Changes <!--REQ-329-->

* File Versions are now visible in the "File Versions" section on the File Entity Page. <!--PRTL-2015-->
* "View Files in Repository" and "View Cases in Repository" button methods were updated to work faster. <!--PRTL-2003 PRTL-2106 -->

### Bugs Fixed Since Last Release

* Fixed warning messages that prompted users to login even when already logged in.  Error warnings now correctly prompt users to reference dbGAP for data access if already signed in. <!--PRTL-1937-->
* Fixed error where you could click Go on Case ID wildcard facet before inputting any data. <!--PRTL-2083-->
* Fixed cart header to be a consistent color for the whole table. <!--PRTL-1943-->
* Fixed error where you could save a set with no name or items, which resulted in an infinite spinner. <!--PRTL-2218--> <!--PRTL-1858-->
* Fixed table width issue when FM-AD was selected as a filter. <!--PRTL-2054-->
* Updated broken help link on Advanced Query. <!--PRTL-2218-->

### Known Issues and Workarounds

*  Advanced Search
    * For advanced search and custom file facet filtering there are some properties that will appear as options that are no longer supported (e.g. file_state). <!--API-530-->
*  Custom facet filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  SIFT and PolyPhen annotations are missing from the Export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    


## Release 1.14.0

* __GDC Product__: GDC Data Portal
* __Release Date__: June 13, 2018

### New Features and Changes <!--REQ-322-->

* Added new Experimental Strategies Diagnostic Slide Image, Bisulfite-Seq, ChIP-Seq, and ATAC-Seq to Case and Project entity pages.

### Bugs Fixed Since Last Release

* Fixed download of clinical and biospecimen data from the Repository when Case table rows are selected.


### Known Issues and Workarounds

*  Custom facet filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  SIFT and PolyPhen annotations are missing from the Export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  When user is logged in and try to download a controlled file he does not have access to, he's prompted to log in. He should be promted to request access. <!-- PRTL-1937 -->
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).


## Release 1.13.0

* __GDC Product__: GDC Data Portal
* __Release Date__: May 21, 2018

### New Features and Changes <!--REQ-322-->

*  Added new image viewer functionality for viewing tissue slide images <!--PRTL-2016-->


### Bugs Fixed Since Last Release

* Updated gene reference labels on gene entity page to adhere to preferred usage <!--PRTL-1911-->
* Fixed issue with user profile displaying all projects twice <!--PRTL-2035-->


### Known Issues and Workarounds

*  Custom facet filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  SIFT and PolyPhen annotations are missing from the Export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  When user is logged in and try to download a controlled file he does not have access to, he's prompted to log in. He should be promted to request access. <!-- PRTL-1937 -->
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    



Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).

## Release 1.12.0

* __GDC Product__: GDC Data Portal
* __Release Date__: February 15, 2018

### New Features and Changes <!--REQ-322-->

*  Provided the ability to export clinical and biospecimen data in a TSV format from the Case, Project, Exploration, Repository and Cart pages.<!--PRTL-1929-->
*  Removed from the Project entity page the sections about mutated genes, somatic mutations and affected Cases and replaced with a button "Explore data" that will open the Exploration page filtered on the project. Indeed the Exploration page provides the same information.  Added a breakdown of cases per primary site for a Project entity page with multiple primary sites (e.g. FM-AD). <!--PRTL-1903-->
*  Added display of coding DNA change and impacts for all the transcripts (instead of canonical transcript only) in the Mutation entity page - Consequences section. In the mutation table (e.g. in Repository), the impacts and consequences are displayed for the canonical transcript only. <!--PRTL-1927-->

### Bugs Fixed Since Last Release

*  Replaced the suggested set name when saving a set with selected items, e.g. for case set the suggested name is now "Custom Case selection". <!--PRTL-1911-->
*  Fixed the protein viewer to indicate when there are overlapping mutations. Mousing over the dot showing multiple mutations will open a right panel with the list of all the corresponding mutations.  <!--SV-750-->
*  Fixed Mutation entity page - Consequences table: the "Coding DNA Change" column is now populated for all the transcripts. <!--SV-751-->
*  Fixed download clinical and download biospecimen actions from TCGA-BRCA project.
*  Fixed facet behavior that did not reset back to showing all options after pressing reset-arrow. <!--PRTL-1928-->
*  Fixed error when user was trying to save a set with no value in the textbox "Save top:".  <!--PRTL-1909-->
*  Removed somatic mutation section from Case entity page for cases with no open-access mutation data (e.g. FM-AD or TARGET cases). <!--PRTL-1926-->
*  Fixed error where a blank page appears after unselecting `Cancer Gene Census` mutation facet. <!--PRTL-1933-->
*  Fixed duplicated date in sample sheet name (e.g. gdc_sample_sheet_YYYY-MM-DD_HH-MM.tsv.YYYY-MM-DD_HH-MM.tsv). <!--SV-942-->
*  Fixed error when annotations were not downloaded along with the file (in File entity page and Cart).  <!--SV-401-->

### Known Issues and Workarounds

*  Custom facet filters
    * Some definitions are missing from the property list when adding custom facet file or case filters. <!--SV-989-->
*  Visualizations
    *  SIFT and PolyPhen annotations are missing from the Export JSON of the mutation table. They are present in the export TSV. <!--PRTL-1990-->
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
*  Repository and Cart
    *  When user is logged in and try to download a controlled file he does not have access to, he's prompted to log in. He should be promted to request access. <!-- PRTL-1937 -->
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only. <!-- LGCY-81 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).


## Release 1.11.0

* __GDC Product__: GDC Data Portal
* __Release Date__: December 21, 2017

### New Features and Changes <!--REQ-322-->

* Updated UI to support SIFT and Polyphen annotations <!--PRTL-1404-->
* A `Sample Sheet` can now be created which allows easy association between file names and the case and sample submitter_id <!--PRTL-1872-->
* Updated Advanced Search page to include options to `Add All Files to Cart`, `Download Manifest`, and `View X Cases in Exploration` <!--PRTL-1796-->
* Provide clear message rather than blank screen if survival plots cannot be calculated for particular cohort comparison <!--PRTL-1842-->
* Display sample_type on associated entities section on file page <!--PRTL-1890-->
* Allows for special characters in case, gene, and mutation set upload (`-, :, >, .`) <!--PRTL-1847-->


### Bugs Fixed Since Last Release

*  Fixed error when trying to download large number of files from the Legacy Archive cart <!--LGCY-75-->
*  Fixed number of annotations displayed in Legacy Archive for particular entities <!--LGCY-74-->
*  Replaced missing bars to indicate proportion of applicable files and cases on project entity page in Cases and File Counts by Data Category table <!--PRTL-1725-->
*  Fixed project page display when projects are selected that contain no mutation data in the facet panel <!--PRTL-1754-->
*  Fixed error where exporting case sets as TSV included fewer cases than the total <!--PRTL-1888-->
*  Fixed error in exploration section when adding custom facets.  Previously selecting 'Only show fields with values' did not result in the expected behavior <!--PRTL-1901-->
*  Fixed error where number of associated entities for a file was showing an incorrect number <!--PRTL-1891-->

### Known Issues and Workarounds

*  Sample sheet will download with a file name including the date duplicated (e.g. gdc_sample_sheet_YYYY-MM-DD_HH-MM.tsv.YYYY-MM-DD_HH-MM.tsv)<!--SV-942-->
*  Custom facet filters
    * Definitions are missing from the property list when adding custom facet file or case filters <!--SV-916-->
*  Visualizations
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
    *  In the protein viewer there may be overlapping mutations.  In this case mousing over a point will just show a single mutation and the other mutations at this location will not be apparent.  <!--SV-750-->
*  Entity page
    *  On the mutation entity page, in the Consequences Table, the "Coding DNA Change" column is not populated for rows that do not correspond to the canonical mutation. <!-- SV-751 -->
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only.
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).



## Release 1.10.0

* __GDC Product__: GDC Data Portal
* __Release Date__: November 16, 2017

### New Features and Changes <!--REQ-322-->

* Support for uploading Case and Mutation sets in Exploration page <!--PRTL-1452, PRTL-1453-->
* Support for saving, editing, removing Case, Gene and Mutation sets in the Exploration page <!--PRTL-1472,PRTL-1464, PRTL-1473, PRTL-1468, PRTL-1465, PRTL-1470,PRTL-1469, PRTL-1466, PRTL-1471-->
* Added a Managed Sets menu where the user can see their saved sets <!--PRTL-1597-->
* Added an Analysis menu with two analyses: Set Operation and Cohort Comparison <!--PRTL-1599, PRTL-1600-->
* Added a User Profile page that shows all the projects and permissions assigned to the user: available in the username dropdown after the user logs in <!--PRTL-1458-->

### Bugs Fixed Since Last Release

*  Project page
    *  On the project page, the Summary Case Count link should open the case tab on the Repository page - instead it opens the file page <!--PRTL-1591-->

### Known Issues and Workarounds

*  Custom facet filters
    * Definitions are missing from the property list when adding custom facet file or case filters <!--SV-916-->
    * Selecting 'Only show fields with values' will show some fields without values in the Repository section.  This works correctly under the Exploration section. <!--SV-917-->
*  Visualizations
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
    *  In the protein viewer there may be overlapping mutations.  In this case mousing over a point will just show a single mutation and the other mutations at this location will not be apparent.  <!--SV-750-->
*  Entity page
    *  On the mutation entity page, in the Consequences Table, the "Coding DNA Change" column is not populated for rows that do not correspond to the canonical mutation. <!-- SV-751 -->
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only.
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).

## Release 1.9.0

* __GDC Product__: GDC Data Portal

* __Release Date__: October 24, 2017

### New Features and Changes <!--REQ-317-->

* Support for projects with multiple primary sites per project <!--PRTL-1478, PRTL-1676, PRTL-1668, PRTL-1675,PRTL-1685,PRTL-1695-->
* Support for slides that are linked to `sample` rather than `portion` <!--PRTL-1696-->

### Bugs Fixed Since Last Release

None

### Known Issues and Workarounds
*  Visualizations
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
    *  In the protein viewer there may be overlapping mutations.  In this case mousing over a point will just show a single mutation and the other mutations at this location will not be apparent.  <!--SV-750-->
*  Project page
    *  On the project page, the Summary Case Count link should open the case tab on the Repository page - instead it opens the file page <!--PRTL-1591-->
*  Entity page
    *  On the mutation entity page, in the Consequences Table, the "Coding DNA Change" column is not populated for rows that do not correspond to the canonical mutation. <!-- SV-751 -->
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only.
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).


## Release 1.8.0

* __GDC Product__: GDC Data Portal
* __Release Date__: August 22, 2017

### New Features and Changes

Major features/changes:

* A feature that links the exploration and repository pages was added. For example:
    - In the exploration page, cases with a specific mutation could be selected. This set could then be linked to the repository page to download the data files associated with these cases.
    - In the repository menu, the user can select cases associated with specific files. The set could then be linked to exploration page to view the variants associated with this set of cases.

*  Users can now upload a custom gene list to the exploration page and leverage the GDC search and visualization features for cases and variants associated with the gene set.

*  Filters added for the gene entity page. For example:
    - Clicking on a mutated gene from the project page will display mutations associated with the gene that are present in this project (filtered protein viewer, etc.).
    - Clicking on a mutated gene from the exploration page will display the mutations associated with the gene filtered by additional search criteria, such as "primary site is Kidney and mutation impact is high".

* UUIDs are now hidden from tables and charts to simplify readability. The UUIDs can still be exported and viewed in the tables using the "arrange columns" feature. In the mutation table, UUIDs are automatically exported.

* Mutation entity page - one consequence per transcript is shown (10 rows by default) in the consequence table. The user should display all rows before exporting the table.

### Bugs Fixed Since Last Release
*  Exploration
    *  Combining "Variant Caller" mutation filter with a case filter will display incorrect counts in the mutation facet. The number of mutations in the resulting mutation table is correct. <!-- API-307 -->
    *  Mutation table: it is difficult to click on the denominator in "#Affected Cases in Cohort" column displayed to the left side of the bar. The user should click at a specific position at the top of the number to be able to go to the corresponding link. <!-- PRTL-1377 -->


### Known Issues and Workarounds
*  Visualizations
    *  Data Portal graphs cannot be exported as PNG images in Internet Explorer. Graphs can be exported in PNG or SVG format from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet Explorer does not display chart legend and title when re-opening previously downloaded SVG files, the recommendation is to open downloaded SVG files with another program.
    *  In the protein viewer there may be overlapping mutations.  In this case mousing over a point will just show a single mutation and the other mutations at this location will not be apparent.  <!--SV-750-->
*  Project page
    *  On the project page, the Summary Case Count link should open the case tab on the Repository page - instead it opens the file page <!--PRTL-1591-->
*  Entity page
    *  On the mutation entity page, in the Consequences Table, the "Coding DNA Change" column is not populated for rows that do not correspond to the canonical mutation. <!-- SV-751 -->
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only.
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).


## Release 1.6.0

* __GDC Product__: GDC Data Portal
* __Release Date__: June 29, 2017

### New Features and Changes

There was a major new release of the GDC Data Portal focused on Data Analysis, Visualization, and Exploration (DAVE). Some important new features include the following:

*  New visual for the Homepage: a human body provides the number of Cases per Primary Site with a link to an advanced Cancer Projects search
*  The Projects menu provides the Top 20 Cancer Genes across the GDC Projects and the Case Distribution per Project
*  A new menu "Exploration" is an advanced Cancer Projects search which provides the ability to apply Case, Gene, and Mutation filters to look for:
    * List of Cases with the largest number of Somatic Mutations
    *	The most frequently mutated Genes
    *	The most frequent Variants
    *	Oncogrid view of mutation frequency
*  Visualizations are provided across the Project, Case, Gene and Mutation entity pages:
    *	List of most frequently mutated genes and most frequent variants
    *	Survival plots for patients with or without specific variants
    *	Survival plots for patients with or without variants in specific genes
    *	Lollipop plots of mutation frequency across protein domains
*  Links to external databases (COSMIC, dbSNP, Uniprot, Ensembl, OMIM, HGNC)
*  Quick Search for Gene and Mutation entity pages
*  The ability to export the current view of a table in TSV
*  Retired GDC cBioPortal

_For detailed updates please review the [Data Portal User Guide](../Users_Guide/Getting_Started/)._

### Bugs Fixed Since Last Release

*  BAM Slicing dialog box does not disappear automatically upon executing the BAM slicing function. The box can be closed manually. <!-- PRTL-282 -->
*  Very long URLs will produce a 400 error.  Users may encounter this after clicking on "source files" on a file page where the target file is derived from hundreds of other files such as for MAF files. <!-- SV-396 / PRTL-342-->
*  If bam slicing produces an error pop-up message it will be obscured behind the original dialog box. <!--SV-419-->
    *   Internet Explorer users are not able to use the "Only show fields with no values" when adding custom facets <!-- PGDC-2467 / PRTL-109 -->
    *   Exporting large tables in the Data Portal may produce a 500 error.  Filtering this list to include fewer cases or files should eliminate the error <!--API-223-->    

### Known Issues and Workarounds
*  New Visualizations
    *  Cannot export Data Portal graphs in PNG in Internet Explorer. Graphs can be exported to PNG or SVG from Chrome or Firefox browsers <!-- PRTL-1325 / PRTL-1114 -->. Internet would not display chart legend and title when re-opening previously downloaded SVG files, recommendation is to open downloaded SVG files with another software.
    *  In the protein viewer there may be overlapping mutations.  In this case mousing over a point will just show a single mutation and the other mutations at this location will not be apparent.  <!--SV-750-->
*  Exploration
    *  Combining "Variant Caller" mutation filter with a case filter will display wrong counts in the mutation facet. The number of mutations in the result mutation table is correct. <!-- API-307 -->
    *  Mutation table: it is difficult to click on the denominator in "#Affected Cases in Cohort" column displayed to the left side of the bar. The user should click at a specific position at the top of the number to be able to go to the corresponding link. <!-- PRTL-1377 -->
*  Entity page
    *  On the mutation entity page, in the Consequences Table, the "Coding DNA Change" column is not populated for rows that do not correspond to the canonical mutation. <!-- SV-751 -->
*  Repository and Cart
    *  The annotation count in File table of Repository and Cart does not link to the Annotations page anymore. The user can navigate to the annotations through the annotation count in Repository - Case table.
*  Legacy Archive
    *	Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.
    *	Exporting the Cart table in JSON will export the GDC Archive file table instead of exporting the files in the Cart only.
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).



## Release 1.5.2

* __GDC Product__: GDC Data Portal
* __Release Date__: May 9, 2017

### New Features and Changes

* Removed link to Data Download Statistics Report <!--PRTL-1081-->
* Updated version numbers of API, GDC Data Portal, and Data Release

### Bugs Fixed Since Last Release

*  None

### Known Issues and Workarounds

*   General
    *   Exporting large tables in the Data Portal may produce a 500 error.  Filtering this list to include fewer cases or files should eliminate the error <!--API-223-->
    *   After successful authentication, the authentication popup does not close for Internet Explorer users running in "Compatibility View". Workaround is to uncheck "Display Intranet sites in Compatibility View" in Internet Explorer options. Alternatively, refreshing the portal will correctly display authentication status. <!-- PGDC-2403 / PRTL-133 -->
    *   BAM Slicing dialog box does not disappear automatically upon executing the BAM slicing function. The box can be closed manually. <!-- PRTL-282 -->
    *   Due to preceding issue, If bam slicing produces an error pop-up message it will be obscured behind the original dialog box. <!--SV-419-->
    *   Very long URLs will produce a 400 error.  Users may encounter this after clicking on "source files" on a file page where the target file is derived from hundreds of other files such as for MAF files.  To produce a list of source files an API call can be used with the search parameter "fields=analysis.input_files.file_name". <!-- SV-396 / PRTL-342-->
		*   Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.


Example

    https://api.gdc.cancer.gov/files/455e26f7-03f2-46f7-9e7a-9c51ac322461?pretty=true&fields=analysis.input_files.file_name




*   Cart
    *   Counts displayed in the top right of the screen, next to the Cart icon, may become inconsistent if files are removed from the server. <!-- PGDC-2403 / PRTL-133 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   Internet Explorer users are not able to use the "Only show fields with no values" when adding custom facets <!-- PGDC-2467 / PRTL-109 -->
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).





## Release 1.4.1

* __GDC Product__: GDC Data Portal
* __Release Date__: October 31, 2016

### New Features and Changes

* Added a search feature to help users select values of interest in certain facets that have many values. <!-- PRTL-21 -->
* Added support for annotation ID queries in quick search. <!-- PRTL-29 -->
* Added a warning when a value greater than 90 is entered in the "Age at Diagnosis" facet. <!-- PRTL-77 -->
* Added Sample Type column to file entity page. <!-- PRTL-42 -->
* Authentication tokens are refreshed every time they are downloaded from the GDC Data Portal. <!-- PRTL-278 -->
* Buttons are inactive when an action is in progress. <!-- PRTL-270 -->
* Improved navigation features in the overview chart on portal homepage. <!-- PRTL-2 -->
* Removed State/Status from File and Case entity pages <!-- PRTL-292 -->
* Removed the "My Projects" feature. <!-- PRTL-174 -->
* Removed "Created" and "Updated" dates from clinical and biospecimen entities. <!-- PRTL-3 -->

### Bugs Fixed Since Last Release

*  Advanced search did not accept negative values for integer fields. <!-- PRTL-283 -->
*  Moving from facet search to advanced search resulted in an incorrect advanced search query. <!-- PRTL-284 -->
*  Some facets were cut off in Internet Explorer and Firefox. <!-- PRTL-290 -->

### Known Issues and Workarounds

*   General
    *   Exporting large tables in the Data Portal may produce a 500 error.  Filtering this list to include fewer cases or files should eliminate the error <!--API-223-->
    *   After successful authentication, the authentication popup does not close for Internet Explorer users running in "Compatibility View". Workaround is to uncheck "Display Intranet sites in Compatibility View" in Internet Explorer options. Alternatively, refreshing the portal will correctly display authentication status. <!-- PGDC-2403 / PRTL-133 -->
    *   BAM Slicing dialog box does not disappear automatically upon executing the BAM slicing function. The box can be closed manually. <!-- PRTL-282 -->
    *   Due to preceding issue, If bam slicing produces an error pop-up message it will be obscured behind the original dialog box. <!--SV-419-->
    *   Very long URLs will produce a 400 error.  Users may encounter this after clicking on "source files" on a file page where the target file is derived from hundreds of other files such as for MAF files.  To produce a list of source files an API call can be used with the search parameter "fields=analysis.input_files.file_name". <!-- SV-396 / PRTL-342-->
		*   Downloading a token in the GDC Legacy Archive does not refresh it. If a user downloads a token in the GDC Data Portal and then attempts to download a token in the GDC Legacy Archive, an old token may be provided. Reloading the Legacy Archive view will allow the user to download the updated token.


Example

    https://api.gdc.cancer.gov/files/455e26f7-03f2-46f7-9e7a-9c51ac322461?pretty=true&fields=analysis.input_files.file_name




*   Cart
    *   Counts displayed in the top right of the screen, next to the Cart icon, may become inconsistent if files are removed from the server. <!-- PGDC-2403 / PRTL-133 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   Internet Explorer users are not able to use the "Only show fields with no values" when adding custom facets <!-- PGDC-2467 / PRTL-109 -->
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).




## Release 1.3.0

* __GDC Product__: GDC Data Portal
* __Release Date__: September 7, 2016

### New Features and Changes

*   A new "Metadata" button on the cart page to download merged clinical, biospecimen, and file metadata in a single consolidated JSON file. **May require clearing browser cache** <!-- PRTL-177 -->
*   Added a banner on the Data Portal to help users find data <!-- PRTL-237 -->
*   Added support for "Enter" key on login button <!-- PRTL-136 -->
*   On the Data page, the browser will remember which facet tab was selected when hitting the "Back" button <!-- PRTL-147 -->
*   In file entity page, if there is a link to one single file, redirect to this file's entity page instead of a list page.  <!-- PRTL-75 -->


### Bugs Fixed Since Last Release

* 	Adding a mix of open and controlled files to the cart from any Case entity pages was creating authorization issues <!-- PRTL-226 -->
*  Opening multiple browser tabs and adding files in those browser tabs was not refreshing the cart in other tabs. <!-- PRTL-181 -->
*   When user logs in from the advanced search page, the login popup does not automatically close <!-- PRTL-88 -->
*   When removing a file from the cart and clicking undo, GDC loses track of permission status of the user towards this file and will ask for the user to log-in again. <!-- PGDC-2496 -->
*   Download File Metadata button produces incomplete JSON output omitting such fields as file_name and submitter_id.  The current workaround includes using the API to return file metadata. <!-- SV-359 / PRTL-177 -->    
*   Annotations notes do not wrap to the next line at the beginning or the end of a word, some words might be split in two lines <!-- PGDC-2474 / PRTL-182-->   
*   Sorting annotations by Case UUID causes error <!-- PRTL-95 -->   

### Known Issues and Workarounds

*   General
    *   When no filters are engaged in the Legacy Archive or Data Portal, clicking the Download Manifest button may produce a 500 error and the message "We are currently experiencing issues. Please try again later.".  To avoid this error the user can first filter by files or cases to reduce the number files added to the manifest.
    *   After successful authentication, the authentication popup does not close for Internet Explorer users running in "Compatibility View". Workaround is to uncheck "Display Intranet sites in Compatibility View" in Internet Explorer options. Alternatively, refreshing the portal will correctly display authentication status. <!-- PGDC-2403 / PRTL-133 -->
    *   BAM Slicing dialog box does not disappear automatically upon executing the BAM slicing function. The box can be closed manually. <!-- PRTL-282 -->
    *   Due to preceding issue, If bam slicing produces an error pop-up message it will be obscured behind the original dialog box. <!--SV-419-->
    *   Very long URLs will produce a 400 error.  Users may encounter this after clicking on "source files" on a file page where the target file is derived from hundreds of other files such as for MAF files.  To produce a list of source files an API call can be used with the search parameter "fields=analysis.input_files.file_name". <!-- SV-396 -->
    *   On the Legacy Archive, searches for "Case Submitter ID Prefix" containing special characters are not displayed correctly above the result list. The result list is correct, however. <!--SV-412 / LGCY-33-->

Example

    https://api.gdc.cancer.gov/files/455e26f7-03f2-46f7-9e7a-9c51ac322461?pretty=true&fields=analysis.input_files.file_name




*   Cart
    *   Counts displayed in the top right of the screen, next to the Cart icon, may become inconsistent if files are removed from the server. <!-- PGDC-2403 / PRTL-133 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   Internet Explorer users are not able to use the "Only show fields with no values" when adding custom facets <!-- PGDC-2467 / PRTL-109 -->
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibility mode. <!-- PGDC-2480 -->    


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).

## Release 1.2.0

* __GDC Product__: GDC Data Portal
* __Release Date__: August 9th, 2016

### New Features and Changes

*   Added a retry (1x) mechanism for API calls <!-- PGDC-2393 -->
*   Added support for ID fields in custom facets <!-- PGDC-1222 -->
*   Added Case Submitter ID to the Annotation entity page <!-- PGDC-747 -->
*   Added a link to Biospeciment in the Case entity page <!-- PGDC-2346 -->

### Bugs Fixed Since Last Release

*   General.
    *   Not possible to use the browser's back button after hitting a 404 page <!-- PGDC-2429 -->
    *   404 page missing from Legacy Archive Portal <!-- PGDC-2477 -->
    *   Table widget icon and export JSON icon should be different <!-- PGDC-2446 -->    
    *   Download SRA XML files from the legacy archive portal might not be possible in some context <!-- PGDC-2457 --> <!-- PGDC-2469 -->
*   Data and facets
    *   Default values for age at diagnosis is showing 0 to 89 instead of 0 to 90 <!-- PGDC-2478 -->
    *   Biospecimen search in the case entity page does not highlight (but does bold and filter) results in yellow when title case is not followed <!-- PGDC-2451 -->
    *   Table sorting icon does not include numbers <!-- PGDC-35 -->   
    *   '--' symbol is missing on empty fields (blank instead), additional missing fields identified since last release.  <!-- PGDC-2447 -->    
### Known Issues and Workarounds

*   General
    *   When no filters are engaged in the Legacy Archive or Data Portal, clicking the Download Manifest button may produce a 500 error and the message "We are currently experiencing issues. Please try again later.".  To avoid this error the user can first filter by files or cases to reduce the number files added to the manifest.
    *   After successful authentication, the authentication popup does not close for Internet Explorer users running in "Compatibility View". This only impact users at the NIH. Workaround is to uncheck "Display Intranet sites in Compatibility View" in Internet Explorer options. Alternatively, refreshing the portal will correctly display authentication status.
    *   When user login from the advanced search page, the login popup does not automatically close <!-- PRTL-88 -->
*   Cart
    *   When removing a file from the cart and clicking undo, GDC looses track of permission status of the user towards this file and will ask for the user to log-in again. <!-- PGDC-2496 -->     
    *   Counts displayed in the top right of the screen, next to the Cart icon, might get inconsistent if files are removed from the server. <!-- PGDC-2403 -->
    *   Download File Metadata button produces incomplete JSON output omitting such fields as file_name and submitter_id.  The current workaround includes using the API to return file metadata. <!-- SV-359 -->
*   Annotations
    *   Annotations notes do not wrap to the next line at the beginning or the end of a word, some words might be split in two lines <!-- PGDC-2474 -->
    *   Sorting annotations by Case UUID causes error <!-- PRTL-95 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   Internet Explorer users are not able to use the "Only show fields with no values" when adding custom facets <!-- PGDC-2467 -->
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibilty mode <!-- PGDC-2480 -->    


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).


## Release 1.1.0

* __GDC Product__: GDC Data Portal
* __Release Date__: June 1st, 2016

### New Features and Changes

*   This is a bug-fixing release, no new features were added.

### Bugs Fixed Since Last Release

*   General
    *  Fixed 508 compliance issues. <!-- PGDC-2497 --><!-- PGDC-2431 -->
    *  Disabled download manifest action on projects without files.  <!-- PGDC-2416 -->
    *  Updated the portal to indicate to the user that his session expired when he tries to download the authentication token. <!-- PGDC-2455 -->   
    *  Unselected "My project" filter after user logs-in. <!-- PGDC-2462 -->  
    *  Fixed missing padding when query includes "My Projects". <!-- PGDC-2420 -->
    *  Enforced "Add to cart" limitation to 10,000 files everywhere on the Data Portal. <!-- PGDC-2409 -->
*   Tables
    *  Improved usability of the "Sort" feature  <!-- PGDC-1771 -->
    *  Updated the "Add all files to cart" button to add all files corresponding to the current query (and not only displayed files). <!-- PGDC-2439 -->
    *  Fixed an issue where Platform would show "0" when selected platform is "Affymetrix SNP 6.0". <!-- PGDC-2419 -->
*   Data
    *  Corrected default values populated when adding a custom range facet. <!-- PGDC-2445 --> <!-- PGDC-2444 -->  <!-- PGDC-2225 -->    
    *  Fixed an issue preventing the user to sort by File Submitter ID in data tables.<!-- PGDC-2430 -->    
*   File Entity Page
    *  Improved "Associated Cases/Biospecimen" table for files associated to a lot of cases.  <!-- PGDC-1993 -->    
    *  Fixed an error when performing BAM Slicing. <!-- PGDC-2433 -->

### Known Issues and Workarounds

*   General.
    *   After successful authentication, the authentication popup does not close for Internet Explorer users running in "Compatibility View". This only impact users at the NIH. Workaround is to uncheck "Display Intranet sites in Compatibility View" in Internet Explorer options. Alternatively, refreshing the portal will correctly display authentication status.
    *   Download SRA XML files from the legacy archive portal might not be possible in some context <!-- PGDC-2457 --> <!-- PGDC-2469 -->  
    *   Not possible to use the browser's back button after hitting a 404 page <!-- PGDC-2429 -->
    *   404 page missing from Legacy Archive Portal <!-- PGDC-2477 -->
    *   Table widget icon and export JSON icon should be different <!-- PGDC-2446 -->     
*   Data and facets
    *   Default values for age at diagnosis is showing 0 to 89 instead of 0 to 90 <!-- PGDC-2478 -->
    *   Biospecimen search in the case entity page does not highlight (but does bold and filter) results in yellow when title case is not followed <!-- PGDC-2451 -->
    *   Table sorting icon does not include numbers <!-- PGDC-35 -->    
    *   '--' symbol is missing on empty fields (blank instead), additional missing fields identified since last release.  <!-- PGDC-2447 -->    
*   Cart
    *   When removing a file from the cart and clicking undo, GDC looses track of permission status of the user towards this file and will ask for the user to log-in again. <!-- PGDC-2496 -->     
    *   Counts displayed in the top right of the screen, next to the Cart icon, might get inconsistent if files are removed from the server. <!-- PGDC-2403 -->
*   Annotations
    *   Annotations notes do not wrap to the next line at the beginning or the end of a word, some words might be split in two lines <!-- PGDC-2474 -->
*   Web Browsers
    *   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).
    *   Internet Explorer users are not able to use the "Only show fields with no values" when adding custom facets <!-- PGDC-2467 -->
    *   The GDC Portals are not compatible with Internet Explorer running in compatibility mode. Workaround is to disable compatibilty mode <!-- PGDC-2480 -->    


Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).


## Release 1.0.1

* __GDC Product__: GDC Data Portal
* __Release Date__: May 18, 2016

### New Features and Changes

*   This is a bug-fixing release, no new features were added.

### Bugs Fixed Since Last Release

*   Tables and Export
    *   Restore default table column arrangement does not restore to the default but it restores to the previous state <!-- PGDC-1769 -->
*   Cart and Download
    *   Make the cart limit warning message more explanatory <!-- PGDC-1952 -->   
    *   In some situations, adding filtered files to the cart might fail <!-- PGDC-1981 -->   
*   Layout, Browser specific and Accessibility
    *   When disabling CSS, footer elements are displayed out of order <!-- PGDC-1972 -->
    *   If javascript is disabled html tags are displayed in the warning message <!-- PGDC-1835 -->
    *   Layout issues when using the browser zoom in function on tables <!-- PGDC-116 -->
    *   Cart download spinner not showing at the proper place <!-- PGDC-2056 -->
    *   Not all facets are expanded by default when loading the app <!-- PGDC-2061 -->

### Known Issues and Workarounds

*   General
    *   If a user has previously logged into the Portal and left a session without logging out, if the user returns to the Portal after the user's sessionID expires, it looks as if the user is still authenticated. The user cannot download the token and gets an error message that would not close. The user should clear the cache to properly log out.
    *   '--' symbol is missing on empty fields (blank instead) <!-- PGDC-2418 -->
    *   Download manifest button is available for TARGET projects with 0 files, resulting in error if user clic on button <!-- PGDC-2416 -->
    *   After successful authentication, the authentication popup does not close for Internet Explorer users running in "Compatibility View". This only impact users at the NIH. Workaround is to uncheck "Display Intranet sites in Compatibility View" in Internet Explorer options. Alternatively, refreshing the portal will correctly display authentication status.
*   Data    
    *   When adding a custom range facet, default values are incorrectly populated <!-- PGDC-2445 --> <!-- PGDC-2444 -->  <!-- PGDC-2225 -->
    *   The portal might return incorrect match between cases and files when using field cases.samples.portions.created_datetime (custom facet or advanced search). Note: this is not a UI issue. <!-- PGDC-2440 -->
    *   Sorting File Submitter ID option on the file tab result in a Data Portal Error <!-- PGDC-2430 -->
*   Tables and Export
    *   Table sorting icon does not include numbers <!-- PGDC-35 -->    
*   Browsers limit the number of concurrent downloads, it is generally recommended to add files to the cart and download large number of files through the GDC Data Transfer Tool, more details can be found on [GDC Website](https://gdc.cancer.gov/about-gdc/gdc-faqs).

Release details are maintained in the [GDC Data Portal Change Log](https://github.com/NCI-GDC/portal-ui/blob/master/CHANGELOG.md).


# TCGA Barcode #
## Description ##
The TCGA barcode is the primary identifier of biospecimen data within the TCGA project.  A copy of the content formerly found at https://wiki.nci.nih.gov/display/TCGA/TCGA+barcode can be found below.

## Overview ##
Historically, the BCR received participant samples and their associated metadata from TSSs. The BCR then assigned human-readable IDs, referred to as TCGA barcodes, representing the metadata of the participants and their samples. TCGA barcodes were used to tie together data that spans the TCGA network, since the IDs uniquely identify a set of results for a particular sample produced by a particular data-generating center (i.e. GCC, GSC or GDAC). The constitutive parts of this barcode provided metadata values for a sample.

Currently the BCR is assigning both a TCGA barcode and a UUID to samples. The UUID is the primary identifier.
For more information on the ID transition, see UUIDs.

### Creating Barcodes ###
All TCGA barcodes are created by the BCR. The following figure illustrates how a sample is processed and assigned a TCGA barcode at each step. Starting from the Tissue Source Site (TSS) and the participant (who donated a tissue sample to the TSS), the barcodes TCGA-02 and TCGA-02-0001 are assigned respectively. The sample itself is also assigned a barcode: TCGA-02-0001-01. The sample is split into vials (e.g. TCGA-02-0001-01B) which are divided into portions (e.g. TCGA-02-0001-01B-02). Analytes (e.g. TCGA-02-0001-01B-02D) are extracted from each portion and distributed across one or more plates (e.g. TCGA-02-0001-01B-02D-0182), where each well is identified as an aliquot (e.g. TCGA-02-0001-01B-02D-0182-06). These plates are sent to GCCs or GSCs for characterization and sequencing.
FIX this below
[![Creating Barcodes](images/creating_barcodes.png)](images/creating_barcodes.png "Click to see the full image.")

### Reading Barcodes ###

A TCGA barcode is composed of a collection of identifiers. Each specifically identifies a TCGA data element. Refer to the following figure for an illustration of how metadata identifiers comprise a barcode. An aliquot barcode, an example of which shows in the illustration, contains the highest number of identifiers.

[![Reading Barcodes](images/barcode.png)](images/barcode.png "Click to see the full image."")

| Label | Identifier for | Value | Value Description | Possible Values |
|---|---|---|---|---|
| Analyte | Molecular type of analyte for analysis | D | The analyte is a DNA sample | See Code Tables Report|
| Plate | Order of plate in a sequence of 96-well plates | 182 | The 182nd plate | 4-digit alphanumeric value |
| Portion | Order of portion in a sequence of 100 - 120 mg sample portions | 1 | The first portion of the sample | 01-99 |
| Vial | Order of sample in a sequence of samples | C | The third vial | A to Z |
| Project | Project name | TCGA | TCGA project | TCGA |
| Sample | Sample type | 1 | A solid tumor | Tumor types range from 01 - 09, normal types from 10 - 19 and control samples from 20 - 29. See Code Tables Report for a complete list of sample codes |
| Center | Sequencing or characterization center that will receive the aliquot for analysis | 1 | The Broad Institute GCC | See Code Tables Report |
| Participant | Study participant | 1 | The first participant from MD Anderson for GBM study | Any alpha-numeric value |
| TSS | Tissue source site | 2 | GBM (brain tumor) sample from MD Anderson | See Code Tables Report |

### Barcode Types ###

Barcodes can also be visualized hierarchically, with TSS barcodes at the top of the tree and aliquot barcodes at the bottom. A parent barcode prefixes any of its descendent barcodes, reflecting the derivation of one biospecimen type from another. For example, samples are collected from a participant and so the corresponding sample barcodes contain the participant barcode from which they were derived.

[![Barcode Hierarchy](images/hierarchy.png)](images/hierarchy.png "Click to see the full image.")

Using the aliquot barcode example from the figure in Reading Barcodes, the following table displays a possible set of related barcodes at each level of the hierarchy:

| Level | Bacode  | Comment |
| Aliquot | TCGA-02-0001-01C-01D-0182-01 | -- |
| Analyte | TCGA-02-0001-01C-01D | Analytes of W and X both refer to analytes derived from whole genome amplification |
| Drug | TCGA-02-0001-C1 | Drug ID is 'C','D','H','I' or 'T' followed by a number |
| Examination | TCGA-02-0001-E3124 | Examination ID is 'E' followed by a number |
| Participant | TCGA-02-0001 | -- |
| Portion | TCGA-02-0001-01C-01 | -- |
| Radiation | TCGA-02-0001-R2 | Radiation ID is 'R' followed by a number |
| Sample | TCGA-02-0001-01 | -- |
| Shipped Portion | TCGA-CM-5341-01A-21-1933-20 | Used in the platform of MDA_RPPA_CORE only |
| Slide | TCGA-02-0001-01C-01-TS1 | Tissue slide ID can be 'TS' ('Top Slide'), 'BS' ('Bottom Slide') or 'MS' ('Middle slide'), followed by a number or letter to indicate slide order |
| Surgery | TCGA-02-0001-S145 | Surgery ID is 'S' followed by a number |
| TSS | TCGA-02 | -- |

## References ##
1. [PDF from original TCGA wiki page](images/TCGA-TCGAbarcode-080518-1750-4378.pdf)


Categories: General


Annotations for TCGA
===========================
This document is retained for reference purposes for TCGA and should not be considered the current GDC standard.  For information on the existing GDC use of annotations please see the [Annotations Encyclopedia Page](/Encyclopedia/pages/Annotations/).

This section includes the following topics:

-   Annotations Overview
-   Annotation Classification and Categories


Annotations Overview
--------------------

TCGA annotations contain important information about TCGA patients and samples needed for complete and accurate
analysis and interpretation of TCGA data. The current general annotation types are: redaction, notification, CenterNotification, and observation. Redactions
and notifications are made under Biospecimen Core Resource (BCR) authority. Each annotation is categorized using controlled vocabulary.

An annotation has multiple components, described in the following table.

| **Component**             | **Description**                                                                                                 |
|---------------------------|-----------------------------------------------------------------------------------------------------------------|
| Item type                 | Patient, Sample, Portion, Slide, Analyte, Aliquot                                                               |
| Disease                   | Tumor type, by [abbreviation](https://gdc.cancer.gov/resources-tcga-users/tcga-code-tables/sample-type-codes) |
| Item Barcode              | [Barcode](TCGA_Barcode.md) of TCGA item                                                              |
| Item UUID                 | [UUID](UUID.md) of TCGA item                                                |
| Annotation Classification | Controlled vocabulary, see the table below                          |
| Annotation Category       | Controlled vocabulary, see the table below                          |
| Notes                     | Collection of free text notes, can be added to annotations after creation                                       |
| Other metadata            | Timestamp, creator                                                                                              |

The following image illustrates an example of annotation components.

![search results page](images/ScreenShot_DOC-719.png)

Annotation Classification and Categories
----------------------------------------

Categories and types of annotations have been explicitly defined, as shown in
the following table.

If an annotation appears, it appears under the authority of a certain component
of TCGA. The authorizing group can be determined from the annotation
classification. For example, Redactions and Notifications possess Program Office
level authority; they are 'official', while Observations are notes from the user
base and are 'unofficial'".

| **Annotation Classification** | **Annotation Category** | **Authority**   | **Admissible for Items** | **Usage notes** |
|-------------------------------|-----------------------------------------------------------------------------|-----------------|------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Redaction                     | Tumor tissue origin incorrect                                               | BCR             | Patient, Sample                          | Use Annotation Note to indicate details, e.g. "Case was of non-ovarian origin"                                                                                       |
| Redaction                     | Tumor type incorrect                                                        | BCR             | Patient, Sample                          | Applies to mislabeling of a disease study, organ or tissue                                                                                                           |
| Redaction                     | Genotype mismatch                                                           | BCR             | (Patient+Analyte), (Sample+Analyte)      | Incorporates conditions such as "Failed SSTR"                                                                                                                        |
| Redaction                     | Subject withdrew consent                                                    | BCR             | Patient                                  |                                                                                                                                                                      |
| Redaction                     | Subject identity unknown                                                    | BCR             | Patient                                  | Applies to patients categorized under the wrong TSS-to-BCR mapping                                                                                                   |
| Redaction                     | Duplicate case                                                              | BCR             | Patient                                  | Applies to the second (and higher) instance(s) of the same patient being accessioned through TCGA. e.g. "duplicated subject 0981"                                    |
| Redaction                     | Administrative Compliance                                                   | BCR             | Patient                                  | Use Annotation Note to indicate details e.g. "Case not meeting regulatory requirements for TCGA"                                                                     |
| Notification                  | Prior malignancy                                                            | BCR             | Patient                                  |                                                                                                                                                                      |
| Notification                  | Neoadjuvant therapy                                                         | BCR             | Patient                                  |                                                                                                                                                                      |
| Notification                  | Qualification metrics changed                                               | BCR             | Any                                      |                                                                                                                                                                      |
| Notification                  | Pathology outside specification                                             | BCR             | Sample, Portion, Slide                   |                                                                                                                                                                      |
| Notification                  | Molecular analysis outside specification                                    | BCR             | Sample, Portion, Analyte, Aliquot        |                                                                                                                                                                      |
| Notification                  | Clinical data insufficient                                                  | BCR             | Patient                                  |                                                                                                                                                                      |
| Notification                  | Item does not meet study protocol                                           | BCR             | Any                                      | This can be used for infrequently encountered situations (e.g, not Prior malignancy or Neoadjuvant therapy); details should be supplied in an Annotation Note.       |
| Notification                  | Item in special subset                                                      | BCR             | Patient, Sample                          | Use to indicate that an item is involved in a "side study". Use free text 'Annotation Note' to indicate study (e.g. "Necrosis study", "Indivumed Prior Malignancy"). |
| Notification                  | Qualified in error                                                          | BCR             | Patient                                  |                                                                                                                                                                      |
| Notification                  | Item is noncanonical                                                        | BCR             | Any                                      |                                                                                                                                                                      |
| Notification                  | New notification type                                                       | BCR             | Any                                      |                                                                                                                                                                      |
| Notification                  | History of unacceptable prior treatment related to a prior/other malignancy | BCR             | Patient                                  |                                                                                                                                                                      |
| Notification                  | History of acceptable prior treatment related to a prior/other malignancy   | BCR             | Patient                                  |                                                                                                                                                                      |
| Notification                  | Case submitted is found to be a recurrence after submission                 | BCR             | Patient                                  |                                                                                                                                                                      |
| Notification                  | Synchronous malignancy                                                      | BCR             | Patient                                  |                                                                                                                                                                      |
| CenterNotification            | Center QC failed                                                            | GSC or GCC      | Patient, Sample, Analyte, Aliquot        | QC failures at GCC/GSC/GDAC level, note describes the issue                                                                                                          |
| CenterNotification            | Item flagged DNU                                                            | GSC or GCC      | Patient, Sample, Analyte, Aliquot        | "Do not use" flag, applied by GCC/GSC/GDAC                                                                                                                           |
| Observation                   | Tumor class but appears normal                                              | Authorized User | Sample, Portion, Slide, Analyte, Aliquot |                                                                                                                                                                      |
| Observation                   | Normal class but appears diseased                                           | Authorized User | Sample, Portion, Slide, Analyte, Aliquot |                                                                                                                                                                      |
| Observation                   | Item may not meet study protocol                                            | Authorized User | Any                                      |                                                                                                                                                                      |
| Observation                   | General                                                                     | Authorized User | Any                                      |                                                                                                                                                                      |
| Observation                   | New observation type                                                        | Authorized User | Any                                      | user-suggested observation type                                                                                                                                      |


# MD5 Checksum #
## Description ##
The MD5 Checksum is a computer algorithm that calculates and verifies 128-bit MD5 hashes generated from a specific file.

## Overview ##
MD5 Checksum is used to verify the integrity of files, as virtually any change to a file will cause its MD5 hash to change. Most commonly, md5sum is used to verify that a file has not changed as a result of a faulty file transfer, a disk error or non-malicious modification.

Every file in the GDC contains an md5sum to ensure file integrity.

When downloading/uploading files using the gdc-client, an md5 checksum will occur to ensure that the file has been transferred successfully and with integrity to the original file.

## References ##
1. N/A

## External Links ##
* N/A

Categories: General


# Variant Call Format (VCF) #

## Description ##
The Variant Call Format (VCF) is a standardized format for storing and reporting genomic sequence variations.

## Overview ##
VCF files are used to report sequence variations (e.g., SNPs, indels and larger structural variants) together with rich annotations. VCF files are modular where the annotations and genotype information for a variant are separated from the call itself. VCF version 4.1 is the currently active format specification<sup>1</sup>.

VCF files are generated at the GDC with one of four variant callers (MuSe, MuTect2, Pindel, VarScan) by comparing a tumor alignment to a normal alignment from the same patient. All GDC VCFs from patients are protected data and require dbGaP credentials to access.

Please note that SomaticSniper was used as a fifth variant caller and VCFs and MAFs were available on the GDC Data Portal before [GDC Data Release 35](https://docs.gdc.cancer.gov/Data/Release_Notes/Data_Release_Notes/#data-release-350).


VCF files are produced on a case-level for each variant caller (four VCF files per case). All VCF files within a project that were produced by a single pipeline are aggregated to produce a MAF file.  

### Structure ###

VCF files are tab-delimited files that report a mutation for each row. VCFs are available at the GDC in a raw format or an annotated format that contains additional information about the location and consequences of each somatic variant.

Details about the structure of the VCF is available in the VCF 4.1 Specification<sup>1</sup>. Changes made to the VCF format in support of the GDC are available in the GDC VCF Format documentation<sup>2</sup>.

## References ##
1. [VCF 4.1 Specification](https://samtools.github.io/hts-specs/VCFv4.1.pdf)
2. [GDC VCF Format](/Data/File_Formats/VCF_Format/)

## External Links ##
* [VCF 4.1 Specification](https://samtools.github.io/hts-specs/VCFv4.1.pdf)

Categories: Data Type


# Variant Type #

## Description ##
Variant callers identify multiple types of variants.  The type of variant can be reported in a VCF or MAF formatted file.

## Overview ##

Variant type is reported by the GDC MAF file in a column named `Variant_Type` (column 10)<sup>1</sup>. The following variant types are reported:

* __SNP:__ Single nucleotide polymorphism -- a substitution in one nucleotide
* __DNP:__ Double nucleotide polymorphism -- a substitution in two consecutive nucleotides
* __TNP:__ Triple nucleotide polymorphism -- a substitution in three consecutive nucleotides
* __ONP:__ Oligo-nucleotide polymorphism -- a substitution in more than three consecutive nucleotides
* __INS:__ Insertion -- the addition of nucleotides
* __DEL:__ Deletion -- the removal of  nucleotides

## References ##
1.[GDC MAF File Format](/Data/File_Formats/MAF_Format/)

## External Links ##
* N/A

Categories: General


# Case #

## Description ##
A case is the collection of all data related to a specific patient in the context of a specific project.

## Overview ##

A case refers to a specific cancer patient or cell line in the context of a project<sup>1</sup>. In the GDC Data Model, cases are associated with: 1) a project, 2) the samples collected from the case, and 3) the clinical data<sup>2</sup>.

Cases are associated with all clinical data elements as well as the clinical and biospecimen supplemental information<sup>3</sup>. The case entity also serves as an API endpoint which allows for case-level data to be easily retrievable using programmatic methods.

Cases are also the basic unit of registration for the GDC Submission Portal. All cases that are to be submitted to a project need to be registered with dbGaP on a case-level basis prior to submission.  

## References ##
1. [GDC Data Dictionary - Case](/Data_Dictionary/viewer/#?view=table-definition-view&id=case)
2. [GDC Data Model](https://gdc.cancer.gov/developers/gdc-data-model/gdc-data-model-components)


Categories: General


# Aggregated Somatic Mutation #

## Description ##
Aggregated Somatic Mutation is a file type that reports somatic mutations for all of the cases in a project and includes information associated with each mutation.

## Overview ##

Aggregated Somatic Mutation files are generated by aggregating all of the case-level annotated VCFs (each from a single case) associated with a project and variant-calling pipeline<sup>1</sup>. The GDC used a modified version of the vcf2maf script to generate aggregated somatic mutation files<sup>2,3</sup>.

### Data Formats ###

Aggregated Somatic Mutation files are available in MAF format. MAF files are tab-delimited and associate each mutation with biologically relevant data. See the GDC MAF File Format documentation for a detailed description of this file type<sup>4</sup>. Aggregated Somatic Mutation files can be downloaded from the GDC Data Portal<sup>5</sup>.

## References ##
1. [GDC Data Dictionary - Aggregated Somatic Mutation](/Data_Dictionary/viewer/#?view=table-definition-view&id=aggregated_somatic_mutation)
2. [GDC DNA-Seq Documentation](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/)
3. [vcf2maf GitHub](https://github.com/mskcc/vcf2maf)
4. [GDC MAF File Format](/Data/File_Formats/MAF_Format/)
5. [GDC Data Portal](https://portal.gdc.cancer.gov/)

## External Links ##
* N/A

Categories: Data Type


# Entity #
## Description ##
An entity in the GDC is a unique component of the GDC Data Model.

## Overview ##
The GDC Data Model is the primary method of organizing all data within the GDC<sup>1</sup>.  More specifically, all the data within the GDC can be thought of as a Directed Acyclic Graph (DAG) composed of interconnected entities.  A graphical representation of the GDC Data Model can be found [here](https://gdc.cancer.gov/developers/gdc-data-model/gdc-data-model-components).

Each entity in the GDC has a set of properties.  An example entity is a `case` (patient). A `case` is linked to a number of other entities in the data model, such as those that contain Biospecimen and Clinical data.  For example, the  `demographic` entity, which is linked to `case`, contains fields for properties such as ethnicity, race, and gender.  The GDC Data Model defines how each of the entities are connected and the GDC Data Dictionary defines the entities and the relationships between them<sup>2</sup>.

Each entity is assigned a unique identifier in the form  of a version 4 UUID.

Data submitters can create and update submittable entities in the GDC Data Model and upload data files registered in the model using the GDC Data Submission Portal, the GDC API, and the GDC Data Transfer Tool<sup>3</sup>.

## References ##
1. [GDC Data Model](https://gdc.cancer.gov/developers/gdc-data-model)
2. [GDC Data Dictionary](https://docs.gdc.cancer.gov/Data_Dictionary/)
3. [GDC Data Submission Portal](https://gdc.cancer.gov/submit-data/gdc-data-submission-portal)

## External Links ##
* N/A


# GDC Web Site #

## Description ##
The GDC Web Site is the primary user facing site that provides access to GDC Resources that support and communicate the GDC mission.

## Overview ##
The GDC Web Site<sup>1</sup> targets data consumers, providers, developers, and general users and provides access to information about the GDC and contributed cancer genomic data sets. The GDC Web Site instructs users on the use of GDC data access and submission tools, provides descriptions of GDC bioinformatics pipelines, and documents GDC data types and file formats. The GDC Web Site also provides access to GDC support resources.

## References ##
1. [GDC Web Site](https://gdc.cancer.gov)

## External Links ##
* N/A

Categories: Tool


# Aligned Reads #
## Description ##

A read is a sequence obtained from a single sequencing experiment. An aligned read, is a sequence that has been aligned to a common reference genome. Typically these reads can number from the hundreds of thousands to tens of millions.

## Overview ##
The GDC supports the submission of aligned reads, in addition to unaligned reads. A data file containing aligned reads can be used as input for most GDC workflows<sup>1</sup>. During harmonization,  reads are aligned to the GRCh38 human genome with standardized protocols based on data type<sup>2,3</sup>. Generated aligned read files also contain unaligned reads to facilitate the retrieval of raw data by end users.

Aligned reads are available at the GDC Data Portal for: Whole Exome Sequencing, Whole Genome Sequencing, Transcriptome Sequencing, Targeted Sequencing, and ATAC-Seq

### Data Formats ###
Aligned reads are maintained in Binary Alignment Map (BAM) format.

## References ##
1. [GDC Data Dictionary - Submitted Aligned Reads](/Data_Dictionary/viewer/#?view=table-definition-view&id=submitted_aligned_reads)
2. [DNA-Seq Documentation](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/)
3. [RNA-Seq Documentation](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/)

## External Links ##
* [BAM File Format](https://samtools.github.io/hts-specs/SAMv1.pdf)

Categories: Data Type


# Genomic Data Commons (GDC) #

## Description ##
The National Cancer Institute's (NCI's) Genomic Data Commons (GDC)<sup>1</sup> is a next generation cancer knowledge network that supports the hosting and standardization of genomic and clinical data from cancer research programs, the harmonization of raw sequence data, and the application of state-of-the art methods for generating high level data (e.g. mutation calls, structural variants, etc.)<sup>2</sup>.

## Overview ##
The NCI Center for Cancer Genomics (CCG) established the GDC to provide the cancer research community with a data service supporting the receipt, quality control, integration, storage, and redistribution of standardized cancer genomic data sets derived from cancer studies.

The mission of the GDC is to provide the cancer research community with a unified data repository that enables data sharing across cancer genomic studies in support of precision medicine. Working towards this mission, the GDC aims to provide a cancer knowledge network that enables the identification of both high- and low-frequency cancer drivers, assists in defining genomic determinants of response to therapy, and informs the composition of clinical trial cohorts sharing targeted genetic lesions<sup>2,3,4</sup>.

## References ##
1. [GDC Web Site](https://gdc.cancer.gov)
2. [GDC Fact Sheet](https://gdc.cancer.gov/gdc-factsheet)
3. [NCI GDC Press Release](https://www.cancer.gov/news-events/press-releases/2014/GenomicDataCommonsNewsNote)
4. [NCI GDC Launch](https://www.cancer.gov/news-events/press-releases/2016/genomic-data-commons-launch)

## External Links ##
* [University of Chicago GDC Press Release](http://www.chicagotribune.com/news/ct-university-chicago-cancer-data-met-20141202-story.html)

Categories: General


# SNP Array-Based Data #
## Description ##

SNP Arrays produce data based on DNA hybridization levels to a set of array probes. Hybridization occurs on the arrays based on the presence or absence of probe-specific SNPs in the chromosome.

## Overview ##

SNP Array-Based data is used at the GDC to perform Copy Number Variation analyses. A copy number segmentation analysis is performed using DNACopy. Chromosomal segments are identified and their copy numbers are estimated based on similarities between adjacent SNP array probe binding sites<sup>1</sup>.   

### Data Formats ###

The GDC stores processed SNP Array-Based data in the active harmonized portal. They are stored as tab-delimited copy number segmentation files, which identify the segment regions and their estimated copy number. The raw array intensity files in CEL format are available in the GDC Data Portal, and the downstream copy number variation and genotyping analyses are available as Birdseed files.

## References ##
1. [GDC Copy Number Variation Documentation](/Data/Bioinformatics_Pipelines/CNV_Pipeline/)

## External Links ##
* N/A

Categories: Data Type


# Data Submitter #
## Description ##

A data submitter uploads approved data to the GDC for harmonization and public release.   

## Overview ##

### Submission Project Approval

A prospective data submitter must first have their project approved by the GDC. Project approval is granted based on the context of the study, the number of samples in the study, and the available data types<sup>1</sup>. Project approval requests can be directed to GDC User Support<sup>2</sup>. Before data can be uploaded to the GDC, a project and all cases must first be registered in dbGaP. Individuals who wish to upload data for a particular project must be registered as a data submitter in dbGaP for that project.

### Data Upload

Data upload must be performed based on the GDC Data Model<sup>3</sup> and Data Dictionary<sup>4</sup>. For example, uploading a sample requires that an associated case is uploaded simultaneously or previously and that the required fields are present. A data submitter can upload data using the GDC API<sup>5</sup> or GDC Data Submission Portal<sup>6</sup>. Once registered, data files must be uploaded to the GDC using the GDC Data Transfer Tool<sup>7</sup>.    

### Additional Data Submitter Roles

After all project data is uploaded and reviewed, the data submitter can release and/or submit the project. Releasing the project gives the GDC permission to distribute data to users. Submitting the project verifies that the data has been reviewed and is ready for harmonization.    

## References ##
1. [Requesting Data Submission](https://gdc.cancer.gov/node/633/)
2. GDC User Support:  __support@nci-gdc.datacommons.io__
3. [GDC Data Model](https://gdc.cancer.gov/developers/gdc-data-model/gdc-data-model-components)
4. [GDC Data Dictionary](/Data_Dictionary/viewer/)
5. [API Submission](/API/Users_Guide/Submission/)
6. [Data Submission Portal Documentation](/Data_Submission_Portal/Users_Guide/Data_Submission_Overview/)
7. [GDC Data Transfer Tool](https://gdc.cancer.gov/access-data/gdc-data-transfer-tool)

## External Links ##
* [dbGaP](https://www.ncbi.nlm.nih.gov/gap)

Categories: General


TCGA Variant Call Format (VCF) 1.1 Specification
================================================

**Document Information**
This document is retained here for reference purposes and should not be considered the current standard.  


**Specification for TCGA Variant Call Format (VCF)**  
Version 1.1


Please note that VCF files are treated as **protected** data and must be
submitted to the DCC only in **Level 2** archives.

About TCGA VCF specification
============================

Variant Call Format (VCF) is a format for storing and reporting genomic sequence
variations. VCF files are modular where the annotations and genotype information
for a variant are separated from the call itself. As of May 2011, VCF version
4.1 (described
[here](http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-41))
is the most recent release. GSCs will generate sequence variation data using
high-throughput sequencing technologies and resulting variations will be
submitted to DCC as VCF files. TCGA has adopted VCF 4.1 with certain
modifications to support supplemental information specific to the project.
Subsequent sections describe the format TCGA VCF files should follow and
validation steps that would have to be implemented at the DCC.

Summary of current version changes
==================================

Following is a summary of additions/modifications for this version and the corresponding validation rule
number is included in parentheses.

**UUID compliance**: All TCGA data is currently in the process of being
converted to be UUID-compliant. Until the conversion is complete and all centers
are prepared to start submitting UUID-compliant data, some of the VCF files may
adhere to UUID-based specification whereas some may still have barcodes.
Non-UUID files will follow the specification described here but for
UUID-compliance, VCF files should satisfy the following criteria.

1.  **SampleUUID** and **SampleTCGABarcode** are required tags in each
    ##SAMPLE declaration. Please note that **SampleName** will not be a
    required tag once submitting center has fully converted to UUIDs

    1.  Metadata represented by SampleTCGABarcode at the DCC should correspond
        to the UUID assigned to SampleUUID

2.  **Individual** is not a required tag in ##SAMPLE declaration

3.  If ##**INDIVIDUAL** is declared in the header, all SampleUUIDs in the
    header must correspond to the same participant, and the corresponding TCGA
    barcode for that participant should be assigned to ##INDIVIDUAL



1.  SampleName is a required tag in ##SAMPLE declaration. The value assigned
    to SampleName should be a valid [aliquot
    barcode](https://docs.gdc.cancer.gov/Encyclopedia/pages/TCGA_Barcode/) / [UUID](https://docs.gdc.cancer.gov/Encyclopedia/pages/UUID/)
    in the database (#15b, #15h)

2.  Header declarations for INFO and FORMAT fields should match the values
    defined in Tables 4 and 5 respectively (#7a)

3.  Following FORMAT fields are required for all variant records in a VCF file:
    (#10c)

    -   Genotype (**GT**)

    -   Read depth (**DP**)

    -   Reads supporting ALT (**AD** or **DP4**)

    -   Average base quality for reads supporting alleles (**BQ**)

    -   Somatic status of the variant (**SS**). SS can be 0, 1, 2, 3, 4 or 5
        depending on whether relative to normal the variant is wildtype,
        germline, somatic, LOH, post-transcriptional modification, or unknown
        respectively (#23)

4.  Values for INFO field **VLS** (validation status relative to non-adjacent
    Normal) will be checked for validity. It can be 0, 1, 2, 3, 4, or 5 based on
    whether the mutation is wildtype, germline, somatic, LOH, post
    transcriptional modification, or unknown respectively (#9c)

5.  Validation of tags in PEDIGREE declaration has changed as follows: (#16)

    -   Name_0, Name_1, etc. do not have to be these literal strings but instead
        represent arbitrary strings

    -   The keys and values used in the <Name_N=Value_N> should be unique
        across assignments in any given PEDIGREE declaration

    -   Value assigned in <Name_N=Value_N> does not have to be defined as a
        SAMPLE in a genotype column or in the header

TCGA-specific customizations
============================

The VCF 4.1 specification has been customized to support TCGA-specific variant
information. While majority of the steps pertaining to the basic structure of
the file remain the same, checks for supplemental information fields have been
introduced. For example, TCGA VCF specification allows for additional fields to
represent data associated with complex rearrangements, RNA-Seq variants, and
sample-specific metadata.

All TCGA-specific additions and modifications in [validation
steps](#validation-rules) are prefixed with a
<TCGA-VCF> tag for convenient comparison with 1000Genomes VCF 4.1. The
following table summarizes TCGA-specific customizations that have been added to
the VCF 4.1 specification. The first column, "Customization type", indicates
whether a new validation step has been introduced or if an existing step has
been modified

**Table 1: TCGA-specific validation steps**

| **Customization type** | **Description**                                                                                                                                                              | **Validation step # in TCGA-VCF 1.1 spec** | **Corresponding validation step # in VCF 4.1 spec** |
|------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------|------------------------------------------------------|
| New                    | Validate that file contains ##tcgaversion HEADER line. Its presence indicates that the file is TCGA VCF and the value assigned to the field contains format version number | \---                                        | \---                                                 |
| New                    | Additional mandatory header lines (Please refer to [Table 2](#TCGAVariantCallFormat(VCF)1.1Specificat))                                                                      | \#1                                         | \#1                                                  |
| New                    | Validation of SAMPLE meta-information lines                                                                                                                                  | \#15                                        | \---                                                 |
| New                    | Validation of PEDIGREE meta-information lines                                                                                                                                | \#16                                        | \---                                                 |
| Modification           | Acceptable value set for CHROM has been modified                                                                                                                             | \#18a,b                                     | \#16a                                                |
| Modification           | Acceptable value set for ALT has been modified                                                                                                                               | \#19                                        | \#17                                                 |
| New                    | Validation for INFO sub-field "VT" has been added                                                                                                                            | \#22                                        | \---                                                 |
| New                    | Validation for FORMAT sub-field "SS" has been added                                                                                                                          | \#23                                        | \---                                                 |
| New                    | Validation for INFO/FORMAT sub-field "DP" has been added                                                                                                                     | \#24                                        | \---                                                 |
| New                    | Validation for complex rearrangement records has been added                                                                                                                  | \#25                                        | \---                                                 |
| New                    | Validation for RNA-Seq annotation fields has been added                                                                                                                      | \#26                                        | \---                                                 |
| New                    | Mandatory FORMAT fields have been added                                                                                                                                      | \#10c                                       | \---                                                 |
| New                    | Check for consistent definitions for INFO and FORMAT fields                                                                                                                  | \#7a                                        | \---                                                 |

File format
===========

The following example (based on [VCF version
4.1)](http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-41)
shows different components of a TCGA VCF file. Any VCF file contains two main
sections. The HEADER section contains meta-information for variant records that
are reported as individual rows in the BODY of the VCF file. Both sections are
described below.

**Case-sensitivity**: Please note that all fields and their associated
validation rules are case-sensitive (as given in the specification) unless noted
otherwise.

**Figure 1: Components of a sample TCGA VCF file**

| ![images](images/vcfExample_VCF.png) |
|------------------------------------------|




HEADER
------

The HEADER contains meta-information lines that provide supplemental information
about variants contained in BODY of the file. HEADER lines could be formatted in
the following two ways:


    ##key=value

    Example:

    ##fileformat=VCFv4.1

    ##fileDate=20090805


or

    ##FIELDTYPE=<key1=value1,key2=value2,...>

    Example:

    ##INFO=<ID=AA,Number=1,Type=String,Description="Ancestral Allele">

Meta-information could be applicable either to all variant records in the file
(e.g., date of creation of file) or to individual variants (e.g., flag to
indicate whether a given variant exists in dbSNP).

### Generic meta-information

**Format**: *##key=value* OR *##FIELDTYPE=<key1=value1,key2=value2,...>*

The following table lists some of the reserved field names. Files can be
customized to contain additional meta-information fields as long as they are not
in conflict with reserved field names. The first field in Table 2 (fileformat)
is mandatory and lists the VCF version number of the file.

**Table 2: Examples of generic meta-information fields**

| **Field**     | **Case-Sensitive**    | **Description** |  **Sample values** | Required (fields in red are TCGA-specific requirements)
| ------------- | --------------------- | --------------- | ------------------ | ----------------------------------------------------- |
| Fileformat    | No                 | Lists the VCF version number the file is based on; must be the first line in the file                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         | ##fileformat=VCFv4.1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | Yes                                                                                                   |
| fileDate      | No                 | Date file was created; should be in yyyymmdd format                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | ##fileDate=20090805                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     | Yes                                                                                                   |
| Tcgaversion   | No                 | Indicates that the file follows TCGA-VCF specification. Format version number is assigned to the field.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       | ##tcgaversion=1.1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       | Yes                                                                                                   |
| Reference     | No                 | Reference build used for variant calling and against which variant coordinates are shown                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      | ##reference=1000GenomesPilot-NCBI36                                                                                                                                                                                                                                                                                                                                                                                                                                                                     | Yes                                                                                                   |
|               |                    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |                                                                                                       |
|               |                    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | OR                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |                                                                                                       |
|               |                    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |                                                                                                       |
|               |                    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | ##reference=<ID=hg18,                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  |                                                                                                       |
|               |                    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | Source=<file://seq/references/1000GenomesPilot-NCBI36.fasta>                                                                                                                                                                                                                                                                                                                                                                                                                                              |                                                                                                       |
| Assembly      | No                 | External assembly file. The field can be assigned a file name if assembly file is included in the archive submitted to the DCC or it can be a URL pointing to the file location.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              | ##assembly=[ftp://ftp-trace.ncbi.nih.gov/](ftp://ftp-trace.ncbi.nih.gov/1000genomes/ftp/release/sv/breakpoint_assemblies.fasta)                                                                                                                                                                                                                                                                                                                                                                         | Yes                                                                                                   |
|               |                    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | [1000genomes/ftp/release/sv/breakpoint_assemblies.fasta](ftp://ftp-trace.ncbi.nih.gov/1000genomes/ftp/release/sv/breakpoint_assemblies.fasta)                                                                                                                                                                                                                                                                                                                                                             | (if a contig from an assembly file is being referred to in the VCF file, especially for breakends)    |
| center        | No                 | Name of the center where VCF file is generated. A comma-separated list can be provided if files from multiple centers are merged.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             | ##center="Broad"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        | Yes                                                                                                   |
|               |                    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |                                                                                                       |
|               |                    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | OR                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |                                                                                                       |
|               |                    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |                                                                                                       |
|               |                    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | ##center="Broad,UCSC,BCM"                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |                                                                                                       |
| phasing       | No                 | Indicates whether genotype calls are partially phased (phasing=partial) or unphased (phasing=none)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            | ##phasing=none                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          | Yes                                                                                                   |
| geneAnno      | No                 | URL of the gene annotation source e.g., Generic Annotation File (GAF)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         | ##geneAnno=[https://gdc.cancer.gov/about-data/data-harmonization-and-generation/gdc-reference-files](https://gdc.cancer.gov/about-data/data-harmonization-and-generation/gdc-reference-files)                                                                                                                                                                                                                                                                                                                                                              | Yes (if annotation tags like GENE, SID and RGN are used) |
| vcfProcessLog | No                 | Lists algorithm, version and settings used to generate variant calls in a VCF file. If multiple VCF files are processed to produce a single merged file, the field records attributes for individual VCF files and the programs used to merge the files along with the associated version, parameters and contact information of the person who produced the merged file.  | | | | | **Note**: If VCF file does not represent a set of merged files, *MergeSoftware*, *MergeParam*, *MergeVer* and *MergeContact* tags will not be applicable and can be omitted.                                          
                                                                |      |      |    | ##vcfProcessLog=<InputVCF=<file1.vcf>,                                                                                                          InputVCFSource=<varCaller1>,                                                                                                                                                                                                                                                                                                                                                                                                                                                                            |                                                                                                       |
|               |                    | **Note**: If VCF file does not represent a set of merged files, *MergeSoftware*, *MergeParam*, *MergeVer* and *MergeContact* tags will not be applicable and can be omitted.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  | InputVCFVer=<1.0>,                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |                                                                                                       |
|               |                    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | InputVCFParam=<a1,c2>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   |                                                                                                       |
|               |                    | **Note**: If multiple parameters need to be declared in *InputVCFParam*, key=value pairs can be used to name these parameters. For example:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   | InputVCFgeneAnno=<anno1.gaf>>                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |                                                                                                       |
|               |                    | InputVCFParam=<key1=value1,key2=value2>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |                                                                                                       |
|               |                    | If there are multiple files for which parameters have to be declared, following format can be used:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | OR                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |                                                                                                       |
|               |                    | InputVCFParam=<key1_1=value1_1,key1_2=value1_2;key2_1=value2_1,key2_2=value2_2>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |                                                                                                       |
|               |                    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | ##vcfProcessLog=<InputVCF=<file1.vcf,file2.vcf,file3.vcf>,                                                                                                                                                                                                                                                                                                                                                                                                                                           |                                                                                                       |
|               |                    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | InputVCFSource=<varCaller1,varCaller2,varCaller3>,                                                                                                                                                                                                                                                                                                                                                                                                                                                      |                                                                                                       |
|               |                    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | InputVCFVer=<1.0,2.1,2.0>,                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |                                                                                                       |
|               |                    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | InputVCFParam=<a1,c2;a1,b1;a1,b1>,                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |                                                                                                       |
|               |                    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | InputVCFgeneAnno=<anno1.gaf,anno2.gaf,anno3.gaf>,                                                                                                                                                                                                                                                                                                                                                                                                                                                       |                                                                                                       |
|               |                    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | MergeSoftware=<sw1,sw2>,                                                                                                                                                                                                                                                                                                                                                                                                                                                                                |                                                                                                       |
|               |                    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | MergeParam=<a1,a2;b1,b2>,                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |                                                                                                       |
|               |                    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | MergeVer=<2.1,3.0>,                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |                                                                                                       |
|               |                    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | MergeContact=<johndoe@xyz.edu>>                                                                                                                                                                                                                                                                                                                                                                            |                      |
| INDIVIDUAL    | No                 | Specifies the individual for which data is presented in the file      | ##INDIVIDUAL=TCGA-24-0980                 | No                             |

### INFO/FORMAT/FILTER meta-information

**Format**: *##FIELDTYPE=<key1=value1,key2=value2,...>*

INFO, FORMAT and FILTER (case-sensitive values) are optional fields that have to
be declared in the HEADER if they are being referred to in BODY of the file.
Different *keys* that can be used to define them are described in Table 3. All
three fields do not use the same set of keys. Please refer to individual field
definitions for further details.

**Important**: TCGA VCF 1.1 requires all VCF files to follow consistent header
declarations for standard INFO and FORMAT sub-fields. Please refer to Tables 4
and 5 for details. If a sub-field exists in these tables and is used in a TCGA
VCF file, then all <key=value> pairs in the definition should match entries in
the corresponding table for the file to pass validation.

**Table 3: Description of keys used in INFO/FORMAT/FILTER meta-information
declarations**

| **Key**     | **Case-sensitive** | **Description** | **Data Type (Possible values)** | **Additional Notes** |
| ----------- | ------------------ | --------------- | ------------------------------- | -------------------- |
| ID          | Yes                | name of the field; also used in BODY of the file to assign values for individual variant records | String, no whitespaces, no comma                                                                                                                         | \---                                                                                                                                                                                                         |
| Number      | Yes                | specifies the number of values that can be associated with the corresponding field               | Set                                                                                                                                                      | Any integer \>= 0 indicating number of values;                                                                                                                                                               |
|             |                    |                                                                                                  | *(Integer \>= 0, "A", "G", ".")*                                                                                                                         | "A", if the field has one value per alternate allele;                                                                                                                                                        |
|             |                    |                                                                                                  |                                                                                                                                                          | "G", if the field has one value per genotype;                                                                                                                                                                |
|             |                    |                                                                                                  |                                                                                                                                                          | ".", if number of values varies, is unknown, or is unbounded                                                                                                                                                 |
| Type        | Yes                | indicates data type of the value associated with the field                                       | Set                                                                                                                                                      | "Flag" type indicates that the field does not contain a value entry, and hence the *Number* should be 0 in this case. FORMAT fields cannot have a "Flag" *Type* assigned to them.                            |
|             |                    |                                                                                                  | *(Integer, Float, Flag, Character, String)*                                                                                                              |                                                                                                                                                                                                              |
| Description | Yes                | provides a brief description of the field                                                        | String, surrounded by double-quotes, cannot itself contain a double-quote, cannot contain trailing whitespace at the end of string before closing quotes | \---                                                                                                                                                                                                         |

#### INFO lines

**Format**: *##INFO=<key1=value1,key2=value2,...>*  
**Required keys**: ID, Type, Number, Description

INFO fields are optional and contain additional annotations for a variant.
Certain INFO fields have already been created and exist as reserved fields in
the current VCF standard. Custom INFO fields can be added based on study
requirements as long as they do not use the reserved field names. If an INFO
field is declared in the header, it needs to be described further using the
following format:

    ##INFO=<ID=ID,Number=number,Type=type,Description=description>

    Example:

    ##INFO=<ID=AA,Number=1,Type=String,Description="Ancestral Allele">

    ##INFO=<ID=AF,Number=A,Type=Float,Description="Allele Frequ

    ency">

#### FORMAT lines

**Format**: *##FORMAT=<key1=value1,key2=value2,...>*  
**Required keys**: ID, Type, Number, Description

FORMAT declaration lines are used when annotations need to be added for
individual genotypes associated with each sample in the file. FORMAT sub-fields
are declared precisely as the INFO sub-fields with the exception that a FORMAT
sub-field cannot be assigned a "Flag" *Type.*

    ##FORMAT=<ID=ID,Number=number,Type=type,Description=description>

    Example:

    ##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">

    ##FORMAT=<ID=GQ,Number=1,Type=Integer,Description="Genotype Quality">

**Important**: TCGA VCF 1.1 requires the following FORMAT sub-fields to be
defined for all variant records. Therefore, these FORMAT lines are not optional
for TCGA VCF files and should be declared in the header. Please refer to Table 5
for definitions for these sub-fields.

-   Genotype (**GT**)

-   Read depth (**DP**)

-   Reads supporting ALT (**AD** or **DP4**). Either AD or DP4 is required to be
    defined although DP4 is preferred.

-   Average base quality for reads supporting alleles (**BQ**)

-   Somatic status of the variant (**SS**). SS can be 0, 1, 2, 3, 4, or 5
    depending on whether relative to normal the variant is wildtype, germline,
    somatic, LOH, post-transcriptional modification, or unknown respectively.

These should be considered as required fields so that they are included by
default unless there is an exceptional scenario where the information for a
field cannot be obtained. In such a case, "." can be used to indicate missing
value.

#### FILTER lines

**Format**: *##FILTER=<key1=value1,key2=value2,...>*  
**Required keys**: ID, Description

FILTER fields are defined to list filtering criteria used for generating variant
calls. Custom filters can be applied as long as a definition is provided in the
HEADER. FILTERs that have been applied to the data should be described as
follows. Please note that FILTER declarations do not include *Type* or *Number*
keys.

    ##FILTER=<ID=ID,Description=description>

    Example:

    ##FILTER=<ID=q10,Description="Quality below 10">

    ##FILTER=<ID=s50,Description="Less than 50% of samples have data">

#### Consistent definitions for reserved INFO and FORMAT fields

To ensure that all TCGA VCF files have consistent definitions for standard
fields and to avoid merging errors due to contradicting definitions, following
header declarations for common fields are proposed. The 'Source' column in the
tables below indicates whether the field is from 1000Genomes VCF or if it is
specific to TCGA-VCF. By adhering to these definitions, we can ensure that a
given field is interpreted the same way across all centers and that same
'Number', 'Type' and 'Description' values are used for these IDs.

##### Table 4: INFO sub-field definitions

| **Sub-field** | **Source** | **Formatted declaration**                                                                                                                                                                           |
|---------------|------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| AA            | VCF        | ##INFO=<ID=AA,Number=1,Type=String,Description="Ancestral Allele">                                                                                                                              |
| AC            | VCF        | ##INFO=<ID=AC,Number=.,Type=Integer,Description="Allele count in genotypes, for each ALT allele, in the same order as listed">                                                                  |
| AF            | VCF        | ##INFO=<ID=AF,Number=.,Type=Float,Description="Allele Frequency in primary data, for each ALT allele, in the same order as listed">                                                             |
| AN            | VCF        | ##INFO=<ID=AN,Number=1,Type=Integer,Description="Total number of alleles in called genotypes">                                                                                                  |
| BQ            | VCF        | ##INFO=<ID=BQ,Number=1,Type=Integer,Description="RMS base quality">                                                                                                                             |
| CIGAR         | VCF        | ##INFO=<ID=CIGAR,Number=1,Type=Integer,Description="Cigar string describing how to align an alternate allele to the reference allele">                                                          |
| DB            | VCF        | ##INFO=<ID=DB,Number=0,Type=Flag,Description="dbSNP membership">                                                                                                                                |
| DP            | VCF        | ##INFO=<ID=DP,Number=1,Type=Integer,Description="Total Depth across samples">                                                                                                                   |
| END           | VCF        | ##INFO=<ID=END,Number=1,Type=Integer,Description="End position of the variant described in this record">                                                                                        |
| H2            | VCF        | ##INFO=<ID=H2,Number=0,Type=Flag,Description="HapMap2 membership">                                                                                                                              |
| H3            | VCF        | ##INFO=<ID=H3,Number=0,Type=Flag,Description="HapMap3 membership">                                                                                                                              |
| MQ            | VCF        | ##INFO=<ID=MQ,Number=1,Type=Integer,Description="RMS Mapping Quality">                                                                                                                          |
| MQ0           | VCF        | ##INFO=<ID=MQ0,Number=1,Type=Integer,Description="Total Mapping Quality Zero Reads">                                                                                                            |
| NS            | VCF        | ##INFO=<ID=NS,Number=1,Type=Integer,Description="Number of Samples With Data">                                                                                                                  |
| SB            | VCF        | ##INFO=<ID=SB,Number=1,Type=Float,Description="Strand bias">                                                                                                                                    |
| SOMATIC       | VCF        | ##INFO=<ID=SOMATIC,Number=0,Type=Flag,Description="Indicates if record is a somatic mutation">                                                                                                  |
| VALIDATED     | VCF        | ##INFO=<ID=VALIDATED,Number=0,Type=Flag,Description="Indicates if variant has been validated by follow-up experiment">                                                                          |
| 1000G         | VCF        | ##INFO=<ID=1000G,Number=0,Type=Flag,Description="Indicates membership in 1000Genomes">                                                                                                          |
| IMPRECISE     | VCF        | ##INFO=<ID=IMPRECISE,Number=0,Type=Flag,Description="Imprecise structural variation">                                                                                                           |
| NOVEL         | VCF        | ##INFO=<ID=NOVEL,Number=0,Type=Flag,Description="Indicates a novel structural variation">                                                                                                       |
| SVTYPE        | VCF        | ##INFO=<ID=SVTYPE,Number=1,Type=String,Description="Type of structural variant">                                                                                                                |
| SVLEN         | VCF        | ##INFO=<ID=SVLEN,Number=.,Type=Integer,Description="Difference in length between REF and ALT alleles">                                                                                          |
| CIPOS         | VCF        | ##INFO=<ID=CIPOS,Number=2,Type=Integer,Description="Confidence interval around POS for imprecise variants">                                                                                     |
| CIEND         | VCF        | ##INFO=<ID=CIEND,Number=2,Type=Integer,Description="Confidence interval around END for imprecise variants">                                                                                     |
| HOMLEN        | VCF        | ##INFO=<ID=HOMLEN,Number=.,Type=Integer,Description="Length of base pair identical micro-homology at event breakpoints">                                                                        |
| HOMSEQ        | VCF        | ##INFO=<ID=HOMSEQ,Number=.,Type=String,Description="Sequence of base pair identical micro-homology at event breakpoints">                                                                       |
| BKPTID        | VCF        | ##INFO=<ID=BKPTID,Number=.,Type=String,Description="ID of the assembled alternate allele in the assembly file">                                                                                 |
| MEINFO        | VCF        | ##INFO=<ID=MEINFO,Number=4,Type=String,Description="Mobile element info of the form NAME,START,END,POLARITY">                                                                                   |
| METRANS       | VCF        | ##INFO=<ID=METRANS,Number=4,Type=String,Description="Mobile element transduction info of the form CHR,START,END,POLARITY">                                                                      |
| DGVID         | VCF        | ##INFO=<ID=DGVID,Number=1,Type=String,Description="ID of this element in Database of Genomic Variation">                                                                                        |
| DBVARID       | VCF        | ##INFO=<ID=DBVARID,Number=1,Type=String,Description="ID of this element in DBVAR"\]>                                                                                                              |
| DBRIPID       | VCF        | ##INFO=<ID=DBRIPID,Number=1,Type=String,Description="ID of this element in DBRIP">                                                                                                              |
| MATEID        | VCF        | ##INFO=<ID=MATEID,Number=.,Type=String,Description="ID of mate breakends">                                                                                                                      |
| PARID         | VCF        | ##INFO=<ID=PARID,Number=1,Type=String,Description="ID of partner breakend">                                                                                                                     |
| EVENT         | VCF        | ##INFO=<ID=EVENT,Number=1,Type=String,Description="ID of event associated to breakend">                                                                                                         |
| CILEN         | VCF        | ##INFO=<ID=CILEN,Number=2,Type=Integer,Description="Confidence interval around the length of the inserted material between breakends">                                                          |
| DPADJ         | VCF        | ##INFO=<ID=DPADJ,Number=.,Type=Integer,Description="Read Depth of adjacency">                                                                                                                   |
| CN            | VCF        | ##INFO=<ID=CN,Number=1,Type=Integer,Description="Copy number of segment containing breakend">                                                                                                   |
| CNADJ         | VCF        | ##INFO=<ID=CNADJ,Number=.,Type=Integer,Description="Copy number of adjacency">                                                                                                                  |
| CICN          | VCF        | ##INFO=<ID=CICN,Number=2,Type=Integer,Description="Confidence interval around copy number for the segment">                                                                                     |
| CICNADJ       | VCF        | ##INFO=<ID=CICNADJ,Number=.,Type=Integer,Description="Confidence interval around copy number for the adjacency">                                                                                |
| VLS           | TCGA-VCF   | ##INFO=<ID=VLS,Number=1,Type=Integer,Description="Final validation status relative to non-adjacent Normal,0=wildtype,1=germline,2=somatic,3=LOH,4=post transcriptional modification,5=unknown"> |
| SID           | TCGA-VCF   | ##INFO=<ID=SID,Number=.,Type=String,Description="Unique identifier from gene annotation source or unknown">                                                                                     |
| GENE          | TCGA-VCF   | ##INFO=<ID=GENE,Number=.,Type=String,Description="HUGO gene symbol or Unknown">                                                                                                                 |
| RGN           | TCGA-VCF   | ##INFO=<ID=RGN,Number=.,Type=String,Description="Region where nucleotide variant occurs in relation to a gene">                                                                                 |
| RE            | TCGA-VCF   | ##INFO=<ID=RE,Number=0,Type=Flag,Description="Position known to have RNA-edits to occur">                                                                                                       |
| VT            | TCGA-VCF   | ##INFO=<ID=VT,Number=1,Type=String,Description="Variant type, can be SNP, INS or DEL">                                                                                                          |

##### Table 5: FORMAT sub-field definitions

| **Sub-field** | **Source** | **Formatted declaration**                                                                                                                                                                   |
|---------------|------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| GT            | VCF        | ##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">                                                                                                                            |
| DP            | VCF        | ##FORMAT=<ID=DP,Number=1,Type=Integer,Description="Read depth at this position in the sample">                                                                                          |
| FT            | VCF        | ##FORMAT=<ID=FT,Number=1,Type=String,Description="Sample genotype filter">                                                                                                              |
| GL            | VCF        | ##FORMAT=<ID=GL,Number=.,Type=Float,Description="Genotype likelihoods">                                                                                                                 |
| PL            | VCF        | ##FORMAT=<ID=PL,Number=3,Type=Integer,Description="Normalized, Phred-scaled likelihoods for AA,AB,BB genotypes where A=ref and B=alt; not applicable if site is not biallelic">         |
| GP            | VCF        | ##FORMAT=<ID=GP,Number=.,Type=Float,Description="Phred-scaled genotype posterior probabilities">                                                                                        |
| GQ            | VCF        | ##FORMAT=<ID=GQ,Number=.,Type=Integer,Description="Conditional Phred-scaled genotype quality">                                                                                          |
| HQ            | VCF        | ##FORMAT=<ID=HQ,Number=2,Type=Integer,Description="Haplotype qualities, two comma separated phred qualities">                                                                           |
| CN            | VCF        | ##FORMAT=<ID=CN,Number=1,Type=Integer,Description="Copy number genotype for imprecise events">                                                                                          |
| CNQ           | VCF        | ##FORMAT=<ID=CNQ,Number=1,Type=Float,Description="Copy number genotype quality for imprecise events">                                                                                   |
| CNL           | VCF        | ##FORMAT=<ID=CNL,Number=.,Type=Float,Description="Copy number genotype likelihood for imprecise events">                                                                                |
| MQ            | VCF        | ##FORMAT=<ID=MQ,Number=1,Type=Integer,Description="Phred style probability score that the variant is novel with respect to the genome's ancestor">                                      |
| HAP           | VCF        | ##FORMAT=<ID=HAP,Number=1,Type=Integer,Description="Unique haplotype identifier">                                                                                                       |
| AHAP          | VCF        | ##FORMAT=<ID=AHAP,Number=1,Type=Integer,Description="Unique identifier of ancestral haplotype">                                                                                         |
| SS            | TCGA-VCF   | ##FORMAT=<ID=SS,Number=1,Type=Integer,Description="Variant status relative to non-adjacent Normal,0=wildtype,1=germline,2=somatic,3=LOH,4=post-transcriptional modification,5=unknown"> |
| TE            | TCGA-VCF   | ##FORMAT=<ID=TE,Number=.,Type=String,Description="Translational effect of the variant in a codon">                                                                                      |
| AD            | TCGA-VCF   | ##FORMAT=<ID=AD,Number=.,Type=Integer,Description="Depth of reads supporting alleles 0/1/2/3...">                                                                                       |
| DP4           | TCGA-VCF   | ##FORMAT=<ID=DP4,Number=4,Type=Integer,Description="Number of high-quality ref-forward, ref-reverse, alt-forward and alt-reverse bases">                                                |
| BQ            | TCGA-VCF   | ##FORMAT=<ID=BQ,Number=.,Type=Integer,Description="Average base quality for reads supporting alleles">                                                                                  |
| VAQ           | TCGA-VCF   | ##FORMAT=<ID=VAQ,Number=1,Type=Integer,Description="Variant allele quality">                                                                                                            |





### TCGA-specific meta-information

#### PEDIGREE lines

**Format**: *##PEDIGREE=<key1=value1,key2=value2,...>*  
**Required keys**: Name_0,..,Name_N where N \>= 1;

PEDIGREE lines are used to specify derivation relationships between different
genomes. *Name_0* is associated with the derived genome and *Name_1* through
*Name_N* represent the genomes from which it is derived. In the case of tumor
clonal populations, one population is clonally derived from another. In the
example below, PRIMARY-TUMOR-GENOME is derived from GERMLINE-GENOME.

    ##PEDIGREE=<Name_0=<G0-ID>,Name_1=<G1-ID>,...,Name_N=<GN-ID>>

    where N is \>= 1;

    Example:

    ##PEDIGREE=<Name_0=PRIMARY-TUMOR-GENOME,Name_1=GERMLINE-GENOME>

#### SAMPLE lines

**Format**: *##SAMPLE=<key1=value1,key2=value2,...>*  
**Required keys**: ID, SampleName, Individual, File, Platform, Source, Accession

For UUID-compliant files, following rules should be followed:

**Required keys**: ID, SampleName, Individual, SampleUUID, SampleTCGABarcode,
File, Platform, Source, Accession

-   Value assigned to "SampleUUID" should be a valid [aliquot
    UUID](https://docs.gdc.cancer.gov/Encyclopedia/pages/UUID/) in the database.

-   Value assigned to "SampleTCGABarcode" should represent the aliquot-level
    metadata associated with SampleUUID. This metadata mapping is originally
    received by the DCC from BCR.

>   Example:

>   ##SAMPLE=<ID=NORMAL,SampleUUID=151626d0-a650-447a-a2b0-1df53c31c682,SampleTCGABarcode=TCGA-A1-A0SD-01A-11R-A114-13,Individual=TCGA-A1-A0SD,Description="Normal",File=TCGA-A1-A0SD-01A-11R-A114-13.bam,SequenceSource=Exome,Platform=Illumina,Source=dbGAP,Accession=1234,Genomes=<Germline,Tumor>,Mixture=<0.1,0.9>,Genome_Description=<"Germline
>   contamination","Tumor genome">>



SAMPLE lines are used to include additional metadata about each sample for which
data is represented in the VCF file. All samples are listed in the column header
line following the FORMAT column (Figure 1). Each of these samples should have
its own HEADER declaration where the sample identifier in the column header
should be the same as the value assigned to "ID" key in the corresponding
declaration. Value assigned to "SampleName" should be a valid [aliquot
barcode](https://docs.gdc.cancer.gov/Encyclopedia/pages/TCGA_Barcode/) / [UUID](https://docs.gdc.cancer.gov/Encyclopedia/pages/UUID/)
in the database. The declaration lists information about the sample (source,
platform, source file, etc.) and can also be used to indicate if the sample is a
mixture of different kind of genomes. In the example below, "Genomes", "Mixture"
and 'Genome_Description" tags represent comma-separated list of different
genomes that a sample contains, proportion of each genome in the sample, and a
brief description of each genome respectively.

##SAMPLE=<ID=id,SampleName=sampleName,Individual=individual,Description="description",
File=bamfile,Platform=platformName,Source=source,Accession=acc,Genomes=<G1-ID,G2-ID,..,GK-ID>,Mixture=
<N1,N2,..,NK>,Genome_Description=<"S1","S2",..,"SK">>

Example:

##SAMPLE=<ID=NORMAL,SampleName=TCGA-06-0881-10A-01W,Individual=TCGA-06-0881,Description="Normal",File=TCGA-06-0881-10A-01W-0421-09.bam,Platform=Illumina,Source=dbGAP,Accession=1234,Genomes=<Germline,Tumor>,Mixture=<0.1,0.9>,Genome_Description=<"Germline
contamination","Tumor genome">>

-   "Description" field for genome mixture has been renamed to
    "Genome_Description" to distinguish it from sample description.

-   Values for tags related to genome mixture (Genomes, Mixture,
    Genome_Description) are within angle brackets.

### Column header meta-information

**Format**: Tab-delimited line starting with "#" and containing headers for all
columns in the BODY as shown below.

This is a mandatory header line where the first 8 fields are fixed and have to
defined in the column header. "FORMAT" onwards are optional and are included to
encapsulate per-sample/genome genotype data.

#CHROM POS ID REF ALT QUAL FILTER INFO FORMAT <SAMPLE1 or GENOME1> <SAMPLE2
or GENOME2> ...

BODY
### Variant records

Data lines are tab-delimited and list information about individual variants and
associated genotypes across samples. The first 8 fields (Figure 1) are required
to be listed in the VCF column header line. Some of these fields require
non-null values (see Table 6) for each record. For the remaining fixed fields,
even if the field does not have an associated value, it still needs to be
specified with a missing value identifier ("." in VCF 4.1). Subsequent fields
are optional.

**Table 6: Description of fields in the BODY of a VCF file**

| **Index** | **Field**  | **Case-sensitive** | **Description** | **Data type** | **Sample values**     | **Required\*** | **Additional notes** |
| --------- | ---------- | ------------------ | --------------- | ------------- | --------------------- | -------------- | -------------------- |
| 1         | CHROM      | Yes                                                                        | *Chromosome*: an identifier from the reference genome or the assembly file defined in the HEADER                                                                                                                                                                                                                             | Alphanumeric string                                                                      | 20                    | Yes            | Chromosome name should not contain "chr" prefix, e.g., "chr10" will be an invalid entry                                                                                                                                                                                                                                                                                                                                                                              |
|           |            |                                                                            |                                                                                                                                                                                                                                                                                                                              | *([1-22], X, Y, MT, <ID>)*                                                             | <ctg1>              |                |                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |
| 2         | POS        | Yes                                                                        | *Position*: The reference position, with the 1st base having position 1                                                                                                                                                                                                                                                      | Non-negative integer                                                                     | 1110696               | Yes            | \---                                                                                                                                                                                                                                                                                                                                                                                                                                                                 |
| 3         | ID         | Yes                                                                        | *Identifier*: Semi-colon separated list of unique identifiers if available                                                                                                                                                                                                                                                   | String, no white-space or semi-colons                                                    | rs6054257_66370       | No             | **Important**: When using an rsID as the variant identifier, please append chromosomal location of the variant to the ID. For example, if the variant is at chr7:6013153 and the corresponding rsID is rs10000, then the variant ID should be rs10000_6013153. This is to ensure that there is a consistent rule for satisfying the condition for unique IDs even if a file contains single rsID that maps to multiple variants                                      |
| 4         | REF        | Yes                                                                        | *Reference allele(s)*: Reference allele at the position                                                                                                                                                                                                                                                                      | String                                                                                   | GTCT                  | Yes            | Value in POS field refers to the position of the first base in the REF string                                                                                                                                                                                                                                                                                                                                                                                        |
|           |            |                                                                            |                                                                                                                                                                                                                                                                                                                              | *([ACGTN]+* )                                                                            |                       |                |                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |
| 5         | ALT        | Yes                                                                        | *Alternate allele(s)*: Comma separated list of alternate non-reference alleles called on at least one of the samples. Angle-bracketed ID String (\<ID\>) can also be used for symbolically representing alternate alleles                                                                                                  | String; no whitespace, commas, or angle-brackets in the ID string                        | G,GTCT                | No             | if ALT==<ID>, ID needs to be defined in the header as                                                                                                                                                                                                                                                                                                                                                                                                              |
|           |            |                                                                            |                                                                                                                                                                                                                                                                                                                              | *([ACGTN]+, <ID>, .)*                                                                  | .                     |                | ##ALT=<ID=*Id*,Description=*"description"*>                                                                                                                                                                                                                                                                                                                                                                                                                      |
|           |            |                                                                            |                                                                                                                                                                                                                                                                                                                              |                                                                                          | <INS:ME:ALU>        |                |                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |
| 6         | QUAL       | Yes                                                                        | *Quality score*: Phred-scaled quality score for the assertion made in ALT                                                                                                                                                                                                                                                    | Integer \>= 0                                                                            | 50                    | No             | Scores should be non-negative integers or missing values                                                                                                                                                                                                                                                                                                                                                                                                             |
| 7         | FILTER     | Yes                                                                        | *Filtering results*: PASS if this position has passed all filters, Otherwise, if the site has not passed all filters, a semicolon-separated list of codes for filters that fail                                                                                                                                              | String, no whitespace or semi-colon                                                      | PASS                  | No             | "0" is reserved and cannot be used as a filter String                                                                                                                                                                                                                                                                                                                                                                                                                |
|           |            |                                                                            |                                                                                                                                                                                                                                                                                                                              |                                                                                          | q10;s50               |                |                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |
| 8         | INFO       | Yes                                                                        | *Additional information*: INFO fields are encoded as a semicolon-separated series of keys (same as ID in an INFO declaration) with optional values in the format                                                                                                                                                           | String, no whitespace, semi-colons, or equal-signs                                       | NS=3;DP=14;           | No             | \---                                                                                                                                                                                                                                                                                                                                                                                                                                                                 |
| 9         | FORMAT     | Yes                                                                        | *Genotype sub-fields*: If genotype data is present in the file, the fixed fields are followed by a FORMAT column. The field contains a colon-separated list of all pre-defined FORMAT sub-fields (same as ID in a FORMAT declaration) that are applicable to all samples that follow                                         | String, no whitespace, sub-fields cannot contain colon                                   | GT:GQ:DP:HQ           | No             | "GT" must be the first sub-field if it is present in the FORMAT field                                                                                                                                                                                                                                                                                                                                                                                                |
| 10        | <SAMPLE> | Case should be same as in "ID" tag of \#\#SAMPLE declaration in the header | *Per-sample genotype information*: An arbitrary number of sample IDs can be added to the column header line and a variant record in the BODY can contain genotype information corresponding to FORMAT column for each sample. Contains a colon-separated list of values assigned to each of the sub-fields in FORMAT column  | String, no whitespace, sub-fields cannot contain colon                                   | 0\|0:48:1:51,51       | No             | Values are assigned to FORMAT sub-fields in the SAME order as specified in "FORMAT" column. All samples in any given row for a variant record MUST contain values for all sub-fields as defined in "FORMAT" column. If any of the fields does not have an associated value, then missing value identifier (".") should be used for that field. However, "." cannot be used as a value for any of the IDs in the FORMAT field (e.g., GT:.:DP would lead to an error). |

* A "Required" field cannot contain missing value identifier for any record
listed in data lines 

Extensions for TCGA data
========================

TCGA data includes but is not limited to SNP's and small indels. A variant
representation format for cancer data should be able to support more complex
variation types such as structural variants, complex rearrangements and RNA-Seq
variants. The following sub-sections present an overview of the extensions that
have been added to clearly describe such variations in a VCF file.

Structural variants
-------------------

A [structural variant](http://www.ncbi.nlm.nih.gov/dbvar/content/overview/) (SV)
can be defined as a region of DNA that includes a variation in the structure of
the chromosome. Such variations could be due to inversions and balanced
translocations or genomic imbalances (insertions and deletions), also referred
to as copy number variants (CNVs). Certain features have been added to the
format in order to clearly describe structural variants in a VCF file. A
detailed description of the extensions is available
[here](http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-41).

Complex rearrangements
----------------------

Chromosomal rearrangements are caused by breakage of DNA double helices at two
different locations. The broken ends in turn rejoin to produce a new chromosomal
arrangement. Complex rearrangements involving more than two breaks are
frequently observed in cancer genomes. Certain modifications need to be made to
the VCF standard to adequately represent such variations in a VCF file. A
detailed specification of the proposed extensions to describe rearrangements in
a VCF file is available
[here](http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-41).
Figure 2 illustrates some of the concepts relevant to VCF records for complex
rearrangements.

**Figure 2: Adjacencies and breakends in a chromosomal rearrangement** (adapted
from VCF 4.1 specification)

| ![media](images/ccr_VCF.png) |
| ------------------------------------------ |



A VCF file has one line for each of the two breakends in an adjacency. Table 7
provides a list of sub-fields that have been added to describe breakends. An
INFO sub-field (**SVTYPE=BND**) is used to indicate a breakend record.
Sub-fields MATEID and PARID are used to represent variant record IDs of
corresponding mates and partners respectively.

**Table 7: Fields added for breakends**

| **Field:Sub-field** | **Description** | **Declaration in HEADER** | **Required** | **(Sample values in BODY)** |
| ------------------- | --------------- | ------------------------- | ------------ | --------------------------- |
| INFO:**SVTYPE**     | Type of structural variant; SVTYPE is set to "BND" for breakend records | ##INFO=<ID=SVTYPE,Number=1,Type=String,Description="Type of structural variant">             | Yes                                  |
|                     |                                                                         | *SVTYPE=BND*                                                                                     | (SVTYPE=BND for breakend records)    |
| INFO:**MATEID**     | ID of corresponding mate of the breakend record                         | ##INFO=<ID=MATEID,Number=.,Type=String,Description="ID of mate breakend">                    | No                                   |
|                     |                                                                         | *MATEID=bnd_U*                                                                                   |                                      |
| INFO:**PARID**      | ID of corresponding partner of the breakend record                      | ##INFO=<ID=PARID,Number=.,Type=String,Description="ID of partner breakend">                  | No                                   |
|                     |                                                                         | *PARID=bnd_V*                                                                                    |                                      |
| INFO:**EVENT**      | ID of event associated to breakend                                      | ##INFO=<ID=EVENT,Number=.,Type=String,Description="ID of breakend event">                    | No                                   |
|                     |                                                                         | *EVENT=RR0*                                                                                      |                                      |

The specification for ALT field deviates from the standard format for breakend
records. ALT field for a breakend record can be represented in four possible
ways based on the type of replacement.

REF ALT Description

s t[p[ piece extending to the right of p is joined after t

s t]p] reverse comp piece extending left of p is joined after t

s ]p]t piece extending to the left of p is joined before t

s [p[t reverse comp piece extending right of p is joined before t

Legend:

s: sequence of REF bases beginning at position POS

t: sequence of bases that replaces "s"

p: position of the breakend mate indicating the first mapped base that joins at
the adjacency; represented as a string of the form "chr:pos"

[]: square brackets indicate direction that the joined sequence continues in,
starting from p

RNA-Seq variants
----------------

VCF specifications have been extended to address expressed variants obtained
from RNA-Seq. Features added for structural variants from genome/exome
sequencing are applicable to RNA-Seq structural variants. However, RNA-Seq
breakends are represented by setting **SVTYPE=FND** instead of BND (Table 8)
since they can be different from those observed in DNA-Seq.

**Table 8: Fields added for RNA-Seq variants**

| **Field:Sub-field** | **Description** | **Declaration in HEADER** | **Required** |
| ------------------- | --------------- | ------------------------- | ------------ |
| INFO:**SVTYPE**     | Type of structural variant; SVTYPE is set to "FND" for breakends associated with RNA-Seq | ##INFO=<ID=SVTYPE,Number=1,Type=String,Description="Type of structural variant">             | Yes                                                    |
|                     |                                                                                          | *SVTYPE=FND*                                                                                     | (required for RNA-Seq breakend records; SVTYPE=FND)    |

VCF files for RNA-Seq variants may include gene-related annotations. However,
this is not a standard feature of VCF files as eventually all VCF variants will
be annotated using information in Generic Annotation File (GAF). Additional INFO
and FORMAT sub-fields have been included to describe the characteristics of
expressed nucleotide variants (Table 8a).

**Table 8a: Annotation fields added for RNA-Seq variants**

| **Field:Sub-field** | **Description** | **Declaration in HEADER** | **Required** |
| ------------------- | --------------- | ------------------------- | ------------ |
| INFO:**SID**        | Unique identifiers from the gene annotation source as specified in ##geneAnno; "unknown" should be used if identifier is not known; comma-separated list of IDs can be used if variant overlaps with multiple features | ##INFO=<ID=SID,Number=.,Type=String,Description=Unique identifier from gene annotation source or unknown>                     | No           |
|                     |                                                                                                                                                                                                                          | *SID=13,198*                                                                                                                        |              |
| INFO:**GENE**       | HUGO gene symbol; "unknown" should be used when gene symbol is unknown; comma-separated list of genes can be used if variant overlaps with multiple transcripts/genes                                                    | ##INFO=<ID=GENE,Number=.,Type=String,Description=HUGO gene symbol>                                                            | No           |
|                     |                                                                                                                                                                                                                          | *GENE=ERBB2,ERBB2*                                                                                                                  |              |
| INFO:**RGN**        | Region where a nucleotide variant occurs in relation to a gene                                                                                                                                                           | ##INFO=<ID=RGN,Number=.,Type=String,Description=Region where nucleotide variant occurs in relation to a gene>                 | No           |
|                     |                                                                                                                                                                                                                          | *RGN=exon,3_utr*                                                                                                                    |              |
| INFO:**RE**         | Flag to indicate if position is known to have RNA-edits occur                                                                                                                                                            | ##INFO=<ID=RE,Number=0,Type=Flag,Description=Position known to have RNA-edits to occur>                                       | No           |
|                     |                                                                                                                                                                                                                          | *RE*                                                                                                                                |              |
| FORMAT:**TE**       | Translational effect of a nucleotide variant in a codon                                                                                                                                                                  | ##FORMAT=<ID=TE,Number=.,Type=String,Description="Translational                                                                  | No           |
|                     |                                                                                                                                                                                                                          | effect of the variant in a codon">                                                                                                 |              |
|                     |                                                                                                                                                                                                                          | *MIS,NA*                                                                                                                            |              |

Including validation status in VCF file
---------------------------------------

Somatic variations are often validated using follow-up experiments to confirm
the variant is not due to sequencing errors. Following points need to be
considered while including validation status in VCF file:

-   A single VCF file will contain sequence data for a single case. The file
    could be the result of merging calls from different centers so validation
    can be performed on a set of variants reported in a merged VCF file.

-   Validation with secondary technology is performed after obtaining results
    from primary sequencing method. Therefore, validation is a confirmation step
    and may or may not be performed before a first-pass VCF file with all
    candidate mutations is generated and submitted to the DCC.

-   A single mutation can be verified with multiple independent methods and the
    results may or may not be in agreement.

-   If results from different methods are in conflict, the final validation
    status of the variant call needs to be inferred based on available
    information. This could be done manually or programatically.

**Format validation**

Since validation data is added as additional genotype/sample columns, the file
will pass validation as long as all existing format rules are followed and
header declarations are correct.

**Sample TCGA VCF file with validation status**

Line1 ##fileformat=VCFv4.1

Line2 ##tcgaversion=1.1

Line3 ##fileDate=20120205

Line4 ##reference=file:///seq/references/1000GenomesPilot-NCBI36.fasta

Line5 ##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">

Line6 ##FORMAT=<ID=GQ,Number=1,Type=Integer,Description="Genotype Quality">

Line7 ##FORMAT=<ID=SS,Number=1,Type=Integer,Description="Variant status
relative to non-adjacent Normal,
0=wildtype,1=germline,2=somatic,3=LOH,4=unknown">

Line8 ##INFO=<ID=VLS,Number=1,Type=Integer,Description="Final validation
status relative to non-adjacent Normal,
0=wildtype,1=germline,2=somatic,3=LOH,4=unknown">

Line9 ##FILTER=<ID=q10,Description="Quality below 10">

Line10
##SAMPLE=<ID=NORMAL,SampleName=TCGA-06-0881-10A-01W,Individual=TCGA-06-0881,Description="Normal",File=TCGA-06-0881-10A-01W-0421-09_illumina.bam,Platform=Illumina,Source=dbGAP,Accession=1234>

Line11
##SAMPLE=<ID=NORMAL_454,SampleName=TCGA-06-0881-10A-01W,Individual=TCGA-06-0881,Description="Validation
normal sample tested with
454",File=TCGA-06-0881-10A-01W-0421-09_454.bam,Platform=454,Source=dbGAP,Accession=245>

Line12
##SAMPLE=<ID=TUMOR,SampleName=TCGA-06-0881-01A-01W,Individual=TCGA-06-0881,Description="Tumor",File=TCGA-06-0881-01A-01W-0421-09.bam,Platform=Illumina,Source=dbGAP,Accession=1234>

Line13
##SAMPLE=<ID=TUMOR_454,SampleName=TCGA-06-0881-01A-01W,Individual=TCGA-06-0881,Description="Validation
tumor sample tested with
454",File=TCGA-06-0881-01A-01W-0421-09_454.bam,Platform=454,Source=dbGAP,Accession=3456>

Line14
##SAMPLE=<ID=TUMOR_Sanger,SampleName=TCGA-06-0881-01A-01W,Individual=TCGA-06-0881,Description="Validation
tumor sample tested with Sanger
seq",File=.,Platform=Sanger_PCR,Source=.,Accession=.>

Line15 #CHROM POS ID REF ALT QUAL FILTER INFO FORMAT NORMAL TUMOR NORMAL_454
TUMOR_454 TUMOR_Sanger

Line16 20 14370 var1 G A 29 PASS VLS=2 GT:GQ:SS 0/0:48:. 0/1:50:2 0/0:20:.
0/1:20:2 0/1:.:2

Line17 5 15000 var2 T C 35 PASS VLS=1 GT:GQ:SS 0/1:48:. 1/1:51:3 0/1:60:.
0/1:50:1 0/1:13:1

Line18 3 170089 var2 G T 30 PASS . GT:GQ:SS 0/1:48:. 0/1:51:1 .:.:. .:.:. .:.:.

The format follows these guidelines:

1.  **Sample columns**

    -   An additional column is included for every line of evidence used for
        validation. In the example above, tumor calls are verified with 454 and
        Sanger sequencing and normal calls are validated with 454. Therefore, 3
        genotype columns exist in addition to the NORMAL and TUMOR sequencing
        calls obtained with the primary sequencing method.

    -   The validation platform name is appended to the original sample to
        distinguish the validation results from primary sequencing.
        <Sample>_<Platform> is used in the example above.

        -   **Note**: \<Platform\> can be obtained from DCC [Code Tables
            Report](http://tcga-data.nci.nih.gov/datareports/codeTablesReport.htm).
            The ##SAMPLE meta-information line also includes a 'Platform' tag
            where platform name is defined.

    -   Each new genotype column header added to the file (e.g., TUMOR_454,
        TUMOR_Sanger) has to be defined in the header using the ##SAMPLE
        meta-information line (e.g., Lines 13 and 14).

    -   As per VCF specification, the order of FORMAT sub-fields is defined by
        the FORMAT column and all calls from primary and validation sequencing
        should comply with this order.

    -   If a sub-field does not apply to any given validation call, it should be
        assigned a missing value (".").

2.  **FORMAT sub-field "SS"**

    -   For any given tumor genotype call, sub-field SS indicates variant status
        with respect to non-adjacent normal counterpart (0, 1, 2, 3, 4 or 5
        based on whether the variant is wildtype, germline, somatic, LOH,
        post-transcriptional modification, or unknown respectively). Therefore,
        each tumor genotype call (primary and secondary sequencing) will have
        its own corresponding SS sub-field.

3.  **INFO sub-field "VLS"**

    -   Sub-field VLS represents an inferred decision for a tumor genotype call
        and is based on the calls obtained with validation. In the example
        above, var1 shows a somatic call (SS=2) for the tumor sample based on
        primary sequencing, and both validation methods confirm this call.
        Therefore, the final validation status of var1 is a somatic variation
        (VLS=2). However, var2 has a LOH variant in tumor sample (SS=3) based on
        primary sequencing whereas both validation methods indicate that it is a
        germline variant (SS=1). In such a case, "VLS" has to be inferred from
        available information and could differ from the SS value assigned to the
        tumor sample based on primary sequencing.

Validation rules
================

At the minimum, every file needs to go through the checks listed below.
Following is an example of a VCF file that shows certain violations cited in the
listed validation steps. Please note that line numbers in the file segment below
are added for illustration purposes alone and are not expected to be found in an
actual VCF file.

Line1 ##fileformat=VCFv4.1

Line2 ##fileDate=20090805

Line3 ##source=myImputationProgramV3.1

Line4 ##reference=file:///seq/references/1000GenomesPilot-NCBI36.fasta

Line5 ##INFO=<ID=NS,Number=1,Type=Integer,Description="Number of Samples With
Data">

Line6 ##INFO=<ID=DB,Number=0,Type=Flag,Description="dbSNP membership">

Line7 ##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">

Line8 ##FORMAT=<ID=GQ,Number=1,Type=Integer,Description="Genotype Quality">

Line9 ##FORMAT=<ID=DP,Number=1,Type=Integer,Description="Read Depth">

Line10 ##FORMAT=<ID=PL,Number=3,Type=Integer,Description=" Normalized
Phred-scaled likelihoods for AA, AB, BB genotypes ">

Line11 ##FILTER=<ID=q10,Description="Quality below 10">

Line12 ##FILTER=<ID=s50,Description="Less than 50% of samples have data">

Line13 FILTER=<ID=c10,Description="Shallow coverage below 10x">

Line14 ##ALT=<ID=DEL:ME:ALU,Description="Deletion of ALU element">

Line15 #CHROM POS ID REF ALT QUAL FILTER INFO FORMAT TCGA-02-0001-01
TCGA-02-0001-02

Line16 20 14370 var1 G A 29 q10 NS=2;DP=14 GT:GQ:DP 0|0:48 0|1:48:3

Line17 19 15000 var2 G A 35 q10;s50 NS=2.5 GQ:GT 48:0|0 51:0|1

Line18 19 16000 var3 C T 30 q10;s10 NS=2 GT:GQ:DP 0/2:48:3 0/1:51:4

Line19 2 14477 rs123 C \<DEL:ME:ALU\> 12 PASS NS=3;DB GT:GQ 0/1:50 1/1:40

Line20 9 13567 . A \<DUP\> 20 PASS NS=3 GT:GQ:PL 0/1:49:42,3 1/1:38:96,47/70

Line21 3 18901 rs456 T C 15 PASS NS=3/DB GT 0/1 1/1

**Important**: A file will be validated as a TCGA VCF file only if it contains
##tcgaversion HEADER line (e.g., ##tcgaversion=1.1). The current acceptable
version is 1.1.

1.  Mandatory [header lines](#TCGAVariantCallFormat(VCF)1.1Specificat) should be
    present.

2.  All meta-information header lines should be prefixed with "\#\#".

3.  [Column header](#TCGAVariantCallFormat(VCF)1.1Specificat) line should be
    prefixed with "\#". A VCF file can contain only a single column header line
    that must contain all required field names.

4.  Any line lacking the "##" or "#" prefix will be assumed to be a BODY data
    line and will have to follow the specified format. For example, Line13 leads
    to a violation as it lacks "##" or "#" but is not a tab-delimited row
    containing variant information.

5.  HEADER lines cannot be present within the BODY of a file and vice-versa.

6.  [INFO](#TCGAVariantCallFormat(VCF)1.1Specificat),
    [FORMAT](#TCGAVariantCallFormat(VCF)1.1Specificat) and
    [FILTER](#TCGAVariantCallFormat(VCF)1.1Specificat)declarations should follow
    the format below where all keys are required but the order of keys is
    irrelevant.

7.  ##INFO=<ID=id,Number=number,Type=type,Description="description">

8.  ##FORMAT=<ID=id,Number=number,Type=type,Description="description">

9.  ##FILTER=<ID=id,Description="description">

10. Values assigned to *ID, Number, Type* and *Description* in INFO, FORMAT or
    FILTER declarations should follow the rules listed below. A detailed
    description of the declaration format is provided
    [here](#TCGAVariantCallFormat(VCF)1.1Specificat).

    1.  If an INFO or FORMAT sub-field exists in Table 4 or 5 respectively (i.e.
        ID of the sub-field matches value in "Sub-field" column of the table)
        then *ID*, *Number*, *Type* and *Description* values for that sub-field
        declaration must match the corresponding value in "Formatted
        declaration" column of the table for that sub-field. (TCGA VCF 1.1)

    2.  ID, Number, Type !\~ /(\\s\|,\|=\|;)/

    3.  *Number* is in {Integer\>=0, "A", "G", "."}

    4.  *Type* is in {Integer, String, Float, Flag, Character}

    5.  *Description* should be within double quotes and cannot itself contain a
        double quote

    6.  *Description* string cannot contain leading or trailing whitespace after
        opening or before closing quotation marks; Line10 shows a violation as
        *Description* string contains leading and trailing whitespace.

    7.  If ID == "FORMAT", then Type != "Flag"

11. Any INFO, FORMAT or FILTER sub-fields used in the BODY are required to be
    defined in the HEADER. For example, var1 (Line16) shows an example of a
    violation as read depth "DP" is assigned a value (DP=14) without being
    defined as an INFO sub-field in the HEADER.

12. Validation of **INFO** sub-fields:

    1.  An INFO sub-field should be included for a variant record in the BODY as
        *\<key=value\>* (e.g., NS=2) where *key*is the "ID" value of the
        sub-field in the HEADER declaration.

        -   **Exception**: An INFO field of "Flag" *Type* will not be assigned a
            value in the BODY. The presence of a flag in INFO column merely
            indicates that the variant record satisfies a condition associated
            with the flag. For example, Line19 has a "DB" flag without a value
            entry in the INFO column. "DB" in the INFO column indicates that the
            variant exists in dbSNP.

    2.  Multiple INFO sub-fields can be associated with a single variant record
        using ";" as a separator (e.g., Line16). Line21 has a violation as "/"
        is used as a separator in INFO column.

    3.  If INFO field "VLS" is defined for a record, its value can only be 0, 1,
        2, 3, 4, or 5 based on whether the mutation is wildtype, germline,
        somatic, LOH, post-transcriptional modification, or unknown.

13. Validation of **FORMAT** sub-fields:

    1.  FORMAT column for a variant record contains a colon-separated list of
        all pre-defined FORMAT sub-fields (identified by "ID" value in the
        HEADER declaration) that are applicable to all samples that follow. A
        ":" is the only valid separator for sub-fields.

    2.  Number of colon-separated sub-fields in FORMAT column should equal to
        number of colon-separated values assigned to each sample. For example,
        var1 (Line16) violates this rule for the sample TCGA-02-0001-01 as there
        are 3 sub-fields in FORMAT column but only 2 values in the sample
        column.

    3.  Following FORMAT fields are required for all variant records in a VCF
        file. Missing value (".") is allowed for these fields.

        -   Genotype (**GT**)

        -   Read depth (**DP**)

        -   Reads supporting ALT (**AD** or **DP4**)

        -   Average base quality for reads supporting alleles (**BQ**)

        -   Somatic status of the variant (**SS**). SS can be 0, 1, 2, 3, 4, or
            5 depending on whether relative to normal the variant is wildtype,
            germline, somatic, LOH, post-transcriptional modification, or
            unknown respectively

    4.  *GT* must be the first sub-field in the string FORMAT. For example, var2
        (Line17) violates this rule as GT is not the first sub-field even though
        it is present in the FORMAT field.

        -   GT is a required sub field for all variants. Missing value (".") is
            allowed for GT. GT is not a required sub field and can be omitted
            for a variant row if none of the samples have genotype calls
            available (TCGA VCF 1.1)

        -   *GT* represents the genotype, encoded as allele values separated by
            either of / (genotype unphased) or \| (genotype phased). The
            allele values are 0 for the reference allele (in REF field), 1 for
            the first allele listed in ALT, 2 for the second allele list in ALT
            and so on. Examples: 0/1, 1\|0, or 1/2, etc.

        -   *GT*is assigned only one allele value for haploid calls (e.g. on Y
            chromosome). Therefore, if CHROM=="Y" then*GT*should have only one
            allele value assigned to it (e.g., "1", "0", ".", etc.) instead of
            two alleles (e.g., "1/1", "0\|0"). If CHROM=="MT" then There is no
            constraint on the number of alleles as long as the number is bounded
            within the alleles listed in REF and/or ALT (e.g., 0/1, 0/1/2, 1 are
            all valid values for MT if REF and ALT have one and two allele
            values respectively).

        -   All samples should have values assigned to *GT* for any given
            variant. If an allele cannot be called for a sample at a given
            locus, . will be specified for each missing allele in the *GT*
            field (for example "./." for a diploid genotype and "." for haploid
            genotype).

        -   Validation should include ensuring that allele number in *GT* is
            within the range of alleles specified in ALT and REF. For example,
            var3 (Line18) violates this rule as it lists GT as "0/2" for sample
            TCGA-02-0001-01 but ALT contains only one allele so the only
            acceptable allele numbers are 0 (REF) and 1 (ALT).

14. If an INFO or FORMAT sub-field is declared in the header AND is assigned a
    value for a variant record in the body, the data type should be consistent
    with the expected type defined in the *Type*key of the corresponding
    declaration. For example, var2 (Line17) violates this rule as the definition
    for "NS" INFO sub-field states the data type is integer whereas the variant
    record contains a float value (2.5) assigned to the sub-field.

    1.  **Exception**: The rule does not apply if *Type* of a field is not
        defined or is incorrectly defined (e.g., field not declared in HEADER,
        *Type* not included in declaration, incorrect value for *Type)*. It also
        does not apply to any missing values (denoted with ".") in the record as
        they do not have an associated data type.

15. Multiple comma-separated values (corresponding to value assigned to *Number*
    key in declaration) can be specified for an INFO or FORMAT sub-field for a
    variant record. No other character can be used as separator. Line20 shows a
    violation as a "/" is used as separator between 2nd and 3rd values for
    *"PL"* FORMAT sub-field in the second sample column.

16. If *Number* tag is assigned a known bounded value (an integer, "A", "G") for
    an INFO/FORMAT sub-field, it should be consistent with number of values
    specified for any variant record in BODY of file. For example, Line20 shows
    a violation as *"PL"* is associated with 3 integer values (Line10) but the
    variant record has only 2 comma-separated integer values (42,3) for
    TCGA-02-0001-01.

17. Validation of **FILTER** sub-fields:

    1.  Valid values for FILTER column are "PASS" or a code for the filter that
        the variant call fails (e.g., "q10" in Line16). The code must correspond
        to the "ID" value of the corresponding FILTER declaration.

    2.  If a call fails multiple filters, FILTER column should contain
        semicolon-separated list of all failed filter codes (e.g., "q10;s50" in
        Line17). A ";" is the only valid separator.

    3.  All codes listed in the FILTER column must have a well-formed
        declaration in the HEADER. Line18 shows a violation as "q10" does not
        have an associated definition in the HEADER.

18. \<TCGA-VCF\> Validation of
    [SAMPLE](#TCGAVariantCallFormat(VCF)1.1Specificat) meta-information lines:

    1.  Each sample ID in the column header (immediately after FORMAT column)
        must have an associated HEADER declaration where value assigned to "ID"
        tag in the declaration is the same as sample ID used in the column name.

    2.  Declaration must contain all required fields.

    3.  Genome mixture tags (Genomes, Mixture, Genome_Description) are enclosed
        within angle brackets (<>) and can have multiple comma-separated
        values.

    4.  If more than one of the genome mixture tags (Genomes, Mixture,
        Genome_Description) are defined in a SAMPLE meta-information line, then
        number of comma-separated values should be the same for all defined
        tags. For example, "Genomes=<G1,G2>,Mixture=<0.1,0.8,0.1>" would
        lead to a violation as Mixture has 3 values while Genomes has only 2
        values.

    5.  Individual values in "Genomes" are strings without white-space, comma or
        angle brackets.

    6.  Individual values in "Mixture" represent proportion (floating point
        number >= 0 and <= 1) of each genome in the sample and all
        comma-separated values should add up to a sum of 1.

    7.  Individual values in "Genome_Description" are strings surrounded by
        double quotes where the string itself cannot contain a double quote.

    8.  The value assigned to "SampleName" must be a valid [aliquot
        barcode](https://docs.gdc.cancer.gov/Encyclopedia/pages/TCGA_Barcode/) / [UUID](https://docs.gdc.cancer.gov/Encyclopedia/pages/UUID/)
        in the database (TCGA VCF 1.1).

19. <TCGA-VCF> Validation of
    [PEDIGREE](#TCGAVariantCallFormat(VCF)1.1Specificat) meta-information lines:

    1.  Declaration line should follow the format:

    2.  ##PEDIGREE=<Name_0=G0-ID,Name_1=G1-ID,...,Name_N=GN-ID>

>   where:

-   N \>= 1

-   Name_0 through Name_N are arbitrary (not literal) strings that cannot
    contain white-space, comma, or angle brackets (TCGA VCF 1.1)

-   G0-ID through GN-ID are strings that cannot contain white-space, comma, or
    angle brackets. Each of these should be a header for the genotype columns
    immediately after FORMAT column and should be defined using "ID" tag in the
    corresponding ##SAMPLE meta-information line. (TCGA VCF 1.1)

-   The keys and values used in the <Name_N=Value_N> should be unique across
    assignments in any given PEDIGREE declaration.

1.  Validation of custom meta-information fields:

    1.  If a user-created custom meta-information declaration is encountered and
        the corresponding key/value structure and content have not been defined
        in this specification, the line should be validated to ensure it follows
        one of the following two formats:

    2.  ##key=value

    3.  Example:

    4.  ##<INDIVIDUAL=TCGA-24-0980>

    5.  OR

    6.  ##FIELDTYPE=<key1=value1,key2=value2,...>

    7.  Example:

    8.  ##contig=<ID=20,length=62435964,assembly=B36,md5=f126cdf8a6e0c7f379d618ff66beb2da,species="Homo
        sapiens",taxonomy=x>

>   where:

-   key !\~ /(\\s\|,\|=\|;)/

-   value !\~ /(\\s\|,\|=\|;)/ UNLESS *value* is within double quotes, in which
    case it cannot itself contain a double quote or leading/trailing whitespace
    OR if *value* is within angle brackets.

1.  *CHROM*, *POS*, and *REF* are required fields and cannot contain missing
    value identifiers. Please refer to [Table
    6](#TCGAVariantCallFormat(VCF)1.1Specificat) for acceptable values.

    1.  <TCGA-VCF> *CHROM* is in {[1-22], X, Y, MT,<chr_ID>} where chr_ID
        cannot contain whitespace or <>

    2.  If CHROM == <chr_ID> then the VCF file MUST have a declaration for
        assembly file in the HEADER. Please note that values assigned to the
        field are currently not being validated.

    3.  ##assembly=url or filename

    4.  Example:

    5.  ##assembly=ftp://ftp-trace.ncbi.nih.gov/1000genomes/ftp/release/sv/breakpoint_assemblies.fasta

    6.  ##assembly=breakpoint_assemblies.fasta

    7.  *POS* is a non-negative integer

    8.  *REF* =\~ /[ACGTN]+/

2.  <TCGA-VCF> *ALT* is in {[ACGTN]+, ".", <ID>, **SV_ALT**};

    1.  String SV_ALT can be in one of the following four formats and can be
        used in the *ALT*field ONLY when the corresponding INFO field has the
        key-value pair "SVTYPE=BND" or "SVTYPE=FND".

    2.  Format Example

    3.  seq[chr:pos[ G[17:198982[

    4.  seq]chr:pos] GC]1:238909]

    5.  ]chr:pos]seq ]\<ctg1\>:235788]GCNA

    6.  [chr:pos[seq [1:2812734[ACT

>   where:

-   *seq* is in {[ACGTN]+, "."}

-   *chr* is in {\<chr_ID\>, [1-22], X, Y, MT} where *chr_ID* is a string

-   *pos* is a non-negative integer

1.  Similar to 18b, if chr == <chr_ID> (where *chr_ID* is a string) then the
    VCF file must have an ##assembly declaration in the HEADER.

2.  If *ALT* is assigned a value in <ID> format, (e.g., rs123 in Line19),
    <ID> should be defined in the HEADER as
    ##ALT=<ID=*ID*,Description="*Description*"> (Line14) where ID cannot
    contain white-space or angle brackets. Line20 shows a violation of this rule
    as *ALT==*<DUP> but there is no corresponding *ALT* declaration in the
    HEADER with <ID=DUP>.

3.  ALT can contain multiple comma-separated values. No other character can be
    used as a separator.

4.  No two records are allowed to have the the same *ID* value. Two records can,
    however, have the same *CHROM* and *POS*values.

    1.  **Exception**: Multiple records in a file are allowed to have the same
        missing value identifier (".") as *ID*.

5.  *QUAL* field can only contain non-negative integers or "." (missing value).

6.  <TCGA-VCF> If INFO sub-field "VT" is declared and used in the BODY, its
    value can only be in {SNP, INS, DEL}

7.  <TCGA-VCF> If FORMAT sub-field "SS" is declared and used in the BODY, its
    value can be 0, 1, 2, 3, 4 or 5 depending on whether relative to normal the
    variant is wildtype, germline, somatic, LOH, post-transcriptional
    modification, or unknown respectively.

8.  <TCGA-VCF> "DP" sub-field for read depth can be defined in INFO (combined
    depth across all samples) or FORMAT (depth in a specific sample) field. If
    both INFO and FORMAT have values for the sub-field, then sum of DP values
    across all FORMAT sample columns should be equal to DP value in the INFO
    field.

9.  <TCGA-VCF> Validation of **complex rearrangement** records:

    1.  If INFO field includes key-value pairs "SVTYPE=BND" or "SVTYPE=FND" and
        has values for "MATEID" and/or "PARID", then the value (or multiple
        comma-separated values) assigned to MATEID or PARID should exist in the
        file as "ID" field for another variant record.

10. <TCGA-VCF> Validation of RNA-Seq **annotation fields**:

    1.  If INFO field includes "SID", "GENE" or "RGN" keys with associated
        values, then file MUST contain a declaration for ##geneAnno in the
        HEADER.

    2.  Number of comma-separated values in the optional INFO sub-fields "SID",
        "GENE" and "RGN" and the FORMAT sub-field "TE" must be the same if more
        than one of these sub-fields are defined for a record.

    3.  INFO sub-field "RGN" is in {5_utr, 3_utr, exon, intron, ncds, sp}.

    4.  FORMAT sub-field "TE" is in {SIL, MIS, NSNS, NSTP, FSH, NA}

    5.  If "RGN" and "TE" have the same number of comma-separated values, then
        "RGN" must be "exon" for "TE" to have any value other than "NA". For
        example, if "RGN=exon,intron,intron" then having "MIS,SIL,NA" for TE
        would lead to a violation as the 2nd value for RGN is "intron" but the
        corresponding TE value is "SIL" instead of "NA".

11. <TCGA-VCF> Validation of
    [vcfProcessLog](#TCGAVariantCallFormat(VCF)1.1Specificat) tags:

##vcfProcessLog=<InputVCF=<file1.vcf>,InputVCFSource=<varCaller1>,InputVCFVer=<1.0>,InputVCFParam=<a1,c2>,InputVCFgeneAnno=<anno1.gaf>>

OR

##vcfProcessLog=<InputVCF=<file1.vcf,file2.vcf,file3.vcf>,InputVCFSource=<varCaller1,varCaller2,varCaller3>,InputVCFVer=<1.0,2.1,2.0>,

InputVCFParam=<a1,c2;a1,b1;a1,b1>,InputVCFgeneAnno=<anno1.gaf,anno2.gaf,anno3.gaf>,

MergeSoftware=<sw1,sw2>,MergeParam=<a1,a2;b1,b2>,MergeVer=<2.1,3.0>,MergeContact=<johndoe@xyz.edu>>

1.  Individual values for each tag are enclosed within angle brackets (<>)
    instead of double quotes.

    1.  If a field contains multiple values, they are separated by comma.
        **Exception**: Separator for multiple values in *InputVCFParam* and
        *MergeParam* is a ";" instead of ",". Individual values within these
        tags can contain comma-separated parameters (e.g., <a1,c2;a1,b1;a1,b1
        in the example given above).

    2.  If *InputVCF* tag has multiple comma-separated values assigned to it
        (please refer to the second example above), then *InputVCFSource,
        InputVCFVer, InputVCFParam,* and *InputVCFgeneAnno* must contain the
        same number of values. If a value is not known, it should be substituted
        with the missing value identifier (".").

    3.  If *InputVCF* contains only a single value, then all tags =\~ /Merge.\*/
        are optional and can either be omitted or can contain missing value
        identifier ("."). The reason is that attribute related to merging VCF
        files are applicable only if multiple input VCF files are being merged.

    4.  If *MergeSoftware* contains multiple comma-separated values,
        *MergeParam* and *MergeVer* should contain the same number of values.
        There is no such constraint for *MergeContact.*

UUID-compliant files should satisfy the following criteria:

1. The value assigned to "SampleUUID" must be a valid aliquot
[UUID](https://docs.gdc.cancer.gov/Encyclopedia/pages/UUID/).

-   The [metadata](https://docs.gdc.cancer.gov/Encyclopedia/pages/TCGA_Barcode/) represented by
    "SampleTCGABarcode" must correspond to SampleUUID at the DCC.

2. If ##INDIVIDUAL is declared in the header, then values assigned to
SampleUUID in all ##SAMPLE declarations should correspond to the same
participant ID, and the TCGA barcode for the participant should be assigned to
##INDIVIDUAL.



Handling failed checks
----------------------

-   A VCF file would be required to pass ALL the checks listed above and any
    violation will lead to a "Failed" validation.

-   Even if a failure is encountered, the file would still need to go through
    all other checks for validation to be complete. Exception to this
    requirement would include cases where execution of one validation check is
    dependent on the success of another prerequisite step. For example, number
    of values associated with a FORMAT field for a variant record cannot be
    validated if the field itself is not declared in the HEADER or has a missing
    *Number* tag.

-   A summary of all failed checks should be provided as an output.

Test files
----------

The following table lists sample files that can be used to test various
validation steps. Parent directory is
<https://tcga-data.nci.nih.gov/tcgafiles/tcgajamboree/Benchmark_VCF>.

**Table 9: Test files known to pass/fail validation steps**

| **Expected result** | **Validation**                                               | **Test file**                                                  |
|---------------------|--------------------------------------------------------------|----------------------------------------------------------------|
| Success             | file with no failures                                        | /vcfFormat/TCGA-24-0980_IlluminaGA-DNASeq_exome.format2.vcf    |
| Failure             | "chr" prefix for chromosome names                            | /BI/20110324/BCM-GBM_solid.TCGA-06-0208.mut.vcf                |
| Failure             | column header line has no "\#" as beginning                  | /genome.wustl.edu/20110420/TCGA-06-0145-01A-01W-0224-08.vcf.gz |
| Failure             | double quotes missing from SAMPLE metadata lines Description | /BCM/TCGA-06-0145_IlluminaGA-DNASeq_exome.vcf                  |
| Failure             | scores in QUAL column are negative integers                  | /UCSC/20110420/TCGA-13-0723_W_capture.vcf                      |

-   missing values for FORMAT fields

-   trailing whitespace at the end of description strings in metadata

-   filter not listed in FILTER metadata line

-   in multi-allelic sites, different alternative alleles in the ALT field are
    separated by "/" instead of ","


# STAR 2-Pass Transcriptome

## Description ##

STAR 2-Pass Transcriptome is an alignment pipeline used in GDC RNA-Seq harmonization.

## Overview ##

STAR 2-Pass Transcriptome is a pipeline used for RNA-Seq reads alignment at the GDC. Alignment is performed with STAR and generates sequencing reads data.

### Input

* Submitted sequencing reads

### Output

* BAM (Data Type: Aligned Reads)

[![STAR Pipeline Diagram](https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/images/RNA-Seq-DR32_Image.png)](https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/images/RNA-Seq-DR32_Image.png "Click to see the full image.")

## References ##

1. [mRNA Analysis Pipeline](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/)
1. [RNA-Seq Alignment Workflow](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/#rna-seq-alignment-workflow)
1. [RNA-Seq Alignment Command Line Parameters](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/#rna-seq-alignment-command-line-parameters)

## External Links ##

* [STAR](https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf)
* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type

# AscatNGS

## Description ##

ASCATNGS is a copy number variation (CNV) analysis pipeline tailored for next-generation sequencing (NGS) data used in GDC whole genome sequencing (WGS) harmonization.

## Overview ##

ASCATNGS is specifically designed to detect CNVs in tumor and normal samples from NGS data. It processes aligned BAM files to generate segmented CNV data, providing high-resolution insights into genomic alterations.

### Input

* Tumor BAM - WGS
* Normal BAM - WGS

### Output

* TSV (Data Type: Gene Level Copy Number)
* TXT (Data Type: Copy Number Segment)

## References ##

1. [Copy Number Variation Analysis Pipeline](/Data/Bioinformatics_Pipelines/CNV_Pipeline/)
1. [ASCAT Pipelines](/Data/Bioinformatics_Pipelines/CNV_Pipeline/#ascat-pipelines)
1. [Whole Genome Sequencing Variant Calling](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/#whole-genome-sequencing-variant-calling)

## External Links ##

* [AscatNGS](https://github.com/cancerit/ascatNgs)
* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type

# Seurat - 10x Chromium

## Description ##

Seurat - 10x Chromium is a secondary gene expression pipeline used in GDC single-cell RNA-Seq (scRNA-Seq) harmonization.

## Overview ##

Seurat - 10x Chromium is a pipeline used for scRNA-Seq for secondary gene expression analysis at the GDC. Secondary gene expression analysis is performed with Seurat using gene expression data and generates transcriptome profiling data.

### Input

* MEX (Filtered Counts) - scRNA-Seq

### Output

* HDF5 (Data Type: Single Cell Analysis)
* TSV (Data Type: Differential Gene Expression, Single Cell Analysis)

## References ##

1. [mRNA Analysis Pipeline](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/)
1. [scRNA-Seq Pipeline (single-nuclei)](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/#scrna-seq-pipeline-single-nuclei)
1. [scRNA Analysis Pipeline](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/#scrna-analysis-pipeline)

## External Links ##

* [Seurat](https://satijalab.org/seurat/)
* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type

# Pindel

## Description ##

Pindel is a somatic variant calling pipeline used in GDC whole genome sequencing (WGS), whole exome sequencing (WXS) and targeted sequencing harmonization.

## Overview ##

Pindel is one of the pipelines used for WGS, WXS and targeted sequencing somatic variant calling at the GDC. Somatic variant calling is performed with Pindel using tumor and normal alignments and generates indel data.

### Input

* Tumor BAM - WGS, WXS or Targeted Sequencing
* Normal BAM - WGS, WXS or Targeted Sequencing

### Output

* VCF (Data Type: Raw Simple Somatic Mutation)

## References ##

1. [Pindel Command Line Parameters at the GDC](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/#pindel)
1. [DNA Seq Processing at the GDC](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/)

## External Links ##
* [Pindel at Washingotn University in St.Louis](https://gmt.genome.wustl.edu/packages/pindel/)
* [Overview of GDC Harmonization Workflows](https://github.com/NCI-GDC/gdc-workflow-overview/blob/master/README.md)
* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type


# MuSE Annotation

## Description ##

MuSE Annotation is a somatic mutation annotation pipeline in GDC whole exome sequencing (WXS) and targeted sequencing harmonization. 

## Overview ##

MuSE Annotation is a workflow in the GDC that annotates somatic variants identified by the MuSE variant calling pipeline.

### Input

* VCF (Data Type:  Raw Simple Somatic Mutation)

### Output

* MAF (Data Type: Annotated Somatic Mutation)
* VCF (Data Type: Annotated Somatic Mutation)

## References ##

1. [DNA-Seq Analysis Pipeline](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/)

## External Links ##

* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type

# Universally Unique Identifier (UUID) #
## Description ##
A UUID (Universal Unique Identifier) is a 128-bit number used to uniquely identify an object or entity in a system.

## Overview ##
In the GDC every entity in the data model is assigned a UUID.  An entity can be a file, case, project, or other node.  Uniquely identifying objects in this manner allows the GDC to have a uniform method of referencing any object in the system<sup>1</sup>.        


## References ##
1. [GDC Entity UUIDs](/API/Users_Guide/Getting_Started/#entity-uuids)

## External Links ##
* [Universally unique identifier](https://en.wikipedia.org/wiki/Universally_unique_identifier)
* [RFC 4122 UUID](http://www.ietf.org/rfc/rfc4122.txt)


Categories: General


# MuTect2 Annotation

## Description ##

MuTect2 Annotation is a somatic mutation annotation pipeline in GDC whole exome sequencing (WXS) and targeted sequencing harmonization.

## Overview ##

MuTect2 Annotation is a workflow in the GDC that annotates somatic variants identified by the MuTect2 variant calling pipeline.

### Input

* VCF (Data Type:  Raw Simple Somatic Mutation)

### Output

* MAF (Data Type: Annotated Somatic Mutation)
* VCF (Data Type: Annotated Somatic Mutation) 

## References ##

1. [DNA-Seq Analysis Pipeline](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/)

## External Links ##

* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type

# Representational state transfer (REST) Application Programming Interface (API) #
## Description ##
A RESTful API is an application program interface (API) that uses HTTP requests to GET, PUT, POST and DELETE data.
## Overview ##
The GDC offers a RESTful API to the client that allows them access to perform all functions currently available on the data portals with the added bonus of accessing additional features like advanced queries to meta data, scripted data downloads, and submission function.  In general, REST-compliant Web services allow requesting systems to access and manipulate textual representations of Web resources using a uniform and predefined set of stateless operations.


## References ##
1. [GDC API](https://gdc.cancer.gov/developers/gdc-application-programming-interface-api)
2. [GDC API User's Guide](/API/Users_Guide/Getting_Started/)


## External Links ##
* [Representational state transfer](https://en.wikipedia.org/wiki/Representational_state_transfer)

Categories: General


# ASCAT2

## Description ##

ASCAT2 is a copy number variation (CNV) analysis pipeline used in GDC genotyping array harmonization.

## Overview ##

ASCAT2 is used to analyze CNVs in tumor and normal samples from genotyping arrays. It processes the raw array files to generate segmented CNV data.

### Input

* Tumor CEL - Genotyping Array
* Normal CEL - Genotyping Array

### Output

* TXT (Data Type: Gene Level Copy Number and Allele-specific Copy Number Segment)

## References ##

1. [Copy Number Variation Analysis Pipeline](/Data/Bioinformatics_Pipelines/CNV_Pipeline/)
1. [ASCAT Pipelines](/Data/Bioinformatics_Pipelines/CNV_Pipeline/#ascat-pipelines)

## External Links ##

* [Example ASCAT analysis](https://github.com/VanLoo-lab/ascat/blob/v2.5/ExampleData/ASCAT_examplePipeline.R)
* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type






# Birdseed

## Description ##

Birdseed is a genotyping algorithm used for the detection of single nucleotide polymorphisms (SNPs) in GDC genotyping array data harmonization.

## Overview ##

Birdseed is utilized for genotyping SNPs from microarray data. This pipeline processes raw array data, identifies SNPs, and generates genotype calls. It is specifically designed for genotyping arrays such as the Affymetrix SNP 6.0 array, providing accurate SNP detection across different sample types. Birdseed files were originally processed and generated by the TCGA program and brought into the GDC Data Portal when the GDC Legacy Archive was retired.

### Input

* Tumor CEL - Genotyping Array
* Normal CEL - Genotyping Array

### Output

* TSV (Data Type: Simple Germline Variation)

## References ##

1. [Copy Number Variation Analysis Pipeline](/Data/Bioinformatics_Pipelines/CNV_Pipeline/)
1. [SNP Array-Based Data](/Encyclopedia/pages/SNP_Array-Based_Data/#data-formats)

## External Links ##

* [Overview of GDC Harmonization Workflows](https://github.com/NCI-GDC/gdc-workflow-overview)
* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type

# BCGSC miRNA Profiling

## Description ##

BCGSC miRNA Profiling is a miRNA profiling pipeline used in GDC for miRNA quantification analysis.

## Overview ##

BCGSC miRNA Profiling is a part of the miRNA Analysis Pipeline in the GDC miRNA quantification analysis. It uses tumor or normal alignments and generates Isoform Expression Quantification and miRNA Expression Quantification data.

### Input

* GDC-aligned BAM - miRNA-Seq

### Output

* TSV/TXT (Data Type: Isoform Expression Quantification, miRNA Expression Quantification)

## References ##

1. [miRNA Analysis Pipeline](/Data/Bioinformatics_Pipelines/miRNA_Pipeline/)
1. [miRNA Expression Workflow](/Data/Bioinformatics_Pipelines/miRNA_Pipeline/#mirna-expression-workflow)

## External Links ##

* [BCGSC's GitHub](https://github.com/bcgsc/mirna)
* ["Large-scale profiling of microRNAs for The Cancer Genome Atlas" Article](http://nar.oxfordjournals.org/content/early/2015/08/13/nar.gkv808.full)
* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type

# Controlled Access #
## Description ##
Data in the GDC is considered either open or controlled access.  Access to controlled-access (i.e. protected) data in the GDC is granted on a per project basis via the database of Genotypes and Phenotypes (dbGaP).

## Overview ##

While much of the data in the GDC is open access, many file types are controlled access. This primarily includes raw sequencing data such as BAM or FASTQ files as well as VCF files and protected MAF files. The informed consent under which the data or samples were collected is the basis for the submitting institution to determine whether the data should be available through unrestricted or controlled access. To gain access to these files a user must apply for access via dbGaP to individual projects<sup>1</sup>. Each project has a Data Access Committee (DAC) that will approve or disapprove data access requests.  Before gaining access through dbGaP users also need to obtain an eRA Commons ID for authentication purposes<sup>2</sup>.  

Genomic data access is governed by the NIH's Genomic Data Sharing Policy.  

## References ##
1. [GDC Granting Access to Controlled Data](https://gdc.cancer.gov/access-data/obtaining-access-controlled-data)

## External Links ##
* [NIH Genomic Data Sharing Policy](https://sharing.nih.gov/genomic-data-sharing-policy)
* [dbGaP](https://www.ncbi.nlm.nih.gov/gap)

Categories: General


# Manifest File #
## Description ##
A GDC manifest file is a file that is created by either the portal or API containing a list of files that the end user has requested.  
## Overview ##
When a user requests files from the GDC they are given the option to create a download manifest file that can be used in conjunction with the Data Transfer Tool to retrieve them.  The download manifest file contains the UUID, MD5 checksum, file size, and file name of each file listed in it.  The download manifest creation process is a function of the GDC API and is made available to GDC users from an API download endpoint or from the GDC Data Portal.

A manifest file can also be used to simplify the upload process to the GDC Submission System.  The format of the submission manifest differs from the download manifest.  A upload manifest file can be generated from either the API or GDC Submission Portal.

## References ##
1. [Manifest endpoint](/API/Users_Guide/Downloading_Files/#manifest-endpoint)
2. [GDC Data Transfer Tool](/Data_Portal/Users_Guide/Repository.md#generating-a-manifest-file-for-the-data-transfer-tool)
3. [Upload Manifest](/API/Users_Guide/Submission/#upload-manifest)
4. [Data Transfer Tool Submission with Manifest ](/Data_Transfer_Tool/Users_Guide/Preparing_for_Data_Download_and_Upload/#obtaining-a-manifest-file-for-data-uploads)

## External Links ##
* N/A

Categories: General


Annotations
===========================

Annotations contain important information about files, cases, or metadata nodes that may be of use to data downloaders when analyzing GDC data.  They should be reviewed prior to running an analysis.  An annotation may include key comments about why particular patients, samples, or files are absent from the GDC or why they may exhibit critical differences from others. Annotations include information that cannot be submitted to the GDC through other existing nodes or properties.

Annotations are automatically downloaded in TSV format with impacted files when using the Data Transfer Tool.  They may also be searched via the [API](/API/Users_Guide/Search_and_Retrieval/#annotations-endpoint) or on the annotations page of the [GDC Data Portal](https://portal.gdc.cancer.gov/annotations).  Instructions on accessing annotations in the GDC Data Portal are found in the [GDC Data Portal User Guide](/Data_Portal/Users_Guide/Repository/#annotations-view).

For information on Annotation structure and content please review the [GDC Data Dictionary](/Data_Dictionary/viewer/#?view=table-definition-view&id=annotation)

For information about TCGA conventions for annotations please see the [TCGA Introduction to Annotations](/Encyclopedia/pages/Annotations_TCGA).  

If a submitter would like to create an annotation, please contact the GDC Support Team (support@nci-gdc.datacommons.io).



## References ##
1. [API User Guide](/API/Users_Guide/Search_and_Retrieval/#annotations-endpoint)
2. [GDC Data Portal](https://portal.gdc.cancer.gov/annotations)
3. [GDC Data Portal User Guide](/Data_Portal/Users_Guide/Repository/#annotations-view)
4. [GDC Data Dictionary](/Data_Dictionary/viewer/#?view=table-definition-view&id=annotation)
5. [TCGA Annotations](Annotations_TCGA/)


## External Links ##
* N/A

Categories: Data Type


# RNA-Seq #
## Description ##

RNA-Seq is a sequencing method used to determine gene expression levels. RNA-Seq data originates from extracted RNA that was reverse transcribed into DNA and sequenced on a next-generation sequencing platform. The number of reads determined to have originated from each transcript (usually by alignment) are proportional to their expression level.

## Overview ##

RNA-Seq data in the GDC is used to generate a gene expression profile for tumor samples across many cancer types and to determine which gene expression levels are responsible for tumor development. The GDC harmonizes RNA-Seq data by aligning raw RNA reads to the GRCh38 reference genome build and calculating gene expression levels with standardized protocols<sup>1</sup>. RNA-Seq data is mostly available for tumor samples, although some normal samples have associated RNA-Seq data.

### Data ###

RNA-Seq data is available as aligned reads (BAM) and expression levels as: raw counts and normalized with TPM, FPKM, or FPKM-UQ. Reads that did not align are also included in BAM files to facilitate the retrieval of the original raw data.   

## References ##
1. [GDC mRNA Expression Pipeline](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/)

## External Links ##
* N/A

Categories: Experimental Strategy


# GDC Documentation Site #
## Description ##
The GDC Documentation Site<sup>1</sup> is the official project documentation site of the GDC which provides users with detailed information on GDC tools and data.

## Overview ##

The GDC Documentation Site<sup>1</sup> contains in-depth information and tutorials on the usage of the GDC features. The website includes tutorials on using the Data Portal, API, Data Transfer Tool, Submission Portal, and Data Dictionary. A guide to interpreting the data available at the GDC, which includes guides to the file formats and bioinformatics pipelines used to process the data, is also included.

## References ##
1. [GDC Documentation Web Site](https://docs.gdc.cancer.gov)

## External Links ##
* N/A

Categories: Tool


# Clinical Data #

## Description ##
Clinical data is a collection of data related to patient diagnosis, demographics, exposures, laboratory tests, and family relationships.  

## Overview ##

A core focus of the GDC is to enable the comparison of cancer genomic data to patient clinical data.  This may include patient diagnosis, exposures, laboratory tests, and family relationships.  To allow for comparisons across projects the GDC adheres to a common set of clinical terms.

Clinical data vocabulary in the GDC is defined in the GDC Data Dictionary<sup>1</sup>.  A simple list of all GDC clinical terms can be found on the GDC Website<sup>2</sup>. Whenever possible each clinical data property is associated with a Common Data Element defined in the [caDSR II Browser](https://cadsr.cancer.gov/onedata/Home.jsp), which is part of the [Center for Biomedical Informatics & Information Technology](https://datascience.cancer.gov/).  CDEs and their component definitions in the [NCI Thesaurus](https://ncit.nci.nih.gov/ncitbrowser/) provide very precise descriptions of clinical data, which allows data consumers and submitters to understand precisely what data in the GDC represents.

Additional information on the goals and origins of the GDC clinical terminology can be found in this GDC scientific report<sup>3</sup>.  Additional information on how the clinical data relates to overall GDC Data Model can be found on the GDC Website<sup>4</sup>.

### Data ###

In the GDC, clinical data is searchable in the API or Data Portal.  This only includes data that has been indexed and aligns with the GDC Data Dictionary<sup>1,2</sup>.  Additional clinical data may be stored in clinical supplement files in different formats depending on the project.  For example, this may include XML or biotab for TCGA or xlsx for TARGET.

## References ##

1. [GDC Clinical Data Dictionary Entries](/Data_Dictionary/viewer/#?view=table-entity-list&anchor=clinical)
2. [GDC Clinical Data Harmonization](https://gdc.cancer.gov/about-data/data-harmonization-and-generation/clinical-data-harmonization)
3. [Selecting Common Cross-Study Clinical Data Elements](https://gdc.cancer.gov/node/777/)
4. [GDC Data Model](https://gdc.cancer.gov/developers/gdc-data-model)

## External Links ##

* [CDE Browser](https://cadsr.cancer.gov/onedata/Home.jsp)
* [Center for Biomedical Informatics & Information Technology](https://datascience.cancer.gov/)

Categories: Data Category


# ASCAT3

## Description ##

ASCAT3 is an advanced copy number variation (CNV) analysis pipeline used in GDC genotyping array harmonization.

## Overview ##

ASCAT3 improves upon previous versions to provide more accurate CNV detection in tumor and normal samples from genotyping arrays. It processes the raw array files to generate high-resolution segmented CNV data.

### Input

* Tumor CEL - Genotyping Array
* Normal CEL - Genotyping Array

### Output

* TXT (Data Type: Allele-specific Copy Number Segment) 
* TSV (Data Type: Gene Level Copy Number)

## References ##

1. [Copy Number Variation Analysis Pipeline](/Data/Bioinformatics_Pipelines/CNV_Pipeline/)
1. [ASCAT Pipelines](/Data/Bioinformatics_Pipelines/CNV_Pipeline/#ascat-pipelines)

## External Links ##

* [Vanloo lab](https://github.com/VanLoo-lab/ascat/tree/master/ReleasedData/TCGA_SNP6_hg38)
* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type

# GDC Data Portal #
## Description ##
The GDC Data Portal<sup>1</sup> is a robust data-driven platform that allows users to search and download harmonized cancer data for analysis using modern web technologies.
## Overview ##
The GDC Data Portal provides users with web-based access to harmonized data from cancer genomics studies. Harmonized data includes standardized biospecimen and clinical data, DNA and RNA sequence data that has been aligned to the latest reference genome build (GRCh38), and derived data generated through GDC pipelines<sup>2</sup>.

Key GDC Data Portal features include:

* Open, granular access to information about all harmonized datasets available in the GDC
* Advanced search and visualization-assisted filtering of data files
* Cart for collecting data files of interest
* Authentication using eRA Commons credentials for access to controlled data files
* Secure data download directly from the cart or using the GDC Data Transfer Tool

## References ##
1. [GDC Data Portal](https://portal.gdc.cancer.gov)
2. [GDC Data Portal User's Guide](/Data_Portal/Users_Guide/quick_start)

## External Links ##
* N/A

Categories: Tool


# Aliquot Ensemble Somatic Variant Merging and Masking

## Description ##

Aliquot Ensemble Somatic Variant Merging and Masking is an aggregation pipeline used in GDC whole exome sequencing (WXS) and targeted sequencing harmonization.

## Overview ##

The Aliquot Ensemble Somatic Variant Merging and Masking workflow in the GDC harmonizes whole exome sequencing (WXS) and targeted sequencing data by merging multiple MAFs that were each generated by one variant caller. This workflow consists of Somatic Aggregation and Masked Somatic Aggregation, which respectively generate comprehensive mutation annotation files (MAFs) and produce filtered, public-access MAFs by removing potentially identifiable germline mutation information.

### Input

* Tumor MAF - WXS or Targeted Sequencing
* Normal MAF - WXS or Targeted Sequencing

### Output

* MAF (Data Type: Masked Somatic Mutation, Aggregated Somatic Mutation)

## References ##

1. [DNA Seq Processing at the GDC](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/)
1. [Somatic Aggregation Workflow](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/#somatic-aggregation-workflow)
1. [Masked Somatic Aggregation Workflow ](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/#masked-somatic-aggregation-workflow)

## External Links ##

* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type

# Genome Data Analysis Network (GDAN) #
## Description ##
The Genomic Data Analysis Network (GDAN) is a network of individuals and institutions that produce the bulk of the analysis required to interpret the data generated by the Genomic Characterization Centers (GCCs), that is housed within the GDC.

## Overview ##
The use of novel technologies, the need to integrate different data types and the immense quantity of data generated by The Cancer Genome Atlas (TCGA) Research Network led to an expansion of the TCGA Research Network to include new centers devoted to data analysis. The Genome Data Analysis Network works hand-in-hand with the Genome Characterization Centers (GCCs) to develop state-of-the-art tools that assist researchers with processing and integrating data analyses across the entire genome.

There are three types of centers in the GDAN, each designed to contribute to a different facet of genomic analysis: Processing, Specialized, and Visualization.

__Processing:__

One processing center, the Broad Institute of MIT and Harvard, does all of the initial processing of the data. After receiving sequencing data from Genome Characterization Centers that have been aligned by the Genomic Data Commons (GDC), the processing center runs an automated analysis of the data using Firehose, a genomic data analysis pipeline. This first-pass produces standard displays of the data, such as DNA mutation plots and maps of changes in the number of copies of genes in the genome, called copy number alterations.

__Specialized:__

The network's eleven specialized centers are comprised of foremost experts in their fields who work with the processed data to illuminate particular patterns in the data, and integrate the data, forming a more complete picture.

__Visualization:__

The network's single visualization center, led by first-time grant recipient Jingchun Zhu, Ph.D., of University of California Santa Cruz, will enable the research community to interact with and manipulate the CCG data. This group developed UCSC Xena, an interactive interface for visualizing genomics data, and will continue to refine their platform as the network's visualization center.

## References ##
1. [GDAN Announcement](https://www.cancer.gov/about-nci/organization/ccg/blog/2016/new-genomic-data-analysis-network)

## External Links ##
* [Xena](https://xenabrowser.net)

Categories: General


# Affymetrix SNP 6.0 #

## Description ##
Affymetrix Genome-Wide Human SNP Array 6.0 is a commercial SNP Array product by Affymetrix containing genetic markers, including single nucleotide polymorphisms (SNPs) and probes for the detection of copy number variation.

## Overview ##
Data from the Affymetrix SNP 6.0 platform is used by the GDC to produce harmonized Copy Number Variation files<sup>2</sup>. Hybridization levels associated with Affymetrix SNP 6.0 probes are used to determine discrete chromosomal regions that show signs of duplication or deletion. Comprehensive documentation of the GDC Copy Number Variation pipeline can be found at the GDC Documentation Website<sup>1</sup>. A list of all probes and the chromosomal regions that associate with them (with GRCh38 coordinates) can be found in the GDC Reference File webpage<sup>2</sup>.

## References ##
1. [GDC Copy Number Variation Documentation](/Data/Bioinformatics_Pipelines/CNV_Pipeline/)
2. [GDC Reference Files](https://gdc.cancer.gov/about-data/data-harmonization-and-generation/gdc-reference-files)

## External Links ##
* [Affymetrix Genome Wide Human SNP 6.0 Array](https://www.affymetrix.com/support/downloads/manuals/genomewidesnp6_manual.pdf)


Categories: Platform


# Cancer Genomics Hub (CGHub) #
## Description ##
The Cancer Genomics Hub (CGHub) was the repository for genomic data for several NCI studies prior to the launch of the GDC.
## Overview ##
The Cancer Genomics Hub was established in August 2011 to provide a repository to The Cancer Genome Atlas (TCGA), the childhood cancer initiative Therapeutically Applicable Research to Generate Effective Treatments (TARGET) and the Cancer Genome Characterization Initiative (CGCI).

The Cancer Genomics Hub (CGHub) was managed by the University of California Santa Cruz and is no longer supported. This data is now migrated to and managed by the Genomic Data Commons (GDC).

## References ##
1. N/A

## External Links ##
* N/A

Categories: General


# Arriba

## Description ##

The Arriba workflow is part of the GDC mRNA quantification analysis pipeline used for detecting gene fusions in RNA-Seq data. 

## Overview ##

Arriba is one of the two pipelines the GDC uses to detect gene fusions. The Arriba gene fusion pipeline detects gene fusions from the RNA-Seq data of tumor samples.

### Input

* Tumor BAM - RNA-Seq
* Normal BAM - RNA-Seq

### Output

* TSV (Data Type: Transcript Fusion)
* BEDPE (Data Type: Transcript Fusion)

## References ##

1. [Arriba Fusion Pipeline](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/#arriba-fusion-pipeline)
1. [mRNA Analysis Pipeline](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/)

## External Links ##

* [Arriba gene fusion pipeline](https://github.com/suhrig/arriba)
* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type

# FPKM #
## Description ##

Fragments Per Kilobase of transcript per Million mapped reads (FPKM) is a simple expression level normalization method. The FPKM normalizes read count based on gene length and the total number of mapped reads.  

## Overview ##

FPKM is implemented at the GDC on gene-level read counts that are produced by STAR<sup>1</sup> and generated using custom scripts<sup>2</sup>. The formula used to generate FPKM values is as follows:

FPKM = [RM<sub>g</sub> * 10<sup>9</sup> ] / [RM<sub>t</sub> * L]

* RM<sub>g</sub>: The number of reads mapped to the gene
* RM<sub>t</sub>: The total number of read mapped to protein-coding sequences in the alignment
* L: The length of the gene in base pairs

The scalar (10<sup>9</sup>) is added to normalize the data to "__kilo__ base" and "__million__ mapped reads."

FPKM files are available as tab delimited files with the Ensembl gene IDs in the first column and the expression values in the second. See [FPKM-UQ](FPKM-UQ.md) for an alternative method of gene expression level normalization.


## References ##
1. [STAR-Fusion pipeline](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/#star-fusion-pipeline)
2. [GDC mRNA-Seq Documentation](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/)


## External Links ##
* [Ensembl Human Genome](http://www.ensembl.org/Homo_sapiens/Info/Annotation)
* [GENCODE 36](https://www.gencodegenes.org/human/release_36.html)

Categories: Workflow Type


# Pindel Annotation

## Description ##

Pindel Annotation is a somatic mutation annotation pipeline used in GDC whole exome sequencing (WXS) and targeted sequencing harmonization.

## Overview ##

Pindel Annotation is a pipeline used for WXS and targeted sequencing somatic mutation annotation at the GDC. Somatic mutation annotation is performed by annotating raw VCFs, which produces annotated somatic mutation files.

### Input

* Raw VCF - WXS or Targeted Sequencing

### Output

* MAF (Data Type: Annotated Somatic Mutation)
* VCF (Data Type: Annotated Somatic Mutation)

## References ##

1. [Pindel Command Line Parameters at the GDC](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/#pindel)
1. [DNA Seq Processing at the GDC](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/)

## External Links ##
* [Pindel at Washingotn University in St.Louis](https://gmt.genome.wustl.edu/packages/pindel/)
* [Overview of GDC Harmonization Workflows](https://github.com/NCI-GDC/gdc-workflow-overview/blob/master/README.md)
* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type

# FM Simple Somatic Mutation

## Description ##

The FM (Foundation Medicine) Simple Somatic Mutation (SSM) workflow is a genomic profile harmonization workflow in the GDC for identifying simple somatic mutations in targeted sequencing data.

## Overview ##

FM Simple Somatic Mutation is the pipeline used for genomic profile harmonization for the FM-AD project at the GDC. Genomic profile harmonization is performed by lifting over simple nucleotide variation data from uploaded genomic profile data.

### Input

* Submitted genomic profiles

### Output

* VCF (Data Type: Raw Simple Somatic Mutation)

## References ##

1. [DNA-Seq Analysis Pipeline](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/)

## External Links ##

* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type

# VCF LiftOver

## Description ##

VCF LiftOver is a pipeline used in GDC whole genome sequencing (WGS) harmonization.

## Overview ##

VCF LiftOver is a pipeline used for harmonization of genomic profiling reports and generates combined nucleotide variation data.

### Input

* Submitted genomic profiles

### Output

* VCF (Data Type: Raw CGI Variant)

## External Links ##

* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type

# Clinical Supplement #

## Description ##

The purpose of a clinical supplement file is to make available clinical information that does not fit into the GDC Data Dictionary.

## Overview ##

The GDC Data Dictionary has an extensive set of clinical elements<sup>1</sup>. However, due to the number and variety of submitting projects the GDC is not able to accommodate all possible clinical elements into the GDC Data Dictionary. The clinical supplement offers a way for submitters to include all existing clinical data even though it may not conform to the GDC Data Dictionary. Note that the content of these files is not searchable via the API, does not fit the GDC Data Dictionary, and may use different CDEs and vocabulary than other projects stored in the GDC.  

For example, whether or not a patient has mutations in the ER, PR and HER2 genes is very important when researching breast cancer. This information would not be very useful in most other cancers, so it can be found in the clinical supplement files for breast cancer patients rather than being incorporated into the data model across all cancers.

While some elements are unique to certain cancers, there is no guarantee that all patients associated with these cancers will have all elements populated.

### Data Formats ###

Clinical Supplement files are available in the GDC Data Portal as case-level XML files. Most elements have an associated CDE-ID that can be queried at the CDE Browser website for additional information<sup>2</sup>.  

## References ##
1. [GDC Clinical Data Elements](https://gdc.cancer.gov/about-data/data-harmonization-and-generation/clinical-data-harmonization)
2. [CDE Browser](https://cdebrowser.nci.nih.gov/cdebrowserClient/cdeBrowser.html#/search)

## External Links ##
* N/A

Categories: Data Type


# Redaction #
## Description ##

A redaction is the removal of data from the GDC at a submitter's request.

## Overview ##

Because the GDC aims to make data available consistently, redactions are generally rare and performed only in specific instances<sup>1</sup>. For example, cases must be redacted when the subject withdraws consent, the TSS/BCR subject link is incorrect ("unknown patient identity"), or in the cases of genotype mismatch (i.e. when a sample is shown to be from the wrong cancer, organ, or tissue)<sup>1</sup>. Redacted cases and their associated files are not displayed in the GDC Data Portal. More generally, a redaction hides the redacted entity and all child entities. Any questions about the redaction status of a case can be directed to GDC User Support<sup>2</sup>.   

## References ##

1. [GDC Policies](https://gdc.cancer.gov/about-gdc/gdc-policies)
2. [GDC Support](https://gdc.cancer.gov/support)

## External Links ##
* N/A

Categories: General


# Database of Genotypes and Phenotypes (dbGaP) #
## Description ##
The Database of Genotypes and Phenotypes (dbGaP) is the National Institutes of Health (NIH) sponsored repository charged to archive, curate and distribute information produced by studies investigating the interaction of genotype and phenotype.
## Overview ##
The dbGaP project serves as an access gateway for researchers seeking to gain access to genotype and phenotype data. The GDC stores and distributes data that is considered "controlled access", meaning it contains some genotype information.  Because of this researchers need to apply for access with dbGaP to gain access to projects<sup>1</sup>. An eRA Commons or NIH iTrust account is needed to authenticate the user. The same process is true for data submission when it contains genotype data. A new project is created in dbGaP and access to granted to their eRA Commons ID.        
## References ##
1. [Obtaining Access to Controlled Data](https://gdc.cancer.gov/access-data/obtaining-access-controlled-data) 

## External Links ##
* [dbGaP Home Page](https://www.ncbi.nlm.nih.gov/gap)
* [dbGaP Authorized Access](https://dbgap.ncbi.nlm.nih.gov/aa/wga.cgi?page=login)

Categories: General


# Harmonized Data #
## Description ##

Harmonized data refers to the collection of raw data from multiple sources that has been normalized so that a valid comparison can be made across these sources.

## Overview ##

Data harmonization is one of the underlying principles on which the GDC is based. Genomic data is typically collected, processed, and analyzed on a project-level basis by many different groups. Even the most similar projects cannot always be compared in a valid way due to small differences across data processing and analysis pipelines. The GDC collects raw data from many cancer projects and processes them using standardized pipelines<sup>1</sup> and the reference genome GRCh38<sup>2</sup>. This gives the advantage of analyzing multiple cancer types or the same cancer type across multiple projects.

GDC Data is harmonized using carefully curated bioinformatics pipelines and produces somatic variant call, gene expression, copy number variation estimation, and methylation data. Clinical and biospecimen data are also harmonized by making a set of elements common to all projects available for download through the API. As new projects are submitted to the GDC, incoming data is reviewed by a team of bioinformaticians who determine how to proceed with harmonization based on the data type, quality, and available computational resources.  

## References ##
1. [GDC Data Harmonization](https://gdc.cancer.gov/about-data/gdc-data-harmonization)
2. [GDC Reference Genome](https://gdc.cancer.gov/about-data/data-harmonization-and-generation/gdc-reference-files)

## External Links ##
* N/A

Categories: General


# STAR - Counts

## Description ##

STAR - Counts is an RNA expression pipeline used in GDC RNA-Seq harmonization.

## Overview ##

STAR - Counts is a pipeline used for to quantify gene expression from RNA-Seq data. Quantification is performed with STAR using tumor or normal alignments and generates transcriptome profiling data.

### Input

* Genomic BAM - RNA-Seq

### Output

* TSV (Data Type: Gene Expression Quantification, Splice Junction Quantification)

[![STAR Pipeline Diagram](https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/images/RNA-Seq-DR32_Image.png)](https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/images/RNA-Seq-DR32_Image.png "Click to see the full image.")

## References ##

1. [mRNA Analysis Pipeline](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/)
1. [mRNA Expression Workflow](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/#mrna-expression-workflow)
1. [mRNA Quantification Command Line Parameters](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/#mrna-quantification-command-line-parameters)

## External Links ##

* [STAR](https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf)
* [Overview of GDC Harmonization Workflows](https://github.com/NCI-GDC/gdc-workflow-overview/blob/master/README.md)
* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type

# VarScan2 Annotation

## Description ##

VarScan2 Annotation is a somatic mutation annotation pipeline in GDC whole exome sequencing (WXS) and targeted sequencing harmonization. 

## Overview ##

VarScan2 Annotation is a workflow in the GDC that annotates somatic variants identified by the VarScan2 variant calling pipeline.

### Input

* VCF (Data Type: Raw Simple Somatic Mutation)

### Output

* MAF (Data Type: Annotated Somatic Mutation)
* VCF (Data Type: Annotated Somatic Mutation)

## References ##

1. [DNA-Seq Analysis Pipeline](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/)
1. [Variant Call Annotation Workflow](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/#variant-call-annotation-workflow)

## External Links ##

* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type

# MuTect2

## Description ##

MuTect2 is a somatic variant calling pipeline used in GDC whole exome sequencing (WXS) and targeted sequencing harmonization.

## Overview ##

MuTect2 is one of the four pipelines used for WXS and targeted sequencing somatic variant calling at the GDC. Somatic variant calling is performed with MuTect2 using tumor and normal alignments and generates single-nucleotide polymorphism (SNP) data.

### Input

* Tumor BAM - WXS or Targeted Sequencing
* Normal BAM - WXS or Targeted Sequencing

### Output

* VCF (Data Type: Raw Simple Somatic Mutation)


## References ##
1. [MuTect2 Command Line Parameters at the GDC](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/#mutect2)
1. [DNA Seq Processing at the GDC](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/)


## External Links ##
* [MuTect2 at the Broad](https://gatk.broadinstitute.org/hc/en-us/articles/360037593851-Mutect2)
* [Overview of GDC Harmonization Workflows](https://github.com/NCI-GDC/gdc-workflow-overview/blob/master/README.md)
* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type


# Latest Data #

## Description ##

The latest data refers to the current collection of files and metadata available at the GDC.       

## Overview ##

A large-scale data commons that provides processed data files will inevitably start to have newer versions of data files due to pipeline improvements, data redactions, and other modifications that can happen during data curation. The default setting for the GDC is to provide the latest version of the data<sup>1</sup>. In some cases older versions of files will be made available upon request, but will not necessarily be discoverable. All versions of files from cases that have been redacted will be made completely unavailable<sup>2</sup>.   

## References ##
1. [GDC Portal](https://portal.gdc.cancer.gov/)
2. [GDC Data Policies](https://gdc.cancer.gov/about-gdc/gdc-policies)

## External Links ##
* N/A

Categories: General


# GDC Data Transfer Tool (DTT) #
## Description ##
The GDC Data Transfer Tool (DTT)<sup>1</sup> provides a standard client-based mechanism in support of high performance data downloads and submission.
## Overview ##
The GDC DTT provides an optimized method of transferring data to and from the GDC, and enables resumption of interrupted transfers. The GDC DTT is the preferred option for downloading or submitting large or numerous files.

Key GDC DTT features include<sup>2</sup>:

* Command-line interface to specify the desired data transfer protocol
* Utilization of a manifest file generated from the GDC Data Portal and GDC Data Submission Portal for multiple downloads and uploads, respectively
* Supports the secure transfer of controlled access data using an authentication key (token)

## References ##
1. [GDC Data Transfer Tool](https://gdc.cancer.gov/access-data/gdc-data-transfer-tool)
2. [GDC Data Transfer Tool User's Guide](/Data_Transfer_Tool/Users_Guide/Getting_Started/)

## External Links ##
* N/A

Categories: Tool


# Release Number #
## Description ##
The release number defines the latest revision of different parts of the GDC.
## Overview ##
The release number keeps track of new releases of the GDC.  The different parts of the GDC that may be updated and their release information (which contains the release number) are summarized below:

* __API__: [Release Notes](/API/Release_Notes/API_Release_Notes/#api-release-notes)
* __Data Portal__: [Release Notes](/Data_Portal/Release_Notes/Data_Portal_Release_Notes/)
* __Data Submission Portal__: [Release Notes](/Data_Submission_Portal/Release_Notes/Data_Submission_Portal_Release_Notes/)
* __Data Transfer Tool__: [Release Notes](/Data_Transfer_Tool/Release_Notes/DTT_Release_Notes/)
* __Data__: [Release Notes](/Data/Release_Notes/Data_Release_Notes/)

## References ##
N/A

## External Links ##
N/A

Categories: General


# Biospecimen Data #

## Description ##
Biospecimen data includes information on how a patient's tissue was processed and subsampled for use in histology and molecular assays. This data can be represented in a hierarchical relationship.

## Overview ##

In the GDC, biospecimen elements refer to biological specimens that originate from cancer patients. For example, a `sample` element (entity) comes directly from a patient and a `portion` that was used to generate a diagnostic slide comes from the `sample`. Each entity has associated attributes that can be used to describe it. For example, an `analyte` entity has the `analyte_type` attribute, which reports the type of biological molecule that was extracted to produce the `analyte`.

Data files hosted at the GDC will be associated with different biospecimen entities depending on their type. For example, clinical data files will be directly associated with a case entity (a patient) because these values are associated with the patient as a whole. The data used to produce an RNA-Seq alignment originates from sequenced RNA extracts and would be associated with an aliquot entity. See the GDC Data Model<sup>1</sup> and Data Dictionary<sup>2</sup> for details on the biospecimen entity types and their properties.

The TCGA program, from which many of the GDC attributes were inherited, includes biospecimen elements that are reflected in the structure of the TCGA Barcode<sup>3</sup>.

### Data ###

Biospecimen data can be downloaded from the GDC in several formats:

* __API Retrieval:__ Specific information about each biospecimen entity can be queried from the API in a tab-delimited or JSON format. This information can be retrieved programatically.
* __Biospecimen Supplements:__ Biospecimen supplements are stored in XML format and can be downloaded from the GDC Portal. Supplements may contain biospecimen fields that do not appear in the API as well as those that do. These can include fields that only apply to certain projects or have not yet been incorporated into the GDC Data Dictionary.
* __Biotab Files:__ Biotab files specific supplemental files that are available in the GDC Data Portal as tab-delimited files on a project-level basis. These may also include fields that are not available in the GDC API. Biotab files can be found under the "Data Format: bcr biotab" filter.

Additionally, the biospecimen data that is available from the API is displayed on the GDC Portal in the summary page for each case.

## References ##
1. [GDC Data Dictionary](/Data_Dictionary/viewer/)
2. [GDC Data Model](https://gdc.cancer.gov/developers/gdc-data-model/gdc-data-model-components)
3. [TCGA Barcode](images/TCGA-TCGAbarcode-080518-1750-4378.pdf)

## External Links ##
* [GDC Biospecimen Harmonization](https://gdc.cancer.gov/about-data/data-harmonization-and-generation/biospecimen-data-harmonization)

Categories: Data Category


# GATK4 MuTect2 Tumor-Only Annotation

## Description ##

GATK4 MuTect2 Tumor-Only Annotation is a workflow used in GDC whole exome sequencing (WXS) and targeted sequencing harmonization.


## Overview ##

GATK4 MuTect2 Tumor-Only Annotation is a workflow utilized in the GDC for annotating somatic variants identified in tumor-only whole exome sequencing (WXS) and targeted sequencing data. 

### Input

* Tumor VCF - WXS or Targeted Sequencing

### Output

* VCF (Data Type: Annotated Somatic Mutation)
* MAF (Data Type: Annotated Somatic Mutation)

## References ##

1. [DNA-Seq Analysis Pipeline](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/)
1. [Tumor-Only Variant Annotation Workflow](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/#tumor-only-variant-annotation-workflow)

## External Links ##

* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type

# Analyte #
## Description ##
An analyte is any substance or sample being analyzed.

## Overview ##
An analyte is the specimen extracted for analysis from a portion or sample using a specific extraction protocol.
The [GDC Data Model](https://gdc.cancer.gov/developers/gdc-data-model/gdc-data-model-components) relates analytes to aliquots or portions or samples but is not a required biospecimen entity.  

## References ##
1. [GDC Data Dictionary - Analyte](/Data_Dictionary/viewer/#?view=table-definition-view&id=analyte)

## External Links ##
* N/A

Categories: General, Biospecimen


# FoundationOne Annotation

## Description ##

FoundationOne Annotation is a somatic mutation annotation workflow used in GDC targeted sequencing harmonization.

## Overview ##

FoundationOne Annotation is a pipeline used to annotate the raw somatic mutation data from the FM-AD project at the GDC.

### Input

* Tumor VCF - Targeted Sequencing

### Output

* VCF (Data Type: Annotated Somatic Mutation)

## External Links ##

* [Foundation Medicine](https://www.foundationmedicine.com/)
* [Foundation Medicine at the GDC](https://gdc.cancer.gov/about-gdc/contributed-genomic-data-cancer-research/foundation-medicine)
* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type

# STAR 2-Pass Chimeric

## Description ##

STAR 2-Pass Chimeric is an alignment pipeline used in GDC RNA-Seq harmonization.

## Overview ##

STAR 2-Pass Chimeric is a pipeline used for RNA-Seq reads alignment at the GDC. Alignment is performed with STAR and generates sequencing reads data.

### Input

* Submitted sequencing reads

### Output

* BAM (Data Type: Aligned Reads)

[![STAR Pipeline Diagram](https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/images/RNA-Seq-DR32_Image.png)](https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/images/RNA-Seq-DR32_Image.png "Click to see the full image.")

## References ##

1. [mRNA Analysis Pipeline](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/)
1. [RNA-Seq Alignment Workflow](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/#rna-seq-alignment-workflow)
1. [RNA-Seq Alignment Command Line Parameters](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/#rna-seq-alignment-command-line-parameters)

## External Links ##

* [STAR](https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf)
* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type

# Mutation Annotation Format (MAF) #
## Description ##
Mutation Annotation Format (MAF) files are tab-delimited files that contain somatic and/or germline mutation annotations. MAF files containing any germline mutation annotations are protected and distributed in the controlled access portion of the GDC Data Portal. MAF files containing only somatic mutations are publicly available.

## Overview ##
Variants are discovered by aligning DNA sequences derived from tumor samples and sequences derived from normal samples to a reference sequence<sup>1</sup>. A MAF file identifies, for all samples in a project, the discovered putative or validated mutations and categorizes those mutations (polymorphism, deletion, or insertion) as somatic (originating in the tumor tissue) or germline (originating from the germline). Mutation annotation are also reported. Note that VCF files report on all transcripts affected by a mutation while MAF files only report on the most affected one.  

MAF files are generated at the GDC using the Variant Aggregation pipeline<sup>2</sup> by aggregating case-level VCF files on a project and pipeline level (one protected MAF per project per pipeline). Protected MAF files are further processed into open-access somatic MAF files by removing low quality or germline mutations.

### Structure ###
The structure of the MAF is available in the [GDC MAF Specification](https://docs.gdc.cancer.gov/Data/File_Formats/MAF_Format/) along with descriptions for each field<sup>3</sup>.

## References ##
1. [GDC DNA-Seq Analysis](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/)
2. [GDC-Dictionary: Somatic Aggregation Pipeline](/Data_Dictionary/viewer/#?view=table-definition-view&id=somatic_aggregation_workflow)



Categories: Data Format


# MuSE

## Description ##

MuSE is a somatic variant calling pipeline used in GDC whole exome sequencing (WXS) and targeted sequencing harmonization.

## Overview ##

MuSE is one of the four pipelines used for WXS and targeted sequencing somatic variant calling at the GDC. Somatic variant calling is performed with MuSE using tumor and normal alignments and generates single-nucleotide polymorphism (SNP) data.

### Input
* Tumor BAM - WXS or Targeted Sequencing
* Normal BAM - WXS or Targeted Sequencing

### Output

* VCF (Data Type: Raw Simple Somatic Mutation)

[![MuSE Pipeline Diagram](https://gdc.cancer.gov/files/public/image/muse-somatic-variant-calling-pipeline.png)](https://gdc.cancer.gov/files/public/image/muse-somatic-variant-calling-pipeline.png "Click to see the full image.")

## References ##

1. [MuSE Command Line Parameters at the GDC](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/#muse)
1. [DNA Seq Processing at the GDC](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/)

## External Links ##

* [MuSE at MDACC](https://bioinformatics.mdanderson.org/public-software/muse/)
* [Overview of GDC Harmonization Workflows](https://github.com/NCI-GDC/gdc-workflow-overview/blob/master/README.md)
* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type

Mutation Annotation Format (MAF) - Legacy TCGA Specification
==============================================


*This definition was taken from the previously public wiki hosted by TCGA and reflects the MAF format
that was available during the active period of the TCGA project.*  




**Document Information**

The spec has been reverted to the June 26th version (version 20). Additional
changes are the removal of the "under construction" banner, changing all text to
black, and fixing a typo in the link to the MAF 2.2 specification.

**Specification for Mutation Annotation Format**  
Version 2.4.1  
June 20, 2014

**Contents**

-   1 Current version changes

-   2 About MAF specifications

    -   2.1 Definition of open access MAF
        data

    -   2.2 Somatic MAF vs. Protected
        MAF

-   3 MAF file fields

    -   3.1 Table 1 - File column
        headers

-   4 MAF file checks

-   5 MAF naming convention

-   6 Previous specification
    versions

Current version changes
=======================

This current revision is **version 2.4.1** of the Mutation Annotation Format
(MAF) specification.

The following items in the specification were added or modified in version 2.4.1
from version 2.4:

-   Header for MAF file is "\#version 2.4.1"

-   "Somatic" and "None" are the only acceptable values for "Mutation_Status"
    for a somatic.MAF (named .somatic.maf). When Mutation_Status is None,
    Validation_Status must be Invalid.

-   Centers need to make sure that Mutations_Status "None" doesn't include
    germline mutation.

-   For a somatic MAF, following rules should be satisfied:  
    SOMATIC = (A AND (B OR C OR D)) OR (E AND F)  
    A: *Mutation_Status* == "Somatic"  
    B: *Validation_Status* == "Valid"  
    C. *Verification_Status* == "Verified"  
    D. *Variant_Classification* is not {Intron, 5'UTR, 3'UTR, 5'Flank, 3'Flank,
    IGR}, which implies that *Variant_Classification* can only be
    \\{Frame_Shift_Del, Frame_Shift_Ins, In_Frame_Del, In_Frame_Ins,
    Missense_Mutation, Nonsense_Mutation, Silent, Splice_Site,
    Translation_Start_Site, Nonstop_Mutation, RNA, Targeted_Region}.  
    E: *Mutations_status == "None"*  
    F: *Validation_status == "Invalid"*

-   Extra validation rules: If Validation_Status == Valid or Invalid, then
    Validation_Method != none (case insensitive).

About MAF specifications
========================

Mutation annotation files should be transferred to the DCC. Those files should
be formatted using the mutation annotation format (MAF) that is described below.
File naming convention is also
[below](#MutationAnnotationFormat(MAF)Specificat).

Following categories of somatic mutations are reported in MAF files:

-   Missense and nonsense

-   Splice site, defined as SNP within 2 bp of the splice junction

-   Silent mutations

-   Indels that overlap the coding region or splice site of a gene or the
    targeted region of a genetic element of interest

-   Frameshift mutations

-   Mutations in regulatory regions

### Definition of open access MAF data

A large proportion of MAFs are submitted as discovery data and sites labeled as
somatic in these files overlap with known germline variants. In order to
minimize germline contamination in putative (unvalidated) somatic calls, certain
filtering criteria have been imposed. Based on current policy, open access MAF
data should:

-   **include** all validated somatic mutation calls

-   **include** all unvalidated somatic mutation calls that overlap with a
    coding region or splice site

-   **exclude** all other types of mutation calls (i.e., non-somatic calls
    (validated or not), unvalidated somatic calls that are not in coding region
    or splice sites, and dbSNP sites that are not annotated as somatic in dbSNP,
    COSMIC or OMIM)



### Somatic MAF vs. Protected MAF

Centers will submit to the DCC MAF archives that contain Somatic MAF
(named**.somatic.maf**) for open access data and an all-inclusive Protected MAF
(named**.protected.maf**) that does not filter any data out and represents the
original super-set of mutation calls. The files will be formatted using the
Mutation Annotation Format (MAF).

The following table lists some of the critical attributes of somatic and
protected MAF files and provides a comparison.

| Attribute       | Somatic MAF | Protected MAF |
| -----------     | ----------- | ------------- |
| **File naming** | Somatic MAFs should be named as**\*.somatic.maf**and cannot contain 'germ' or 'protected' in file name. | Protected MAFs should be named as**\*.protected.maf**and should not contain 'somatic' in the file name. |
| **Mutation category**  | Somatic MAFs can only contain entries where*Mutation_Status*is "Somatic". If any other value is assigned to the field, the archive will fail. Experimentally validated or unvalidated (see next row) somatic mutations can be included in the file. | There is no such restriction for protected MAF. The file should contain all mutation calls including those from which .somatic.maf is derived. |
| **Filtering criteria** | In order to minimize germline contamination, somatic MAFs can contain unvalidated somatic mutations only from coding regions and splice sites, which implies: | There are no such constraints for mutations in protected MAF. |
|                        | If *Validation_Status* **is**"Unknown",*V a riant_Classification* **cannot** be 3'UTR, 3'Flank, 5'UTR, 5'Flank, IGR, or Intron.*Variant_Classification*can only be \\{Frame_Shift_Del, Frame_Shift_Ins, In_Frame_Del, In_Frame_Ins, Missense_Mutation, Nonsense_Mutation, Silent, Splice_Site, Translation_Start_Site, Nonstop_Mutation, RNA, Targeted_Region, De_novo_Start_InFrame, De_novo_Start_OutOfFrame\\}.                                                  |                                                                                                                                                |
|                        | There is no such constraint for experimentally validated (*Validation_Status*is "Valid") somatic mutations.                                                                  |                                                                                                                                                |
|                        |                                                                                |                                                                                                                                                |
|   | dbSNP sites that are not annotated as somatic in dbSNP, COSMIC or OMIM must be removed from somatic MAFs.                                                                         |                                                                                                                                                |
| **Access level**       | These files are deployed as open access data.    | These files are deployed as protected data.                                                                                                    |

MAF file fields
===============

The format of a MAF file is tab-delimited columns. Those columns are described
in Table 1 and are required in every MAF file. The order of the columns will be
validated by the DCC. Column headers and values **are** case sensitive where
specified. Columns may allow null values (i.e.\_ blank cells) and/or have
enumerated values. **The validator looks for a header stating the version of the
specification to validate against (e.g. \#version 2.4). If not, validation
fails.** Any columns that come after the columns described in Table 1 are
optional. Optional columns are not validated by the DCC and can be in any order.



Table 1 - File column headers
-----------------------------



| **Index** | **MAF Column Header** | **Description of Values** | **Example** | **Case Sensitive** | **Null** | **Enumerated** |                                    
| --------- | --------------------- | ------------------------- | ----------- | ------------------ | -------- | -------------- |
| 1         | Hugo_Symbol           | HUGO symbol for the gene (HUGO symbols are *always* in all caps). If no gene exists within 3kb enter "Unknown".  |EGFR                                           | Yes                   | No                                                                                                                                  | Set or Unknown                                                                                                                                                                                                                          |                                      |   |   |   |   |   |   |   |   |
|              |                               | Source: <http://genenames.org>                                                                                                                                                                                                                                                                                         |                                                |                       |                                                                                                                                     |                                                                                                                                                                                                                                         |                                      |   |   |   |   |   |   |   |   |
| 2            | Entrez_Gene_Id                | Entrez gene ID (an integer). If no gene exists within 3kb enter "0".                                                                                                                                                                                                                                                   | 1956                                           | No                    | No                                                                                                                                  | Set                                                                                                                                                                                                                                     |                                      |   |   |   |   |   |   |   |   |
|              |                               | Source: <http://ncbi.nlm.nih.gov/sites/entrez?db=gene>                                                                                                                                                                                                                                                                 |                                                |                       |                                                                                                                                     |                                                                                                                                                                                                                                         |                                      |   |   |   |   |   |   |   |   |
| 3            | Center                        | Genome sequencing center reporting the variant. If multiple institutions report the same mutation separate list using semicolons. Non-GSC centers will be also supported if center name is an accepted center name.                                                                                                    | hgsc.bcm.edu;genome.wustl.edu                  | Yes                   | No                                                                                                                                  | Set                                                                                                                                                                                                                                     |                                      |   |   |   |   |   |   |   |   |
| 4            | NCBI_Build                    | Any TGCA accepted genome identifier. Can be string, integer or a float.                                                                                                                                                                                                                                               | hg18, hg19, GRCh37, GRCh37-lite, 36, 36.1, 37, | No                    | No                                                                                                                                  | Set and Enumerated.                                                                                                                                                                                                                     |                                      |   |   |   |   |   |   |   |   |
| 5            | Chromosome                    | Chromosome number without "chr" prefix that contains the gene.                                                                                                                                                                                                                                                         | X, Y, M, 1, 2, etc.                            | Yes                   | No                                                                                                                                  | Set                                                                                                                                                                                                                                     |                                      |   |   |   |   |   |   |   |   |
| 6            | Start_Position                | Lowest numeric position of the reported variant on the genomic reference sequence. Mutation start coordinate (1-based coordinate system).                                                                                                                                                                              | 999                                            | No                    | No                                                                                                                                  | Set                                                                                                                                                                                                                                     |                                      |   |   |   |   |   |   |   |   |
| 7            | End_Position                  | Highest numeric genomic position of the reported variant on the genomic reference sequence. Mutation end coordinate (inclusive, 1-based coordinate system).                                                                                                                                                            | 1000                                           | No                    | No                                                                                                                                  | Set                                                                                                                                                                                                                                     |                                      |   |   |   |   |   |   |   |   |
| 8            | Strand                        | Genomic strand of the reported allele. Variants should always be reported on the positive genomic strand. (Currently, only the positive strand is an accepted value).                                                                                                                                                  | \+                                             | No                    | No                                                                                                                                  | \+                                                                                                                                                                                                                                      |                                      |   |   |   |   |   |   |   |   |
| 9            | Variant_Classification        | Translational effect of variant allele.                                                                                                                                                                                                                                                                                | Missense_Mutation                              | Yes                   | No                                                                                                                                  | Frame_Shift_Del, Frame_Shift_Ins, In_Frame_Del, In_Frame_Ins, Missense_Mutation, Nonsense_Mutation, Silent, Splice_Site, Translation_Start_Site, Nonstop_Mutation, 3'UTR, 3'Flank, 5'UTR, 5'Flank, IGR *(See Notes Section #1)* , Intron, RNA, Targeted_Region |                                      |   |   |   |   |   |   |   |   |
| 10           | Variant_Type                  | Type of mutation. TNP (tri-nucleotide polymorphism) is analogous to DNP but for 3 consecutive nucleotides. ONP (oligo-nucleotide polymorphism) is analogous to TNP but for consecutive runs of 4 or more.                                                                                                              | INS                                            | Yes                   | No                                                                                                                                  | SNP, DNP, TNP, ONP, INS, DEL, or Consolidated *(See Notes Section #2)* )                                                                                                                                                                                         |                                      |   |   |   |   |   |   |   |   |
| 11           | Reference_Allele              | The plus strand reference allele at this position. Include the sequence deleted for a deletion, or "-" for an insertion.                                                                                                                                                                                               | A                                              | Yes                   | No                                                                                                                                  | A,C,G,T and/or -                                                                                                                                                                                                                        |                                      |   |   |   |   |   |   |   |   |
| 12           | Tumor_Seq_Allele1             | Primary data genotype. Tumor sequencing (discovery) allele 1. " -" for a deletion represent a variant. "-" for an insertion represents wild-type allele. Novel inserted sequence for insertion should not include flanking reference bases.                                                                            | C                                              | Yes                   | No                                                                                                                                  | A,C,G,T and/or -                                                                                                                                                                                                                        |                                      |   |   |   |   |   |   |   |   |
| 13           | Tumor_Seq_Allele2             | Primary data genotype. Tumor sequencing (discovery) allele 2. " -" for a deletion represents a variant. "-" for an insertion represents wild-type allele. Novel inserted sequence for insertion should not include flanking reference bases.                                                                           | G                                              | Yes                   | No                                                                                                                                  | A,C,G,T and/or -                                                                                                                                                                                                                        |                                      |   |   |   |   |   |   |   |   |
| 14           | dbSNP_RS                      | Latest dbSNP rs ID (dbSNP_ID) or "novel" if there is no dbSNP record. source: <http://ncbi.nlm.nih.gov/projects/SNP/>                                                                                                                                                                                                  | rs12345                                        | Yes                   | Yes                                                                                                                                 | Set or "novel"                                                                                                                                                                                                                          |                                      |   |   |   |   |   |   |   |   |
| 15           | dbSNP_Val_Status              | dbSNP validation status. Semicolon- separated list of validation statuses.                                                                                                                                                                                                                                             | by2Hit2Allele;byCluster                        | No                    | Yes                                                                                                                                 | by1000genomes;by2Hit2Allele; byCluster; byFrequency; byHapMap; byOtherPop; bySubmitter; alternate_allele *(See Notes Section #3)* **Note that "none" will no longer be an acceptable value.**                                                                  |                                      |   |   |   |   |   |   |   |   |
| 16           | Tumor_Sample_Barcode          | BCR aliquot barcode for the tumor sample including the two additional fields indicating plate and well position. i.e. TCGA-SiteID-PatientID-SampleID-PortionID-PlateID-CenterID. The full TCGA Aliquot ID.                                                                                                             | TCGA-02-0021-01A-01D-0002-04                   | Yes                   | No                                                                                                                                  | Set                                                                                                                                                                                                                                     |                                      |   |   |   |   |   |   |   |   |
| 17           | Matched_Norm_Sample_Barcode   | BCR aliquot barcode for the matched normal sample including the two additional fields indicating plate and well position. i.e. TCGA-SiteID-PatientID-SampleID-PortionID-PlateID-CenterID. The full TCGA Aliquot ID; e.g. TCGA-02-0021-10A-01D-0002-04 (compare portion ID '10A' normal sample, to '01A' tumor sample). | TCGA-02-0021-10A-01D-0002-04                   | Yes                   | No                                                                                                                                  | Set                                                                                                                                                                                                                                     |                                      |   |   |   |   |   |   |   |   |
| 18           | Match_Norm_Seq_Allele1        | Primary data. Matched normal sequencing allele 1. "-" for deletions; novel inserted sequence for INS not including flanking reference bases.                                                                                                                                                                           | T                                              | Yes                   | Yes                                                                                                                                 | A,C,G,T and/or -                                                                                                                                                                                                                        |                                      |   |   |   |   |   |   |   |   |
| 19           | Match_Norm_Seq_Allele2        | Primary data. Matched normal sequencing allele 2. "-" for deletions; novel inserted sequence for INS not including flanking reference bases.                                                                                                                                                                           | ACGT                                           | Yes                   | Yes                                                                                                                                 | A,C,G,T and/or -                                                                                                                                                                                                                        |                                      |   |   |   |   |   |   |   |   |
| 20           | Tumor_Validation_Allele1      | Secondary data from orthogonal technology. Tumor genotyping (validation) for allele 1. "-" for deletions; novel inserted sequence for INS not including flanking reference bases.                                                                                                                                      | \-                                             | Yes                   | Yes                                                                                                                                 | A,C,G,T and/or -                                                                                                                                                                                                                        |                                      |   |   |   |   |   |   |   |   |
| 21           | Tumor_Validation_Allele2      | Secondary data from orthogonal technology. Tumor genotyping (validation) for allele 2. "-" for deletions; novel inserted sequence for INS not including flanking reference bases.                                                                                                                                      | A                                              | Yes                   | Yes                                                                                                                                 | A,C,G,T and/or -                                                                                                                                                                                                                        |                                      |   |   |   |   |   |   |   |   |
| 22           | Match_Norm_Validation_Allele1 | Secondary data from orthogonal technology. Matched normal genotyping (validation) for allele 1. "-" for deletions; novel inserted sequence for INS not including flanking reference bases.                                                                                                                             | C                                              | Yes                   | Yes                                                                                                                                 | A,C,G,T and/or -                                                                                                                                                                                                                        |                                      |   |   |   |   |   |   |   |   |
| 23           | Match_Norm_Validation_Allele2 | Secondary data from orthogonal technology. Matched normal genotyping (validation) for allele 2. "-" for deletions; novel inserted sequence for INS not including flanking reference bases.                                                                                                                             | G                                              | Yes                   | Yes                                                                                                                                 | A,C,G,T and/or -                                                                                                                                                                                                                        |                                      |   |   |   |   |   |   |   |   |
| 24           | Verification_Status *(See Notes Section #4)*         | Second pass results from independent attempt using same methods as primary data source. Generally reserved for 3730 Sanger Sequencing.                                                                                                                                                                                 | Verified                                       | Yes                   | Yes                                                                                                                                 | Verified, Unknown                                                                                                                                                                                                                       |                                      |   |   |   |   |   |   |   |   |
| 25           | Validation_Status *(See Notes Section #5)*           | Second pass results from orthogonal technology.                                                                                                                                                                                                                                                                        | Valid                                          | Yes                   | No                                                                                                                                  | Untested, Inconclusive, Valid, Invaild                                                                                                                                                                                                                                |                                      |   |   |   |   |   |   |   |   |
| 26           | Mutation_Status               | Updated to reflect validation or verification status and to be in agreement with the [VCF VLS](https://wiki.nci.nih.gov/x/2gcYAw) field. The values allowed in this field are constrained by the value in the Validation_Status field.                                                                                 | Somatic                                        | Yes                   | No                                                                                                                                  | **Validation_Status values:** Untested, Inconslusive, Valid, Invalid - **Allowed Mutations_Status Values for Untested and Inconclusive:** *(See Notes Seciton #6)* None, Germline, Somatic, LOH, Post-transcriptional modification **Unknown Allowed Mutation_status Values for Valid:** *(See Notes Seciton #6)* Germline, Somatic, LOH, Post-transcriptional modification, Unknown - **Allowed Mutations_Status Values for Invalid:** *(See Notes Seciton #6)* none                                                                                                                                                                                                             |  |   |   |   |   |   |   |   |   |
 |                                                                                                                                                                                                                                                                                                                        |                                                |                       |                                                                                                                                     |                                                                                                                                                                                                                                         |                                      |   |   |   |   |   |   |   |   |
| 27           | Sequencing_Phase              | TCGA sequencing phase. Phase should change under any circumstance that the targets under consideration change.                                                                                                                                                                                                         | Phase_I                                        | No                    | Yes                                                                                                                                 | No                                                                                                                                                                                                                                      |                                      |   |   |   |   |   |   |   |   |
| 28           | Sequence_Source               | Molecular assay type used to produce the analytes used for sequencing. Allowed values are a subset of the [SRA 1.5](http://www.ncbi.nlm.nih.gov/viewvc/v1/trunk/sra/doc/SRA_1-5/) library_strategy field values. This subset matches those used at CGHub.                                                              | WGS;WXS                                        | Yes                   | No                                                                                                                                  | **Common TCGA values:** WGS, WGA, WXS, RNA-Seq, miRNA-Seq, Bisulfite-Seq, VALIDATION, Other **Other allowed values (per SRA 1.5)**  ncRNA-Seq, WCS, CLONE, POOLCLONE, AMPLICON, CLONEEND, FINISHING, ChIP-Seq, MNase-Seq, DNase-Hypersensitivity, EST, FL-cDNA, CTS, MRE-Seq, MeDIP-Seq, MBD-Seq, Tn-Seq, FAIRE-seq, SELEX, RIP-Seq, ChIA-PET
                                                                                                                                                                                                                       |                                      |   |   |   |   |   |   |   |   |
| 29           | Validation_Method             | The assay platforms used for the validation call. Examples: Sanger_PCR_WGA, Sanger_PCR_gDNA, 454_PCR_WGA, 454_PCR_gDNA; separate multiple entries using semicolons.                                                                                                                                                    | Sanger_PCR_WGA;Sanger_PCR_gDNA                 | No                    | **NO**. I**f Validation_Status = Untested then "none"** If Validation_Status = Valid or Invalid, then not "none" (case insensitive) | No                                                                                                                                                                                                                                      |                                      |   |   |   |   |   |   |   |   |
| 30           | Score                         | Not in use.                                                                                                                                                                                                                                                                                                            | NA                                             | No                    | Yes                                                                                                                                 | No                                                                                                                                                                                                                                      |                                      |   |   |   |   |   |   |   |   |
| 31           | BAM_File                      | Not in use.                                                                                                                                                                                                                                                                                                            | NA                                             | No                    | Yes                                                                                                                                 | No                                                                                                                                                                                                                                      |                                      |   |   |   |   |   |   |   |   |
| 32           | Sequencer                     | Instrument used to produce primary data. Separate multiple entries using semicolons.                                                                                                                                                                                                                                   | Illumina GAIIx;SOLID                           | Yes                   | No                                                                                                                                  | Illumina GAIIx, Illumina HiSeq, SOLID, 454, ABI 3730xl, Ion Torrent PGM, Ion Torrent Proton, PacBio RS, Illumina MiSeq, Illumina HiSeq 2500, 454 GS FLX Titanium, AB SOLiD 4 System                                                                                                                                                                                                                         |                                      |   |   |   |   |   |   |   |   |
| 33           | Tumor_Sample_UUID             | BCR aliquot UUID for tumor sample                                                                                                                                                                                                                                                                                      | 550e8400-e29b-41d4-a716-446655440000           | Yes                   | No                                                                                                                                  |                                                                                                                                                                                                                                         |                                      |   |   |   |   |   |   |   |   |
| 34           | Matched_Norm_Sample_UUID      | BCR aliquot UUID for matched normal                                                                                                                                                                                                                                                                                    | 567e8487-e29b-32d4-a716-446655443246           | Yes                   | No                                                       |                                                                                                                                                  

**Notes**<br/>
*1 Intergenic Region.* <br/>
*2 Consolidationd is used to indicate a site that was initially reported as a variant but subsequently removed from further analysis because it was consolidated into a new variant. For example, a SNP variant incorporated into a TNP variant.* <br/>
*3 Used when the discovered varieant differs from that of dbSNP.*<br/>
*4 These MAF headers describe the technology that was used to confirm a mutation, whether the same technology ("verification") or a different technology ("validation") is used to prove that a variant is germline or a somatic mutation.*<br/>
*5 These MAF headers describe the technology that was used toconfirm a mutation, whether the same technology (verification) or a different technology (validation) is used to prove that a variant is germline or a somatic mutation.*<br/>
*6 Explanation of some Validation Status-Mutation Status combinations.*<br/>

| Validation Status | Mutation Status | Explanation |
| ------------------ | --------------- | ----------- |
| Valid              | Unknown         | a valid variant with unknown somatic status due to lack of data from matched normal tissue. |
| Invalid            | None            | validation attempted, tumor and normal are homozygous reference (formerly described as Wildtype) |
| Inconclusive       | Unknown         | validation failed, neither the genotype nor its somatic status is certain due to lack of data from matched normal tissue |
| Inconclusive       | None            | validation failed, tumor genotype appears to be homozygous reference |

    Important Criteria

    **Index column indicates the order in which the columns are expected**. **All
    headers are case sensitive.** The Case Sensitive column specifies which values
    are case sensitive. The Null column indicates which MAF columns are allowed to
    have null values. The Enumerated column indicates which MAF columns have
    specified values: an Enumerated value of "No" indicates that there are no
    specified values for that column; other values indicate the specific values
    listed allowed; a value of "Set" indicates that the MAF column values come from
    a specified set of known values (*e.g.*HUGO gene symbols).


MAF file checks
===============

The DCC Archive Validator checks the integrity of a MAF file. Validation will
fail if any of the below are not true for a MAF file:

1.  Column header text (including case) and order must match specification
    (Table 1) exactly

2.  Values under column headers listed in the specification (Table 1) as not
    null must have values

3.  Values that are specified in Table 1 as Case Sensitive must be

4.  If column headers are listed in the specification as having *enumerated*
    values (*i.e.* a "Yes" in the "Enumerated" column), then the values under
    those column must come from the enumerated values listed under "Enumerated"

5.  If column headers are listed in the specification as having *set* values
    (*i.e.* a "Set" in the "Enumerated" column), then the values under those
    column must come from the enumerated values of that domain (*e.g.* HUGO gene
    symbols)

6.  All Allele-based columns must contain- (deletion), or a string composed of
    the following capitalized letters: A, T, G, C

7.  IfValidation_Status== "Untested"
    thenTumor_Validation_Allele1,Tumor_Validation_Allele2,Match_Norm_Validation_Allele1,Match_Norm_Validation_Allele2can
    be null (depending onValidation_Status)

    1.  IfValidation_Status== "Inconclusive"
        thenTumor_Validation_Allele1,Tumor_Validation_Allele2,Match_Norm_Validation_Allele1,Match_Norm_Validation_Allele2can
        be null (depending onValidation_Status)

8.  If Validation_Status == Valid, then Validated_Tumor_Allele1 and
    Validated_Tumor_Allele2must be populated (one of A, C, G, T, and -)

    1.  If Validation_Status == "Valid" then Tumor_Validation_Allele1,
        Tumor_Validation_Allele2, Match_Norm_Validation_Allele1,
        Match_Norm_Validation_Allele2 cannot be null

    2.  IfValidation_Status== "Invalid"
        thenTumor_Validation_Allele1,Tumor_Validation_Allele2,Match_Norm_Validation_Allele1,Match_Norm_Validation_Allele2cannot
        be null AND Tumor_Validation_Allelle1 ==
        Match_Norm_Validation_Allele1AND Tumor_Validation_Allelle2 ==
        Match_Norm_Validation_Allele2 (Added as a replacement for 8a as a
        result of breakdown)

9.  Check allele values against Mutation_Status:  
    Check allele values against Validation_status:

    1.  If Mutation_Status == "Germline" and Validation_Status == "Valid", then
        Tumor_Validation_Allele1 == Match_Norm_Validation_Allele1 and
        Tumor_Validation_Allele2 == Match_Norm_Validation_Allele2

    2.  If Mutation_Status == "Somatic" and Validation_Status == "Valid", then
        Match_Norm_Validation_Allele1 == Match_Norm_Validation_Allele2 ==
        Reference_Allele and (Tumor_Validation_Allele1 or
        Tumor_Validation_Allele2) != Reference_Allele

    3.  If Mutation_Status == "LOH" and Validation_Status=="Valid", then
        Tumor_Validation_Allele1 == Tumor_Validation_Allele2 and
        Match_Norm_Validation_Allele1 != Match_Norm_Validation_Allele2 and
        Tumor_Validation_Allele1 == (Match_Norm_Validation_Allele1 or
        Match_Norm_Validation_Allele2)

10. Check that Start_position \<= End_position

11. Check for the Start_position and End_position against Variant_Type:

    1.  If Variant_Type == "INS", then (End_position - Start_position + 1 ==
        length (Reference_Allele) or End_position - Start_position == 1) and
        length(Reference_Allele) \<= length(Tumor_Seq_Allele1 and
        Tumor_Seq_Allele2)

    2.  If Variant_Type == "DEL", then End_position - Start_position + 1 ==
        length (Reference_Allele), then length(Reference_Allele) \>=
        length(Tumor_Seq_Allele1 and Tumor_Seq_Allele2)

    3.  If Variant_Type == "SNP", then length(Reference_Allele and
        Tumor_Seq_Allele1 and Tumor_Seq_Allele2) == 1 and (Reference_Allele and
        Tumor_Seq_Allele1 and Tumor_Seq_Allele2) != "-"

    4.  If Variant_Type == "DNP", then length(Reference_Allele and
        Tumor_Seq_Allele1 and Tumor_Seq_Allele2) == 2 and (Reference_Allele and
        Tumor_Seq_Allele1 and Tumor_Seq_Allele2) !contain "-"

    5.  If Variant_Type == "TNP", then length(Reference_Allele and
        Tumor_Seq_Allele1 and Tumor_Seq_Allele2) == 3 and (Reference_Allele and
        Tumor_Seq_Allele1 and Tumor_Seq_Allele2) !contain "-"

    6.  If Variant_Type == "ONP", then length(Reference_Allele) ==
        length(Tumor_Seq_Allele1) == length(Tumor_Seq_Allele2) \> 3 and
        (Reference_Allele and Tumor_Seq_Allele1 and Tumor_Seq_Allele2) !contain
        "-"

12. Validation for UUID-based files:

    1.  Column \#33 must be Tumor_Sample_UUID containing UUID of the BCR aliquot
        for tumor sample

    2.  Column \#34 must be Matched_Norm_Sample_UUID containing UUID of the BCR
        aliquot for matched normal sample

    3.  Metadata represented by Tumor_Sample_Barcode and
        Matched_Norm_Sample_Barcode should correspond to the UUIDs assigned to
        Tumor_Sample_UUID and Matched_Norm_Sample_UUID respectively

13. If Validation_Status == "Valid" or "Invalid", then Validation_Method !=
    "none" (case insensitive)

MAF naming convention
=====================

In archives uploaded to the DCC, the MAF file name should relate to the
containing archive name in the following way:

If the archive has the name

    \<domain\>_\<disease_abbrev\>.\<platform\>.Level_2.\<serial_index\>.\<revision\>.0.tar.gz

then a somatic MAF file with the archive should be named according to

    \<domain\>_\<disease_abbrev\>.\<platform\>.Level_2.\<serial_index\>[.\<optional_tag\>].somatic.maf

and a protected MAF with the archive should be named according to

    \<domain\>_\<disease_abbrev\>.\<platform\>.Level_2.\<serial_index\>[.\<optional_tag\>].protected.maf

The \<optional_tag\> may consist of alphanumeric characters, dash, and
underscore; no spaces or periods; or it may be left out altogether. The purpose
of the optional tag is to impart some brief annotation.

*Example*

For the archive

    genome.wustl.edu_OV.IlluminaGA_DNASeq.Level_2.7.6.0.tar.gz

the following are examples of valid maf names

    genome.wustl.edu_OV.IlluminaGA_DNASeq.Level_2.7.somatic.maf
    genome.wustl.edu_OV.IlluminaGA_DNASeq.Level_2.7.protected.maf


# FoundationOne Variant Aggregation and Masking

## Description ##

FoundationOne Variant Aggregation and Masking is a somatic mutation aggregation workflow used in GDC targeted sequencing harmonization.

## Overview ##

FoundationOne Variant Aggregation and Masking is a pipeline used for targeted sequencing somatic mutation aggregation at the GDC. Somatic mutation aggregation is performed by aggregating annotated somatic mutations from the FM-AD project in the GDC by case primary site.

### Input

* Tumor VCF - Targeted Sequencing

### Output

* MAF (Data Type: Aggregated Somatic Mutation)

## External Links ##

* [Foundation Medicine](https://www.foundationmedicine.com/)
* [Foundation Medicine at the GDC](https://gdc.cancer.gov/about-gdc/contributed-genomic-data-cancer-research/foundation-medicine)
* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type

# FPKM-UQ #
## Description ##

Fragments Per Kilobase of transcript per Million mapped reads upper quartile (FPKM-UQ) is a RNA-Seq-based expression normalization method.  The FPKM-UQ is based on a modified version of the [FPKM](FPKM.md) normalization method.  

## Overview ##

FPKM-UQ is implemented at the GDC on gene-level read counts that are produced by STAR<sup>1</sup> and generated using custom scripts<sup>2</sup>. The formula used to generate FPKM-UQ values is as follows:

FPKM = [RM<sub>g</sub> * 10<sup>9</sup> ] / [RM<sub>75</sub> * L]

* RM<sub>g</sub>: The number of reads mapped to the gene
* RM<sub>75</sub>: The number of read mapped to the 75th percentile gene in the alignment
* L: The length of the gene in base pairs

FPKM-UQ files are available as tab delimited files with the Ensembl gene IDs in the first column and the expression values in the second.

### Notes
- The scalar (10<sup>9</sup>) is added to normalize the values to "__kilo__ base" and "__million__ mapped reads"
- FPKM-UQ values tend to be much higher than FPKM values because of the large difference between the total mapped number of reads in an alignment and the mapped number of reads to one gene 

## References ##
1. [STAR-Fusion pipeline](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/#star-fusion-pipeline)
2. [GDC mRNA-Seq Documentation](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/)


## External Links ##
* [Ensembl Human Genome](http://www.ensembl.org/Homo_sapiens/Info/Annotation)
* [GENCODE 36](https://www.gencodegenes.org/human/release_36.html)

Categories: Workflow Type


# ABSOLUTE LiftOver

## Description ##

ABSOLUTE LiftOver is a copy number variation (CNV) pipeline used in GDC genotyping array harmonization.

## Overview ##

The ABSOLUTE LiftOver workflow in the GDC is a copy number variation (CNV) pipeline used for genotyping array harmonization. It converts genomic coordinates from hg19 to GRCh38, ensuring high-quality, curated CNV data for downstream analysis. This pipeline also provides gene-level copy numbers, along with purity and ploidy measurements.


### Input

* Tumor CEL - Genotyping Array

### Output

* TSV (Data Type: Gene Level Copy Number)

## References ##

1. [Copy Number Variation Analysis Pipeline](/Data/Bioinformatics_Pipelines/CNV_Pipeline/)
1. [ABSOLUTE Copy Number](/Data/Bioinformatics_Pipelines/CNV_Pipeline/#absolute-copy-number)

## External Links ##

* [TCGA PanCancer analysis papers](https://doi.org/10.1016/j.ccell.2018.03.007)
* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type

# The Cancer Imaging Archive (TCIA) #
## Description ##
The Cancer Imaging Archive is an open-access database of medical images for cancer research.
## Overview ##
The Cancer Imaging Archive is an open-access database of medical images for cancer research.  It provides access to radiological imaging data sets in DICOM format from the TCGA cases.
## References ##
1. N/A

## External Links ##
* [The Cancer Imaging Archive](http://www.cancerimagingarchive.net)

Categories: General


# Aliquot #
## Description ##
An aliquot pertains to a portion of the whole, especially a sample taken for biological analysis or other treatment.

## Overview ##
Aliquots are typically the products shipped by a Biospecimen Core Resource (BCR) to a Genome Characterization Center (GCC) or analysis center. The [GDC Data Model](https://gdc.cancer.gov/developers/gdc-data-model/gdc-data-model-components) relates aliquots to samples and/or analytes for capturing the biospecimen information and read groups for associating experimental data.

## References ##
1. [GDC Data Dictionary - Aliquot](/Data_Dictionary/viewer/#?view=table-definition-view&id=aliquot)
2. [GDC Biospecimen Data Encyclopedia Page](/Encyclopedia/pages/Biospecimen_Data/)

## External Links ##
* N/A

Categories: General, Biospecimen


# SeSAMe Methylation Beta Estimation

## Description ##

SeSAMe Methylation Beta Estimation is a workflow used in GDC methylation array harmonization.

## Overview ##

SeSAMe Methylation Beta Estimation is used for methylation array harmonization at the GDC. Methylation array harmonization is performed with SeSAMe using raw tumor or normal methylation array files and generates beta values and masked intensities data, which removes potential genotype information.

### Input

* Raw Methylation Array

### Output

* IDAT (Data Type: Masked Intensities)
* TXT (Data Type: Methylation Beta Value)

## References ##

1. [Methylation Analysis](/Data/Bioinformatics_Pipelines/Methylation_Pipeline/)
1. [SeSAMe Methylation Beta Values File Format](/Data/Bioinformatics_Pipelines/Methylation_Pipeline/#sesame-methylation-beta-values-file-format)

## External Links ##

* [SeSAMe - SEnsible Step-wise Analysis of Methylation data](https://github.com/zwdzwd/sesame)
* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type

# STAR 2-Pass Genome

## Description ##

STAR 2-Pass Genome is an alignment pipeline used in GDC RNA-Seq harmonization.

## Overview ##

STAR 2-Pass Genome is a pipeline used for RNA-Seq reads alignment at the GDC. Alignment is performed with STAR and generates sequencing reads data.

### Input

* Submitted sequencing reads

### Output

* BAM (Data Type: Aligned Reads)

[![STAR Pipeline Diagram](https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/images/RNA-Seq-DR32_Image.png)](https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/images/RNA-Seq-DR32_Image.png "Click to see the full image.")

## References ##

1. [mRNA Analysis Pipeline](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/)
1. [RNA-Seq Alignment Workflow](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/#rna-seq-alignment-workflow)
1. [RNA-Seq Alignment Command Line Parameters](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/#rna-seq-alignment-command-line-parameters)

## External Links ##

* [STAR](https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf)
* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type

# STAR-Fusion

## Description ##

STAR-Fusion is a structural variant calling pipeline used in GDC RNA-Seq harmonization.

## Overview ##

STAR-Fusion is a pipeline used for RNA-Seq structural variant calling at the GDC. Structural variant calling is performed with STAR-Fusion using tumor or normal alignments and generates structural variation data.

### Input

* BAM - RNA-Seq

### Output

* BEDPE (Data Type: Transcript Fusion)
* TSV (Data Type: Transcript Fusion)

[![STAR Pipeline Diagram](https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/images/RNA-Seq-DR32_Image.png)](https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/images/RNA-Seq-DR32_Image.png "Click to see the full image.")

## References ##

1. [mRNA Analysis Pipeline](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/)
1. [STAR-Fusion Pipeline](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/#star-fusion-pipeline)

## External Links ##

* [STAR](https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf)
* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type

# GDC Application Programming Interface (API) #
## Description ##
The GDC Application Programming Interface (API) is an external facing REST interface for querying and downloading GDC data, and submitting data to the GDC.
## Overview ##
The GDC API drives the GDC Data Portal, the GDC Submission Portal, and the GDC Data Transfer Tool (DTT) and is made accessible to external users for programmatic access to the same functionality found through GDC Portals. This includes searching, viewing, submitting and downloading subsets of data files, metadata, and annotations based on specific parameters. The GDC API also provides remote BAM slicing functionality that enables downloading of specific parts of a BAM file instead of the whole file.

Communicating with the GDC API involves making calls to API endpoints that represent specific API functionality:

* __status:__ Get the API status and version information
* __projects:__ Search all data generated by a project
* __cases:__ Find all files related to a specific case, or sample donor
* __files:__ Find all files with specific characteristics such as file_name, md5sum, data_format and others
* __annotations:__ Search annotations added to data after curation
* __data:__ Used to download GDC data
* __manifest:__ Generates manifests for use with GDC Data Transfer Tool
* __slicing:__ Allows remote slicing of BAM format objects
* __submission:__ Returns the available resources at the top level above programs i.e., registered programs

The GDC API uses JSON as its communication format, and standard HTTP methods like GET, PUT, POST and DELETE.

## References ##
1. [GDC API](https://gdc.cancer.gov/developers/gdc-application-programming-interface-api)
2. [GDC API User's Guide](/API/Users_Guide/Getting_Started/)

## External Links ##
* [JSON](http://www.json.org/)

Categories: Tool


# VarScan2

## Description ##

VarScan2 is a somatic variant calling pipeline used in GDC whole exome sequencing (WXS) and targeted sequencing harmonization.

## Overview ##

VarScan2 is one of the four pipelines used for WXS and targeted sequencing somatic variant calling at the GDC. Somatic variant calling is performed with VarScan2 using tumor and normal alignments and generates single-nucleotide polymorphism (SNP) data.

### Input

* Tumor BAM - WXS or Targeted Sequencing
* Normal BAM - WXS or Targeted Sequencing

### Output

* VCF (Data Type: Raw Simple Somatic Mutation)

[![VarScan2 Pipeline Diagram](https://gdc.cancer.gov/system/files/public/image/varscan-somatic-variant-calling-pipeline.png)](https://gdc.cancer.gov/system/files/public/image/varscan-somatic-variant-calling-pipeline.png "Click to see the full image.")

## References ##

1. [DNA Seq Processing at the GDC](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/)
1. [Somatic Variant Calling Workflow](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/#somatic-variant-calling-workflow)

## External Links ##

* [VarScan](https://dkoboldt.github.io/varscan/)
* [VarScan 2: somatic mutation and copy number alteration discovery in cancer by exome sequencing.](http://genome.cshlp.org/content/22/3/568.short)
* [Overview of GDC Harmonization Workflows](https://github.com/NCI-GDC/gdc-workflow-overview/blob/master/README.md)
* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type

# GDC Data Submission Portal #
## Description ##
The GDC Data Submission Portal<sup>1</sup> is a platform that allows researchers to submit and release data into the GDC.
## Overview ##

Key GDC Data Submission Portal features include<sup>2</sup>:

* __Upload and Validate Data:__ Project data can be uploaded to the GDC project workspace. The GDC will validate the data against the GDC Data Dictionary.
* __Review and Submit Data:__ Prior to submission, data can be reviewed to check for accuracy. Once the review is complete, the data can be submitted to the GDC for processing through Data Harmonization.
* __Release Data:__ After harmonization, data can be released to the research community for access through GDC Data Access Tools
* __Download Data:__ Data that has been uploaded into the project workspace can be downloaded for review or update. Data can then be re-uploaded before it is released for access through GDC Data Access Tools
* __Browse Data:__ Data that has been uploaded to the project workspace can be browsed to ensure that the project is ready for processing
* __Status and Alerts:__ Visual cues are implemented to easily identify incomplete submissions

## References ##
1. [GDC Data Submission Portal](https://portal.gdc.cancer.gov/submission/)
2. [GDC Data Submission Portal User's Guide](/Data_Submission_Portal/Users_Guide/Data_Submission_Overview/)

## External Links ##
* N/A

Categories: Tool


# BRASS

## Description ##

BRASS is a structural variant calling pipeline used in GDC whole genome sequencing (WGS) harmonization.

## Overview ##

Structural variant calling is performed with BRASS using tumor and normal alignments and generates Structural Rearrangement data.

### Input

* Tumor BAM - WGS
* Normal BAM - WGS

### Output

* BEDPE (Data Type: Structural Rearrangement)
* VCF (Data Type: Structural Rearrangement)

## References ##

1. [DNA-Seq Analysis Pipeline](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/)
1. [Whole Genome Sequencing Variant Calling](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/#whole-genome-sequencing-variant-calling)

## External Links ##

* [GDC Data Portal](https://portal.gdc.cancer.gov)
* [Workflow generated by the Sanger Institute](https://github.com/cancerit/dockstore-cgpwgs)

Categories: Workflow Type

# Center for Cancer Genomics (CCG) #
## Description ##
The Center for Cancer Genomics (CCG) is an organization within the National Cancer Institute (NCI) that unifies NCI's activities in cancer genomics. The CCG aims to synthesize research in different fields of cancer genomics - structural, functional, and computational - to improve patient outcomes<sup>1</sup>.

## Overview ##
CCG and the offices it oversees-The Cancer Genome Atlas (TCGA) and the Office of Cancer Genomics (OCG)-manage multiple programs. These offices serve to advance CCG's goal of ushering in a modern era of diagnosis, treatment, and prevention based on the study of genomes.
CCG programs and collaborations generate cancer genomic and clinical data, and make this data available for widespread use by the research community.

CCG established the NCI Genomic Data Commons (GDC) to provide the cancer research community with a data service supporting the receipt, quality control, integration, storage, and redistribution of standardized cancer genomic data sets derived from cancer studies.

## References ##
1. [Center for Cancer Genomics](https://www.cancer.gov/about-nci/organization/ccg)

## External Links ##
* N/A

Categories: General


# Data Access Policy #
## Description ##
A Data Access Policy is an established control put in place to ensure that data protection requirements are followed. The GDC Data Access Policy<sup>1</sup> ensures that personally identifiable information in data made available in the GDC is kept from unauthorized users. GDC access control policies adhere to the [National Institutes of Health (NIH) Genomic Data Sharing Policy (GDS) Policy](https://sharing.nih.gov/genomic-data-sharing-policy/about-genomic-data-sharing)<sup>2</sup> as well as the [NCI GDS Policy](https://www.cancer.gov/grants-training/grants-management/nci-policies/genomic-data)<sup>3</sup>. The GDC requires that users obtain authorization from the [National Center for Biotechnology Information (NCBI) Database of Genotypes and Phenotypes (dbGaP)](https://www.ncbi.nlm.nih.gov/gap) for accessing controlled data.

## Overview ##
Any user accessing GDC open data must adhere to the NIH Genomic Data Sharing (GDS) Policy<sup>2</sup> which indicates that investigators who download unrestricted-access data from NIH-designated data repositories should:

* Not attempt to identify individual human research participants from whom the data were obtained
* Acknowledge in all oral or written presentations, disclosures, or publications the specific dataset(s) or applicable accession number(s) and the NIH-designated data repositories through which the investigator accessed any data

Any user requesting access to GDC controlled data must apply for dbGaP authorization, which is granted by an NIH Data Access Committee (DAC). DACs review and approve or disapprove all requests from the research community for data access.  Decisions to grant access are made based on whether the request conforms to the specifications within the NIH Genomic Data Sharing Policy and program specific requirements or procedures (if any). All uses proposed for controlled-access data must be consistent with the data use limitations for the data set as indicated by the submitting institution and identified on the public website for dbGaP. DACs also review and approve or disapprove all requests for access to dbGaP data for programmatic oversight by NIH employees.

## References ##
1. [GDC Data Access Policies](https://gdc.cancer.gov/access-data/data-access-policies)
2. [NIH GDS Policy](https://grants.nih.gov/grants/guide/notice-files/not-od-14-124.html)
3. [NCI GDS Policy](https://www.cancer.gov/grants-training/grants-management/nci-policies/genomic-data)

## External Links ##
* [NCBI dbGaP](https://www.ncbi.nlm.nih.gov/gap)
* [dbGaP Application for Controlled Access Data](https://dbgap.ncbi.nlm.nih.gov/aa/wga.cgi?page=login)

Categories: General


# GATK4 MuTect2 Tumor-Only

## Description ##

GATK4 MuTect2 Tumor-Only is a somatic variant calling pipeline utilized in the GDC for analyzing tumor-only whole exome sequencing (WXS) and targeted sequencing data without normal samples. 

## Overview ##

GATK4 MuTect2 Tumor-Only is a pipeline used for WXS and targeted sequencing somatic variant calling at the GDC. Somatic variant calling is performed with GATK4 MuTect2 Tumor-Only using tumor alignments and generates single nucleotide variants (SNVs) data.

### Input

* Tumor BAM - WXS or Targeted Sequencing

### Output

* VCF (Data Type: Raw Simple Somatic Mutation)

## References ##

1. [DNA-Seq Analysis Pipeline](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/)
1. [Tumor-Only Variant Calling Workflow](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/#tumor-only-variant-calling-workflow)

## External Links ##

* [Overview of GDC Harmonization Workflows](https://github.com/NCI-GDC/gdc-workflow-overview/blob/master/README.md)
* [GDC Data Portal](https://portal.gdc.cancer.gov)

Categories: Workflow Type

---
hide:
    - navigation
    - toc
---

<script src="https://code.jquery.com/jquery-3.7.1.min.js" integrity="sha256-/JqT3SQfawRcv/BIHPThkBvs0OEvtFFmqPF/lYI/Cxo=" crossorigin="anonymous"></script>
<script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/js/bootstrap.bundle.min.js" integrity="sha384-YvpcrYf0tY3lHB60NNkmXc5s9fDVZLESaAA55NDzOxhy9GkcIdslK1eN7N6jIeHz" crossorigin="anonymous"></script>
<script type="text/javascript" src="lib/js/jquery-ui.min.js"></script>
<script type="text/javascript" src="lib/js/pagination.min.js"></script>
<script type="text/javascript" src="lib/js/jsrender.min.js"></script>
<!-- boostrap used by bundle.js for tooltips -->
<script type="text/javascript" src="dist/bundle.js"></script>
<link rel="stylesheet" href="lib/css/jquery-ui.min.css">
<link rel="stylesheet" href="lib/css/pagination.min.css">
<link rel="stylesheet" href="dist/styles.css">


<!-- GDCMVS APP -->

<h1 class="hide">GDC Metadata Validation Services</h1>
<div class="search-box">
    <div class="search-bar">
      <span id="suggestWidth" class="suggest__width"></span>
      <div class="input-group search-bar__group">
        <input id="keywords" type="text" class="form-control search-bar__input" aria-label="keywords" placeholder="Please enter keyword" autocomplete="off">
        <div id="search-bar-options" class="search-bar__options dropdown" style="display: none;">
          <a href="#" data-toggle="dropdown" class="dropdown-toggle search-bar__option" aria-label="boolean operators"><i class="fa fa-ellipsis-h"></i></a>
          <div class="dropdown-menu search-bar__dropdown">
              <a class="search-bar__boolean" data-boolean="AND" href="#">AND</a>
              <a class="search-bar__boolean" data-boolean="OR" href="#">OR</a>
              <a class="search-bar__boolean" data-boolean="NOT" href="#">NOT</a>
          </div>
          <a href="#" id="searchclear" class="search-bar__option" aria-label="clear search bar"><i class="fa fa-times"></i></a>
        </div>
        <span class="input-group-btn">
          <button id="search" class="btn search-bar__btn" type="button">Search</button>
        </span>
      </div>
      <div class="suggest">
        <div id="suggestBox" class="suggest__listbox"></div>
      </div>
    </div>
    <div class="search-options">
      <div class="checkbox">
        <label class="checkbox__label checkbox__label--padding">
          <input id="i_ematch" class="checkbox__input" type="checkbox" value="" tabindex="0">
          <span class="checkbox__btn"><i class="checkbox__icon fa fa-check"></i></span> Exact match
        </label>
        <label class="checkbox__label">
          <input id="i_desc" class="checkbox__input" type="checkbox" value="" tabindex="0">
          <span class="checkbox__btn"><i class="checkbox__icon fa fa-check"></i></span> Property description
        </label>
        <label class="checkbox__label">
          <input id="i_syn" class="checkbox__input" type="checkbox" value="" tabindex="0">
          <span class="checkbox__btn"><i class="checkbox__icon fa fa-check"></i></span> Synonyms
        </label>
      </div>
      <div class="ref-box">
        <a href="https://ncit.nci.nih.gov/" class="ref-box__link" target="_blank">Search in NCIt</a>
        <a href="https://cadsr.cancer.gov/onedata/Home.jsp"
          class="ref-box__link" target="_blank">Search in caDSR</a>
      </div>
    </div>
</div>

<div id="gdc-loading-icon" class="loadingContainer" style="display: none;">
  <div class="spinParticleContainer">
      <div class="particle red"></div>
      <div class="particle grey other-particle"></div>
      <div class="particle blue other-other-particle"></div>
  </div>
  <div>Loading GDC Data...</div>
</div>

<div id="root"></div>

<div id="info-content" class="info-content">
    <div id="unofficial-term"></div>
    <div id="version-content" class="version-content">GDC Dictionary Version</div>
</div>
<div id="alert-error" class="alert alert__error alert-info" role="alert">Error: Undefined</div>
<div id="alert-suggest" class="alert alert__suggest alert-info" role="alert">
  Coming soon: users will be able to suggest new vocabulary terms to the GDC via this link.
</div>

<!-- END GDCMVS APP -->


---
hide:
    - navigation
    - toc
---

# Data Dictionary Viewer

<link rel="stylesheet" href="dictionary.css">
<script src="https://code.jquery.com/jquery-3.7.1.min.js" integrity="sha256-/JqT3SQfawRcv/BIHPThkBvs0OEvtFFmqPF/lYI/Cxo=" crossorigin="anonymous"></script>
<script src="https://cdn.jsdelivr.net/npm/lodash@4.17.21/lodash.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/d3/7.9.0/d3.min.js" integrity="sha512-vc58qvvBdrDR4etbxMdlTt4GBQk1qjvyORR2nrsPsFPyrs+/u5c3+1Ct6upOgdZoIl7eq6k3a1UPDSNAQi/32A==" crossorigin="anonymous" referrerpolicy="no-referrer"></script>
<script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/js/bootstrap.bundle.min.js" integrity="sha384-YvpcrYf0tY3lHB60NNkmXc5s9fDVZLESaAA55NDzOxhy9GkcIdslK1eN7N6jIeHz" crossorigin="anonymous"></script>

<script src="dictionary-init.js"></script>
<script src="dictionary.js"></script>
<script src="dictionary-views.js"></script>
<p id="dictionary-preamble" style="margin: 1rem auto 2rem auto; width: 95%;">
The GDC data dictionary viewer is a user-friendly interface for accessing the <a href="../">GDC Data Dictionary</a>.
</p>
<div id="dictionary-loading-icon" class="loadingContainer">
    <div class="spinParticleContainer">
        <div class="particle red"></div>
        <div class="particle grey other-particle"></div>
        <div class="particle blue other-other-particle"></div>
    </div>
  <div>
  Loading Dictionary Data...
  </div>
</div>
<div id="dictionary-app-container"></div>


# Data Dictionary Release Notes

| Version | Date |
|---|---|
| [v.3.2.0](Data_Dictionary_Release_Notes.md#v320) | September XX, 2024 |
| [v.3.1.0](Data_Dictionary_Release_Notes.md#v310) | May 30, 2024 |
| [v.3.0.0](Data_Dictionary_Release_Notes.md#v300) | October 30, 2023 |
| [v.2.6.6](Data_Dictionary_Release_Notes.md#v266) | June 16, 2023 |
| [v.2.6.0](Data_Dictionary_Release_Notes.md#v260) | February 2, 2023 |
| [v.2.5.0](Data_Dictionary_Release_Notes.md#v250) | July 8, 2022 |
| [v.2.4.1](Data_Dictionary_Release_Notes.md#v241) | August 23, 2021 |
| [v.2.4.0](Data_Dictionary_Release_Notes.md#v240) | June 21, 2021 |
| [v.2.3.0](Data_Dictionary_Release_Notes.md#v230) | January 5, 2021 |
| [v.2.2.0](Data_Dictionary_Release_Notes.md#v220) | July 2, 2020 |
| [v.2.1.0](Data_Dictionary_Release_Notes.md#v210) | March 10, 2020 |
| [v.2.0.0](Data_Dictionary_Release_Notes.md#v200) | January 30, 2020 |
| [v.1.18.1](Data_Dictionary_Release_Notes.md#v1181) | November 6, 2019 |
| [v.1.18](Data_Dictionary_Release_Notes.md#v118) | July 31, 2019 |
| [v.1.17](Data_Dictionary_Release_Notes.md#v117) | June 5, 2019 |
| [v.1.16](Data_Dictionary_Release_Notes.md#v116) | April 17, 2019 |
| [v.1.15](Data_Dictionary_Release_Notes.md#v115) | December 18, 2018 |
| [v.1.14](Data_Dictionary_Release_Notes.md#v114) | September 27, 2018 |
| [v1.13](Data_Dictionary_Release_Notes.md#v113) | May 21, 2018 |
| [v1.12.1](Data_Dictionary_Release_Notes.md#v1121) | April 26, 2018 |
| [v1.12](Data_Dictionary_Release_Notes.md#v112) | April 23, 2018 |
| [v1.11](Data_Dictionary_Release_Notes.md#v111) | January 20, 2018 |
| [v1.10.0](Data_Dictionary_Release_Notes.md#release-with-api-v1100) | August 22, 2017 |
| [v1.7.1](Data_Dictionary_Release_Notes.md#release-with-api-v171) | March 16, 2017 |
| [v1.3.1](Data_Dictionary_Release_Notes.md#release-with-api-v131) | September 7, 2016 |

## v3.2.0

* __GDC Product__: GDC Data Dictionary
* __Release Date__: September XX, 2024

### New Features and Changes

* Altered `demographic` Entity
	* New property: `population_group`
* Altered `diagnosis` Entity
	* New property: `ajcc_serum_tumor_markers`
	* Changes made to `method_of_diagnosis`
		* New permissible value: `Orchiectomy`
	* Changes made to `primary_diagnosis`
		* New permissible value: `Adenocarcinoma in tubulovillous adenoma`
		* New permissible value: `Myxofibrosarcoma`
		* New deprecated value: `Adenocarcinoma in tubolovillous adenoma`
	* Changes made to `sites_of_involvement`
		* New permissible value: `Breast, Left Lower Inner`
		* New permissible value: `Breast, Left Lower Outer`
		* New permissible value: `Breast, Left Upper Inner`
		* New permissible value: `Breast, Left Upper Outer`
		* New permissible value: `Breast, Right Lower Inner`
		* New permissible value: `Breast, Right Lower Outer`
		* New permissible value: `Breast, Right Upper Inner`
		* New permissible value: `Breast, Right Upper Outer`
		* New permissible value: `Eye, Choroid`
		* New permissible value: `Eye, Ciliary Body`
		* New permissible value: `Eye, Iris`
		* New permissible value: `Hilar Fat`
		* New permissible value: `Rete Testis`
		* New permissible value: `Scrotum`
		* New permissible value: `Skin, Extremities`
		* New permissible value: `Skin, Groin`
		* New permissible value: `Skin, Head and Neck`
		* New permissible value: `Skin, Trunk`
		* New permissible value: `Spermatic Cord`
		* New permissible value: `Tunica Albuginea`
		* New permissible value: `Tunica Vaginalis`
* Altered `follow_up` Entity
	* New property: `histologic_progression`
	* Changes made to `evidence_of_progression_type`
		* New permissible value: `Biopsy with Histologic Confirmation`
		* New permissible value: `Physical Examination`
		* New permissible value: `Positive Biomarker(s)`
	* Changes made to `imaging_anatomic_site`
		* New permissible value: `Cervix Uteri`
	* Changes made to `imaging_findings`
		* New permissible value: `Bladder Involvement`
		* New permissible value: `Extra-Pelvic Metastatic Disease`
		* New permissible value: `Paraaortic Lymph Node Involvement`
		* New permissible value: `Parametrium Involvement`
		* New permissible value: `Pelvic Lymph Node Involvement`
		* New permissible value: `Supraclavicular Lymph Node Involvement`
	* Changes made to `timepoint_category`
		* New permissible value: `After Chemotherapy`
		* New permissible value: `After Study Enrollment`
		* New permissible value: `End of Consolidation Therapy`
		* New permissible value: `End of Treatment Course`
		* New permissible value: `End of Treatment Course 1`
		* New permissible value: `End of Treatment Course 2`
		* New permissible value: `First Complete Response`
		* New permissible value: `First Treatment`
		* New permissible value: `Post Initial Treatment`
		* New permissible value: `Prior to Chemotherapy`
		* New permissible value: `Prior to Procurement`
		* New permissible value: `Prior to Study Enrollment`
		* New permissible value: `Prior to Study Registration`
		* New permissible value: `Progression`
		* New permissible value: `Recurrence`
		* New permissible value: `Sample Procurement`
* Altered `molecular_test` Entity
	* Changes made to `hpv_strain`
		* New permissible value: `HPV63`
		* New permissible value: `HPV70`
	* Changes made to `test_value_range`
		* New permissible value: `1-4%`
		* New permissible value: `5-9%`
	* Changes made to `timepoint_category`
		* New permissible value: `Adjuvant Therapy`
		* New permissible value: `Adolescence`
		* New permissible value: `Adulthood`
		* New permissible value: `After Chemotherapy`
		* New permissible value: `After Study Enrollment`
		* New permissible value: `Childhood`
		* New permissible value: `End of Treatment Course`
		* New permissible value: `First Complete Response`
		* New permissible value: `First Treatment`
		* New permissible value: `Follow-up`
		* New permissible value: `Last Contact`
		* New permissible value: `Post Adjuvant Therapy`
		* New permissible value: `Post Hormone Therapy`
		* New permissible value: `Post Initial Treatment`
		* New permissible value: `Post Secondary Therapy`
		* New permissible value: `Postoperative`
		* New permissible value: `Prior to Adjuvant Therapy`
		* New permissible value: `Prior to Chemotherapy`
		* New permissible value: `Prior to Diagnosis`
		* New permissible value: `Prior to Procurement`
		* New permissible value: `Prior to Study Enrollment`
		* New permissible value: `Prior to Study Registration`
		* New permissible value: `Progression`
		* New permissible value: `Recurrence`
		* New permissible value: `Recurrence/Progression`
		* New permissible value: `Within 2 Months After Completion of First-Course Treatment`
		* New permissible value: `Within 3 Months of Surgery`
		* New permissible value: `Other`
* Altered `other_clinical_attribute` Entity
	* New property: `undescended_testis_corrected_age_range`
	* Changes made to `comorbidities`
		* New permissible value: `Adenomyosis`
		* New permissible value: `Adenosis (Atypical Adenomatous Hyperplasia)`
		* New permissible value: `Alcoholic Liver Disease`
		* New permissible value: `Allergy, Animal, NOS`
		* New permissible value: `Allergy, Ant`
		* New permissible value: `Allergy, Bee`
		* New permissible value: `Allergy, Cat`
		* New permissible value: `Allergy, Dairy or Lactose`
		* New permissible value: `Allergy, Dog`
		* New permissible value: `Allergy, Eggs`
		* New permissible value: `Allergy, Food, NOS`
		* New permissible value: `Allergy, Fruit`
		* New permissible value: `Allergy, Meat`
		* New permissible value: `Allergy, Mold or Dust`
		* New permissible value: `Allergy, Nuts`
		* New permissible value: `Allergy, Processed Foods`
		* New permissible value: `Allergy, Seafood`
		* New permissible value: `Allergy, Wasp`
		* New permissible value: `Alpha-1 Antitrypsin Deficiency`
		* New permissible value: `Altered Mental Status`
		* New permissible value: `Androgen Excess`
		* New permissible value: `Autoimmune Atrophic Chronic Gastritis`
		* New permissible value: `BAP1 Tumor Predisposition Syndrome`
		* New permissible value: `Benign Prostatic Hyperplasia`
		* New permissible value: `Birt-Hogg-Dube Syndrome`
		* New permissible value: `BK Virus`
		* New permissible value: `Chronic Kidney Disease`
		* New permissible value: `Colonization, Bacterial`
		* New permissible value: `Colonization, Fungal`
		* New permissible value: `Common Variable Immune Deficiency (CVID)`
		* New permissible value: `Cortisol Excess`
		* New permissible value: `Cowden Syndrome`
		* New permissible value: `Cyst(s)`
		* New permissible value: `Dementia`
		* New permissible value: `Diabetes, NOS`
		* New permissible value: `Diabetes, Type I`
		* New permissible value: `Diverticulosis`
		* New permissible value: `Endometriosis`
		* New permissible value: `Endosalpingiosis`
		* New permissible value: `Epithelial Dysplasia`
		* New permissible value: `Epithelial Hyperplasia`
		* New permissible value: `Estrogen Excess`
		* New permissible value: `Gastric Polyp(s)`
		* New permissible value: `Gilbert's Syndrome`
		* New permissible value: `Glomerular Disease`
		* New permissible value: `Hay Fever`
		* New permissible value: `Helicobacter pylori-Associated Gastritis`
		* New permissible value: `Hematologic Disorder, NOS`
		* New permissible value: `Hemochromatosis`
		* New permissible value: `Hepatic Encephalopathy`
		* New permissible value: `Hepatitis, NOS`
		* New permissible value: `Hereditary Breast Cancer`
		* New permissible value: `Hereditary Hemorrhagic Telangiectasia`
		* New permissible value: `Hereditary Kidney Oncocytoma`
		* New permissible value: `Hereditary Leiomyomatosis and Renal Cell Carcinoma`
		* New permissible value: `Hereditary Ovarian Cancer`
		* New permissible value: `Hereditary Papillary Renal Cell Carcinoma`
		* New permissible value: `Hereditary Prostate Cancer`
		* New permissible value: `Hereditary Renal Cell Carcinoma`
		* New permissible value: `High Grade Dysplasia`
		* New permissible value: `High-grade Prostatic Intraepithelial Neoplasia (PIN)`
		* New permissible value: `Human Herpesvirus-6 (HHV-6)`
		* New permissible value: `Human Herpesvirus-8 (HHV-8)`
		* New permissible value: `Inflammation`
		* New permissible value: `Inflammation, Hyperkeratosis`
		* New permissible value: `Inherited Genetic Syndrome, NOS`
		* New permissible value: `Intestinal Metaplasia`
		* New permissible value: `Low Grade Dysplasia`
		* New permissible value: `Mineralcorticoids Excess`
		* New permissible value: `Motor / Movement Change`
		* New permissible value: `Myelodysplastic Syndrome`
		* New permissible value: `Neurocystericerosis`
		* New permissible value: `Nevus of Ota`
		* New permissible value: `Nodular Prostatic Hyperplasia`
		* New permissible value: `Nonalcoholic Fatty Liver Disease`
		* New permissible value: `Parasitic Disease of Biliary Tract`
		* New permissible value: `Parkinson's Disease`
		* New permissible value: `Recurrent Pyogenic Cholangitis`
		* New permissible value: `Rubella`
		* New permissible value: `Scarlet Fever`
		* New permissible value: `Sensory Changes`
		* New permissible value: `Serous Tubal Intraepithelial Carcinoma (STIC)`
		* New permissible value: `Sialadenitis`
		* New permissible value: `Skin Rash`
		* New permissible value: `Squamous Metaplasia`
		* New permissible value: `Succinate Dehydrogenase-Deficient Renal Cell Carcinoma`
		* New permissible value: `Thyroid Nodular Hyperplasia`
		* New permissible value: `Tobacco, NOS`
		* New permissible value: `Tobacco, Smokeless`
		* New permissible value: `Tobacco, Smoking`
		* New permissible value: `Tuberous Sclerosis`
		* New permissible value: `Tubulointerstitial Disease`
		* New permissible value: `Tumor-Associated Lymphoid Proliferation`
		* New permissible value: `Undescended Testis`
		* New permissible value: `Vascular Disease`
		* New permissible value: `Vision Changes`
		* New permissible value: `Von Hippel-Lindau Syndrome`
		* New deprecated value: `Alpha-1 Antitrypsin`
		* New deprecated value: `Diabetes`
		* New deprecated value: `Gastroesophageal Reflux Disease`
		* New deprecated value: `Hepatitis`
		* New deprecated value: `Kidney Disease`
		* New deprecated value: `Smoking`
	* Changes made to `number_of_pregnancies`
		* New permissible value: `0`
	* Changes made to `risk_factors`
		* New permissible value: `Adenocarcinoma`
		* New permissible value: `Adenomatous Polyposis Coli`
		* New permissible value: `Allergies`
		* New permissible value: `Arthritis`
		* New permissible value: `Atrial Fibrillation`
		* New permissible value: `Basal Cell Carcinoma`
		* New permissible value: `Biliary Disorder`
		* New permissible value: `Bronchitis`
		* New permissible value: `Calcium Channel Blockers`
		* New permissible value: `Cataracts`
		* New permissible value: `Celiac Disease`
		* New permissible value: `Cerebrovascular Disease`
		* New permissible value: `Chronic Fatigue Syndrome`
		* New permissible value: `Clonal Hematopoiesis`
		* New permissible value: `CNS Infection`
		* New permissible value: `Common Variable Immunodeficiency`
		* New permissible value: `Congestive Heart Failure (CHF)`
		* New permissible value: `Connective Tissue Disorder`
		* New permissible value: `COPD`
		* New permissible value: `Coronary Artery Disease`
		* New permissible value: `Crohn's Disease`
		* New permissible value: `Cryptogenic Organizing Pneumonia`
		* New permissible value: `Dyslipidemia`
		* New permissible value: `GERD`
		* New permissible value: `Glaucoma`
		* New permissible value: `Glycogen Storage Disease`
		* New permissible value: `Gout`
		* New permissible value: `Heart Disease`
		* New permissible value: `Hepatitis, Chronic`
		* New permissible value: `Hereditary Non-Polyposis Colon Cancer`
		* New permissible value: `Herpes`
		* New permissible value: `Herpes Simplex Virus`
		* New permissible value: `High Grade Liver Dysplastic Nodule`
		* New permissible value: `HIV/AIDS`
		* New permissible value: `Hyperglycemia`
		* New permissible value: `Hyperlipidemia`
		* New permissible value: `Inflammatory Bowel Disease`
		* New permissible value: `Interstitial Pneumonitis or ARDS`
		* New permissible value: `Intraductal Papillary Mucinous Neoplasm`
		* New permissible value: `Ischemic Heart Disease`
		* New permissible value: `ITP`
		* New permissible value: `Liver Cirrhosis (Liver Disease)`
		* New permissible value: `Low Grade Liver Dysplastic Nodule`
		* New permissible value: `Lupus`
		* New permissible value: `Myocardial Infarction`
		* New permissible value: `Neuroendocrine Tumor`
		* New permissible value: `Nevus of Ota`
		* New permissible value: `Organ Transplant (Site)`
		* New permissible value: `Osteoarthritis`
		* New permissible value: `Peptic Ulcer (Ulcer)`
		* New permissible value: `Peripheral Vascular Disease`
		* New permissible value: `Peutz-Jeghers Disease`
		* New permissible value: `Pregnancy in Patient or Partner`
		* New permissible value: `Psoriasis`
		* New permissible value: `Pulmonary Fibrosis`
		* New permissible value: `Renal Failure (Requiring Dialysis)`
		* New permissible value: `Renal Insufficiency`
		* New permissible value: `Rheumatologic Disease`
		* New permissible value: `Sleep Apnea`
		* New permissible value: `Staph Osteomyelitis`
		* New permissible value: `Thyroid Disease, Non-Cancer`
		* New permissible value: `Tyrosinemia`
		* New permissible value: `Urinary Tract Infection`
		* New deprecated value: `Chronic Hepatitis`
		* New deprecated value: `Cirrhosis`
		* New deprecated value: `HIV`
		* New deprecated value: `Reflux Disease`
	* Changes made to `timepoint_category`
		* New permissible value: `After Chemotherapy`
		* New permissible value: `After Study Enrollment`
		* New permissible value: `End of Consolidation Therapy`
		* New permissible value: `End of Treatment Course`
		* New permissible value: `End of Treatment Course 1`
		* New permissible value: `End of Treatment Course 2`
		* New permissible value: `First Complete Response`
		* New permissible value: `First Treatment`
		* New permissible value: `Post Initial Treatment`
		* New permissible value: `Prior to Adjuvant Therapy`
		* New permissible value: `Prior to Chemotherapy`
		* New permissible value: `Prior to Procurement`
		* New permissible value: `Prior to Study Enrollment`
		* New permissible value: `Prior to Study Registration`
		* New permissible value: `Progression`
		* New permissible value: `Recurrence`
		* New permissible value: `Within 2 Months After Completion of First-Course Treatment`
	* New deprecated property: `undescended_testis_corrected_age`
* Altered `pathology_detail` Entity
	* New property: `epithelioid_cell_percent_range`
	* New property: `extraocular_nodule_size`
	* New property: `extrascleral_extension_present`
	* New property: `intratubular_germ_cell_neoplasia_present`
	* New property: `spindle_cell_percent_range`
	* New property: `timepoint_category`
	* New property: `tumor_basal_diameter`
	* Changes made to `lymph_node_dissection_site`
		* New permissible value: `Retroperitoneal`
	* Changes made to `lymph_node_involved_site`
		* New permissible value: `Intra-Abdominal, NOS`
	* New deprecated property: `epithelioid_cell_percent`
	* New deprecated property: `extrascleral_extension`
	* New deprecated property: `size_extraocular_nodule`
	* New deprecated property: `spindle_cell_percent`
* Altered `sample` Entity
		* Changes made to `biospecimen_anatomic_site`
			* New permissible value: `Brainstem`
			* New permissible value: `Common Bile Duct`
			* New deprecated value: `Brain Stem`
			* New deprecated value: `Common Duct`
		* Changes made to `method_of_sample_procurement`
			* New permissible value: `Indeterminate`
			* New deprecated value: `Indeterminant`
* Altered `treatment` Entity
	* New property: `margin_distance`
	* New property: `margin_status`
	* New property: `margins_involved_site`
	* New property: `pretreatment`
	* Changes made to `drug_category`
		* New permissible value: `Platinum`
	* Changes made to `therapeutic_agents`
		* New permissible value: `Cidofovir`
		* New permissible value: `Cyanocobalamin`
		* New permissible value: `Poly ICLC`
	* Changes made to `timepoint_category`
		* New permissible value: `Adjuvant Therapy`
		* New permissible value: `Adolescence`
		* New permissible value: `Adulthood`
		* New permissible value: `After Chemotherapy`
		* New permissible value: `After Study Enrollment`
		* New permissible value: `Childhood`
		* New permissible value: `End of Treatment Course 1`
		* New permissible value: `End of Treatment Course 2`
		* New permissible value: `Follow-up`
		* New permissible value: `Initial Diagnosis`
		* New permissible value: `Last Contact`
		* New permissible value: `Post Adjuvant Therapy`
		* New permissible value: `Post Hormone Therapy`
		* New permissible value: `Post Initial Treatment`
		* New permissible value: `Post Secondary Therapy`
		* New permissible value: `Prior to Adjuvant Therapy`
		* New permissible value: `Prior to Chemotherapy`
		* New permissible value: `Prior to Study Enrollment`
		* New permissible value: `Recurrence/Progression`
		* New permissible value: `Sample Procurement`
		* New permissible value: `Within 2 Months After Completion of First-Course Treatment`
		* New permissible value: `Within 3 Months of Surgery`
		* New permissible value: `Other`
	* Changes made to `treatment_anatomic_sites`
		* New permissible value: `Epiglottis`
	* Changes made to `treatment_intent_type`
		* New permissible value: `Re-Excision`
		* New permissible value: `Salvage`
	* Changes made to `treatment_type`
		* New permissible value: `Hysterectomy, NOS`
	* New deprecated property: `therapeutic_level_achieved`

### Bugs Fixed Since Last Release

* N/A 

### Known Issues and Workarounds

* The enum `Head - Face Or Neck, Nos` has been deprecated for the property `treatment_anatomic_sites` on the `treatment` entity, but still appears in the Data Dictionary viewer. This will be resolved in a future release. <!--SV-2310--->

## v3.1.0

* __GDC Product__: GDC Data Dictionary
* __Release Date__: May 30, 2024

### New Features and Changes

* New Entity: `germline_mutation_index`
* New Entity: `submitted_expression_array`
* Altered `aggregated_somatic_mutation` Entity
	* New property: `platform`
* Altered `aligned_reads` Entity
	* Added `minimum` and/or `maximum` values to properties <!-- summarized this for readability -->
		* `average_base_quality`
		* `average_insert_size`
		* `average_read_length`
		* `contamination`
		* `contamination_error`
		* `mean_coverage`
		* `msi_score`
		* `pairs_on_diff_chr`
		* `proportion_base_mismatch`
		* `proportion_coverage_10x`
		* `proportion_coverage_30x`
		* `proportion_reads_duplicated`
		* `proportion_reads_mapped`
		* `proportion_targets_no_coverage`
		* `total_reads`
		* `tumor_ploidy`
		* `tumor_purity`
* Altered `annotated_somatic_mutation` Entity
	* New property: `platform`
* Altered `annotation` Entity
	* Changes made to `links`
		* `germline_mutation_indexes` added to subgroup
		* `submitted_expression_arrays` added to subgroup
	* Changes made to `properties`
		* New property: `germline_mutation_indexes`
		* New property: `submitted_expression_arrays`
* Altered `copy_number_auxiliary_file` Entity <!-- excluded new internal prop `curated_model_index` -->
	* Changes made to `data_format`
		* New permissible value: `PDF`
	* Changes made to `data_type`
		* New permissible value: `CNV Model`
* Altered `copy_number_segment` Entity
	* New property: `cancer_dna_fraction`
	* New property: `genome_doubling`
	* New property: `subclonal_genome_fraction`
	* New property: `tumor_ploidy`
	* New property: `tumor_purity`
* Altered `demographic` Entity
	* Added permissible values `Unknown` and/or `Not Reported` to properties
		* `country_of_birth`
		* `country_of_residence_at_enrollment`
		* `education_level`
		* `marital_status`
* Altered `diagnosis` Entity
	* New property: `calgb_risk_group`
	* New property: `tumor_of_origin`
	* Changes made to `classification_of_tumor`
		* New permissible value: `Subsequent Primary`
	* Changes made to `primary_diagnosis`
		* New permissible value: `Diffuse intrinsic pontine glioma, H3 K27M-mutant`
		* Removed permissible value: `Hairy cell leukaemia variant`
	* Changes made to `sites_of_involvement`
		* New permissible value: `Dura Mater`
		* New permissible value: `Esophagus, Lower Third`
		* New permissible value: `Esophagus, Middle Third`
		* New permissible value: `Esophagus, NOS`
		* New permissible value: `Esophagus, Upper Third`
		* New permissible value: `Lymph Node, Cervical`
		* New permissible value: `Lymph Node, Femoral`
		* New permissible value: `Lymph Node, Hilar`
		* New permissible value: `Lymph Node, Iliac-Common`
		* New permissible value: `Lymph Node, Iliac-External`
		* New permissible value: `Lymph Node, Mediastinal`
		* New permissible value: `Lymph Node, Mesenteric`
		* New permissible value: `Lymph Node, Para-Aortic`
		* New permissible value: `Lymph Node, Retroperitoneal`
		* New permissible value: `Lymph Node, Splenic`
		* New permissible value: `Lymph Node, Submandibular`
		* New permissible value: `Lymph Node, Supraclavicular`
		* New permissible value: `Pancreas Body`
		* New permissible value: `Pancreas Duct`
		* New permissible value: `Pancreas Head`
		* New permissible value: `Pancreas Tail`
		* New permissible value: `Pancreas, NOS`
		* New permissible value: `Parametrium, NOS`
		* New deprecated value: `Esophagus`
		* New deprecated value: `Pancreas`
	* Added permissible values `Unknown` and/or `Not Reported` to properties
		* `adrenal_hormone`
		* `ann_arbor_b_symptoms_described`
		* `best_overall_response`
		* `cancer_detection_method`
		* `clark_level`
		* `contiguous_organ_invaded`
		* `double_expressor_lymphoma`
		* `double_hit_lymphoma`
		* `ensat_clinical_m`
		* `ensat_pathologic_n`
		* `ensat_pathologic_stage`
		* `ensat_pathologic_t`
		* `gleason_grade_group`
		* `gleason_grade_tertiary`
		* `international_prognostic_index`
		* `margins_involved_site`
		* `masaoka_stage`
		* `melanoma_known_primary`
		* `pediatric_kidney_staging`
		* `primary_gleason_grade`
		* `secondary_gleason_grade`
		* `tumor_grade_category`
		* `ulceration_indicator`
		* `weiss_assessment_findings`
		* `weiss_assessment_score`
* Altered `exposure` Entity
	* Added permissible values `Other`, `Unknown` and/or `Not Reported` to properties
		* `alcohol_frequency`
		* `asbestos_exposure_type`
		* `chemical_exposure_type`
		* `environmental_tobacco_smoke_exposure`
		* `exposure_source`
		* `exposure_type`
		* `occupation_type`
		* `parent_with_radiation_exposure`
		* `smoking_frequency`
		* `time_between_waking_and_first_smoke`
		* `type_of_smoke_exposure`
* Altered `family_history` Entity
	* Added permissible values `Unknown` to properties
		* `relative_deceased`
		* `relative_smoker`
* Altered `filtered_copy_number_segment` Entity
	* New property: `platform`
* Altered `follow_up` Entity
	* New property: `treatment_emergent_adverse_event`
	* Changes made to `adverse_event`
		* New permissible value: `Cardiac Ischemia`
		* New permissible value: `Glomerular Filtration Rate Decreased`
		* New permissible value: `Joint Pain`
		* New permissible value: `Rash/Desquamation`
		* New permissible value: `Renal Failure`
		* New permissible value: `Ureteral Obstruction`
		* New permissible value: `Unknown`
		* New permissible value: `Not Reported`
	* Changes made to `first_event`
		* New permissible value: `Recurrence`
		* New permissible value: `Unknown`
	* Changes made to `imaging_anatomic_site`
		* New permissible value: `Pleura`
		* New permissible value: `Unknown`
		* New permissible value: `Not Reported`
	* Changes made to `imaging_findings`
		* New permissible value: `Distant Metastasis`
		* New permissible value: `Equivocal`
		* New permissible value: `Extraprostatic Extension, Localized`
		* New permissible value: `Extraprostatic Extension, Regional Lymphadenopathy`
		* New permissible value: `No Evidence of Extraprostatic Extension`
		* New permissible value: `Unknown`
	* Changes made to `imaging_type`
		* New permissible value: `Bone Scan, NOS`
		* New permissible value: `Unknown`
		* New permissible value: `Not Reported`
	* Changes made to `timepoint_category`
		* New permissible value: `Prior to Adjuvant Therapy`
		* New permissible value: `Within 2 Months After Completion of First-Course Treatment`
		* New permissible value: `Unknown`
	* Added permissible values `Unknown` and/or `Not Reported` to properties
		* `adverse_event_grade`
		* `evidence_of_progression_type`
		* `evidence_of_recurrence_type`
		* `history_of_tumor`
		* `history_of_tumor_type`
		* `peritoneal_washing_results`
		* `scan_tracer_used`
	* New deprecated property: `days_to_risk_factor`
* Altered `gene_expression` Entity
	* New property: `platform`
	* Changes made to `experimental_strategy`
		* New permissible value: `Expression Array`
* Altered `germline_mutation_calling_workflow` Entity
	* Changes made to `workflow_type`
		* New permissible value: `Strelka2 Germline`
* Altered `masked_somatic_mutation` Entity
	* New property: `platform`
* Altered `mirna_expression` Entity
	* New property: `platform`
* Altered `molecular_test` Entity
	* Changes made to `chromosomal_translocation`
		* New permissible value: `t(4;11)`
		* New permissible value: `t(9;11)`
		* New permissible value: `t(9;22)`
		* New permissible value: `t(15;17)`
	* Changes made to `gene_symbol`
		* New permissible value: `MPO`
		* New permissible value: `RAS, NOS`
	* Changes made to `laboratory_test`
		* New permissible value: `Abnormal Cells`
	* Changes made to `molecular_analysis_method`
		* New permissible value: `Immunophenotyping, NOS`
	* Changes made to `second_gene_symbol`
		* New permissible value: `MPO`
		* New permissible value: `RAS, NOS`
	* Changes made to `test_units`
		* New permissible value: `IU`
		* New permissible value: `nmol/L`
		* New permissible value: `Seconds`
		* New permissible value: `x10^6 cells/mcL`
	* Changes made to `test_value_range`
		* New permissible value: `0-25%`
		* New permissible value: `26-50%`
		* New permissible value: `51-75%`
		* New permissible value: `76-100%`
		* New permissible value: `Unknown`
		* New deprecated value: `0-25`
		* New deprecated value: `26-50`
		* New deprecated value: `51-75`
		* New deprecated value: `76-100`
	* Changes made to `timepoint_category`
		* New permissible value: `Initial Diagnosis`
		* New permissible value: `Sample Procurement`
		* New permissible value: `Unknown`
	* Added permissible values `Unknown` and/or `Not Reported` to properties
		* `aneuploidy`
		* `chromosome_arm`
		* `clonality`
		* `hpv_strain`
		* `molecular_consequence`
		* `mutation_codon`
		* `pathogenicity`
		* `staining_intensity_scale`
		* `staining_intensity_value`
		* `variant_origin`
* Altered `other_clinical_attribute` Entity
	* New property: `days_to_risk_factor`
	* New property: `hormonal_replacement_therapy_status`
	* New property: `number_of_pregnancies`
	* New property: `viral_hepatitis_serology_tests`
	* Changes made to `immunosuppressive_treatment_type`
		* New permissible value: `Immunoglobulin`
		* New permissible value: `Prednisone`
	* Changes made to `pregnancy_outcome`
		* New permissible value: `Full Term Birth, NOS`
	* Changes made to `risk_factors`
		* New permissible value: `BK Virus`
		* New permissible value: `Diverticulosis`
		* New permissible value: `Hypercholesterolemia`
		* New permissible value: `Hypertension`
		* New permissible value: `Scarlet Fever`
		* New permissible value: `Ulcerative Colitis`
	* Changes made to `timepoint_category`
		* New permissible value: `Sample Procurement`
		* New permissible value: `Unknown`
	* Added permissible values `Unknown` and/or `Not Reported` to properties
		* `eye_color`
		* `nononcologic_therapeutic_agents`
		* `oxygen_use_type`
		* `risk_factor_method_of_diagnosis`
		* `undescended_testis_corrected`
		* `undescended_testis_corrected_laterality`
		* `undescended_testis_corrected_method`
		* `undescended_testis_history`
		* `undescended_testis_history_laterality`
	* Removed `maximum` value for properties
		* `dlco_ref_predictive_percent`
		* `fev1_fvc_post_bronch_percent`
		* `fev1_ref_post_bronch_percent`
		* `fev1_fvc_pre_bronch_percent`
		* `fev1_ref_pre_bronch_percent`
	* New deprecated property: `pregnancy_count`
	* New deprecated property: `viral_hepatitis_serologies`
* Altered `pathology_detail` Entity
	* New property: `prcc_type`
	* New property: `tumor_burden`
	* Added permissible values `Unknown` and/or `Not Reported` to properties
		* `consistent_pathology_review`
		* `extracapsular_extension`
		* `extracapsular_extension_present`
		* `extranodal_extension`
		* `extrathyroid_extension`
		* `lymph_node_dissection_site`
		* `lymph_nodes_removed`
		* `measurement_type`
		* `measurement_unit`
		* `morphologic_architectural_pattern`
		* `necrosis_present`
		* `residual_tumor`
		* `residual_tumor_measurement`
		* `rhabdoid_present`
		* `sarcomatoid_present`
		* `tumor_depth_descriptor`
		* `tumor_infiltrating_lymphocytes`
		* `tumor_infiltrating_macrophages`
		* `tumor_shape`
		* `zone_of_origin_prostate`
* Altered `rna_expression_workflow` Entity
	* Changes made to `links`
		* `submitted_expression_arrays` added to subgroup
	* New property: `submitted_expression_arrays`
	* Changes made to `workflow_type`
		* New permissible value: `Expression Array Quantification`
* Altered `sample` Entity
	* New deprecated property: `tumor_code`
	* Updated definitions for properties
		* `days_to_collection`
		* `days_to_sample_procurement`
		* `biospecimen_anatomic_site`
* Altered `secondary_expression_analysis` Entity
	* New property: `platform`
* Altered `simple_germline_variation` Entity
	* Changes made to `data_format`
		* New permissible value: `gVCF`
	* Changes made to `platform`
		* New permissible value: `Illumina`
* Altered `simple_somatic_mutation` Entity
	* New property: `platform`
* Altered `somatic_aggregation_workflow` Entity
	* Changes made to `workflow_type`
		* New permissible value: `Tumor-Only Somatic Variant Merging and Masking`
* Altered `somatic_annotation_workflow` Entity
	* Changes made to `workflow_type`
		* New permissible value: `Manta Indel Annotation`
		* New permissible value: `Strelka2 Annotation`
		* New permissible value: `SvABA Indel Annotation`
* Altered `somatic_copy_number_workflow` Entity
	* Changes made to `workflow_type`
		* New permissible value: `ABSOLUTE GATK4 CNV Auto`
		* New permissible value: `ABSOLUTE GATK4 CNV Curated`
* Altered `somatic_mutation_calling_workflow` Entity
	* Changes made to `workflow_type`
		* New permissible value: `Manta Indel`
		* New permissible value: `Strelka2`
		* New permissible value: `SvABA Indel`
* Altered `somatic_mutation_index` Entity:
	* Updated definition for Entity
* Altered `structural_variant_calling_workflow` Entity
	* Changes made to `workflow_type`
		* New permissible value: `Manta`
* Altered `structural_variation` Entity
	* New property: `platform`
* Altered `treatment` Entity
	* New property: `prescribed_dose_units`
	* Changes made to `reason_treatment_not_given`
		* New permissible value: `Patient Ineligible`
	* Changes made to `treatment_type`
		* New permissible value: `Ablation or Embolization, NOS`
	* Added permissible values `Unknown` and/or `Not Reported` to properties
		* `drug_category`
		* `embolic_agent`
		* `protocol_identifier`
		* `reason_treatment_ended`
		* `residual_disease`
		* `therapeutic_target_level`
		* `timepoint_category`
		* `treatment_dose_units`

### Bugs Fixed Since Last Release

* The [GDC Data Dictionary Viewer](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/) on the [GDC Documentation Site](https://docs.gdc.cancer.gov) correctly displays links to terms for properties that were previously listed as 'null'. <!--DICT-375--->

### Known Issues and Workarounds

* The enum `Head - Face Or Neck, Nos` has been deprecated for the property `treatment_anatomic_sites` on the `treatment` entity, but still appears in the Data Dictionary viewer. This will be resolved in a future release. <!--SV-2310--->

## v3.0.0

* __GDC Product__: GDC Data Dictionary
* __Release Date__: October 30, 2023

### New Features and Changes

* New Entity: `other_clinical_attribute`
* Altered `annotation` Entity
	* Changes made to `links`
		* `other_clinical_attributes` added to subgroup
* Altered `demographic` Entity
	* New deprecated property: `premature_at_birth`
	* New deprecated property: `weeks_gestation_at_birth`
* Altered `diagnosis` Entity
	* Changes made to `morphology`
		* New permissible value: `8980/6`
	* Changes made to `primary_diagnosis`
		* New permissible value: `Diffuse Pediatric-type High Grade Glioma`
	* Changes made to `sites_of_involvement`
		* New permissible value: `Thyroid Gland Isthmus`
	* Changes made to `tumor_grade`
		* Removed deprecated value: `not reported`
	* New deprecated property `pregnant_at_diagnosis`
	* Removed property: `measurement_type`
* Altered `exposure` Entity
	* Changes made to `exposure_source`
		* New permissible value: `Home`
		* New permissible value: `Social`
		* New permissible value: `Work`
* Altered `follow_up` Entity
	* Changes made to `imaging_anatomic_site`
		* New permissible value: `Lymph Node`
	* Changes made to `imaging_type`
		* New permissible value: `Ultrasound`
	* New deprecated property: `aids_risk_factors`
	* New deprecated property: `bmi`
	* New deprecated property: `body_surface_area`
	* New deprecated property: `cd4_count`
	* New deprecated property: `cdc_hiv_risk_factors`
	* New deprecated property: `comorbidity_method_of_diagnosis`
	* New deprecated property: `comorbidities`
	* New deprecated property: `days_to_comorbidity`
	* New deprecated property: `diabetes_treatment_type`
	* New deprecated property: `dlco_ref_predictive_percent`
	* New deprecated property: `eye_color`
	* New deprecated property: `fev1_fvc_post_bronch_percent`
	* New deprecated property: `fev1_fvc_pre_bronch_percent`
	* New deprecated property: `fev1_ref_post_bronch_percent`
	* New deprecated property: `fev1_ref_pre_bronch_percent`
	* New deprecated property: `haart_treatment_indicator`
	* New deprecated property: `height`
	* New deprecated property: `hepatitis_sustained_virological_response`
	* New deprecated property: `hiv_viral_load`
	* New deprecated property: `hormonal_contraceptive_type`
	* New deprecated property: `hormonal_contraceptive_use`
	* New deprecated property: `hormone_replacement_therapy_type`
	* New deprecated property: `hysterectomy_margins_involved`
	* New deprecated property: `hysterectomy_type`
	* New deprecated property: `immunosuppressive_treatment_type`
	* New deprecated property: `menopause_status`
	* New deprecated property: `nadir_cd4_count`
	* New deprecated property: `pancreatitis_onset_year`
	* New deprecated property: `pregnancy_count`
	* New deprecated property: `pregnancy_outcome`
	* New deprecated property: `reflux_treatment_type`
	* New deprecated property: `risk_factor_method_of_diagnosis`
	* New deprecated property: `risk_factor_treatment`
	* New deprecated property: `risk_factors`
	* New deprecated property: `undescended_testis_corrected`
	* New deprecated property: `undescended_testis_corrected_age`
	* New deprecated property: `undescended_testis_corrected_laterality`
	* New deprecated property: `undescended_testis_corrected_method`
	* New deprecated property: `undescended_testis_history`
	* New deprecated property: `undescended_testis_history_laterality`
	* New deprecated property: `viral_hepatitis_serologies`
	* New deprecated property: `weight`
* Altered `molecular_test` Entity
	* New property: `staining_intensity_value`
	* Changes made to `antigen`
		* New permissible value: `CD43`
	* Changes made to `molecular_analysis_method`
		* New permissible value: `Microscopy, NOS`
	* Changes made to `test_result`
		* New permissible value: `Mitotic Count Reported`
		* New permissible value: `Staining Intensity Value Reported`
	* Changes made to `test_units`
		* Removed deprecated value: `count x10^9/L`
	* Changes made to `timepoint_category`
		* New permissible value: `Prior to Treatment`
* Altered `pathology_detail` Entity
	* Changes made to `lymph_node_involved_site`
		* New permissible value: `Cervical, Central`
		* New permissible value: `Cervical, Lateral`
		* New permissible value: `Cervical, NOS`
		* New deprecated value: `Cervical`
	* Changes made to `measurement_type`
		* New permissible value: `Echographic`
	* New property: `days_to_pathology_detail`
	* New property: `extracapsular_extension_present`
	* New property: `extrathyroid_extension`
	* New property: `histologic_progression_type`
	* New property: `micrometastasis_present`
	* Removed property: `tumor_measurement_method`
* Altered `read_group` Entity
	* Changes made to `instrument_model`
		* New permissible value: `Illumina NovaSeq X`
* Altered `sample` Entity
	* Changes made to `biospecimen_anatomic_site`
		* New permissible value: `Head, Face or Neck, NOS`
		* New deprecated value: `Head - Face Or Neck, Nos`
* Altered `submitted_aligned_reads` Entity
	* Changes made to `experimental_strategy`
		* New deprecated value: `scRNA-Seq`
* Altered `treatment` Entity
	* Changes made to `timepoint_category`
		* New permissible value: `First Treatment`
		* New permissible value: `Prior to Treatment`
	* Changes made to `treatment_anatomic_sites`
		* New deprecated value: `Head - Face Or Neck, Nos`
		* New permissible value: `Head, Face or Neck, NOS`
	* New property: `radiosensitizing_agent`
	* New property: `reason_treatment_not_given`
	* New property: `therapeutic_levels_achieved`

### Bugs Fixed Since Last Release

* The [GDC Data Dictionary Viewer](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/) on the [GDC Documentation Site](https://docs.gdc.cancer.gov) correctly displays links to terms for properties that were previously listed as 'null'.

### Known Issues and Workarounds

* The enum `Head - Face Or Neck, Nos` has been deprecated for the property `treatment_anatomic_sites` on the `treatment` entity, but still appears in the Data Dictionary viewer. This will be resolved in a future release.

## v2.6.6

* __GDC Product__: GDC Data Dictionary
* __Release Date__: June 16, 2023

### New Features and Changes

* Updated CDE links to point to newer version of caDSR website.
* Altered `aligned_reads` Entity
	* New property: `wgs_coverage`
* Altered `case` Entity
	* New required property: `disease_type`
	* New required property: `primary_site`
	* Changes made to `primary_site`
		* New permissible value: `Not Applicable`
* Altered `demographic` Entity
	* New property: `country_of_birth`
	* New property: `education_level`
	* New property: `marital_status`
* Altered `diagnosis` Entity
	* New deprecated property: `metastasis_at_diagnosis_site`
	* New property: `clark_level`
	* New property: `contiguous_organ_invaded`
	* New property: `ensat_clinical_m`
	* New property: `ensat_pathologic_n`
	* New property: `ensat_pathologic_stage`
	* New property: `ensat_pathologic_t`
	* New property: `first_symptom_longest_duration`
	* New property: `measurement_type`
	* New property: `melanoma_known_primary`
	* New property: `tumor_burden`
	* New property: `ulceration_indicator`
	* Changes made to `sites_of_involvement`
		* New permissible value: `Ascites`
		* New permissible value: `Common or Superficial Femoral Vein`
		* New permissible value: `Distant Nodes`
		* New permissible value: `Distant Organ`
		* New permissible value: `Gastrointestinal Tract`
		* New permissible value: `Great Blood Vessel`
		* New permissible value: `Groin`
		* New permissible value: `Head, Face or Neck, NOS`
		* New permissible value: `Inferior Vena Cava`
		* New permissible value: `Lymph Node, Axillary`
		* New permissible value: `Lymph Node, Inguinal`
		* New permissible value: `Lymph Node, Regional`
		* New permissible value: `Lymph Node, Subcarinal`
		* New permissible value: `Mediastinum`
		* New permissible value: `Pelvic Vein, Common, External or Iliac`
		* New permissible value: `Peritoneal Cavity`
		* New permissible value: `Renal Vein`
		* New permissible value: `Scalp`
		* New permissible value: `Spinal Cord`
		* New permissible value: `Urethra`
		* New permissible value: `Vertebral Canal`
		* New permissible value: `Vulva, NOS`
* Altered `exposure` Entity
	* New property: `exposure_duration_hrs_per_day`
	* Changes made to `exposure_type`
		* New permissible value: `Dust, NOS`
* Altered `family_history` Entity
	* New property: `relative_deceased`
	* New property: `relative_smoker`
* Altered `follow_up` Entity
	* New property: `discontiguous_lesion_count`
	* Changes made to `comorbidities`
		* New permissible value: `Pneumonia, NOS`
	* Changes made to `first_event`
		* New permissible value: `Presented with Metastases`
	* Changes made to `risk_factors`
		* New permissible value: `Pneumonia, NOS`
	* Changes made to `timepoint_category`
		* New permissible value: `Within 3 Months of Surgery`
* Altered `masked_methylation_array` Entity
	* Changes made to `platform`
		* New permissible value: `Illumina Methylation Epic v2`
* Altered `methylation_beta_value` Entity
	* Changes made to `platform`
		* New permissible value: `Illumina Methylation Epic v2`
* Altered `molecular_test` Entity
	* New property: `staining_intensity_scale`
	* Changes made to `gene_symbol` and `second_gene_symbol`
		* New permissible value: `ACACA`
		* New permissible value: `BRD1`
		* New permissible value: `BTBD18`
		* New permissible value: `CBFA2T2`
		* New permissible value: `CEP164`
		* New permissible value: `CEP170B`
		* New permissible value: `CTDP1`
		* New permissible value: `DOT1L`
		* New permissible value: `FRYL`
		* New permissible value: `GLIS2`
		* New permissible value: `HMGB3`
		* New permissible value: `HNRNPH1`
		* New permissible value: `INO80D`
		* New permissible value: `JARID2`
		* New permissible value: `PHF23`
		* New permissible value: `PIM3`
		* New permissible value: `PTP4A1`
		* New permissible value: `RPS15`
		* New permissible value: `SETD1A`
		* New permissible value: `SLC66A2 (aka PQLC1)`
		* New permissible value: `TOP2B`
		* New permissible value: `UBB`
		* New permissible value: `ZEB2`
	* Changes made to `test_value_range`
		* New permissible value: `<10%`
		* New permissible value: `10-19%`
		* New permissible value: `20-29%`
		* New permissible value: `30-39%`
		* New permissible value: `40-49%`
		* New permissible value: `50-59%`
		* New permissible value: `60-69%`
		* New permissible value: `70-79%`
		* New permissible value: `80-89%`
		* New permissible value: `90-99%`
		* New permissible value: `<1%`
		* New permissible value: `1-49%`
		* New permissible value: `>=50%`
	* Changes made to `timepoint_category`
		* New permissible value: `Preoperative`
* Altered `pathology_detail` Entity
	* New property: `epithelioid_cell_percent`
	* New property: `measurement_type`
	* New property: `measurement_unit`
	* New property: `spindle_cell_percent`
	* New property: `tumor_depth_descriptor`
	* New property: `tumor_depth_measurement`
	* New property: `tumor_infiltrating_lymphocytes`
	* New property: `tumor_infiltrating_macrophages`
	* New property: `tumor_length_measurement`
	* New property: `tumor_measurement_method`
	* New property: `tumor_shape`
	* New property: `tumor_width_measurement`
	* Changes made to `additional_pathology_findings`
		* New permissible value: `Extravascular Matrix Loops`
		* New permissible value: `Other Complex Extravascular Matrix Patterns`
		* New permissible value: `Poorly Differentiated`
		* New permissible value: `Well Differentiated`
* Altered `raw_methylation_array` Entity
	* Changes made to `platform`
		* New permissible value: `Illumina Methylation Epic v2`
* Altered `sample` Entity
	* Changes made to `preservation_method`
		* New permissible value: `EDTA`
	* Changes made to `specimen_type`
		* New permissible value: `Derived Cell Lines and Sorted Cells`
* Altered `treatment` Entity
	* New deprecated property: `treatment_anatomic_site`
	* New property: `treatment_anatomic_sites`
	* Changes made to `protocol_identifier`
		* New permissible value: `P9407`


### Bugs Fixed Since Last Release

* The [GDC Data Dictionary Viewer](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/) on the [GDC Documentation Site](https://docs.gdc.cancer.gov) correctly displays permissible values for array-type properties.

## v2.6.0

* __GDC Product__: GDC Data Dictionary
* __Release Date__: February 2, 2023

### New Features and Changes

* Altered `aliquot` Entity
	* New deprecated property: `analyte_type_id`
* Altered `analyte` Entity
	* New deprecated property: `analyte_type_id`
* Altered `annotation` Entity
	* Changes made to `links`
		* `copy_number_auxiliary_files` added to subgroup
	* New property: `copy_number_auxiliary_files`
* Altered `exposure` Entity
	* New deprecated property: `coal_dust_exposure`
	* New property: `alcohol_frequency`
	* New property: `chemical_exposure_type`
	* New property: `occupation_duration_years`
	* New property: `occupation_type`
	* Changes made to `exposure_type`
		* New permissable value: `Coal Dust`
	* Changes made to `tobacco_smoking_status`
		* New permissable value: `Current Reformed Smoker, Duration Not Specified`
		* New permissable value: `Current Reformed Smoker for < or = 15 yrs`
		* New permissable value: `Current Reformed Smoker for > 15 yrs`
		* New permissable value: `Current Smoker`
		* New permissable value: `Lifelong Non-Smoker`
		* New permissable value: `Smoker at Diagnosis`
		* New permissable value: `Smoking history not documented`
		* New deprecated value: `1`
		* New deprecated value: `2`
		* New deprecated value: `3`
		* New deprecated value: `4`
		* New deprecated value: `5`
		* New deprecated value: `6`
		* New deprecated value: `7`
* Altered `diagnosis` Entity
	* New property: `cancer_detection_method`
	* New property: `fab_morphology_code`
	* New property: `gleason_score`
	* New property: `pediatric_kidney_staging`
	* New property: `sites_of_involvement_count`
	* New property: `tumor_grade_category`
	* New property: `uicc_staging_system_edition`
	* New property: `weiss_assessment_findings`
	* Changes made to `inpc_grade`
		* New permissable value: `Undifferentiated or Poorly Differentiated`
	* Changes made to `sites_of_involvement`
		* Changes made to `items`
			* New permissable value: `Brain, Brain stem`
			* New permissable value: `Brain, Cerebellum, NOS`
			* New permissable value: `Brain, Cerebrum`
			* New permissable value: `Brain, Frontal lobe`
			* New permissable value: `Brain, Occipital lobe`
			* New permissable value: `Brain, Parietal`
			* New permissable value: `Brain, Temporal lobe`
			* New permissable value: `Brain, Ventricle, NOS`
			* New permissable value: `Central Lung`
			* New permissable value: `Gastrosplenic Ligament`
			* New permissable value: `Parametrium, Left`
			* New permissable value: `Parametrium, Right`
			* New permissable value: `Pelvic Lymph Node(s)`
			* New permissable value: `Peri-aortic Lymph Node(s)`
			* New permissable value: `Peripheral Lung`
			* New permissable value: `Thorax, NOS`
* Altered `family_history` Entity
	* Changes made to `relationship_type`
		* New permissable value: `First Degree Relative, NOS`
* Altered `follow_up` Entity
	* New deprecated property: `comorbidity`
	* New deprecated property: `risk_factor`
	* New property: `comorbidities`
	* New property: `days_to_first_event`
	* New property: `days_to_risk_factor`
	* New property: `evidence_of_progression_type`
	* New property: `first_event`
	* New property: `imaging_anatomic_site`
	* New property: `imaging_findings`
	* New property: `imaging_suv`
	* New property: `imaging_suv_max`
	* New property: `peritoneal_washing_results`
	* New property: `pregnancy_count`
	* New property: `risk_factors`
	* New property: `risk_factor_method_of_diagnosis`
	* New property: `timepoint_category`
	* New property: `year_of_follow_up`
	* Changes made to `evidence_of_recurrence_type`
		* New permissable value: `Histologic Confirmation`
	* Changes made to `history_of_tumor_type`
		* New permissable value: `Colorectal Cancer`
		* New permissable value: `Lower Grade Glioma`
	* Changes made to `pregnancy_outcome`
		* New permissable value: `Spontaneous Abortion`
	* Changes made to `progression_or_recurrence_type`
		* New permissable value: `Locoregional`
* Altered `gene_expression` Entity
	* Changes made to `experimental_strategy`
		* New permissable value: `m6A MeRIP-Seq`
* Altered `molecular_test` Entity
	* New property: `aneuploidy`
	* New property: `chromosomal_translocation`
	* New property: `chromosome_arm`
	* New property: `mutation_codon`
	* New property: `test_value_range`
	* New property: `timepoint_category`
	* Changes made to `antigen`
		* New permissable value: `Immunoglobulin, Cytoplasmic`
		* New permissable value: `Immunoglobulin, Surface`
	* Changes made to `laboratory_test`
		* New permissable value: `Blast Count`
		* New permissable value: `Metaphase Nucleus Count`
		* New permissable value: `Minimal Residual Disease`
		* New permissable value: `Monocytes`
	* Changes made to `test_result`
		* New permissable value: `Amplified`
		* New permissable value: `Elevated`
		* New permissable value: `Not Amplified`
	* Changes made to `variant_type`
		* New permissable value: `Fusion`
		* New permissable value: `Genetic Polymorphism`
		* New permissable value: `Internal Tandem Duplication`
		* New permissable value: `Mutation, NOS`
* Altered `pathology_detail` Entity
	* New property: `extracapsular_extension`
	* New property: `extranodal_extension`
	* New property: `extrascleral_extension`
	* New property: `lymph_node_dissection_method`
	* New property: `lymph_node_dissection_site`
	* New property: `lymph_nodes_removed`
	* New property: `percent_tumor_nuclei`
	* New property: `residual_tumor_measurement`
	* Changes made to `additional_pathology_findings`
		* New permissable value: `Bone marrow concordant histology`
		* New permissable value: `Bone marrow discordant histology`
	* Changes made to `dysplasia_type`
		* New permissable value: `Esophageal Mucosa Columnar Dysplasia`
	* Changes made to `lymph_node_involved_site`
		* New permissable value: `Aortic`
		* New permissable value: `Pelvis, NOS`
* Altered `read_group` Entity
	* Changes made to `target_capture_kit`
		* New property: `Custom SureSelect CGCI-BLGSP Panel - 7.8 Mb`
* Altered `sample` Entity
	* New required property: `preservation_method`
	* New required property: `specimen_type`
	* New required property: `tumor_descriptor`
	* New deprecated property: `composition`
	* New deprecated property: `sample_type`
	* New deprecated property: `sample_type_id`
	* New property: `specimen_type`
	* Changes made to `tumor_descriptor`
		* New permissable value: `New Primary`
* Altered `simple_germline_variation` Entity
	* Changes made to `experimental_strategy`
		* New permissable value: `Genotyping Array`
* Altered `somatic_copy_number_workflow` Entity
	* Changes made to `workflow_type`
		* New permissable value: `ABSOLUTE LiftOver`
		* New permissable value: `ASCAT3`
		* New permissable value: `GATK4 CNV`
* Altered `structural_variant_calling_workflow` Entity
	* Changes made to `workflow_type`
		* New permissable value: `SvABA`
* Altered `submitted_genotyping_array` Entity
	* Changed `aliquot` link from `one-to-one` to `many-to-one`
* Altered `treatment` Entity
	* New property: `residual_disease`
	* New property: `timepoint_category`
	* New property: `treatment_duration`
	* New property: `treatment_outcome_duration`
	* New property: `therapeutic_level_achieved`
	* New property: `therapeutic_target_level`
	* Changes made to `initial_disease_status`
		* New permissable value: `Persistent Disease`
		* New permissable value: `Refractory Disease`
	* Changes made to `protocol_identifier`
		* New permissable value: `901`
		* New permissable value: `911`
		* New permissable value: `914`
		* New permissable value: `925`
		* New permissable value: `935`
		* New permissable value: `937`
		* New permissable value: `3881`
		* New permissable value: `3891`
		* New permissable value: `4941`
		* New permissable value: `8605`
		* New permissable value: `9047`
		* New permissable value: `9082`
		* New permissable value: `9340`
		* New permissable value: `9341`
		* New permissable value: `9342`
		* New permissable value: `9343`
		* New permissable value: `9464`
		* New permissable value: `9640`
		* New permissable value: `9906`
		* New permissable value: `321P2`
		* New permissable value: `321P3`
		* New permissable value: `323P`
		* New permissable value: `A3961`
		* New permissable value: `A3973`
		* New permissable value: `AADM01P1`
		* New permissable value: `AALL0031`
		* New permissable value: `AALL0232`
		* New permissable value: `AALL0331`
		* New permissable value: `AALL03B1`
		* New permissable value: `AALL0434`
		* New permissable value: `AALL07P4`
		* New permissable value: `AALL08P1`
		* New permissable value: `AALL1131`
		* New permissable value: `AAML00P2`
		* New permissable value: `AAML03P1`
		* New permissable value: `AAML0531`
		* New permissable value: `AAML0631`
		* New permissable value: `AAML1031`
		* New permissable value: `AB9804`
		* New permissable value: `ACCL0331`
		* New permissable value: `ACCL0431`
		* New permissable value: `ACCL05C1`
		* New permissable value: `ACCL0934`
		* New permissable value: `ACCL1031`
		* New permissable value: `ADVL0018`
		* New permissable value: `ADVL0212`
		* New permissable value: `ADVL0214`
		* New permissable value: `ADVL0215`
		* New permissable value: `ADVL0421`
		* New permissable value: `ADVL0524`
		* New permissable value: `ADVL0525`
		* New permissable value: `ADVL06B1`
		* New permissable value: `ADVL0714`
		* New permissable value: `ADVL0812`
		* New permissable value: `ADVL0813`
		* New permissable value: `ADVL0821`
		* New permissable value: `ADVL0911`
		* New permissable value: `ADVL0912`
		* New permissable value: `ADVL0918`
		* New permissable value: `ADVL0921`
		* New permissable value: `ADVL1011`
		* New permissable value: `ADVL1111`
		* New permissable value: `ADVL1112`
		* New permissable value: `ADVL1115`
		* New permissable value: `ADVL1213`
		* New permissable value: `ADVL1412`
		* New permissable value: `AEPI07N1`
		* New permissable value: `ALTE03N1`
		* New permissable value: `ALTE05N1`
		* New permissable value: `ANBL0032`
		* New permissable value: `ANBL00B1`
		* New permissable value: `ANBL00P1`
		* New permissable value: `ANBL00P3`
		* New permissable value: `ANBL02P1`
		* New permissable value: `ANBL0321`
		* New permissable value: `ANBL0322`
		* New permissable value: `ANBL0421`
		* New permissable value: `ANBL0531`
		* New permissable value: `ANBL0532`
		* New permissable value: `ANBL0621`
		* New permissable value: `ANBL0931`
		* New permissable value: `ANBL09P1`
		* New permissable value: `ANBL1021`
		* New permissable value: `ANBL1221`
		* New permissable value: `ANUR1131`
		* New permissable value: `AOST0121`
		* New permissable value: `AOST0331`
		* New permissable value: `AOST06B1`
		* New permissable value: `AOST06P1`
		* New permissable value: `AREN03B2`
		* New permissable value: `B003`
		* New permissable value: `B903`
		* New permissable value: `B947`
		* New permissable value: `B954`
		* New permissable value: `B973`
		* New permissable value: `BCM`
		* New permissable value: `CCG2961`
		* New permissable value: `D9902`
		* New permissable value: `E04`
		* New permissable value: `E18`
		* New permissable value: `GBCTTO/99`
		* New permissable value: `GLATO 2006`
		* New permissable value: `I03`
		* New permissable value: `IHRT`
		* New permissable value: `INT-0133`
		* New permissable value: `N891`
		* New permissable value: `NWTS-4`
		* New permissable value: `NWTS-5`
		* New permissable value: `OSTEO 2006`
		* New permissable value: `Not Reported`
		* New permissable value: `P9462`
		* New permissable value: `P9641`
		* New permissable value: `P9754`
		* New permissable value: `P9851`
		* New permissable value: `P9906`
		* New permissable value: `P9963`
		* New permissable value: `R9702`
		* New permissable value: `S31`
		* New permissable value: `S921`
		* New permissable value: `STB`
	* Changes made to `treatment_anatomic_site`
		* New permissable value: `Locoregional Site`
	* Changes made to `treatment_intent_type`
		* New permissable value: `Consolidation Therapy`
		* New permissable value: `First-Line Therapy`
		* New permissable value: `Induction`
		* New permissable value: `Radiation Boost`
	* Changes made to `treatment_outcome`
		* New permissable value: `Normalization of Tumor Markers`
	* Changes made to `treatment_type`
		* New permissable value: `Biopsy, Excisional`
		* New permissable value: `Biopsy, Incisional`
		* New permissable value: `Distal Pancreatectomy`
		* New permissable value: `Surgery, NOS`
		* New permissable value: `Surgery, Minimally Invasive`
		* New permissable value: `Surgery, Open`
		* New permissable value: `Total Pancreatectomy`
		* New permissable value: `Whipple`
		* New deprecated value: `Cryoablation`
		* New deprecated value: `Surgery`
	* Changes made to `therapeutic_agents`
		* New permissable value: `TLR9 Agonist SD-101`
* New Entity: `copy_number_auxiliary_file`

### Known Issues and Workarounds

* The [GDC Data Dictionary Viewer](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/) on the [GDC Documentation Site](https://docs.gdc.cancer.gov) does not currently display permissible values for array type properties. This does not impact submission of permissible values for these properties. Data submitters that would like to submit data for these properties can contact the GDC Helpdesk (support@nci-gdc.datacommons.io) for a list of permissible values for the affected properties. This will be addressed in a future release.

## v2.5.0

* __GDC Product__: GDC Data Dictionary
* __Release Date__: July 8, 2022

### New Features and Changes

* Altered `diagnosis` Entity
	* New property: `double_expressor_lymphoma`
	* New property: `double_hit_lymphoma`
	* New property: `max_tumor_bulk_site`
	* New property: `uicc_clinical_m`
	* New property: `uicc_clinical_n`
	* New property: `uicc_clinical_stage`
	* New property: `uicc_clinical_t`
	* New property: `uicc_pathologic_m`
	* New property: `uicc_pathologic_n`
	* New property: `uicc_pathologic_stage`
	* New property: `uicc_pathologic_t`
	* Changes made to `ajcc_pathologic_n`
		* New permissible value: `N2mi`
	* Changes made to `ajcc_pathologic_t`
		* New permissible value: `T1c2`
	* Changes made to `classification_of_tumor`
		* New permissible value: `Prior primary`
		* New permissible value: `Synchronous primary`
	* Changes made to `metastasis_at_diagnosis_site`
		* New permissible value: `Spleen`
		* New permissible value: `Stomach`
	* Changes made to `morphology`
		* New permissible value: `8211/6`
		* New permissible value: `8460/6`
		* New permissible value: `8520/6`
		* New permissible value: `9715/3`
	* Changes made to `primary_diagnosis`
		* New permissible value: `Breast implant-associated anaplastic large cell lymphoma`
		* New permissible value: `Indolent T-cell lymphoproliferative disorder of gastrointestinal tract`
	* Changes made to `sites_of_involvement`
		* New permissible value: `Abdomen`
		* New permissible value: `Adrenal gland, NOS`
		* New permissible value: `Adnexa`
		* New permissible value: `Anus, NOS`
		* New permissible value: `Appendix`
		* New permissible value: `Bladder, NOS`
		* New permissible value: `Bone, NOS`
		* New permissible value: `Bone marrow`
		* New permissible value: `Brain, NOS`
		* New permissible value: `Breast, NOS`
		* New permissible value: `Bronchus`
		* New permissible value: `Buccal mucosa`
		* New permissible value: `Central nervous system`
		* New permissible value: `Cerebrospinal fluid`
		* New permissible value: `Chest wall`
		* New permissible value: `Colon, NOS`
		* New permissible value: `Diaphragm`
		* New permissible value: `Ear`
		* New permissible value: `Epididymis`
		* New permissible value: `Esophagus`
		* New permissible value: `Eye`
		* New permissible value: `Gallbladder`
		* New permissible value: `Heart`
		* New permissible value: `Kidney, NOS`
		* New permissible value: `Larynx, NOS`
		* New permissible value: `Leptomeninges`
		* New permissible value: `Liver`
		* New permissible value: `Lower lobe, lung`
		* New permissible value: `Lung, NOS`
		* New permissible value: `Lymph node, NOS`
		* New permissible value: `Mandible`
		* New permissible value: `Maxilla`
		* New permissible value: `Mediastinal soft tissue`
		* New permissible value: `Mesentery`
		* New permissible value: `Middle lobe, lung`
		* New permissible value: `Mouth`
		* New permissible value: `Nasopharynx`
		* New permissible value: `Nasal soft tissue`
		* New permissible value: `Neck`
		* New permissible value: `Ocular orbits`
		* New permissible value: `Oropharynx`
		* New permissible value: `Pancreas`
		* New permissible value: `Parotid gland`
		* New permissible value: `Pelvis`
		* New permissible value: `Pericardium`
		* New permissible value: `Perineum`
		* New permissible value: `Peri-orbital soft tissue`
		* New permissible value: `Peripheral blood`
		* New permissible value: `Peripheral nervous system`
		* New permissible value: `Peritoneum, NOS`
		* New permissible value: `Pleura`
		* New permissible value: `Prostate`
		* New permissible value: `Rectum`
		* New permissible value: `Retroperitoneum`
		* New permissible value: `Salivary gland`
		* New permissible value: `Sigmoid colon`
		* New permissible value: `Sinus`
		* New permissible value: `Skin`
		* New permissible value: `Small intestine`
		* New permissible value: `Soft tissue`
		* New permissible value: `Spleen`
		* New permissible value: `Stomach`
		* New permissible value: `Testes`
		* New permissible value: `Thyroid`
		* New permissible value: `Tongue`
		* New permissible value: `Tonsil`
		* New permissible value: `Trachea`
		* New permissible value: `Transplanted kidney`
		* New permissible value: `Transverse colon`
		* New permissible value: `Upper lobe, lung`
		* New permissible value: `Vagina`
		* New permissible value: `Vertebrae`
	* Changes made to `tumor_depth`
		* New property: `minimum`
* Altered `submitted_unaligned_reads` Entity
	* Changes made to `experimental_strategy`
		* New permissible value: `m6A MeRIP-Seq`
		* Removed permissible value: `m6A RNA Methylation`
* Altered `pathology_detail` Entity
	* New property: `tumor_level_prostate`
	* New property: `zone_of_origin_prostate`
	* Changes made to `additional_pathology_findings`
		* New permissible value: `Percent follicular component <= 10%`
		* New permissible value: `Percent follicular component > 10%`
* Altered `exposure` Entity
	* New property: `asbestos_exposure_type`
	* New property: `age_at_last_exposure`
	* New property: `exposure_source`
	* New property: `use_per_day`
	* Removed property: `marijuana_use_per_week`
	* Removed property: `smokeless_tobacco_quit_age`
	* Removed property: `tobacco_use_per_day`
	* Changes made to `alcohol_intensity`
		* New permissible value: `Social Drinker`
	* Added maximum to `exposure_duration_years`
	* Changes made to `exposure_type`
		* New permissible value: `Asbestos`
		* New permissible value: `Chemical`
		* New permissible value: `Radon`
		* New permissible value: `Respirable Crystalline Silica`
		* New permissible value: `Smokeless Tobacco`
* Altered `case` Entity
	* Added min/max to `days_to_consent`
* Altered `analyte` Entity
	* Changes made to `analyte_type`
		* New permissible value: `m6A Enriched RNA`
* Altered `read_group` Entity
	* Changes made to `library_strategy`
		* New permissible value: `m6A MeRIP-Seq`
		* Removed permissible value: `m6A RNA Methylation`
	* Changes made to `target_capture_kit`
		* New permissible value: `Custom SeqCap EZ BeatAML Panel - 12.5 Mb`
		* New permissible value: `Custom Twist MP2PRT-WT Panel - 52 Kb`
* Altered `aligned_reads` Entity
	* Changes made to `experimental_strategy`
		* New permissible value: `m6A MeRIP-Seq`
* Altered `follow_up` Entity
	* Changes made to `comorbidity`
		* New permissible value: `CNS Infection`
		* New permissible value: `EBV Lymphoproliferation`
		* New permissible value: `Hodgkin Lymphoma`
		* New permissible value: `Lymphamatoid Papulosis`
		* New permissible value: `Methicillin-Resistant Staphylococcus aureus (MRSA)`
		* New permissible value: `Staph Osteomyelitis`
		* New permissible value: `Thyroid Disease, Non-Cancer`
		* New permissible value: `Urinary Tract Infection`
	* Changes made to `diabetes_treatment_type`
		* New permissible value: `Linagliptin`
	* Changes made to `risk_factor`
		* New permissible value: `Adenomyosis`
		* New permissible value: `Human Herpesvirus-6 (HHV-6)`
		* New permissible value: `Human Herpesvirus-8 (HHV-8)`
	* Changes made to `undescended_testis_corrected_age`
		* New property: `maximum`
* Altered `molecular_test` Entity
	* New property: `hpv_strain`
	* Changes made to `antigen`
		* New permissible value: `TAG-72`
	* Changes made to `gene_symbol`
		* New permissible value: `TLR2`
	* Changes made to `laboratory_test`
		* New permissible value: `Erythrocyte Sedimentation Rate`
		* New permissible value: `Prothrombin Time`
	* Changes made to `second_gene_symbol`
		* New permissible value: `TLR2`
	* Changes made to `test_result`
		* New permissible value: `Stable`
	* Changes made to `test_units`
		* New permissible value: `cells/mL`
		* New permissible value: `mcg/L`
		* New permissible value: `mg/24 hr`
* Altered `somatic_annotation_workflow` Entity
	* Changes made to `workflow_type`
		* New permissible value: `GATK4 MuTect2 Tumor-Only Annotation`
* Altered `treatment` Entity
	* New property: `clinical_trial_indicator`
	* New property: `course_number`
	* New property: `drug_category`
	* New property: `embolic_agent`
	* New property: `lesions_treated_number`
	* New property: `number_of_fractions`
	* New property: `prescribed_dose`
	* New property: `protocol_identifier`
	* New property: `treatment_dose_max`
	* `route_of_administration`changed to array type.
	* Changes made to `treatment_dose_units`
		* New permissible value: `AUC`
		* New permissible value: `g/day`
		* New permissible value: `g/m2`
		* New permissible value: `IU/kg`
		* New permissible value: `IU/mg`
		* New permissible value: `mCi`
		* New permissible value: `mEq`
		* New permissible value: `mg/day`
		* New permissible value: `mg/dL`
		* New permissible value: `mg/kg`
		* New permissible value: `mg/kg/day`
		* New permissible value: `mg/m2`
		* New permissible value: `mg/m2/day`
		* New permissible value: `mg/m2/wk`
		* New permissible value: `mg/mL`
		* New permissible value: `mg/wk`
		* New permissible value: `mIU`
		* New permissible value: `mL`
		* New permissible value: `ug`
		* New permissible value: `ug/m2`
		* New permissible value: `Wafer`
	* Changes made to `treatment_type`
		* New permissible value: `Peptide Receptor Radionuclide Therapy (PRRT)`
	* Changes made to `therapeutic_agents`
		* New permissible value: `Canakinumab`
		* New permissible value: `Rivaroxaban`
* Altered `somatic_mutation_calling_workflow` Entity
	* Changes made to `workflow_type`
		* New permissible value: `GATK4 MuTect2 Tumor-Only`
* Altered `submitted_aligned_reads` Entity
	* Changes made to `experimental_strategy`
		* New permissible value: `m6A MeRIP-Seq`
		* Removed permissible value: `m6A RNA Methylation`

### Known Issues and Workarounds

* The `mitotic_count` field in `diagnosis` is erroneously set as "deprecated" and does not appear in the dictionary viewer.  This field can be uploaded successfully without issue and will appear in the dictionary viewer at a later release. <!--SV-1939-->

## v2.4.1

* __GDC Product__: GDC Data Dictionary
* __Release Date__: August 23, 2021

### New Features and Changes

* Altered `pathology_detail` Entity
	* Changes made to `additional_pathology_findings`
		* New permissible value: `Adenomyosis`
		* New permissible value: `Atrophic endometrium`
		* New permissible value: `Atypical hyperplasia/Endometrial intraepithelial neoplasia (EIN)`
		* New permissible value: `Autoimmune atrophic chronic gastritis`
		* New permissible value: `Asbestos bodies`
		* New permissible value: `Benign endocervical polyp`
		* New permissible value: `Bilateral ovaries with endometriotic cyst and surface adhesions`
		* New permissible value: `Carcinoma in situ`
		* New permissible value: `Cirrhosis`
		* New permissible value: `Clostridioides difficile (c. diff)`
		* New permissible value: `Colonization; bacterial`
		* New permissible value: `Colonization; fungal`
		* New permissible value: `Cyst(s)`
		* New permissible value: `Diffuse and early nodular diabetic glomerulosclerosis`
		* New permissible value: `Dysplasia; high grade`
		* New permissible value: `Dysplasia; low grade`
		* New permissible value: `Endometrial polyp`
		* New permissible value: `Endometriosis`
		* New permissible value: `Endometroid carcinoma with local mucinous differentiation`
		* New permissible value: `Endosalpingiosis`
		* New permissible value: `Epithelial dysplasia`
		* New permissible value: `Epithelial hyperplasia`
		* New permissible value: `Gallbladder adenomyomatosis`
		* New permissible value: `Glomerular disease`
		* New permissible value: `Hyperkeratosis`
		* New permissible value: `Inflammation`
		* New permissible value: `Intestinal metaplasia`
		* New permissible value: `Keratinizing dysplasia; mild`
		* New permissible value: `Keratinizing dysplasia; moderate`
		* New permissible value: `Keratinizing dysplasia; severe (carcinoma in situ)`
		* New permissible value: `Leiomyoma`
		* New permissible value: `Leiomyomata w/ degenerative changes`
		* New permissible value: `Nonkeratinizing dysplasia; mild`
		* New permissible value: `Nonkeratinizing dysplasia; moderate`
		* New permissible value: `Nonkeratinizing dysplasia; severe (carcinoma in situ)`
		* New permissible value: `Other`
		* New permissible value: `PD-L1 CPS (223C LDT) - 20%`
		* New permissible value: `Platinum-resistant`
		* New permissible value: `Pleural plaque`
		* New permissible value: `Pulmonary interstitial fibrosis`
		* New permissible value: `Sialadenitis`
		* New permissible value: `Sinonasal papilloma`
		* New permissible value: `Squamous metaplasia`
		* New permissible value: `Squamous papilloma; solitary`
		* New permissible value: `Squamous papillomatosis`
		* New permissible value: `Tubular (papillary) adenoma(s)`
		* New permissible value: `Tumor-associated lymphoid proliferation`
		* New permissible value: `Tumor has rough spikey edges`
		* Removed permissible value: `Pleurodesis, Talc`
		* Removed permissible value: `Pleurodesis, NOS`
* Altered `diagnosis` Entity
	* `primary_disease` field changed to `diagnosis_is_primary_disease` for clarity.
* Altered `treatment` Entity
	* Changes made to `treatment_type`
		* New permissible value: `Pleurodesis, Talc`
		* New permissible value: `Pleurodesis, NOS`
* Altered `molecular_test` Entity
	* Changes made to `test_units`
		* New permissible value: `mm^2`
		* New permissible value: `ng/mL`
		* New permissible value: `percent`
		* New permissible value: `count x10^9/L`

### Known Issues and Workarounds

* The `mitotic_count` field in `diagnosis` is erroneously set as "deprecated" and does not appear in the dictionary viewer.  This field can be uploaded successfully without issue and will appear in the dictionary viewer at a later release. <!--SV-1939-->


## v2.4.0

* __GDC Product__: GDC Data Dictionary
* __Release Date__: June 21, 2021

### New Features and Changes

* Added min-max limitations on property values in `demographic`, `portion`, `aliquot`, `family_history`, `slide`, `follow_up`, `read_group`,
`sample`, `analyte`, `exposure`, `diagnosis`, `treatment`, `molecular_test`
* Altered `submitted_unaligned_reads`, `submitted_aligned_reads`, `annotated_somatic_mutation`, `simple_somatic_mutation`, `masked_somatic_mutation`, `submitted_genomic_profile`, `aligned_reads`, `aggregated_somatic_mutation`, `simple_germline_variation` Entities
	* Changes made to `experimental_strategy`
		* Removed permissible value: `Low Pass WGS`
* Altered `follow_up` Entity
	* New property: `eye_color`
	* New property: `history_of_tumor`
	* New property: `history_of_tumor_type`
	* New property: `undescended_testis_corrected`
	* New property: `undescended_testis_corrected_age`
	* New property: `undescended_testis_corrected_laterality`
	* New property: `undescended_testis_corrected_method`
	* New property: `undescended_testis_history`
	* New property: `undescended_testis_history_laterality`
	* Changes made to `comorbidity`
		* New permissible value: `Dermatomyosis`
		* New permissible value: `Herpes Zoster`
		* New permissible value: `Varicella Zoster Virus`
	* Changes made to `risk_factor`
		* New permissible value: `Dermatomyosis`
		* New permissible value: `Herpes Zoster`
		* New permissible value: `Varicella Zoster Virus`
* Altered `read_group` Entity
	* Changes made to `instrument_model`
		* New permissible value: `Illumina NovaSeq 6000`
	* Changes made to `single_cell_library`
		* New permissible value: `Chromium scATAC v1 Library`
	* Changes made to `target_capture_kit`
		* New permissible value: `Twist Human Comprehensive Exome`
* Altered `somatic_aggregation_workflow` Entity
	* Added link: `simple_somatic_mutations`
	* Changes made to `workflow_type`
		* New permissible value: `CaVEMan Variant Aggregation and Masking`
* Altered `sample` Entity
	* Changes made to `sample_type`
		* New permissible value: `Next Generation Cancer Model Expanded Under Non-conforming Conditions`
	* Changes made to `sample_type_id`
		* New permissible value: `87`
* Altered `germline_mutation_calling_workflow` Entity
	* New link: `submitted_genotyping_arrays`
	* Changes made to `workflow_type`
		* New permissible value: `Birdseed`
* Altered `slide_image` Entity
	* Changes made to `data_format`
		* New permissible value: `JPEG 2000`
* Altered `exposure` Entity
	* New property: `exposure_duration_years`
	* New property: `parent_with_radiation_exposure`
	* Changes made to `exposure_type`
		* New permissible value: `Radiation`
* Altered `simple_germline_variation` Entity
	* New property: `platform`
	* Changes made to `data_format`
		* New permissible value: `TSV`
* Altered `pathology_detail` Entity
	* New property: `consistent_pathology_review`
	* New property: `residual_tumor`
	* New property: `size_extraocular_nodule`
	* New property: `tumor_thickness`
* Altered `diagnosis` Entity
	* New property: `adrenal_hormone`
	* New property: `primary_disease`
	* Changes made to `ajcc_pathologic_stage`
		* New permissible value: `Stage IIIA1`
		* New permissible value: `Stage IIIA2`
	* Changes made to `metastasis_at_diagnosis_site`
		* New permissible value: `Bladder`
		* New permissible value: `Bronchus`
		* New permissible value: `Head, Face or Neck, NOS`
		* New permissible value: `Lymph Node, Regional`
		* New permissible value: `Lymph Node, Subcarinal`
	* Changes made to `method_of_diagnosis`
		* New permissible value: `Exoresection`
	* Changes made to `morphology`
		* New permissible value: `8246/6`
		* New permissible value: `8380/6`
		* New permissible value: `8461/6`
		* New permissible value: `8522/6`
	* Changes made to `sites_of_involvement`
		* New permissible value: `Mesothelium`
* Altered `treatment` Entity
	* New property: `route_of_administration`
	* Changes made to `treatment_arm`
		* New permissible value: `A081801`
	* Changes made to `therapeutic_agents`
		* New permissible value: `Interferon Alfa-2B`
		* New permissible value: `Levoleucovorin Calcium`
		* New permissible value: `Mistletoe Extract`
* Altered `somatic_mutation_calling_workflow` Entity
	* Changes made to `workflow_type`
		* New permissible value: `Strelka2 RNA`
* Altered `molecular_test` Entity
	* New property: `days_to_test`
	* New link: `diagnoses`
	* Changes made to `antigen`
		* New permissible value: `FMC-7`
		* New permissible value: `Kappa, Surface`
		* New permissible value: `Lambda, Surface`
	* Changes made to `gene_symbol`
		* New permissible value: `AQP1`
		* New permissible value: `CALB2`
		* New permissible value: `DNTT`
		* New permissible value: `EPCAM`
		* New permissible value: `GCET1`
		* New permissible value: `PDPN`
		* New permissible value: `PTGS2`
	* Changes made to `laboratory_test`
		* New permissible value: `BG8`
		* New permissible value: `Circulating Endothelial Cells`
		* New permissible value: `Cytokeratin 5`
		* New permissible value: `Cytokeratin 6`
		* New permissible value: `Dopamine-Secreting`
		* New permissible value: `Epinephrine-Secreting`
		* New permissible value: `Metanephrine-Secreting`
		* New permissible value: `Methoxytyramine-Secreting`
		* New permissible value: `Microsatellite Instability`
		* New permissible value: `Norepinephrine-Secreting`
		* New permissible value: `Normetanephrine-Secreting`
		* New permissible value: `Serum Mesothelin`
		* New permissible value: `TAG-72`

### Known Issues and Workarounds

* The `mitotic_count` field in `diagnosis` is erroneously set as "deprecated" and does not appear in the dictionary viewer.  This field can be uploaded successfully without issue and will appear in the dictionary viewer at a later release. <!--SV-1939-->


## v2.3.0

* __GDC Product__: GDC Data Dictionary
* __Release Date__: January 5, 2021

### New Features and Changes
* Altered `submitted_unaligned_reads` Entity
	* Changes made to `experimental_strategy`
		* New permissible value: `HiChIP`
		* New permissible value: `m6A RNA Methylation`
		* New permissible value: `scATAC-Seq`
	* Changes made to `read_pair_number`
		* New permissible value: `R3`
* Altered `submitted_aligned_reads` Entity
	* Changes made to `experimental_strategy`
		* New permissible value: `HiChIP`
		* New permissible value: `m6A RNA Methylation`
		* New permissible value: `scATAC-Seq`
* Altered `follow_up` Entity
	* Changes made to `comorbidity`
		* New permissible value: `Abnormal Glucose Level`
		* New permissible value: `Chronic Fatigue Syndrome`
		* New permissible value: `Clonal Hematopoiesis`
		* New permissible value: `Fibromyalgia`
		* New permissible value: `Gastritis`
	* Changes made to `risk_factor`
		* New permissible value: `Abnormal Glucose Level`
		* New permissible value: `Chronic Kidney Disease`
		* New permissible value: `Escherichia coli`
		* New permissible value: `Gastritis`
		* New permissible value: `Skin Rash`
* New Entity: `masked_methylation_array`
* Altered `read_group` Entity
	* New property: `chipseq_antibody`
	* New property: `fragmentation_enzyme`
	* Removed property: `RIN`
	* Changes made to `library_strategy`
		* New permissible value: `HiChIP`
		* New permissible value: `m6A RNA Methylation`
		* New permissible value: `scATAC-Seq`
* Altered `sample` Entity
	* New property: `sample_ordinal`
	* Changes made to `composition`
		* New permissible value: `Mixed Adherent Suspension`
* Altered `analyte` Entity
	* New property: `experimental_protocol_type`
	* New property: `rna_integrity_number`
* Altered `pathology_detail` Entity
	* New property: `additional_pathology_findings`
	* New property: `necrosis_percent`
	* New property: `necrosis_present`
	* New property: `rhabdoid_percent`
	* New property: `rhabdoid_present`
	* New property: `sarcomatoid_percent`
	* New property: `sarcomatoid_present`
	* Changes made to `dysplasia_degree`
		* New permissible value: `Mild`
		* New permissible value: `Moderate`
		* New permissible value: `Severe`
	* Changes made to `dysplasia_type`
		* New permissible value: `Epithelial`
		* New permissible value: `Keratinizing`
		* New permissible value: `Nonkeratinizing`
	* Changes made to `lymph_node_involvement`
* Altered `diagnosis` Entity
	* New property: `ann_arbor_b_symptoms_described`
	* Changes made to `ajcc_clinical_stage`
		* New permissible value: `Stage IA3`
	* Changes made to `ajcc_pathologic_m`
		* New permissible value: `M1d`
	* Changes made to `metastasis_at_diagnosis_site`
		* New permissible value: `Gastrointestinal Tract`
		* New permissible value: `Heart`
		* New permissible value: `Neck`
		* New permissible value: `Retroperitoneum`
		* New permissible value: `Urethra`
		* New permissible value: `Uterine Adnexa`
		* New permissible value: `Vertebral Canal`
		* New permissible value: `Vulva, NOS`
	* Changes made to `morphology`
		* New permissible value: `8249/6`
		* New permissible value: `8800/6`
	* Changes made to `supratentorial_localization`
		* New permissible value: `Frontal lobe`
		* New permissible value: `Occipital lobe`
		* New permissible value: `Parietal lobe`
		* New permissible value: `Temporal lobe`
* Altered `treatment` Entity
	* Changes made to `treatment_dose_units`
		* New permissible value: `mg`
	* Changes made to `treatment_type`
		* New permissible value: `Radiation, Hypofractionated`
		* New permissible value: `Radiation, Mixed Photon Beam`
		* New permissible value: `Radiation, Photon Beam`
	* Changes made to `therapeutic_agents`
		* Updated enum list.
* Altered `case` Entity
	* Updated `disease_type` and `primary_site` enum values.
* Altered `rna_expression_workflow` Entity
	* Changes made to `workflow_type`
		* New permissible value: `STAR - Smart-Seq2 Raw Counts`
		* New permissible value: `STAR - Smart-Seq2 Filtered Counts`
* Altered `molecular_test` Entity
	* New link: `slides`
	* Changes made to `antigen`
		* New permissible value: `Ki67`
	* Changes made to `gene_symbol`
		* New permissible value: `CHGA`
		* New permissible value: `SYP`
	* Changes made to `molecular_consequence`
		* New permissible value: `Exon Variant`
	* Changes made to `second_gene_symbol`
		* New permissible value: `CHGA`
		* New permissible value: `SYP`
* Altered `aligned_reads` Entity
	* Changes made to `experimental_strategy`
		* New permissible value: `HiChIP`
		* New permissible value: `scATAC-Seq`

### Bugs Fixed Since Last Release
* None

## v2.2.0

* __GDC Product__: GDC Data Dictionary
* __Release Date__: July 2, 2020

### New Features and Changes
* Added NCIt codes associated with enumeration values in `diagnosis` entity type
* Added `pathology_detail` entity <!--DICT-16-->
* Modified `treatment` entity <!--DICT-1-->
    - Added new enumerations for `therapeutic_agents` property
        * `Itraconazole`
        * `Tipiracil`
        * `Tipiracil Hydrochloride`
        * `Zirconium Zr 89 Panitumumab`
        * `Estradiol mustard`
        * `Progestational IUD`
        * `PD-1 Inhibitor`
        * `IGF-1R Inhibitor`
        * `CDK4/6 Inhibitor`
        * `ALK Inhibitor`
* Modified `annotation` entity <!--DICT-2 -->
    - Added `many_to_many` link to `copy_number_estimate` entity
* Modified `diagnosis` entity <!--DICT-6-->
    - Added new properties:
        * `eln_risk_classification`
        * `satellite_nodule_present`
        * `who_cns_grade`
        * `who_nte_grade`
        *  `sites_of_involvement`
    - Added new permissible values to `ajcc_pathologic_stage` property
        * `Stage IA3`
    - Added new permissible values to `classification_of_tumor` property
        * `Progression`
    - Added new permissible values to `metastasis_at_diagnosis_site` property
        * `Esophagus`
    - Added new permissible values to `morphology` property
    - Removed properties as `required`
        * `days_to_last_follow_up`
        * `tumor_grade`
        * `progression_or_recurrence`
        * `days_to_recurrence`
        * `days_to_last_known_disease_status`
        * `last_known_disease_status`
    - Deprecated properties
        * `mitotic_count`
        * `papillary_renal_cell_type`
        * `micropapillary_features`
        * `non_nodal_regional_disease`
        * `non_nodal_tumor_deposits`
* Modified `sample` entity <!--DICT-7-->
    - Changed description of properties
        * `days_to_collection`
        * `days_to_sample_procurement`
    - Deprecated properties
        * `is_ffpe`
        * `oct_embedded`
    - Added new permissible values to `sample_type` property
        * `Mixed Adherent Suspension Saliva`
* Modified `follow_up` entity <!--DICT-11-->
    - Added new properties
        * `procedures_performed`
        * `hormonal_contraceptive_type`
        * `hormone_replacement_therapy_type`
    - Added new permissible values to `comorbidity` property
    - Added new permissible values to `risk_factor` property
    - Added new permissible values to `evidence_of_recurrence_type` property
    - Added new permissible values to `aids_risk_factors` property
* Modified `exposure` entity <!--DICT-12-->
    - Added new properties
        * `smokeless_tobacco_quit_age`
        * `alcohol_type`
    - Added new permissible values to `exposure_type` property
        * `Wood Dust`
        * `Smoke`
    - Added new permissible value to `type_of_smoke_exposure` property
        *  `Tobacco smoke, NOS`
* Modified `submitted_unaligned_reads` property <!--DICT-13-->
    - Removed permissible value from `read_pair_number` property
        * `I1`
* Modified `aliquot` entity <!--DICT-14-->
    - Added new permissible value to `analyte_type` property
        * `Nuclei RNA`
* Modified `rna_expression_workflow` entity <!--DICT-15-->
    - Removed permissible value from `workflow_type` property
        * `STAR - Smart-Seq2 Counts`
    - Added new permissible values to `workflow_type` property
        * `STAR - Smart-Seq2 Gene Counts`
        * `STAR - Smart-Seq2 GeneFull Counts`
* Modified `molecular_test` entity <!--DICT-17-->
    - Added new properties
        * `mitotic_count`
        * `mitotic_total_area`
        * `biospecimen_volume`
    - Added new permissible values to `gene_symbol` property
    - Added new permissible values to `second_gene_symbol` property
    - Added new permissible values to `antigen` property
    - Added new permissible values to `laboratory_test` property 	  

### Bugs Fixed Since Last Release
* None

## v2.1.0

* __GDC Product__: GDC Data Dictionary
* __Release Date__: March 10, 2020

### New Features and Changes

* Added NCIt codes associated with enumeration values in `exposure`, `family_history`, and `demographic` entity types
* Restructured dictionary for consistency across types
* __read_group entity__
* Added 5 new target capture kits <!--DAT-2755-->
    - `Custom Twist Broad PanCancer Panel - 396 Genes`
    - `Nextera DNA Exome`
    - `Custom Twist Broad Exome v1.0 - 35.0 Mb`
    - `Custom SureSelect CGCI-HTMCP-CC KMT2D And Hotspot Panel - 37.0 Kb`
    - `TruSeq RNA Exome`
* Add one new enum to `library_strategy` <!--DAT-2793-->
    - `scRNA-Seq`
* __structural_variation entity__ <!--DAT-2703-->
* Added `VCF` to `data_format` property
* Added links to `structural_variation` from `somatic_mutation_index` <!--DAT-2704-->
* Added links to `aligned_reads` from `somatic_copy_number` workflow <!--DAT-2705-->
* __copy_number_segment entity__
* Added new permissible values to `experimental_strategy` property <!--DAT-2706-->
    - `WGS`
    - `WXS`
* Added new enum to `experimental_strategy` <!--DAT-2788-->
    - `WGS`
* __aligned_reads entity__
* Added 2 new properties <!--DAT-2787-->
    - `tumor_ploidy`
    - `tumor_purity`
* __treatment entity__
* Added enumeration to `therapeutic_agent` property <!--DAT-2727-->
* __copy_number_estimate entity__ <!--DAT-2729-->
* Added new enum to `data_format`
    - `TSV`
* __demographic entity__	 <!--DAT-2733-->
* Added new property `country_of_residence_at_enrollment`
* __family_history entity__ <!--DAT-2738-->
* Added new permissible values to `relationship_primary_diagnosis` property
*  __follow_up entity__	 <!--DAT-2739-->
* Added new properties
    - `body_surface_area`
    - `recist_targeted_regions_number`
    - `recist_targeted_regions_sum`
    - `adverse_event_grade`
    - `cd4_count`
    - `imaging_type`
    - `scan_tracer_used`
    - `nadir_cd4_count`
    - `hiv_viral_load`
    - `aids_risk_factors`
    - `haart_treatment_indicator`
    - `immunosuppressive_treatment_type`
    - `evidence_of_recurrence_type`
    - `imaging_result`
    - `hormonal_contraceptive_use`
    - `pregnancy_outcome`
    - `hysterectomy_type`
    - `hysterectomy_margins_involved`
    - `days_to_imaging`
    - `cdc_hiv_risk_factors`
    - `risk_factor`
* Added new enum to `days_to_follow_up`
    - `null`
* __molecular_test entity__ <!--DAT-2741-->
* Added new permissible values to various properties
    - `laboratory_test`
    - `second_gene_symbol`
    - `molecular_consequence`
    - `biospecimen_type`
    - `molecular_analysis_method`
    - `gene_symbol`
    - `clonality`
* __sample entity__ <!--DAT-2742-->
* Added new permissible values to various properties
    - `method_of_sample_procurement`
    - `biospecimen_anatomic_site`
    - `tumor_descriptor`
* Added new property
    - `tissue_collection_type`
* __treatment entity__ <!--DAT-2745-->
* Added new properties
    - `treatment_arm`
    - `reason_treatment_ended`
    - `number_of_cycles`
    - `treatment_effect_indicator`
    - `treatment_dose`
    - `treatment_dose_units`
    - `treatment_frequency`
    - `chemo_concurent_to_radiation`
* Added new permissible values to various properties
    - `therapeutic_agent`
    - `treatment_effect`
    - `treatment_intent_type`
* __slide entity__ <!--DAT-2747-->
* Added new properties
    - `percent_sarcomatoid_features`
    - `percent_rhabdoid_features`
    - `prostatic_chips_total_count`
    - `prostatic_chips_positive_count`
    - `prostatic_involvement_percent`
    - `bone_marrow_malignant_cells`
    - `percent_follicular_component`
    - `tissue_microarray_coordinates`
* __diagnosis entity__ <!--DAT-2735-->
* Added new properties
    - `tumor_depth`
    - `margin_distance`
    - `transglottic_extension`
    - `margins_involved_site`
    - `gleason_grade_tertiary`
    - `papillary_renal_cell_type`
    - `gleason_patterns_percent`
    - `greatest_tumor_dimension`
    - `lymph_node_involved_site`
    - `pregnant_at_diagnosis`
    - `figo_staging_edition_year`
* Added new permissible values to various properties
    - `classification_of_tumor`
    - `figo_stage`
    - `tumor_grade`
    - `metastasis_at_diagnosis_site`
    - `gleason_grade_group`
    - `tissue_or_organ_of_origin`
    - `morphology`
* __exposure entity__ <!--DAT-2737-->
* Added new properties
    - `secondhand_smoke_as_child`
    - `exposure_type`
    - `type_of_tobacco_used`
    - `exposure_duration`
    - `tobacco_use_per_day`
    - `age_at_onset`
    - `marijuana_use_per_week`
    - `tobacco_smoking_status`
* __case entity__ <!--DAT-2746-->
* Added new properties
    - `consent_type`
    - `days_to_consent`
* Added new permissible values to `index_date`
* Added `scRNA-Seq` as new enum to `experimental_strategy` in 4 entities <!--DAT-2794-->
    - `submitted_unaligned_reads`
    - `submitted_aligned_reads`
    - `aligned_reads`
    - `gene_expression`
* Added 3 new permissible values to `workflow_type` in alignment_workflow <!--2795-->
    - `CellRanger - 10x Chromium`
    - `STAR - Smart-Seq2`
    - `zUMIs - Smart-Seq2`
* Added 4 new permissible values to `workflow_type` in rna_expression_workflow <!--DAT-2796-->
    - `CellRanger - 10x Raw Counts`
    - `CellRanger - 10x Filtered Counts`
    - `STAR - Smart-Seq2 Counts`
    - `zUMIs - Smart-Seq2 Counts`
* Add one new integer property to `read_group` <!--DAT-2798-->
    -  `number_expect_cells`
* Add one new enum to `read_pair_number` in `submitted_unaligned_reads` <!--DAT-2799-->
    - `I1`
* Add one new enum property in `read_group` node <!--DAT-2800-->
    - `Chromium 3' Gene Expression v2 Library`
    - `Chromium 3' Gene Expression v3 Library`
    - `Smart-Seq2`
* Created new node `expression_analysis_workflow` <!--DAT-2802-->
* Created new node `secondary_expression_analysis` <!--2803-->
* Add one new enum to `data_format` in `gene_expression` <!--DAT-2804-->
    -  `MEX`

### Bugs Fixed Since Last Release
* None

## v2.0.0 <!--REQ-393-->

* __GDC Product__: GDC Data Dictionary
* __Release Date__: January 30, 2020

### New Features and Changes
* The API that includes the GDC data dictionary now uses Python 3.

### Bugs Fixed Since Last Release

* None


## v.1.18.1

* __GDC Product__: GDC Data Dictionary
* __Release Date__: November 6, 2019

### New Features and Changes
* Added new permissible value `deleted` to property `file_state`

### Bugs Fixed Since Last Release

* None


## v.1.18

* __GDC Product__: GDC Data Dictionary
* __Release Date__: July 31, 2019

### New Features and Changes

* Added new entities <!--DAT-2450-->
    - `protein_expression_quantification`  <!--DAT-2237-->
    - `submitted_genotyping_array`  <!--DAT-2327-->
    - `somatic_copy_number_workflow` <!--DAT-2330-->
* Add links in data model to `somatic_copy_number_workflow` from `copy_number_segment`  <!--DAT-2331-->
* Add links in data model to `somatic_copy_number_workflow` from `copy_number_estimate`	 <!--DAT-2415-->
* Add links in data model from `annotated_somatic_mutation` to `genomic_profile_harmonization_workflow` <!--DAT-2333-->
* Modified `copy_number_segment` entity
    - Add new data type <!--DAT-2332-->
        - `Allele-specific Copy Number Segment`
* Update data dictionary to support new annotation classifications <!--DAT-2238-->
* Fixed typo in `sample` entity schema <!--DAT-2247-->
* Unrequired project link for aliquot level MAFs  <!--DAT-2260-->
* Added NCIt codes for gender values  <!--DAT-2314-->
* Changed description of `masked_somatic_mutation` and `aggregated_somatic_mutation` nodes to be the same  <!--DAT-2311-->
* Modified `archive` entity <!--DAT-2259-->
    - Set `downloadable` property to `true`
* Modified `publication`  entity<!--DAT-2259-->
    - Set `downloadable` property to `true`
* Modified `filtered_copy_number_segment`  entity<!--DAT-2259-->
    - Set `downloadable` property to `true`
* Modified `aligned_reads` entity
    - Added `MSI properties` as new property
* Modified `read_group` entity
    - Added `Custom SureSelect Human All Exon v1.1 Plus 3 Boosters` as new permissible value for `target_capture_kit` field <!--DAT-2003-->
    - Added `SeqCap EZ Human Exome v3.0` as new permissible value for `target_capture_kit` field <!--DAT-2310-->
    - Added new permissible values for `instrument_model`
        - `Unknown` <!--DAT-2411-->
        - `Not Reported` <!--DAT-2411-->
        - `Ion Torrent S5` <!--DAT-2338-->
* Modified `biospecimen_supplement` entity <!--  DAT-2266-->
    - Added `CDC JSON` as new permissible data format
* Modified `demographic` entity <!--DAT-2303-->
    - Added new property
        - `age_is_obfuscated`
* Modified `demographic` entity <!--DAT-2303-->
    - Added new properties
        - `cause_of_death_source` <!--DAT-2341-->
        - `occupation_duration_years` <!--DAT-2342-->
* Modified `diagnosis` entity
    - Added new properties
        - `non_nodal_regional_disease` <!--DAT-2353-->
        - `non_nodal_tumor_deposits` <!--DAT-2354-->
        - `ovarian_specimen_status` <!--DAT-2356-->
        - `ovarian_surface_involvement` <!--DAT-2357-->
        - `percent_tumor_invasion` <!--DAT-2360-->
        - `peritoneal_fluid_cytological_status` <!-- DAT-2361-->      
        - `breslow_thickness` <!--DAT-2345-->
        - `international_prognostic_index` <!--DAT-2349-->
        - `largest_extrapelvic_peritoneal_focus` <!--DAT-2350-->
        - `mitotic_count` <!--DAT-2352-->     
    - Removed permissible values from `primary_diagnosis`  <!--DAT-2377 -->
    - Removed permissible values from `site_of_resection_or_biopsy` <!--DAT-2378-->
    - Removed `tumor_stage` as required property <!--DAT-2397-->
    - Added new permissible values for
        - `ajcc_pathologic_stage` <!--DAT-2368-->
        - `metastasis_at_diagnosis_site` <!--DAT-2370-->
    - Migrated values not part of permissible values to `Not Reported` for following properties
          - `primary_diagnosis` <!--DAT-2412-->
          - `site_of_resection_or_biopsy` <!--DAT-2413 -->
          - `tumor_grade` <!--DAT-2414-->
          - `tissue_or_organ_of_origin` <!--DAT-2438-->
    - Removed permissible values from
          - `tissue_or_organ_of_origin` <!--DAT-2426-->
          - `morphology` <!--  DAT-2427-->
- Modified `structural_variant_calling_workflow` entity <!--DAT-233-->
    - Added new `workflow_type`
- Modified `structural_variant` entity
    - Added `BEDPE` to `data_format` as permissible value <!--DAT-2336-->
- Modified `molecular_test` entity
    - Added new permissible values for `gene_symbol` <!--DAT-2373-->
    - Added new permissible values for `test_result`<!--DAT-2374-->
    - Added new permissible values for `antigen`<!--DAT-2372-->
    - Added new property `pathogenicity`  <!--DAT-2366-->
- Modified `follow_up` entity
    - Added new permissible value for `risk_factor` <!--DAT-2371-->
- Modified `sample` entity
    - Added new permissible values for `method_of_sample_procurement` <!--DAT-2376-->
- Modified `genomic_profile_harmonization_workflow` entity
    - Added new `workflow_type` permissible values <!--DAT-2388-->
- Modified `somatic_mutation_calling_workflow` entity
    - Added new `workflow_type` permissible values <!--DAT-2389-->
    - Modified `somatic_annotation_workflow` entity
    - Added new `workflow_type` permissible values <!--DAT-2391-->
- Modified `case` entity
    - Added new permissible value to `disease_type` <!--DAT-2390-->
        - `Not Applicable`
- Modified `copy_number_estimate` entity
    - Added new permissible value to `experimental_strategy` <!--DAT-2334-->
        -  `WXS`
- Modified `family_history` entity
    - Added new property <!--DAT-2365-->
        - `relatives_with_cancer_history_count`
- Modified `sample` entity <!--DAT-2418-->
    - Added new permissible values
        - `sample_type`
        - `sample_type_id`

### Bugs Fixed Since Last Release

* None

## v.1.17

* __GDC Product__: GDC Data Dictionary
* __Release Date__: June 5, 2019


### New Features and Changes

* Deleted vital status, days_to_birth, and days_to_death from Diagnosis node.  Data submission and data requests should all be directed to the corresponding properties on the Demographic Node.

### Bugs Fixed Since Last Release

* None



## v.1.16

* __GDC Product__: GDC Data Dictionary
* __Release Date__: April 17, 2019


### New Features and Changes

* Updates to the Data Dictionary Search Tool
* Added new bioinformatics workflow for methylation arrays (Sesame) <!--DAT-2025-->
* Changed `somatic_mutation_calling_workflow` link from `one_to_many` to `many_to_many`	<!--DAT-1697-->
* Modified `read_group` entity
    - Added `SeqCap EZ Human Exome v2.0` as new permissible value for `target_capture_kit` field <!--DAT-1827-->
    - Added `Custom SureSelect Human All Exon v1.1 Plus 3 Boosters` as new permissible value `target_capture_kit` field <!--DAT-2002-->
    - Added `Custom SureSelect CGCI-HTMCP-CC Panel - 19.7 Mb`  as new permissible value `target_capture_kit` field <!--DAT-2013-->
* Modified `case` entity
    - Updated the description for the `primary_site` field <!--DAT-1928-->
    - Added new permissible value to `lost_to_followup` field <!--DAT-2133-->
* Modified `molecular_test` entity
    - Removed properties with genomic coordinates <!--DAT-1991-->
    - Add new permissible values to `test_result`	<!--TT-918-->
    - Added `second_exon` as new property<!--TT-919-->
* Modified `aligned_reads_index` entity
    - Made these files not submittable <!--DAT-1985-->
* Modified `somatic_mutation_index` entity
    - Made these files not submittable <!--DAT-1986-->
* Modified `sample` entity
    - Added new permissible values for `sample_type` <!--DAT-2017-->
        - `Blood Derived Cancer - Bone Marrow`
        - `Blood Derived Cancer - Peripheral Blood`
    - Added new permissible values to `sample_type_id` <!--DAT-2116-->
* Modified `diagnosis` entity
    - Added 6 new staging and grading properties for TCGA <!--DAT-2056-->
        - `igcccg_stage`
        - `masaoka_stage`
        - `gleason_grade_group`
        - `primary_gleason_grade`
        - `secondary_gleason_grade`
        - `weiss_assessment_score`
    - Made `vital_status` an optional field <!--DAT-2157-->
    - Removed deprecated properties <!--TT-939--><!--TT-938-->
        - `days_to_death`
        - `days_to_birth`
        - `cause_of_death`
        - `hiv_positive`
        - `days_to_hiv_diagnosis`
        - `ldh_normal_range_upper`
        - `new_event_type`
        - `hpv_status`
        - `hpv_positive_type`
        - `colon_polyps_history`
        - `progression_free_survival`
        - `progression_free_survival_event`
        - `overall_survival`
        - `days_to_treatment`
        - `ldh_level_at_diagnosis`
    - Added `vital_status` property to deprecated list<!--DAT-2158-->
* Modified `somatic_aggregation_workflow` entity
    - Added `Aliquot Ensemble Somatic Variant Merging and Masking` as new permissible value to `workflow_type`
* Modified `slide` entity
    - Updated the description for the `magnification` field <!--DAT-2125-->
* Modified `aliquot` entity
    - Updated the the description for several fields <!--DAT-2126-->
        - `selected_normal_low_pass_wgs`
        - `selected_normal_targeted_sequencing`
        - `selected_normal_wgs`
        - `selected_normal_wxs`
* Modified `follow-up` entity
    - Added new field `days_to_progression_free` <!--DAT-2135-->
* Modified `demographic` entity
    - Made `vital_status` a required field <!--DAT-2157-->
* Modified `exposure` entity <!--TT-926-->
    - Added new properties
        - `environmental_tobacco_smoke_exposure`
        - `respirable_crystalline_silica_exposure`
        - `coal_dust_exposure`
        - `type_of_smoke_exposure`
        - `type_of_tobacco_used`
        - `smoking_frequency`
        - `time_between_waking_and_first_smoke`
    - Removed `cigarettes_per_day` property from deprecated list<!--DAT-2062-->
* Modified `annotation` entity
    - Modified permissible values to `status` <!--TT-930-->
        - Approved
        - Rescinded

### Bugs Fixed Since Last Release

* None


## v.1.15

* __GDC Product__: GDC Data Dictionary
* __Release Date__: December 18, 2018


### New Features and Changes

* Removed `Raw Sequencing Data` and `Sequencing Data` as permissible values from `submitted_aligned_reads`, `submitted_unaligned_reads`, and `aligned_reads` <!--DAT-42--> <!--DAT-1904-->
* Deleted `aligned_reads_metrics` entity <!--DAT-1754-->
* Created new `raw_methylation_array` entity <!--DAT-1854-->
* Add regex validation to property `md5sum` for following entities: <!--DAT-1899-->
    - `slide_image`
    - `analysis_metadata`
    - `clinical_supplement`
    - `experiment_metadata`
    - `pathology_report`
    - `run_metadata`
    - `biospecimen_supplement`
    - `submitted_aligned_reads`
    - `submitted_genomic_profile`
    - `submitted_methylation_beta_value`
    - `submitted_tangent_copy_number`
    - `submitted_unaligned_reads`
* Modified `molecular_test` entity
    - Migrated data from `blood_test` to `laboratory_test` and `biospecimen_type` for all entities<!--TT-754-->
    - Added new property `intron` <!--DAT-1847-->
    - Deleted `blood_test` entity <!--DAT-1639-->
    - Added new permissible values for `gene_symbol`<!--DAT-1553-->
    - Added new permissible values for `antigen`<!--DAT-1662-->
    - Added new permissible values for `molecular_analysis_method` <!--DAT-1663-->
    - Added new permissible values for `variant_type` <!--DAT-1664-->
    - Added new permissible values for `test_result` <!--DAT-1665-->
    - Added new permissible values for `molecular_consequence` <!--DAT-1666-->
    - Added regex validation to property `transcript` <!--DAT-1916-->
    - Added regex validation to property `locus` <!--DAT-1874-->
    - Changed data type of `exon` property to be `string` with regex validation <!--DAT-1890-->
* Modified `diagnosis` entity
    - Added new fields
      - `tumor_focality`<!-- DAT-1832-->
      - `tumor_regression_grade` <!--DAT-1833-->
      - `lymph_nodes_tested` <!--DAT-1834-->
    - Added new permissible value for `primary_diagnosis` field<!--DAT-1879-->
    - Added min and max values to time-based properties <!--DAT-1885-->
    - Added new permissible value for `morphology` field <!--TT-818-->
* Modified `follow_up` entity
    - Added new permissible values for `ecog_performance_status`<!--DAT-1684-->
    - Added new permissible values for `comorbidity` <!--DAT-1766-->
    - Added new permissible values for `disease_response`<!--DAT-1840-->
    - Added new permissible values for `risk_factor`<!-- DAT-1841-->
    - Added min and max values to time-based properties <!--DAT-1884-->
    - Added new property:
      - `hepatitis_sustained_virological_response` <!--DAT-1845-->
    - Updated CDE, CDE version, description and URL for `comorbidity`<!--DAT-1911-->
    - Added a CDE for `days_to_comorbidity` <!--DAT-1912-->
    - Removed `reflux_treatment` property <!--DAT-1913-->
    - Add a new property:<!--DAT-1843-->
      - `risk_factor_treatment`
* Modified `aligned_reads` entity
  - Added new contamination properties <!--DAT-1749-->
    - `contamination`
    - `contamination_error`
* Modified `read_group` entity
  - Added new permissible values for `target_capture_kit` <!--DAT-1757-->
  - Updated description for property `instrument_model` <!--DAT-1763-->
  - Added new permissible values for `target_capture_kit` <!--DAT-1799-->
  - Added new permissible values for `library_strategy`<!--DAT-1814-->
  - Added regex validation to property `adapter_sequence` <!--DAT-1895-->
  - Added regex validation to property `multiplex_barcode`<!--DAT-1897-->
  - Allow users to enter null for property `read_length` <!--DAT-1908-->
  - Allow users to enter null for property `is_paired_end` <!--DAT-1909-->
* Modified `family_history` entity
  - Added new permissible values for `relationship_primary_diagnosis` <!--DAT-1765-->
  - Added min and max values to properties <!--DAT-1887-->
* Modified `case` entity
  - Add min and max values to properties <!--DAT-1888-->
  - Delete permissible value from `primary_site` <!-- DAT-1772-->
    - `Unknown Primary Site`
* Modified `analyte` entity
  - Corrected the description for fields `analyte_volume` to include microliters as unit <!--DAT-1801-->
* Modified `exposure` entity
  - Added new properties
    - `asbestos_exposure` <!-- DAT-1836-->
    - `radon_exposure` <!-- DAT-1837-->
* Modified `sample` entity
  - Added new permissible values to `method_of_sample_procurement` <!--DAT-1849-->
  - Added regex validation to `pathology_report_uuid` <!--DAT-1893-->
  - Change type from string to number for properties:
    - `intermediate_dimension` <!--DAT-1861-->
    - `longest_dimension` <!--DAT-1863-->
    - `shortest_dimension` <!--DAT-1865-->
    - `time_between_clamping_and_freezing` <!--DAT-1867-->
    - `time_between_excision_and_freezing` <!--DAT-1869-->
  - Add min and max to properties on the sample node <!--DAT-1883-->
  - Populated sample nodes that have no value for `tissue_type` to "Not Reported" <!--TT-658-->
* Modified `treatment` entity
  - Added a new property
    - `prior_treatment_effect` <!--DAT-1850-->
  - Add min and max values to properties <!--DAT-1889-->
* Modified `aliquot` entity
  - Corrected the description for fields `analyte_volume` to include microliters as unit <!--DAT-1859-->
* Modified `demographic` entity
  - Added min and max to properties <!--DAT-1886-->


### Bugs Fixed Since Last Release

* Fixed value of `pathology_report_uuid` on sample entity `7b29b034-86e4-4266-8657-036e96e04430` to satisfy regex requirements <!--DAT-1939-->
* Migrated a few unsupported values for sample.pathology_report_uuid, read_group.adapter_sequence, read_group.multiplex_barcode <!--DAT-1941-->

## v.1.14

* __GDC Product__: GDC Data Dictionary
* __Release Date__: September 27, 2018

### New Features and Changes

* Added description and type information to generic properties <!--TT-1347-->
* Modify the dictionary viewer to be case sensitive for deprecated fields <!--PRTL-2180-->
* Deleted `sample_level_maf` entity  <!--TT-666-->
* Added new field for all nodes: <!--TT-556-->
    - `previous_versions_downloadable`
* Modified `molecular_test` entity
    - Migrated data from `blood_test` to `laboratory_test` and `biospecimen_type` <!--TT-754-->
    - Added new fields: <!--DAT-1644-->
        - `laboratory_test`
        - `biospecimen_type`
    - Added new permissible values for `gene_symbol` fields <!--DAT-1553-->
        - `Not Applicable`
    - Deleted field `blood_test` <!--DAT-1639-->
    - Added new permissible values for `antigen` field <!--DAT-1662-->
    - Added new permissible values to `molecular_analysis_method` <!--DAT-1663-->
    - Added new permissible values for `variant_type` field <!--DAT-1664-->
    - Added new permissible values to `test_result` <!--DAT-1665-->
* Modified `case` entity
    - Modified permissible values on `index_date`
        - Added new value `Initial Genomic Sequencing` <!--TT-1461-->
    - Updated CDE codes for properties <!--DAT-1590-->
    - Add new permissible value to `index_date` <!--DAT-1669-->
        - `Sample Procurement` <!--TT-678-->
* Modified `aliquot` entity
    - Changed `exclusive` to `True` <!--DAT-1519-->
* Modified `read_group_qc` entity
    - Made following fields not `required` and added `Unknown`/`Not Reported` as permissible values: <!--DAT-1519--><!--DAT-1525-->
        - `adapter_content`
        - `basic_statistics`
        - `encoding`
        - `kmer_content`
        - `overrepresented_sequences`
        - `per_base_sequence_quality`
        - `per_tile_sequence_quality`
        - `per_sequence_quality_score`
        - `per_base_sequence_content`
        - `per_sequence_gc_content`
        - `per_base_n_content`
        - `percent_gc_content`
        - `sequence_length_distribution`
        - `sequence_duplication_levels`
        - `total_sequences`
* Modified `aligned_reads` entity
    - Added new permissible values to `experimental_strategy` field <!--DAT-1526-->
        - `Targeted Sequencing`
        - `Bisulfite-Seq`
        - `ChIP-Seq`
        - `ATAC-Seq`
* Modified `read_group` entity
    - Added new field `days_to_sequencing` <!--DAT-1533-->
    - Add new permissible values to `target_capture_kit` <!--DAT-667-->
        - `Custom Personalis ACEcp VAREPOP-APOLLO Panel v2`
        - `xGen Exome Research Panel v1.0`
        - `SureSelect Human All Exon v5 + UTR`
* Modified `treatment` entity
    - Updated CDE codes for properties <!--DAT-1596-->
    - Updated description for `treatment_or_therapy` <!--DAT-1548-->
    - Updated description for `therapeutic_agents` <!--DAT-1549-->
    - Added new field:
        - `initial_disease_status` <!--DAT-1647-->
    - Updated permissible values for `treatment_type` <!--TT-672-->
    - Added new permissible values for `treatment_outcome`  <!--TT-680-->
        - `No Measurable Disease`
        - `Persistent Disease`
    - Deleted values from `treatment_intent_type`  <!--TT-681-->
* Modified `annotated_somatic_mutation` entity
    - Added new permissible values to `experimental_strategy` field <!--DAT-1554-->
        - `RNA-Seq`
        - `miRNA-Seq`
        - `Bisulfite-Seq`
        - `ChIP-Seq`
        - `ATAC-Seq`
    - Add new permissible value to `data_format` <!--TT-663-->
        - `MAF`
    - Add new permissible value to `data_type` <!--TT-665-->
        - `Masked Annotated Somatic Mutation`
* Modified `masked_somatic_mutation` entity
    - Added new permissible values to `experimental_strategy` field <!--DAT-1555-->
        - `RNA-Seq`
        - `miRNA-Seq`
        - `Bisulfite-Seq`
        - `ChIP-Seq`
        - `ATAC-Seq`
    - Removed `Validation` as permissible from `experimental_strategy` <!--TT-532-->
* Modified `simple_germlne_variation` entity
    - Added new permissible values to `experimental_strategy` field <!--DAT-1556-->
        - `RNA-Seq`
        - `miRNA-Seq`
        - `Bisulfite-Seq`
        - `ChIP-Seq`
        - `ATAC-Seq`
* Modified `simple_somatic_mutation` entity
    - Added new permissible values to `experimental_strategy` field <!--DAT-1558-->
        - `RNA-Seq`
        - `miRNA-Seq`
        - `Bisulfite-Seq`
        - `ChIP-Seq`
        - `ATAC-Seq`    
* Modified `submitted_genomic_profile` entity
    - Added new permissible values to `experimental_strategy` field <!--DAT-1558-->
        - `WXS`
        - `Low Pass WGS`
        - `RNA-Seq`
        - `miRNA-Seq`
        - `Bisulfite-Seq`
        - `ChIP-Seq`
        - `ATAC-Seq`
* Modified `diagnosis` entity
    - Updated CDE codes for properties <!--DAT-1592-->
    - Corrected the reference to `ubiquitous_properties` <!--DAT-1634-->
    - Deleted permissible values from `vascular_invasion_present` field <!--DAT-1560-->
        - `Extramural`
        - `Intramural`
    - Added permissible value to `ajcc_clinical_stage` <!--DAT-1571-->
        - `Stage IIIC1`
    - Added permissible values to `method_of_diagnosis` <!--DAT-1567-->
        - `Imaging`
        - `Physical Exam`
        - `Pathologic Review`
    - Added permissible value to `vascular_invasion_type` <!--DAT-1579-->
        - `Extramural`
        - `Intramural`  
    - Added permissible value to `metastasis_at_diagnosis_site`  <!--TT-673-->
    - Updated deprecated values list <!--TT-682-->
    - Deleted field `ldh_level` <!--TT-686-->
* Modified `follow_up` entity
    - Updated CDE codes for properties <!--DAT-1594-->
    - Added permissible value to `progression_or_recurrence_type` <!--DAT-1581-->
        - `Biochemical`
    - Added permissible values to `hpv_positive_type` <!--DAT-1583-->
        - `63`
        - `70`
    - Added permissible value to `disease_response` <!--DAT-1584-->
        - `BED-Biochemical Evidence of Disease`
        - `PDM-Persistent Distant Metastasis`
        - `PLD-Persistent Locoregional Disease`
        - `TF-Tumor Free`
        - `WT-With Tumor`
    - Added permissible value to `comorbidity` <!--DAT-1585-->
        - `Hepatitis A Infection`
        - `Herpes`      
    - Added permissible values to `risk_factor` field  <!--TT-674-->
* Modified `demographic` entity
    - Updated CDE codes for properties <!--DAT-1591-->
    - Modified permissible values for `cause_of_death` field   <!--TT-676-->
        - `Cardiovascular Disorder, NOS`
        - `Renal Disorder, NOS`
        - `Surgical Complications`
    - Updated `days_to` descriptions  <!--TT-683-->
* Modified `family_history` entity
    - Updated CDE codes for properties <!--DAT-1635-->
    - Added new fields: <!--DAT-1635-->
        - `relationship_primary_diagnosis`
        - `relationship_gender`
    - Updated CDE for field `relative_with_cancer_history` <!--TT-254-->
* Modified `submitted_unaligned_reads` entity
    - Make field `read_pair_number` required  <!--DAT-1648-->
    - Added permissible values to `read_pair_number` <!--DAT-1651-->
        - ` Not Applicable`
* Modified `aligned_reads` entity <!--DAT-1668-->
    - Added new fields:
        - `average_base_quality`
        - `average_insert_size`
        - `average_read_length`
        - `mean_coverage`
        - `pairs_on_diff_chr`
        - `proportion_base_mismatch`
        - `proportion_coverage_10X`
        - `proportion_coverage_30X`
        - `proportion_reads_duplicated`
        - `proportion_reads_mapped`
        - `proportion_targets_no_coverage`
        - `total_reads`
    - Changed nodes with `experimental_strategy` as `Validation` to `Targeted Sequencing` <!--TT-704-->
* Modified `sample` entity
    - Made `tissue_type` a required field <!--TT-657-->
    - Populated `tissue_type` with `Not Reported` for all nodes with no value  <!--TT-658-->
* Modified `alignment_workflow` entity
    - Added new field to `workflow_type` <!--TT-659-->
        - `STAR 2-Pass Chimeric`
* Modified `rna_expression_workflow` entity
    - Added new field to `workflow_type` <!--TT-660-->
        - `STAR - FPKM`
* Modified `gene_expression` entity
    - Add new permissible value to `data_type` <!--TT-663-->
        - `Splice Junction Quantification`
* Modified `aggregated_somatic_mutation` entity
    - Updated field `experimental_strategy` <!--TT-694-->
* Modified `somatic_mutation_calling_workflow` entity
    - Added `workflow_type` field <!--TT-661-->
* Modified `somatic_annotation_workflow` entity
    - Added `workflow_type` field <!--TT-662-->


### Bugs Fixed Since Last Release

* None

## v.1.13

* __GDC Product__: GDC Data Dictionary
* __Release Date__: May 21, 2018

### New Features and Changes

* Submitted Genomic Profile to Read Group relationship is updated to be many-to-many <!--DAT-1425-->
* Added requirement for submitter_id on additional nodes <!--TT-609-->

### Bugs Fixed Since Last Release

* None

## v.1.12.1

* __GDC Product__: GDC Data Dictionary
* __Release Date__: April 26, 2018

### New Features and Changes

* None

### Bugs Fixed Since Last Release

* Moved read_pair_number from Read Group to Submitted Unaligned Reads node <!--SV-1089-->

## v.1.12

* __GDC Product__: GDC Data Dictionary
* __Release Date__: April 23, 2018


### New Features and Changes

* Updates to data dictionary viewer:
    - Data dictionary viewer displays `true/false` instead of `boolean` <!--DOC-75-->
    - Updated `md5` description in data dictionary viewer to be more complete  <!--TT-400-->
* Creation of new entities:
    - `Copy Number Estimate` entity for copy number variation data <!--TT-439-->
    - `Copy Number Variation Workflow` entity for copy number variation pipeline metadata <!--TT-440-->
    - `Molecular Test` entity <!--TT-464-->
* Updates to all entities:
    - Updated all entities to include `batch_id` field for submission <!--TT-434-->
    - Updated all entities to include `downloadable` field <!--TT-501-->
    - Modified all file entities `file_size` field to be an integer <!--TT-457-->  <!--TT-403-->
    - Added support for creation of annotations on all entities in data model <!--DAT-1331-->
* Added new links / updated links between entities:
    - Created optional link between `follow_up` and `diagnosis` entities <!--TT-401-->
    - Created many-to-many link between `genomic_profile_harmonization_workflows` and `masked_somatic_mutations` <!--TT-530-->
    - Remove the restriction that prevents having alignment workflows to point to both `submitted_aligned_reads` and `submitted_unaligned_reads` <!--TT-422-->
* Updated the BCR XML endpoint to support submission of new entity relationships <!--DAT-1207-->
* Fixed description of `prior_malignancy` on `diagnosis` entity <!--TT-412-->
* Modified `sample` entity
    - Added new fields  <!--TT-399-->
        - `growth_rate`
        - `passage_count`
        - `catalog_reference`
        - `distributor_reference`
        - `distance_normal_to_tumor`
        - `biospecimen_laterality`
    - Added new permissible values to `composition` field  <!--TT-527-->
        - `Sorted Cells`
    - Added new permissible value to `sample_type` field <!--TT-509-->
        - `Next Generation Cancer Model`
    - Added new permissible values to `method_of_sample_procurement` field <!--TT-528-->
        - `Pancreatectomy`
        - `Whipple Procedure`
        - `Paracentesis`
    - Added new permissible values to `biospecimen_anatomic_site` field <!--TT-529-->
        - `Esophageal; Distal`
        - `Esophageal; Mid`
        - `Esophageal; Proximal`
        - `Hepatic Flexure`
        - `Rectosigmoid Junction`
    - Removed permissible values on `sample_type` <!--TT-510-->
        - `2D Classical Conditionally Reprogrammed Cells`
        - `2D Modified Conditionally Reprogrammed Cells`
        - `3D Organoid`
        - `3D Air-Liquid Interface Organoid`
        - `3D Neurosphere`
        - `Adherent Cell Line`
        - `Liquid Suspension Cell Line`
* Modified `aliquot` entity <!--TT-416-->
    - Added new fields
        - `no_matched_normal_wxs`
        - `no_matched_normal_wgs`
        - `no_matched_normal_targeted_sequencing`
        - `no_matched_normal_low_pass_wgs`
* Modified `read_group` entity
    - Updated descriptions on `read_group` entity  <!--TT-463-->
    - Added new field  <!--TT-415-->
        - `target_capture_kit`
    - Made `library_selection` a required field <!--TT-472-->
    - Removed duplicate `validators` field  <!--TT-484-->
    - Modified permissible values on `library_selection` field
        - Replaced `Affinity_Enrichment` with `Affinity Enrichment` <!--TT-447-->
        - Replaced `Poly-T_Enrichment` with `Poly-T Enrichment` <!--TT-449-->
        - Replaced `Hybrid_Selection` with `Hybrid Selection` <!--TT-451-->
        - Replaced `RNA_Depletion` with `rRNA Depletion` <!--TT-453-->
        - Replaced `Targeted Sequencing` with `Hybrid Selection` for FM-AD  <!--TT-505-->
    - Added new permissible value on `library_strand` field
        - `Not Applicable` <!--TT-402-->
    - Removed permissible values in `library_strategy` field <!--TT-442-->
        - `Amplicon`
        - `Validation`
        - `Other`
* Modified `diagnosis` entity
    - Added new fields  <!--TT-380-->
        - `ajcc_staging_system_edition`
        - `anaplasia_present`
        - `anaplasia_present_type`
        - `child_pugh_classification`
        - `cog_neuroblastoma_risk_group`
        - `cog_rhabdomyosarcoma_risk_group`
        - `enneking_msts_grade`
        - `enneking_msts_metastasis`
        - `enneking_msts_stage`
        - `enneking_msts_tumor_site`
        - `esophageal_columnar_dysplasia_degree`
        - `esophageal_columnar_metaplasia_present`
        - `first_symptom_prior_to_diagnosis`
        - `gastric_esophageal_junction_involvement`
        - `goblet_cells_columnar_mucosa_present`
        - `inpc_grade`
        - `inss_stage`
        - `irs_group`
        - `ishak_fibrosis_score`
        - `micropapillary_features`
        - `meduloblastoma_molecular_classification`
        - `metastasis_at_diagnosis`
        - `mitosis_karyorrhexis_index`
        - `peripancreatic_lymph_nodes_positive`
        - `peripancreatic_lymph_nodes_tested`
        - `synchronous_malignancy`
        - `supratentorial_localization`
        - `tumor_confined_to_organ_of_origin`
    - Added new permissible values to `vascular_invasion_present` field <!--TT-495-->
        - `Extramural`
        - `Intramural`
    - Updated `ann_arbor_clinical_stage` and `ann_arbor_pathologic_stage` fields  <!--TT-524-->
* Modified `aliquot` entity
    - Added new fields <!--TT-425-->
        - `selected_normal_wxs`
        - `selected_normal_wgs`
        - `selected_normal_targeted_sequencing`
        - `selected_normal_low_pass_wgs`
* Modified `project` entity
    - Added new fields
        - `request_submission`
        - `in_review` <!--TT-428-->
        - `submission_enabled` <!--TT-429-->
    - Removed duplicate `intended_release_date` field <!--TT-483-->
    - Made fields `not required` <!--TT-465-->
        - `disease_type`
        - `primary_site`
* Modified `case` entity
    - Updated descriptions  <!--TT-488-->
        - `disease_type`
        - `primary_site`
* Modified `demographic` entity
    - Added new fields <!--TT-379-->
        - `premature_at_birth`
        - `weeks_gestation_at_birth`
    - Made `submitter_id` a required field <!--TT-544-->
    - Changed type of `year_of_birth` from number to integer <!--TT-489-->
    - Changed type of `year_of_death` from number to integer <!--TT-490-->
* Modified `treatment` entity
    - Updated CDE code for `treatment_anatomic_site` <!--TT-492-->
    - Modified permissible values to `treatment_anatomic_site` field  <!--TT-493--> <!--TT-506-->
    - Made `submitter_id` a required field <!--TT-544-->
    - Added new permissible value to `treatment_intent_type` <!--TT-494-->
        - `Neoadjuvant`
    - Added new permissible value to `treatment_outcome` <!--TT-498-->
        - `Very Good Partial Response`
        - `Mixed Response`
        - `No Response`
    - Added new permissible value to `treatment_type` <!--TT-499-->
        - `Brachytherapy, High Dose`
        - `Brachytherapy, Low Dose`
        - `Radiation, 2D Conventional`
        - `Radiation, 3D Conformal`
        - `Radiation, Intensity-Modulated Radiotherapy`
        - `Radiation, Proton Beam`
        - `Radiation, Stereotactic Body`
        - `Radiation Therapy, NOS`
        - `Stereotactic Radiosurgery`
    - Removed field
        - `days_to_treatment`
    - Removed permissible values for `treatment_type` <!--TT-523-->
        - `Radiation`
        - `Radiation Therapy`
* Modified `analyte` entity
    - Removed duplicate `project` field <!--TT-486-->
* Modified `follow_up` entity
    - Added new permissible values to `comorbidity` field <!--TT-500-->
    - Added new fields <!--TT-365-->
        - `progression_or_recurrence_type`
        - `diabetes_treatment_type`
        - `reflux_treatment_type`
        - `barretts_esophagus_goblet_cells_present`
        - `karnofsky__performance_status`
        - `menopause_status`
        - `viral_hepatitis_serologies`
        - `reflux_treatment`
        - `pancreatitis_onset_year`
        - `comorbidity_method_of_diagnosis`
        - `risk_factor`
    - Removed fields <!--TT-517-->
        - `absolute_neutrophil`
        - `albumin`
        - `beta_2_microglobulin`
        - `bun`
        - `calcium`
        - `cea_level`
        - `colon_polyps_history`
        - `creatinine`
        - `crp`
        - `days_to_hiv_diagnosis`
        - `estrogen_receptor_percent_positive_ihc`
        - `estrogen_receptor_result_ihc`
        - `glucose`
        - `hemoglobin`
        - `her2_erbb2_percent_positive_ihc`
        - `her2_erbb2_result_fish`
        - `her2_erbb2_result_ihc`
        - `hiv_positive`
        - `hpv_status`
        - `iga`
        - `igg`
        - `igl_kappa`
        - `igl_lambda`
        - `igm`
        - `ldh_level`
        - `ldh_normal_range_upper`
        - `m_protein`
        - `microsatellite_instability_abnormal`
        - `platelet_count`
        - `progesterone_receptor_percent_positive_ihc`
        - `progesterone_receptor_result_ihc`
        - `total_protein`
        - `wbc`
* Modified `exposure` entity
    - Added new fields  <!--TT-383-->
        - `alcohol_days_per week`
        - `alcohol_drinks_per_day`
* Added `molecular_test` entity
    - Added new fields  <!--TT-384-->
        - `gene_symbol`
        - `second_gene_symbol`
        - `test_analyte_type`
        - `test_result`
        - `molecular_analysis_method`
        - `variant_type`
        - `molecular_consequence`
        - `chromosome`
        - `cytoband`
        - `exon`
        - `transcript`
        - `locus`
        - `dna_change`
        - `aa_change`
        - `rna_change`
        - `zygosity`
        - `histone_family`
        - `histone_variant`
        - `copy_number`
        - `antigen`
        - `test_value`
        - `test_units`
        - `specialized_molecular_test`
        - `ploidy`
        - `cell_count`
        - `loci_count`
        - `loci_abnormal_count`
        - `mismatch_repair_mutation`
        - `blood_test`
        - `blood_test_normal_range_upper`
        - `blood_test_normal_range_lower`
* Modified `biospecimen_supplement` entity
    - Added new permissible values to `data_format` field <!--TT-394-->
        - `TSV`
        - `BCR Biotab`
        - `BCR SSF XML`
        - `BCR PPS XML`
        - `FoundationOne XML`
        - `BCR Auxiliary XML`
    - Removed permissible values on `data_format` field <!--TT-397-->
        - `SSF`
        - `PPS`
* Modified `clinical_supplement` entity
    - Added new permissible values to `data_format` field <!--TT-395-->
        - `TSV`
        - `BCR OMF XML`
        - `BCR Biotab`
    - Removed field
        - `data_format`
* Modified `submitted_aligned_reads` entity
    - Added new permissible values to `experimental_strategy` field <!--TT-404-->
        - `Targeted Sequencing`
        - `Bisulfite-Seq`
        - `ChIP-Seq`
        - `ATAC-Seq`
    - Removed permissible values from `experimental_strategy` field <!--TT-404-->
        - `Validation`
        - `Total RNA-Seq`
* Modified `submitted_unaligned_reads` entity <!--TT-411-->
    - Added new field
        - `read_pair_number`
    - Added new permissible values to `experimental_strategy` field <!--TT-405-->
        - `Targeted Sequencing`
        - `Bisulfite-Seq`
        - `ChIP-Seq`
        - `ATAC-Seq`
* Modified `alignment_workflow` entity
    - Added new permissible values to `workflow_type` field  
        - `BWA with Mark Duplicates and BQSR` <!--TT-426-->
        - `BWA with BQSR` <!--TT-479-->
* Modified `copy_number_segment` entity
    - Removed permissible values for `data_type` field  <!--TT-441-->
        - `Gene Level Copy Number`
        - `Gene Level Copy Number Scores`
* Modified `submitted_genomic_profile` entity <!--TT-461-->
    - Added new permissible values for `data_type`
        - `Raw GCI Variant`
    - Added new permissible values for `data_category`
        - `Combined Nucleotide Variation`
    - Added new permissible values for `experimental_strategy`
        - `WGS`
    - Added new permissible values for `data_format`
        - `VCF`
* Modified `genomic_profile_harmonization_workflow` entity  <!--TT-466-->
    - Added new permissible value for `workflow_type`
        - `VCF LiftOver`
* Modified `simple_somatic_mutation` entity   <!--TT-467-->
    - Added new permissible values for `data_type`
        - `Raw GCI Variant`
    - Added new permissible values for `data_category`
        - `Combined Nucleotide Variation`
* Modified `masked_somatic_mutations` entity
    - Added new permissible value for `experimental_strategy` <!--TT-533-->
        - `Targeted Sequencing`
    - Removed permissible value for `experimental_strategy` <!--TT-532-->
        - `Validation`

### Bugs Fixed Since Last Release

* Fixed issue when `file_size` is specified as a float in submitted json file

## v.1.11


* __GDC Product__: GDC Data Dictionary
* __Release Date__: January 20, 2018


### New Features and Changes

* Added a link between the `sample` and `analyte` entities in the data model <!--TT-260--> <!--TT-259-->
* Added a link between the `sample` entity and other `sample` entities in the data model <!--TT-261-->
* Created `structural_variant_calling_workflow` entity <!--TT-270-->
* Created `structural_variation` entity <!--TT-207-->
* Removed `clinical_test` entity  <!--TT-232-->
* Removed `exon_expression` entity <!--TT-203-->
* Modified relationships of entities
    * Changed relationship between `submitted_genomic_profile` and `read_group` to one-to-many <!--TT-75-->
    * Changed relationship between `projects` and `masked_somatic_mutations` from one-to-one to many_to_one <!--TT-172-->
    * Changed relationship between `projects` and `aggregated_somatic_mutations` from one-to-one to many_to_one <!--TT-173-->
    * Changed relationship of `rna_expression_workflow` to downstream entities from one-to-one to one-to-many <!--TT-171-->
    * Changed relationship of `alignment_workflow` to downstream entities from many-to-one to many-to-many <!--TT-174-->
* Modified `project` entity
    - Added new state
        - `processed` <!--TT-266-->
    - Added new fields
        - `release_requested` <!--TT-267-->
        - `awg_review` <!--TT-267-->
        - `is_legacy` <!--TT-252-->
* Modified `clinical_supplement` entity
    - Added new `data_format` field <!--TT-271-->
        - `CDC JSON`
* Modified `case` entity
    - Enumerated `primary_site` field <!--TT-189-->
* Modified `diagnosis` entity
    - Enumerated fields
        - `primary_diagnosis` <!--TT-185-->
        - `tissue_or_organ_of_origin` <!--TT-190-->
        - `site_of_resection_or_biopsy` <!--TT-192-->
        - `tumor_grade` <!--TT-193-->
        - `morphology` <!--TT-186-->
        - `ajcc_clinical_stage` <!--TT-238-->
        - `ajcc_pathologic_stage` <!--TT-239-->
        - `laterality` <!--TT-242-->
        - `method_of_diagnosis` <!--TT-244-->
    - Changed type of fields
        - `year_of_diagnosis`: changed type to `int`<!--TT-121-->
        - `age_at_diagnosis`: changed type to `int` <!--TT-121-->
    - Added new permissible values to fields
        - `figo_stage`
            - `Stage IIC` <!--TT-241-->
        - `lymphatic_invasion_present`
            - `Not Reported` <!--TT-243-->
        - `perineural_invasion_present`
            - `Not Reported` <!--TT-245-->
        - `ann_arbor_clinical_stage`
            - `Not Reported` <!--TT-222-->  
            - `Unknown`<!--TT-240-->
        - `residual_disease`<!--TT-265-->
            - `Not Reported`
            - `Unknown`
        - `vascular_invasion_type`<!--TT-246-->
            - `Macro`
            - `Micro`
            - `No Vascular Invasion`
            - `Not Reported`
            - `Unknown`    
* Modified `exposure` entity
    - Enumerated fields
        - `alcohol_intensity` <!--TT-191-->
        - `alcohol_history` <!--TT-247-->
        - `cause_of_death` <!--TT-264-->
    - Removed `6` as a valid field from `tobacco_smoking_status` <!--TT-214-->
* Modified `demographic` entity
    - Enumerated field
        - `comorbidity` <!--TT-290-->
    - Added new permissible values to field
        - `cause_of_death` <!--TT-264-->
            - `Infection`
            - `Toxicity`
            - `Spinal Muscular Atrophy`
            - `End-stage Renal Disease`
* Modified `family history` entity
    - Enumerated fields
        - `relationship_primary_diagnosis` <!--TT-67-->
        - `relationship_type` <!--TT-67-->
* Modified `treatment` entity
    - Enumerated fields
        - `treatment_anatomic_site` <!--TT-221-->
        - `treatment_type` <!--TT-226-->
    - Changed type of fields
        - `days_to_treatment`: changed type to `int`<!--TT-213-->
        - `days_to_treatment_end`: changed type to `int`<!--TT-213-->
        - `days_to_treatment_start`: changed type to `int`<!--TT-213-->
* Modified `follow_up` entity
    - Enumerated field
        - `comorbidity` <!--TT-225-->
    - Removed `None` as valid field from `comorbidity` <!--TT-290-->
* Modified `demographic` entity
    - Added new field
        - `age_at_index` <!--TT-181-->
* Modified read_group entity <!--TT-200-->
    - Added new fields
        - `fragment_minimum_length`
        - `fragment_maximum_length`
        - `fragment_mean_length`
        - `fragment_standard_deviation_length`
* Modified `slide image` entity <!--TT-262-->
    - Added new data_formats
        - `JPEG` and `TIFF`
    - Added new data_type
        - `Cell Culture`
    - Added new experimental strategy
        - `Cell Culture`
    - Added new property
        - `Magnification`
    - Added new field
        - `date_time`
* Modified `sample` entity
    - Added new permissible values to fields <!--TT-274-->
        - `method_of_sample_procurement`
            - `Autopsy`  
        - `sample_type` <!--TT-257-->
            - `2D Classical Conditionally Reprogrammed Cells`
            - `2D Modified Conditionally Reprogrammed Cells`
            - `3D Organoid`
            - `3D Air-Liquid Interface Organoid`
            - `3D Neurosphere`
            - `Adherent Cell Line`
            - `Liquid Suspension Cell Line`
        - `composition` <!--TT-258-->
            - `2D Classical Conditionally Reprogrammed Cells`
            - `2D Modified Conditionally Reprogrammed Cells`
            - `3D Organoid`
            - `3D Air-Liquid Interface Organoid`
            - `3D Neurosphere`
            - `Adherent Cell Line`
            - `Liquid Suspension Cell Line`
* Modified `genomic_profile_harmonization_workflow` entity
    - Added new permissible values to field
        - `workflow_type` <!--TT-117-->
            - `GENIE Simple Somatic Mutation`
            - `GENIE Copy Number Variation`
* Modified `somatic_mutation_calling_workflow` entity
    - Added new permissible values to field
        - `workflow_type`<!--TT-206-->
            - `Pindel`
* Modified `rna_expression_workflow` entity <!--TT-202--> <!--TT-208-->
    - Added new permissible values to field
        - `workflow_type`
            - `RSEM - Quantification`
            - `STAR - Counts`
            - `RNA-SeQC - Counts`
            - `RNA-SeQC - FPKM`
            - `Kallisto - Quantification`
            - `Kallisto - HDF5`
* Modified `somatic_aggregation_workflow` <!--TT-202--> <!--TT-208-->
    - Added new permissible values to field
        - `workflow_type`
            - `Pindel Variant Aggregation and Masking`
            - `GENIE Variant Aggregation and Masking`
* Modified `gene_expression entity` <!--TT-204-->
    - Added new data_types
        - `Isoform Expression Quantification`
        - `Exon Expression Quantification`
    - Added new data_format
        - `HDF5`


### Bugs Fixed Since Last Release
* Fixed issue when submitting `tobacco_smoking_status` via tsv <!--TT-205-->

### Known Issues and Workarounds



## Release with API v1.10.0

* __GDC Product__: GDC Data Dictionary
* __Release Date__: August 22, 2017


### New Features and Changes

* Created `follow_up` entity to support longitudinal clinical data <!--TT-65-->
* Deprecated `clinical_test` entity <!--DOC-67-->
* Modified acceptable values for Read Group properties <!--TT-9,TT-76-->
    - `library_selection` : "Hybrid Selection, Affinity Enrichment, Poly-T Enrichment, Random, rRNA Depletion, miRNA Size Fractionation, Targeted Sequencing"
    - `library_strategy` : "Targeted Sequencing"
* Modified Diagnosis entity <!--TT-64-->
    - Added field `iss_stage`
    - Added field `best_overall_response`
    - Added field `days_to_best_overall_response`
    - Added field `progression_free_survival`
    - Added field `progression_free_survival_event`
    - Added field `overall_survival`
    - Added field `days_to_diagnosis` <!--TT-91-->
* Modified Treatment entity <!--TT-66-->
    - Added field `regimen_or_line_of_therapy`
* Modified Demographic entity <!--TT-83-->
    - Added field `cause_of_death`
    - Added field `days_to_birth`
    - Added field `days_to_death`
    - Added field `vital_status`
* Modified Case entity <!--TT-84-->
    - Added field `days_to_lost_to_followup`
    - Added field `lost_to_followup`
    - Added field `index_date`
* Added new tumor code, tumor id, and sample types to Sample entity to support OCG <!--TT-85, TT-68-->
    - `tumor_code` : "Acute leukemia of Ambiguous Lineage (ALAL), Lymphoid Normal, Tumor Adjacent Normal - Post Neo Adjuvant Therapy"
    - `tumor_code_id` : "15, 17, 18"
* Created `somatic_mutation_index` entity <!--TT-92-->
* Updated caDSR CDE links in data dictionary <!--DAT-794-->
* Added new `sample_type` : `tumor` to sample entity <!--TT-77-->
* Made `classification_of_tumor` on diagnosis entity non-required <!--DAT-203-->
* Added support for FM-AD to Genomic Profile Harmonization Workflow entity <!--DAT-985-->
* Added `data_type` : `Gene Level Copy Number Scores` to Copy Number Segment entity <!--TT-94-->

### Bugs Fixed Since Last Release

None

### Known Issues and Workarounds

*  Portion `weight` property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->

## Release with API v1.7.1

* __GDC Product__: GDC Data Dictionary
* __Release Date__: March 16, 2017


### New Features and Changes

* Added "submittable" property to all entities <!--DAT-215-->
* Changed Read Group to category biospecimen <!--DAT-216-->
* Added many new clinical properties available for submission <!--DAT-210, DAT-31, DAT-226, DAT-205-->
* Added sample codes from Office of Cancer Genomics (OCG) to analyte and aliquot <!--DAT-170-->
* Slides can now be attached to sample rather than just portion <!--DAT-205-->
* `sample_type_id` is no longer required when submitting sample entities <!--DAT-233-->
* `analyte_type_id` is no longer required when submitting aliquot and analyte entities<!--DAT-255-->
* Clinical Test Entity is created for storing results of a variety of potential clinical tests related to the diagnosis - <!--DAT-223-->
* Genomic Profiling Report entity created for storing particular derived sequencing results <!--DAT-229-->
* Structural Variation entity created <!--DAT-229-->
* Project entity includes new field "Intended Release Date" <!--API-143-->
* Project entity includes new field "Releasable" <!--API-157-->

### Bugs Fixed Since Last Release

None

### Known Issues and Workarounds

None

## Release with API v1.3.1

* __GDC Product__: GDC Data Dictionary
* __Release Date__: September 7, 2016


### New Features and Changes

* Clinical Supplement entities can have `data_format` set to OMF.
* Biospecimen Supplement entities can have `data_format` set to SSF or PPS.
* Read group `instrument_model` can be set to "Illumina HiSeq 4000".
* Category of Slide entities in the GDC Data Model has changed from `data_bundle` to `biospecimen`.

### Bugs Fixed Since Last Release

None

### Known Issues and Workarounds

None


# Authentication

## Requirements

Accessing the GDC Data Submission Portal requires eRA Commons credentials with appropriate dbGaP authorization. To learn more about obtaining the required credentials and authorization, see [Obtaining Access to Submit Data]( https://gdc.cancer.gov/submit-data/obtaining-access-submit-data).

## Authentication via eRA Commons

Users can log into the GDC Data Submission Portal with eRA Commons credentials by clicking the "Login" button. If authentication is successful, the user will be redirected to the GDC Data Submission Portal front page and the user's eRA Commons username will be displayed in the upper right corner of the screen.

## GDC Authentication Tokens

The GDC Data Submission Portal provides authentication tokens for use with the GDC Data Transfer Tool or GDC API. To download a token:

1. Log into the GDC using eRA Commons credentials
1. Click the username in the top right corner of the screen
1. Select the "Download Token" option

![Token Download Link](images/GDC_Submission_Portal_Token_Download.png)

For more information about authentication tokens, see [Data Security](../../Data/Data_Security/Data_Security/#authentication-tokens).

**NOTE:** The authentication token should be kept in a secure location, as it allows access to all data accessible by the associated user account.

## Logging Out

To log out of the GDC, click the username in the top right corner of the screen and select the "Logout" option. Users will automatically be logged out after 15 minutes of inactivity.


# Data Upload Walkthrough

This guide details step-by-step procedures for different aspects of the GDC Data Submission process and how they relate to the GDC Data Model and structure. The first sections of this guide break down the submission process and associate each step with the Data Model. Additional sections are detailed below for strategies on expediting data submission, using features of the GDC Data Submission Portal, and best practices used by the GDC.

## GDC Data Model Basics

Pictured below is the submittable subset of the GDC Data Model: a roadmap for GDC data submission. Each oval node in the graphic represents an entity: a logical unit of data related to a specific clinical, biospecimen, or file facet in the GDC. An entity includes a set of fields, the associated values, and information about its related node associations. All submitted entities require a connection to another entity type, based on the GDC Data Model, and a `submitter_id` as an identifier. This walkthrough will go through the submission of different entities. The completed (submitted) portion of the entity process will be highlighted in __blue__.

[![GDC Data Model 1](images/GDC-Data-Model-None.png)](images/GDC-Data-Model-None.png "Click to see the full image.")

# Case Submission

The `case` is the center of the GDC Data Model and usually describes a specific patient. Each `case` is connected to a `project`.  Different types of clinical data, such as `diagnoses` and `exposures`, are connected to the `case` to describe the case's attributes and medical information.   

[![GDC Data Model 2](images/GDC-Data-Model-Case.png)](images/GDC-Data-Model-Case.png "Click to see the full image.")

The main entity of the GDC Data Model is the `case`, each of which must be registered beforehand with [dbGaP](https://www.ncbi.nlm.nih.gov/sra/docs/submitdbgap) under a unique `submitter_id`. The first step to submitting a `case` is to consult the [Data Dictionary](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#data-dictionary-viewer), which details the fields that are associated with a `case`, the fields that are required to submit a `case`, and the values that can populate each field. Dictionary entries are available for all entities in the GDC Data Model.

[![Dictionary Case](images/Dictionary_Case.png)](images/Dictionary_Case.png "Click to see the full image.")

Submitting a [__Case__](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=case) entity requires:

* __`submitter_id`:__ A unique key to identify the `case`
* __`projects.code`:__ A link to the `project`

The submitter ID is different from the universally unique identifier (UUID), which is based on the [UUID Version 4 Naming Convention](https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_4_.28random.29). The UUID can be accessed under the `<entity_type>_id` field for each entity. For example, the `case` UUID can be accessed under the `case_id` field. The UUID is either assigned to each entity automatically or can be submitted by the user. Submitter-generated UUIDs cannot be uploaded in `submittable_data_file` entity types. See the [Data Model Users Guide](https://docs.gdc.cancer.gov/Data/Data_Model/GDC_Data_Model/#gdc-identifiers) for more details about GDC identifiers.

The `projects.code` field connects the `case` entity to the `project` entity.  The rest of the entity connections use the `submitter_id` field instead.  

The `case` entity can be added in JSON or TSV format. A template for any entity in either of these formats can be found in the Data Dictionary at the top of each page. Templates populated with `case` metadata in both formats are displayed below.  

=== "JSON"

    ```JSON
    {
        "type": "case",
        "submitter_id": "PROJECT-INTERNAL-000055",
        "projects": {
            "code": "INTERNAL"
        }
    }
    ```

=== "TSV"

    ```TSV
    type  submitter_id  projects.code
    case  PROJECT-INTERNAL-000055 INTERNAL   
    ```

>__Note:__ JSON and TSV formats handle links between entities (`case` and `project`) differently.  JSON includes the `code` field nested within `projects` while TSV appends `code` to `projects` with a period.  


## Uploading the Case Submission File

The file detailed above can be uploaded using the GDC Data Submission Portal and the GDC API as described below:

### Upload Using the GDC Data Submission Portal

An example of a `case` upload is detailed below. The [GDC Data Submission Portal](https://gdc.cancer.gov/submit-data/gdc-data-submission-portal) is equipped with a wizard window to facilitate the upload and validation of entities.

#### 1. Upload Files

Choosing _'UPLOAD'_ from the project dashboard will open the Upload Data Wizard.

[![GDC Submission Wizard Upload Files](images/GDC_Submission_Wizard_Upload_2.png)](images/GDC_Submission_Wizard_Upload_2.png "Click to see the full image.")

Files containing one or more entities can be added either by clicking on `CHOOSE FILE(S)` or using drag and drop. Files can be removed from the Upload Data Wizard by clicking on the garbage can icon that is displayed next to the file after the file is selected for upload.

#### 2. Validate Entities

The __Validate Entities__ stage acts as a safeguard against submitting incorrectly formatted data to the GDC Data Submission Portal. During the validation stage, the GDC API will validate the content of uploaded entities against the Data Dictionary to detect potential errors. Invalid entities will not be processed and must be corrected by the user and re-uploaded before being accepted. A validation error report provided by the system can be used to isolate and correct errors.

When the first file is added, the wizard will move to the Validate section and the user can continue to add files. When all files have been added, choosing `VALIDATE` will run a test to check if the entities are valid for submission.

[![GDC Submission Wizard Validate Files](images/GDC_Submission_Portal_Validate.png)](images/GDC_Submission_Portal_Validate.png "Click to see the full image.")

#### 3. Commit or Discard Files
If the upload contains valid entities, a new transaction will appear in the latest transactions panel with the option to `COMMIT` or `DISCARD` the data. Entities contained in these files can be committed (applied) to the project or discarded using these two buttons.

If the upload contains invalid files, a transaction will appear with a FAILED status. Invalid files will need to be either corrected and re-uploaded or removed from the submission. If more than one file is uploaded and at least one is not valid, the validation step will fail for all files.  

[![Commit_Discard](images/GDC_Submission_CommitDiscard.png)](images/GDC_Submission_CommitDiscard.png "Click to see the full image.")


### Upload Using the GDC API

The API has a much broader range of functionality than the Data Wizard. Entities can be created, updated, and deleted through the API. See the [API Submission User Guide](https://docs.gdc.cancer.gov/API/Users_Guide/Submission/#creating-and-updating-entities) for a more detailed explanation and for the rest of the functionalities of the API. Generally, uploading an entity through the API can be performed using a command similar to the following:

=== "Shell"

    ```Shell
    curl --header "X-Auth-Token: $token" --request POST --data @CASE.json https://api.gdc.cancer.gov/v0/submission/GDC/INTERNAL/_dry_run?async=true
    ```

CASE.json is detailed below.

=== "JSON"

    ```json
    {
        "type": "case",
        "submitter_id": "PROJECT-INTERNAL-000055",
        "projects": {
            "code": "INTERNAL"
        }
    }
    ```

In this example, the `_dry_run` marker is used to determine if the entities can be validated, but without committing any information. If a command passed through the `_dry_run` works, the command will work when it is changed to `commit`. For more information please go to [Dry Run Transactions](https://docs.gdc.cancer.gov/API/Users_Guide/Submission/#dry-run-transactions).

>__Note:__ Submission of TSV files is also supported by the GDC API.

Next, the file can either be committed (applied to the project) through the Data Submission Portal as before, or another API query can be performed that will commit the file to the project. The transaction number in the URL (467) is printed to the console during the first step of API submission and can also be retrieved from the [Transactions](Data_Submission_Process.md#transactions) tab in the Data Submission Portal.

=== "Shell"

    ```Shell
    curl --header "X-Auth-Token: $token" --request POST https://api.gdc.cancer.gov/v0/submission/GDC/INTERNAL/transactions/467/commit?async=true
    ```

# Clinical Data Submission

Typically, a submission project will include additional information about a `case` such as `demographic`, `diagnosis`, or `exposure` data.

## Clinical Data Requirements

For the GDC to release a project there is a minimum number of clinical properties that are required.  Minimal GDC requirements for each project includes age, gender, and diagnosis information.  Other [requirements](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-entity-list&anchor=clinical) may be added when the submitter is approved for submission to the GDC.

[![GDC Data Model Clinical](images/GDC-Data-Model-Clinical.png)](images/GDC-Data-Model-Clinical.png "Click to see the full image.")

## Submitting a Demographic Entity to a Case

The `demographic` entity contains information that characterizes the `case` entity.  

Submitting a [__Demographic__](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=demographic) entity requires:

* __`submitter_id`:__ A unique key to identify the `demographic` entity.
* __`cases.submitter_id`:__ The unique key that was used for the `case` that links the `demographic` entity to the `case`.
* __`ethnicity`:__ An individual's self-described social and cultural grouping, specifically whether an individual describes themselves as Hispanic or Latino. The provided values are based on the categories defined by the U.S. Office of Management and Business and used by the U.S. Census Bureau.
* __`gender`:__ Text designations that identify gender. Gender is described as the assemblage of properties that distinguish people on the basis of their societal roles.
* __`race`:__ An arbitrary classification of a taxonomic group that is a division of a species. It usually arises as a consequence of geographical isolation within a species and is characterized by shared heredity, physical attributes and behavior, and in the case of humans, by common history, nationality, or geographic distribution. The provided values are based on the categories defined by the U.S. Office of Management and Business and used by the U.S. Census Bureau.

=== "JSON"

    ```JSON
    {
        "type": "demographic",
        "submitter_id": "PROJECT-INTERNAL-000055-DEMOGRAPHIC-1",
        "cases": {
            "submitter_id": "PROJECT-INTERNAL-000055"
        },
        "ethnicity": "not hispanic or latino",
        "gender": "male",
        "race": "asian",
    }
    ```

=== "TSV"

    ```TSV
    type	cases.submitter_id	ethnicity	gender	race
    demographic	PROJECT-INTERNAL-000055	not hispanic or latino	male	asian
    ```

## Submitting a Diagnosis Entity to a Case

Submitting a [__Diagnosis__](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=diagnosis) entity requires:

* __`submitter_id`:__ A unique key to identify the `diagnosis` entity.
* __`cases.submitter_id`:__ The unique key that was used for the `case` that links the `diagnosis` entity to the `case`.
* __`age_at_diagnosis`:__ Age at the time of diagnosis expressed in number of days since birth.
* __`days_to_last_follow_up`:__  Time interval from the date of last follow up to the date of initial pathologic diagnosis, represented as a calculated number of days.
* __`days_to_last_known_disease_status`:__ Time interval from the date of last follow up to the date of initial pathologic diagnosis, represented as a calculated number of days.
* __`days_to_recurrence`:__ Time interval from the date of new tumor event including progression, recurrence and new primary malignancies to the date of initial pathologic diagnosis, represented as a calculated number of days.
* __`last_known_disease_status`:__  The state or condition of an individual's neoplasm at a particular point in time.
* __`morphology`:__  The third edition of the International Classification of Diseases for Oncology, published in 2000 used principally in tumor and cancer registries for coding the site (topography) and the histology (morphology) of neoplasms. The study of the structure of the cells and their arrangement to constitute tissues and, finally, the association among these to form organs. In pathology, the microscopic process of identifying normal and abnormal morphologic characteristics in tissues, by employing various cytochemical and immunocytochemical stains. A system of numbered categories for representation of data.
* __`primary_diagnosis`:__  Text term for the structural pattern of cancer cells used to define a microscopic diagnosis.
* __`progression_or_recurrence`:__ Yes/No/Unknown indicator to identify whether a patient has had a new tumor event after initial treatment.
* __`site_of_resection_or_biopsy`:__ The third edition of the International Classification of Diseases for Oncology, published in 2000, used principally in tumor and cancer registries for coding the site (topography) and the histology (morphology) of neoplasms. The description of an anatomical region or of a body part. Named locations of, or within, the body. A system of numbered categories for representation of data.
* __`tissue_or_organ_of_origin`:__ Text term that describes the anatomic site of the tumor or disease.
* __`tumor_grade`:__ Numeric value to express the degree of abnormality of cancer cells, a measure of differentiation and aggressiveness.
* __`tumor_stage`:__ The extent of a cancer in the body. Staging is usually based on the size of the tumor, whether lymph nodes contain cancer, and whether the cancer has spread from the original site to other parts of the body. The accepted values for tumor_stage depend on the tumor site, type, and accepted staging system. These items should accompany the tumor_stage value as associated metadata.
* __`vital_status`:__ The survival state of the person registered on the protocol.

=== "JSON"

    ```JSON
    {
        "type": "diagnosis",
        "submitter_id": "PROJECT-INTERNAL-000055-DIAGNOSIS-1",
        "cases": {
            "submitter_id": "GDC-INTERNAL-000099"
        },
        "age_at_diagnosis": 10256,
        "days_to_last_follow_up": 34,
        "days_to_last_known_disease_status": 34,
        "days_to_recurrence": 45,
        "last_known_disease_status": "Tumor free",
        "morphology": "8260/3",
        "primary_diagnosis": "ACTH-producing tumor",
        "progression_or_recurrence": "no",
        "site_of_resection_or_biopsy": "Lung, NOS",
        "tissue_or_organ_of_origin": "Lung, NOS",
        "tumor_grade": "Not Reported",
        "tumor_stage": "stage i",
        "vital_status": "alive"
    }
    ```

=== "TSV"

    ```TSV
    type	submitter_id	cases.submitter_id	age_at_diagnosis    days_to_last_follow_up	days_to_last_known_disease_status	days_to_recurrence	last_known_disease_status	morphology	primary_diagnosis	progression_or_recurrence	site_of_resection_or_biopsy	tissue_or_organ_of_origin	tumor_grade	tumor_stage	vital_status
    diagnosis	PROJECT-INTERNAL-000055-DIAGNOSIS-1	GDC-INTERNAL-000099	10256	34	34	45	Tumor free	8260/3	ACTH-producing tumor	no	Lung, NOS	Lung, NOS	not reported	stage i	alive
    ```

### Submitting an Exposure Entity to a Case

Submitting an [__Exposure__](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=exposure) entity does not require any information besides a link to the `case` and a `submitter_id`.  The following fields are optionally included:  

* __`alcohol_history`:__ A response to a question that asks whether the participant has consumed at least 12 drinks of any kind of alcoholic beverage in their lifetime.
* __`alcohol_intensity`:__ Category to describe the patient's current level of alcohol use as self-reported by the patient.
* __`alcohol_days_per_week`:__ Numeric value used to describe the average number of days each week that a person consumes an alchoolic beverage.
* __`years_smoked`:__ Numeric value (or unknown) to represent the number of years a person has been smoking.
* __`tobacco_smoking_onset_year`:__ The year in which the participant began smoking.
* __`tobacco_smoking_quit_year`:__ The year in which the participant quit smoking.

=== "JSON"

    ```JSON
    {
        "type": "exposure",
        "submitter_id": "PROJECT-INTERNAL-000055-EXPOSURE-1",
        "cases": {
            "submitter_id": "PROJECT-INTERNAL-000055"
        },
        "alcohol_history": "yes",
        "alcohol_intensity": "Drinker",
        "alcohol_days_per_week": 2,
        "years_smoked": 5,
        "tobacco_smoking_onset_year": 2007,
        "tobacco_smoking_quit_year": 2012
    }
    ```

=== "TSV"

    ```TSV
    type	submitter_id	cases.submitter_id	alcohol_history alcohol_intensity   alcohol_days_per_week years_smoked  tobacco_smoking_onset_year  tobacco_smoking_quit_year
    exposure	PROJECT-INTERNAL-000055-EXPOSURE-1	PROJECT-INTERNAL-000055	yes Drinker 2 5 2007    2012
    ```

>__Note:__ Submitting a clinical entity uses the same conventions as submitting a `case` entity (detailed above).


# Biospecimen Submission

One of the main features of the GDC is the genomic data harmonization workflow. Genomic data is connected the case through biospecimen entities.  The `sample` entity describes a biological piece of matter that originated from a `case`.  Subsets of the `sample` such as `portions` and `analytes` can optionally be described.  The `aliquot` originates from a `sample` or `analyte` and describes the nucleic acid extract that was sequenced. The `read_group` entity describes the resulting set of reads from one sequencing lane.

## Sample Submission

[![GDC Data Model 3](images/GDC-Data-Model-Sample.png)](images/GDC-Data-Model-Sample.png "Click to see the full image.")

A `sample` submission has the same general structure as a `case` submission as it will require a unique key and a link to the `case`.  However, `sample` entities require four additional values:  

* `tissue_type`
* `tumor_descriptor`
* `specimen_type`
* `preservation_method`

This peripheral data is required because it is necessary for the data to be interpreted. For example, an investigator using this data would need to know whether the `sample` came from tumor or normal tissue.  

[![Dictionary Sample](images/Dictionary_Sample.png)](images/Dictionary_Sample.png "Click to see the full image.")

Submitting a [__Sample__](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=sample) entity requires:

* __`submitter_id`:__ A unique key to identify the `sample`.
* __`cases.submitter_id`:__ The unique key that was used for the `case` that links the `sample` to the `case`.
* __`tissue_type`:__ Text term that represents a description of the kind of tissue collected with respect to disease status or proximity to tumor tissue.
* __`tumor_descriptor`:__ Text that describes the kind of disease present in the tumor specimen as related to a specific timepoint.
* __`specimen_type`:__ The type of a material sample taken from a biological entity for testing, diagnostic, propagation, treatment or research purposes. This includes particular types of cellular molecules, cells, tissues, organs, body fluids, embryos, and body excretory substances.
* __`preservation_method`:__ Text term that represents the method used to preserve the sample.

>__Note:__ The `case` must be "committed" to the project before a `sample` can be linked to it.  This also applies to all other links between entities.

=== "JSON"

    ```JSON
    {
        "type": "sample",
        "cases": {
            "submitter_id": "PROJECT-INTERNAL-000055"
        },
        "submitter_id": "Blood-00001SAMPLE_55",
        "tissue_type": "Normal",
        "tumor_descriptor": "Not Applicable",
        "specimen_type": "Peripheral Blood NOS",
        "preservation_method": "Frozen"
    }
    ```
=== "TSV"

    ```TSV
    type	cases.submitter_id	submitter_id	tissue_type	specimen_type	tumor_descriptor	preservation_method
    sample	PROJECT-INTERNAL-000055	Blood-00001SAMPLE_55	Normal	Peripheral Blood NOS	Not Applicable	Frozen
    ```

## Portion, Analyte and Aliquot Submission

[![GDC Data Model 4](images/GDC-Data-Model-Aliquot.png)](images/GDC-Data-Model-Aliquot.png "Click to see the full image.")

Submitting a [__Portion__](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=portion) entity requires:

* __`submitter_id`:__ A unique key to identify the `portion`.
* __`samples.submitter_id`:__ The unique key that was used for the `sample` that links the `portion` to the `sample`.

=== "JSON"

    ```JSON
    {
        "type": "portion",
        "submitter_id": "Blood-portion-000055",
        "samples": {
            "submitter_id": "Blood-00001SAMPLE_55"
        }
    }    
    ```

=== "TSV"

    ```TSV
    type	submitter_id	samples.submitter_id
    portion	Blood-portion-000055	Blood-00001SAMPLE_55
    ```

Submitting an [__Analyte__](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=analyte) entity requires:

* __`submitter_id`:__ A unique key to identify the `analyte`.
* __`portions.submitter_id`:__ The unique key that was used for the `portion` that links the `analyte` to the `portion`.
* __`analyte_type`:__ Text term that represents the kind of molecular specimen analyte.

=== "JSON"

    ```JSON
    {
        "type": "analyte",
        "portions": {
            "submitter_id": "Blood-portion-000055"
        },
        "analyte_type": "DNA",
        "submitter_id": "Blood-analyte-000055"
    }
    ```

=== "TSV"

    ```TSV
    type	portions.submitter_id	analyte_type	submitter_id
    analyte	Blood-portion-000055	DNA	Blood-analyte-000055
    ```

Submitting an [__Aliquot__](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=aliquot) entity requires:

* __`submitter_id`:__ A unique key to identify the `aliquot`.
* __`analytes.submitter_id`:__ The unique key that was used for the `analyte` that links the `aliquot` to the `analyte`.

=== "JSON"

    ```JSON
    {
        "type": "aliquot",
        "submitter_id": "Blood-00021-aliquot55",
        "analytes": {
            "submitter_id": "Blood-analyte-000055"
        }
    }
    ```

=== "TSV"

    ```TSV
    type	submitter_id	analytes.submitter_id
    aliquot	Blood-00021-aliquot55	Blood-analyte-000055
    ```

>__Note:__ `aliquot` entities can be directly linked to `sample` entities via the `samples.submitter_id`. The `portion` and `analyte` entities are not required for submission.

## Read Group Submission

[![GDC Data Model 5](images/GDC-Data-Model-RG.png)](images/GDC-Data-Model-RG.png "Click to see the full image.")

Information about sequencing reads is necessary for downstream analysis, thus the `read_group` entity requires more fields than the other Biospecimen entities (`sample`, `portion`, `analyte`, `aliquot`).

Submitting a [__Read Group__](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=read_group) entity requires:

* __`submitter_id`:__ A unique key to identify the `read_group`.
* __`aliquots.submitter_id`:__ The unique key that was used for the `aliquot` that links the `read_group` to the `aliquot`.
* __`experiment_name`:__ Submitter-defined name for the experiment.
* __`is_paired_end`:__ Are the reads paired end? (Boolean value: `true` or `false`).
* __`library_name`:__ Name of the library.
* __`library_strategy`:__ Library strategy.
* __`platform`:__ Name of the platform used to obtain data.
* __`read_group_name`:__ The name of the `read_group`.
* __`read_length`:__ The length of the reads (integer).
* __`sequencing_center`:__ Name of the center that provided the sequence files.
* __`library_selection`:__ Library Selection Method.
* __`target_capture_kit`:__ Description that can uniquely identify a target capture kit. Suggested value is a combination of vendor, kit name, and kit version.

=== "JSON"
    
    ```JSON
    {
        "type": "read_group",
        "submitter_id": "Blood-00001-aliquot_lane1_barcodeACGTAC_55",
        "experiment_name": "Resequencing",
        "is_paired_end": true,
        "library_name": "Solexa-34688",
        "library_strategy": "WXS",
        "platform": "Illumina",
        "read_group_name": "205DD.3-2",
        "read_length": 75,
        "sequencing_center": "BI",
        "library_selection": "Hybrid Selection",
        "target_capture_kit": "Custom MSK IMPACT Panel - 468 Genes",
        "aliquots":
            {
                "submitter_id": "Blood-00021-aliquot55"
            }    
    }
    ```

=== "TSV"

    ```TSV
    type	submitter_id	experiment_name	is_paired_end	library_name    library_selection	library_strategy	platform	read_group_name	read_length	sequencing_center   target_capture_kit	aliquots.submitter_id
    read_group	Blood-00001-aliquot_lane1_barcodeACGTAC_55	Resequencing	true	Solexa-34688    Hybrid Selection	WXS	Illumina	205DD.3-2	75	BI	Custom MSK IMPACT Panel - 468 Genes Blood-00021-aliquot55
    ```

>__Note:__ Submitting a biospecimen entity uses the same conventions as submitting a `case` entity (detailed above).

# Experiment Data Submission

Several types of experiment data can be uploaded to the GDC.  The `submitted_aligned_reads` and `submitted_unaligned_reads` files are associated with the `read_group` entity, while the array-based files such as the `submitted_tangent_copy_number` are associated with the `aliquot` entity.  Each of these file types are described in their respective entity submission and are uploaded separately using the [GDC API](https://docs.gdc.cancer.gov/API/Users_Guide/Getting_Started/) or the [GDC Data Transfer Tool](https://gdc.cancer.gov/access-data/gdc-data-transfer-tool).  

[![GDC Data Model 6](images/GDC-Data-Model-Reads.png)](images/GDC-Data-Model-Reads.png "Click to see the full image.")

Before the experiment data file can be submitted, the GDC requires that the user provides information about the file as a `submittable_data_file` entity. This includes file-specific data needed to validate the file and assess which analyses should be performed. Sequencing data files can be submitted as `submitted_aligned_reads` or `submitted_unaligned_reads`.

Submitting a [__Submitted Aligned-Reads__](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=submitted_aligned_reads) ([__Submitted Unaligned-Reads__](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=submitted_unaligned_reads)) entity requires:

* __`submitter_id`:__ A unique key to identify the `submitted_aligned_reads`.
* __`read_groups.submitter_id`:__ The unique key that was used for the `read_group` that links the `submitted_aligned_reads` to the `read_group`.
* __`data_category`:__ Broad categorization of the contents of the data file.
* __`data_format`:__ Format of the data files.
* __`data_type`:__ Specific content type of the data file. (must be "Aligned Reads").
* __`experimental_strategy`:__ The sequencing strategy used to generate the data file.
* __`file_name`:__ The name (or part of a name) of a file (of any type).
* __`file_size`:__ The size of the data file (object) in bytes.
* __`md5sum`:__ The 128-bit hash value expressed as a 32 digit hexadecimal number used as a file's digital fingerprint.


=== "JSON"
    
    ```JSON
    {
        "type": "submitted_aligned_reads",
        "submitter_id": "Blood-00001-aliquot_lane1_barcodeACGTAC_55.bam",
        "data_category": "Raw Sequencing Data",
        "data_format": "BAM",
        "data_type": "Aligned Reads",
        "experimental_strategy": "WGS",
        "file_name": "test.bam",
        "file_size": 38,
        "md5sum": "aa6e82d11ccd8452f813a15a6d84faf1",
        "read_groups": [
            {
                "submitter_id": "Primary_Tumor_RG_86-1"
            }
        ]
    }
    ```

=== "TSV"

    ```TSV
    type	submitter_id	data_category	data_format	data_type	experimental_strategy	file_name	file_size	md5sum	read_groups.submitter_id#1
    submitted_aligned_reads	Blood-00001-aliquot_lane1_barcodeACGTAC_55.bam	Raw Sequencing Data	BAM	Aligned Reads	WGS	test.bam	38	aa6e82d11ccd8452f813a15a6d84faf1	Primary_Tumor_RG_86-1
    ```

>__Note:__ For details on submitting experiment data associated with more than one `read_group` entity, see the [Tips for Complex Submissions](#submitting-complex-data-model-relationships) section.    

## Uploading the Submittable Data File to the GDC

The submittable data file can be uploaded when it is registered with the GDC. A submittable data file is registered when its corresponding entity (e.g. `submitted_unaligned_reads`) is uploaded and committed. It is important to note that the Harmonization process does not occur on these submitted files until the user clicks the [`Request Submission`](Data_Submission_Process.md#release) button. Uploading the file can be performed with either the [GDC Data Transfer Tool](https://gdc.cancer.gov/access-data/gdc-data-transfer-tool) or the [GDC API](https://docs.gdc.cancer.gov/API/Users_Guide/Getting_Started/). Other types of data files such as clinical supplements, biospecimen supplements, and pathology reports are uploaded to the GDC in the same way. Supported data file formats are listed at the [GDC Data Dictionary](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-entity-list&anchor=submittable_data_file).

__GDC Data Transfer Tool:__ A file can be uploaded using its UUID (which can be retrieved from the GDC Submission Portal or API) once it is registered.

[![UUID Location](images/GDC_Submission_UUID_location.png)](images/GDC_Submission_UUID_location.png "Click to see the full image.")

The following command can be used to upload the file:

=== "Shell"

    ```shell
    gdc-client upload --project-id PROJECT-INTERNAL --identifier a053fad1-adc9-4f2d-8632-923579128985 -t $token -f $path_to_file
    ```

Additionally a manifest can be downloaded from the Submission Portal and passed to the Data Transfer Tool. This will allow for the upload of more than one `submittable_data_file`:

=== "Shell"

    ```Shell
    gdc-client upload -m manifest.yml -t $token
    ```

__API Upload:__  A `submittable_data_file` can be uploaded through the API by using the `/submission/$PROGRAM/$PROJECT/files` endpoint.  The following command would be typically used to upload a file:  

=== "Shell"

    ```Shell
    curl --request PUT --header "X-Auth-Token: $token" https://api.gdc.cancer.gov/v0/submission/PROJECT/INTERNAL/files/6d45f2a0-8161-42e3-97e6-e058ac18f3f3 -d $path_to_file
    ```

For more details on how to upload a `submittable_data_file` to a project see the [API Users Guide](https://docs.gdc.cancer.gov/API/Users_Guide/Submission/) and the [Data Transfer Tool Users Guide](https://docs.gdc.cancer.gov/Data_Transfer_Tool/Users_Guide/Data_Download_and_Upload/).  

## Annotation Submission

The GDC Data Portal supports the use of annotations for any submitted entity or file.  An annotation entity may include comments about why particular patients or samples are not present or why they may exhibit critical differences from others.  Annotations include information that cannot be submitted to the GDC through other existing nodes or properties.

If a submitter would like to create an annotation, please contact the GDC Support Team (support@nci-gdc.datacommons.io).

## Deleting Submitted Entities

The GDC Data Submission Portal allows users to delete submitted entities from the project when the project is in an "OPEN" state. Files cannot be deleted while in the "SUBMITTED" state.  This section applies to entities that have been committed to the project. Entities that have not been committed can be removed from the project by choosing the `DISCARD` button.  Entities can also be deleted using the API. See the [API Submission Documentation](https://docs.gdc.cancer.gov/API/Users_Guide/Submission/#deleting-entities) for specific instructions.

>__NOTE:__  Entities associated with files uploaded to the GDC object store cannot be deleted until the associated file has been deleted. Users must utilize the [GDC Data Transfer Tool](https://docs.gdc.cancer.gov/Data_Transfer_Tool/Users_Guide/Data_Download_and_Upload/#deleting-previously-uploaded-data) to delete these files first.

### Simple Deletion

If an entity was uploaded and has no related entities, it can be deleted from the [Browse](Data_Submission_Process.md#browse) tab. Once the entity to be deleted is selected, choose the `DELETE` button in the right panel under "ACTIONS".


[![GDC Delete Unassociated Case](images/GDC-Delete-Case-Unassociated.png)](images/GDC-Delete-Case-Unassociated.png "Click to see the full image.")


A message will then appear asking if you are sure about deleting the entity.  Choosing the `YES, DELETE` button will remove the entity from the project, whereas choosing the `NO, CANCEL` button will return the user to the previous screen.  


[![GDC Yes or No](images/GDC-Delete-Sure.png)](images/GDC-Delete-Sure.png "Click to see the full image.")


### Deletion with Dependents

If an entity has related entities, such as a `case` with multiple `samples` and `aliquots`, deletion takes one extra step.  


[![GDC Delete Associated Case](images/GDC-Delete-Case-Associated.png)](images/GDC-Delete-Case-Associated.png "Click to see the full image.")


Follow the [Simple Deletion](Data_Submission_Walkthrough.md#simple-deletion) method until the end. This action will appear in the [Transactions](Data_Submission_Process.md#transactions) tab as "Delete" with a "FAILED" state.  


[![GDC Delete Failed](images/GDC-Failed-Transaction.png)](images/GDC-Failed-Transaction.png "Click to see the full image.")


Choose the failed transaction and the right panel will show the list of entities related to the entity that was going to be deleted.  


[![GDC Error Related](images/GDC-Error-Related.png)](images/GDC-Error-Related.png "Click to see the full image.")


Selecting the `DELETE ALL` button at the bottom of the list will delete all of the related entities, their descendants, and the original entity.


### Submitted Data File Deletion

The [`submittable_data_files`](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-entity-list&anchor=submittable_data_file) that were uploaded erroneously are deleted separately from their associated entity using the GDC Data Transfer Tool. See the section on [Deleting Data Files](https://docs.gdc.cancer.gov/Data_Transfer_Tool/Users_Guide/Data_Download_and_Upload/#deleting-previously-uploaded-data) in the Data Transfer Tool users guide for specific instructions.  

## Updating Uploaded Entities

Before harmonization occurs, entities can be modified to update, add, or delete information. These methods are outlined below.

### Updating or Adding Fields

Updated or additional fields can be applied to entities by re-uploading them through the GDC Data Submission portal or API. See below for an example of a case upload with a `primary_site` field being added and a `disease_type` field being updated.

=== "Before"

    ```json
    {
    "type":"case",
    "submitter_id":"GDC-INTERNAL-000043",
    "projects":{
      "code":"INTERNAL"
    },
    "disease_type": "Myomatous Neoplasms"
    }
    ```

=== "After"
    
    ```json
    {
    "type":"case",
    "submitter_id":"GDC-INTERNAL-000043",
    "projects":{
      "code":"INTERNAL"
    },
    "disease_type": "Myxomatous Neoplasms",
    "primary_site": "Pancreas"
    }
    ```

__Guidelines:__

* The newly uploaded entity must contain the `submitter_id` of the existing entity so that the system updates the correct one.
* All newly updated entities will be validated by the GDC Dictionary.  All required fields must be present in the newly updated entity.
* Fields that are not required do not need to be re-uploaded and will remain unchanged in the entity unless they are updated.

### Deleting Optional Fields

It may be necessary to delete fields from uploaded entities. This can be performed through the API and can only be applied to optional fields. It also requires the UUID of the entity, which can be retrieved from the submission portal or using a GraphQL query.

In the example below, the `primary_site` and `disease_type` fields are removed from a `case` entity:

=== "Shell"

    ```Shell
    curl --header "X-Auth-Token: $token_string" --request DELETE  --header "Content-Type: application/json" "https://api.gdc.cancer.gov/v0/submission/EXAMPLE/PROJECT/entities/7aab7578-34ff-5651-89bb-57aefdc4c4f8?fields=primary_site,disease_type"
    ```

=== "Before"

    ```json
    {
    "type":"case",
    "submitter_id":"GDC-INTERNAL-000043",
    "projects":{
      "code":"INTERNAL"
    },
    "disease_type": "Germ Cell Neoplasms",
    "primary_site": "Pancreas"
    }
    ```

=== "After"

    ```json
    {
    "type":"case",
    "submitter_id":"GDC-INTERNAL-000043",
    "projects":{
      "code":"INTERNAL"
    }
    }
    ```

### Versioning
Changes to entities will create versions. For more information on this, please go to [Uploading New Versions of Data Files](https://docs.gdc.cancer.gov/API/Users_Guide/Submission/#uploading-new-versions-of-data-files).

## Strategies for Submitting in Bulk

Each submission in the previous sections was broken down by component to demonstrate the GDC Data Model structure. However, the submission of multiple entities at once is supported and encouraged. Here two strategies for submitting data in an efficient manner are discussed.   

### Registering a BAM File: One Step

Registering a BAM file (or any other type) can be performed in one step by including all of the entities, from `case` to `submitted_aligned_reads`, in one file.  See the example below:

=== "JSON"

    ```
    [{
        "type": "case",
        "submitter_id": "PROJECT-INTERNAL-000055",
        "projects": {
            "code": "INTERNAL"
        }
    },
    {
        "type": "sample",
        "cases": {
            "submitter_id": "PROJECT-INTERNAL-000055"
        },
        "submitter_id": "Blood-00001SAMPLE_55",
        "tissue_type": "Normal",
        "tumor_descriptor": "Not Applicable",
        "specimen_type": "Peripheral Blood NOS",
        "preservation_method": "Frozen"
    },
    {
        "type": "portion",
        "submitter_id": "Blood-portion-000055",
        "samples": {
            "submitter_id": "Blood-00001SAMPLE_55"
        }
    },
    {
        "type": "analyte",
        "portions": {
            "submitter_id": "Blood-portion-000055"
        },
        "analyte_type": "DNA",
        "submitter_id": "Blood-analyte-000055"
    },
    {
        "type": "aliquot",
        "submitter_id": "Blood-00021-aliquot55",
        "analytes": {
            "submitter_id": "Blood-analyte-000055"
        }
    },
    {
        "type": "read_group",
        "submitter_id": "Blood-00001-aliquot_lane1_barcodeACGTAC_55",
        "experiment_name": "Resequencing",
        "is_paired_end": true,
        "library_name": "Solexa-34688",
        "library_selection":"Hybrid Selection",
        "library_strategy": "WXS",
        "platform": "Illumina",
        "read_group_name": "205DD.3-2",
        "read_length": 75,
        "sequencing_center": "BI",
        "aliquots":
            {
                "submitter_id": "Blood-00021-aliquot55"
            }    
    },
    {
        "type": "submitted_aligned_reads",
        "submitter_id": "Blood-00001-aliquot_lane1_barcodeACGTAC_55.bam",
        "data_category": "Raw Sequencing Data",
        "data_format": "BAM",
        "data_type": "Aligned Reads",
        "experimental_strategy": "WGS",
        "file_name": "test.bam",
        "file_size": 38,
        "md5sum": "aa6e82d11ccd8452f813a15a6d84faf1",
        "read_groups": [
            {
                "submitter_id": "Blood-00001-aliquot_lane1_barcodeACGTAC_55"
            }
        ]
    }]
    ```

All of the entities are placed into a JSON list object:

`[{"type": "case","submitter_id": "PROJECT-INTERNAL-000055","projects": {"code": "INTERNAL"}}}, entity-2, entity-3]`

The entities need not be in any particular order as they are validated together.

>__Note:__ Tab-delimited format is not recommended for 'one-step' submissions due to an inability of the format to accommodate multiple 'types' in one row.  

### Submitting Numerous Cases

The GDC understands that submitters will have projects that comprise more entities than would be reasonable to individually parse into JSON formatted files. Additionally, many investigators store large amounts of data in a tab-delimited format (TSV).  For instances like this, we recommend parsing all entities of the same type into separate TSVs and submitting them on a type-basis.  

For example, a user may want to submit 100 Cases associated with 100 `samples`, 100 `portions`, 100 `analytes`, 100 `aliquots`, and 100 `read_groups`. Constructing and submitting 100 JSON files would be tedious and difficult to organize. The solution is submitting one `case` TSV containing the 100 `cases`, one `sample` TSV containing the 100 `samples`, so on and so forth. Doing this would only require six TSVs and these files can be formatted in programs such as Microsoft Excel or Google Spreadsheets.  

See the following example TSV files:

* [Cases.tsv](Cases.tsv)
* [Samples.tsv](Samples.tsv)
* [Portions.tsv](Portions.tsv)
* [Analytes.tsv](Analytes.tsv)
* [Aliquots.tsv](Aliquots.tsv)
* [Read-Groups.tsv](Readgroups.tsv)

### Download Previously Uploaded Metadata Files

The [transaction](Data_Submission_Process.md#transactions) page lists all previous transactions in the project. The user can download metadata files uploaded to the GDC workspace in the details section of the screen by selecting one transaction and scrolling to the "DOCUMENTS" section.


[![Transaction Original Files](images/GDC_Submission_Transactions_Original_Files_2.png)](images/GDC_Submission_Transactions_Original_Files_2.png "Click to see the full image.")

### Download Previously Uploaded Data Files

The only supported method to download data files previously uploaded to the GDC Submission Portal that have not been release yet is to use the API or the [Data Transfer Tool](https://docs.gdc.cancer.gov/Data_Transfer_Tool/Users_Guide/Getting_Started/). To retrieve data previous upload to the submission portal you will need to retrieve the data file's UUID.  The UUIDs for submitted data files are located in the submission portal under the file's Summary section as well as the manifest file located on the file's Summary page.

[![Submission Portal Summary View](images/gdc-submission__image2_submission_UUID.png)](images/gdc-submission__image2_submission_UUID.png "Click to see the full image.")  

Once the UUID(s) have been retrieved, the download process is the same as it is for downloading data files at the [GDC Portal using UUIDs](https://docs.gdc.cancer.gov/Data_Transfer_Tool/Users_Guide/Data_Download_and_Upload/#downloading-data-using-gdc-file-uuids).

 >__Note:__ When submittable data files are uploaded through the Data Transfer Tool they are not displayed as transactions.


# Release
Project release occurs after the data has been harmonized and allows users to access this data with the GDC Data Portal. The GDC will release data according to [GDC Data Sharing Policies](https://gdc.cancer.gov/submit-data/data-submission-policies). Data may be released after six months from the date of upload, or the submitter may request earlier release using the "Request Release" function.  A project can only be released once. If additional data is added to the project after it is released, the data will be released automatically after harmonization.  

Upon release, harmonized data will be available to GDC users through the [GDC Data Portal](https://portal.gdc.cancer.gov/) and other [GDC Data Access Tools](https://gdc.cancer.gov/access-data/data-access-processes-and-tools).

**Note:** To release data to the GDC Data Portal, the user must have release privileges.

[![GDC Submission Release Tab](images/GDC_Submission_Landing_Submitter_4.png)](images/GDC_Submission_Landing_Submitter_4.png "Click to see the full image.")

When the user clicks on the action _'Request Release'_, the following Release popup is displayed:

[![GDC Submission Release Popup](images/GDC_Submission_Submit_Release_Release_Popup.png)](images/GDC_Submission_Submit_Release_Release_Popup.png "Click to see the full image.")

After the user clicks on "Release Submitted and Processed Data", the project release state becomes "Release Requested":

[![GDC Submission Project State](images/GDC_Submission_Submit_Release_Project_State_3.png)](images/GDC_Submission_Submit_Release_Project_State_3.png "Click to see the full image.")


__Note__: Released cases and/or files can be redacted from the GDC. Redaction is performed by GDC administrators at the case level through synchronization with dbGaP and at the file level at the submitter's request usually after a data quality issue is identified. The GDC Data Submission Portal itself currently does not support redaction through the web interface.


# Reports

## Overview

The GDC Data Submission Portal provides access to the different reports listed below. Access to the reports is granted to users based on their project permissions.

The reports are available from the [Homepage](Homepage.md) and the [Project Dashboard](Dashboard.md).

[![GDC Submission Reports](images/GDC_Submission_Reports.png)](images/GDC_Submission_Reports.png "Click to see the full image.")

## Case Overview Report

This report provides an overview of all cases associated with one or more projects and is generated daily.

The report is downloadable in TSV format.

```TSV
Program	Project	dbGaP Study ID	# Cases Released	# Cases Unreleased	# Cases Expected	Date Expected for Release	Initial Upload Date	Project Last Update Date	# Cases Registered in dbGaP	# Cases with Clinical	# Cases with Sample	# Cases with Aliquot	# Cases with Portion	# Cases with Analyte	# Cases with WGS Data	# Cases with WXS Data	# Cases with RNA-Seq Data	# Cases with ChIP-Seq Data	# Cases with MiRNA-Seq Data	# Cases with Bisulfite-Seq Data	# Cases with Validation Data	# Cases with Amplicon Data	# Cases with Other Data
TCGA	TEST	TEST	256	0	Not Available	Not Available	2015-11-10 10:40:15.689597-06:00	2015-11-10 10:40:15.689597-06:00	256	0	11	10	10	10	0	4	0	0	0	0	0	0	0
TCGA	DEV1	phs000178	1	15	Not Available	Not Available	2015-11-18 09:27:36.990449-06:00	2015-11-18 09:27:36.990449-06:00	16	0	16	16	16	16	0	0	0	0	0	0	0	0	0
TCGA	DEV2	phs000178	1	7	Not Available	Not Available	2015-11-17 20:14:02.194718-06:00	2015-11-17 20:14:02.194718-06:00	8	0	8	8	0	0	0	0	0	0	0	0	0	0	0
TCGA	DEV3	phs000178	64	0	Not Available	Not Available	2015-12-11 15:41:29.760763-06:00	2015-12-11 15:41:29.760763-06:00	64	0	25	25	0	0	0	2	0	0	0	0	0	0	0
```

Previous versions of the report are available for comparison. To download a previous version, the user should click on the "Previous Snapshots" button. This will open a calendar to allow the user to select the date of the report for download.

## Data Validation Report

The Data Validation Report provides a live view of quality metrics coming from data validation performed by the GDC when validating a file upload through the GDC Data Transfer Tool.

The report can be downloaded in TSV format.

```TSV
program	project	dbgap_accession_number	file_format_fail	file_size_fail	md5sum_fail	processed	processing	registered	submitted	uploaded	uploading	validated	validating
TCGA	DEV3		0	1	0	0	0	3	0	0	0	1	0
```


# Dashboard

## Overview

The GDC Data Submission Portal dashboard provides details about a specific project.

[![GDC Submission Dashboard Page-2](images/GDC_Submission_Dashboard_3.png)](images/GDC_Submission_Dashboard_3.png "Click to see the full image.")

The dashboard contains various visual elements to guide the user through all stages of submission, from viewing the [Data Dictionary](https://docs.gdc.cancer.gov/Data_Dictionary/) in support of data upload to submitting a project for harmonization.

To better understand the information displayed on the dashboard and the available actions, please refer to the [Submission Workflow](Submission_Workflow.md).

## Project Overview
The Project Overview sections of the dashboard displays the project state (open / review / submitted / processing) and the GDC Release, which is the date in which the project was released to the GDC.

The search field at the top of the dashboard allows for submitted entities to be searched by partial or whole `submitter_id`.  When a search term is entered into the field, a list of entities matching the term is updated in real time.  Selecting one of these entities links to its details in the [Browse Tab](Browse_Data.md)

The remaining part of the top section of the dashboard is broken down into five status charts:

* __QC Errors__: The number of QC errors identified within the data that has been uploaded. Refer to [QC Reports](/Data_Submission_Portal/Users_Guide/Data_Submission_Process/#qc-reports) for more information.
* __Cases with Clinical__: The number of `cases` for which Clinical data has been uploaded.
* __Cases with Biospecimen__: The number of `cases` for which Biospecimen data has been uploaded.
* __Cases with Submittable Data Files__: The number of `cases` for which experimental data has been uploaded.
* __Submittable Data Files__: The number of files uploaded through the GDC Data Transfer Tool. For more information on this status chart, please refer to [File Status Lifecycle](Submission_Workflow/#file-status-lifecycle).
The _'DOWNLOAD MANIFEST'_ button below this status chart allows the user to download a manifest for registered files in this project that have not yet been uploaded.

\* Note that the QC Errors will not be immediately updated after submission of a new file.

Status charts are constantly updated to reflect the current state of the selected project.

## Action Panels

There are two action panels available below the Project Overview.

* [UPLOAD DATA TO YOUR WORKSPACE](Data_Upload_UG.md): Allows a submitter to upload project data to the GDC project workspace. The GDC will validate the uploaded data against the [GDC Data Dictionary](https://docs.gdc.cancer.gov/Data_Dictionary/). This panel also contains a table that displays details about the five latest transactions. Clicking the IDs in the first column will bring up a window with details about the transaction, which are documented in the [transactions](Transactions.md) page. This panel will also allow the user to commit file submissions to the project.
* [REVIEW AND SUBMIT YOUR WORKSPACE DATA TO THE GDC](Submit_Data.md#review-and-submit): Allows a submitter to review project data which will lock the project to ensure that additional data cannot be uploaded while in review. Once the review is complete, the data can be submitted to the GDC for processing through the [GDC Harmonization Process](https://gdc.cancer.gov/submit-data/gdc-data-harmonization).

These actions and associated features are further detailed in their respective sections of the documentation.


# Pre-Release Data Portal

The [GDC Pre-Release Data Portal](https://portal.awg.gdc.cancer.gov/) provides users with web-based access to pre-released data from cancer genomics studies that have been harmonized by the GDC, but not yet released in the main GDC Data Portal.

## Navigation

[Pre-Release Data Portal](https://portal.awg.gdc.cancer.gov/) will appear similar to the GDC Active Portal, but the Pre-Release Data Portal features are a subset of what can be found in the GDC Data Portal.

[![GDC Views](images/AWG_Portal.png)](images/AWG_Portal.png "Click to see the full image.")

For more information on any of these general features please review the [GDC Data Portal User Guide](/Data_Portal/Users_Guide/getting_started/).

## Authentication

### Relationship between GDC Data Portal and Pre-Release Data Portal Tokens

The GDC Pre-Release Data Portal provides access to datasets prior to release to a group of users specified by the data submitter.  This area is only available to data submitters (or their designees) for reviewing pre-release data.  Users must be granted access as specified in the GDC Pre-Release Data Admin Portal section and have downloader access within dbGaP for the specified project. To learn more about obtaining the required credentials and authorization, see [Obtaining Access to Submit Data]( https://gdc.cancer.gov/submit-data/obtaining-access-submit-data).

The tokens used to download files from the GDC Data Portal and Pre-Release Data Portal are related but distinct. Specifically, the token generated in the Pre-Release Data Portal contains a longer version of the regular GDC Authentication Token downloaded from the GDC Data Portal. Because of this, the GDC Data Portal token will not function for downloading data from the Pre-release Data Portal environment using the Data Transfer Tool or API. However, the Pre-Release Data Portal token will function for downloading data from the GDC Data Portal using the API or Data Transfer Tool. Finally, if a new token is generated in the Pre-release Data Portal this will invalidate the token downloaded from the GDC Data Portal and vice versa.  

## Data Transfer Tool

As with the GDC Data Portal, downloads of large or numerous files is best performed using the GDC Data Transfer Tool.  Information on the GDC Data Transfer Tool is available in the [GDC Data Transfer Tool User's Guide](https://docs.gdc.cancer.gov/Data_Transfer_Tool/Users_Guide/Getting_Started/).  An important distinction for use with the Pre-Release Data Portal is that it must always be used with a token and with the option `-s https://api.awg.gdc.cancer.gov`.

## GDC Pre-Release Data Admin Portal

The GDC Pre-Release Data Admin Portal allows admins to create and maintain Pre-Release Data Groups and associated projects, as well as grant appropriate access to users within these groups. To gain access to the Pre-Release Data Admin Portal please contact the GDC Helpdesk (support@nci-gdc.datacommons.io).

[![GDC Pre-Release Data Portal Main Page](images/AWG_Admin.png)](images/AWG_Admin.png "Click to see the full image.")

The Pre-Release Data Admin Portal is broken into two views on the left-most panel:

* __Users__: Allows admin to create, view, edit Pre-Release Data Portal user profiles.
* __Groups__: Allows admin to manage groups projects / users.

#### Definitions

| Entity | Definition |
|---|---|
| __User__  | An individual with an eRA Commons account. |
| __Project__  | A collection of files and observations that are contained in the GDC database and have been registered in dbGaP as a project. Only certain projects are designated as Pre-Release Data projects.|
| __Group__  | A collection of users and projects.  When a user is assigned to a group, they will have access to the projects in that group when they login to the Pre-Release Data portal as long as they have downloader access to the project in dbGaP.|

### Users

The __Users__ section of the GDC Pre-Release Data Admin portal allows admins to manage and create Pre-Release Data users.

[![GDC Pre-Release Data Portal Main Page](images/AWG_Admin.png)](images/AWG_Admin.png "Click to see the full image.")

#### Creating Users

To create a new user in the Pre-Release Data Admin Portal, click on the `Create` button on the far-right panel.

[![GDC Pre-Release Data Portal Main Page](images/AWG_Admin_Create_User.png)](images/AWG_Admin_Create_User.png "Click to see the full image.")

Then the following information must be supplied, before clicking the `Save` button:

* __eRA Commons ID__: The eRA Commons ID of the user to be added.
* __Role__: Choose between `Admin` or `User` roles.
* __Group (Optional)__: Choose existing groups to add the user to.

After clicking `Save`, the user should appear in the list of users in the center panel.  Also clicking on the user in the list will display information about that user and gives the options to `Edit` the user profile, or `Delete` the user.

[![GDC Pre-Release Data Portal Main Page](images/AWG_Admin_New_User.png)](images/AWG_Admin_New_User.png "Click to see the full image.")

### Groups

The __Groups__ section of the GDC Pre-Release Data Admin portal allows admins to manage and create groups for which users and projects may be added.

[![GDC Pre-Release Data Portal Main Page](images/AWG_Admin_Group.png)](images/AWG_Admin_Group.png "Click to see the full image.")

#### Creating Groups

To create a new group in the Pre-Release Data Admin Portal, click on the `Create` button on the far-right panel.

[![GDC Pre-Release Data Portal Main Page](images/AWG_Admin_Groups_Add.png)](images/AWG_Admin_Groups_Add.png "Click to see the full image.")

Then the following information must be supplied, before clicking the `Save` button:

* __Name__: The name of the group.
* __Description__: The description of the group.
* __Users (Optional)__: Choose existing users to add to the group.
* __Projects(Optional)__: Choose existing projects to add to the group.

After clicking `Save`, the group should appear in the list of groups in the center panel.  Also clicking on the group in the list will display information about that group and gives the options to `Edit` or `Delete` the group.

[![GDC Pre-Release Data Portal Main Page](images/AWG_Admin_New_Group.png)](images/AWG_Admin_New_Group.png "Click to see the full image.")

## AWG API

API functionality is similar to what is available for the main GDC Data Portal.  You can read more about the GDC API in general in the [API User Guide](/API/Users_Guide/Getting_Started/).  Important differences for the Analysis Working Group (AWG) API include the following:

*  The base URL is different. Instead use https://api.awg.gdc.cancer.gov/
*  An authorization token must always be passed with every query rather than just for downloading controlled access data.


# Data Submission Troubleshooting

This guide is intended to assist in problem-solving when encountering data upload errors. Please contact the GDC Help Desk at __<support@nci-gdc.datacommons.io>__ if you have any questions or concerns regarding a submission project.

## Transactions Page

After attempting data upload, the Data Submission Portal's [transaction page](https://docs.gdc.cancer.gov/Data_Submission_Portal/Users_Guide/Data_Submission_Process/#transactions) will indicate the state of each transaction as `SUCCEEDED`, `PENDING`, or `FAILED`. 

[![GDC Submission Transaction Page](images/DSP_Transaction_Page.png)](images/DSP_Transaction_Page.png "Click to see the full image.")

### Transaction Details

Clicking on a failed upload will open the [details panel](https://docs.gdc.cancer.gov/Data_Submission_Portal/Users_Guide/Data_Submission_Process/#transactions-details), which includes a Documents section containing an Error Report. 

[![GDC Submission Transaction Details](images/DSP_Transaction_Details.png)](images/DSP_Transaction_Details.png "Click to see the full image.")

## Error Report

The Error Report is generated in tab-delimited format and describes the reason(s) that the transaction was not successful.

### Error Report Table

The Error Report table displays the following information:

|Column|Description|
| --- | --- |
| __entityType__ | Entity type that caused the error |
| __key__ | Property that caused the error|
| __errorType__ | The type of error |
| __message__ | Detailed description of the error |
| __id__ | UUID of the errored entity |

[![GDC Submission Transaction Details](images/DSP_Error_Report.png)](images/DSP_Error_Report.png "Click to see the full image.")

### Data Upload Error Messages

Each error type can have numerous error messages which are detailed in the following sections. If data upload is attempted through the API, the command line output will indicate whether the transaction was successful or failed, and, if applicable, the same error message(s) that would be included in the Data Submission Error Report.

#### ERROR Messages

|Message|Explanation|Solution|
| --- | --- | --- |
| __'{Value}' is not one of [{acceptable values}]__ | The value is not accepted by the GDC for the designated property | Ensure the property value is acceptable by reviewing the [GDC Data Dictionary](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/) |
| __'{Entity}' with {'project_id': '{project_id}', 'submitter_id': '{entity submitter_id}'} already exists in the GDC__ | An entity with that submitter_id has already been uploaded to the designated project | Ensure the submitter_id is unique to the project |
| __Additional properties are not allowed ('{property}' or '{list of properties}') was/were unexpected__ | The given property or properties are not accepted for the designated entity | Ensure entity accepts the properties by reviewing the [GDC Data Dictionary](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/) |
| __{value} is less than the minimum of -32872__ | The amount is less than the minimum accepted value | Ensure the value is greater than or equal to the minimum value and view the best practices section on [Date Obfuscation](https://docs.gdc.cancer.gov/Data_Submission_Portal/Users_Guide/Best_Practices/#date-obfuscation) if applicable |

Example 1:
=== "Request1"

    ```json
    { 
      "submitter_id": "Diagnosis_000093",
      "tissue_or_organ_of_origin": "Abdomen, NOS",
      "primary_diagnosis": "Adenoma, NOS", 
      "morphology": "8000/9", 
      "type": "diagnosis", 
      "cases": {
        "submitter_id": "GDC-INTERNAL-000093"
      }, 
      "age_at_diagnosis": 35, 
      "diagnosis_is_primary_disease": true, 
      "site_of_resection_or_biopsy": "Abdomen, NOS",
      "gleason_grade_group": "Group 6"
    }
    ```

=== "Response1"

    ```json
    {
      "cases_related_to_created_entities_count": 1,
      "cases_related_to_updated_entities_count": 0,
      "code": 400,
      "created_entity_count": 0,
      "entities":
        [
            {
                "action":"create",
                "errors":
                    [
                        {
                            "keys":["gleason_grade_group"],
                            "message":"'Group 6' is not one of ['Group 1', 'Group 2', 'Group 3', 'Group 4', 'Group 5', 'Unknown', 'Not Reported']",
                            "type":"ERROR"
                        }
                    ],
                "id":"61f17908-73fe-4961-b760-516d588105e6",
                "related_cases":[],
                "type":"diagnosis",
                "unique_keys":
                    [
                        {
                            "project_id":"GDC-INTERNAL",
                            "submitter_id":"Diagnosis_000093"
                        }
                    ],
                "valid":false,
                "warnings":[]
            }
        ],
        "entity_error_count":1,
        "message":"Transaction aborted due to 1 invalid entity.",
        "success":false,
        "transaction_id":6372895,
        "transactional_error_count":0,
        "transactional_errors":[],
        "updated_entity_count":0
    }
    ```
=== "Request2"

    ```json
    { 
      "submitter_id": "Diagnosis_000093",
      "tissue_or_organ_of_origin": "Abdomen, NOS",
      "primary_diagnosis": "Adenoma, NOS", 
      "morphology": "8000/9", 
      "type": "diagnosis", 
      "cases": {
        "submitter_id": "GDC-INTERNAL-000093"
      }, 
      "age_at_diagnosis": 35, 
      "diagnosis_is_primary_disease": true, 
      "site_of_resection_or_biopsy": "Abdomen, NOS",
      "gleason_grade_group": "Group 5"
    }
    ```

=== "Response2"

    ```json
    {
        "cases_related_to_created_entities_count":1,
        "cases_related_to_updated_entities_count":0,
        "code":201,
        "created_entity_count":1,
        "entities":
            [
                {
                    "action":"create",
                    "errors":[],
                    "id":"a9d70ba0-a547-4936-889d-8da9f903816f",
                    "related_cases":
                        [
                            {
                                "id":"a00f076e-d694-47dd-8e50-24c28e90fd6a",
                                "submitter_id":"GDC-INTERNAL-000093"
                            }
                        ],
                    "type":"diagnosis",
                    "unique_keys":
                        [
                            {
                                "project_id":"GDC-INTERNAL",
                                "submitter_id":"Diagnosis_000093"
                            }
                        ],
                    "valid":true,
                    "warnings":[]
                }
            ],
        "entity_error_count":0,
        "message":"Transaction successful.",
        "success":true,
        "transaction_id":6372896,
        "transactional_error_count":0,
        "transactional_errors":[],
        "updated_entity_count":0
    }
    ```

Example 2:
=== "Request1"

    ```json
    { 
      "submitter_id": "demographic_test",
      "ethnicity": "not reported",
      "gender": "female", 
      "race": "black or african american", 
      "type": "demographic", 
      "cases": {
        "submitter_id": "GDC-INTERNAL-000093"
      }, 
      "vital_status": "Alive", 
      "days_to_birth": -42875
    }
    ```

=== "Response1"

    ```json
    {
      "cases_related_to_created_entities_count": 0,
      "cases_related_to_updated_entities_count": 1,
      "code": 400,
      "created_entity_count": 0,
      "entities":
        [
            {
                "action":"update",
                "errors":
                    [
                        {
                            "keys":["days_to_birth"],
                            "message":"-42875 is less than the minimum of -32872",
                            "type":"ERROR"
                        }
                    ],
                "id":"6bef9467-28d6-43ee-a294-cfbab2808d1b",
                "related_cases":[
                    {
                        "id":"a00f076e-d694-47dd-8e50-24c28e90fd6a","submitter_id":"GDC-INTERNAL-000093"
                    }
                ],
                "type":"demographic",
                "unique_keys":
                    [
                        {
                            "project_id":"GDC-INTERNAL",
                            "submitter_id":"demographic_test"
                        }
                    ],
                "valid":false,
                "warnings":[]
            }
        ],
        "entity_error_count":1,
        "message":"Transaction aborted due to 1 invalid entity.",
        "success":false,
        "transaction_id":6372898,
        "transactional_error_count":0,
        "transactional_errors":[],
        "updated_entity_count":0
    }
    ```

=== "Request2"

    ```json
    { 
      "submitter_id": "demographic_test",
      "ethnicity": "not reported",
      "gender": "female", 
      "race": "black or african american", 
      "type": "demographic", 
      "cases": {
        "submitter_id": "GDC-INTERNAL-000093"
      }, 
      "vital_status": "Alive", 
      "days_to_birth": -22875
    }
    ```

=== "Response2"

    ```json
    {
        "cases_related_to_created_entities_count":0,
        "cases_related_to_updated_entities_count":1,
        "code":200,
        "created_entity_count":0,
        "entities":
            [
                {
                    "action":"update",
                    "errors":[],
                    "id":"6bef9467-28d6-43ee-a294-cfbab2808d1b",
                    "related_cases":
                        [
                            {
                                "id":"a00f076e-d694-47dd-8e50-24c28e90fd6a",
                                "submitter_id":"GDC-INTERNAL-000093"
                            }
                        ],
                    "type":"demographic",
                    "unique_keys":
                        [
                            {
                                "project_id":"GDC-INTERNAL",
                                "submitter_id":"demographic_test"
                            }
                        ],
                    "valid":true,
                    "warnings":[]
                }
            ],
        "entity_error_count":0,
        "message":"Transaction successful.",
        "success":true,
        "transaction_id":6372899,
        "transactional_error_count":0,
        "transactional_errors":[],
        "updated_entity_count":1
    }
    ```

#### INVALID_LINK Messages

|Message|Explanation|Solution|
| --- | --- | --- |
| __'{Parent entity}' link has to be one_to_one, target node {parent entity} already has {child entity}__ | The parent entity can only have one child entity, and the child entity can only have one parent entity | Ensure the parent entity is only linked to one child entity and vice versa |
| __'{Parent entity}' link has to be one_to_many, target node {parent entity} already has a {child entity}__ | The parent entity cannot have multiple child entities | Ensure each parent entity is only linked to one child entity |
| __'{Parent entity}' link has to be many_to_one__ | The child entity cannot have multiple parent entities | Ensure the child entity links to only one parent entity |
| __More than one link destination found for {parent entity}__ | A new entity's submitter_id is already in use, such that multiple entities with the same submitter_id would exist had the upload not failed | Make sure the submitter_ids for each new entity are unique |
| __Entity is missing required link to {parent entity}__ | The child entity is not linked to a parent node | Link the child entity to at least one parent entity, based on their relationship (many_to_one, one_to_many, or one_to_one) |
| __No link destination found for {}__ | The parent entity's submitter_id does not exist | Make sure the child entity is linking to the correct parent entity and that the submitter_id is accurate |

Example 1:
=== "Request1"
    
    ```json
    {
      "read_groups": {
        "submitter_id": "Read_group_00093"
      }, 
      "data_type": "Aligned Reads", 
      "md5sum": "aa6e82d11ccd8452f813a15a6d84faf1", 
      "type": "submitted_aligned_reads", 
      "data_category": "Sequencing Reads", 
      "data_format": "BAM",
      "project_id": "GDC-INTERNAL", 
      "file_size": 928374, 
      "file_name": "test.bam", 
      "experimental_strategy": "WGS", 
      "submitter_id": "Blood-00009-aliquot01_lane1_barcodeACGTAC_55.bam"
    }
    ```

=== "Response1"

    ```json
    {
      "cases_related_to_created_entities_count": 0,
      "cases_related_to_updated_entities_count": 1,
      "code": 400,
      "created_entity_count": 0,
      "entities":
        [
            {
                "action":"update",
                "errors":
                    [
                        {
                            "keys":["read_groups"],
                            "message":"'read_groups' link has to be one_to_many, target node read_group already has submitted_aligned_reads_files",
                            "type":"INVALID_LINK"
                        }
                    ],
                "id":
                    [
                        {
                            "id":"a00f076e-d694-47dd-8e50-24c28e90fd6a","submitter_id":"GDC-INTERNAL-000093"
                        },
                        {
                            "id":"8c7e2ce7-a772-441c-a33b-e0e3bdbd4f78","submitter_id":"GDC-INTERNAL-000078"
                        }
                    ],
                "related_cases":[],
                "type":"submitted_aligned_reads",
                "unique_keys":
                    [
                        {
                            "project_id":"GDC-INTERNAL",
                            "submitter_id":"Blood-00009-aliquot01_lane1_barcodeACGTAC_55.bam"
                        }
                    ],
                "valid":false,
                "warnings":[]
            }
        ],
        "entity_error_count":1,
        "message":"Transaction aborted due to 1 invalid entity.",
        "success":false,
        "transaction_id":6372988,
        "transactional_error_count":0,
        "transactional_errors":[],
        "updated_entity_count":0
    }
    ```

Example 2:
=== "Request1"

    ```json
    [
        {
            "data_category": "Sequencing Reads",
            "data_format": "BAI",
            "data_type": "Aligned Reads Index",
            "project_id": "CPTAC-3",
            "submitter_id": "Aligned_Reads_Index_000000",
            "file_name": "1c5321a4-3b33-4787-9393-3c999940284f.rna_seq.genomic.gdc_realn.bam.bai",
            "file_size": 5106896,
            "md5sum": "250a27e69c3556311934d0d8daab2207",
            "aligned_reads_files": {
                "submitter_id": "Aligned_Reads_000000"
            },
            "type": "aligned_reads_index"
        }
    ]
    ```

=== "Response1"

    ```json
    {
      "cases_related_to_created_entities_count": 0,
      "cases_related_to_updated_entities_count": 0,
      "code": 400,
      "created_entity_count": 0,
      "entities":
        [
            {
                "action":"create",
                "errors":
                    [
                        {
                            "keys":["aligned_reads_files"],
                            "message":"No link destination found for aligned_reads_files, unique_keys='[{'project_id': 'GDC-INTERNAL', 'submitter_id': 'Aligned_Reads_000000'}]'",
                            "type":"INVALID_LINK"
                        }
                    ],
                "id":
                    [
                        {
                            "id":"001812ff-15e0-4c29-a3bb-582f81b46b2c"
                        }
                    ],
                "related_cases":[],
                "type":"aligned_reads_index",
                "unique_keys":
                    [
                        {
                            "project_id":"CPTAC-3",
                            "submitter_id":"Aligned_Reads_Index_000000"
                        }
                    ],
                "valid":false,
                "warnings":[]
            }
        ],
        "entity_error_count":1,
        "message":"Transaction aborted due to 1 invalid entity.",
        "success":false,
        "transaction_id":6372995,
        "transactional_error_count":0,
        "transactional_errors":[],
        "updated_entity_count":0
    }
    ```

#### INVALID_PROPERTY Messages

|Message|Explanation|Solution|
| --- | --- | --- |
| __Key '{property}' is not a valid property for type '{entity}'__ | The designated entity does not accept that property | Ensure the property is accepted for the entity by reviewing the [GDC Data Dictionary](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/) |

Example 1:
=== "Request1"

    ```json
    {
      "projects": {
        "code": "INTERNAL"
      }, 
      "disease_type": "Adenomas and Adenocarcinomas", 
      "primary_site": "Bladder",
      "type": "case",
      "project_id": "GDC-INTERNAL", 
      "submitter_id": "GDC-INTERNAL-000093",
      "consent_typ": "Informed Consent"
    }
    ```

=== "Response1"

    ```json
    {
      "cases_related_to_created_entities_count": 0,
      "cases_related_to_updated_entities_count": 0,
      "code": 400,
      "created_entity_count": 0,
      "entities":
        [
            {
                "action":"update",
                "errors":
                    [
                        {
                            "keys":["consent_typ"],
                            "message":"Key 'consent_typ' is not a valid property for type 'case'. Did you mean 'consent_type'?",
                            "type":"INVALID_PROPERTY"
                        },
                        {
                            "keys":[],
                            "message":"Additional properties are not allowed ('consent_typ' was unexpected)",
                            "type":"ERROR"
                        }
                    ],
                "id":
                    [
                        {
                            "id":"a00f076e-d694-47dd-8e50-24c28e90fd6a"
                        }
                    ],
                "related_cases":[],
                "type":"case",
                "unique_keys":
                    [
                        {
                            "project_id":"GDC-INTERNAL",
                            "submitter_id":"GDC-INTERNAL-000093"
                        }
                    ],
                "valid":false,
                "warnings":[]
            }
        ],
        "entity_error_count":1,
        "message":"Transaction aborted due to 1 invalid entity.",
        "success":false,
        "transaction_id":6372996,
        "transactional_error_count":0,
        "transactional_errors":[],
        "updated_entity_count":0
    }
    ```

=== "Request2"

    ```json
    {
      "projects": {
        "code": "INTERNAL"
      }, 
      "disease_type": "Adenomas and Adenocarcinomas", 
      "primary_site": "Bladder",
      "type": "case",
      "project_id": "GDC-INTERNAL", 
      "submitter_id": "GDC-INTERNAL-000093",
      "consent_type": "Informed Consent"
    }
    ```

=== "Response2"

    ```json
    {
      "cases_related_to_created_entities_count": 0,
      "cases_related_to_updated_entities_count": 0,
      "code": 200,
      "created_entity_count": 0,
      "entities":
        [
            {
                "action":"update",
                "errors":[],
                "id":
                    [
                        {
                            "id":"a00f076e-d694-47dd-8e50-24c28e90fd6a"
                        }
                    ],
                "related_cases":[],
                "type":"case",
                "unique_keys":
                    [
                        {
                            "project_id":"GDC-INTERNAL",
                            "submitter_id":"GDC-INTERNAL-000093"
                        }
                    ],
                "valid":true,
                "warnings":[]
            }
        ],
        "entity_error_count":0,
        "message":"Transaction successful.",
        "success":true,
        "transaction_id":6372997,
        "transactional_error_count":0,
        "transactional_errors":[],
        "updated_entity_count":1
    }
    ```

#### INVALID_TYPE Messages

|Message|Explanation|Solution|
| --- | --- | --- |
| __missing 'type'__ | This is a vague error message that frequently does not encapsulate the reason that the data upload is failing, and may be indicative of multiple errors occurring simultaneously | Scrutinize the file for other issues such as a formatting problem (e.g. an extra column in the TSV, incorrect JSON formatting, the node is already in state=submitted, or a case is not registered with dbGaP) |

Example 1:
=== "Request1"
    
    ```json
    {
      "projects": {
        "code": "INTERNAL"
      }, 
      "disease_type": "Adenomas and Adenocarcinomas", 
      "primary_site": "Bladder",
      "project_id": "GDC-INTERNAL", 
      "submitter_id": "GDC-INTERNAL-000099"
    }
    ```

=== "Response1"

    ```json
    {
      "cases_related_to_created_entities_count": 0,
      "cases_related_to_updated_entities_count": 0,
      "code": 400,
      "created_entity_count": 0,
      "entities":
        [
            {
                "action":null,
                "errors":
                    [
                        {
                            "keys":["type"],
                            "message":"missing 'type'",
                            "type":"INVALID_TYPE"
                        },
                        {
                            "keys":["type"],
                            "message":"'type' is a required property",
                            "type":"MISSING_PROPERTY"
                        }
                    ]
                "id":
                    [
                        {
                            "id":null
                        }
                    ],
                "related_cases":[],
                "type":null,
                "unique_keys":[],
                "valid":false,
                "warnings":[]
            }
        ],
        "entity_error_count":1,
        "message":"Transaction aborted due to 1 invalid entity.",
        "success":false,
        "transaction_id":6372998,
        "transactional_error_count":0,
        "transactional_errors":[],
        "updated_entity_count":0
    }
    ```

=== "Request2"

    ```json
    {
      "projects": {
        "code": "INTERNAL"
      }, 
      "disease_type": "Adenomas and Adenocarcinomas", 
      "primary_site": "Bladder",
      "project_id": "GDC-INTERNAL", 
      "type": "case",
      "submitter_id": "GDC-INTERNAL-000099"
    }
    ```

=== "Response2"

    ```json
    {
      "cases_related_to_created_entities_count": 0,
      "cases_related_to_updated_entities_count": 0,
      "code": 200,
      "created_entity_count": 0,
      "entities":
        [
            {
                "action":"update",
                "errors":[],
                "id":
                    [
                        {
                            "id":"f0c68d72-ee54-4ad9-89e8-b08398b5aa0b"
                        }
                    ],
                "related_cases":[],
                "type":"case",
                "unique_keys":
                    [
                        {
                            "project_id":"GDC-INTERNAL",
                            "submitter_id":"GDC-INTERNAL-000099"
                        }
                    ],
                "valid":true,
                "warnings":[]
            }
        ],
        "entity_error_count":0,
        "message":"Transaction successful.",
        "success":true,
        "transaction_id":6372999,
        "transactional_error_count":0,
        "transactional_errors":[],
        "updated_entity_count":1
    }
    ```

Example 2:
=== "Request1"

    ```json
    {
      "read_groups": 
        {                                                                    
            "submitter_id":"Read_group_000093"                                            
        },                                                                               
      "data_type": "Unaligned Reads",                                                  
      "file_name": "test_filename.txt",                                                
      "md5sum": "b026324c6904b2a9cb4b88d6d61c81d1",                                    
      "data_format": "FASTQ",                                                          
      "submitter_id": "SUR_000093",                                           
      "data_category": "Sequencing Reads",                   
      "file_size": 1,                                                                  
      "project_id": "GDC-INTERNAL",                                                    
      "type": "submitted_unaligned_reads",                                             
      "experimental_strategy": "WXS",
      "read_pair_number": "R1"                                                 
    } 
    ```

=== "Response1"

    ```json
    {
      "cases_related_to_created_entities_count": 0,
      "cases_related_to_updated_entities_count": 0,
      "code": 400,
      "created_entity_count": 0,
      "entities":
        [
            {
                "action":null,
                "errors":
                    [
                        {
                            "keys":["type"],
                            "message":"missing 'type'",
                            "type":"INVALID_TYPE"
                        },
                        {
                            "keys":["type"],
                            "message":"'type' is a required property",
                            "type":"MISSING_PROPERTY"
                        }
                    ]
                "id":
                    [
                        {
                            "id":null
                        }
                    ],
                "related_cases":[],
                "type":null,
                "unique_keys":[],
                "valid":false,
                "warnings":[]
            }
        ],
        "entity_error_count":1,
        "message":"Transaction aborted due to 1 invalid entity.",
        "success":false,
        "transaction_id":6373011,
        "transactional_error_count":0,
        "transactional_errors":[],
        "updated_entity_count":0
    }
    ```

=== "Request2"

    ```json
    {
      "read_groups": 
        {                                                                    
            "submitter_id":"Read_group_000093"                                            
        },                                                                               
      "data_type": "Unaligned Reads",                                                  
      "file_name": "test_filename.txt",                                                
      "md5sum": "b026324c6904b2a9cb4b88d6d61c81d1",                                    
      "data_format": "FASTQ",                                                          
      "submitter_id": "SUR_000093",                                           
      "data_category": "Sequencing Reads",                   
      "file_size": 1,                                                                  
      "project_id": "GDC-INTERNAL",                                                    
      "type": "submitted_unaligned_reads",                                             
      "experimental_strategy": "WXS",
      "read_pair_number": "R1"                                                 
    }
    ```

=== "Response2"

    ```json
    {
      "cases_related_to_created_entities_count": 1,
      "cases_related_to_updated_entities_count": 0,
      "code": 200,
      "created_entity_count": 1,
      "entities":
        [
            {
                "action":"update",
                "errors":[],
                "id":
                    [
                        {
                            "id":"f0c79d72-ed55-4ac4-82b8-b18335b5ea0e"
                        }
                    ],
                "related_cases":[],
                "type":"case",
                "unique_keys":
                    [
                        {
                            "project_id":"GDC-INTERNAL",
                            "submitter_id":"SUR_000093"
                        }
                    ],
                "valid":true,
                "warnings":[]
            }
        ],
        "entity_error_count":0,
        "message":"Transaction successful.",
        "success":true,
        "transaction_id":6373012,
        "transactional_error_count":0,
        "transactional_errors":[],
        "updated_entity_count":1
    }
    ```

#### INVALID_VALUE Messages

|Message|Explanation|Solution|
| --- | --- | --- |
| __None is not of type 'string'__ | If a property accepts string values, null is not an accepted value | Update the entity to have a string value for the given property, change null to a string "null", or remove the property if it is not required |
| __{value} is not valid under any of the given schemas: {value} is less than the minimum of {minimum value} and {value} is not of type 'null'__ | The amount does not fall within the accepted minimum and maximum values | Ensure the value falls within the accepted range |

Example 1:
=== "Request1"

    ```json
    {
        "channel": "Green",
        "data_category": "DNA Methylation", 
        "data_format": "IDAT", 
        "data_type": "Masked Intensities", 
        "experimental_strategy": "Methylation Array",
        "platform": "Illumina Human Methylation 450", 
        "project_id": "GDC-INTERNAL", 
        "submitter_id": "masked_methylation_array_test1", 
        "file_name": "28daf457-7098-4254-a330-a411da32cd5e_noid_Grn.idat", 
        "file_size": 8095272, "md5sum": "857c7e54f211d0fc6f6bee05b4b9968e",
        "methylation_array_harmonization_workflows": 
            {
                "submitter_id": "test_methylation_array_harmonization_workflow1"
            }, 
        "type": "masked_methylation_array",
        "chip_position": null
    }
    ```

=== "Response1"

    ```json
    {
      "cases_related_to_created_entities_count": 0,
      "cases_related_to_updated_entities_count": 0,
      "code": 400,
      "created_entity_count": 0,
      "entities":
        [
            {
                "action":"create",
                "errors":
                    [
                        {
                            "keys":["chip_position"],
                            "message":"None is not of type 'string'",
                            "type":"INVALID_VALUE"
                        }
                    ]
                "id":
                    [
                        {
                            "id":"fd3f2f5b-ce6d-4028-b5c9-226044c51565"
                        }
                    ],
                "related_cases":[],
                "type":"masked_methylation_array",
                "unique_keys":[
                    {
                        "project_id":"GDC-INTERNAL","submitter_id":"masked_methylation_array_test1"
                    }
                ],
                "valid":false,
                "warnings":[]
            }
        ],
        "entity_error_count":1,
        "message":"Transaction aborted due to 1 invalid entity.",
        "success":false,
        "transaction_id":6373000,
        "transactional_error_count":0,
        "transactional_errors":[],
        "updated_entity_count":0
    }
    ```

=== "Request2"

    ```json
    {
        "channel": "Green",
        "data_category": "DNA Methylation", 
        "data_format": "IDAT", 
        "data_type": "Masked Intensities", 
        "experimental_strategy": "Methylation Array",
        "platform": "Illumina Human Methylation 450", 
        "project_id": "GDC-INTERNAL", 
        "submitter_id": "masked_methylation_array_test1", 
        "file_name": "28daf457-7098-4254-a330-a411da32cd5e_noid_Grn.idat", 
        "file_size": 8095272, "md5sum": "857c7e54f211d0fc6f6bee05b4b9968e",
        "methylation_array_harmonization_workflows": 
            {
                "submitter_id": "test_methylation_array_harmonization_workflow1"
            }, 
        "type": "masked_methylation_array",
        "chip_position": "null"
    }
    ```

=== "Response2"

    ```json
    {
        "cases_related_to_created_entities_count":1,
        "cases_related_to_updated_entities_count":0,
        "code":201,
        "created_entity_count":1,
        "entities":
            [
                {
                    "action":"create",
                    "errors":[],
                    "id":"a7c70ba0-a447-4946-888e-8ea9e903816d",
                    "related_cases":
                        [
                            {
                                "id":"a10f066d-d594-48ed-8d50-25c38e91ed6b",
                                "submitter_id":"GDC-INTERNAL-000093"
                            }
                        ],
                    "type":"diagnosis",
                    "unique_keys":
                        [
                            {
                                "project_id":"GDC-INTERNAL",
                                "submitter_id":"masked_methylation_array_test1"
                            }
                        ],
                    "valid":true,
                    "warnings":[]
                }
            ],
        "entity_error_count":0,
        "message":"Transaction successful.",
        "success":true,
        "transaction_id":6373001,
        "transactional_error_count":0,
        "transactional_errors":[],
        "updated_entity_count":0
    }
    ```

#### MISSING_PROPERTY Messages

|Message|Explanation|Solution|
| --- | --- | --- |
| __'{value}' is a required property__ | The upload is missing a required property | Ensure the required property is included in the file |

Example 1:
=== "Request1"

    ```json
    {
      "projects": {
        "code": "INTERNAL"
      }, 
      "disease_type": "Adenomas and Adenocarcinomas", 
      "primary_site": "Bladder",
      "project_id": "GDC-INTERNAL", 
      "submitter_id": "GDC-INTERNAL-000099"
    }
    ```

=== "Response1"

    ```json
    {
      "cases_related_to_created_entities_count": 0,
      "cases_related_to_updated_entities_count": 0,
      "code": 400,
      "created_entity_count": 0,
      "entities":
        [
            {
                "action":null,
                "errors":
                    [
                        {
                            "keys":["type"],
                            "message":"missing 'type'",
                            "type":"INVALID_TYPE"
                        },
                        {
                            "keys":["type"],
                            "message":"'type' is a required property",
                            "type":"MISSING_PROPERTY"
                        }
                    ]
                "id":
                    [
                        {
                            "id":null
                        }
                    ],
                "related_cases":[],
                "type":null,
                "unique_keys":[],
                "valid":false,
                "warnings":[]
            }
        ],
        "entity_error_count":1,
        "message":"Transaction aborted due to 1 invalid entity.",
        "success":false,
        "transaction_id":6372998,
        "transactional_error_count":0,
        "transactional_errors":[],
        "updated_entity_count":0
    }
    ```

=== "Request2"

    ```json
    {
      "projects": {
        "code": "INTERNAL"
      }, 
      "disease_type": "Adenomas and Adenocarcinomas", 
      "primary_site": "Bladder",
      "project_id": "GDC-INTERNAL", 
      "type": "case",
      "submitter_id": "GDC-INTERNAL-000099"
    }
    ```

=== "Response2"

    ```json
    {
      "cases_related_to_created_entities_count": 0,
      "cases_related_to_updated_entities_count": 0,
      "code": 200,
      "created_entity_count": 0,
      "entities":
        [
            {
                "action":"update",
                "errors":[],
                "id":
                    [
                        {
                            "id":"f0c68d72-ee54-4ad9-89e8-b08398b5aa0b"
                        }
                    ],
                "related_cases":[],
                "type":"case",
                "unique_keys":
                    [
                        {
                            "project_id":"GDC-INTERNAL",
                            "submitter_id":"GDC-INTERNAL-000099"
                        }
                    ],
                "valid":true,
                "warnings":[]
            }
        ],
        "entity_error_count":0,
        "message":"Transaction successful.",
        "success":true,
        "transaction_id":6372999,
        "transactional_error_count":0,
        "transactional_errors":[],
        "updated_entity_count":1
    }
    ```

#### NOT_FOUND Messages

|Message|Explanation|Solution|
| --- | --- | --- |
| __Cannot create node with new submitter id specified. If update action (PUT) requested, Ensure specified id/submitter id exists__ | The submitter_id does not exist so the entity cannot be versioned and replaced with an entity with the new_submitter_id | If the new_submitter_id already exists and the md5sum is the same, this entity has already been uploaded and can be skipped. If neither the submitter_id nor the new_submitter_id exists, remove submitter_id and change "new_submitter_id" to "submitter_id" |
| __Unable to validate case against dbGaP. {}__ | The case submitter_id is not registered in dbGaP | Ensure that there are no typos in the submitter_id or submit additional cases to dbGaP |

Example 1:
=== "Request1"

    ```json
    [
        {
            "data_category": "Sequencing Reads",
            "data_format": "BAM",
            "data_type": "Aligned Reads",
            "project_id": "GDC-INTERNAL",
            "experimental_strategy": "WXS",
            "submitter_id": "Aligned_Reads_000089",
            "new_submitter_id": "Aligned_Reads_000089_v2",
            "file_name": "aligned_reads_00089_test.bam",
            "file_size": 298374,
            "md5sum": "250a27e69c3556312934e0e8eabb2218",
            "alignment_workflows": {
                "submitter_id": "Alignment_Workflow_000088"
            },
            "type": "aligned_reads"
        }
    ]
    ```

=== "Response1"

    ```json
    {
      "cases_related_to_created_entities_count": 0,
      "cases_related_to_updated_entities_count": 0,
      "code": 400,
      "created_entity_count": 0,
      "entities":
        [
            {
                "action":null,
                "errors":
                    [
                        {
                            "keys":[[["GDC-INTERNAL","Aligned_Reads_000089"]]],"message":"Cannot create node with new submitter id specified. If update action (PUT) requested, make sure specified id/submitter id exists","type":"NOT_FOUND"
                        }
                    ]
                "id":
                    [
                        {
                            "id":null
                        }
                    ],
                "related_cases":[],
                "type":"aligned_reads",
                "unique_keys":[
                    {
                        "project_id":"GDC-INTERNAL",
                        "submitter_id":"Aligned_Reads_000089"
                    }
                ],
                "valid":false,
                "warnings":[]
            }
        ],
        "entity_error_count":1,
        "message":"Transaction aborted due to 1 invalid entity.",
        "success":false,
        "transaction_id":6373004,
        "transactional_error_count":0,
        "transactional_errors":[],
        "updated_entity_count":0
    }
    ```

=== "Request2"

    ```json
    [
        {
            "data_category": "Sequencing Reads",
            "data_format": "BAM",
            "data_type": "Aligned Reads",
            "project_id": "GDC-INTERNAL",
            "experimental_strategy": "WXS",
            "platform": "Illumina",
            "submitter_id": "Aligned_Reads_000089_v2",
            "file_name": "aligned_reads_00089_test.bam",
            "file_size": 298374,
            "md5sum": "250a27e69c3556312934e0e8eabb2218",
            "alignment_workflows": {
                "submitter_id": "Alignment_Workflow_000088"
            },
            "type": "aligned_reads"
        }
    ]
    ```

=== "Response2"

    ```json
    {
      "cases_related_to_created_entities_count": 1,
      "cases_related_to_updated_entities_count": 0,
      "code": 200,
      "created_entity_count": 1,
      "entities":
        [
            {
                "action":"create",
                "errors":[],
                "id":
                    [
                        {
                            "id":"ad19702e-7951-47a8-b547-00b077b1547a"
                        }
                    ],
                "related_cases":[
                    {
                        "id":"221db987-6008-4839-a735-63869761b4f9","submitter_id":"GDC-INTERNAL-000088"
                    }
                ],
                "type":"aligned_reads",
                "unique_keys":
                    [
                        {
                            "project_id":"GDC-INTERNAL",
                            "submitter_id":"Aligned_Reads_000089_v2"
                        }
                    ],
                "valid":true,
                "warnings":[]
            }
        ],
        "entity_error_count":0,
        "message":"Transaction successful.",
        "success":true,
        "transaction_id":6373005,
        "transactional_error_count":0,
        "transactional_errors":[],
        "updated_entity_count":1
    }
    ```

Example 2:
=== "Request1"

    ```json
    {
      "projects": {
        "code": "INTERNAL"
      }, 
      "disease_type": "Adenomas and Adenocarcinomas", 
      "primary_site": "Bladder",
      "project_id": "GDC-INTERNAL", 
      "type": "case",
      "submitter_id": "GDC-INTERNAL-000101"
    }
    ```

=== "Reponse1"

    ```json
    {
      "cases_related_to_created_entities_count": 0,
      "cases_related_to_updated_entities_count": 0,
      "code": 400,
      "created_entity_count": 0,
      "entities":
        [
            {
                "action":"create",
                "errors":
                    [
                        {
                            "keys":["submitter_id"],
                            "message":"Unable to validate case against dbGaP. [500] - Internal server error: Unable to cross reference cases with dbGaP. Either this project is not registered in dbGaP or we were temporarily unable to communicate with dbGaP. Please try again later.",
                            "type":"NOT_FOUND"
                        }
                    ]
                "id":
                    [
                        {
                            "id":"GDC-INTERNAL-000101"
                        }
                    ],
                "related_cases":[],
                "type":"case",
                "unique_keys":[
                    {
                        "project_id":"GDC-INTERNAL"
                    }
                ],
                "valid":false,
                "warnings":[]
            }
        ],
        "entity_error_count":1,
        "message":"Transaction aborted due to 1 invalid entity.",
        "success":false,
        "transaction_id":6373005,
        "transactional_error_count":0,
        "transactional_errors":[],
        "updated_entity_count":0
    }
    ```

#### ID Guide

Several errors, described below, stem from the mislabelling of entity IDs (submitter_ids and uuids).

|Message|Explanation|Solution|
| --- | --- | --- |
| __Cannot create an entity with an id that already exists__ | The submitter_id already exists within your project | Make sure the submitter_id is unique within your project |
| __Existing {} entity found with type different from {}__ | The submitter_id already exists for a different entity type in your project | Make sure the submitter_id is unique within your project |
| __Cannot create entity that already exists. Try updating entity (PUT instead of POST)__ | The submitter_id already exists | If attempting to create a new entity, make sure the submitter_id is unique; if attempting to update an existing entity, use PUT instead of POST in the API call |
| __Entity is not unique, (({project_id}, {submitter_id}),)__ | An entity with that submitter_id already exists in the given project | Make sure the submitter_id is unique |

Example 1:
=== "Request1"

    ```json
    {
      "projects": {
        "code": "INTERNAL"
      }, 
      "disease_type": "Adenomas and Adenocarcinomas", 
      "primary_site": "Bladder",
      "project_id": "GDC-INTERNAL", 
      "type": "case",
      "submitter_id": "GDC-INTERNAL-000078"
    }
    ```

=== "Response1"

    ```json
    {
      "cases_related_to_created_entities_count": 0,
      "cases_related_to_updated_entities_count": 0,
      "code": 400,
      "created_entity_count": 0,
      "entities":
        [
            {
                "action":null,
                "errors":
                    [
                        {
                            "keys":["id"],
                            "message":"Cannot create an entity with an id that already exists.",
                            "type":"NOT_UNIQUE"
                        }
                    ]
                "id":null
                "related_cases":[],
                "type":"case",
                "unique_keys":
                    [
                        {
                            "project_id":"GDC-INTERNAL","submitter_id":"GDC-INTERNAL-000078"
                        }
                    ]
                "valid":false,
                "warnings":[]
            }
        ],
        "entity_error_count":1,
        "message":"Transaction aborted due to 1 invalid entity.",
        "success":false,
        "transaction_id":6466260,
        "transactional_error_count":0,
        "transactional_errors":[],
        "updated_entity_count":0
    }
    ```

=== "Request2"

    ```json
    {
      "projects": {
        "code": "INTERNAL"
      }, 
      "disease_type": "Adenomas and Adenocarcinomas", 
      "primary_site": "Bladder",
      "project_id": "GDC-INTERNAL", 
      "type": "case",
      "submitter_id": "GDC-INTERNAL-000028"
    }
    ```

=== "Response2"

    ```
    {
      "cases_related_to_created_entities_count": 0,
      "cases_related_to_updated_entities_count": 0,
      "code": 201,
      "created_entity_count": 1,
      "entities":
        [
            {
                "action":"create",
                "errors":[],
                "id":"3cb0dc61-2375-4edc-b8b9-7962e2362a65",
                "related_cases":[],
                "type":"case",
                "unique_keys":
                    [
                        {
                            "project_id":"GDC-INTERNAL","submitter_id":"GDC-INTERNAL-000028"
                        }
                    ]
                "valid":true,
                "warnings":[]
            }
        ],
        "entity_error_count":0,
        "message":"Transaction successful.",
        "success":true,
        "transaction_id":6466261,
        "transactional_error_count":0,
        "transactional_errors":[],
        "updated_entity_count":0
    }
    ```

Example 2:
=== "Request1"

    ```json
    [
      {
        "type": "case",
        "submitter_id": "GDC-INTERNAL-000029",
        "primary_site": "Base of tongue",
        "disease_type": "Acinar Cell Neoplasms",
        "projects": {
          "code": "INTERNAL"
        }
      },
      {
        "type": "family_history",
        "submitter_id": "GDC-INTERNAL-000029",
        "relative_deceased": "Yes",
        "relative_smoker": "No",
        "cases": {
          "submitter_id": "GDC-INTERNAL-000029"
        }
      }
    ]
    ```

=== "Response1"

    ```json
    {
      "cases_related_to_created_entities_count": 0,
      "cases_related_to_updated_entities_count": 0,
      "code": 400,
      "created_entity_count": 0,
      "entities":
        [
            {
                "action":null,
                "errors":
                    [
                        {
                            "keys":["id"],
                            "message":"Existing case entity found with type different from family_history.",
                            "type":"NOT_UNIQUE"
                        }
                    ]
                "id":null
                "related_cases":[],
                "type":"family_history",
                "unique_keys":
                    [
                        {
                            "project_id":"GDC-INTERNAL","submitter_id":"GDC-INTERNAL-000029"
                        }
                    ]
                "valid":false,
                "warnings":[]
            }
        ],
        "entity_error_count":1,
        "message":"Transaction aborted due to 1 invalid entity.",
        "success":false,
        "transaction_id":6466263,
        "transactional_error_count":0,
        "transactional_errors":[],
        "updated_entity_count":0
    }
    ```

=== "Request2"

    ```json
    {
        "type": "case",
        "submitter_id": "GDC-INTERNAL-000029",
        "primary_site": "Base of tongue",
        "disease_type": "Acinar Cell Neoplasms",
        "projects": {
          "code": "INTERNAL"
        }
      },
      {
        "type": "family_history",
        "submitter_id": "GDC-INTERNAL-000029_family_history",
        "relative_deceased": "Yes",
        "relative_smoker": "No",
        "cases": {
          "submitter_id": "GDC-INTERNAL-000029"
        }
    }
    ```

=== "Response2"

    ```json
    {
      "cases_related_to_created_entities_count": 0,
      "cases_related_to_updated_entities_count": 0,
      "code": 201,
      "created_entity_count": 1,
      "entities":
        [
            {
                "action":"create",
                "errors":[],
                "id":"3ae1eb12-1724-4cba-b6c6-1283b1872c16",
                "related_cases":[],
                "type":"case",
                "unique_keys":
                    [
                        {
                            "project_id":"GDC-INTERNAL","submitter_id":"GDC-INTERNAL-000028_family_history"
                        }
                    ]
                "valid":true,
                "warnings":[]
            }
        ],
        "entity_error_count":0,
        "message":"Transaction successful.",
        "success":true,
        "transaction_id":6466262,
        "transactional_error_count":0,
        "transactional_errors":[],
        "updated_entity_count":0
    }
    ```


# Submission Workflow

## Overview

The workflow diagram below represents the data submission process implemented by the GDC Data Submission Portal. The submitter logs into the GDC Data Submission Portal, uploads data into the project workspace, and validates the data. When the data is ready for processing, the submitter reviews the data and submits it to the GDC. The GDC harmonizes the data through the [GDC Harmonization Process](https://gdc.cancer.gov/submit-data/gdc-data-harmonization). The submitter can then release the harmonized data to the [GDC Data Portal](https://portal.gdc.cancer.gov/) and other [GDC Data Access Tools](https://gdc.cancer.gov/access-data/data-access-processes-and-tools) according to [GDC Data Sharing Policies](https://gdc.cancer.gov/submit-data/data-submission-policies).

[![GDC Data Submission Portal Workflow](images/gdc-submission-portal-submission-workflow.png)](images/gdc-submission-portal-submission-workflow.png "Click to see the full image.")

### Upload and Validate Data

The submitter uploads Clinical and Biospecimen data to the project workspace using GDC templates that are available in the [GDC Data Dictionary](https://docs.gdc.cancer.gov/Data_Dictionary/). The GDC will validate the uploaded data against the GDC Data Dictionary.

To upload submittable data files, such as sequence data in BAM or FASTQ format, the submitter must register file metadata with the GDC using a method similar to uploading Clinical and Biospecimen data. When files are registered, the submitter downloads a manifest from the GDC Data Submission Portal and uses it with the [GDC Data Transfer Tool](https://gdc.cancer.gov/access-data/gdc-data-transfer-tool) to upload the files.

[![GDC Data Submission Portal Workflow Upload](images/gdc-submission-portal-data-upload-workflow.png)](images/gdc-submission-portal-data-upload-workflow.png "Click to see the full image.")

### Review and Submit Data

When all necessary data and files have been uploaded, the submitter reviews the dataset and submits it to the GDC for processing through the [GDC Data Harmonization Pipeline](https://gdc.cancer.gov/submit-data/gdc-data-harmonization).

### Release Data

The GDC will release data according to [GDC data sharing policies](https://gdc.cancer.gov/submit-data/data-submission-policies). A project may be released after six months from the date of upload or the submitter may request earlier release using the "Release Project" function.  

Harmonized data will be released to GDC users through the [GDC Data Portal](https://gdc-portal.nci.nih.gov/) and other [GDC Data Access Tools](https://gdc.cancer.gov/access-data/data-access-processes-and-tools). Once a project is released, all additional submitted data will automatically be released after harmonization.

## Project Lifecycle

The lifecycle of a project in the GDC describes the workflow throughout the data submission process. The project lifecycle starts with the upload and validation of data into the project and ends with the release of the harmonized data to the GDC Data Portal and other GDC data access tools. Throughout the lifecycle, the project transitions through various states in which the project is open for uploading data, in review, and processing. This lifecycle is continuous as new project data becomes available.

### Project State
The diagram below demonstrates the transition of a project through the various states. Initially the project is OPEN for data upload and validation. Any changes to the data must be made while the project status is OPEN. When the data is uploaded and ready for review, the submitter changes the project state to REVIEW by clicking the _'REVIEW'_ button on the dashboard. During the REVIEW state, the project is locked so that additional data cannot be uploaded. If data changes are needed during the review period, the project can be re-opened and the state changes back to OPEN. When review has been completed and the submitter submits the data for GDC processing, the project state changes to SUBMITTED. When the data has been processed, the project state changes back to OPEN for new data to be submitted to the project.

[![GDC Data Submission Portal Project State](images/gdc-submission-portal-project-states.png)](images/gdc-submission-portal-project-states.png "Click to see the full image.")


## File Status Lifecycle

This section describes states pertaining to submittable data files throughout the data submission process. A submittable data file could contain data such as genomic sequences (such as a BAM or FASTQ) or pathology slide images. The file lifecycle starts when a submitter uploads metadata for a file to the GDC Data Submission Portal. Upload of file metadata, as a GDC Data Model entity, registers a description of the file in the GDC. The submitter can then use the [GDC Data Transfer Tool](https://gdc.cancer.gov/access-data/gdc-data-transfer-tool) to upload the actual file. Throughout the lifecycle, the file status transitions through various states from when the file is initially registered through file submission and processing. The diagram below details these status transitions.   

[![GDC Data Submission Portal File Status](images/gdc-submission-portal-file-state-vs-state.png)](images/gdc-submission-portal-file-state-vs-state.png "Click to see the full image.")


# Upload Data

This guide details step-by-step procedures for different aspects of the GDC Data Submission process and how they relate to the GDC Data Model and structure. The first sections of this guide break down the submission process and associate each step with the Data Model. Additional sections are detailed below for strategies on expediting data submission and using features of the Submission Portal.

## GDC Data Model Basics

Pictured below is the submittable subset of the GDC Data Model: a roadmap for GDC data submission. The entities that make up the completed (submitted) portion of the submission process will be highlighted in __blue__.

[![GDC Data Model 1](images/GDC-Data-Model-None.png)](images/GDC-Data-Model-None.png "Click to see the full image.")

Each entity type is represented with an oval in the above graphic. All submitted entities require a connection to another entity type, based on the GDC Data Model, and a `submitter_id` as an identifier.

### The Case Entity and Clinical Data

The `case` is the center of the GDC Data Model and usually describes a specific patient. Each `case` is connected to a `project`.  Different types of clinical data, such as `diagnoses` and `exposures`, are connected to the `case` to describe the case's attributes and medical information.   

### Biospecimen Data

One of the main features of the GDC is the genomic data harmonization workflow. Genomic data is connected the the case through biospecimen entities.  The `sample` entity describes a biological piece of matter that originated from a `case`.  Subsets of the `sample` such as `portions` and `analytes` can optionally be described.  The `aliquot` originates from a `sample` or `analyte` and describes the nucleic acid extract that was sequenced. The `read_group` entity describes the resulting set of reads from one sequencing lane.

### Experiment Data

Several types of experiment data can be uploaded to the GDC.  The `submitted_aligned_reads` and `submitted_unaligned_reads` files are associated with the `read_group` entity. While the array-based files such as the `submitted_tangent_copy_number` are associated with the `aliquot` entity.  Each of these file types are described in their respective entity submission and are uploaded separately using the API or the GDC Data Transfer Tool.  

## Program and Project Registration

Before submission can begin, the Program and Project must be approved and set by the GDC.

### Program and Project Approval

Each new project must [request submission access](https://gdc.cancer.gov/submit-data/requesting-data-submission) from the GDC.  Before submission can commence the project must be registered at dbGaP along with the eRA commons IDs for all those users who will be uploading data. All cases (i.e. patients) must also be registered in dbGaP for that particular project.  Once these steps are complete the GDC will grant submission access and create program and project names in consultation with the user based on the rules outlined below.  

### Program and Project Naming Conventions

The program is assigned a `program.name`, which uniquely identifies that program. Each program may have multiple projects and will be assigned a `project.code`, which uniquely identifies each project. The `project_id` is the main identifier for the project in the GDC system and comprises the `program.name` with the `project.code` appended to it with a dash.  For example:

```Example
program.name = TCGA
project.code = BRCA
project.project_id = TCGA-BRCA
```  

## Case Submission

[![GDC Data Model 2](images/GDC-Data-Model-Case.png)](images/GDC-Data-Model-Case.png "Click to see the full image.")

The main entity of the GDC Data Model is the `case`, each of which must be registered beforehand with dbGaP under a unique `submitter_id`. The first step to submitting a `case` is to consult the [Data Dictionary](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#data-dictionary-viewer), which details the fields that are associated with a `case`, the fields that are required to submit a `case`, and the values that can populate each field. Dictionary entries are available for all entities in the GDC Data Model.

[![Dictionary Case](images/Dictionary_Case.png)](images/Dictionary_Case.png "Click to see the full image.")

Submitting a [__Case__](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=case) entity requires:

* __`submitter_id`:__ A unique key to identify the `case`
* __`projects.code`:__ A link to the `project`

The submitter ID is different from the universally unique identifier (UUID), which is based on the [UUID Version 4 Naming Convention](https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_4_.28random.29). The UUID can be accessed under the `<entity_type>_id` field for each entity. For example, the `case` UUID can be accessed under the `case_id` field. The UUID is either assigned to each entity automatically or can be submitted by the user. Submitter-generated UUIDs cannot be uploaded in `submittable_data_file` entity types. See the [Data Model Users Guide](https://docs.gdc.cancer.gov/Data/Data_Model/GDC_Data_Model/#gdc-identifiers) for more details about GDC identifiers.

The `projects.code` field connects the `case` entity to the `project` entity.  The rest of the entity connections use the `submitter_id` field instead.  

The `case` entity can be added in JSON or TSV format. A template for any entity in either of these formats can be found in the Data Dictionary at the top of each page. Templates populated with `case` metadata in both formats are displayed below.  

```JSON
{
    "type": "case",
    "submitter_id": "PROJECT-INTERNAL-000055",
    "projects": {
        "code": "INTERNAL"
    }
}
```
```TSV
type  submitter_id  projects.code
case  PROJECT-INTERNAL-000055 INTERNAL   
```

__Note:__ JSON and TSV formats handle links between entities (`case` and `project`) differently.  JSON includes the `code` field nested within `projects` while TSV appends `code` to `projects` with a period.  


## Uploading the Case Submission File

The file detailed above can be uploaded using the Data Submission Portal and the API as described below:

### Upload - Data Submission Portal

#### 1. Upload Files

An example of a `case` upload is detailed below. The GDC Data Submission Portal is equipped with a wizard window to facilitate the upload and validation of entities. The Upload Data Wizard comprises two stages:

* __Upload Entities__: Upload an entity into the user's browser, at this point nothing is submitted to the project workspace.
* __Validate Entities__: Send an entity to the GDC backend to validate its content (see below)

The __Validate Entities__ stage acts as a safeguard against submitting incorrectly formatted data to the GDC Data Submission Portal. During the validation stage, the GDC API will validate the content of uploaded entities against the Data Dictionary to detect potential errors. Invalid entities will not be processed and must be corrected by the user and re-uploaded before being accepted. A validation error report provided by the system can be used to isolate and correct errors.

Choosing _'UPLOAD'_ from the project dashboard will open the Upload Data Wizard.

[![GDC Submission Wizard Upload Files](images/GDC_Submission_Wizard_Upload_2.png)](images/GDC_Submission_Wizard_Upload_2.png "Click to see the full image.")

Files containing one or more entities can be added either by clicking on _'CHOOSE FILE(S)'_ or using drag and drop. Files can be removed from the Upload Data Wizard by clicking on the garbage can icon next to the file.

#### 2. Validate Entities

When the first file is added, the wizard will move to the _'VALIDATE'_ section and the user can continue to add files.

[![GDC Submission Wizard Validate Files](images/GDC_Submission_Portal_Validate.png)](images/GDC_Submission_Portal_Validate.png "Click to see the full image.")

When all files have been added, choosing _'VALIDATE'_ will run a test to check if the entities are valid for submission.

#### 3. Commit or Discard Files
If the upload contains valid entities, a new transaction will appear in the latest transactions panel with the option to _'COMMIT'_ or _'DISCARD'_ the data. Entities contained in these files can be committed (applied) to the project or discarded using these two buttons.

If the upload contains invalid files, a transaction will appear with a FAILED status. Invalid files will need to be either corrected and re-uploaded or removed from the submission. If more than one file is uploaded and at least one is not valid, the validation step will fail for all files.  

[![Commit_Discard](images/GDC_Submission_CommitDiscard.png)](images/GDC_Submission_CommitDiscard.png "Click to see the full image.")


### Upload - API

The API has a much broader range of functionality than the Data Wizard. Entities can be created, updated, and deleted through the API. See the [API Submission User Guide](API/Users_Guide/Submission/#creating-and-updating-entities) for a more detailed explanation and for the rest of the functionalities of the API. Generally, uploading an entity through the API can be performed using a command similar to the following:

```Shell
curl --header "X-Auth-Token: $token" --request POST --data @CASE.json https://api.gdc.cancer.gov/v0/submission/GDC/INTERNAL/_dry_run?async=true
```
CASE.json is detailed below.
```json
{
    "type": "case",
    "submitter_id": "PROJECT-INTERNAL-000055",
    "projects": {
        "code": "INTERNAL"
    }
}
```

__Note:__ Submission of TSV files is also supported by the GDC API.

Next, the file can either be committed (applied to the project) through the Data Submission Portal as before, or another API query can be performed that will commit the file to the project. The transaction number in the URL (467) is printed to the console during the first step of API submission and can also be retrieved from the 'Transactions' tab in the Data Submission Portal.

```Shell
curl --header "X-Auth-Token: $token" --request POST https://api.gdc.cancer.gov/v0/submission/GDC/INTERNAL/transactions/467/commit?async=true
```

## Clinical Submission

[![GDC Data Model Clinical](images/GDC-Data-Model-Clinical.png)](images/GDC-Data-Model-Clinical.png "Click to see the full image.")

Typically a submission project will include additional information about a `case` such as `demographic`, `diagnosis`, or `exposure` data.

### Submitting a Demographic Entity to a Case

The `demographic` entity contains information that characterizes the `case` entity.  

Submitting a [__Demographic__](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=demographic) entity requires:

* __`submitter_id`:__ A unique key to identify the `demographic` entity
* __`cases.submitter_id`:__ The unique key that was used for the `case` that links the `demographic` entity to the `case`
* __`ethnicity`:__ An individual's self-described social and cultural grouping, specifically whether an individual describes themselves as Hispanic or Latino. The provided values are based on the categories defined by the U.S. Office of Management and Business and used by the U.S. Census Bureau
* __`gender`:__ Text designations that identify gender. Gender is described as the assemblage of properties that distinguish people on the basis of their societal roles
* __`race`:__ An arbitrary classification of a taxonomic group that is a division of a species. It usually arises as a consequence of geographical isolation within a species and is characterized by shared heredity, physical attributes and behavior, and in the case of humans, by common history, nationality, or geographic distribution. The provided values are based on the categories defined by the U.S. Office of Management and Business and used by the U.S. Census Bureau
* __`year_of_birth`:__ Numeric value to represent the calendar year in which an individual was born

```JSON
{
    "type": "demographic",
    "submitter_id": "PROJECT-INTERNAL-000055-DEMOGRAPHIC-1",
    "cases": {
        "submitter_id": "PROJECT-INTERNAL-000055"
    },
    "ethnicity": "not hispanic or latino",
    "gender": "male",
    "race": "asian",
    "year_of_birth": 1946
}
```
```TSV
type	cases.submitter_id	ethnicity	gender	race	year_of_birth
demographic	PROJECT-INTERNAL-000055	not hispanic or latino	male	asian	1946
```

### Submitting a Diagnosis Entity to a Case

Submitting a [__Diagnosis__](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=diagnosis) entity requires:

* __`submitter_id`:__ A unique key to identify the `diagnosis` entity
* __`cases.submitter_id`:__ The unique key that was used for the `case` that links the `diagnosis` entity to the `case`
* __`age_at_diagnosis`:__ Age at the time of diagnosis expressed in number of days since birth
* __`classification_of_tumor`:__ Text that describes the kind of disease present in the tumor specimen as related to a specific timepoint
* __`days_to_last_follow_up`:__  Time interval from the date of last follow up to the date of initial pathologic diagnosis, represented as a calculated number of days
* __`days_to_last_known_disease_status`:__ Time interval from the date of last follow up to the date of initial pathologic diagnosis, represented as a calculated number of days
* __`days_to_recurrence`:__ Time interval from the date of new tumor event including progression, recurrence and new primary malignancies to the date of initial pathologic diagnosis, represented as a calculated number of days
* __`last_known_disease_status`:__  The state or condition of an individual's neoplasm at a particular point in time
* __`morphology`:__  The third edition of the International Classification of Diseases for Oncology, published in 2000 used principally in tumor and cancer registries for coding the site (topography) and the histology (morphology) of neoplasms. The study of the structure of the cells and their arrangement to constitute tissues and, finally, the association among these to form organs. In pathology, the microscopic process of identifying normal and abnormal morphologic characteristics in tissues, by employing various cytochemical and immunocytochemical stains. A system of numbered categories for representation of data
* __`primary_diagnosis`:__  Text term for the structural pattern of cancer cells used to define a microscopic diagnosis
* __`progression_or_recurrence`:__ Yes/No/Unknown indicator to identify whether a patient has had a new tumor event after initial treatment
* __`site_of_resection_or_biopsy`:__ The third edition of the International Classification of Diseases for Oncology, published in 2000, used principally in tumor and cancer registries for coding the site (topography) and the histology (morphology) of neoplasms. The description of an anatomical region or of a body part. Named locations of, or within, the body. A system of numbered categories for representation of data
* __`tissue_or_organ_of_origin`:__ Text term that describes the anatomic site of the tumor or disease
* __`tumor_grade`:__ Numeric value to express the degree of abnormality of cancer cells, a measure of differentiation and aggressiveness
* __`tumor_stage`:__ The extent of a cancer in the body. Staging is usually based on the size of the tumor, whether lymph nodes contain cancer, and whether the cancer has spread from the original site to other parts of the body. The accepted values for tumor_stage depend on the tumor site, type, and accepted staging system. These items should accompany the tumor_stage value as associated metadata
* __`vital_status`:__ The survival state of the person registered on the protocol

```JSON
{
    "type": "diagnosis",
    "submitter_id": "PROJECT-INTERNAL-000055-DIAGNOSIS-1",
    "cases": {
        "submitter_id": "GDC-INTERNAL-000099"
    },
    "age_at_diagnosis": 10256,
    "classification_of_tumor": "not reported",
    "days_to_last_follow_up": 34,
    "days_to_last_known_disease_status": 34,
    "days_to_recurrence": 45,
    "last_known_disease_status": "Tumor free",
    "morphology": "8260/3",
    "primary_diagnosis": "ACTH-producing tumor",
    "progression_or_recurrence": "no",
    "site_of_resection_or_biopsy": "Lung, NOS",
    "tissue_or_organ_of_origin": "Lung, NOS",
    "tumor_grade": "Not Reported",
    "tumor_stage": "stage i",
    "vital_status": "alive"
}
```
```TSV
type	submitter_id	cases.submitter_id	age_at_diagnosis	classification_of_tumor	days_to_last_follow_up	days_to_last_known_disease_status	days_to_recurrence	last_known_disease_status	morphology	primary_diagnosis	progression_or_recurrence	site_of_resection_or_biopsy	tissue_or_organ_of_origin	tumor_grade	tumor_stage	vital_status
diagnosis	PROJECT-INTERNAL-000055-DIAGNOSIS-1	GDC-INTERNAL-000099	10256	not reported	34	34	45	Tumor free	8260/3	ACTH-producing tumor	no	Lung, NOS	Lung, NOS	not reported	stage i	alive
```
__Note:__ For information on submitting time-based data for patients over 90 years old see the [GDC Submission Best Practices](Best_Practices.md) Guide.

### Submitting an Exposure Entity to a Case

Submitting an [__Exposure__](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=exposure) entity does not require any information besides a link to the `case` and a `submitter_id`.  The following fields are optionally included:  

* __`alcohol_history`:__ A response to a question that asks whether the participant has consumed at least 12 drinks of any kind of alcoholic beverage in their lifetime
* __`alcohol_intensity`:__ Category to describe the patient's current level of alcohol use as self-reported by the patient
* __`bmi`:__ The body mass divided by the square of the body height expressed in units of kg/m^2
* __`cigarettes_per_day`:__ The average number of cigarettes smoked per day (number)
* __`height`:__ The height of the individual in cm (number)
* __`weight`:__ The weight of the individual in kg (number)
* __`years_smoked`:__ Numeric value (or unknown) to represent the number of years a person has been smoking

```JSON
{
    "type": "exposure",
    "submitter_id": "PROJECT-INTERNAL-000055-EXPOSURE-1",
    "cases": {
        "submitter_id": "PROJECT-INTERNAL-000055"
    },
    "alcohol_history": "yes",
    "bmi": 27.5,
    "cigarettes_per_day": 20,
    "height": 190,
    "weight": 100,
    "years_smoked": 5
}
```
```TSV
type	submitter_id	cases.submitter_id	alcohol_history	bmi	cigarettes_per_day	height	weight	years_smoked
exposure	PROJECT-INTERNAL-000055-EXPOSURE-1	PROJECT-INTERNAL-000055	yes	27.5	20	190	100	5
```

__Note:__ Submitting a clinical entity uses the same conventions as submitting a `case` entity (detailed above).


## Biospecimen Submission

### Sample Submission

[![GDC Data Model 3](images/GDC-Data-Model-Sample.png)](images/GDC-Data-Model-Sample.png "Click to see the full image.")

A `sample` submission has the same general structure as a `case` submission as it will require a unique key and a link to the `case`.  However, `sample` entities require one additional value:  `sample_type`. This peripheral data is required because it is necessary for the data to be interpreted. For example, an investigator using this data would need to know whether the `sample` came from tumor or normal tissue.  


[![Dictionary Sample](images/Dictionary_Sample.png)](images/Dictionary_Sample.png "Click to see the full image.")

Submitting a [__Sample__](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=sample) entity requires:

* __`submitter_id`:__ A unique key to identify the `sample`
* __`cases.submitter_id`:__ The unique key that was used for the `case` that links the `sample` to the `case`
* __`sample_type`:__ Type of the `sample`. Named for its cellular source, molecular composition, and/or therapeutic treatment

__Note:__ The `case` must be "committed" to the project before a `sample` can be linked to it.  This also applies to all other links between entities.

```JSON
{
    "type": "sample",
    "cases": {
        "submitter_id": "PROJECT-INTERNAL-000055"
    },
    "sample_type": "Blood Derived Normal",
    "submitter_id": "Blood-00001SAMPLE_55"
}
```
```TSV
type	cases.submitter_id	submitter_id	sample_type
sample	PROJECT-INTERNAL-000055	Blood-00001SAMPLE_55	Blood Derived Normal  
```



### Aliquot, Portion, and Analyte Submission

[![GDC Data Model 4](images/GDC-Data-Model-Aliquot.png)](images/GDC-Data-Model-Aliquot.png "Click to see the full image.")


Submitting an [__Aliquot__](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=aliquot) entity requires:

* __`submitter_id`:__ A unique key to identify the `aliquot`
* __`analytes.submitter_id`:__ The unique key that was used for the `analyte` that links the `aliquot` to the `analyte`

```JSON
{
    "type": "aliquot",
    "submitter_id": "Blood-00021-aliquot55",
    "samples": {
        "submitter_id": "Blood-00001SAMPLE_55"
    }
}

```
```TSV
type	submitter_id	analytes.submitter_id
aliquot	Blood-00021-aliquot55	Blood-00001SAMPLE_55
```

__Note:__ `aliquot` entities can be directly linked to `sample` entities. The `portion` and `analyte` entities described below are not required for submission.

Submitting a [__Portion__](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=portion) entity requires:

* __`submitter_id`:__ A unique key to identify the `portion`
* __`samples.submitter_id`:__ The unique key that was used for the `sample` that links the `portion` to the `sample`

```JSON
{
    "type": "portion",
    "submitter_id": "Blood-portion-000055",
    "samples": {
        "submitter_id": "Blood-00001SAMPLE_55"
    }
}

```
```TSV
type	submitter_id	samples.submitter_id
portion	Blood-portion-000055	Blood-00001SAMPLE_55
```

Submitting an [__Analyte__](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=analyte) entity requires:

* __`submitter_id`:__ A unique key to identify the `analyte`
* __`portions.submitter_id`:__ The unique key that was used for the `portion` that links the `analyte` to the `portion`
* __`analyte_type`:__ Protocol-specific molecular type of the specimen

```JSON
{
    "type": "analyte",
    "portions": {
        "submitter_id": "Blood-portion-000055"
    },
    "analyte_type": "DNA",
    "submitter_id": "Blood-analyte-000055"
}

```
```TSV
type	portions.submitter_id	analyte_type	submitter_id
analyte	Blood-portion-000055	DNA	Blood-analyte-000055
```

### Read Group Submission

[![GDC Data Model 5](images/GDC-Data-Model-RG.png)](images/GDC-Data-Model-RG.png "Click to see the full image.")

Because information about sequencing reads is necessary for downstream analysis, the `read_group` entity requires more fields than the other Biospecimen entities (`sample`, `portion`, `analyte`, `aliquot`).

Submitting a [__Read Group__](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=read_group) entity requires:

* __`submitter_id`:__ A unique key to identify the `read_group`
* __`aliquot.submitter_id`:__ The unique key that was used for the `aliquot` that links the `read_group` to the `aliquot`
* __`experiment_name`:__ Submitter-defined name for the experiment
* __`is_paired_end`:__ Are the reads paired end? (Boolean value: `true` or `false`)
* __`library_name`:__ Name of the library  
* __`library_strategy`:__ Library strategy
* __`platform`:__ Name of the platform used to obtain data
* __`read_group_name`:__ The name of the `read_group`
* __`read_length`:__ The length of the reads (integer)
* __`sequencing_center`:__ Name of the center that provided the sequence files  
* __`library_selection`:__ Library Selection Method
* __`target_capture_kit`:__ Description that can uniquely identify a target capture kit. Suggested value is a combination of vendor, kit name, and kit version.
```JSON
{
    "type": "read_group",
    "submitter_id": "Blood-00001-aliquot_lane1_barcodeACGTAC_55",
    "experiment_name": "Resequencing",
    "is_paired_end": true,
    "library_name": "Solexa-34688",
    "library_strategy": "WXS",
    "platform": "Illumina",
    "read_group_name": "205DD.3-2",
    "read_length": 75,
    "sequencing_center": "BI",
    "library_selection": "Hybrid Selection",
    "target_capture_kit": "Custom MSK IMPACT Panel - 468 Genes",
    "aliquots":
        {
            "submitter_id": "Blood-00021-aliquot55"
        }    
}

```
```TSV
type	submitter_id	experiment_name	is_paired_end	library_name	library_strategy	platform	read_group_name	read_length	sequencing_center library_selection target_capture_kit	aliquots.submitter_id
read_group	Blood-00001-aliquot_lane1_barcodeACGTAC_55	Resequencing	true	Solexa-34688	WXS	Illumina	205DD.3-2	75	BI	Hybrid Selection Custom MSK IMPACT Panel - 468 Genes Blood-00021-aliquot55
```

__Note:__ Submitting a biospecimen entity uses the same conventions as submitting a `case` entity (detailed above).

## Experiment Data Submission

[![GDC Data Model 6](images/GDC-Data-Model-Reads.png)](images/GDC-Data-Model-Reads.png "Click to see the full image.")

Before the experiment data file can be submitted, the GDC requires that the user provides information about the file as a `submittable_data_file` entity. This includes file-specific data needed to validate the file and assess which analyses should be performed. Sequencing data files can be submitted as `submitted_aligned_reads` or `submitted_unaligned_reads`.

Submitting a [__Submitted Aligned-Reads__](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=submitted_aligned_reads) entity requires:

* __`submitter_id`:__ A unique key to identify the `submitted_aligned_reads`
* __`read_groups.submitter_id`:__ The unique key that was used for the `read_group` that links the `submitted_aligned_reads` to the `read_group`
* __`data_category`:__ Broad categorization of the contents of the data file
* __`data_format`:__ Format of the data files
* __`data_type`:__ Specific content type of the data file. (must be "Aligned Reads")
* __`experimental_strategy`:__ The sequencing strategy used to generate the data file
* __`file_name`:__ The name (or part of a name) of a file (of any type)
* __`file_size`:__ The size of the data file (object) in bytes
* __`md5sum`:__ The 128-bit hash value expressed as a 32 digit hexadecimal number used as a file's digital fingerprint


```JSON
{
    "type": "submitted_aligned_reads",
    "submitter_id": "Blood-00001-aliquot_lane1_barcodeACGTAC_55.bam",
    "data_category": "Raw Sequencing Data",
    "data_format": "BAM",
    "data_type": "Aligned Reads",
    "experimental_strategy": "WGS",
    "file_name": "test.bam",
    "file_size": 38,
    "md5sum": "aa6e82d11ccd8452f813a15a6d84faf1",
    "read_groups": [
        {
            "submitter_id": "Primary_Tumor_RG_86-1"
        }
    ]
}
```
```TSV
type	submitter_id	data_category	data_format	data_type	experimental_strategy	file_name	file_size	md5sum	read_groups.submitter_id#1
submitted_aligned_reads	Blood-00001-aliquot_lane1_barcodeACGTAC_55.bam	Raw Sequencing Data	BAM	Aligned Reads	WGS	test.bam	38	aa6e82d11ccd8452f813a15a6d84faf1	Primary_Tumor_RG_86-1
```

Submitting a [__Submitted Unaligned-Reads__](https://gdc-docs.nci.nih.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=submitted_unaligned_reads) entity requires:

* __`submitter_id`:__ A unique key to identify the `submitted_unaligned_reads`
* __`read_groups.submitter_id`:__ The unique key that was used for the `read_group` that links the `submitted_unaligned_reads` to the `read_group`
* __`data_category`:__ Broad categorization of the contents of the data file
* __`data_format`:__ Format of the data files
* __`data_type`:__ Specific content type of the data file. (must be "Unaligned Reads")
* __`experimental_strategy`:__ The sequencing strategy used to generate the data file
* __`file_name`:__ The name (or part of a name) of a file (of any type)
* __`file_size`:__ The size of the data file (object) in bytes
* __`md5sum`:__ The 128-bit hash value expressed as a 32 digit hexadecimal number used as a file's digital fingerprint


```JSON
{
    "type": "submitted_unaligned_reads",
    "submitter_id": "Blood-00001-aliquot_lane2_barcodeACGTAC_55.fastq",
    "data_category": "Raw Sequencing Data",
    "data_format": "FASTQ",
    "data_type": "Unaligned Reads",
    "experimental_strategy": "WGS",
    "file_name": "test.fastq",
    "file_size": 38,
    "md5sum": "901d48b862ea5c2bcdf376da82f2d22f",
    "read_groups": [
        {
            "submitter_id": "Primary_Tumor_RG_86-1"
        }
    ]
}
```
```TSV
type	submitter_id	data_category	data_format	data_type	experimental_strategy	file_name	file_size	md5sum	read_groups.submitter_id
submitted_unaligned_reads	Blood-00001-aliquot_lane2_barcodeACGTAC_55.fastq	Raw Sequencing Data	FASTQ	Unaligned Reads	WGS	test.fastq	38	901d48b862ea5c2bcdf376da82f2d22f
Primary_Tumor_RG_86-1
```

__Note:__ For details on submitting experiment data associated with more than one `read_group` entity, see the [GDC Submission Best Practices](Best_Practices.md) Guide.    

### Uploading the Submittable Data File to the GDC

The submittable data file can be uploaded when it is registered with the GDC. An submittable data file is registered when its corresponding entity (e.g. `submitted_unaligned_reads`) is uploaded and committed. Uploading the file can be performed with either the GDC Data Transfer Tool or the API. Other types of data files such as clinical supplements, biospecimen supplements, and pathology reports are uploaded to the GDC in the same way. Supported data file formats are listed at the GDC [Submitted Data Types and File Formats](https://gdc.cancer.gov/about-data/data-types-and-file-formats/submitted-data-types-and-file-formats) website.

__GDC Data Transfer Tool:__ A file can be uploaded using its UUID (which can be retrieved from the portal or API) once it is registered. The following command can be used to upload the file:

```Shell
gdc-client upload --project-id PROJECT-INTERNAL --identifier a053fad1-adc9-4f2d-8632-923579128985 -t $token -f $path_to_file
```   

Additionally a manifest can be downloaded from the Submission Portal and passed to the Data Transfer Tool, this will allow for the upload of more than one `submittable_data_file`:

```Shell
gdc-client upload -m manifest.yml -t $token
```
__API Upload:__  A `submittable_data_file` can be uploaded through the API by using the `/submission/program/project/files` endpoint.  The following command would be typically used to upload a file:  

```Shell
curl --request PUT --header "X-Auth-Token: $token" https://api.gdc.cancer.gov/v0/submission/PROJECT/INTERNAL/files/6d45f2a0-8161-42e3-97e6-e058ac18f3f3 -d@$path_to_file

```

For more details on how to upload a `submittable_data_file` to a project see the [API Users Guide](API/Users_Guide/Submission/) and the [Data Transfer Tool Users Guide](Data_Transfer_Tool/Users_Guide/Data_Download_and_Upload/).  


## Metadata File Submission

[![GDC Data Model Metadata](images/GDC-Data-Model-Metadata.png)](images/GDC-Data-Model-Metadata.png "Click to see the full image.")


The `experiment_metadata` entity contains information about the experiment that was performed to produce each `read_group`. Unlike the previous two entities outlined, only information about the `experiment_metadata` file itself (SRA XML) is applied to the entity (indexed) and the `experiment_metadata` file is submitted in the same way that a BAM file would be submitted.

Submitting an [__Experiment Metadata__](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=experiment_metadata) entity requires:

* __`submitter_id`:__ A unique key to identify the `experiment_metadata` entity
* __`read_groups.submitter_id`:__ The unique key that was used for the `read_group` that links the `experiment_metadata` entity to the `read_group`
* __`data_category`:__ Broad categorization of the contents of the data file
* __`data_format`:__ Format of the data files. (must be "SRA XML")
* __`data_type`:__ Specific contents of the data file. (must be "Experiment Metadata")
* __`file_name`:__ The name (or part of a name) of a file (of any type)
* __`file_size`:__ The size of the data file (object) in bytes
* __`md5sum`:__ The 128-bit hash value expressed as a 32 digit hexadecimal number used as a file's digital fingerprint

```JSON
{
    "type": "experiment_metadata",
    "submitter_id": "Blood-001-aliquot_lane1_barcodeACGTAC_55-EXPERIMENT-1",
    "read_groups": {
        "submitter_id": "Primary_Tumor_RG_86-1"
    },
    "data_category": "Sequencing Data",
    "data_format": "SRA XML",
    "data_type": "Experiment Metadata",
    "file_name": "Experimental-data.xml",
    "file_size": 65498,
    "md5sum": "d79997e4de03b5a0311f0f2fe608c11d"
}
```

```TSV
type	submitter_id	cases.submitter_id	data_category	data_format	data_type	file_name	file_size	md5sum
experiment_metadata	Blood-00001-aliquot_lane1_barcodeACGTAC_55-EXPERIMENT-1	Primary_Tumor_RG_86-1	Sequencing Data	SRA XML	Experiment Metadata	Experimental-data.xml	65498	d79997e4de03b5a0311f0f2fe608c11d
```
## Annotation Submission

The GDC Data Portal supports the use of annotations for any submitted entity or file.  An annotation entity may include comments about why particular patients or samples are not present or why they may exhibit critical differences from others.  Annotations include information that cannot be submitted to the GDC through other existing nodes or properties.

If a submitter would like to create an annotation please contact the GDC Support Team (support@nci-gdc.datacommons.io).

## Deleting Submitted Entities

The GDC Data Submission Portal allows users to delete submitted entities from the project when the project is in an "OPEN" state. Files cannot be deleted while in the 'SUBMITTED' stated.  This section applies to entities that have been committed to the project. Entities that have not been committed can be removed from the project by choosing the `DISCARD` button.  Entities can also be deleted using the API. See the [API Submission Documentation](../../API/Users_Guide/Submission/#deleting-entities) for specific instructions.

__NOTE:__  Entities associated with files uploaded to the GDC object store cannot be deleted until the associated file has been deleted. Users must utilize the [GDC Data Transfer Tool](https://docs.gdc.cancer.gov/Data_Transfer_Tool/Users_Guide/Data_Download_and_Upload/#deleting-previously-uploaded-data) to delete these files first.

### Simple Deletion

If an entity was uploaded and has no related entities, it can be deleted from the [Browse](Browse_Data.md) tab. Once the entity to be deleted is selected, choose the `DELETE` button in the right panel under "ACTIONS".

---

[![GDC Delete Unassociated Case](images/GDC-Delete-Case-Unassociated.png)](images/GDC-Delete-Case-Unassociated.png "Click to see the full image.")

---

A message will then appear asking if you are sure about deleting the entity.  Choosing the `YES, DELETE` button will remove the entity from the project, whereas choosing the `NO, CANCEL` button will return the user to the previous screen.  

---

[![GDC Yes or No](images/GDC-Delete-Sure.png)](images/GDC-Delete-Sure.png "Click to see the full image.")

---

### Deletion with Dependents

If an entity has related entities, such as a `case` with multiple `samples` and `aliquots`, deletion takes one extra step.  

---

[![GDC Delete Associated Case](images/GDC-Delete-Case-Associated.png)](images/GDC-Delete-Case-Associated.png "Click to see the full image.")

---

Follow the 'Simple Deletion' method until the end. This action will appear in the [Transactions](Transactions.md) tab as "Delete" with a "FAILED" state.  

---

[![GDC Delete Failed](images/GDC-Failed-Transaction.png)](images/GDC-Failed-Transaction.png "Click to see the full image.")

---

Choose the failed transaction and the right panel will show the list of entities related to the entity that was going to be deleted.  

---

[![GDC Error Related](images/GDC-Error-Related.png)](images/GDC-Error-Related.png "Click to see the full image.")

---

Selecting the `DELETE ALL` button at the bottom of the list will delete all of the related entities, their descendants, and the original entity.


### Submitted Data File Deletion

The `submittable_data_file` that were uploaded erroneously are deleted separately from their associated entity using the GDC Data Transfer Tool. See the section on [Deleting Data Files](https://docs.gdc.cancer.gov/Data_Transfer_Tool/Users_Guide/Data_Download_and_Upload/#deleting-previously-uploaded-data) in the Data Transfer Tool users guide for specific instructions.  

## Updating Uploaded Entities

Before harmonization occurs, entities can be modified to update, add, or delete information. Below, these methods are outlined.

### Updating or Adding Fields

Updated or additional fields can applied to entities by reuploading them through the submission portal or API. See below for an example of a case upload with a `primary_site` field being added and a `disease_type` field being updated.

Existing Entity:

```JSON
{
"type":"case",
"submitter_id":"GDC-INTERNAL-000043",
"projects":{
  "code":"INTERNAL"
},
"disease_type": "Neuroblastoma"
}
```
The following entity would be submitted to update the existing one:

```JSON
{
"type":"case",
"submitter_id":"GDC-INTERNAL-000043",
"projects":{
  "code":"INTERNAL"
},
"disease_type": "Germ Cell Neoplasms",
"primary_site": "Pancreas"
}
```
__Guidelines:__

* The newly uploaded entity must contain the `submitter_id` of the existing entity so that the system updates the correct one.
* All newly updated entities will be validated by the GDC Dictionary.  All required fields must be present in the newly updated entity.
* Fields that are not required do not need to be re-uploaded and will remain unchanged in the entity unless they are updated.

### Deleting Optional Fields

It may be necessary to delete fields from uploaded entities. This can be performed through the API and can only be applied to optional fields. It also requires the UUID of the entity, which can be retrieved from the submission portal or using a GraphQL query.

In the example below, the `primary_site` and `disease_type` fields are removed from a `case` entity:

```Shell
curl --header "X-Auth-Token: $token_string" --request DELETE  --header "Content-Type: application/json" "https://api.gdc.cancer.gov/v0/submission/EXAMPLE/PROJECT/entities/7aab7578-34ff-5651-89bb-57aefdc4c4f8?fields=primary_site,disease_type"
```

```Before
{
"type":"case",
"submitter_id":"GDC-INTERNAL-000043",
"projects":{
  "code":"INTERNAL"
},
"disease_type": "Germ Cell Neoplasms",
"primary_site": "Pancreas"
}
```
```After
{
"type":"case",
"submitter_id":"GDC-INTERNAL-000043",
"projects":{
  "code":"INTERNAL"
}
}
```

## Strategies for Submitting in Bulk

Each submission in the previous sections was broken down by component to demonstrate the GDC Data Model structure. However, the submission of multiple entities at once is supported and encouraged. Here two strategies for submitting data in an efficient manner are discussed.   

### Registering a BAM File: One Step

Registering a BAM file (or any other type) can be performed in one step by including all of the entities, from `case` to `submitted_aligned_reads`, in one file.  See the example below:

```JSON
[{
    "type": "case",
    "submitter_id": "PROJECT-INTERNAL-000055",
    "projects": {
        "code": "INTERNAL"
    }
},
{
    "type": "sample",
    "cases": {
        "submitter_id": "PROJECT-INTERNAL-000055"
    },
    "sample_type": "Blood Derived Normal",
    "submitter_id": "Blood-00001_55"
},
{
    "type": "portion",
    "submitter_id": "Blood-portion-000055",
    "samples": {
        "submitter_id": "Blood-00001_55"
    }
},
{
    "type": "analyte",
    "portions": {
        "submitter_id": "Blood-portion-000055"
    },
    "analyte_type": "DNA",
    "submitter_id": "Blood-analyte-000055"
},
{
    "type": "aliquot",
    "submitter_id": "Blood-00021-aliquot55",
    "analytes": {
        "submitter_id": "Blood-analyte-000055"
    }
},
{
    "type": "read_group",
    "submitter_id": "Blood-00001-aliquot_lane1_barcodeACGTAC_55",
    "experiment_name": "Resequencing",
    "is_paired_end": true,
    "library_name": "Solexa-34688",
    "library_strategy": "WXS",
    "platform": "Illumina",
    "read_group_name": "205DD.3-2",
    "read_length": 75,
    "sequencing_center": "BI",
    "aliquots":
        {
            "submitter_id": "Blood-00021-aliquot55"
        }    
},
{
    "type": "submitted_aligned_reads",
    "submitter_id": "Blood-00001-aliquot_lane1_barcodeACGTAC_55.bam",
    "data_category": "Raw Sequencing Data",
    "data_format": "BAM",
    "data_type": "Aligned Reads",
    "experimental_strategy": "WGS",
    "file_name": "test.bam",
    "file_size": 38,
    "md5sum": "aa6e82d11ccd8452f813a15a6d84faf1",
    "read_groups": [
        {
            "submitter_id": "Blood-00001-aliquot_lane1_barcodeACGTAC_55"
        }
    ]
}]
```

All of the entities are placed into a JSON list object:

`[{"type": "case","submitter_id": "PROJECT-INTERNAL-000055","projects": {"code": "INTERNAL"}}}, entity-2, entity-3]`

The entities need not be in any particular order as they are validated together.

__Note:__ Tab-delimited format is not recommended for 'one-step' submissions due to an inability of the format to accommodate multiple 'types' in one row.  

### Submitting Numerous Cases

The GDC understands that submitters will have projects that comprise more entities than would be reasonable to individually parse into JSON formatted files. Additionally, many investigators store large amounts of data in a tab-delimited format (TSV).  For instances like this, we recommend parsing all entities of the same type into separate TSVs and submitting them on a type-basis.  

For example, a user may want to submit 100 Cases associated with 100 `samples`, 100 `portions`, 100 `analytes`, 100 `aliquots`, and 100 `read_groups`. Constructing and submitting 100 JSON files would be tedious and difficult to organize. Submitting one `case` TSV containing 100 `cases`, one `sample` TSV containing 100 `samples`, and the rest would require six TSVs and can be formatted in programs such as Microsoft Excel or Google Spreadsheets.  

See the following example TSV files:

* [Cases.tsv](Cases.tsv)
* [Samples.tsv](Samples.tsv)
* [Portions.tsv](Portions.tsv)
* [Analytes.tsv](Analytes.tsv)
* [Aliquots.tsv](Aliquots.tsv)
* [Read-Groups.tsv](Readgroups.tsv)

### Download Previously Uploaded Metadata Files

The [transaction](Transactions.md) page lists all previous transactions in the project. The user can download metadata files uploaded to the GDC workspace in the details section of the screen by selecting one transaction and scrolling to the "DOCUMENTS" section.


[![Transaction Original Files](images/GDC_Submission_Transactions_Original_Files_2.png)](images/GDC_Submission_Transactions_Original_Files_2.png "Click to see the full image.")

### Download Previously Uploaded Data Files

The only supported method to download data files previously uploaded to the GDC Submission Portal that have not been release yet is to use the API or the [Data Transfer Tool](https://docs.gdc.cancer.gov/Data_Transfer_Tool/Users_Guide/Getting_Started/). To retrieve data previous upload to the submission portal you will need to retrieve the data file's UUID.  The UUIDs for submitted data files are located in the submission portal under the file's Summary section.    

[![Submission Portal Summary View](images/gdc-submission__image2_submission_UUID.png)](images/gdc-submission__image2_submission_UUID.png "Click to see the full image.")  
The UUIDs can also be found in the manifest file accessible from the manifest button located on the file's  Summary page.   

Once the UUID(s) have been retrieved the download process is the same as it is for data on our main portal.  If the process is unfamiliar please refer to the [Downloading data using GDC file UUIDs](https://docs.gdc.cancer.gov/Data_Transfer_Tool/Users_Guide/Data_Download_and_Upload/#downloading-data-using-gdc-file-uuids) for downloading instructions.

 __Note:__ When submittable data files are uploaded through the Data Transfer Tool they are not displayed as transactions.


# Submit Your Workspace Data to the GDC

## Overview

The GDC Data Submission process is detailed on the [Data Submission Processes and Tools](https://gdc.cancer.gov/submit-data/data-submission-processes-and-tools) section of the GDC Website.

## Review and Submit

When data is uploaded to the project workspace (see previous section: [Upload Data](Data_Upload_UG.md)) the submitter should review the data to ensure that it is ready for processing by the GDC [Harmonization Process](https://gdc.cancer.gov/submit-data/gdc-data-harmonization). Setting the project to the 'REVIEW' state will lock the project and prevent users from uploading additional data. During this period, the submitter can browse the data in the Submission Portal or download it.

If the project is ready for processing, the submitter will submit data to the GDC. If the project is not ready for processing, the project can be re-opened. Then the submitter will be able to upload more data to the project workspace.

The GDC requests that users submit their data to the GDC within six months from the first upload to the project workspace.

### Review

To review and submit data to the GDC, the user must have release privileges. The user will be able to view the section below on the dashboard. The "REVIEW" button is available only if the project is in "OPEN" state.

[![GDC Submission Review Tab](images/GDC_Submission_Submit_Release_Review_tab_2_v2.png)](images/GDC_Submission_Submit_Release_Review_tab_2_v2.png "Click to see the full image.")

Reviewing the project will prevent other users from uploading data to the project. Once the review is complete, the user can submit data to the GDC.

Once the user clicks on "REVIEW", the project state will change to "REVIEW":

[![GDC Submission Review State](images/GDC_Submission_Submit_Release_Project_State_Review_2.png)](images/GDC_Submission_Submit_Release_Project_State_Review_2.png "Click to see the full image.")



### Submit to the GDC

The _'Request Submission'_ button is available only if the project is in "REVIEW" state. At this point, the user can decide whether to re-open the project to upload more data or to request submission of the data to the GDC. When the project is in "REVIEW" the following panel appears on the dashboard:



[![GDC Submission Submit Tab](images/GDC_Submission_Submit_Release_Submit_tab_2_v3.png)](images/GDC_Submission_Submit_Release_Submit_tab_2_v3.png "Click to see the full image.")

Once the user submits data to the GDC, they __cannot upload additional data until the harmonization process is complete__.

[![GDC Submission Submission Tab](images/GDC_SUBMIT_TO_GDC_v2.png)](images/GDC_SUBMIT_TO_GDC_v2.png "Click to see the full image.")

When the user clicks on the action _'Request Submission'_ on the dashboard, the following submission popup is displayed:

[![GDC Submission Submit Popup](images/GDC_Submission_Submit_Release_Submit_Popup_v2.png)](images/GDC_Submission_Submit_Release_Submit_Popup_v2.png "Click to see the full image.")


After the user clicks on "Submit Validated Data to the GDC", the project state becomes "Submission Requested":
[![GDC Submission Project State](images/GDC_Submission_Submit_Release_Project_State.png)](images/GDC_Submission_Submit_Release_Project_State_v2.png "Click to see the full image.")


# Data

## Overview

The Data section of the GDC Data Submission Portal provides access to Experimental Data, it contains all molecular data (read groups), and will likely host images and pathology reports at a later stage.

[![GDC Submission Read Groups](images/GDC_Submission_Data.png)](images/GDC_Submission_Data.png "Click to see the full image.")

## Read Groups List View

The read groups list view displays the following information:

|Column|Description|
| --- | --- |
| Submitter ID | Submitter ID of the read group |
| Type | TO BE COMPLETED |
| Case | Case associated to the read group |
| Library Strategy | TO BE COMPLETED |
| Submission Status | TO BE COMPLETED |
| ## Files | TO BE COMPLETED |
| Scientific Pre-Alignement QC Report | TO BE COMPLETED |
| Last Updated | TO BE COMPLETED |

At the top left section of the screen, the user can download data about the selected read group or all read groups in the project.

## Read Group Details

Clicking on a read group will open the details panel. Data in this panel is broken down into multiple sections.

[![GDC Submission Read Group Details](images/GDC_Submission_Data_Read_Group_Details.png)](images/GDC_Submission_Data_Read_Group_Details.png "Click to see the full image.")

Navigation between these sections can be done either by scrolling down or by clicking on the section icon on the left side of the details panel.

### Details

Provides details about the read group itself, such as its UUID, status, project and creation date.

[![GDC Submission Read Group Details](images/GDC_Submission_Data_Read_Group_Details_Details.png)](images/GDC_Submission_Data_Read_Group_Details_Details.png "Click to see the full image.")

### Hierarchy

List entities (cases, clinical, biospecimen, annotations) attached to a read-group in a tree-like view. Clicking on an entity redirects the user to its corresponding details page, easing navigation between entities.

[![GDC Submission Read Group Details Hierarchy](images/GDC_Submission_Data_Read_Group_Details_Hierarchy.png)](images/GDC_Submission_Data_Read_Group_Details_Hierarchy.png "Click to see the full image.")

### Files

Lists files submitted for the read group.

[![GDC Submission Read Group Details Files](images/GDC_Submission_Data_Read_Group_Details_Files.png)](images/GDC_Submission_Data_Read_Group_Details_Files.png "Click to see the full image.")

### Download

List files available for download. It could be Metadata, a Manifest to be used by the GDC Data Transfer Tool or a QC Report.

[![GDC Submission Read Group Details Download](images/GDC_Submission_Data_Read_Group_Details_Download.png)](images/GDC_Submission_Data_Read_Group_Details_Download.png "Click to see the full image.")

### Transactions

Lists all the transactions associated to this read group. Clicking on an transaction ID will redirect the user to the transaction details page.

[![GDC Submission Read Group Details Transactions](images/GDC_Submission_Data_Read_Group_Details_Transactions.png)](images/GDC_Submission_Data_Read_Group_Details_Transactions.png "Click to see the full image.")


# Upload Data to Your Workspace

## Overview

The GDC Data Submission Portal allows users to upload and validate Clinical, Biospecimen and file metadata using the portal, register the Submittable Data Files, and submit the data files using the GDC Data Transfer Tool or API. The following diagram provides an overview of the process:

[![GDC Data Submission Portal Workflow Upload](images/gdc-submission-portal-data-upload-workflow.png)](images/gdc-submission-portal-data-upload-workflow.png "Click to see the full image.")


## Supported File Formats

### Metadata Files

The GDC API accepts project metadata in JSON and TSV formats for the purpose of creating entities in the GDC Data Model. This includes Clinical and Biospecimen metadata such as tumor classification, age at diagnosis, sample type, and details about the available data files. Upon successful data submission and project release, this metadata is stored in the GDC and becomes available to access through queries by data users with the GDC Data Portal and the GDC API. Before submitting metadata files:

* Review the [GDC Data Model](https://gdc.cancer.gov/developers/gdc-data-model) and the [GDC Data Dictionary](/Data_Dictionary/) to understand accepted metadata elements.
* Download metadata submission templates from the [GDC Data Dictionary](/Data_Dictionary/).




### Data Relationships

All submitted entities must be linked to existing entities in the [GDC Data Model](../../Data/Data_Model/GDC_Data_Model.md). In order to create the relationship, the user must include a reference to the related entity using its Submitter ID or Universally Unique Identifier (UUID).

For example, a Demographic entity describes a Case entity. The user is required to provide the case Submitter ID or UUID in the Demographic metadata.

For a new submission project, cases are the first entities to be created, and linked to the project. Other entities are then created according to the [GDC Data Model](../../Data/Data_Model/GDC_Data_Model.md).

### Download Previously Uploaded Metadata Files

The [transaction](Transactions.md) page lists all previous transactions in the project. The user can download metadata files uploaded to the GDC workspace in the details section of the screen by selecting one transaction and scrolling to the "DOCUMENTS" section.


[![Transaction Original Files](images/GDC_Submission_Transactions_Original_Files_2.png)](images/GDC_Submission_Transactions_Original_Files_2.png "Click to see the full image.")

### Download Previously Uploaded Files

The only supported method to download case data files previously uploaded to the GDC Submission Portal is to use the data transfer tool. For more information on downloading data using the data transfer tool please refer to [Downloading data using GDC file UUIDs](https://docs.gdc.cancer.gov/Data_Transfer_Tool/Users_Guide/Data_Download_and_Upload/#downloading-data-using-gdc-file-uuids) BFrom the browse tab link located on the project's home page navigate to the "Submittable Data Files" section and click on the submitter id associated with the file.  The UUID associated file will appear under the "SUMMARY" section in the file's information pain located on the right site of the page. [Submission Portal Summary View](images/gdc-submission-portal_image_summary_submission_UUID.png)  

For more information on downloading data using the data transfer tool with a UUID please refer to [Downloading data using GDC file UUIDs](https://docs.gdc.cancer.gov/Data_Transfer_Tool/Users_Guide/Data_Download_and_Upload/#downloading-data-using-gdc-file-uuids)

__Note:__ When submittable data files are uploaded through the Data Transfer Tool they are not displayed as transactions.

## Deleting Previously Uploaded Files

The GDC Data Submission Portal does not support the deletion of entities at this time. This can be performed using the API. See the [API Submission Documentation](../../API/Users_Guide/Submission/#deleting-entities) for specific instructions.


# Data Submission Overview

## Overview
This section will walk users through two parts of the submission process. The first portion will be the steps taken by the users to go through the submission process from start to finish. The second portion will describe the lifecycle of a project and a file throughout the data submission process.

## GDC Data Submission Workflow

The diagram below illustrates the process from uploading through releasing data in the GDC Data Submission Portal. To review the steps needed before beginning submission see [Before Submitting Data to the GDC Portal](https://docs.gdc.cancer.gov/Data_Submission_Portal/Users_Guide/Checklist/)

[![GDC Data Submission Portal Workflow Upload](images/GDC_Data_Submission_Workflow-updated_20190301.jpg)](images/GDC_Data_Submission_Workflow-updated_20190301.jpg "Click to see the full image.")

### Review GDC Dictionary and GDC Data Model - Submitter Activity

It is suggested that all submitters review the [GDC Dictionary](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/) and [GDC Data Model](https://gdc.cancer.gov/developers/gdc-data-model/gdc-data-model-components). It is beneficial for submitters to know which nodes will need metadata submission, how these nodes relate to each other, and what information is required for each node in the model.

### Download Templates - Submitter Activity

After determining the required nodes for the submission, go to each node page in the [GDC Dictionary](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/). There will be a "Download Template" drop down list. Select the file format, either TSV or JSON, and download the template for the node. If [numerous entries](Data_Submission_Walkthrough.md#submitting-numerous-cases) are being submitted all at one time, it is suggested that the user uses a TSV template. At this point, it is suggested to go through the template and remove fields that will not be populated by the metadata submission, but make sure to complete all fields that are required for the node. For more information about the Data Dictionary, please visit [here](../../../Data_Dictionary/).

[`See GDC Data Dictionary here.`](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/)

### Upload Case Information Including dbGaP Submitted Subject IDs - Submitter Activity

After registering the study in [dbGaP](https://gdc.cancer.gov/submit-data/obtaining-access-submit-data), the first node to be created in the data model is the [`case` node](Data_Submission_Walkthrough.md#case-submission). The `case` node is important as it will contain a unique `submitter_id` that is registered in dbGaP under a particular project. This will connect the two databases, dbGaP and GDC, and allows for access to be granted to a controlled data set based on the study and its cases.

To [submit the `case`](Data_Submission_Walkthrough.md#uploading-the-case-submission-file) nodes, a user must be able to [login](Data_Submission_Process.md#authentication) and access the [GDC Submission Portal](https://portal.gdc.cancer.gov/submission/) for their respective project. Metadata for all nodes are uploaded via the [API](https://docs.gdc.cancer.gov/API/Users_Guide/Submission/#creating-and-updating-entities) or through the [Submission Portal](Data_Submission_Walkthrough.md#upload-using-the-gdc-data-submission-portal).

[`See case example here.`](Data_Submission_Walkthrough.md#case-submission)

[`See metadata upload example here.`](Data_Submission_Walkthrough.md#upload-using-the-gdc-data-submission-portal)

### Upload Clinical and Biospecimen Data - Submitter Activity

With the creation of `case` nodes, other nodes in the [data model](https://gdc.cancer.gov/developers/gdc-data-model/gdc-data-model-components) can be [uploaded](Data_Submission_Walkthrough.md#upload-using-the-gdc-data-submission-portal). This includes the [Clinical](Data_Submission_Walkthrough.md#clinical-data-submission) and [Biospecimen](Data_Submission_Walkthrough.md#biospecimen-submission) nodes, with examples for each that can be found in the [Data Upload Walkthrough](Data_Submission_Walkthrough.md).

[`See clinical example here.`](Data_Submission_Walkthrough.md#clinical-data-submission)

[`See biospecimen example here.`](Data_Submission_Walkthrough.md#biospecimen-submission)

[`See metadata upload example here.`](Data_Submission_Walkthrough.md#upload-using-the-gdc-data-submission-portal)

### Register Data Files - Submitter Activity

Registering data files is necessary before they can be uploaded. This allows the GDC to later validate the uploads against the user-supplied md5sum and file size. The [submission](Data_Submission_Walkthrough.md#experiment-data-submission) of these files can range from clinical and biospecimen supplements to `submitted_aligned_reads` and `submitted_unaligned_reads`.

[`See experiment data example here.`](Data_Submission_Walkthrough.md#experiment-data-submission)

### Upload Data Using Data Transfer Tool - Submitter Activity

Before uploading the submittable data files to the GDC, a user will need to determine if the correct nodes have been created and the information within them are correct. This is accomplished using the [Browse](Data_Submission_Process.md#browse) page in the [Data Submission Portal](https://portal.gdc.cancer.gov/submission). Here you can find the metadata and file_state, which must have progressed to `registered` for an associated file to be uploaded.  You can find more about the file life cycle [here](#file-lifecycle).

Once the submitter has verified that the submittable data files have been registered, the user can obtain the submission manifest file that is found on the [Project Overview](Data_Submission_Process.md#project-overview) page. From this point the submission process is described in the ["Uploading the Submittable Data File to the GDC"](Data_Submission_Walkthrough.md#uploading-the-submittable-data-file-to-the-gdc) section.

For strategies on data upload, further documentation for the GDC Data Submission process is detailed on the [Data Submission Processes and Tools](https://gdc.cancer.gov/submit-data/data-submission-processes-and-tools) section of the GDC Website.

[`See submittable data file upload example here.`](Data_Submission_Walkthrough.md#uploading-the-submittable-data-file-to-the-gdc)

### Verify Accuracy and Completeness of Project Data (Project QC) - Submitter Activity

The submitter is responsible for reviewing the data uploaded to the project workspace and ensuring there are no critical QC errors, see [Data Submission Walkthrough](Data_Submission_Walkthrough.md), and ensuring that it is ready for processing by the GDC [Harmonization Process](https://gdc.cancer.gov/submit-data/gdc-data-harmonization). A user should be able to go through the [Pre-Harmonization Checklist](Data_Submission_Process.md#pre-harmonization-checklist), and verify that their submission meets these criteria.

[`See pre-harmonization checklist here.`](Data_Submission_Process.md#pre-harmonization-checklist)

### Request Data Harmonization - Submitter Activity

When the project is complete and ready for processing, the submitter will [request harmonization](Data_Submission_Process.md#submit-your-workspace-data-to-the-gdc). If the project is not ready for processing, the project can be re-opened and the submitter will be able to upload more data to the project workspace.

[`See harmonization request example here.`](Data_Submission_Process.md#submit-your-workspace-data-to-the-gdc)

> __NOTE:__ The GDC requests that users submit their data to the GDC within six months from the first upload of data to the project workspace.

### GDC Review/QC Submitted Data - GDC Activity

The Bioinformatics Team at the GDC runs the Quality Control pipeline on the submitted data. This pipeline mirrors the [Pre-Harmonization Checklist](Data_Submission_Process.md#pre-harmonization-checklist) and will determine if the submission is complete and is ready for the Harmonization pipeline. If the submission does contain problems, the GDC will contact the user to "Re-Open" the project and fix the errors in their submission.

Once the review is complete, all validated nodes will be changed to state "submitted".  At this point users can submit more files to a project, but they will be considered as a different batch for harmonization.

### GDC Harmonize Data - GDC Activity

After the submission passes the GDC Quality Control pipeline, it will be queued for the [GDC Harmonization pipeline](https://gdc.cancer.gov/about-data/gdc-data-harmonization).

### Submitter Review/QC of Harmonized Data - Submitter Activity

After the data is processed in the Harmonization pipeline, the GDC asks submitters to [verify the quality](https://portal.gdc.cancer.gov/submission/login?next=%2Fsubmission%2F) of their harmonized data.  It is the user's responsibility to notify the GDC of any errors in their harmonized data sets. The GDC will then work with the user to correct the issue and rerun the Harmonization pipeline if needed.

### Release Data Within Six Months - Submitter Activity

Project release occurs after the data has been harmonized, and allows users to access this data with the [GDC Data Portal](https://portal.gdc.cancer.gov/) and other [GDC Data Access Tools](https://gdc.cancer.gov/access-data/data-access-processes-and-tools). The GDC will release data according to [GDC Data Sharing Policies](https://gdc.cancer.gov/submit-data/data-submission-policies). Data must be released within six months after GDC data processing has been completed, or the submitter may request earlier release.

[`See release example here.`](Data_Submission_Process.md#release)

>__Note__: Released cases and/or files can be redacted from the GDC. For more information, visit the [GDC Policies page (under GDC Data Sharing Policies)](https://gdc.cancer.gov/submit-data/data-submission-policies).

### GDC Releases Data - GDC Activity

GDC data releases are not continuous, but instead are released in discrete data updates. Once harmonized data is approved and release request is approved, data will be available in an upcoming GDC Data Release.

## Project and File Lifecycles

### Project Lifecycle
The diagram of the project lifecycle below demonstrates the transition of a project through the various states. Initially the project is open for data upload and validation. Any changes to the data must be made while the project status is open. When the data is uploaded and ready for review, the submitter changes the project state to review. During the review state, the project is locked and additional data cannot be uploaded. If data changes are needed during the review period, the project has to be re-opened.

The process of Harmonization does not occur immediately after submitted files are uploaded. After the submission is complete and all the necessary data and files have been uploaded, the user submits the data to the GDC for processing through the [GDC Data Harmonization Pipelines](https://gdc.cancer.gov/submit-data/gdc-data-harmonization) and the project state changes to submitted. When the data has been processed, the project state changes back to open for new data to be submitted to the project and the submitter can review the processed data. After review of the processed data, the submitter can then release the harmonized data to the [GDC Data Portal](https://portal.gdc.cancer.gov/) and other [GDC Data Access Tools](https://gdc.cancer.gov/access-data/data-access-processes-and-tools) according to [GDC Data Sharing Policies](https://gdc.cancer.gov/submit-data/data-submission-policies).  

[![GDC Data Submission Portal Workflow](images/Submission.png)](images/Submission.png "Click to see the full image.")

### File Lifecycle

This section describes states pertaining to submittable data files throughout the data submission process. A submittable data file could contain data such as genomic sequences (such as a BAM or FASTQ) or pathology slide images. The file lifecycle starts when a submitter uploads metadata for a file to the [GDC Data Submission Portal](https://portal.gdc.cancer.gov/submission/). This metadata file registers a description of the file as an entity on the GDC, the status for this is known as "state" and is represented by __purple__ cirlces. The submitter can then use the [GDC Data Transfer Tool](https://gdc.cancer.gov/access-data/gdc-data-transfer-tool) to upload the actual file, which is represeneted by __red__ circles. Throughout the lifecycle, the file and its associated entity transition through various states from when they are initially registered through file submission and processing. The diagram below details these state transitions.   

[![GDC Data Submission Portal File Status](images/gdc-submission-portal-file-state-vs-state.png)](images/gdc-submission-portal-file-state-vs-state.png "Click to see the full image.")


# Homepage

## Overview

After [authentication](Authentication.md), users are redirected to a homepage. The homepage acts as the entry point for GDC data submission and provides submitters with access to a list of authorized projects, reports, and transactions.

[![GDC Submitter Home Page](images/GDC-HomePage-Submit_v2.png)](images/GDC-HomePage-Submit_v2.png "Click to see the full image.")

Content on the homepage varies based on the user profile (e.g. submitter, program office).

## Reports

Project Summary Reports can be downloaded at the Submission Portal Homepage at three different levels: CASE OVERVIEW, ALIQUOT OVERVIEW, and DATA VALIDATION.  Each report is generated in tab-delimited format in which each row represents an active project.  

* __CASE OVERVIEW:__ This report describes the number of cases with associated biospecimen data, clinical data, or submittable data files (broken down by type) for each project.
* __ALIQUOT OVERVIEW:__ This report describes the number of aliquots in a project with associated data files. Aliquot numbers are broken down by tissue sample type.
* __DATA VALIDATION:__ This report categorizes all submittable data files associated with a project by their file status.

## Projects

The projects section in the homepage lists the projects that the user has access to along with basic information about each. The button used to [release](Submit_Data.md) each project is located on this screen. For users with access to a large number of projects, this table can be filtered with the 'FILTER PROJECTS' field.  

Selecting a project ID will direct the user to the project's [Dashboard](Dashboard.md).

## Release data

Data Release can be performed from the homepage. See the section about [Data Release](Release_Data.md) for details.


# Clinical Data

## Overview

The _"Clinical"_ section lists all the clinical data entities available in a project. Clicking on a particular clinical entity will open the details panel providing more details about the selected entity.

[![GDC Submission Cases Default View](images/GDC_Submission_Clinical_Default.png)](images/GDC_Submission_Clinical_Default.png "Click to see the full image.")

## Clinical Data Filters

Clinical data entities can be accessed from the menu through multiple filters automatically created based on the [GDC Data Dictionary](../../Dictionary/index.md).

The number of entities corresponding to those filters is displayed on the right side of the filter.

## Clinical List View

The clinical list view displays the following information:

|Column|Description|
| --- | --- |
| Submitter ID | Submitter ID of the case |
| Type | Type of Clinical Data (corresponds to the selected filter)|
| Case ID | Case ID this entity is attached to |
| Status | Status of the entity |
| Last Updated | Last time the entity was updated. |

On the top left section of the screen, the user can download a tar.gz file containing all entities on the project. The user has the choice between TSV or JSON format.

## Clinical Details

Clicking on a case will open the details panel. Data in this panel is broken down into multiple sections.

Navigation between these sections can be done either by scrolling down or by clicking on the section icon on the left side of the details panel.

### Summary

Provides details about the entity itself, such as its UUID, status, project and creation date.

|Value|Description|
| --- | --- |
| Type | Entity type (Case in this situation)  |
| UUID | Entity's [UUID](https://en.wikipedia.org/wiki/Universally_unique_identifier) |
| Project ID | Project ID associated with the Entity |
| Submitter ID | Submitter ID associated with the Entity |
| Created Datetime | Date and time the Entity was created |
| State | State of the Entity |
| Updated Datetime | Date and time the Entity was last updated |

### Details

Table providing details about the entity's content.

This table contains the following columns.

|Column|Description|
| --- | --- |
| Ethnicity | Category of the Entity (Clinical, Biospecimen, Experiment Data)  |
| Type | Type of Entity (based on Data Dictionary)  |
| Count | Number of occurences of an Entity of this type |

Clicking on the count will open a list page listing those entities.


### Transactions

Lists the 10 most recent transactions associated with this entity ordered by date. Clicking on a transaction will open it in the list page.


# Dictionary Viewer

## Overview

The Data Dictionary viewer provides access to GDC-wide dictionaries as well as specific dictionaries that can be applied to a specific project.

More details about dictionaries can be found in a dedicated section of the documentation [=>LINK TO GDC DICTIONARY DOCUMENTATION].

The Data Dictionary viewer is automatically updated from the GDC Data Dictionary and can considered an up-to-date view of the Data Dictionary.

[![GDC Submission Dictionary Viewer](images/GDC_Submission_Dictionary_Viewer.png)](images/GDC_Submission_Dictionary_Viewer.png "Click to see the full image.")

## Available Dictionaries

The exact list of available dictionaries will vary from one project to another

## Data Dictionary View

After selecting a Data Dictionary in the left side of the screen, the user can view the content of this Data Dictionary. The following columns are available:

* _Property_: Name of the field to be used during submission.
* _Description_: Description of the field.
* _Acceptable Types or Values_: Details what is expected by the system and rules for data validation. This field can take different values such as string, number, enum (list of values).
* _Required?_: Indicates whether this field is required for the submission to be valid.

## Download Template

After selecting a Data Dictionary, its template can be downloaded from the system in TSV format.


# Browse Data

## Overview

The _"Browse"_ menu provides access to all of a project's content. Most content is driven by the GDC Data Dictionary and the interface is dynamically generated to accommodate the content.

Please refer to the [GDC Data Dictionary Viewer](../../Data_Dictionary/viewer.md) for specific details about dictionary-generated fields, columns, and filters.

[![GDC Submission Cases Default View](images/GDC_Submission_Cases_Default_2.png)](images/GDC_Submission_Cases_Default_2.png "Click to see the full image.")

## Main Interface Elements

### Filters

A wide set of filters are available for the user to select the type of entity to be displayed. These filters are dynamically created based on the [GDC Data Dictionary](../../Data_Dictionary/index.md).

Current filters are:

|Filter|Description|
| --- | --- |
| __Cases__ | Display all `cases` associated with the project. |
| __Clinical Entities__ | Display all Clinical data uploaded to the project workspace. This is divided into subgroups including `demographics`, `diagnoses`, `exposures`, `family histories`, and `treatments`. |
| __Biospecimen Data__ | Display all Biospecimen data uploaded to the project workspace. This is divided into subgroups including `samples`, `portions`, `analytes`, `aliquots`, and `read groups`. |
| __Submittable Data Files__ | Displays all data files that have been registered with the project. This includes files that have been uploaded and those that have been registered but not uploaded yet. This category is divided into groups by file type. |
| __Annotations__ | Lists all annotations associated with the project. An annotation provides an explanatory comment associated with data in the project. |


### List View

The list view is a paginated list of all entities corresponding to the selected filter.

On the top-right section of the screen, the user can download data about all entities associated with the selected filter.

* For the case filter, it will download all Clinical data or all Metadata 
* For all other filters, it will download the corresponding metadata (e.g., for the `demographic` filter, it will download all `demographic` data).

[![GDC Submission Case Summary Download](images/GDC_Submission_Cases_Summary_Download_4.png)](images/GDC_Submission_Cases_Summary_Download_2.png "Click to see the full image.")

### Details Panel

Clicking on an entity will open the details panel. Data in this panel is broken down into multiple sections depending on the entity type. The main sections are:

* __Actions__: Actions that can be performed relating the entity. This includes downloading the metadata (JSON or TSV) or submittable data file pertaining to the entity and deleting the entity. See the [Deleting Entities](https://docs.gdc.cancer.gov/Data_Submission_Portal/Users_Guide/Data_Upload_UG/#deleting-submitted-entities) guide for more information.  
* __Summary__: A list of IDs and system properties associated with the entity.
* __Details__: Properties of the entity (not associated with cases).
* __Hierarchy__ or __Related Entities__: A list of associated entities.
* __Annotations__: A list of annotations associated with the entity.
* __Transactions__: A list of previous transactions that affect the entity.

[![GDC Submission Case Details](images/GDC_Submission_Cases_Details_2.png)](images/GDC_Submission_Cases_Details_2.png "Click to see the full image.")

The sections listed above can be navigated either by scrolling down or by clicking on the section icon on the left side of the details panel.

#### Related Entities

The Related Entities table lists all entities, grouped by type, related to the selected `case`. This section is only available at the `case` level.

[![GDC Submission Cases Related Entities](images/GDC_Submission_Cases_Summary_Related_Entities_2.png)](images/GDC_Submission_Cases_Summary_Related_Entities_2.png "Click to see the full image.")


This table contains the following columns:

* __Category__: category of the entity (Clinical, Biospecimen, submittable data file).
* __Type__: type of entity (based on Data Dictionary).
* __Count:__ number of occurrences of an entity associated with the `case`. Clicking on the count will open a window listing those entities within the Browse page.

#### Hierarchy

The hierarchy section is available for entities at any level (e.g., Clinical, Biospecimen, etc.), except for `case`. The user can use the hierarchy section to navigate through entities.

The hierarchy shows:

* The `case` associated with the entity.
* The __direct__ parents of the entity.
* The __direct__ children of the entity.


[![GDC Submission Cases Details Hierarchy](images/GDC_Submission_Cases_Summary_Hierarchy_2.png)](images/GDC_Submission_Cases_Summary_Hierarchy_2.png "Click to see the full image.")


# Before Submitting Data to the GDC Portal

## Overview
The National Cancer Institute (NCI) Genomic Data Commons (GDC) Data Submission Portal User's Guide is the companion documentation for the [GDC Data Submission Portal](https://gdc.cancer.gov/submit-data/gdc-data-submission-portal) and provides detailed information and instructions for its use.

## Steps to Submit Data to the GDC
The following tasks are required to submit data to the [GDC Data Submission Portal](https://portal.gdc.cancer.gov/).

1.  Complete the GDC Data [Submission Request Form](https://gdc.cancer.gov/data-submission-request-form). After submission, the reqest will be reviewed by the GDC Data Submission Review Committee. During this time, create an [eRA Commons account](https://era.nih.gov/registration_accounts.cfm) if you do not already have one.

2.  If the study is approved, contact a [Genomic Program Administrator (GPA)](https://sharing.nih.gov/contacts-and-help#gds_support) to register the approved study in [dbGaP](https://www.ncbi.nlm.nih.gov/sra/docs/submitdbgap).  This includes registering the project as a GDC Trusted Partner study, registering cases, and adding authorized data submitters. For more information, see the [Data Submission Process](https://gdc.cancer.gov/submit-data/data-submission-processes-and-tools).

3.  Contact GDC User Services to create a submission project.  The User Services team will require a project ID, which is a two-part identifier, where the first portion is the __Program__ followed by a hyphen (__-__) and the second portion is the __Project__.  This must be alphanumeric and all caps only.  An example would be `TCGA-BRCA`.  You must also create a project name, which can be longer and has fewer requirements on length or character usage.  An example would be `Breast Invasive Carcinoma`.

## Key Features
The GDC Data Submission Portal is a platform that allows researchers to submit and release data to the GDC. The key features of the GDC Data Submission Portal are:

* __Upload and Validate Data__: Project data can be uploaded to the GDC project workspace. The GDC will validate the data against the [GDC Data Dictionary](../../Data_Dictionary/viewer.md).
* __Browse Data__: Data that has been uploaded to the project workspace can be browsed to ensure that the project is ready for processing.
* __Download Data__: Data that has been uploaded into the project workspace can be downloaded for review or update by using the [API](https://docs.gdc.cancer.gov/API/Users_Guide/Downloading_Files/) or the [Data Transfer Tool](https://gdc.cancer.gov/access-data/gdc-data-transfer-tool).
* __Review and Submit Data__: Prior to submission, data can be reviewed to check for accuracy and completeness. Once the review is complete, the data can be submitted to the GDC for processing through [Data Harmonization](https://gdc.cancer.gov/submit-data/gdc-data-harmonization).
* __Release Data__: After harmonization, data can be released to the research community for access through the [GDC Data Portal](https://portal.gdc.cancer.gov/) and other [GDC Data Access Tools](https://gdc.cancer.gov/access-data/data-access-processes-and-tools).
* __Status and Alerts__: Visual cues are implemented in the GDC Data Submission Portal Dashboard to easily identify incomplete submissions via panel displays summarizing submitted data and associated data elements.
* __Transactions__: A list of all actions performed in a project is provided with detailed information for each action.

## Sections to the Data Submission Portal Guide

* [__Data Submission Overview__](Data_Submission_Overview.md): Graphical explanations used to display the life cycle of projects and their data.
* [__Data Submission Process__](Data_Submission_Process.md): An overview of the data submission process using the GDC Data Submission Portal.
* [__Data Submission Walkthrough__](Data_Submission_Walkthrough.md): Step-by-step instructions on GDC data submission and their relationship to the GDC Data Model.

## HIPAA Compliance

The GDC will not accept any data for patients age 90 and over including any follow-up events in which the event occurs after a patient turns 90 to ensure that HIPAA compliance is maintained. To comply with these requirements data submitters may omit any data (entire cases or specific nodes) that would violate this rule or obfuscate associated dates. Please see the [Date Obfuscation](/Data_Submission_Portal/Users_Guide/Best_Practices/#date-obfuscation) section for more information.

## Release Notes

The [Release Notes](../../Data_Submission_Portal/Release_Notes/Data_Submission_Portal_Release_Notes.md) section of this User's Guide contains details about new features, bug fixes, and known issues.


# Cases

## Overview

The _"Case"_ section list all cases available in the project. Clicking on a particular case will open the details panel, providing more details about the selected case.

[![GDC Submission Cases Default View](images/GDC_Submission_Cases_Default.png)](images/GDC_Submission_Cases_Default.png "Click to see the full image.")

## Case Filters

Cases can be accessed from the menu through multiple filters.

|Filter|Description|
| --- | --- |
| All Cases | Display all cases associated with the project |
| Missing Clinical Data | Display only cases with missing Clinical Data |
| Missing Samples Data | Display only cases with missing Samples Data|

The number of cases corresponding to each filter is displayed on the right side of the filter name.

## Cases List View

The cases list view displays the following information:

|Column|Description|
| --- | --- |
| Submitter ID | Submitter ID of the case |
| Status | Status of the submission, can take the following values: unreleased or released.|
| Last Updated | Last time the case was updated. |
| Warnings | Using icons, displayed if clinical or biospecimen elements are missing for the case. |

On the top left section of the screen, the user can download data about all cases associated to the selected filter.

## Case Details

Clicking on a case will open the details panel. Data in this panel is broken down into multiple sections.

[![GDC Submission Case Details](images/GDC_Submission_Cases_Details.png)](images/GDC_Submission_Cases_Details.png "Click to see the full image.")

Navigation between these sections can be done either by scrolling down or by clicking on the section icon on the left side of the details panel.

[![GDC Submission Cases Details Navigation](images/GDC_Submission_Cases_Details_Navigation.png)](images/GDC_Submission_Cases_Details_Navigation.png "Click to see the full image.")

### Details

Provides details about the case itself, such as its UUID, status, project and creation date.

|Value|Description|
| --- | --- |
| Type | Entity type (Case in this situation)  |
| UUID | Entity's [UUID](https://en.wikipedia.org/wiki/Universally_unique_identifier) |
| Project ID | Project ID associated with the Entity |
| Submitter ID | Submitter ID associated with the Entity |
| Created Datetime | Date and time the entity was created |
| Updated Datetime | Date and time the entity was last updated |

### Related Entities

Table listing all entities, grouped by type, related to the selected case.

This table contains the following columns.

|Column|Description|
| --- | --- |
| Category | Category of the Entity (Clinical, Biospecimen, Experiment Data)  |
| Type | Type of entity (based on Data Dictionary)  |
| Count | Number of occurences of an entity of this type |

Clicking on the count will open a list page listing those entities.


### Transactions

List the 10 most recent transactions associated with this entity ordered by date. Clicking on a transaction will open it in the list page.


# Features

## Homepage

[![GDC Data Submission Portal Homepage](images/GDC_Submission_Homepage_2.png)](images/GDC_Submission_Homepage_2.png "Click to see the full image.")

The homepage is displayed when the submitter logs into the GDC Data Submission Portal. The homepage provides access to project reports on previously submitted data and a list of projects that the submitter is authorized to submit data into. Documentation links are also provided.

To submit data for a project, select the project from the list. This displays the project dashboard.

## Dashboard

[![GDC Data Submission Portal Toolbar](images/GDC_Submission_Dashboard.png)](images/GDC_Submission_Dashboard.png "Click to see the full image.")

The dashboard provides quick access to all the actions a submitter or project owner can perform:

* __Charts__: Provide information on the content of project data (e.g. 28 registered cases in this project have clinical data).
* __Actions__: Provides the list of actions that the user can perform as part of the submission process such as uploading, validating, reviewing, and submitting data. This list varies depending on the user's privileges.
* __Transactions Tab__: Displays all data submission transactions that have occurred on the project.
* __Browse Tab__: Provides a link to browse submitted data as described in the Browse section below.

## Browse

### Layout

[![GDC Data Submission Portal Layout](images/GDC_Submission_Portal_Layout_2.png)](images/GDC_Submission_Portal_Layout_2.png "Click to see the full image.")

The GDC Data Submission Portal browse menu consists of three separate panels:

* __Navigation Panel__: A panel composed of links to different items to filter by case, clinical, biospecimen, or submittable data file entities.
* __Summary Panel__: A table listing search results corresponding to the current selection in the navigation panel filter.
* __Details Panel__: A panel displaying details and previous activities related to a particular entity selected in the summary panel.

### Navigation Panel

[![GDC Data Submission Portal Navigation Panel](images/GDC_Submission_Navigation_2.png)](images/GDC_Submission_Navigation_2.png "Click to see the full image.")

The navigation panel provides access to the following elements:

* __Entity List__: Includes filters for the cases, clinical, biospecimen, and submittable data files uploaded to the project workspace.
* __Annotations__: Provides access to a List of annotations submitted on an entity such as a case or aliquot.

### Summary Panel

The summary panel lists items in a table based on the filter selected in the navigation panel or on the results of a search. Columns in the table vary depending on the selected filter in the navigation panel. The table view supports pagination and column sorting.

[![Cases with no clinical data](images/GDC_Submission_Cases_with_no_Clinical_Data_2.png)](images/GDC_Submission_Cases_with_no_Clinical_Data_2.png "Click to see the full image.")

### Details Panel

The third panel, on the right-hand side of the page, provides more details about a selected entity. The exact content in this panel depends on the type of entity (e.g., case, sample, transaction) selected in the navigation panel.

[![Browse Detail Panel](images/GDC_Submission_Details_Panel_2.png)](images/GDC_Submission_Details_Panel_2.png "Click to see the full image.")

The transactions section in the panel lists all past transactions associated with an entity, clicking on a transaction will redirect the user to that particular entry in the Transactions tab.

## Quick Search

The quick search feature available at the top of the dashboard and browse menu will provide the ability to search for an entity and display the results in the browse menu. To search for an entity, the submitter can begin entering the entity ID in the search box and the autocomplete feature will provide a list of suggested entities for selection.

[![GDC Submission Portal Quick Search](images/GDC_Submission_Quick_Search_2.png)](images/GDC_Submission_Quick_Search_2.png "Click to see the full image.")


# Biospecimen

## Overview

The biospecimen section of the navigation panel provides access to Samples, Portions, Analytes and Aliquots.

[![GDC Submission Biospecimen Panel](images/GDC_Submission_Biospecimen_Panel.png)](images/GDC_Submission_Biospecimen_Panel.png "Click to see the full image.")

Samples, Portions, Analytes and Aliquots on the navigation panel (left side of the screen) are actually filters applied to the list of biospecimens in the project. Screen layout will be identical in-between those four filters.

## Biospecimen List View

The biospecimen list view displays the following information:

|Column|Description|
| --- | --- |
| Submitter ID | Submitter ID of the biospecimen |
| Type | The type of biospecimen, can take the following values: Samples, Portions, Analytes, Aliquots.|
| Last Updated | Last time the case was updated. |

On the top left section of the screen, the user can download data about the selected biospecimen or all biospecimen corresponding to the selected filter.

Clicking on a biospecimen will open the details panel. Data in this panel is broken down in multiple sections.

## Details

Provides details about the biospeciment itself, such as its UUID, status, project and creation date.

[![GDC Submission Biospecimen Details Navigation](images/GDC_Submission_Cases_Details_Details.png)](images/GDC_Submission_Cases_Details_Details.png "Click to see the full image.")

### Hierarchy

List entities (clinical, biospecimen, annotations) attached to a case in a tree-like view. Clicking on an entity redirect to its corresponding details page, easing navigation between entities.

[![GDC Submission Cases Details Hierarchy](images/GDC_Submission_Cases_Details_Hierarchy.png)](images/GDC_Submission_Cases_Details_Hierarchy.png "Click to see the full image.")

### Annotations

Lists annotations attached to an entity.

[![GDC Submission Cases Details Annotations](images/GDC_Submission_Cases_Details_Annotations.png)](images/GDC_Submission_Cases_Details_Annotations.png "Click to see the full image.")

### Transactions

Lists all transactions associated to a case. Clicking on an transaction ID will redirect to the transaction details page.

[![GDC Submission Cases Details Transactions](images/GDC_Submission_Cases_Details_Transactions.png)](images/GDC_Submission_Cases_Details_Transactions.png "Click to see the full image.")


# Data Submission Portal

## Overview

This section will walk users through the submission process using the [GDC Data Submission Portal](https://portal.gdc.cancer.gov/submission/) to upload files to the GDC.

## Authentication

### Requirements

Accessing the GDC Data Submission Portal requires eRA Commons credentials with appropriate dbGaP authorization.  To learn more about obtaining the required credentials and authorization, see [Obtaining Access to Submit Data]( https://gdc.cancer.gov/submit-data/obtaining-access-submit-data).

### Authentication via eRA Commons

Users can log into the GDC Data Submission Portal with eRA Commons credentials by clicking the "Login" button. If authentication is successful, the user will be redirected to the GDC Data Submission Portal front page and the user's eRA Commons username will be displayed in the upper right corner of the screen.

#### GDC Authentication Tokens

The GDC Data Portal provides authentication tokens for use with the GDC Data Transfer Tool or the GDC API. To download a token:

1. Log into the GDC using your eRA Commons credentials.
2. Click the username in the top right corner of the screen.
3. Select the "Download Token" option.

![Token Download Button](images/gdc-data-portal-token-download.png)

A new token is generated each time the `Download Token` button is clicked.

For more information about authentication tokens, see [Data Security](../../Data/Data_Security/Data_Security.md#authentication-tokens).

>**NOTE:** The authentication token should be kept in a secure location, as it allows access to all data accessible by the associated user account.

#### Logging Out

To log out of the GDC, click the username in the top right corner of the screen, and select the Logout option. Users will automatically be logged out after 15 minutes of inactivity.

![Logout link](images/gdc-data-portal-token-download.png)

## Homepage

After authentication, users are redirected to a homepage. The homepage acts as the entry point for GDC data submission and provides submitters with access to a list of authorized projects, reports, and transactions. Content on the homepage varies based on the user profile (e.g. submitter, program office).

[![GDC Submitter Home Page](images/GDC-HomePage-Submit_v2.png)](images/GDC-HomePage-Submit_v2.png "Click to see the full image.")

### Reports

Project summary reports can be downloaded at the Submission Portal homepage at three different levels: `CASE OVERVIEW`, `ALIQUOT OVERVIEW`, and `DATA VALIDATION`.  Each report is generated in tab-delimited format in which each row represents an active project.  

* __`CASE OVERVIEW`:__ This report describes the number of cases with associated biospecimen data, clinical data, or submittable data files (broken down by data type) for each project.
* __`ALIQUOT OVERVIEW`:__ This report describes the number of aliquots in a project with associated data files. Aliquot numbers are broken down by sample tissue type.
* __`DATA VALIDATION`:__ This report categorizes all submittable data files associated with a project by their file status.

### Projects

The projects section in the homepage lists the projects that the user has access to along with basic information about each project. For users with access to a large number of projects, this table can be filtered using the 'FILTER PROJECTS' field. Selecting a project ID will direct the user to the project's [Dashboard](#dashboard). The button used to release data for each project is also located on this screen, see [Release](#release) for details.

## Dashboard

The GDC Data Submission Portal dashboard provides details about a specific project.

[![GDC Submission Dashboard Page](images/Submission_portal_homepage.png)](images/Submission_portal_homepage.png "Click to see the full image.")

The dashboard contains various visual elements to guide the user through all stages of submission, from viewing the [Data Dictionary](https://docs.gdc.cancer.gov/Data_Dictionary/), support of data upload, to submitting a project for harmonization.

To better understand the information displayed on the dashboard and the available actions, please refer to the [Data Submission Walkthrough](Data_Submission_Walkthrough.md).

### Project Overview
The Project Overview sections of the dashboard displays the most current project state (open / review / submitted / processing) and the GDC Release, which is the date in which the project was released to the GDC.

The search field at the top of the dashboard allows for submitted entities to be searched by partial or whole `submitter_id`.  When a search term is entered into the field, a list of entities matching the term is updated in real time.  Selecting one of these entities links to its details in the [Browse Tab](#browse).

The remaining part of the top section of the dashboard is broken down into four status charts:

* __QC Errors__: The number of errors found in the uploaded data.  For more details please refer to the [QC Report Section](/Data_Submission_Portal/Users_Guide/Data_Submission_Process/#qc-reports).
* __Cases with Clinical__: The number of `cases` for which Clinical data has been uploaded.
* __Cases with Biospecimen__: The number of `cases` for which Biospecimen data has been uploaded.
* __Cases with Submittable Data Files__: The number of `cases` for which experimental data has been uploaded.
* __Submittable Data Files__: The number of registered submittable data files that have been successfully uploaded through the GDC Data Transfer Tool. Totals do not include files that have been submitted for harmonization. For more information on this status chart, please refer to [File Lifecycle](Data_Submission_Overview.md#file-lifecycle).
    * __`DOWNLOAD MANIFEST`:__ This button below the status chart allows the user to download a manifest for registered files in this project that have not yet been uploaded.

### Action Panels

There are two action panels available below the Project Overview.

* [UPLOAD DATA TO YOUR WORKSPACE](Data_Submission_Walkthrough.md): Allows a submitter to upload project data to the GDC project workspace. The GDC will validate the uploaded data against the [GDC Data Dictionary](https://docs.gdc.cancer.gov/Data_Dictionary/). This panel also contains a table that displays details about the five latest transactions. Clicking the IDs in the first column will bring up a window with details about the transaction, which are documented in the [transactions](#transactions) page. This panel will also allow the user to commit file uploads to the project.
* [REVIEW AND SUBMIT YOUR WORKSPACE DATA TO THE GDC](#submit-your-workspace-data-to-the-gdc): Allows a submitter to review project data which will lock the project to ensure that additional data cannot be uploaded while in review. Once the review is complete, the data can be submitted to the GDC for processing through the [GDC Harmonization Process](https://gdc.cancer.gov/submit-data/gdc-data-harmonization).

These actions and associated features are further detailed in their respective sections of the documentation.

## Transactions

The transactions page lists all of the project's transactions. The transactions page can be accessed by choosing the Transactions tab at the top of the dashboard or by choosing "View All Data Upload Transactions" in the first panel of the dashboard.

[![GDC Submission Transactions](images/GDC_Submission_Transactions_2.png)](images/GDC_Submission_Transactions_2.png "Click to see the full image.")

The types of transactions are the following:

* __Upload:__ The user uploads data to the project workspace. Note that submittable data files uploaded using the GDC Data Transfer tool do not appear as transactions. Uploaded submittable data can be viewed in the Browse tab.
* __Delete:__ The user deletes data from the project workspace.
* __Review:__ The user reviews the project before submitting data to the GDC.
* __Open:__ The user re-opens the project if it was under review. This allows the upload of new data to the project workspace.
* __Submit:__ The user submits uploaded data to the GDC. This triggers the data harmonization process.
* __Release:__ The user releases harmonized data to be available through the GDC Data Portal and other GDC data access tools.

### Transactions List View

The transactions list view displays the following information:

|Column|Description|
| --- | --- |
| __ID__ | Identifier of the transaction |
| __Type__ | Type of the transaction (see the list of transaction types in the previous section)|
| __Step__ | The step of the submission process that each file is currently in. This can be Validate or Commit. "Validate" represents files that have not yet been committed but have been uploaded using the submission portal or the API. |
| __DateTime__ | Date and Time that the transaction was initiated |
| __User__ | The username of the submitter that performed the transaction |
| __State__ | 	Indicates the status of the transaction: `SUCCEEDED`, `PENDING`, or `FAILED` |
| __Commit/Discard__ | Two buttons appear when data has been uploaded using the API or the submission portal. This allows for validated data to be incorporated into the project or discarded. This column will then display the transaction number for commited uploads and "Discarded" for the uploads that are discarded.|

### Transaction Filters

Choosing from the drop-down menu at the top of the table allows the transactions to be filtered by those that are in progress, to be committed, succeeded, failed, or discarded. The drop-down menu also allows for the transactions to be filtered by type and step.  

### Transactions Details

Clicking on a transaction will open the details panel. Data in this panel is organized into multiple sections including actions, details, types, and documents as described below.

[![GDC Submission Transactions](images/GDC_Submission_Transactions_Details_3.png)](images/GDC_Submission_Transactions_Details_3.png "Click to see the full image.")

Navigation between the sections can be performed by either scrolling down or by clicking on the section icon displayed on the left side of the details panel.

#### Actions

The Actions section allows a user to perform an action for transactions that provide actions. For example, if a user uploads read groups and file metadata, a corresponding manifest file will be available for download from the transaction. This manifest is used to upload the actual files through the [GDC Data Transfer Tool](https://gdc.cancer.gov/access-data/gdc-data-transfer-tool).

[![GDC Submission Transactions Details Action](images/GDC_Submission_Transactions_Details_Action_2.png)](images/GDC_Submission_Transactions_Details_Action_2.png "Click to see the full image.")

#### Details

The Details section provides details about the transaction itself, such as its project, type, and number of affected cases.

[![GDC Submission Transactions Details](images/GDC_Submission_Transactions_Details_Details_2.png)](images/GDC_Submission_Transactions_Details_Details_2.png "Click to see the full image.")

#### Types

The Types section lists the type of files submitted and the number of affected cases and entities.

[![GDC Submission Transactions Types](images/GDC_Submission_Transactions_Details_Types_2.png)](images/GDC_Submission_Transactions_Details_Types_2.png "Click to see the full image.")

#### Documents

The Documents section lists the files submitted during the transaction.
The user can download the original files from the transaction, a report detailing the transaction, or the errors that originated from the transaction that has failed.

[![GDC Submission Transactions Documents](images/GDC_Submission_Transactions_Details_Documents_2.png)](images/GDC_Submission_Transactions_Details_Documents_2.png "Click to see the full image.")

## Browse

The `Browse` menu provides access to all of a project's content. Most content is driven by the GDC Data Dictionary and the interface is dynamically generated to accommodate the content.

Please refer to the [GDC Data Dictionary Viewer](../../Data_Dictionary/viewer.md) for specific details about dictionary-generated fields, columns, and filters.

[![GDC Submission Cases Default View](images/GDC_Submission_Cases_Default_2.png)](images/GDC_Submission_Cases_Default_2.png "Click to see the full image.")

### Main Interface Elements

#### Filters

A wide set of filters are available for the user to select the type of entity to be displayed. These filters are dynamically created based on the [GDC Data Dictionary](../../Data_Dictionary/index.md).

Current filters are:

|Filter|Description|
| --- | --- |
| __Cases__ | Display all `Cases` associated with the project. |
| __Clinical__ | Display all Clinical data uploaded to the project workspace. This is divided into subgroups including `Demographics`, `Diagnoses`, `Exposures`, `Family Histories`, `Follow_up`, `Molecular_tests`, and `Treatments`. |
| __Biospecimen__ | Display all Biospecimen data uploaded to the project workspace. This is divided into subgroups including `Samples`, `Portions`, `Slides`, `Analytes`, `Aliquots`, and `Read Groups`. |
| __Submittable Data Files__ | Displays all data files that have been registered with the project. This includes files that have been uploaded and those that have been registered but not uploaded yet. This category is divided into groups by file type. |
| __Annotations__ | Lists all annotations associated with the project. An annotation provides an explanatory comment associated with data in the project. |
| __Harmonized Data Files__ | Lists all data files that have been harmonized by the GDC. This category is divided into groups by generated data. |

#### List View

The list view is a paginated list of all entities corresponding to the selected filter.

On the top-right section of the screen, the user can download data about all entities associated with the selected filter.

* For the case filter, it will download all Clinical data or all Metadata.
* For all other filters, it will download the corresponding metadata (e.g., for the `demographic` filter, it will download all `demographic` data).

[![GDC Submission Case Summary Download](images/GDC_Submission_Cases_Summary_Download_2.png)](images/GDC_Submission_Cases_Summary_Download_2.png "Click to see the full image.")

#### Details Panel

Clicking on an entity will open the details panel. Data in this panel is broken down into multiple sections depending on the entity type. The main sections are:

* __Actions__: Actions that can be performed relating the entity. This includes downloading the metadata (JSON or TSV) or submittable data file pertaining to the entity and deleting the entity. See the [Deleting Entities](Data_Submission_Walkthrough.md#deleting-submitted-entities) guide for more information.  
* __Summary__: A list of IDs and system properties associated with the entity.
* __Details__: Properties of the entity (not associated with cases).
* __Hierarchy__ or __Related Entities__: A list of associated entities.
* __Annotations__: A list of annotations associated with the entity.
* __Transactions__: A list of previous transactions that affect the entity.

[![GDC Submission Case Details](images/GDC_Submission_Cases_Details_2.png)](images/GDC_Submission_Cases_Details_2.png "Click to see the full image.")

The sections listed above can be navigated either by scrolling down or by clicking on the section icon on the left side of the details panel.

#### Related Entities

The Related Entities table lists all entities, grouped by type, related to the selected `case`. This section is only available at the `case` level.

[![GDC Submission Cases Related Entities](images/GDC_Submission_Cases_Summary_Related_Entities_2.png)](images/GDC_Submission_Cases_Summary_Related_Entities_2.png "Click to see the full image.")


This table contains the following columns:

* __Category__: category of the entity (Clinical, Biospecimen, submittable data file).
* __Type__: type of entity (based on Data Dictionary).
* __Count:__ number of occurrences of an entity associated with the `case`. Clicking on the count will open a window listing those entities within the Browse page.

#### Hierarchy

The hierarchy section is available for entities at any level (e.g., Clinical, Biospecimen, etc.), except for `case`. The user can use the hierarchy section to navigate through entities.

The hierarchy shows:

* The `case` associated with the entity.
* The __direct__ parents of the entity.
* The __direct__ children of the entity.

[![GDC Submission Cases Details Hierarchy](images/GDC_Submission_Cases_Summary_Hierarchy_2.png)](images/GDC_Submission_Cases_Summary_Hierarchy_2.png "Click to see the full image.")

After uploading data to the workspace on the GDC Data Submission Portal, data will need to be [reviewed by the submitter](#pre-harmonization-checklist) and then [submitted to the GDC](#submit-to-the-gdc) for processing.

## QC Reports

The QC Reports section allows users to see errors identified by the GDC for the current data that has not yet been submitted for harmonization.  This includes all nodes in state `validated`.  Data with error type `Critical` indicates errors that must be fixed before a submitter can Request Harmonization.  Errors with error type `Warning` should be reviewed by the submitter as they may indicate discrepancies or problematic data.

You can see in the QC Reports Tab highlights of what data are present and the types of errors found in the project.

[![QC Report](images/QC_Report_tab.png)](images/QC_Report_tab.png "Click to see the full image.")

To find specific details for any node that contains errors you can click on the facet panel on the left to see those errors and to download a list of errors for that respective node. All potential errors are listed in the [Pre-harmonization Checklist](/Data_Submission_Portal/Users_Guide/Data_Submission_Process/#pre-harmonization-checklist).

[![QC Errors for Submitted Unaligned Reads](images/SUR_QC_errors.png)](images/SUR_QC_errors.png "Click to see the full image.")

## Submit Your Workspace Data to the GDC

The GDC Data Submission process is detailed on the [Data Submission Processes and Tools](https://gdc.cancer.gov/submit-data/data-submission-processes-and-tools) section of the GDC Website.

### Review

The submitter is responsible for reviewing the data uploaded to the project workspace (see [Data Submission Walkthrough](Data_Submission_Walkthrough.md)), and ensuring that it is ready for processing by the GDC [Harmonization Process](https://gdc.cancer.gov/submit-data/gdc-data-harmonization).

The user will be able to view the section below on the dashboard. The `REVIEW` button is available only if the project is in "OPEN" state.

[![GDC Submission Review Tab](images/GDC_Submission_Submit_Release_Review_tab_2_v2.png)](images/GDC_Submission_Submit_Release_Review_tab_2_v2.png "Click to see the full image.")

Setting the project to the "REVIEW" state will lock the project and prevent users from uploading additional data. During this period, the submitter can browse the data in the Data Submission Portal or download it. Once the review is complete, the user can request to submit data to the GDC.

Once the user clicks on `REVIEW`, the project state will change to "REVIEW":

[![GDC Submission Review State](images/GDC_Submission_Submit_Release_Project_State_Review_3.png)](images/GDC_Submission_Submit_Release_Project_State_Review_3.png "Click to see the full image.")

### Pre-Harmonization Checklist

The Harmonization step is __NOT__ an automatic process that occurs when data is uploaded to the GDC. The GDC performs batch processing of submitted data for Harmonization only after verifying that the submission is complete.

QC checks are automatically run on all supplied metadata and data files.  The results are displayed within the [QC Reports](/Data_Submission_Portal/Users_Guide/Data_Submission_Process/#qc-reports).  These errors fall into two categories: Critical or Warning.  If an error is deemed Critical it must be resolved before a submitter can request harmonization.  If an error is categorized as Warning then the submitter should review this to verify the data have been submitted correctly.  A list of the errors and their meanings are found in the table below:

#### __Critical Errors__

| Error Message | Description | How to Fix / Error Meaning |
|---|---|---|
|INVALID_CHARACTER  | This entity submitter_id includes invalid characters | Upload new entity without invalid characters.  The acceptable characters are alphanumeric characters [a-z, A-Z, 0-9] and `_`, `.`, `-`. Any other characters will interfere with the Harmonization workflow. |
| MORE_THAN_ONE_SAMPLE_TYPE  | The aliquot is associated with more than one sample type | Ensure there is no `aliquot` attached to multiple `sample` nodes of more than one sample_type. |
| TWO_NODE_TYPES  | The aliquot is associated with two or more node types| Ensure aliquot is only connected to a single type of node. |
| PE_FASTQ_FILE_COUNT  | The number of FASTQ files for PE readgroup is not 2| Ensure that if a read group is paired end, that it has two FASTQ files. For the `read_group` node, make sure that the `is_paired_end` is set to `true` for paired end sequencing and `false` for single end sequencing.|
| SE_FASTQ_FILE_COUNT  | The number of FASTQ files for SE readgroup is not 1| Ensure that if a read group is single end, that it has one FASTQ file. For the `read_group` node, make sure that the `is_paired_end` is set to `true` for paired end sequencing and `false` for single end sequencing.|
| CAPTURE_KIT_INADEQUATE  | WXS/Targeted Sequencing ReadGroup lacks valid target capture kit| Modify read group entity to have a valid target capture kit from data dictionary. The `target_capture_kit` property is completed when the selected `library_strategy` is `WXS`. Errors will occur if `Not Applicable` or `Unknown` is selected. |
| TARGET_SEQ_LIBRARY_SELECTION  | ReadGroup has library strategy Targeted Sequencing but does not have PCR or Hybrid Selection as its library selection|  If library strategy is Target Sequencing, modify library selection to be either PCR or Hybrid Selection |
| WXS_LIBRARY_SELECTION  | ReadGroup has library strategy WXS but does not have Hybrid Selection as its library selection| Modify library selection to be Hybrid Selection for WXS read groups |  
| WGS_LIBRARY_SELECTION  | ReadGroup has library strategy WGS but does not have Random as its library selection| For WGS read groups, ensure library strategy is set to Random |
| NO_READ_PAIR_NUMBER  | The FASTQ is paired but has no read_pair_number| Include a read_pair_number for paired end FASTQ files |
| DUPLICATE_MD5S  | Two or more files have the same md5sum| This means there are duplicate files in the submission.  You must delete one of these files |

#### __Warning Errors__
| Error Message | Description | How to Fix / Error Meaning |
|---|---|---|
| FILE_BAD_STATE  | The file node is in a bad state |  There are some files in a bad file_state. All files that are registered must been uploaded and validated. If file_state is `Error` You will have to delete the file using the data transfer tool, and re-upload it, or upload a file if the state is `Registered`|
| INCONSISTENT_READGROUPS  | ReadGroups sharing a library_strategy under a given aliquot have properties that do not match| Verify the properties of shared read groups under the same aliquot are consistent.|
| NO_CLINICAL_SUPPLEMENT  | The case has no associated clinical supplement| Upload an optional clinical supplement file.  This is a file that contains clinical data about one or more cases in a user specified format |
| NO_BIOSPECIMEN_SUPPLEMENT  | The case has no associated biospecimen supplement| Upload an optional biospecimen supplement file.  This is a file that contains biospecimen data about one or more cases in a user specified format |
| NO_DEMOGRAPHIC  | The case has no associated demographic information| Provide demographic information on the case.  This will be required before data can be released. |
| NO_DIAGNOSIS  | The case has no associated diagnosis information | Provide diagnosis information on the case.  This will be required before data can be released. |
| MORE_THAN_ONE_SAMPLE  | The aliquot is associated with more than one sample| Review whether multiple samples were actually combined to make a single aliquot.  This is uncommon, but potentially correct. |
| MULTIPLE_ALIGNED_BAMS  | The read_group has multiple submitted aligned BAMs| Review whether one read group actually appears in multiple BAM files.  This is uncommon. |
| NO_MULTIPLEX_BARCODE  | The read_group has no multiplex barcode| Provide multiplex barcode for the read_group. |
| NO_FLOWCELL_BARCODE  | The read_group has no flowcell barcode| Provide flowcell barcode for the read_group |
| NO_LANE_NUMBER  | The read_group has no lane number| Provide lane number for the read_group |
| MULTIPLE_SARS_ON_ALIQUOT  | Multiple submitted aligned reads of the same experimental strategy are associated with one aliquot.|  Each `aliquot` node is only associated with one `submitted_aligned_reads` file of the same `experimental_strategy`. |
| FASTQ_UNKNOWN_EXTENSION  | The FASTQ filename has an unknown extension| FASTQ file extension should be `.fq` or `.fq.gz`. Impermissible extensions are `tar.gz` and `tar`. |
| MULTIPLE_FASTQ_READGROUPS  | Submitted FASTQ file has links to multiple read groups| Ensure `submitted_unaligned_reads` of data_format `FASTQ` is not linked to multiple `read_group` nodes. |
| INVALID_FASTQ_EXTENSION  | Submitted FASTQ file name has an invalid extension| FASTQ file extension should be `.fq` or `.fq.gz`. Impermissible extensions are `tar.gz` and `tar`.|
| FASTQ_TOO_LARGE  | FASTQ exceeds 10GB in size| The `submitted_unaligned_reads` file is larger than 10 GB. |
| NO_ASSOCIATED_FILES  | ReadGroup has no associated genomic files| Ensure that all read groups have genomic files attached - or delete them if they are no longer needed |

Once user review is complete and all Critical errors are resolved, clicking the `REQUEST HARMONIZATION` button will indicate to the GDC Team and pipeline automation system that data processing can begin.

### Submit to the GDC for Harmonization

When the project is ready for processing, the submitter will request to submit data to the GDC for Harmonization. If the project is not ready for processing, the project can be re-opened. Then the submitter will be able to upload more data to the project workspace.

The `REQUEST HARMONIZATION` button is available only if the project is in "REVIEW" state. At this point, the user can decide whether to re-open the project to upload more data or to request harmonization of the data to the GDC. When the project is in "REVIEW" the following panel appears on the dashboard:

[![GDC Submission Submit Tab](images/GDC_Submission_Submit_Release_Submit_tab_2_v4.png)](images/GDC_Submission_Submit_Release_Submit_tab_2_v4.png "Click to see the full image.")

Once the user submits data to the GDC, they cannot modify the submitted nodes and files while harmonization is underway.  Additional project data can be added during this period and will be considered a separate batch.  To process an additional batch the user must again review the data and select `REQUEST HARMONIZATION`.

[![GDC Submission Submission Tab](images/GDC_SUBMIT_TO_GDC_v3.png)](images/GDC_SUBMIT_TO_GDC_v3.png "Click to see the full image.")

When the user clicks on the action `REQUEST HARMONIZATION` on the dashboard, the following popup is displayed:

[![GDC Submission Submit Popup](images/GDC_Submission_Submit_Release_Submit_Popup_v2.png)](images/GDC_Submission_Submit_Release_Submit_Popup_v2.png "Click to see the full image.")


After the user clicks on `SUBMIT VALIDATED DATA TO THE GDC`, the project state becomes "Harmonization Requested":

[![GDC Submission Project State](images/GDC_Submission_Submit_Release_Project_State_v3.png)](images/GDC_Submission_Submit_Release_Project_State_v3.png "Click to see the full image.")

The GDC requests that users submit their data to the GDC for harmonization within six months from the first upload of data to the project workspace.

### Reviewing Harmonized Data
After harmonization and prior to release, the GDC provides data submitters with access to their harmonized data. This allows the submitter to perform a check of the data, and let the GDC know if anything is incorrect before the data are released to the GDC Data Portal. How and in what detail the submitter wants to perform such a review is up to them, but here are a few suggestions for what a submitter may want to check.

Are all expected data present? More specifically, you could review the following questions:
   * Are the number of cases correct?
   * Are the number of cases associated with a given experimental strategy correct?
   * Are there any cases or experimental strategies I want to hold back that are still within the 6 month embargo period?
   * Does the clinical data appear as I expect?
   * Do the alignment statistics look acceptable? The GDC produces alignment metrics which are available via the API. This will allow users to see whether coverage, alignment, and other statistics are in line with expectations. [The complete list can be found here.](https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/Aligned_reads_summary_metrics/)

If users have access to other derived data files, like called variants or expression levels, there is another level of QC that is possible.

If you have access to this data you could also investigate the following:
   * Are expected variants present for a given tumor-normal pair? Note, due to differences between the GDC and user workflows (e.g. reference genome, variant calling pipelines, variant filtering, etc.) the exact list of variants may differ significantly between MAFs generated by users and those generated by the GDC.
   * Does gene expression correlate with previously generated expression data from the same aliquot? Note, the GDC performs non-stranded expression quantification for HTSeq workflows. To review strand-specific results please review STAR output.

Once these user reviews have been completed, the user will need to contact the GDC and inform them that the project is ready for release.

## Release
Project release occurs after the data has been harmonized, and allows users to access this data with the [GDC Data Portal](https://portal.gdc.cancer.gov/) and other [GDC Data Access Tools](https://gdc.cancer.gov/access-data/data-access-processes-and-tools). The GDC will release data according to [GDC Data Sharing Policies](https://gdc.cancer.gov/submit-data/data-submission-policies). Data must be released within six months after GDC data processing has been completed, or the submitter may request earlier release using the "Request Release" function.  A project can only be released once.

[![GDC Submission Release Tab](images/GDC_Submission_Landing_Submitter_4.png)](images/GDC_Submission_Landing_Submitter_4.png "Click to see the full image.")

When the user clicks on the action `REQUEST RELEASE`, the following Release popup is displayed:

[![GDC Submission Release Popup](images/GDC_Submission_Submit_Release_Release_Popup.png)](images/GDC_Submission_Submit_Release_Release_Popup.png "Click to see the full image.")

After the user clicks on `RELEASE SUBMITTED AND PROCESSED DATA`, the project release state becomes "Release Requested":

[![GDC Submission Project State](images/GDC_Submission_Submit_Release_Project_State_3.png)](images/GDC_Submission_Submit_Release_Project_State_3.png "Click to see the full image.")


>__Note__: Released cases and/or files can be redacted from the GDC. For more information, visit the [GDC Policies page (under GDC Data Sharing Policies)](https://gdc.cancer.gov/about-gdc/gdc-policies).


# Transactions

## Overview

The transactions page lists all of the project's transactions. The transactions page can be accessed by choosing the "Transactions" tab at the top of the dashboard or by choosing "View All Data Upload Transactions" in the first panel of the dashboard.

[![GDC Submission Transactions](images/GDC_Submission_Transactions_2.png)](images/GDC_Submission_Transactions_2.png "Click to see the full image.")

The types of transactions are the following:

* __Upload:__ The user uploads data to the project workspace. Note that submittable data files uploaded using the GDC Data Transfer tool do not appear as transactions. Uploaded submittable can be viewed in the [Browse](Browse_Data.md) tab.
* __Review:__ The user reviews the project before submitting data to the GDC.
* __Open:__ The user re-opens the project if it was under review. This allows the upload of new data to the project workspace.
* __Submit:__ The user submits uploaded data to the GDC. This triggers the data harmonization process.
* __Release:__ The user releases harmonized data to be available through the GDC Data Portal and other GDC data access tools.

__Note:__ When submittable data files are uploaded through the Data Transfer Tool they are not displayed as transactions.  

## Transactions List View

The transactions list view displays the following information:

|Column|Description|
| --- | --- |
| __ID__ | Identifier of the transaction |
| __Type__ | Type of the transaction (see the list of transaction types in the previous section)|
| __Step__ | The step of the submission process that each file is currently in. This can be Validate or Commit. "Validate" represents files that have not yet been committed but have been submitted using the submission portal or the API . |
| __DateTime__ | Date and Time that the transaction was initiated |
| __User__ | The username of the submitter that performed the transaction |
| __Status__ | 	Indicates the status of the transaction: `SUCCEEDED`, `PENDING`, or `FAILED` |
| __Commit/Discard__ | Two buttons that appear when data has been uploaded using the API or the submission portal.  This allows for validated data to be incorporated into the project or discarded. |

## Transaction Filters

Choosing the radio buttons at the top of the table allows the transactions to be filtered by those that are in progress, to be committed, succeeded, failed, or discarded. The drop-down menu also allows for the transactions to be filtered by type.  

## Transactions Details

Clicking on a transaction will open the details panel. Data in this panel is organized into multiple sections including actions, details, types, and documents as described below.

[![GDC Submission Transactions](images/GDC_Submission_Transactions_Details_2.png)](images/GDC_Submission_Transactions_Details_2.png "Click to see the full image.")

Navigation between the sections can be performed by either scrolling down or by clicking on the section icon displayed on the left side of the details panel.

### Actions

The Actions section allows a user to perform an action for transactions that provide actions. For example, if a user uploads read groups and file metadata, a corresponding manifest file will be available for download from the transaction. This manifest is used to upload the actual files through the [GDC Data Transfer Tool](https://gdc.cancer.gov/access-data/gdc-data-transfer-tool).

[![GDC Submission Transactions Details Action](images/GDC_Submission_Transactions_Details_Action_2.png)](images/GDC_Submission_Transactions_Details_Action_2.png "Click to see the full image.")

### Details

The Details section provides details about the transaction itself, such as its project, type, and number of affected cases.

[![GDC Submission Transactions Details](images/GDC_Submission_Transactions_Details_Details_2.png)](images/GDC_Submission_Transactions_Details_Details_2.png "Click to see the full image.")

### Types

The Types section lists the type of files submitted and the number of affected cases and entities.

[![GDC Submission Transactions Types](images/GDC_Submission_Transactions_Details_Types_2.png)](images/GDC_Submission_Transactions_Details_Types_2.png "Click to see the full image.")

### Documents

The Documents section lists the files submitted during the transaction.
The user can download the original files from the transaction, a report detailing the transaction, or the errors that originated from the transaction (if the transaction had failed).

[![GDC Submission Transactions Documents](images/GDC_Submission_Transactions_Details_Documents_2.png)](images/GDC_Submission_Transactions_Details_Documents_2.png "Click to see the full image.")


# Submission Best Practices

Because of the data types and relationships included in the GDC, data submission can become a complex procedure. The purpose of this section is to present guidelines that will aid in the incorporation and harmonization of submitters' data. Please contact the GDC Help Desk at __<support@nci-gdc.datacommons.io>__ if you have any questions or concerns regarding a submission project.

## Date Obfuscation

The GDC is committed to providing accurate and useful information as well as protecting the privacy of  patients if necessary. Following this, the GDC accepts time intervals that were transformed to remove information that could identify an individual but preserve clinically useful timelines. The GDC recommends following a series of HIPAA regulations regarding the reporting of age-related information, which can be downloaded [here](BCRInformatics-DateObfuscator-291116-1100-2.pdf) as a PDF.

### General Guidelines

Actual calendar dates are not reported in GDC clinical fields but the lengths of time between events are preserved. Time points are reported based on the number of days since the patient's initial diagnosis. Events that occurred after the initial diagnosis are reported as positive and events that occurred before are reported as negative. Dates are not automatically obfuscated by the GDC validation system and submitters are required to make these changes in their clinical data. This affects these fields: `days_to_birth`, `days_to_death`, `days_to_last_follow_up`, `days_to_last_known_disease_status`, `days_to_recurrence`, `days_to_treatment`

>__Note:__ The day-based fields take leap years into account.  

### Patients Older than 90 Years and Clinical Events

Because of the low population number within the demographic of patients over 90 years old, it becomes more likely that patients can potentially be identified by a combination of their advanced age and publicly available clinical data. Because of this, patients over 90 years old are reported as exactly 90 years or 32,872 days old.

Following this, clinical events that occur over 32,872 days are also capped at 32,872 days. When timelines are capped, the priority should be to shorten the post-diagnosis values to preserve the accuracy of the age of the patient (except for patients who were diagnosed at over 90 years old). Values such as `days_to_death` and `days_to_recurrence` should be compressed before `days_to_birth` is compressed.

### Examples Timelines

__Example 1:__ An 88 year old patient is diagnosed with cancer and dies 13 years later.  The `days_to_birth` value is less than 32,872 days, so it can be accurately reported.  However, between the initial diagnosis and death, the patient turned 90 years old. Since 32,872 is the maximum, `days_to_death` would be calculated as 32872 - 32142 = 730.

>__Dates__

>* _Date of Birth:_ 01-01-1900
>* _Date of Initial Diagnosis:_ 01-01-1988
>* _Date of Death:_ 01-01-2001

>__Actual-Values__

>* _days_to_birth:_ -32142
>* _days_to_death:_ 4748

>__Obfuscated-Values__

>* _days_to_birth:_ -32142
>* _days_to_death:_ 730


__Example 2:__ A 98 year old patient is diagnosed with cancer and dies three years later.  Because `days_to_X` values are counted from initial diagnosis, days will be at their maximum value of 32,872 upon initial diagnosis. This will compress the later dates and reduce `days_to_birth` to -32,872 and `days_to_death` to zero.  

>__Dates__

>* _Date of Birth:_ 01-01-1900
>* _Date of Initial Diagnosis:_ 01-01-1998
>* _Date of Death:_ 01-01-2001

>__Actual-Values__

>* _days_to_birth:_ -35794
>* _days_to_death:_ 1095

>__Obfuscated-Values__

>* _days_to_birth:_ -32872
>* _days_to_death:_ 0

## Array Submission

Certain fields in the GDC, such as diagnosis.sites_of_involvement, are of type "array".  This allows multiple values to be submitted on one property. These values need to be uploaded in a `|`-delimited format for TSV formatted uploads and a JSON-type array for JSON formatted uploads. See the example below.  

*  __Example (TSV):__ `Kidney, Upper Pole|Kidney, Middle` (would appear under sites_of_involvement header)
*  __Example (JSON):__ `"sites_of_involvement" : ["Kidney, Upper Pole", "Kidney, Middle"]`

## Submitting Complex Data Model Relationships

The GDC Data Model includes relationships in which more than one entity of one type can be associated with one entity of another type. For example, more than one `read_group` entity can be associated with a `submitted_aligned_reads` entity. JSON-formatted files, in which a list object can be used, are well-suited to represent this type of relationship. Tab-delimited (TSV) files require additional syntax to demonstrate these relationships. For example, associating a `submitted_aligned_reads` entity to three read groups would require three `read_groups.submitter_id` columns, each with the `#` symbol and a number appended to them. See the two files below:

=== "TSV"

    ```TSV
    type    submitter_id    data_category   data_format data_type   experimental_strategy   file_name   file_size   md5sum  read_groups.submitter_id#1 read_groups.submitter_id#2  read_groups.submitter_id#3
    submitted_aligned_reads Alignment.bam  Raw Sequencing Data BAM Aligned Reads   WGS test_alignment.bam    123456789  aa6e82d11ccd8452f813a15a6d84faf1    READ_GROUP_1  READ_GROUP_2  READ_GROUP_3  

    ```

=== "JSON"

    ```JSON
    {
        "type": "submitted_aligned_reads",
        "submitter_id": "Alignment.bam",
        "data_category": "Raw Sequencing Data",
        "data_format": "BAM",
        "data_type": "Aligned Reads",
        "experimental_strategy": "WGS",
        "file_name": "test_alignment.bam",
        "file_size": 123456789,
        "md5sum": "aa6e82d11ccd8452f813a15a6d84faf1",
        "read_groups": [
            {"submitter_id": "READ_GROUP_1"},
            {"submitter_id": "READ_GROUP_2"},
            {"submitter_id": "READ_GROUP_3"}
        ]
    }
    ```

### Read groups

#### Submitting Read Group Names

The `read_group` entity requires a `read_group_name` field for submission.  If the `read_group` entity is associated with a BAM file, the submitter should use the `@RG` ID present in the BAM header as the `read_group_name`. This is important for the harmonization process and will reduce the possibility of errors.  

#### Multiple FASTQs from One Read Group

To align reads according to their direction and pair, the GDC requires that unaligned forward and reverse reads are submitted as "submitted_unaligned_reads."  When more than one FASTQ exists for a read group direction, the GDC requires that the FASTQ files are concatenated for each direction. In other words, each paired-end read group should be associated with exactly two FASTQ files (submitted_unaligned_reads).  

#### Minimal and Recommended Read Group Information

In addition to the required properties on `read_group` we also recommend submitting `flow_cell_barcode`, `lane_number` and `multiplex_barcode`.  This information can be used by our bioinformatics team and data downloaders to construct a `Platform Unit` (`PU`), which is a universally unique identifier that can be used to model various sequencing technical artifacts.  More information can be found in the [SAM specification PDF](https://github.com/samtools/hts-specs/blob/master/SAMv1.pdf).

For projects with library strategies of targeted sequencing or WXS we also require information on the target capture protocol included on `target_capture_kit`.

If this information is not provided it may cause a delay in the processing of submitted data.

Additional read group information will benefit data users.  Such information can be used by bioinformatics pipelines and will aid understanding and mitigation of batch effects.  If available, you should also provide as many of the remaining read group properties as possible.

## Submission File Quality Control

The GDC harmonization pipelines include multiple quality control steps to ensure the integrity of harmonized files and the efficient use of computational resources. For fastest turnaround of data processing we recommend that submitters perform basic QC of their data files prior to upload to identify any underlying data quality issues. This may include such tests as verifying expected genome coverage levels and sequencing quality.

Except for [miRNA data](https://docs.gdc.cancer.gov/Data_Submission_Portal/Users_Guide/Best_Practices/#mirna-submission) submission, sequencing data for all other experimental strategy types (i.e. whole exome sequencing, whole genome sequencing, targeted sequencing and mRNA sequencing) do not need to be trimmed by submitters prior to submission, as tools used in GDC alignment workflows are capable of handling adaptors and low quality bases correctly.

## Target Capture Kit Q and A

1. What is a Target Capture Kit?  
Target capture kits contain reagents designed to enrich for and thus increase sequencing depth in certain genomic regions before library preparation. Two of the major methods to enrich genomic regions are by Hybridization and by PCR (Amplicon).

2. Why do we need Target Capture Kit information?  
Target region information is important for DNA-Seq variant calling and filtering, and essential for Copy Number Alternation and other analyses. This information is only needed for the Experimental Strategies of WXS or Targeted Sequencing.

3. How do submitters provide this information?  
There are 3 steps  
    * Step 1. The submitter should contact GDC User Service about any new Target Capture Kits that do not already exist in the GDC Dictionary. The GDC Bioinformatics and User Services teams will work together with the submitter to create a meaningful name for the kit and import this name and Target Region Bed File into the GDC.
    * Step 2. The submitter can then select one and only one GDC Target Capture Kit for each read group during molecular data submission.
    * Step 3. The submitter should also select the appropriate `library_strategy` and `library_selection` on the read_group entity.

4. What is a Target Region Bed File?  
A Target Region Bed File is tab-delimited file describing the kit target region in [bed format](https://genome.ucsc.edu/FAQ/FAQformat.html#format1). The first 3 columns of such files are chrom, chromStart, and chromEnd.
Note that by definition, bed files are 0-based or "left-open, right-closed", which means bed interval "chr1    10    20" only contains 10 bases on chr1, from the 11th to the 20th.
In addition, submitters should also let GDC know the genome build (hg18, hg19 or GRCh38) of their bed files.

5. Is a Target Capture Kit uniquely defined by its Target Region Bed File?  
Not necessarily. Sometimes, users or manufactures may want to augment an existing kit with additional probes, in order to capture more regions or simply improve the quality of existing regions. In the latter case, the bed file stays the same, but it is now a different Target Capture Kit and should be registered separately as described in Step 3 above.

## Specifying Tumor Normal Pairs for Analysis

It is critical for many cancer bioinformatics pipelines to specify which normal sample to use to factor out germline variation.  In particular, this is a necessary specification for all tumor normal paired variant calling pipelines.  The following details describe how the GDC determines which normal sample to use for variant calling.

*  Every tumor aliquot will be used for variant calling.  For example, if 10 WXS tumor aliquots are submitted, the GDC will produce 10 alignments and 10 VCFs for each variant calling pipeline.
*  If there is only one normal we will use that normal for variant calling
*  If there are multiple normals of the same experimental_strategy for a case:
    *  Users can specify which normal to use by specifying on the aliquot.  To do so one of the following should be set to `TRUE` for the specified experimental strategy: `selected_normal_low_pass_wgs`, `selected_normal_targeted_sequencing`, `selected_normal_wgs`, or `selected_normal_wxs`.
    *  Or if no normal is specified the GDC will select the best normal for that patient based on the following criteria.  This same logic will also be used if multiple normal are selected.
        * If a case has blood cancer we will use sample type in the following priority order:

            Blood Derived Normal > Bone Marrow Normal > Mononuclear Cells from Bone Marrow Normal > Fibroblasts from Bone Marrow Normal > Lymphoid Normal > Buccal Cell Normal > Solid Tissue Normal > EBV Immortalized Normal

        * If a case does not have blood cancer we will use sample type in the following priority order:

            Solid Tissue Normal > Buccal Cell Normal > Lymphoid Normal > Fibroblasts from Bone Marrow Normal > Mononuclear Cells from Bone Marrow Normal > Bone Marrow Normal > Blood Derived Normal > EBV Immortalized Normal

        * If there are still ties, we will choose the aliquot submitted first.
* If there are no normals.
    * The GDC will not run tumor only variant calling pipeline by default.  The submitter must specify one of the following properties as TRUE: `no_matched_normal_low_pass_wgs`, `no_matched_normal_targeted_sequencing`, `no_matched_normal_wgs`, `no_matched_normal_wxs`.

Note that we will only run variant calling for a particular tumor aliquot per experimental strategy once. You must make sure that the appropriate normal control is uploaded to the GDC when Requesting Submission. Uploading a different normal sample later will not result in reanalysis by the GDC.

## Submission of Single-Cell RNA-Seq Data

For any submitter that is uploading scRNA-Seq data, please follow these guidelines:

* If the data is single-nuclei RNA-Seq, please populate the associated aliquot field `analyte_type` with `Nuclei RNA`.
* Please only submit the molecular files as `submitted_unaligned_reads` in FASTQ format.
* When submitting molecular files, please submit files with file names that are in the acceptable format for CellRanger input. The acceptable format follows this regular expression: SampleName_S[\d*]\_L00[\d]\_R{1,2}\_001.fastq.gz, for example: `SampleName_S1_L001_R1_001.fastq.gz`

## Clinical Data Requirements

For the GDC to release a project there is a minimum number of clinical properties that are required.  Minimal cross-project GDC requirements include age, gender, and diagnosis information.  Other requirements may be added when the submitter is approved for submission to the GDC.

## miRNA Submission

The GDC requires that miRNA reads be adapter-trimmed before being uploaded to the GDC because miRNA datasets can have different trimming schemas. Uploading untrimmed miRNA reads will result in unusably low miRNA quantifications.  

## Slide Image Submission

To submit slide images to GDC, it is a requirement that images should not contain any label/captions as well as no macro views for compliance with patient information confidentiality.



# Pre-release Data Portal Release Notes


## Release 1.0

* __GDC Product__: GDC Pre-Release Data Portal
* __Release Date__: July 12, 2018

### New Features and Changes

*  Support for group based Authorization on top of dbGaP for all API and portal features
*  Admin Portal for managing membership of groups
*  Updated UI focused on repository capabilities

### Bugs Fixed Since Last Release

*  None

### Known Issues and Workarounds

*  Cannot log into AWG Portal on Internet Explorer.  The workaround is to use Edge. <!--SV-1133-->
*  The Project facet on the case panel will show access to more projects than the user has access to.  The user cannot see any further details of those project however. <!--PRTL-2194-->
*  Can not delete users/projects from group if the group has more than 5 users/projects <!--PRTL-2217-->
*  Logging into AWG portal without project credentials causes the portal to continuously refreshes <!--SV-1129-->


# Data Submission Portal Release Notes

| Version | Date |
|---|---|
| [v2.6.0](Data_Submission_Portal_Release_Notes.md#release-260) | July 8, 2022 |
| [v2.5.1](Data_Submission_Portal_Release_Notes.md#release-251) | August 14, 2020 |
| [v2.5.0](Data_Submission_Portal_Release_Notes.md#release-250) | July 2, 2020 |
| [v2.4.1](Data_Submission_Portal_Release_Notes.md#release-241) | March 9, 2020 |
| [v2.4.0](Data_Submission_Portal_Release_Notes.md#release-240) | November 6, 2019 |
| [v2.3.0](Data_Submission_Portal_Release_Notes.md#release-230) | June 5, 2019 |
| [v2.2.0](Data_Submission_Portal_Release_Notes.md#release-220) | February 20, 2019 |
| [v2.1.0](Data_Submission_Portal_Release_Notes.md#release-210) | November 7, 2018 |
| [v2.0.0](Data_Submission_Portal_Release_Notes.md#release-200) | August 23, 2018 |
| [v1.9.0](Data_Submission_Portal_Release_Notes.md#release-190) | May 21, 2018 |
| [v1.8.0](Data_Submission_Portal_Release_Notes.md#release-180) | February 15, 2018 |
| [v1.7.0](Data_Submission_Portal_Release_Notes.md#release-170) | November 16, 2017 |
| [v1.6.0](Data_Submission_Portal_Release_Notes.md#release-160) | August 22, 2017 |
| [v1.5.1](Data_Submission_Portal_Release_Notes.md#release-151) | March 16, 2017 |
| [v1.3.0](Data_Submission_Portal_Release_Notes.md#release-130) | October 31, 2016 |
| [v1.2.2](Data_Submission_Portal_Release_Notes.md#release-122) | September 23, 2016 |
| [v1.1.0](Data_Submission_Portal_Release_Notes.md#release-110) | May 20th, 2016 |
| [v0.3.24.1](Data_Submission_Portal_Release_Notes.md#release-03241) | February 26, 2016 |
| [v0.3.21](Data_Submission_Portal_Release_Notes.md#release-0321) | January 27, 2016 |
| [v0.2.18.3](Data_Submission_Portal_Release_Notes.md#release-02183) | November 30, 2015 |

---
## Release 2.6.0

* __GDC Product__: GDC Data Submission Portal
* __Release Date__: July 8, 2022

### New Features and Changes

*  Text on the QC tab was changed to clarify which errors require attention from the submitter. <!--DEV-747-->

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

*  If a project ID has a character that is not alphanumeric, a dash, or an underscore, submission portal users may experience errors.
*  When creating entities in the Submission Portal, occasionally an extra transaction will appear with status error. This does not seem to impact that actual transaction, which is recorded as occurring successfully.
<!--API-219-->


## Release 2.5.1

* __GDC Product__: GDC Data Submission Portal
* __Release Date__: August 14, 2020

### New Features and Changes

*  Enhancements were made to the submission API to increase performance and usability.

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

*  When creating entities in the Submission Portal, occasionally an extra transaction will appear with status error. This does not seem to impact that actual transaction, which is recorded as occurring successfully.
<!--API-219-->

## Release 2.5.0

* __GDC Product__: GDC Data Submission Portal
* __Release Date__: July 2, 2020

### New Features and Changes

*  None.

### Bugs Fixed Since Last Release

* Fixed bug where the Details pane in the QC Report was displaying crowded, non-uniform buttons. <!--SUBP-605-->

### Known Issues and Workarounds

*  When creating entities in the Submission Portal, occasionally an extra transaction will appear with status error. This does not seem to impact that actual transaction, which is recorded as occurring successfully.
<!--API-219-->

## Release 2.4.1

* __GDC Product__: GDC Data Submission Portal
* __Release Date__: March 9, 2020

### New Features and Changes

*  Removed duplicate queries from various pages in the Submission Portal to optimize data retrieval and rendering. <!--SUBP-584-->

### Bugs Fixed Since Last Release

* Fixed bug where the right-hand detail pane in the Transactions and QC Report tabs was being cut off and not scrollable in the viewport for Windows environments (all browsers). <!--SUBP-594-->
* Fixed bug in the PDF file downloaded from the QC Report tab's Project Summary, where text was being cut off when browsing in Firefox or Microsoft Edge. <!--SUBP-588-->
* Fixed bug where the TSV and JSON download buttons completely disappear and cannot be scrolled to in the Project Data Download modal, if it is shrunk beyond a certain threshold. <!--SUBP-585-->
* Fixed bug in the Manifest download button that was trying to capture certain incorrect or unnecessary file states. <!--SUBP-570-->
* Fixed incorrect DTT hyperlink in the GDC Apps menu. <!--SUBP-568-->
* Fixed bug where the banner warning users that ERA Commons login was currently not working, would only appear after the user logged in, thus defeating the purpose of the warning in the first place. <!--SUBP-545-->

### Known Issues and Workarounds

*  When creating entities in the Submission Portal, occasionally an extra transaction will appear with status error. This does not seem to impact that actual transaction, which is recorded as occurring successfully.
<!--API-219-->

## Release 2.4.0

* __GDC Product__: GDC Data Submission Portal
* __Release Date__: November 6, 2019

### New Features and Changes

*  Added new QC Report tab that allows users to view and download QC errors detected on the current set of unsubmitted data.  Users must examine these errors and fix them appropriately before re-uploading the data and requesting harmonization.  New donut added to the Dashboard tab to display a quick breakdown of CRITICAL vs WARNING errors across the project.<!--FEAT-545-->

### Bugs Fixed Since Last Release

*  Fixed the Project Data download button in the Project Overview, so that the JSON option is selectable in the modal.<!--SUBP-519-->
*  Fixed the trash can icon for the Delete button in the Submitter Detail pane, so that it is fully visible and no longer cutoff<!--SUBP-540-->
*  Fixed a Section 508 Accessibility violation in the Submitter Detail pane.<!--SUBP-566-->
*  Increased the global transaction polling interval to 15 seconds across the portal to improve performance.<!--SUBP-571-->

### Known Issues and Workarounds

*  When creating entities in the Submission Portal, occasionally an extra transaction will appear with status error. This does not seem to impact that actual transaction, which is recorded as occurring successfully.
<!--API-219-->

## Release 2.3.0

* __GDC Product__: GDC Data Submission Portal
* __Release Date__: June 5, 2019

### New Features and Changes

*  Added Aligned Reads, Gene Expressions, miRNA Expression to the Harmonized Data Files list in the Browse tab. <!--SUBP-505-->

### Bugs Fixed Since Last Release

*  Fixed logic on the Submittable Data Files donut to more accurately display number of submittable files that have been validated.  Specifically, both the file state and corresponding note state must be validated.  Also updated the corresponding tooltip text.<!--SUBP-510-->

### Known Issues and Workarounds

*  When creating entities in the Submission Portal, occasionally an extra transaction will appear with status error. This does not seem to impact that actual transaction, which is recorded as occurring successfully.
<!--API-219-->

## Release 2.2.0

* __GDC Product__: GDC Data Submission Portal
* __Release Date__: February 20, 2019

### New Features and Changes

* Renamed the "Request Submission" button to "Request Harmonization" to make the purpose of this action more clear. <!--SUBP-500-->

### Bugs Fixed Since Last Release

*  Fixed the right scroll bar in the records list on the Browse page so that it works in Firefox. <!--SUBP-493-->
*  Fixed a dead link to the Submission Portal User Guide on the Dashboard. <!--SUBP-494-->

### Known Issues and Workarounds

*  When creating entities in the Submission Portal, occasionally an extra transaction will appear with status error. This does not seem to impact that actual transaction, which is recorded as occurring successfully.
<!--API-219-->

## Release 2.1.0

* __GDC Product__: GDC Data Submission Portal
* __Release Date__: November 7, 2018

### New Features and Changes

* Updated the project columns to include a Release column in addition to the Batch Submit column. <!--SUBP-487-->

### Bugs Fixed Since Last Release

*  Fixed quick search so that projects with a dash in the name will no longer break the search. <!--SUBP-465-->
*  PO reports will now return the latest data for each project that has completed running. <!--TT-826-->

### Known Issues and Workarounds

*  When creating entities in the Submission Portal, occasionally an extra transaction will appear with status error. This does not seem to impact that actual transaction, which is recorded as occurring successfully.
<!--API-219-->

## Release 2.0.0

* __GDC Product__: GDC Data Submission Portal
* __Release Date__: August 23, 2018

### New Features and Changes

* The submission process has been updated to request submission and release. <!--SUBP-443-->
   * Users can "Request Submission" once data has been reviewed.  Previously the button that said "Submit" now says "Request Submission".
   * Users can "Request Release" once data has processed by the GDC.  Previously the button that said "Release" now says "Request Release".

### Bugs Fixed Since Last Release

*  Fixed Download All Manifest button being much larger than other buttons. <!--SUBP-461-->
*  Fixed bug where all tables said Showing x of xx projects instead of the correct entity. <!--SUBP-462-->

### Known Issues and Workarounds

*  When creating entities in the Submission Portal, occasionally an extra transaction will appear with status error. This does not seem to impact that actual transaction, which is recorded as occurring successfully.
<!--API-219-->


## Release 1.9.0

* __GDC Product__: GDC Data Submission Portal
* __Release Date__: May 21, 2018

### New Features and Changes

*  Added the ability to download all case metadata in the Browse Cases view <!--SUBP-443-->

### Bugs Fixed Since Last Release

*  Time outs when loading submission portal project list <!--SUBP-407-->
*  Missing PO reports for CPTAC-3 project <!--SUBP-407-->
*  Windows - Scroll bar interfering with browser scroll bar <!--SUBP-424-->
*  Donut "Cases with submittable data files" always shows 0 <!--SUBP-425-->
*  For Biospecimen entities, the "Download all" button does not take filtering into account <!--SUBP-423-->
*  For Clinical entities, the "Download all" button was downloading all clinical entities instead of selected clinical type <!--SUBP-442-->

### Known Issues and Workarounds

*  When creating entities in the Submission Portal, occasionally an extra transaction will appear with status error. This does not seem to impact that actual transaction, which is recorded as occurring successfully.
<!--API-219-->

## Release 1.8.0

* __GDC Product__: GDC Data Submission Portal
* __Release Date__: February 15, 2018

### New Features and Changes

*  Added the ability to support banners with hyperlinks <!--SUBP-411-->

### Bugs Fixed Since Last Release

*  Fixed 508 compliance issues <!--SUBP-408-->

### Known Issues and Workarounds

*  When creating entities in the Submission Portal, occasionally an extra transaction will appear with status error. This does not seem to impact that actual transaction, which is recorded as occurring successfully.
<!--API-219-->

## Release 1.7.0

* __GDC Product__: GDC Data Submission Portal
* __Release Date__: November 16, 2017

### New Features and Changes

*  None

### Bugs Fixed Since Last Release

*  Fixed bug where error would be produced even while project was successfully submitted <!--SUBP-394-->

### Known Issues and Workarounds

*  When creating entities in the Submission Portal, occasionally an extra transaction will appear with status error. This does not seem to impact that actual transaction, which is recorded as occurring successfully.
<!--API-219-->

## Release 1.6.0

* __GDC Product__: GDC Data Submission Portal
* __Release Date__: August 22, 2017

### New Features and Changes

*	 Added ability to see metadata for particular harmonized data files in the Submission Portal <!--SUBP-393-->

### Bugs Fixed Since Last Release

None

### Known Issues and Workarounds

*  When creating entities in the Submission Portal, occasionally an extra transaction will appear with status error. This does not seem to impact that actual transaction, which is recorded as occurring successfully.
<!--API-219-->


## Release 1.5.1

* __GDC Product__: GDC Data Submission Portal
* __Release Date__: March 16, 2017

### New Features and Changes

*	 Added ability to delete an entity.  Read more about this [here](https://docs.gdc.cancer.gov/Data_Submission_Portal/Users_Guide/Data_Submission_Walkthrough/#deleting-submitted-entities) <!-- SUBP-144 -->
*  Added Project Reports in the projects list page.  Read more about this [here](https://docs.gdc.cancer.gov/Data_Submission_Portal/Users_Guide/Data_Submission_Process/#reports). <!-- SUBP-281, DAT-286, DAT-287, DAT-289 -->
*  To avoid confusion, renamed "Status" to "State" in the Browse section <!-- SUBP-237 -->
*  Added tooltip over Hierarchy title when reviewing an entity <!-- SUBP-238 -->
*  Restrict the upload window to only supported data formats (JSON and TSV) <!-- SUBP-245 -->

### Bugs Fixed Since Last Release

*  File status column should not be displayed for any clinical or biospecimen entities but only for submittable data files. <!--SUBP-226-->
*	Diagnosis / Treatment detail: Submitter ID (of the child / parent) is missing in the Details -> Hierarchy view. <!--SUBP-227-->
*	In some situations tooltip entries remain on-screen. Workaround is to refresh the page. <!--SUBP-229-->
*	In Browse tab, "Submittable Data Files" filter, clicking on "Download All" currently returns case and clinical informations instead of returning file informations. Workaround is to download information from the file the details panel. <!--SUBP-230-->
*	In Dashboard, the donut chart for number of cases with submittable data files is always empty. A workaround is to visit the Browse, detailed case view section to see, case by case, if it has submittable data files.<!--SUBP-231 / SUBP-156-->
*	In Transactions tab, after clicking on Commit or Discard, status is not automatically refreshed. <!--SUBP-232-->
*  Added the API version in the Data Submission Portal footer on the project list page. <!--SUBP-235-->
*  Inconsistent behavior when clicking on a Transaction ID on the Dashboard. <!--SUBP-239-->
*  Empty transactions created when submitting files in an incorrect format. <!--SUBP-242-->
*  JSON file downloaded from the Data Submission Portal cannot be used to resubmit data. <!--SUBP-243-->
*  "Submitted data files" donut chart and "Download Manifest" button do not get refreshed after committing a transaction. <!--SUBP-248-->
*  Release information on the Dashboard creates confusion. <!--SUBP-252-->
*  Missing Boolean fields from details panel. <!--SUBP-254-->
*  No message displayed if no results are found via the top menu search. <!--SUBP-256-->
*  "Invalid Date" on IE11 and Firefox ESR 45.x <!--SUBP-257-->
*  Download option truncated in the details panel. <!--SUBP-258-->
*  Download All from the browse section returns too many records. <!--SUBP-266-->

### Known Issues and Workarounds

*  When creating entities in the Submission Portal, occasionally an extra transaction will appear with status error. This does not seem to impact that actual transaction, which is recorded as occurring successfully.
<!--API-219-->

## Release 1.3.0

* __GDC Product__: GDC Data Submission Portal
* __Release Date__: October 31, 2016

### New Features and Changes

Not Applicable

### Bugs Fixed Since Last Release

*	Adding a call to the backend to ensure a refreshed token is being downloaded when user clicks on "Download Token". <!-- SUBP-170 -->
*	Fixed an issue with some buttons not working in Firefox <!-- SUBP-60 -->
*	Disabled Download Clinical button if the file has no clinical data <!-- SUBP-173 -->
*	Disabled Download Manifest button while the page is loading  <!-- SUBP-206 -->
* 	Fixed an issue with some dropdowns being cut off <!-- SUBP-212 / SUBP-214 -->

### Known Issues and Workarounds

*	Project submission and release is currently disabled. <!--SUBP-201-->
*	Diagnosis / Treatment detail: Submitter ID (of the child / parent) is missing in the Details -> Hierarchy view. <!--SUBP-227-->
* File status column should not be displayed for any clinical or biospecimen entities but only for submittable data files. <!--SUBP-226-->
*	In some situations tooltip entries remain on-screen. Workaround is to refresh the page. <!--SUBP-229-->
*	In Browse tab, "Submittable Data Files" filter, clicking on "Download All" currently returns case and clinical informations instead of returning file informations. Workaround is to download information from the file the details panel. <!--SUBP-230-->
*	In Dashboard, the donut chart for number of cases with submittable data files is always empty. A workaround is to visit the Browse, detailed case view section to see, case by case, if it has submittable data files.<!--SUBP-231 / SUBP-156-->
*	In Transactions tab, after clicking on Commit or Discard, status is not automatically refreshed. Workaround is to refresh the page after clicking on Commit or Discard. This does not affect the transaction section of the project dashboard. <!--SUBP-232-->
*  Reports are currently not available in the Data Submission Portal and will be added back in an upcoming version:
    *   Data Validation Report: The rows in the report are sometimes duplicated and #Files in error are not showing up in the report. The user should go to Project > Browse > Submitted Files to see the files in error and the error type<!--PGDC-1025 and PGDC-997-->.
    *   The Scientific Pre-alignment QC Report is not available.

## Release 1.2.2

* __GDC Product__: GDC Data Submission Portal
* __Release Date__: September 23, 2016

### New Features and Changes

This version contains major improvements to the GDC Data Submission Portal in both usability, performance and reliability.

Some known issues and workarounds listed in previous release notes have been made redundant due to this refactoring effort, thus are not listed anymore.

Please refer to the GDC Data Submission Portal User's Guide for more details about the features.

* Submission-related actions have been made Asynchronous.
* Fully revamped the dashboard layout and features to clarify the submission process and give easier access to key features.
* Created a transactions list page with options to take actions on transactions (in particular committing an upload)
* Improved performance of the Browse tab.
* Added GDC Apps to the header section.

### Bugs Fixed Since Last Release

* Data submitted to the project can be downloaded from each project page by clicking on "PROJECT DATA" from the project page.<!--PGDC-774-->
* When uploading multiple files at once, validation will fail if a child entity is listed before its parent. <!--PGDC-861-->
* In Browse > Case > Details, Experimental Data (renamed to Submittable Data Files) are not listed in "Related Entities" section. <!--PGDC-850-->
* In the upload report, the number of affected cases is incorrect (show 0) when entities are created. <!--PGDC-838-->


### Known Issues and Workarounds

*	Project submission and release is currently disabled. <!--SUBP-201-->
*	If case has no clinical data, the "Download Clinical" button is not disabled. The downloaded TSV will not contain Clinical Data. <!--SUBP-173-->
*	Download Manifest button is available while page is loading or when no files are in "registered" state. Clicking on Download will return a file with an error message. <!--SUBP-206-->
*	In Internet Explorer, GDC APPs and File dropdown are incorrectly aligned, making some elements only partially visible. <!--SUBP-212--><!--SUBP-214-->
*	Diagnosis / Treatment detail: Submitter ID (of the child / parent) is missing in the Details -> Hierarchy view. <!--SUBP-227-->
* File status column should not be displayed for any clinical or biospecimen entities but only for submittable data files. <!--SUBP-226-->
*	In some situations tooltip entries remain on-screen. Workaround is to refresh the page. <!--SUBP-229-->
*	In Browse tab, "Submittable Data Files" filter, clicking on "Download All" currently returns case and clinical informations instead of returning file informations. Workaround is to download information from the file the details panel. <!--SUBP-230-->
*	In Dashboard, the donut chart for number of cases with submittable data files is always empty. A workaround is to visit the Browse, detailed case view section to see, case by case, if it has submittable data files.<!--SUBP-231-->
*	In Transactions tab, after clicking on Commit or Discard, status is not automatically refreshed. Workaround is to refresh the page after clicking on Commit or Discard. This does not affect the transaction section of the project dashboard. <!--SUBP-232-->
*  Reports are currently not available in the Data Submission Portal and will be added back in an upcoming version:
    *   Data Validation Report: The rows in the report are sometimes duplicated and #Files in error are not showing up in the report. The user should go to Project > Browse > Submitted Files to see the files in error and the error type<!--PGDC-1025 and PGDC-997-->.
    *   The Scientific Pre-alignment QC Report is not available.

## Release 1.1.0

* __GDC Product__: GDC Data Submission Portal
* __Release Date__: May 20th, 2016

### New Features and Changes

* Updated login text

### Bugs Fixed Since Last Release

* Improved 508 compliance of the landing page

### Known Issues and Workarounds

*	Some actions on the data submission portal are resource intensive and might result in timeout or errors. The team is currently working on a major update to the data submission portal expected to be available towards the end of the summer. This version will address performance issues and improve overall user experience.

## Release 0.3.24.1

* __GDC Product__: GDC Data Submission Portal
* __Release Date__: February 26, 2016

### New Features and Changes

* Fully revamped dashboard, improved widgets with more details.
* Improved the release process and provided guidance.
* Removed dictionary viewer (moved to Documentation site).
* Added reports.

### Bugs Fixed Since Last Release

* Issues with transactions list page have been addressed.
* Search feature has been improved.
* Release processes (submit and release) have been finalized and implemented.

### Known Issues and Workarounds

* In Browse > Submitted Files, "DOWNLOAD ALL" downloads cases and clinical data instead of submitted files. A workaround is to initiate the download from the Dashboard<!--PGDC-970-->.
* In Browse > Case > Details, experimental data are not listed in "Related Entities" section<!--PGDC-850-->.
* When uploading multiple files at once, validation will fail if a child entity is listed before its parent. A workaround is to ensure parent entities are listed before their children in the Upload wizard<!--PGDC-861-->.
* Submission status column is inconsistent between Submitted Files and Read Group. Submission Status says "Validated" when it should say "Submitted" when users submit data to the GDC<!--PGDC-959-->.
* In the upload report, the number of affected cases is incorrect (show 0) when entities are created<!--PGDC-838-->.
* In Dashboard > Release, Download of submitted data returns data from the project workspace but not from snapshot (submitted files)<!--PGDC-774-->.
* In Browse > Read Groups, the data completeness property is incorrect when multiple files are in the bundle<!--PGDC-916-->.
* In Browse > Diagnosis/Treatment > Details, the hierarchy section is missing elements<!--PGDC-1023-->.
* Data Validation Report: The rows in the report are sometimes duplicated and #Files in error are not showing up in the report. The user should go to Project > Browse > Submitted Files to see the files in error and the error type<!--PGDC-1025 and PGDC-997-->.
* The Scientific Pre-alignment QC Report is not available.
* The Submission Portal is not meant to support XML file submission, users have to submit files through the GDC API.


## Release 0.3.21

* __GDC Product__: GDC Data Submission Portal
* __Release Date__: January 27, 2016

### New Features and Changes

*   New landing page.
*   Updated dashboard with new widgets, updated instructions and added a transactions list.
*   Improved user experience in the browse section of the portal (new layout, revamped detailed section, updated icons, etc.).

### Bugs Fixed Since Last Release

*   Some sections of the GDC Data Submission Portal are not compatible with Internet Explorer 10 and 11 (Note: support has been dropped for IE 10).
*   Case count coverage report is currently not available. Case overview report is available from the landing page.
*   If a page does not return data, the Submission Portal does not indicate that there is no data.
*   The projects dropdown show legacy projects. It should only show projects active for submission.

### Known Issues and Workarounds

*   Users may experience issues (e.g., unable to access a specific transaction ID) when navigating the transactions list page. The workaround is to refresh the page.
*   Search feature partially implemented, some items are not searchable.
*   Implementation of Submit and Release not finalized.
*   XML file submission is returning a bad request error. The workaround is to submit files through the GDC API.
*   Data Bundles - Lane Level Sequence: read group ID not unique, the message generated is not user friendly.

## Release 0.2.18.3

* __GDC Product__: GDC Data Submission Portal
* __Release Date__: November 30, 2015

### New Features and Changes

*   Dashboard to provide high-level details about one or more projects.
*   Submission wizard to guide user through a three stage submission:
    *   Upload file to the GDC.
    *   Validate uploaded files.
    *   Submit validated files to the GDC.
*   Support submission of JSON, TSV and XML files.
*   Tables and reports to identify key elements of the submission:
    *   List of Cases, Samples, Portions, Analytes, Aliquots, Lane Level Sequence.
    *   Cases missing clinical data.
    *   Cases missing samples.
    *   Aliquots missing data bundles.
    *   Submitted data bundle and associated QC metrics.
    *   Recent and all transactions.
    *   Case count coverage.
*   Manifest to facilitate submission of molecular data via the GDC Data Transfer Tool.
*   Per-project dictionary viewer.
*   Download authentication token.

### Bugs Fixed Since Last Release

*   Initial Release - Not Applicable

### Known Issues and Workarounds

*   Some sections of the GDC Data Submission Portal are not compatible with Internet Explorer 10 and 11.
*   Case count coverage report is currently not available.
*   Unable to use the search feature (implementation of this feature is not complete).
*   If the page does not return data, the Submission Portal does not mention that there is no data.
*   Case is not releasable when submitting a partial lane level seq data bundle.
    <u>Note</u>: This feature is currently being revised.
*   XML file submission is returning bad request error.
    A workaround is to submit files through the API.
*   Data Bundles - Lane Level Sequence: read group ID not unique, message generated is not user friendly.
*   The projects dropdown show legacy projects, it should only show projects active for submission.


# Data Analysis

The GDC data analysis endpoints allow API users to programmatically explore data in the GDC using advanced filters at a gene and mutation level. Survival analysis data is also available.  

## Endpoints

The following data analysis endpoints are available from the GDC API:

|__Node__| __Endpoint__ | __Description__ |
|---|---|---|
|__Genes__| __/genes__ | Allows users to access summary information about each gene using its Ensembl ID. |
|__Gene Expression__|__/gene_expression/availability__|Allows users to retrieve the availability of gene expression data for specific cases and/or genes.|
||__/gene_expression/values__|Get gene expression values for specified cases and genes.|
||__/gene_expression/gene_selection__|Select the most variably expressed genes for a collection of cases and genes.|
|__SSMS__| __/ssms__ | Allows users to access information about each somatic mutation. For example, a `ssm` would represent the transition of C to T at position 52000 of chromosome 1. |
||__/ssms/`<ssm_id>`__|Get information about a specific ssm using a `<ssm_id>`, often supplemented with the `expand` option to show fields of interest. |
|| __/ssm_occurrences__ | A `ssm` entity as applied to a single instance (case). An example of a `ssm occurrence` would be that the transition of C to T at position 52000 of chromosome 1 occurred in patient TCGA-XX-XXXX. |
||__/ssm_occurrences/`<ssm_occurrences_id>`__|Get information about a specific ssm occurrence using a `<ssm_occurrences_id>`, often supplemented with the `expand` option to show fields of interest. |
|__CNVS__|__/cnvs__|Allows users to access data about copy number variations (cnvs). This data will be specifc to cnvs and not a specific case. |
||__/cnvs/`<cnv_id>`__|Get information about a specific copy number variation using a `<cnv_id>`, often supplemented with the `expand` option to show fields of interest. |
||__/cnvs/ids__|This endpoint will retrieve nodes that contain the queried cnv_id. This is accomplished by adding the query parameter: /cnvs/ids?query=`<cnv_id>`.|
||__/cnv_occurrences__|A `cnv` entity as applied to a single case.|
||__/cnv_occurrences/`<cnv_occurrence_id>`__|Get information about a specific copy number variation occurrence using a `<cnv_occurrence_id>`, often supplemented with the `expand` option to show fields of interest. |
||__/cnv_occurrences/ids__|This endpoint will retrieve nodes that contain the queried cnv_occurrence_id. This is accomplished by adding the query parameter: /cnv_occurrences/ids?query=`<cnv_occurrences_id>`|
|__Analysis__|__/analysis/top_cases_counts_by_genes__| Returns the number of cases with a mutation in each gene listed in the gene_ids parameter for each project. Note that this endpoint cannot be used with the `format` or `fields` parameters.|
||__/analysis/top_mutated_genes_by_project__| Returns a list of genes that have the most mutations within a given project. |
||__/analysis/top_mutated_cases_by_gene__| Generates information about the cases that are most affected by mutations in a given number of genes |
||__/analysis/mutated_cases_count_by_project__| Returns counts for the number of cases that have associated `ssm` data in each project. The number of affected cases can be found under "case_with_ssm": {"doc_count": $case_count}.|
||__/analysis/survival__| Survival plots can be generated in the Data Portal for different subsets of data, based upon many query factors such as variants, disease type and projects. This endpoint can be used to programmatically retrieve the raw data to generate these plots and apply different filters to the data. (see Survival Example)|


The methods for retrieving information from these endpoints are very similar to those used for the `cases` and `files` endpoints. These methods are explored in depth in the [API Search and Retrieval](https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/) documentation. The `_mapping` parameter can also be used with each of these endpoints to generate a list of potential fields.  For example:

`https://api.gdc.cancer.gov/ssms/_mapping`

While it is not an endpoint, the `observation` entity is featured in the visualization section of the API. The `observation` entity provides information from the MAF file, such as read depth and normal genotype, that supports the validity of the associated `ssm`. An example is demonstrated below:

=== "Shell"

    ```Shell
    curl "https://api.gdc.cancer.gov/ssms/57bb3f2e-ec05-52c2-ab02-7065b7d24849?expand=occurrence.case.observation.read_depth&pretty=true"
    ```

=== "Response"

    ```json
    {
      "data": {
        "ncbi_build": "GRCh38",
        "occurrence": [
          {
            "case": {
              "observation": [
                {
                  "read_depth": {
                    "t_ref_count": 321,
                    "t_alt_count": 14,
                    "t_depth": 335,
                    "n_depth": 115
                  }
                }
              ]
            }
          }
        ],
        "tumor_allele": "G",
        "mutation_type": "Simple Somatic Mutation",
        "end_position": 14304578,
        "reference_allele": "C",
        "ssm_id": "57bb3f2e-ec05-52c2-ab02-7065b7d24849",
        "start_position": 14304578,
        "mutation_subtype": "Single base substitution",
        "cosmic_id": null,
        "genomic_dna_change": "chr5:g.14304578C>G",
        "gene_aa_change": [
          "TRIO L229V",
          "TRIO L437V",
          "TRIO L447V",
          "TRIO L496V"
        ],
        "chromosome": "chr5"
      },
      "warnings": {}
    }
    ```

## Genes Endpoint Examples

__Example 1:__ A user would like to access information about the gene `ZMPSTE24`, which has an Ensembl gene ID of `ENSG00000084073`.  This would be accomplished by appending `ENSG00000084073` (`gene_id`) to the `genes` endpoint.

=== "Shell"

    ```Shell
    curl "https://api.gdc.cancer.gov/genes/ENSG00000084073?pretty=true"
    ```

=== "Response"

    ```json
    {
      "data": {
        "canonical_transcript_length": 3108,
        "description": "This gene encodes a member of the peptidase M48A family. The encoded protein is a zinc metalloproteinase involved in the two step post-translational proteolytic cleavage of carboxy terminal residues of farnesylated prelamin A to form mature lamin A. Mutations in this gene have been associated with mandibuloacral dysplasia and restrictive dermopathy. [provided by RefSeq, Jul 2008]",
        "cytoband": [
          "1p34.2"
        ],
        "gene_start": 40258107,
        "canonical_transcript_length_genomic": 36078,
        "gene_id": "ENSG00000084073",
        "gene_strand": 1,
        "canonical_transcript_length_cds": 1425,
        "gene_chromosome": "1",
        "synonyms": [
          "FACE-1",
          "HGPS",
          "PRO1",
          "STE24",
          "Ste24p"
        ],
        "is_cancer_gene_census": null,
        "biotype": "protein_coding",
        "gene_end": 40294184,
        "canonical_transcript_id": "ENST00000372759",
        "symbol": "ZMPSTE24",
        "name": "zinc metallopeptidase STE24"
      },
      "warnings": {}
    }
    ```

__Example 2:__ A user wants a subset of elements such as a list of coordinates for all genes on chromosome 7. The query can be filtered for only results from chromosome 7 using a JSON-formatted query that is URL-encoded.

=== "Shell"

    ```Shell
    curl "https://api.gdc.cancer.gov/genes?pretty=true&fields=gene_id,symbol,gene_start,gene_end&format=tsv&size=2000&filters=%7B%0D%0A%22op%22%3A%22in%22%2C%0D%0A%22content%22%3A%7B%0D%0A%22field%22%3A%22gene_chromosome%22%2C%0D%0A%22value%22%3A%5B%0D%0A%227%22%0D%0A%5D%0D%0A%7D%0D%0A%7D"
    ```

=== "Response"

    ```
    gene_start      gene_end        symbol  id
    28995231        29195451        CPVL    ENSG00000106066
    33014114        33062797        NT5C3A  ENSG00000122643
    143052320       143053347       OR6V1   ENSG00000225781
    100400826       100428992       ZCWPW1  ENSG00000078487
    73861159        73865893        WBSCR28 ENSG00000175877
    64862999        64864370        EEF1DP4 ENSG00000213640
    159231435       159233377       PIP5K1P2        ENSG00000229435
    141972631       141973773       TAS2R38 ENSG00000257138
    16646131        16706523        BZW2    ENSG00000136261
    149239651       149255609       ZNF212  ENSG00000170260
    57405025        57405090        MIR3147 ENSG00000266168
    130393771       130442433       CEP41   ENSG00000106477
    150800403       150805120       TMEM176A        ENSG00000002933
    93591573        93911265        GNGT1   ENSG00000127928
    117465784       117715971       CFTR    ENSG00000001626
    5879827 5886362 OCM     ENSG00000122543
    144118461       144119360       OR2A15P ENSG00000239981
    30424527        30478784        NOD1    ENSG00000106100
    137227341       137343865       PTN     ENSG00000105894
    84876554        84876956        HMGN2P11        ENSG00000232605
    107470018       107475659       GPR22   ENSG00000172209
    31330711        31330896        RP11-463M14.1   ENSG00000271027
    78017057        79453574        MAGI2   ENSG00000187391
    55736779        55739605        CICP11  ENSG00000237799
    142111749       142222324       RP11-1220K2.2   ENSG00000257743
    (truncated)
    ```

## Gene Expression Examples

### Gene Expression Availability Endpoint

The purpose of this endpoint is to retrieve the availability of gene expression data for cases, genes, or both. The availability response informs the user if gene expression data exists for each case or gene, which are specified with case and gene IDs. Gene expression data is only available for protein-coding genes.

__Example 1__: A user wants to get the availability of gene expression data for a set of cases and genes.

=== "Filter"

    ```json
    {
      "case_ids": [
        "6d4f38db-a97b-4dc0-8dc5-2ac7f2cc5e38",
        "e3b32485-b204-43a7-93a5-601408fcdf96"
      ],
      "gene_ids": [
        "ENSG00000141510",
        "ENSG00000181143"
      ]
    }
    ```

=== "Shell"

    ```json
    curl -X 'POST' \
      'https://api.gdc.cancer.gov/gene_expression/availability' \
      -H 'accept: application/json' \
      -H 'Content-Type: application/json' \
      -d '{
      "case_ids": [
        "6d4f38db-a97b-4dc0-8dc5-2ac7f2cc5e38",
        "e3b32485-b204-43a7-93a5-601408fcdf96"
      ],
      "gene_ids": [
        "ENSG00000141510",
        "ENSG00000181143"
      ]
    }'
    ```

=== "Response"

    ```json
    {
        "cases": {
            "details": [
                {
                    "case_id": "6d4f38db-a97b-4dc0-8dc5-2ac7f2cc5e38",
                    "has_gene_expression_values": false
                },
                {
                    "case_id": "e3b32485-b204-43a7-93a5-601408fcdf96",
                    "has_gene_expression_values": true
                }
            ],
            "with_gene_expression_count": 1,
            "without_gene_expression_count": 1
        },
        "genes": {
            "details": [
                {
                    "gene_id": "ENSG00000141510",
                    "has_gene_expression_values": true
                },
                {
                    "gene_id": "ENSG00000181143",
                    "has_gene_expression_values": true
                }
            ],
            "with_gene_expression_count": 2,
            "without_gene_expression_count": 0
        }
    }
    ```

### Gene Expression Values Endpoint

The purpose of this endpoint is to retrieve the gene expression values for the given cases and genes. The response is a TSV containing the expression values for genes to cases.
The `tsv_units` of gene expression data must be defined by exactly one of the following:

* `uqfpkm` - FPKM-UQ values. More information on calculations can be found [here](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/#calculations).
* `median_centered_log2_uqfpkm` - Median-centered log2(FPKM-UQ+1) values.

The `median_centered_log2_uqfpkm` is calculated through the following steps:

1. Calculate the Median: Determine the median of all provided log2(uqfpkm + 1) values.
1. Compute Median-Centered Values: Subtract the median from each log2(uqfpkm + 1) value.
1. Generate the Result Sequence: Create a new sequence with the median-centered values, preserving the original order.

__Example 1__: A user wants to get expression values using case IDs and gene IDs.

=== "Filter"

    ```json
    {
        "case_ids": [
            "6d4f38db-a97b-4dc0-8dc5-2ac7f2cc5e38",
            "e3b32485-b204-43a7-93a5-601408fcdf96",
            "000ead0d-abf5-4606-be04-1ea31b999840",
            "001ab32d-f924-4753-ad67-4366fb845ae6"
        ],
        "gene_ids": [
            "ENSG00000141510",
            "ENSG00000181143"
        ],
        "tsv_units": "median_centered_log2_uqfpkm",
        "format": "tsv"
    }
    ```

=== "Shell"

    ```shell
    curl -X 'POST' \
      'https://api.gdc.cancer.gov/gene_expression/values' \
      -H 'accept: text/tab-separated-values' \
      -H 'Content-Type: application/json' \
      -d '{
        "case_ids": [
            "6d4f38db-a97b-4dc0-8dc5-2ac7f2cc5e38",
            "e3b32485-b204-43a7-93a5-601408fcdf96",
            "000ead0d-abf5-4606-be04-1ea31b999840",
            "001ab32d-f924-4753-ad67-4366fb845ae6"
        ],
        "gene_ids": [
            "ENSG00000141510",
            "ENSG00000181143"
        ],
        "tsv_units": "median_centered_log2_uqfpkm",
        "format": "tsv"
    }'
    ```

=== "Response"

    ```
    gene_id	000ead0d-abf5-4606-be04-1ea31b999840	001ab32d-f924-4753-ad67-4366fb845ae6	e3b32485-b204-43a7-93a5-601408fcdf96
    ENSG00000141510	-0.58248	1.75830	0.00000
    ENSG00000181143	-0.02529	0.00000	3.52293
    ```

### Gene Expression Gene Selection Endpoint

Select the most variably expressed genes for a collection of cases and collection of genes. The request must define a collection of cases, a collection of genes, and a selection size. A minimum expression value may optionally be defined.

A collection of cases must be defined by case IDs.

A collection of genes must be defined by exactly one of the following:

* `gene_ids`
* `gene_type` which has only one value: `protein_coding`.

A selection size (`selection_size`) defines the maximum number of genes to select.

An optional threshold (`min_median_log2_uqfpkm`) defines a minimum value for expression. Defaults to `1`.

__Example 1__: A user wants to get the most variably expressed genes for a list of case UUIDs and a list of Ensembl gene IDs.

=== "Filter"

    ```json
    {
        "case_ids": [
            "000ead0d-abf5-4606-be04-1ea31b999840",
            "001ab32d-f924-4753-ad67-4366fb845ae6",
            "0024c94c-88ff-49d9-8dc4-bf77f832d85e",
            "003f4f85-3244-4132-8c9d-c29f09382269",
            "005d0639-c923-470f-a179-02a4dbb5cdf2",
            "006931bb-f5b1-4aa4-b0a8-af517a912db0",
            "0084e8b6-57fc-48b6-aa77-fec6e45161d2",
            "008d3744-e7f0-41a5-a419-702960cf1ccb",
            "0094e07c-1595-402e-9d38-68b9cac71e7b",
            "00bd58bd-223d-433e-b60a-5bf355f342b1"
        ],
        "gene_ids": [
            "ENSG00000141510",
            "ENSG00000181143"
        ],
        "selection_size": 1
    }
    ```

=== "Shell"

    ```shell
    curl -X 'POST' \
      'https://api.gdc.cancer.gov/gene_expression/gene_selection' \
      -H 'accept: application/json' \
      -H 'Content-Type: application/json' \
      -d '{
        "case_ids": [
            "000ead0d-abf5-4606-be04-1ea31b999840",
            "001ab32d-f924-4753-ad67-4366fb845ae6",
            "0024c94c-88ff-49d9-8dc4-bf77f832d85e",
            "003f4f85-3244-4132-8c9d-c29f09382269",
            "005d0639-c923-470f-a179-02a4dbb5cdf2",
            "006931bb-f5b1-4aa4-b0a8-af517a912db0",
            "0084e8b6-57fc-48b6-aa77-fec6e45161d2",
            "008d3744-e7f0-41a5-a419-702960cf1ccb",
            "0094e07c-1595-402e-9d38-68b9cac71e7b",
            "00bd58bd-223d-433e-b60a-5bf355f342b1"
        ],
        "gene_ids": [
            "ENSG00000141510",
            "ENSG00000181143"
        ],
        "selection_size": 1
    }'
    ```

=== "Response"

    ```json
    {
        "gene_selection": [
            {
                "log2_uqfpkm_stddev": 0.9962971125913709,
                "log2_uqfpkm_median": 2.904457107848132,
                "gene_id": "ENSG00000141510",
                "symbol": "TP53"
            }
        ]
    }
    ```

## Simple Somatic Mutation Endpoint Examples

__Example 1__: Similar to the `/genes` endpoint, a user would like to retrieve information about the mutation based on its COSMIC ID. This would be accomplished by creating a JSON filter, which will then be encoded to URL for the `curl` command.

=== "Filter"

    ```json
     {
       "op":"in",
       "content":{
          "field":"cosmic_id",
          "value":[
             "COSM1135366"
          ]
       }
    }
    ```

=== "Shell"

    ```Shell
    curl 'https://api.gdc.cancer.gov/ssms?pretty=true&filters=%7B%0A%22op%22%3A%22in%22%2C%0A%22content%22%3A%7B%0A%22field%22%3A%22cosmic_id%22%2C%0A%2value%22%3A%5B%0A%22COSM1135366%22%0A%5D%0A%7D%0A%7D%0A'
    ```

=== "Response"

    ```json
    {
      "data": {
        "hits": [
          {
            "id": "edd1ae2c-3ca9-52bd-a124-b09ed304fcc2",
            "start_position": 25245350,
            "gene_aa_change": [
              "KRAS G12D"
            ],
            "reference_allele": "C",
            "ncbi_build": "GRCh38",
            "cosmic_id": [
              "COSM1135366",
              "COSM521"
            ],
            "mutation_subtype": "Single base substitution",
            "mutation_type": "Simple Somatic Mutation",
            "chromosome": "chr12",
            "ssm_id": "edd1ae2c-3ca9-52bd-a124-b09ed304fcc2",
            "genomic_dna_change": "chr12:g.25245350C>T",
            "tumor_allele": "T",
            "end_position": 25245350
          }
        ],
        "pagination": {
          "count": 1,
          "total": 1,
          "size": 10,
          "from": 0,
          "sort": "",
          "page": 1,
          "pages": 1
        }
      },
      "warnings": {}
    }
    ```

__Example 2:__ Based on the previous example's `ssm_id` (`8b3c1a7a-e4e0-5200-9d46-5767c2982145`), a user would like to look at the consequences and the VEP impact due to this ssm.

=== "Shell"

    ```Shell
    curl 'https://api.gdc.cancer.gov/ssms/edd1ae2c-3ca9-52bd-a124-b09ed304fcc2?pretty=true&expand=consequence.transcript&fields=consequence.transcript.annotation.vep_impact'
    ```

=== "JSON"

    ```JSON
    {
      "data": {
        "consequence": [
          {
            "transcript": {
              "annotation": {
                "vep_impact": "MODERATE"
              },
              "transcript_id": "ENST00000557334",
              "aa_end": 12,
              "consequence_type": "missense_variant",
              "aa_start": 12,
              "is_canonical": false,
              "aa_change": "G12D",
              "ref_seq_accession": ""
            }
          },
          {
            "transcript": {
              "annotation": {
                "vep_impact": "MODERATE"
              },
              "transcript_id": "ENST00000256078",
              "aa_end": 12,
              "consequence_type": "missense_variant",
              "aa_start": 12,
              "is_canonical": true,
              "aa_change": "G12D",
              "ref_seq_accession": "NM_001369786.1&NM_033360.4"
            }
          },
          {
            "transcript": {
              "annotation": {
                "vep_impact": "MODERATE"
              },
              "transcript_id": "ENST00000311936",
              "aa_end": 12,
              "consequence_type": "missense_variant",
              "aa_start": 12,
              "is_canonical": false,
              "aa_change": "G12D",
              "ref_seq_accession": "NM_001369787.1&NM_004985.5"
            }
          },
          {
            "transcript": {
              "annotation": {
                "vep_impact": "MODERATE"
              },
              "transcript_id": "ENST00000556131",
              "aa_end": 12,
              "consequence_type": "missense_variant",
              "aa_start": 12,
              "is_canonical": false,
              "aa_change": "G12D",
              "ref_seq_accession": ""
            }
          }
        ]
      },
      "warnings": {}
    }
    ```

## Simple Somatic Mutation Occurrence Endpoint Examples

__Example 1:__ A user wants to determine the chromosome in case `TCGA-DU-6407` that contains the greatest number of `ssms`. As this relates to mutations that are observed in a case, the `ssm_occurrences` endpoint is used.

=== "Filter"

    ```json
    {  
       "op":"in",
       "content":{  
          "field":"case.submitter_id",
          "value":["TCGA-DU-6407"]
       }
    }
    ```

=== "Shell"

    ```Shell
    curl "https://api.gdc.cancer.gov/ssm_occurrences?format=tsv&fields=ssm.chromosome&size=5000&filters=%7B%0D%0A%22op%22%3A%22in%22%2C%0D%0A%22content%22%3A%7B%0D%0A%22field%22%3A%22case.submitter_id%22%2C%0D%0A%22value%22%3A%5B%0D%0A%22TCGA-DU-6407%22%0D%0A%5D%0D%0A%7D%0D%0A%7D"
    ```

=== "Tsv"

    ```tsv
    id	ssm.chromosome
    105e7811-4601-5ccb-ae93-e7107923599e	chr16
    faee73a9-4804-58ea-a91f-18c3d901774f	chr2
    99b3aad4-d368-506d-99d6-047cbe5dff0f	chr2
    2cb06277-993e-5502-b2c5-263037c45d18	chr10
    f08dcc53-eadc-5ceb-bf31-f6b38629e4cb	chr19
    97c5b38b-fc96-57f5-8517-cc702b3aa70a	chr6
    19ca262d-b354-54a0-b582-c4719e37e91d	chrX
    b4822fc9-f0cc-56fd-9d97-f916234e309d	chr2
    22a07c7c-16ba-51df-a9a9-1e41e2a45225	chrX
    0010a89d-9434-5d97-8672-36ee394767d0	chr17
    3a023e72-da92-54f7-aa18-502c1076b2b0	chr5
    391011ff-c1fd-5e2a-a128-652bc660f64c	chr10
    3548ecfe-5186-51e7-8f40-37f4654cd260	chr2
    b67f31b5-0341-518e-8fcc-811cd2e36af1	chr3
    4a93d7a5-988d-5055-80da-999dc3b45d80	chr1
    9dc3f7cd-9efa-530a-8524-30d067e49d54	chr13
    552c09d1-69b1-5c04-b543-524a6feae3eb	chr3
    dbc5eafa-ea26-5f1c-946c-b6974a345b69	chr12
    d25129ad-3ad7-584f-bdeb-fba5c3881d32	chr17
    1378cbc4-af88-55bb-b2e5-185bb4246d7a	chr10
    c44a93a1-5c73-5cff-b40e-98ce7e5fe57b	chr19
    1267330b-ae6d-5e25-b19e-34e98523679e	chr21
    1476a543-2951-5ec4-b165-67551b47d810	chr17
    727c9d57-7b74-556f-aa5b-e1ca1f76d119	chr10
    94abd5fd-d539-5a4a-8719-9615cf7cec5d	chr1
    a76469cb-973c-5d4d-bf82-7cf4e8f6c129	chr17
    ```

__Example 2:__ A user has retrieved a `ssm_occurrence`, and would like to determine if that case also has diagnostic information.

=== "Shell"

    ```Shell
    curl 'https://api.gdc.cancer.gov/ssm_occurrences/6fd8527d-5c40-5604-8fa9-0ce798eec231?pretty=true&expand=case.diagnoses'
    ```

=== "Json"

    ```Json
    {
      "data": {
        "ssm_occurrence_id": "6fd8527d-5c40-5604-8fa9-0ce798eec231",
        "case": {
          "diagnoses": [
            {
              "ajcc_pathologic_t": "T3b",
              "synchronous_malignancy": "No",
              "morphology": "8720/3",
              "ajcc_pathologic_stage": "Stage IIB",
              "ajcc_pathologic_n": "N0",
              "ajcc_pathologic_m": "M0",
              "submitter_id": "TCGA-Z2-A8RT_diagnosis",
              "days_to_diagnosis": 0,
              "last_known_disease_status": "not reported",
              "tissue_or_organ_of_origin": "Skin, NOS",
              "days_to_last_follow_up": 839.0,
              "age_at_diagnosis": 15342,
              "primary_diagnosis": "Malignant melanoma, NOS",
              "classification_of_tumor": "not reported",
              "prior_malignancy": "no",
              "year_of_diagnosis": 2012,
              "diagnosis_id": "1d06a202-c51a-52e2-805f-eeb5f7fac14e",
              "icd_10_code": "C44.6",
              "site_of_resection_or_biopsy": "Skin of upper limb and shoulder",
              "prior_treatment": "No",
              "state": "released",
              "tumor_grade": "Not Reported",
              "progression_or_recurrence": "not reported",
              "ajcc_staging_system_edition": "7th"
            }
          ]
        }
      },
      "warnings": {}
    }
    ```

## Copy Number Variation Endpoint Examples

__Example 1:__ A user is interested in finding the first 30 cnvs found on chromosome 4 that have a cnv loss.

=== "Filter"

    ```json
    {
        "op": "and",
        "content": [
            {
                "op": "in",
                "content": {
                    "field": "chromosome",
                    "value": [
                        "4"
                    ]
                }
            },
            {
                "op": "in",
                "content": {
                    "field": "cnv_change",
                    "value": [
                        "Loss"
                    ]
                }
            }
        ]
    }
    ```

=== "Shell"

    ```Shell
    curl 'https://api.gdc.cancer.gov/cnvs?filters=%7B%0D%0A+++%22op%22%3A+%22and%22%2C%0D%0A++++%22content%22%3A+%5B%0D%0A++++++++%7B%0D%0A++++++++++++%22op%22%3A+%22in%22%2C%0D%0A++++++++++++%22content%22%3A+%7B%0D%0A++++++++++++++++%22field%22%3A+%22chromosome%22%2C%0D%0A++++++++++++++++%22value%22%3A+%5B%0D%0A++++++++++++++++++++%224%22%0D%0A++++++++++++++++%5D%0D%0A++++++++++++%7D%0D%0A++++++++%7D%2C%0D%0A++++++++%7B%0D%0A++++++++++++%22op%22%3A+%22in%22%2C%0D%0A++++++++++++%22content%22%3A+%7B%0D%0A++++++++++++++++%22field%22%3A+%22cnv_change%22%2C%0D%0A++++++++++++++++%22value%22%3A+%5B%0D%0A++++++++++++++++++++%22Loss%22%0D%0A++++++++++++++++%5D%0D%0A++++++++++++%7D%0D%0A++++++++%7D%0D%0A++++%5D%0D%0A%7D&size=30&sort=start_position&format=tsv'
    ```

=== "Tsv"

    ```tsv
    chromosome	cnv_change	cnv_id	end_position	gene_level_cn	id	ncbi_build	start_position
    4	Loss	11381600-f064-5c42-90d2-a5c79c8b23e1	88208	True	11381600-f064-5c42-90d2-a5c79c8b23e1	GRCh38	53286
    4	Loss	edef0f2f-c1a7-507c-842f-e1f8a568df9d	202303	True	edef0f2f-c1a7-507c-842f-e1f8a568df9d	GRCh38	124501
    4	Loss	eba92f9a-b045-54a8-948a-451e439ed418	305474	True	eba92f9a-b045-54a8-948a-451e439ed418	GRCh38	270675
    4	Loss	89319453-2a3f-5ebe-be30-8af0426e0343	384868	True	89319453-2a3f-5ebe-be30-8af0426e0343	GRCh38	337814
    4	Loss	6567929c-4b6f-582b-aedf-acde2c0ec736	499156	True	6567929c-4b6f-582b-aedf-acde2c0ec736	GRCh38	425815
    4	Loss	2daff58b-5065-50cd-8239-253180eaee81	540200	True	2daff58b-5065-50cd-8239-253180eaee81	GRCh38	499210
    4	Loss	2b42c8d4-6d85-5352-96e1-9e52e722c248	576295	True	2b42c8d4-6d85-5352-96e1-9e52e722c248	GRCh38	573880
    4	Loss	2646cdc7-7602-59a4-ae4f-d171352bae88	670782	True	2646cdc7-7602-59a4-ae4f-d171352bae88	GRCh38	625573
    4	Loss	c11ad392-949f-593f-a3ab-d834b2f82809	674330	True	c11ad392-949f-593f-a3ab-d834b2f82809	GRCh38	672436
    4	Loss	f31be658-4de0-549e-81be-e79759879acf	682033	True	f31be658-4de0-549e-81be-e79759879acf	GRCh38	673580
    4	Loss	d72c62f2-fc29-5b83-9839-7f6b03970aff	689271	True	d72c62f2-fc29-5b83-9839-7f6b03970aff	GRCh38	681829
    4	Loss	45448d47-6e13-5d30-824d-96150a7f55c6	770640	True	45448d47-6e13-5d30-824d-96150a7f55c6	GRCh38	705748
    4	Loss	517e65ea-9084-54c2-abe0-b1b47e9f872c	826129	True	517e65ea-9084-54c2-abe0-b1b47e9f872c	GRCh38	784957
    4	Loss	b5a09c9b-d842-5b76-a500-56f18252c29d	932373	True	b5a09c9b-d842-5b76-a500-56f18252c29d	GRCh38	849276
    4	Loss	e3a3b61d-2881-5ad4-90bf-58ef29ae9ecb	958656	True	e3a3b61d-2881-5ad4-90bf-58ef29ae9ecb	GRCh38	932387
    4	Loss	8630a1b6-3215-5b71-903a-ad9845505afc	986895	True	8630a1b6-3215-5b71-903a-ad9845505afc	GRCh38	958887
    4	Loss	f748b06f-1fb7-53a9-a7d6-2c22a3ae6de5	993440	True	f748b06f-1fb7-53a9-a7d6-2c22a3ae6de5	GRCh38	979073
    4	Loss	a5e4a63f-c5f6-5f0f-a6b6-f51bfb643533	1004564	True	a5e4a63f-c5f6-5f0f-a6b6-f51bfb643533	GRCh38	986997
    4	Loss	73f6fbbe-6fd9-524c-a7c8-a7cf3f08ada4	1026898	True	73f6fbbe-6fd9-524c-a7c8-a7cf3f08ada4	GRCh38	1009936
    4	Loss	adad579a-b002-5022-823a-570c59549065	1113564	True	adad579a-b002-5022-823a-570c59549065	GRCh38	1056250
    4	Loss	d5a5c45e-594b-5cbc-97d5-75fc5155d021	1208962	True	d5a5c45e-594b-5cbc-97d5-75fc5155d021	GRCh38	1166932
    4	Loss	6c910993-faa8-5abc-b433-b3afcc5e9e11	1249953	True	6c910993-faa8-5abc-b433-b3afcc5e9e11	GRCh38	1211445
    4	Loss	4453b4cb-7d8a-5e26-a856-eac62eec287a	1340147	True	4453b4cb-7d8a-5e26-a856-eac62eec287a	GRCh38	1289887
    4	Loss	6db1001a-a41b-518d-9491-2bf41544d90f	1395989	True	6db1001a-a41b-518d-9491-2bf41544d90f	GRCh38	1345691
    4	Loss	6bef981a-ead1-5aa7-8a69-8d38e576e5c0	1406442	True	6bef981a-ead1-5aa7-8a69-8d38e576e5c0	GRCh38	1402932
    4	Loss	af6e0b49-922a-587e-b353-4b9414605cf1	1684261	True	af6e0b49-922a-587e-b353-4b9414605cf1	GRCh38	1617915
    4	Loss	400352ad-8526-562a-bbf4-29b90a48f46f	1712344	True	400352ad-8526-562a-bbf4-29b90a48f46f	GRCh38	1692731
    4	Loss	8811414d-2434-56c6-afe5-a998c9b18d47	1745171	True	8811414d-2434-56c6-afe5-a998c9b18d47	GRCh38	1712891
    4	Loss	169c4409-0256-5841-9314-f1a4dd2bcc38	1721358	True	169c4409-0256-5841-9314-f1a4dd2bcc38	GRCh38	1715952
    4	Loss	1712ccac-6e70-5fb3-b71e-1a029eaf047c	1808872	True	1712ccac-6e70-5fb3-b71e-1a029eaf047c	GRCh38	1793293
    ```

__Example 2:__ A user wants to determine the location and identity of the gene affected by the cnv `544c4896-0152-5787-8d77-894a16f0ded0`, and determine whether the gene is found within the Cancer Gene Census.

=== "Shell"

    ```Shell
    curl 'https://api.gdc.cancer.gov/cnvs/544c4896-0152-5787-8d77-894a16f0ded0?pretty=true&expand=consequence.gene'
    ```

=== "Json" 

    ```Json
    {
      "data": {
        "start_position": 27100354,
        "consequence": [
          {
            "gene": {
              "biotype": "protein_coding",
              "symbol": "HOXA2",
              "gene_id": "ENSG00000105996"
            }
          }
        ],
        "gene_level_cn": true,
        "cnv_change": "Gain",
        "ncbi_build": "GRCh38",
        "chromosome": "7",
        "cnv_id": "544c4896-0152-5787-8d77-894a16f0ded0",
        "end_position": 27102686
      },
      "warnings": {}
    }
    ```

## Copy Number Variation Occurrence Enpoint Examples

__Example 1:__ A user is interested in finding cases that have both cnv and ssm data for females diagnosed with Squamous Cell Neoplasms and have a cnv gain change on chromosome 9. It is important to note that for a case like this, where multiple arguments are need for one filtered field, it is easier for the API to have multiple filters for the same field, `case.available_variation_data` in this example, than having one filter with multiple arguments.

=== "Filter"

    ```json
    {
        "op": "and",
        "content": [
            {
                "op": "in",
                "content": {
                    "field": "cnv.cnv_change",
                    "value": [
                        "Gain"
                    ]
                }
            },
            {
                "op": "in",
                "content": {
                    "field": "case.demographic.gender",
                    "value": [
                        "female"
                    ]
                }
            },
            {
                "op": "in",
                "content": {
                    "field": "case.available_variation_data",
                    "value": [
                        "cnv"
                    ]
                }
            },
            {
                "op": "in",
                "content": {
                    "field": "case.available_variation_data",
                    "value": [
                        "ssm"
                    ]
                }
            },
            {
                "op": "in",
                "content": {
                    "field": "cnv.chromosome",
                    "value": [
                        "9"
                    ]
                }
            },
            {
                "op": "in",
                "content": {
                    "field": "case.disease_type",
                    "value": [
                        "Squamous Cell Neoplasms"
                    ]
                }
            }
        ]
    }

    ```
=== "Shell"

    ```shell
    curl 'https://api.gdc.cancer.gov/cnv_occurrences?filters=%7B%0D%0A++++%22op%22%3A+%22and%22%2C%0D%0A++++%22content%22%3A+%5B%0D%0A++++++++%7B%0D%0A++++++++++++%22op%22%3A+%22in%22%2C%0D%0A++++++++++++%22content%22%3A+%7B%0D%0A++++++++++++++++%22field%22%3A+%22cnv.cnv_change%22%2C%0D%0A++++++++++++++++%22value%22%3A+%5B%0D%0A++++++++++++++++++++%22Gain%22%0D%0A++++++++++++++++%5D%0D%0A++++++++++++%7D%0D%0A++++++++%7D%2C%0D%0A++++++++%7B%0D%0A++++++++++++%22op%22%3A+%22in%22%2C%0D%0A++++++++++++%22content%22%3A+%7B%0D%0A++++++++++++++++%22field%22%3A+%22case.demographic.gender%22%2C%0D%0A++++++++++++++++%22value%22%3A+%5B%0D%0A++++++++++++++++++++%22female%22%0D%0A++++++++++++++++%5D%0D%0A++++++++++++%7D%0D%0A++++++++%7D%2C%0D%0A++++++++%7B%0D%0A++++++++++++%22op%22%3A+%22in%22%2C%0D%0A++++++++++++%22content%22%3A+%7B%0D%0A++++++++++++++++%22field%22%3A+%22case.available_variation_data%22%2C%0D%0A++++++++++++++++%22value%22%3A+%5B%0D%0A++++++++++++++++++++%22cnv%22%0D%0A++++++++++++++++%5D%0D%0A++++++++++++%7D%0D%0A++++++++%7D%2C%0D%0A++++++++%7B%0D%0A++++++++++++%22op%22%3A+%22in%22%2C%0D%0A++++++++++++%22content%22%3A+%7B%0D%0A++++++++++++++++%22field%22%3A+%22case.available_variation_data%22%2C%0D%0A++++++++++++++++%22value%22%3A+%5B%0D%0A++++++++++++++++++++%22ssm%22%0D%0A++++++++++++++++%5D%0D%0A++++++++++++%7D%0D%0A++++++++%7D%2C%0D%0A++++++++%7B%0D%0A++++++++++++%22op%22%3A+%22in%22%2C%0D%0A++++++++++++%22content%22%3A+%7B%0D%0A++++++++++++++++%22field%22%3A+%22cnv.chromosome%22%2C%0D%0A++++++++++++++++%22value%22%3A+%5B%0D%0A++++++++++++++++++++%229%22%0D%0A++++++++++++++++%5D%0D%0A++++++++++++%7D%0D%0A++++++++%7D%2C%0D%0A++++++++%7B%0D%0A++++++++++++%22op%22%3A+%22in%22%2C%0D%0A++++++++++++%22content%22%3A+%7B%0D%0A++++++++++++++++%22field%22%3A+%22case.disease_type%22%2C%0D%0A++++++++++++++++%22value%22%3A+%5B%0D%0A++++++++++++++++++++%22Squamous+Cell+Neoplasms%22%0D%0A++++++++++++++++%5D%0D%0A++++++++++++%7D%0D%0A++++++++%7D%0D%0A++++%5D%0D%0A%7D&fields=case.available_variation_data,case.case_id&format=tsv'
    ```

=== "Tsv"

    ```tsv
    case.available_variation_data.0	case.available_variation_data.1	case.case_id	id
    cnv	ssm	da30a845-c4d3-4c78-b8b0-210239224f8f	3caf6e3b-024f-57b6-bdd9-3b67e423cc11
    cnv	ssm	0809ba8b-4ab6-4f43-934c-c1ccbc014a7e	e6afe58e-c99c-5c8d-920e-8ba4daad4d89
    cnv	ssm	8e0e456e-85ee-4de5-8f0b-72393d6acde0	9d983d9c-8320-53f1-9054-e46926c5b834
    cnv	ssm	64a195f6-2212-4e81-bccc-e39c77a10908	8caeaecc-ad68-539d-8b3c-8320b3684763
    cnv	ssm	2f6a0e87-1e6c-41f3-93e0-3e505fa654b0	4862c166-0f37-5e3c-ae4e-a2964de01cea
    cnv	ssm	f0daf315-8909-4cda-886d-a2770b08db94	099ff6cd-bd28-56f4-a181-6b02f3ba7503
    cnv	ssm	ff3808e4-eece-4046-819b-fe1019317f8e	0c936aa2-393e-5463-a431-3613b4510021
    cnv	ssm	9205dc07-93f5-4b5e-924e-8e097616160f	133d27a7-fdc6-5082-a1f3-022b89f4e851
    cnv	ssm	79ae5209-f476-4d65-a6c0-ebc18d7c8942	7a5e6bb1-8af3-5964-a3cc-c53602c8b099
    cnv	ssm	ff7099e1-8ff9-48e4-842d-46e98076e7e6	fb27fa8f-aa31-5e20-84da-8f45bb675405
    ```

__Example 2:__ A user is interested in the first cnv occurrence (`3b9f7ecc-2280-5b89-80f9-ec8d6c5e604e`) from the previous example, and would like to know more about the case exposures and demographics.

=== "Shell"

    ```Shell
    curl 'https://api.gdc.cancer.gov/cnv_occurrences/3b9f7ecc-2280-5b89-80f9-ec8d6c5e604e?pretty=true&expand=cnv,case,case.exposures,case.demographic'
    ```

=== "Json"

    ```Json
    {
      "data": {
        "cnv": {
          "start_position": 68815994,
          "gene_level_cn": true,
          "cnv_change": "Gain",
          "ncbi_build": "GRCh38",
          "chromosome": "4",
          "variant_status": "Tumor Only",
          "cnv_id": "1a889109-30d5-51e3-848f-9f615c69f407",
          "end_position": 68832023
        },
        "cnv_occurrence_id": "3b9f7ecc-2280-5b89-80f9-ec8d6c5e604e",
        "case": {
          "exposures": [
            {
              "cigarettes_per_day": 5.47945205479452,
              "alcohol_history": "Not Reported",
              "exposure_id": "f7b08a8e-d22b-5cb0-be9f-b922c9ca87d2",
              "submitter_id": "TCGA-38-4629_exposure",
              "state": "released",
              "pack_years_smoked": 100.0
            }
          ],
          "primary_site": "Bronchus and lung",
          "disease_type": "Adenomas and Adenocarcinomas",
          "available_variation_data": [
            "cnv",
            "ssm"
          ],
          "case_id": "127bf818-f7e5-46b5-a9de-39f6d96b8b83",
          "submitter_id": "TCGA-38-4629",
          "state": "released",
          "demographic": {
            "demographic_id": "9ea1f795-9510-5acc-a9a5-bf1379e6635a",
            "ethnicity": "not hispanic or latino",
            "gender": "male",
            "race": "white",
            "vital_status": "Dead",
            "age_at_index": 68,
            "submitter_id": "TCGA-38-4629_demographic",
            "days_to_death": 864,
            "days_to_birth": -25104,
            "state": "released",
            "year_of_death": 2005,
            "year_of_birth": 1935
          }
        }
      },
      "warnings": {}
    }
    ```

## Analysis Endpoints

In addition to the `ssms`, `ssm_occurrences`, and `genes` endpoints mentioned previously, several `/analysis` endpoints were designed to quickly retrieve specific datasets used for visualization display.  

__Example 1:__ The `/analysis/top_cases_counts_by_genes` endpoint gives the number of cases with a mutation in each gene listed in the `gene_ids` parameter for each project. Note that this endpoint cannot be used with the `format` or `fields` parameters. In this instance, the query will produce the number of cases in each projects with mutations in the gene `ENSG00000155657`.

```Shell
curl "https://api.gdc.cancer.gov/analysis/top_cases_counts_by_genes?gene_ids=ENSG00000155657&pretty=true"
```


This JSON-formatted output is broken up by project. For an example, see the following text:

```json
$ curl "https://api.gdc.cancer.gov/analysis/top_cases_counts_by_genes?gene_ids=ENSG00000155657&pretty=true"
{
  "took": 6,
  "timed_out": false,
  "_shards": {
    "total": 12,
    "successful": 12,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": {
      "value": 5967,
      "relation": "eq"
    },
    "max_score": null,
    "hits": []
  },
  "aggregations": {
    "projects": {
      "doc_count_error_upper_bound": 0,
      "sum_other_doc_count": 0,
      "buckets": [
        {
          "key": "TCGA-BRCA",
          "doc_count": 425,
          "genes": {
            "doc_count": 4031450,
            "my_genes": {
              "doc_count": 425,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 425
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-LUSC",
          "doc_count": 423,
          "genes": {
            "doc_count": 4123089,
            "my_genes": {
              "doc_count": 423,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 423
                  }
                ]
              }
            }
          }
        },
        {
          "key": "CPTAC-3",
          "doc_count": 421,
          "genes": {
            "doc_count": 251552,
            "my_genes": {
              "doc_count": 421,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 421
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-SKCM",
          "doc_count": 391,
          "genes": {
            "doc_count": 3040929,
            "my_genes": {
              "doc_count": 391,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 391
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-LUAD",
          "doc_count": 345,
          "genes": {
            "doc_count": 3188761,
            "my_genes": {
              "doc_count": 345,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 345
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-OV",
          "doc_count": 341,
          "genes": {
            "doc_count": 3728561,
            "my_genes": {
              "doc_count": 341,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 341
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-STAD",
          "doc_count": 300,
          "genes": {
            "doc_count": 2145783,
            "my_genes": {
              "doc_count": 300,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 300
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-UCEC",
          "doc_count": 297,
          "genes": {
            "doc_count": 1637055,
            "my_genes": {
              "doc_count": 297,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 297
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-HNSC",
          "doc_count": 293,
          "genes": {
            "doc_count": 2325617,
            "my_genes": {
              "doc_count": 293,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 293
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-COAD",
          "doc_count": 288,
          "genes": {
            "doc_count": 1695280,
            "my_genes": {
              "doc_count": 288,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 288
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-BLCA",
          "doc_count": 280,
          "genes": {
            "doc_count": 2466835,
            "my_genes": {
              "doc_count": 280,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 280
                  }
                ]
              }
            }
          }
        },
        {
          "key": "MMRF-COMMPASS",
          "doc_count": 181,
          "genes": {
            "doc_count": 45977,
            "my_genes": {
              "doc_count": 181,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 181
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-LIHC",
          "doc_count": 167,
          "genes": {
            "doc_count": 1216775,
            "my_genes": {
              "doc_count": 167,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 167
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-CESC",
          "doc_count": 161,
          "genes": {
            "doc_count": 1103281,
            "my_genes": {
              "doc_count": 161,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 161
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-KIRC",
          "doc_count": 161,
          "genes": {
            "doc_count": 842546,
            "my_genes": {
              "doc_count": 161,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 161
                  }
                ]
              }
            }
          }
        },
        {
          "key": "CPTAC-2",
          "doc_count": 131,
          "genes": {
            "doc_count": 72575,
            "my_genes": {
              "doc_count": 131,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 131
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-GBM",
          "doc_count": 131,
          "genes": {
            "doc_count": 756809,
            "my_genes": {
              "doc_count": 131,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 131
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-ESCA",
          "doc_count": 129,
          "genes": {
            "doc_count": 1210888,
            "my_genes": {
              "doc_count": 129,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 129
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-PRAD",
          "doc_count": 101,
          "genes": {
            "doc_count": 379949,
            "my_genes": {
              "doc_count": 101,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 101
                  }
                ]
              }
            }
          }
        },
        {
          "key": "HCMI-CMDC",
          "doc_count": 99,
          "genes": {
            "doc_count": 54829,
            "my_genes": {
              "doc_count": 99,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 99
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-READ",
          "doc_count": 98,
          "genes": {
            "doc_count": 726313,
            "my_genes": {
              "doc_count": 98,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 98
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-LGG",
          "doc_count": 95,
          "genes": {
            "doc_count": 424689,
            "my_genes": {
              "doc_count": 95,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 95
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-KIRP",
          "doc_count": 93,
          "genes": {
            "doc_count": 521936,
            "my_genes": {
              "doc_count": 93,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 93
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-SARC",
          "doc_count": 93,
          "genes": {
            "doc_count": 903111,
            "my_genes": {
              "doc_count": 93,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 93
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-TGCT",
          "doc_count": 51,
          "genes": {
            "doc_count": 524456,
            "my_genes": {
              "doc_count": 51,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 51
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TARGET-ALL-P2",
          "doc_count": 50,
          "genes": {
            "doc_count": 1882,
            "my_genes": {
              "doc_count": 50,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 50
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-KICH",
          "doc_count": 43,
          "genes": {
            "doc_count": 353674,
            "my_genes": {
              "doc_count": 43,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 43
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-PAAD",
          "doc_count": 43,
          "genes": {
            "doc_count": 300427,
            "my_genes": {
              "doc_count": 43,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 43
                  }
                ]
              }
            }
          }
        },
        {
          "key": "CGCI-HTMCP-CC",
          "doc_count": 37,
          "genes": {
            "doc_count": 3606,
            "my_genes": {
              "doc_count": 37,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 37
                  }
                ]
              }
            }
          }
        },
        {
          "key": "CDDP_EAGLE-1",
          "doc_count": 32,
          "genes": {
            "doc_count": 16980,
            "my_genes": {
              "doc_count": 32,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 32
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-ACC",
          "doc_count": 29,
          "genes": {
            "doc_count": 283969,
            "my_genes": {
              "doc_count": 29,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 29
                  }
                ]
              }
            }
          }
        },
        {
          "key": "CMI-MBC",
          "doc_count": 28,
          "genes": {
            "doc_count": 3581,
            "my_genes": {
              "doc_count": 28,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 28
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-THCA",
          "doc_count": 28,
          "genes": {
            "doc_count": 89120,
            "my_genes": {
              "doc_count": 28,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 28
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-UCS",
          "doc_count": 28,
          "genes": {
            "doc_count": 283673,
            "my_genes": {
              "doc_count": 28,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 28
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-MESO",
          "doc_count": 21,
          "genes": {
            "doc_count": 137002,
            "my_genes": {
              "doc_count": 21,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 21
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-PCPG",
          "doc_count": 19,
          "genes": {
            "doc_count": 99444,
            "my_genes": {
              "doc_count": 19,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 19
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TARGET-NBL",
          "doc_count": 15,
          "genes": {
            "doc_count": 829,
            "my_genes": {
              "doc_count": 15,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 15
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-UVM",
          "doc_count": 12,
          "genes": {
            "doc_count": 68201,
            "my_genes": {
              "doc_count": 12,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 12
                  }
                ]
              }
            }
          }
        },
        {
          "key": "EXCEPTIONAL_RESPONDERS-ER",
          "doc_count": 11,
          "genes": {
            "doc_count": 10617,
            "my_genes": {
              "doc_count": 11,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 11
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-THYM",
          "doc_count": 11,
          "genes": {
            "doc_count": 59647,
            "my_genes": {
              "doc_count": 11,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 11
                  }
                ]
              }
            }
          }
        },
        {
          "key": "BEATAML1.0-COHORT",
          "doc_count": 10,
          "genes": {
            "doc_count": 279,
            "my_genes": {
              "doc_count": 10,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 10
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TARGET-OS",
          "doc_count": 10,
          "genes": {
            "doc_count": 414,
            "my_genes": {
              "doc_count": 10,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 10
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-LAML",
          "doc_count": 10,
          "genes": {
            "doc_count": 10175,
            "my_genes": {
              "doc_count": 10,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 10
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-DLBC",
          "doc_count": 9,
          "genes": {
            "doc_count": 63497,
            "my_genes": {
              "doc_count": 9,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 9
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TCGA-CHOL",
          "doc_count": 8,
          "genes": {
            "doc_count": 52960,
            "my_genes": {
              "doc_count": 8,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 8
                  }
                ]
              }
            }
          }
        },
        {
          "key": "CMI-MPC",
          "doc_count": 7,
          "genes": {
            "doc_count": 365,
            "my_genes": {
              "doc_count": 7,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 7
                  }
                ]
              }
            }
          }
        },
        {
          "key": "CMI-ASC",
          "doc_count": 6,
          "genes": {
            "doc_count": 5745,
            "my_genes": {
              "doc_count": 6,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 6
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TARGET-WT",
          "doc_count": 3,
          "genes": {
            "doc_count": 51,
            "my_genes": {
              "doc_count": 3,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 3
                  }
                ]
              }
            }
          }
        },
        {
          "key": "TARGET-ALL-P3",
          "doc_count": 2,
          "genes": {
            "doc_count": 66,
            "my_genes": {
              "doc_count": 2,
              "gene_id": {
                "doc_count_error_upper_bound": 0,
                "sum_other_doc_count": 0,
                "buckets": [
                  {
                    "key": "ENSG00000155657",
                    "doc_count": 2
                  }
                ]
              }
            }
          }
        }
      ]
    }
  }

```

This portion of the output shows TCGA-GBM including 45 cases that have `ssms` in the gene `ENSG00000155657`.

__Example 2:__ The following demonstrates a use of the `/analysis/top_mutated_genes_by_project` endpoint.  This will output the genes that are mutated in the most cases in "TCGA-DLBC" and will count the mutations that have a `HIGH` or `MODERATE` impact on gene function. Note that the `score` field does not represent the number of mutations in a given gene, but a calculation that is used to determine which genes have the greatest number of unique mutations.  

=== "Json"

    ```json
    {  
       "op":"AND",
       "content":[  
          {  
             "op":"in",
             "content":{  
                "field":"case.project.project_id",
                "value":[  
                   "TCGA-DLBC"
                ]
             }
          },
          {  
             "op":"in",
             "content":{  
                "field":"case.ssm.consequence.transcript.annotation.vep_impact",
                "value":[  
                   "HIGH",
                   "MODERATE"
                ]
             }
          }
       ]
    }
    ```

=== "Shell"

    ```Shell
    curl "https://api.gdc.cancer.gov/analysis/top_mutated_genes_by_project?fields=gene_id,symbol&filters=%7B%20%20%0A%20%20%20%22op%22%3A%22AND%20%20%20%22content%22%3A%5B%20%20%0A%20%20%20%20%20%20%7B%20%20%0A%20%20%20%20%20%20%20%20%20%22op%22%3A%22in%22%2C%0A%20%20%20%20%20%20%20%20%20%22content%22%3A%7B%20%20%0A%20%20%20%20%20%20%20%20%20%20%20%20%22field%22%3A%22case.project.project_id%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%22value%22%3A%5B%20%20%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%22TCGA-DLBC%22%0A%20%20%20%20%20%20%20%20%20%20%20%20%5D%0A%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%7D%2C%0A%20%20%20%20%20%20%7B%20%20%0A%20%20%20%20%20%20%20%20%20%22op%22%3A%22in%22%2C%0A%20%20%20%20%20%20%20%20%20%22content%22%3A%7B%20%20%0A%20%20%20%20%20%20%20%20%20%20%20%20%22field%22%3A%22case.ssm.consequence.transcript.annotation.vep_impact%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%22value%22%3A%5B%20%20%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%22HIGH%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%22MODERATE%22%0A%20%20%20%20%20%20%20%20%20%20%20%20%5D%0A%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%7D%0A%20%20%20%5D%0A%7D%0A&pretty=true"
    ```

=== "Response"

    ```json
    {
      "data": {
        "hits": [
          {
            "symbol": "KMT2D",
            "gene_id": "ENSG00000167548",
            "_score": 13.0
          },
          {
            "symbol": "BTG2",
            "gene_id": "ENSG00000159388",
            "_score": 13.0
          },
          {
            "symbol": "B2M",
            "gene_id": "ENSG00000166710",
            "_score": 11.0
          },
          {
            "symbol": "PIM1",
            "gene_id": "ENSG00000137193",
            "_score": 10.0
          },
          {
            "symbol": "IGHG1",
            "gene_id": "ENSG00000211896",
            "_score": 10.0
          },
          {
            "symbol": "CARD11",
            "gene_id": "ENSG00000198286",
            "_score": 10.0
          },
          {
            "symbol": "H1-4",
            "gene_id": "ENSG00000168298",
            "_score": 9.0
          },
          {
            "symbol": "PCLO",
            "gene_id": "ENSG00000186472",
            "_score": 9.0
          },
          {
            "symbol": "IGHG2",
            "gene_id": "ENSG00000211893",
            "_score": 9.0
          },
          {
            "symbol": "FAT4",
            "gene_id": "ENSG00000196159",
            "_score": 8.0
          }
        ],
        "pagination": {
          "count": 10,
          "total": 3500,
          "size": 10,
          "from": 0,
          "sort": "None",
          "page": 1,
          "pages": 350
        }
      },
      "warnings": {}
    }
    ```

__Example 3:__ The `/analysis/top_mutated_cases_by_gene` endpoint will generate information about the cases that are most affected by mutations in a given number of genes. Below, the file count for each category is given for the cases most affected by mutations in these 50 genes.  The size of the output is limited to two cases with the `size=2` parameter, but a higher value can be set by the user.

=== "Shell"

    ```Shell
    curl "https://api.gdc.cancer.gov/analysis/top_mutated_cases_by_gene?fields=diagnoses.age_at_diagnosis,diagnoses.primary_diagnosis,demographic.gender,demographic.race,demographic.ethnicity,case_id,summary.data_categories.file_count,summary.data_categories.data_category&filters=%7B%22op%22%3A%22and%22%2C%22content%22%3A%5B%7B%22op%22%3A%22%3D%22%2C%22content%22%3A%7B%22field%22%3A%22cases.project.project_id%22%2C%22value%22%3A%22TCGA-DLBC%22%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22genes.gene_id%22%2C%22value%22%3A%5B%22ENSG00000166710%22%2C%22ENSG00000005339%22%2C%22ENSG00000083857%22%2C%22ENSG00000168769%22%2C%22ENSG00000100906%22%2C%22ENSG00000184677%22%2C%22ENSG00000101680%22%2C%22ENSG00000101266%22%2C%22ENSG00000028277%22%2C%22ENSG00000140968%22%2C%22ENSG00000181827%22%2C%22ENSG00000116815%22%2C%22ENSG00000275221%22%2C%22ENSG00000139083%22%2C%22ENSG00000112851%22%2C%22ENSG00000112697%22%2C%22ENSG00000164134%22%2C%22ENSG00000009413%22%2C%22ENSG00000071626%22%2C%22ENSG00000135407%22%2C%22ENSG00000101825%22%2C%22ENSG00000104814%22%2C%22ENSG00000166415%22%2C%22ENSG00000142867%22%2C%22ENSG00000254585%22%2C%22ENSG00000139718%22%2C%22ENSG00000077721%22%2C%22ENSG00000130294%22%2C%22ENSG00000117245%22%2C%22ENSG00000117318%22%2C%22ENSG00000270550%22%2C%22ENSG00000163637%22%2C%22ENSG00000166575%22%2C%22ENSG00000065526%22%2C%22ENSG00000156453%22%2C%22ENSG00000128191%22%2C%22ENSG00000055609%22%2C%22ENSG00000204469%22%2C%22ENSG00000187605%22%2C%22ENSG00000185875%22%2C%22ENSG00000110888%22%2C%22ENSG00000007341%22%2C%22ENSG00000173198%22%2C%22ENSG00000115568%22%2C%22ENSG00000163714%22%2C%22ENSG00000125772%22%2C%22ENSG00000080815%22%2C%22ENSG00000189079%22%2C%22ENSG00000120837%22%2C%22ENSG00000143951%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22ssms.consequence.transcript.annotation.vep_impact%22%2C%22value%22%3A%5B%22HIGH%22%5D%7D%7D%5D%7D&pretty=true&size=2"
    ```

=== "Response"

    ```json
    {
      "data": {
        "hits": [
          {
            "summary": {
              "data_categories": [
                {
                  "file_count": 6,
                  "data_category": "Sequencing Reads"
                },
                {
                  "file_count": 14,
                  "data_category": "Biospecimen"
                },
                {
                  "file_count": 8,
                  "data_category": "Copy Number Variation"
                },
                {
                  "file_count": 16,
                  "data_category": "Simple Nucleotide Variation"
                },
                {
                  "file_count": 4,
                  "data_category": "Transcriptome Profiling"
                },
                {
                  "file_count": 3,
                  "data_category": "DNA Methylation"
                },
                {
                  "file_count": 8,
                  "data_category": "Clinical"
                },
                {
                  "file_count": 4,
                  "data_category": "Structural Variation"
                },
                {
                  "file_count": 1,
                  "data_category": "Proteome Profiling"
                }
              ]
            },
            "case_id": "eda9496e-be80-4a13-bf06-89f0cc9e937f",
            "diagnoses": [
              {
                "age_at_diagnosis": 18691,
                "primary_diagnosis": "Malignant lymphoma, large B-cell, diffuse, NOS"
              }
            ],
            "demographic": {
              "ethnicity": "hispanic or latino",
              "gender": "male",
              "race": "white"
            },
            "_score": 7.0
          },
          {
            "summary": {
              "data_categories": [
                {
                  "file_count": 4,
                  "data_category": "Sequencing Reads"
                },
                {
                  "file_count": 13,
                  "data_category": "Biospecimen"
                },
                {
                  "file_count": 8,
                  "data_category": "Copy Number Variation"
                },
                {
                  "file_count": 16,
                  "data_category": "Simple Nucleotide Variation"
                },
                {
                  "file_count": 2,
                  "data_category": "Transcriptome Profiling"
                },
                {
                  "file_count": 3,
                  "data_category": "DNA Methylation"
                },
                {
                  "file_count": 8,
                  "data_category": "Clinical"
                },
                {
                  "file_count": 4,
                  "data_category": "Structural Variation"
                }
              ]
            },
            "case_id": "7a589441-11ef-4158-87e7-3951d86bc2aa",
            "diagnoses": [
              {
                "age_at_diagnosis": 20812,
                "primary_diagnosis": "Malignant lymphoma, large B-cell, diffuse, NOS"
              }
            ],
            "demographic": {
              "ethnicity": "not hispanic or latino",
              "gender": "female",
              "race": "white"
            },
            "_score": 4.0
          }
        ],
        "pagination": {
          "count": 2,
          "total": 32,
          "size": 2,
          "from": 0,
          "sort": "None",
          "page": 1,
          "pages": 16
        }
      },
      "warnings": {}
    }
    ```

__Example 4:__  The `/analysis/mutated_cases_count_by_project` endpoint produces counts for the number of cases that have associated `ssm` data in each project. The number of affected cases can be found under `"case_with_ssm": {"doc_count": $case_count}`.  

=== "Shell"

    ```Shell
    curl "https://api.gdc.cancer.gov/analysis/mutated_cases_count_by_project?size=0&pretty=true"
    ```

=== "Response"

    ```json
    {
      "took": 9,
      "timed_out": false,
      "_shards": {
        "total": 12,
        "successful": 12,
        "skipped": 0,
        "failed": 0
      },
      "hits": {
        "total": {
          "value": 44451,
          "relation": "eq"
        },
        "max_score": null,
        "hits": []
      },
      "aggregations": {
        "projects": {
          "doc_count_error_upper_bound": 0,
          "sum_other_doc_count": 0,
          "buckets": [
            {
              "key": "FM-AD",
              "doc_count": 18004,
              "case_summary": {
                "doc_count": 54012,
                "case_with_ssm": {
                  "doc_count": 18004
                }
              }
            },
            {
              "key": "TARGET-AML",
              "doc_count": 2492,
              "case_summary": {
                "doc_count": 10780,
                "case_with_ssm": {
                  "doc_count": 22
                }
              }
            },
            {
              "key": "TARGET-ALL-P2",
              "doc_count": 1587,
              "case_summary": {
                "doc_count": 5562,
                "case_with_ssm": {
                  "doc_count": 717
                }
              }
            },
            {
              "key": "MP2PRT-ALL",
              "doc_count": 1510,
              "case_summary": {
                "doc_count": 10472,
                "case_with_ssm": {
                  "doc_count": 1508
                }
              }
            },
            {
              "key": "CPTAC-3",
              "doc_count": 1235,
              "case_summary": {
                "doc_count": 8508,
                "case_with_ssm": {
                  "doc_count": 1218
                }
              }
            },
            {
              "key": "TARGET-NBL",
              "doc_count": 1132,
              "case_summary": {
                "doc_count": 3007,
                "case_with_ssm": {
                  "doc_count": 220
                }
              }
            },
            {
              "key": "TCGA-BRCA",
              "doc_count": 1098,
              "case_summary": {
                "doc_count": 9735,
                "case_with_ssm": {
                  "doc_count": 1098
                }
              }
            },
            {
              "key": "MMRF-COMMPASS",
              "doc_count": 995,
              "case_summary": {
                "doc_count": 3528,
                "case_with_ssm": {
                  "doc_count": 959
                }
              }
            },
            {
              "key": "BEATAML1.0-COHORT",
              "doc_count": 826,
              "case_summary": {
                "doc_count": 2891,
                "case_with_ssm": {
                  "doc_count": 759
                }
              }
            },
            {
              "key": "TARGET-WT",
              "doc_count": 652,
              "case_summary": {
                "doc_count": 3045,
                "case_with_ssm": {
                  "doc_count": 631
                }
              }
            },
            {
              "key": "TCGA-GBM",
              "doc_count": 617,
              "case_summary": {
                "doc_count": 3867,
                "case_with_ssm": {
                  "doc_count": 600
                }
              }
            },
            {
              "key": "TCGA-OV",
              "doc_count": 608,
              "case_summary": {
                "doc_count": 5028,
                "case_with_ssm": {
                  "doc_count": 599
                }
              }
            },
            {
              "key": "TCGA-LUAD",
              "doc_count": 585,
              "case_summary": {
                "doc_count": 4825,
                "case_with_ssm": {
                  "doc_count": 571
                }
              }
            },
            {
              "key": "TCGA-UCEC",
              "doc_count": 560,
              "case_summary": {
                "doc_count": 4559,
                "case_with_ssm": {
                  "doc_count": 559
                }
              }
            },
            {
              "key": "TCGA-KIRC",
              "doc_count": 537,
              "case_summary": {
                "doc_count": 4768,
                "case_with_ssm": {
                  "doc_count": 534
                }
              }
            },
            {
              "key": "TCGA-HNSC",
              "doc_count": 528,
              "case_summary": {
                "doc_count": 4577,
                "case_with_ssm": {
                  "doc_count": 528
                }
              }
            },
            {
              "key": "TCGA-LGG",
              "doc_count": 516,
              "case_summary": {
                "doc_count": 4570,
                "case_with_ssm": {
                  "doc_count": 516
                }
              }
            },
            {
              "key": "TCGA-THCA",
              "doc_count": 507,
              "case_summary": {
                "doc_count": 4442,
                "case_with_ssm": {
                  "doc_count": 507
                }
              }
            },
            {
              "key": "TCGA-LUSC",
              "doc_count": 504,
              "case_summary": {
                "doc_count": 4425,
                "case_with_ssm": {
                  "doc_count": 504
                }
              }
            },
            {
              "key": "TCGA-PRAD",
              "doc_count": 500,
              "case_summary": {
                "doc_count": 4365,
                "case_with_ssm": {
                  "doc_count": 500
                }
              }
            },
            {
              "key": "NCICCR-DLBCL",
              "doc_count": 489,
              "case_summary": {
                "doc_count": 1451,
                "case_with_ssm": {
                  "doc_count": 0
                }
              }
            },
            {
              "key": "TCGA-SKCM",
              "doc_count": 470,
              "case_summary": {
                "doc_count": 4125,
                "case_with_ssm": {
                  "doc_count": 470
                }
              }
            },
            {
              "key": "TCGA-COAD",
              "doc_count": 461,
              "case_summary": {
                "doc_count": 3888,
                "case_with_ssm": {
                  "doc_count": 461
                }
              }
            },
            {
              "key": "TCGA-STAD",
              "doc_count": 443,
              "case_summary": {
                "doc_count": 3884,
                "case_with_ssm": {
                  "doc_count": 443
                }
              }
            },
            {
              "key": "REBC-THYR",
              "doc_count": 440,
              "case_summary": {
                "doc_count": 2456,
                "case_with_ssm": {
                  "doc_count": 380
                }
              }
            },
            {
              "key": "TCGA-BLCA",
              "doc_count": 412,
              "case_summary": {
                "doc_count": 3645,
                "case_with_ssm": {
                  "doc_count": 412
                }
              }
            },
            {
              "key": "TARGET-OS",
              "doc_count": 383,
              "case_summary": {
                "doc_count": 1276,
                "case_with_ssm": {
                  "doc_count": 97
                }
              }
            },
            {
              "key": "TCGA-LIHC",
              "doc_count": 377,
              "case_summary": {
                "doc_count": 3204,
                "case_with_ssm": {
                  "doc_count": 377
                }
              }
            },
            {
              "key": "CPTAC-2",
              "doc_count": 342,
              "case_summary": {
                "doc_count": 1349,
                "case_with_ssm": {
                  "doc_count": 328
                }
              }
            },
            {
              "key": "TRIO-CRU",
              "doc_count": 339,
              "case_summary": {
                "doc_count": 339,
                "case_with_ssm": {
                  "doc_count": 0
                }
              }
            },
            {
              "key": "CGCI-BLGSP",
              "doc_count": 324,
              "case_summary": {
                "doc_count": 2076,
                "case_with_ssm": {
                  "doc_count": 262
                }
              }
            },
            {
              "key": "TCGA-CESC",
              "doc_count": 307,
              "case_summary": {
                "doc_count": 2623,
                "case_with_ssm": {
                  "doc_count": 306
                }
              }
            },
            {
              "key": "TCGA-KIRP",
              "doc_count": 291,
              "case_summary": {
                "doc_count": 2568,
                "case_with_ssm": {
                  "doc_count": 291
                }
              }
            },
            {
              "key": "HCMI-CMDC",
              "doc_count": 278,
              "case_summary": {
                "doc_count": 2420,
                "case_with_ssm": {
                  "doc_count": 277
                }
              }
            },
            {
              "key": "TCGA-TGCT",
              "doc_count": 263,
              "case_summary": {
                "doc_count": 2124,
                "case_with_ssm": {
                  "doc_count": 262
                }
              }
            },
            {
              "key": "TCGA-SARC",
              "doc_count": 261,
              "case_summary": {
                "doc_count": 2309,
                "case_with_ssm": {
                  "doc_count": 261
                }
              }
            },
            {
              "key": "CGCI-HTMCP-CC",
              "doc_count": 212,
              "case_summary": {
                "doc_count": 1452,
                "case_with_ssm": {
                  "doc_count": 206
                }
              }
            },
            {
              "key": "CMI-MBC",
              "doc_count": 200,
              "case_summary": {
                "doc_count": 653,
                "case_with_ssm": {
                  "doc_count": 174
                }
              }
            },
            {
              "key": "TCGA-LAML",
              "doc_count": 200,
              "case_summary": {
                "doc_count": 1533,
                "case_with_ssm": {
                  "doc_count": 200
                }
              }
            },
            {
              "key": "TARGET-ALL-P3",
              "doc_count": 191,
              "case_summary": {
                "doc_count": 782,
                "case_with_ssm": {
                  "doc_count": 86
                }
              }
            },
            {
              "key": "TCGA-ESCA",
              "doc_count": 185,
              "case_summary": {
                "doc_count": 1623,
                "case_with_ssm": {
                  "doc_count": 185
                }
              }
            },
            {
              "key": "TCGA-PAAD",
              "doc_count": 185,
              "case_summary": {
                "doc_count": 1720,
                "case_with_ssm": {
                  "doc_count": 185
                }
              }
            },
            {
              "key": "TCGA-PCPG",
              "doc_count": 179,
              "case_summary": {
                "doc_count": 1512,
                "case_with_ssm": {
                  "doc_count": 179
                }
              }
            },
            {
              "key": "OHSU-CNL",
              "doc_count": 176,
              "case_summary": {
                "doc_count": 494,
                "case_with_ssm": {
                  "doc_count": 158
                }
              }
            },
            {
              "key": "TCGA-READ",
              "doc_count": 172,
              "case_summary": {
                "doc_count": 1414,
                "case_with_ssm": {
                  "doc_count": 171
                }
              }
            },
            {
              "key": "TCGA-THYM",
              "doc_count": 124,
              "case_summary": {
                "doc_count": 1078,
                "case_with_ssm": {
                  "doc_count": 124
                }
              }
            },
            {
              "key": "TCGA-KICH",
              "doc_count": 113,
              "case_summary": {
                "doc_count": 705,
                "case_with_ssm": {
                  "doc_count": 66
                }
              }
            },
            {
              "key": "WCDT-MCRPC",
              "doc_count": 101,
              "case_summary": {
                "doc_count": 299,
                "case_with_ssm": {
                  "doc_count": 0
                }
              }
            },
            {
              "key": "TCGA-ACC",
              "doc_count": 92,
              "case_summary": {
                "doc_count": 809,
                "case_with_ssm": {
                  "doc_count": 92
                }
              }
            },
            {
              "key": "APOLLO-LUAD",
              "doc_count": 87,
              "case_summary": {
                "doc_count": 510,
                "case_with_ssm": {
                  "doc_count": 83
                }
              }
            },
            {
              "key": "TCGA-MESO",
              "doc_count": 87,
              "case_summary": {
                "doc_count": 813,
                "case_with_ssm": {
                  "doc_count": 87
                }
              }
            },
            {
              "key": "EXCEPTIONAL_RESPONDERS-ER",
              "doc_count": 84,
              "case_summary": {
                "doc_count": 412,
                "case_with_ssm": {
                  "doc_count": 20
                }
              }
            },
            {
              "key": "TCGA-UVM",
              "doc_count": 80,
              "case_summary": {
                "doc_count": 700,
                "case_with_ssm": {
                  "doc_count": 80
                }
              }
            },
            {
              "key": "CGCI-HTMCP-DLBCL",
              "doc_count": 70,
              "case_summary": {
                "doc_count": 465,
                "case_with_ssm": {
                  "doc_count": 50
                }
              }
            },
            {
              "key": "ORGANOID-PANCREATIC",
              "doc_count": 70,
              "case_summary": {
                "doc_count": 225,
                "case_with_ssm": {
                  "doc_count": 57
                }
              }
            },
            {
              "key": "TARGET-RT",
              "doc_count": 69,
              "case_summary": {
                "doc_count": 404,
                "case_with_ssm": {
                  "doc_count": 0
                }
              }
            },
            {
              "key": "CMI-MPC",
              "doc_count": 63,
              "case_summary": {
                "doc_count": 199,
                "case_with_ssm": {
                  "doc_count": 60
                }
              }
            },
            {
              "key": "MATCH-I",
              "doc_count": 60,
              "case_summary": {
                "doc_count": 345,
                "case_with_ssm": {
                  "doc_count": 57
                }
              }
            },
            {
              "key": "TCGA-DLBC",
              "doc_count": 58,
              "case_summary": {
                "doc_count": 441,
                "case_with_ssm": {
                  "doc_count": 50
                }
              }
            },
            {
              "key": "TCGA-UCS",
              "doc_count": 57,
              "case_summary": {
                "doc_count": 504,
                "case_with_ssm": {
                  "doc_count": 57
                }
              }
            },
            {
              "key": "BEATAML1.0-CRENOLANIB",
              "doc_count": 56,
              "case_summary": {
                "doc_count": 107,
                "case_with_ssm": {
                  "doc_count": 51
                }
              }
            },
            {
              "key": "MP2PRT-WT",
              "doc_count": 52,
              "case_summary": {
                "doc_count": 361,
                "case_with_ssm": {
                  "doc_count": 51
                }
              }
            },
            {
              "key": "TCGA-CHOL",
              "doc_count": 51,
              "case_summary": {
                "doc_count": 378,
                "case_with_ssm": {
                  "doc_count": 51
                }
              }
            },
            {
              "key": "CDDP_EAGLE-1",
              "doc_count": 50,
              "case_summary": {
                "doc_count": 384,
                "case_with_ssm": {
                  "doc_count": 50
                }
              }
            },
            {
              "key": "CTSP-DLBCL1",
              "doc_count": 45,
              "case_summary": {
                "doc_count": 201,
                "case_with_ssm": {
                  "doc_count": 0
                }
              }
            },
            {
              "key": "MATCH-W",
              "doc_count": 45,
              "case_summary": {
                "doc_count": 265,
                "case_with_ssm": {
                  "doc_count": 44
                }
              }
            },
            {
              "key": "MATCH-Z1A",
              "doc_count": 45,
              "case_summary": {
                "doc_count": 262,
                "case_with_ssm": {
                  "doc_count": 43
                }
              }
            },
            {
              "key": "CGCI-HTMCP-LC",
              "doc_count": 39,
              "case_summary": {
                "doc_count": 292,
                "case_with_ssm": {
                  "doc_count": 34
                }
              }
            },
            {
              "key": "CMI-ASC",
              "doc_count": 36,
              "case_summary": {
                "doc_count": 124,
                "case_with_ssm": {
                  "doc_count": 36
                }
              }
            },
            {
              "key": "MATCH-Z1D",
              "doc_count": 36,
              "case_summary": {
                "doc_count": 212,
                "case_with_ssm": {
                  "doc_count": 34
                }
              }
            },
            {
              "key": "MATCH-Q",
              "doc_count": 35,
              "case_summary": {
                "doc_count": 203,
                "case_with_ssm": {
                  "doc_count": 34
                }
              }
            },
            {
              "key": "MATCH-B",
              "doc_count": 33,
              "case_summary": {
                "doc_count": 187,
                "case_with_ssm": {
                  "doc_count": 32
                }
              }
            },
            {
              "key": "MATCH-Y",
              "doc_count": 31,
              "case_summary": {
                "doc_count": 181,
                "case_with_ssm": {
                  "doc_count": 30
                }
              }
            },
            {
              "key": "TARGET-ALL-P1",
              "doc_count": 24,
              "case_summary": {
                "doc_count": 86,
                "case_with_ssm": {
                  "doc_count": 0
                }
              }
            },
            {
              "key": "MATCH-U",
              "doc_count": 23,
              "case_summary": {
                "doc_count": 137,
                "case_with_ssm": {
                  "doc_count": 22
                }
              }
            },
            {
              "key": "MATCH-H",
              "doc_count": 21,
              "case_summary": {
                "doc_count": 122,
                "case_with_ssm": {
                  "doc_count": 21
                }
              }
            },
            {
              "key": "MATCH-N",
              "doc_count": 21,
              "case_summary": {
                "doc_count": 120,
                "case_with_ssm": {
                  "doc_count": 21
                }
              }
            },
            {
              "key": "TARGET-CCSK",
              "doc_count": 13,
              "case_summary": {
                "doc_count": 100,
                "case_with_ssm": {
                  "doc_count": 0
                }
              }
            },
            {
              "key": "VAREPOP-APOLLO",
              "doc_count": 7,
              "case_summary": {
                "doc_count": 14,
                "case_with_ssm": {
                  "doc_count": 7
                }
              }
            }
          ]
        }
      }
    }
    ```

### Survival Analysis Endpoint

[Survival plots](/Data_Portal/Users_Guide/mutation_frequency/#survival-plot-for-mutated-genes-and-mutations) are generated for different subsets of data, based on variants or projects, in the GDC Data Portal. The `/analysis/survival` endpoint can be used to programmatically retrieve the raw data used to generate these plots and apply different filters. Note that the `fields` and `format` parameters cannot be modified.

 __Example 1:__ A user wants to download data to generate a survival plot for cases from the project TCGA-DLBC.

=== "Shell"

    ```Shell
    curl "https://api.gdc.cancer.gov/analysis/survival?filters=%5B%7B%22op%22%3A%22%3D%22%2C%22content%22%3A%7B%22field%22%3A%22cases.project.project_id%22%2C%22value%22%3A%22TCGA-DLBC%22%7D%7D%5D&pretty=true"
    ```

=== "Response"

    ```json
    {
      "results": [
        {
          "meta": {
            "id": 139834474037000
          },
          "donors": [
            {
              "time": 1.0,
              "censored": true,
              "survivalEstimate": 1,
              "id": "dc87a809-95de-4eb7-a1c2-2650475f2d7e",
              "submitter_id": "TCGA-RQ-A6JB",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 17.0,
              "censored": true,
              "survivalEstimate": 1,
              "id": "4dd86ebd-ef16-4b2b-9ea0-5d1d7afef257",
              "submitter_id": "TCGA-RQ-AAAT",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 58,
              "censored": false,
              "survivalEstimate": 1,
              "id": "0bf573ac-cd1e-42d8-90cf-b30d7b08679c",
              "submitter_id": "TCGA-FA-A6HN",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 126.0,
              "censored": true,
              "survivalEstimate": 0.9777777777777777,
              "id": "f978cb0f-d319-4c01-b4c5-23ae1403a106",
              "submitter_id": "TCGA-FF-8047",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 132.0,
              "censored": true,
              "survivalEstimate": 0.9777777777777777,
              "id": "1843c82e-7a35-474f-9f79-c0a9af9aa09c",
              "submitter_id": "TCGA-FA-A4BB",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 132.0,
              "censored": true,
              "survivalEstimate": 0.9777777777777777,
              "id": "a43e5f0e-a21f-48d8-97e0-084d413680b7",
              "submitter_id": "TCGA-FA-8693",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 248,
              "censored": false,
              "survivalEstimate": 0.9777777777777777,
              "id": "0030a28c-81aa-44b0-8be0-b35e1dcbf98c",
              "submitter_id": "TCGA-FA-A7Q1",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 298.0,
              "censored": true,
              "survivalEstimate": 0.9539295392953929,
              "id": "f553f1a9-ecf2-4783-a609-6adca7c4c597",
              "submitter_id": "TCGA-FF-A7CQ",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 313,
              "censored": false,
              "survivalEstimate": 0.9539295392953929,
              "id": "f784bc3a-751b-4025-aab2-0af2f6f24266",
              "submitter_id": "TCGA-FF-A7CR",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 385.0,
              "censored": true,
              "survivalEstimate": 0.929469807518588,
              "id": "29e3d122-15a1-4235-a356-b1a9f94ceb39",
              "submitter_id": "TCGA-FF-A7CX",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 391,
              "censored": false,
              "survivalEstimate": 0.929469807518588,
              "id": "0e251c03-bf86-4ed8-b45d-3cbc97160502",
              "submitter_id": "TCGA-GS-A9U4",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 427.0,
              "censored": true,
              "survivalEstimate": 0.9043490019099776,
              "id": "e6365b38-bc44-400c-b4aa-18ce8ff5bfce",
              "submitter_id": "TCGA-FA-A82F",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 553.0,
              "censored": true,
              "survivalEstimate": 0.9043490019099776,
              "id": "b56bdbdb-43af-4a03-a072-54dd22d7550c",
              "submitter_id": "TCGA-FA-A86F",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 595,
              "censored": false,
              "survivalEstimate": 0.9043490019099776,
              "id": "31bbad4e-3789-42ec-9faa-1cb86970f723",
              "submitter_id": "TCGA-G8-6907",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 679.0,
              "censored": true,
              "survivalEstimate": 0.8777505018538018,
              "id": "0e9fcccc-0630-408d-a121-2c6413824cb7",
              "submitter_id": "TCGA-FF-8062",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 708,
              "censored": false,
              "survivalEstimate": 0.8777505018538018,
              "id": "a5b188f0-a6d3-4d4a-b04f-36d47ec05338",
              "submitter_id": "TCGA-FA-A4XK",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 719.0,
              "censored": true,
              "survivalEstimate": 0.8503207986708705,
              "id": "ed746cb9-0f2f-48ce-923a-3a9f9f00b331",
              "submitter_id": "TCGA-FA-A7DS",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 730.0,
              "censored": true,
              "survivalEstimate": 0.8503207986708705,
              "id": "c85f340e-584b-4f3b-b6a5-540491fc8ad2",
              "submitter_id": "TCGA-GS-A9TV",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 749.0,
              "censored": true,
              "survivalEstimate": 0.8503207986708705,
              "id": "69f23725-adca-48ac-9b33-80a7aae24cfe",
              "submitter_id": "TCGA-FA-A6HO",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 751.0,
              "censored": true,
              "survivalEstimate": 0.8503207986708705,
              "id": "67325322-483f-443f-9ffa-2a20d108a2fb",
              "submitter_id": "TCGA-FF-8046",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 765.0,
              "censored": true,
              "survivalEstimate": 0.8503207986708705,
              "id": "eda9496e-be80-4a13-bf06-89f0cc9e937f",
              "submitter_id": "TCGA-GS-A9TZ",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 788.0,
              "censored": true,
              "survivalEstimate": 0.8503207986708705,
              "id": "25ff86af-beb4-480c-b706-f3fe0306f7cf",
              "submitter_id": "TCGA-RQ-A68N",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 791.0,
              "censored": true,
              "survivalEstimate": 0.8503207986708705,
              "id": "1d0db5d7-39ca-466d-96b3-0d278c5ea768",
              "submitter_id": "TCGA-FF-A7CW",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 832.0,
              "censored": true,
              "survivalEstimate": 0.8503207986708705,
              "id": "c8cde9ea-89e9-4ee8-8a46-417a48f6d3ab",
              "submitter_id": "TCGA-FF-8061",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 946.0,
              "censored": true,
              "survivalEstimate": 0.8503207986708705,
              "id": "f0a326d2-1f3e-4a5d-bca8-32aaccc52338",
              "submitter_id": "TCGA-FF-8042",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 965.0,
              "censored": true,
              "survivalEstimate": 0.8503207986708705,
              "id": "a8e2df1e-4042-42af-9231-3a00e83489f0",
              "submitter_id": "TCGA-FF-8043",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 972.0,
              "censored": true,
              "survivalEstimate": 0.8503207986708705,
              "id": "e56e4d9c-052e-4ec6-a81b-dbd53e9c8ffe",
              "submitter_id": "TCGA-FM-8000",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 982.0,
              "censored": true,
              "survivalEstimate": 0.8503207986708705,
              "id": "45b0cf9f-a879-417f-8f39-7770552252c0",
              "submitter_id": "TCGA-GS-A9TQ",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 1081.0,
              "censored": true,
              "survivalEstimate": 0.8503207986708705,
              "id": "1f971af1-6772-4fe6-8d35-bbe527a037fe",
              "submitter_id": "TCGA-FF-8041",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 1163.0,
              "censored": true,
              "survivalEstimate": 0.8503207986708705,
              "id": "33365d22-cb83-4d8e-a2d1-06b675f75f6e",
              "submitter_id": "TCGA-GS-A9TT",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 1252,
              "censored": false,
              "survivalEstimate": 0.8503207986708705,
              "id": "6a21c948-cd85-4150-8c01-83017d7dc1ed",
              "submitter_id": "TCGA-G8-6324",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 1299.0,
              "censored": true,
              "survivalEstimate": 0.8003019281608192,
              "id": "f855dad1-6ffc-493e-ba6c-970874bc9210",
              "submitter_id": "TCGA-GR-A4D5",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 1334.0,
              "censored": true,
              "survivalEstimate": 0.8003019281608192,
              "id": "c1c06604-5ae2-4a53-b9c0-eb210d38e3f0",
              "submitter_id": "TCGA-GR-A4D6",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 1373.0,
              "censored": true,
              "survivalEstimate": 0.8003019281608192,
              "id": "58e66976-4507-4552-ac53-83a49a142dde",
              "submitter_id": "TCGA-GS-A9TX",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 1581.0,
              "censored": true,
              "survivalEstimate": 0.8003019281608192,
              "id": "ea54dbad-1b23-41cc-9378-d4002a8fca51",
              "submitter_id": "TCGA-G8-6325",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 1581.0,
              "censored": true,
              "survivalEstimate": 0.8003019281608192,
              "id": "d7df78b5-24f1-4ff4-bd9b-f0e6bec8289a",
              "submitter_id": "TCGA-GR-A4D4",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 1617.0,
              "censored": true,
              "survivalEstimate": 0.8003019281608192,
              "id": "29aff186-c321-4ff9-b81b-105e27e620ff",
              "submitter_id": "TCGA-GS-A9TW",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 1739.0,
              "censored": true,
              "survivalEstimate": 0.8003019281608192,
              "id": "5eff68ff-f6c3-40c9-9fc8-00e684a7b712",
              "submitter_id": "TCGA-GR-A4D9",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 2131.0,
              "censored": true,
              "survivalEstimate": 0.8003019281608192,
              "id": "f8cf647b-1447-4ac3-8c43-bef07765cabf",
              "submitter_id": "TCGA-G8-6326",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 2616.0,
              "censored": true,
              "survivalEstimate": 0.8003019281608192,
              "id": "6e9437f0-a4ed-475c-ab0e-bf1431c70a90",
              "submitter_id": "TCGA-GS-A9TY",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 2983.0,
              "censored": true,
              "survivalEstimate": 0.8003019281608192,
              "id": "c3d662ee-48d0-454a-bb0c-77d3338d3747",
              "submitter_id": "TCGA-GR-7353",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 3394.0,
              "censored": true,
              "survivalEstimate": 0.8003019281608192,
              "id": "fdecb74f-ac4e-46b1-b23a-5f7fde96ef9f",
              "submitter_id": "TCGA-GS-A9U3",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 3553,
              "censored": false,
              "survivalEstimate": 0.8003019281608192,
              "id": "a468e725-ad4b-411d-ac5c-2eacc68ec580",
              "submitter_id": "TCGA-G8-6909",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 3897.0,
              "censored": true,
              "survivalEstimate": 0.6402415425286554,
              "id": "1ea575f1-f731-408b-a629-f5f4abab569e",
              "submitter_id": "TCGA-GS-A9TU",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 4578.0,
              "censored": true,
              "survivalEstimate": 0.6402415425286554,
              "id": "7a589441-11ef-4158-87e7-3951d86bc2aa",
              "submitter_id": "TCGA-GR-7351",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 5980.0,
              "censored": true,
              "survivalEstimate": 0.6402415425286554,
              "id": "3622cf29-600f-4410-84d4-a9afeb41c475",
              "submitter_id": "TCGA-G8-6914",
              "project_id": "TCGA-DLBC"
            },
            {
              "time": 6425,
              "censored": false,
              "survivalEstimate": 0.6402415425286554,
              "id": "3f5a897d-1eaa-4d4c-8324-27ac07c90927",
              "submitter_id": "TCGA-G8-6906",
              "project_id": "TCGA-DLBC"
            }
          ]
        }
      ],
      "overallStats": {}
    }
    ```

__Example 2:__ Here the survival endpoint is used to compare two survival plots for TCGA-BRCA cases.  One plot will display survival information about cases with a particular mutation (in this instance: `chr3:g.179234297A>G`) and the other plot will display information about cases without that mutation. This type of query will also print the results of a chi-squared analysis between the two subsets of cases.  

=== "Json"

    ```json
    [  
      {  
        "op":"and",
        "content":[  
          {  
            "op":"=",
            "content":{  
              "field":"cases.project.project_id",
              "value":"TCGA-BRCA"
            }
          },
          {  
            "op":"=",
            "content":{  
              "field":"gene.ssm.ssm_id",
              "value":"edd1ae2c-3ca9-52bd-a124-b09ed304fcc2"
            }
          }
        ]
      },
      {  
        "op":"and",
        "content":[  
          {  
            "op":"=",
            "content":{  
              "field":"cases.project.project_id",
              "value":"TCGA-BRCA"
            }
          },
          {  
            "op":"excludeifany",
            "content":{  
              "field":"gene.ssm.ssm_id",
              "value":"edd1ae2c-3ca9-52bd-a124-b09ed304fcc2"
            }
          }
        ]
      }
    ]
    ```

=== "Shell"

    ```Shell
    curl "https://api.gdc.cancer.gov/analysis/survival?filters=%5B%7B%22op%22%3A%22and%22%2C%22content%22%3A%5B%7B%22op%22%3A%22%3D%22%2C%22content%22%3A%7B%22field%22%3A%22cases.project.project_id%22%2C%22value%22%3A%22TCGA-BRCA%22%7D%7D%2C%7B%22op%22%3A%22%3D%22%2C%22content%22%3A%7B%22field%22%3A%22gene.ssm.ssm_id%22%2C%22value%22%3A%22edd1ae2c-3ca9-52bd-a124-b09ed304fcc2%22%7D%7D%5D%7D%2C%7B%22op%22%3A%22and%22%2C%22content%22%3A%5B%7B%22op%22%3A%22%3D%22%2C%22content%22%3A%7B%22field%22%3A%22cases.project.project_id%22%2C%22value%22%3A%22TCGA-BRCA%22%7D%7D%2C%7B%22op%22%3A%22excludeifany%22%2C%22content%22%3A%7B%22field%22%3A%22gene.ssm.ssm_id%22%2C%22value%22%3A%22edd1ae2c-3ca9-52bd-a124-b09ed304fcc2%22%7D%7D%5D%7D%5D&pretty=true"
    ```

=== "Json2"     

    ```json
    {
      "overallStats": {
        "degreesFreedom": 1,
        "chiSquared": 0.8577589072612264,
        "pValue": 0.35436660628146011
      },
      "results": [
        {
          "donors": [
            {
              "survivalEstimate": 1,
              "id": "a991644b-3ee6-4cda-acf0-e37de48a49fc",
              "censored": true,
              "time": 10
            },
            {
              "survivalEstimate": 1,
              "id": "2e1e3bf0-1708-4b65-936c-48b89eb8966a",
              "censored": true,
              "time": 19
            },
    (truncated)
    ],
    "meta": {
      "id": 140055251282040
    }
    },
    {
    "donors": [
      {
        "survivalEstimate": 1,
        "id": "5e4187c9-98f8-4bdb-a8da-6a914e96f47a",
        "censored": true,
        "time": -31
      },
    (truncated)
    ```

The output represents two sets of coordinates delimited as objects with the `donors` tag. One set of coordinates will generate a survival plot representing TCGA-BRCA cases that have the mutation of interest and the other will generate a survival plot for the remaining cases in TCGA-BRCA.

__Example 3:__ Custom survival plots can be generated using the GDC API.  For example, a user could generate survival plot data comparing patients with a mutation in genes associated with a biological pathway with patients without mutations in that pathway. The following example compares a patient with at least one mutation in either gene `ENSG00000141510` or `ENSG00000155657` with patients that do not have mutations in these genes.

=== "Query"

    ``` json
    [  
       {  
          "op":"and",
          "content":[  
             {  
                "op":"=",
                "content":{  
                   "field":"cases.project.project_id",
                   "value":"TCGA-BRCA"
                }
             },
             {  
                "op":"=",
                "content":{  
                   "field":"gene.gene_id",
                   "value":["ENSG00000141510","ENSG00000155657"]
                }
             }
          ]
       },
       {  
          "op":"and",
          "content":[  
             {  
                "op":"=",
                "content":{  
                   "field":"cases.project.project_id",
                   "value":"TCGA-BRCA"
                }
             },
             {  
                "op":"excludeifany",
                "content":{  
                   "field":"gene.gene_id",
                   "value":["ENSG00000141510","ENSG00000155657"]
                }
             }
          ]
       }
    ]
    ```

=== "Shell"

    ```Shell
    curl "https://api.gdc.cancer.gov/analysis/survival?filters=%5B%0D%0A%7B%0D%0A%22op%22%3A%22and%22%2C%0D%0A%22content%22%3A%5B%0D%0A%7B%0D%0A%22op%22%3A%22%3D%22%2C%0D%0A%22content%22%3A%7B%0D%0A%22field%22%3A%22cases.project.project_id%22%2C%0D%0A%22value%22%3A%22TCGA-BRCA%22%0D%0A%7D%0D%0A%7D%2C%0D%0A%7B%0D%0A%22op%22%3A%22%3D%22%2C%0D%0A%22content%22%3A%7B%0D%0A%22field%22%3A%22gene.gene_id%22%2C%0D%0A%22value%22%3A%5B%22ENSG00000141510%22%2C%22ENSG00000155657%22%5D%0D%0A%7D%0D%0A%7D%0D%0A%5D%0D%0A%7D%2C%0D%0A%7B%0D%0A%22op%22%3A%22and%22%2C%0D%0A%22content%22%3A%5B%0D%0A%7B%0D%0A%22op%22%3A%22%3D%22%2C%0D%0A%22content%22%3A%7B%0D%0A%22field%22%3A%22cases.project.project_id%22%2C%0D%0A%22value%22%3A%22TCGA-BRCA%22%0D%0A%7D%0D%0A%7D%2C%0D%0A%7B%0D%0A%22op%22%3A%22excludeifany%22%2C%0D%0A%22content%22%3A%7B%0D%0A%22field%22%3A%22gene.gene_id%22%2C%0D%0A%22value%22%3A%5B%22ENSG00000141510%22%2C%22ENSG00000155657%22%5D%0D%0A%7D%0D%0A%7D%0D%0A%5D%0D%0A%7D%0D%0A%5D&pretty=true"
    ```

__Example 4:__ Survival plots can be even more customizable when sets of case IDs are used. Two sets of case IDs (or barcodes) can be retrieved in a separate step based on custom criteria and compared in a survival plot. See below for an example query.

=== "Json"

    ```Json
    [{  
       "op":"=",
       "content":{  
          "field":"cases.submitter_id",
          "value":["TCGA-HT-A74J","TCGA-43-A56U","TCGA-GM-A3XL","TCGA-A1-A0SQ","TCGA-K1-A6RV","TCGA-J2-A4AD","TCGA-XR-A8TE"]
       }
    },
    {  
       "op":"=",
       "content":{  
          "field":"cases.submitter_id",
          "value":["TCGA-55-5899","TCGA-55-6642","TCGA-55-7907","TCGA-67-6216","TCGA-75-5146","TCGA-49-4510","TCGA-78-7159"]
       }
    }]
    ```

=== "Shell"

    ```Shell
    curl "https://api.gdc.cancer.gov/analysis/survival?filters=%5B%7B%22op%22%3A%22%3D%22%2C%22content%22%3A%7B%22field%22%3A%22cases.submitter_id%22%2C%22value%22%3A%5B%22TCGA-HT-A74J%22%2C%22TCGA-43-A56U%22%2C%22TCGA-GM-A3XL%22%2C%22TCGA-A1-A0SQ%22%2C%22TCGA-K1-A6RV%22%2C%22TCGA-J2-A4AD%22%2C%22TCGA-XR-A8TE%22%5D%7D%7D%2C%7B%22op%22%3A%22%3D%22%2C%22content%22%3A%7B%22field%22%3A%22cases.submitter_id%22%2C%22value%22%3A%5B%22TCGA-55-5899%22%2C%22TCGA-55-6642%22%2C%22TCGA-55-7907%22%2C%22TCGA-67-6216%22%2C%22TCGA-75-5146%22%2C%22TCGA-49-4510%22%2C%22TCGA-78-7159%22%5D%7D%7D%5D"
    ```


# BAM Slicing

The GDC API provides remote BAM slicing functionality that enables downloading of specific parts of a BAM file instead of the whole file. This functionality can be accessed at the `slicing` endpoint, using a syntax similar to that of widely used bioinformatics tools such as `samtools`.

## About the slicing endpoint

The `slicing` endpoint accepts HTTP GET requests in the form of a URL, and HTTP POST requests that carry a JSON payload. POST requests are more appropriate in cases where query parameters make the GET URL very long.

The response will be a BAM-formatted file containing the header of the source BAM file, as well as any alignment records that are found to overlap the specified regions, sorted by chromosomal coordinate.

Please note the following:

* The functionality of this API differs from the usual functionality of `samtools` in that alignment records that overlap multiple regions will not be returned multiple times.
* A request with no region or gene specified will return the BAM header, which makes it easy to inspect the references to which the alignment records were aligned.
* A request for regions that are not included in the source BAM is not considered an error, and is treated the same as if no records existed for the region.
* Examples provided for BAM slicing functionality are intended for use with GDC harmonized data (i.e. BAM files available in the GDC Data Portal).
* Bam slicing does not create an associated bam index (.bai) file.  For applications requiring a .bai file users will need to generate this file from the bam slice using a tool and command such as `samtools index`.

### Query Parameters

The following query parameters and JSON fields are supported:

| Description | Query Parameter | JSON Field | Query format |
|---|---|---|---|
| entire chromosome, or a position or region on the chromosome, specified using chromosomal coordinates | region | regions | region=<chr>(:<start>(-<stop>)?)?</stop></start></chr> |
| region specified using a [HGNC](http://www.genenames.org/) / [GENCODE v36](http://www.gencodegenes.org/) gene name |  gencode | gencode | gencode=<gene_name> |

>**NOTE:** The successfully sliced BAM will contain all reads that overlap (entirely or partially) with the specified region or gene. It is possible to specify an open-ended region, e.g. `chr2:10000`, which would return all reads that (completely or partially) overlap with the region of chromosome 2 from position 10,000 to the end of the chromosome.

### JSON Schema

JSON payloads can be syntactically verified using the following JSON schema:

	{
	  "$schema": "http://json-schema.org/schema#",
	  "type": "object",
	  "properties": {
	    "regions": {
	      "type": "array",
	      "items": {
	        "type": "string",
	        "pattern": "^[a-zA-Z0-9]+(:([0-9]+)?(-[0-9]+)?)?$"
	      }
	    },
	    "gencode": {
	      "type": "array",
	      "items": {
	        "type": "string"
	      }
	    }
	  }
	}



## Examples: Specifying a region

The following two requests are examples of BAM slicing using region(s).

=== "Regions_GET"

    ```shell

    token=$(<gdc-token-text-file.txt)

    curl --header "X-Auth-Token: $token" 'https://api.gdc.cancer.gov/slicing/view/2912e314-f6a7-4f4a-94ac-20db2c8f793b?region=chr1&region=chr2:10000&region=chr3:10000-20000' --output get_regions_slice.bam
    ```

=== "Regions_POST_Payload"

    ```json
    {
        "regions": [
            "chr1",
            "chr2:10000",
            "chr3:10000-20000"
        ]
    }
    ```

=== "Regions_POST"

    ```shell
    token=$(<gdc-token-text-file.txt)

    curl --header "X-Auth-Token: $token" --request POST https://api.gdc.cancer.gov/slicing/view/2912e314-f6a7-4f4a-94ac-20db2c8f793b --header "Content-Type: application/json" -d@Payload --output post_regions_slice.bam
    ```

=== "Response"

    ```
    Response:
    HTTP/1.1 206

    <bam_data_stream>
    ```

## Examples: Specifying a gene

The following two requests are examples of BAM slicing using HGNC / GENCODE v36 gene name(s).

=== "Gencode_GET"

    ```shell
    token=$(<gdc-token-text-file.txt)

    curl --header "X-Auth-Token: $token" 'https://api.gdc.cancer.gov/slicing/view/2912e314-f6a7-4f4a-94ac-20db2c8f793b?gencode=BRCA1' --output get_brca1_slice.bam
    ```

=== "Gencode_POST_Payload"

    ```json
    {
        "gencode": [
            "BRCA1",
            "BRCA2"
        ]
    }
    ```

=== "Gencode_POST"

    ```shell
    curl --header "X-Auth-Token: $token" --request POST https://api.gdc.cancer.gov/slicing/view/2912e314-f6a7-4f4a-94ac-20db2c8f793b --header "Content-Type: application/json" -d@Payload --output post_brca12_slice.bam
    ```

=== "Response"

    ```
    Response:
    HTTP/1.1 206

    <bam_data_stream>
    ```

## Examples: Specifying unmapped reads

Unmapped reads are found in GDC BAM files. You may request these reads by using the following commands.

=== "GET"

    ```shell
    token=$(<gdc-token-text-file.txt)

    curl --header "X-Auth-Token: $token" 'https://api.gdc.cancer.gov/slicing/view/dc87e1b8-d8b7-4837-88ea-fb7f017b3c69?region=unmapped' --output get_regions_slice.bam
    ```


=== "POST_Payload"

    ```json
    {
        "regions": [
            "unmapped"
        ]
    }
    ```

=== "POST"

    ```shell
    curl --header "X-Auth-Token: $token" --request POST https://api.gdc.cancer.gov/slicing/view/dc87e1b8-d8b7-4837-88ea-fb7f017b3c69 --header "Content-Type: application/json" -d@Payload --output get_regions_slice.bam
    ```

=== "Response"

    ```
    Response:
    HTTP/1.1 206

    <bam_data_stream>
    ```



After downloading, the sliced BAM file can be converted to SAM using the following command if `samtools` is installed on the user's system:

	samtools view -h brca1_slice.bam -o brca1_slice.sam

## Errors

When slicing cannot be performed, the GDC API will provide JSON error responses and HTTP error codes.

### JSON Error Responses

JSON error responses have the following structure:

	{
	    "error": "<error-message>"
	}

For example, when making a request for a protected BAM without supplying a GDC authentication token:

=== "Shell"

    ```Shell
    curl https://api.gdc.cancer.gov/v0/slicing/view/15b0bf8e-ff20-41ab-8366-a495c11b30be
    ```

=== "Response"
    ```
    HTTP/1.1 403 FORBIDDEN
    {
        "error": "Please specify a X-Auth-Token"
    }
    ```

### HTTP error codes

Potential HTTP error codes include:

Error Code | Description
---------- | -------
400 | Bad Request -- The regions specified are malformed
403 | Unauthorized -- The user could not be authenticated
403 | Unauthorized -- The user is not authorized for access to the source BAM
404 | Not Found -- No BAM is specified
404 | Not Found -- No BAM can be found for the specified GDC BAM ID
504 | BAI Not Found -- No BAI can be found for the BAM


### Transfer Errors

In the case that an error occurs during transfer of the resulting BAM, the BGZF EOF marker will not be present. This early truncation of the BAM file will cause errors if the file is used as input to other programs. For example, `samtools` will provide the error "EOF marker is absent".

Early truncation can arise when connection is interrupted or when slicing fails due to BAM corruption.


# System Information

## Overview

The GDC API offers endpoints that provide information about the system. These endpoints are described below.


## GDC Notifications Endpoint

The `notifications` endpoint provides current user-facing notifications.

GDC notifications have a corresponding `level` with the following meanings:

| Level   | Meaning                                                              |
|---------|----------------------------------------------------------------------|
| INFO    | Non-essential information, e.g. regarding a new dataset              |
| WARNING | Important user information, e.g. regarding a dataset to be removed   |
| ERROR   | Important system information, e.g. regarding a GDC component         |
| DEBUG   | Unimportant system information, e.g. testing the notification system |

Notifications will indicate the GDC `components` to which they apply:

| Component   | Description                                                              |
|---------|----------------------------------------------------------------------|
| PORTAL   | The GDC Data Portal         |
| SUBMISSION   | The GDC Data Submission Portal |
| DOCUMENTATION | The GDC documentation site that contains GDC user guides, release notes, and the GDC Data Dictionary    |
| WEBSITE    | The GDC project website that includes information about the system. This does not include any of the above-listed GDC components.           |

### Sample Request

=== "Shell"

    ```Shell
    curl --request GET https://api.gdc.cancer.gov/v0/notifications  
    ```

=== "Response"

    ```json
    {
      "data": [
        {
          "level": "INFO",
          "components": [
            "SUBMISSION_API"
            "LEGACY_API"
          ],
          "message": "The system is up!"
        }
      ]
    }
    ```

## API Status Endpoint

The `status` endpoint provides information about the current status and version of the GDC API.

### Sample Request

=== "Shell"

    ``` shell
    curl https://api.gdc.cancer.gov/status
    ```

=== "Python"

    ``` python
    import requests
    import json

    status_endpt = 'https://api.gdc.cancer.gov/status'
    response = requests.get(status_endpt)
    print json.dumps(response.json(), indent=2)
    ```

=== "Response"

    ``` json
    {
      "commit": "74e1e3583c0f39fbf2149322addb7378206be3b9",
      "status": "OK",
      "tag": "1.2.0",
      "version": 1
    }
    ```


# Getting Started

## The GDC Application Programming Interface (API): An Overview

The GDC API drives the GDC Data and Submission Portals and provides programmatic access to GDC functionality. This includes searching for, downloading, and submitting data and metadata. The GDC API uses JSON as its communication format, and standard HTTP methods like `GET`, `PUT`, `POST` and `DELETE`.

This guide explains how to construct and execute API requests and interpret API responses.

## Tools for communicating with the GDC API

Many third-party tools can be used for communicating with the GDC API and for preparing and visualizing API calls.

Examples of tools for communicating with the GDC API:

| Tool        | Type     |
| ------------- |-------------|
| [Curl](http://curl.haxx.se/docs/manpage.html) 		| Command line tool |
| [HTTPie](http://httpie.org) 	| Command line tool |
| [Postman REST Client](http://www.getpostman.com/) 														| App for Google Chrome and OS X |
| [DHC REST Client](http://restlet.com/products/dhc/)           | Google Chrome extension |
| [Google Chrome](http://www.google.com/chrome/) 	  | Google Chrome web browser |

Examples of tools that can help build GDC API calls:

| Tool        | Description     |
| ------------- |-------------|
| [JSONLint](http://jsonlint.com/)| Validate JSON |
| [JSON Formatter](http://jsonformatter.org/) | Format, validate, and convert JSON to other formats |
| [Percent-(URL)-encoding tool](https://codebeautify.org/url-encode-string)| Tool for percent-encoding strings |

>**NOTE:** When using the command line tool curl be sure to use the bash shell only for compatibility reasons.  All examples using curl are using the bash shell.  

The authentication token should be kept in a secure location, as it allows access to all data accessible by the associated user account.


## API Endpoints

Communicating with the GDC API involves making calls to API endpoints. Each GDC API endpoint represents specific API functionality, as summarized in the following table:

| Endpoint | Type | Description |
| --- | --- | --- |
| status | Status | Get the API status and version information |
| projects | Search & Retrieval | Search all data generated by a project |
| cases | Search & Retrieval | Find all files related to a specific case, or sample donor. |
| files | Search & Retrieval | Find all files with specific characteristics such as file_name, md5sum, data_format and others. |
| annotations | Search & Retrieval | Search annotations added to data after curation |
| data | Download | Used to download GDC data |
| manifest | Download | Generates manifests for use with GDC Data Transfer Tool |
| slicing | BAM Slicing | Allows remote slicing of BAM format objects |
| submission | Submission | Returns the available resources at the top level above programs i.e., registered programs |

The HTTP URL that corresponds to the latest version of a GDC API endpoint is `https://api.gdc.cancer.gov/<endpoint>`, where `<endpoint>` is the name of the endpoint.

The HTTP URL of an endpoint corresponding to a specific major version of the GDC API is `https://api.gdc.cancer.gov/<version>/<endpoint>`, where `<endpoint>` is the name of the endpoint and `<version>` is the GDC API version.

For example, the address of the latest version of the `status` endpoint is `https://api.gdc.cancer.gov/status`, whereas the address of the `status` endpoint corresponding to version 0 of GDC API is `https://api.gdc.cancer.gov/v0/status`.

## Entity UUIDs

All objects (*entities*) in the GDC are assigned a unique identifier in the form of a [version 4 universally unique identifier (UUID)](https://en.wikipedia.org/wiki/Universally_unique_identifier). The UUID uniquely identifies the entity in the GDC, and is stored in the entity's `id` property.

UUIDs are frequently used in GDC API requests and responses to identify specific entities like files, cases, and samples.

See [GDC Data Model](../../Data/Data_Model/GDC_Data_Model.md) for details.

## Sample Request

The following is an example of a request to the `files` endpoint, which retrieves information about a MAF file stored in the GDC.

=== "Shell"
  
    ``` shell
    curl https://api.gdc.cancer.gov/files/cb92f61d-041c-4424-a3e9-891b7545f351?pretty=true
    ```

=== "Python"

    ``` python
    import requests
    import json
    
    file_endpt = 'https://api.gdc.cancer.gov/files/'
    file_uuid = 'cb92f61d-041c-4424-a3e9-891b7545f351'
    response = requests.get(file_endpt + file_uuid)
    
    # OUTPUT METHOD 1: Write to a file.
    file = open("sample_request.json", "w")
    file.write(response.text)
    file.close()
    
    # OUTPUT METHOD 2: View on screen.
    print(json.dumps(response.json(), indent=2))
    ```

[Download Script](scripts/Sample_Request.py)
## Authentication

Authentication is required for downloading controlled-access data, and for all data submission functionality. The GDC API uses tokens for authentication.

Users can obtain authentication tokens from the [GDC Data Portal](https://portal.gdc.cancer.gov) and the [GDC Data Submission Portal](https://portal.gdc.cancer.gov/submission). See the [GDC Data Submission Portal User's Guide](../../Data_Submission_Portal/Users_Guide/Data_Submission_Process.md#authentication) for instructions.

### Using Authentication Tokens

All API requests that require authentication must include a token as an `X-Auth-Token` custom HTTP header.

In the following example, an authentication token is saved as an environment variable and passed to `curl` to download a controlled-access file:

=== "Shell"
    
    ```shell
    token=$(cat <gdc-token-text-file.txt>)
    
    curl -O -J -H "X-Auth-Token: $token" 'https://api.gdc.cancer.gov/data/fd89bfa5-b3a7-4079-bf90-709580c006e5'
    ```

=== "Output"

    ```
    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                    Dload  Upload   Total   Spent    Left  Speed
    100 4161M  100 4161M    0     0   281k      0  4:12:45  4:12:45 --:--:-- 1894k
    
    ```

=== "Python"

    ```Python
    import requests
    import json
    import re
    
    '''
     This script will not work until $TOKEN_FILE_PATH
     is replaced with an actual path.
    '''
    
    with open("$TOKEN_FILE_PATH","r") as token:
        token_string = str(token.read().strip())
    
    headers = {
               'X-Auth-Token': token_string
              }
    
    data_endpt = 'https://api.gdc.cancer.gov/data/'
    data_uuid = 'fd89bfa5-b3a7-4079-bf90-709580c006e5'
    headers = {
               'X-Auth-Token': token_string
              }
    response = requests.get(data_endpt + data_uuid, headers=headers)
    
    # The file name can be found in the header within the Content-Disposition key.
    response_head_cd = response.headers["Content-Disposition"]
    
    file_name = re.findall("filename=(.+)", response_head_cd)[0]
    
    with open(file_name, "wb") as output_file:
        output_file.write(response.content)
    ```

[Download Python Script](scripts/Authentication_Tokens.py)


For more information about authentication tokens, including token expiration and rotation, see [Data Security](../../Data/Data_Security/Data_Security.md#authentication-tokens).

>**NOTE:** The authentication token should be kept in a secure location, as it allows access to all data accessible by the associated user account.


# Introduction to GDC GraphQL
[GraphQL](https://graphql.org/) is a query language for APIs. The [GDC REST API](https://docs.gdc.cancer.gov/API/Users_Guide/Getting_Started/) has structured and specifically defined query parameters as well as endpoints that have set requests and responses. The GDC GraphQL provides advanced GDC users greater flexibility to specify the data they would like to be returned. This allows queries to be cleaner and easier to understand, especially when combining multiple queries into one request.

To produce queries in a visual interface, the GDC recommends using [GraphiQL](https://github.com/graphql/graphiql). See below for the correct endpoint URLs.

## Using GDC GraphQL vs GDC REST API

If the query requires only a subset of the data to be returned, GDC GraphQL may speed up requests as GraphQL queries return only the specified data. This may require less work on the GDC server-side to fulfill those requests. Conversely, if an entire data-set is required for each request, the GDC REST API may be a better fit. No matter which method is used, the data returned by the GDC REST API and the GraphQL query will be identical as they query the same source.

## GDC GraphQL Overview
GraphQL is not a storage model or a database query language. The graph refers to graph structures defined in the schema, where nodes define objects and edges define relationships between objects. The API traverses and returns application data based on the schema definitions, independent of how the data is stored.

## GDC GraphQL Endpoints

The GDC GraphQL has only two endpoints:
* __GDC Search and Retrieval Endpoint:__ https://api.gdc.cancer.gov/v0/graphql
* __GDC Submission Endpoint:__ https://api.gdc.cancer.gov/v0/submission/graphql

This page covers the search and retrieval endpoint, see the [GDC Submission API](Submission.md) for additional details on the submission endpoint.

## GDC GraphQL Schema
All GDC GraphQL queries are validated and executed against the [GDC GraphQL schema]( https://github.com/NCI-GDC/portal-ui/blob/92f0dfa17838746093c3c011141d08391016da91/data/schema.graphql). Because the GraphQL parameters are discoverable, the GDC GraphQL schema can be queried for details about itself.

The `__schema` keyword can be queried to list all types defined in the schema and retrieve details about each:

=== "GraphQL"

    ```graphql
    {
      __schema {
        types {
          name
          kind
          fields {
            name
          }
        }
      }
    }
    ```

The `__type` keyword can also be queried to retrieve details about any type such as "Explore" or "Case":

=== "Explore"

    ```graphql

    {
      __type(name: "Explore") {
        name
        kind
        description
        fields {
          name
        }
      }
    }
    ```

=== "Case"

    ```graphql
    {
      __type(name: "Case") {
        name
        kind
        description
        fields {
          name
        }
      }
    }
    ```

## Basic GraphQL queries in GDC
The two types of allowed operations in GDC GraphQL API are queries and mutations. Comparing GraphQL to REST, queries operate like `GET` requests, while mutations operate like `POST`/`PATCH`/`DELETE`.

__Note:__ This guide does not cover GDC GraphQL mutation operations.

GraphQL queries return only the data that is specified. Queries are built by specifying fields within fields (also called nested *subfields*) until only scalars are returned.  Scalars are primitive values such as: `Int`, `Float`, `String`, `Boolean`, or `ID`.

## Anatomy of a typical GDC GraphQL Query

 [![GraqhQL-Query](images/graphql-query.png)](images/graphql-query.png "Click to see the full image.")

- __Operation type:__ Describes what type of operation that is being performed, such as query, mutation, or subscription
- __Operation name:__ Similar to a function name, gives queries meaningful names
- __Field:__ Denotes the specific fields on objects that will be included with the response data
- __Arguments:__ A set of key-value pairs associated with a specific field. The parameters can be literal values or variables. __NOTE:__ Arguments can appear on any field, even fields nested deep in an operation.
- __Variable definitions:__ As GraphQL is strong typed, it validates the variable being passed dynamically. __NOTE:__ Variables are passed separately from the query document as JSON such as:

```json
    { "filters_1": {"op":"in","content":{"field":"projects.program.name","value":["TARGET"]}}}
```

## GDC GraphQL Examples
### Nodes And Edges Example
A very powerful feature of GDC GraphQL API is that the graph structures defined in the [GDC GraphQL schema]( https://github.com/NCI-GDC/portal-ui/blob/92f0dfa17838746093c3c011141d08391016da91/data/schema.graphql ) can be queried and traversed. In these queries, nodes define objects and edges define relationships between objects.

=== "GraphQL"

    ```graphql
    query PROJECTS_EDGES($filters_1: FiltersArgument) {
      projects {
        hits(filters: $filters_1) {
          total
          edges {
            node {
              primary_site
              disease_type
              project_id
              dbgap_accession_number
            }
          }
        }
      }
    }

        variable:
        { "filters_1": {"op": "in", "content": {"field": "projects.primary_site", "value": ["Kidney"]}}}
    ```

### Query Case File Counts

=== "GraphQL"

    ```graphql
    query CaseFileCounts($filters: FiltersArgument) {
      viewer {
        repository {
          cases {
            hits(first: 1, filters: $filters) {
              edges {
                node {
                  case_id
                  files {
                    hits(first: 0) {
                      total
                    }
                  }
                  summary {
                    experimental_strategies {
                      experimental_strategy
                      file_count
                    }
                    data_categories {
                      data_category
                      file_count
                    }
                  }
                }
              }
            }
          }
        }
      }
    }

    variable:
    {"filters":{"op":"in","content":{"field":"cases.case_id","value":["dcd5860c-7e3a-44f3-a732-fe92fe3fe300"]}}}
    ```

### Query Simple Static Mutations Based on Gene IDs

=== "GraphQL"

    ```graphql
    query PROJECTS_EDGES($filters_2: FiltersArgument) {
      explore {
        ssms {
          hits(filters: $filters_2) {
            total
            edges {
              node {
                ssm_id
                gene_aa_change
              }
            }
          }
        }
      }
    }

    variable:
    {"filters_2": {"op":"in","content":{"field":"consequence.transcript.gene.gene_id","value":["ENSG00000155657"]}}}
    ```


# Additional Examples

## Data Search and Retrieval

### Endpoint Examples

This section contains additional examples for using endpoints.

#### Project Endpoint Example

This example is a query for Projects contained in GDC. It returns only the first five projects sorted by project name.

=== "Query"

    ```shell
    curl 'https://api.gdc.cancer.gov/projects?from=0&size=5&sort=project.name:asc&pretty=true'
    ```

=== "Response"

    ```json
    {
      "data": {
        "hits": [
          {
            "state": "legacy",
            "project_id": "TARGET-AML",
            "primary_site": "Blood",
            "disease_type": "Acute Myeloid Leukemia",
            "name": "Acute Myeloid Leukemia"
          },
          {
            "state": "legacy",
            "project_id": "TCGA-LAML",
            "primary_site": "Blood",
            "disease_type": "Acute Myeloid Leukemia",
            "name": "Acute Myeloid Leukemia"
          },
          {
            "state": "legacy",
            "project_id": "TARGET-AML-IF",
            "primary_site": "Blood",
            "disease_type": "Acute Myeloid Leukemia Induction Failure",
            "name": "Acute Myeloid Leukemia Induction Failure"
          },
          {
            "state": "legacy",
            "project_id": "TARGET-ALL-P2",
            "primary_site": "Blood",
            "disease_type": "Acute Lymphoblastic Leukemia",
            "name": "Acute Lymphoblastic Leukemia - Phase II"
          },
          {
            "state": "legacy",
            "project_id": "TARGET-ALL-P1",
            "primary_site": "Blood",
            "disease_type": "Acute Lymphoblastic Leukemia",
            "name": "Acute Lymphoblastic Leukemia - Phase I"
          }
        ],
        "pagination": {
          "count": 5,
          "sort": "project.name:asc",
          "from": 0,
          "pages": 10,
          "total": 46,
          "page": 1,
          "size": 5
        }
      },
      "warnings": {}
    }
    ```
#### Files Endpoint Example

This example is a query for files contained in GDC. It returns only the first two files, sorted by file size, from smallest to largest.

=== "Query"

    ``` shell
    curl 'https://api.gdc.cancer.gov/files?from=0&size=2&sort=file_size:asc&pretty=true'
    ```

=== "Response"

    ```json
    {
      "data": {
        "hits": [
          {
            "data_type": "Raw Simple Somatic Mutation",
            "updated_datetime": "2017-03-04T16:45:40.925270-06:00",
            "file_name": "9f78a291-2d50-472c-8f56-5f8fbd09ab2a.snp.Somatic.hc.vcf.gz",
            "submitter_id": "TCGA-13-0757-01A-01W-0371-08_TCGA-13-0757-10A-01W-0371-08_varscan",
            "file_id": "9f78a291-2d50-472c-8f56-5f8fbd09ab2a",
            "file_size": 1120,
            "id": "9f78a291-2d50-472c-8f56-5f8fbd09ab2a",
            "created_datetime": "2016-05-04T14:50:54.560567-05:00",
            "md5sum": "13c1ceb3519615e2c67128b350365fbf",
            "data_format": "VCF",
            "acl": [
              "phs000178"
            ],
            "access": "controlled",
            "state": "live",
            "data_category": "Simple Nucleotide Variation",
            "type": "simple_somatic_mutation",
            "file_state": "submitted",
            "experimental_strategy": "WXS"
          },
          {
            "data_type": "Raw Simple Somatic Mutation",
            "updated_datetime": "2017-03-04T16:45:40.925270-06:00",
            "file_name": "7780009b-abb6-460b-903d-accdac626c2e.snp.Somatic.hc.vcf.gz",
            "submitter_id": "TCGA-HC-8261-01A-11D-2260-08_TCGA-HC-8261-10A-01D-2260-08_varscan",
            "file_id": "7780009b-abb6-460b-903d-accdac626c2e",
            "file_size": 1237,
            "id": "7780009b-abb6-460b-903d-accdac626c2e",
            "created_datetime": "2016-05-08T13:54:38.369393-05:00",
            "md5sum": "fd9bb46c8022b96af730c48dc00e2c41",
            "data_format": "VCF",
            "acl": [
              "phs000178"
            ],
            "access": "controlled",
            "state": "live",
            "data_category": "Simple Nucleotide Variation",
            "type": "simple_somatic_mutation",
            "file_state": "submitted",
            "experimental_strategy": "WXS"
          }
        ],
        "pagination": {
          "count": 2,
          "sort": "file_size:asc",
          "from": 0,
          "page": 1,
          "total": 274724,
          "pages": 137362,
          "size": 2
        }
      },
      "warnings": {}
    }
    ```

#### Cases Endpoint Examples

This example is a query for cases contained in GDC. It returns only the first five files.

=== "Query"

    ```shell
    curl 'https://api.gdc.cancer.gov/cases?from=0&size=5&pretty=true'
    ```

=== "Response"

    ```json
    {
      "data": {
        "hits": [
          {
            "updated_datetime": "2017-03-09T10:01:14.834935-06:00",
            "submitter_analyte_ids": [
              "TCGA-ER-A193-06A-12D",
              "TCGA-ER-A193-06A-12R",
              "TCGA-ER-A193-06A-12W",
              "TCGA-ER-A193-10A-01W",
              "TCGA-ER-A193-10A-01D"
            ],
            "analyte_ids": [
              "62e14ca4-95f5-4af3-848f-83f7273c3b70",
              "6178b8aa-6afb-4951-bc92-bf9bfc57b9c7",
              "e16b701c-7809-4fb5-a9e0-4ff71e5d1d84",
              "5bfa8c9f-6797-4b2b-9122-854f8ab3bbba",
              "9b73d64e-c973-45b6-be31-a486fb8d1708"
            ],
            "submitter_id": "TCGA-ER-A193",
            "case_id": "8ab09143-daf6-40a9-85d3-0fe9de7b3e06",
            "id": "8ab09143-daf6-40a9-85d3-0fe9de7b3e06",
            "disease_type": "Skin Cutaneous Melanoma",
            "sample_ids": [
              "378b3d8a-adbb-4912-a0bf-6b74a282113e",
              "7a384d44-8b05-4197-9921-7d020ada2437"
            ],
            "portion_ids": [
              "6680bbf2-9cf1-4f93-9ec3-04318cffb5ba",
              "690d3b12-a61d-42fd-af2a-5a7a9a3e5de8",
              "824d724e-6836-423e-a751-fee3260ef4d2"
            ],
            "submitter_portion_ids": [
              "TCGA-ER-A193-06A-21-A20N-20",
              "TCGA-ER-A193-10A-01",
              "TCGA-ER-A193-06A-12"
            ],
            "created_datetime": null,
            "slide_ids": [
              "d2751354-a8b7-4f7a-a4f1-d062de5ceb14"
            ],
            "state": "live",
            "aliquot_ids": [
              "dc9f9544-6c76-4b45-b5c3-dd2fecd5acfe",
              "390b3574-ba23-4ecb-acf8-f5ad8a958bd2",
              "33f43961-b32d-46fc-ba11-264f1101e78d",
              "cd17367c-3270-42ae-8ac5-941a3453ea33",
              "b17269a2-79aa-459e-9c3d-589b7efe6fd9",
              "28a7d729-7555-4545-924b-3dec49b54230",
              "13256e77-0b0b-49e3-9959-3b6730d68732",
              "87ca642a-dd4c-47ea-b81f-2d3402f2157a",
              "8a1bfe0e-c97a-41c4-815f-cf5bb5cfc69f",
              "5e1e9c82-99fd-49de-9dfb-a349d4d8ac94",
              "67f00459-e423-4900-be23-9283b0478620",
              "d939c477-a01f-4d54-bcfb-c9fdd957f2ec"
            ],
            "primary_site": "Skin",
            "submitter_aliquot_ids": [
              "TCGA-ER-A193-06A-12D-A18Y-02",
              "TCGA-ER-A193-10A-01D-A193-01",
              "TCGA-ER-A193-10A-01D-A190-02",
              "TCGA-ER-A193-06A-12D-A197-08",
              "TCGA-ER-A193-06A-12R-A18S-07",
              "TCGA-ER-A193-06A-12W-A20H-08",
              "TCGA-ER-A193-10A-01D-A199-08",
              "TCGA-ER-A193-10A-01D-A38R-08",
              "TCGA-ER-A193-10A-01W-A20J-08",
              "TCGA-ER-A193-06A-12R-A18V-13",
              "TCGA-ER-A193-06A-12D-A19C-05",
              "TCGA-ER-A193-06A-12D-A191-01"
            ],
            "submitter_sample_ids": [
              "TCGA-ER-A193-10A",
              "TCGA-ER-A193-06A"
            ],
            "submitter_slide_ids": [
              "TCGA-ER-A193-06A-01-TSA"
            ]
          },
          {
            "updated_datetime": "2017-03-04T16:39:19.244769-06:00",
            "submitter_analyte_ids": [
              "TCGA-VR-AA4G-10A-01W",
              "TCGA-VR-AA4G-01A-11R",
              "TCGA-VR-AA4G-10A-01D",
              "TCGA-VR-AA4G-01A-11D",
              "TCGA-VR-AA4G-01A-11W"
            ],
            "analyte_ids": [
              "152d7d7a-c746-4b58-8c3f-4252454c7b7c",
              "9090d556-bd2e-4851-8a0c-46e22cc61408",
              "7118f4c3-b635-4428-8240-8db85281f2d9",
              "1d8223ff-685a-4427-a3d1-f53887f2a19d",
              "60dfb30a-bea0-426d-b11d-d5813ba39cfc"
            ],
            "submitter_id": "TCGA-VR-AA4G",
            "case_id": "df5bd25c-d70b-4126-89cb-6c838044ae3b",
            "id": "df5bd25c-d70b-4126-89cb-6c838044ae3b",
            "disease_type": "Esophageal Carcinoma",
            "sample_ids": [
              "21456849-38a9-4190-9ece-ed69b3c24fda",
              "6ee6d239-2af6-41cd-bc32-c5cdaf7742b0"
            ],
            "portion_ids": [
              "484b40d5-d77c-4e6f-9e80-1ef27ffbc8a5",
              "fdc56e67-52ab-44fd-823a-5a3124876ff7"
            ],
            "submitter_portion_ids": [
              "TCGA-VR-AA4G-10A-01",
              "TCGA-VR-AA4G-01A-11"
            ],
            "created_datetime": null,
            "slide_ids": [
              "e950eba2-7d6e-4ffd-a2d5-e0eb6486848a"
            ],
            "state": "live",
            "aliquot_ids": [
              "db6beed3-a5a2-469f-8dc8-00d838c1f37f",
              "f5db4d36-034b-429b-a7be-26a872b702ee",
              "16421a96-b843-4f7e-9f7c-64d2fb5b2a25",
              "5d938cb5-7064-40bc-877d-57faa94c3333",
              "d231404d-ece5-43c0-a8a3-e9f294ceb777",
              "8c77dc3e-2ea3-4626-88f5-e74f242bedf3",
              "993624d4-1c28-41a5-a0b6-094a0e442c36",
              "105a18c9-df7e-4573-b1a2-6a987e57d553",
              "af81c3bb-3b9e-41cb-b85a-b55c6437d05b",
              "38938066-5fd9-415c-b00e-65efff14085e",
              "20139afe-ad04-4571-b779-0c4a51e74ada"
            ],
            "primary_site": "Esophagus",
            "submitter_aliquot_ids": [
              "TCGA-VR-AA4G-10A-01W-A44M-09",
              "TCGA-VR-AA4G-01A-11D-A37B-01",
              "TCGA-VR-AA4G-01A-11D-A37D-05",
              "TCGA-VR-AA4G-10A-01D-A37F-09",
              "TCGA-VR-AA4G-01A-11D-A37R-26",
              "TCGA-VR-AA4G-01A-11R-A37J-13",
              "TCGA-VR-AA4G-01A-11R-A37I-31",
              "TCGA-VR-AA4G-01A-11D-A37C-09",
              "TCGA-VR-AA4G-10A-01D-A37R-26",
              "TCGA-VR-AA4G-10A-01D-A37E-01",
              "TCGA-VR-AA4G-01A-11W-A44L-09"
            ],
            "submitter_sample_ids": [
              "TCGA-VR-AA4G-01A",
              "TCGA-VR-AA4G-10A"
            ],
            "submitter_slide_ids": [
              "TCGA-VR-AA4G-01A-01-TS1"
            ]
          },
          {
            "updated_datetime": "2017-03-04T16:39:19.244769-06:00",
            "submitter_analyte_ids": [
              "TCGA-D1-A174-01A-11D",
              "TCGA-D1-A174-01A-11W",
              "TCGA-D1-A174-10A-01D",
              "TCGA-D1-A174-10A-01W",
              "TCGA-D1-A174-01A-11R"
            ],
            "analyte_ids": [
              "96203028-f824-4a90-9758-22340285062c",
              "f4878e33-b773-43b5-83a5-9fd8e539e668",
              "8627ccd0-0575-4d03-b589-ca45642d523d",
              "1183f7c6-992d-4084-946e-adce7c52f9cc",
              "5343f6a8-8ac2-4446-ace5-a27d21e76844"
            ],
            "submitter_id": "TCGA-D1-A174",
            "case_id": "fc7315b0-9f48-4206-b197-2268c0518eb4",
            "id": "fc7315b0-9f48-4206-b197-2268c0518eb4",
            "disease_type": "Uterine Corpus Endometrial Carcinoma",
            "sample_ids": [
              "df9a1f44-9b3f-48b2-96af-54aaabdfd243",
              "ad5a9cb6-b3f9-4651-b6d1-13c78010bd88"
            ],
            "portion_ids": [
              "79dd516c-bae3-4f6e-b4cb-901de030acb7",
              "6e55e6d9-902f-439b-b6f1-ca296c123fd3"
            ],
            "submitter_portion_ids": [
              "TCGA-D1-A174-01A-11",
              "TCGA-D1-A174-10A-01"
            ],
            "created_datetime": null,
            "slide_ids": [
              "7602727e-b46d-40fc-bd03-5ccf631041f8"
            ],
            "state": "live",
            "aliquot_ids": [
              "5c15542b-cd63-44b5-b278-e211410fb0aa",
              "d661cfb9-248a-49e6-b0db-865ca257e8dc",
              "83bd3bdb-9bd3-46fa-888c-f6f5efec530f",
              "c46551c9-c0d0-4140-8d0a-946b53e504e2",
              "96b511df-3a69-4168-908c-662060b4f976",
              "0182d4e1-f835-46b5-a8f0-53decf5868de",
              "e9563a06-0b86-4986-976e-43d4040f1d61",
              "6bb2de6e-5b85-4e97-a930-1f2c6bf663a1",
              "f6ee5558-a1b6-4b11-8f48-c17186fff39a",
              "67f6f0d9-6581-4946-a9c7-a6629da86888",
              "39e9a948-054a-4b50-b108-7d7aee686363",
              "ddb4ca26-655d-4bdc-a00d-7caf26cadafe"
            ],
            "primary_site": "Uterus",
            "submitter_aliquot_ids": [
              "TCGA-D1-A174-01A-11D-A12F-02",
              "TCGA-D1-A174-01A-01D-YYYY-23",
              "TCGA-D1-A174-01A-11W-A139-09",
              "TCGA-D1-A174-10A-01W-A139-09",
              "TCGA-D1-A174-01A-11D-A12K-05",
              "TCGA-D1-A174-10A-01D-A12F-02",
              "TCGA-D1-A174-10A-01D-A12G-01",
              "TCGA-D1-A174-01A-11R-A12I-07",
              "TCGA-D1-A174-01A-11D-A12J-09",
              "TCGA-D1-A174-10A-01D-A12J-09",
              "TCGA-D1-A174-01A-11R-A12H-13",
              "TCGA-D1-A174-01A-11D-A12G-01"
            ],
            "submitter_sample_ids": [
              "TCGA-D1-A174-01A",
              "TCGA-D1-A174-10A"
            ],
            "submitter_slide_ids": [
              "TCGA-D1-A174-01A-01-TS1"
            ]
          },
          {
            "updated_datetime": "2017-03-04T16:39:19.244769-06:00",
            "submitter_analyte_ids": [
              "TCGA-XM-A8RL-10A-01D",
              "TCGA-XM-A8RL-01A-11R",
              "TCGA-XM-A8RL-01A-11D"
            ],
            "analyte_ids": [
              "2c483e72-92b0-425d-ac1b-b75a169cf531",
              "57f88d4f-8b3a-4349-88b0-3d2e58a95ed9",
              "499bfbe1-639c-479c-abaa-42cbb11c0568"
            ],
            "submitter_id": "TCGA-XM-A8RL",
            "case_id": "dd240b82-b1d6-4c0f-aa3e-6fcfe1364ec1",
            "id": "dd240b82-b1d6-4c0f-aa3e-6fcfe1364ec1",
            "disease_type": "Thymoma",
            "sample_ids": [
              "cb091cc1-7bbe-43a4-8460-01215af3aa21",
              "cabc9729-c1e1-4f08-9959-985dcb7a00d5"
            ],
            "portion_ids": [
              "e8ea57c9-729e-46ea-b1da-2db7a00b02bc",
              "8e2edb92-753f-4cb0-a5b8-8c45dbefaf36",
              "650fa4f2-9fa2-4d3a-8b63-ff4a9bd8c33e"
            ],
            "submitter_portion_ids": [
              "TCGA-XM-A8RL-01A-21-A45R-20",
              "TCGA-XM-A8RL-10A-01",
              "TCGA-XM-A8RL-01A-11"
            ],
            "created_datetime": null,
            "slide_ids": [
              "08cedd34-aafd-4b47-891f-cf66ee1f627b"
            ],
            "state": "live",
            "aliquot_ids": [
              "df9d8553-8d5b-4c65-8b28-74030a8f8e76",
              "47b7f634-b36f-49e9-a4dc-d8f5508fdc0a",
              "e692ebed-9721-40db-8986-fcaba07d68f1",
              "189ee080-95d1-4ccb-8618-955605c7bd55",
              "83af7ff3-45be-4378-a8b5-5dff3584e95d",
              "42ebb1f0-e236-48ae-847f-69a153969903",
              "e8a4938f-6b93-4ad1-9324-31c97dd1d477"
            ],
            "primary_site": "Thymus",
            "submitter_aliquot_ids": [
              "TCGA-XM-A8RL-10A-01D-A426-09",
              "TCGA-XM-A8RL-01A-11D-A423-09",
              "TCGA-XM-A8RL-01A-11D-A422-01",
              "TCGA-XM-A8RL-01A-11R-A42C-07",
              "TCGA-XM-A8RL-10A-01D-A425-01",
              "TCGA-XM-A8RL-01A-11R-A42W-13",
              "TCGA-XM-A8RL-01A-11D-A424-05"
            ],
            "submitter_sample_ids": [
              "TCGA-XM-A8RL-10A",
              "TCGA-XM-A8RL-01A"
            ],
            "submitter_slide_ids": [
              "TCGA-XM-A8RL-01A-01-TSA"
            ]
          },
          {
            "updated_datetime": "2017-03-04T16:39:19.244769-06:00",
            "submitter_analyte_ids": [
              "TCGA-B0-5120-01A-01W",
              "TCGA-B0-5120-01A-01D",
              "TCGA-B0-5120-01A-01R",
              "TCGA-B0-5120-11A-01W",
              "TCGA-B0-5120-11A-01D"
            ],
            "analyte_ids": [
              "996336e6-fad7-4100-96ae-60adb5c276f1",
              "0eb7da02-0b90-4f6d-abd2-b048a9cb2995",
              "fa2861b9-67c1-486a-a1e0-95d8f8adf65b",
              "7e9f5639-a462-493e-98f8-1b7aeee383c7",
              "d51e9fd4-0c99-49ec-9de5-db3946b0bf43"
            ],
            "submitter_id": "TCGA-B0-5120",
            "case_id": "c5bf474c-6919-47b4-ba59-34ab20c087d5",
            "id": "c5bf474c-6919-47b4-ba59-34ab20c087d5",
            "disease_type": "Kidney Renal Clear Cell Carcinoma",
            "sample_ids": [
              "b50d3c6f-fdec-488b-ab26-a9b690fad34f",
              "f3148210-ecae-4314-b5f8-9bee2315a093"
            ],
            "portion_ids": [
              "b8fcbf00-4c5a-42c3-95e9-fb6e169a8da9",
              "34443e91-0210-4477-9511-53026ae62b38",
              "e466f011-79a1-4158-b796-f8e9dda32d68"
            ],
            "submitter_portion_ids": [
              "TCGA-B0-5120-01A-01",
              "TCGA-B0-5120-11A-01",
              "TCGA-B0-5120-01A-21-1740-20"
            ],
            "created_datetime": null,
            "slide_ids": [
              "e5a29e92-4125-4acb-a797-86822b4961a2",
              "78d873e0-037f-4aef-8725-7c651598b1f8",
              "43d8cec7-f5a0-45d5-a5f8-cc77d6b7b539"
            ],
            "state": "live",
            "aliquot_ids": [
              "b35280fe-dbfa-4e45-8f49-3d0489e68743",
              "a2e3a2f2-c32b-44a1-9b29-911145d700b8",
              "a064d108-e8b2-46fa-b277-0a7a89904a3a",
              "59be71a1-50e3-4565-852a-173afc8a6851",
              "136dff0e-b181-49c9-8305-b3289625ea2e",
              "8fbb983b-53ad-44a9-976a-7945628eaa51",
              "cecf40f8-7301-4db9-b276-a14317d4dd59",
              "fac8b066-bf2c-4f08-b42b-251035596a28",
              "fa55c92f-54e8-436b-b8c4-04cb68a24e93",
              "007e3098-aaf9-4ee7-9ae1-f94b131a5ae0",
              "6ce58fbc-6742-4ade-84b0-cd025266e030",
              "9668e15e-a3fa-4ead-ad42-322c5700e0db",
              "c1167003-0730-41d5-bdd5-1cbf501c1463",
              "73aab074-cbd1-45f2-8266-9ef6f7c559bc"
            ],
            "primary_site": "Kidney",
            "submitter_aliquot_ids": [
              "TCGA-B0-5120-11A-01D-1416-02",
              "TCGA-B0-5120-11A-01D-2099-10",
              "TCGA-B0-5120-11A-01D-1418-05",
              "TCGA-B0-5120-01A-01W-1475-10",
              "TCGA-B0-5120-01A-01D-1421-08",
              "TCGA-B0-5120-01A-01D-1416-02",
              "TCGA-B0-5120-01A-01R-1419-13",
              "TCGA-B0-5120-01A-01R-1420-07",
              "TCGA-B0-5120-11A-01D-1421-08",
              "TCGA-B0-5120-01A-01D-1417-01",
              "TCGA-B0-5120-01A-01D-1418-05",
              "TCGA-B0-5120-11A-01W-1475-10",
              "TCGA-B0-5120-01A-01D-2099-10",
              "TCGA-B0-5120-11A-01D-1417-01"
            ],
            "submitter_sample_ids": [
              "TCGA-B0-5120-11A",
              "TCGA-B0-5120-01A"
            ],
            "submitter_slide_ids": [
              "TCGA-B0-5120-11A-01-TS1",
              "TCGA-B0-5120-01A-01-BS1",
              "TCGA-B0-5120-01A-01-TS1"
            ]
          }
        ],
        "pagination": {
          "count": 5,
          "sort": "",
          "from": 0,
          "page": 1,
          "total": 14551,
          "pages": 2911,
          "size": 5
        }
      },
      "warnings": {}
    }
    ```

This example is a query for cases with a height (follow_ups.other_clinical_attributes.height) equal to or greater than 225, and return the height and project_id the case belongs to. Note: height is expressed in centimeters. 

=== "Filter"

```json
{
            "op":">=",
            "content":{
                "field": "follow_ups.other_clinical_attributes.height",
                "value": 225
    }
}
```

=== "Query"

```shell
curl 'https://api.gdc.cancer.gov/cases?filters=%7B%22op%22%3A%20%22%3E%3D%22%2C%20%22content%22%3A%20%7B%22field%22%3A%20%22follow_ups.other_clinical_attributes.height%22%2C%20%22value%22%3A%20225%7D%7D&fields=follow_ups.other_clinical_attributes.height,project.project_id&pretty=true'
```

=== "Response"

```json
{
  "data": {
    "hits": [
      {
        "id": "56a06787-58e9-4cd4-a027-0c0fabcc0983", 
        "follow_ups": [
          {
            "other_clinical_attributes": [
              {
                "height": 243
              }
            ]
          }
        ], 
        "project": {
          "project_id": "MMRF-COMMPASS"
        }
      }
    ], 
    "pagination": {
      "count": 1, 
      "total": 1, 
      "size": 10, 
      "from": 0, 
      "sort": "", 
      "page": 1, 
      "pages": 1
    }
  }, 
  "warnings": {}
}
```

This example is a query to filter for first 5 cases that have Other Clinical Attribute entities with data and to display the Other Clinical Attribute data and project_id. Note: the `not` operator filters for fields that are *not* missing or *not* empty. 

=== "Filter"

```json
{
            "op":"not",
            "content":{
                "field": "follow_ups.other_clinical_attributes.submitter_id",
    }
}
```

=== "Query"

```shell
curl 'https://api.gdc.cancer.gov/cases?filters=%7B%22op%22%3A%20%22not%22%2C%20%22content%22%3A%20%7B%22field%22%3A%20%22follow_ups.other_clinical_attributes.submitter_id%22%7D%7D&fields=project.project_id&expand=follow_ups.other_clinical_attributes&size=5&pretty=true'
```

=== "Response"

```json
{
  "data": {
    "hits": [
      {
        "id": "4ad1f10e-b45d-4bed-bc99-f3d48254623d", 
        "follow_ups": [
          {
            "other_clinical_attributes": [
              {
                "timepoint_category": "Initial Diagnosis", 
                "updated_datetime": "2024-07-09T13:45:54.127250-05:00", 
                "other_clinical_attribute_id": "b1828570-fe96-4b74-8e35-1d7b7c5b25b8", 
                "weight": 91, 
                "submitter_id": "CTSP-AD2E_other_clinical_attribute", 
                "state": "released", 
                "created_datetime": "2023-11-30T10:08:31.042703-06:00", 
                "bmi": 39.4, 
                "height": 152
              }
            ]
          }
        ], 
        "project": {
          "project_id": "CTSP-DLBCL1"
        }
      }, 
      {
        "id": "1cfe090d-9a34-4984-a0d8-392ab83635b3", 
        "follow_ups": [
          {
            "other_clinical_attributes": [
              {
                "timepoint_category": "Initial Diagnosis", 
                "updated_datetime": "2024-07-09T13:45:54.127250-05:00", 
                "other_clinical_attribute_id": "c8098f60-394a-422b-a3aa-a3a1c0d9dae7", 
                "weight": 58, 
                "submitter_id": "CTSP-AD4C_other_clinical_attribute", 
                "state": "released", 
                "created_datetime": "2023-11-30T10:19:30.283698-06:00", 
                "bmi": 22.4, 
                "height": 161
              }
            ]
          }
        ], 
        "project": {
          "project_id": "CTSP-DLBCL1"
        }
      }, 
      {
        "id": "297b1cb6-df7d-4e75-a17c-592179be7b71", 
        "follow_ups": [
          {
            "other_clinical_attributes": [
              {
                "timepoint_category": "Initial Diagnosis", 
                "updated_datetime": "2024-07-09T13:45:54.127250-05:00", 
                "other_clinical_attribute_id": "069f5ef5-a488-454d-8b28-ca1f81375006", 
                "weight": 132, 
                "submitter_id": "CTSP-AD1S_other_clinical_attribute", 
                "state": "released", 
                "created_datetime": "2023-11-30T10:01:36.617920-06:00", 
                "bmi": 45.7, 
                "height": 170
              }
            ]
          }
        ], 
        "project": {
          "project_id": "CTSP-DLBCL1"
        }
      }, 
      {
        "id": "dd6f8769-ee57-4171-a353-ceb762541845", 
        "follow_ups": [
          {
            "other_clinical_attributes": [
              {
                "timepoint_category": "Initial Diagnosis", 
                "updated_datetime": "2024-07-09T13:45:54.127250-05:00", 
                "other_clinical_attribute_id": "5c9c4fe0-aedc-4016-90c6-1075f82a139b", 
                "weight": 81, 
                "submitter_id": "CTSP-AD1Z_other_clinical_attribute", 
                "state": "released", 
                "created_datetime": "2023-11-30T10:00:47.753276-06:00", 
                "bmi": 30.1, 
                "height": 164
              }
            ]
          }
        ], 
        "project": {
          "project_id": "CTSP-DLBCL1"
        }
      }, 
      {
        "id": "19ec7505-20c7-4ba2-bf08-4a8875cb2639", 
        "follow_ups": [
          {
            "other_clinical_attributes": [
              {
                "timepoint_category": "Initial Diagnosis", 
                "updated_datetime": "2024-07-09T13:45:54.127250-05:00", 
                "other_clinical_attribute_id": "89938ef7-aa86-49e1-98b6-fd6ee6a4278a", 
                "weight": 47, 
                "submitter_id": "CTSP-AD1R_other_clinical_attribute", 
                "state": "released", 
                "created_datetime": "2023-11-30T10:01:55.498836-06:00", 
                "bmi": 20.3, 
                "height": 152
              }
            ]
          }
        ], 
        "project": {
          "project_id": "CTSP-DLBCL1"
        }
      }
    ], 
    "pagination": {
      "count": 5, 
      "total": 3306, 
      "size": 5, 
      "from": 0, 
      "sort": "", 
      "page": 1, 
      "pages": 662
    }
  }, 
  "warnings": {}
}
```


#### Annotations Endpoint Example

This example is a query for annotations contained in the GDC. It returns only the first two annotations.

=== "Query"

    ```shell
    curl 'https://api.gdc.cancer.gov/annotations?from=0&size=2&pretty=true'
    ```

=== "Response"

    ```json
    {
      "data": {
        "hits": [
          {
            "category": "History of unacceptable prior treatment related to a prior/other malignancy",
            "status": "Approved",
            "entity_id": "51c37449-6a2e-4c3d-a7cc-06f901e1224f",
            "classification": "Notification",
            "entity_type": "case",
            "created_datetime": "2014-06-16T00:00:00",
            "annotation_id": "3d086829-de62-5d08-b848-ce0724188ff0",
            "notes": "unknown treatment history",
            "updated_datetime": "2017-03-09T12:32:36.305475-06:00",
            "submitter_id": "20743",
            "state": "submitted",
            "case_id": "51c37449-6a2e-4c3d-a7cc-06f901e1224f",
            "case_submitter_id": "TCGA-AG-A014",
            "entity_submitter_id": "TCGA-AG-A014",
            "id": "3d086829-de62-5d08-b848-ce0724188ff0"
          },
          {
            "category": "Center QC failed",
            "status": "Approved",
            "entity_id": "733f0607-6c6b-4385-9868-fa6f155a9a2e",
            "classification": "CenterNotification",
            "entity_type": "aliquot",
            "created_datetime": "2012-07-20T00:00:00",
            "annotation_id": "5cf05f41-ce70-58a3-8ecb-6bfaf6264437",
            "notes": "RNA-seq:INSUFFICIENT INPUT MATERIAL,LOW SEQUENCE YIELD/DIVERSITY;LOW 5/3 COVERAGE RATIO",
            "updated_datetime": "2017-03-09T13:51:45.396638-06:00",
            "submitter_id": "8764",
            "state": "submitted",
            "case_id": "3e8a51bf-7e1f-4eab-af83-3c60d04db1bf",
            "case_submitter_id": "TCGA-13-0913",
            "entity_submitter_id": "TCGA-13-0913-02A-01R-1564-13",
            "id": "5cf05f41-ce70-58a3-8ecb-6bfaf6264437"
          }
        ],
        "pagination": {
          "count": 2,
          "sort": "",
          "from": 0,
          "page": 1,
          "total": 2361,
          "pages": 1181,
          "size": 2
        }
      },
      "warnings": {}
    }
    ```

### Filters Examples

This section contains additional examples for using the `filters` parameter.

#### Example: Basic syntax

The following is an example of `filters` syntax, including the JSON object passed to the `filters` parameter, the corresponding API query, and the JSON object returned by the API. The example finds projects where the primary site is Blood.

=== "Filter"
    
    ```json
    {
      "op": "and",
      "content": [
        {
          "op": "in",
          "content": {
            "field": "primary_site",
            "value": [
              "Blood"
            ]
          }
        }
      ]
    }
    ```

=== "Query"

    ```shell
    curl 'https://api.gdc.cancer.gov/projects?filters=%7b%0d%0a++%22op%22%3a+%22and%22%2c%0d%0a++%22content%22%3a+%5b%0d%0a++++%7b%0d%0a++++++%22op%22%3a+%22in%22%2c%0d%0a++++++%22content%22%3a+%7b%0d%0a++++++++%22field%22%3a+%22primary_site%22%2c%0d%0a++++++++%22value%22%3a+%5b%0d%0a++++++++++%22Blood%22%0d%0a++++++++%5d%0d%0a++++++%7d%0d%0a++++%7d%0d%0a++%5d%0d%0a%7d&pretty=true'
    ```

=== "Response"

    ```json
    {
      "data": {
        "hits": [
          {
            "dbgap_accession_number": "phs000465",
            "disease_type": [
              "Acute Myeloid Leukemia"
            ],
            "released": true,
            "state": "legacy",
            "primary_site": [
              "Blood"
            ],
            "project_id": "TARGET-AML",
            "id": "TARGET-AML",
            "name": "Acute Myeloid Leukemia"
          }
        ],
        "pagination": {
          "count": 1,
          "sort": "",
          "from": 0,
          "page": 1,
          "total": 1,
          "pages": 1,
          "size": 10
        }
      },
      "warnings": {}
    }
    ```

#### Example: Filter cases keeping only 'male'

This is an example of a value-based filter:

=== "Filter"

    ```json
    {
       "op" : "=" ,
       "content" : {
           "field" : "cases.demographic.gender" ,
           "value" : [ "male" ]
       }
    }
    ```

=== "Query"

    ```shell
    curl 'https://api.gdc.cancer.gov/cases?filters=%7b%0d%0a+++%22op%22+%3a+%22%3d%22+%2c%0d%0a+++%22content%22+%3a+%7b%0d%0a+++++++%22field%22+%3a+%22cases.demographic.gender%22+%2c%0d%0a+++++++%22value%22+%3a+%5b+%22male%22+%5d%0d%0a+++%7d%0d%0a%7d%0d%0a&fields=demographic.gender,case_id&pretty=true'
    ```

#### Example: Filter using a range

This is an example of filtering for age at diagnosis. The request is for cases where the age at diagnosis is between 40 and 70 years. >**Note:** `age_at_diagnosis` is expressed in days.

=== "Filter"

    ```json
    {
        "op": "and",
        "content": [
            {
                "op": ">=",
                "content": {
                    "field": "cases.diagnoses.age_at_diagnosis",
                    "value": [
                        14600
                    ]
                }
            },
            {
                "op": "<=",
                "content": {
                    "field": "cases.diagnoses.age_at_diagnosis",
                    "value": [
                        25550
                    ]
                }
            }
        ]
    }
    ```

=== "Query"

    ```shell
    curl 'https://api.gdc.cancer.gov/cases?filters=%7B%22op%22:%22and%22,%22content%22:%5B%7B%22op%22:%22%3E%3D%22,%22content%22:%7B%22field%22:%22cases.diagnoses.age_at_diagnosis%22,%22value%22:%5B14600%5D%7D%7D,%7B%22op%22:%22%3C%3D%22,%22content%22:%7B%22field%22:%22cases.diagnoses.age_at_diagnosis%22,%22value%22:%5B25550%5D%7D%7D%5D%7D&fields=diagnoses.age_at_diagnosis,case_id&pretty=true'
    ```


#### Example: Multiple fields

Filter projects for primary_site being Kidney or Brain and program.name being TCGA

=== "Filter"

    ```json
    {
         "op" : "and" ,
         "content" : [{
                 "op" : "in" ,
                 "content" : {
                     "field" : "primary_site" ,
                     "value" : [
                         "Kidney" ,
                         "Brain"
                     ]
                 }
             }, {
                 "op" : "in" ,
                 "content" : {
                     "field" : "program.name" ,
                     "value" : [
                         "TCGA"
                     ]
                 }
             }]
    }
    ```

=== "Query"

    ```shell
    curl 'https://api.gdc.cancer.gov/projects?filters=%7B%22op%22%3A%22and%22%2C%22content%22%3A%5B%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22primary_site%22%2C%22value%22%3A%5B%22Kidney%22%2C%22Brain%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22program.name%22%2C%22value%22%3A%5B%22TCGA%22%5D%7D%7D%5D%7D&pretty=true'
    ```


# Appendix C: Format of Submission Requests and Responses

### Format of Submission Request

The general format of JSON objects submitted to the GDC API is as follows:

	{
	    "type": string,
	    "id": string,
	    "submitter_id": string,
	    "<properties>": any type,
	    "<relationship_name>": [
	        {
	            "id": string,
	            "submitter_id": string
	        },
	        ...
	    ]
	}

The request must specify either an `id` or a `submitter_id`.

**`id`**: A string specifying the `id` of the node that the user is creating or updating. This is the persistent GDC UUID4 for the node. If it is preferred to refer to the node using a custom id, users can do so with the `submitter_id` field (described below).

**`submitter_id`**: A string specifying the custom id of the object the user is creating or updating. This is not the official GDC ID for the node.

**`<properties>`**: These key-value pairs will be applied to properties on the referenced node.

**`<relationship_name>`**: A JSON object that specifies a relationship (link) between the node and other nodes. Links are typically established using the `submitter_id` or `id` of the neighboring node.

### Format of API Response to a Submission Request

The following fields are included in all API responses to submission requests.

	{
	  "cases_related_to_created_entities_count": int,
	  "cases_related_to_updated_entities_count": int,
	  "code": int,
	  "created_entity_count": int,
	  "entities": [entities],
	  "entity_error_count": int,
	  "message": string,
	  "success": boolean,
	  "transaction_id": string,
	  "transactional_error_count": int,
	  "transactional_errors": [transactional_errors],
	  "updated_entity_count": int
	}

**`cases_related_to_created_entities_count`**: Number of cases related to the created entities.

**`cases_related_to_updated_entities_count`**: Number of cases related to the updated entities.

**`code`**: The HTTP status code of the response message.

**`created_entity_count`**: Number of entities created.

**`entities`**: A list of entities of the form:

	{
	  "action": string,
	  "errors": [entity_errors],
	  "id": string,
	  "related_cases": [object],
	  "type": string,
	  "unique_keys": [unique_keys],
	  "valid": boolean,
	  "warnings": [object]
	}

*`entity_errors`*: A list of errors that occurred while parsing, validating, or performing a CRUD operation on a
specific entity. Entity errors are of the form:

	{
		"keys": [string],
		"message": string
	}


*`unique_keys`*: Properties, or combinations of properties, that can be used to uniquely identify the node in the GDC.  Unique_keys are of the form:


	{
		"project_id": string,
		"submitter_id": string
	}


**`entity_error_count`**: Number of entities that were not successful.

**`message`**: A human-readable message describing the transaction.

**`success`**: A boolean value stating whether the transaction was successful. If the value is False, then no changes will be made to the database.

**`transaction_id`**: A string specifying the transaction id.

**`transactional_error_count`**: Number of transactional errors that occurred.

**`transactional_errors`**: A list of transactional errors that have occurred. These errors are errors that are not specific to an individual entity. Transactional errors are of the form:

	{
		"message": string
	}

**`updated_entity_count`**: Number of existing entities updated by the transaction.


### Error Types

**`EntityNotFoundError`** A referenced entity was not found among existing entities and entities specified in the transaction.

**`MissingPropertyError`** A required property was not provided.

**`ValidationError`** A provided property did not pass a validation test.

## Status Messages

API responses will contain a status for each entity specified in the request:

**`success`**: The desired transaction was sucessful and the entity's state was modified in the GDC.

**`valid`**: The desired transaction was not sucessful, but the trasaction was not aborted because of this entity.

**`error`**: The desired transaction was not sucessful, and the transaction was aborted because of this entity. This entity did not pass validation or an internal error occured when attempting to complete the transaction. The error state will be accompanied by a list of errors recorded about the entity (see label-error-messages).

>**Note:** GDC API requests are transactional. An error with processing a node specified in the transaction will abort the transaction and will result in no changes being applied for any node involved in the transaction.


# Appendix B: Key Terms

The following table provides definitions and explanations for terms and acronyms relevant to the content presented within this document.


| Term | Definition |
| --- | --- |
| ACL | Access Control List |
| API | Application Programming Interface |
| BAM | Bidirectional Associative Memory |
| BCR | Biospecimen Core Resource |
| CSV | Comma Seperated Values |
| DCC | Data Coordinating Center |
| eRA | Electronic Research Administration |
| GDC | Genomic Data Commons |
| HTTP | Hypertext Transfer Protocol |
| IDE | Integrated Development Environment |
| JSON | JavaScript Object Notation |
| MAGE-TAB | Microarray Gene Expression - Tabular format |
| NCI | National Cancer Institute |
| NIH | National Institutes of Health |
| Op | Operations |
| REST | Representational State Transfer |
| TARGET | Therapeutically Applicable Research to Generate Effective Treatments |
| TCGA | The Cancer Genome Atlas |
| TSV | Tab Seperated Values |
| UUID | Universally Unique Identifier |
| URL | Universal Resource Locator |
| XML | Extensible Markup Language |


# Search and Retrieval

## Introducing Search and Retrieval Requests

The GDC API provides endpoints that search and retrieve information stored in the GDC according to the [GDC Data Model](../../Data/Data_Model/GDC_Data_Model.md). The general format of requests to search & retrieval endpoints is described below.

>**Note:** Queries described in this section work for datasets that have been released to the GDC Data Portal. Unreleased data that is in the process of being submitted to GDC cannot be queried using these methods. See [Submission](Submission.md) to learn how to query unreleased data using GraphQL.

### Components of a Request

A typical search and retrieval API request specifies the following parameters:

- a `filters` parameter, that specifies the search terms for the query
- several parameters that specify the API response, such as:
	- `format` &mdash; specifies response format (JSON, TSV, XML)
	- `fields` &mdash; specifies the which data elements should be returned in the response, if available
	- `size` &mdash; specifies the the maximum number of results to include in the response
	- other parameters are described below.

Requests can be executed using HTTP GET or HTTP POST. GET requests are limited by maximum URL length, so the POST method is recommended for large queries.

### POST Example

The following is an example of an HTTP POST request to the `files` endpoint of the GDC API. It looks for Gene Expression Quantification files associated with specific TCGA cases (represented by TCGA barcodes) and retrieves the associated biospecimen metadata in TSV format.

#### Request

	curl --request POST --header "Content-Type: application/json" --data @Payload 'https://api.gdc.cancer.gov/files' > response.tsv

#### Payload

	{
	    "filters":{
	        "op":"and",
	        "content":[
	            {
	                "op":"in",
	                "content":{
	                    "field":"cases.submitter_id",
	                    "value":[
	                        "TCGA-CK-4948",
	                        "TCGA-D1-A17N",
	                        "TCGA-4V-A9QX",
	                        "TCGA-4V-A9QM"
	                    ]
	                }
	            },
	            {
	                "op":"=",
	                "content":{
	                    "field":"files.data_type",
	                    "value":"Gene Expression Quantification"
	                }
	            }
	        ]
	    },
	    "format":"tsv",
	    "fields":"file_id,file_name,cases.submitter_id,cases.case_id,data_category,data_type,cases.samples.tumor_descriptor,cases.samples.tissue_type,cases.samples.sample_type,cases.samples.submitter_id,cases.samples.sample_id,analysis.workflow_type,cases.project.project_id,cases.samples.portions.analytes.aliquots.aliquot_id,cases.samples.portions.analytes.aliquots.submitter_id",
	    "size":"1000"
	}

Each component of the request is explained below.

### GET Example

The above request can be executed as an HTTP GET:

	https://api.gdc.cancer.gov/files?filters=%7B%22op%22%3A%22and%22%2C%22content%22%3A%5B%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22cases.submitter_id%22%2C%22value%22%3A%5B%22TCGA-CK-4948%22%2C%22TCGA-D1-A17N%22%2C%22TCGA-4V-A9QX%22%2C%22TCGA-4V-A9QM%22%5D%7D%7D%2C%7B%22op%22%3A%22%3D%22%2C%22content%22%3A%7B%22field%22%3A%22files.data_type%22%2C%22value%22%3A%22Gene%20Expression%20Quantification%22%7D%7D%5D%7D&format=tsv&fields=file_id,file_name,cases.submitter_id,cases.case_id,data_category,data_type,cases.samples.tumor_descriptor,cases.samples.tissue_type,cases.samples.sample_type,cases.samples.submitter_id,cases.samples.sample_id,analysis.workflow_type,cases.project.project_id,cases.samples.portions.analytes.aliquots.aliquot_id,cases.samples.portions.analytes.aliquots.submitter_id&size=1000

Each component of the request is explained below.


## Endpoints

The following search and retrieval endpoints are available in the GDC API:

| Endpoints | Description |
| --- | --- |
| [files](/API/Users_Guide/Search_and_Retrieval/#files-endpoint) | Information about files stored in the GDC |
| [cases](/API/Users_Guide/Search_and_Retrieval/#cases-endpoint) | Information related to cases, or sample donors |
| [history](/API/Users_Guide/Search_and_Retrieval/#history-endpoint) | Information related to file version history |
| [projects](/API/Users_Guide/Search_and_Retrieval/#project-endpoint) | Information about projects |
| [annotations](/API/Users_Guide/Search_and_Retrieval/#annotations-endpoint) | Information about annotations to GDC data |
| \_mapping | Information about elements that can be used to query other endpoints |

The choice of endpoint determines what is listed in the search results. The `files` endpoint will generate a list of files, whereas the `cases` endpoint will generate a list of cases. Each of the above endpoints, other than `_mapping`, can query and return any of the related fields in the [GDC Data Model](../../Data/Data_Model/GDC_Data_Model.md). So the `cases` endpoint can be queried for file fields (e.g. to look for cases that have certain types of experimental data), and the `files` endpoint can be queried for clinical metadata associated with a case (e.g. to look for files from cases diagnosed with a specific cancer type).

### `Project` Endpoint
The `projects` endpoint provides access to project records, the highest level of data organization in the GDC.

#### Example
This example is a query for projects contained in the GDC. It uses the [from](#from), [size](#size), [sort](#sort), and [pretty](#pretty) parameters, and returns the first two projects sorted by project id.

=== "Shell"

    ```shell
    curl 'https://api.gdc.cancer.gov/projects?from=0&size=2&sort=project_id:asc&pretty=true'
    ```

=== "Output"

    ```json
    {
      "data": {
        "hits": [
          {
            "id": "APOLLO-LUAD",
            "primary_site": [
              "Bronchus and lung"
            ],
            "dbgap_accession_number": "phs003011",
            "project_id": "APOLLO-LUAD",
            "disease_type": [
              "Adenomas and Adenocarcinomas"
            ],
            "name": "APOLLO1: Proteogenomic characterization of lung adenocarcinoma",
            "releasable": false,
            "state": "open",
            "released": true
          },
          {
            "id": "BEATAML1.0-COHORT",
            "primary_site": [
              "Hematopoietic and reticuloendothelial systems"
            ],
            "dbgap_accession_number": "phs001657",
            "project_id": "BEATAML1.0-COHORT",
            "disease_type": [
              "Myelodysplastic Syndromes",
              "Leukemias, NOS",
              "Unknown",
              "Myeloid Leukemias",
              "Plasma Cell Tumors",
              "Chronic Myeloproliferative Disorders"
            ],
            "name": "Functional Genomic Landscape of Acute Myeloid Leukemia",
            "releasable": true,
            "state": "open",
            "released": true
          }
        ],
        "pagination": {
          "count": 2,
          "total": 78,
          "size": 2,
          "from": 0,
          "sort": "None",
          "page": 1,
          "pages": 39
        }
      },
      "warnings": {}
    }
    ```

#### Retrieval of project metadata using project_id

The `project` endpoint supports a simple query format that retrieves the metadata of a single project using its `project_id`:

=== "Shell"

    ```shell
    curl 'https://api.gdc.cancer.gov/projects/TARGET-NBL?expand=summary,summary.experimental_strategies,summary.data_categories&pretty=true'
    ```

=== "Response"    

    ```json
    {
      "data": {
        "summary": {
          "file_count": 5705,
          "data_categories": [
            {
              "file_count": 943,
              "case_count": 278,
              "data_category": "Sequencing Reads"
            },
            {
              "file_count": 3080,
              "case_count": 220,
              "data_category": "Simple Nucleotide Variation"
            },
            {
              "file_count": 3,
              "case_count": 1119,
              "data_category": "Clinical"
            },
            {
              "file_count": 705,
              "case_count": 225,
              "data_category": "DNA Methylation"
            },
            {
              "file_count": 2,
              "case_count": 1132,
              "data_category": "Biospecimen"
            },
            {
              "file_count": 324,
              "case_count": 155,
              "data_category": "Transcriptome Profiling"
            },
            {
              "file_count": 648,
              "case_count": 155,
              "data_category": "Structural Variation"
            }
          ],
          "experimental_strategies": [
            {
              "file_count": 1458,
              "case_count": 155,
              "experimental_strategy": "RNA-Seq"
            },
            {
              "file_count": 15,
              "case_count": 8,
              "experimental_strategy": "WGS"
            },
            {
              "file_count": 3522,
              "case_count": 222,
              "experimental_strategy": "WXS"
            },
            {
              "file_count": 705,
              "case_count": 225,
              "experimental_strategy": "Methylation Array"
            }
          ],
          "case_count": 1132,
          "file_size": 16968781125824
        },
        "primary_site": [
          "Stomach",
          "Bones, joints and articular cartilage of limbs",
          "Heart, mediastinum, and pleura",
          "Peripheral nerves and autonomic nervous system",
          "Uterus, NOS",
          "Bones, joints and articular cartilage of other and unspecified sites",
          "Other endocrine glands and related structures",
          "Renal pelvis",
          "Retroperitoneum and peritoneum",
          "Liver and intrahepatic bile ducts",
          "Meninges",
          "Connective, subcutaneous and other soft tissues",
          "Adrenal gland",
          "Unknown",
          "Spinal cord, cranial nerves, and other parts of central nervous system",
          "Skin",
          "Other and ill-defined sites",
          "Kidney",
          "Lymph nodes",
          "Hematopoietic and reticuloendothelial systems"
        ],
        "dbgap_accession_number": "phs000467",
        "project_id": "TARGET-NBL",
        "disease_type": [
          "Neuroepitheliomatous Neoplasms",
          "Not Applicable"
        ],
        "name": "Neuroblastoma",
        "releasable": true,
        "state": "open",
        "released": true
      },
      "warnings": {}
    }

    ```

### `Files` Endpoint

The GDC Files Endpoint `https://api.gdc.cancer.gov/files` enables search and retrieval of information relating to files stored in the GDC, including file properties such as `file_name`, `md5sum`, `data_format`, and others.

#### Example

This example is a query for files contained in the GDC. It uses the [from](#from), [size](#size), [sort](#sort), and [pretty](#pretty) parameters, and returns only the first two files, sorted by file size, from smallest to largest.

=== "Shell"

    ```shell
    curl 'https://api.gdc.cancer.gov/files?from=0&size=2&sort=file_size:asc&pretty=true'
    ```
=== "Output"

    ```json 
    {
      "data": {
        "hits": [
          {
            "id": "0ab5e358-b1ff-4433-8959-c37c5890d9aa",
            "data_format": "BEDPE",
            "access": "controlled",
            "file_name": "090e2828-079c-48e6-97cb-735c763da8d3.wgs.BRASS.rerun_structural_variation.bedpe.gz",
            "submitter_id": "247c3c9a-58b9-4b70-bda8-cb197acb5609",
            "data_category": "Somatic Structural Variation",
            "acl": [
              "phs001287"
            ],
            "type": "structural_variation",
            "file_size": 20,
            "created_datetime": "2022-04-08T20:27:04.633842-05:00",
            "updated_datetime": "2022-07-07T11:02:27.204310-05:00",
            "file_id": "0ab5e358-b1ff-4433-8959-c37c5890d9aa",
            "data_type": "Structural Rearrangement",
            "state": "released",
            "experimental_strategy": "WGS",
            "version": "1",
            "data_release": "34.0 - 37.0"
          },
          {
            "id": "a8bc2405-b57d-48bb-b241-18b3e28caa56",
            "data_format": "BEDPE",
            "access": "controlled",
            "file_name": "eae76f14-8aa7-427f-a90c-4e0ed095e0c2.wgs.BRASS.rerun_structural_variation.bedpe.gz",
            "submitter_id": "618cd251-ddcb-4a7e-9a6d-efb132b0bd7a",
            "data_category": "Somatic Structural Variation",
            "acl": [
              "phs001287"
            ],
            "type": "structural_variation",
            "file_size": 20,
            "created_datetime": "2022-04-08T20:43:16.505747-05:00",
            "updated_datetime": "2022-07-07T11:00:43.345766-05:00",
            "file_id": "a8bc2405-b57d-48bb-b241-18b3e28caa56",
            "data_type": "Structural Rearrangement",
            "state": "released",
            "experimental_strategy": "WGS",
            "version": "1",
            "data_release": "34.0 - 37.0"
          }
        ],
        "pagination": {
          "count": 2,
          "total": 931947,
          "size": 2,
          "from": 0,
          "sort": "None",
          "page": 1,
          "pages": 465974
        }
      },
      "warnings": {}
    }
    
    ```

#### Retrieval of file metadata using individual UUIDs:

The `/files` endpoint supports a simple query format that retrieves the metadata of a single file using its UUID.  Note that the `/files` endpoint is inactive when querying for earlier file versions.  In that case, the `/history` or `/files/versions` endpoints should be used instead.

=== "Shell"

    ```Shell
    curl 'https://api.gdc.cancer.gov/files/20f45e04-3c10-4f11-b57b-719880eab69e?pretty=true'
    ```

=== "Output"

    ```json
    {
      "data": {
        "data_format": "VCF",
        "access": "controlled",
        "file_name": "TCGA_BRCA.8d9cb5ae-e568-41fc-8b53-14467c2623dc.wxs.MuTect2.somatic_annotation.vcf.gz",
        "submitter_id": "675f31dd-70e5-4a72-8139-423b14b31564",
        "data_category": "Simple Nucleotide Variation",
        "acl": [
          "phs000178"
        ],
        "type": "annotated_somatic_mutation",
        "file_size": 6894331,
        "created_datetime": "2022-02-07T08:48:39.178606-06:00",
        "updated_datetime": "2022-02-09T12:11:12.781445-06:00",
        "file_id": "20f45e04-3c10-4f11-b57b-719880eab69e",
        "data_type": "Annotated Somatic Mutation",
        "state": "released",
        "experimental_strategy": "WXS",
        "version": "2",
        "data_release": "32.0 - 37.0"
      },
      "warnings": {}
    }
    ```

>__Note:__ The `file_size` field associated with each file is reported in bytes.  


#### Example of retrieving file version information:

The `https://api.gdc.cancer.gov/files/versions` endpoint enables search and retrieval of version information about a file.  A file may be versioned if a file is updated by the GDC (e.g. using a new alignment algorithm or fixing a file that contained an error). `Version` refers to the instance of a particular file.  Inputs can either be a list of UUIDs as shown in example 1 or a download manifest as shown in example 2.  Output includes information about the current and latest version for any given file.  While `/files` also returns information about a file version this endpoint will only work for the most recent version of a file whereas `/files/versions` will work for all previous and current versions of a file.  In both examples below the output format can be modified by adding the `format=tsv` parameter.

=== "Shell1"

    ```shell
    curl 'https://api.gdc.cancer.gov/files/versions/1dd28069-5777-4ff9-bd2b-d1ba68e88b06,2a03abac-f1a2-49a9-a57c-7543739dd862?pretty=true'
    ```

=== "Output1"

    ```json 
    [
      {
        "id": "1dd28069-5777-4ff9-bd2b-d1ba68e88b06",
        "filename": "1dd28069-5777-4ff9-bd2b-d1ba68e88b06.vcf.gz",
        "version": "1",
        "md5": "c2f9b196e154906a70c7ec46492a859d",
        "size": 332092,
        "state": "validated",
        "release": "12.0",
        "latest_id": "76b3f4d8-c6b7-4662-ac42-1d27d4684281",
        "latest_filename": "def1cc5b-55f0-4372-a3ff-df3ea93cf3e7.wxs.somaticsniper.raw_somatic_mutation.vcf.gz",
        "latest_version": "2",
        "latest_size": 357706,
        "latest_state": "validated",
        "latest_release": [
          "32.0",
          "33.0",
          "33.1",
          "34.0",
          "35.0",
          "36.0",
          "37.0"
        ]
      },
      {
        "id": "2a03abac-f1a2-49a9-a57c-7543739dd862",
        "filename": "a5d86cde-32ca-4ed6-b1a5-5a47575f2ac6_gdc_realn_rehead.bam",
        "version": "1",
        "md5": "48686fcd84ac713d44261ca9e26b89fb",
        "size": 6653119038,
        "state": "validated",
        "release": "12.0",
        "latest_id": "de0ce84d-c286-405c-a556-39dac14c7c74",
        "latest_filename": "d45c33cc-88e2-4de5-a578-f7e31a6c0738.rna_seq.genomic.gdc_realn.bam",
        "latest_version": "2",
        "latest_size": 6223445806,
        "latest_state": "validated",
        "latest_release": [
          "32.0",
          "33.0",
          "33.1",
          "34.0",
          "35.0",
          "36.0",
          "37.0"
        ]
      }
    ]
    ```

=== "Shell2"

    ```shell
    curl --request POST --header "Content-Type: text/tsv"  https://api.gdc.cancer.gov/files/versions/manifest?pretty=true --data-binary @gdc_manifest_20180809_154816.txt
    ```
    
=== "Output2"

    ```json
    [
      {
        "id": "0b20e27c-9a09-4f15-923f-d5b4f185dc22",
        "filename": "nationwidechildrens.org_clinical.TCGA-13-1500.xml",
        "version": "1",
        "md5": "597aa4df24c4d544b6c25cbd8b25a33e",
        "size": 44857,
        "state": "validated",
        "release": "12.0",
        "latest_id": "0b20e27c-9a09-4f15-923f-d5b4f185dc22",
        "latest_filename": "nationwidechildrens.org_clinical.TCGA-13-1500.xml",
        "latest_version": "1",
        "latest_md5": "597aa4df24c4d544b6c25cbd8b25a33e",
        "latest_size": 44857,
        "latest_state": "validated",
        "latest_release": [
          "12.0",
          "13.0",
          "14.0",
          "15.0",
          "16.0",
          "17.0",
          "18.0",
          "19.0",
          "20.0",
          "21.0",
          "22.0",
          "23.0",
          "24.0",
          "25.0",
          "26.0",
          "27.0",
          "28.0",
          "29.0",
          "30.0",
          "31.0",
          "32.0",
          "33.0",
          "33.1",
          "34.0",
          "35.0",
          "36.0",
          "37.0"
        ]
      },
      {
        "id": "3edc7084-013c-4493-8507-c00b0e9962d8",
        "filename": "BUCKS_p_TCGA_272_273_N_GenomeWideSNP_6_G05_1320676.grch38.seg.v2.txt",
        "version": "1",
        "md5": "35a18d990a05eedfaf96e753bee0b96d",
        "size": 27620,
        "state": "validated",
        "release": "12.0",
        "latest_id": "3edc7084-013c-4493-8507-c00b0e9962d8",
        "latest_filename": "BUCKS_p_TCGA_272_273_N_GenomeWideSNP_6_G05_1320676.grch38.seg.v2.txt",
        "latest_version": "1",
        "latest_md5": "35a18d990a05eedfaf96e753bee0b96d",
        "latest_size": 27620,
        "latest_state": "validated",
        "latest_release": [
          "12.0",
          "13.0",
          "14.0",
          "15.0",
          "16.0",
          "17.0",
          "18.0",
          "19.0",
          "20.0",
          "21.0",
          "22.0",
          "23.0",
          "24.0",
          "25.0",
          "26.0",
          "27.0",
          "28.0",
          "29.0",
          "30.0",
          "31.0",
          "32.0",
          "33.0",
          "33.1",
          "34.0",
          "35.0",
          "36.0",
          "37.0"
        ]
      },
      {
        "id": "a22f5e32-b16e-458f-a412-7e438056ece6",
        "filename": "a22f5e32-b16e-458f-a412-7e438056ece6.vep.vcf.gz",
        "version": "1",
        "md5": "68b2433b31679bbbc6681919a1b81762",
        "size": 2346,
        "state": "validated",
        "release": "12.0",
        "latest_id": "55491171-6170-45cb-af9d-d99345b289e5",
        "latest_filename": "4b89bb97-41f6-43c4-a481-287556f7bb4a.targeted_sequencing.annotated_somatic_mutation.vcf.gz",
        "latest_version": "2",
        "latest_size": 2618,
        "latest_state": "validated",
        "latest_release": [
          "32.0",
          "33.0",
          "33.1",
          "34.0",
          "35.0",
          "36.0",
          "37.0"
        ]
      }
    ]
    ```

### `Cases` Endpoint

The GDC Cases Endpoint `https://api.gdc.cancer.gov/cases` enables search and retrieval of information related to a specific case.

The `cases` endpoint is designed to retrieve the metadata associated with one or more cases, including all nested biospecimen entities. Filters can be applied to retrieve information for entire cases, but not for lower-level biospecimen entities. For example, a sample within a case cannot be used to query for aliquots that are associated only with that sample. All aliquots associated with the case would be retrieved.


#### Example

This example is a query for files contained in GDC. It returns case where submitter id is `TCGA-BH-A0EA`, using the [pretty](#pretty) and [filters](#filters) parameters and the following [filtering operators](#filtering-operators):

	{"op":"and","content":[{"op":"in","content":{"field":"submitter_id","value":["TCGA-BH-A0EA"]}}]}

Command:

=== "Shell"

    ```shell
    curl 'https://api.gdc.cancer.gov/cases?filters=%7B%22op%22%3A%22and%22%2C%22content%22%3A%5B%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22submitter_id%22%2C%22value%22%3A%5B%22TCGA-BH-A0EA%22%5D%7D%7D%5D%7D%0A%0A&pretty=true'
    ```

=== "Output"

    ```json 
    {
      "data": {
        "hits": [
          {
            "id": "1f601832-eee3-48fb-acf5-80c4a454f26e",
            "slide_ids": [
              "a0826f0d-986a-491b-8c6f-b34f8929f3ee",
              "90154ea1-6b76-4445-870e-d531d6fa1239",
              "1dd1cab5-5a81-428a-8153-91e8c4cf9905"
            ],
            "submitter_slide_ids": [
              "TCGA-BH-A0EA-01Z-00-DX1",
              "TCGA-BH-A0EA-01A-01-MSA",
              "TCGA-BH-A0EA-01A-01-TSA"
            ],
            "disease_type": "Ductal and Lobular Neoplasms",
            "analyte_ids": [
              "f19f408a-815f-43d9-8032-e9482b796371",
              "fe678556-acf4-4bde-a95e-860bb0150a95",
              "69ddc092-88a0-4839-a2bb-9f1c9e760409",
              "66ed0f86-5ca5-4dec-ba76-7ee4dcf31831",
              "30cb470f-66d4-4085-8c30-83a42e8453d4"
            ],
            "submitter_id": "TCGA-BH-A0EA",
            "submitter_analyte_ids": [
              "TCGA-BH-A0EA-10A-01D",
              "TCGA-BH-A0EA-01A-11D",
              "TCGA-BH-A0EA-01A-11R",
              "TCGA-BH-A0EA-10A-01W",
              "TCGA-BH-A0EA-01A-11W"
            ],
            "aliquot_ids": [
              "cde982b7-3b0a-49eb-8710-a599cb0e44c1",
              "b1a3739d-d554-4202-b96f-f25a444e2042",
              "eef9dce1-6ba6-432b-bbe2-53c7dbe64fe7",
              "97c64d6a-7dce-4d0f-9cb3-b3e4eb4719c5",
              "262715e1-835c-4f16-8ee7-6900e26f7cf5",
              "cfbd5476-e83a-401d-9f9a-639c73a0e35b",
              "bcb7fc6d-60a0-48b7-aa81-14c0dda72d76",
              "561b8777-801a-49ed-a306-e7dafeb044b6",
              "edad5bd3-efe0-4c5f-b05c-2c0c2951c45a",
              "42d050e4-e8ee-4442-b9c0-0ee14706b138",
              "2beb34c4-d493-4a73-b21e-de77d43251ff",
              "ca71ca96-cbb7-4eab-9487-251dda34e107"
            ],
            "submitter_aliquot_ids": [
              "TCGA-BH-A0EA-10A-01W-A12U-09",
              "TCGA-BH-A0EA-01A-11D-A111-01",
              "TCGA-BH-A0EA-01A-11D-A314-09",
              "TCGA-BH-A0EA-01A-11D-A10X-02",
              "TCGA-BH-A0EA-10A-01D-A10Z-02",
              "TCGA-BH-A0EA-10A-01D-A110-09",
              "TCGA-BH-A0EA-01A-11D-A10Y-09",
              "TCGA-BH-A0EA-10A-01D-A113-01",
              "TCGA-BH-A0EA-01A-11D-A112-05",
              "TCGA-BH-A0EA-01A-11R-A115-07",
              "TCGA-BH-A0EA-01A-11W-A12T-09",
              "TCGA-BH-A0EA-01A-11R-A114-13"
            ],
            "created_datetime": null,
            "diagnosis_ids": [
              "84654ad5-2a2c-5c3b-8340-ecac6a5550fe"
            ],
            "sample_ids": [
              "55864d86-dab8-47bb-a3e3-8cfb198b06c1",
              "9a6c71a6-82cd-42b1-a93f-f569370848d6",
              "7f791228-dd77-4ab0-8227-d784a4c7fea1"
            ],
            "submitter_sample_ids": [
              "TCGA-BH-A0EA-01A",
              "TCGA-BH-A0EA-01Z",
              "TCGA-BH-A0EA-10A"
            ],
            "primary_site": "Breast",
            "submitter_diagnosis_ids": [
              "TCGA-BH-A0EA_diagnosis"
            ],
            "updated_datetime": "2019-08-06T14:15:54.128069-05:00",
            "case_id": "1f601832-eee3-48fb-acf5-80c4a454f26e",
            "state": "released",
            "portion_ids": [
              "cb6086d1-3416-4310-b109-e8fa6e8b72d4",
              "8629bf5a-cdaf-4f6a-90bb-27dd4a7565c5",
              "ae4f5816-f97a-4605-9b05-9ab820467dee"
            ],
            "submitter_portion_ids": [
              "TCGA-BH-A0EA-10A-01",
              "TCGA-BH-A0EA-01A-21-A13C-20",
              "TCGA-BH-A0EA-01A-11"
            ]
          }
        ],
        "pagination": {
          "count": 1,
          "total": 1,
          "size": 10,
          "from": 0,
          "sort": "",
          "page": 1,
          "pages": 1
        }
      },
      "warnings": {}
    }
    ```

#### Retrieval of case metadata using individual UUIDs:

The `cases` endpoint supports a simple query format that retrieves the metadata of a single case using its UUID:

=== "Shell"

    ```shell
    curl 'https://api.gdc.cancer.gov/cases/1f601832-eee3-48fb-acf5-80c4a454f26e?pretty=true&expand=diagnoses'
    ```

=== "Response"

    ```json
    
      "data": {
        "slide_ids": [
          "90154ea1-6b76-4445-870e-d531d6fa1239",
          "1dd1cab5-5a81-428a-8153-91e8c4cf9905",
          "a0826f0d-986a-491b-8c6f-b34f8929f3ee"
        ],
        "submitter_slide_ids": [
          "TCGA-BH-A0EA-01A-01-MSA",
          "TCGA-BH-A0EA-01A-01-TSA",
          "TCGA-BH-A0EA-01Z-00-DX1"
        ],
        "disease_type": "Ductal and Lobular Neoplasms",
        "analyte_ids": [
          "fe678556-acf4-4bde-a95e-860bb0150a95",
          "66ed0f86-5ca5-4dec-ba76-7ee4dcf31831",
          "30cb470f-66d4-4085-8c30-83a42e8453d4",
          "69ddc092-88a0-4839-a2bb-9f1c9e760409",
          "f19f408a-815f-43d9-8032-e9482b796371"
        ],
        "submitter_id": "TCGA-BH-A0EA",
        "submitter_analyte_ids": [
          "TCGA-BH-A0EA-01A-11D",
          "TCGA-BH-A0EA-01A-11R",
          "TCGA-BH-A0EA-10A-01W",
          "TCGA-BH-A0EA-01A-11W",
          "TCGA-BH-A0EA-10A-01D"
        ],
        "aliquot_ids": [
          "eef9dce1-6ba6-432b-bbe2-53c7dbe64fe7",
          "2beb34c4-d493-4a73-b21e-de77d43251ff",
          "b1a3739d-d554-4202-b96f-f25a444e2042",
          "262715e1-835c-4f16-8ee7-6900e26f7cf5",
          "cfbd5476-e83a-401d-9f9a-639c73a0e35b",
          "edad5bd3-efe0-4c5f-b05c-2c0c2951c45a",
          "bcb7fc6d-60a0-48b7-aa81-14c0dda72d76",
          "42d050e4-e8ee-4442-b9c0-0ee14706b138",
          "97c64d6a-7dce-4d0f-9cb3-b3e4eb4719c5",
          "561b8777-801a-49ed-a306-e7dafeb044b6",
          "ca71ca96-cbb7-4eab-9487-251dda34e107",
          "cde982b7-3b0a-49eb-8710-a599cb0e44c1"
        ],
        "submitter_aliquot_ids": [
          "TCGA-BH-A0EA-01A-11R-A115-07",
          "TCGA-BH-A0EA-01A-11D-A112-05",
          "TCGA-BH-A0EA-10A-01W-A12U-09",
          "TCGA-BH-A0EA-01A-11D-A10X-02",
          "TCGA-BH-A0EA-10A-01D-A113-01",
          "TCGA-BH-A0EA-10A-01D-A110-09",
          "TCGA-BH-A0EA-01A-11D-A314-09",
          "TCGA-BH-A0EA-01A-11D-A10Y-09",
          "TCGA-BH-A0EA-01A-11D-A111-01",
          "TCGA-BH-A0EA-10A-01D-A10Z-02",
          "TCGA-BH-A0EA-01A-11R-A114-13",
          "TCGA-BH-A0EA-01A-11W-A12T-09"
        ],
        "diagnoses": [
          {
            "synchronous_malignancy": "Not Reported",
            "ajcc_pathologic_stage": "Stage IIA",
            "days_to_diagnosis": 0,
            "created_datetime": null,
            "last_known_disease_status": "not reported",
            "tissue_or_organ_of_origin": "Breast, NOS",
            "days_to_last_follow_up": null,
            "age_at_diagnosis": 26548,
            "primary_diagnosis": "Infiltrating duct carcinoma, NOS",
            "updated_datetime": "2019-08-08T16:25:42.215495-05:00",
            "prior_malignancy": "yes",
            "year_of_diagnosis": 2008,
            "state": "released",
            "prior_treatment": "No",
            "days_to_last_known_disease_status": null,
            "ajcc_staging_system_edition": "6th",
            "ajcc_pathologic_t": "T1c",
            "days_to_recurrence": null,
            "morphology": "8500/3",
            "ajcc_pathologic_n": "N1a",
            "ajcc_pathologic_m": "M0",
            "submitter_id": "TCGA-BH-A0EA_diagnosis",
            "classification_of_tumor": "not reported",
            "diagnosis_id": "84654ad5-2a2c-5c3b-8340-ecac6a5550fe",
            "icd_10_code": "C50.9",
            "site_of_resection_or_biopsy": "Breast, NOS",
            "tumor_grade": "Not Reported",
            "progression_or_recurrence": "not reported"
          }
        ],
        "created_datetime": null,
        "diagnosis_ids": [
          "84654ad5-2a2c-5c3b-8340-ecac6a5550fe"
        ],
        "sample_ids": [
          "55864d86-dab8-47bb-a3e3-8cfb198b06c1",
          "7f791228-dd77-4ab0-8227-d784a4c7fea1",
          "9a6c71a6-82cd-42b1-a93f-f569370848d6"
        ],
        "submitter_sample_ids": [
          "TCGA-BH-A0EA-01A",
          "TCGA-BH-A0EA-01Z",
          "TCGA-BH-A0EA-10A"
        ],
        "primary_site": "Breast",
        "submitter_diagnosis_ids": [
          "TCGA-BH-A0EA_diagnosis"
        ],
        "updated_datetime": "2019-08-06T14:15:54.128069-05:00",
        "case_id": "1f601832-eee3-48fb-acf5-80c4a454f26e",
        "state": "released",
        "portion_ids": [
          "1ef8b20e-43e5-49d7-ac9a-03ce14f58daa",
          "cb6086d1-3416-4310-b109-e8fa6e8b72d4",
          "8629bf5a-cdaf-4f6a-90bb-27dd4a7565c5",
          "ae4f5816-f97a-4605-9b05-9ab820467dee"
        ],
        "submitter_portion_ids": [
          "TCGA-BH-A0EA-01A-21-A13C-20",
          "TCGA-BH-A0EA-10A-01",
          "TCGA-BH-A0EA-01A-21",
          "TCGA-BH-A0EA-01A-11"
        ]
      },
      "warnings": {}
    }
    ```

### `Annotations` Endpoint

The GDC Annotation Endpoint `https://api.gdc.cancer.gov/annotations` enables search and retrieval of annotations stored in the GDC.


#### Example

This example is a query for any annotations **directly** associated with the following GDC entities:

* the case with UUID e0d36cc0-652c-4224-bb10-09d15c7bd8f1
* the sample with UUID 25ebc29a-7598-4ae4-ba7f-618d448882cc
* the aliquot with UUID fe660d7c-2746-4b50-ab93-b2ed99960553

The query uses the [filters](#filters) parameter to specify entity UUIDs. Code samples below include the bare and percent-encoded filter JSON.

=== "Filter-json"

    ```json
    {
       "op":"in",
       "content":{
          "field":"entity_id",
          "value":[
             "e0d36cc0-652c-4224-bb10-09d15c7bd8f1",
             "25ebc29a-7598-4ae4-ba7f-618d448882cc",
             "fe660d7c-2746-4b50-ab93-b2ed99960553"
          ]
       }
    }
    ```

=== "Filter-JSON-percent-encoded"

    ```
    %7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22entity_id%22%2C%22value%22%3A%5B%22e0d36cc0-652c-4224-bb10-09d15c7bd8f1%22%2C%2225ebc29a-7598-4ae4-ba7f-618d448882cc%22%2C%22fe660d7c-2746-4b50-ab93-b2ed99960553%22%5D%7D%7D
    ```

=== "Shell"

    ```shell
    curl 'https://api.gdc.cancer.gov/annotations?filters=%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22entity_id%22%2C%22value%22%3A%5B%22e0d36cc0-652c-4224-bb10-09d15c7bd8f1%22%2C%2225ebc29a-7598-4ae4-ba7f-618d448882cc%22%2C%22fe660d7c-2746-4b50-ab93-b2ed99960553%22%5D%7D%7D&pretty=true'
    ```

=== "Output"

    ```json 
    {
      "data": {
        "hits": [
          {
            "category": "Item flagged DNU",
            "status": "Approved",
            "entity_id": "fe660d7c-2746-4b50-ab93-b2ed99960553",
            "classification": "CenterNotification",
            "entity_type": "aliquot",
            "created_datetime": "2015-09-28T00:00:00",
            "annotation_id": "5ddadefe-8b57-5ce2-b8b2-918d63d99a59",
            "notes": "The aliquot failed Broad pipeline QC and not all files are suitable for use. Consult the SDRF file to determine which files are usable.",
            "updated_datetime": "2017-03-09T13:20:38.962182-06:00",
            "submitter_id": "29087",
            "state": "submitted",
            "case_id": "41b59716-116f-4942-8b63-409870a87e26",
            "case_submitter_id": "TCGA-DK-A3IM",
            "entity_submitter_id": "TCGA-DK-A3IM-10A-01D-A20B-01",
            "id": "5ddadefe-8b57-5ce2-b8b2-918d63d99a59"
          },
          {
            "category": "Item is noncanonical",
            "status": "Approved",
            "entity_id": "25ebc29a-7598-4ae4-ba7f-618d448882cc",
            "classification": "Notification",
            "entity_type": "sample",
            "created_datetime": "2012-07-12T00:00:00",
            "annotation_id": "d6500f94-618f-5334-a810-ade76b887ec9",
            "notes": "No Matching Normal",
            "updated_datetime": "2017-03-09T13:47:18.182075-06:00",
            "submitter_id": "8009",
            "state": "submitted",
            "case_id": "bd114e05-5a97-41e2-a0d5-5d39a1e9d461",
            "case_submitter_id": "TCGA-08-0514",
            "entity_submitter_id": "TCGA-08-0514-01A",
            "id": "d6500f94-618f-5334-a810-ade76b887ec9"
          },
          {
            "category": "Prior malignancy",
            "status": "Approved",
            "entity_id": "e0d36cc0-652c-4224-bb10-09d15c7bd8f1",
            "classification": "Notification",
            "entity_type": "case",
            "created_datetime": "2013-03-12T00:00:00",
            "annotation_id": "33336cdf-2cf0-5af2-bb52-fecd3427f180",
            "notes": "Patient had a prior lymphoma. Unknown radiation or systemic chemotherapy.",
            "updated_datetime": "2017-03-09T12:11:31.786013-06:00",
            "submitter_id": "15630",
            "state": "submitted",
            "case_id": "e0d36cc0-652c-4224-bb10-09d15c7bd8f1",
            "case_submitter_id": "TCGA-FS-A1ZF",
            "entity_submitter_id": "TCGA-FS-A1ZF",
            "id": "33336cdf-2cf0-5af2-bb52-fecd3427f180"
          }
        ],
        "pagination": {
          "count": 3,
          "sort": "",
          "from": 0,
          "page": 1,
          "total": 3,
          "pages": 1,
          "size": 10
        }
      },
      "warnings": {}
    }
    ```

### `History` Endpoint

The GDC History Endpoint `https://api.gdc.cancer.gov/history` enables search and retrieval of version and release information about a file.  This endpoint will return the entire provenance of all versions of a file.  A file may be versioned if a file is updated by the GDC (e.g. using a new alignment algorithm or fixing a file that contained an error). `Version` refers to the instance of a particular file. `Release` refers to which data release a file was part of.  A file may be a part of many different data releases with no change in version number or content.  

#### Example

This example is a query for versioning information associated with the follow with file `1dd28069-5777-4ff9-bd2b-d1ba68e88b06`.

=== "Shell"

    ```shell
    curl 'https://api.gdc.cancer.gov/history/1dd28069-5777-4ff9-bd2b-d1ba68e88b06'
    ```
=== "Output"

    ```json 
    [{"uuid": "1dd28069-5777-4ff9-bd2b-d1ba68e88b06", "version": "1", "file_change": "superseded", "release_date": "2018-08-23", "data_release": "12.0"}, {"uuid": "76b3f4d8-c6b7-4662-ac42-1d27d4684281", "version": "2", "file_change": "released", "release_date": "2022-03-29", "data_release": "32.0"}]

    ```


### `_mapping` Endpoint

Each search and retrieval endpoint is equipped with a ```_mapping``` endpoint that provides information about available fields. For example, `files/_mapping` endpoint provides information about fields and field groups available at the `files` endpoint: `https://api.gdc.cancer.gov/files/_mapping`.

The high-level structure of a response to a `_mapping` query is as follows:

	"_mapping": {}
	, "defaults": []
	, "expand": []
	, "fields": []
	, "multi": []
	, "nested": []

[//]: # (_)

Each part of the response is described below:

| Part | Description |
|------|-------------|
| `_mapping` | All available fields and their descriptions. The endpoint-agnostic field names provided here are compatible with the `filters` parameter but are not always compatible with the `fields` parameter |
| `defaults` | The default set of fields included in the API response when the `fields` parameter is not used in the request |
| `expand` | Field group names for use with the `expand` parameter |
| `fields` | All available fields in an endpoint-specific format that is compatible with both the `filters` and `fields` parameters |
| `multi` | GDC internal use |
| `nested` | Nested fields |


#### Example

=== "Shell"

    ```shell
    curl 'https://api.gdc.cancer.gov/projects/_mapping'
    ```

=== "Output"

    ```json
    This output was put thought a json format application for easier viewability.
    {
        ...
    
          "_mapping": {
            "projects.disease_type": {
              "doc_type": "projects",
              "field": "disease_type",
              "type": "id"
            },
            "projects.name": {
              "doc_type": "projects",
              "field": "name",
              "type": "id"
            }
          }
    
        ...
    
    }
    ```

Similar information can be obtained using the `fields` parameter; `fields` queries provide additional information in the response, such as the name of the Elastic Search document (`doc_type`), the field name and the type of value. A list of supported types (such as `string`, `long`, `float`, ...) can be obtained from [Elastic Search Documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-types.html).


## Request Parameters

The GDC API supports the following search & retrieval request parameters:

Parameter | Default | Description
--------- | ------- | -----------
filters| null | Specifies search parameters
format | JSON | Specifies the API response format: JSON, XML, or TSV
pretty | false | Returns response with indentations and line breaks in a human-readable format
fields | null | Specifies which fields to include in the response
expand | null | Returns multiple related fields
size | 10 | Specifies the number of results to return
from   | 0 | Specifies the first record to return from a set of search results
sort | null | Specifies sorting for the search results
facets | null | Provides all existing values for a given field and the number of records having this value.


### Filters: Specifying the Query

The `filters` parameter enables passing of complex search queries to the GDC API. The parameter carries a query in the form of a JSON object.

#### Query Format

A `filters` query consists of an operator (or a nested set of operators) with a set of `field` and `value` operands.

The following `filters` query operators are supported by the GDC API:

| Operator | Description                                      | Number of Operands | Logic example                                                |
|----------|--------------------------------------------------|--------------------|--------------------------------------------------------------|
| =        | equals (string or number)                        | one                | gender = "female"                                            |
| !=       | does not equal (string or number)                | one                | project_id != "TARGET-AML"                                   |
| <        | less than (number)                               | one                | age at diagnosis < 90y                                       |
| <=       | less than or equal (number)                      | one                | age at diagnosis <= 17                                       |
| >        | greater than (number)                            | one                | age at diagnosis > 50                                        |
| >=       | greater than or equal (number)                   | one                | age at diagnosis >= 18                                       |
| is       | is (missing)                                     | one                | gender is missing                                            |
| not      | not (missing)                                    | one                | race not missing                                             |
| in       | matches a string or number in (a list)           | multiple           | primary_site in [Brain, Lung]                                |
| exclude  | does not match any strings or values in (a list) | multiple           | experimental_strategy exclude [WXS, WGS, "Genotyping array"] |
| and      | (operation1) and (operation2)                    | multiple           | {primary_site in [Brain, Lung]} and {gender = "female"}      |
| or       | (operation1) or (operation2)                     | multiple           | {project_id != "TARGET-AML"} or {age at diagnosis < 90y}     |

The `field` operand specifies a field that corresponds to a property defined in the [GDC Data Dictionary](../../Data_Dictionary/viewer.md). A list of supported fields is provided in [Appendix A](Appendix_A_Available_Fields.md); the list can also be accessed programmatically at the [_mapping endpoint](#95mapping-endpoint).

The `value` operand specifies the search terms. Users can get a list of available values for a specific property by making a call to the appropriate API endpoint using the `facets` parameter, e.g. `https://api.gdc.cancer.gov/v0/cases?facets=demographic.gender&size=0&pretty=true`. See [Facets](#facets) for details.

A simple query with a single operator looks like this:

	{
	    "op":"=",
	    "content":{
	        "field":"cases.demographic.gender",
	        "value":[
	            "male"
	        ]
	    }
	}

A more complex query with multiple operators looks like this:

	{
	    "op":"and",
	    "content":[
	        {
	            "op":"in",
	            "content":{
	                "field":"cases.submitter_id",
	                "value":[
	                    "TCGA-CK-4948",
	                    "TCGA-D1-A17N",
	                    "TCGA-4V-A9QX",
	                    "TCGA-4V-A9QM"
	                ]
	            }
	        },
	        {
	            "op":"=",
	            "content":{
	                "field":"files.data_type",
	                "value":"Gene Expression Quantification"
	            }
	        }
	    ]
	}


#### Example: HTTP GET Request

This example requests `male` cases using HTTP GET.

The JSON object to be passed to the GDC API looks like:

    {
        "op": "=",
        "content": {
            "field": "cases.demographic.gender",
            "value": [
                "male"
           ]
        }
    }

URL-encoding the above JSON object using [Percent-(URL)-encoding tool](https://www.freeformatter.com/url-encoder.html) results in the following string:

    %7B%0D%0A++++%22op%22%3A+%22%3D%22%2C%0D%0A++++%22content%22%3A+%7B%0D%0A++++++++%22field%22%3A+%22cases.demographic.gender%22%2C%0D%0A++++++++%22value%22%3A+%5B%0D%0A++++++++++++%22male%22%0D%0A++++++++%5D%0D%0A++++%7D%0D%0A%7D

The above string can now be passed to the GDC API using the `filters` parameter:

=== "Shell"

    ```shell
     curl  'https://api.gdc.cancer.gov/cases?filters=%7b%22op%22%3a+%22%3d%22%2c%0d%0a++++++%22content%22%3a+%7b%0d%0a++++++++++%22field%22%3a+%22cases.demographic.gender%22%2c%0d%0a++++++++++%22value%22%3a+%5b%22male%22%5d%0d%0a++++++%7d%0d%0a%7d&pretty=true'
    ```

=== "Python"

    ```python
    import requests
    import json
    cases_endpt = 'https://api.gdc.cancer.gov/cases'
    filt = {"op":"=",
            "content":{
                "field": "cases.demographic.gender",
                "value": ["male"]
            }
    }
    params = {'filters':json.dumps(filt), 'sort':'demographic.gender:asc'}
    # requests URL-encodes automatically
    response = requests.get(cases_endpt, params = params)
    print(json.dumps(response.json(), indent=2))
    ```

=== "Output"

    ```json 
    {
      "data": {
        "hits": [
          {
            "id": "03974dc9-0162-4de8-9897-09f88693681a",
            "lost_to_followup": null,
            "days_to_lost_to_followup": null,
            "disease_type": "Nevi and Melanomas",
            "analyte_ids": [
              "9747b614-624b-410a-8b94-854a16cd143a",
              "c8974764-4836-4a34-aeb8-52b491f78d0e",
              "bcea1ed5-b9cb-4a92-ad80-598d8a223fb3"
            ],
            "submitter_id": "HCM-BROD-0334-C43",
            "submitter_analyte_ids": [
              "HCM-BROD-0334-C43-10A-01D",
              "HCM-BROD-0334-C43-85M-01D",
              "HCM-BROD-0334-C43-85M-01R"
            ],
            "days_to_consent": null,
            "aliquot_ids": [
              "dcd74e48-12f3-4a86-a829-c7e055c215b7",
              "ea182abf-041d-474a-bc53-f6fdd05cd999",
              "33f3ba0a-c902-4288-9fa7-5696d959e51d"
            ],
            "submitter_aliquot_ids": [
              "HCM-BROD-0334-C43-85M-01R-A79O-41",
              "HCM-BROD-0334-C43-10A-01D-A79L-36",
              "HCM-BROD-0334-C43-85M-01D-A79L-36"
            ],
            "created_datetime": "2020-05-21T08:55:40.814734-05:00",
            "diagnosis_ids": [
              "3d666f1b-58c2-451f-8ebf-87b5caa02aaf",
              "fedc3533-85f7-4fc6-b996-a1f596e021df"
            ],
            "sample_ids": [
              "cd88baf4-b6eb-4df5-9b42-d55f3aad739c",
              "eb79f8b4-1cc3-4a32-ad51-cfea8cf150f0"
            ],
            "consent_type": null,
            "submitter_sample_ids": [
              "HCM-BROD-0334-C43-10A",
              "HCM-BROD-0334-C43-85M"
            ],
            "primary_site": "Skin",
            "submitter_diagnosis_ids": [
              "HCM-BROD-0334-C43_diagnosis2",
              "HCM-BROD-0334-C43_diagnosis"
            ],
            "updated_datetime": "2021-03-03T15:15:08.075155-06:00",
            "case_id": "03974dc9-0162-4de8-9897-09f88693681a",
            "index_date": "Diagnosis",
            "state": "released",
            "portion_ids": [
              "bd0bc175-5b54-47c1-96fc-c6d8afc0c115"
            ],
            "submitter_portion_ids": [
              "HCM-BROD-0334-C43-10A-01"
            ]
          },
          {
            "id": "03bfeb7c-cecf-4691-8263-33cdfe391ea9",
            "lost_to_followup": null,
            "days_to_lost_to_followup": null,
            "disease_type": "Adenomas and Adenocarcinomas",
            "analyte_ids": [
              "db8132c3-47a8-49ec-9b78-bc7d18debf67",
              "f74bf217-dae2-4554-92e4-8707068ea7a7",
              "01764f17-2a97-442e-a08b-8a21303b4770",
              "c9884c81-3c8f-4ad9-a962-42c7459a2276",
              "d4f1a9f8-f748-4f45-aa06-da4d760c4fab"
            ],
            "submitter_id": "HCM-BROD-0124-C25",
            "submitter_analyte_ids": [
              "HCM-BROD-0124-C25-85A-01D",
              "HCM-BROD-0124-C25-01A-01D",
              "HCM-BROD-0124-C25-10A-01D",
              "HCM-BROD-0124-C25-85A-01R",
              "HCM-BROD-0124-C25-01A-01R"
            ],
            "aliquot_ids": [
              "fe8e2565-749a-470a-b843-7afbe95ded81",
              "092656af-b279-46b8-9ccf-b1eabfbd1d6f",
              "677edb2c-fac3-4878-a28d-cf4e0d7873d7",
              "8bb9fff4-24f8-426e-9f2a-4cb30a4ac5c2",
              "f1c6f71d-b125-47bb-91ad-90d7cbff0012"
            ],
            "submitter_aliquot_ids": [
              "HCM-BROD-0124-C25-01A-01D-A78W-36",
              "HCM-BROD-0124-C25-85A-01D-A786-36",
              "HCM-BROD-0124-C25-01A-01R-A78X-41",
              "HCM-BROD-0124-C25-10A-01D-A78W-36",
              "HCM-BROD-0124-C25-85A-01R-A787-41"
            ],
            "created_datetime": "2019-04-16T11:21:56.471158-05:00",
            "diagnosis_ids": [
              "00184ed8-780a-4acf-b5f1-b1fcd6b08dcf"
            ],
            "sample_ids": [
              "e6bc6b9d-553f-4c78-bc11-ebcf7b0d4f27",
              "539593d1-bd9b-4379-8c86-16cdf607cd4e",
              "b0c0b5b0-cf6d-4281-8f4d-43dc77e88bc6"
            ],
            "submitter_sample_ids": [
              "HCM-BROD-0124-C25-85A",
              "HCM-BROD-0124-C25-01A",
              "HCM-BROD-0124-C25-10A"
            ],
            "primary_site": "Pancreas",
            "submitter_diagnosis_ids": [
              "HCM-BROD-0124-C25_diagnosis"
            ],
            "updated_datetime": "2021-07-12T12:25:55.528644-05:00",
            "case_id": "03bfeb7c-cecf-4691-8263-33cdfe391ea9",
            "index_date": "Diagnosis",
            "state": "released",
            "portion_ids": [
              "303512c0-382b-4442-a5a7-2699ca8b1384",
              "f07a4dae-2878-452e-9836-6f39c594d38d"
            ],
            "submitter_portion_ids": [
              "HCM-BROD-0124-C25-10A-01",
              "HCM-BROD-0124-C25-01A-01"
            ]
          },
          {
            "id": "05f41641-ee22-4d41-bb87-2bfa47cd983f",
            "lost_to_followup": null,
            "slide_ids": [
              "775be57f-6df8-40c3-9c4e-c06dd900a237",
              "7bd1dea3-7819-43e5-a9e4-5fe0a189cc87"
            ],
            "submitter_slide_ids": [
              "HCM-BROD-0095-C15-06A-01-S2-HE",
              "HCM-BROD-0095-C15-06A-01-S1-HE"
            ],
            "days_to_lost_to_followup": null,
            "disease_type": "Adenomas and Adenocarcinomas",
            "analyte_ids": [
              "3cd4c8b4-8c23-4c20-bffb-97e0f5e5ac0a",
              "76ed5a0c-8129-4a73-bbd2-08d9b36bee62",
              "d2acb3a5-7b51-4d86-8c2f-18b3f886a001",
              "e6f06e9e-bf13-44fb-990c-d64b1096cd7c",
              "331fc610-2b43-4b0a-a9b2-3a8a665cb000"
            ],
            "submitter_id": "HCM-BROD-0095-C15",
            "submitter_analyte_ids": [
              "HCM-BROD-0095-C15-06A-11R",
              "HCM-BROD-0095-C15-85A-01R",
              "HCM-BROD-0095-C15-85A-01D",
              "HCM-BROD-0095-C15-10B-01D",
              "HCM-BROD-0095-C15-06A-11D"
            ],
            "days_to_consent": null,
            "aliquot_ids": [
              "24b11e4f-5e04-4d28-875e-a242515f6d07",
              "4af130bc-16ef-42c0-9f01-04dc601f4165",
              "9bd07008-27df-4e7e-be83-2d0fbeb2db94",
              "11efd698-fb14-4a52-aaec-0084afc5bbe0",
              "3f7b05f1-a8cd-48f2-89f1-4a1e4eab92c5"
            ],
            "submitter_aliquot_ids": [
              "HCM-BROD-0095-C15-06A-11R-A79D-41",
              "HCM-BROD-0095-C15-85A-01R-A79D-41",
              "HCM-BROD-0095-C15-10B-01D-A79C-36",
              "HCM-BROD-0095-C15-06A-11D-A79C-36",
              "HCM-BROD-0095-C15-85A-01D-A79C-36"
            ],
            "created_datetime": "2019-04-04T14:07:27.780827-05:00",
            "diagnosis_ids": [
              "cd0da3ad-189b-4be0-a8c6-55e0294e7c73"
            ],
            "sample_ids": [
              "5fcb1b3c-6711-4caa-9c6c-31ed2c0fc238",
              "71568fd7-b545-4979-a907-ef8bf41e76db",
              "d9db85d9-5ea2-41cd-96d2-d50462b5d4b6"
            ],
            "consent_type": null,
            "submitter_sample_ids": [
              "HCM-BROD-0095-C15-85A",
              "HCM-BROD-0095-C15-06A",
              "HCM-BROD-0095-C15-10B"
            ],
            "primary_site": "Esophagus",
            "submitter_diagnosis_ids": [
              "HCM-BROD-0095-C15_diagnosis"
            ],
            "updated_datetime": "2021-01-06T22:55:10.531130-06:00",
            "case_id": "05f41641-ee22-4d41-bb87-2bfa47cd983f",
            "index_date": "Diagnosis",
            "state": "released",
            "portion_ids": [
              "0ef29dec-13e5-4faf-997c-a9c9502a353b",
              "c7660224-cef0-4d7d-b532-a4e9d4a7fb7c"
            ],
            "submitter_portion_ids": [
              "HCM-BROD-0095-C15-06A-11",
              "HCM-BROD-0095-C15-10B-01"
            ]
          },
          {
            "id": "07a067d0-7dfc-4817-b4c5-9200da20a59f",
            "lost_to_followup": null,
            "days_to_lost_to_followup": null,
            "disease_type": "Adenomas and Adenocarcinomas",
            "analyte_ids": [
              "3a6a1f63-f9d3-44f0-8659-3b3cd19a42a0",
              "6671382a-20f0-41d7-a35d-c09c7275d08a",
              "d5034c0e-2790-4fb5-82b0-e36c0626b57c"
            ],
            "submitter_id": "HCM-SANG-0268-C18",
            "submitter_analyte_ids": [
              "HCM-SANG-0268-C18-85A-01D",
              "HCM-SANG-0268-C18-85A-01R",
              "HCM-SANG-0268-C18-10A-01D"
            ],
            "days_to_consent": null,
            "aliquot_ids": [
              "e60dec0b-a2db-4b66-9e77-e8f0b8856a53",
              "2a558468-cca8-4fe9-a745-8c76436ce6cb",
              "ae8d0e5c-340b-488e-ba9e-d0912869ee8d",
              "745432c8-ea7f-48e0-bd53-28ead61a7ec0",
              "e2d32e2a-3161-40ab-88ef-0a4dfd3a72b4",
              "c2328729-0a45-4740-a9cb-3b8ff31807b3",
              "6b3085c9-63b8-4b82-8413-f775dbc6d993"
            ],
            "submitter_aliquot_ids": [
              "HCM-SANG-0268-C18-85A-01D-A80T-32-aliquot",
              "HCM-SANG-0268-C18-85A-01R-A80W-32-aliquot",
              "HCM-SANG-0268-C18-10A-01D-A79M-36",
              "HCM-SANG-0268-C18-10A-01D-A80T-32-aliquot",
              "HCM-SANG-0268-C18-85A-01R-A79O-41",
              "HCM-SANG-0268-C18-01A-01D-A80T-32-aliquot",
              "HCM-SANG-0268-C18-85A-01D-A79M-36"
            ],
            "created_datetime": "2020-05-29T12:07:57.849637-05:00",
            "diagnosis_ids": [
              "78f3dc9e-9b51-42b2-b809-405e438f7f68"
            ],
            "sample_ids": [
              "20b5f03e-26b9-4902-9bdd-667168727e6d",
              "17766175-a251-454b-9263-76af20b77290",
              "6a530ae1-79fa-4536-bbf5-86b14a80563e",
              "9eb44d2a-80c0-43d2-822a-0598a7d8e68c",
              "a3817d45-dbaa-4295-8d17-712c0c2438e4",
              "475ac2f5-327f-4d94-bebf-e03058911b59"
            ],
            "consent_type": null,
            "submitter_sample_ids": [
              "HCM-SANG-0268-C18-10A-01D-A80T-32",
              "HCM-SANG-0268-C18-85A-01D-A80T-32",
              "HCM-SANG-0268-C18-01A-01D-A80T-32",
              "HCM-SANG-0268-C18-10A",
              "HCM-SANG-0268-C18-85A-01R-A80W-32",
              "HCM-SANG-0268-C18-85A"
            ],
            "primary_site": "Colon",
            "submitter_diagnosis_ids": [
              "HCM-SANG-0268-C18_diagnosis"
            ],
            "updated_datetime": "2023-02-22T07:39:25.979291-06:00",
            "case_id": "07a067d0-7dfc-4817-b4c5-9200da20a59f",
            "index_date": "Sample Procurement",
            "state": "released",
            "portion_ids": [
              "4d2aeabd-3a5a-42fc-8bcd-301829a32883"
            ],
            "submitter_portion_ids": [
              "HCM-SANG-0268-C18-10A-01"
            ]
          },
          {
            "id": "0a6a14db-ca5c-4bf9-9125-611d672bc67b",
            "lost_to_followup": null,
            "days_to_lost_to_followup": null,
            "disease_type": "Adenomas and Adenocarcinomas",
            "analyte_ids": [
              "4c154bcd-3cd5-4e3a-bb90-948cf4c92965",
              "e8c8e41f-0a24-4d16-bcdd-3d7422150c4d",
              "7ce86045-0051-465e-8744-f6299a007bce",
              "f32ed211-100b-47ca-b0b0-774a15d58e60",
              "a70c036b-ea29-46d5-ba7f-8152947146f8",
              "117f4f00-2978-4a67-8a18-3acbd77fe930"
            ],
            "submitter_id": "HCM-SANG-0271-D12",
            "submitter_analyte_ids": [
              "HCM-SANG-0271-D12-86A-01R",
              "HCM-SANG-0271-D12-86A-01D",
              "HCM-SANG-0271-D12-85A-01R",
              "HCM-SANG-0271-D12-31B-01D",
              "HCM-SANG-0271-D12-10A-01D",
              "HCM-SANG-0271-D12-85B-01D"
            ],
            "days_to_consent": null,
            "aliquot_ids": [
              "d59f00c0-696c-4444-8e7f-c0140dbcd8e3",
              "d1848dd2-dccb-43c6-bd29-bb7f331089ef",
              "f7125c62-3864-412e-9afa-dc4970e43d05",
              "3bfc0b7c-58c7-47c9-9664-5c2d087f4485",
              "3109b83d-36ed-4850-9847-9710f3921413",
              "8dda870c-374e-49ae-8f0b-44b50ef567cd",
              "80d9347a-3684-4348-b1de-14bdca1471fd",
              "0bc14c0a-9fb9-480d-bfb5-3b7d89c83efd",
              "fe6747dd-12a0-4bb4-b2f3-9dd6f047df02",
              "98284402-4f76-4aa8-893c-95c72b5fb395"
            ],
            "submitter_aliquot_ids": [
              "HCM-SANG-0271-D12-85A-01R-A80W-32-aliquot",
              "HCM-SANG-0271-D12-31B-01D-A80U-36",
              "HCM-SANG-0271-D12-86A-01D-A85C-36",
              "HCM-SANG-0271-D12-85A-01D-A80T-32-aliquot",
              "HCM-SANG-0271-D12-01A-01D-A80T-32-aliquot",
              "HCM-SANG-0271-D12-85B-01D-A80U-36",
              "HCM-SANG-0271-D12-10A-01D-A80U-36",
              "HCM-SANG-0271-D12-10A-01D-A80T-32-aliquot",
              "HCM-SANG-0271-D12-85A-01R-A80V-41",
              "HCM-SANG-0271-D12-86A-01R-A85D-41"
            ],
            "created_datetime": "2020-07-08T11:54:16.081928-05:00",
            "diagnosis_ids": [
              "17a41ef7-aafc-4b70-8d8f-484dc5cd27bd"
            ],
            "sample_ids": [
              "ae2b48ea-7255-4d3d-ba00-1def687c3606",
              "b16a3a6b-b0b1-4749-afbc-aea8f4eb3a5d",
              "dc021e65-03cd-4210-a9b7-cdc971c22223",
              "bc55eaf9-56fb-4f98-9fd8-2e80796b2873",
              "5d4ac7e0-5a91-4e32-bf47-1febc9cc37d7",
              "371a0350-a942-4a63-ac59-873da0cd1e86",
              "4d8a7679-067e-4af7-9b11-d7722adc35ba",
              "2ccc3dc4-042a-477a-8b3f-b8b7ff838762",
              "fa202911-ce39-476e-9631-e81ffb46a402"
            ],
            "consent_type": null,
            "submitter_sample_ids": [
              "HCM-SANG-0271-D12-85A-01R-A80W-32",
              "HCM-SANG-0271-D12-85B",
              "HCM-SANG-0271-D12-01A-01D-A80T-32",
              "HCM-SANG-0271-D12-85A",
              "HCM-SANG-0271-D12-10A-01D-A80T-32",
              "HCM-SANG-0271-D12-10A",
              "HCM-SANG-0271-D12-31B",
              "HCM-SANG-0271-D12-86A",
              "HCM-SANG-0271-D12-85A-01D-A80T-32"
            ],
            "primary_site": "Colon",
            "submitter_diagnosis_ids": [
              "HCM-SANG-0271-D12_diagnosis"
            ],
            "updated_datetime": "2023-02-22T07:39:25.979291-06:00",
            "case_id": "0a6a14db-ca5c-4bf9-9125-611d672bc67b",
            "index_date": "Sample Procurement",
            "state": "released",
            "portion_ids": [
              "bcb558ac-e5bf-4b6c-af72-9b78b7b76722",
              "a9aec562-c531-458d-95f1-768e2610aae7"
            ],
            "submitter_portion_ids": [
              "HCM-SANG-0271-D12-10A-01",
              "HCM-SANG-0271-D12-31B-01"
            ]
          },
          {
            "id": "0cf7d1fe-e9c7-4e84-9497-df13ca2ed2c9",
            "lost_to_followup": null,
            "days_to_lost_to_followup": null,
            "disease_type": "Adenomas and Adenocarcinomas",
            "analyte_ids": [
              "6ef5c20b-50c3-4fda-bd1e-de4876ccab7b",
              "4a18990e-a0dc-4466-aaed-0053ffa0656a",
              "0a62151e-6c10-4e23-80ff-93970fa20f7a",
              "9b91bd41-f1f6-4712-84c9-54afa62d089b",
              "038fc540-f8e2-44b7-944c-ac6f25cac665"
            ],
            "submitter_id": "HCM-SANG-0287-C20",
            "submitter_analyte_ids": [
              "HCM-SANG-0287-C20-85A-01D",
              "HCM-SANG-0287-C20-01A-01D",
              "HCM-SANG-0287-C20-85A-01R",
              "HCM-SANG-0287-C20-10A-01D",
              "HCM-SANG-0287-C20-01A-01R"
            ],
            "aliquot_ids": [
              "43bb9903-2b44-494f-9275-62dd88803739",
              "c0b86e70-1df5-4acb-9667-cfada31b7c5e",
              "feb5f601-cd08-46bf-a8b3-4553d6ce24ba",
              "1e9d473d-c7b7-4eca-82c5-4d8b40e482a8",
              "22bc24d9-719b-4f3c-8587-e394cf409117",
              "77b7d070-0fd0-4d33-9204-b6f8cd95c415",
              "89ab0d1b-b692-4248-abc6-7dc698b54e67",
              "1870408c-6d85-4e00-9c64-178350135992",
              "48f6c991-9d58-4d1f-96e4-c62dcf0c6bf8"
            ],
            "submitter_aliquot_ids": [
              "HCM-SANG-0287-C20-01A-01D-A78U-36",
              "HCM-SANG-0287-C20-85A-01R-A80W-32-aliquot",
              "HCM-SANG-0287-C20-85A-01R-A78V-41",
              "HCM-SANG-0287-C20-10A-01D-A80T-32-aliquot",
              "HCM-SANG-0287-C20-85A-01D-A80T-32-aliquot",
              "HCM-SANG-0287-C20-01A-01D-A80T-32-aliquot",
              "HCM-SANG-0287-C20-01A-01R-A78V-41",
              "HCM-SANG-0287-C20-10A-01D-A78U-36",
              "HCM-SANG-0287-C20-85A-01D-A78U-36"
            ],
            "created_datetime": "2019-10-14T10:45:59.013881-05:00",
            "diagnosis_ids": [
              "c91c84dc-0e49-412f-9c97-8c69b195cf02"
            ],
            "sample_ids": [
              "a716e59b-6876-4c5e-85c9-a53869482d95",
              "06c2d962-ede5-4f2f-81dd-dc9253a9ddf5",
              "459eaa9f-e97f-4008-8cf6-f6671985fd30",
              "31866d02-8351-497c-8c73-1fc5fef584aa",
              "38243f28-93e3-4a8a-bf92-8019a07b3bec",
              "62ec2e41-acac-4449-b9fe-fd2c938cc811",
              "fa630ad7-f709-43ca-bf5c-d3e83dd51779"
            ],
            "submitter_sample_ids": [
              "HCM-SANG-0287-C20-01A",
              "HCM-SANG-0287-C20-85A-01D-A80T-32",
              "HCM-SANG-0287-C20-85A-01R-A80W-32",
              "HCM-SANG-0287-C20-10A",
              "HCM-SANG-0287-C20-01A-01D-A80T-32",
              "HCM-SANG-0287-C20-85A",
              "HCM-SANG-0287-C20-10A-01D-A80T-32"
            ],
            "primary_site": "Rectum",
            "submitter_diagnosis_ids": [
              "HCM-SANG-0287-C20_diagnosis"
            ],
            "updated_datetime": "2023-02-22T07:39:25.979291-06:00",
            "case_id": "0cf7d1fe-e9c7-4e84-9497-df13ca2ed2c9",
            "index_date": "Sample Procurement",
            "state": "released",
            "portion_ids": [
              "003cfb21-0ffb-44e2-9961-ebdd5c39f361",
              "a7236e0c-decf-48c3-a977-d069436420b7"
            ],
            "submitter_portion_ids": [
              "HCM-SANG-0287-C20-01A-01",
              "HCM-SANG-0287-C20-10A-01"
            ]
          },
          {
            "id": "0e9a9e97-f0bf-4f4a-84cc-73eccfc627b1",
            "lost_to_followup": null,
            "days_to_lost_to_followup": null,
            "disease_type": "Adenomas and Adenocarcinomas",
            "analyte_ids": [
              "78c3017c-6de1-4ccb-bf4d-27922b0b1f38",
              "f877946f-8a20-40dd-b5e0-5d4a350c2528",
              "c70fdba0-7324-4ea1-b15c-8de62a459a50",
              "9c7ccff1-5150-46a4-94dd-a928df9db3e7",
              "9bc12019-74f2-4784-9322-0398a2c1b3d1"
            ],
            "submitter_id": "HCM-CSHL-0376-D37",
            "submitter_analyte_ids": [
              "HCM-CSHL-0376-D37-31A-11D",
              "HCM-CSHL-0376-D37-10A-01D",
              "HCM-CSHL-0376-D37-31A-11R",
              "HCM-CSHL-0376-D37-85P-01D",
              "HCM-CSHL-0376-D37-85P-01R"
            ],
            "aliquot_ids": [
              "790b5310-35ac-4382-b062-db3078e8b20a",
              "33019b27-f718-4436-95fd-ff4d8a28c923",
              "6c69fc6a-6606-4f79-8ad8-4bf6bfaab56d",
              "aa16d6e6-6a3d-487f-82b1-9ffa258b851e",
              "5facfb50-5793-43a6-be50-5c1afe446dcc"
            ],
            "submitter_aliquot_ids": [
              "HCM-CSHL-0376-D37-31A-11R-A78V-41",
              "HCM-CSHL-0376-D37-85P-01D-A78T-36",
              "HCM-CSHL-0376-D37-31A-11D-A78T-36",
              "HCM-CSHL-0376-D37-85P-01R-A78V-41",
              "HCM-CSHL-0376-D37-10A-01D-A78T-36"
            ],
            "created_datetime": "2019-10-14T13:24:10.078043-05:00",
            "diagnosis_ids": [
              "9ce37f22-bbde-4447-bbf0-28e85f4f2837"
            ],
            "sample_ids": [
              "ac11b2da-38a9-444e-a922-3b78f15be942",
              "cbfc2432-c011-41ee-81fd-5efd8c0dac79",
              "76b129b9-954b-4471-949b-e118dc778e2d"
            ],
            "submitter_sample_ids": [
              "HCM-CSHL-0376-D37-10A",
              "HCM-CSHL-0376-D37-85P",
              "HCM-CSHL-0376-D37-31A"
            ],
            "primary_site": "Colon",
            "submitter_diagnosis_ids": [
              "HCM-CSHL-0376-D37_diagnosis"
            ],
            "updated_datetime": "2023-02-22T07:39:25.979291-06:00",
            "case_id": "0e9a9e97-f0bf-4f4a-84cc-73eccfc627b1",
            "index_date": "Diagnosis",
            "state": "released",
            "portion_ids": [
              "d0c90c2f-d6d9-4601-b2f8-6ca30b48f405",
              "f2b394b2-35f1-492b-969e-df863a2714cc"
            ],
            "submitter_portion_ids": [
              "HCM-CSHL-0376-D37-10A-01",
              "HCM-CSHL-0376-D37-31A-11"
            ]
          },
          {
            "id": "149a8565-e0c5-4474-a693-d44f1b445c0c",
            "lost_to_followup": "Yes",
            "slide_ids": [
              "846c933d-4bca-4995-ac45-1caf38ee481b",
              "4be18133-d1ff-400f-b57b-dded404249c0",
              "8f24d92e-cfee-4cce-8eec-e4f302766255",
              "08650e03-96e9-4213-8892-b3da6928171f"
            ],
            "submitter_slide_ids": [
              "HCM-BROD-0199-C71-01A-01-S2-HE",
              "HCM-BROD-0199-C71-02A-01-S2-HE",
              "HCM-BROD-0199-C71-01A-01-S1-HE",
              "HCM-BROD-0199-C71-02A-01-S1-HE"
            ],
            "days_to_lost_to_followup": null,
            "disease_type": "Gliomas",
            "analyte_ids": [
              "3b8a2f07-a6d0-4998-8fe8-c2138372e191",
              "ae4eb402-7c13-46f0-bdf8-361c7fdc9430",
              "b720a33e-c858-47f4-9e51-e58748004e96",
              "fdf06863-7d14-4f79-ad2c-b46d443d235c",
              "a4f88ff3-7c9e-4edc-808f-aa01378ec68d",
              "cca66f05-89e0-48a7-8948-53e556fee5be",
              "e658c087-c37d-47ac-9472-3b6b6eed7188",
              "6be6aff1-9694-4c7f-8f43-f0f902b95849",
              "39f9ce44-944c-49b8-b215-05b117d5e62d"
            ],
            "submitter_id": "HCM-BROD-0199-C71",
            "submitter_analyte_ids": [
              "HCM-BROD-0199-C71-85A-01D",
              "HCM-BROD-0199-C71-01A-11D",
              "HCM-BROD-0199-C71-85S-01R",
              "HCM-BROD-0199-C71-85A-01R",
              "HCM-BROD-0199-C71-02A-11R",
              "HCM-BROD-0199-C71-02A-11D",
              "HCM-BROD-0199-C71-01A-11R",
              "HCM-BROD-0199-C71-10A-01D",
              "HCM-BROD-0199-C71-85R-01D"
            ],
            "aliquot_ids": [
              "9f18daaa-cd67-436e-85a1-1e9b3e1e2135",
              "eff144c4-4e2a-497d-998a-5ae6bccf2576",
              "7ec23cac-61f4-4f8f-afa7-ed8a3ca1493f",
              "6ea15978-902f-45ba-b148-cc4247341882",
              "529f3cbf-f65d-4a60-a562-b3fcfbd7d4c9",
              "22542d68-476b-47fd-91da-03db887756d6",
              "7f4e7f6b-33b0-49da-95bd-88643d5e14ff",
              "fc0121b9-5de8-4af6-90db-b36dd8207ebf",
              "c9dd10db-bf23-4bb2-a6db-fec1b45549c8"
            ],
            "submitter_aliquot_ids": [
              "HCM-BROD-0199-C71-85R-01D-A80U-36",
              "HCM-BROD-0199-C71-02A-11R-A80V-41",
              "HCM-BROD-0199-C71-02A-11D-A80U-36",
              "HCM-BROD-0199-C71-01A-11D-A786-36",
              "HCM-BROD-0199-C71-10A-01D-A786-36",
              "HCM-BROD-0199-C71-01A-11R-A787-41",
              "HCM-BROD-0199-C71-85A-01D-A786-36",
              "HCM-BROD-0199-C71-85S-01R-A80V-41",
              "HCM-BROD-0199-C71-85A-01R-A787-41"
            ],
            "created_datetime": "2019-04-04T15:00:32.807421-05:00",
            "diagnosis_ids": [
              "df0230e6-d07c-4814-9aee-5be560d1ce58",
              "2d72ecab-4038-4ef7-b921-79b88ad62722"
            ],
            "sample_ids": [
              "b5919dd1-039e-4ab4-b6d5-37b9d483893f",
              "769c5e1b-89e3-431c-9420-31de5efe5a22",
              "5e0faf77-f1b3-4b1a-8b6a-8066270eddb4",
              "ee0e94ad-b5b0-4ad6-912c-c4010f1a1d26",
              "600a5b42-f871-47a2-8124-68184edf10bd",
              "a1dbbc0c-173c-4455-a780-8682dd2e258a"
            ],
            "submitter_sample_ids": [
              "HCM-BROD-0199-C71-85R",
              "HCM-BROD-0199-C71-02A",
              "HCM-BROD-0199-C71-85S",
              "HCM-BROD-0199-C71-85A",
              "HCM-BROD-0199-C71-10A",
              "HCM-BROD-0199-C71-01A"
            ],
            "primary_site": "Brain",
            "submitter_diagnosis_ids": [
              "HCM-BROD-0199-C71_diagnosis",
              "HCM-BROD-0199-C71_diagnosis2"
            ],
            "updated_datetime": "2021-01-06T22:55:10.531130-06:00",
            "case_id": "149a8565-e0c5-4474-a693-d44f1b445c0c",
            "index_date": "Diagnosis",
            "state": "released",
            "portion_ids": [
              "9dccf7ac-58a0-4c9e-9fb6-4ead8c37d48b",
              "5afaa056-b09e-4328-ae5b-feb70faa3595",
              "77827fa6-18a9-4e1a-b49f-800cec351fa8"
            ],
            "submitter_portion_ids": [
              "HCM-BROD-0199-C71-10A-01",
              "HCM-BROD-0199-C71-02A-11",
              "HCM-BROD-0199-C71-01A-11"
            ]
          },
          {
            "id": "19b1e69a-355a-4dd7-9c56-d701f6c2c5a0",
            "lost_to_followup": null,
            "days_to_lost_to_followup": null,
            "disease_type": "Adenomas and Adenocarcinomas",
            "analyte_ids": [
              "181d5e6e-026d-4983-97e8-f4d9e28a0cfe",
              "293683aa-f2ef-4d14-8ec7-681c870d3b71",
              "c1f6530c-004c-4b10-bae5-a571542aabd2",
              "2140e379-dd8b-4440-a626-bb0e01f8fc00",
              "4d337e44-6468-43b8-bf2f-4301750dab99"
            ],
            "submitter_id": "HCM-SANG-0299-C15",
            "submitter_analyte_ids": [
              "HCM-SANG-0299-C15-10B-01D",
              "HCM-SANG-0299-C15-85X-01R",
              "HCM-SANG-0299-C15-85A-01R",
              "HCM-SANG-0299-C15-85X-01D",
              "HCM-SANG-0299-C15-85A-01D"
            ],
            "aliquot_ids": [
              "37fee31c-8669-4057-ae95-424586fa2a05",
              "d17fd7af-fd14-4cb7-a6c1-c23339589288",
              "65b7ffa3-0f1f-4918-a75f-2343720fe40c",
              "a331201c-3aef-4eec-83d5-d38f4211c1b1",
              "3ef40a4d-21cd-4be4-8c57-cba30c8e0778",
              "b8af2a87-5b55-4c73-88f1-90ce5e1d05f5",
              "a1988273-c9f8-4cd1-b1a6-5daa2c7a3e51",
              "b535f87b-0a61-49bd-828b-bb5e3c11f2f7",
              "3bb67aa8-5155-425e-9afc-9aca5120b0f0",
              "fa20b306-a835-46d8-ab03-f8ea0b585381",
              "65ec4e30-e782-4b6f-b985-33e9c7e72a0a"
            ],
            "submitter_aliquot_ids": [
              "HCM-SANG-0299-C15-85B-01D-A80T-32-aliquot",
              "HCM-SANG-0299-C15-10A-01D-A80T-32-aliquot",
              "HCM-SANG-0299-C15-85A-01R-A78V-41",
              "HCM-SANG-0299-C15-85A-01D-A78U-36",
              "HCM-SANG-0299-C15-85A-01R-A80W-32-aliquot",
              "HCM-SANG-0299-C15-85X-01D-A78U-36",
              "HCM-SANG-0299-C15-85X-01R-A78V-41",
              "HCM-SANG-0299-C15-85A-01D-A80T-32-aliquot",
              "HCM-SANG-0299-C15-01A-01D-A80T-32-aliquot",
              "HCM-SANG-0299-C15-85B-01R-A80W-32-aliquot",
              "HCM-SANG-0299-C15-10B-01D-A78U-36"
            ],
            "created_datetime": "2019-10-14T10:46:36.257369-05:00",
            "diagnosis_ids": [
              "7c6aa4ce-6661-4491-a827-c0a8045743a6",
              "c1a6f70f-b871-4b4e-a292-aef67c2d4776"
            ],
            "sample_ids": [
              "c56f3f94-deda-4d21-8f8e-658108995dfa",
              "ce46b9f4-5244-432d-aa87-026e0a27d71a",
              "cf134dfb-5126-4bee-bcab-39d584335a21",
              "34e0d6ee-97f5-420f-9b34-4784098125f7",
              "ac8476c1-53bf-43b5-8695-db441ed1a720",
              "9000b366-a14b-44bc-a782-484e09765b2d",
              "8541bc01-57e6-4bf7-a42e-da1c8e790633",
              "277d7ed9-4e6a-429b-8f4e-438ff0d2ba7a",
              "d12d9184-5317-4730-af2e-a8428456a2a7"
            ],
            "submitter_sample_ids": [
              "HCM-SANG-0299-C15-85X",
              "HCM-SANG-0299-C15-85B-01D-A80T-32",
              "HCM-SANG-0299-C15-85A-01R-A80W-32",
              "HCM-SANG-0299-C15-10B",
              "HCM-SANG-0299-C15-01A-01D-A80T-32",
              "HCM-SANG-0299-C15-85A",
              "HCM-SANG-0299-C15-85A-01D-A80T-32",
              "HCM-SANG-0299-C15-85B-01R-A80W-32",
              "HCM-SANG-0299-C15-10A-01D-A80T-32"
            ],
            "primary_site": "Esophagus",
            "submitter_diagnosis_ids": [
              "HCM-SANG-0299-C15_diagnosis2",
              "HCM-SANG-0299-C15_diagnosis"
            ],
            "updated_datetime": "2023-02-22T07:39:25.979291-06:00",
            "case_id": "19b1e69a-355a-4dd7-9c56-d701f6c2c5a0",
            "index_date": "Sample Procurement",
            "state": "released",
            "portion_ids": [
              "b8192c57-9cda-4dca-a590-1e13beadf2a0"
            ],
            "submitter_portion_ids": [
              "HCM-SANG-0299-C15-10B-01"
            ]
          },
          {
            "id": "19f1d344-4c14-4733-abbd-c2db6737e210",
            "lost_to_followup": null,
            "days_to_lost_to_followup": null,
            "disease_type": "Ductal and Lobular Neoplasms",
            "analyte_ids": [
              "a21d5f05-af57-41bf-ab60-32d2c869611b",
              "696f4b03-5f0d-4eaa-975d-a9feb64dae07",
              "c8d4fb23-c55d-4bb6-bd3d-fc1f159d7a33",
              "5db09eb8-fecc-433f-aa34-a12a7c9333dd",
              "51598aba-be8e-42a0-bf6c-aca34776fc1f"
            ],
            "submitter_id": "HCM-CSHL-0081-C25",
            "submitter_analyte_ids": [
              "HCM-CSHL-0081-C25-11A-11D",
              "HCM-CSHL-0081-C25-85A-01R",
              "HCM-CSHL-0081-C25-85B-01D",
              "HCM-CSHL-0081-C25-01A-11R",
              "HCM-CSHL-0081-C25-01A-11D"
            ],
            "aliquot_ids": [
              "0a65e1bd-6af3-44ba-924c-193eb8e099d6",
              "61112c39-c838-4f84-89c7-a33bfe5dea88",
              "25cd08c3-09a9-406d-8ec3-ab4946224cf1",
              "973d4fa3-4fa7-4ff9-8176-cc51c53b7079",
              "9aa21acf-1e24-4b7a-a3c1-73354bdd81b6"
            ],
            "submitter_aliquot_ids": [
              "HCM-CSHL-0081-C25-11A-11D-A78M-36",
              "HCM-CSHL-0081-C25-01A-11R-A78N-41",
              "HCM-CSHL-0081-C25-85B-01D-A78M-36",
              "HCM-CSHL-0081-C25-01A-11D-A78M-36",
              "HCM-CSHL-0081-C25-85A-01R-A78N-41"
            ],
            "created_datetime": "2019-09-19T08:58:31.776805-05:00",
            "diagnosis_ids": [
              "c1cca3a7-e0ac-40a1-9db7-6902b48d3c62"
            ],
            "sample_ids": [
              "d9f23187-9a29-426a-9ead-4bb3a2ce6cf9",
              "3709004e-b04d-4473-aa29-8dd84176d17d",
              "adcc54e3-074b-4ca6-b179-0a5df8efeb36",
              "05478d15-885a-4c44-a46a-81bbe6c9ee11"
            ],
            "submitter_sample_ids": [
              "HCM-CSHL-0081-C25-85B",
              "HCM-CSHL-0081-C25-01A",
              "HCM-CSHL-0081-C25-85A",
              "HCM-CSHL-0081-C25-11A"
            ],
            "primary_site": "Pancreas",
            "submitter_diagnosis_ids": [
              "HCM-CSHL-0081-C25_diagnosis"
            ],
            "updated_datetime": "2023-02-22T07:39:25.979291-06:00",
            "case_id": "19f1d344-4c14-4733-abbd-c2db6737e210",
            "index_date": "Diagnosis",
            "state": "released",
            "portion_ids": [
              "d9588542-d7ef-413e-900c-3f816b583525",
              "b77573f1-e6a3-43c1-a56d-a207c39e18c4"
            ],
            "submitter_portion_ids": [
              "HCM-CSHL-0081-C25-01A-11",
              "HCM-CSHL-0081-C25-11A-11"
            ]
          }
        ],
        "pagination": {
          "count": 10,
          "total": 40232,
          "size": 10,
          "from": 0,
          "sort": "",
          "page": 1,
          "pages": 4024
        }
      },
      "warnings": {}
    }
    ```



#### Example: HTTP POST Request

This example demonstrates how to obtain metadata in TSV format for a set of files using their UUIDs (e.g. UUIDs obtained from a [download manifest file generated by the GDC Data Portal](/Data_Portal/Users_Guide/Repository.md#generating-a-manifest-file-for-the-data-transfer-tool)).

The first step is to construct a JSON query object, including `filters`, `fields`, `format`, and `size` parameters. The object is then submitted as HTTP POST payload to the GDC API using curl, in order to retrieve a TSV file with the requested metadata.

=== "Payload_txt"

    ```json
    {
        "filters":{
            "op":"in",
            "content":{
                "field":"files.file_id",
                "value":[
                    "0001801b-54b0-4551-8d7a-d66fb59429bf",
                    "002c67f2-ff52-4246-9d65-a3f69df6789e",
                    "003143c8-bbbf-46b9-a96f-f58530f4bb82",
                    "0043d981-3c6b-463f-b512-ab1d076d3e62",
                    "004e2a2c-1acc-4873-9379-ef1aa12283b6",
                    "005239a8-2e63-4ff1-9cd4-714f81837a61",
                    "006b8839-31e5-4697-b912-8e3f4124dd15",
                    "006ce9a8-cf38-462e-bb99-7f08499244ab",
                    "007ce9b5-3268-441e-9ffd-b40d1127a319",
                    "0084a614-780b-42ec-b85f-7a1b83128cd3",
                    "00a5e471-a79f-4d56-8a4c-4847ac037400",
                    "00ab2b5a-b59e-4ec9-b297-76f74ff1d3fb",
                    "00c5f14e-a398-4076-95d1-25f320ee3a37",
                    "00c74a8b-10aa-40cc-991e-3365ea1f3fce",
                    "00df5a50-bce3-4edf-a078-641e54800dcb"
                ]
            }
        },
        "format":"TSV",
        "fields":"file_id,file_name,cases.submitter_id,cases.case_id,data_category,data_type,cases.samples.tumor_descriptor,cases.samples.tissue_type,cases.samples.sample_type,cases.samples.submitter_id,cases.samples.sample_id,cases.samples.portions.analytes.aliquots.aliquot_id,cases.samples.portions.analytes.aliquots.submitter_id",
        "size":"100"
    }
    ```

=== "Shell"

    ```shell
    curl --request POST --header "Content-Type: application/json" --data @Payload.txt 'https://api.gdc.cancer.gov/files' > File_metadata.txt
    ```

=== "File_metadata_txt"

    ```
    cases_0_submitter_id	cases_0_case_id	data_type	cases_0_samples_0_sample_type	cases_0_samples_0_tissue_type	file_name	cases_0_samples_0_submitter_id	cases_0_samples_0_portions_0_analytes_0_aliquots_0_aliquot_id	cases_0_samples_0_sample_id	file_id	data_category	cases_0_samples_0_tumor_descriptor	cases_0_samples_0_portions_0_analytes_0_aliquots_0_submitter_id
    TCGA-B0-5094	8aaa4e25-5c12-4ace-96dc-91aaa0c4457c	Aligned Reads	Solid Tissue Normal		C345.TCGA-B0-5094-11A-01D-1421-08.5_gdc_realn.bam	TCGA-B0-5094-11A	b4e4630a-b38c-4b62-b0e8-d73f0e3b4e47	7519d7a8-c3ee-417b-9cfc-111bc5ad0637	0001801b-54b0-4551-8d7a-d66fb59429bf	Raw Sequencing Data		TCGA-B0-5094-11A-01D-1421-08
    TCGA-B0-5117	ae55b2d3-62a1-419e-9f9a-5ddfac356db4	Aligned Reads	Solid Tissue Normal		C345.TCGA-B0-5117-11A-01D-1421-08.5_gdc_realn.bam	TCGA-B0-5117-11A	45c68b6b-0bed-424d-9a77-4f87bbaa3649	b1116541-bece-4df3-b3dd-cec50aeb277b	003143c8-bbbf-46b9-a96f-f58530f4bb82	Raw Sequencing Data		TCGA-B0-5117-11A-01D-1421-08
    TCGA-G7-6790	e7a1cbe2-793c-4747-8412-8be794f2382b	Aligned Reads	Blood Derived Normal		C489.TCGA-G7-6790-10A-01D-1962-08.2_gdc_realn.bam	TCGA-G7-6790-10A	66cbb40f-14b3-40c0-a332-e8a8e21bca11	4be83d0f-8b09-4e9e-8318-358371d34332	004e2a2c-1acc-4873-9379-ef1aa12283b6	Raw Sequencing Data		TCGA-G7-6790-10A-01D-1962-08
    TCGA-B9-A69E	a4225cb2-7b4b-4122-b6b9-629c26e3ea56	Aligned Reads	Blood Derived Normal		TCGA-B9-A69E-10A-01D-A31X-10_Illumina_gdc_realn.bam	TCGA-B9-A69E-10A	f4799bdc-b207-4053-9a4b-5a26ebf8ab91	5d6d6cd4-6a7b-499d-936a-1be9bf74b07f	0084a614-780b-42ec-b85f-7a1b83128cd3	Raw Sequencing Data		TCGA-B9-A69E-10A-01D-A31X-10
    TCGA-EE-A2GU	24faa36a-268d-4a13-b3ae-eacd431a2bcc	Aligned Reads	Blood Derived Normal		C828.TCGA-EE-A2GU-10A-01D-A198-08.2_gdc_realn.bam	TCGA-EE-A2GU-10A	c3feacc2-5a26-4bb2-a312-8b2ee53ccad1	cc4a5ed8-376a-4842-a25d-ffb07d8e1ca0	00c74a8b-10aa-40cc-991e-3365ea1f3fce	Raw Sequencing Data		TCGA-EE-A2GU-10A-01D-A198-08
    TCGA-CE-A484	e62a728d-390f-428a-bea1-fc8c9814fb11	Aligned Reads	Blood Derived Normal		C499.TCGA-CE-A484-10A-01D-A23U-08.3_gdc_realn.bam	TCGA-CE-A484-10A	641a0220-6eec-434a-b606-e256113b65da	27a8008e-044a-4966-b518-cc6905e292ca	00df5a50-bce3-4edf-a078-641e54800dcb	Raw Sequencing Data		TCGA-CE-A484-10A-01D-A23U-08
    TCGA-DA-A1IB	8fc9cc74-f388-49f0-b957-debb62638634	Aligned Reads	Blood Derived Normal		C828.TCGA-DA-A1IB-10A-01D-A198-08.2_gdc_realn.bam	TCGA-DA-A1IB-10A	30919a1a-df9f-4604-835e-f66ac7bcacdf	432952c5-6505-4220-a581-f65270a45281	00ab2b5a-b59e-4ec9-b297-76f74ff1d3fb	Raw Sequencing Data		TCGA-DA-A1IB-10A-01D-A198-08
    TCGA-AX-A2HG	7a2cf5ce-8317-4fff-946e-b9937afab815	Aligned Reads	Blood Derived Normal		6c2a8ea343da8d6cc0fd2043492f16df_gdc_realn.bam	TCGA-AX-A2HG-10A	8c34ffe2-9012-4b4a-b610-a42a9c6a9780	ef4b80ec-b453-48ec-8ad8-ccac83e1e4db	00c5f14e-a398-4076-95d1-25f320ee3a37	Raw Sequencing Data		TCGA-AX-A2HG-10A-01D-A17D-09
    TCGA-EC-A24G	b5c1e511-baf2-45b3-9919-110e8941e3c2	Aligned Reads	Blood Derived Normal		671333b193812fc2bd2744053b383459_gdc_realn.bam	TCGA-EC-A24G-10A	2a8cb8fe-b64f-453e-8139-7ede12f3fc51	61cf2e54-1b8d-40a0-9c73-a7449cbd570a	00a5e471-a79f-4d56-8a4c-4847ac037400	Raw Sequencing Data		TCGA-EC-A24G-10A-01D-A16D-09
    TCGA-B5-A0K0	29c8f468-5ac1-4d6c-8376-e36e6d246926	Aligned Reads	Blood Derived Normal		TCGA-B5-A0K0-10A-01W-A062-09_IlluminaGA-DNASeq_exome_gdc_realn.bam	TCGA-B5-A0K0-10A	02e65074-ffda-4795-b8f5-1bfd20bd1019	1df69e2e-f392-465f-8e61-4671ba2fcd35	007ce9b5-3268-441e-9ffd-b40d1127a319	Raw Sequencing Data		TCGA-B5-A0K0-10A-01W-A062-09
    TCGA-C8-A27B	f0d8a1fe-e313-44f1-99cc-b965cbeeff0e	Aligned Reads	Blood Derived Normal		3c99d98ea8eb6acbf819e67fc77623d9_gdc_realn.bam	TCGA-C8-A27B-10A	922226ba-6244-4953-ad42-f4daa474c288	31139082-7978-45aa-9d8f-ac4789ac5cec	006b8839-31e5-4697-b912-8e3f4124dd15	Raw Sequencing Data		TCGA-C8-A27B-10A-01D-A167-09
    TCGA-E9-A295	fec0da58-1047-44d2-b6d1-c18cceed43dc	Aligned Reads	Blood Derived Normal		fd4421a6bbf3efd4e3d5c17fdd610314_gdc_realn.bam	TCGA-E9-A295-10A	cd761feb-9a20-4495-8943-c6243532a5cf	e74183e1-f0b4-412a-8dac-a62d404add78	002c67f2-ff52-4246-9d65-a3f69df6789e	Raw Sequencing Data		TCGA-E9-A295-10A-01D-A16D-09
    TCGA-EB-A44O	c787c4da-c564-44f1-89eb-dd9da107acb1	Aligned Reads	Blood Derived Normal		C828.TCGA-EB-A44O-10A-01D-A25O-08.3_gdc_realn.bam	TCGA-EB-A44O-10A	c723584a-c404-4c88-bfea-e40f5dbba542	5b738547-1825-4684-81bd-864bf2eb43ef	006ce9a8-cf38-462e-bb99-7f08499244ab	Raw Sequencing Data		TCGA-EB-A44O-10A-01D-A25O-08
    TCGA-A2-A3XX	53886143-c1c6-40e9-88e6-e4e5e0271fc8	Aligned Reads	Blood Derived Normal		b40998d4778f18ed80d6dd8bff0eb761_gdc_realn.bam	TCGA-A2-A3XX-10A	e96d5811-4736-40dd-966d-e0e172aeb0af	c6eb6218-ad71-40a6-88b7-a4f1a015b816	0043d981-3c6b-463f-b512-ab1d076d3e62	Raw Sequencing Data		TCGA-A2-A3XX-10A-01D-A23C-09
    TCGA-EB-A3XB	a9255dcb-b236-4777-ac43-555e3a5386c3	Aligned Reads	Blood Derived Normal		C828.TCGA-EB-A3XB-10B-01D-A23B-08.1_gdc_realn.bam	TCGA-EB-A3XB-10B	9f4ffc2f-d006-4d86-b3b1-b25020481893	0e1d4c7c-204d-4765-b090-68ed4cd83835	005239a8-2e63-4ff1-9cd4-714f81837a61	Raw Sequencing Data		TCGA-EB-A3XB-10B-01D-A23B-08
    ```


### Format

Specifies the format of the API response: JSON (default), `TSV` or `XML`.

#### Examples

=== "Shell1"

    ```shell
    curl  'https://api.gdc.cancer.gov/cases?fields=submitter_id&size=5&format=TSV'
    ```

=== "Python1"

    ```python
    import requests

    cases_endpt = 'https://api.gdc.cancer.gov/cases'
    params = {'fields':'submitter_id',
              'format':'TSV'}
    response = requests.get(cases_endpt, params = params)
    print(response.content)
    ```

=== "Response1"

    ```
    id	submitter_id
    0286c31b-a704-4d7d-99e3-0bc4e8975b8b	HCM-CSHL-0084-C25
    02f6d684-b6b5-419a-b0e1-b74d0a384a30	HCM-BROD-0408-C71
    03974dc9-0162-4de8-9897-09f88693681a	HCM-BROD-0334-C43
    03bfeb7c-cecf-4691-8263-33cdfe391ea9	HCM-BROD-0124-C25
    04cbceab-f945-482b-956b-840756a17a4a	HCM-BROD-0421-C71

    ```

=== "Shell 2"

    ```shell
    curl  'https://api.gdc.cancer.gov/cases?fields=submitter_id&size=5&format=XML&pretty=true'
    ```

=== "Python2"

    ```python
    import requests

    cases_endpt = 'https://api.gdc.cancer.gov/cases'
    params = {'fields':'submitter_id',
              'format':'XML',
              'pretty':'true'}
    response = requests.get(cases_endpt, params = params)
    print(response.content)
    ```

=== "Output2"

    ```xml
    <?xml version="1.0" ?>
    <response>
    	<data>
    		<hits>
    			<item>
    				<id>0286c31b-a704-4d7d-99e3-0bc4e8975b8b</id>
    				<submitter_id>HCM-CSHL-0084-C25</submitter_id>
    			</item>
    			<item>
    				<id>02f6d684-b6b5-419a-b0e1-b74d0a384a30</id>
    				<submitter_id>HCM-BROD-0408-C71</submitter_id>
    			</item>
    			<item>
    				<id>03974dc9-0162-4de8-9897-09f88693681a</id>
    				<submitter_id>HCM-BROD-0334-C43</submitter_id>
    			</item>
    			<item>
    				<id>03bfeb7c-cecf-4691-8263-33cdfe391ea9</id>
    				<submitter_id>HCM-BROD-0124-C25</submitter_id>
    			</item>
    			<item>
    				<id>04cbceab-f945-482b-956b-840756a17a4a</id>
    				<submitter_id>HCM-BROD-0421-C71</submitter_id>
    			</item>
    		</hits>
    		<pagination>
    			<count>5</count>
    			<total>86962</total>
    			<size>5</size>
    			<from>0</from>
    			<sort/>
    			<page>1</page>
    			<pages>17393</pages>
    		</pagination>
    	</data>
    	<warnings/>
    </response>
    ```

### Pretty

Returns when the `pretty` parameter is set to `true`, the API response is formatted with additional whitespace to improve legibility.

#### Example

=== "Request1"

    ```shell
    curl  'https://api.gdc.cancer.gov/cases?fields=submitter_id&sort=submitter_id:asc&size=5'
    ```

=== "Response1"

    ```json
    {"data": {"hits": [{"id": "be37f1f7-2f98-4f74-bc04-6dd2ae2afcad", "submitter_id": "01BR001"}, {"id": "e6915db0-7c89-484d-8f9f-15cca68b82fc", "submitter_id": "01BR008"}, {"id": "16614d46-172b-479c-992b-e80a8e9a2c59", "submitter_id": "01BR009"}, {"id": "567fc9e3-17a6-42b1-a896-5e9a9507d1d8", "submitter_id": "01BR010"}, {"id": "54e89878-a1bc-4f5a-9d68-4842a469586e", "submitter_id": "01BR015"}], "pagination": {"count": 5, "total": 86962, "size": 5, "from": 0, "sort": "None", "page": 1, "pages": 17393}}, "warnings": {}}
    ```

=== "Request2"

    ```shell
    curl  'https://api.gdc.cancer.gov/cases?fields=submitter_id&sort=submitter_id:asc&size=5&pretty=true'
    ```

=== "Response2"

    ```json
    {
      "data": {
        "hits": [
          {
            "id": "be37f1f7-2f98-4f74-bc04-6dd2ae2afcad",
            "submitter_id": "01BR001"
          },
          {
            "id": "e6915db0-7c89-484d-8f9f-15cca68b82fc",
            "submitter_id": "01BR008"
          },
          {
            "id": "16614d46-172b-479c-992b-e80a8e9a2c59",
            "submitter_id": "01BR009"
          },
          {
            "id": "567fc9e3-17a6-42b1-a896-5e9a9507d1d8",
            "submitter_id": "01BR010"
          },
          {
            "id": "54e89878-a1bc-4f5a-9d68-4842a469586e",
            "submitter_id": "01BR015"
          }
        ],
        "pagination": {
          "count": 5,
          "total": 86962,
          "size": 5,
          "from": 0,
          "sort": "None",
          "page": 1,
          "pages": 17393
        }
      },
      "warnings": {}
    }
    ```

### Fields

This query parameter specifies which fields are to be included in the API response. The fields in the API response will be unordered. A listing of available fields for each endpoint is provided in [Appendix A](Appendix_A_Available_Fields.md).

#### Example

The following example requests case submitter ID, file UUID, file name and file size from the `files` endpoint.

=== "Shell"

    ```shell
    curl 'https://api.gdc.cancer.gov/files?fields=cases.submitter_id,file_id,file_name,file_size&pretty=true'
    ```

=== "Python"    

    ```python
    import requests
    import json
    
    files_endpt = 'https://api.gdc.cancer.gov/files'
    params = {'fields':'cases.submitter_id,file_id,file_name,file_size'}
    response = requests.get(files_endpt, params = params)
    print(json.dumps(response.json(), indent=2))
    ```

=== "Response"

    ```json
    {
      "data": {
        "hits": [
          {
            "id": "d570eccc-3c1c-4c4f-ae04-96be71fbe016",
            "cases": [
              {
                "submitter_id": "TCGA-AN-A0FL"
              }
            ],
            "file_name": "TCGA-AN-A0FL-01Z-00-DX1.20A041C6-A306-4599-A7D1-65032A252AA9.svs",
            "file_id": "d570eccc-3c1c-4c4f-ae04-96be71fbe016",
            "file_size": 1055798681
          },
          {
            "id": "0f8d8202-a1ca-4ea1-98b2-c20a6b08479a",
            "cases": [
              {
                "submitter_id": "TCGA-AN-A0FL"
              }
            ],
            "file_name": "nationwidechildrens.org_ssf.TCGA-AN-A0FL.xml",
            "file_id": "0f8d8202-a1ca-4ea1-98b2-c20a6b08479a",
            "file_size": 15519
          },
          {
            "id": "b76f87b3-99c5-4297-b2df-8cbea8ecaf61",
            "cases": [
              {
                "submitter_id": "TCGA-BH-A18F"
              }
            ],
            "file_name": "7c4e4c2a-a0b1-424f-97d8-359825674429.wxs.aliquot_ensemble_masked.maf.gz",
            "file_id": "b76f87b3-99c5-4297-b2df-8cbea8ecaf61",
            "file_size": 21571
          },
          {
            "id": "be6d269d-4305-4643-b98e-af703a067761",
            "cases": [
              {
                "submitter_id": "TCGA-BH-A18F"
              }
            ],
            "file_name": "HITCH_p_TCGASNP_b93_N_GenomeWideSNP_6_E11_741424.CEL",
            "file_id": "be6d269d-4305-4643-b98e-af703a067761",
            "file_size": 69084893
          },
          {
            "id": "fed73119-1d5e-4f7e-9713-183d1916422b",
            "cases": [
              {
                "submitter_id": "TCGA-BH-A18F"
              }
            ],
            "file_name": "3b928f83-14a7-4bd6-a9b0-744b414d4495.wxs.varscan2.raw_somatic_mutation.vcf.gz",
            "file_id": "fed73119-1d5e-4f7e-9713-183d1916422b",
            "file_size": 35903
          },
          {
            "id": "6877b045-91f1-4030-82ff-b90507e11e17",
            "cases": [
              {
                "submitter_id": "TCGA-BH-A18F"
              }
            ],
            "file_name": "5057e3cb-25cd-4a67-8d31-6ac8508ba3c7.methylation_array.sesame.level3betas.txt",
            "file_id": "6877b045-91f1-4030-82ff-b90507e11e17",
            "file_size": 770500
          },
          {
            "id": "07e8cdc7-d228-4752-ad19-800abd507277",
            "cases": [
              {
                "submitter_id": "TCGA-BH-A0BM"
              }
            ],
            "file_name": "TCGA-BRCA.28dcad29-448e-4bcb-911d-556c6f4a5573.star_fusion.rna_fusion.tsv",
            "file_id": "07e8cdc7-d228-4752-ad19-800abd507277",
            "file_size": 234
          },
          {
            "id": "fef57b45-ede1-49b0-b60d-957a55a15e0e",
            "cases": [
              {
                "submitter_id": "TCGA-BH-A0BM"
              }
            ],
            "file_name": "nationwidechildrens.org_biospecimen.TCGA-BH-A0BM.xml",
            "file_id": "fef57b45-ede1-49b0-b60d-957a55a15e0e",
            "file_size": 127218
          },
          {
            "id": "81a1b323-88b6-4837-bccf-ac84a79828b6",
            "cases": [
              {
                "submitter_id": "TCGA-BH-A0BM"
              }
            ],
            "file_name": "TCGA-BRCA.4570b87f-8116-48bf-86d3-b993536c88db.gene_level_copy_number.v36.tsv",
            "file_id": "81a1b323-88b6-4837-bccf-ac84a79828b6",
            "file_size": 3446816
          },
          {
            "id": "c6bf94a6-9940-4155-86b4-bbb10875dbdb",
            "cases": [
              {
                "submitter_id": "TCGA-BH-A18F"
              }
            ],
            "file_name": "TCGA-BRCA.88cae21a-4890-4fdd-a678-c4864620942c.star_fusion.rna_fusion.bedpe",
            "file_id": "c6bf94a6-9940-4155-86b4-bbb10875dbdb",
            "file_size": 229
          }
        ],
        "pagination": {
          "count": 10,
          "total": 931947,
          "size": 10,
          "from": 0,
          "sort": "",
          "page": 1,
          "pages": 93195
        }
      },
      "warnings": {}
    }
    ```

### Expand

The `expand` parameter provides a shortcut to request multiple related fields (field groups) in the response. Instead of specifying each field using the `fields` parameter, users can specify a field group name using the `expand` parameter to request all fields in the group. Available field groups are listed in [Appendix A](Appendix_A_Available_Fields.md#field-group-listing-by-endpoint); the list can also be accessed programmatically at the [_mapping endpoint](#95mapping-endpoint). The `fields` and `expand` parameters can be used together to request custom combinations of field groups and individual fields.

#### Example

=== "Shell"

    ```Shell
    curl 'https://api.gdc.cancer.gov/files/573ee7e9-b8bd-419e-808b-a027c4311731?expand=cases.samples&pretty=true'
    ```

=== "Response"

    ```json
    {
      "data": {
        "proportion_reads_mapped": 0.9648433596149857,
        "access": "controlled",
        "proportion_base_mismatch": 0.004117986,
        "contamination_error": 0,
        "acl": [
          "phs000178"
        ],
        "type": "aligned_reads",
        "platform": "Illumina",
        "created_datetime": "2022-05-12T14:42:10.014925-05:00",
        "updated_datetime": "2022-11-01T11:52:54.136033-05:00",
        "pairs_on_diff_chr": 1170013,
        "state": "released",
        "data_format": "BAM",
        "total_reads": 379313036,
        "proportion_coverage_30x": 0.000109,
        "cases": [
          {
            "samples": [
              {
                "sample_type_id": "10",
                "tumor_descriptor": "Not Reported",
                "sample_id": "4e128a37-be58-477a-a01f-448179360b7c",
                "sample_type": "Blood Derived Normal",
                "tumor_code": null,
                "created_datetime": null,
                "time_between_excision_and_freezing": null,
                "composition": "Not Reported",
                "updated_datetime": "2022-04-28T22:05:09.013808-05:00",
                "days_to_collection": 6755,
                "state": "released",
                "initial_weight": null,
                "preservation_method": null,
                "intermediate_dimension": null,
                "time_between_clamping_and_freezing": null,
                "freezing_method": null,
                "pathology_report_uuid": null,
                "submitter_id": "TCGA-B6-A0RI-10A",
                "tumor_code_id": null,
                "shortest_dimension": null,
                "oct_embedded": "false",
                "days_to_sample_procurement": null,
                "longest_dimension": null,
                "current_weight": null,
                "is_ffpe": false,
                "tissue_type": "Not Reported"
              }
            ]
          }
        ],
        "file_name": "c9478f7d-bfe3-4e80-8161-39b3d440fa16_wgs_gdc_realn.bam",
        "mean_coverage": 5.452655,
        "proportion_reads_duplicated": 0.009253781617987946,
        "submitter_id": "a4e380e5-420e-49af-986d-e721601065fb",
        "data_category": "Sequencing Reads",
        "proportion_coverage_10x": 0.07674,
        "file_size": 42958286722,
        "contamination": 0,
        "average_base_quality": 32,
        "file_id": "573ee7e9-b8bd-419e-808b-a027c4311731",
        "data_type": "Aligned Reads",
        "average_insert_size": 207,
        "average_read_length": 51,
        "experimental_strategy": "WGS",
        "version": "1",
        "data_release": "36.0 - 37.0"
      },
      "warnings": {}
    }
    ```

### Size and From

GDC API provides a pagination feature that limits the number of results returned by the API. It is implemented using `size` and `from` query parameters.

The `size` query parameter specifies the maximum number of results to return. Default `size` is 10. If the number of query results is greater than `size`, only some of the results will be returned.

The `from` query parameter specifies the first record to return out of the set of results. For example, if there are 20 cases returned from the `cases` endpoint, then setting `from` to `11` will return results 12 to 20. The `from` parameter can be used in conjunction with the `size` parameter to return a specific subset of results.


#### Example

=== "Shell1"

    ``` Shell
    curl 'https://api.gdc.cancer.gov/files?fields=file_name&from=0&size=2&pretty=true'
    ```

===  "Python1"

    ``` Python
    import requests
    import json

    files_endpt = 'https://api.gdc.cancer.gov/files'
    params = {'fields':'file_name',
              'from':0, 'size':2}
    response = requests.get(files_endpt, params = params)
    print(json.dumps(response.json(), indent=2))

    ```

=== "Response1"

    ```json
    {
      "data": {
        "hits": [
          {
            "id": "d570eccc-3c1c-4c4f-ae04-96be71fbe016",
            "file_name": "TCGA-AN-A0FL-01Z-00-DX1.20A041C6-A306-4599-A7D1-65032A252AA9.svs"
          },
          {
            "id": "0f8d8202-a1ca-4ea1-98b2-c20a6b08479a",
            "file_name": "nationwidechildrens.org_ssf.TCGA-AN-A0FL.xml"
          }
        ],
        "pagination": {
          "count": 2,
          "total": 931947,
          "size": 2,
          "from": 0,
          "sort": "",
          "page": 1,
          "pages": 465974
        }
      },
      "warnings": {}
    }
    ```

=== "Shell2"

    ``` Shell
    curl 'https://api.gdc.cancer.gov/files?fields=file_name&from=101&size=5&pretty=true'
    ```

=== "Python2"

    ``` Python
    import requests
    import json

    files_endpt = 'https://api.gdc.cancer.gov/files'
    params = {'fields':'file_name',
              'from':101, 'size':5}
    response = requests.get(files_endpt, params = params)
    print(json.dumps(response.json(), indent=2))
    ```

=== "Output2"

    ```json
    {
      "data": {
        "hits": [
          {
            "id": "297933f5-1316-4cb6-b53f-9dbfa7f3d7ed",
            "file_name": "TCGA-B6-A0RH-01A-02-TSB.ea83f31e-defb-4436-8a58-5b66b18d13b5.svs"
          },
          {
            "id": "2f31e897-b3e8-49f1-a400-ccf9f00f294a",
            "file_name": "URAEI_p_TCGASNP_b85_N_GenomeWideSNP_6_F01_735050.grch38.seg.v2.txt"
          },
          {
            "id": "ebd6cf90-4f6b-4193-887a-22fdb5645fbc",
            "file_name": "TCGA-BRCA.5994c06d-ee9b-4ead-b3d1-2e1f286f7d6d.ascat2.allelic_specific.seg.txt"
          },
          {
            "id": "aebd6b5a-e676-4357-93df-523b31b55ea0",
            "file_name": "TCGA-BRCA.c737131c-636f-4e1b-89b8-bb2d6ddd8164.star_fusion.rna_fusion.bedpe"
          },
          {
            "id": "aa83a7e7-e9cc-4330-a7be-ca750cffb74c",
            "file_name": "URAEI_p_TCGASNP_b85_N_GenomeWideSNP_6_F01_735050.birdseed.data.txt"
          }
        ],
        "pagination": {
          "count": 5,
          "total": 931947,
          "size": 5,
          "from": 101,
          "sort": "",
          "page": 21,
          "pages": 186390
        }
      },
      "warnings": {}
    }
    ```

### Sort

The `sort` query parameter sorts the results by a specific field, and with the sort direction specified using the `:asc` (ascending) or `:desc` (descending) prefix, e.g. `sort=field:desc`. A list of all valid _field_ names is available in [Appendix A](Appendix_A_Available_Fields.md); the list can also be accessed programmatically at the [_mapping endpoint](#95mapping-endpoint).

#### Example

Sort cases by `submitter_id` in ascending order:

=== "Shell"

    ``` shell
    curl  'https://api.gdc.cancer.gov/cases?fields=submitter_id&sort=submitter_id:asc&pretty=true'
    ```

=== "Python"

    ``` python
    import requests
    import json

    cases_endpt = 'https://api.gdc.cancer.gov/cases'
    params = {'fields':'submitter_id',
              'sort':'submitter_id:asc'}
    response = requests.get(cases_endpt, params = params)
    print(json.dumps(response.json(), indent=2))

    ```

=== "Output"
    
    ``` json
    {
      "data": {
        "hits": [
          {
            "id": "be37f1f7-2f98-4f74-bc04-6dd2ae2afcad",
            "submitter_id": "01BR001"
          },
          {
            "id": "e6915db0-7c89-484d-8f9f-15cca68b82fc",
            "submitter_id": "01BR008"
          },
          {
            "id": "16614d46-172b-479c-992b-e80a8e9a2c59",
            "submitter_id": "01BR009"
          },
          {
            "id": "567fc9e3-17a6-42b1-a896-5e9a9507d1d8",
            "submitter_id": "01BR010"
          },
          {
            "id": "54e89878-a1bc-4f5a-9d68-4842a469586e",
            "submitter_id": "01BR015"
          },
          {
            "id": "a1c7b7b9-b8c8-48c3-9420-55497f9318fd",
            "submitter_id": "01BR017"
          },
          {
            "id": "ce3c8b98-e275-4cfd-a379-940d675a564b",
            "submitter_id": "01BR018"
          },
          {
            "id": "e4ce89ef-bcaa-418a-8a6b-3602793b9bbf",
            "submitter_id": "01BR020"
          },
          {
            "id": "19d3c861-8a5f-49a2-acc0-b55b25465c35",
            "submitter_id": "01BR023"
          },
          {
            "id": "afae8dce-294a-4108-bb28-376f804ae5c4",
            "submitter_id": "01BR025"
          }
        ],
        "pagination": {
          "count": 10,
          "total": 86962,
          "size": 10,
          "from": 0,
          "sort": "None",
          "page": 1,
          "pages": 8697
        }
      },
      "warnings": {}
    }
    ```

### Facets
The `facets` parameter provides aggregate information for a specified field. It provides all values that exist for that field, and the number of entities (cases, projects, files, or annotations) that this value. The primary intended use of this parameter is for displaying aggregate information in the GDC Data Portal.

The `facets` parameter can be used in conjunction with the `filters` parameter to get aggregate information for a set of search results. The following limitations apply when using `facets` and `filters` together:

1. The `filters` object's top level operator must be `and`, and the internal filters must be limited to: `=`, `!=`, `in`, `exclude`, `is`, and `not`.
2. The information provided by `facets` for a given field will disregard any filters applied to that same field.

#### Example

This is an example of a request for a count of projects in each program.

=== "Shell"

    ```shell
    curl  'https://api.gdc.cancer.gov/projects?facets=program.name&from=0&size=0&sort=program.name:asc&pretty=true'
    ```

=== "Python"

    ```python
    import requests
    import json

    projects_endpt = 'https://api.gdc.cancer.gov/projects'
    params = {'facets':'program.name',
              'from':0, 'size':0,
              'sort':'program.name:asc'}
    response = requests.get(projects_endpt, params = params)
    print(json.dumps(response.json(), indent=2))
    ```

=== "Response"

    ```
    {
      "data": {
        "hits": [],
        "aggregations": {
          "program.name": {
            "buckets": [
              {
                "doc_count": 33,
                "key": "TCGA"
              },
              {
                "doc_count": 10,
                "key": "MATCH"
              },
              {
                "doc_count": 9,
                "key": "TARGET"
              },
              {
                "doc_count": 4,
                "key": "CGCI"
              },
              {
                "doc_count": 3,
                "key": "CMI"
              },
              {
                "doc_count": 2,
                "key": "BEATAML1.0"
              },
              {
                "doc_count": 2,
                "key": "CPTAC"
              },
              {
                "doc_count": 2,
                "key": "MP2PRT"
              },
              {
                "doc_count": 1,
                "key": "APOLLO"
              },
              {
                "doc_count": 1,
                "key": "CDDP_EAGLE"
              },
              {
                "doc_count": 1,
                "key": "CTSP"
              },
              {
                "doc_count": 1,
                "key": "EXCEPTIONAL_RESPONDERS"
              },
              {
                "doc_count": 1,
                "key": "FM"
              },
              {
                "doc_count": 1,
                "key": "HCMI"
              },
              {
                "doc_count": 1,
                "key": "MMRF"
              },
              {
                "doc_count": 1,
                "key": "NCICCR"
              },
              {
                "doc_count": 1,
                "key": "OHSU"
              },
              {
                "doc_count": 1,
                "key": "ORGANOID"
              },
              {
                "doc_count": 1,
                "key": "REBC"
              },
              {
                "doc_count": 1,
                "key": "TRIO"
              },
              {
                "doc_count": 1,
                "key": "VAREPOP"
              },
              {
                "doc_count": 1,
                "key": "WCDT"
              }
            ]
          }
        },
        "pagination": {
          "count": 0,
          "total": 79,
          "size": 0,
          "from": 0,
          "sort": "None",
          "page": 1,
          "pages": 79
        }
      },
      "warnings": {}
    }
    ```

#### Example

In this sample POST request, both `filters` and `facets` parameters are used. Note that `facets` ignores the `primary_site` filter.

=== "Payload"

    ```json
    {
        "filters":{
            "op":"and",
            "content":[
                {
                    "op":"=",
                    "content":{
                        "field":"cases.project.primary_site",
                        "value":"Kidney"
                    }
                },
                {
                    "op":"=",
                    "content":{
                        "field":"project.program.name",
                        "value":"TCGA"
                    }
                }
            ]
        },
        "size":"0",
        "facets":"project.primary_site",
        "pretty":"true"
    }
    ```

=== "Shell"

    ```Shell
    curl --request POST --header "Content-Type: application/json" --data @Payload 'https://api.gdc.cancer.gov/v0/cases'
    ```

=== "Response"

    ```json
    {
      "data": {
        "hits": [],
        "aggregations": {
          "project.primary_site": {
            "buckets": [
              {
                "doc_count": 1202,
                "key": "kidney"
              },
              {
                "doc_count": 1191,
                "key": "brain"
              },
              {
                "doc_count": 1176,
                "key": "bronchus and lung"
              },
              {
                "doc_count": 1156,
                "key": "breast"
              },
              {
                "doc_count": 952,
                "key": "colon"
              },
              {
                "doc_count": 947,
                "key": "stomach"
              },
              {
                "doc_count": 878,
                "key": "uterus, nos"
              },
              {
                "doc_count": 869,
                "key": "ovary"
              },
              {
                "doc_count": 821,
                "key": "corpus uteri"
              },
              {
                "doc_count": 789,
                "key": "other and unspecified parts of tongue"
              },
              {
                "doc_count": 670,
                "key": "connective, subcutaneous and other soft tissues"
              },
              {
                "doc_count": 633,
                "key": "rectosigmoid junction"
              },
              {
                "doc_count": 586,
                "key": "bones, joints and articular cartilage of other and unspecified sites"
              },
              {
                "doc_count": 565,
                "key": "thyroid gland"
              },
              {
                "doc_count": 528,
                "key": "base of tongue"
              },
              {
                "doc_count": 528,
                "key": "floor of mouth"
              },
              {
                "doc_count": 528,
                "key": "gum"
              },
              {
                "doc_count": 528,
                "key": "hypopharynx"
              },
              {
                "doc_count": 528,
                "key": "larynx"
              },
              {
                "doc_count": 528,
                "key": "lip"
              },
              {
                "doc_count": 528,
                "key": "oropharynx"
              },
              {
                "doc_count": 528,
                "key": "other and ill-defined sites in lip, oral cavity and pharynx"
              },
              {
                "doc_count": 528,
                "key": "other and unspecified parts of mouth"
              },
              {
                "doc_count": 528,
                "key": "palate"
              },
              {
                "doc_count": 528,
                "key": "tonsil"
              },
              {
                "doc_count": 500,
                "key": "prostate gland"
              },
              {
                "doc_count": 498,
                "key": "retroperitoneum and peritoneum"
              },
              {
                "doc_count": 470,
                "key": "skin"
              },
              {
                "doc_count": 448,
                "key": "heart, mediastinum, and pleura"
              },
              {
                "doc_count": 428,
                "key": "liver and intrahepatic bile ducts"
              },
              {
                "doc_count": 412,
                "key": "bladder"
              },
              {
                "doc_count": 307,
                "key": "cervix uteri"
              },
              {
                "doc_count": 271,
                "key": "adrenal gland"
              },
              {
                "doc_count": 261,
                "key": "bones, joints and articular cartilage of limbs"
              },
              {
                "doc_count": 261,
                "key": "meninges"
              },
              {
                "doc_count": 261,
                "key": "other and unspecified male genital organs"
              },
              {
                "doc_count": 261,
                "key": "peripheral nerves and autonomic nervous system"
              },
              {
                "doc_count": 258,
                "key": "hematopoietic and reticuloendothelial systems"
              },
              {
                "doc_count": 208,
                "key": "testis"
              },
              {
                "doc_count": 185,
                "key": "esophagus"
              },
              {
                "doc_count": 185,
                "key": "pancreas"
              },
              {
                "doc_count": 179,
                "key": "other and ill-defined sites"
              },
              {
                "doc_count": 179,
                "key": "other endocrine glands and related structures"
              },
              {
                "doc_count": 179,
                "key": "spinal cord, cranial nerves, and other parts of central nervous system"
              },
              {
                "doc_count": 172,
                "key": "rectum"
              },
              {
                "doc_count": 172,
                "key": "unknown"
              },
              {
                "doc_count": 124,
                "key": "thymus"
              },
              {
                "doc_count": 80,
                "key": "eye and adnexa"
              },
              {
                "doc_count": 58,
                "key": "lymph nodes"
              },
              {
                "doc_count": 58,
                "key": "other and unspecified major salivary glands"
              },
              {
                "doc_count": 58,
                "key": "small intestine"
              },
              {
                "doc_count": 51,
                "key": "gallbladder"
              },
              {
                "doc_count": 51,
                "key": "other and unspecified parts of biliary tract"
              }
            ]
          }
        },
        "pagination": {
          "count": 0,
          "total": 1202,
          "size": 0,
          "from": 0,
          "sort": "",
          "page": 1,
          "pages": 1202
        }
      },
      "warnings": {}
    }
    ```


## Alternative Request Format

The GDC API also supports POST requests with `Content-Type: application/x-www-form-urlencoded` (curl default), which require payloads in the following format:
```
	filters=%7B%0A%20%20%20%20%22op%22%3A%22in%22%2C%0A%20%20%20%20%22content%22%3A%7B%0A%20%20%20%20%20%20%20%20%22field%22%3A%22files.file_id%22%2C%0A%20%20%20%20%20%20%20%20%22value%22%3A%5B%0A%20%20%20%20%20%20%20%20%20%20%20%20%220001801b-54b0-4551-8d7a-d66fb59429bf%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%22002c67f2-ff52-4246-9d65-a3f69df6789e%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%22003143c8-bbbf-46b9-a96f-f58530f4bb82%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%220043d981-3c6b-463f-b512-ab1d076d3e62%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%22004e2a2c-1acc-4873-9379-ef1aa12283b6%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%22005239a8-2e63-4ff1-9cd4-714f81837a61%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%22006b8839-31e5-4697-b912-8e3f4124dd15%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%22006ce9a8-cf38-462e-bb99-7f08499244ab%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%22007ce9b5-3268-441e-9ffd-b40d1127a319%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%220084a614-780b-42ec-b85f-7a1b83128cd3%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%2200a5e471-a79f-4d56-8a4c-4847ac037400%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%2200ab2b5a-b59e-4ec9-b297-76f74ff1d3fb%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%2200c5f14e-a398-4076-95d1-25f320ee3a37%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%2200c74a8b-10aa-40cc-991e-3365ea1f3fce%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%2200df5a50-bce3-4edf-a078-641e54800dcb%22%0A%20%20%20%20%20%20%20%20%5D%0A%20%20%20%20%7D%0A%7D&fields=file_id,file_name,cases.submitter_id,cases.case_id,data_category,data_type,cases.samples.tumor_descriptor,cases.samples.tissue_type,cases.samples.sample_type,cases.samples.submitter_id,cases.samples.sample_id&format=tsv&size=100
```
## Using Wildcards

The GDC API supports the use of the wildcard character, an asterisk (\*), in the `value` fields of a JSON query.  For example, if a user wanted to retrieve information about projects with a disease type that ended in "Adenocarcinoma" a query for `"disease_type": "*Adenocarcinoma"` would be appropriate. See below:

```
{  
   "size":"20000",
   "pretty":"TRUE",
   "fields":"submitter_id,disease_type",
   "format":"TSV",
   "filters":{  
      "op":"=",
      "content":{  
         "field":"disease_type",
         "value":"*Adenocarcinoma"
      }
   }
}
```

## Quicksearch Endpoint

The GDC Portal has a quicksearch functionality that allows for a project, case, or file to be queried from a search box. This function calls the `/v0/all` endpoint, which retrieves the top cases, files, and projects that match to the query. The quicksearch can also be used programmatically through the API.  For example, a search term of 'TCGA' would produce the following query:  

=== "Shell"

    ```Shell
    curl "https://api.gdc.cancer.gov/v0/all?query=TCGA&size=5"
    ```

=== "Response"   

    ```json
    {"data":{"query":{"hits":[{"disease_type":["Adenomas and Adenocarcinomas"],"id":"UHJvamVjdDpUQ0dBLUFDQw==","name":"Adrenocortical Carcinoma","primary_site":["Adrenal gland"],"project_id":"TCGA-ACC","project_quicksearch":"Adrenocortical Carcinoma"},{"disease_type":["Adenomas and Adenocarcinomas"],"id":"UHJvamVjdDpUQ0dBLUtJQ0g=","name":"Kidney Chromophobe","primary_site":["Kidney"],"project_id":"TCGA-KICH","project_quicksearch":"Kidney Chromophobe"},{"disease_type":["Adenomas and Adenocarcinomas"],"id":"UHJvamVjdDpUQ0dBLUxJSEM=","name":"Liver Hepatocellular Carcinoma","primary_site":["Liver and intrahepatic bile ducts"],"project_id":"TCGA-LIHC","project_quicksearch":"Liver Hepatocellular Carcinoma"},{"disease_type":["Myeloid Leukemias"],"id":"UHJvamVjdDpUQ0dBLUxBTUw=","name":"Acute Myeloid Leukemia","primary_site":["Hematopoietic and reticuloendothelial systems"],"project_id":"TCGA-LAML","project_quicksearch":"Acute Myeloid Leukemia"},{"disease_type":["Adenomas and Adenocarcinomas"],"id":"UHJvamVjdDpUQ0dBLUtJUlA=","name":"Kidney Renal Papillary Cell Carcinoma","primary_site":["Kidney"],"project_id":"TCGA-KIRP","project_quicksearch":"Kidney Renal Papillary Cell Carcinoma"}],"total":183550}}}
    ```

This endpoint can be used to quickly retrieve information about a file.  For example, if a user wanted to know the UUID for `nationwidechildrens.org_biospecimen.TCGA-EL-A4K1.xml`, the following query could be used to quickly retrieve it programmatically:

=== "Shell"

    ```Shell
    curl "https://api.gdc.cancer.gov/v0/all?query=nationwidechildrens.org_biospecimen.TCGA-EL-A4K1.xml&size=5"
    ```

=== "Response"

    ```json
    {"data":{"query":{"hits":[{"file_id":"a74abfec-db78-4ed4-9e4b-604b66e30e30","file_name":"nationwidechildrens.org_biospecimen.TCGA-EL-A4K1.xml","id":"RmlsZTphNzRhYmZlYy1kYjc4LTRlZDQtOWU0Yi02MDRiNjZlMzBlMzA=","submitter_id":"nationwidechildrens.org_biospecimen.TCGA-EL-A4K1.xml"}],"total":1}}}
    ```

## Additional Examples

More examples of API functionality described in this section are provided in [Additional Examples](Additional_Examples.md).


# Downloading Files

The GDC API implements file download functionality using `data` and `manifest` endpoints. The `data` endpoint allows users to download files stored in the GDC by specifying file UUID(s). The `manifest` endpoint generates a download manifest file that can be used with the GDC Data Transfer Tool to transfer large volumes of data.

>**Note:** Downloading controlled access data requires the use of an authentication token. See [Getting Started: Authentication](Getting_Started.md#authentication) for details.

## Data endpoint

To download a file, users can pass UUID(s) to the `data` endpoint.  If a single UUID is provided, the API will return the associated file. If a comma-separated list of UUIDs is provided, the API will return an archive file containing the requested files.

The `data` endpoint supports GET and POST requests as demonstrated in the following examples.


### Downloading a Single File using GET

This example demonstrates downloading a single file from the GDC. Here we pass the file's UUID to the `data` endpoint with a GET request.

=== "Shell"

    ```shell
    curl --remote-name --remote-header-name 'https://api.gdc.cancer.gov/data/5b2974ad-f932-499b-90a3-93577a9f0573'
    ```

=== "Output"

    ```
    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                    Dload  Upload   Total   Spent    Left  Speed
    100 6111k  100 6111k    0     0   414k      0  0:00:14  0:00:14 --:--:--  412k
    ```

### Related Files

If the `related_files=true` parameter is specified, the following related files, if available, will be included in the download package by the GDC API:

* BAM index files (BAI files)
* VCF index files (TBI files)

For example, this request will download a BAM file and its associated BAI file:

=== "Shell"

    ```shell
    curl --remote-name --remote-header-name -H "x-auth-token: $token" "https://api.gdc.cancer.gov/data/f587ef82-acbe-44f9-ad5a-6207e148f61f?related_files=true"
    ```

=== "Output"

    ```
    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                Dload  Upload   Total   Spent    Left  Speed
    100 63.4M    0 63.4M    0     0  7541k      0 --:--:--  0:00:08 --:--:--  9.9M
    ```


### Downloading Multiple Files using GET

This example demonstrates downloading multiple files from the GDC using a GET request. The GDC API returns a `.tar.gz` archive containing the downloaded files.

=== "Shell"

    ```shell
    curl --remote-name --remote-header-name 'https://api.gdc.cancer.gov/data/e3228020-1c54-4521-9182-1ea14c5dc0f7,18e1e38e-0f0a-4a0e-918f-08e6201ea140'
    ```

=== "Output"

    ```
    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                   Dload  Upload   Total   Spent    Left  Speed
    100  287k    0  287k    0     0  30131      0 --:--:--  0:00:09 --:--:-- 42759
    
    ```

>**Note:** This method supports downloading a limited number of files at one time. To download a large number of files, please use [POST](#downloading-multiple-files-using-post).

#### Downloading an Uncompressed Group of Files

If the `?tarfile` parameter is specified to a data endpoint download query all files requested in the download string will be bundled in a single tar file rather than a tar.gz file which is the default behavior.  

=== "Shell"

    ```shell
    curl --remote-name --remote-header-name 'https://api.gdc.cancer.gov/data/1da7105a-f0ff-479d-9f82-6c1d94456c91,77e73cc4-ff31-449e-8e3c-7ae5ce57838c?tarfile'
    ```      

### Downloading Multiple Files using POST

The following two examples demonstrate downloading multiple files from the GDC using a POST request that contains a payload in one of two formats: percent-encoded form data or JSON. The GDC API returns a `.tar.gz` archive containing the downloaded files.

#### POST request with form data payload

POST requests that carry a payload of percent-encoded form data must include the HTTP header `Content-Type: application/x-www-form-urlencoded`.

The payload is a string in the following format:

	ids=UUID1&ids=UUID2&ids=UUID3...

where UUID# corresponds to the UUIDs of the files to be downloaded.

In this example we use `curl` to download a set of files from the GDC Data Portal. The payload is stored in a plain text file named `Payload`; `curl` includes the `Content-Type: application/x-www-form-urlencoded` header by default.

=== "Payload"

    ```
    ids=59eb3fc5-9172-4828-8dec-0d9988073103&ids=869b7d7c-ff35-482a-aa8d-1a8675c161d3&ids=b8ffff40-aa0e-4534-b05f-9311f16c2f6b&ids=51e14969-30a7-42d9-8168-4a5ea422ca4a&ids=adcfc856-990b-40fc-8f1e-67dfc2343fb7&ids=7f1e9aee-eb4e-4c79-8626-b603c9be124d&ids=62a8feb5-c660-4261-bcd6-67fbb79bb422
    ```

=== "Shell"

    ```Shell
    curl --remote-name --remote-header-name --request POST 'https://api.gdc.cancer.gov/data' --data @Payload
    ```

=== "Output"

    ```
    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                    Dload  Upload   Total   Spent    Left  Speed
    100 6804k    0 6804k  100   286   245k     10  0:00:28  0:00:27  0:00:01  357k
    ```

#### POST request with JSON payload

POST requests that carry a JSON payload must include the HTTP header `Content-Type: application/json`.

The payload is a string in the following format:

	{
	    "ids":[
	        "UUID1",
	        "UUID2",
					...
	        "UUID3"
	    ]
	}


where UUID# corresponds to the UUIDs of the files to be downloaded.

In this example we use `curl` to download a set of files from the GDC Portal; the payload is stored in a plain text file named `Payload`.

=== "Payload"

    ```json
    {
        "ids":[
            "0451fc55-33ef-4151-a68c-cac59be716dc",
            "0cc3d450-2c60-4cb0-a073-d92dc979fa5e",
            "0de9bc40-3ef8-4fe7-b7d6-80a9339b0bf8",
            "0f8d8202-a1ca-4ea1-98b2-c20a6b08479a"
        ]
    }
    ```

=== "Shell"

    ```Shell
    curl --remote-name --remote-header-name --request POST --header 'Content-Type: application/json' --data @request.txt 'https://api.gdc.cancer.gov/data'
    ```

=== "Output"

    ```
    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                    Dload  Upload   Total   Spent    Left  Speed
    100 5878k    0 5878k  100   205   290k     10  0:00:20  0:00:20 --:--:--  198k
    ```

### Downloading Controlled-access Files

To download controlled-access files, a valid authentication token must be passed to the GDC API using the `X-Auth-Token` HTTP header:

=== "Shell"

    ```shell
    token=$(<gdc-token-text-file.txt)

    curl --remote-name --remote-header-name --header "X-Auth-Token: $token" 'https://api.gdc.cancer.gov/data/be6d269d-4305-4643-b98e-af703a067761'
    ```

=== "Output"

    ```
    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
    100 65.8M  100 65.8M    0     0   271k      0  0:04:08  0:04:08 --:--:--  288k
    ```

## Manifest endpoint

The `manifest` endpoint generates a download manifest file that can be used with the [GDC Data Transfer Tool](../../Data_Transfer_Tool/Users_Guide/Getting_Started.md). The Data Transfer Tool is recommended for transferring large volumes of data. The GDC API can also generate a download manifest from a list of results that match a [Search and Retrieval](Search_and_Retrieval.md) query. To do this, append `&return_type=manifest` to the end of the query. Note that the "size" parameter does not work with the manifest endpoint and will return the entire set of files.


### Using the manifest endpoint

The `manifest` endpoint allows users to create a download manifest, which can be used with the GDC Data Transfer Tool to download a large volume of data. The `manifest` endpoint generates a manifest file from a comma-separated list of UUIDs.

=== "Shell"

    ```shell
    curl --remote-name --remote-header-name 'https://api.gdc.cancer.gov/v0/manifest/a751cc7e-d2ff-4e9a-8645-09bf12612f1a,9c97e3fe-1610-4a92-9a24-ab3b9e4000e2'
    ```

=== "Output"

    ```
      % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                     Dload  Upload   Total   Spent    Left  Speed
    100   274  100   274    0     0   1042      0 --:--:-- --:--:-- --:--:--  1041

    ```

The `manifest` endpoint also supports HTTP POST requests in the same format as the `data` endpoint; see [above](#post-request-with-json-payload) for details.


### Using return_type=manifest

Alternatively, users can create a manifest by appending `&return_type=manifest` to a [Search and Retrieval](Search_and_Retrieval.md) query. In this example, we generate a download manifest for RNA-seq data files from solid tissue normal samples, that are part of the TCGA-KIRC project:

=== "Shell"

    ```Shell
    curl --remote-name --remote-header-name 'https://api.gdc.cancer.gov/files?filters=%7B%22op%22%3A%22and%22%2C%22content%22%3A%5B%7B%22op%22%3A%22%3D%22%2C%22content%22%3A%7B%22field%22%3A%22experimental_strategy%22%2C%22value%22%3A%5B%22RNA-Seq%22%5D%7D%7D%2C%7B%22op%22%3A%22%3D%22%2C%22content%22%3A%7B%22field%22%3A%22cases.project.project_id%22%2C%22value%22%3A%5B%22TCGA-KIRC%22%5D%7D%7D%2C%7B%22op%22%3A%22%3D%22%2C%22content%22%3A%7B%22field%22%3A%22cases.samples.sample_type%22%2C%22value%22%3A%5B%22Solid+Tissue+Normal%22%5D%7D%7D%5D%7D&return_type=manifest'
    ```

=== "Output"

    ```
    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                    Dload  Upload   Total   Spent    Left  Speed
    100  100k    0  100k    0     0   277k      0 --:--:-- --:--:-- --:--:--  282k
    ```


# Appendix A: Available Fields

The GDC API's [search and retrieval endpoints](Search_and_Retrieval.md) provide access to fields that correspond to properties defined in the [GDC Data Dictionary](../../Data_Dictionary/viewer.md). This appendix contains a list of fields available at each endpoint, and a list of field groups accessible via the [expand parameter](Search_and_Retrieval.md#expand).

## Field Listing by Endpoint

### Project Fields

| Field Name |
| --- | -------- |
| dbgap_accession_number |
| disease_type |
| name |  
| primary_site |
| project_id |
| released |
| state |
| program.dbgap_accession_number |
| program.name |  
| program.program_id |
| summary.case_count |
| summary.file_count |
| summary.file_size |
| summary.data_categories.case_count |
| summary.data_categories.data_category |
| summary.data_categories.file_count |
| summary.experimental_strategies.case_count |  
| summary.experimental_strategies.experimental_strategy |
| summary.experimental_strategies.file_count |



### Case Fields


| Field Name |
| --- | -------- |
| aliquot_ids |
| analyte_ids |
| case_id |
| created_datetime |  
| days_to_index |  
| portion_ids |
| sample_ids |
| slide_ids |
| state |
| submitter_aliquot_ids |
| submitter_analyte_ids |
| submitter_id |
| submitter_portion_ids |
| submitter_sample_ids |
| submitter_slide_ids |
| updated_datetime |  
| annotations.annotation_id |
| annotations.case_id |
| annotations.case_submitter_id |
| annotations.category |
| annotations.classification |
| annotations.created_datetime |
| annotations.creator |
| annotations.entity_id |
| annotations.entity_submitter_id |
| annotations.entity_type |
| annotations.legacy_created_datetime |
| annotations.legacy_updated_datetime |
| annotations.notes |
| annotations.state |
| annotations.status |
| annotations.submitter_id |
| annotations.updated_datetime |
| demographic.created_datetime |
| demographic.demographic_id |
| demographic.ethnicity |
| demographic.gender |
| demographic.race |
| demographic.state |
| demographic.submitter_id |
| demographic.updated_datetime |
| demographic.year_of_birth |
| demographic.year_of_death |
| diagnoses.age_at_diagnosis |
| diagnoses.classification_of_tumor |
| diagnoses.created_datetime |
| diagnoses.days_to_birth |
| diagnoses.days_to_death |
| diagnoses.days_to_last_follow_up |
| diagnoses.days_to_last_known_disease_status |
| diagnoses.days_to_recurrence |
| diagnoses.diagnosis_id |
| diagnoses.last_known_disease_status |
| diagnoses.morphology |
| diagnoses.primary_diagnosis |
| diagnoses.prior_malignancy |
| diagnoses.progression_or_recurrence |
| diagnoses.site_of_resection_or_biopsy |
| diagnoses.state |
| diagnoses.submitter_id |
| diagnoses.tissue_or_organ_of_origin |
| diagnoses.tumor_grade |
| diagnoses.tumor_stage |
| diagnoses.updated_datetime |
| diagnoses.vital_status |
| diagnoses.treatments.created_datetime |
| diagnoses.treatments.days_to_treatment |
| diagnoses.treatments.state |
| diagnoses.treatments.submitter_id |
| diagnoses.treatments.therapeutic_agents |
| diagnoses.treatments.treatment_id |
| diagnoses.treatments.treatment_intent_type |
| diagnoses.treatments.treatment_or_therapy |
| diagnoses.treatments.updated_datetime |
| exposures.alcohol_history |
| exposures.alcohol_intensity |
| exposures.bmi |
| exposures.cigarettes_per_day |
| exposures.created_datetime |
| exposures.exposure_id |
| exposures.height |
| exposures.state |
| exposures.submitter_id |
| exposures.updated_datetime |
| exposures.weight |
| exposures.years_smoked |
| family_histories.created_datetime |
| family_histories.family_history_id |  
| family_histories.relationship_age_at_diagnosis |
| family_histories.relationship_gender |
| family_histories.relationship_primary_diagnosis |
| family_histories.relationship_type |
| family_histories.relative_with_cancer_history |
| family_histories.state |
| family_histories.submitter_id |
| family_histories.updated_datetime |
| files.access |
| files.acl |
| files.created_datetime |
| files.data_category |
| files.data_format |
| files.data_type |
| files.error_type |
| files.experimental_strategy |
| files.file_id |
| files.file_name |
| files.file_size |
| files.file_state |
| files.md5sum |
| files.origin |
| files.platform |
| files.revision |
| files.state |
| files.state_comment |
| files.submitter_id |
| files.tags |
| files.type |
| files.updated_datetime |
| files.analysis.analysis_id |
| files.analysis.analysis_type |
| files.analysis.created_datetime |
| files.analysis.state |
| files.analysis.submitter_id |
| files.analysis.updated_datetime |
| files.analysis.workflow_end_datetime |
| files.analysis.workflow_link |
| files.analysis.workflow_start_datetime |
| files.analysis.workflow_type |
| files.analysis.workflow_version |
| files.analysis.input_files.access |
| files.analysis.input_files.created_datetime |
| files.analysis.input_files.data_category |
| files.analysis.input_files.data_format |
| files.analysis.input_files.data_type |
| files.analysis.input_files.error_type |
| files.analysis.input_files.experimental_strategy |
| files.analysis.input_files.file_id |
| files.analysis.input_files.file_name |
| files.analysis.input_files.file_size |
| files.analysis.input_files.file_state |
| files.analysis.input_files.md5sum |
| files.analysis.input_files.platform |
| files.analysis.input_files.revision |
| files.analysis.input_files.state |
| files.analysis.input_files.state_comment |
| files.analysis.input_files.submitter_id |
| files.analysis.input_files.updated_datetime |
| files.analysis.metadata.read_groups.adapter_name |
| files.analysis.metadata.read_groups.adapter_sequence |
| files.analysis.metadata.read_groups.base_caller_name |
| files.analysis.metadata.read_groups.base_caller_version |
| files.analysis.metadata.read_groups.created_datetime |
| files.analysis.metadata.read_groups.experiment_name |
| files.analysis.metadata.read_groups.flow_cell_barcode |
| files.analysis.metadata.read_groups.includes_spike_ins |
| files.analysis.metadata.read_groups.instrument_model |
| files.analysis.metadata.read_groups.is_paired_end |
| files.analysis.metadata.read_groups.library_name |
| files.analysis.metadata.read_groups.library_preparation_kit_catalog_number |
| files.analysis.metadata.read_groups.library_preparation_kit_name |
| files.analysis.metadata.read_groups.library_preparation_kit_vendor |
| files.analysis.metadata.read_groups.library_preparation_kit_version |
| files.analysis.metadata.read_groups.library_selection |
| files.analysis.metadata.read_groups.library_strand |
| files.analysis.metadata.read_groups.library_strategy |
| files.analysis.metadata.read_groups.platform |
| files.analysis.metadata.read_groups.read_group_id |
| files.analysis.metadata.read_groups.read_group_name |
| files.analysis.metadata.read_groups.read_length |
| files.analysis.metadata.read_groups.RIN |
| files.analysis.metadata.read_groups.sequencing_center |
| files.analysis.metadata.read_groups.sequencing_date |
| files.analysis.metadata.read_groups.size_selection_range |
| files.analysis.metadata.read_groups.spike_ins_concentration |
| files.analysis.metadata.read_groups.spike_ins_fasta |
| files.analysis.metadata.read_groups.state |
| files.analysis.metadata.read_groups.submitter_id |
| files.analysis.metadata.read_groups.target_capture_kit_catalog_number |
| files.analysis.metadata.read_groups.target_capture_kit_name |
| files.analysis.metadata.read_groups.target_capture_kit_target_region |
| files.analysis.metadata.read_groups.target_capture_kit_vendor |
| files.analysis.metadata.read_groups.target_capture_kit_version |
| files.analysis.metadata.read_groups.to_trim_adapter_sequence |
| files.analysis.metadata.read_groups.updated_datetime |
| files.analysis.metadata.read_groups.read_group_qcs.adapter_content |
| files.analysis.metadata.read_groups.read_group_qcs.basic_statistics |
| files.analysis.metadata.read_groups.read_group_qcs.created_datetime |
| files.analysis.metadata.read_groups.read_group_qcs.encoding |
| files.analysis.metadata.read_groups.read_group_qcs.fastq_name |
| files.analysis.metadata.read_groups.read_group_qcs.kmer_content |
| files.analysis.metadata.read_groups.read_group_qcs.overrepresented_sequences |
| files.analysis.metadata.read_groups.read_group_qcs.per_base_n_content |
| files.analysis.metadata.read_groups.read_group_qcs.per_base_sequence_content |
| files.analysis.metadata.read_groups.read_group_qcs.per_base_sequence_quality |
| files.analysis.metadata.read_groups.read_group_qcs.per_sequence_gc_content |
| files.analysis.metadata.read_groups.read_group_qcs.per_sequence_quality_score |
| files.analysis.metadata.read_groups.read_group_qcs.per_tile_sequence_quality |
| files.analysis.metadata.read_groups.read_group_qcs.percent_gc_content |
| files.analysis.metadata.read_groups.read_group_qcs.read_group_qc_id |
| files.analysis.metadata.read_groups.read_group_qcs.sequence_duplication_levels |
| files.analysis.metadata.read_groups.read_group_qcs.sequence_length_distribution |
| files.analysis.metadata.read_groups.read_group_qcs.state |
| files.analysis.metadata.read_groups.read_group_qcs.submitter_id |
| files.analysis.metadata.read_groups.read_group_qcs.total_sequences |
| files.analysis.metadata.read_groups.read_group_qcs.updated_datetime |
| files.analysis.metadata.read_groups.read_group_qcs.workflow_end_datetime |
| files.analysis.metadata.read_groups.read_group_qcs.workflow_link |
| files.analysis.metadata.read_groups.read_group_qcs.workflow_start_datetime |
| files.analysis.metadata.read_groups.read_group_qcs.workflow_type |
| files.analysis.metadata.read_groups.read_group_qcs.workflow_version |
| files.archive.archive_id |
| files.archive.created_datetime |
| files.archive.data_category |
| files.archive.data_format |
| files.archive.data_type |
| files.archive.error_type |
| files.archive.file_name |
| files.archive.file_size |
| files.archive.file_state |
| files.archive.md5sum |
| files.archive.revision |
| files.archive.state |
| files.archive.state_comment |
| files.archive.submitter_id |
| files.archive.updated_datetime |
| files.cases.aliquot_ids |
| files.cases.analyte_ids |
| files.cases.case_id |
| files.cases.created_datetime |
| files.cases.days_to_index |
| files.cases.portion_ids |
| files.cases.sample_ids |
| files.cases.slide_ids |
| files.cases.state |
| files.cases.submitter_aliquot_ids |
| files.cases.submitter_analyte_ids |
| files.cases.submitter_id |
| files.cases.submitter_portion_ids |
| files.cases.submitter_sample_ids |
| files.cases.submitter_slide_ids |
| files.cases.updated_datetime |
| files.cases.annotations.annotation_id |
| files.cases.annotations.case_id |
| files.cases.annotations.case_submitter_id |
| files.cases.annotations.category |
| files.cases.annotations.classification |
| files.cases.annotations.created_datetime |
| files.cases.annotations.creator |
| files.cases.annotations.entity_id |
| files.cases.annotations.entity_submitter_id |
| files.cases.annotations.entity_type |
| files.cases.annotations.legacy_created_datetime |
| files.cases.annotations.legacy_updated_datetime |
| files.cases.annotations.notes |
| files.cases.annotations.state |
| files.cases.annotations.status |
| files.cases.annotations.submitter_id |
| files.cases.annotations.updated_datetime |
| files.cases.demographic.created_datetime |
| files.cases.demographic.demographic_id |
| files.cases.demographic.ethnicity |
| files.cases.demographic.gender |
| files.cases.demographic.race |
| files.cases.demographic.state |
| files.cases.demographic.submitter_id |
| files.cases.demographic.updated_datetime |
| files.cases.demographic.year_of_birth |
| files.cases.demographic.year_of_death |
| files.cases.diagnoses.age_at_diagnosis |
| files.cases.diagnoses.classification_of_tumor |
| files.cases.diagnoses.created_datetime |
| files.cases.diagnoses.days_to_birth |
| files.cases.diagnoses.days_to_death |
| files.cases.diagnoses.days_to_last_follow_up |
| files.cases.diagnoses.days_to_last_known_disease_status |
| files.cases.diagnoses.days_to_recurrence |
| files.cases.diagnoses.diagnosis_id |
| files.cases.diagnoses.last_known_disease_status |
| files.cases.diagnoses.morphology |
| files.cases.diagnoses.primary_diagnosis |
| files.cases.diagnoses.prior_malignancy |
| files.cases.diagnoses.progression_or_recurrence |
| files.cases.diagnoses.site_of_resection_or_biopsy |
| files.cases.diagnoses.state |
| files.cases.diagnoses.submitter_id |
| files.cases.diagnoses.tissue_or_organ_of_origin |
| files.cases.diagnoses.tumor_grade |
| files.cases.diagnoses.tumor_stage |
| files.cases.diagnoses.updated_datetime |
| files.cases.diagnoses.vital_status |
| files.cases.diagnoses.treatments.created_datetime |
| files.cases.diagnoses.treatments.days_to_treatment |
| files.cases.diagnoses.treatments.state |
| files.cases.diagnoses.treatments.submitter_id |
| files.cases.diagnoses.treatments.therapeutic_agents |
| files.cases.diagnoses.treatments.treatment_id |
| files.cases.diagnoses.treatments.treatment_intent_type |
| files.cases.diagnoses.treatments.treatment_or_therapy |
| files.cases.diagnoses.treatments.updated_datetime |
| files.cases.exposures.alcohol_history |
| files.cases.exposures.alcohol_intensity |
| files.cases.exposures.bmi |
| files.cases.exposures.cigarettes_per_day |
| files.cases.exposures.created_datetime |
| files.cases.exposures.exposure_id |
| files.cases.exposures.height |
| files.cases.exposures.state |
| files.cases.exposures.submitter_id |
| files.cases.exposures.updated_datetime |
| files.cases.exposures.weight |
| files.cases.exposures.years_smoked |
| files.cases.family_histories.created_datetime |
| files.cases.family_histories.family_history_id |
| files.cases.family_histories.relationship_age_at_diagnosis |
| files.cases.family_histories.relationship_gender |
| files.cases.family_histories.relationship_primary_diagnosis |
| files.cases.family_histories.relationship_type |
| files.cases.family_histories.relative_with_cancer_history |
| files.cases.family_histories.state |
| files.cases.family_histories.submitter_id |
| files.cases.family_histories.updated_datetime |
| files.cases.files.created_datetime |
| files.cases.files.error_type |
| files.cases.files.file_id |
| files.cases.files.file_name |
| files.cases.files.file_size |
| files.cases.files.file_state |
| files.cases.files.md5sum |
| files.cases.files.state |
| files.cases.files.state_comment |
| files.cases.files.submitter_id |
| files.cases.files.updated_datetime |
| files.cases.project.dbgap_accession_number |
| files.cases.project.disease_type |
| files.cases.project.name |
| files.cases.project.primary_site |
| files.cases.project.project_id |
| files.cases.project.released |
| files.cases.project.state |
| files.cases.project.program.dbgap_accession_number |
| files.cases.project.program.name |
| files.cases.project.program.program_id |
| files.cases.samples.composition |
| files.cases.samples.created_datetime |
| files.cases.samples.current_weight |
| files.cases.samples.days_to_collection |
| files.cases.samples.days_to_sample_procurement |
| files.cases.samples.freezing_method |
| files.cases.samples.initial_weight |
| files.cases.samples.intermediate_dimension |
| files.cases.samples.is_ffpe |
| files.cases.samples.longest_dimension |
| files.cases.samples.oct_embedded |
| files.cases.samples.pathology_report_uuid |
| files.cases.samples.preservation_method |
| files.cases.samples.sample_id |
| files.cases.samples.sample_type |
| files.cases.samples.sample_type_id |
| files.cases.samples.shortest_dimension |
| files.cases.samples.state |
| files.cases.samples.submitter_id |
| files.cases.samples.time_between_clamping_and_freezing |
| files.cases.samples.time_between_excision_and_freezing |
| files.cases.samples.tissue_type |
| files.cases.samples.tumor_code |
| files.cases.samples.tumor_code_id |
| files.cases.samples.tumor_descriptor |
| files.cases.samples.updated_datetime |
| files.cases.samples.annotations.annotation_id |
| files.cases.samples.annotations.case_id |
| files.cases.samples.annotations.case_submitter_id |
| files.cases.samples.annotations.category |
| files.cases.samples.annotations.classification |
| files.cases.samples.annotations.created_datetime |
| files.cases.samples.annotations.creator |
| files.cases.samples.annotations.entity_id |
| files.cases.samples.annotations.entity_submitter_id |
| files.cases.samples.annotations.entity_type |
| files.cases.samples.annotations.legacy_created_datetime |
| files.cases.samples.annotations.legacy_updated_datetime |
| files.cases.samples.annotations.notes |
| files.cases.samples.annotations.state |
| files.cases.samples.annotations.status |
| files.cases.samples.annotations.submitter_id |
| files.cases.samples.annotations.updated_datetime |
| files.cases.samples.portions.created_datetime |
| files.cases.samples.portions.creation_datetime |
| files.cases.samples.portions.is_ffpe |
| files.cases.samples.portions.portion_id |
| files.cases.samples.portions.portion_number |
| files.cases.samples.portions.state |  
| files.cases.samples.portions.submitter_id |  
| files.cases.samples.portions.updated_datetime |
| files.cases.samples.portions.weight |
| files.cases.samples.portions.analytes.a260_a280_ratio |
| files.cases.samples.portions.analytes.amount |
| files.cases.samples.portions.analytes.analyte_id |
| files.cases.samples.portions.analytes.analyte_type |
| files.cases.samples.portions.analytes.analyte_type_id |
| files.cases.samples.portions.analytes.concentration |
| files.cases.samples.portions.analytes.created_datetime |
| files.cases.samples.portions.analytes.spectrophotometer_method |
| files.cases.samples.portions.analytes.state |
| files.cases.samples.portions.analytes.submitter_id |
| files.cases.samples.portions.analytes.updated_datetime |
| files.cases.samples.portions.analytes.well_number |
| files.cases.samples.portions.analytes.aliquots.aliquot_id |
| files.cases.samples.portions.analytes.aliquots.amount |
| files.cases.samples.portions.analytes.aliquots.analyte_type |
| files.cases.samples.portions.analytes.aliquots.analyte_type_id |
| files.cases.samples.portions.analytes.aliquots.concentration |
| files.cases.samples.portions.analytes.aliquots.created_datetime |
| files.cases.samples.portions.analytes.aliquots.source_center |
| files.cases.samples.portions.analytes.aliquots.state |
| files.cases.samples.portions.analytes.aliquots.submitter_id |
| files.cases.samples.portions.analytes.aliquots.updated_datetime |
| files.cases.samples.portions.analytes.aliquots.annotations.annotation_id |
| files.cases.samples.portions.analytes.aliquots.annotations.case_id |
| files.cases.samples.portions.analytes.aliquots.annotations.case_submitter_id |
| files.cases.samples.portions.analytes.aliquots.annotations.category |
| files.cases.samples.portions.analytes.aliquots.annotations.classification |
| files.cases.samples.portions.analytes.aliquots.annotations.created_datetime |
| files.cases.samples.portions.analytes.aliquots.annotations.creator |
| files.cases.samples.portions.analytes.aliquots.annotations.entity_id |
| files.cases.samples.portions.analytes.aliquots.annotations.entity_submitter_id |
| files.cases.samples.portions.analytes.aliquots.annotations.entity_type |
| files.cases.samples.portions.analytes.aliquots.annotations.legacy_created_datetime |
| files.cases.samples.portions.analytes.aliquots.annotations.legacy_updated_datetime |
| files.cases.samples.portions.analytes.aliquots.annotations.notes |
| files.cases.samples.portions.analytes.aliquots.annotations.state |
| files.cases.samples.portions.analytes.aliquots.annotations.status |
| files.cases.samples.portions.analytes.aliquots.annotations.submitter_id |
| files.cases.samples.portions.analytes.aliquots.annotations.updated_datetime |
| files.cases.samples.portions.analytes.aliquots.center.center_id |
| files.cases.samples.portions.analytes.aliquots.center.center_type |
| files.cases.samples.portions.analytes.aliquots.center.code |  
| files.cases.samples.portions.analytes.aliquots.center.name |  
| files.cases.samples.portions.analytes.aliquots.center.namespace |  
| files.cases.samples.portions.analytes.aliquots.center.short_name |
| files.cases.samples.portions.analytes.annotations.annotation_id |
| files.cases.samples.portions.analytes.annotations.case_id |
| files.cases.samples.portions.analytes.annotations.case_submitter_id |
| files.cases.samples.portions.analytes.annotations.category |
| files.cases.samples.portions.analytes.annotations.classification |
| files.cases.samples.portions.analytes.annotations.created_datetime |
| files.cases.samples.portions.analytes.annotations.creator |
| files.cases.samples.portions.analytes.annotations.entity_id |
| files.cases.samples.portions.analytes.annotations.entity_submitter_id |
| files.cases.samples.portions.analytes.annotations.entity_type |
| files.cases.samples.portions.analytes.annotations.legacy_created_datetime |
| files.cases.samples.portions.analytes.annotations.legacy_updated_datetime |
| files.cases.samples.portions.analytes.annotations.notes |
| files.cases.samples.portions.analytes.annotations.state |
| files.cases.samples.portions.analytes.annotations.status |
| files.cases.samples.portions.analytes.annotations.submitter_id |
| files.cases.samples.portions.analytes.annotations.updated_datetime |
| files.cases.samples.portions.annotations.annotation_id |
| files.cases.samples.portions.annotations.case_id |
| files.cases.samples.portions.annotations.case_submitter_id |
| files.cases.samples.portions.annotations.category |
| files.cases.samples.portions.annotations.classification |
| files.cases.samples.portions.annotations.created_datetime |
| files.cases.samples.portions.annotations.creator |
| files.cases.samples.portions.annotations.entity_id |
| files.cases.samples.portions.annotations.entity_submitter_id |
| files.cases.samples.portions.annotations.entity_type |
| files.cases.samples.portions.annotations.legacy_created_datetime |
| files.cases.samples.portions.annotations.legacy_updated_datetime |
| files.cases.samples.portions.annotations.notes |
| files.cases.samples.portions.annotations.state |
| files.cases.samples.portions.annotations.status |
| files.cases.samples.portions.annotations.submitter_id |
| files.cases.samples.portions.annotations.updated_datetime |
| files.cases.samples.portions.center.center_id |
| files.cases.samples.portions.center.center_type |
| files.cases.samples.portions.center.code |
| files.cases.samples.portions.center.name |
| files.cases.samples.portions.center.namespace |
| files.cases.samples.portions.center.short_name |
| files.cases.samples.portions.slides.created_datetime |
| files.cases.samples.portions.slides.number_proliferating_cells |
| files.cases.samples.portions.slides.percent_eosinophil_infiltration |
| files.cases.samples.portions.slides.percent_granulocyte_infiltration |
| files.cases.samples.portions.slides.percent_inflam_infiltration |
| files.cases.samples.portions.slides.percent_lymphocyte_infiltration |
| files.cases.samples.portions.slides.percent_monocyte_infiltration |
| files.cases.samples.portions.slides.percent_necrosis |
| files.cases.samples.portions.slides.percent_neutrophil_infiltration |
| files.cases.samples.portions.slides.percent_normal_cells |
| files.cases.samples.portions.slides.percent_stromal_cells |
| files.cases.samples.portions.slides.percent_tumor_cells |
| files.cases.samples.portions.slides.percent_tumor_nuclei |
| files.cases.samples.portions.slides.section_location |
| files.cases.samples.portions.slides.slide_id |
| files.cases.samples.portions.slides.state |
| files.cases.samples.portions.slides.submitter_id |
| files.cases.samples.portions.slides.updated_datetime |
| files.cases.samples.portions.slides.annotations.annotation_id |
| files.cases.samples.portions.slides.annotations.case_id |
| files.cases.samples.portions.slides.annotations.case_submitter_id |
| files.cases.samples.portions.slides.annotations.category |
| files.cases.samples.portions.slides.annotations.classification |
| files.cases.samples.portions.slides.annotations.created_datetime |
| files.cases.samples.portions.slides.annotations.creator |
| files.cases.samples.portions.slides.annotations.entity_id |
| files.cases.samples.portions.slides.annotations.entity_submitter_id |
| files.cases.samples.portions.slides.annotations.entity_type |
| files.cases.samples.portions.slides.annotations.legacy_created_datetime |
| files.cases.samples.portions.slides.annotations.legacy_updated_datetime |
| files.cases.samples.portions.slides.annotations.notes |
| files.cases.samples.portions.slides.annotations.state |
| files.cases.samples.portions.slides.annotations.status |
| files.cases.samples.portions.slides.annotations.submitter_id |
| files.cases.samples.portions.slides.annotations.updated_datetime |
| files.cases.summary.file_count |
| files.cases.summary.file_size |
| files.cases.summary.data_categories.data_category |
| files.cases.summary.data_categories.file_count |
| files.cases.summary.experimental_strategies.experimental_strategy |
| files.cases.summary.experimental_strategies.file_count |
| files.cases.tissue_source_site.bcr_id |
| files.cases.tissue_source_site.code |
| files.cases.tissue_source_site.name |
| files.cases.tissue_source_site.project |
| files.cases.tissue_source_site.tissue_source_site_id |
| files.center.center_id |
| files.center.center_type |
| files.center.code |
| files.center.name |
| files.center.namespace |
| files.center.short_name |
| files.downstream_analyses.analysis_id |
| files.downstream_analyses.analysis_type |
| files.downstream_analyses.created_datetime |
| files.downstream_analyses.state |
| files.downstream_analyses.submitter_id |
| files.downstream_analyses.updated_datetime |
| files.downstream_analyses.workflow_end_datetime |
| files.downstream_analyses.workflow_link |
| files.downstream_analyses.workflow_start_datetime |
| files.downstream_analyses.workflow_type |
| files.downstream_analyses.workflow_version |
| files.downstream_analyses.output_files.access |
| files.downstream_analyses.output_files.created_datetime |
| files.downstream_analyses.output_files.data_category |
| files.downstream_analyses.output_files.data_format |
| files.downstream_analyses.output_files.data_type |
| files.downstream_analyses.output_files.error_type |
| files.downstream_analyses.output_files.experimental_strategy |
| files.downstream_analyses.output_files.file_id |
| files.downstream_analyses.output_files.file_name |
| files.downstream_analyses.output_files.file_size |
| files.downstream_analyses.output_files.file_state |
| files.downstream_analyses.output_files.md5sum |
| files.downstream_analyses.output_files.platform |
| files.downstream_analyses.output_files.revision |
| files.downstream_analyses.output_files.state |
| files.downstream_analyses.output_files.state_comment |
| files.downstream_analyses.output_files.submitter_id |
| files.downstream_analyses.output_files.updated_datetime |
| files.index_files.access |
| files.index_files.created_datetime |
| files.index_files.data_category |
| files.index_files.data_format |
| files.index_files.data_type |
| files.index_files.error_type |
| files.index_files.experimental_strategy |
| files.index_files.file_id |
| files.index_files.file_name |
| files.index_files.file_size |
| files.index_files.file_state |
| files.index_files.md5sum |
| files.index_files.platform |
| files.index_files.revision |
| files.index_files.state |
| files.index_files.state_comment |
| files.index_files.submitter_id |
| files.index_files.updated_datetime |
| files.metadata_files.access |
| files.metadata_files.created_datetime |
| files.metadata_files.data_category |
| files.metadata_files.data_format |
| files.metadata_files.data_type |
| files.metadata_files.error_type |
| files.metadata_files.file_id |
| files.metadata_files.file_name |
| files.metadata_files.file_size |
| files.metadata_files.file_state |
| files.metadata_files.md5sum |
| files.metadata_files.state |
| files.metadata_files.state_comment |
| files.metadata_files.submitter_id |
| files.metadata_files.type |
| files.metadata_files.updated_datetime |
| project.dbgap_accession_number |
| project.disease_type |
| project.name |
| project.primary_site |
| project.project_id |
| project.released |
| project.state |
| project.program.dbgap_accession_number |
| project.program.name |
| project.program.program_id |
| samples.composition |
| samples.created_datetime |
| samples.current_weight |
| samples.days_to_collection |
| samples.days_to_sample_procurement |
| samples.freezing_method |
| samples.initial_weight |
| samples.intermediate_dimension |
| samples.is_ffpe |
| samples.longest_dimension |
| samples.oct_embedded |
| samples.pathology_report_uuid |
| samples.preservation_method |
| samples.sample_id |
| samples.sample_type |
| samples.sample_type_id |
| samples.shortest_dimension |
| samples.state |
| samples.submitter_id |
| samples.time_between_clamping_and_freezing |
| samples.time_between_excision_and_freezing |
| samples.tissue_type |
| samples.tumor_code |
| samples.tumor_code_id |
| samples.tumor_descriptor |
| samples.updated_datetime |
| samples.annotations.annotation_id |
| samples.annotations.case_id |
| samples.annotations.case_submitter_id |
| samples.annotations.category |
| samples.annotations.classification |
| samples.annotations.created_datetime |
| samples.annotations.creator |
| samples.annotations.entity_id |
| samples.annotations.entity_submitter_id |
| samples.annotations.entity_type |
| samples.annotations.legacy_creat
| samples.annotations.legacy_updated_datetime |
| samples.annotations.notes |
| samples.annotations.state |
| samples.annotations.status |
| samples.annotations.submitter_id |
| samples.annotations.updated_datetime |
| samples.portions.created_datetime |
| samples.portions.creation_datetime |
| samples.portions.is_ffpe |
| samples.portions.portion_id |
| samples.portions.portion_number |
| samples.portions.state |
| samples.portions.submitter_id |
| samples.portions.updated_datetime |
| samples.portions.weight |
| samples.portions.analytes.a260_a280_ratio |
| samples.portions.analytes.amount |
| samples.portions.analytes.analyte_id |
| samples.portions.analytes.analyte_type |
| samples.portions.analytes.analyte_type_id |
| samples.portions.analytes.concentration |
| samples.portions.analytes.created_datetime |
| samples.portions.analytes.spectrophotometer_method |
| samples.portions.analytes.state |
| samples.portions.analytes.submitter_id |
| samples.portions.analytes.updated_datetime |
| samples.portions.analytes.well_number |
| samples.portions.analytes.aliquots.aliquot_id |
| samples.portions.analytes.aliquots.amount |
| samples.portions.analytes.aliquots.analyte_type |
| samples.portions.analytes.aliquots.analyte_type_id |
| samples.portions.analytes.aliquots.concentration |
| samples.portions.analytes.aliquots.created_datetime |
| samples.portions.analytes.aliquots.source_center |
| samples.portions.analytes.aliquots.state |
| samples.portions.analytes.aliquots.submitter_id |
| samples.portions.analytes.aliquots.updated_datetime |
| samples.portions.analytes.aliquots.annotations.annotation_id |
| samples.portions.analytes.aliquots.annotations.case_id |
| samples.portions.analytes.aliquots.annotations.case_submitter_id |
| samples.portions.analytes.aliquots.annotations.category |
| samples.portions.analytes.aliquots.annotations.classification |
| samples.portions.analytes.aliquots.annotations.created_datetime |
| samples.portions.analytes.aliquots.annotations.creator |
| samples.portions.analytes.aliquots.annotations.entity_id |
| samples.portions.analytes.aliquots.annotations.entity_submitter_id |
| samples.portions.analytes.aliquots.annotations.entity_type |
| samples.portions.analytes.aliquots.annotations.legacy_created_datetime |
| samples.portions.analytes.aliquots.annotations.legacy_updated_datetime |
| samples.portions.analytes.aliquots.annotations.notes |
| samples.portions.analytes.aliquots.annotations.state |
| samples.portions.analytes.aliquots.annotations.status |
| samples.portions.analytes.aliquots.annotations.submitter_id |
| samples.portions.analytes.aliquots.annotations.updated_datetime |
| samples.portions.analytes.aliquots.center.center_id |
| samples.portions.analytes.aliquots.center.center_type |
| samples.portions.analytes.aliquots.center.code |
| samples.portions.analytes.aliquots.center.name |
| samples.portions.analytes.aliquots.center.namespace |
| samples.portions.analytes.aliquots.center.short_name |
| samples.portions.analytes.annotations.annotation_id |
| samples.portions.analytes.annotations.case_id |
| samples.portions.analytes.annotations.case_submitter_id |
| samples.portions.analytes.annotations.category |
| samples.portions.analytes.annotations.classification |
| samples.portions.analytes.annotations.created_datetime |
| samples.portions.analytes.annotations.creator |
| samples.portions.analytes.annotations.entity_id |
| samples.portions.analytes.annotations.entity_submitter_id |
| samples.portions.analytes.annotations.entity_type |
| samples.portions.analytes.annotations.legacy_created_datetime |
| samples.portions.analytes.annotations.legacy_updated_datetime |
| samples.portions.analytes.annotations.notes |
| samples.portions.analytes.annotations.state |
| samples.portions.analytes.annotations.status |
| samples.portions.analytes.annotations.submitter_id |
| samples.portions.analytes.annotations.updated_datetime |
| samples.portions.annotations.annotation_id |
| samples.portions.annotations.case_id |
| samples.portions.annotations.case_submitter_id |
| samples.portions.annotations.category |
| samples.portions.annotations.classification |
| samples.portions.annotations.created_datetime |
| samples.portions.annotations.creator |
| samples.portions.annotations.entity_id |
| samples.portions.annotations.entity_submitter_id |
| samples.portions.annotations.entity_type |
| samples.portions.annotations.legacy_created_datetime |
| samples.portions.annotations.legacy_updated_datetime |
| samples.portions.annotations.notes |
| samples.portions.annotations.state |
| samples.portions.annotations.status |
| samples.portions.annotations.submitter_id |
| samples.portions.annotations.updated_datetime |
| samples.portions.center.center_id |
| samples.portions.center.center_type |
| samples.portions.center.code |
| samples.portions.center.name |
| samples.portions.center.namespace |
| samples.portions.center.short_name |
| samples.portions.slides.created_datetime |
| samples.portions.slides.number_proliferating_cells |
| samples.portions.slides.percent_eosinophil_infiltration |
| samples.portions.slides.percent_granulocyte_infiltration |
| samples.portions.slides.percent_inflam_infiltration |
| samples.portions.slides.percent_lymphocyte_infiltration |
| samples.portions.slides.percent_monocyte_infiltration |
| samples.portions.slides.percent_necrosis |
| samples.portions.slides.percent_neutrophil_infiltration |
| samples.portions.slides.percent_normal_cells |
| samples.portions.slides.percent_stromal_cells |
| samples.portions.slides.percent_tumor_cells |
| samples.portions.slides.percent_tumor_nuclei |
| samples.portions.slides.section_location |
| samples.portions.slides.slide_id |
| samples.portions.slides.state |
| samples.portions.slides.submitter_id |
| samples.portions.slides.updated_datetime |
| samples.portions.slides.annotations.annotation_id |
| samples.portions.slides.annotations.case_id |
| samples.portions.slides.annotations.case_submitter_id |
| samples.portions.slides.annotations.category |
| samples.portions.slides.annotations.classification |
| samples.portions.slides.annotations.created_datetime |
| samples.portions.slides.annotations.creator |
| samples.portions.slides.annotations.entity_id |
| samples.portions.slides.annotations.entity_submitter_id |
| samples.portions.slides.annotations.entity_type |
| samples.portions.slides.annotations.legacy_created_datetime |
| samples.portions.slides.annotations.legacy_updated_datetime |
| samples.portions.slides.annotations.notes |
| samples.portions.slides.annotations.state |
| samples.portions.slides.annotations.status |
| samples.portions.slides.annotations.submitter_id |
| samples.portions.slides.annotations.updated_datetime |
| summary.file_count |
| summary.file_size |
| summary.data_categories.data_category |
| summary.data_categories.file_count |
| summary.experimental_strategies.experimental_strategy |
| summary.experimental_strategies.file_count |
| tissue_source_site.bcr_id |
| tissue_source_site.code |
| tissue_source_site.name |
| tissue_source_site.project |
| tissue_source_site.tissue_source_site_id |


### File Fields

| Field Name |
| --- | ------ |
| access |
| acl |
| created_datetime |
| data_category |
| data_format |
| data_type |
| error_type |
| experimental_strategy |
| file_id |
| file_name |
| file_size |
| file_state |
| md5sum |
| origin |
| platform |
| revision |
| state |
| state_comment |
| submitter_id |
| tags |
| type |
| updated_datetime |
| analysis.analysis_id |
| analysis.analysis_type |
| analysis.created_datetime |
| analysis.state |
| analysis.submitter_id |
| analysis.updated_datetime |
| analysis.workflow_end_datetime |
| analysis.workflow_link |
| analysis.workflow_start_datetime |
| analysis.workflow_type |
| analysis.workflow_version |
| analysis.input_files.access |
| analysis.input_files.created_datetime |
| analysis.input_files.data_category |
| analysis.input_files.data_format |
| analysis.input_files.data_type |
| analysis.input_files.error_type |
| analysis.input_files.experimental_strategy |
| analysis.input_files.file_id |
| analysis.input_files.file_name |
| analysis.input_files.file_size |
| analysis.input_files.file_state |
| analysis.input_files.md5sum |
| analysis.input_files.platform |
| analysis.input_files.revision |
| analysis.input_files.state |
| analysis.input_files.state_comment |
| analysis.input_files.submitter_id |
| analysis.input_files.updated_datetime |
| analysis.metadata.read_groups.adapter_name |
| analysis.metadata.read_groups.adapter_sequence |
| analysis.metadata.read_groups.base_caller_name |
| analysis.metadata.read_groups.base_caller_version |
| analysis.metadata.read_groups.created_datetime |
| analysis.metadata.read_groups.experiment_name |
| analysis.metadata.read_groups.flow_cell_barcode |
| analysis.metadata.read_groups.includes_spike_ins |
| analysis.metadata.read_groups.instrument_model |
| analysis.metadata.read_groups.is_paired_end |
| analysis.metadata.read_groups.library_name |
| analysis.metadata.read_groups.library_preparation_kit_catalog_number |
| analysis.metadata.read_groups.library_preparation_kit_name |
| analysis.metadata.read_groups.library_preparation_kit_vendor |
| analysis.metadata.read_groups.library_preparation_kit_version |
| analysis.metadata.read_groups.library_selection |
| analysis.metadata.read_groups.library_strand |
| analysis.metadata.read_groups.library_strategy |
| analysis.metadata.read_groups.platform |
| analysis.metadata.read_groups.read_group_id |
| analysis.metadata.read_groups.read_group_name |
| analysis.metadata.read_groups.read_length |
| analysis.metadata.read_groups.RIN |
| analysis.metadata.read_groups.sequencing_center |
| analysis.metadata.read_groups.sequencing_date |
| analysis.metadata.read_groups.size_selection_range |
| analysis.metadata.read_groups.spike_ins_concentration |
| analysis.metadata.read_groups.spike_ins_fasta |
| analysis.metadata.read_groups.state |
| analysis.metadata.read_groups.submitter_id |
| analysis.metadata.read_groups.target_capture_kit_catalog_number |
| analysis.metadata.read_groups.target_capture_kit_name |
| analysis.metadata.read_groups.target_capture_kit_target_region |
| analysis.metadata.read_groups.target_capture_kit_vendor |
| analysis.metadata.read_groups.target_capture_kit_version |
| analysis.metadata.read_groups.to_trim_adapter_sequence |
| analysis.metadata.read_groups.updated_datetime |
| analysis.metadata.read_groups.read_group_qcs.adapter_content |
| analysis.metadata.read_groups.read_group_qcs.basic_statistics |
| analysis.metadata.read_groups.read_group_qcs.created_datetime |
| analysis.metadata.read_groups.read_group_qcs.encoding |
| analysis.metadata.read_groups.read_group_qcs.fastq_name |
| analysis.metadata.read_groups.read_group_qcs.kmer_content |
| analysis.metadata.read_groups.read_group_qcs.overrepresented_sequences |
| analysis.metadata.read_groups.read_group_qcs.per_base_n_content |
| analysis.metadata.read_groups.read_group_qcs.per_base_sequence_content |
| analysis.metadata.read_groups.read_group_qcs.per_base_sequence_quality |
| analysis.metadata.read_groups.read_group_qcs.per_sequence_gc_content |
| analysis.metadata.read_groups.read_group_qcs.per_sequence_quality_score |
| analysis.metadata.read_groups.read_group_qcs.per_tile_sequence_quality |
| analysis.metadata.read_groups.read_group_qcs.percent_gc_content |
| analysis.metadata.read_groups.read_group_qcs.read_group_qc_id |
| analysis.metadata.read_groups.read_group_qcs.sequence_duplication_levels |
| analysis.metadata.read_groups.read_group_qcs.sequence_length_distribution |
| analysis.metadata.read_groups.read_group_qcs.state |
| analysis.metadata.read_groups.read_group_qcs.submitter_id |
| analysis.metadata.read_groups.read_group_qcs.total_sequences |
| analysis.metadata.read_groups.read_group_qcs.updated_datetime |
| analysis.metadata.read_groups.read_group_qcs.workflow_end_datetime |
| analysis.metadata.read_groups.read_group_qcs.workflow_link |
| analysis.metadata.read_groups.read_group_qcs.workflow_start_datetime |
| analysis.metadata.read_groups.read_group_qcs.workflow_type |
| analysis.metadata.read_groups.read_group_qcs.workflow_version |
| annotations.annotation_id |
| annotations.case_id |
| annotations.case_submitter_id |
| annotations.category |
| annotations.classification |
| annotations.created_datetime |
| annotations.creator |
| annotations.entity_id |
| annotations.entity_submitter_id |
| annotations.entity_type |
| annotations.legacy_created_datetime |
| annotations.legacy_updated_datetime |
| annotations.notes |
| annotations.state |
| annotations.status |
| annotations.submitter_id |
| annotations.updated_datetime |
| archive.archive_id |
| archive.created_datetime |
| archive.data_category |
| archive.data_format |
| archive.data_type |
| archive.error_type |
| archive.file_name |
| archive.file_size |
| archive.file_state |
| archive.md5sum |
| archive.revision |
| archive.state |
| archive.state_comment |
| archive.submitter_id |
| archive.updated_datetime |
| associated_entities.case_id |
| associated_entities.entity_id |
| associated_entities.entity_submitter_id |
| associated_entities.entity_type |
| cases.aliquot_ids |
| cases.analyte_ids |
| cases.case_id |
| cases.created_datetime |
| cases.days_to_index |
| cases.portion_ids |
| cases.sample_ids |
| cases.slide_ids |
| cases.state |
| cases.submitter_aliquot_ids |
| cases.submitter_analyte_ids |
| cases.submitter_id |
| cases.submitter_portion_ids |
| cases.submitter_sample_ids |
| cases.submitter_slide_ids |
| cases.updated_datetime |
| cases.annotations.annotation_id |
| cases.annotations.case_id |
| cases.annotations.case_submitter_id |
| cases.annotations.category |
| cases.annotations.classification |
| cases.annotations.created_datetime |
| cases.annotations.creator |
| cases.annotations.entity_id |
| cases.annotations.entity_submitter_id |
| cases.annotations.entity_type |
| cases.annotations.legacy_created_datetime |
| cases.annotations.legacy_updated_datetime |
| cases.annotations.notes |
| cases.annotations.state |
| cases.annotations.status |
| cases.annotations.submitter_id |
| cases.annotations.updated_datetime |
| cases.demographic.created_datetime |
| cases.demographic.demographic_id |
| cases.demographic.ethnicity |
| cases.demographic.gender |
| cases.demographic.race |
| cases.demographic.state |
| cases.demographic.submitter_id |
| cases.demographic.updated_datetime |
| cases.demographic.year_of_birth |
| cases.demographic.year_of_death |
| cases.diagnoses.age_at_diagnosis |
| cases.diagnoses.classification_of_tumor |
| cases.diagnoses.created_datetime |
| cases.diagnoses.days_to_birth |
| cases.diagnoses.days_to_death |
| cases.diagnoses.days_to_last_follow_up |
| cases.diagnoses.days_to_last_known_disease_status |
| cases.diagnoses.days_to_recurrence |
| cases.diagnoses.diagnosis_id |
| cases.diagnoses.last_known_disease_status |
| cases.diagnoses.morphology |
| cases.diagnoses.primary_diagnosis |
| cases.diagnoses.prior_malignancy |
| cases.diagnoses.progression_or_recurrence |
| cases.diagnoses.site_of_resection_or_biopsy |
| cases.diagnoses.state |
| cases.diagnoses.submitter_id |
| cases.diagnoses.tissue_or_organ_of_origin |
| cases.diagnoses.tumor_grade |
| cases.diagnoses.tumor_stage |
| cases.diagnoses.updated_datetime |
| cases.diagnoses.vital_status |
| cases.diagnoses.treatments.created_datetime |
| cases.diagnoses.treatments.days_to_treatment |
| cases.diagnoses.treatments.state |
| cases.diagnoses.treatments.submitter_id |
| cases.diagnoses.treatments.therapeutic_agents |
| cases.diagnoses.treatments.treatment_id |
| cases.diagnoses.treatments.treatment_intent_type |
| cases.diagnoses.treatments.treatment_or_therapy |
| cases.diagnoses.treatments.updated_datetime |
| cases.exposures.alcohol_history |
| cases.exposures.alcohol_intensity |
| cases.exposures.bmi |
| cases.exposures.cigarettes_per_day |
| cases.exposures.created_datetime |
| cases.exposures.exposure_id |
| cases.exposures.height |
| cases.exposures.state |
| cases.exposures.submitter_id |
| cases.exposures.updated_datetime |
| cases.exposures.weight |
| cases.exposures.years_smoked |
| cases.family_histories.created_datetime |
| cases.family_histories.family_history_id |
| cases.family_histories.relationship_age_at_diagnosis |
| cases.family_histories.relationship_gender |
| cases.family_histories.relationship_primary_diagnosis |
| cases.family_histories.relationship_type |
| cases.family_histories.relative_with_cancer_history |
| cases.family_histories.state |
| cases.family_histories.submitter_id |
| cases.family_histories.updated_datetime |
| cases.files.created_datetime |
| cases.files.error_type |
| cases.files.file_id |
| cases.files.file_name |
| cases.files.file_size |
| cases.files.file_state |
| cases.files.md5sum |
| cases.files.state |
| cases.files.state_comment |
| cases.files.submitter_id |
| cases.files.updated_datetime |
| cases.project.dbgap_accession_number |
| cases.project.disease_type |
| cases.project.name |
| cases.project.primary_site |
| cases.project.project_id |
| cases.project.released |
| cases.project.state |
| cases.project.program.dbgap_accession_number |
| cases.project.program.name |
| cases.project.program.program_id |
| cases.samples.composition |
| cases.samples.created_datetime |
| cases.samples.current_weight |
| cases.samples.days_to_collection |
| cases.samples.days_to_sample_procurement |
| cases.samples.freezing_method |
| cases.samples.initial_weight |
| cases.samples.intermediate_dimension |
| cases.samples.is_ffpe |
| cases.samples.longest_dimension |
| cases.samples.oct_embedded |
| cases.samples.pathology_report_uuid |
| cases.samples.preservation_method |
| cases.samples.sample_id |
| cases.samples.sample_type |
| cases.samples.sample_type_id |
| cases.samples.shortest_dimension |
| cases.samples.state |
| cases.samples.submitter_id |
| cases.samples.time_between_clamping_and_freezing |
| cases.samples.time_between_excision_and_freezing |
| cases.samples.tissue_type |
| cases.samples.tumor_code |
| cases.samples.tumor_code_id |
| cases.samples.tumor_descriptor |
| cases.samples.updated_datetime |
| cases.samples.annotations.annotation_id |
| cases.samples.annotations.case_id |
| cases.samples.annotations.case_submitter_id |
| cases.samples.annotations.category |
| cases.samples.annotations.classification |
| cases.samples.annotations.created_datetime |
| cases.samples.annotations.creator |
| cases.samples.annotations.entity_id |
| cases.samples.annotations.entity_submitter_id |
| cases.samples.annotations.entity_type |
| cases.samples.annotations.legacy_created_datetime |
| cases.samples.annotations.legacy_updated_datetime |
| cases.samples.annotations.notes |
| cases.samples.annotations.state |
| cases.samples.annotations.status |
| cases.samples.annotations.submitter_id |
| cases.samples.annotations.updated_datetime |
| cases.samples.portions.created_datetime |
| cases.samples.portions.creation_datetime |
| cases.samples.portions.is_ffpe |
| cases.samples.portions.portion_id |
| cases.samples.portions.portion_number |
| cases.samples.portions.state |
| cases.samples.portions.submitter_id |
| cases.samples.portions.updated_datetime |
| cases.samples.portions.weight |
| cases.samples.portions.analytes.a260_a280_ratio |
| cases.samples.portions.analytes.amount |
| cases.samples.portions.analytes.analyte_id |
| cases.samples.portions.analytes.analyte_type |
| cases.samples.portions.analytes.analyte_type_id |
| cases.samples.portions.analytes.concentration |
| cases.samples.portions.analytes.created_datetime |
| cases.samples.portions.analytes.spectrophotometer_method |
| cases.samples.portions.analytes.state |
| cases.samples.portions.analytes.submitter_id |
| cases.samples.portions.analytes.updated_datetime |
| cases.samples.portions.analytes.well_number |
| cases.samples.portions.analytes.aliquots.aliquot_id |
| cases.samples.portions.analytes.aliquots.amount |
| cases.samples.portions.analytes.aliquots.analyte_type |
| cases.samples.portions.analytes.aliquots.analyte_type_id |
| cases.samples.portions.analytes.aliquots.concentration |
| cases.samples.portions.analytes.aliquots.created_datetime |
| cases.samples.portions.analytes.aliquots.source_center |
| cases.samples.portions.analytes.aliquots.state |
| cases.samples.portions.analytes.aliquots.submitter_id |
| cases.samples.portions.analytes.aliquots.updated_datetime |
| cases.samples.portions.analytes.aliquots.annotations.annotation_id |
| cases.samples.portions.analytes.aliquots.annotations.case_id |
| cases.samples.portions.analytes.aliquots.annotations.case_submitter_id |
| cases.samples.portions.analytes.aliquots.annotations.category |
| cases.samples.portions.analytes.aliquots.annotations.classification |
| cases.samples.portions.analytes.aliquots.annotations.created_datetime |
| cases.samples.portions.analytes.aliquots.annotations.creator |
| cases.samples.portions.analytes.aliquots.annotations.entity_id |
| cases.samples.portions.analytes.aliquots.annotations.entity_submitter_id |
| cases.samples.portions.analytes.aliquots.annotations.entity_type |
| cases.samples.portions.analytes.aliquots.annotations.legacy_created_datetime |
| cases.samples.portions.analytes.aliquots.annotations.legacy_updated_datetime |
| cases.samples.portions.analytes.aliquots.annotations.notes |
| cases.samples.portions.analytes.aliquots.annotations.state |
| cases.samples.portions.analytes.aliquots.annotations.status |
| cases.samples.portions.analytes.aliquots.annotations.submitter_id |
| cases.samples.portions.analytes.aliquots.annotations.updated_datetime |
| cases.samples.portions.analytes.aliquots.center.center_id |
| cases.samples.portions.analytes.aliquots.center.center_type |
| cases.samples.portions.analytes.aliquots.center.code |
| cases.samples.portions.analytes.aliquots.center.name |
| cases.samples.portions.analytes.aliquots.center.namespace |
| cases.samples.portions.analytes.aliquots.center.short_name |
| cases.samples.portions.analytes.annotations.annotation_id |
| cases.samples.portions.analytes.annotations.case_id |
| cases.samples.portions.analytes.annotations.case_submitter_id |
| cases.samples.portions.analytes.annotations.category |
| cases.samples.portions.analytes.annotations.classification |
| cases.samples.portions.analytes.annotations.created_datetime |
| cases.samples.portions.analytes.annotations.creator |
| cases.samples.portions.analytes.annotations.entity_id |
| cases.samples.portions.analytes.annotations.entity_submitter_id |
| cases.samples.portions.analytes.annotations.entity_type |
| cases.samples.portions.analytes.annotations.legacy_created_datetime |
| cases.samples.portions.analytes.annotations.legacy_updated_datetime |
| cases.samples.portions.analytes.annotations.notes |
| cases.samples.portions.analytes.annotations.state |
| cases.samples.portions.analytes.annotations.status |
| cases.samples.portions.analytes.annotations.submitter_id |
| cases.samples.portions.analytes.annotations.updated_datetime |
| cases.samples.portions.annotations.annotation_id |
| cases.samples.portions.annotations.case_id |
| cases.samples.portions.annotations.case_submitter_id |
| cases.samples.portions.annotations.category |
| cases.samples.portions.annotations.classification |
| cases.samples.portions.annotations.created_datetime |
| cases.samples.portions.annotations.creator |
| cases.samples.portions.annotations.entity_id |
| cases.samples.portions.annotations.entity_submitter_id |
| cases.samples.portions.annotations.entity_type |
| cases.samples.portions.annotations.legacy_created_datetime |
| cases.samples.portions.annotations.legacy_updated_datetime |
| cases.samples.portions.annotations.notes |
| cases.samples.portions.annotations.state |
| cases.samples.portions.annotations.status |
| cases.samples.portions.annotations.submitter_id |
| cases.samples.portions.annotations.updated_datetime |
| cases.samples.portions.center.center_id |
| cases.samples.portions.center.center_type |
| cases.samples.portions.center.code |
| cases.samples.portions.center.name |
| cases.samples.portions.center.namespace |
| cases.samples.portions.center.short_name |
| cases.samples.portions.slides.created_datetime |
| cases.samples.portions.slides.number_proliferating_cells |
| cases.samples.portions.slides.percent_eosinophil_infiltration |
| cases.samples.portions.slides.percent_granulocyte_infiltration |
| cases.samples.portions.slides.percent_inflam_infiltration |
| cases.samples.portions.slides.percent_lymphocyte_infiltration |
| cases.samples.portions.slides.percent_monocyte_infiltration |
| cases.samples.portions.slides.percent_necrosis |
| cases.samples.portions.slides.percent_neutrophil_infiltration |
| cases.samples.portions.slides.percent_normal_cells |
| cases.samples.portions.slides.percent_stromal_cells |
| cases.samples.portions.slides.percent_tumor_cells |
| cases.samples.portions.slides.percent_tumor_nuclei |
| cases.samples.portions.slides.section_location |
| cases.samples.portions.slides.slide_id |
| cases.samples.portions.slides.state |
| cases.samples.portions.slides.submitter_id |
| cases.samples.portions.slides.updated_datetime |
| cases.samples.portions.slides.annotations.annotation_id |
| cases.samples.portions.slides.annotations.case_id |
| cases.samples.portions.slides.annotations.case_submitter_id |
| cases.samples.portions.slides.annotations.category |
| cases.samples.portions.slides.annotations.classification |
| cases.samples.portions.slides.annotations.created_datetime |
| cases.samples.portions.slides.annotations.creator |
| cases.samples.portions.slides.annotations.entity_id |
| cases.samples.portions.slides.annotations.entity_submitter_id |
| cases.samples.portions.slides.annotations.entity_type |
| cases.samples.portions.slides.annotations.legacy_created_datetime |
| cases.samples.portions.slides.annotations.legacy_updated_datetime |
| cases.samples.portions.slides.annotations.notes |
| cases.samples.portions.slides.annotations.state |
| cases.samples.portions.slides.annotations.status |
| cases.samples.portions.slides.annotations.submitter_id |
| cases.samples.portions.slides.annotations.updated_datetime |
| cases.summary.file_count |
| cases.summary.file_size |
| cases.summary.data_categories.data_category |
| cases.summary.data_categories.file_count |
| cases.summary.experimental_strategies.experimental_strategy |
| cases.summary.experimental_strategies.file_count |
| cases.tissue_source_site.bcr_id |
| cases.tissue_source_site.code |
| cases.tissue_source_site.name |
| cases.tissue_source_site.project |
| cases.tissue_source_site.tissue_source_site_id |
| center.center_id |
| center.center_type |
| center.code |
| center.name |
| center.namespace |
| center.short_name |
| downstream_analyses.analysis_id |
| downstream_analyses.analysis_type |
| downstream_analyses.created_datetime |
| downstream_analyses.state |
| downstream_analyses.submitter_id |
| downstream_analyses.updated_datetime |
| downstream_analyses.workflow_end_datetime |
| downstream_analyses.workflow_link |
| downstream_analyses.workflow_start_datetime |
| downstream_analyses.workflow_type |
| downstream_analyses.workflow_version |
| downstream_analyses.output_files.access |
| downstream_analyses.output_files.created_datetime |
| downstream_analyses.output_files.data_category |
| downstream_analyses.output_files.data_format |
| downstream_analyses.output_files.data_type |
| downstream_analyses.output_files.error_type |
| downstream_analyses.output_files.experimental_strategy |
| downstream_analyses.output_files.file_id |
| downstream_analyses.output_files.file_name |
| downstream_analyses.output_files.file_size |
| downstream_analyses.output_files.file_state |
| downstream_analyses.output_files.md5sum |
| downstream_analyses.output_files.platform |
| downstream_analyses.output_files.revision |
| downstream_analyses.output_files.state |
| downstream_analyses.output_files.state_comment |
| downstream_analyses.output_files.submitter_id |
| downstream_analyses.output_files.updated_datetime |
| index_files.access |
| index_files.created_datetime |
| index_files.data_category |
| index_files.data_format |
| index_files.data_type |
| index_files.error_type |
| index_files.experimental_strategy |
| index_files.file_id |
| index_files.file_name |
| index_files.file_size |
| index_files.file_state |
| index_files.md5sum |
| index_files.platform |
| index_files.revision |
| index_files.state |
| index_files.state_comment |
| index_files.submitter_id |
| index_files.updated_datetime |
| metadata_files.access |
| metadata_files.created_datetime |
| metadata_files.data_category |
| metadata_files.data_format |
| metadata_files.data_type |
| metadata_files.error_type |
| metadata_files.file_id |
| metadata_files.file_name |
| metadata_files.file_size |
| metadata_files.file_state |
| metadata_files.md5sum |
| metadata_files.state |
| metadata_files.state_comment |
| metadata_files.submitter_id |
| metadata_files.type |
| metadata_files.updated_datetime |



### Annotation Fields


| Field Name |
| --- |
| annotation_id |
| case_id |
| case_submitter_id |
| category |
| classification |
| created_datetime |
| entity_id |
| entity_submitter_id |
| entity_type |
| legacy_created_datetime |
| legacy_updated_datetime |
| notes |
| state |
| status |
| submitter_id |
| updated_datetime |
| project.code |
| project.dbgap_accession_number |
| project.disease_type |
| project.name |
| project.primary_site |
| project.program.dbgap_accession_number |
| project.program.name |
| project.program.program_id |
| project.project_id |
| project.released |
| project.state |

## Field Group Listing by Endpoint

### Projects Field Groups

| Field Group Name |
| --- |
| program |
| summary |
| summary.data_categories |
| summary.experimental_strategies |

### Cases Field Groups

| Field Group Name |
| --- |
| annotations |
| demographic |
| diagnoses |
| diagnoses.treatments |
| exposures |
| family_histories |
| files |
| files.analysis |
| files.analysis.input_files |
| files.analysis.metadata |
| files.analysis.metadata.read_groups |
| files.analysis.metadata.read_groups.read_group_qcs |
| files.archive |
| files.cases |
| files.cases.annotations |
| files.cases.demographic |
| files.cases.diagnoses |
| files.cases.diagnoses.treatments |
| files.cases.exposures |
| files.cases.family_histories |
| files.cases.files |
| files.cases.project |
| files.cases.project.program |
| files.cases.samples |
| files.cases.samples.annotations |
| files.cases.samples.portions |
| files.cases.samples.portions.analytes |
| files.cases.samples.portions.analytes.aliquots |
| files.cases.samples.portions.analytes.aliquots.annotations |
| files.cases.samples.portions.analytes.aliquots.center |
| files.cases.samples.portions.analytes.annotations |
| files.cases.samples.portions.annotations |
| files.cases.samples.portions.center |
| files.cases.samples.portions.slides |
| files.cases.samples.portions.slides.annotations |
| files.cases.summary |
| files.cases.summary.data_categories |
| files.cases.summary.experimental_strategies |
| files.cases.tissue_source_site |
| files.center |
| files.downstream_analyses |
| files.downstream_analyses.output_files |
| files.index_files |
| files.metadata_files |
| project |
| project.program |
| samples |
| samples.annotations |
| samples.portions |
| samples.portions.analytes |
| samples.portions.analytes.aliquots |
| samples.portions.analytes.aliquots.annotations |
| samples.portions.analytes.aliquots.center |
| samples.portions.analytes.annotations |
| samples.portions.annotations |
| samples.portions.center |
| samples.portions.slides |
| samples.portions.slides.annotations |
| summary |
| summary.data_categories |
| summary.experimental_strategies |
| tissue_source_site |

### Files Field Groups

| Field Group Name |
| --- |
| analysis |
| analysis.input_files |
| analysis.metadata |
| analysis.metadata.read_groups |
| analysis.metadata.read_groups.read_group_qcs |
| annotations |
| archive |
| associated_entities |
| cases |
| cases.annotations |
| cases.demographic |
| cases.diagnoses |
| cases.diagnoses.treatments |
| cases.exposures |
| cases.family_histories |
| cases.files |
| cases.project |
| cases.project.program |
| cases.samples |
| cases.samples.annotations |
| cases.samples.portions |
| cases.samples.portions.analytes |
| cases.samples.portions.analytes.aliquots |
| cases.samples.portions.analytes.aliquots.annotations |
| cases.samples.portions.analytes.aliquots.center |
| cases.samples.portions.analytes.annotations |
| cases.samples.portions.annotations |
| cases.samples.portions.center |
| cases.samples.portions.slides |
| cases.samples.portions.slides.annotations |
| cases.summary |
| cases.summary.data_categories |
| cases.summary.experimental_strategies |
| cases.tissue_source_site |
| center |
| downstream_analyses |
| downstream_analyses.output_files |
| index_files |
| metadata_files |

### Annotations Field Groups

| Field Group Name |
| --- |
| project |
| project.program |

### GDC Conventions Supplemental
| Field Group Name |
| --- | -------- |
| File size | Listed in bytes |
| Submitter id | Supplied string by submitter |
| Datetime | A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm] |
| Id | Is the supplied GDC UUID |  


# Using Python to Query the GDC API

Python can be a versatile tool for retrieving information from the GDC API and performing downstream processing. This page details some examples that demonstrate the basic API queries using Python. The examples in this guide will use the [requests](https://pypi.org/project/requests/) Python library and should be compatible with Python3.

## Querying Metadata

Python can be used with the GDC API to retrieve metadata that is indexed in the GDC Database. See the [Search and Retrieval](Search_and_Retrieval.md) section of the API documentation for specific details about parameters and usage.

### A Basic Query

This example passes some basic parameters (fields, format, size) to the `cases` endpoint and prints the results in a tab-delimited format. Note that the `fields` parameter needs to be a string comprising comma-delimited field names.  

```Python
import requests
import json

cases_endpt = 'https://api.gdc.cancer.gov/cases'

# The 'fields' parameter is passed as a comma-separated string of single names.
fields = [
    "submitter_id",
    "case_id",
    "primary_site",
    "disease_type",
    "diagnoses.vital_status"
    ]

fields = ','.join(fields)

params = {
    "fields": fields,
    "format": "TSV",
    "size": "100"
    }

response = requests.get(cases_endpt, params = params)

print(response.content)
```
[Download Script](scripts/Basic_Query.py)

### A Filtered Query

In the next example, a `filters` parameter is added to the script. This parameter is passed as a Python dictionary object. The filter used in this example will only display cases that come from a kidney disease study (`primary_site: Kidney`).

```Python
import requests
import json

fields = [
    "submitter_id",
    "case_id",
    "primary_site",
    "disease_type",
    "diagnoses.vital_status"
    ]

fields = ",".join(fields)

cases_endpt = "https://api.gdc.cancer.gov/cases"

filters = {
    "op": "in",
    "content":{
        "field": "primary_site",
        "value": ["Kidney"]
        }
    }

# With a GET request, the filters parameter needs to be converted
# from a dictionary to JSON-formatted string

params = {
    "filters": json.dumps(filters),
    "fields": fields,
    "format": "TSV",
    "size": "100"
    }

response = requests.get(cases_endpt, params = params)

print(response.content)
```
[Download Script](scripts/Filter_Query.py)

### Complex Filters

The following example uses the `and` operator in the filter to returns information about files that 1) were produced using RNA-Seq, 2) are downloadable in BAM format, and 3) originate from lung cancer patients. Note that these three filters are nested within a list in the highest level `content` key.  

```Python
import requests
import json

fields = [
    "file_name",
    "cases.submitter_id",
    "cases.samples.sample_type",
    "cases.disease_type",
    "cases.project.project_id"
    ]

fields = ",".join(fields)

files_endpt = "https://api.gdc.cancer.gov/files"

# This set of filters is nested under an 'and' operator.
filters = {
    "op": "and",
    "content":[
        {
        "op": "in",
        "content":{
            "field": "cases.project.primary_site",
            "value": ["Lung"]
            }
        },
        {
        "op": "in",
        "content":{
            "field": "files.experimental_strategy",
            "value": ["RNA-Seq"]
            }
        },
        {
        "op": "in",
        "content":{
            "field": "files.data_format",
            "value": ["BAM"]
            }
        }
    ]
}

# A POST is used, so the filter parameters can be passed directly as a Dict object.
params = {
    "filters": filters,
    "fields": fields,
    "format": "TSV",
    "size": "2000"
    }

# The parameters are passed to 'json' rather than 'params' in this case
response = requests.post(files_endpt, headers = {"Content-Type": "application/json"}, json = params)

print(response.content.decode("utf-8"))
```
[Download Script](scripts/Complex_Query.py)


## Downloading Files

GDC files can also be downloaded from the API and saved locally using Python scripts. See the [File Download](Downloading_Files.md) section of the API documentation for more information.  

### A Simple Download Request

An open-access GDC file can be downloaded by appending the file UUID to the `data` endpoint URL.  

```Python
import requests
import json
import re

file_id = "b658d635-258a-4f6f-8377-767a43771fe4"

data_endpt = "https://api.gdc.cancer.gov/data/{}".format(file_id)

response = requests.get(data_endpt, headers = {"Content-Type": "application/json"})

# The file name can be found in the header within the Content-Disposition key.
response_head_cd = response.headers["Content-Disposition"]

file_name = re.findall("filename=(.+)", response_head_cd)[0]

with open(file_name, "wb") as output_file:
    output_file.write(response.content)
```
[Download Script](scripts/Download_Files.py)

### Passing a Token to Download a Controlled-Access File

A token can be passed to the script by specifying a plain text file that contains only the GDC token. A token can be downloaded by logging into the GDC Data Portal. See the [Data Security](../../../Data/Data_Security/Data_Security) documentation for more details.  

```Python
import requests
import json

'''
 This script will not work until $TOKEN_FILE_PATH
 is replaced with an actual path.
'''
token_file = "$TOKEN_FILE_PATH"

file_id = "11443f3c-9b8b-4e47-b5b7-529468fec098"

data_endpt = "https://api.gdc.cancer.gov/slicing/view/{}".format(file_id)

with open(token_file,"r") as token:
    token_string = str(token.read().strip())

params = {"gencode": ["BRCA1", "BRCA2"]}

response = requests.post(data_endpt,
                        data = json.dumps(params),
                        headers = {
                            "Content-Type": "application/json",
                            "X-Auth-Token": token_string
                            })

file_name = "brca_slices.bam"

with open(file_name, "wb") as output_file:
    output_file.write(response.content)
```
[Download Script](scripts/Download_Files_Token.py)

### Post Request to Download Multiple Files  

This example uses a Python list to specify a set of file UUIDs. The list in the example was populated manually but could potentially be populated programmatically from an external list or API call.  

```Python
import requests
import json
import re

data_endpt = "https://api.gdc.cancer.gov/data"

ids = [
    "b658d635-258a-4f6f-8377-767a43771fe4",
    "3968213d-b293-4b3d-8033-5b5a0ca07b6c"
    ]

params = {"ids": ids}

response = requests.post(data_endpt,
                        data = json.dumps(params),
                        headers={
                            "Content-Type": "application/json"
                            })

response_head_cd = response.headers["Content-Disposition"]

file_name = re.findall("filename=(.+)", response_head_cd)[0]

with open(file_name, "wb") as output_file:
    output_file.write(response.content)
```
[Download Script](scripts/Download_Files_Post.py)

### Downloading a Set of Files Based on a Filter

Here a list of files based on a set of filters are downloaded.  File UUIDs are retrieved based on the filters. These UUIDs are then passed to the `data` endpoint to download the correct files.   

```Python
import requests
import json
import re

files_endpt = "https://api.gdc.cancer.gov/files"

filters = {
    "op": "and",
    "content":[
        {
        "op": "in",
        "content":{
            "field": "cases.project.primary_site",
            "value": ["Lung"]
            }
        },
        {
        "op": "in",
        "content":{
            "field": "cases.demographic.race",
            "value": ["white"]
            }
        },
        {
        "op": "in",
        "content":{
            "field": "cases.demographic.gender",
            "value": ["female"]
            }
        },
        {
        "op": "in",
        "content":{
            "field": "files.analysis.workflow_type",
            "value": ["HTSeq - FPKM"]
            }
        }
    ]
}

# Here a GET is used, so the filter parameters should be passed as a JSON string.

params = {
    "filters": json.dumps(filters),
    "fields": "file_id",
    "format": "JSON",
    "size": "1000"
    }

response = requests.get(files_endpt, params = params)

file_uuid_list = []

# This step populates the download list with the file_ids from the previous query
for file_entry in json.loads(response.content.decode("utf-8"))["data"]["hits"]:
    file_uuid_list.append(file_entry["file_id"])

data_endpt = "https://api.gdc.cancer.gov/data"

params = {"ids": file_uuid_list}

response = requests.post(data_endpt, data = json.dumps(params), headers = {"Content-Type": "application/json"})

response_head_cd = response.headers["Content-Disposition"]

file_name = re.findall("filename=(.+)", response_head_cd)[0]

with open(file_name, "wb") as output_file:
    output_file.write(response.content)
```
[Download Script](scripts/Download_Files_Filter.py)

### BAM Slicing

The GDC [BAM Slicing](BAM_Slicing.md) feature can also be accessed through Python. Below is an example of a basic BAM slicing command.  

```Python
import requests
import json

'''
 This script will not work until $TOKEN_FILE_PATH
 is replaced with an actual path.
'''
token_file = "$TOKEN_FILE_PATH"

file_id = "11443f3c-9b8b-4e47-b5b7-529468fec098"

data_endpt = "https://api.gdc.cancer.gov/slicing/view/{}".format(file_id)

with open(token_file,"r") as token:
    token_string = str(token.read().strip())

params = {"gencode": ["BRCA1", "BRCA2"]}

response = requests.post(data_endpt,
                        data = json.dumps(params),
                        headers = {
                            "Content-Type": "application/json",
                            "X-Auth-Token": token_string
                            })

file_name = "brca_slices.bam"

with open(file_name, "wb") as output_file:
    output_file.write(response.content)
```
[Download Script](scripts/BAM_Slice.py)

The same region(s) across multiple BAM files can be retrieved using a for-loop within a Python script.

```Python
import requests
import json

'''
 This script will not work until $TOKEN_FILE_PATH
 is replaced with an actual path.
'''
token_file = "$TOKEN_FILE_PATH"

file_ids = [
    "11443f3c-9b8b-4e47-b5b7-529468fec098",
    "1f103620-bb34-46f1-b565-94f0027e396d",
    "ca549554-a244-4209-9086-92add7bb7109"
    ]

for file_id in file_ids:

    data_endpt = "https://api.gdc.cancer.gov/slicing/view/{}".format(file_id)

    with open(token_file, "r") as token:
        token_string = str(token.read().strip())

    params = {
        "regions": ["chr1:1-20000", "chr10:129000-160000"]
        }

    response = requests.post(data_endpt,
                            data = json.dumps(params),
                            headers = {
                                "Content-Type": "application/json",
                                "X-Auth-Token": token_string
                                })

    file_name = "{}_region_slices.bam".format(file_id)

    with open(file_name, "wb") as output_file:
        output_file.write(response.content)
```
[Download Script](scripts/BAM_Slice_Multiple.py)

## Basic Troubleshooting

The following script should produce an unformatted JSON string with information about the API status. Run this script to verify that a valid connection is being made to the GDC API.  

```Python
import requests
status_endpt = "https://api.gdc.cancer.gov/status"
response = requests.get(status_endpt)

# OUTPUT METHOD 1: Write to a file.
file = open("api_status.json", "w")
file.write(response.text)
file.close()

# OUTPUT METHOD 2: View on screen.
print(response.content)
```
[Download Script](scripts/Basic_Troubleshooting.py)


# Appendix A: Available Fields

The GDC API's [search and retrieval endpoints](Search_and_Retrieval.md) provide access to fields that correspond to properties defined in the [GDC Data Dictionary](../../Data_Dictionary/viewer.md). This appendix contains a list of fields available at each endpoint, and a list of field groups accessible via the [expand parameter](Search_and_Retrieval.md#expand).

## Field Listing by Endpoint

### Project Fields

| Field Name |
| --- | -------- |
| dbgap_accession_number |
| disease_type |
| name |  
| primary_site |
| project_id |
| released |
| state |
| program.dbgap_accession_number |
| program.name |  
| program.program_id |
| summary.case_count |
| summary.file_count |
| summary.file_size |
| summary.data_categories.case_count |
| summary.data_categories.data_category |
| summary.data_categories.file_count |
| summary.experimental_strategies.case_count |  
| summary.experimental_strategies.experimental_strategy |
| summary.experimental_strategies.file_count |



### Case Fields


| Field Name |
| --- | -------- |
| aliquot_ids |
| analyte_ids |
| case_id |
| created_datetime |  
| days_to_index |  
| portion_ids |
| sample_ids |
| slide_ids |
| state |
| submitter_aliquot_ids |
| submitter_analyte_ids |
| submitter_id |
| submitter_portion_ids |
| submitter_sample_ids |
| submitter_slide_ids |
| updated_datetime |  
| annotations.annotation_id |
| annotations.case_id |
| annotations.case_submitter_id |
| annotations.category |
| annotations.classification |
| annotations.created_datetime |
| annotations.creator |
| annotations.entity_id |
| annotations.entity_submitter_id |
| annotations.entity_type |
| annotations.legacy_created_datetime |
| annotations.legacy_updated_datetime |
| annotations.notes |
| annotations.state |
| annotations.status |
| annotations.submitter_id |
| annotations.updated_datetime |
| demographic.created_datetime |
| demographic.demographic_id |
| demographic.ethnicity |
| demographic.gender |
| demographic.race |
| demographic.state |
| demographic.submitter_id |
| demographic.updated_datetime |
| demographic.year_of_birth |
| demographic.year_of_death |
| diagnoses.age_at_diagnosis |
| diagnoses.classification_of_tumor |
| diagnoses.created_datetime |
| diagnoses.days_to_birth |
| diagnoses.days_to_death |
| diagnoses.days_to_last_follow_up |
| diagnoses.days_to_last_known_disease_status |
| diagnoses.days_to_recurrence |
| diagnoses.diagnosis_id |
| diagnoses.last_known_disease_status |
| diagnoses.morphology |
| diagnoses.primary_diagnosis |
| diagnoses.prior_malignancy |
| diagnoses.progression_or_recurrence |
| diagnoses.site_of_resection_or_biopsy |
| diagnoses.state |
| diagnoses.submitter_id |
| diagnoses.tissue_or_organ_of_origin |
| diagnoses.tumor_grade |
| diagnoses.tumor_stage |
| diagnoses.updated_datetime |
| diagnoses.vital_status |
| diagnoses.treatments.created_datetime |
| diagnoses.treatments.days_to_treatment |
| diagnoses.treatments.state |
| diagnoses.treatments.submitter_id |
| diagnoses.treatments.therapeutic_agents |
| diagnoses.treatments.treatment_id |
| diagnoses.treatments.treatment_intent_type |
| diagnoses.treatments.treatment_or_therapy |
| diagnoses.treatments.updated_datetime |
| exposures.alcohol_history |
| exposures.alcohol_intensity |
| exposures.bmi |
| exposures.cigarettes_per_day |
| exposures.created_datetime |
| exposures.exposure_id |
| exposures.height |
| exposures.state |
| exposures.submitter_id |
| exposures.updated_datetime |
| exposures.weight |
| exposures.years_smoked |
| family_histories.created_datetime |
| family_histories.family_history_id |  
| family_histories.relationship_age_at_diagnosis |
| family_histories.relationship_gender |
| family_histories.relationship_primary_diagnosis |
| family_histories.relationship_type |
| family_histories.relative_with_cancer_history |
| family_histories.state |
| family_histories.submitter_id |
| family_histories.updated_datetime |
| files.access |
| files.acl |
| files.created_datetime |
| files.data_category |
| files.data_format |
| files.data_type |
| files.error_type |
| files.experimental_strategy |
| files.file_id |
| files.file_name |
| files.file_size |
| files.file_state |
| files.md5sum |
| files.origin |
| files.platform |
| files.revision |
| files.state |
| files.state_comment |
| files.submitter_id |
| files.tags |
| files.type |
| files.updated_datetime |
| files.analysis.analysis_id |
| files.analysis.analysis_type |
| files.analysis.created_datetime |
| files.analysis.state |
| files.analysis.submitter_id |
| files.analysis.updated_datetime |
| files.analysis.workflow_end_datetime |
| files.analysis.workflow_link |
| files.analysis.workflow_start_datetime |
| files.analysis.workflow_type |
| files.analysis.workflow_version |
| files.analysis.input_files.access |
| files.analysis.input_files.created_datetime |
| files.analysis.input_files.data_category |
| files.analysis.input_files.data_format |
| files.analysis.input_files.data_type |
| files.analysis.input_files.error_type |
| files.analysis.input_files.experimental_strategy |
| files.analysis.input_files.file_id |
| files.analysis.input_files.file_name |
| files.analysis.input_files.file_size |
| files.analysis.input_files.file_state |
| files.analysis.input_files.md5sum |
| files.analysis.input_files.platform |
| files.analysis.input_files.revision |
| files.analysis.input_files.state |
| files.analysis.input_files.state_comment |
| files.analysis.input_files.submitter_id |
| files.analysis.input_files.updated_datetime |
| files.analysis.metadata.read_groups.adapter_name |
| files.analysis.metadata.read_groups.adapter_sequence |
| files.analysis.metadata.read_groups.base_caller_name |
| files.analysis.metadata.read_groups.base_caller_version |
| files.analysis.metadata.read_groups.created_datetime |
| files.analysis.metadata.read_groups.experiment_name |
| files.analysis.metadata.read_groups.flow_cell_barcode |
| files.analysis.metadata.read_groups.includes_spike_ins |
| files.analysis.metadata.read_groups.instrument_model |
| files.analysis.metadata.read_groups.is_paired_end |
| files.analysis.metadata.read_groups.library_name |
| files.analysis.metadata.read_groups.library_preparation_kit_catalog_number |
| files.analysis.metadata.read_groups.library_preparation_kit_name |
| files.analysis.metadata.read_groups.library_preparation_kit_vendor |
| files.analysis.metadata.read_groups.library_preparation_kit_version |
| files.analysis.metadata.read_groups.library_selection |
| files.analysis.metadata.read_groups.library_strand |
| files.analysis.metadata.read_groups.library_strategy |
| files.analysis.metadata.read_groups.platform |
| files.analysis.metadata.read_groups.read_group_id |
| files.analysis.metadata.read_groups.read_group_name |
| files.analysis.metadata.read_groups.read_length |
| files.analysis.metadata.read_groups.RIN |
| files.analysis.metadata.read_groups.sequencing_center |
| files.analysis.metadata.read_groups.sequencing_date |
| files.analysis.metadata.read_groups.size_selection_range |
| files.analysis.metadata.read_groups.spike_ins_concentration |
| files.analysis.metadata.read_groups.spike_ins_fasta |
| files.analysis.metadata.read_groups.state |
| files.analysis.metadata.read_groups.submitter_id |
| files.analysis.metadata.read_groups.target_capture_kit_catalog_number |
| files.analysis.metadata.read_groups.target_capture_kit_name |
| files.analysis.metadata.read_groups.target_capture_kit_target_region |
| files.analysis.metadata.read_groups.target_capture_kit_vendor |
| files.analysis.metadata.read_groups.target_capture_kit_version |
| files.analysis.metadata.read_groups.to_trim_adapter_sequence |
| files.analysis.metadata.read_groups.updated_datetime |
| files.analysis.metadata.read_groups.read_group_qcs.adapter_content |
| files.analysis.metadata.read_groups.read_group_qcs.basic_statistics |
| files.analysis.metadata.read_groups.read_group_qcs.created_datetime |
| files.analysis.metadata.read_groups.read_group_qcs.encoding |
| files.analysis.metadata.read_groups.read_group_qcs.fastq_name |
| files.analysis.metadata.read_groups.read_group_qcs.kmer_content |
| files.analysis.metadata.read_groups.read_group_qcs.overrepresented_sequences |
| files.analysis.metadata.read_groups.read_group_qcs.per_base_n_content |
| files.analysis.metadata.read_groups.read_group_qcs.per_base_sequence_content |
| files.analysis.metadata.read_groups.read_group_qcs.per_base_sequence_quality |
| files.analysis.metadata.read_groups.read_group_qcs.per_sequence_gc_content |
| files.analysis.metadata.read_groups.read_group_qcs.per_sequence_quality_score |
| files.analysis.metadata.read_groups.read_group_qcs.per_tile_sequence_quality |
| files.analysis.metadata.read_groups.read_group_qcs.percent_gc_content |
| files.analysis.metadata.read_groups.read_group_qcs.read_group_qc_id |
| files.analysis.metadata.read_groups.read_group_qcs.sequence_duplication_levels |
| files.analysis.metadata.read_groups.read_group_qcs.sequence_length_distribution |
| files.analysis.metadata.read_groups.read_group_qcs.state |
| files.analysis.metadata.read_groups.read_group_qcs.submitter_id |
| files.analysis.metadata.read_groups.read_group_qcs.total_sequences |
| files.analysis.metadata.read_groups.read_group_qcs.updated_datetime |
| files.analysis.metadata.read_groups.read_group_qcs.workflow_end_datetime |
| files.analysis.metadata.read_groups.read_group_qcs.workflow_link |
| files.analysis.metadata.read_groups.read_group_qcs.workflow_start_datetime |
| files.analysis.metadata.read_groups.read_group_qcs.workflow_type |
| files.analysis.metadata.read_groups.read_group_qcs.workflow_version |
| files.archive.archive_id |
| files.archive.created_datetime |
| files.archive.data_category |
| files.archive.data_format |
| files.archive.data_type |
| files.archive.error_type |
| files.archive.file_name |
| files.archive.file_size |
| files.archive.file_state |
| files.archive.md5sum |
| files.archive.revision |
| files.archive.state |
| files.archive.state_comment |
| files.archive.submitter_id |
| files.archive.updated_datetime |
| files.cases.aliquot_ids |
| files.cases.analyte_ids |
| files.cases.case_id |
| files.cases.created_datetime |
| files.cases.days_to_index |
| files.cases.portion_ids |
| files.cases.sample_ids |
| files.cases.slide_ids |
| files.cases.state |
| files.cases.submitter_aliquot_ids |
| files.cases.submitter_analyte_ids |
| files.cases.submitter_id |
| files.cases.submitter_portion_ids |
| files.cases.submitter_sample_ids |
| files.cases.submitter_slide_ids |
| files.cases.updated_datetime |
| files.cases.annotations.annotation_id |
| files.cases.annotations.case_id |
| files.cases.annotations.case_submitter_id |
| files.cases.annotations.category |
| files.cases.annotations.classification |
| files.cases.annotations.created_datetime |
| files.cases.annotations.creator |
| files.cases.annotations.entity_id |
| files.cases.annotations.entity_submitter_id |
| files.cases.annotations.entity_type |
| files.cases.annotations.legacy_created_datetime |
| files.cases.annotations.legacy_updated_datetime |
| files.cases.annotations.notes |
| files.cases.annotations.state |
| files.cases.annotations.status |
| files.cases.annotations.submitter_id |
| files.cases.annotations.updated_datetime |
| files.cases.demographic.created_datetime |
| files.cases.demographic.demographic_id |
| files.cases.demographic.ethnicity |
| files.cases.demographic.gender |
| files.cases.demographic.race |
| files.cases.demographic.state |
| files.cases.demographic.submitter_id |
| files.cases.demographic.updated_datetime |
| files.cases.demographic.year_of_birth |
| files.cases.demographic.year_of_death |
| files.cases.diagnoses.age_at_diagnosis |
| files.cases.diagnoses.classification_of_tumor |
| files.cases.diagnoses.created_datetime |
| files.cases.diagnoses.days_to_birth |
| files.cases.diagnoses.days_to_death |
| files.cases.diagnoses.days_to_last_follow_up |
| files.cases.diagnoses.days_to_last_known_disease_status |
| files.cases.diagnoses.days_to_recurrence |
| files.cases.diagnoses.diagnosis_id |
| files.cases.diagnoses.last_known_disease_status |
| files.cases.diagnoses.morphology |
| files.cases.diagnoses.primary_diagnosis |
| files.cases.diagnoses.prior_malignancy |
| files.cases.diagnoses.progression_or_recurrence |
| files.cases.diagnoses.site_of_resection_or_biopsy |
| files.cases.diagnoses.state |
| files.cases.diagnoses.submitter_id |
| files.cases.diagnoses.tissue_or_organ_of_origin |
| files.cases.diagnoses.tumor_grade |
| files.cases.diagnoses.tumor_stage |
| files.cases.diagnoses.updated_datetime |
| files.cases.diagnoses.vital_status |
| files.cases.diagnoses.treatments.created_datetime |
| files.cases.diagnoses.treatments.days_to_treatment |
| files.cases.diagnoses.treatments.state |
| files.cases.diagnoses.treatments.submitter_id |
| files.cases.diagnoses.treatments.therapeutic_agents |
| files.cases.diagnoses.treatments.treatment_id |
| files.cases.diagnoses.treatments.treatment_intent_type |
| files.cases.diagnoses.treatments.treatment_or_therapy |
| files.cases.diagnoses.treatments.updated_datetime |
| files.cases.exposures.alcohol_history |
| files.cases.exposures.alcohol_intensity |
| files.cases.exposures.bmi |
| files.cases.exposures.cigarettes_per_day |
| files.cases.exposures.created_datetime |
| files.cases.exposures.exposure_id |
| files.cases.exposures.height |
| files.cases.exposures.state |
| files.cases.exposures.submitter_id |
| files.cases.exposures.updated_datetime |
| files.cases.exposures.weight |
| files.cases.exposures.years_smoked |
| files.cases.family_histories.created_datetime |
| files.cases.family_histories.family_history_id |
| files.cases.family_histories.relationship_age_at_diagnosis |
| files.cases.family_histories.relationship_gender |
| files.cases.family_histories.relationship_primary_diagnosis |
| files.cases.family_histories.relationship_type |
| files.cases.family_histories.relative_with_cancer_history |
| files.cases.family_histories.state |
| files.cases.family_histories.submitter_id |
| files.cases.family_histories.updated_datetime |
| files.cases.files.created_datetime |
| files.cases.files.error_type |
| files.cases.files.file_id |
| files.cases.files.file_name |
| files.cases.files.file_size |
| files.cases.files.file_state |
| files.cases.files.md5sum |
| files.cases.files.state |
| files.cases.files.state_comment |
| files.cases.files.submitter_id |
| files.cases.files.updated_datetime |
| files.cases.project.dbgap_accession_number |
| files.cases.project.disease_type |
| files.cases.project.name |
| files.cases.project.primary_site |
| files.cases.project.project_id |
| files.cases.project.released |
| files.cases.project.state |
| files.cases.project.program.dbgap_accession_number |
| files.cases.project.program.name |
| files.cases.project.program.program_id |
| files.cases.samples.composition |
| files.cases.samples.created_datetime |
| files.cases.samples.current_weight |
| files.cases.samples.days_to_collection |
| files.cases.samples.days_to_sample_procurement |
| files.cases.samples.freezing_method |
| files.cases.samples.initial_weight |
| files.cases.samples.intermediate_dimension |
| files.cases.samples.is_ffpe |
| files.cases.samples.longest_dimension |
| files.cases.samples.oct_embedded |
| files.cases.samples.pathology_report_uuid |
| files.cases.samples.preservation_method |
| files.cases.samples.sample_id |
| files.cases.samples.sample_type |
| files.cases.samples.sample_type_id |
| files.cases.samples.shortest_dimension |
| files.cases.samples.state |
| files.cases.samples.submitter_id |
| files.cases.samples.time_between_clamping_and_freezing |
| files.cases.samples.time_between_excision_and_freezing |
| files.cases.samples.tissue_type |
| files.cases.samples.tumor_code |
| files.cases.samples.tumor_code_id |
| files.cases.samples.tumor_descriptor |
| files.cases.samples.updated_datetime |
| files.cases.samples.annotations.annotation_id |
| files.cases.samples.annotations.case_id |
| files.cases.samples.annotations.case_submitter_id |
| files.cases.samples.annotations.category |
| files.cases.samples.annotations.classification |
| files.cases.samples.annotations.created_datetime |
| files.cases.samples.annotations.creator |
| files.cases.samples.annotations.entity_id |
| files.cases.samples.annotations.entity_submitter_id |
| files.cases.samples.annotations.entity_type |
| files.cases.samples.annotations.legacy_created_datetime |
| files.cases.samples.annotations.legacy_updated_datetime |
| files.cases.samples.annotations.notes |
| files.cases.samples.annotations.state |
| files.cases.samples.annotations.status |
| files.cases.samples.annotations.submitter_id |
| files.cases.samples.annotations.updated_datetime |
| files.cases.samples.portions.created_datetime |
| files.cases.samples.portions.creation_datetime |
| files.cases.samples.portions.is_ffpe |
| files.cases.samples.portions.portion_id |
| files.cases.samples.portions.portion_number |
| files.cases.samples.portions.state |  
| files.cases.samples.portions.submitter_id |  
| files.cases.samples.portions.updated_datetime |
| files.cases.samples.portions.weight |
| files.cases.samples.portions.analytes.a260_a280_ratio |
| files.cases.samples.portions.analytes.amount |
| files.cases.samples.portions.analytes.analyte_id |
| files.cases.samples.portions.analytes.analyte_type |
| files.cases.samples.portions.analytes.analyte_type_id |
| files.cases.samples.portions.analytes.concentration |
| files.cases.samples.portions.analytes.created_datetime |
| files.cases.samples.portions.analytes.spectrophotometer_method |
| files.cases.samples.portions.analytes.state |
| files.cases.samples.portions.analytes.submitter_id |
| files.cases.samples.portions.analytes.updated_datetime |
| files.cases.samples.portions.analytes.well_number |
| files.cases.samples.portions.analytes.aliquots.aliquot_id |
| files.cases.samples.portions.analytes.aliquots.amount |
| files.cases.samples.portions.analytes.aliquots.analyte_type |
| files.cases.samples.portions.analytes.aliquots.analyte_type_id |
| files.cases.samples.portions.analytes.aliquots.concentration |
| files.cases.samples.portions.analytes.aliquots.created_datetime |
| files.cases.samples.portions.analytes.aliquots.source_center |
| files.cases.samples.portions.analytes.aliquots.state |
| files.cases.samples.portions.analytes.aliquots.submitter_id |
| files.cases.samples.portions.analytes.aliquots.updated_datetime |
| files.cases.samples.portions.analytes.aliquots.annotations.annotation_id |
| files.cases.samples.portions.analytes.aliquots.annotations.case_id |
| files.cases.samples.portions.analytes.aliquots.annotations.case_submitter_id |
| files.cases.samples.portions.analytes.aliquots.annotations.category |
| files.cases.samples.portions.analytes.aliquots.annotations.classification |
| files.cases.samples.portions.analytes.aliquots.annotations.created_datetime |
| files.cases.samples.portions.analytes.aliquots.annotations.creator |
| files.cases.samples.portions.analytes.aliquots.annotations.entity_id |
| files.cases.samples.portions.analytes.aliquots.annotations.entity_submitter_id |
| files.cases.samples.portions.analytes.aliquots.annotations.entity_type |
| files.cases.samples.portions.analytes.aliquots.annotations.legacy_created_datetime |
| files.cases.samples.portions.analytes.aliquots.annotations.legacy_updated_datetime |
| files.cases.samples.portions.analytes.aliquots.annotations.notes |
| files.cases.samples.portions.analytes.aliquots.annotations.state |
| files.cases.samples.portions.analytes.aliquots.annotations.status |
| files.cases.samples.portions.analytes.aliquots.annotations.submitter_id | Submitter supplied identifier information
| files.cases.samples.portions.analytes.aliquots.annotations.updated_datetime |
| files.cases.samples.portions.analytes.aliquots.center.center_id |
| files.cases.samples.portions.analytes.aliquots.center.center_type |
| files.cases.samples.portions.analytes.aliquots.center.code |  Numeric code for the center
| files.cases.samples.portions.analytes.aliquots.center.name |  Name of the center (e.g. Center For Data Intensive Science at the University of Chicago)
| files.cases.samples.portions.analytes.aliquots.center.namespace |  Domain name of the center (e.g. cdis.uchicago.edu)
| files.cases.samples.portions.analytes.aliquots.center.short_name |  Shortened name of the center (e.g. BI)
| files.cases.samples.portions.analytes.annotations.annotation_id |
| files.cases.samples.portions.analytes.annotations.case_id |
| files.cases.samples.portions.analytes.annotations.case_submitter_id | Submitter supplied identifier information
| files.cases.samples.portions.analytes.annotations.category |  Top level characterization of the annotation
| files.cases.samples.portions.analytes.annotations.classification |  Top level classification of the annotation
| files.cases.samples.portions.analytes.annotations.created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| files.cases.samples.portions.analytes.annotations.creator |  Name of the person or entity responsible for the creation of the annotation
| files.cases.samples.portions.analytes.annotations.entity_id |
| files.cases.samples.portions.analytes.annotations.entity_submitter_id | Submitter supplied identifier information
| files.cases.samples.portions.analytes.annotations.entity_type |
| files.cases.samples.portions.analytes.annotations.legacy_created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| files.cases.samples.portions.analytes.annotations.legacy_updated_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| files.cases.samples.portions.analytes.annotations.notes |
| files.cases.samples.portions.analytes.annotations.state |
| files.cases.samples.portions.analytes.annotations.status |
| files.cases.samples.portions.analytes.annotations.submitter_id | Submitter supplied identifier information
| files.cases.samples.portions.analytes.annotations.updated_datetime |
| files.cases.samples.portions.annotations.annotation_id |
| files.cases.samples.portions.annotations.case_id |
| files.cases.samples.portions.annotations.case_submitter_id | Submitter supplied identifier information
| files.cases.samples.portions.annotations.category |  Top level characterization of the annotation
| files.cases.samples.portions.annotations.classification |  Top level classification of the annotation
| files.cases.samples.portions.annotations.created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| files.cases.samples.portions.annotations.creator |  Name of the person or entity responsible for the creation of the annotation
| files.cases.samples.portions.annotations.entity_id |
| files.cases.samples.portions.annotations.entity_submitter_id | Submitter supplied identifier information
| files.cases.samples.portions.annotations.entity_type |
| files.cases.samples.portions.annotations.legacy_created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| files.cases.samples.portions.annotations.legacy_updated_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| files.cases.samples.portions.annotations.notes |
| files.cases.samples.portions.annotations.state |
| files.cases.samples.portions.annotations.status |
| files.cases.samples.portions.annotations.submitter_id | Submitter supplied identifier information
| files.cases.samples.portions.annotations.updated_datetime |
| files.cases.samples.portions.center.center_id |
| files.cases.samples.portions.center.center_type |
| files.cases.samples.portions.center.code |  Numeric code for the center
| files.cases.samples.portions.center.name |  Name of the center (e.g. Center For Data Intensive Science at the University of Chicago)
| files.cases.samples.portions.center.namespace |  Domain name of the center (e.g. cdis.uchicago.edu)
| files.cases.samples.portions.center.short_name |  Shortened name of the center (e.g. BI)
| files.cases.samples.portions.slides.created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| files.cases.samples.portions.slides.number_proliferating_cells |  Numeric value that represents the count of proliferating cells determined during pathologic review of the sample slide(s)
| files.cases.samples.portions.slides.percent_eosinophil_infiltration |  Numeric value that represents the count of proliferating cells determined during pathologic review of the sample slide(s).
| files.cases.samples.portions.slides.percent_granulocyte_infiltration |
| files.cases.samples.portions.slides.percent_inflam_infiltration |  Numeric value to represent local response to cellular injury, marked by capillary dilatation, edema and leukocyte infiltration; clinically, inflammation is manifest by reddness, heat, pain, swelling and loss of function, with the need to heal damaged tissue
| files.cases.samples.portions.slides.percent_lymphocyte_infiltration |
| files.cases.samples.portions.slides.percent_monocyte_infiltration |
| files.cases.samples.portions.slides.percent_necrosis |
| files.cases.samples.portions.slides.percent_neutrophil_infiltration |
| files.cases.samples.portions.slides.percent_normal_cells |
| files.cases.samples.portions.slides.percent_stromal_cells |
| files.cases.samples.portions.slides.percent_tumor_cells |
| files.cases.samples.portions.slides.percent_tumor_nuclei |
| files.cases.samples.portions.slides.section_location |
| files.cases.samples.portions.slides.slide_id |
| files.cases.samples.portions.slides.state |  The current state of the object
| files.cases.samples.portions.slides.submitter_id |  The legacy barcode used before prior to the use UUIDs, varies by project
| files.cases.samples.portions.slides.updated_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]\
| files.cases.samples.portions.slides.annotations.annotation_id |
| files.cases.samples.portions.slides.annotations.case_id |
| files.cases.samples.portions.slides.annotations.case_submitter_id | Submitter supplied identifier information
| files.cases.samples.portions.slides.annotations.category |  Top level characterization of the annotation
| files.cases.samples.portions.slides.annotations.classification |  Top level classification of the annotation
| files.cases.samples.portions.slides.annotations.created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| files.cases.samples.portions.slides.annotations.creator |  Name of the person or entity responsible for the creation of the annotation
| files.cases.samples.portions.slides.annotations.entity_id |
| files.cases.samples.portions.slides.annotations.entity_submitter_id | Submitter supplied identifier information
| files.cases.samples.portions.slides.annotations.entity_type |
| files.cases.samples.portions.slides.annotations.legacy_created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| files.cases.samples.portions.slides.annotations.legacy_updated_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| files.cases.samples.portions.slides.annotations.notes |  Open entry for any further description or characterization of the data
| files.cases.samples.portions.slides.annotations.state |  The current state of the object
| files.cases.samples.portions.slides.annotations.status |  Status of the annotation
| files.cases.samples.portions.slides.annotations.submitter_id | Submitter supplied identifier information
| files.cases.samples.portions.slides.annotations.updated_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| files.cases.summary.file_count |
| files.cases.summary.file_size |
| files.cases.summary.data_categories.data_category |
| files.cases.summary.data_categories.file_count |
| files.cases.summary.experimental_strategies.experimental_strategy |
| files.cases.summary.experimental_strategies.file_count |
| files.cases.tissue_source_site.bcr_id |
| files.cases.tissue_source_site.code |
| files.cases.tissue_source_site.name |
| files.cases.tissue_source_site.project |
| files.cases.tissue_source_site.tissue_source_site_id |
| files.center.center_id |
| files.center.center_type |
| files.center.code |  Numeric code for the center
| files.center.name |  Name of the center (e.g. Center For Data Intensive Science at the University of Chicago)
| files.center.namespace |  Domain name of the center (e.g. cdis.uchicago.edu)
| files.center.short_name |  Shortened name of the center (e.g. BI)
| files.downstream_analyses.analysis_id |
| files.downstream_analyses.analysis_type |
| files.downstream_analyses.created_datetime |
| files.downstream_analyses.state |
| files.downstream_analyses.submitter_id | Submitter supplied identifier information
| files.downstream_analyses.updated_datetime |
| files.downstream_analyses.workflow_end_datetime |
| files.downstream_analyses.workflow_link |
| files.downstream_analyses.workflow_start_datetime |
| files.downstream_analyses.workflow_type |
| files.downstream_analyses.workflow_version |
| files.downstream_analyses.output_files.access |
| files.downstream_analyses.output_files.created_datetime |
| files.downstream_analyses.output_files.data_category |
| files.downstream_analyses.output_files.data_format |
| files.downstream_analyses.output_files.data_type |
| files.downstream_analyses.output_files.error_type |
| files.downstream_analyses.output_files.experimental_strategy |
| files.downstream_analyses.output_files.file_id |
| files.downstream_analyses.output_files.file_name |
| files.downstream_analyses.output_files.file_size |
| files.downstream_analyses.output_files.file_state |
| files.downstream_analyses.output_files.md5sum |
| files.downstream_analyses.output_files.platform |
| files.downstream_analyses.output_files.revision |
| files.downstream_analyses.output_files.state |
| files.downstream_analyses.output_files.state_comment |
| files.downstream_analyses.output_files.submitter_id | Submitter supplied identifier information
| files.downstream_analyses.output_files.updated_datetime |
| files.index_files.access |
| files.index_files.created_datetime |
| files.index_files.data_category |
| files.index_files.data_format |
| files.index_files.data_type |
| files.index_files.error_type |
| files.index_files.experimental_strategy |
| files.index_files.file_id |
| files.index_files.file_name |
| files.index_files.file_size |
| files.index_files.file_state |
| files.index_files.md5sum |
| files.index_files.platform |
| files.index_files.revision |
| files.index_files.state |
| files.index_files.state_comment |
| files.index_files.submitter_id | Submitter supplied identifier information
| files.index_files.updated_datetime |
| files.metadata_files.access |
| files.metadata_files.created_datetime |
| files.metadata_files.data_category |
| files.metadata_files.data_format |
| files.metadata_files.data_type |
| files.metadata_files.error_type |
| files.metadata_files.file_id |
| files.metadata_files.file_name |
| files.metadata_files.file_size |
| files.metadata_files.file_state |
| files.metadata_files.md5sum |
| files.metadata_files.state |
| files.metadata_files.state_comment |
| files.metadata_files.submitter_id | Submitter supplied identifier information
| files.metadata_files.type |
| files.metadata_files.updated_datetime |
| project.dbgap_accession_number |  The dbgap accession number provided for the program
| project.disease_type |
| project.name |
| project.primary_site |
| project.project_id |
| project.released |
| project.state |
| project.program.dbgap_accession_number |
| project.program.name |
| project.program.program_id |
| samples.composition |
| samples.created_datetime |
| samples.current_weight |
| samples.days_to_collection |
| samples.days_to_sample_procurement |
| samples.freezing_method |
| samples.initial_weight |
| samples.intermediate_dimension |
| samples.is_ffpe |
| samples.longest_dimension |
| samples.oct_embedded |
| samples.pathology_report_uuid |
| samples.preservation_method |
| samples.sample_id |
| samples.sample_type |
| samples.sample_type_id |
| samples.shortest_dimension |
| samples.state |
| samples.submitter_id | Submitter supplied identifier information
| samples.time_between_clamping_and_freezing |
| samples.time_between_excision_and_freezing |
| samples.tissue_type |
| samples.tumor_code |
| samples.tumor_code_id |
| samples.tumor_descriptor |
| samples.updated_datetime |
| samples.annotations.annotation_id |
| samples.annotations.case_id |
| samples.annotations.case_submitter_id | Submitter supplied identifier information
| samples.annotations.category |  Top level characterization of the annotation
| samples.annotations.classification |  Top level classification of the annotation
| samples.annotations.created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| samples.annotations.creator |  Name of the person or entity responsible for the creation of the annotation
| samples.annotations.entity_id |
| samples.annotations.entity_submitter_id | Submitter supplied identifier information
| samples.annotations.entity_type |
| samples.annotations.legacy_created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| samples.annotations.legacy_updated_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| samples.annotations.notes |
| samples.annotations.state |
| samples.annotations.status |
| samples.annotations.submitter_id | Submitter supplied identifier information
| samples.annotations.updated_datetime |
| samples.portions.created_datetime |
| samples.portions.creation_datetime |
| samples.portions.is_ffpe |
| samples.portions.portion_id |
| samples.portions.portion_number |  Numeric value that represents the sequential number assigned to a portion of the sample.
| samples.portions.state |
| samples.portions.submitter_id | Submitter supplied identifier information
| samples.portions.updated_datetime |
| samples.portions.weight |
| samples.portions.analytes.a260_a280_ratio |
| samples.portions.analytes.amount |
| samples.portions.analytes.analyte_id |
| samples.portions.analytes.analyte_type |
| samples.portions.analytes.analyte_type_id |
| samples.portions.analytes.concentration |
| samples.portions.analytes.created_datetime |
| samples.portions.analytes.spectrophotometer_method |
| samples.portions.analytes.state |
| samples.portions.analytes.submitter_id | Submitter supplied identifier information
| samples.portions.analytes.updated_datetime |
| samples.portions.analytes.well_number |
| samples.portions.analytes.aliquots.aliquot_id |
| samples.portions.analytes.aliquots.amount |
| samples.portions.analytes.aliquots.analyte_type |
| samples.portions.analytes.aliquots.analyte_type_id |
| samples.portions.analytes.aliquots.concentration |
| samples.portions.analytes.aliquots.created_datetime |
| samples.portions.analytes.aliquots.source_center |
| samples.portions.analytes.aliquots.state |
| samples.portions.analytes.aliquots.submitter_id | Submitter supplied identifier information
| samples.portions.analytes.aliquots.updated_datetime |
| samples.portions.analytes.aliquots.annotations.annotation_id |
| samples.portions.analytes.aliquots.annotations.case_id |
| samples.portions.analytes.aliquots.annotations.case_submitter_id | Submitter supplied identifier information
| samples.portions.analytes.aliquots.annotations.category |  Top level characterization of the annotation
| samples.portions.analytes.aliquots.annotations.classification |  Top level classification of the annotation
| samples.portions.analytes.aliquots.annotations.created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| samples.portions.analytes.aliquots.annotations.creator |  Name of the person or entity responsible for the creation of the annotation
| samples.portions.analytes.aliquots.annotations.entity_id |
| samples.portions.analytes.aliquots.annotations.entity_submitter_id | Submitter supplied identifier information
| samples.portions.analytes.aliquots.annotations.entity_type |
| samples.portions.analytes.aliquots.annotations.legacy_created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| samples.portions.analytes.aliquots.annotations.legacy_updated_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| samples.portions.analytes.aliquots.annotations.notes |
| samples.portions.analytes.aliquots.annotations.state |
| samples.portions.analytes.aliquots.annotations.status |
| samples.portions.analytes.aliquots.annotations.submitter_id | Submitter supplied identifier information
| samples.portions.analytes.aliquots.annotations.updated_datetime |
| samples.portions.analytes.aliquots.center.center_id |
| samples.portions.analytes.aliquots.center.center_type |
| samples.portions.analytes.aliquots.center.code |  Numeric code for the center
| samples.portions.analytes.aliquots.center.name |  Name of the center (e.g. Center For Data Intensive Science at the University of Chicago)
| samples.portions.analytes.aliquots.center.namespace |  Domain name of the center (e.g. cdis.uchicago.edu)
| samples.portions.analytes.aliquots.center.short_name |  Shortened name of the center (e.g. BI)
| samples.portions.analytes.annotations.annotation_id |
| samples.portions.analytes.annotations.case_id |
| samples.portions.analytes.annotations.case_submitter_id | Submitter supplied identifier information
| samples.portions.analytes.annotations.category |  Top level characterization of the annotation
| samples.portions.analytes.annotations.classification |  Top level classification of the annotation
| samples.portions.analytes.annotations.created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| samples.portions.analytes.annotations.creator |  Name of the person or entity responsible for the creation of the annotation
| samples.portions.analytes.annotations.entity_id |
| samples.portions.analytes.annotations.entity_submitter_id | Submitter supplied identifier information
| samples.portions.analytes.annotations.entity_type |
| samples.portions.analytes.annotations.legacy_created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| samples.portions.analytes.annotations.legacy_updated_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| samples.portions.analytes.annotations.notes |
| samples.portions.analytes.annotations.state |
| samples.portions.analytes.annotations.status |
| samples.portions.analytes.annotations.submitter_id | Submitter supplied identifier information
| samples.portions.analytes.annotations.updated_datetime |
| samples.portions.annotations.annotation_id |
| samples.portions.annotations.case_id |
| samples.portions.annotations.case_submitter_id | Submitter supplied identifier information
| samples.portions.annotations.category |  Top level characterization of the annotation
| samples.portions.annotations.classification |  Top level classification of the annotation
| samples.portions.annotations.created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| samples.portions.annotations.creator |  Name of the person or entity responsible for the creation of the annotation
| samples.portions.annotations.entity_id |
| samples.portions.annotations.entity_submitter_id | Submitter supplied identifier information
| samples.portions.annotations.entity_type |
| samples.portions.annotations.legacy_created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| samples.portions.annotations.legacy_updated_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| samples.portions.annotations.notes |
| samples.portions.annotations.state |
| samples.portions.annotations.status |
| samples.portions.annotations.submitter_id | Submitter supplied identifier information
| samples.portions.annotations.updated_datetime |
| samples.portions.center.center_id |
| samples.portions.center.center_type |
| samples.portions.center.code |  Numeric code for the center
| samples.portions.center.name |  Name of the center (e.g. Center For Data Intensive Science at the University of Chicago)
| samples.portions.center.namespace |  Domain name of the center (e.g. cdis.uchicago.edu)
| samples.portions.center.short_name |  Shortened name of the center (e.g. BI)
| samples.portions.slides.created_datetime |
| samples.portions.slides.number_proliferating_cells |  Numeric value that represents the count of proliferating cells determined during pathologic review of the sample slide(s)
| samples.portions.slides.percent_eosinophil_infiltration |
| samples.portions.slides.percent_granulocyte_infiltration |
| samples.portions.slides.percent_inflam_infiltration |
| samples.portions.slides.percent_lymphocyte_infiltration |
| samples.portions.slides.percent_monocyte_infiltration |
| samples.portions.slides.percent_necrosis |
| samples.portions.slides.percent_neutrophil_infiltration |
| samples.portions.slides.percent_normal_cells |
| samples.portions.slides.percent_stromal_cells |
| samples.portions.slides.percent_tumor_cells |
| samples.portions.slides.percent_tumor_nuclei |
| samples.portions.slides.section_location |
| samples.portions.slides.slide_id |
| samples.portions.slides.state |
| samples.portions.slides.submitter_id | Submitter supplied identifier information
| samples.portions.slides.updated_datetime |
| samples.portions.slides.annotations.annotation_id |
| samples.portions.slides.annotations.case_id |
| samples.portions.slides.annotations.case_submitter_id | Submitter supplied identifier information
| samples.portions.slides.annotations.category |  Top level characterization of the annotation
| samples.portions.slides.annotations.classification |  Top level classification of the annotation
| samples.portions.slides.annotations.created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| samples.portions.slides.annotations.creator |  Name of the person or entity responsible for the creation of the annotation
| samples.portions.slides.annotations.entity_id |
| samples.portions.slides.annotations.entity_submitter_id | Submitter supplied identifier information
| samples.portions.slides.annotations.entity_type |
| samples.portions.slides.annotations.legacy_created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| samples.portions.slides.annotations.legacy_updated_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| samples.portions.slides.annotations.notes |  Open entry for any further description or characterization of the data
| samples.portions.slides.annotations.state |  The current state of the object
| samples.portions.slides.annotations.status |  Status of the annotation
| samples.portions.slides.annotations.submitter_id | Submitter supplied identifier information
| samples.portions.slides.annotations.updated_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| summary.file_count |
| summary.file_size |
| summary.data_categories.data_category |
| summary.data_categories.file_count |
| summary.experimental_strategies.experimental_strategy |
| summary.experimental_strategies.file_count |
| tissue_source_site.bcr_id |
| tissue_source_site.code |
| tissue_source_site.name |
| tissue_source_site.project |
| tissue_source_site.tissue_source_site_id |


### File Fields

| Field Name |
| --- | ------ |
| access |
| acl |
| created_datetime |
| data_category |
| data_format |
| data_type |
| error_type |
| experimental_strategy |
| file_id |
| file_name |
| file_size |
| file_state |
| md5sum |
| origin |
| platform |
| revision |
| state |
| state_comment |
| submitter_id | Submitter supplied identifier information
| tags |
| type |
| updated_datetime |
| analysis.analysis_id |
| analysis.analysis_type |
| analysis.created_datetime |
| analysis.state |
| analysis.submitter_id | Submitter supplied identifier information
| analysis.updated_datetime |
| analysis.workflow_end_datetime |
| analysis.workflow_link |
| analysis.workflow_start_datetime |
| analysis.workflow_type |
| analysis.workflow_version |
| analysis.input_files.access |
| analysis.input_files.created_datetime |
| analysis.input_files.data_category |
| analysis.input_files.data_format |
| analysis.input_files.data_type |
| analysis.input_files.error_type |
| analysis.input_files.experimental_strategy |
| analysis.input_files.file_id |
| analysis.input_files.file_name |
| analysis.input_files.file_size |
| analysis.input_files.file_state |
| analysis.input_files.md5sum |
| analysis.input_files.platform |
| analysis.input_files.revision |
| analysis.input_files.state |
| analysis.input_files.state_comment |
| analysis.input_files.submitter_id | Submitter supplied identifier information
| analysis.input_files.updated_datetime |
| analysis.metadata.read_groups.adapter_name |
| analysis.metadata.read_groups.adapter_sequence |
| analysis.metadata.read_groups.base_caller_name |
| analysis.metadata.read_groups.base_caller_version |
| analysis.metadata.read_groups.created_datetime |
| analysis.metadata.read_groups.experiment_name |
| analysis.metadata.read_groups.flow_cell_barcode |
| analysis.metadata.read_groups.includes_spike_ins |
| analysis.metadata.read_groups.instrument_model |
| analysis.metadata.read_groups.is_paired_end |
| analysis.metadata.read_groups.library_name |
| analysis.metadata.read_groups.library_preparation_kit_catalog_number |
| analysis.metadata.read_groups.library_preparation_kit_name |
| analysis.metadata.read_groups.library_preparation_kit_vendor |
| analysis.metadata.read_groups.library_preparation_kit_version |
| analysis.metadata.read_groups.library_selection |
| analysis.metadata.read_groups.library_strand |
| analysis.metadata.read_groups.library_strategy |
| analysis.metadata.read_groups.platform |
| analysis.metadata.read_groups.read_group_id |
| analysis.metadata.read_groups.read_group_name |
| analysis.metadata.read_groups.read_length |
| analysis.metadata.read_groups.RIN |
| analysis.metadata.read_groups.sequencing_center |
| analysis.metadata.read_groups.sequencing_date |
| analysis.metadata.read_groups.size_selection_range |
| analysis.metadata.read_groups.spike_ins_concentration |
| analysis.metadata.read_groups.spike_ins_fasta |
| analysis.metadata.read_groups.state |
| analysis.metadata.read_groups.submitter_id | Submitter supplied identifier information
| analysis.metadata.read_groups.target_capture_kit_catalog_number |
| analysis.metadata.read_groups.target_capture_kit_name |
| analysis.metadata.read_groups.target_capture_kit_target_region |
| analysis.metadata.read_groups.target_capture_kit_vendor |
| analysis.metadata.read_groups.target_capture_kit_version |
| analysis.metadata.read_groups.to_trim_adapter_sequence |
| analysis.metadata.read_groups.updated_datetime |
| analysis.metadata.read_groups.read_group_qcs.adapter_content |
| analysis.metadata.read_groups.read_group_qcs.basic_statistics |
| analysis.metadata.read_groups.read_group_qcs.created_datetime |
| analysis.metadata.read_groups.read_group_qcs.encoding |
| analysis.metadata.read_groups.read_group_qcs.fastq_name |
| analysis.metadata.read_groups.read_group_qcs.kmer_content |
| analysis.metadata.read_groups.read_group_qcs.overrepresented_sequences |
| analysis.metadata.read_groups.read_group_qcs.per_base_n_content |
| analysis.metadata.read_groups.read_group_qcs.per_base_sequence_content |
| analysis.metadata.read_groups.read_group_qcs.per_base_sequence_quality |
| analysis.metadata.read_groups.read_group_qcs.per_sequence_gc_content |
| analysis.metadata.read_groups.read_group_qcs.per_sequence_quality_score |
| analysis.metadata.read_groups.read_group_qcs.per_tile_sequence_quality |
| analysis.metadata.read_groups.read_group_qcs.percent_gc_content |
| analysis.metadata.read_groups.read_group_qcs.read_group_qc_id |
| analysis.metadata.read_groups.read_group_qcs.sequence_duplication_levels |
| analysis.metadata.read_groups.read_group_qcs.sequence_length_distribution |
| analysis.metadata.read_groups.read_group_qcs.state |
| analysis.metadata.read_groups.read_group_qcs.submitter_id | Submitter supplied identifier information
| analysis.metadata.read_groups.read_group_qcs.total_sequences |
| analysis.metadata.read_groups.read_group_qcs.updated_datetime |
| analysis.metadata.read_groups.read_group_qcs.workflow_end_datetime |
| analysis.metadata.read_groups.read_group_qcs.workflow_link |
| analysis.metadata.read_groups.read_group_qcs.workflow_start_datetime |
| analysis.metadata.read_groups.read_group_qcs.workflow_type |
| analysis.metadata.read_groups.read_group_qcs.workflow_version |
| annotations.annotation_id |
| annotations.case_id |
| annotations.case_submitter_id | Submitter supplied identifier information
| annotations.category |  Characterization of the annotation
| annotations.classification |  Top level classification of the annotation
| annotations.created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| annotations.creator |  Name of the person or entity responsible for the creation of the annotation
| annotations.entity_id |
| annotations.entity_submitter_id | Submitter supplied identifier information
| annotations.entity_type |
| annotations.legacy_created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| annotations.legacy_updated_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| annotations.notes |
| annotations.state |
| annotations.status |
| annotations.submitter_id | Submitter supplied identifier information
| annotations.updated_datetime |
| archive.archive_id |
| archive.created_datetime |
| archive.data_category |
| archive.data_format |
| archive.data_type |
| archive.error_type |
| archive.file_name |
| archive.file_size |
| archive.file_state |
| archive.md5sum |
| archive.revision |
| archive.state |
| archive.state_comment |
| archive.submitter_id | Submitter supplied identifier information
| archive.updated_datetime |
| associated_entities.case_id |
| associated_entities.entity_id |
| associated_entities.entity_submitter_id | Submitter supplied identifier information
| associated_entities.entity_type |
| cases.aliquot_ids |
| cases.analyte_ids |
| cases.case_id |
| cases.created_datetime |
| cases.days_to_index |
| cases.portion_ids |
| cases.sample_ids |
| cases.slide_ids |
| cases.state |
| cases.submitter_aliquot_ids |
| cases.submitter_analyte_ids |
| cases.submitter_id | Submitter supplied identifier information
| cases.submitter_portion_ids |
| cases.submitter_sample_ids |
| cases.submitter_slide_ids |
| cases.updated_datetime |
| cases.annotations.annotation_id |
| cases.annotations.case_id |
| cases.annotations.case_submitter_id | Submitter supplied identifier information
| cases.annotations.category |  Characterization of the annotation
| cases.annotations.classification |  Top level classification of the annotation
| cases.annotations.created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| cases.annotations.creator |  Name of the person or entity responsible for the creation of the annotation
| cases.annotations.entity_id |
| cases.annotations.entity_submitter_id | Submitter supplied identifier information
| cases.annotations.entity_type |
| cases.annotations.legacy_created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| cases.annotations.legacy_updated_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| cases.annotations.notes |
| cases.annotations.state |
| cases.annotations.status |
| cases.annotations.submitter_id | Submitter supplied identifier information
| cases.annotations.updated_datetime |
| cases.demographic.created_datetime |
| cases.demographic.demographic_id |
| cases.demographic.ethnicity |
| cases.demographic.gender |
| cases.demographic.race |
| cases.demographic.state |
| cases.demographic.submitter_id | Submitter supplied identifier information
| cases.demographic.updated_datetime |
| cases.demographic.year_of_birth |
| cases.demographic.year_of_death |
| cases.diagnoses.age_at_diagnosis |
| cases.diagnoses.classification_of_tumor |
| cases.diagnoses.created_datetime |
| cases.diagnoses.days_to_birth |
| cases.diagnoses.days_to_death |
| cases.diagnoses.days_to_last_follow_up |
| cases.diagnoses.days_to_last_known_disease_status |
| cases.diagnoses.days_to_recurrence |
| cases.diagnoses.diagnosis_id |
| cases.diagnoses.last_known_disease_status |
| cases.diagnoses.morphology |
| cases.diagnoses.primary_diagnosis |
| cases.diagnoses.prior_malignancy |
| cases.diagnoses.progression_or_recurrence |
| cases.diagnoses.site_of_resection_or_biopsy |
| cases.diagnoses.state |
| cases.diagnoses.submitter_id | Submitter supplied identifier information
| cases.diagnoses.tissue_or_organ_of_origin |
| cases.diagnoses.tumor_grade |
| cases.diagnoses.tumor_stage |
| cases.diagnoses.updated_datetime |
| cases.diagnoses.vital_status |
| cases.diagnoses.treatments.created_datetime |
| cases.diagnoses.treatments.days_to_treatment |
| cases.diagnoses.treatments.state |
| cases.diagnoses.treatments.submitter_id | Submitter supplied identifier information
| cases.diagnoses.treatments.therapeutic_agents |
| cases.diagnoses.treatments.treatment_id |
| cases.diagnoses.treatments.treatment_intent_type |
| cases.diagnoses.treatments.treatment_or_therapy |
| cases.diagnoses.treatments.updated_datetime |
| cases.exposures.alcohol_history |
| cases.exposures.alcohol_intensity |
| cases.exposures.bmi |
| cases.exposures.cigarettes_per_day |
| cases.exposures.created_datetime |
| cases.exposures.exposure_id |
| cases.exposures.height |
| cases.exposures.state |
| cases.exposures.submitter_id | Submitter supplied identifier information
| cases.exposures.updated_datetime |
| cases.exposures.weight |
| cases.exposures.years_smoked |
| cases.family_histories.created_datetime |
| cases.family_histories.family_history_id |
| cases.family_histories.relationship_age_at_diagnosis |
| cases.family_histories.relationship_gender |
| cases.family_histories.relationship_primary_diagnosis |
| cases.family_histories.relationship_type |
| cases.family_histories.relative_with_cancer_history |
| cases.family_histories.state |
| cases.family_histories.submitter_id | Submitter supplied identifier information
| cases.family_histories.updated_datetime |
| cases.files.created_datetime |
| cases.files.error_type |
| cases.files.file_id |
| cases.files.file_name |
| cases.files.file_size |
| cases.files.file_state |
| cases.files.md5sum |
| cases.files.state |
| cases.files.state_comment |
| cases.files.submitter_id | Submitter supplied identifier information
| cases.files.updated_datetime |
| cases.project.dbgap_accession_number |
| cases.project.disease_type |
| cases.project.name |
| cases.project.primary_site |
| cases.project.project_id |
| cases.project.released |
| cases.project.state |
| cases.project.program.dbgap_accession_number |
| cases.project.program.name |
| cases.project.program.program_id |
| cases.samples.composition |
| cases.samples.created_datetime |
| cases.samples.current_weight |
| cases.samples.days_to_collection |
| cases.samples.days_to_sample_procurement |
| cases.samples.freezing_method |
| cases.samples.initial_weight |
| cases.samples.intermediate_dimension |
| cases.samples.is_ffpe |
| cases.samples.longest_dimension |
| cases.samples.oct_embedded |
| cases.samples.pathology_report_uuid |
| cases.samples.preservation_method |  Text term that represents the method used to preserve the sample
| cases.samples.sample_id |
| cases.samples.sample_type |  Text term to describe the source of a biospecimen used for a laboratory test
| cases.samples.sample_type_id |  The accompanying sample type id for the sample type
| cases.samples.shortest_dimension |  Numeric value that represents the shortest dimension of the sample, measured in millimeters
| cases.samples.state |
| cases.samples.submitter_id | Submitter supplied identifier information
| cases.samples.time_between_clamping_and_freezing |
| cases.samples.time_between_excision_and_freezing |  Numeric representation of the elapsed time between the excision and freezing of the sample, measured in minutes
| cases.samples.tissue_type |  Text term that represents a description of the kind of tissue collected with respect to disease status or proximity to tumor tissue
| cases.samples.tumor_code |  Diagnostic tumor code of the tissue sample source
| cases.samples.tumor_code_id |  BCR-defined id code for the tumor sample
| cases.samples.tumor_descriptor |  Text that describes the kind of disease present in the tumor specimen as related to a specific timepoint
| cases.samples.updated_datetime |
| cases.samples.annotations.annotation_id |
| cases.samples.annotations.case_id |
| cases.samples.annotations.case_submitter_id | Submitter supplied identifier information
| cases.samples.annotations.category | Top level characterization of the annotation
| cases.samples.annotations.classification |  Top level classification of the annotation
| cases.samples.annotations.created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| cases.samples.annotations.creator |  Name of the person or entity responsible for the creation of the annotation
| cases.samples.annotations.entity_id |
| cases.samples.annotations.entity_submitter_id | Submitter supplied identifier information
| cases.samples.annotations.entity_type |
| cases.samples.annotations.legacy_created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| cases.samples.annotations.legacy_updated_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| cases.samples.annotations.notes |
| cases.samples.annotations.state |
| cases.samples.annotations.status |
| cases.samples.annotations.submitter_id | Submitter supplied identifier information
| cases.samples.annotations.updated_datetime |
| cases.samples.portions.created_datetime |
| cases.samples.portions.creation_datetime |
| cases.samples.portions.is_ffpe |
| cases.samples.portions.portion_id |
| cases.samples.portions.portion_number |  Numeric value that represents the sequential number assigned to a portion of the sample.
| cases.samples.portions.state |  The current state of the object
| cases.samples.portions.submitter_id |  The legacy barcode used before prior to the use UUIDs, varies by project. For TCGA this is bcrportionbarcode
| cases.samples.portions.updated_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]\
| cases.samples.portions.weight |  Numeric value that represents the sample portion weight, measured in milligrams
| cases.samples.portions.analytes.a260_a280_ratio |
| cases.samples.portions.analytes.amount |
| cases.samples.portions.analytes.analyte_id |
| cases.samples.portions.analytes.analyte_type |
| cases.samples.portions.analytes.analyte_type_id |
| cases.samples.portions.analytes.concentration |
| cases.samples.portions.analytes.created_datetime |
| cases.samples.portions.analytes.spectrophotometer_method |
| cases.samples.portions.analytes.state |
| cases.samples.portions.analytes.submitter_id | Submitter supplied identifier information
| cases.samples.portions.analytes.updated_datetime |
| cases.samples.portions.analytes.well_number |
| cases.samples.portions.analytes.aliquots.aliquot_id |
| cases.samples.portions.analytes.aliquots.amount |
| cases.samples.portions.analytes.aliquots.analyte_type |
| cases.samples.portions.analytes.aliquots.analyte_type_id |
| cases.samples.portions.analytes.aliquots.concentration |
| cases.samples.portions.analytes.aliquots.created_datetime |
| cases.samples.portions.analytes.aliquots.source_center |
| cases.samples.portions.analytes.aliquots.state |
| cases.samples.portions.analytes.aliquots.submitter_id | Submitter supplied identifier information
| cases.samples.portions.analytes.aliquots.updated_datetime |
| cases.samples.portions.analytes.aliquots.annotations.annotation_id |
| cases.samples.portions.analytes.aliquots.annotations.case_id |
| cases.samples.portions.analytes.aliquots.annotations.case_submitter_id | Submitter supplied identifier information
| cases.samples.portions.analytes.aliquots.annotations.category |  Top level characterization of the annotation
| cases.samples.portions.analytes.aliquots.annotations.classification |  Top level classification of the annotation
| cases.samples.portions.analytes.aliquots.annotations.created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| cases.samples.portions.analytes.aliquots.annotations.creator |  Name of the person or entity responsible for the creation of the annotation
| cases.samples.portions.analytes.aliquots.annotations.entity_id |
| cases.samples.portions.analytes.aliquots.annotations.entity_submitter_id | Submitter supplied identifier information
| cases.samples.portions.analytes.aliquots.annotations.entity_type |
| cases.samples.portions.analytes.aliquots.annotations.legacy_created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| cases.samples.portions.analytes.aliquots.annotations.legacy_updated_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| cases.samples.portions.analytes.aliquots.annotations.notes |
| cases.samples.portions.analytes.aliquots.annotations.state |
| cases.samples.portions.analytes.aliquots.annotations.status |
| cases.samples.portions.analytes.aliquots.annotations.submitter_id | Submitter supplied identifier information
| cases.samples.portions.analytes.aliquots.annotations.updated_datetime |
| cases.samples.portions.analytes.aliquots.center.center_id |
| cases.samples.portions.analytes.aliquots.center.center_type |
| cases.samples.portions.analytes.aliquots.center.code |  Numeric code for the center
| cases.samples.portions.analytes.aliquots.center.name |  Name of the center (e.g. Center For Data Intensive Science at the University of Chicago)
| cases.samples.portions.analytes.aliquots.center.namespace |  Domain name of the center (e.g. cdis.uchicago.edu)
| cases.samples.portions.analytes.aliquots.center.short_name |  Shortened name of the center (e.g. BI)
| cases.samples.portions.analytes.annotations.annotation_id |
| cases.samples.portions.analytes.annotations.case_id |
| cases.samples.portions.analytes.annotations.case_submitter_id | Submitter supplied identifier information
| cases.samples.portions.analytes.annotations.category |  Top level characterization of the annotation
| cases.samples.portions.analytes.annotations.classification |  Top level classification of the annotation
| cases.samples.portions.analytes.annotations.created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| cases.samples.portions.analytes.annotations.creator |  Name of the person or entity responsible for the creation of the annotation
| cases.samples.portions.analytes.annotations.entity_id |
| cases.samples.portions.analytes.annotations.entity_submitter_id | Submitter supplied identifier information
| cases.samples.portions.analytes.annotations.entity_type |
| cases.samples.portions.analytes.annotations.legacy_created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| cases.samples.portions.analytes.annotations.legacy_updated_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| cases.samples.portions.analytes.annotations.notes |
| cases.samples.portions.analytes.annotations.state |
| cases.samples.portions.analytes.annotations.status |
| cases.samples.portions.analytes.annotations.submitter_id | Submitter supplied identifier information
| cases.samples.portions.analytes.annotations.updated_datetime |
| cases.samples.portions.annotations.annotation_id |
| cases.samples.portions.annotations.case_id |
| cases.samples.portions.annotations.case_submitter_id | Submitter supplied identifier information
| cases.samples.portions.annotations.category |  Top level characterization of the annotation
| cases.samples.portions.annotations.classification |  Top level classification of the annotation
| cases.samples.portions.annotations.created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| cases.samples.portions.annotations.creator |  Name of the person or entity responsible for the creation of the annotation
| cases.samples.portions.annotations.entity_id |
| cases.samples.portions.annotations.entity_submitter_id | Submitter supplied identifier information
| cases.samples.portions.annotations.entity_type |
| cases.samples.portions.annotations.legacy_created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| cases.samples.portions.annotations.legacy_updated_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| cases.samples.portions.annotations.notes |
| cases.samples.portions.annotations.state |
| cases.samples.portions.annotations.status |
| cases.samples.portions.annotations.submitter_id | Submitter supplied identifier information
| cases.samples.portions.annotations.updated_datetime |
| cases.samples.portions.center.center_id |
| cases.samples.portions.center.center_type |
| cases.samples.portions.center.code |  Numeric code for the center
| cases.samples.portions.center.name |  Name of the center (e.g. Center For Data Intensive Science at the University of Chicago)  
| cases.samples.portions.center.namespace |  Domain name of the center (e.g. cdis.uchicago.edu)
| cases.samples.portions.center.short_name |  Shortened name of the center (e.g. BI)
| cases.samples.portions.slides.created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| cases.samples.portions.slides.number_proliferating_cells |  Numeric value that represents the count of proliferating cells determined during pathologic review of the sample slide(s).
| cases.samples.portions.slides.percent_eosinophil_infiltration |  Numeric value that represents the count of proliferating cells determined during pathologic review of the sample slide(s).
| cases.samples.portions.slides.percent_granulocyte_infiltration |  Numeric value to represent the percentage of infiltration by granulocytes in a tumor sample or specimen
| cases.samples.portions.slides.percent_inflam_infiltration |  Numeric value to represent local response to cellular injury, marked by capillary dilatation, edema and leukocyte infiltration; clinically, inflammation is manifest by reddness, heat, pain, swelling and loss of function, with the need to heal damaged tissue
| cases.samples.portions.slides.percent_lymphocyte_infiltration |  Numeric value to represent the percentage of infiltration by lymphocytes in a solid tissue normal sample or specimen
| cases.samples.portions.slides.percent_monocyte_infiltration |  Numeric value to represent the percentage of monocyte infiltration in a sample or specimen
| cases.samples.portions.slides.percent_necrosis |  Numeric value to represent the percentage of cell death in a malignant tumor sample or specimen
| cases.samples.portions.slides.percent_neutrophil_infiltration |  Numeric value to represent the percentage of infiltration by neutrophils in a tumor sample or specimen
| cases.samples.portions.slides.percent_normal_cells |  Numeric value to represent the percentage of normal cell content in a malignant tumor sample or specimen
| cases.samples.portions.slides.percent_stromal_cells |  Numeric value to represent the percentage of reactive cells that are present in a malignant tumor sample or specimen but are not malignant such as fibroblasts, vascular structures, etc.\
| cases.samples.portions.slides.percent_tumor_cells |  Numeric value that represents the percentage of infiltration by granulocytes in a sample
| cases.samples.portions.slides.percent_tumor_nuclei |  Numeric value to represent the percentage of tumor nuclei in a malignant neoplasm sample or specimen
| cases.samples.portions.slides.section_location |  Tissue source of the slide
| cases.samples.portions.slides.slide_id |
| cases.samples.portions.slides.state |  The current state of the object
| cases.samples.portions.slides.submitter_id |  The legacy barcode used before prior to the use UUIDs, varies by project
| cases.samples.portions.slides.updated_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]\
| cases.samples.portions.slides.annotations.annotation_id |
| cases.samples.portions.slides.annotations.case_id |
| cases.samples.portions.slides.annotations.case_submitter_id | Submitter supplied identifier information
| cases.samples.portions.slides.annotations.category |  Top level characterization of the annotation
| cases.samples.portions.slides.annotations.classification |  Top level classification of the annotation
| cases.samples.portions.slides.annotations.created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| cases.samples.portions.slides.annotations.creator |  Name of the person or entity responsible for the creation of the annotation
| cases.samples.portions.slides.annotations.entity_id |
| cases.samples.portions.slides.annotations.entity_submitter_id | Submitter supplied identifier information
| cases.samples.portions.slides.annotations.entity_type |
| cases.samples.portions.slides.annotations.legacy_created_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| cases.samples.portions.slides.annotations.legacy_updated_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| cases.samples.portions.slides.annotations.notes |  Open entry for any further description or characterization of the data
| cases.samples.portions.slides.annotations.state |  The current state of the object
| cases.samples.portions.slides.annotations.status |  Status of the annotation
| cases.samples.portions.slides.annotations.submitter_id | Submitter supplied identifier information
| cases.samples.portions.slides.annotations.updated_datetime |  A combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]
| cases.summary.file_count |
| cases.summary.file_size |
| cases.summary.data_categories.data_category |
| cases.summary.data_categories.file_count |
| cases.summary.experimental_strategies.experimental_strategy |
| cases.summary.experimental_strategies.file_count |
| cases.tissue_source_site.bcr_id |  TCGA-provided BCR id
| cases.tissue_source_site.code |  TCGA-provided TSS code
| cases.tissue_source_site.name |  Name of the source site
| cases.tissue_source_site.project |  Study name of the project
| cases.tissue_source_site.tissue_source_site_id |
| center.center_id |
| center.center_type |
| center.code |  Numeric code for the center
| center.name |  Name of the center (e.g. Center For Data Intensive Science at the University of Chicago)
| center.namespace |  Domain name of the center (e.g. cdis.uchicago.edu)
| center.short_name |  Shortened name of the center (e.g. BI)
| downstream_analyses.analysis_id |
| downstream_analyses.analysis_type |
| downstream_analyses.created_datetime |
| downstream_analyses.state |
| downstream_analyses.submitter_id | Submitter supplied identifier information
| downstream_analyses.updated_datetime |
| downstream_analyses.workflow_end_datetime |
| downstream_analyses.workflow_link |
| downstream_analyses.workflow_start_datetime |
| downstream_analyses.workflow_type |
| downstream_analyses.workflow_version |
| downstream_analyses.output_files.access |
| downstream_analyses.output_files.created_datetime |
| downstream_analyses.output_files.data_category |
| downstream_analyses.output_files.data_format |
| downstream_analyses.output_files.data_type |
| downstream_analyses.output_files.error_type |
| downstream_analyses.output_files.experimental_strategy |
| downstream_analyses.output_files.file_id |
| downstream_analyses.output_files.file_name |
| downstream_analyses.output_files.file_size |
| downstream_analyses.output_files.file_state |
| downstream_analyses.output_files.md5sum |
| downstream_analyses.output_files.platform |
| downstream_analyses.output_files.revision |
| downstream_analyses.output_files.state |
| downstream_analyses.output_files.state_comment |
| downstream_analyses.output_files.submitter_id | Submitter supplied identifier information
| downstream_analyses.output_files.updated_datetime |
| index_files.access |
| index_files.created_datetime |
| index_files.data_category |
| index_files.data_format |
| index_files.data_type |
| index_files.error_type |
| index_files.experimental_strategy |
| index_files.file_id |
| index_files.file_name |
| index_files.file_size |
| index_files.file_state |
| index_files.md5sum |
| index_files.platform |
| index_files.revision |
| index_files.state |
| index_files.state_comment |
| index_files.submitter_id | Submitter supplied identifier information
| index_files.updated_datetime |
| metadata_files.access |
| metadata_files.created_datetime |
| metadata_files.data_category |
| metadata_files.data_format |
| metadata_files.data_type |
| metadata_files.error_type |
| metadata_files.file_id |
| metadata_files.file_name |
| metadata_files.file_size |
| metadata_files.file_state |
| metadata_files.md5sum |
| metadata_files.state |
| metadata_files.state_comment |
| metadata_files.submitter_id | Submitter supplied identifier information
| metadata_files.type |
| metadata_files.updated_datetime |



### Annotation Fields


| Field Name |
| --- |
| annotation_id |
| case_id |
| case_submitter_id | Submitter supplied identifier information
| category |
| classification |
| created_datetime |
| entity_id |
| entity_submitter_id | Submitter supplied identifier information
| entity_type |
| legacy_created_datetime |
| legacy_updated_datetime |
| notes |
| state |
| status |
| submitter_id | Submitter supplied identifier information
| updated_datetime |
| project.code |
| project.dbgap_accession_number |
| project.disease_type |
| project.name |
| project.primary_site |
| project.program.dbgap_accession_number |
| project.program.name |
| project.program.program_id |
| project.project_id |
| project.released |
| project.state |

## Field Group Listing by Endpoint

### Projects Field Groups

| Field Group Name |
| --- |
| program |
| summary |
| summary.data_categories |
| summary.experimental_strategies |

### Cases Field Groups

| Field Group Name |
| --- |
| annotations |
| demographic |
| diagnoses |
| diagnoses.treatments |
| exposures |
| family_histories |
| files |
| files.analysis |
| files.analysis.input_files |
| files.analysis.metadata |
| files.analysis.metadata.read_groups |
| files.analysis.metadata.read_groups.read_group_qcs |
| files.archive |
| files.cases |
| files.cases.annotations |
| files.cases.demographic |
| files.cases.diagnoses |
| files.cases.diagnoses.treatments |
| files.cases.exposures |
| files.cases.family_histories |
| files.cases.files |
| files.cases.project |
| files.cases.project.program |
| files.cases.samples |
| files.cases.samples.annotations |
| files.cases.samples.portions |
| files.cases.samples.portions.analytes |
| files.cases.samples.portions.analytes.aliquots |
| files.cases.samples.portions.analytes.aliquots.annotations |
| files.cases.samples.portions.analytes.aliquots.center |
| files.cases.samples.portions.analytes.annotations |
| files.cases.samples.portions.annotations |
| files.cases.samples.portions.center |
| files.cases.samples.portions.slides |
| files.cases.samples.portions.slides.annotations |
| files.cases.summary |
| files.cases.summary.data_categories |
| files.cases.summary.experimental_strategies |
| files.cases.tissue_source_site |
| files.center |
| files.downstream_analyses |
| files.downstream_analyses.output_files |
| files.index_files |
| files.metadata_files |
| project |
| project.program |
| samples |
| samples.annotations |
| samples.portions |
| samples.portions.analytes |
| samples.portions.analytes.aliquots |
| samples.portions.analytes.aliquots.annotations |
| samples.portions.analytes.aliquots.center |
| samples.portions.analytes.annotations |
| samples.portions.annotations |
| samples.portions.center |
| samples.portions.slides |
| samples.portions.slides.annotations |
| summary |
| summary.data_categories |
| summary.experimental_strategies |
| tissue_source_site |

### Files Field Groups

| Field Group Name |
| --- |
| analysis |
| analysis.input_files |
| analysis.metadata |
| analysis.metadata.read_groups |
| analysis.metadata.read_groups.read_group_qcs |
| annotations |
| archive |
| associated_entities |
| cases |
| cases.annotations |
| cases.demographic |
| cases.diagnoses |
| cases.diagnoses.treatments |
| cases.exposures |
| cases.family_histories |
| cases.files |
| cases.project |
| cases.project.program |
| cases.samples |
| cases.samples.annotations |
| cases.samples.portions |
| cases.samples.portions.analytes |
| cases.samples.portions.analytes.aliquots |
| cases.samples.portions.analytes.aliquots.annotations |
| cases.samples.portions.analytes.aliquots.center |
| cases.samples.portions.analytes.annotations |
| cases.samples.portions.annotations |
| cases.samples.portions.center |
| cases.samples.portions.slides |
| cases.samples.portions.slides.annotations |
| cases.summary |
| cases.summary.data_categories |
| cases.summary.experimental_strategies |
| cases.tissue_source_site |
| center |
| downstream_analyses |
| downstream_analyses.output_files |
| index_files |
| metadata_files |

### Annotations Field Groups

| Field Group Name |
| --- |
| project |
| project.program |


# Submission

## Overview

The GDC Submission API uses methods and endpoints that are distinct from those that drive the functionality of the GDC Data Portal. In particular, data and metadata that are in the process of being submitted can only be queried using [GraphQL](#querying-submitted-data-and-metadata-using-graphql), and not the methods described in [Search and Retrieval](Search_and_Retrieval.md).

This section describes the GDC API's submission functionality, including methods for submitting, deleting, updating, searching, and retrieving data and metadata.

## Submission endpoint

### Constructing the endpoint URL

The endpoint for submitting data to a specific project in the GDC is constructed as follows:

	https://api.gdc.cancer.gov/[API_version/]submission/Program.name/Project.code

where __[API_version/]__ is the optional API version component (see [Getting Started](Getting_Started.md)).

The values of `Program.name` and `Project.code` can be obtained from the project URL on the GDC Data Submission Portal:

	https://portal.gdc.cancer.gov/submission/Program.name/Project.code/dashboard

For more information about program name and project code see [The GDC Data Model section](/Data/Data_Model/GDC_Data_Model/#program-name-project-code-and-project-id).

#### Example

The following are URL examples for a project with `Program.name` "TCGA" and `Project.code` "ALCH":

* Submission Portal URL: `https://portal.gdc.cancer.gov/submission/TCGA/ALCH/dashboard`
* API submission endpoint (versioned): `https://api.gdc.cancer.gov/v0/submission/TCGA/ALCH`
* API submission endpoint (unversioned): `https://api.gdc.cancer.gov/submission/TCGA/ALCH`

## Submission Formats

### Metadata Formats

#### JSON and TSV

The GDC API accepts project metadata in JSON and TSV formats for the purpose of creating entities in the GDC Data Model. This includes clinical and biospecimen metadata such as disease name and stage, patient age, sample type, and certain details about the types of data collected. Upon successful data submission and project release, this metadata is indexed and becomes available for queries by data users via the GDC Data Portal and the GDC API. See [GDC Data Model](#gdc-data-model) (below) for information on accepted metadata elements and instructions for obtaining templates for metadata submission.

##### Content-Type Header

JSON is the default format for metadata submission. Submission API calls with JSON payloads should include the HTTP header `Content-Type: application/json`. Requests with TSV payloads must instead include the header `Content-Type: text/tsv`.

##### Binary Mode

Metadata files must be uploaded in raw, unencoded form. Binary mode should be used, if available, to ensure that file contents are not encoded by the upload tool before transmission. For example, when using the `curl` command-line tool, the `--data-binary` switch should be used instead of `--data`. The `--data-binary` switch is required for uploading TSV files.

### Data File Formats

The GDC API accepts a variety of data files after their metadata has been registered: BAM and FASTQ files, clinical and biospecimen supplements, slide images, and other file types. Supported data file formats are listed on the [GDC Data Dictionary](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-entity-list&anchor=submittable_data_file).

## GDC Data Model

Submitters should review the [GDC Data Model documentation](../../Data/Data_Model/GDC_Data_Model.md) and the [GDC Data Dictionary](../../Data_Dictionary/index.md) before initiating submission.

### UUIDs

Submitters can assign UUIDs to all submittable entities other than those that correspond to files (entities in categories `data_file` or `metadata_file`). If the submitter does not provide a UUID, it will be assigned by the GDC and returned in the API response upon successful completion of the submission transaction. See [Appendix C](Appendix_C_Format_of_Submission_Requests_and_Responses.md) for details of the API response format. To learn more about UUIDs see the [GDC Data Model documentation](../../Data/Data_Model/GDC_Data_Model.md#uuids).

### Submitter IDs

In addition to `id`, many entities also include a `submitter_id` field. This field can contain any string (e.g. a "barcode") that the submitter wishes to use to identify the entity. Typically this string identifies a corresponding entry in submitter's records. The GDC's only requirement with respect to `submitter_id` is that it be a string that is unique for all entities within a project. The GDC Submission API requires a `submitter_id` for most entities.

>**Note:** For `case` entities, `submitter_id` must correspond to a `submitted_subject_id` of a study participant registered with the project in dbGaP.

### GDC Data Dictionary Endpoints

Information in the [GDC Data Dictionary](../../Data_Dictionary/index.md) can be accessed programmatically as described below.

#### Submission Templates

Submission templates are accessible programmatically at the `templates` endpoint. Template format (`json`, `tsv` or `csv`) is specified using the `format` parameter.

For example, the JSON template for `case` entities can be obtained from:

	https://api.gdc.cancer.gov/v0/submission/template/case?format=json

In addition to `case`, templates for the following entities can be downloaded


__Biospecimen:__
```
sample
portion
analyte
aliquot
read_group
```
__Clinical:__
```
slide
demographic
diagnosis
exposure
family_history
treatment
follow_up
molecular_test
```

__Data Files:__
```
analysis_metadata
biospecimen_supplement
clinical_supplement
experiment_metadata
pathology_report
run_metadata
slide_image
submitted_unaligned_reads
submitted_aligned_reads
submitted_genomic_profile
```

#### Entity JSON Schemas

The entire collection of GDC entity schemas can be downloaded from the `dictionary` endpoint:

	https://api.gdc.cancer.gov/v0/submission/_dictionary/_all

Individual schemas can be downloaded by specifying entity type. For example, the JSON schema for `case` entities can be found at:

	https://api.gdc.cancer.gov/v0/submission/_dictionary/case


## Making Requests to the Submission API

Requests to create or update entities in the GDC must specify the entity `type`, the entity `id` or `submitter_id`, relationships (links) that the entity has to existing entities in the [GDC Data Model](../../Data/Data_Model/GDC_Data_Model.md), and entity properties as defined by the [GDC Data Dictionary](../../Data_Dictionary/index.md). To delete entities, only the `id` property is required. The general format of GDC API submission requests and responses is provided in [Appendix C](Appendix_C_Format_of_Submission_Requests_and_Responses.md).

## Submission Transactions

Submission of data to the GDC involves a series of transactions initiated by the submitter, that create and link entities according to the [GDC Data Model](../../Data/Data_Model/GDC_Data_Model.md). With the exception of `program`, which is an administrative entity created by the GDC, all new entities must be linked, at creation, to existing entities or to new entities being created in the same transaction. For example, a submitter cannot create a `portion` entity unless the submitter either (1) has previously created the corresponding `case` and `sample` entities, or (2) is creating those entities in the same transaction. This also means that entities cannot be deleted if they have "child" entities attached to them.

If multiple entities are being created and/or updated in a transaction, and an error is encountered for one of the entities, then the transaction will fail and no changes will be made to the GDC.

### Dry Run Transactions

The `submission` endpoint provides a `_dry_run` mode that simulates submission transactions without making changes to the GDC. This mode is activated by appending `/_dry_run` to the end of a submission endpoint.

The following is an example of a POST request, that simulates creating an entity in dry run mode:

=== "Request"

    ```json
    {
      "project_id": "GDC-INTERNAL",
      "type": "case",
      "submitter_id": "GDC-INTERNAL-000093",
      "disease_type": "Blood Vessel Tumors",
      "primary_site": "Base of tongue",
       "projects": {
         "code": "INTERNAL"
      }
    }

    ```

=== "Command"

    ```shell
    token=$(<gdc-token-text-file.txt)

    curl --header "X-Auth-Token: $token" --request POST --data-binary @Request --header 'Content-Type: application/json' https://api.gdc.cancer.gov/v0/submission/GDC/INTERNAL/_dry_run
    ```

=== "Response"

    ```json
    {
      "cases_related_to_created_entities_count": 0,
      "cases_related_to_updated_entities_count": 0,
      "code": 200,
      "created_entity_count": 1,
      "entities": [
        {
          "action": "create",
          "errors": [],
          "id": "bfc1fb29-28db-4137-8379-3d1693ce3423",
          "related_cases": [],
          "type": "case",
          "unique_keys": [
            {
              "project_id": "GDC-INTERNAL",
              "submitter_id": "GDC-INTERNAL-000093"
            }
          ],
          "valid": true,
          "warnings": []
        }
      ],
      "entity_error_count": 0,
      "message": "Transaction would have been successful. User selected dry run option, transaction aborted, no data written to database.",
      "success": true,
      "transaction_id": 5834800,
      "transactional_error_count": 0,
      "transactional_errors": [],
      "updated_entity_count": 0
    }
    ```

#### Dry Run Commit

For convenience, the GDC enables users to commit earlier `_dry_run` transactions instead of uploading the same data again to execute the changes. This `commit` action is allowed on transactions that (1) have not been previously committed and (2) were successful `dry_run` transactions.

Note that the `commit` action is a separate transaction with its own transaction id, and it can be executed [asynchronously](#asynchronous-transactions). If the state of the submission project has changed in a way that would make the original `_dry_run` transaction invalid if it were run again (e.g. entities with the same `submitter_id` have since been created in another transaction), then the `commit` action will fail.

To commit a transaction, submit a POST or PUT request to `/submission/Program.name/Project.code/transactions/transaction_id/commit`, replacing `Program.name`, `Project.code`, and `transaction_id` with values associated with the transaction.

=== "Command"

    ```shell
    token=$(<gdc-token-text-file.txt)

    curl --header "X-Auth-Token: $token" --request POST https://api.gdc.cancer.gov/v0/submission/GDC/INTERNAL/transactions/467/commit?async=true
    ```

=== "Response"

    ```json
    {
      "code": 200,
      "message": "Transaction submitted.",
      "transaction_id": 468,
    }
    ```



#### Dry Run Close

The GDC Submission API also provides a `close` action on `_dry_run` transactions. This `close` action is allowed on `_dry_run` transactions that have not been previously closed. Closing a `_dry_run` transaction prevents it from being committed in the future.

To close a transaction, submit a POST or PUT request to `/submission/Program.name/Project.code/transactions/transaction_id/close`, replacing `Program.name`, `Project.code`, and `transaction_id` with values associated with the transaction.

=== "Command"

    ```shell
    token=$(<gdc-token-text-file.txt)

    curl --header "X-Auth-Token: $token" --request POST https://api.gdc.cancer.gov/v0/submission/GDC/INTERNAL/transactions/467/close
    ```

=== "Response"

    ```json
    {
        "code": 200,
        "message": "Closed transaction.",
        "transaction_id": 467
    }
    ```


### Asynchronous Transactions

The `submission` endpoint provides an asynchronous mode that provides immediate response and executes submission transactions in the background. This mode is activated by appending `?async=true` to the end of a submission endpoint.  The API will respond with the `transaction_id` which can be used to look up the result of the transaction at a later time via the [GraphQL](#querying-submitted-data-and-metadata-using-graphql) endpoint.  If the server has too many asynchronous jobs scheduled already, your request to schedule a transaction may fail.

#### Example

The following is an example of a PUT request, that creates a case asynchronously:

=== "Request"

    ```json
    {
      "project_id": "GDC-INTERNAL",
      "type": "case",
      "submitter_id": "GDC-INTERNAL-000093",
      "disease_type": "Blood Vessel Tumors",
      "primary_site": "Base of tongue",
       "projects": {
         "code": "INTERNAL"
      }
    }
    ```

=== "Command"

    ```shell
    token=$(<gdc-token-text-file.txt)

    curl --header "X-Auth-Token: $token" --request POST --data-binary @Request --header 'Content-Type: application/json' https://api.gdc.cancer.gov/v0/submission/GDC/INTERNAL?async=true
    ```

=== "Response"

    ```json
    {
      "code": 200,
      "message": "Transaction submitted.",
      "transaction_id": 467,
    }
    ```

The following is a [GraphQL](#querying-submitted-data-and-metadata-using-graphql) request that looks up the state of the above transaction:

=== "GraphQL_Request"

    ```GraphQL
    query {
      transaction_log(id: 467) {
        is_dry_run
        committed_by
        state
      }
    }
    ```

=== "GraphQL_Response"

    ```GraphQL
    {
      "data": {
        "transaction_log": [
          {
            "committed_by": null,
            "is_dry_run": false,
            "state": "FAILED"
          }
        ]
      }
    }
    ```

#### Transaction Status

The following transaction fields can be queried using [GraphQL](#querying-submitted-data-and-metadata-using-graphql) and are helpful in determining the status of a transaction:

|Field|Type|Description|
|---|---|---|
|`id`|ID|Transaction identifier|
|`is_dry_run`|Boolean|Indicates whether the transaction is a dry run|
|`closed`|Boolean|For dry run transactions, indicates whether the transaction has been closed to prevent it from being committed in the future.|
|`committable`|Boolean|Indicates whether the transaction can be committed (i.e. it is a successful dry run transaction that has not been committed previously and has not been closed)|
|`state`|String|Indicates the state of the transaction: `PENDING`, `SUCCEEDED`, `FAILED` (due to user error), or `ERRORED` (due to system error)|
|`committed_by`|ID|The ID of the transaction that committed this transaction|

>**Note:** To check whether a dry run transaction was committed successfully, check the `state` of the transaction that executed the commit. The `state` of the dry run transaction itself does not represent the status of a subsequent commit.

## Creating and Updating Entities

The GDC Submission API supports HTTP POST and HTTP PUT methods for creating entities:

* **POST** will create entities that do not exist, and will fail if any of the entities in the transaction already exist in the GDC.

* **PUT** will create new entities and update existing entities, and identify which entities were created or updated in the API response.

The GDC suggests using POST for creating new entities, and using PUT only for updating entities. This helps to avoid inadvertent entity updates that can occur when using PUT for creating entities.

>**Note:** Once a relationship has been created between two entities, it cannot be removed by updating an entity. To remove a relationship, the child entity must be [deleted](#deleting-entities).


### Example: Creating and Updating Case Entities (JSON)

In this example, a case entity is created using POST. Then an attempt is made to create the same entity again using POST, resulting in an error. Then the originally created entity is updated (with the same information) using PUT.

The JSON in the request was generated using the `case` JSON template that can be obtained from the [GDC Data Dictionary Viewer](../../Data_Dictionary/index.md) and from `https://api.gdc.cancer.gov/v0/submission/template/case?format=json`.

>**Note:** For `case` entities, `submitter_id` must correspond to a `submitted_subject_id` of a study participant registered with the project in dbGaP.


=== "Request1"

    ```json
    {
      "project_id": "GDC-INTERNAL",
      "type": "case",
      "submitter_id": "GDC-INTERNAL-000093",
      "disease_type": "Blood Vessel Tumors",
      "primary_site": "Base of tongue",
       "projects": {
         "code": "INTERNAL"
      }
    }
    ```

=== "Command1"

    ```shell
    token=$(<gdc-token-text-file.txt)

    curl --header "X-Auth-Token: $token" --request POST --data-binary @Request --header 'Content-Type: application/json' https://api.gdc.cancer.gov/v0/submission/GDC/INTERNAL
    ```

=== "Response1"

    ```json
    {
      "cases_related_to_created_entities_count": 0,
      "cases_related_to_updated_entities_count": 0,
      "code": 200,
      "created_entity_count": 1,
      "entities": [
        {
          "action": "create",
          "errors": [],
          "id": "bfc1fb29-28db-4137-8379-3d1693ce3423",
          "related_cases": [],
          "type": "case",
          "unique_keys": [
            {
              "project_id": "GDC-INTERNAL",
              "submitter_id": "GDC-INTERNAL-000093"
            }
          ],
          "valid": true,
          "warnings": []
        }
      ],
      "entity_error_count": 0,
      "message": "Transaction would have been successful. User selected dry run option, transaction aborted, no data written to database.",
      "success": true,
      "transaction_id": 5834800,
      "transactional_error_count": 0,
      "transactional_errors": [],
      "updated_entity_count": 0
    }
    ```

=== "Command2"

    ```shell
    curl --header "X-Auth-Token: $token" --request POST --data-binary @Request --header 'Content-Type: application/json' https://api.gdc.cancer.gov/v0/submission/TCGA/ALCH
    ```

=== "Response2"

    ```json
    {
      "cases_related_to_created_entities_count": 0,
      "cases_related_to_updated_entities_count": 0,
      "code": 400,
      "created_entity_count": 0,
      "entities": [
        {
          "action": null,
          "errors": [
            {
              "keys": [
                "id"
              ],
              "message": "Cannot create an entity with an id that already exists.",
              "type": "NOT_UNIQUE"
            }
          ],
          "id": null,
          "related_cases": [],
          "type": "case",
          "unique_keys": [
            {
              "project_id": "GDC-INTERNAL",
              "submitter_id": "GDC-INTERNAL-000093"
            }
          ],
          "valid": false,
          "warnings": []
        }
      ],
      "entity_error_count": 1,
      "message": "Transaction aborted due to 1 invalid entity.",
      "success": false,
      "transaction_id": 5834802,
      "transactional_error_count": 0,
      "transactional_errors": [],
      "updated_entity_count": 0
    }
    ```

=== "Command3"

    ```shell
    curl --header "X-Auth-Token: $token" --request PUT --data-binary @Request --header 'Content-Type: application/json' https://api.gdc.cancer.gov/v0/submission/GDC/INTERNAL
    ```

=== "Response3"

    ```json
    {
      "cases_related_to_created_entities_count": 0,
      "cases_related_to_updated_entities_count": 0,
      "code": 200,
      "created_entity_count": 0,
      "entities": [
        {
          "action": "update",
          "errors": [],
          "id": "a35b8e26-3b43-4203-9d33-44c2b351f177",
          "related_cases": [],
          "type": "case",
          "unique_keys": [
            {
              "project_id": "GDC-INTERNAL",
              "submitter_id": "GDC-INTERNAL-000093"
            }
          ],
          "valid": true,
          "warnings": []
        }
      ],
      "entity_error_count": 0,
      "message": "Transaction successful.",
      "success": true,
      "transaction_id": 5834803,
      "transactional_error_count": 0,
      "transactional_errors": [],
      "updated_entity_count": 1
    }
    ```



### Example: Creating an Aliquot Entity (JSON)

In this example, an `aliquot` entity and a `sample` entity are created in a single transaction. The `aliquot` is linked to `sample` which is linked to `case`. The first request is an example of using `submitter_id` properties to link entities together. The second request is an example of using UUIDs for creating the links.

#### Request 1: Creating Links Using submitter_id

=== "Request"

    ```json
    [
      {
        "type": "sample",
        "submitter_id": "GDC-INTERNAL-000093-SAMPLE000093",
        "tissue_type": "Tumor",
        "preservation_method": "Fresh",
        "specimen_type": "Whole Bone Marrow",
        "tumor_descriptor": "Primary",
        "cases": {
          "submitter_id": "GDC-INTERNAL-000093"
        }
      },
      {
        "type": "aliquot",
        "submitter_id": "GDC-INTERNAL-000093-SAMPLE000093-ALIQUOT000093",
        "samples": {
          "submitter_id": "GDC-INTERNAL-000093-SAMPLE000093"
        }
      }
    ]
    ```

=== "Command"

    ```shell
    token=$(<gdc-token-text-file.txt)
    
    curl --header "X-Auth-Token: $token" --request POST --data-binary @Request --header 'Content-Type: application/json' https://api.gdc.cancer.gov/v0/submission/GDC/INTERNAL
    ```

=== "Response"

    ```json
    {
      "cases_related_to_created_entities_count": 1,
      "cases_related_to_updated_entities_count": 0,
      "code": 201,
      "created_entity_count": 2,
      "entities": [
        {
          "action": "create",
          "errors": [],
          "id": "0a877533-0c85-4a7e-9309-733ccf295c1b",
          "related_cases": [
            {
              "id": "a00f076e-d694-47dd-8e50-24c28e90fd6a",
              "submitter_id": "GDC-INTERNAL-000093"
            }
          ],
          "type": "sample",
          "unique_keys": [
            {
              "project_id": "GDC-INTERNAL",
              "submitter_id": "GDC-INTERNAL-000093-SAMPLE000093"
            }
          ],
          "valid": true,
          "warnings": []
        },
        {
          "action": "create",
          "errors": [],
          "id": "45c57067-c92d-453b-8b6d-14a3fe08f802",
          "related_cases": [
            {
              "id": "a00f076e-d694-47dd-8e50-24c28e90fd6a",
              "submitter_id": "GDC-INTERNAL-000093"
            }
          ],
          "type": "aliquot",
          "unique_keys": [
            {
              "project_id": "GDC-INTERNAL",
              "submitter_id": "GDC-INTERNAL-000093-SAMPLE000093-ALIQUOT000093"
            }
          ],
          "valid": true,
          "warnings": []
        }
      ],
      "entity_error_count": 0,
      "message": "Transaction successful.",
      "success": true,
      "transaction_id": 5835160,
      "transactional_error_count": 0,
      "transactional_errors": [],
      "updated_entity_count": 0
    }
    ```

#### Request 2: Creating Links Using UUID

=== "Request"
    
    ```json
    [
      {
        "type": "sample",
        "submitter_id": "GDC-INTERNAL-000093-SAMPLE000093",
        "tissue_type": "Tumor",
        "preservation_method": "Fresh",
        "specimen_type": "Whole Bone Marrow",
        "tumor_descriptor": "Primary",
        "cases": {
          "id": "a00f076e-d694-47dd-8e50-24c28e90fd6a"
        }
      },
      {
        "type": "aliquot",
        "submitter_id": "GDC-INTERNAL-000093-SAMPLE000093-ALIQUOT000093",
        "samples": {
          "submitter_id": "GDC-INTERNAL-000093-SAMPLE000093"
        }
      }
    ]
    ```

=== "Command"

    ```shell
    token=$(<gdc-token-text-file.txt)
    
    curl --header "X-Auth-Token: $token" --request POST --data-binary @Request --header 'Content-Type: application/json' https://api.gdc.cancer.gov/submission/GDC/INTERNAL
    ```

=== "Response"

    ```json
    {
      "cases_related_to_created_entities_count": 1,
      "cases_related_to_updated_entities_count": 0,
      "code": 201,
      "created_entity_count": 2,
      "entities": [
        {
          "action": "create",
          "errors": [],
          "id": "9684fd7c-97b5-42a2-b350-2d86d41bbfdb",
          "related_cases": [
            {
              "id": "a00f076e-d694-47dd-8e50-24c28e90fd6a",
              "submitter_id": "GDC-INTERNAL-000093"
            }
          ],
          "type": "sample",
          "unique_keys": [
            {
              "project_id": "GDC-INTERNAL",
              "submitter_id": "GDC-INTERNAL-000093-SAMPLE000093"
            }
          ],
          "valid": true,
          "warnings": []
        },
        {
          "action": "create",
          "errors": [],
          "id": "cd5613ef-acb8-4b56-af4b-8bf3ab0e09d8",
          "related_cases": [
            {
              "id": "a00f076e-d694-47dd-8e50-24c28e90fd6a",
              "submitter_id": "GDC-INTERNAL-000093"
            }
          ],
          "type": "aliquot",
          "unique_keys": [
            {
              "project_id": "GDC-INTERNAL",
              "submitter_id": "GDC-INTERNAL-000093-SAMPLE000093-ALIQUOT000093"
            }
          ],
          "valid": true,
          "warnings": []
        }
      ],
      "entity_error_count": 0,
      "message": "Transaction successful.",
      "success": true,
      "transaction_id": 5835208,
      "transactional_error_count": 0,
      "transactional_errors": [],
      "updated_entity_count": 0
    }
    ```

### Example: Creating Two Samples (TSV)

In this example, a TSV file containing metadata for two samples is uploaded to the GDC in dry run mode.

=== "Request"

    ```
    type	project_id	submitter_id	cases.submitter_id	specimen_type	tissue_type	tumor_descriptor	preservation_method
    sample	GDC-INTERNAL	GDC-INTERNAL-000093-sampleA	GDC-INTERNAL-000093	Solid Tissue	Tumor	Primary	Frozen
    sample	GDC-INTERNAL	GDC-INTERNAL-000093-sampleB	GDC-INTERNAL-000093	Solid Tissue	Normal	Not Reported	Frozen
    ```

=== "Command"

    ```shell
    curl --header "X-Auth-Token: $token" --header 'Content-Type: text/tsv' --request PUT --data-binary @Samples.tsv 'https://api.gdc.cancer.gov/submission/GDC/INTERNAL/_dry_run'
    ```

=== "Reponse"

    ```json
    {
      "cases_related_to_created_entities_count": 1,
      "cases_related_to_updated_entities_count": 0,
      "code": 200,
      "created_entity_count": 2,
      "entities": [
        {
          "action": "create",
          "errors": [],
          "id": "c3d401f1-d505-4240-b801-dc2b389ddea1",
          "related_cases": [
            {
              "id": "a00f076e-d694-47dd-8e50-24c28e90fd6a",
              "submitter_id": "GDC-INTERNAL-000093"
            }
          ],
          "type": "sample",
          "unique_keys": [
            {
              "project_id": "GDC-INTERNAL",
              "submitter_id": "GDC-INTERNAL-000093-sampleA"
            }
          ],
          "valid": true,
          "warnings": []
        },
        {
          "action": "create",
          "errors": [],
          "id": "d3c4be95-9c69-4e8e-9f37-61db455ded7a",
          "related_cases": [
            {
              "id": "a00f076e-d694-47dd-8e50-24c28e90fd6a",
              "submitter_id": "GDC-INTERNAL-000093"
            }
          ],
          "type": "sample",
          "unique_keys": [
            {
              "project_id": "GDC-INTERNAL",
              "submitter_id": "GDC-INTERNAL-000093-sampleB"
            }
          ],
          "valid": true,
          "warnings": []
        }
      ],
      "entity_error_count": 0,
      "message": "Transaction would have been successful. User selected dry run option, transaction aborted, no data written to database.",
      "success": true,
      "transaction_id": 5835321,
      "transactional_error_count": 0,
      "transactional_errors": [],
      "updated_entity_count": 0
    }
    ```

### Example: Updating a Sample Entity (JSON)

Entities can be updated using a very similar process to what is shown above.  

#### Updating a sample

New nodes are created in Request1.  Nodes in state `validated` are updated in Request2.

=== "Request1"

    ```json
    [
       {
        "type": "case",
        "submitter_id": "QA-REGRESSION-0002",  
        "projects": {
        "code": "REGRESSION"
      }
        },
        {
        "type": "sample",
        "submitter_id": "QA-REGRESSION-0002-SAMPLE000001",
        "sample_type": "Primary Tumor",
        "sample_type_id": "01",
        "cases": {
          "submitter_id": "QA-REGRESSION-0002"
        }
      },
      {
        "type": "aliquot",
        "submitter_id": "QA-REGRESSION-0002-SAMPLE000001-ALIQUOT000001",
        "samples": {
          "submitter_id": "QA-REGRESSION-0002-SAMPLE000001"
        }
      }
    ]
    ```
    
=== "Conmmand1"

    ```shell
    token=$(<gdc-token-text-file.txt)

    curl --header "X-Auth-Token: $token" --request POST --data-binary @sample.json --header 'Content-Type: application/json' https://api.gdc.cancer.gov/v0/submission/QA/REGRESSION
    ```

=== "Response1"

    ```json
    {
      "cases_related_to_created_entities_count": 1,
      "cases_related_to_updated_entities_count": 0,
      "code": 201,
      "created_entity_count": 3,
      "entities": [
        {
          "action": "create",
          "errors": [],
          "id": "3a750ae8-8e63-472e-852e-8e514a0c1550",
          "related_cases": [],
          "type": "case",
          "unique_keys": [
            {
              "project_id": "QA-REGRESSION",
              "submitter_id": "QA-REGRESSION-0002"
            }
          ],
          "valid": true,
          "warnings": []
        },
        {
          "action": "create",
          "errors": [],
          "id": "8a1872e6-c5e6-4f39-b9fe-15ecf45715c7",
          "related_cases": [
            {
              "id": "3a750ae8-8e63-472e-852e-8e514a0c1550",
              "submitter_id": "QA-REGRESSION-0002"
            }
          ],
          "type": "sample",
          "unique_keys": [
            {
              "project_id": "QA-REGRESSION",
              "submitter_id": "QA-REGRESSION-0002-SAMPLE000001"
            }
          ],
          "valid": true,
          "warnings": []
        },
        {
          "action": "create",
          "errors": [],
          "id": "e9279137-92b4-41ab-be28-a03e32e6fac7",
          "related_cases": [
            {
              "id": "3a750ae8-8e63-472e-852e-8e514a0c1550",
              "submitter_id": "QA-REGRESSION-0002"
            }
          ],
          "type": "aliquot",
          "unique_keys": [
            {
              "project_id": "QA-REGRESSION",
              "submitter_id": "QA-REGRESSION-0002-SAMPLE000001-ALIQUOT000001"
            }
          ],
          "valid": true,
          "warnings": []
        }
      ],
      "entity_error_count": 0,
      "message": "Transaction successful.",
      "success": true,
      "transaction_id": 920117,
      "transactional_error_count": 0,
      "transactional_errors": [],
      "updated_entity_count": 0
    }
    ```

=== "Request2"

    ```json
    [
       {
        "type": "case",
        "submitter_id": "QA-REGRESSION-0002",  
        "projects": {
        "code": "REGRESSION"
      }
        },
        {
        "type": "sample",
        "submitter_id": "QA-REGRESSION-0002-SAMPLE000001",
        "sample_type": "Primary Tumor",
        "days_to_collection":5,
        "sample_type_id": "01",
        "cases": {
          "submitter_id": "QA-REGRESSION-0002"
        }
      },
      {
        "type": "aliquot",diff
        "submitter_id": "QA-REGRESSION-0002-SAMPLE000001-ALIQUOT000001",
        "samples": {
          "submitter_id": "QA-REGRESSION-0002-SAMPLE000001"
        }
      }
    ]
    ```

=== "Command2"

    ```shell
    token=$(<gdc-token-text-file.txt)

    curl --header "X-Auth-Token: $token" --request PUT --data-binary @sample2.json --header 'Content-Type: application/json' https://api.gdc.cancer.gov/v0/submission/QA/REGRESSION
    ```
    
=== "Response2"

    ```json
    {
      "cases_related_to_created_entities_count": 0,
      "cases_related_to_updated_entities_count": 1,
      "code": 200,
      "created_entity_count": 0,
      "entities": [
        {
          "action": "update",
          "errors": [],
          "id": "3a750ae8-8e63-472e-852e-8e514a0c1550",
          "related_cases": [],
          "type": "case",
          "unique_keys": [
            {
              "project_id": "QA-REGRESSION",
              "submitter_id": "QA-REGRESSION-0002"
            }
          ],
          "valid": true,
          "warnings": []
        },
        {
          "action": "update",
          "errors": [],
          "id": "8a1872e6-c5e6-4f39-b9fe-15ecf45715c7",
          "related_cases": [
            {
              "id": "3a750ae8-8e63-472e-852e-8e514a0c1550",
              "submitter_id": "QA-REGRESSION-0002"
            }
          ],
          "type": "sample",
          "unique_keys": [
            {
              "project_id": "QA-REGRESSION",
              "submitter_id": "QA-REGRESSION-0002-SAMPLE000001"
            }
          ],
          "valid": true,
          "warnings": []
        },
        {
          "action": "update",
          "errors": [],
          "id": "e9279137-92b4-41ab-be28-a03e32e6fac7",
          "related_cases": [
            {
              "id": "3a750ae8-8e63-472e-852e-8e514a0c1550",
              "submitter_id": "QA-REGRESSION-0002"
            }
          ],
          "type": "aliquot",
          "unique_keys": [
            {
              "project_id": "QA-REGRESSION",
              "submitter_id": "QA-REGRESSION-0002-SAMPLE000001-ALIQUOT000001"
            }
          ],
          "valid": true,
          "warnings": []
        }
      ],
      "entity_error_count": 0,
      "message": "Transaction successful.",
      "success": true,
      "transaction_id": 920120,
      "transactional_error_count": 0,
      "transactional_errors": [],
      "updated_entity_count": 3
    }
    ```

## Retrieving Entities

### Entities Endpoint

JSON objects representing submitted entities can be retrieved with a GET request to the `entities` endpoint. This endpoint retrieves entities by UUID. A single UUID or a comma-separated list of UUIDs can be passed to this endpoint as a query.

=== "Command"
    
    ```shell
    token=$(<gdc-token-text-file.txt)
    
    curl --header "X-Auth-Token: $token" https://api.gdc.cancer.gov/v0/submission/TCGA/ALCH/entities/fbf69646-5904-4f95-92d6-692bde658f05
    ```

=== "Response"

    ```json
    {
      "entities": [
        {
          "program": "TCGA",
          "project": "ALCH",
          "properties": {
            "created_datetime": "2016-04-14T08:44:43.361800-05:00",
            "id": "fbf69646-5904-4f95-92d6-692bde658f05",
            "project_id": "TCGA-ALCH",
            "projects": [
              {
                "id": "d9906779-f1da-5d9f-9caa-6d5ecb2e3cd6",
                "submitter_id": null
              }
            ],
            "state": "validated",
            "submitter_id": "TCGA-ALCH-000001",
            "type": "case",
            "updated_datetime": "2016-04-14T21:29:28.401212-05:00"
          }
        }
      ]
    }
    ```

### Export Endpoint

The `export` endpoint provides additional functionality for exporting entities from the GDC submission system. The `ids` parameter accepts a UUID or a comma-separated list of UUIDs. The `format` parameter allows the user to specify the preferred format of the API response: JSON, TSV, or CSV. When the `with_children` parameter is set to `with_children`, the response includes the metadata stored in all "child" entities of the entity being requested. The `export` endpoint accepts GET requests.

=== "Command"

    ```shell
    token=$(<gdc-token-text-file.txt)


    curl --header "X-Auth-Token: $token" 'https://api.gdc.cancer.gov/v0/submission/TCGA/ALCH/export?ids=11f8321-832f-4a8b-8384-a2f6256557e0&format=json&with_children=with_children'
    ```

=== "Response"

    ```json
    {
      "case": [
        {
          "tissue_source_sites": [],
          "submitter_id": "TCGA-ALCH-000026",
          "project_id": "TCGA-ALCH",
          "type": "case",
          "id": "11f83251-832f-4a8b-8384-a2f6256557e0",
          "projects": [
            {
              "code": "ALCH",
              "id": "d9906779-f1da-5d9f-9caa-6d5ecb2e3cd6"
            }
          ]
        }
      ],
      "sample": [
        {
          "sample_type_id": "10",
          "time_between_excision_and_freezing": null,
          "oct_embedded": "false",
          "tumor_code_id": null,
          "submitter_id": "Blood-00001_api26",
          "intermediate_dimension": null,
          "id": "23308708-6a63-471e-947c-6a93c6e85983",
          "time_between_clamping_and_freezing": null,
          "pathology_report_uuid": null,
          "tumor_descriptor": null,
          "sample_type": "Blood Derived Normal",
          "project_id": "TCGA-ALCH",
          "current_weight": null,
          "composition": null,
          "is_ffpe": null,
          "shortest_dimension": null,
          "tumor_code": null,
          "tissue_type": null,
          "days_to_sample_procurement": null,
          "cases": [
            {
              "id": "11f83251-832f-4a8b-8384-a2f6256557e0",
              "submitter_id": "TCGA-ALCH-000026"
            }
          ],
          "freezing_method": null,
          "type": "sample",
          "preservation_method": null,
          "days_to_collection": null,
          "initial_weight": null,
          "longest_dimension": null
        }
      ],
      "read_group": [
        {
          "library_name": "Solexa-34688",
          "is_paired_end": true,
          "size_selection_range": null,
          "adapter_sequence": null,
          "library_strand": null,
          "submitter_id": "Blood-00001-aliquot_lane1_barcode26",
          "library_preparation_kit_name": null,
          "adapter_name": null,
          "target_capture_kit_name": null,
          "includes_spike_ins": null,
          "library_preparation_kit_version": null,
          "id": "90163202-cfd7-4f6a-8214-e7e4e924d3a6",
          "spike_ins_concentration": null,
          "target_capture_kit_vendor": null,
          "read_length": 75,
          "sequencing_date": "2010-08-04",
          "spike_ins_fasta": null,
          "to_trim_adapter_sequence": null,
          "RIN": null,
          "platform": "Illumina",
          "library_selection": "Hybrid_Selection",
          "library_strategy": "WXS",
          "library_preparation_kit_catalog_number": null,
          "target_capture_kit_target_region": null,
          "fastq_name": null,
          "target_capture_kit_version": null,
          "aliquots": [
            {
              "id": "e66dee54-5f4c-4471-9e08-dba0f6cdaaa4",
              "submitter_id": "Blood-00001-aliquot26"
            }
          ],
          "read_group_name": "205DD.3-2",
          "library_preparation_kit_vendor": null,
          "project_id": "TCGA-ALCH",
          "type": "read_group",
          "target_capture_kit_catalog_number": null,
          "instrument_model": "Illumina HiSeq 2000",
          "base_caller_name": null,
          "experiment_name": "Resequencing",
          "flow_cell_barcode": "205DDABXX",
          "sequencing_center": "BI",
          "base_caller_version": null
        }
      ],
      "aliquot": [
        {
          "source_center": "23",
          "centers": [],
          "analytes": [],
          "submitter_id": "Blood-00001-aliquot26",
          "amount": 10,
          "samples": [
            {
              "id": "23308708-6a63-471e-947c-6a93c6e85983",
              "submitter_id": "Blood-00001_api26"
            }
          ],
          "concentration": 0.07,
          "project_id": "TCGA-ALCH",
          "type": "aliquot",
          "id": "e66dee54-5f4c-4471-9e08-dba0f6cdaaa4"
        }
      ],
      "submitted_unaligned_reads": [
        {
          "read_groups": [
            {
              "id": "90163202-cfd7-4f6a-8214-e7e4e924d3a6",
              "submitter_id": "Blood-00001-aliquot_lane1_barcode26"
            }
          ],
          "data_type": "Unaligned Reads",
          "file_name": "dummy.fastq",
          "md5sum": "70c48a8a670ed2a02327601a10038d06",
          "data_format": "FASTQ",
          "submitter_id": "Blood-00001-aliquot_lane1_barcode26.fastq",
          "state_comment": null,
          "data_category": "Sequencing Data",
          "file_size": 38,
          "project_id": "TCGA-ALCH",
          "type": "submitted_unaligned_reads",
          "id": "6d45f2a0-8161-42e3-97e6-e058ac18f3f3",
          "experimental_strategy": "WGS"
        },
        {
          "read_groups": [
            {
              "id": "90163202-cfd7-4f6a-8214-e7e4e924d3a6",
              "submitter_id": "Blood-00001-aliquot_lane1_barcode26"
            }
          ],
          "data_type": "Unaligned Reads",
          "file_name": "dummy.fastq",
          "md5sum": "70c48a8a670ed2a02327601a10038d06",
          "data_format": "FASTQ",
          "submitter_id": "Blood-00001-aliquot_lane1_barcode27.fastq",
          "state_comment": null,
          "data_category": "Sequencing Data",
          "file_size": 38,
          "project_id": "TCGA-ALCH",
          "type": "submitted_unaligned_reads",
          "id": "4faabdd6-45bb-4259-8868-13d5b1149748",
          "experimental_strategy": "WGS"
        }
      ]
    }
    ```

### GraphQL

Submitters can use the GraphQL query language for advanced search and retrieval of data from the GDC Submission Portal. See [GraphQL](#querying-submitted-data-using-graphql) for more information.

## Patching Entitites

The GDC Submission API supports the HTTP PATCH method for updating existing entities with additional fields.

**PATCH** can be used to add extra fields to an existing entity, without requiring the submission of required fields.

The PATCH method cannot be used to create new entities, and the provided submitter_id must match an existing submitter_id.

#### Example: Creating a new demographic entity using POST

=== "Request1"

    ```json
    {
      "type": "demographic",
      "submitter_id": "demographic7892",
      "cases": {
        "submitter_id": "GDC-INTERNAL-000073"
      }, 
      "ethnicity": "not reported",
      "gender": "male",
      "race": "white",
      "vital_status": "Dead"
    }
    ```

=== "Command1"

  ```shell
  token=$(<gdc-token-text-file.txt)

  curl --header "X-Auth-Token: $token" --request POST --data-binary @Request --header 'Content-Type: application/json' https://api.gdc.cancer.gov/v0/submission/GDC/INTERNAL
  ```

=== "Response1"

    ```json
    {
      "cases_related_to_created_entities_count": 1,
      "cases_related_to_updated_entities_count": 0,
      "code": 201,
      "created_entity_count": 1,
      "entities": [
        {
          "action": "create",
          "errors": [],
          "id": "4e4f29a3-5325-47ef-a583-a251677ed29a",
          "related_cases": [
            {
              "id":"71d17c1f-8985-4b2f-bb63-1c39cb6562d5",
              "submitter_id":"GDC-INTERNAL-000073"
            }
          ],
          "type": "demographic",
          "unique_keys": [
            {
              "project_id": "GDC-INTERNAL",
              "submitter_id": "demographic7892"
            }
          ],
          "valid": true,
          "warnings": []
        }
      ],
      "entity_error_count": 0,
      "message": "Transaction successful.",
      "success": true,
      "transaction_id": 6357750,
      "transactional_error_count": 0,
      "transactional_errors": [],
      "updated_entity_count": 0
    }
    ```

#### Example: Updating the existing demographic entity using PATCH

=== "Request2"

  ```json
  {
    "type": "demographic",
    "submitter_id": "demographic7892",
    "cause_of_death": "Infection", 
    "cause_of_death_source": "Death Certificate", 
    "country_of_birth": "Antigua and Barbuda", 
    "country_of_residence_at_enrollment": "Antigua and Barbuda"
  }
  ```

=== "Command2"

    ```shell
    token=$(<gdc-token-text-file.txt)

    curl --header "X-Auth-Token: $token" --request PATCH --data-binary @Request --header 'Content-Type: application/json' https://api.gdc.cancer.gov/v0/submission/GDC/INTERNAL
    ```

"Response2"

    ```json
    {
      "cases_related_to_created_entities_count": 0,
      "cases_related_to_updated_entities_count": 1,
      "code": 200,
      "created_entity_count": 0,
      "entities": [
        {
          "action": "update",
          "errors": [],
          "id": "4e4f29a3-5325-47ef-a583-a251677ed29a",
          "related_cases": [
            {
              "id":"71d17c1f-8985-4b2f-bb63-1c39cb6562d5",
              "submitter_id":"GDC-INTERNAL-000073"
            }
          ],
          "type": "demographic",
          "unique_keys": [
            {
              "project_id": "GDC-INTERNAL",
              "submitter_id": "demographic7892"
            }
          ],
          "valid": true,
          "warnings": []
        }
      ],
      "entity_error_count": 0,
      "message": "Transaction successful.",
      "success": true,
      "transaction_id": 6357751,
      "transactional_error_count": 0,
      "transactional_errors": [],
      "updated_entity_count": 1
    }
    ```


## Deleting Entities

The `entities` endpoint can also be used to delete entities. This is accomplished using a DELETE request to the endpoint, specifying the entity's UUID. If an entity cannot be deleted because it is linked to child entities, the GDC Submission API will respond with an error providing a list of entities that must be deleted prior to deleting the subject entity.

A subgraph (a parent along with all of its child entities) can be deleted in a single transaction by passing a comma-separated list of UUIDs to the `entities` endpoint.

Entities in submitted state (assigned when the project has been submitted) cannot be deleted.

=== "Shell"

    ```Shell
    token=$(<gdc-token-text-file.txt)

    curl --header "X-Auth-Token: $token" --request DELETE https://api.gdc.cancer.gov/v0/submission/TCGA/ALCH/entities/67782964-0065-491d-b051-2ae404bb734d
    ```

=== "Response"

    ```json
    {
      "code": 200,
      "deleted_entity_count": 1,
      "dependent_ids": "",
      "entities": [
        {
          "action": "delete",
          "errors": [],
          "id": "67782964-0065-491d-b051-2ae404bb734d",
          "related_cases": [],
          "type": "case",
          "valid": true,
          "warnings": []
        }
      ],
      "entity_error_count": 0,
      "message": "Successfully deleted 1 entities",
      "success": true,
      "transaction_id": 192,
      "transactional_error_count": 0,
      "transactional_errors": []
    }
    ```

## Working With Files

### Uploading Data Files

Experimental data files like BAM and FASTQ can be uploaded directly to the API using the `files` endpoint, by specifying the UUID of the corresponding `data_file` entity. Binary upload mode must be used if available. Uploading large files may be more efficiently performed using the [GDC Data Transfer Tool](/Data_Transfer_Tool/Users_Guide/Getting_Started.md).

```
token=$(<gdc-token-text-file.txt)

curl --header "X-Auth-Token: $token" --output needed_to_show_progress_bar.log --request PUT --data-binary @GDC-INTERNAL-000084-S1-Q1-RG1.fastq.zip https://api.gdc.cancer.gov/v0/submission/GDC/INTERNAL/files/c414a205-376e-4993-af48-2a4689eb433e && rm needed_to_show_progress_bar.log

	# "&& rm needed_to_show_progress_bar.log" at the end of the command above
	# removes the temporary file required to show upload progress bar. This
	# will not work on Windows platforms. Windows users must remove this
	# string and can delete the file manually.
```

#### Upload Manifest

The `manifest` endpoint generates a manifest for uploading files using the GDC Data Transfer Tool. It requires a comma-separated list of file UUIDs to generate a manifest.

```
https://api.gdc.cancer.gov/v0/submission/PROGRAM/PROJECT/manifest?ids=bf0751ca-fc3b-4760-b876-0fefce040be5,90163202-cfd7-4f6a-8214-e7e4e924d3a6
```
### Uploading New Versions of Data Files

The GDC Submission system supports submitting updated versions of files.  For example, you may want to submit an updated version of a clinical supplement file that contains new clinical information about a patient.  If a file is in file_state `validated` then you would simply delete and upload a new copy of this file.  No additional version of the file will be created in this case.  The UUID of the node stays the same.

However, if a file is in file_state `submitted` or `validated` and state `released` then a different process is required.  In this situation simply upload a new template containing updated metadata (e.g. md5sum or file_size).  A new node (with a new UUID) will automatically be created that is linked to the previous version.  Once this new file is indexed and released to users they will be able to query the new UUID in the /files endpoint and both versions' UUID in the files/versions or /history endpoint.  In the example below we register a file, upload the file, and register a new version of this file.

=== "Request"

    ```json
    [
      {
        "data_type": "Clinical Supplement",
        "file_name": "nationwidechildrens.org_clinical.TCGA-4G-AAZT-.xml",
        "md5sum": "ecaaa87613ba03c971bfefdb6f693959",
        "data_format": "BCR XML",
        "submitter_id": "nationwidechildrens.org_CHOL.bio.Level_1.428.25.0.tar.gz_nationwidechildrens.org_clinical.TCGA-4G-AAZT-.xml",
        "archives": [],
        "data_category": "Clinical",
        "file_size": 39195,
        "cases": [
          {
            "id": "b10c64c2-7fd2-4210-b975-034affb14b57",
            "submitter_id": "TCGA-4G-AAZT"
          }
        ],
        "project_id": "TCGA-CHOL",
        "type": "clinical_supplement"
      }
    ]
    ```

=== "Command"

    ```shell
    curl --header "X-Auth-Token: $token" --header 'Content-Type: json' --request PUT --data-binary @clin.json 'https://api.gdc.cancer.gov/submission/TCGA/CHOL'
    ```

=== "Response"

    ```json
    {
      "cases_related_to_created_entities_count": 1,
      "cases_related_to_updated_entities_count": 0,
      "code": 200,
      "created_entity_count": 1,
      "entities": [
        {
          "action": "create",
          "errors": [],
          "id": "d65c15d9-9e33-4a0b-863d-605ad6155506",
          "related_cases": [
            {
              "id": "b10c64c2-7fd2-4210-b975-034affb14b57",
              "submitter_id": "TCGA-4G-AAZT"
            }
          ],
          "type": "clinical_supplement",
          "unique_keys": [
            {
              "project_id": "TCGA-CHOL",
              "submitter_id": "nationwidechildrens.org_CHOL.bio.Level_1.428.25.0.tar.gz_nationwidechildrens.org_clinical.TCGA-4G-AAZT-.xml"
            }
          ],
          "valid": true,
          "warnings": []
        }
      ],
      "entity_error_count": 0,
      "message": "Transaction successful.",
      "success": true,
      "transaction_id": 922606,
      "transactional_error_count": 0,
      "transactional_errors": [],
      "updated_entity_count": 0
    }
    ```

=== "Request2"

    ```json
    [
      {
        "data_type": "Clinical Supplement",
        "file_name": "nationwidechildrens.org_clinical.TCGA-4G-AAZT-.xml",
        "md5sum": "93e306e5e621d3cacb363e2be96ca3cd",
        "data_format": "BCR XML",
        "submitter_id": "nationwidechildrens.org_CHOL.bio.Level_1.428.25.0.tar.gz_nationwidechildrens.org_clinical.TCGA-4G-AAZT-.xml",
        "archives": [],
        "data_category": "Clinical",
        "file_size": 39197,
        "cases": [
          {
            "id": "b10c64c2-7fd2-4210-b975-034affb14b57",
            "submitter_id": "TCGA-4G-AAZT"
          }
        ],
        "project_id": "TCGA-CHOL",
        "type": "clinical_supplement"
      }
    ]
    ```

=== "Command2"

    ```shell
    curl --header "X-Auth-Token: $token" --header 'Content-Type: json' --request PUT --data-binary @clin_v2.json 'https://api.gdc.cancer.gov/submission/TCGA/CHOL'
    ```

=== "Response2"

    ```json
    {
      "cases_related_to_created_entities_count": 0,
      "cases_related_to_updated_entities_count": 0,
      "code": 200,
      "created_entity_count": 0,
      "entities": [
        {
          "action": "version",
          "errors": [],
          "id": "32e9fd2c-877a-4700-a06f-bb34e0590ca5",
          "related_cases": [
            {
              "id": "b10c64c2-7fd2-4210-b975-034affb14b57",
              "submitter_id": "TCGA-4G-AAZT"
            }
          ],
          "type": "clinical_supplement",
          "unique_keys": [
            {
              "project_id": "TCGA-CHOL",
              "submitter_id": "nationwidechildrens.org_CHOL.bio.Level_1.428.25.0.tar.gz_nationwidechildrens.org_clinical.TCGA-4G-AAZT-.xml"
            }
          ],
          "valid": true,
          "warnings": []
        }
      ],
      "entity_error_count": 0,
      "message": "Transaction successful.",
      "success": true,
      "transaction_id": 922607,
      "transactional_error_count": 0,
      "transactional_errors": [],
      "updated_entity_count": 0
    }
    ```

### Downloading Files

Files in `file_state = validated` can be downloaded by the submitter using the API or the Data Transfer Tool. This is done in a similar manner as files available in the Data Portal, but will require submission access to the particular project in dbGaP as opposed to downloader access.  File UUIDs can be found in the original upload manifest file, the submission portal, or by API calls.  See [Downloading Files](Downloading_Files.md) for details.

### Deleting Files

Uploaded files must be deleted using a two step process.  First, the file is deleted using the Data Transfer Tool.  See [Deleting Previously Uploaded Data](/Data_Transfer_Tool/Users_Guide/Data_Download_and_Upload/#deleting-previously-uploaded-data) for details.

Second, the file node can be deleted or modified. See [Deleting Entities](#deleting-entities) for details.

## Querying Submitted Data Using GraphQL

### GraphQL Overview

[GraphQL](https://graphql.org/) is a query language that makes it easy to search and retrieve data from graph data structures such as the GDC Data Model.

Unlike the methods outlined in [Search and Retrieval](Search_and_Retrieval.md), which provide access to public releases (or snapshots) of GDC data, the `/graphql` endpoint of GDC Submission API makes it possible for submitters to access "live" data, which provides a real-time view of the state of entities in a project.

>**NOTE:** Access to GDC Submission API GraphQL service is limited to authorized and authenticated submitters. Submitters may only access data in their own project using GraphQL.


### GraphQL IDE

The GDC GraphQL IDE is an instance of [GraphiQL](https://github.com/graphql/graphiql), an in-browser GraphQL IDE that facilitates construction and execution of GraphQL queries. The GDC GraphQL IDE provides tab-completion and syntax checking using schema from the GDC Data Dictionary. It can be found at [https://portal.gdc.cancer.gov/submission/graphiql](https://portal.gdc.cancer.gov/submission/graphiql).

Before interacting directly with the GDC Submission API's GraphQL endpoint, users are encouraged to become familiar with executing queries using the GDC GraphQL IDE.

### GraphQL Endpoint

GDC data submitters can access the GDC Submission API GraphQL endpoint at:

	https://api.gdc.cancer.gov/[API_version/]submission/graphql

where __[API_version/]__ is the optional API version component (see [Getting Started](Getting_Started.md)).

>**NOTE:** An authentication token is required for all requests to the `graphql` endpoint. Queries are restricted to those projects for which the submitter has obtained authorization.


### Constructing a Query

When sending GraphQL requests to the API directly, the bare GraphQL query must be wrapped in a "query" JSON object as shown below:

<pre>
{
	"query": "<b>{Bare_GraphQL_Query}</b>",
	"variables": null
}
</pre>

When using the GDC GraphQL IDE, the bare JSON query must be used without a JSON wrapper.

#### Bare GraphQL query

In its simplest form, a GraphQL query is a **selection set** (curly brackets) that encloses a set of **fields**. The selection set defines the set of information that is to be retrieved. Furthermore, in GraphQL fields are conceptually equivalent to functions that retrieve additional fields and, in some cases, can take arguments. So each field in a selection set can have its own selection set, thereby creating a nested query structure that can navigate complex data relationships. See [GraphQL Specification](https://facebook.github.io/graphql/) for further details.

In GDC GraphQL IDE, a root field (field within the outermost/umbrella selection set) typically corresponds to an entity, whereas fields inside nested selection sets are typically a combination of entities and entity properties.

The "Docs" panel on the right-hand side of the GDC GraphQL IDE allows users to discover the fields that can be queried with GraphQL. Note that the panel contains a lot of information and users may experience a delay before it is displayed.

A simple GraphQL query looks like this:

	{
	  case (project_id: "TCGA-ALCH", first: 0) {
	    id
	    submitter_id

	  }
	  _case_count (project_id: "TCGA-ALCH")
	}

[//]: # (this is just a comment ignore me I beg of you_)


The query above has two root fields: `case` and `_case_count`. The `case` field corresponds to the `case` entity in the GDC Data Model. The query supplies two arguments to the field:

1. `project_id: "TCGA-ALCH"`, which requests only cases in the TCGA-ALCH project.
2. `first: 0`, which requests that the API provide all results in the response, without pagination ( a nonzero positive integer value of `first` specifies the number of results to return, 10 by default; "pages" are selected using `offset`).

The `_case_count` field is a special field that returns the number of cases that match the supplied argument.

The bare query above can be used as is in the GraphQL IDE. In order to pass this query to the GDC API directly, it needs to be further processed as described below.

#### Passing GraphQL queries to GDC API directly

Before a bare GraphQL query is passed to the GDC API, it must be processed as follows:

1. [Escape](https://www.freeformatter.com/json-escape.html) the query using JSON string rules
2. Wrap the query in a ["query" JSON object](#constructing-a-query).
3. Pass the query to the `graphql` endpoint in an HTTP POST request.

Using the `case` and `_case_count` example above as the starting point, the results are as follows:

=== "bare_GraphQL"

    ```GraphQL
    {
    	case (project_id: "TCGA-ALCH", first: 0) {
    		id
    		submitter_id

    	}
    	_case_count (project_id: "TCGA-ALCH")
    }
    ```

=== "escaped_GraphQL"

    ```GraphQL
    {\n\tcase (project_id: \"TCGA-ALCH\", first: 0) {\n\t\tid\n\t\tsubmitter_id\n\n\t}\n\t_case_count (project_id: \"TCGA-ALCH\")\n}
    ```
    
=== "Query_json"

    ```json
    {
    	"query": "{\n\tcase (project_id: \"TCGA-ALCH\", first: 0) {\n\t\tid\n\t\tsubmitter_id\n\n\t}\n\t_case_count (project_id: \"TCGA-ALCH\")\n}",
    	"variables": null
    }
    ```

=== "Shell_command"

    ```shell
    token=$(<gdc-token-text-file.txt)

    curl --request POST --header "X-Auth-Token: $token" 'https://api.gdc.cancer.gov/v0/submission/graphql' --data-binary @Query_json
    ```

=== "API_Response"

    ```json
    {
      "data": {
        "_case_count": 20,
        "case": [
          {
            "id": "700d1110-b6b4-4251-89d4-fa6f0698e3f8",
            "submitter_id": "TCGA-ALCH-000004"
          },
          {
            "id": "be01357d-7348-40b4-a997-8a61ae7af17d",
            "submitter_id": "TCGA-ALCH-000005"
          },
          {
            "id": "e5638697-6ef3-4bf8-a373-102519093f33",
            "submitter_id": "TCGA-ALCH-000008"
          },
          {
            "id": "4871d41a-680e-4fd0-901c-b06f06ecae33",
            "submitter_id": "TCGA-ALCH-000007"
          },
          {
            "id": "2f18c2c1-bff2-43b6-9702-e138c72d8c6b",
            "submitter_id": "TCGA-ALCH-000009"
          },
          {
            "id": "ec83e038-4f01-47a6-bc69-47fb297d0282",
            "submitter_id": "TCGA-ALCH-000006"
          },
          {
            "id": "e4642952-d259-4be1-9c53-ed95aa1fc50b",
            "submitter_id": "TCGA-ALCH-000011"
          },
          {
            "id": "8bcaf0b3-21d0-45c6-87ee-c997efb417dc",
            "submitter_id": "TCGA-ALCH-000010"
          },
          {
            "id": "83de027e-bcbf-4239-975b-7e8ced82448e",
            "submitter_id": "TCGA-ALCH-000013"
          },
          {
            "id": "bbd91cc1-06e2-4e60-8b93-e09c3b16f00c",
            "submitter_id": "TCGA-ALCH-000014"
          },
          {
            "id": "574fd163-4368-440c-9548-d76a0fbc9056",
            "submitter_id": "TCGA-ALCH-000015"
          },
          {
            "id": "47c92cdd-ff11-4c25-b0f0-0f7671144271",
            "submitter_id": "TCGA-ALCH-000016"
          },
          {
            "id": "9f13caab-1fda-4b2a-b500-f79dc978c6c1",
            "submitter_id": "TCGA-ALCH-000017"
          },
          {
            "id": "9418f194-8741-44db-bd8f-36f4fd8c3bf2",
            "submitter_id": "TCGA-ALCH-000018"
          },
          {
            "id": "6fb2a018-c5f3-45e5-81d3-e58e7e4bf921",
            "submitter_id": "TCGA-ALCH-000019"
          },
          {
            "id": "70236972-e796-414a-9b7a-3b29b849ba7c",
            "submitter_id": "TCGA-ALCH-000020"
          },
          {
            "id": "6f78e86f-9e31-4af5-a0d9-b8970ece476d",
            "submitter_id": "TCGA-ALCH-000021"
          },
          {
            "id": "c6fcb2f0-c6bb-4b40-a761-bae3e63869cb",
            "submitter_id": "TCGA-ALCH-000002"
          },
          {
            "id": "67782964-0065-491d-b051-2ae404bb734d",
            "submitter_id": "TCGA-ALCH-000001"
          },
          {
            "id": "b45d2891-ba81-4ecc-a250-c58060934227",
            "submitter_id": "TCGA-ALCH-000012"
          }
        ]
      }
    }
    ```

### Additional Examples

#### Example: File UUID

GraphQL query to find the file UUID based on file `submitter_id`:

=== "bare_GraphQL"

    ```GraphQL
    {

      submitted_unaligned_reads (project_id: "GDC-INTERNAL", submitter_id: "Blood-00001-aliquot_lane1_barcode23.fastq") {
        id
        submitter_id
        file_name
        project_id
    }
    }
    ```

=== "escaped_GraphQL"

    ```GraphQL
    {
        "query": "{\n \n  submitted_unaligned_reads (project_id: \"GDC-INTERNAL\", submitter_id: \"Blood-00001-aliquot_lane1_barcode23.fastq\") {\n    id\n    submitter_id\n    file_name\n    project_id\n}\n}",
        "variables": null
    }
    ```

=== "Shell"

    ```Shell
    curl --request POST --header "X-Auth-Token: $token" 'https://api.gdc.cancer.gov/v0/submission/graphql' --data-binary @escaped_GraphQL
    ```

=== "Response"

    ```json
    {
      "data": {
        "submitted_unaligned_reads": [
          {
            "file_name": "dummy.fastq",
            "id": "616eab2f-791a-4641-8cd6-ee195a10a201",
            "project_id": "GDC-INTERNAL",
            "submitter_id": "Blood-00001-aliquot_lane1_barcode23.fastq"
          }
        ]
      }
    }
    ```



#### Example: Case Without Diagnosis

GraphQL query for any one case in 'TCGA-LUAD' without Diagnosis information:

=== "bare_GraphQL"
    
    ```GraphQL
    {
      case (project_id: "TCGA-LUAD", without_links: ["diagnoses"], first: 1) {
          submitter_id
      }
    }
    ```

=== "Response"

    ```json
    {
      "data": {
        "case": [
          {
            "submitter_id": "TCGA-17-Z050"
          }
        ]
      }
    }
    ```

#### Example: Number of Cases Without Diagnosis

GraphQL query for the number of cases in 'TCGA-LUAD' without Diagnosis information:

=== "bare_GraphQL"

    ```GraphQL
    {
      _case_count (project_id: "TCGA-LUAD", without_links: ["diagnoses"])
    }
    ```

=== "Response"

    ```json
    {
      "data": {
        "_case_count": 5
      }
    }
    ```

#### Example: Aliquot State

Query for the `state` of aliquots belonging to case with `submitter_id: "TCGA-ALCH-000001"`:

=== "bare_GraphQL"
    
    ```GraphQL
    {
      aliquot(with_path_to: {type: "case", submitter_id:"TCGA-ALCH-000001"}) {
        id release_state
      }
    }
    ```

=== "Response"

    ```json
    {
      "data": {
        "aliquot": [
          {
            "id": "7af58da0-cb3e-43e2-a074-4bd8f27565ba",
            "state": "validated"
          }
        ]
      }
    }
    ```

#### Example: Aliases

GraphQL query that uses a GraphQL fragment to get specific properties from two portions and give them aliases in the response:

=== "bare_GraphQL"
    
    ```GraphQL
    {
      some_portion: portion (first: 1) {
        ...portionProperties
      }
      specific_portion: portion(submitter_id: "TCGA-67-6217-01A-13-2191-20") {
        ...portionProperties
      }
    }

    fragment portionProperties on portion {
      submitter_id
      is_ffpe
    }
    ```

=== "Response"

    ```json
    {
      "data": {
        "some_portion": [
          {
            "is_ffpe": false,
            "submitter_id": "TCGA-62-A471-10A-01"
          }
        ],
        "specific_portion": [
          {
            "is_ffpe": false,
            "submitter_id": "TCGA-67-6217-01A-13-2191-20"
          }
        ]
      }
    }
    ```

#### Example: Biospecimen Tree

GraphQL Query for a case in "TCGA-LUAD" and return a biospecimen tree:

=== "bare_GraphQL"

    ```graphQL
    {
      case(project_id: "TCGA-LUAD", first: 1) {
        id
        samples(first: 1) {
          id
          portions(first: 1) {
            id
            analytes(first: 1) {
              id
              aliquots(first: 1) {
                id
              }
            }
          }
        }
      }
    }
    ```

=== "Response"

    ```json
    {
      "data": {
        "case": [
          {
            "id": "19ca36e6-2154-4224-89b1-117a4a4407f6",
            "samples": [
              {
                "id": "5e2625d2-290d-48cd-af5c-27dc8e3c8b6a",
                "portions": [
                  {
                    "analytes": [
                      {
                        "aliquots": [
                          {
                            "id": "8e1820d5-dcd8-4760-9962-221e2b71d4b9"
                          }
                        ],
                        "id": "6449533c-e52a-4e58-bae7-0732f48153ef"
                      }
                    ],
                    "id": "26b75643-8fcd-445e-a0e0-9868cac589ea"
                  }
                ]
              }
            ]
          }
        ]
      }
    }
    ```


# Appendix A: Available Fields

The GDC API's [search and retrieval endpoints](Search_and_Retrieval.md) provide access to fields that correspond to properties defined in the [GDC Data Dictionary](../../Data_Dictionary/viewer.md). This appendix contains a list of fields available at each endpoint, and a list of field groups accessible via the [expand parameter](Search_and_Retrieval.md#expand).

## Field Listing by Endpoint

### Project Fields

| Field Name |
| --- |
| dbgap_accession_number |
| disease_type |
| name |  
| primary_site |
| project_id |
| released |
| state |
| program.dbgap_accession_number |
| program.name |  
| program.program_id |
| summary.case_count |
| summary.file_count |
| summary.file_size |
| summary.data_categories.case_count |
| summary.data_categories.data_category |
| summary.data_categories.file_count |
| summary.experimental_strategies.case_count |  
| summary.experimental_strategies.experimental_strategy |
| summary.experimental_strategies.file_count |



### Case Fields


| Field Name |
| --- |
| aliquot_ids |
| analyte_ids |
| case_id |
| created_datetime |  
| days_to_index |  
| portion_ids |
| sample_ids |
| slide_ids |
| state |
| submitter_aliquot_ids |
| submitter_analyte_ids |
| submitter_id |
| submitter_portion_ids |
| submitter_sample_ids |
| submitter_slide_ids |
| updated_datetime |  
| annotations.annotation_id |
| annotations.case_id |
| annotations.case_submitter_id |
| annotations.category |
| annotations.classification |
| annotations.created_datetime |
| annotations.creator |
| annotations.entity_id |
| annotations.entity_submitter_id |
| annotations.entity_type |
| annotations.legacy_created_datetime |
| annotations.legacy_updated_datetime |
| annotations.notes |
| annotations.state |
| annotations.status |
| annotations.submitter_id |
| annotations.updated_datetime |
| demographic.created_datetime |
| demographic.demographic_id |
| demographic.ethnicity |
| demographic.gender |
| demographic.race |
| demographic.state |
| demographic.submitter_id |
| demographic.updated_datetime |
| demographic.year_of_birth |
| demographic.year_of_death |
| diagnoses.age_at_diagnosis |
| diagnoses.classification_of_tumor |
| diagnoses.created_datetime |
| diagnoses.days_to_birth |
| diagnoses.days_to_death |
| diagnoses.days_to_last_follow_up |
| diagnoses.days_to_last_known_disease_status |
| diagnoses.days_to_recurrence |
| diagnoses.diagnosis_id |
| diagnoses.last_known_disease_status |
| diagnoses.morphology |
| diagnoses.primary_diagnosis |
| diagnoses.prior_malignancy |
| diagnoses.progression_or_recurrence |
| diagnoses.site_of_resection_or_biopsy |
| diagnoses.state |
| diagnoses.submitter_id |
| diagnoses.tissue_or_organ_of_origin |
| diagnoses.tumor_grade |
| diagnoses.tumor_stage |
| diagnoses.updated_datetime |
| diagnoses.vital_status |
| diagnoses.treatments.created_datetime |
| diagnoses.treatments.days_to_treatment |
| diagnoses.treatments.state |
| diagnoses.treatments.submitter_id |
| diagnoses.treatments.therapeutic_agents |
| diagnoses.treatments.treatment_id |
| diagnoses.treatments.treatment_intent_type |
| diagnoses.treatments.treatment_or_therapy |
| diagnoses.treatments.updated_datetime |
| exposures.alcohol_history |
| exposures.alcohol_intensity |
| exposures.bmi |
| exposures.cigarettes_per_day |
| exposures.created_datetime |
| exposures.exposure_id |
| exposures.height |
| exposures.state |
| exposures.submitter_id |
| exposures.updated_datetime |
| exposures.weight |
| exposures.years_smoked |
| family_histories.created_datetime |
| family_histories.family_history_id |  
| family_histories.relationship_age_at_diagnosis |
| family_histories.relationship_gender |
| family_histories.relationship_primary_diagnosis |
| family_histories.relationship_type |
| family_histories.relative_with_cancer_history |
| family_histories.state |
| family_histories.submitter_id |
| family_histories.updated_datetime |
| files.access |
| files.acl |
| files.created_datetime |
| files.data_category |
| files.data_format |
| files.data_type |
| files.error_type |
| files.experimental_strategy |
| files.file_id |
| files.file_name |
| files.file_size |
| files.file_state |
| files.md5sum |
| files.origin |
| files.platform |
| files.revision |
| files.state |
| files.state_comment |
| files.submitter_id |
| files.tags |
| files.type |
| files.updated_datetime |
| files.analysis.analysis_id |
| files.analysis.analysis_type |
| files.analysis.created_datetime |
| files.analysis.state |
| files.analysis.submitter_id |
| files.analysis.updated_datetime |
| files.analysis.workflow_end_datetime |
| files.analysis.workflow_link |
| files.analysis.workflow_start_datetime |
| files.analysis.workflow_type |
| files.analysis.workflow_version |
| files.analysis.input_files.access |
| files.analysis.input_files.created_datetime |
| files.analysis.input_files.data_category |
| files.analysis.input_files.data_format |
| files.analysis.input_files.data_type |
| files.analysis.input_files.error_type |
| files.analysis.input_files.experimental_strategy |
| files.analysis.input_files.file_id |
| files.analysis.input_files.file_name |
| files.analysis.input_files.file_size |
| files.analysis.input_files.file_state |
| files.analysis.input_files.md5sum |
| files.analysis.input_files.platform |
| files.analysis.input_files.revision |
| files.analysis.input_files.state |
| files.analysis.input_files.state_comment |
| files.analysis.input_files.submitter_id |
| files.analysis.input_files.updated_datetime |
| files.analysis.metadata.read_groups.adapter_name |
| files.analysis.metadata.read_groups.adapter_sequence |
| files.analysis.metadata.read_groups.base_caller_name |
| files.analysis.metadata.read_groups.base_caller_version |
| files.analysis.metadata.read_groups.created_datetime |
| files.analysis.metadata.read_groups.experiment_name |
| files.analysis.metadata.read_groups.flow_cell_barcode |
| files.analysis.metadata.read_groups.includes_spike_ins |
| files.analysis.metadata.read_groups.instrument_model |
| files.analysis.metadata.read_groups.is_paired_end |
| files.analysis.metadata.read_groups.library_name |
| files.analysis.metadata.read_groups.library_preparation_kit_catalog_number |
| files.analysis.metadata.read_groups.library_preparation_kit_name |
| files.analysis.metadata.read_groups.library_preparation_kit_vendor |
| files.analysis.metadata.read_groups.library_preparation_kit_version |
| files.analysis.metadata.read_groups.library_selection |
| files.analysis.metadata.read_groups.library_strand |
| files.analysis.metadata.read_groups.library_strategy |
| files.analysis.metadata.read_groups.platform |
| files.analysis.metadata.read_groups.read_group_id |
| files.analysis.metadata.read_groups.read_group_name |
| files.analysis.metadata.read_groups.read_length |
| files.analysis.metadata.read_groups.RIN |
| files.analysis.metadata.read_groups.sequencing_center |
| files.analysis.metadata.read_groups.sequencing_date |
| files.analysis.metadata.read_groups.size_selection_range |
| files.analysis.metadata.read_groups.spike_ins_concentration |
| files.analysis.metadata.read_groups.spike_ins_fasta |
| files.analysis.metadata.read_groups.state |
| files.analysis.metadata.read_groups.submitter_id |
| files.analysis.metadata.read_groups.target_capture_kit_catalog_number |
| files.analysis.metadata.read_groups.target_capture_kit_name |
| files.analysis.metadata.read_groups.target_capture_kit_target_region |
| files.analysis.metadata.read_groups.target_capture_kit_vendor |
| files.analysis.metadata.read_groups.target_capture_kit_version |
| files.analysis.metadata.read_groups.to_trim_adapter_sequence |
| files.analysis.metadata.read_groups.updated_datetime |
| files.analysis.metadata.read_groups.read_group_qcs.adapter_content |
| files.analysis.metadata.read_groups.read_group_qcs.basic_statistics |
| files.analysis.metadata.read_groups.read_group_qcs.created_datetime |
| files.analysis.metadata.read_groups.read_group_qcs.encoding |
| files.analysis.metadata.read_groups.read_group_qcs.fastq_name |
| files.analysis.metadata.read_groups.read_group_qcs.kmer_content |
| files.analysis.metadata.read_groups.read_group_qcs.overrepresented_sequences |
| files.analysis.metadata.read_groups.read_group_qcs.per_base_n_content |
| files.analysis.metadata.read_groups.read_group_qcs.per_base_sequence_content |
| files.analysis.metadata.read_groups.read_group_qcs.per_base_sequence_quality |
| files.analysis.metadata.read_groups.read_group_qcs.per_sequence_gc_content |
| files.analysis.metadata.read_groups.read_group_qcs.per_sequence_quality_score |
| files.analysis.metadata.read_groups.read_group_qcs.per_tile_sequence_quality |
| files.analysis.metadata.read_groups.read_group_qcs.percent_gc_content |
| files.analysis.metadata.read_groups.read_group_qcs.read_group_qc_id |
| files.analysis.metadata.read_groups.read_group_qcs.sequence_duplication_levels |
| files.analysis.metadata.read_groups.read_group_qcs.sequence_length_distribution |
| files.analysis.metadata.read_groups.read_group_qcs.state |
| files.analysis.metadata.read_groups.read_group_qcs.submitter_id |
| files.analysis.metadata.read_groups.read_group_qcs.total_sequences |
| files.analysis.metadata.read_groups.read_group_qcs.updated_datetime |
| files.analysis.metadata.read_groups.read_group_qcs.workflow_end_datetime |
| files.analysis.metadata.read_groups.read_group_qcs.workflow_link |
| files.analysis.metadata.read_groups.read_group_qcs.workflow_start_datetime |
| files.analysis.metadata.read_groups.read_group_qcs.workflow_type |
| files.analysis.metadata.read_groups.read_group_qcs.workflow_version |
| files.archive.archive_id |
| files.archive.created_datetime |
| files.archive.data_category |
| files.archive.data_format |
| files.archive.data_type |
| files.archive.error_type |
| files.archive.file_name |
| files.archive.file_size |
| files.archive.file_state |
| files.archive.md5sum |
| files.archive.revision |
| files.archive.state |
| files.archive.state_comment |
| files.archive.submitter_id |
| files.archive.updated_datetime |
| files.cases.aliquot_ids |
| files.cases.analyte_ids |
| files.cases.case_id |
| files.cases.created_datetime |
| files.cases.days_to_index |
| files.cases.portion_ids |
| files.cases.sample_ids |
| files.cases.slide_ids |
| files.cases.state |
| files.cases.submitter_aliquot_ids |
| files.cases.submitter_analyte_ids |
| files.cases.submitter_id |
| files.cases.submitter_portion_ids |
| files.cases.submitter_sample_ids |
| files.cases.submitter_slide_ids |
| files.cases.updated_datetime |
| files.cases.annotations.annotation_id |
| files.cases.annotations.case_id |
| files.cases.annotations.case_submitter_id |
| files.cases.annotations.category |
| files.cases.annotations.classification |
| files.cases.annotations.created_datetime |
| files.cases.annotations.creator |
| files.cases.annotations.entity_id |
| files.cases.annotations.entity_submitter_id |
| files.cases.annotations.entity_type |
| files.cases.annotations.legacy_created_datetime |
| files.cases.annotations.legacy_updated_datetime |
| files.cases.annotations.notes |
| files.cases.annotations.state |
| files.cases.annotations.status |
| files.cases.annotations.submitter_id |
| files.cases.annotations.updated_datetime |
| files.cases.demographic.created_datetime |
| files.cases.demographic.demographic_id |
| files.cases.demographic.ethnicity |
| files.cases.demographic.gender |
| files.cases.demographic.race |
| files.cases.demographic.state |
| files.cases.demographic.submitter_id |
| files.cases.demographic.updated_datetime |
| files.cases.demographic.year_of_birth |
| files.cases.demographic.year_of_death |
| files.cases.diagnoses.age_at_diagnosis |
| files.cases.diagnoses.classification_of_tumor |
| files.cases.diagnoses.created_datetime |
| files.cases.diagnoses.days_to_birth |
| files.cases.diagnoses.days_to_death |
| files.cases.diagnoses.days_to_last_follow_up |
| files.cases.diagnoses.days_to_last_known_disease_status |
| files.cases.diagnoses.days_to_recurrence |
| files.cases.diagnoses.diagnosis_id |
| files.cases.diagnoses.last_known_disease_status |
| files.cases.diagnoses.morphology |
| files.cases.diagnoses.primary_diagnosis |
| files.cases.diagnoses.prior_malignancy |
| files.cases.diagnoses.progression_or_recurrence |
| files.cases.diagnoses.site_of_resection_or_biopsy |
| files.cases.diagnoses.state |
| files.cases.diagnoses.submitter_id |
| files.cases.diagnoses.tissue_or_organ_of_origin |
| files.cases.diagnoses.tumor_grade |
| files.cases.diagnoses.tumor_stage |
| files.cases.diagnoses.updated_datetime |
| files.cases.diagnoses.vital_status |
| files.cases.diagnoses.treatments.created_datetime |
| files.cases.diagnoses.treatments.days_to_treatment |
| files.cases.diagnoses.treatments.state |
| files.cases.diagnoses.treatments.submitter_id |
| files.cases.diagnoses.treatments.therapeutic_agents |
| files.cases.diagnoses.treatments.treatment_id |
| files.cases.diagnoses.treatments.treatment_intent_type |
| files.cases.diagnoses.treatments.treatment_or_therapy |
| files.cases.diagnoses.treatments.updated_datetime |
| files.cases.exposures.alcohol_history |
| files.cases.exposures.alcohol_intensity |
| files.cases.exposures.bmi |
| files.cases.exposures.cigarettes_per_day |
| files.cases.exposures.created_datetime |
| files.cases.exposures.exposure_id |
| files.cases.exposures.height |
| files.cases.exposures.state |
| files.cases.exposures.submitter_id |
| files.cases.exposures.updated_datetime |
| files.cases.exposures.weight |
| files.cases.exposures.years_smoked |
| files.cases.family_histories.created_datetime |
| files.cases.family_histories.family_history_id |
| files.cases.family_histories.relationship_age_at_diagnosis |
| files.cases.family_histories.relationship_gender |
| files.cases.family_histories.relationship_primary_diagnosis |
| files.cases.family_histories.relationship_type |
| files.cases.family_histories.relative_with_cancer_history |
| files.cases.family_histories.state |
| files.cases.family_histories.submitter_id |
| files.cases.family_histories.updated_datetime |
| files.cases.files.created_datetime |
| files.cases.files.error_type |
| files.cases.files.file_id |
| files.cases.files.file_name |
| files.cases.files.file_size |
| files.cases.files.file_state |
| files.cases.files.md5sum |
| files.cases.files.state |
| files.cases.files.state_comment |
| files.cases.files.submitter_id |
| files.cases.files.updated_datetime |
| files.cases.follow_ups.adverse_event |
| files.cases.follow_ups.adverse_event_grade |
| files.cases.follow_ups.aids_risk_factors |
| files.cases.follow_ups.barretts_esophagus_goblet_cells_present |
| files.cases.follow_ups.bmi |
| files.cases.follow_ups.body_surface_area |
| files.cases.follow_ups.cause_of_response |
| files.cases.follow_ups.cd4_count |
| files.cases.follow_ups.cdc_hiv_risk_factors |
| files.cases.follow_ups.comorbidities |
| files.cases.follow_ups.comorbidity |
| files.cases.follow_ups.comorbidity_method_of_diagnosis |
| files.cases.follow_ups.created_datetime |
| files.cases.follow_ups.days_to_adverse_event |
| files.cases.follow_ups.days_to_comorbidity |
| files.cases.follow_ups.days_to_first_event |
| files.cases.follow_ups.days_to_follow_up |
| files.cases.follow_ups.days_to_imaging |
| files.cases.follow_ups.days_to_progression |
| files.cases.follow_ups.days_to_progression_free |
| files.cases.follow_ups.days_to_recurrence |
| files.cases.follow_ups.days_to_risk_factor |
| files.cases.follow_ups.diabetes_treatment_type |
| files.cases.follow_ups.discontiguous_lesion_count |
| files.cases.follow_ups.disease_response |
| files.cases.follow_ups.dlco_ref_predictive_percent |
| files.cases.follow_ups.ecog_performance_status |
| files.cases.follow_ups.evidence_of_progression_type |
| files.cases.follow_ups.evidence_of_recurrence_type |
| files.cases.follow_ups.eye_color |
| files.cases.follow_ups.fev1_fvc_post_bronch_percent |
| files.cases.follow_ups.fev1_fvc_pre_bronch_percent |
| files.cases.follow_ups.fev1_ref_post_bronch_percent |
| files.cases.follow_ups.fev1_ref_pre_bronch_percent |
| files.cases.follow_ups.first_event |
| files.cases.follow_ups.follow_up_id |
| files.cases.follow_ups.haart_treatment_indicator |
| files.cases.follow_ups.height |
| files.cases.follow_ups.hepatitis_sustained_virological_response |
| files.cases.follow_ups.history_of_tumor |
| files.cases.follow_ups.history_of_tumor_type |
| files.cases.follow_ups.hiv_viral_load |
| files.cases.follow_ups.hormonal_contraceptive_type |
| files.cases.follow_ups.hormonal_contraceptive_use |
| files.cases.follow_ups.hormone_replacement_therapy_type |
| files.cases.follow_ups.hpv_positive_type |
| files.cases.follow_ups.hysterectomy_margins_involved |
| files.cases.follow_ups.hysterectomy_type |
| files.cases.follow_ups.imaging_anatomic_site |
| files.cases.follow_ups.imaging_findings |
| files.cases.follow_ups.imaging_result |
| files.cases.follow_ups.imaging_suv |
| files.cases.follow_ups.imaging_suv_max |
| files.cases.follow_ups.imaging_type |
| files.cases.follow_ups.immunosuppressive_treatment_type |
| files.cases.follow_ups.karnofsky_performance_status |
| files.cases.follow_ups.menopause_status |
| files.cases.follow_ups.molecular_tests.aa_change |
| files.cases.follow_ups.molecular_tests.aneuploidy |
| files.cases.follow_ups.molecular_tests.antigen |
| files.cases.follow_ups.molecular_tests.biospecimen_type |
| files.cases.follow_ups.molecular_tests.biospecimen_volume |
| files.cases.follow_ups.molecular_tests.blood_test_normal_range_lower |
| files.cases.follow_ups.molecular_tests.blood_test_normal_range_upper |
| files.cases.follow_ups.molecular_tests.cell_count |
| files.cases.follow_ups.molecular_tests.chromosomal_translocation |
| files.cases.follow_ups.molecular_tests.chromosome |
| files.cases.follow_ups.molecular_tests.chromosome_arm |
| files.cases.follow_ups.molecular_tests.clonality |
| files.cases.follow_ups.molecular_tests.copy_number |
| files.cases.follow_ups.molecular_tests.created_datetime |
| files.cases.follow_ups.molecular_tests.cytoband |
| files.cases.follow_ups.molecular_tests.days_to_test |
| files.cases.follow_ups.molecular_tests.exon |
| files.cases.follow_ups.molecular_tests.gene_symbol |
| files.cases.follow_ups.molecular_tests.histone_family |
| files.cases.follow_ups.molecular_tests.histone_variant |
| files.cases.follow_ups.molecular_tests.hpv_strain |
| files.cases.follow_ups.molecular_tests.intron |
| files.cases.follow_ups.molecular_tests.laboratory_test |
| files.cases.follow_ups.molecular_tests.loci_abnormal_count |
| files.cases.follow_ups.molecular_tests.loci_count |
| files.cases.follow_ups.molecular_tests.locus |
| files.cases.follow_ups.molecular_tests.mismatch_repair_mutation |
| files.cases.follow_ups.molecular_tests.mitotic_count |
| files.cases.follow_ups.molecular_tests.mitotic_total_area |
| files.cases.follow_ups.molecular_tests.molecular_analysis_method |
| files.cases.follow_ups.molecular_tests.molecular_consequence |
| files.cases.follow_ups.molecular_tests.molecular_test_id |
| files.cases.follow_ups.molecular_tests.mutation_codon |
| files.cases.follow_ups.molecular_tests.pathogenicity |
| files.cases.follow_ups.molecular_tests.ploidy |
| files.cases.follow_ups.molecular_tests.second_exon |
| files.cases.follow_ups.molecular_tests.second_gene_symbol |
| files.cases.follow_ups.molecular_tests.specialized_molecular_test |
| files.cases.follow_ups.molecular_tests.staining_intensity_scale |
| files.cases.follow_ups.molecular_tests.staining_intensity_value |
| files.cases.follow_ups.molecular_tests.state |
| files.cases.follow_ups.molecular_tests.submitter_id |
| files.cases.follow_ups.molecular_tests.test_analyte_type |
| files.cases.follow_ups.molecular_tests.test_result |
| files.cases.follow_ups.molecular_tests.test_units |
| files.cases.follow_ups.molecular_tests.test_value |
| files.cases.follow_ups.molecular_tests.test_value_range |
| files.cases.follow_ups.molecular_tests.timepoint_category |
| files.cases.follow_ups.molecular_tests.transcript |
| files.cases.follow_ups.molecular_tests.updated_datetime |
| files.cases.follow_ups.molecular_tests.variant_origin |
| files.cases.follow_ups.molecular_tests.variant_type |
| files.cases.follow_ups.molecular_tests.zygosity |
| files.cases.follow_ups.nadir_cd4_count |
| files.cases.follow_ups.other_clinical_attributes.aids_risk_factors |
| files.cases.follow_ups.other_clinical_attributes.bmi |
| files.cases.follow_ups.other_clinical_attributes.body_surface_area |
| files.cases.follow_ups.other_clinical_attributes.cd4_count |
| files.cases.follow_ups.other_clinical_attributes.cdc_hiv_risk_factors |
| files.cases.follow_ups.other_clinical_attributes.comorbidities |
| files.cases.follow_ups.other_clinical_attributes.comorbidity_method_of_diagnosis |
| files.cases.follow_ups.other_clinical_attributes.created_datetime |
| files.cases.follow_ups.other_clinical_attributes.days_to_comorbidity |
| files.cases.follow_ups.other_clinical_attributes.days_to_risk_factor |
| files.cases.follow_ups.other_clinical_attributes.diabetes_treatment_type |
| files.cases.follow_ups.other_clinical_attributes.dlco_ref_predictive_percent |
| files.cases.follow_ups.other_clinical_attributes.exercise_frequency_weekly |
| files.cases.follow_ups.other_clinical_attributes.eye_color |
| files.cases.follow_ups.other_clinical_attributes.fev1_fvc_post_bronch_percent |
| files.cases.follow_ups.other_clinical_attributes.fev1_fvc_pre_bronch_percent |
| files.cases.follow_ups.other_clinical_attributes.fev1_ref_post_bronch_percent |
| files.cases.follow_ups.other_clinical_attributes.fev1_ref_pre_bronch_percent |
| files.cases.follow_ups.other_clinical_attributes.haart_treatment_indicator |
| files.cases.follow_ups.other_clinical_attributes.height |
| files.cases.follow_ups.other_clinical_attributes.hepatitis_sustained_virological_response |
| files.cases.follow_ups.other_clinical_attributes.hiv_viral_load |
| files.cases.follow_ups.other_clinical_attributes.hormonal_contraceptive_type |
| files.cases.follow_ups.other_clinical_attributes.hormonal_contraceptive_use |
| files.cases.follow_ups.other_clinical_attributes.hormonal_replacement_therapy_status |
| files.cases.follow_ups.other_clinical_attributes.hormone_replacement_therapy_type |
| files.cases.follow_ups.other_clinical_attributes.hysterectomy_margins_involved |
| files.cases.follow_ups.other_clinical_attributes.hysterectomy_type |
| files.cases.follow_ups.other_clinical_attributes.immunosuppressive_treatment_type |
| files.cases.follow_ups.other_clinical_attributes.menopause_status |
| files.cases.follow_ups.other_clinical_attributes.myasthenia_gravis_classification |
| files.cases.follow_ups.other_clinical_attributes.nadir_cd4_count |
| files.cases.follow_ups.other_clinical_attributes.nononcologic_therapeutic_agents |
| files.cases.follow_ups.other_clinical_attributes.number_of_pregnancies |
| files.cases.follow_ups.other_clinical_attributes.other_clinical_attribute_id |
| files.cases.follow_ups.other_clinical_attributes.oxygen_use_indicator |
| files.cases.follow_ups.other_clinical_attributes.oxygen_use_type |
| files.cases.follow_ups.other_clinical_attributes.pancreatitis_onset_year |
| files.cases.follow_ups.other_clinical_attributes.pregnancy_count |
| files.cases.follow_ups.other_clinical_attributes.pregnancy_outcome |
| files.cases.follow_ups.other_clinical_attributes.pregnant_at_diagnosis |
| files.cases.follow_ups.other_clinical_attributes.premature_at_birth |
| files.cases.follow_ups.other_clinical_attributes.reflux_treatment_type |
| files.cases.follow_ups.other_clinical_attributes.risk_factor_method_of_diagnosis |
| files.cases.follow_ups.other_clinical_attributes.risk_factor_treatment |
| files.cases.follow_ups.other_clinical_attributes.risk_factors |
| files.cases.follow_ups.other_clinical_attributes.state |
| files.cases.follow_ups.other_clinical_attributes.submitter_id |
| files.cases.follow_ups.other_clinical_attributes.timepoint_category |
| files.cases.follow_ups.other_clinical_attributes.treatment_frequency |
| files.cases.follow_ups.other_clinical_attributes.undescended_testis_corrected |
| files.cases.follow_ups.other_clinical_attributes.undescended_testis_corrected_age |
| files.cases.follow_ups.other_clinical_attributes.undescended_testis_corrected_laterality |
| files.cases.follow_ups.other_clinical_attributes.undescended_testis_corrected_method |
| files.cases.follow_ups.other_clinical_attributes.undescended_testis_history |
| files.cases.follow_ups.other_clinical_attributes.undescended_testis_history_laterality |
| files.cases.follow_ups.other_clinical_attributes.updated_datetime |
| files.cases.follow_ups.other_clinical_attributes.viral_hepatitis_serologies |
| files.cases.follow_ups.other_clinical_attributes.viral_hepatitis_serology_tests |
| files.cases.follow_ups.other_clinical_attributes.weeks_gestation_at_birth |
| files.cases.follow_ups.other_clinical_attributes.weight |
| files.cases.follow_ups.pancreatitis_onset_year |
| files.cases.follow_ups.peritoneal_washing_results |
| files.cases.follow_ups.pregnancy_count |
| files.cases.follow_ups.pregnancy_outcome |
| files.cases.follow_ups.procedures_performed |
| files.cases.follow_ups.progression_or_recurrence |
| files.cases.follow_ups.progression_or_recurrence_anatomic_site |
| files.cases.follow_ups.progression_or_recurrence_type |
| files.cases.follow_ups.recist_targeted_regions_number |
| files.cases.follow_ups.recist_targeted_regions_sum |
| files.cases.follow_ups.reflux_treatment_type |
| files.cases.follow_ups.risk_factor |
| files.cases.follow_ups.risk_factor_method_of_diagnosis |
| files.cases.follow_ups.risk_factor_treatment |
| files.cases.follow_ups.risk_factors |
| files.cases.follow_ups.scan_tracer_used |
| files.cases.follow_ups.state |
| files.cases.follow_ups.submitter_id |
| files.cases.follow_ups.timepoint_category |
| files.cases.follow_ups.treatment_emergent_adverse_event |
| files.cases.follow_ups.undescended_testis_corrected |
| files.cases.follow_ups.undescended_testis_corrected_age |
| files.cases.follow_ups.undescended_testis_corrected_laterality |
| files.cases.follow_ups.undescended_testis_corrected_method |
| files.cases.follow_ups.undescended_testis_history |
| files.cases.follow_ups.undescended_testis_history_laterality |
| files.cases.follow_ups.updated_datetime |
| files.cases.follow_ups.viral_hepatitis_serologies |
| files.cases.follow_ups.weight |
| files.cases.follow_ups.year_of_follow_up |
| files.cases.project.dbgap_accession_number |
| files.cases.project.disease_type |
| files.cases.project.name |
| files.cases.project.primary_site |
| files.cases.project.project_id |
| files.cases.project.released |
| files.cases.project.state |
| files.cases.project.program.dbgap_accession_number |
| files.cases.project.program.name |
| files.cases.project.program.program_id |
| files.cases.samples.composition |
| files.cases.samples.created_datetime |
| files.cases.samples.current_weight |
| files.cases.samples.days_to_collection |
| files.cases.samples.days_to_sample_procurement |
| files.cases.samples.freezing_method |
| files.cases.samples.initial_weight |
| files.cases.samples.intermediate_dimension |
| files.cases.samples.is_ffpe |
| files.cases.samples.longest_dimension |
| files.cases.samples.oct_embedded |
| files.cases.samples.pathology_report_uuid |
| files.cases.samples.preservation_method |
| files.cases.samples.sample_id |
| files.cases.samples.sample_type |
| files.cases.samples.sample_type_id |
| files.cases.samples.shortest_dimension |
| files.cases.samples.state |
| files.cases.samples.submitter_id |
| files.cases.samples.time_between_clamping_and_freezing |
| files.cases.samples.time_between_excision_and_freezing |
| files.cases.samples.tissue_type |
| files.cases.samples.tumor_code |
| files.cases.samples.tumor_code_id |
| files.cases.samples.tumor_descriptor |
| files.cases.samples.updated_datetime |
| files.cases.samples.annotations.annotation_id |
| files.cases.samples.annotations.case_id |
| files.cases.samples.annotations.case_submitter_id |
| files.cases.samples.annotations.category |
| files.cases.samples.annotations.classification |
| files.cases.samples.annotations.created_datetime |
| files.cases.samples.annotations.creator |
| files.cases.samples.annotations.entity_id |
| files.cases.samples.annotations.entity_submitter_id |
| files.cases.samples.annotations.entity_type |
| files.cases.samples.annotations.legacy_created_datetime |
| files.cases.samples.annotations.legacy_updated_datetime |
| files.cases.samples.annotations.notes |
| files.cases.samples.annotations.state |
| files.cases.samples.annotations.status |
| files.cases.samples.annotations.submitter_id |
| files.cases.samples.annotations.updated_datetime |
| files.cases.samples.portions.created_datetime |
| files.cases.samples.portions.creation_datetime |
| files.cases.samples.portions.is_ffpe |
| files.cases.samples.portions.portion_id |
| files.cases.samples.portions.portion_number |
| files.cases.samples.portions.state |  
| files.cases.samples.portions.submitter_id |  
| files.cases.samples.portions.updated_datetime |
| files.cases.samples.portions.weight |
| files.cases.samples.portions.analytes.a260_a280_ratio |
| files.cases.samples.portions.analytes.amount |
| files.cases.samples.portions.analytes.analyte_id |
| files.cases.samples.portions.analytes.analyte_type |
| files.cases.samples.portions.analytes.analyte_type_id |
| files.cases.samples.portions.analytes.concentration |
| files.cases.samples.portions.analytes.created_datetime |
| files.cases.samples.portions.analytes.spectrophotometer_method |
| files.cases.samples.portions.analytes.state |
| files.cases.samples.portions.analytes.submitter_id |
| files.cases.samples.portions.analytes.updated_datetime |
| files.cases.samples.portions.analytes.well_number |
| files.cases.samples.portions.analytes.aliquots.aliquot_id |
| files.cases.samples.portions.analytes.aliquots.amount |
| files.cases.samples.portions.analytes.aliquots.analyte_type |
| files.cases.samples.portions.analytes.aliquots.analyte_type_id |
| files.cases.samples.portions.analytes.aliquots.concentration |
| files.cases.samples.portions.analytes.aliquots.created_datetime |
| files.cases.samples.portions.analytes.aliquots.source_center |
| files.cases.samples.portions.analytes.aliquots.state |
| files.cases.samples.portions.analytes.aliquots.submitter_id |
| files.cases.samples.portions.analytes.aliquots.updated_datetime |
| files.cases.samples.portions.analytes.aliquots.annotations.annotation_id |
| files.cases.samples.portions.analytes.aliquots.annotations.case_id |
| files.cases.samples.portions.analytes.aliquots.annotations.case_submitter_id |
| files.cases.samples.portions.analytes.aliquots.annotations.category |
| files.cases.samples.portions.analytes.aliquots.annotations.classification |
| files.cases.samples.portions.analytes.aliquots.annotations.created_datetime |
| files.cases.samples.portions.analytes.aliquots.annotations.creator |
| files.cases.samples.portions.analytes.aliquots.annotations.entity_id |
| files.cases.samples.portions.analytes.aliquots.annotations.entity_submitter_id |
| files.cases.samples.portions.analytes.aliquots.annotations.entity_type |
| files.cases.samples.portions.analytes.aliquots.annotations.legacy_created_datetime |
| files.cases.samples.portions.analytes.aliquots.annotations.legacy_updated_datetime |
| files.cases.samples.portions.analytes.aliquots.annotations.notes |
| files.cases.samples.portions.analytes.aliquots.annotations.state |
| files.cases.samples.portions.analytes.aliquots.annotations.status |
| files.cases.samples.portions.analytes.aliquots.annotations.submitter_id |
| files.cases.samples.portions.analytes.aliquots.annotations.updated_datetime |
| files.cases.samples.portions.analytes.aliquots.center.center_id |
| files.cases.samples.portions.analytes.aliquots.center.center_type |
| files.cases.samples.portions.analytes.aliquots.center.code |  
| files.cases.samples.portions.analytes.aliquots.center.name |  
| files.cases.samples.portions.analytes.aliquots.center.namespace |  
| files.cases.samples.portions.analytes.aliquots.center.short_name |
| files.cases.samples.portions.analytes.annotations.annotation_id |
| files.cases.samples.portions.analytes.annotations.case_id |
| files.cases.samples.portions.analytes.annotations.case_submitter_id |
| files.cases.samples.portions.analytes.annotations.category |
| files.cases.samples.portions.analytes.annotations.classification |
| files.cases.samples.portions.analytes.annotations.created_datetime |
| files.cases.samples.portions.analytes.annotations.creator |
| files.cases.samples.portions.analytes.annotations.entity_id |
| files.cases.samples.portions.analytes.annotations.entity_submitter_id |
| files.cases.samples.portions.analytes.annotations.entity_type |
| files.cases.samples.portions.analytes.annotations.legacy_created_datetime |
| files.cases.samples.portions.analytes.annotations.legacy_updated_datetime |
| files.cases.samples.portions.analytes.annotations.notes |
| files.cases.samples.portions.analytes.annotations.state |
| files.cases.samples.portions.analytes.annotations.status |
| files.cases.samples.portions.analytes.annotations.submitter_id |
| files.cases.samples.portions.analytes.annotations.updated_datetime |
| files.cases.samples.portions.annotations.annotation_id |
| files.cases.samples.portions.annotations.case_id |
| files.cases.samples.portions.annotations.case_submitter_id |
| files.cases.samples.portions.annotations.category |
| files.cases.samples.portions.annotations.classification |
| files.cases.samples.portions.annotations.created_datetime |
| files.cases.samples.portions.annotations.creator |
| files.cases.samples.portions.annotations.entity_id |
| files.cases.samples.portions.annotations.entity_submitter_id |
| files.cases.samples.portions.annotations.entity_type |
| files.cases.samples.portions.annotations.legacy_created_datetime |
| files.cases.samples.portions.annotations.legacy_updated_datetime |
| files.cases.samples.portions.annotations.notes |
| files.cases.samples.portions.annotations.state |
| files.cases.samples.portions.annotations.status |
| files.cases.samples.portions.annotations.submitter_id |
| files.cases.samples.portions.annotations.updated_datetime |
| files.cases.samples.portions.center.center_id |
| files.cases.samples.portions.center.center_type |
| files.cases.samples.portions.center.code |
| files.cases.samples.portions.center.name |
| files.cases.samples.portions.center.namespace |
| files.cases.samples.portions.center.short_name |
| files.cases.samples.portions.slides.created_datetime |
| files.cases.samples.portions.slides.number_proliferating_cells |
| files.cases.samples.portions.slides.percent_eosinophil_infiltration |
| files.cases.samples.portions.slides.percent_granulocyte_infiltration |
| files.cases.samples.portions.slides.percent_inflam_infiltration |
| files.cases.samples.portions.slides.percent_lymphocyte_infiltration |
| files.cases.samples.portions.slides.percent_monocyte_infiltration |
| files.cases.samples.portions.slides.percent_necrosis |
| files.cases.samples.portions.slides.percent_neutrophil_infiltration |
| files.cases.samples.portions.slides.percent_normal_cells |
| files.cases.samples.portions.slides.percent_stromal_cells |
| files.cases.samples.portions.slides.percent_tumor_cells |
| files.cases.samples.portions.slides.percent_tumor_nuclei |
| files.cases.samples.portions.slides.section_location |
| files.cases.samples.portions.slides.slide_id |
| files.cases.samples.portions.slides.state |
| files.cases.samples.portions.slides.submitter_id |
| files.cases.samples.portions.slides.updated_datetime |
| files.cases.samples.portions.slides.annotations.annotation_id |
| files.cases.samples.portions.slides.annotations.case_id |
| files.cases.samples.portions.slides.annotations.case_submitter_id |
| files.cases.samples.portions.slides.annotations.category |
| files.cases.samples.portions.slides.annotations.classification |
| files.cases.samples.portions.slides.annotations.created_datetime |
| files.cases.samples.portions.slides.annotations.creator |
| files.cases.samples.portions.slides.annotations.entity_id |
| files.cases.samples.portions.slides.annotations.entity_submitter_id |
| files.cases.samples.portions.slides.annotations.entity_type |
| files.cases.samples.portions.slides.annotations.legacy_created_datetime |
| files.cases.samples.portions.slides.annotations.legacy_updated_datetime |
| files.cases.samples.portions.slides.annotations.notes |
| files.cases.samples.portions.slides.annotations.state |
| files.cases.samples.portions.slides.annotations.status |
| files.cases.samples.portions.slides.annotations.submitter_id |
| files.cases.samples.portions.slides.annotations.updated_datetime |
| files.cases.summary.file_count |
| files.cases.summary.file_size |
| files.cases.summary.data_categories.data_category |
| files.cases.summary.data_categories.file_count |
| files.cases.summary.experimental_strategies.experimental_strategy |
| files.cases.summary.experimental_strategies.file_count |
| files.cases.tissue_source_site.bcr_id |
| files.cases.tissue_source_site.code |
| files.cases.tissue_source_site.name |
| files.cases.tissue_source_site.project |
| files.cases.tissue_source_site.tissue_source_site_id |
| files.center.center_id |
| files.center.center_type |
| files.center.code |
| files.center.name |
| files.center.namespace |
| files.center.short_name |
| files.downstream_analyses.analysis_id |
| files.downstream_analyses.analysis_type |
| files.downstream_analyses.created_datetime |
| files.downstream_analyses.state |
| files.downstream_analyses.submitter_id |
| files.downstream_analyses.updated_datetime |
| files.downstream_analyses.workflow_end_datetime |
| files.downstream_analyses.workflow_link |
| files.downstream_analyses.workflow_start_datetime |
| files.downstream_analyses.workflow_type |
| files.downstream_analyses.workflow_version |
| files.downstream_analyses.output_files.access |
| files.downstream_analyses.output_files.created_datetime |
| files.downstream_analyses.output_files.data_category |
| files.downstream_analyses.output_files.data_format |
| files.downstream_analyses.output_files.data_type |
| files.downstream_analyses.output_files.error_type |
| files.downstream_analyses.output_files.experimental_strategy |
| files.downstream_analyses.output_files.file_id |
| files.downstream_analyses.output_files.file_name |
| files.downstream_analyses.output_files.file_size |
| files.downstream_analyses.output_files.file_state |
| files.downstream_analyses.output_files.md5sum |
| files.downstream_analyses.output_files.platform |
| files.downstream_analyses.output_files.revision |
| files.downstream_analyses.output_files.state |
| files.downstream_analyses.output_files.state_comment |
| files.downstream_analyses.output_files.submitter_id |
| files.downstream_analyses.output_files.updated_datetime |
| files.index_files.access |
| files.index_files.created_datetime |
| files.index_files.data_category |
| files.index_files.data_format |
| files.index_files.data_type |
| files.index_files.error_type |
| files.index_files.experimental_strategy |
| files.index_files.file_id |
| files.index_files.file_name |
| files.index_files.file_size |
| files.index_files.file_state |
| files.index_files.md5sum |
| files.index_files.platform |
| files.index_files.revision |
| files.index_files.state |
| files.index_files.state_comment |
| files.index_files.submitter_id |
| files.index_files.updated_datetime |
| files.metadata_files.access |
| files.metadata_files.created_datetime |
| files.metadata_files.data_category |
| files.metadata_files.data_format |
| files.metadata_files.data_type |
| files.metadata_files.error_type |
| files.metadata_files.file_id |
| files.metadata_files.file_name |
| files.metadata_files.file_size |
| files.metadata_files.file_state |
| files.metadata_files.md5sum |
| files.metadata_files.state |
| files.metadata_files.state_comment |
| files.metadata_files.submitter_id |
| files.metadata_files.type |
| files.metadata_files.updated_datetime |
| follow_ups.adverse_event |
| follow_ups.adverse_event_grade |
| follow_ups.aids_risk_factors |
| follow_ups.barretts_esophagus_goblet_cells_present |
| follow_ups.bmi |
| follow_ups.body_surface_area |
| follow_ups.cause_of_response |
| follow_ups.cd4_count |
| follow_ups.cdc_hiv_risk_factors |
| follow_ups.comorbidities |
| follow_ups.comorbidity |
| follow_ups.comorbidity_method_of_diagnosis |
| follow_ups.created_datetime |
| follow_ups.days_to_adverse_event |
| follow_ups.days_to_comorbidity |
| follow_ups.days_to_first_event |
| follow_ups.days_to_follow_up |
| follow_ups.days_to_imaging |
| follow_ups.days_to_progression |
| follow_ups.days_to_progression_free |
| follow_ups.days_to_recurrence |
| follow_ups.days_to_risk_factor |
| follow_ups.diabetes_treatment_type |
| follow_ups.discontiguous_lesion_count |
| follow_ups.disease_response |
| follow_ups.dlco_ref_predictive_percent |
| follow_ups.ecog_performance_status |
| follow_ups.evidence_of_progression_type |
| follow_ups.evidence_of_recurrence_type |
| follow_ups.eye_color |
| follow_ups.fev1_fvc_post_bronch_percent |
| follow_ups.fev1_fvc_pre_bronch_percent |
| follow_ups.fev1_ref_post_bronch_percent |
| follow_ups.fev1_ref_pre_bronch_percent |
| follow_ups.first_event |
| follow_ups.follow_up_id |
| follow_ups.haart_treatment_indicator |
| follow_ups.height |
| follow_ups.hepatitis_sustained_virological_response |
| follow_ups.history_of_tumor |
| follow_ups.history_of_tumor_type |
| follow_ups.hiv_viral_load |
| follow_ups.hormonal_contraceptive_type |
| follow_ups.hormonal_contraceptive_use |
| follow_ups.hormone_replacement_therapy_type |
| follow_ups.hpv_positive_type |
| follow_ups.hysterectomy_margins_involved |
| follow_ups.hysterectomy_type |
| follow_ups.imaging_anatomic_site |
| follow_ups.imaging_findings |
| follow_ups.imaging_result |
| follow_ups.imaging_suv |
| follow_ups.imaging_suv_max |
| follow_ups.imaging_type |
| follow_ups.immunosuppressive_treatment_type |
| follow_ups.karnofsky_performance_status |
| follow_ups.menopause_status |
| follow_ups.molecular_tests.aa_change |
| follow_ups.molecular_tests.aneuploidy |
| follow_ups.molecular_tests.antigen |
| follow_ups.molecular_tests.biospecimen_type |
| follow_ups.molecular_tests.biospecimen_volume |
| follow_ups.molecular_tests.blood_test_normal_range_lower |
| follow_ups.molecular_tests.blood_test_normal_range_upper |
| follow_ups.molecular_tests.cell_count |
| follow_ups.molecular_tests.chromosomal_translocation |
| follow_ups.molecular_tests.chromosome |
| follow_ups.molecular_tests.chromosome_arm |
| follow_ups.molecular_tests.clonality |
| follow_ups.molecular_tests.copy_number |
| follow_ups.molecular_tests.created_datetime |
| follow_ups.molecular_tests.cytoband |
| follow_ups.molecular_tests.days_to_test |
| follow_ups.molecular_tests.exon |
| follow_ups.molecular_tests.gene_symbol |
| follow_ups.molecular_tests.histone_family |
| follow_ups.molecular_tests.histone_variant |
| follow_ups.molecular_tests.hpv_strain |
| follow_ups.molecular_tests.intron |
| follow_ups.molecular_tests.laboratory_test |
| follow_ups.molecular_tests.loci_abnormal_count |
| follow_ups.molecular_tests.loci_count |
| follow_ups.molecular_tests.locus |
| follow_ups.molecular_tests.mismatch_repair_mutation |
| follow_ups.molecular_tests.mitotic_count |
| follow_ups.molecular_tests.mitotic_total_area |
| follow_ups.molecular_tests.molecular_analysis_method |
| follow_ups.molecular_tests.molecular_consequence |
| follow_ups.molecular_tests.molecular_test_id |
| follow_ups.molecular_tests.mutation_codon |
| follow_ups.molecular_tests.pathogenicity |
| follow_ups.molecular_tests.ploidy |
| follow_ups.molecular_tests.second_exon |
| follow_ups.molecular_tests.second_gene_symbol |
| follow_ups.molecular_tests.specialized_molecular_test |
| follow_ups.molecular_tests.staining_intensity_scale |
| follow_ups.molecular_tests.staining_intensity_value |
| follow_ups.molecular_tests.state |
| follow_ups.molecular_tests.submitter_id |
| follow_ups.molecular_tests.test_analyte_type |
| follow_ups.molecular_tests.test_result |
| follow_ups.molecular_tests.test_units |
| follow_ups.molecular_tests.test_value |
| follow_ups.molecular_tests.test_value_range |
| follow_ups.molecular_tests.timepoint_category |
| follow_ups.molecular_tests.transcript |
| follow_ups.molecular_tests.updated_datetime |
| follow_ups.molecular_tests.variant_origin |
| follow_ups.molecular_tests.variant_type |
| follow_ups.molecular_tests.zygosity |
| follow_ups.nadir_cd4_count |
| follow_ups.other_clinical_attributes.aids_risk_factors |
| follow_ups.other_clinical_attributes.bmi |
| follow_ups.other_clinical_attributes.body_surface_area |
| follow_ups.other_clinical_attributes.cd4_count |
| follow_ups.other_clinical_attributes.cdc_hiv_risk_factors |
| follow_ups.other_clinical_attributes.comorbidities |
| follow_ups.other_clinical_attributes.comorbidity_method_of_diagnosis |
| follow_ups.other_clinical_attributes.created_datetime |
| follow_ups.other_clinical_attributes.days_to_comorbidity |
| follow_ups.other_clinical_attributes.days_to_risk_factor |
| follow_ups.other_clinical_attributes.diabetes_treatment_type |
| follow_ups.other_clinical_attributes.dlco_ref_predictive_percent |
| follow_ups.other_clinical_attributes.exercise_frequency_weekly |
| follow_ups.other_clinical_attributes.eye_color |
| follow_ups.other_clinical_attributes.fev1_fvc_post_bronch_percent |
| follow_ups.other_clinical_attributes.fev1_fvc_pre_bronch_percent |
| follow_ups.other_clinical_attributes.fev1_ref_post_bronch_percent |
| follow_ups.other_clinical_attributes.fev1_ref_pre_bronch_percent |
| follow_ups.other_clinical_attributes.haart_treatment_indicator |
| follow_ups.other_clinical_attributes.height |
| follow_ups.other_clinical_attributes.hepatitis_sustained_virological_response |
| follow_ups.other_clinical_attributes.hiv_viral_load |
| follow_ups.other_clinical_attributes.hormonal_contraceptive_type |
| follow_ups.other_clinical_attributes.hormonal_contraceptive_use |
| follow_ups.other_clinical_attributes.hormonal_replacement_therapy_status |
| follow_ups.other_clinical_attributes.hormone_replacement_therapy_type |
| follow_ups.other_clinical_attributes.hysterectomy_margins_involved |
| follow_ups.other_clinical_attributes.hysterectomy_type |
| follow_ups.other_clinical_attributes.immunosuppressive_treatment_type |
| follow_ups.other_clinical_attributes.menopause_status |
| follow_ups.other_clinical_attributes.myasthenia_gravis_classification |
| follow_ups.other_clinical_attributes.nadir_cd4_count |
| follow_ups.other_clinical_attributes.nononcologic_therapeutic_agents |
| follow_ups.other_clinical_attributes.number_of_pregnancies |
| follow_ups.other_clinical_attributes.other_clinical_attribute_id |
| follow_ups.other_clinical_attributes.oxygen_use_indicator |
| follow_ups.other_clinical_attributes.oxygen_use_type |
| follow_ups.other_clinical_attributes.pancreatitis_onset_year |
| follow_ups.other_clinical_attributes.pregnancy_count |
| follow_ups.other_clinical_attributes.pregnancy_outcome |
| follow_ups.other_clinical_attributes.pregnant_at_diagnosis |
| follow_ups.other_clinical_attributes.premature_at_birth |
| follow_ups.other_clinical_attributes.reflux_treatment_type |
| follow_ups.other_clinical_attributes.risk_factor_method_of_diagnosis |
| follow_ups.other_clinical_attributes.risk_factor_treatment |
| follow_ups.other_clinical_attributes.risk_factors |
| follow_ups.other_clinical_attributes.state |
| follow_ups.other_clinical_attributes.submitter_id |
| follow_ups.other_clinical_attributes.timepoint_category |
| follow_ups.other_clinical_attributes.treatment_frequency |
| follow_ups.other_clinical_attributes.undescended_testis_corrected |
| follow_ups.other_clinical_attributes.undescended_testis_corrected_age |
| follow_ups.other_clinical_attributes.undescended_testis_corrected_laterality |
| follow_ups.other_clinical_attributes.undescended_testis_corrected_method |
| follow_ups.other_clinical_attributes.undescended_testis_history |
| follow_ups.other_clinical_attributes.undescended_testis_history_laterality |
| follow_ups.other_clinical_attributes.updated_datetime |
| follow_ups.other_clinical_attributes.viral_hepatitis_serologies |
| follow_ups.other_clinical_attributes.viral_hepatitis_serology_tests |
| follow_ups.other_clinical_attributes.weeks_gestation_at_birth |
| follow_ups.other_clinical_attributes.weight |
| follow_ups.pancreatitis_onset_year |
| follow_ups.peritoneal_washing_results |
| follow_ups.pregnancy_count |
| follow_ups.pregnancy_outcome |
| follow_ups.procedures_performed |
| follow_ups.progression_or_recurrence |
| follow_ups.progression_or_recurrence_anatomic_site |
| follow_ups.progression_or_recurrence_type |
| follow_ups.recist_targeted_regions_number |
| follow_ups.recist_targeted_regions_sum |
| follow_ups.reflux_treatment_type |
| follow_ups.risk_factor |
| follow_ups.risk_factor_method_of_diagnosis |
| follow_ups.risk_factor_treatment |
| follow_ups.risk_factors |
| follow_ups.scan_tracer_used |
| follow_ups.state |
| follow_ups.submitter_id |
| follow_ups.timepoint_category |
| follow_ups.treatment_emergent_adverse_event |
| follow_ups.undescended_testis_corrected |
| follow_ups.undescended_testis_corrected_age |
| follow_ups.undescended_testis_corrected_laterality |
| follow_ups.undescended_testis_corrected_method |
| follow_ups.undescended_testis_history |
| follow_ups.undescended_testis_history_laterality |
| follow_ups.updated_datetime |
| follow_ups.viral_hepatitis_serologies |
| follow_ups.weight |
| follow_ups.year_of_follow_up |
| project.dbgap_accession_number |
| project.disease_type |
| project.name |
| project.primary_site |
| project.project_id |
| project.released |
| project.state |
| project.program.dbgap_accession_number |
| project.program.name |
| project.program.program_id |
| samples.composition |
| samples.created_datetime |
| samples.current_weight |
| samples.days_to_collection |
| samples.days_to_sample_procurement |
| samples.freezing_method |
| samples.initial_weight |
| samples.intermediate_dimension |
| samples.is_ffpe |
| samples.longest_dimension |
| samples.oct_embedded |
| samples.pathology_report_uuid |
| samples.preservation_method |
| samples.sample_id |
| samples.sample_type |
| samples.sample_type_id |
| samples.shortest_dimension |
| samples.state |
| samples.submitter_id |
| samples.time_between_clamping_and_freezing |
| samples.time_between_excision_and_freezing |
| samples.tissue_type |
| samples.tumor_code |
| samples.tumor_code_id |
| samples.tumor_descriptor |
| samples.updated_datetime |
| samples.annotations.annotation_id |
| samples.annotations.case_id |
| samples.annotations.case_submitter_id |
| samples.annotations.category |
| samples.annotations.classification |
| samples.annotations.created_datetime |
| samples.annotations.creator |
| samples.annotations.entity_id |
| samples.annotations.entity_submitter_id |
| samples.annotations.entity_type |
| samples.annotations.legacy_creat
| samples.annotations.legacy_updated_datetime |
| samples.annotations.notes |
| samples.annotations.state |
| samples.annotations.status |
| samples.annotations.submitter_id |
| samples.annotations.updated_datetime |
| samples.portions.created_datetime |
| samples.portions.creation_datetime |
| samples.portions.is_ffpe |
| samples.portions.portion_id |
| samples.portions.portion_number |
| samples.portions.state |
| samples.portions.submitter_id |
| samples.portions.updated_datetime |
| samples.portions.weight |
| samples.portions.analytes.a260_a280_ratio |
| samples.portions.analytes.amount |
| samples.portions.analytes.analyte_id |
| samples.portions.analytes.analyte_type |
| samples.portions.analytes.analyte_type_id |
| samples.portions.analytes.concentration |
| samples.portions.analytes.created_datetime |
| samples.portions.analytes.spectrophotometer_method |
| samples.portions.analytes.state |
| samples.portions.analytes.submitter_id |
| samples.portions.analytes.updated_datetime |
| samples.portions.analytes.well_number |
| samples.portions.analytes.aliquots.aliquot_id |
| samples.portions.analytes.aliquots.amount |
| samples.portions.analytes.aliquots.analyte_type |
| samples.portions.analytes.aliquots.analyte_type_id |
| samples.portions.analytes.aliquots.concentration |
| samples.portions.analytes.aliquots.created_datetime |
| samples.portions.analytes.aliquots.source_center |
| samples.portions.analytes.aliquots.state |
| samples.portions.analytes.aliquots.submitter_id |
| samples.portions.analytes.aliquots.updated_datetime |
| samples.portions.analytes.aliquots.annotations.annotation_id |
| samples.portions.analytes.aliquots.annotations.case_id |
| samples.portions.analytes.aliquots.annotations.case_submitter_id |
| samples.portions.analytes.aliquots.annotations.category |
| samples.portions.analytes.aliquots.annotations.classification |
| samples.portions.analytes.aliquots.annotations.created_datetime |
| samples.portions.analytes.aliquots.annotations.creator |
| samples.portions.analytes.aliquots.annotations.entity_id |
| samples.portions.analytes.aliquots.annotations.entity_submitter_id |
| samples.portions.analytes.aliquots.annotations.entity_type |
| samples.portions.analytes.aliquots.annotations.legacy_created_datetime |
| samples.portions.analytes.aliquots.annotations.legacy_updated_datetime |
| samples.portions.analytes.aliquots.annotations.notes |
| samples.portions.analytes.aliquots.annotations.state |
| samples.portions.analytes.aliquots.annotations.status |
| samples.portions.analytes.aliquots.annotations.submitter_id |
| samples.portions.analytes.aliquots.annotations.updated_datetime |
| samples.portions.analytes.aliquots.center.center_id |
| samples.portions.analytes.aliquots.center.center_type |
| samples.portions.analytes.aliquots.center.code |
| samples.portions.analytes.aliquots.center.name |
| samples.portions.analytes.aliquots.center.namespace |
| samples.portions.analytes.aliquots.center.short_name |
| samples.portions.analytes.annotations.annotation_id |
| samples.portions.analytes.annotations.case_id |
| samples.portions.analytes.annotations.case_submitter_id |
| samples.portions.analytes.annotations.category |
| samples.portions.analytes.annotations.classification |
| samples.portions.analytes.annotations.created_datetime |
| samples.portions.analytes.annotations.creator |
| samples.portions.analytes.annotations.entity_id |
| samples.portions.analytes.annotations.entity_submitter_id |
| samples.portions.analytes.annotations.entity_type |
| samples.portions.analytes.annotations.legacy_created_datetime |
| samples.portions.analytes.annotations.legacy_updated_datetime |
| samples.portions.analytes.annotations.notes |
| samples.portions.analytes.annotations.state |
| samples.portions.analytes.annotations.status |
| samples.portions.analytes.annotations.submitter_id |
| samples.portions.analytes.annotations.updated_datetime |
| samples.portions.annotations.annotation_id |
| samples.portions.annotations.case_id |
| samples.portions.annotations.case_submitter_id |
| samples.portions.annotations.category |
| samples.portions.annotations.classification |
| samples.portions.annotations.created_datetime |
| samples.portions.annotations.creator |
| samples.portions.annotations.entity_id |
| samples.portions.annotations.entity_submitter_id |
| samples.portions.annotations.entity_type |
| samples.portions.annotations.legacy_created_datetime |
| samples.portions.annotations.legacy_updated_datetime |
| samples.portions.annotations.notes |
| samples.portions.annotations.state |
| samples.portions.annotations.status |
| samples.portions.annotations.submitter_id |
| samples.portions.annotations.updated_datetime |
| samples.portions.center.center_id |
| samples.portions.center.center_type |
| samples.portions.center.code |
| samples.portions.center.name |
| samples.portions.center.namespace |
| samples.portions.center.short_name |
| samples.portions.slides.created_datetime |
| samples.portions.slides.number_proliferating_cells |
| samples.portions.slides.percent_eosinophil_infiltration |
| samples.portions.slides.percent_granulocyte_infiltration |
| samples.portions.slides.percent_inflam_infiltration |
| samples.portions.slides.percent_lymphocyte_infiltration |
| samples.portions.slides.percent_monocyte_infiltration |
| samples.portions.slides.percent_necrosis |
| samples.portions.slides.percent_neutrophil_infiltration |
| samples.portions.slides.percent_normal_cells |
| samples.portions.slides.percent_stromal_cells |
| samples.portions.slides.percent_tumor_cells |
| samples.portions.slides.percent_tumor_nuclei |
| samples.portions.slides.section_location |
| samples.portions.slides.slide_id |
| samples.portions.slides.state |
| samples.portions.slides.submitter_id |
| samples.portions.slides.updated_datetime |
| samples.portions.slides.annotations.annotation_id |
| samples.portions.slides.annotations.case_id |
| samples.portions.slides.annotations.case_submitter_id |
| samples.portions.slides.annotations.category |
| samples.portions.slides.annotations.classification |
| samples.portions.slides.annotations.created_datetime |
| samples.portions.slides.annotations.creator |
| samples.portions.slides.annotations.entity_id |
| samples.portions.slides.annotations.entity_submitter_id |
| samples.portions.slides.annotations.entity_type |
| samples.portions.slides.annotations.legacy_created_datetime |
| samples.portions.slides.annotations.legacy_updated_datetime |
| samples.portions.slides.annotations.notes |
| samples.portions.slides.annotations.state |
| samples.portions.slides.annotations.status |
| samples.portions.slides.annotations.submitter_id |
| samples.portions.slides.annotations.updated_datetime |
| summary.file_count |
| summary.file_size |
| summary.data_categories.data_category |
| summary.data_categories.file_count |
| summary.experimental_strategies.experimental_strategy |
| summary.experimental_strategies.file_count |
| tissue_source_site.bcr_id |
| tissue_source_site.code |
| tissue_source_site.name |
| tissue_source_site.project |
| tissue_source_site.tissue_source_site_id |


### File Fields

| Field Name |
| --- |
| access |
| acl |
| created_datetime |
| data_category |
| data_format |
| data_type |
| error_type |
| experimental_strategy |
| file_id |
| file_name |
| file_size |
| file_state |
| md5sum |
| origin |
| platform |
| revision |
| state |
| state_comment |
| submitter_id |
| tags |
| type |
| updated_datetime |
| analysis.analysis_id |
| analysis.analysis_type |
| analysis.created_datetime |
| analysis.state |
| analysis.submitter_id |
| analysis.updated_datetime |
| analysis.workflow_end_datetime |
| analysis.workflow_link |
| analysis.workflow_start_datetime |
| analysis.workflow_type |
| analysis.workflow_version |
| analysis.input_files.access |
| analysis.input_files.created_datetime |
| analysis.input_files.data_category |
| analysis.input_files.data_format |
| analysis.input_files.data_type |
| analysis.input_files.error_type |
| analysis.input_files.experimental_strategy |
| analysis.input_files.file_id |
| analysis.input_files.file_name |
| analysis.input_files.file_size |
| analysis.input_files.file_state |
| analysis.input_files.md5sum |
| analysis.input_files.platform |
| analysis.input_files.revision |
| analysis.input_files.state |
| analysis.input_files.state_comment |
| analysis.input_files.submitter_id |
| analysis.input_files.updated_datetime |
| analysis.metadata.read_groups.adapter_name |
| analysis.metadata.read_groups.adapter_sequence |
| analysis.metadata.read_groups.base_caller_name |
| analysis.metadata.read_groups.base_caller_version |
| analysis.metadata.read_groups.created_datetime |
| analysis.metadata.read_groups.experiment_name |
| analysis.metadata.read_groups.flow_cell_barcode |
| analysis.metadata.read_groups.includes_spike_ins |
| analysis.metadata.read_groups.instrument_model |
| analysis.metadata.read_groups.is_paired_end |
| analysis.metadata.read_groups.library_name |
| analysis.metadata.read_groups.library_preparation_kit_catalog_number |
| analysis.metadata.read_groups.library_preparation_kit_name |
| analysis.metadata.read_groups.library_preparation_kit_vendor |
| analysis.metadata.read_groups.library_preparation_kit_version |
| analysis.metadata.read_groups.library_selection |
| analysis.metadata.read_groups.library_strand |
| analysis.metadata.read_groups.library_strategy |
| analysis.metadata.read_groups.platform |
| analysis.metadata.read_groups.read_group_id |
| analysis.metadata.read_groups.read_group_name |
| analysis.metadata.read_groups.read_length |
| analysis.metadata.read_groups.RIN |
| analysis.metadata.read_groups.sequencing_center |
| analysis.metadata.read_groups.sequencing_date |
| analysis.metadata.read_groups.size_selection_range |
| analysis.metadata.read_groups.spike_ins_concentration |
| analysis.metadata.read_groups.spike_ins_fasta |
| analysis.metadata.read_groups.state |
| analysis.metadata.read_groups.submitter_id |
| analysis.metadata.read_groups.target_capture_kit_catalog_number |
| analysis.metadata.read_groups.target_capture_kit_name |
| analysis.metadata.read_groups.target_capture_kit_target_region |
| analysis.metadata.read_groups.target_capture_kit_vendor |
| analysis.metadata.read_groups.target_capture_kit_version |
| analysis.metadata.read_groups.to_trim_adapter_sequence |
| analysis.metadata.read_groups.updated_datetime |
| analysis.metadata.read_groups.read_group_qcs.adapter_content |
| analysis.metadata.read_groups.read_group_qcs.basic_statistics |
| analysis.metadata.read_groups.read_group_qcs.created_datetime |
| analysis.metadata.read_groups.read_group_qcs.encoding |
| analysis.metadata.read_groups.read_group_qcs.fastq_name |
| analysis.metadata.read_groups.read_group_qcs.kmer_content |
| analysis.metadata.read_groups.read_group_qcs.overrepresented_sequences |
| analysis.metadata.read_groups.read_group_qcs.per_base_n_content |
| analysis.metadata.read_groups.read_group_qcs.per_base_sequence_content |
| analysis.metadata.read_groups.read_group_qcs.per_base_sequence_quality |
| analysis.metadata.read_groups.read_group_qcs.per_sequence_gc_content |
| analysis.metadata.read_groups.read_group_qcs.per_sequence_quality_score |
| analysis.metadata.read_groups.read_group_qcs.per_tile_sequence_quality |
| analysis.metadata.read_groups.read_group_qcs.percent_gc_content |
| analysis.metadata.read_groups.read_group_qcs.read_group_qc_id |
| analysis.metadata.read_groups.read_group_qcs.sequence_duplication_levels |
| analysis.metadata.read_groups.read_group_qcs.sequence_length_distribution |
| analysis.metadata.read_groups.read_group_qcs.state |
| analysis.metadata.read_groups.read_group_qcs.submitter_id |
| analysis.metadata.read_groups.read_group_qcs.total_sequences |
| analysis.metadata.read_groups.read_group_qcs.updated_datetime |
| analysis.metadata.read_groups.read_group_qcs.workflow_end_datetime |
| analysis.metadata.read_groups.read_group_qcs.workflow_link |
| analysis.metadata.read_groups.read_group_qcs.workflow_start_datetime |
| analysis.metadata.read_groups.read_group_qcs.workflow_type |
| analysis.metadata.read_groups.read_group_qcs.workflow_version |
| annotations.annotation_id |
| annotations.case_id |
| annotations.case_submitter_id |
| annotations.category |
| annotations.classification |
| annotations.created_datetime |
| annotations.creator |
| annotations.entity_id |
| annotations.entity_submitter_id |
| annotations.entity_type |
| annotations.legacy_created_datetime |
| annotations.legacy_updated_datetime |
| annotations.notes |
| annotations.state |
| annotations.status |
| annotations.submitter_id |
| annotations.updated_datetime |
| archive.archive_id |
| archive.created_datetime |
| archive.data_category |
| archive.data_format |
| archive.data_type |
| archive.error_type |
| archive.file_name |
| archive.file_size |
| archive.file_state |
| archive.md5sum |
| archive.revision |
| archive.state |
| archive.state_comment |
| archive.submitter_id |
| archive.updated_datetime |
| associated_entities.case_id |
| associated_entities.entity_id |
| associated_entities.entity_submitter_id |
| associated_entities.entity_type |
| cases.aliquot_ids |
| cases.analyte_ids |
| cases.case_id |
| cases.created_datetime |
| cases.days_to_index |
| cases.portion_ids |
| cases.sample_ids |
| cases.slide_ids |
| cases.state |
| cases.submitter_aliquot_ids |
| cases.submitter_analyte_ids |
| cases.submitter_id |
| cases.submitter_portion_ids |
| cases.submitter_sample_ids |
| cases.submitter_slide_ids |
| cases.updated_datetime |
| cases.annotations.annotation_id |
| cases.annotations.case_id |
| cases.annotations.case_submitter_id |
| cases.annotations.category |
| cases.annotations.classification |
| cases.annotations.created_datetime |
| cases.annotations.creator |
| cases.annotations.entity_id |
| cases.annotations.entity_submitter_id |
| cases.annotations.entity_type |
| cases.annotations.legacy_created_datetime |
| cases.annotations.legacy_updated_datetime |
| cases.annotations.notes |
| cases.annotations.state |
| cases.annotations.status |
| cases.annotations.submitter_id |
| cases.annotations.updated_datetime |
| cases.demographic.created_datetime |
| cases.demographic.demographic_id |
| cases.demographic.ethnicity |
| cases.demographic.gender |
| cases.demographic.race |
| cases.demographic.state |
| cases.demographic.submitter_id |
| cases.demographic.updated_datetime |
| cases.demographic.year_of_birth |
| cases.demographic.year_of_death |
| cases.diagnoses.age_at_diagnosis |
| cases.diagnoses.classification_of_tumor |
| cases.diagnoses.created_datetime |
| cases.diagnoses.days_to_birth |
| cases.diagnoses.days_to_death |
| cases.diagnoses.days_to_last_follow_up |
| cases.diagnoses.days_to_last_known_disease_status |
| cases.diagnoses.days_to_recurrence |
| cases.diagnoses.diagnosis_id |
| cases.diagnoses.last_known_disease_status |
| cases.diagnoses.morphology |
| cases.diagnoses.primary_diagnosis |
| cases.diagnoses.prior_malignancy |
| cases.diagnoses.progression_or_recurrence |
| cases.diagnoses.site_of_resection_or_biopsy |
| cases.diagnoses.state |
| cases.diagnoses.submitter_id |
| cases.diagnoses.tissue_or_organ_of_origin |
| cases.diagnoses.tumor_grade |
| cases.diagnoses.tumor_stage |
| cases.diagnoses.updated_datetime |
| cases.diagnoses.vital_status |
| cases.diagnoses.treatments.created_datetime |
| cases.diagnoses.treatments.days_to_treatment |
| cases.diagnoses.treatments.state |
| cases.diagnoses.treatments.submitter_id |
| cases.diagnoses.treatments.therapeutic_agents |
| cases.diagnoses.treatments.treatment_id |
| cases.diagnoses.treatments.treatment_intent_type |
| cases.diagnoses.treatments.treatment_or_therapy |
| cases.diagnoses.treatments.updated_datetime |
| cases.exposures.alcohol_history |
| cases.exposures.alcohol_intensity |
| cases.exposures.bmi |
| cases.exposures.cigarettes_per_day |
| cases.exposures.created_datetime |
| cases.exposures.exposure_id |
| cases.exposures.height |
| cases.exposures.state |
| cases.exposures.submitter_id |
| cases.exposures.updated_datetime |
| cases.exposures.weight |
| cases.exposures.years_smoked |
| cases.family_histories.created_datetime |
| cases.family_histories.family_history_id |
| cases.family_histories.relationship_age_at_diagnosis |
| cases.family_histories.relationship_gender |
| cases.family_histories.relationship_primary_diagnosis |
| cases.family_histories.relationship_type |
| cases.family_histories.relative_with_cancer_history |
| cases.family_histories.state |
| cases.family_histories.submitter_id |
| cases.family_histories.updated_datetime |
| cases.files.created_datetime |
| cases.files.error_type |
| cases.files.file_id |
| cases.files.file_name |
| cases.files.file_size |
| cases.files.file_state |
| cases.files.md5sum |
| cases.files.state |
| cases.files.state_comment |
| cases.files.submitter_id |
| cases.files.updated_datetime |
| cases.follow_ups.adverse_event |
| cases.follow_ups.adverse_event_grade |
| cases.follow_ups.aids_risk_factors |
| cases.follow_ups.barretts_esophagus_goblet_cells_present |
| cases.follow_ups.bmi |
| cases.follow_ups.body_surface_area |
| cases.follow_ups.cause_of_response |
| cases.follow_ups.cd4_count |
| cases.follow_ups.cdc_hiv_risk_factors |
| cases.follow_ups.comorbidities |
| cases.follow_ups.comorbidity |
| cases.follow_ups.comorbidity_method_of_diagnosis |
| cases.follow_ups.created_datetime |
| cases.follow_ups.days_to_adverse_event |
| cases.follow_ups.days_to_comorbidity |
| cases.follow_ups.days_to_first_event |
| cases.follow_ups.days_to_follow_up |
| cases.follow_ups.days_to_imaging |
| cases.follow_ups.days_to_progression |
| cases.follow_ups.days_to_progression_free |
| cases.follow_ups.days_to_recurrence |
| cases.follow_ups.days_to_risk_factor |
| cases.follow_ups.diabetes_treatment_type |
| cases.follow_ups.discontiguous_lesion_count |
| cases.follow_ups.disease_response |
| cases.follow_ups.dlco_ref_predictive_percent |
| cases.follow_ups.ecog_performance_status |
| cases.follow_ups.evidence_of_progression_type |
| cases.follow_ups.evidence_of_recurrence_type |
| cases.follow_ups.eye_color |
| cases.follow_ups.fev1_fvc_post_bronch_percent |
| cases.follow_ups.fev1_fvc_pre_bronch_percent |
| cases.follow_ups.fev1_ref_post_bronch_percent |
| cases.follow_ups.fev1_ref_pre_bronch_percent |
| cases.follow_ups.first_event |
| cases.follow_ups.follow_up_id |
| cases.follow_ups.haart_treatment_indicator |
| cases.follow_ups.height |
| cases.follow_ups.hepatitis_sustained_virological_response |
| cases.follow_ups.history_of_tumor |
| cases.follow_ups.history_of_tumor_type |
| cases.follow_ups.hiv_viral_load |
| cases.follow_ups.hormonal_contraceptive_type |
| cases.follow_ups.hormonal_contraceptive_use |
| cases.follow_ups.hormone_replacement_therapy_type |
| cases.follow_ups.hpv_positive_type |
| cases.follow_ups.hysterectomy_margins_involved |
| cases.follow_ups.hysterectomy_type |
| cases.follow_ups.imaging_anatomic_site |
| cases.follow_ups.imaging_findings |
| cases.follow_ups.imaging_result |
| cases.follow_ups.imaging_suv |
| cases.follow_ups.imaging_suv_max |
| cases.follow_ups.imaging_type |
| cases.follow_ups.immunosuppressive_treatment_type |
| cases.follow_ups.karnofsky_performance_status |
| cases.follow_ups.menopause_status |
| cases.follow_ups.molecular_tests.aa_change |
| cases.follow_ups.molecular_tests.aneuploidy |
| cases.follow_ups.molecular_tests.antigen |
| cases.follow_ups.molecular_tests.biospecimen_type |
| cases.follow_ups.molecular_tests.biospecimen_volume |
| cases.follow_ups.molecular_tests.blood_test_normal_range_lower |
| cases.follow_ups.molecular_tests.blood_test_normal_range_upper |
| cases.follow_ups.molecular_tests.cell_count |
| cases.follow_ups.molecular_tests.chromosomal_translocation |
| cases.follow_ups.molecular_tests.chromosome |
| cases.follow_ups.molecular_tests.chromosome_arm |
| cases.follow_ups.molecular_tests.clonality |
| cases.follow_ups.molecular_tests.copy_number |
| cases.follow_ups.molecular_tests.created_datetime |
| cases.follow_ups.molecular_tests.cytoband |
| cases.follow_ups.molecular_tests.days_to_test |
| cases.follow_ups.molecular_tests.exon |
| cases.follow_ups.molecular_tests.gene_symbol |
| cases.follow_ups.molecular_tests.histone_family |
| cases.follow_ups.molecular_tests.histone_variant |
| cases.follow_ups.molecular_tests.hpv_strain |
| cases.follow_ups.molecular_tests.intron |
| cases.follow_ups.molecular_tests.laboratory_test |
| cases.follow_ups.molecular_tests.loci_abnormal_count |
| cases.follow_ups.molecular_tests.loci_count |
| cases.follow_ups.molecular_tests.locus |
| cases.follow_ups.molecular_tests.mismatch_repair_mutation |
| cases.follow_ups.molecular_tests.mitotic_count |
| cases.follow_ups.molecular_tests.mitotic_total_area |
| cases.follow_ups.molecular_tests.molecular_analysis_method |
| cases.follow_ups.molecular_tests.molecular_consequence |
| cases.follow_ups.molecular_tests.molecular_test_id |
| cases.follow_ups.molecular_tests.mutation_codon |
| cases.follow_ups.molecular_tests.pathogenicity |
| cases.follow_ups.molecular_tests.ploidy |
| cases.follow_ups.molecular_tests.second_exon |
| cases.follow_ups.molecular_tests.second_gene_symbol |
| cases.follow_ups.molecular_tests.specialized_molecular_test |
| cases.follow_ups.molecular_tests.staining_intensity_scale |
| cases.follow_ups.molecular_tests.staining_intensity_value |
| cases.follow_ups.molecular_tests.state |
| cases.follow_ups.molecular_tests.submitter_id |
| cases.follow_ups.molecular_tests.test_analyte_type |
| cases.follow_ups.molecular_tests.test_result |
| cases.follow_ups.molecular_tests.test_units |
| cases.follow_ups.molecular_tests.test_value |
| cases.follow_ups.molecular_tests.test_value_range |
| cases.follow_ups.molecular_tests.timepoint_category |
| cases.follow_ups.molecular_tests.transcript |
| cases.follow_ups.molecular_tests.updated_datetime |
| cases.follow_ups.molecular_tests.variant_origin |
| cases.follow_ups.molecular_tests.variant_type |
| cases.follow_ups.molecular_tests.zygosity |
| cases.follow_ups.nadir_cd4_count |
| cases.follow_ups.other_clinical_attributes.aids_risk_factors |
| cases.follow_ups.other_clinical_attributes.bmi |
| cases.follow_ups.other_clinical_attributes.body_surface_area |
| cases.follow_ups.other_clinical_attributes.cd4_count |
| cases.follow_ups.other_clinical_attributes.cdc_hiv_risk_factors |
| cases.follow_ups.other_clinical_attributes.comorbidities |
| cases.follow_ups.other_clinical_attributes.comorbidity_method_of_diagnosis |
| cases.follow_ups.other_clinical_attributes.created_datetime |
| cases.follow_ups.other_clinical_attributes.days_to_comorbidity |
| cases.follow_ups.other_clinical_attributes.days_to_risk_factor |
| cases.follow_ups.other_clinical_attributes.diabetes_treatment_type |
| cases.follow_ups.other_clinical_attributes.dlco_ref_predictive_percent |
| cases.follow_ups.other_clinical_attributes.exercise_frequency_weekly |
| cases.follow_ups.other_clinical_attributes.eye_color |
| cases.follow_ups.other_clinical_attributes.fev1_fvc_post_bronch_percent |
| cases.follow_ups.other_clinical_attributes.fev1_fvc_pre_bronch_percent |
| cases.follow_ups.other_clinical_attributes.fev1_ref_post_bronch_percent |
| cases.follow_ups.other_clinical_attributes.fev1_ref_pre_bronch_percent |
| cases.follow_ups.other_clinical_attributes.haart_treatment_indicator |
| cases.follow_ups.other_clinical_attributes.height |
| cases.follow_ups.other_clinical_attributes.hepatitis_sustained_virological_response |
| cases.follow_ups.other_clinical_attributes.hiv_viral_load |
| cases.follow_ups.other_clinical_attributes.hormonal_contraceptive_type |
| cases.follow_ups.other_clinical_attributes.hormonal_contraceptive_use |
| cases.follow_ups.other_clinical_attributes.hormonal_replacement_therapy_status |
| cases.follow_ups.other_clinical_attributes.hormone_replacement_therapy_type |
| cases.follow_ups.other_clinical_attributes.hysterectomy_margins_involved |
| cases.follow_ups.other_clinical_attributes.hysterectomy_type |
| cases.follow_ups.other_clinical_attributes.immunosuppressive_treatment_type |
| cases.follow_ups.other_clinical_attributes.menopause_status |
| cases.follow_ups.other_clinical_attributes.myasthenia_gravis_classification |
| cases.follow_ups.other_clinical_attributes.nadir_cd4_count |
| cases.follow_ups.other_clinical_attributes.nononcologic_therapeutic_agents |
| cases.follow_ups.other_clinical_attributes.number_of_pregnancies |
| cases.follow_ups.other_clinical_attributes.other_clinical_attribute_id |
| cases.follow_ups.other_clinical_attributes.oxygen_use_indicator |
| cases.follow_ups.other_clinical_attributes.oxygen_use_type |
| cases.follow_ups.other_clinical_attributes.pancreatitis_onset_year |
| cases.follow_ups.other_clinical_attributes.pregnancy_count |
| cases.follow_ups.other_clinical_attributes.pregnancy_outcome |
| cases.follow_ups.other_clinical_attributes.pregnant_at_diagnosis |
| cases.follow_ups.other_clinical_attributes.premature_at_birth |
| cases.follow_ups.other_clinical_attributes.reflux_treatment_type |
| cases.follow_ups.other_clinical_attributes.risk_factor_method_of_diagnosis |
| cases.follow_ups.other_clinical_attributes.risk_factor_treatment |
| cases.follow_ups.other_clinical_attributes.risk_factors |
| cases.follow_ups.other_clinical_attributes.state |
| cases.follow_ups.other_clinical_attributes.submitter_id |
| cases.follow_ups.other_clinical_attributes.timepoint_category |
| cases.follow_ups.other_clinical_attributes.treatment_frequency |
| cases.follow_ups.other_clinical_attributes.undescended_testis_corrected |
| cases.follow_ups.other_clinical_attributes.undescended_testis_corrected_age |
| cases.follow_ups.other_clinical_attributes.undescended_testis_corrected_laterality |
| cases.follow_ups.other_clinical_attributes.undescended_testis_corrected_method |
| cases.follow_ups.other_clinical_attributes.undescended_testis_history |
| cases.follow_ups.other_clinical_attributes.undescended_testis_history_laterality |
| cases.follow_ups.other_clinical_attributes.updated_datetime |
| cases.follow_ups.other_clinical_attributes.viral_hepatitis_serologies |
| cases.follow_ups.other_clinical_attributes.viral_hepatitis_serology_tests |
| cases.follow_ups.other_clinical_attributes.weeks_gestation_at_birth |
| cases.follow_ups.other_clinical_attributes.weight |
| cases.follow_ups.pancreatitis_onset_year |
| cases.follow_ups.peritoneal_washing_results |
| cases.follow_ups.pregnancy_count |
| cases.follow_ups.pregnancy_outcome |
| cases.follow_ups.procedures_performed |
| cases.follow_ups.progression_or_recurrence |
| cases.follow_ups.progression_or_recurrence_anatomic_site |
| cases.follow_ups.progression_or_recurrence_type |
| cases.follow_ups.recist_targeted_regions_number |
| cases.follow_ups.recist_targeted_regions_sum |
| cases.follow_ups.reflux_treatment_type |
| cases.follow_ups.risk_factor |
| cases.follow_ups.risk_factor_method_of_diagnosis |
| cases.follow_ups.risk_factor_treatment |
| cases.follow_ups.risk_factors |
| cases.follow_ups.scan_tracer_used |
| cases.follow_ups.state |
| cases.follow_ups.submitter_id |
| cases.follow_ups.timepoint_category |
| cases.follow_ups.treatment_emergent_adverse_event |
| cases.follow_ups.undescended_testis_corrected |
| cases.follow_ups.undescended_testis_corrected_age |
| cases.follow_ups.undescended_testis_corrected_laterality |
| cases.follow_ups.undescended_testis_corrected_method |
| cases.follow_ups.undescended_testis_history |
| cases.follow_ups.undescended_testis_history_laterality |
| cases.follow_ups.updated_datetime |
| cases.follow_ups.viral_hepatitis_serologies |
| cases.follow_ups.weight |
| cases.follow_ups.year_of_follow_up |
| cases.project.dbgap_accession_number |
| cases.project.disease_type |
| cases.project.name |
| cases.project.primary_site |
| cases.project.project_id |
| cases.project.released |
| cases.project.state |
| cases.project.program.dbgap_accession_number |
| cases.project.program.name |
| cases.project.program.program_id |
| cases.samples.composition |
| cases.samples.created_datetime |
| cases.samples.current_weight |
| cases.samples.days_to_collection |
| cases.samples.days_to_sample_procurement |
| cases.samples.freezing_method |
| cases.samples.initial_weight |
| cases.samples.intermediate_dimension |
| cases.samples.is_ffpe |
| cases.samples.longest_dimension |
| cases.samples.oct_embedded |
| cases.samples.pathology_report_uuid |
| cases.samples.preservation_method |
| cases.samples.sample_id |
| cases.samples.sample_type |
| cases.samples.sample_type_id |
| cases.samples.shortest_dimension |
| cases.samples.state |
| cases.samples.submitter_id |
| cases.samples.time_between_clamping_and_freezing |
| cases.samples.time_between_excision_and_freezing |
| cases.samples.tissue_type |
| cases.samples.tumor_code |
| cases.samples.tumor_code_id |
| cases.samples.tumor_descriptor |
| cases.samples.updated_datetime |
| cases.samples.annotations.annotation_id |
| cases.samples.annotations.case_id |
| cases.samples.annotations.case_submitter_id |
| cases.samples.annotations.category |
| cases.samples.annotations.classification |
| cases.samples.annotations.created_datetime |
| cases.samples.annotations.creator |
| cases.samples.annotations.entity_id |
| cases.samples.annotations.entity_submitter_id |
| cases.samples.annotations.entity_type |
| cases.samples.annotations.legacy_created_datetime |
| cases.samples.annotations.legacy_updated_datetime |
| cases.samples.annotations.notes |
| cases.samples.annotations.state |
| cases.samples.annotations.status |
| cases.samples.annotations.submitter_id |
| cases.samples.annotations.updated_datetime |
| cases.samples.portions.created_datetime |
| cases.samples.portions.creation_datetime |
| cases.samples.portions.is_ffpe |
| cases.samples.portions.portion_id |
| cases.samples.portions.portion_number |
| cases.samples.portions.state |
| cases.samples.portions.submitter_id |
| cases.samples.portions.updated_datetime |
| cases.samples.portions.weight |
| cases.samples.portions.analytes.a260_a280_ratio |
| cases.samples.portions.analytes.amount |
| cases.samples.portions.analytes.analyte_id |
| cases.samples.portions.analytes.analyte_type |
| cases.samples.portions.analytes.analyte_type_id |
| cases.samples.portions.analytes.concentration |
| cases.samples.portions.analytes.created_datetime |
| cases.samples.portions.analytes.spectrophotometer_method |
| cases.samples.portions.analytes.state |
| cases.samples.portions.analytes.submitter_id |
| cases.samples.portions.analytes.updated_datetime |
| cases.samples.portions.analytes.well_number |
| cases.samples.portions.analytes.aliquots.aliquot_id |
| cases.samples.portions.analytes.aliquots.amount |
| cases.samples.portions.analytes.aliquots.analyte_type |
| cases.samples.portions.analytes.aliquots.analyte_type_id |
| cases.samples.portions.analytes.aliquots.concentration |
| cases.samples.portions.analytes.aliquots.created_datetime |
| cases.samples.portions.analytes.aliquots.source_center |
| cases.samples.portions.analytes.aliquots.state |
| cases.samples.portions.analytes.aliquots.submitter_id |
| cases.samples.portions.analytes.aliquots.updated_datetime |
| cases.samples.portions.analytes.aliquots.annotations.annotation_id |
| cases.samples.portions.analytes.aliquots.annotations.case_id |
| cases.samples.portions.analytes.aliquots.annotations.case_submitter_id |
| cases.samples.portions.analytes.aliquots.annotations.category |
| cases.samples.portions.analytes.aliquots.annotations.classification |
| cases.samples.portions.analytes.aliquots.annotations.created_datetime |
| cases.samples.portions.analytes.aliquots.annotations.creator |
| cases.samples.portions.analytes.aliquots.annotations.entity_id |
| cases.samples.portions.analytes.aliquots.annotations.entity_submitter_id |
| cases.samples.portions.analytes.aliquots.annotations.entity_type |
| cases.samples.portions.analytes.aliquots.annotations.legacy_created_datetime |
| cases.samples.portions.analytes.aliquots.annotations.legacy_updated_datetime |
| cases.samples.portions.analytes.aliquots.annotations.notes |
| cases.samples.portions.analytes.aliquots.annotations.state |
| cases.samples.portions.analytes.aliquots.annotations.status |
| cases.samples.portions.analytes.aliquots.annotations.submitter_id |
| cases.samples.portions.analytes.aliquots.annotations.updated_datetime |
| cases.samples.portions.analytes.aliquots.center.center_id |
| cases.samples.portions.analytes.aliquots.center.center_type |
| cases.samples.portions.analytes.aliquots.center.code |
| cases.samples.portions.analytes.aliquots.center.name |
| cases.samples.portions.analytes.aliquots.center.namespace |
| cases.samples.portions.analytes.aliquots.center.short_name |
| cases.samples.portions.analytes.annotations.annotation_id |
| cases.samples.portions.analytes.annotations.case_id |
| cases.samples.portions.analytes.annotations.case_submitter_id |
| cases.samples.portions.analytes.annotations.category |
| cases.samples.portions.analytes.annotations.classification |
| cases.samples.portions.analytes.annotations.created_datetime |
| cases.samples.portions.analytes.annotations.creator |
| cases.samples.portions.analytes.annotations.entity_id |
| cases.samples.portions.analytes.annotations.entity_submitter_id |
| cases.samples.portions.analytes.annotations.entity_type |
| cases.samples.portions.analytes.annotations.legacy_created_datetime |
| cases.samples.portions.analytes.annotations.legacy_updated_datetime |
| cases.samples.portions.analytes.annotations.notes |
| cases.samples.portions.analytes.annotations.state |
| cases.samples.portions.analytes.annotations.status |
| cases.samples.portions.analytes.annotations.submitter_id |
| cases.samples.portions.analytes.annotations.updated_datetime |
| cases.samples.portions.annotations.annotation_id |
| cases.samples.portions.annotations.case_id |
| cases.samples.portions.annotations.case_submitter_id |
| cases.samples.portions.annotations.category |
| cases.samples.portions.annotations.classification |
| cases.samples.portions.annotations.created_datetime |
| cases.samples.portions.annotations.creator |
| cases.samples.portions.annotations.entity_id |
| cases.samples.portions.annotations.entity_submitter_id |
| cases.samples.portions.annotations.entity_type |
| cases.samples.portions.annotations.legacy_created_datetime |
| cases.samples.portions.annotations.legacy_updated_datetime |
| cases.samples.portions.annotations.notes |
| cases.samples.portions.annotations.state |
| cases.samples.portions.annotations.status |
| cases.samples.portions.annotations.submitter_id |
| cases.samples.portions.annotations.updated_datetime |
| cases.samples.portions.center.center_id |
| cases.samples.portions.center.center_type |
| cases.samples.portions.center.code |
| cases.samples.portions.center.name |
| cases.samples.portions.center.namespace |
| cases.samples.portions.center.short_name |
| cases.samples.portions.slides.created_datetime |
| cases.samples.portions.slides.number_proliferating_cells |
| cases.samples.portions.slides.percent_eosinophil_infiltration |
| cases.samples.portions.slides.percent_granulocyte_infiltration |
| cases.samples.portions.slides.percent_inflam_infiltration |
| cases.samples.portions.slides.percent_lymphocyte_infiltration |
| cases.samples.portions.slides.percent_monocyte_infiltration |
| cases.samples.portions.slides.percent_necrosis |
| cases.samples.portions.slides.percent_neutrophil_infiltration |
| cases.samples.portions.slides.percent_normal_cells |
| cases.samples.portions.slides.percent_stromal_cells |
| cases.samples.portions.slides.percent_tumor_cells |
| cases.samples.portions.slides.percent_tumor_nuclei |
| cases.samples.portions.slides.section_location |
| cases.samples.portions.slides.slide_id |
| cases.samples.portions.slides.state |
| cases.samples.portions.slides.submitter_id |
| cases.samples.portions.slides.updated_datetime |
| cases.samples.portions.slides.annotations.annotation_id |
| cases.samples.portions.slides.annotations.case_id |
| cases.samples.portions.slides.annotations.case_submitter_id |
| cases.samples.portions.slides.annotations.category |
| cases.samples.portions.slides.annotations.classification |
| cases.samples.portions.slides.annotations.created_datetime |
| cases.samples.portions.slides.annotations.creator |
| cases.samples.portions.slides.annotations.entity_id |
| cases.samples.portions.slides.annotations.entity_submitter_id |
| cases.samples.portions.slides.annotations.entity_type |
| cases.samples.portions.slides.annotations.legacy_created_datetime |
| cases.samples.portions.slides.annotations.legacy_updated_datetime |
| cases.samples.portions.slides.annotations.notes |
| cases.samples.portions.slides.annotations.state |
| cases.samples.portions.slides.annotations.status |
| cases.samples.portions.slides.annotations.submitter_id |
| cases.samples.portions.slides.annotations.updated_datetime |
| cases.summary.file_count |
| cases.summary.file_size |
| cases.summary.data_categories.data_category |
| cases.summary.data_categories.file_count |
| cases.summary.experimental_strategies.experimental_strategy |
| cases.summary.experimental_strategies.file_count |
| cases.tissue_source_site.bcr_id |
| cases.tissue_source_site.code |
| cases.tissue_source_site.name |
| cases.tissue_source_site.project |
| cases.tissue_source_site.tissue_source_site_id |
| center.center_id |
| center.center_type |
| center.code |
| center.name |
| center.namespace |
| center.short_name |
| downstream_analyses.analysis_id |
| downstream_analyses.analysis_type |
| downstream_analyses.created_datetime |
| downstream_analyses.state |
| downstream_analyses.submitter_id |
| downstream_analyses.updated_datetime |
| downstream_analyses.workflow_end_datetime |
| downstream_analyses.workflow_link |
| downstream_analyses.workflow_start_datetime |
| downstream_analyses.workflow_type |
| downstream_analyses.workflow_version |
| downstream_analyses.output_files.access |
| downstream_analyses.output_files.created_datetime |
| downstream_analyses.output_files.data_category |
| downstream_analyses.output_files.data_format |
| downstream_analyses.output_files.data_type |
| downstream_analyses.output_files.error_type |
| downstream_analyses.output_files.experimental_strategy |
| downstream_analyses.output_files.file_id |
| downstream_analyses.output_files.file_name |
| downstream_analyses.output_files.file_size |
| downstream_analyses.output_files.file_state |
| downstream_analyses.output_files.md5sum |
| downstream_analyses.output_files.platform |
| downstream_analyses.output_files.revision |
| downstream_analyses.output_files.state |
| downstream_analyses.output_files.state_comment |
| downstream_analyses.output_files.submitter_id |
| downstream_analyses.output_files.updated_datetime |
| index_files.access |
| index_files.created_datetime |
| index_files.data_category |
| index_files.data_format |
| index_files.data_type |
| index_files.error_type |
| index_files.experimental_strategy |
| index_files.file_id |
| index_files.file_name |
| index_files.file_size |
| index_files.file_state |
| index_files.md5sum |
| index_files.platform |
| index_files.revision |
| index_files.state |
| index_files.state_comment |
| index_files.submitter_id |
| index_files.updated_datetime |
| metadata_files.access |
| metadata_files.created_datetime |
| metadata_files.data_category |
| metadata_files.data_format |
| metadata_files.data_type |
| metadata_files.error_type |
| metadata_files.file_id |
| metadata_files.file_name |
| metadata_files.file_size |
| metadata_files.file_state |
| metadata_files.md5sum |
| metadata_files.state |
| metadata_files.state_comment |
| metadata_files.submitter_id |
| metadata_files.type |
| metadata_files.updated_datetime |



### Annotation Fields


| Field Name |
| --- |
| annotation_id |
| case_id |
| case_submitter_id |
| category |
| classification |
| created_datetime |
| entity_id |
| entity_submitter_id |
| entity_type |
| legacy_created_datetime |
| legacy_updated_datetime |
| notes |
| state |
| status |
| submitter_id |
| updated_datetime |
| project.code |
| project.dbgap_accession_number |
| project.disease_type |
| project.name |
| project.primary_site |
| project.program.dbgap_accession_number |
| project.program.name |
| project.program.program_id |
| project.project_id |
| project.released |
| project.state |

## Field Group Listing by Endpoint

### Projects Field Groups

| Field Group Name |
| --- |
| program |
| summary |
| summary.data_categories |
| summary.experimental_strategies |

### Cases Field Groups

| Field Group Name |
| --- |
| annotations |
| demographic |
| diagnoses |
| diagnoses.treatments |
| exposures |
| family_histories |
| files |
| files.analysis |
| files.analysis.input_files |
| files.analysis.metadata |
| files.analysis.metadata.read_groups |
| files.analysis.metadata.read_groups.read_group_qcs |
| files.archive |
| files.cases |
| files.cases.annotations |
| files.cases.demographic |
| files.cases.diagnoses |
| files.cases.diagnoses.treatments |
| files.cases.exposures |
| files.cases.family_histories |
| files.cases.files |
| files.cases.project |
| files.cases.project.program |
| files.cases.samples |
| files.cases.samples.annotations |
| files.cases.samples.portions |
| files.cases.samples.portions.analytes |
| files.cases.samples.portions.analytes.aliquots |
| files.cases.samples.portions.analytes.aliquots.annotations |
| files.cases.samples.portions.analytes.aliquots.center |
| files.cases.samples.portions.analytes.annotations |
| files.cases.samples.portions.annotations |
| files.cases.samples.portions.center |
| files.cases.samples.portions.slides |
| files.cases.samples.portions.slides.annotations |
| files.cases.summary |
| files.cases.summary.data_categories |
| files.cases.summary.experimental_strategies |
| files.cases.tissue_source_site |
| files.center |
| files.downstream_analyses |
| files.downstream_analyses.output_files |
| files.index_files |
| files.metadata_files |
| project |
| project.program |
| samples |
| samples.annotations |
| samples.portions |
| samples.portions.analytes |
| samples.portions.analytes.aliquots |
| samples.portions.analytes.aliquots.annotations |
| samples.portions.analytes.aliquots.center |
| samples.portions.analytes.annotations |
| samples.portions.annotations |
| samples.portions.center |
| samples.portions.slides |
| samples.portions.slides.annotations |
| summary |
| summary.data_categories |
| summary.experimental_strategies |
| tissue_source_site |

### Files Field Groups

| Field Group Name |
| --- |
| analysis |
| analysis.input_files |
| analysis.metadata |
| analysis.metadata.read_groups |
| analysis.metadata.read_groups.read_group_qcs |
| annotations |
| archive |
| associated_entities |
| cases |
| cases.annotations |
| cases.demographic |
| cases.diagnoses |
| cases.diagnoses.treatments |
| cases.exposures |
| cases.family_histories |
| cases.files |
| cases.project |
| cases.project.program |
| cases.samples |
| cases.samples.annotations |
| cases.samples.portions |
| cases.samples.portions.analytes |
| cases.samples.portions.analytes.aliquots |
| cases.samples.portions.analytes.aliquots.annotations |
| cases.samples.portions.analytes.aliquots.center |
| cases.samples.portions.analytes.annotations |
| cases.samples.portions.annotations |
| cases.samples.portions.center |
| cases.samples.portions.slides |
| cases.samples.portions.slides.annotations |
| cases.summary |
| cases.summary.data_categories |
| cases.summary.experimental_strategies |
| cases.tissue_source_site |
| center |
| downstream_analyses |
| downstream_analyses.output_files |
| index_files |
| metadata_files |

### Annotations Field Groups

| Field Group Name |
| --- |
| project |
| project.program |

### GDC Conventions Supplemental
| Field Group Name |
| --- | --- |
| File size | Listed in bytes |
| Submitter id | String supplied by submitter |
| Datetime | Combination of date and time of day in the form [-]CCYY-MM-DDThh:mm:ss |    
| Id | GDC supplied UUID ID |  


# API Release Notes


| Version | Date |
|---|---|
| [v7.4.1](API_Release_Notes.md#v741) | July 30, 2024 |
| [v7.3.0](API_Release_Notes.md#v730) | June 26, 2024 |
| [v4.0.0](API_Release_Notes.md#v400) | July 31, 2023 |
| [v3.28.0](API_Release_Notes.md#v3280) | May 11, 2023 |
| [v3.5.0](API_Release_Notes.md#v350) | July 8, 2022 |
| [v3.3.0](API_Release_Notes.md#v330) | May 17, 2021 |
| [v3.0.0](API_Release_Notes.md#v300) | August 14, 2020 |
| [v2.1.2](API_Release_Notes.md#v212) | May 7, 2020 |
| [v2.1.0](API_Release_Notes.md#v210) | March 10, 2020 |
| [v2.0.0](API_Release_Notes.md#v200) | January 30, 2020 |
| [v1.23.0](API_Release_Notes.md#v1230) | November 6, 2019 |
| [v1.22.0](API_Release_Notes.md#v1220) | July 31, 2019 |
| [v1.21.0](API_Release_Notes.md#v1210) | June 5, 2019 |
| [v1.20.0](API_Release_Notes.md#v1200) | April 17, 2019 |
| [v1.19.0](API_Release_Notes.md#v1190) | February 20, 2019 |
| [v1.18.0](API_Release_Notes.md#v1180) | December 18, 2018 |
| [v1.17.0](API_Release_Notes.md#v1170) | November 7, 2018 |
| [v1.16.0](API_Release_Notes.md#v1160) | September 27, 2018 |
| [v1.15.0](API_Release_Notes.md#v1150) | August 23, 2018 |
| [v1.14.1](API_Release_Notes.md#v1141) | May 21, 2018 |
| [v1.14.0](API_Release_Notes.md#v1140) | April 23, 2018 |
| [v1.13.0](API_Release_Notes.md#v1130) | February 15, 2018 |
| [v1.12.0](API_Release_Notes.md#v1120) | December 21, 2017 |
| [v1.11.0](API_Release_Notes.md#v1110) | November 16, 2017 |
| [v1.10.0](API_Release_Notes.md#v1100) | August 22, 2017 |
| [v1.9.0](API_Release_Notes.md#v190) | June 29, 2017 |
| [v1.8.0](API_Release_Notes.md#v180) | May 9, 2017 |
| [v1.7.1](API_Release_Notes.md#v171) | March 16, 2017 |
| [v1.5.0](API_Release_Notes.md#v150) | October 31, 2016 |
| [v1.4.0](API_Release_Notes.md#v140) | September 23, 2016 |
| [v1.3.1](API_Release_Notes.md#v131) | September 7, 2016 |
| [v1.2.0](API_Release_Notes.md#v120) | August 9, 2016 |
| [v1.1.0](API_Release_Notes.md#v110) | May 25, 2016 |
| [v1.0.1](API_Release_Notes.md#v101) | May 16, 2016 |

## v7.4.1

* __GDC Product__: Application Programming Interface (API)
* __Release Date__:  July 30, 2024

### New Features and Changes

* The API indices will include fields from the "other clinical attribute" entity types. This will be visible in the next data release.

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

* Some fields may be repeated in the API or data portal due to being in a transitional states. As a workaround, select all of the fields of interest, and the results will demonstrate which are currently populated.
* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->

## v7.3.0

* __GDC Product__: Application Programming Interface (API)
* __Release Date__:  June 26, 2024

### New Features and Changes

* The GDC submission API now supports the [PATCH method](/API/Users_Guide/Submission/#patching-entitites), which allow users to upload without required fields while updating already-existing entities.
* The gene expression API is now available to be queried by users. See the [Gene Expression API](/API/Users_Guide/Data_Analysis/#gene-expression-examples) documentation for more information.

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->

## v4.0.0

* __GDC Product__: Application Programming Interface (API)
* __Release Date__:  July 31, 2023

### New Features and Changes

* The GDC API no longer supports download of Legacy Archive files.

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->

## v3.28.0

* __GDC Product__: Application Programming Interface (API)
* __Release Date__:  May 11, 2023

### New Features and Changes

* The `/legacy/files` endpoint has been deprecated and is no longer available.  
* Features to support Data Portal bugfixes were added.

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->

## v3.5.0

* __GDC Product__: Application Programming Interface (API)
* __Release Date__:  July 8, 2022

### New Features and Changes

* Features to support Data Portal performance were added.

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->


## v3.3.0 <!--REQ-422-->

* __GDC Product__: Application Programming Interface (API)
* __Release Date__:  May 17, 2021

### New Features and Changes

* Features to better support batch tracking for submitted data were added.

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->


## v3.0.0 <!--REQ-408-->

* __GDC Product__: Application Programming Interface (API)
* __Release Date__:  August 14, 2020

### New Features and Changes

* Enhancements were made to increase performance.

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->

## v2.1.2 <!--REQ-395-->

* __GDC Product__: Application Programming Interface (API)
* __Release Date__:  May 7, 2020

### New Features and Changes

* An update to improve usability in the homepage quicksearch <!--DEV-173-->

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->

## v2.1.0 <!--REQ-395-->

* __GDC Product__: Application Programming Interface (API)
* __Release Date__:  March 10, 2020

### New Features and Changes

* New data dictionary changes. <!--REQ-395-->

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->

## v2.0.0 <!--REQ-393-->

* __GDC Product__: Application Programming Interface (API)
* __Release Date__:  January 30, 2020

### New Features and Changes

* API code now uses Python 3. <!--REQ-393-->

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->

## v1.23.0 <!--REQ-390-->

* __GDC Product__: Application Programming Interface (API)
* __Release Date__:  November 6, 2019

### New Features and Changes

* QC Tests added for Submission <!--TT-748-->
* BAM slicing is now supported for unmapped reads <!--TT-1061-->
* API now includes data from molecular_test and follow_up nodes.  This impacts what is displayed on the GDC Data Portal <!--API-750-->
* Better handling of concurrent transactions <!--API-767-->
* CIViC annotations now included on the ssms endpoint.  You can read more about CIViC annotations [here](https://civicdb.org/home) <!--API-733-->

### Bugs Fixed Since Last Release

* Fixed API memory leak <!--API-719, API-775-->
* Fixed data offset issue returned by clinical.tar endpoint <!--API-732-->

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->



## v1.22.0 <!--REQ-387-->

* __GDC Product__: Application Programming Interface (API)
* __Release Date__:  July 31, 2019

### New Features and Changes

* Updated BCR Clinical XML parser code to support future indexing of additional clinical data.  Parser code can be found [here](https://github.com/NCI-GDC/gdcdatamodel/tree/develop/gdcdatamodel/xml_mappings) <!--DAT-2315-->

    - Added a mapping for:
        - `ann_arbor_b_symptoms` <!--DAT-2399-->
        - `ann_arbor_extranodal_involvement` <!--DAT-2399-->
        - `ajcc_pathologic_t` <!--DAT-2399-->
        - `ajcc_pathologic_n` <!--DAT-2399-->
        - `ajcc_pathologic_m` <!--DAT-2399-->
        - `ajcc_clinical_t` <!--DAT-2399-->
        - `ajcc_clinical_n` <!--DAT-2399-->
        - `ajcc_clinical_m` <!--DAT-2399-->
        - `ajcc_staging_system_edition`	<!--DAT-2398-->
        - `figo_stage` <!--DAT-2401-->
        - `ajcc_clinical_stage` <!--DAT-2403-->
        - `primary_gleason_grade` <!--DAT-2404-->
        - `secondary_gleason_graade` <!--DAT-2405-->
        - `igcccg_stage` <!--DAT-2406-->
        - `masaoka_stage` <!--DAT-2407-->

    - Updated the mapping for:
        - `primary_diagnosis`, `morphology`, `tissue_or_organ_of_origin`, and `site_of_resection_or_biopsy`  <!--DAT-2409-->

### Bugs Fixed Since Last Release

* Fixed a bug preventing multipart uploads <!--API-748, SV-1472-->

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->


## v1.21.0 <!--REQ-383-->

* __GDC Product__: Application Programming Interface (API)
* __Release Date__:  June 5, 2019

### New Features and Changes

* Mutation indexer update to accommodate data model changes <!--API-677-->
* Updates to when in the release cycle downloaders and submitters have access to files <!--TT-973-->


### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->


## v1.20.0 <!--REQ-385-->

* __GDC Product__: Application Programming Interface (API)
* __Release Date__:  April 17, 2019

### New Features and Changes

* Disallowed creation of multiple file versions in the same data release <!--API-585-->
* Improved API concurrency performance and addressed random failures/4XX/5XX responses <!--GDCDEVOPS-187,GP-15-->
* Improved API/Portal performances for querying large sets of data from the Elasticsearch indices <!--API-613-->
* Updated BCR Clinical XML parser code to support future indexing of additional clinical data.  Parser code can be found [here](https://github.com/NCI-GDC/gdcdatamodel/tree/develop/gdcdatamodel/xml_mappings) <!--DAT-2076-->
    - Updated the mapping for:
        - `days_to_death`	<!--DAT-2083-->
        - `days_to_birth`	<!--DAT-2084-->
        - `vital_status`	<!--DAT-2085-->
        - `prior_malignancy` and `synchronous_malignancy`	<!--DAT-2094-->
    - Added a mapping for:
        - `pack_years_smoked`	<!--DAT-2086-->
        - `prior_treatment`	<!--DAT-2095-->
        - `age_at_index`	<!--DAT-2096-->
        - `days_to_diagnosis`	<!--DAT-2097-->
        - `icd_10_code`	<!--DAT-2098-->
        - `year_of_diagnosis`	<!--DAT-2099-->
    - Remove calculation for:
        - `cigarettes_per_day`	<!--DAT-2091-->
        - `year_of_birth`	<!--DAT-2092-->
        - `year_of_death`	<!--DAT-2093-->
        - `bmi`  <!--DAT-2087-->
    - Updated XML Parser to pull the most Up-to-Date Survival Information	<!--DAT-2063-->

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->



## v1.19.0 <!--REQ-381-->

* __GDC Product__: Application Programming Interface (API)
* __Release Date__:  February 20, 2019

### New Features and Changes

* Added API features to support controlled access DAVE <!--TT-393-->
* Updated API query endpoints to handle filtering of queries based on tokens <!--TT-641-->
* Created login notification Endpoint <!--API-586-->
* Added hashing and logging for similar ES queries <!--API-569-->

### Bugs Fixed Since Last Release

* Fixed bug where quick search ES query grows with each request <!--API-580-->
* Fixed bug where new file versions could be created when exactly the same existing metadata is uploaded <!--API-571-->
* Fixed bug where submitting to specific projects produced error that data already existed <!--API-568-->

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->


## v1.18.0 <!--REQ-335-->

* __GDC Product__: Application Programming Interface (API)
* __Release Date__:  December 18, 2018

### New Features and Changes

* Update to auth for GDC Pre-Release Data Portal <!--TT-739-->

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->



## v1.17.0

* __GDC Product__: Application Programming Interface (API)
* __Release Date__:  November 7, 2018

### New Features and Changes

* Created new index cnv_centric <!--TT743-->
* Created new index cnv_occurrence_centric <!--TT-744-->
* Created new REST API endpoints for CNV <!--TT-757-->
* Created mapping from aliquot to case for occurrence on cnv_centric <!--TT-762-->
* Created new graphql endpoints for CNV <!--TT-769-->
* Updated index case_centric to add cnv <!--TT-745-->
* Updated index gene_centric to add cnv <!--TT-747-->

### Bugs Fixed Since Last Release

* Fixed bug to prevent users from deleting files in state submitted or released <!--API-560-->

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->



## v1.16.0

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: September 27, 2018

### New Features and Changes

* Tissue_type is now parsed from the BCR Biospecimen submission <!--TT-753-->


### Bugs Fixed Since Last Release

* Re-attach the /files path to the legacy blueprints in gdcapi <!--SV-1195-->
* Fix Internal Server Error when exporting JSON in Data Portal <!--SV-1209-->

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->




## v1.15.0

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: August 23, 2018

### New Features and Changes

* Added support for file versioning <!--TT-482-->
* Added new API endpoints in support of versioning: history, files/versions, and files/versions/manifest  <!--PRTL-2015-->
* Updated the submit and release process <!--TT-249-->
* Created admin endpoint to allow for modifying file and project flags and states <!--API-368-->
* Populate version and release information for all existing files <!--TT-154-->

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->


## v1.14.1

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: May 21, 2018

### New Features and Changes

* None

### Bugs Fixed Since Last Release

* Update to XML submission to correctly import biospecimen relationnships <!--TT-578-->
* GraphQL submission queries return incorrect information <!--SV-1086-->

### Known Issues and Workarounds

* The error message for not including a token is incorrect for controlled access API requests (e.g. bam slicing).  It states that your token is invalid, but it should request that a token be supplied. <!--API-495-->
* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->


## v1.14.0

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: April 23, 2018

### New Features and Changes

* Updated support of BCR XML submission for Next Generation Cancer Models <!--DAT-1207-->


### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->


## v1.13.0

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: February 15, 2018

### New Features and Changes

* Reduced number of submission portal timeouts by optimizing graphql API requests <!--API-317-->

### Bugs Fixed Since Last Release

* Fixed bug preventing submission of clinical data via BCR XML <!--SV-985-->

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->

## v1.12.0

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: December 21, 2017

### New Features and Changes

* None

### Bugs Fixed Since Last Release

* Updated submit project button on submission portal to work more quickly for projects with large numbers of nodes <!--API-344-->

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->



## v1.11.0

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: November 16, 2017

### New Features and Changes

* Added support for histogram aggregation for numeric fields <!--API-355-->

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->



## v1.10.0

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: August 22, 2017

### New Features and Changes

* Longitudinal clinical data can now be submitted to the GDC
* Created new entities and reorganized the location of many properties in the GDC Dictionary.  A list of deprecated properties and their new locations can be found [here](../Release_Notes/DeprecatedFields_072017.txt).  Note that the deprecated properties still contain all information that they previously did.  In a later release we will map the data from the old location to the new.  Eventually, we will suspend submission to the old properties and will remove them entirely from the GDC.
* GDC Data Dictionary Changes Details
    * Created follow_up entity to support longitudinal clinical data <!--TT-65-->
    * Deprecated clinical test entity <!--DOC-67-->
    * Modified acceptable values for Read Group properties <!--TT-9,TT-76-->
    * Modified Diagnosis entity <!--TT-64-->  
    * Modified Treatment entity <!--TT-66-->
    * Modified Demographic entity <!--TT-83-->
    * Modified Case entity <!--TT-84-->
    * Added new tumor code, tumor id, and sample types to Sample entity to support OCG <!--TT-85, TT-68-->
    * Added property `days_to_diagnosis` to Diagnosis entity <!--TT-91-->
    * Created Somatic Mutation Index entity <!--TT-92-->
    * Updated CaDSR CDE links in data dictionary <!--DAT-794-->
    * Added new sample type `tumor` to sample entity <!--TT-77-->
    * Made classification_of_tumor on diagnosis entity non-required <!--DAT-203-->
    * Added support for FM-AD to Genomic Profile Harmonization Workflow entity <!--DAT-985-->
    * Added data type `Gene Level Copy Number Scores` to Copy Number Segment entity <!--TT-94-->



### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->


## v1.9.0

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: June 29, 2017

### New Features and Changes

* Updated ability to search data based on genes and mutations have been released. Discussion of the new endpoints and their use can be found in the Data Portal User Guide. These feature are limited to those projects for which the GDC contains open-access MAF files.
<!--API-112, API-114, API-116, API-118, API-119, APIT-179, API-180, API-181, API-181-->
* API request parameter `from` now returns first record when `from=0`.  Previously, `from=1` was the first record.  `from=0` is now the default.

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->


## v1.8.0

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: May 9, 2017

### New Features and Changes

* Users can now control whether a set of files will be compressed or not when downloading.  For further details see the [API Users Guide](/API/Users_Guide/Downloading_Files/#downloading-an-uncompressed-group-of-files). <!--API-175-->

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->


## v1.7.1

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: March 16, 2017

### New Features and Changes

* Submission: Due to Data Dictionary updates new submission templates may be required for users submitting JSON and TSV formats
* Submission: Entities in submitted state (assigned when the project has been submitted) cannot be deleted.  <!-- API-129 -->
* Submission: When attempting to delete an entity that has child entities not specified in the request, an error message is generated that will include all of the child entities' UUIDs.  <!-- API-130 -->
* Submission: Entities associated with files uploaded to the GDC object store cannot be deleted until the associated file has been deleted. <!-- API-132 -->
* Re-enable Review, Submit, and Release functions for submission <!--API-73, API-138, API-159-->
* GDC Data Dictionary Changes
  * Added "submittable" property to all entities <!--DAT-215-->
  * Changed Read Group to category biospecimen <!--DAT-216-->
  * Added many new clinical properties available for submission <!--DAT-210, DAT-31, DAT-226, DAT-205-->
  * Added sample codes from Office of Cancer Genomics (OCG) to analyte and aliquot <!--DAT-170-->
  * Slides can now be attached to sample rather than just portion <!--DAT-205-->
  * `sample_type_id` is no longer required when submitting sample entities <!--DAT-233-->
  * `analyte_type_id` is no longer required when submitting aliquot and analyte entities<!--DAT-255-->
  * Clinical Test Entity is created for storing results of a variety of potential clinical tests related to the diagnosis - <!--DAT-223-->
  * Genomic Profiling Report entity created for storing particular derived sequencing results <!--DAT-229-->
  * Structural Variation entity created <!--DAT-229-->
  * Project entity includes new field "Intended Release Date" <!--API-143-->
  * Project entity includes new field "Releasable" <!--API-157-->


### Bugs Fixed Since Last Release

  * Fixed bug where boolean properties were not accepted with TSV submission <!--API-168-->

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->


## v1.5.0

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: October 31, 2016

### New Features and Changes

* API responds with an error when the request specifies an unsupported combination of `filters` and `facets`. <!-- API-34 -->
* In TSV submissions, trailing and leading whitespace, including non-ASCII whitespace characters, are stripped from property names and values. <!-- API-63, API-29-->
* For released projects, any updates to previously submitted entities (i.e. `"state": "submitted"`) will be included in the following GDC data release. <!-- API-81 -->
* Performance improvements for manifest generation. <!-- API-58 -->

### Bugs Fixed Since Last Release

* Uploading certain unsupported metadata files caused the associated submission transactions to remain stuck in pending state. <!-- API-78 -->

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->




## v1.4.0

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: September 23, 2016

### New Features and Changes

* Submission transaction log includes additional information to assist in tracking. <!-- API-40 -->
* Submission project state transitions are disabled temporarily while project release features are being improved. <!-- API-56 -->
* GDC data dictionary changes:
    * The `submittable` property was added to all entity types in the GDC data model. It indicates whether the entity type can be submitted by users. <!-- DM-16 -->
    * Category of Read Group entities in the GDC Data Model has changed from `data_bundle` to `biospecimen`.
    * Analyte entities support an expanded set of `analyte_type` values.

### Bugs Fixed Since Last Release

* None to report

### Known Issues and Workarounds

* API search & retrieval queries that do not include a `sort` parameter may return results in different order each time they are executed. This is a particular problem for paginated responses (i.e. responses to queries for which the number of results is greater than the `size` parameter). <!-- FEAT-120 -->
    * **Workaround:** Include a `sort` parameter in API search & retrieval queries.
* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->



## v1.3.1

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: September 7, 2016

### New Features and Changes

* Successful `_dry_run` submission transactions can be committed to the GDC data model without having to re-upload metadata. The transactions can also be closed to prevent them from being committed in the future.
* Submission transactions can be submitted in asynchronous mode. In this mode, the GDC API will issue an immediate acknowledgement of the transaction, along with the `transaction_id`. The status of the transaction can be verified by the user at a later time by specifying the `transaction_id`. Users submitting large transactions may find this mode helpful.
* New submission transaction properties can be queried with GraphQL
* GDC Data Dictionary changes:
    * Clinical Supplement entities can have `data_format` set to OMF.
    * Biospecimen Supplement entities can have `data_format` set to SSF or PPS.
    * Read group `instrument_model` can be set to "Illumina HiSeq 4000".
    * Category of Slide entities in the GDC Data Model has changed from `data_bundle` to `biospecimen`.

### Bugs Fixed Since Last Release

* Incorrect BMI calculation in the import of BCR XML files.

### Known Issues and Workarounds

* API search & retrieval queries that do not include a `sort` parameter may return results in different order each time they are executed. This is a particular problem for paginated responses (i.e. responses to queries for which the number of results is greater than the `size` parameter). <!-- FEAT-120 -->
    * **Workaround:** Include a `sort` parameter in API search & retrieval queries.
* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->




## v1.2.0

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: August 9, 2016

### New Features and Changes

* Tarballs generated by the `data` endpoint in response to multi-file data download requests now include a folder structure that puts each file in a folder whose name is the file's UUID. <!-- PGDC-2581 -->
* UUIDs in clinical XML files are no longer treated in a case-sensitive way by the `submission` endpoint. <!-- API-14 -->
* Improved performance of `submission` endpoint for transactions that include many cases. <!-- https://github.com/NCI-GDC/gdcapi/pull/495 -->
* Speed improvements for the `submission` endpoint.
* BCR XML is no longer validated against its XSD at submission.

### Bugs Fixed Since Last Release

* Fixed handling of `POST` requests to address problems with cart functionality in older versions of Firefox <!-- PGDC-2555 -->
* Files of category `related_files` can now be downloaded from the `data` endpoint. <!-- PGDC-2542 -->
* Allowed submission by projects in certain dbGaP registration states that were previously blocked. <!--API-3 -->

### Known Issues and Workarounds

* API search & retrieval queries that do not include a `sort` parameter may return results in different order each time they are executed. This is a particular problem for paginated responses (i.e. responses to queries for which the number of results is greater than the `size` parameter). <!-- FEAT-120 -->
    * **Workaround:** Include a `sort` parameter in API search & retrieval queries.
* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING". <!-- PGDC-2530 // https://github.com/NCI-GDC/gdcapi/pull/524  -->
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests. <!-- PGDC-2411 -->




## v1.1.0

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: May 25, 2016

### New Features and Changes

* BAM index files (.bai) are now automatically downloaded with parent BAM.

### Bugs Fixed Since Last Release

* None to report

### Bugs Fixed Since Last Release

* Sorting by file `submitter_id` no longer causes an internal server error
* BAM index files are now included with harmonized BAM files
* Certain very long API requests will time out.  It is recommended to break up into a series of smaller requests.

### Known Issues and Workarounds

* Fields are not counted as missing if parent field is also missing.  This may occur with queries of nested fields in the Data Portal Advanced Search or an API query using a filter.  This behavior could impact results reported using search parameters of "IS MISSING" or "NOT MISSING".
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests.




## v1.0.1

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: May 16, 2016

### New Features and Changes

* HTTP interface that uses JSON as the primary data exchange format
* Programmatic access to functionality provided by GDC Data and Submission portals, via `projects`, `cases`, `files`, `annotations`, `data`, `slicing`, `status`, and `submission` endpoints
* Programmatic access to GDC Legacy Archive via `legacy` endpoint
* Token-based authentication for secure access to controlled data and to submission functionality
* RESTful search that supports simple and complex queries via `filters`, `fields`, and `facets` parameters, and `project`, `files`, `cases`, and `annotations` endpoints.
* Search results can be sorted using `sort` parameter, paginated using `size` and `from` parameters, and output in JSON, TSV, and XML using `format` and `pretty` parameters.
* `_mapping` endpoint enables user discovery of fields available for data search and retrieval operations
* Support for downloading of individual files and of archives containing multiple files
* Generation of download and upload manifests for use with the GDC Data Transfer Tool
* BAM slicing functionality for downloading part(s) of a BAM file specified using chromosomal coordinates or HGNC gene names
* Transactional submission system that links individual data elements according to a graph-based GDC Data Model
* Two data entity identifiers: UUIDs, which are consistent across GDC, and Submitter IDs, for compatibility with submitters' tracking systems

### Bugs Fixed Since Last Release

* None to report


### Known Issues and Workarounds

* Use of non-ascii characters in token passed to Data Transfer Tool will produce incorrect error message "Internal server error: Auth service temporarily unavailable".
* Use of a decimal in an integer search field produces unexpected error.
* Certain very large API requests will time out.  It is recommended to break up very large requests into a series of smaller requests.


# API Release Notes Archive


## v0.3.24.3

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: May 2, 2016

### New Features and Changes

* BAM slicing: regions can be specified using a GENCODE v22 (HGNC) gene names
* BAM slicing: disabled for files in GDC legacy archive
* New YAML format of upload manifests
* Annotations no longer include `creator` information
* `legacy` endpoint added; provides access to data in legacy archive
* Certain API responses now include MD5 checksums, using `Content-MD5` HTTP header

### Bugs Fixed Since Last Release

* Authentication system fixed to allow download of harmonized files
* BCR clinical XML `follow_up` element is not required
* Index files can be downloaded
* Downloaded data files are named correctly


### Known Issues and Workarounds

* None to report

Release details are maintained in the [GDC API change log](https://github.com/NCI-GDC/gdcapi/blob/develop/CHANGELOG.md)






## v0.3.24-spr5

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: April 19, 2016

### New Features and Changes

* None to report

### Bugs Fixed Since Last Release

* API responds with an error when multiple `submitter_id` properties are submitted for the same entity in a transaction.
* API download functionality now provides indication when when no related files (e.g. SRA XML) exist.

### Known Issues and Workarounds

* None to report

Release details are maintained in the [GDC API change log](https://github.com/NCI-GDC/gdcapi/blob/develop/CHANGELOG.md)






## v0.3.24.2

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: April 5, 2016

### New Features and Changes

* Updates to GDC data model
* File metadata can be submitted before file upload
* GQL improvements
* `/download/metadata` endpoint allows download of MAGE-TAB and SRA XML
* `/submission/<Program.name>/<Project.code>/entities/<uuid>` endpoint allows retrieval of Submission Portal entities by UUID

### Bugs Fixed Since Last Release

* Submission: files in `validated`, `submitted`, `processing`, and `processed` states can no longer be updated. To change these files the user must delete or redact them.
* `_mapping` endpoint now available for `annotations` endpoint
* Order of files submitted in same transaction does not matter
* Fixed incorrect response for bulk transactions

### Known Issues and Workarounds

* None to report

Release details are maintained in the [GDC API change log](https://github.com/NCI-GDC/gdcapi/blob/develop/CHANGELOG.md)






## v0.3.24.1

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: March 1, 2016

### New Features and Changes

* New entities: *Slide Data Bundle* and *Pathology Data Bundle*
* New `_dictionary/_all` endpoint displays the entire GDC dictionary for the `submission` endpoint, and project-specific dictionaries for project endpoints; e.g. `https://api.gdc.cancer.gov/v0/submission/_dictionary/_all` and `https://api.gdc.cancer.gov/v0/submission/TCGA/SKCM/_dictionary/_all`.
* Support for releases
* Project `state` property definition has been updated. The property can now have one of six values: `open` (default), `review`, `submitted`, `processing`, `closed`, and `legacy`.
* `days_to_index` property has been removed from `Case`
* `template` endpoint for submissions returns archive of all templates, e.g. `https://api.gdc.cancer.gov/v0/submission/TCGA/SKCM/template`
* `template` endpoint can generate JSON using `/template?format=json`
* `export` endpoint now supports POST requests
* GraphQL: lists are accepted as top level property filters
* GraphQL: `not` filter was added

### Bugs Fixed Since Last Release

* Portion is now linked to Center

### Known Issues and Workarounds

* None to report

Release details are maintained in the [GDC API change log](https://github.com/NCI-GDC/gdcapi/blob/develop/CHANGELOG.md)






## v0.3.20.1

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: January 28, 2016

### New Features and Changes

* Metadata export support for TSV, CSV and JSON formats, including:
  *  Clinical metadata
  *  Biospecimen metadata
  *  Project and case entities
  *  Annotations
* Export of metadata can be filtered by category
* Support for generating a TSV template
* Additional details are provided in JSON parsing error messages

### Bugs Fixed Since Last Release

* Biospecimen XML submissions with shipment_portion nodes now parsed correctly
* Correct HTTP code returned upon authentication failure
* API URLs work with and without trailing slash
* Miscellaneous bug fixes

### Known Issues and Workarounds

*   None to report

Release details are maintained in the [GDC API change log](https://github.com/NCI-GDC/gdcapi/blob/develop/CHANGELOG.md)






## v0.2.18

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: November 18, 2015

### New Features and Changes

*   Added support for molecular data upload.
  *   Users can submit data bundle entities to specify the molecular data and relations to be uploaded.
  *   Users can update existing data by submitting update data bundles.
  *   Manifest files can be downloaded to allow for molecular data upload via the API or Data Transfer Tool.
*   Users can delete previously submitted entities.

### Bugs Fixed Since Last Release

*   None to report

### Known Issues and Workarounds

*   None to report

Release details are maintained in the [GDC API change log](https://github.com/NCI-GDC/gdcapi/blob/develop/CHANGELOG.md)






## v0.2.18-spr1

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: November 2, 2015

### New Features and Changes

*   Entity Metadata submission added with support for JSON, CSV, TSV, and XML formats and the following data entities:
    *   Clinical metadata
    *   Biospecimen metadata
    *   Project and case entities
    *   Annotations
*   Dry run option available for metadata submission to validate metadata files prior to submission
*   Submission integrated with authorization to ensure read, create, update, and admin roles can be controlled at the project level

### Bugs Fixed Since Last Release

*   None to report

### Known Issues and Workarounds

*   None to report

Release details are maintained in the [GDC API change log](https://github.com/NCI-GDC/gdcapi/blob/develop/CHANGELOG.md)






## v0.2.15-oicr1

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: August 7, 2015


### New Features and Changes

*   None to report

### Bugs Fixed Since Last Release

*   Addressed an issue in the range aggregation feature (used by facets)

### Known Issues and Workarounds

*   Query and Download Service
    *   Universal Resource Locator (URL) length is limited.  As such, download requests with many files have to be divided into multiple requests. A feature that allows a payload to specify the file identifiers to download to overcome this limitation is planned for a future release.
    *   All TARGET, CGCI, and CCLE data is currently not available via the GDC API but will be made available as data is continously imported into the GDC
*   Authentication and Authorization Service
    *   Obtaining a token requires browser access. This is a limitation of eRA Commons and not the GDC authorization service. The Security Assertion Markup Language  (SAML) based protocol eRA Commons uses does not support [ECP](https://wiki.shibboleth.net/confluence/display/CONCEPT/ECP) (or similar)

Release details are maintained in the [GDC API change log](https://github.com/NCI-GDC/gdcapi/blob/develop/CHANGELOG.md)






## v0.2.13

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: July 23, 2015


### New Features and Changes

*   Query and Download Service
    *   Implemented server-side related files
    *   Created a quick search endpoint
*   Misc
    *   Added POST support
    *   Added range support (used by range facet)
    *   Hooked-up reports to new reports index

### Bugs Fixed Since Last Release

*   Query and Download Service
    *   Fixed error (500) on annotations without notes
    *   Correct structure for related files
    *   Fixed error (404) with annotation download
    *   Fixed issue with support for IS NOT operator

### Known Issues and Workarounds

*   Query and Download Service
    *   Universal Resource Locator (URL) length is limited.  As such, download requests with many files have to be ivided into multiple requests. A feature that allows a payload to specify the file identifiers to download to zovercome this limitation is planned for a future release.
    *   All TARGET, CGCI, and CCLE data is currently not available via the GDC API but will be made available as data is continously imported into the GDC
*   Authentication and Authorization Service
    *   Obtaining a token requires browser access. This is a limitation of eRA Commons and not the GDC authorization service. The Security Assertion Markup Language  (SAML) based protocol eRA Commons uses does not support [ECP](https://wiki.shibboleth.net/confluence/display/CONCEPT/ECP) (or similar)

Release details are maintained in the [GDC API change log](https://github.com/NCI-GDC/gdcapi/blob/develop/CHANGELOG.md)






## v0.1.10

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: March 18, 2015


### New Features and Changes

*   Query and Download Service
    *   Supports query for files, participants, annotations, and projects by metadata. Interfaces with ElasticSearch for efficient queries.
    *   Provides a RESTful interface to support the download of data, metadata, and annotations
    *   Supports the download of multiple files, manifests, or annotations in compressed (gzip) format by default
*   Authentication and Authorization Service
    *   Authenticates users using the GDC authorization service which authorizes against dbGaP
    *   Manages tokens and token-based authorization using OpenStack Keystone
    *   Implements federated authentication with eRA Commons to generate a Keystone token
    *   Downloads Comma Separated Values (CSVs) from dbGaP at regular intervals and verifies authorization using eRA Commons credentials

### Bugs Fixed Since Last Release

*   Query and Download Service
    *   Initial Release - Not Applicable
*   Authentication and Authorization Service
    *   Further developed the prototype into a production ready service

### Known Issues and Workarounds

*   Query and Download Service
    *   Universal Resource Locator (URL) length is limited.  As such, download requests with many files have to be divided into multiple requests. A feature that allows a payload to specify the file identifiers to download to overcome this limitation is planned for a future release.
    *   All TARGET, CGCI, and CCLE data is currently not available via the GDC API but will be made available as data is continously imported into the GDC
*   Authentication and Authorization Service
    *   Obtaining a token requires browser access. This is a limitation of eRA Commons and not the GDC authorization service. The Security Assertion Markup Language  (SAML) based protocol eRA Commons uses does not support [ECP](https://wiki.shibboleth.net/confluence/display/CONCEPT/ECP) (or similar)

Release details are maintained in the [GDC API change log](https://github.com/NCI-GDC/gdcapi/blob/develop/CHANGELOG.md)






## v0.1.8

* __GDC Product__: Application Programming Interface (API)
* __Release Date__: January 22, 2015


### New Features and Changes

*   Identity Service ([Signpost](https://github.com/NCI-GDC/signpost))
    *   Provides an internal identity service used for assigning Digital IDs to all GDC data
    *   Tracks the location of data in GDC object stores
*   Authentication and Authorization Service
    *   Provides a prototype service for authenticating using eRA Commons as the identity provider and dbGaP for authorization

### Bugs Fixed Since Last Release

*   None to report

### Known Issues and Workarounds

*   Identify Service ([Signpost](https://github.com/NCI-GDC/signpost))
    *   The internal identity service does not support authentication for create, read, update, and delete operations as this is an internal service
*   Authentication and Authorization Service
*   Obtaining a token requires browser access. This is a limitation of eRA Commons, not the GDC authorization service. The SAML based protocol eRA Commons uses does not support [ECP](https://wiki.shibboleth.net/confluence/display/CONCEPT/ECP) (or similar)

Release details are maintained in the [GDC API change log](https://github.com/NCI-GDC/gdcapi/blob/develop/CHANGELOG.md).


# Introduction

The Genomic Data Commons receives, processes, and distributes genomic, clinical, and biospecimen data from cancer research programs. General information about data in the GDC can be found [on the GDC website](https://gdc.cancer.gov/about-data).

This document provides details about data included in the Genomic Data Commons, including information about the GDC data model, data formats, data processing, data security, and data releases.


# GDC VCF Format

## Introduction

The GDC DNA-Seq somatic variant-calling pipeline compares a set of matched tumor/normal alignments and produces a VCF file. VCF files report the somatic variants that were detected by each of the four variant callers. Four raw VCFs (Data Type: Raw Simple Somatic Mutation) are produced for each tumor/normal pair of BAMs. Four additional annotated VCFs (Data Type: Annotated Somatic Mutation) are produced by adding biologically relevant information about each variant.  

The GDC VCF file format follows standards of the [Variant Call Format (VCF) Version 4.1 Specification](https://samtools.github.io/hts-specs/VCFv4.1.pdf). Raw Simple Somatic Mutation VCF files are unannotated, whereas Annotated Somatic Mutation VCF files include extensive, consistent, and pipeline-agnostic annotation of somatic variants.

## VCF file structure

### Metadata header

A VCF file starts with lines of metadata that begin with `##`. Some key components of this section include:

*  __gdcWorkflow:__ Information on the pipelines that were used by the GDC to generate the VCF file. Annotated VCF files contain two *gdcWorkflow* lines, one that reports the variant calling process and one that reports the variant annotation process.
*  __INDIVIDUAL:__ information about the study participant (`case`), including:
 	* *NAME:* Submitter ID (barcode) associated with the participant
	* *ID:* GDC case UUID
*  __SAMPLE:__ sample information, including:
	*  *ID:* NORMAL or TUMOR
	*  *NAME:* Submitter ID (barcode) of the aliquot
	*  *ALIQUOT_ID:* GDC aliquot UUID
	*  *BAM_ID:* The UUID for the BAM file used to produce the VCF  
*   __INFO:__ Format of *additional information* fields
	*  NOTE: GDC Annotated VCFs may contain multiple INFO lines. The last INFO line contains information about annotation fields generated by the [Somatic Annotation Workflow](/Data_Dictionary/viewer/#?view=table-definition-view&id=somatic_annotation_workflow) (see [GDC INFO Fields](#gdc-info-fields) below).
*   __FILTER:__ Description of filters that have been applied to the variants
*   __FORMAT:__ Description of genotype fields
*   __reference:__ The reference genome used to generate the VCF file (GRCh38.d1.vd1.fa)
*   __contig:__ A list of IDs for the contiguous DNA sequences that appear in the reference genome used to produce VCF files
	*  NOTE: Annotated VCFs include contig information for autosomes, sex chromosomes, and mitochondrial DNA. Unplaced, unlocalized, human decoy, and viral genome sequences are not included.
*   __VEP:__ the VEP command used by the [Somatic Annotation Workflow](/Data_Dictionary/viewer/#?view=table-definition-view&id=somatic_annotation_workflow) to generate the annotated VCF file.

### Column Header Line

Each variant is represented by a row in the VCF file. Below each of the columns are described:

0. __CHROM:__ The chromosome on which the variant is located
0. __POS:__ The position of the variant on the chromosome.  Refers to the first position if the variant includes more than one base
0. __ID:__ A unique identifier for the variant; usually a dbSNP rs number if applicable
0. __REF:__ The base(s) exhibited by the reference genome at the variant's position
0. __ALT:__ The alternate allele(s), comma-separated if there are more than one
0. __QUAL:__ Not populated
0. __FILTER:__ The names of the filters that have flagged this variant. The types of filters used will depend on the variant caller used.
0. __INFO:__ Additional information about the variant. This includes the annotation applied by the VEP.
0. __FORMAT:__ The format of the sample genotype data in the next two columns. This includes descriptions of the colon-separated values.
0. __NORMAL:__ Colon-separated values that describe the normal sample
0. __TUMOR:__ Colon-separated values that describe the tumor sample

See [Variant Call Format (VCF) Version 4.1 Specification](https://samtools.github.io/hts-specs/VCFv4.1.pdf) for details.

## GDC INFO fields

The following variant annotation fields are currently included in Annotated Somatic Mutation VCF files.  Please refer to the [DNA-Seq Analysis Pipeline](../Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline.md) documentation for details on how this information is generated. [VEP Documentation](http://useast.ensembl.org/info/docs/tools/vep/vep_formats.html#output) provides additional information about some of these fields.


| Field | Description |
|--|--|
| Allele | The variant allele used to calculate the consequence |
| Consequence | Consequence type of this variant  |
| IMPACT | The impact modifier for the consequence type  |
| SYMBOL | The HUGO gene symbol |
| Gene | Ensembl stable ID of the affected gene |
| Feature_type | Type of feature. Currently one of Transcript, RegulatoryFeature, MotifFeature. |
| Feature | Ensembl stable ID of the feature |
| BIOTYPE | The type of transcript or regulatory feature (e.g. protein_coding) |
| EXON | Exon number (out of total exons) |
| INTRON | Intron number (out of total introns) |
| HGVSc | The HGVS coding sequence name |
| HGVSp | The HGVS protein sequence name |
| cDNA_position | Relative position of base pair in cDNA sequence |
| CDS_position | Relative position of base pair in coding sequence |
| Protein_position | Relative position of the affected amino acid in protein |
| Amino_acids | Change in amino acids (only given if the variant affects the protein-coding sequence) |
| Codon | The affected codons with the variant base in upper case |
| Existing_variation | Known identifier of existing variant; usually a dbSNP rs number if applicable |
| ALLELE_NUM | Allele number from input; 0 is reference, 1 is first alternate, etc. |
| DISTANCE | Shortest distance from variant to transcript |
| STRAND | The DNA strand (1 or -1) on which the transcript/feature lies |
| FLAGS | Transcript quality flags |
| VARIANT_CLASS | Sequence Ontology variant class |
| SYMBOL_SOURCE | The source of the gene symbol |
| HGNC_ID | HGNC gene ID |
| CANONICAL | A flag indicating if the transcript is denoted as the canonical transcript for this gene |
| TSL | Transcript support level |
| APPRIS | APPRIS isoform annotation |
| CCDS | The CCDS identifer for this transcript, where applicable |
| ENSP | The Ensembl protein identifier of the affected transcript |
| SWISSPROT | UniProtKB/Swiss-Prot identifier of protein product |
| TREMBL | UniProtKB/TrEMBL identifier of protein product |
| UNIPARC | UniParc identifier of protein product |
| RefSeq | RefSeq gene ID |
| GENE_PHENO | Indicates if the gene is associated with a phenotype, disease or trait |
| SIFT | The SIFT prediction and/or score, with both given as prediction (score) |
| PolyPhen | The PolyPhen prediction and/or score |
| DOMAINS | The source and identifier of any overlapping protein domains |
| HGVS_OFFSET | Indicates by how many bases the HGVS notations for this variant have been shifted |
| GMAF | Non-reference allele and frequency of existing variant in 1000 Genomes |
| AFR_MAF | Non-reference allele and frequency of existing variant in 1000 Genomes combined African population |
| AMR_MAF | Non-reference allele and frequency of existing variant in 1000 Genomes combined American population |
| EAS_MAF | Non-reference allele and frequency of existing variant in 1000 Genomes combined East Asian population |
| EUR_MAF | Non-reference allele and frequency of existing variant in 1000 Genomes combined European population |
| SAS_MAF | Non-reference allele and frequency of existing variant in 1000 Genomes combined South Asian population |
| AA_MAF | Non-reference allele and frequency of existing variant in NHLBI-ESP African American population |
| EA_MAF | Non-reference allele and frequency of existing variant in NHLBI-ESP European American population |
| ExAC_MAF | Frequency of existing variant in ExAC combined population |
| ExAC_Adj_MAF | Adjusted frequency of existing variant in ExAC combined population |
| ExAC_AFR_MAF | Frequency of existing variant in ExAC African/American population |
| ExAC_AMR_MAF | Frequency of existing variant in ExAC American population |
| ExAC_EAS_MAF | Frequency of existing variant in ExAC East Asian population |
| ExAC_FIN_MAF | Frequency of existing variant in ExAC Finnish population |
| ExAC_NFE_MAF | Frequency of existing variant in ExAC Non-Finnish European population |
| ExAC_OTH_MAF | Frequency of existing variant in ExAC combined other combined populations |
| ExAC_SAS_MAF | Frequency of existing variant in ExAC South Asian population |
| CLIN_SIG | Clinical significance of variant from dbSNP |
| SOMATIC | Somatic status of existing variant(s) |
| PHENO | Indicates if existing variant is associated with a phenotype, disease or trait |
| PUBMED | Pubmed ID(s) of publications that cite existing variant |
| MOTIF_NAME | The source and identifier of a transcription factor binding profile aligned at this position |
| MOTIF_POS | The relative position of the variation in the aligned TFBP |
| HIGH_INF_POS | A flag indicating if the variant falls in a high information position of a transcription factor binding profile (TFBP) |
| MOTIF_SCORE_CHANGE | The difference in motif score of the reference and variant sequences for the TFBP |
| ENTREZ | Entrez ID |
| EVIDENCE | Evidence that the variant exists |


# GDC MAF Format v.1.0.0

## Introduction

Mutation Annotation Format (MAF) is a tab-delimited text file with aggregated mutation information from [VCF Files](VCF_Format.md) and are generated on a project-level. MAF files are produced through the [Somatic Aggregation Workflow](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=somatic_aggregation_workflow&_top=1). The GDC produces MAF files at two permission levels: __protected__ and __somatic__ (or open-access). One MAF file is produced per variant calling pipeline per GDC project. MAFs are produced by aggregating the GDC annotated VCF files generated from one pipeline for one project.

Annotated VCF files often have variants reported on multiple transcripts whereas the MAF files generated from the VCFs (\*protected.maf) only report the most critically affected one. Somatic MAFs (\*somatic.maf), which are also known as [Masked Somatic Mutation](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=masked_somatic_mutation) files, are further processed to remove lower quality and potential germline variants. For tumor samples that contain variants from multiple combinations of tumor-normal aliquot pairs, only one pair is selected in the Somatic MAF based on their sample type. Somatic MAFs are publicly available and can be freely distributed within the boundaries of the [GDC Data Access Policies](https://gdc.cancer.gov/access-data/data-access-policies).

The GDC MAF file format is based on the TCGA Mutation Annotation Format specifications, with additional columns included.

__Note:__ The criteria for allowing mutations into open-access are purposefully implemented to overcompensate and filter out germline variants. If omission of true-positive somatic mutations is a concern, the GDC recommends using protected MAFs.  

## Somatic MAF File Generation

The process for modifying a protected MAF into a somatic MAF is as follows:

*  Aliquot Selection: only one tumor-normal pair are selected for each tumor sample based on the plate number, sample type, analyte type and other features extracted from tumor TCGA aliquot barcode.
*  Low quality variant filtering and germline masking:
    1. Variants with ```Mutation_Status != 'Somatic'``` or ```GDC_FILTER = 'Gapfiller'```, ```'ContEst'```, ```'multiallelic'```, ```'nonselectedaliquot'```, ```'BCR_Duplicate'``` or ```'BadSeq'``` are __removed__.
    2. Remaining variants with ```GDC_Valid_Somatic = True``` are __included__ in the Somatic MAF.
    3. Remaining variants with ```FILTER != 'panel_of_normals'``` or ```PASS``` are __removed__. Note that the `FILTER != panel_of_normals` value is only relevant for the variants generated from the MuTect2 pipeline.
    4. Remaining variants with ```MC3_Overlap = True``` are __included__ in the Somatic MAF.
    5. Remaining variants with ```GDC_FILTER = 'ndp'```, ```'NonExonic'```, ```'bitgt'```, ```'gdc_pon'``` are __removed__.
    6. Remaining variants with ```SOMATIC != null``` are __included__ in the Somatic MAF.
    7. Remaining variants with ```dbSNP_RS = 'novel'``` or ```null``` are __included__ in the Somatic MAF.
    8. Remaining variants are __removed__.
* Removal of the following columns:
    * vcf_region
    * vcf_info
    * vcf_format
    * vcf_tumor_gt
    * vcf_normal_gt
    * GDC_Valid_Somatic
* Set values to be blank in the following columns that may contain information about germline genotypes:
    * Match_Norm_Seq_Allele1
    * Match_Norm_Seq_Allele2
    * Match_Norm_Validation_Allele1
    * Match_Norm_Validation_Allele2
    * n_ref_count
    * n_alt_count

---

![Somatic MAF Generation](images/ProtectedMAF4.png)

---

## Protected MAF File Structure

The table below describes the columns in a protected MAF and their definitions. Note that the somatic (open-access) MAF structure is the same except for having the last six columns removed.

| Column | Description |
|---|---|
| 1 - Hugo_Symbol | [HUGO](http://www.genenames.org/) symbol for the gene (HUGO symbols are always in all caps). "Unknown" is used for regions that do not correspond to a gene  |
| 2 - Entrez_Gene_Id | [Entrez gene](https://www.ncbi.nlm.nih.gov/gene) ID (an integer).  "0" is used for regions that do not correspond to a gene region or Ensembl ID |
| 3 - Center| One or more genome sequencing center reporting the variant |
| 4 - NCBI_Build| The reference genome used for the alignment (GRCh38) |
| 5 - Chromosome | The affected chromosome (chr1) |
| 6 - Start_Position | Lowest numeric position of the reported variant on the genomic reference sequence. Mutation start coordinate |
| 7 - End_Position | Highest numeric genomic position of the reported variant on the genomic reference sequence. Mutation end coordinate |
| 8 - Strand | Genomic strand of the reported allele. Currently, all variants will report the positive strand: '+' |
| 9 - Variant_Classification | Translational effect of variant allele  |
| 10 - Variant_Type | Type of mutation. TNP (tri-nucleotide polymorphism) is analogous to DNP (di-nucleotide polymorphism) but for three consecutive nucleotides. ONP (oligo-nucleotide polymorphism) is analogous to TNP but for consecutive runs of four or more (SNP, DNP, TNP, ONP, INS, DEL, or Consolidated)|
| 11 - Reference_Allele | The plus strand reference allele at this position. Includes the deleted sequence for a deletion or "-" for an insertion |
| 12 - Tumor_Seq_Allele1 | Primary data genotype for tumor sequencing (discovery) allele 1. A "-" symbol for a deletion represents a variant. A "-" symbol for an insertion represents wild-type allele. Novel inserted sequence for insertion does not include flanking reference bases  |
| 13 - Tumor_Seq_Allele2 | Tumor sequencing (discovery) allele 2  |
| 14 - dbSNP_RS| The rs-IDs from the [dbSNP](https://www.ncbi.nlm.nih.gov/projects/SNP/) database, "novel" if not found in any database used, or null if there is no dbSNP record, but it is found in other databases|
| 15 - dbSNP_Val_Status | The dbSNP validation status is reported as a semicolon-separated list of statuses. The union of all rs-IDs is taken when there are multiple |
| 16 - Tumor_Sample_Barcode| Aliquot barcode for the tumor sample |
| 17 - Matched_Norm_Sample_Barcode| Aliquot barcode for the matched normal sample |
| 18 - Match_Norm_Seq_Allele1 | Primary data genotype. Matched normal sequencing allele 1. A "-" symbol for a deletion represents a variant. A "-" symbol for an insertion represents wild-type allele. Novel inserted sequence for insertion does not include flanking reference bases (cleared in somatic MAF) |
| 19 - Match_Norm_Seq_Allele2| Matched normal sequencing allele 2 |
| 20 - Tumor_Validation_Allele1| Secondary data from orthogonal technology. Tumor genotyping (validation) for allele 1. A "-" symbol for a deletion represents a variant. A "-" symbol for an insertion represents wild-type allele. Novel inserted sequence for insertion does not include flanking reference bases  |
| 21 - Tumor_Validation_Allele2 | Secondary data from orthogonal technology. Tumor genotyping (validation) for allele 2 |
| 22 - Match_Norm_Validation_Allele1 | Secondary data from orthogonal technology. Matched normal genotyping (validation) for allele 1. A "-" symbol for a deletion represents a variant. A "-" symbol for an insertion represents wild-type allele. Novel inserted sequence for insertion does not include flanking reference bases (cleared in somatic MAF) |
| 23 - Match_Norm_Validation_Allele2 | Secondary data from orthogonal technology. Matched normal genotyping (validation) for allele 2 (cleared in somatic MAF) |
| 24 - Verification_Status| Second pass results from independent attempt using same methods as primary data source. Generally reserved for 3730 Sanger Sequencing |
| 25 - Validation_Status | Second pass results from orthogonal technology |
| 26 - Mutation_Status | An assessment of the mutation as somatic, germline, LOH, post transcriptional modification, unknown, or none. The values allowed in this field are constrained by the value in the Validation_Status field |
| 27 - Sequencing_Phase | TCGA sequencing phase (if applicable). Phase should change under any circumstance that the targets under consideration change |
| 28 - Sequence_Source | Molecular assay type used to produce the analytes used for sequencing. Allowed values are a subset of the SRA 1.5 library_strategy field values. This subset matches those used at CGHub |
| 29 - Validation_Method | The assay platforms used for the validation call |
| 30 - Score | Not in use |
| 31 - BAM_File | Not in use |
| 32 - Sequencer | Instrument used to produce primary sequence data |
| 33 - Tumor_Sample_UUID | GDC aliquot UUID for tumor sample |
| 34 - Matched_Norm_Sample_UUID | GDC aliquot UUID for matched normal sample |
| 35 - HGVSc | The coding sequence of the variant in HGVS recommended format |
| 36 - HGVSp | The protein sequence of the variant in HGVS recommended format. "p.=" signifies no change in the protein |
| 37 - HGVSp_Short | Same as the HGVSp column, but using 1-letter amino-acid codes |
| 38 - Transcript_ID| [Ensembl](http://useast.ensembl.org/index.html) ID of the transcript affected by the variant |
| 39 - Exon_Number | The exon number (out of total number) |
| 40 - t_depth | Read depth across this locus in tumor BAM |
| 41 - t_ref_count | Read depth supporting the reference allele in tumor BAM |
| 42 - t_alt_count | Read depth supporting the variant allele in tumor BAM |
| 43 - n_depth | Read depth across this locus in normal BAM |
| 44 - n_ref_count | Read depth supporting the reference allele in normal BAM (cleared in somatic MAF) |
| 45 - n_alt_count | Read depth supporting the variant allele in normal BAM (cleared in somatic MAF) |
| 46 - all_effects | A semicolon delimited list of all possible variant effects, sorted by priority ([Symbol,Consequence,HGVSp_Short,Transcript_ID,RefSeq,HGVSc,Impact,Canonical,Sift,PolyPhen,Strand]) |
|47 - Allele| The variant allele used to calculate the consequence |
|48 - Gene|Stable Ensembl ID of affected gene |
|49 - Feature|Stable Ensembl ID of feature (transcript, regulatory, motif) |
|50 - Feature_type| Type of feature. Currently one of Transcript, RegulatoryFeature, MotifFeature (or blank) |
|51 - One_Consequence | The single consequence of the canonical transcript in [sequence ontology](http://www.sequenceontology.org/) terms |
|52 - Consequence | Consequence type of this variant; [sequence ontology](http://www.sequenceontology.org/) terms |
|53 - cDNA_position | Relative position of base pair in the cDNA sequence as a fraction. A "-" symbol is displayed as the numerator if the variant does not appear in cDNA |
|54 - CDS_position | Relative position of base pair in coding sequence. A "-" symbol is displayed as the numerator if the variant does not appear in coding sequence |
|55 - Protein_position | Relative position of affected amino acid in protein. A "-" symbol is displayed as the numerator if the variant does not appear in coding sequence |
|56 - Amino_acids | Only given if the variation affects the protein-coding sequence |
|57 - Codons | The alternative codons with the variant base in upper case |
|58 - Existing_variation | Known identifier of existing variation |
|59 - ALLELE_NUM | Allele number from input; 0 is reference, 1 is first alternate etc. |
|60 - DISTANCE | Shortest distance from the variant to transcript |
|61 - TRANSCRIPT_STRAND | The DNA strand (1 or -1) on which the transcript/feature lies |
|62 - SYMBOL | The gene symbol |
|63 - SYMBOL_SOURCE | The source of the gene symbol |
|64 - HGNC_ID | Gene identifier from the HUGO Gene Nomenclature Committee if applicable |
|65 - BIOTYPE | Biotype of transcript |
|66 - CANONICAL | A flag (YES) indicating that the VEP-based canonical transcript, the longest translation, was used for this gene. If not, the value is null |
|67 - CCDS | The [CCDS](https://www.ncbi.nlm.nih.gov/projects/CCDS/CcdsBrowse.cgi) identifier for this transcript, where applicable |
|68 - ENSP | The Ensembl protein identifier of the affected transcript |
|69 - SWISSPROT | [UniProtKB/Swiss-Prot](http://www.uniprot.org/) accession |
|70 - TREMBL | UniProtKB/TrEMBL identifier of protein product |
|71 - UNIPARC | UniParc identifier of protein product |
|72 - RefSeq | RefSeq identifier for this transcript |
|73 - SIFT | The [SIFT](http://sift.jcvi.org/) prediction and/or score, with both given as prediction (score) |
|74 - PolyPhen | The [PolyPhen](http://genetics.bwh.harvard.edu/pph2/) prediction and/or score |
|75 - EXON | The exon number (out of total number) |
|76 - INTRON | The intron number (out of total number) |
|77 - DOMAINS | The source and identifier of any overlapping protein domains |
|78 - GMAF | Non-reference allele and frequency of existing variant in [1000 Genomes](http://www.internationalgenome.org/) |
|79 - AFR_MAF | Non-reference allele and frequency of existing variant in 1000 Genomes combined African population |
|80 - AMR_MAF | Non-reference allele and frequency of existing variant in 1000 Genomes combined American population |
|81 - ASN_MAF | Non-reference allele and frequency of existing variant in 1000 Genomes combined Asian population |
|82 - EAS_MAF | Non-reference allele and frequency of existing variant in 1000 Genomes combined East Asian population |
|83 - EUR_MAF | Non-reference allele and frequency of existing variant in 1000 Genomes combined European population |
|84 - SAS_MAF | Non-reference allele and frequency of existing variant in 1000 Genomes combined South Asian population |
|85 - AA_MAF | Non-reference allele and frequency of existing variant in [NHLBI-ESP](https://genome.ucsc.edu/cgi-bin/hgTables?db=hg19&hgta_group=varRep&hgta_track=evsEsp6500&hgta_table=evsEsp6500&hgta_doSchema=describe+table+schema) African American population |
|86 - EA_MAF | Non-reference allele and frequency of existing variant in NHLBI-ESP European American population|
|87 - CLIN_SIG | Clinical significance of variant from dbSNP as annotated in ClinVar |
|88 - SOMATIC |Somatic status of each ID reported under Existing_variation (0, 1, or null) |
|89 - PUBMED | Pubmed ID(s) of publications that cite existing variant |
|90 - MOTIF_NAME |The source and identifier of a transcription factor binding profile aligned at this position |
|91 - MOTIF_POS | The relative position of the variation in the aligned TFBP |
|92 - HIGH_INF_POS | A flag indicating if the variant falls in a high information position of a transcription factor binding profile (TFBP) (Y, N, or null) |
| 93 - MOTIF_SCORE_CHANGE | The difference in motif score of the reference and variant sequences for the TFBP |
| 94 - IMPACT | The impact modifier for the consequence type |
| 95 - PICK | Indicates if this block of consequence data was picked by VEP's [pick feature](http://useast.ensembl.org/info/docs/tools/vep/script/vep_options.html#opt_pick) (1 or null) |
| 96 - VARIANT_CLASS | Sequence Ontology variant class |
| 97 - TSL | [Transcript support level](http://useast.ensembl.org/Help/Glossary?id=492), which is based on independent RNA analyses |
| 98 - HGVS_OFFSET | Indicates by how many bases the HGVS notations for this variant have been shifted |
| 99 - PHENO|Indicates if existing variant is associated with a phenotype, disease or trait (0, 1, or null) |
| 100 - MINIMISED | Alleles in this variant have been converted to minimal representation before consequence calculation (1 or null) |
| 101 - ExAC_AF | Global Allele Frequency from [ExAC](http://exac.broadinstitute.org/) |
| 102 - ExAC_AF_Adj | Adjusted Global Allele Frequency from ExAC |
| 103 - ExAC_AF_AFR | African/African American Allele Frequency from ExAC |
| 104 - ExAC_AF_AMR | American Allele Frequency from ExAC |
| 105 - ExAC_AF_EAS | East Asian Allele Frequency from ExAC |
| 106 - ExAC_AF_FIN | Finnish Allele Frequency from ExAC |
| 107 - ExAC_AF_NFE | Non-Finnish European Allele Frequency from ExAC |
| 108 - ExAC_AF_OTH | Other Allele Frequency from ExAC |
| 109 - ExAC_AF_SAS | South Asian Allele Frequency from ExAC |
| 110 - GENE_PHENO | Indicates if gene that the variant maps to is associated with a phenotype, disease or trait (0, 1, or null) |
| 111 - FILTER | Copied from input VCF.  This includes filters implemented directly by the variant caller and other external software used in the DNA-Seq pipeline. See below for additional details. |
| 112 - CONTEXT | The reference allele per VCF specs, and its five flanking base pairs |
| 113 - src_vcf_id | GDC UUID for the input VCF file |
| 114 - tumor_bam_uuid | GDC UUID for the tumor bam file |
| 115 - normal_bam_uuid | GDC UUID for the normal bam file |
| 116 - case_id | GDC UUID for the case |
| 117 - GDC_FILTER | GDC filters applied universally across all MAFs |
| 118 - COSMIC | Overlapping COSMIC variants |
| 119 - MC3_Overlap| Indicates whether this region overlaps with an MC3 variant for the same sample pair |
| 120 - GDC_Validation_Status | GDC implementation of validation checks. See notes section (#5) below for details |
| 121 - GDC_Valid_Somatic | True or False (not in somatic MAF) |
| 122 - vcf_region | Colon separated string containing the CHROM, POS, ID, REF, and ALT columns from the VCF file (e.g., chrZ:20:rs1234:A:T) (not in somatic MAF) |
| 123 - vcf_info | INFO column from VCF (not in somatic MAF) |
| 124 - vcf_format | FORMAT column from VCF (not in somatic MAF) |
| 125 - vcf_tumor_gt | Tumor sample genotype column from VCF (not in somatic MAF) |
| 126 - vcf_normal_gt | Normal sample genotype column from VCF (not in somatic MAF) |  

### Notes About GDC MAF Implementation

1. Column #4: __NCBI_Build__ is GRCh38 by default
2. Column #32: __Sequencer__ includes the sequencers used. If different sequencers were used to generate normal and tumor data, the normal sequencer is listed first.
3. Column #61: VEP name "STRAND" is changed to __TRANSCRIPT_STRAND__ to avoid confusion with Column#8 "Strand"
4. Column #94: __IMPACT__ categories are defined by the VEP software and do not necessarily reflect the relative biological influence of each mutation.
4. Column #122-125: __vcf_info, vcf_format, vcf_tumor_gt, and vcf_normal_gt__ are the corresponding columns from the VCF files. Including them facilitates parsing specific variant information.
5. Column #120: __GDC_Validation_Status__: GDC also collects TCGA validation sequences. It compares these with variants derived from Next-Generation Sequencing data from the same sample and populates the comparison result in "GDC_Validation_Status".
    * "Valid", if the alternative allele(s) in the tumor validation sequence is(are) the same as GDC variant call
    * "Invalid", if none of the alternative allele(s) in the tumor validation sequence is the same as GDC variant call
    * "Inconclusive" if two alternative allele exists, and one matches while the other does not
    * "Unknown" if no validation sequence exists
6. Column #121: __GDC_Valid_Somatic__ is TRUE if GDC_Validation_Status is "Valid" and the variant is "Somatic" in validation calls.  It is FALSE if these criteria are not met

### FILTER Value Definitions (column 111)

* __oxog :__ Signifies that this variant was determined to be an OxoG artifact. This was calculated with D-ToxoG.
* __bPcr :__ Signifies that this variant was determined to be an artifact of bias on the PCR template strand.  This was calculated with the [DKFZ Bias Filter](https://github.com/eilslabs/DKFZBiasFilter).
* __bSeq :__ Signifies that this variant was determined to be an artifact of bias on the forward/reverse strand. This was also calculated with the [DKFZ Bias Filter](https://github.com/eilslabs/DKFZBiasFilter).

## Impact Categories

### VEP

* __HIGH (H)__: The variant is assumed to have high (disruptive) impact in the protein, probably causing protein truncation, loss of function, or triggering nonsense mediated decay
* __MODERATE (M)__: A non-disruptive variant that might change protein effectiveness
* __LOW (L)__: Assumed to be mostly harmless or unlikely to change protein behavior
* __MODIFIER (MO)__: Usually non-coding variants or variants affecting non-coding genes, where predictions are difficult or there is no evidence of impact

### PolyPhen

* __probably damaging (PR)__: It is with high confidence supposed to affect protein function or structure
* __possibly damaging (PO)__: It is supposed to affect protein function or structure
* __benign (BE)__: Most likely lacking any phenotypic effect
* __unknown (UN)__: When in some rare cases, the lack of data does not allow PolyPhen to make a prediction

### SIFT

* __tolerated__: Not likely to have a phenotypic effect
* __tolerated_low_confidence__: More likely to have a phenotypic effect than 'tolerated'
* __deleterious__:  Likely to have a phenotypic effect
* __deleterious_low_confidence__: Less likely to have a phenotypic effect than 'deleterious'


# Data Release Notes

| Version | Date |
|---|---|
| [v41.0](Data_Release_Notes.md#data-release-410) | August 28, 2024 |
| [v40.0](Data_Release_Notes.md#data-release-400) | March 29, 2024 |
| [v39.0](Data_Release_Notes.md#data-release-390) | December 4, 2023 |
| [v38.0](Data_Release_Notes.md#data-release-380) | August 31, 2023 |
| [v37.0](Data_Release_Notes.md#data-release-370) | March 29, 2023 |
| [v36.0](Data_Release_Notes.md#data-release-360) | December 12, 2022 |
| [v35.0](Data_Release_Notes.md#data-release-350) | September 28, 2022 |
| [v34.0](Data_Release_Notes.md#data-release-340) | July 27, 2022 |
| [v33.1](Data_Release_Notes.md#data-release-331) | May 31, 2022 |
| [v33.0](Data_Release_Notes.md#data-release-330) | May 3, 2022 |
| [v32.0](Data_Release_Notes.md#data-release-320) | March 29, 2022 |
| [v31.0](Data_Release_Notes.md#data-release-310) | October 29, 2021 |
| [v30.0](Data_Release_Notes.md#data-release-300) | September 23, 2021 |
| [v29.0](Data_Release_Notes.md#data-release-290) | March 31, 2021 |
| [v28.0](Data_Release_Notes.md#data-release-280) | February 2, 2021 |
| [v27.0-fix](Data_Release_Notes.md#data-release-270-bug-fix) | November 9, 2020 |
| [v27.0](Data_Release_Notes.md#data-release-270) | October 29, 2020 |
| [v26.0](Data_Release_Notes.md#data-release-260) | September 8, 2020 |
| [v25.0](Data_Release_Notes.md#data-release-250) | July 22, 2020 |
| [v24.0](Data_Release_Notes.md#data-release-240) | May 7, 2020 |
| [v23.0](Data_Release_Notes.md#data-release-230) | April 7, 2020 |
| [v22.0](Data_Release_Notes.md#data-release-220) | January 16, 2020 |
| [v21.0](Data_Release_Notes.md#data-release-210) | December 10, 2019 |
| [v20.0](Data_Release_Notes.md#data-release-200) | November 11, 2019 |
| [v19.1](Data_Release_Notes.md#data-release-191) | November 6, 2019 |
| [v19.0](Data_Release_Notes.md#data-release-190) | September 17, 2019 |
| [v18.0](Data_Release_Notes.md#data-release-180) | July 8, 2019 |
| [v17.1](Data_Release_Notes.md#data-release-171) | June 12, 2019 |
| [v17.0](Data_Release_Notes.md#data-release-170) | June 5, 2019 |
| [v16.0](Data_Release_Notes.md#data-release-160) | March 26, 2019 |
| [v15.0](Data_Release_Notes.md#data-release-150) | February 20, 2019 |
| [v14.0](Data_Release_Notes.md#data-release-140) | December 18, 2018 |
| [v13.0](Data_Release_Notes.md#data-release-130) | September 27, 2018 |
| [v12.0](Data_Release_Notes.md#data-release-120) | June 13, 2018 |
| [v11.0](Data_Release_Notes.md#data-release-110) | May 21, 2018 |
| [v10.1](Data_Release_Notes.md#data-release-101) | February 15, 2018 |
| [v10.0](Data_Release_Notes.md#data-release-100) | December 21, 2017 |
| [v9.0](Data_Release_Notes.md#data-release-90) | October 24, 2017 |
| [v8.0](Data_Release_Notes.md#data-release-80) | August 22, 2017 |
| [v7.0](Data_Release_Notes.md#data-release-70) | June 29, 2017 |
| [v6.0](Data_Release_Notes.md#data-release-60) | May 9, 2017 |
| [v5.0](Data_Release_Notes.md#data-release-50) | March 16, 2017 |
| [v4.0](Data_Release_Notes.md#data-release-40) | October 31, 2016 |
| [v3.0](Data_Release_Notes.md#data-release-30) | September 16, 2016 |
| [v2.0](Data_Release_Notes.md#data-release-20) | August 9, 2016 |
| [v1.0](Data_Release_Notes.md#initial-data-release-10) | June 6, 2016 |

## Data Release 41.0

* __GDC Product__: Data
* __Release Date__: August 28, 2024

### New Updates

* New Projects
    * MATCH-C1
        * 11 cases
        * WXS, RNA-Seq
    * MATCH-P
        * 28 cases
        * WXS, RNA-Seq
    * MATCH-Z1B
        * 29 cases
        * WXS, RNA-Seq

* New Cases from Existing Projects
    * CPTAC-3 - 31 cases

* New Data Sets
    * TARGET-AML Tumor-Only Targeted Sequencing - 163 variant call sets
    * TCGA U133 Submitted Expression Arrays
        * TCGA-GBM - 560 aliquots
        * TCGA-LAML - 183 aliquots
        * TCGA-LUSC - 135 aliquots
        * TCGA-OV - 548 aliquots
    * TCGA-LUAD Methylation Data - 53 aliquots
    * CDDP_EAGLE-1 Slide Images - 49 cases
    * HCMI-CMDC
        * Tumor-Only WGS Data - 2 aliquot BAMs, 2 variant call sets
        * Tumor-Only WXS Data - 3 aliquot BAMs, 3 variant call sets
        * Updated clinical supplements
    * BEATAML1.0-COHORT scRNA-Seq Data - 8 aliquots

* Data Updates
    * Indexing of [ABSOLUTE Liftover](https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/CNV_Pipeline/) copy number variation data
    * Release of data for [Other Clinical Attribute](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=other_clinical_attribute) clinical entities
    * `platform` field populated for harmonized data files, can be used as a filter in `Repository`

A complete list of files included in the GDC Data Portal can be found below:

* [gdc_manifest_20240826_data_release_41.0_active.tsv.gz](https://api.gdc.cancer.gov/data/7e1bb66c-c944-4057-85b0-4bfe3bf3acd1)
* [DR41 Project Level Manifests](https://api.gdc.cancer.gov/data/339aef11-7176-45c5-811f-81bee1534ea8)
* [DR41 New Files Manifest](https://api.gdc.cancer.gov/data/3f7f2d3a-73a3-487c-ac9d-7aa2aa3aaaf1)

### Bugs Fixed Since Last Release

* Fixed 4 TARGET-NBL gene expression sets that pointed to multiple cases/aliquots <!--SV-1332-->
* Fixed multiple expression files per aliquot for several TARGET-AML RNA-Seq aliquots <!--DAT-3711-->

### Known Issues and Workarounds

* The slide image viewer does not display for any non-TCGA slides. At this time, these slides will need to be downloaded and viewed locally. Additionally, the slide image viewer does not display properly for 14 TCGA slides, which are identified [here](missing_tiling.txt).
* Pathology reports do not have any associated case/biospecimen information in the portal. This information can be found in the reports themselves. <!--SV-2118-->  
* 397 alignments from the TCGA program were found to have contamination values over 0.04 ([alignment list](Contaminated_Alignments.dr32.tsv)). The ensemble MAFs produced by these alignments were removed from the Data Portal.
* One methylation aliquot from the TCGA-COAD project, TCGA-D5-6930-01A-11D-1926-05, was not added to the portal and will be added in a future release.
* Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* No data from TARGET-MDLS is available.
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Data Portal<!--DAT-52-->. These annotations can be found [here](tcga-annotations-unavailable-20170315.json).
    * Tumor_grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->

## Data Release 40.0

* __GDC Product__: Data
* __Release Date__: March 29, 2024

### New Updates

* New Projects
    * MATCH-R - Genomic Characterization CS-MATCH-0007 Arm R - phs002029
        *  28 cases
        *  WXS, RNA-Seq
    * MATCH-S1 - Genomic Characterization CS-MATCH-0007 Arm S1 - phs002153
        *  41 cases
        *  WXS, RNA-Seq
    * MATCH-S2 - Genomic Characterization CS-MATCH-0007 Arm S2 - phs002178
        *  3 cases
        *  WXS, RNA-Seq
    * MATCH-Z1I - Genomic Characterization CS-MATCH-0007 Arm Z1I - phs002058
        *  26 cases
        *  WXS, RNA-Seq

* New Cases from Existing Projects
    * CPTAC-3 - 79 cases
    * REBC-THYR - 9 cases

* New Data Sets
    * Targeted Sequencing
        * TARGET-AML - 1,596 aliquot BAMs, 769 variant calls
        * TARGET-NBL - 998 aliquot BAMs, 476 variant calls
        * TARGET-OS - 233 aliquot BAMs, 65 variant calls
    * TCGA WGS
        * 57 alignments
        * 486 variant call aliquot pairs
    * REBC-THYR
        * WGS - 90 aliquot BAMs, 69 variant calls
        * miRNA-Seq - 177 aliquots
        * RNA-Seq - 78 aliquots
        * RNA-Seq - Addition of STAR-Fusion data to existing aliquots
    * HCMI-CMDC
        * Slide images for released cases
        * Updated clinical supplements
    * TCGA-GBM
        * miRNA-Seq - 8 aliquots
        * RNA-Seq - 1 aliquot

A complete list of files included in the GDC Data Portal can be found below:

* [gdc_manifest_27Mar2024_data_release_40.0_active.tsv.gz](https://api.gdc.cancer.gov/data/ddc77ead-d6d5-4732-80f9-2b4f7f76534d)
* [DR40 Project Level Manifests](https://api.gdc.cancer.gov/data/63826234-fd63-403e-aaeb-e68cc284fb8a)
* [DR40 New Files Manifest](https://api.gdc.cancer.gov/data/eecc9c5b-0fa0-4815-9f43-4bc0d885cf3d)

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

* The slide image viewer does not display for any non-TCGA slides. At this time, these slides will need to be downloaded and viewed locally. Additionally, the slide image viewer does not display properly for 14 TCGA slides, which are identified [here](missing_tiling.txt).
* Pathology reports do not have any associated case/biospecimen information in the portal. This information can be found in the reports themselves. <!--SV-2118-->  
* 397 alignments from the TCGA program were found to have contamination values over 0.04 ([alignment list](Contaminated_Alignments.dr32.tsv)). The ensemble MAFs produced by these alignments were removed from the Data Portal.
* One methylation aliquot from the TCGA-COAD project, TCGA-D5-6930-01A-11D-1926-05, was not added to the portal and will be added in a future release.
* Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* No data from TARGET-MDLS is available.
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Data Portal<!--DAT-52-->. These annotations can be found [here](tcga-annotations-unavailable-20170315.json).
    * Tumor_grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->


## Data Release 39.0

* __GDC Product__: Data
* __Release Date__: December 4, 2023

### New Updates

* New Projects
    * MATCH-H - Genomic Characterization CS-MATCH-0007 Arm H - phs001888
        *  21 cases
        *  WXS, RNA-Seq
    * MATCH-I - Genomic Characterization CS-MATCH-0007 Arm I - phs002181
        *  60 cases
        *  WXS, RNA-Seq
    * MATCH-U - Genomic Characterization CS-MATCH-0007 Arm U - phs002179
        *  23 cases
        *  WXS, RNA-Seq
    * MATCH-W - Genomic Characterization CS-MATCH-0007 Arm W - phs001948
        *  45 cases
        *  WXS, RNA-Seq
    * MATCH-Z1A - Genomic Characterization CS-MATCH-0007 Arm Z1A - phs001973
        *  45 cases
        *  WXS, RNA-Seq

* New Cases from Existing Projects
    * HCMI-CMDC - 19 cases

* New Data Sets
    * 6,957 WGS alignments from the TCGA program
    * 1,002 sets of WGS variants from TCGA
    * MP2PRT-ALL:  WXS and RNA-Seq data
    * Tumor-only data produced with a new pipeline. This includes raw and annotated VCFs and MAFs for the following projects. Note that all tumor-only variants are controlled-access:
        * BEATAML1.0-COHORT
        * BEATAML1.0-CRENOLANIB
        * CGCI-BLGSP
        * CPTAC-3
        * HCMI-CMDC
        * MATCH-B
        * MATCH-H
        * MATCH-I
        * MATCH-N
        * MATCH-Q
        * MATCH-U
        * MATCH-W
        * MATCH-Y
        * MATCH-Z1A
        * MATCH-Z1D
        * OHSU-CNL
        * ORGANOID-PANCREATIC
        * TARGET-ALL-P3
        * TARGET-WT
        * VAREPOP-APOLLO


* New Metadata
    * Sample type refactoring:
        * Four fields (tissue_type, specimen_type, preservation_method, tumor_descriptor) have been populated to contain the information that was previously populated in the sample_type field
        * The new field, specimen_type, is now available in the API to accommodate information about the biological makeup of the sample
    * The follow up data for CPTAC-3 has been updated

* Other Updates
    * CNV mutations are now available on the exploration page for projects that only had ASCAT CNV data from WGS files. This includes CNV mutations for the following projects:
        * APOLLO-LUAD
        * CDDP_EAGLE-1
        * CGCI-BLGSP
        * CGCI-HTMCP-CC
        * CGCI-HTMCP-DLBCL
        * CGCI-HTMCP-LC
        * CPTAC-3
        * HCMI-CMDC
        * MP2PRT-ALL
        * REBC-THYR
    * The GENIE program was removed from the GDC Portal because it was not representative of the latest version of GENIE
        * GENIE data can be accessed from the [AACR Repositories](https://www.aacr.org/professionals/research/aacr-project-genie/aacr-project-genie-data/)


A complete list of files included in the GDC Data Portal can be found below:

* [gdc_manifest_20231204_data_release_39.0_active.tsv.gz](https://api.gdc.cancer.gov/data/c5171766-a5b4-4347-9db2-645bc4c91659)
* [DR39 Project Level Manifests](https://api.gdc.cancer.gov/data/55e541f8-dde1-4d02-bd4b-c09aa4d36740)

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

* The slide image viewer does not display for any non-TCGA slides. At this time, these slides will need to be downloaded and viewed locally. Additionally, the slide image viewer does not display properly for 14 TCGA slides, which are identified [here](https://api.gdc.cancer.gov/data/3363f51f-2250-491c-a1ec-6a563a9527ad).
* Pathology reports do not have any associated case/biospecimen information in the portal. This information can be found in the reports themselves. <!--SV-2118-->  
* 397 alignments from the TCGA program were found to have contamination values over 0.04 ([alignment list](https://api.gdc.cancer.gov/data/a206b677-8396-4178-8ee2-36df4538aaf5)). The ensemble MAFs produced by these alignments were removed from the Data Portal.
* One methylation aliquot from the TCGA-COAD project, TCGA-D5-6930-01A-11D-1926-05, was not added to the portal and will be added in a future release.
* The Copy Number Estimate files in GENIE are labeled on the portal as TXT while the files are actually in TSV format.  <!--DAT-2728-->
* Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
    * Tumor grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->
* TARGET projects
    * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
    * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
    * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
    * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
    * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
    * Some TARGET files are not connected to all related aliquots <!--SV-929-->
    * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
    * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the NCI's webpage on [Using TARGET Data](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
        *  All microarray data and metadata
        *  All sequencing analyzed data and metadata
        *  1180 of 12063 sequencing runs of raw data
    * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
    * No data from TARGET-MDLS is available.

## Data Release 38.0

* __GDC Product__: Data
* __Release Date__: August 31, 2023

### New Updates

* New Projects
    * MP2PRT-ALL - Molecular Profiling to Predict Response to Treatment for Acute Lymphoblastic Leukemia - phs002005
        *  1,507 cases
        *  WGS
    * CGCI-HTMCP-DLBCL - HIV+ Tumor Molecular Characterization Project - Diffuse Large B-Cell Lymphoma - phs000235
        *  70 cases
        *  WGS, RNA-Seq, miRNA-Seq, Tissue Slide Images
    * MATCH-B - Genomic Characterization CS-MATCH-0007 Arm B - phs002028
        *  33 cases
        *  WXS, RNA-Seq
    * MATCH-N - Genomic Characterization CS-MATCH-0007 Arm N - phs002151
        *  21 cases
        *  WXS, RNA-Seq

* New Cases from Existing Projects
    * CPTAC-3 - GBM and Kidney cohorts - 50 cases
    * HCMI-CMDC - 31 cases
    * CGCI-BLGSP - 204 cases
    * TCGA-TGCT - 113 cases

* New Data Sets
    * 9,368 WGS alignments from the TCGA program
        * 4,676 Cases
        * 9,368 Aliquots
    * All methylation files that were produced with the SeSAMe pipeline was replaced with a new version.
    * TCGA SNP6 data processed with the ASCAT3 and ABSOLUTE pipelines
    * 172 CEL and birdseed files from TCGA SNP6
    * Release of remaining data for CGCI projects CGCI-BGLSP and CGCI-HTMCP-CC

* New Metadata
    * The `wgs_coverage` field is now populated for most BAMs and will allow for WGS BAMs to be queried by coverage range category.
    * The QC metrics for applicable BAMs are now queryable through the GDC Data Portal and API.
    * The `msi_status` and `msi_score` fields, which were produced using MSISensor2, are now queryable through the GDC Data Portal and API


A complete list of files included in the GDC Data Portal can be found below:

* [gdc_manifest_20230830_data_release_38.0_active.tsv.gz](https://api.gdc.cancer.gov/data/14573c1f-5dd2-4e00-b4d8-5da6af6e4baa)
* [DR38 Project Level Manifests](https://api.gdc.cancer.gov/data/6e85885d-a3f2-4bcc-a16f-9a34cfef4a67)

### Bugs Fixed Since Last Release

* The files produced with the SeSAMe pipeline had unfiltered methylation beta values that should be set as N/A for quality reasons.  These files were replaced.
* A bug in which certain files were shown to be associated with more aliquots than usual has been fixed.  

### Known Issues and Workarounds

* The slide image viewer does not display for any non-TCGA slides. At this time, these slides will need to be downloaded and viewed locally. Additionally, the slide image viewer does not display properly for 14 TCGA slides, which are identified [here](https://api.gdc.cancer.gov/data/3363f51f-2250-491c-a1ec-6a563a9527ad).
* Pathology reports do not have any associated case/biospecimen information in the portal. This information can be found in the reports themselves. <!--SV-2118-->  
* 397 alignments from the TCGA program were found to have contamination values over 0.04 ([alignment list](https://api.gdc.cancer.gov/data/a206b677-8396-4178-8ee2-36df4538aaf5)). The ensemble MAFs produced by these alignments were removed from the Data Portal.
* One methylation aliquot from the TCGA-COAD project, TCGA-D5-6930-01A-11D-1926-05, was not added to the portal and will be added in a future release.
* The Copy Number Estimate files in GENIE are labeled on the portal as TXT while the files are actually in TSV format.  <!--DAT-2728-->
* Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
    * Tumor grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->
* TARGET projects
    * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
    * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
    * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
    * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
    * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
    * Some TARGET files are not connected to all related aliquots <!--SV-929-->
    * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
    * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the NCI's webpage on [Using TARGET Data](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
        *  All microarray data and metadata
        *  All sequencing analyzed data and metadata
        *  1180 of 12063 sequencing runs of raw data
    * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
    * No data from TARGET-MDLS is available.

## Data Release 37.0

* __GDC Product__: Data
* __Release Date__: March 29, 2023

### New Updates

* New Projects
    * APOLLO-LUAD - Proteogenomic characterization of lung adenocarcinoma - phs003011
        *  87 cases
        *  WGS, RNA-Seq
    * CGCI-HTMCP-LC - HIV+ Tumor Molecular Characterization Project - Lung Cancer - phs000530
        *  39 cases
        *  WGS, RNA-Seq, miRNA-Seq, Slide Images
    * MATCH-Q - Genomic Characterization CS-MATCH-0007 Arm Q - phs001926
        *  35 cases
        *  WXS, RNA-Seq
    * MATCH-Y - Genomic Characterization CS-MATCH-0007 Arm Y - phs001904
        *  31 cases
        *  WXS, RNA-Seq

* New Data from Existing Projects
    * CPTAC-3 - 139 new cases and two new snRNA-Seq samples
    * HCMI-CMDC - 118 new cases
    * TCGA-THCA - 941 new WGS alignments
    * TARGET-OS and TARGET-ALL-P2 - Masked Somatic Mutation MAFs are now open access and their mutations now appear in the exploration portal.

* Data Migrated from the Legacy Archive to Active Portal
    * Birdseed files that were generated from Affymetrix SNP6 arrays
    * Additional WGS Alignments are now available for TCGA projects
    * Additional samples from RNA-Seq and WXS are now available for TCGA projects


A complete list of files included in the GDC Data Portal can be found below:

* [gdc_manifest_20230329_data_release_37.0_active.tsv.gz](https://api.gdc.cancer.gov/data/838d2fe9-6540-441e-a666-2fd8af47ebf0)
* [DR37 Project Level Manifests](https://api.gdc.cancer.gov/data/27fe396d-c40f-4d6a-b3c4-bb9dea0f2148)

### Unavailable Files

* 56 CPTAC-3 snRNA-Seq files are currently unavailable for download. A list of the affected files can be found [here](https://api.gdc.cancer.gov/data/e89608f1-bc1a-405e-a0d2-7da563183f67). These files will be restored for download by the next data release. <!--DAT-3433-->


### Bugs Fixed Since Last Release

* Outcome data for the CPTAC program has been updated.
* The `age_at_index` field was incorrectly reported in days in the GENIE program.  These values have been removed as it contained the same information as the `days_to_birth` field. <!--SV-2032-->


### Known Issues and Workarounds

* The current files produced with the SeSAMe pipeline have unfiltered methylation beta values that should be set as N/A for quality reasons.  These files will be replaced in a future release.
* Pathology reports do not have any associated case/biospecimen information in the portal. This information can be found in the reports themselves. <!--SV-2118-->  
* 397 alignments from the TCGA program were found to have contamination values over 0.04 ([alignment list](https://api.gdc.cancer.gov/data/a206b677-8396-4178-8ee2-36df4538aaf5)). The ensemble MAFs produced by these alignments were removed from the Data Portal.
* One methylation aliquot from the TCGA-COAD project, TCGA-D5-6930-01A-11D-1926-05, was not added to the portal and will be added in a future release.
* The clinical supplement for TARGET-ALL-P1 is not currently available. It will be made available in a future release.
* The slide image viewer does not display properly for 14 slides, which are identified [here](https://api.gdc.cancer.gov/data/3363f51f-2250-491c-a1ec-6a563a9527ad).  The full slide image can be downloaded as an SVS file.
* The Copy Number Estimate files in GENIE are labeled on the portal as TXT while the files are actually in TSV format.  <!--DAT-2728-->
* Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release. <!--DAT-1589-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
    * Tumor grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->
* TARGET projects
    * TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
        * TARGET-20-PASJGZ-04A-02D
        * TARGET-30-PAPTLY-01A-01D
        * TARGET-20-PAEIKD-09A-01D
        * TARGET-20-PASMYS-14A-02D
        * TARGET-20-PAMYAS-14A-02D
        * TARGET-10-PAPZST-09A-01D
    * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
    * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
    * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
    * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
    * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
    * Some TARGET files are not connected to all related aliquots <!--SV-929-->
    * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
    * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
        *  All microarray data and metadata
        *  All sequencing analyzed data and metadata
        *  1180 of 12063 sequencing runs of raw data
    * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
    * No data from TARGET-MDLS is available.
* Issues in the Legacy Archive
    * The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
    * SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
    * Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
    * SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
    * TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->


## Data Release 36.0

* __GDC Product__: Data
* __Release Date__: December 12, 2022

### New Updates

* New Projects
    * MATCH-Z1D - Genomic Characterization CS-MATCH-0007 Arm Z1D - phs001859
        *  36 cases
        *  WXS, RNA-Seq
    * CDDP_EAGLE-1 - CDDP Integrative Analysis of Lung Adenocarcinoma (Phase 2) - phs001239
        *  50 cases
        *  WXS, WGS, RNA-Seq

* New Data from Existing Projects
    * CMI-MPC - new RNA-Seq and WXS data

* Data Migrated from the Legacy Archive to Active Portal
    * WGS Alignments are now available for 25 TCGA Projects
    * Pathology reports from TCGA
    * Affymetrix SNP6 Genotyping Array CEL files
    * A set of WXS and RNA-Seq samples from TCGA and TARGET that failed harmonization at launch have been rerun and are now available in the active portal.
    * TCGA Bisulfite-Seq files can be downloaded using the following manifests:
        * [TARGET-RT](https://api.gdc.cancer.gov/data/ec12eb66-2b44-40e7-9cde-9598e6839a44)
        * [TCGA-BLCA](https://api.gdc.cancer.gov/data/3482eb71-130c-4cf9-b187-5feea0a0c6f9)
        * [TCGA-BRCA](https://api.gdc.cancer.gov/data/78dd8adc-af9b-4d32-ac9b-9ab12c4f633c)
        * [TCGA-COAD](https://api.gdc.cancer.gov/data/8416eb46-11e3-4936-a7e5-61cb549a6bf8)
        * [TCGA-GBM](https://api.gdc.cancer.gov/data/2827b5aa-842c-4511-b639-e5b31f096858)
        * [TCGA-LUAD](https://api.gdc.cancer.gov/data/244ec7d5-3f2b-4157-8259-1de7c6c717c6)
        * [TCGA-LUSC](https://api.gdc.cancer.gov/data/a9612080-3ced-4d66-9469-afe291a1cb59)
        * [TCGA-READ](https://api.gdc.cancer.gov/data/ec423eac-a8e1-4bdf-a19a-01b920386864)
        * [TCGA-STAD](https://api.gdc.cancer.gov/data/147b3449-6f4b-4189-b5e3-1e2fc0b8491e)
        * [TCGA-UCEC](https://api.gdc.cancer.gov/data/005b4d7d-a5bb-49ad-b072-c8194e6260ff)

A complete list of files included in the GDC Data Portal can be found below:

* [gdc_manifest_20221212_data_release_36.0_active.tsv.gz](https://api.gdc.cancer.gov/data/27a4fae0bb-f1fc-4e6f-b8cc-05aa9c28f8bc)

### Unavailable Files

* None


### Bugs Fixed Since Last Release

* The copy number variation data is now available on the GDC Exploration portal.
* The mutations on GDC Exploration were re-built with the correct gene model.


### Known Issues and Workarounds

* Outcome data for the CPTAC program is not up-to-date. Please visit the [Proteomic Data Commons](https://proteomic.datacommons.cancer.gov/pdc/) for updated outcome data for CPTAC.
* Pathology reports do not have any associated case/biospecimen information in the portal. This information can be found in the reports themselves. <!--SV-2118-->  
* 397 alignments from the TCGA program were found to have contamination values over 0.04 ([alignment list](https://api.gdc.cancer.gov/data/a206b677-8396-4178-8ee2-36df4538aaf5)). The ensemble MAFs produced by these alignments were removed from the Data Portal.
* One methylation aliquot from the TCGA-COAD project, TCGA-D5-6930-01A-11D-1926-05, was not added to the portal and will be added in a future release.
* The clinical supplement for TARGET-ALL-P1 is not currently available. It will be made available in a future release.
* The slide image viewer does not display properly for 14 slides, which are identified [here](https://api.gdc.cancer.gov/data/3363f51f-2250-491c-a1ec-6a563a9527ad).  The full slide image can be downloaded as an SVS file.
* The Copy Number Estimate files in GENIE are labeled on the portal as TXT while the files are actually in TSV format.  <!--DAT-2728-->
* Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release. <!--DAT-1589-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
    * Tumor grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->
* TARGET projects
    * TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
        * TARGET-20-PASJGZ-04A-02D
        * TARGET-30-PAPTLY-01A-01D
        * TARGET-20-PAEIKD-09A-01D
        * TARGET-20-PASMYS-14A-02D
        * TARGET-20-PAMYAS-14A-02D
        * TARGET-10-PAPZST-09A-01D
    * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
    * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
    * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
    * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
    * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
    * Some TARGET files are not connected to all related aliquots <!--SV-929-->
    * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
    * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
        *  All microarray data and metadata
        *  All sequencing analyzed data and metadata
        *  1180 of 12063 sequencing runs of raw data
    * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
    * No data from TARGET-MDLS is available.
* Issues in the Legacy Archive
    * The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
    * SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
    * Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
    * SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
    * TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->

## Data Release 35.0

* __GDC Product__: Data
* __Release Date__: September 28, 2022

### New Updates

* The SomaticSniper variant calling pipeline was deprecated.  To support this, the following changes were made:
    * All SomaticSniper files no longer appear in the portal, but still can be downloaded using the Data Transfer Tool or API using the original UUID.
    * The aggregated somatic mutation and masked somatic mutation files (multi-caller MAFs) have been replaced to reflect the absence of variants from the SomaticSniper pipeline.
    * The mutations on the exploration portal reflect the above-mentioned masked somatic mutation files.
* 10 snRNA-Seq samples were released from the CPTAC-3 project.
* Additional RNA-Seq samples from 2,082 additional cases are now available for the TARGET-AML project.
* Demographic data has been added for 94 cases in TARGET-ALL-P2 and TARGET-ALL-P3 projects. A list of the updated cases can be found
[here](https://api.gdc.cancer.gov/data/ce79d116-f397-4cfa-8636-8e482fce8631).

A complete list of files included in the GDC Data Portal can be found below:

* [gdc_manifest_20220928_data_release_35.0_active.tsv.gz](https://api.gdc.cancer.gov/data/e7808c04-d03f-48f4-afb6-916073cbac98)

### Unavailable Files

* None


### Bugs Fixed Since Last Release

* Data from two HCMI-CMDC aliquots (HCM-BROD-0100-C15-85A-01D-A786-36 and HCM-BROD-0679-C43-85M-01D-A80U-36) were incorrectly selected for inclusion into the Exploration Page in Data Release 32 and has been replaced with the correct aliquots (HCM-BROD-0100-C15-01A-11D-A786-36 and HCM-BROD-0679-C43-06A-11D-A80U-36). <!--DAT-3220-->

### Known Issues and Workarounds

* The mutations on GDC Exploration were built with an incorrect gene model.
    * The mutations are still correct in terms of the gene affected, coordinates, DNA changes, amino acid changes, and impact.
    * Mutations associated with genes that were present in GENCODE v36 and not GENCODE v22 are not displayed. This affects less than 1% of mutations.
    * Files downloaded from the the GDC Repository are not affected by this issue.  This only affects mutations that are downloaded from GDC Exploration.
* Pathology reports do not have any associated case/biospecimen information in the portal. This information can be found in the reports themselves. <!--SV-2118-->  
* 397 alignments from the TCGA program were found to have contamination values over 0.04 ([alignment list](https://api.gdc.cancer.gov/data/a206b677-8396-4178-8ee2-36df4538aaf5)). The ensemble MAFs produced by these alignments were removed from the Data Portal.
* One methylation aliquot from the TCGA-COAD project, TCGA-D5-6930-01A-11D-1926-05, was not added to the portal and will be added in a future release.
* The clinical supplement for TARGET-ALL-P1 is not currently available. It will be made available in a future release.
* Copy number variations currently do not appear in the Exploration page.  This will be restored in a future release.
* Mutations from SomaticSniper were erroneously labelled as LOH (loss of heterozygosity). This affects the VCF files, MAF files, and may cause SomaticSniper mutations to be absent from ensemble MAFs.
* The slide image viewer does not display properly for 14 slides, which are identified [here](https://api.gdc.cancer.gov/data/3363f51f-2250-491c-a1ec-6a563a9527ad).  The full slide image can be downloaded as an SVS file.
* The Copy Number Estimate files in GENIE are labeled on the portal as TXT while the files are actually in TSV format.  <!--DAT-2728-->
* Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release. <!--DAT-1589-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
    * Tumor grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->
* TARGET projects
    * TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
        * TARGET-20-PASJGZ-04A-02D
        * TARGET-30-PAPTLY-01A-01D
        * TARGET-20-PAEIKD-09A-01D
        * TARGET-20-PASMYS-14A-02D
        * TARGET-20-PAMYAS-14A-02D
        * TARGET-10-PAPZST-09A-01D
    * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
    * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
    * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
    * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
    * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
    * Some TARGET files are not connected to all related aliquots <!--SV-929-->
    * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
    * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
        *  All microarray data and metadata
        *  All sequencing analyzed data and metadata
        *  1180 of 12063 sequencing runs of raw data
    * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
    * No data from TARGET-MDLS is available.
* Issues in the Legacy Archive
    * The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
    * SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
    * Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
    * SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
    * TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->

## Data Release 34.0

* __GDC Product__: Data
* __Release Date__: July 27, 2022

### New updates

* 251 cases from the CPTAC-3 project were added to the portal.  This includes all files associated with these cases.
* 243 cases from the BEATAML1.0-COHORT project were added to the portal.  This includes most of the files associated with these cases.
    * The raw tumor-only VCFs from BEATAML1.0-COHORT are downloadable from the BEATAML1.0-COHORT (2022) publication page [here](https://gdc.cancer.gov/about-data/publications/BEATAML1-0-COHORT-2022) and will be added to the Data Portal in a future release.
* WXS mutations from the BEATAML1.0-COHORT project are now available in the Exploration portal.
* Transcript fusion files are now available for the following projects:
    * BEATAML1.0-COHORT
    * CMI-ASC
    * CMI-MBC
    * CPTAC-2
    * CTSP-DLBCL1
    * MMRF-COMMPASS
    * NCICCR-DLBCL
    * OHSU-CNL
    * ORGANOID-PANCREATIC
    * WCDT-MCRPC

A complete list of files included in the GDC Data Portal can be found below:

* [gdc_manifest_20220727_data_release_34.0_active.tsv.gz](https://api.gdc.cancer.gov/data/f0f04445-c20a-43ea-8c55-9ff1e51aff96)

### Unavailable Files

* None


### Bugs Fixed Since Last Release

* Data from two HCMI-CMDC aliquots (HCM-BROD-0100-C15-85A-01D-A786-36 and HCM-BROD-0679-C43-85M-01D-A80U-36) were incorrectly selected for inclusion into the Exploration Page in Data Release 32 and has been replaced with the correct aliquots (HCM-BROD-0100-C15-01A-11D-A786-36 and HCM-BROD-0679-C43-06A-11D-A80U-36). <!--DAT-3220-->

### Known Issues and Workarounds

* Pathology reports do not have any associated case/biospecimen information in the portal. This information can be found in the reports themselves. <!--SV-2118-->  
* 397 alignments from the TCGA program were found to have contamination values over 0.04 ([alignment list](https://api.gdc.cancer.gov/data/a206b677-8396-4178-8ee2-36df4538aaf5)). The ensemble MAFs produced by these alignments were removed from the Data Portal.
* One methylation aliquot from the TCGA-COAD project, TCGA-D5-6930-01A-11D-1926-05, was not added to the portal and will be added in a future release.
* The clinical supplement for TARGET-ALL-P1 is not currently available. It will be made available in a future release.
* Copy number variations currently do not appear in the Exploration page.  This will be restored in a future release.
* Mutations from SomaticSniper were erroneously labelled as LOH (loss of heterozygosity). This affects the VCF files, MAF files, and may cause SomaticSniper mutations to be absent from ensemble MAFs.
* The slide image viewer does not display properly for 14 slides, which are identified [here](https://api.gdc.cancer.gov/data/3363f51f-2250-491c-a1ec-6a563a9527ad).  The full slide image can be downloaded as an SVS file.
* The Copy Number Estimate files in GENIE are labeled on the portal as TXT while the files are actually in TSV format.  <!--DAT-2728-->
* Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release. <!--DAT-1589-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
    * Tumor grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->
* TARGET projects
    * TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
        * TARGET-20-PASJGZ-04A-02D
        * TARGET-30-PAPTLY-01A-01D
        * TARGET-20-PAEIKD-09A-01D
        * TARGET-20-PASMYS-14A-02D
        * TARGET-20-PAMYAS-14A-02D
        * TARGET-10-PAPZST-09A-01D
    * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
    * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
    * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
    * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
    * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
    * Some TARGET files are not connected to all related aliquots <!--SV-929-->
    * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
    * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
        *  All microarray data and metadata
        *  All sequencing analyzed data and metadata
        *  1180 of 12063 sequencing runs of raw data
    * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
    * No data from TARGET-MDLS is available.
* Issues in the Legacy Archive
    * The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
    * SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
    * Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
    * SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
    * TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->


## Data Release 33.1

* __GDC Product__: Data
* __Release Date__: May 31, 2022

### New updates

* None, see "Bugs Fixed Since Last Release" section below.

A complete list of files included in the GDC Data Portal can be found below:

* [gdc_manifest_20220531_data_release_33.1_active.tsv.gz](https://api.gdc.cancer.gov/data/0e16d92d-9627-4ed5-b7d2-80b6d78e1d35)

### Unavailable Files

* None


### Bugs Fixed Since Last Release

* 32 cases from the EXCEPTIONAL_RESPONDERS-ER project were released as they were missing from the previous release.
* All mutations from EXCEPTIONAL_RESPONDERS-ER in the exploration portal come from WXS data, whereas they were previously a mixture of WXS and Targeted Sequencing.

### Known Issues and Workarounds

* Pathology reports do not have any associated case/biospecimen information in the portal. This information can be found in the reports themselves. <!--SV-2118-->  
* 397 alignments from the TCGA program were found to have contamination values over 0.04 ([alignment list](https://api.gdc.cancer.gov/data/a206b677-8396-4178-8ee2-36df4538aaf5)). The ensemble MAFs produced by these alignments were removed from the Data Portal.
* One methylation aliquot from the TCGA-COAD project, TCGA-D5-6930-01A-11D-1926-05, was not added to the portal and will be added in a future release.
* The clinical supplement for TARGET-ALL-P1 is not currently available. It will be made available in a future release.
* Copy number variations currently do not appear in the Exploration page.  This will be restored in a future release.
* Mutations from SomaticSniper were erroneously labelled as LOH (loss of heterozygosity). This affects the VCF files, MAF files, and may cause SomaticSniper mutations to be absent from ensemble MAFs.
* The slide image viewer does not display properly for 14 slides, which are identified [here](https://api.gdc.cancer.gov/data/3363f51f-2250-491c-a1ec-6a563a9527ad).  The full slide image can be downloaded as an SVS file.
* The Copy Number Estimate files in GENIE are labeled on the portal as TXT while the files are actually in TSV format.  <!--DAT-2728-->
* Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release. <!--DAT-1589-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
    * Tumor grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->
* TARGET projects
    * TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
        * TARGET-20-PASJGZ-04A-02D
        * TARGET-30-PAPTLY-01A-01D
        * TARGET-20-PAEIKD-09A-01D
        * TARGET-20-PASMYS-14A-02D
        * TARGET-20-PAMYAS-14A-02D
        * TARGET-10-PAPZST-09A-01D
    * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
    * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
    * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
    * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
    * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
    * Some TARGET files are not connected to all related aliquots <!--SV-929-->
    * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
    * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
        *  All microarray data and metadata
        *  All sequencing analyzed data and metadata
        *  1180 of 12063 sequencing runs of raw data
    * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
    * No data from TARGET-MDLS is available.
* Issues in the Legacy Archive
    * The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
    * SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
    * Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
    * SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
    * TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->


## Data Release 33.0

* __GDC Product__: Data
* __Release Date__: May 3, 2022

### New updates

1.  New Project: NCI Exceptional Responders Initiative (EXCEPTIONAL_RESPONDERS-ER, phs001145)
    * RNA-Seq - 45 Cases
    * WXS - 50 Cases
    * Targeted Sequencing - 41 Cases
    * Mutations from WXS and Targeted Sequencing are present in the exploration page.
2.  New Project: Molecular Profiling to Predict Response to Treatment - Wilms Tumor (MP2PRT-WT, phs001965)
    * WGS - 52 Cases
    * RNA-Seq - 52 Cases
    * miRNA-Seq - 52 Cases
3.  Methylation files from the SeSAMe pipeline are now available for CGCI-HTMCP-CC and the TARGET projects.

A complete list of files for this release are listed for the GDC Data Portal and the GDC Legacy Archive are found below:

* [gdc_manifest_20220503_data_release_33.0_active.tsv.gz](https://api.gdc.cancer.gov/data/684e304b-02e6-47e9-99e6-30ecdb4fe644)
* [gdc_manifest_20220503_data_release_33.0_legacy.tsv.gz](https://api.gdc.cancer.gov/data/0dadd9da-d96c-454e-9d4a-65a64730637c)

### Unavailable Files

* The Arriba pipeline failed for one aliquot from EXCEPTIONAL-RESPONDERS-ER and is documented [here](https://api.gdc.cancer.gov/data/80cf3148-1dbb-4fda-834d-118d3f8bedb7).


### Bugs Fixed Since Last Release

* Gene-level copy number files from TCGA-THCA and TCGA-UCEC were set as controlled-access files.  These have been corrected to be available as open-access files.
* Due to a problem with the columns generated by the pipeline, all scRNA-Seq files have been replaced with a new version.

### Known Issues and Workarounds

* 397 alignments from the TCGA program were found to have contamination values over 0.04 ([alignment list](https://api.gdc.cancer.gov/data/a206b677-8396-4178-8ee2-36df4538aaf5)). The ensemble MAFs produced by these alignments were removed from the Data Portal.
* One methylation aliquot from the TCGA-COAD project, TCGA-D5-6930-01A-11D-1926-05, was not added to the portal and will be added in a future release.
* The clinical supplement for TARGET-ALL-P1 is not currently available. It will be made available in a future release.
* Copy number variations currently do not appear in the Exploration page.  This will be restored in a future release.
* Mutations from SomaticSniper were erroneously labelled as LOH (loss of heterozygosity). This affects the VCF files, MAF files, and may cause SomaticSniper mutations to be absent from ensemble MAFs.
* The slide image viewer does not display properly for 14 slides, which are identified [here](https://api.gdc.cancer.gov/data/3363f51f-2250-491c-a1ec-6a563a9527ad).  The full slide image can be downloaded as an SVS file.
* The Copy Number Estimate files in GENIE are labeled on the portal as TXT while the files are actually in TSV format.  <!--DAT-2728-->
* Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release. <!--DAT-1589-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
    * Tumor grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->
* TARGET projects
    * TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
        * TARGET-20-PASJGZ-04A-02D
        * TARGET-30-PAPTLY-01A-01D
        * TARGET-20-PAEIKD-09A-01D
        * TARGET-20-PASMYS-14A-02D
        * TARGET-20-PAMYAS-14A-02D
        * TARGET-10-PAPZST-09A-01D
    * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
    * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
    * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
    * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
    * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
    * Some TARGET files are not connected to all related aliquots <!--SV-929-->
    * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
    * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
        *  All microarray data and metadata
        *  All sequencing analyzed data and metadata
        *  1180 of 12063 sequencing runs of raw data
    * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
    * No data from TARGET-MDLS is available.
* Issues in the Legacy Archive
    * The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
    * SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
    * Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
    * SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
    * TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->

## Data Release 32.0

* __GDC Product__: Data - GENCODE v36 Release
* __Release Date__: March 29, 2022

### New updates

#### New data files
1.  The following data types have been replaced with new GENCODE v36 versions
    * RNA-Seq: all files, including alignments, gene expression files, and transcript fusion files.
    * WXS and Targeted Sequencing: annotated VCFs, single-caller MAFs, Ensemble MAFs.
    * WGS: BEDPE-format structural variants and gene-level copy number variants.
    * GENIE Targeted Sequencing files.
    * FM-AD Targeted Sequencing files.
        * The primary-site-level FM-AD MAF files have been replaced with aliquot-level MAF files.
1. RNA-Seq STAR-Counts files now contain additional normalized counts such as FPKM, FPKM-UQ, and TPM.
1. All WXS files for TCGA have been replaced with new versions. Alignments will contain QC metrics and variants were produced using the same pipelines as all other GDC projects.
1. TCGA RNA-Seq has been changed to contain three alignments (genomic, transcriptome, and chimeric), STAR-counts files, and transcript fusion files for each aliquot.
1. The project-level MAFs in TCGA and FM-AD have been replaced with aliquot-level MAFs.
1. GENCODE v22 derived files (not BAM) that no longer appear in the portal will be downloadable as previous versions of v36 files.  
1. Methylation data produced from the SeSAMe pipeline is now available for all TCGA projects.
1. Note that miRNA-Seq data remains unchanged. The miRNA-Seq pipeline uses the miRBase database, which is not affected by the GENCODE version change.
1. A set of manifests were generated at the project-level that map each v22 file to its corresponding v36 file.  These can be used to help users transition from v22 to v36 and can be downloaded [here](https://github.com/NCI-GDC/gdc-docs/tree/develop/docs/Data/Release_Notes/GCv36_Manifests).

#### Removed data files and pipelines
1. Files from the HTSeq pipeline are no longer supported and will no longer appear in the portal. Normalized counts can now be found in the STAR-Counts files.  
1. Files that originated from the methylation liftover pipeline are no longer supported and will no longer appear in the portal.
1. GENCODE v22 BAM files that no longer appear in the portal will be available for six months past this release. They may not be available after that.
1. New variant calling tumor-normal pairing was implemented in TCGA, which results in certain aliquots no longer being available as a v36 version (see the aliquots labeled "Unpaired Aliquots" [here](https://api.gdc.cancer.gov/data/80cf3148-1dbb-4fda-834d-118d3f8bedb7)).
1. Some aliquots failed harmonization when the new v36 gene model was used, which results in some new versions no longer being available (see the aliquots labeled "Failed Harmonization" [here](https://api.gdc.cancer.gov/data/80cf3148-1dbb-4fda-834d-118d3f8bedb7)).
1. Some aliquots were found to contain a cross-patient contamination level of over 0.04 as measured by GATK4 CalculateContamination (see the aliquots labeled "Contamination" [here](https://api.gdc.cancer.gov/data/80cf3148-1dbb-4fda-834d-118d3f8bedb7)).  

#### Data Portal Exploration Data
1. The Data Portal Exploration Page is now populated based on open-access mutations from analyses that used GENCODE v36.
1. Mutations from SomaticSniper will not appear on the Exploration page.
1. Due to the copy number variation pipeline transition from GISTIC to ASCAT, the CNV data was not included in the GDC Exploration page. This will be replaced in a future release once visualization of the new pipeline is fully assessed.
1. The TCGA program mutations have been processed using the same pipeline as all other projects, which resulted in a 26% reduction in the number of open-access mutations. Some points on this change are listed below with TCGA-BRCA as the benchmark project:
    * 97% of the previously released open-access mutations are still discoverable in the new GDC controlled-access MAFs. This number increases to 99.95% when focusing only on mutations that were also called by [MC3](https://gdc.cancer.gov/about-data/publications/pancanatlas).
    * Somatic mutations will now be removed from the Data Portal Exploration Page unless they are detected by more than one variant calling software. This accounts for 40% of the total reduction.
    * Somatic mutations will now be removed from the Data Portal Exploration Page if they are detected outside of the target capture region, while previously out-of-target mutations detected from the TCGA Gene Annotation File (GAF) regions were allowed. This accounts for 36% of the total reduction.  
    * Some TCGA-specific variant-rescue steps have been removed in favor of a more robust and uniform filtering pipeline.
    * Some other minor changes due to updates in the gene model or other databases (e.g., the ExAC germline variant database was replaced with gnomAD in DR32).


A complete list of files for this release are listed for the GDC Data Portal and the GDC Legacy Archive are found below:

* [gdc_manifest_20220316_data_release_32.0_active.tsv.gz](https://api.gdc.cancer.gov/data/a79c578c-d601-4113-8afd-3b36b410133c)
* [gdc_manifest_20220316_data_release_32.0_legacy.tsv.gz](https://api.gdc.cancer.gov/data/65788d0e-b415-431e-b53e-5e59dc6198a5)

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

* 397 alignments from the TCGA program were found to have contamination values over 0.04 ([alignment list](https://api.gdc.cancer.gov/data/a206b677-8396-4178-8ee2-36df4538aaf5)). The ensemble MAFs produced by these alignments were removed from the Data Portal.
* One methylation aliquot from the TCGA-COAD project, TCGA-D5-6930-01A-11D-1926-05, was not added to the portal and will be added in a future release.
* The clinical supplement for TARGET-ALL-P1 is not currently available. It will be made available in a future release.
* Copy number variations currently do not appear in the Exploration page.  This will be restored in a future release.
* Mutations from SomaticSniper were erroneously labelled as LOH (loss of heterozygosity). This affects the VCF files, MAF files, and may cause SomaticSniper mutations to be absent from ensemble MAFs.
* The slide image viewer does not display properly for 14 slides, which are identified [here](https://api.gdc.cancer.gov/data/3363f51f-2250-491c-a1ec-6a563a9527ad).  The full slide image can be downloaded as an SVS file.
* The Copy Number Estimate files in GENIE are labeled on the portal as TXT while the files are actually in TSV format.  <!--DAT-2728-->
* Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release. <!--DAT-1589-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
    * Tumor grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->
* TARGET projects
    * TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
        * TARGET-20-PASJGZ-04A-02D
        * TARGET-30-PAPTLY-01A-01D
        * TARGET-20-PAEIKD-09A-01D
        * TARGET-20-PASMYS-14A-02D
        * TARGET-20-PAMYAS-14A-02D
        * TARGET-10-PAPZST-09A-01D
    * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
    * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
    * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
    * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
    * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
    * Some TARGET files are not connected to all related aliquots <!--SV-929-->
    * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
    * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
        *  All microarray data and metadata
        *  All sequencing analyzed data and metadata
        *  1180 of 12063 sequencing runs of raw data
    * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
    * No data from TARGET-MDLS is available.
* Issues in the Legacy Archive
    * The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
    * SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
    * Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
    * SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
    * TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->




## Data Release 31.0

* __GDC Product__: Data
* __Release Date__: October 29, 2021

### New updates

1.  TCGA Slide Images:
    * All TCGA slide images that were removed earlier this year have been restored.
    * Note that the UUIDs for most TCGA slide images have changed.  Older manifest files may not work when downloading slide images.
2. CPTAC-3 clinical data has been refreshed and includes new follow up entities.
3. REBC-THYR
    * The clinical and biospecimen XML files were removed as they were not intended for release in DR 30.
    * The case REBC-ADL5 was added, which includes one WGS pair.

A complete list of files for this release are listed for the GDC Data Portal and the GDC Legacy Archive are found below:

* [gdc_manifest_20211029_data_release_31.0_active.tsv.gz](https://api.gdc.cancer.gov/data/4a5ba7bc-b53f-40ee-b536-7a4c4ad30eaa)
* [gdc_manifest_20211029_data_release_31.0_legacy.tsv.gz](https://api.gdc.cancer.gov/data/036c711f-77f5-4c6a-9426-00f7f4fe39ef)

### Bugs Fixed Since Last Release

* One file from a previous version of the methylation pipeline appeared in the data portal (bd2f864a-3f00-47b5-815d-bd01ca21ef61; CPTAC-3).  This file should no longer appear in the data portal. <!--SV-2012-->

### Known Issues and Workarounds

* The slide image viewer does not display properly for 14 slides, which are identified [here](https://api.gdc.cancer.gov/data/3363f51f-2250-491c-a1ec-6a563a9527ad).  The full slide image can be downloaded as an SVS file.
* The Copy Number Estimate files in GENIE are labeled on the portal as TXT while the files are actually in TSV format.  <!--DAT-2728-->
* Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release <!--DAT-1589-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
    * Tumor grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->
* TARGET projects
    * TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
        * TARGET-20-PASJGZ-04A-02D
        * TARGET-30-PAPTLY-01A-01D
        * TARGET-20-PAEIKD-09A-01D
        * TARGET-20-PASMYS-14A-02D
        * TARGET-20-PAMYAS-14A-02D
        * TARGET-10-PAPZST-09A-01D
    * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
    * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
    * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
    * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
    * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
    * Some TARGET files are not connected to all related aliquots <!--SV-929-->
    * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
    * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
        *  All microarray data and metadata
        *  All sequencing analyzed data and metadata
        *  1180 of 12063 sequencing runs of raw data
    * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
    * No data from TARGET-MDLS is available.
* Issues in the Legacy Archive
    * The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
    * SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
    * Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
    * SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
    * TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->

## Data Release 30.0

* __GDC Product__: Data
* __Release Date__: September 23, 2021

### New updates

1.  New Projects:
    * TRIO-CRU (phs001163) - Ukrainian National Research Center for Radiation Medicine Trio Study
        * WGS Alignments
    * REBC-THYR (phs001134) - Comprehensive genomic characterization of radiation-related papillary thyroid cancer in the Ukraine
        * miRNA-Seq
        * RNA-Seq
        * WGS
2.  CPTAC Program
    * CPTAC-3 methylation data produced from the SeSAMe pipeline is now available.
    * CPTAC-2 miRNA-Seq files have been replaced with better quality data.
3. HCMI-CMDC
    * 31 New cases have been released to the GDC Data Portal.
    * Methylation data produced from the SeSAMe pipeline is now available.
4. TCGA
    * Protein expression data (RPPA) is now available for 32 projects.
    * RNA-Seq data for TCGA-TGCT was replaced with files from an updated pipeline.
5. TARGET-AML - New RNA-Seq and miRNA-Seq aliquots have been released.

A complete list of files for this release are listed for the GDC Data Portal and the GDC Legacy Archive are found below:

* [gdc_manifest_20210923_data_release_30.0_active.tsv.gz](https://api.gdc.cancer.gov/data/74597c3a-d048-4f5a-8e41-fa95c00f667a)
* [gdc_manifest_20210923_data_release_30.0_legacy.tsv.gz](https://api.gdc.cancer.gov/data/362bbf4b-a4a8-4591-a796-cbf4930322bc)

### Bugs Fixed Since Last Release

*  None

### Known Issues and Workarounds

* One file from a previous version of the methylation pipeline appears in the data portal (bd2f864a-3f00-47b5-815d-bd01ca21ef61; CPTAC-3).  This file cannot be downloaded, but may cause bulk downloads to fail.  Remove this file from any manifest or cart you plan on downloading. <!--SV-2012-->
* The Copy Number Estimate files in GENIE are labeled on the portal as TXT while the files are actually in TSV format.  <!--DAT-2728-->
* Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release <!--DAT-1589-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
    * Tumor grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->
* TARGET projects
    * TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
        * TARGET-20-PASJGZ-04A-02D
        * TARGET-30-PAPTLY-01A-01D
        * TARGET-20-PAEIKD-09A-01D
        * TARGET-20-PASMYS-14A-02D
        * TARGET-20-PAMYAS-14A-02D
        * TARGET-10-PAPZST-09A-01D
    * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
    * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
    * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
    * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
    * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
    * Some TARGET files are not connected to all related aliquots <!--SV-929-->
    * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
    * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
        *  All microarray data and metadata
        *  All sequencing analyzed data and metadata
        *  1180 of 12063 sequencing runs of raw data
    * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
    * No data from TARGET-MDLS is available.
* Issues in the Legacy Archive
    * The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
    * SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
    * Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
    * SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
    * TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->


## Data Release 29.0

* __GDC Product__: Data
* __Release Date__: March 31, 2021

### New updates

1.  Count Me In Program
    * Aliquot-level MAFs are now available for projects CMI-ASC, CMI-MBC, and CMI-MPC.
    * Somatic mutation are now explorable for projects CMI-ASC, CMI-MBC, and CMI-MPC
2.  CPTAC Program
    * CPTAC-2 open-access somatic mutations are now browsable through the GDC Exploration Portal.
    * MSI data is now browsable through the faceted search for CPTAC-2 and CPTAC-3.
3. HCMI-CMDC - Data files and explorable mutations for 18 new cases are now available.

A complete list of files for this release are listed for the GDC Data Portal and the GDC Legacy Archive are found below:

* [gdc_manifest_20210331_data_release_29.0_active.tsv.gz](https://api.gdc.cancer.gov/data/d14fe1c9-a30d-40f6-87c5-71e678e57a57)
* [gdc_manifest_20210331_data_release_29.0_legacy.tsv.gz](https://api.gdc.cancer.gov/data/501cb79d-ea36-48c8-a790-ebb5c6f73fcb)

### Bugs Fixed Since Last Release

*  The aggregated and masked MAF files that were missing for seven pancreatic cases in CPTAC-3 have been restored to the data portal.
* The missing RNA-Seq data files for the seven normal pancreatic cases in CPTAC-3 have been restored to the data portal.

### Known Issues and Workarounds

* The Copy Number Estimate files in GENIE are labeled on the portal as TXT while the files are actually in TSV format.  <!--DAT-2728-->
* Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release <!--DAT-1589-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
    * Tumor grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->
* TARGET projects
    * TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
        * TARGET-20-PASJGZ-04A-02D
        * TARGET-30-PAPTLY-01A-01D
        * TARGET-20-PAEIKD-09A-01D
        * TARGET-20-PASMYS-14A-02D
        * TARGET-20-PAMYAS-14A-02D
        * TARGET-10-PAPZST-09A-01D
    * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
    * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
    * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
    * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
    * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
    * Some TARGET files are not connected to all related aliquots <!--SV-929-->
    * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
    * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
        *  All microarray data and metadata
        *  All sequencing analyzed data and metadata
        *  1180 of 12063 sequencing runs of raw data
    * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
    * No data from TARGET-MDLS is available.
* Issues in the Legacy Archive
    * The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
    * SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
    * Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
    * SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
    * TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->

    ## Data Release 28.0

    * __GDC Product__: Data
    * __Release Date__: February 2, 2021

    ### New updates

    1.  New Project: CMI-MPC - Count Me In - The Metastatic Prostate Cancer Project
        * WXS alignments and variant calls (VCFs) are available.
    2.  New Data Type: Single nuclei (snRNA-Seq) data is now available for 18 CPTAC-3 cases. See the [RNA-Seq](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/#scrna-seq-pipeline) documentation for details.
    3.  CPTAC-3
        * Data files for 147 new cases from the pancreatic cohort are now available.
        * CPTAC-3 open-access somatic mutations are now browsable through the GDC Exploration Portal.
        * RNA-Seq transcript fusion files are now available.
        * Targeted Sequencing alignments and raw tumor-only variant calls (VCF) are now available.
    4. HCMI-CMDC
        * Data files for 22 new cases are now available.
        * The HCMI-CMDC open-access somatic mutations have been refreshed on the GDC Exploration Portal to reflect all newly released cases.

    A complete list of files for this release are listed for the GDC Data Portal and the GDC Legacy Archive are found below:

    * [gdc_manifest_20210202_data_release_28.0_active.tsv.gz](https://api.gdc.cancer.gov/data/f554526d-6a0d-448e-ad56-1f757d651820)
    * [gdc_manifest_20210202_data_release_28.0_legacy.tsv.gz](https://api.gdc.cancer.gov/data/fca3d58a-09d7-4f91-a2d0-9ce98e2f32ea)

    ### Bugs Fixed Since Last Release

    *  None

    ### Known Issues and Workarounds

    * The aggregated and masked MAF files for seven pancreatic cases in CPTAC-3 do not appear in the Data Portal. See below for download instructions.
        - [This manifest](https://api.gdc.cancer.gov/data/1a229f9a-1492-43c7-9cf4-20e52b72626e) can be used to download the files.  
        - To download the raw aggregated MAF files, dbGaP access to CPTAC-3 (phs001287) is required.  The masked MAF files are open-access.
        - The seven cases are as follows: C3L-04027, C3L-04080, C3N-02585, C3N-02768, C3N-02971, C3N-03754, and C3N-03839. The case the each file is associated with is denoted in the manifest.
    * The RNA-Seq data files for the seven normal pancreatic cases in CPTAC-3 do not appear in the Data Portal. See below for download instructions.
        - [This manifest](https://api.gdc.cancer.gov/data/322db59a-50b2-4903-8a94-6b7280ea6954) can be used to download the files.  
        - To download the alignments or splice-junction files, dbGaP access to CPTAC-3 (phs001287) is required.  The other gene expression files are open-access.
        - The seven cases are as follows: C3L-03513, C3L-07032, C3L-07033, C3L-07034, C3L-07035, C3L-07036, C3L-07037. The case the each file is associated with is denoted in the manifest.
    * The Copy Number Estimate files in GENIE are labeled on the portal as TXT while the files are actually in TSV format.  <!--DAT-2728-->
    * Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
    * The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
    * Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release <!--DAT-1589-->
    * Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
    * Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
    * Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
    * BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
    * Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
    * TCGA Projects
        * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
        * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
        * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
        * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
        * Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
        * Tumor grade property is not populated <!--SV-585-->
        * Progression_or_recurrence property is not populated <!--SV-584-->
    * TARGET projects
        * TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
            * TARGET-20-PASJGZ-04A-02D
            * TARGET-30-PAPTLY-01A-01D
            * TARGET-20-PAEIKD-09A-01D
            * TARGET-20-PASMYS-14A-02D
            * TARGET-20-PAMYAS-14A-02D
            * TARGET-10-PAPZST-09A-01D
        * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
        * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
        * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
        * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
        * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
        * Some TARGET files are not connected to all related aliquots <!--SV-929-->
        * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
        * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
            *  All microarray data and metadata
            *  All sequencing analyzed data and metadata
            *  1180 of 12063 sequencing runs of raw data
        * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
        * No data from TARGET-MDLS is available.
    * Issues in the Legacy Archive
        * The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
        * SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
        * Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
        * SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
        * TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->



## Data Release 27.0 Bug Fix

* __GDC Product__: Data
* __Release Date__: November 9, 2020

### New updates

1.  None, see bug fix section below.

A complete list of files for this release are listed for the GDC Data Portal and the GDC Legacy Archive are found below:

* [gdc_manifest_20201109_data_release_27.0_active.tsv.gz](https://api.gdc.cancer.gov/data/6bbf882a-a24f-420f-8f6b-d526c47af488)
* [gdc_manifest_20201109_data_release_27.0_legacy.tsv.gz](https://api.gdc.cancer.gov/data/765000d4-3592-4b4d-8f1e-422a46dcd448)

### Bugs Fixed Since Last Release

*  Some files in projects CGCI-BLGSP, CGCI-HTMCP-CC, and HCMI-CMDC were marked on the portal as controlled-access, when they were supposed to be open-access. These are now downloadable as open-access files.

### Known Issues and Workarounds

* The Copy Number Estimate files in GENIE are labeled on the portal as TXT while the files are actually in TSV format.  <!--DAT-2728-->
* Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release <!--DAT-1589-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
    * Tumor grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->
* TARGET projects
    * TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
        * TARGET-20-PASJGZ-04A-02D
        * TARGET-30-PAPTLY-01A-01D
        * TARGET-20-PAEIKD-09A-01D
        * TARGET-20-PASMYS-14A-02D
        * TARGET-20-PAMYAS-14A-02D
        * TARGET-10-PAPZST-09A-01D
    * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
    * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
    * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
    * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
    * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
    * Some TARGET files are not connected to all related aliquots <!--SV-929-->
    * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
    * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
        *  All microarray data and metadata
        *  All sequencing analyzed data and metadata
        *  1180 of 12063 sequencing runs of raw data
    * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
    * No data from TARGET-MDLS is available.
* Issues in the Legacy Archive
    * The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
    * SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
    * Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
    * SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
    * TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->

## Data Release 27.0 <!--REQ-413-->

* __GDC Product__: Data
* __Release Date__: October 29, 2020

### New updates

1.  Initial release for the WGS variant calling pipeline. See the [documentation](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/#whole-genome-sequencing-variant-calling) on WGS variant calling for more details on the available files.  This includes data from the following projects:
    * CGCI-BLGSP <!--SPT7-75-->
    * CGCI-HTMCP-CC <!--SPT7-27-->
    * HCMI-CMDC <!--DAT-2948-->
2. RNA-Seq transcript fusion files are available for the following projects: <!--DAT-2947-->
    * CGCI-BLGSP
    * CGCI-HTMCP-CC
    * HCMI-CMDC
3. Aliquot level MAFs were released for CGCI-HTMCP-CC Targeted Sequencing variants. Open access MAFs are included. <!--SPT7-27-->
4. 17 new cases were released for the HCMI-CMDC project.  This includes WGS, WXS, and RNA-Seq data. <!--DAT-2949-->
5.  WGS alignments were released for 99 TCGA-LUAD cases (196 files). <!--DAT-1831-->
6. Therapeutic agents (treatment) and tumor stage (diagnosis) properties were migrated to remove deprecated values and better adhere to a standardized set of values. <!--DAT-2831--> <!--DAT-2880-->

A complete list of files for DR27.0 are listed for the GDC Data Portal and the GDC Legacy Archive are found below:

* [gdc_manifest_20201029_data_release_27.0_active.tsv.gz](https://api.gdc.cancer.gov/data/6bbf882a-a24f-420f-8f6b-d526c47af488)
* [gdc_manifest_20201029_data_release_27.0_legacy.tsv.gz](https://api.gdc.cancer.gov/data/765000d4-3592-4b4d-8f1e-422a46dcd448)

### Bugs Fixed Since Last Release

*  None

### Known Issues and Workarounds

* Some files in projects CGCI-BLGSP, CGCI-HTMCP-CC, and HCMI-CMDC are marked on the portal as controlled-access. These files are publicly downloadable using the Data Transfer Tool or API. All files from the following data types should be open-access within the previously specified projects: Biospecimen Supplement, Clinical Supplement, Gene Expression Quantification, Masked Somatic Mutation
* The Copy Number Estimate files in GENIE are labeled on the portal as TXT while the files are actually in TSV format.  <!--DAT-2728-->
* Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release <!--DAT-1589-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
    * Tumor grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->
* TARGET projects
    * TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
        * TARGET-20-PASJGZ-04A-02D
        * TARGET-30-PAPTLY-01A-01D
        * TARGET-20-PAEIKD-09A-01D
        * TARGET-20-PASMYS-14A-02D
        * TARGET-20-PAMYAS-14A-02D
        * TARGET-10-PAPZST-09A-01D
    * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
    * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
    * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
    * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
    * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
    * Some TARGET files are not connected to all related aliquots <!--SV-929-->
    * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
    * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
        *  All microarray data and metadata
        *  All sequencing analyzed data and metadata
        *  1180 of 12063 sequencing runs of raw data
    * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
    * No data from TARGET-MDLS is available.
* Issues in the Legacy Archive
    * The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
    * SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
    * Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
    * SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
    * TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->

## Data Release 26.0 <!--REQ-409-->

* __GDC Product__: Data
* __Release Date__: September 8, 2020

### New updates

1.  New program released:
    * Count Me In (CMI)
        * CMI-ASC - The Angiosarcoma Project
            * RNA-Seq
            * WXS
        * CMI-MBC - The Metastatic Breast Cancer Project
            * RNA-Seq
            * WXS
2. Somatic mutations are now available on the exploration portal for the following projects:
    * MMRF-COMMPASS
    * TARGET-ALL-P3
    * TARGET-AML
    * TARGET-NBL
    * TARGET-WT
3. Primary sites and disease types were updated for multiple projects to correspond to GDC Dictionary updates.

A complete list of files for DR26.0 are listed for the GDC Data Portal and the GDC Legacy Archive are found below:

* [gdc_manifest_20200908_data_release_26.0_active.tsv.gz](https://api.gdc.cancer.gov/data/9d66bc89-7f99-480e-9a6a-e4114d4c58a3)
* [gdc_manifest_20200908_data_release_26.0_legacy.tsv.gz](https://api.gdc.cancer.gov/data/8f44d8bb-09e0-43d5-bf12-6133e27c2e86)

### Bugs Fixed Since Last Release

*  The CPTAC-3 head and neck cohort can now be queried by choosing the head and neck anatomic site on the GDC home page.

### Known Issues and Workarounds

* The Copy Number Estimate files in GENIE are labeled on the portal as TXT while the files are actually in TSV format.  <!--DAT-2728-->
* Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release <!--DAT-1589-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
    * Tumor grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->
* TARGET projects
    * TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
        * TARGET-20-PASJGZ-04A-02D
        * TARGET-30-PAPTLY-01A-01D
        * TARGET-20-PAEIKD-09A-01D
        * TARGET-20-PASMYS-14A-02D
        * TARGET-20-PAMYAS-14A-02D
        * TARGET-10-PAPZST-09A-01D
    * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
    * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
    * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
    * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
    * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
    * Some TARGET files are not connected to all related aliquots <!--SV-929-->
    * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
    * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
        *  All microarray data and metadata
        *  All sequencing analyzed data and metadata
        *  1180 of 12063 sequencing runs of raw data
    * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
    * No data from TARGET-MDLS is available.
* Issues in the Legacy Archive
    * The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
    * SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
    * Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
    * SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
    * TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->

    ## Data Release 25.0 <!--REQ-403-->

    * __GDC Product__: Data
    * __Release Date__: July 22, 2020

    ### New updates

    1.  New data types released:
        * RNA-Seq Transcript Fusion files were released for the following projects:
            * TARGET-ALL-P1
            * TARGET-ALL-P2
            * TARGET-ALL-P3
            * TARGET-CCSK
            * TARGET-NBL
            * TARGET-OS
            * TARGET-RT
            * TARGET-WT
        * The msi_status and msi_score properties can be queried on the GDC Portal for the CPTAC-3 project.  
            * To query for these fields: go to the [GDC Repository](https://portal.gdc.cancer.gov/repository), click on "Add a File Filter" at the top left of the screen, type msi_score or msi_status in the field, and click on "msi_score" or "msi_status".  This should bring up the corresponding filters to use on the portal.
    2. 108 cases from the CPTAC-3 LSCC Cohort were released. Includes the following data types:
        * WXS
        * WGS
        * RNA-Seq
        * miRNA-Seq
    3. Aliquot level MAFs were released for MMRF-COMMPASS WXS variants. Open access MAFs are included.
    4. HCMI-CMDC open-access somatic mutations were released to the [Exploration Portal](https://portal.gdc.cancer.gov/exploration).  

    A complete list of files for DR25.0 are listed for the GDC Data Portal and the GDC Legacy Archive are found below:

    * [gdc_manifest_20200722_data_release_25.0_active.tsv.gz](https://api.gdc.cancer.gov/data/515b9ae2-3f52-4b92-9475-ff59a4b846b5)
    * [gdc_manifest_20200722_data_release_25.0_legacy.tsv.gz](https://api.gdc.cancer.gov/fef95b15-7541-438c-a887-126e4e04dc94)

    ### Bugs Fixed Since Last Release

    *  A few supplements from CGCI-BLGSP are now associated with their correct versions. <!--SV-1709-->

    ### Known Issues and Workarounds

    * Currently the CPTAC-3 HNSCC cohort does not appear when the "Head and Neck" primary site is selected from the GDC home page. This cohort can be queried by clicking [here](https://portal.gdc.cancer.gov/repository?facetTab=cases&filters=%7B%22op%22%3A%22and%22%2C%22content%22%3A%5B%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22cases.primary_site%22%2C%22value%22%3A%5B%22other%20and%20ill-defined%20sites%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22cases.project.project_id%22%2C%22value%22%3A%5B%22CPTAC-3%22%5D%7D%7D%5D%7D)
    * The Copy Number Estimate files in GENIE are labeled on the portal as TXT while the files are actually in TSV format.  <!--DAT-2728-->
    * Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
    * The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
    * Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release <!--DAT-1589-->
    * Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
    * Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
    * Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
    * BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
    * Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
    * TCGA Projects
        * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
        * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
        * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
        * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
        * Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
        * Tumor grade property is not populated <!--SV-585-->
        * Progression_or_recurrence property is not populated <!--SV-584-->
    * TARGET projects
        * TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
            * TARGET-20-PASJGZ-04A-02D
            * TARGET-30-PAPTLY-01A-01D
            * TARGET-20-PAEIKD-09A-01D
            * TARGET-20-PASMYS-14A-02D
            * TARGET-20-PAMYAS-14A-02D
            * TARGET-10-PAPZST-09A-01D
        * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
        * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
        * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
        * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
        * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
        * Some TARGET files are not connected to all related aliquots <!--SV-929-->
        * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
        * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
            *  All microarray data and metadata
            *  All sequencing analyzed data and metadata
            *  1180 of 12063 sequencing runs of raw data
        * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
        * No data from TARGET-MDLS is available.
    * Issues in the Legacy Archive
        * The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
        * SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
        * Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
        * SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
        * TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->


## Data Release 24.0 <!--REQ-401-->

* __GDC Product__: Data
* __Release Date__: May 7, 2020

### New updates

1.  New project released: CGCI-HTMCP-CC - HIV+ Tumor Molecular Characterization Project - Cervical Cancer
    * RNA-Seq: Alignments and gene expression levels
    * miRNA-Seq: Alignments and miRNA expression levels
    * WGS: Alignments
    * Targeted Sequencing: Alignments

2. 110 new cases were released from the HNSCC cohort of CPTAC-3.  This includes WXS, WGS, RNA-Seq and miRNA-Seq data.

3. Aliquot-level WXS MAFs are now available from the following projects:
    * CPTAC-2
    * CPTAC-3

A complete list of files for DR24.0 are listed for the GDC Data Portal and the GDC Legacy Archive are found below:

* [gdc_manifest_20200507_data_release_24.0_active.tsv.gz](https://api.gdc.cancer.gov/data/8bc60b7b-8b13-4622-af1a-c684b9fa74c7)
* [gdc_manifest_20200507_data_release_24.0_legacy.tsv.gz](https://api.gdc.cancer.gov/data/a3966236-f6ba-4cba-8324-22b5f16963ee)


### Bugs Fixed Since Last Release

*  None

### Known Issues and Workarounds

* Currently the CPTAC-3 HNSCC cohort does not appear when the "Head and Neck" primary site is selected from the GDC home page. This cohort can be queried by clicking [here](https://portal.gdc.cancer.gov/repository?facetTab=cases&filters=%7B%22op%22%3A%22and%22%2C%22content%22%3A%5B%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22cases.primary_site%22%2C%22value%22%3A%5B%22other%20and%20ill-defined%20sites%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22cases.project.project_id%22%2C%22value%22%3A%5B%22CPTAC-3%22%5D%7D%7D%5D%7D)
* The Copy Number Estimate files in GENIE are labeled on the portal as TXT while the files are actually in TSV format.  <!--DAT-2728-->
* Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release <!--DAT-1589-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
    * Tumor grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->
* TARGET projects
    * TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
        * TARGET-20-PASJGZ-04A-02D
        * TARGET-30-PAPTLY-01A-01D
        * TARGET-20-PAEIKD-09A-01D
        * TARGET-20-PASMYS-14A-02D
        * TARGET-20-PAMYAS-14A-02D
        * TARGET-10-PAPZST-09A-01D
    * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
    * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
    * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
    * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
    * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
    * Some TARGET files are not connected to all related aliquots <!--SV-929-->
    * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
    * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
        *  All microarray data and metadata
        *  All sequencing analyzed data and metadata
        *  1180 of 12063 sequencing runs of raw data
    * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
    * No data from TARGET-MDLS is available.
* Issues in the Legacy Archive
    * The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
    * SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
    * Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
    * SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
    * TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->

## Data Release 23.0 <!--REQ-397-->

* __GDC Product__: Data
* __Release Date__: April 7, 2020

### New updates

1.  New data types released:
    * Aliquot-level MAFs: MAF Files with mutations derived from one tumor/normal pair <!--DAT-2732-->
        * HCMI-CMDC
        * TARGET-ALL-P2
        * TARGET-ALL-P3
        * TARGET-AML
        * TARGET-NBL
        * TARGET-OS
        * TARGET-WT
        * Note: Previously released TARGET project level MAFs can be downloaded with the following manifest: [TARGET_Project-Level-MAF_GDC-Manifest.txt](https://api.gdc.cancer.gov/data/64be0d29-ba73-45c3-b4ec-646a5aa63980)
    * Copy number segment and estimate files from SNP6 ASCAT <!--BINF-293-->
        * All TCGA Projects
        * TARGET-ALL-P2
        * TARGET-AML

2.  To accommodate users who prefer to use project-level MAFs, a MAF aggregation tool was developed by the GDC: <!--DEV-89-->
    * [Github Release](https://github.com/NCI-GDC/gdc-maf-tool/releases)

3.  New RNA-Seq data was released from HCMI-CMDC for nine additional cases.  

4.  Clinical updates were performed for the following projects
    * CGCI-BLGSP
    * HCMI-CMDC <!--SV-1660-->
    * WCDT-MCRPC <!--DAT-2782-->

A complete list of files for DR23.0 are listed for the GDC Data Portal and the GDC Legacy Archive are found below:

* [gdc_manifest_20200407_data_release_23.0_active.tsv.gz](https://api.gdc.cancer.gov/data/e04d9cbf-3759-4af5-805b-a3ebbad0d7e3)
* [gdc_manifest_20200407_data_release_23.0_legacy.tsv.gz](https://api.gdc.cancer.gov/data/328e7829-1cd0-474a-8ff3-a68a2aa91f28)


### Bugs Fixed Since Last Release

*  The 6 HCMI-CMDC cases without clinical data now have clinical data. <!--SV-1660-->
*  Most of the "associated_entities" fields in CGCI-BLGSP were not populated correct, this has been resolved.  <!--SV-1701-->

### Known Issues and Workarounds

* The Copy Number Estimate files in GENIE are labeled on the portal as TXT while the files are actually in TSV format.  <!--DAT-2728-->
* Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release <!--DAT-1589-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
    * Tumor grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->
* TARGET projects
    * TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
        * TARGET-20-PASJGZ-04A-02D
        * TARGET-30-PAPTLY-01A-01D
        * TARGET-20-PAEIKD-09A-01D
        * TARGET-20-PASMYS-14A-02D
        * TARGET-20-PAMYAS-14A-02D
        * TARGET-10-PAPZST-09A-01D
    * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
    * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
    * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
    * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
    * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
    * Some TARGET files are not connected to all related aliquots <!--SV-929-->
    * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
    * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
        *  All microarray data and metadata
        *  All sequencing analyzed data and metadata
        *  1180 of 12063 sequencing runs of raw data
    * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
    * No data from TARGET-MDLS is available.
* Issues in the Legacy Archive
    * The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
    * SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
    * Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
    * SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
    * TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->

## Data Release 22.0 <!--REQ-396-->

* __GDC Product__: Data
* __Release Date__: January 16, 2020

### New updates

1.  New projects released:
    * WCDT-MCRPC - Genomic Characterization of Metastatic Castration Resistant Prostate Cancer (phs001648)
        * RNA-Seq; WGS Data <!--SPT7-4--> <!--SPT7-38-->
2. New data from HCMI-CMDC <!--DAT-2758--> <!--SPT7-33-->
    * 16 New Cases
    * Includes WXS, WGS, and RNA-Seq data
3. New data from CPTAC-3 <!--DAT-2723-->
    * 108 New Cases
    * Includes WXS, WGS, and RNA-Seq data
    * miRNA-Seq data for currently released cases

A complete list of files for DR22.0 are listed for the GDC Data Portal and the GDC Legacy Archive are found below:

* [gdc_manifest_20200116_data_release_22.0_active.tsv.gz](https://api.gdc.cancer.gov/data/7a4c2f58-0936-44d3-848e-39ffa59a3683)
* [gdc_manifest_20200116_data_release_22.0_legacy.tsv.gz](https://api.gdc.cancer.gov/data/896aa2d9-e7c5-42d1-9943-c5bd285f5d03)


### Bugs Fixed Since Last Release

*  None

### Known Issues and Workarounds

* The Copy Number Estimate files in GENIE are labeled on the portal as TXT while the files are actually in TSV format.  <!--DAT-2728-->
* 6 of the HCMI-CMDC cases are missing clinical nodes <!--SV-1660-->
    * HCM-CSHL-0060-C18
    * HCM-CSHL-0089-C25
    * HCM-CSHL-0090-C25
    * HCM-CSHL-0092-C25
    * HCM-CSHL-0091-C25
    * HCM-CSHL-0057-C18
* Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release <!--DAT-1589-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
    * Tumor grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->
* TARGET projects
    * TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
        * TARGET-20-PASJGZ-04A-02D
        * TARGET-30-PAPTLY-01A-01D
        * TARGET-20-PAEIKD-09A-01D
        * TARGET-20-PASMYS-14A-02D
        * TARGET-20-PAMYAS-14A-02D
        * TARGET-10-PAPZST-09A-01D
    * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
    * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
    * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
    * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
    * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
    * Some TARGET files are not connected to all related aliquots <!--SV-929-->
    * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
    * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
        *  All microarray data and metadata
        *  All sequencing analyzed data and metadata
        *  1180 of 12063 sequencing runs of raw data
    * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
    * No data from TARGET-MDLS is available.
* Issues in the Legacy Archive
    * The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
    * SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
    * Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
    * SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
    * TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->

## Data Release 21.0 <!--REQ-396-->

* __GDC Product__: Data
* __Release Date__: December 10, 2019

### New updates

1.  New projects released:
    *  GENIE - AACR Project Genomics Evidence Neoplasia Information Exchange (phs001337) <!--SPT-173-->
        *  Includes Targeted Sequencing, Transcript Fusion, Copy Number Estimate from GENIE 5.0
    *  AACR Project GENIE is divided by sequencing center:
        * GENIE-MSK
	    * GENIE-DFCI
	    * GENIE-MDA
	    * GENIE-JHU
	    * GENIE-UHN
	    * GENIE-VICC
	    * GENIE-GRCC
	    * GENIE-NKI

A complete list of files for DR21.0 are listed for the GDC Data Portal and the GDC Legacy Archive are found below:

* [gdc_manifest_20191210_data_release_21.0_active.txt.gz](https://api.gdc.cancer.gov/data/1b08d5fd-7510-4efb-a6ff-4f564a671ffd)
* [gdc_manifest_20191210_data_release_21.0_legacy.txt.gz](https://api.gdc.cancer.gov/data/482409bc-f1a4-4d10-8456-6b154266335b)


### Bugs Fixed Since Last Release

*  None

### Known Issues and Workarounds

* The Copy Number Estimate files in GENIE are labeled on the portal as TXT while the files are actually in TSV format.  <!--DAT-2728-->
* Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release <!--DAT-1589-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
    * Tumor grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->
* TARGET projects
    * TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
        * TARGET-20-PASJGZ-04A-02D
        * TARGET-30-PAPTLY-01A-01D
        * TARGET-20-PAEIKD-09A-01D
        * TARGET-20-PASMYS-14A-02D
        * TARGET-20-PAMYAS-14A-02D
        * TARGET-10-PAPZST-09A-01D
    * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
    * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
    * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
    * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
    * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
    * Some TARGET files are not connected to all related aliquots <!--SV-929-->
    * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
    * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
        *  All microarray data and metadata
        *  All sequencing analyzed data and metadata
        *  1180 of 12063 sequencing runs of raw data
    * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
    * No data from TARGET-MDLS is available.
* Issues in the Legacy Archive
    * The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
    * SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
    * Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
    * SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
    * TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->

## Data Release 20.0 <!--REQ-392-->

* __GDC Product__: Data
* __Release Date__: November 11, 2019

### New updates

1.  New projects released:
    *  CPTAC-2 - CPTAC Proteogenomic Confirmatory Study (phs000892) <!--SPT-825-->
        *  Includes WXS, RNA-Seq, and miRNA-Seq
    *  OHSU-CNL - Genomic landscape of Neutrophilic Leukemias of Ambiguous Diagnosis (phs001799) <!--SPT-755-->
        *  Includes WXS and RNA-Seq
        *  No VCF files will be included at this time.  They will follow in a later release.
2.  New TARGET data released
    *  TARGET-OS: WGS, WXS <!--SPT-394, SPT-226-->
    *  TARGET-NBL: WGS <!--SPT-412-->
    *  TARGET-AML: miRNA <!--SV-1518-->
3.  CGCI-BLGSP miRNA-Seq released <!-- SPT-441-->


A complete list of files for DR20.0 are listed for the GDC Data Portal and the GDC Legacy Archive are found below:


* [gdc_manifest_20191111_data_release_20.0_active.txt.gz](https://api.gdc.cancer.gov/data/c802c963-e974-422e-88bc-793366026d6a)
* [gdc_manifest_20191111_data_release_20.0_legacy.txt.gz](https://api.gdc.cancer.gov/data/402be47c-a39a-4f5b-9845-102c58a3335e)


### Bugs Fixed Since Last Release

*  None

### Known Issues and Workarounds

* Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release <!--DAT-1589-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
    * Tumor grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->
* TARGET projects
    * TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
        * TARGET-20-PASJGZ-04A-02D
        * TARGET-30-PAPTLY-01A-01D
        * TARGET-20-PAEIKD-09A-01D
        * TARGET-20-PASMYS-14A-02D
        * TARGET-20-PAMYAS-14A-02D
        * TARGET-10-PAPZST-09A-01D
    * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
    * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
    * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
    * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
    * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
    * Some TARGET files are not connected to all related aliquots <!--SV-929-->
    * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
    * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
        *  All microarray data and metadata
        *  All sequencing analyzed data and metadata
        *  1180 of 12063 sequencing runs of raw data
    * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
    * No data from TARGET-MDLS is available.
* Issues in the Legacy Archive
    * The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
    * SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
    * Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
    * SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
    * TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->




## Data Release 19.1

* __GDC Product__: Data
* __Release Date__: November 6, 2019

### New updates

* The following cases are no longer available in the GDC Data Portal.  They had no data files associated with them in DR 19 so there are no changes in file availability in this release.
    * TARGET-00-NAAENF
    * TARGET-00-NAAENG
    * TARGET-00-NAAENH
    * TARGET-00-NAAENI
    * TARGET-00-NAAENJ
    * TARGET-00-NAAENK
    * TARGET-00-NAAENL
    * TARGET-00-NAAENM
    * TARGET-00-NAAENN
    * TARGET-00-NAAENP
    * TARGET-00-NAAENR
    * TARGET-00-NAAEPE

A complete list of files for DR19.1 are listed for the GDC Data Portal and the GDC Legacy Archive are found below:

* [gdc_manifest_20190917_data_release_19.0_active.txt.gz](https://api.gdc.cancer.gov/data/04800e48-6404-4d44-ad64-5b58aa2b78a0)
* [gdc_manifest_20190917_data_release_19.0_legacy.txt.gz](https://api.gdc.cancer.gov/data/a0a68dca-fc58-4d1c-aaad-1357807dcbd7)


### Bugs Fixed Since Last Release

*  None

### Known Issues and Workarounds

* Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release <!--DAT-1589-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
    * Tumor grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->
* TARGET projects
    * TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
        * TARGET-20-PASJGZ-04A-02D
        * TARGET-30-PAPTLY-01A-01D
        * TARGET-20-PAEIKD-09A-01D
        * TARGET-20-PASMYS-14A-02D
        * TARGET-20-PAMYAS-14A-02D
        * TARGET-10-PAPZST-09A-01D
    * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
    * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
    * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
    * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
    * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
    * Some TARGET files are not connected to all related aliquots <!--SV-929-->
    * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
    * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
        *  All microarray data and metadata
        *  All sequencing analyzed data and metadata
        *  1180 of 12063 sequencing runs of raw data
    * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
    * No data from TARGET-MDLS is available.
* Issues in the Legacy Archive
    * The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
    * SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
    * Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
    * SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
    * TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->


## Data Release 19.0 <!--REQ-389-->

* __GDC Product__: Data
* __Release Date__: September 17, 2019

### New updates

1.  New projects released:
    *  BEATAML1.0-COHORT - Functional Genomic Landscape of Acute Myeloid Leukemia (phs001657)
        *  Includes WXS and RNA-Seq
2.  New TARGET data released
    *  TARGET-ALL-P1 RNA-Seq
    *  TARGET-ALL-P2 RNA-Seq, WXS, and miRNA-Seq
    *  TARGET-ALL-P3 miRNA-Seq
    *  TARGET-AML WXS, WGS, and miRNA-Seq
    *  TARGET-NBL WXS and RNA-Seq
    *  TARGET-RT WGS and RNA-Seq
    *  TARGET-WT WGS, WXS, and RNA-Seq
3.  Additional CGCI-BLGSP WGS data released
4.  Pindel VCFs released for TARGET-ALL-P2, TARGET-ALL-P3, TARGET-AML, TARGET-NBL, TARGET-WT, MMRF-COMMPASS, HCMI-CMDC, and CPTAC-3
5.  Disease-specific staging properties for many projects were released <!--DAT-2480,  DAT-2503-->


A complete list of files for DR19.0 are listed for the GDC Data Portal and the GDC Legacy Archive are found below:

* [gdc_manifest_20190917_data_release_19.0_active.txt.gz](https://api.gdc.cancer.gov/data/04800e48-6404-4d44-ad64-5b58aa2b78a0)
* [gdc_manifest_20190917_data_release_19.0_legacy.txt.gz](https://api.gdc.cancer.gov/data/a0a68dca-fc58-4d1c-aaad-1357807dcbd7)


### Bugs Fixed Since Last Release

*  None

### Known Issues and Workarounds

* Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release <!--DAT-1589-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
    * Tumor grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->
* TARGET projects
    * TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
        * TARGET-20-PASJGZ-04A-02D
        * TARGET-30-PAPTLY-01A-01D
        * TARGET-20-PAEIKD-09A-01D
        * TARGET-20-PASMYS-14A-02D
        * TARGET-20-PAMYAS-14A-02D
        * TARGET-10-PAPZST-09A-01D
    * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
    * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
    * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
    * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
    * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
    * Some TARGET files are not connected to all related aliquots <!--SV-929-->
    * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
    * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
        *  All microarray data and metadata
        *  All sequencing analyzed data and metadata
        *  1180 of 12063 sequencing runs of raw data
    * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
    * No data from TARGET-MDLS is available.
* Issues in the Legacy Archive
    * The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
    * SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
    * Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
    * SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
    * TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->


## Data Release 18.0

* __GDC Product__: Data
* __Release Date__: July 8, 2019

### New updates

1.  New Projects released
    *   MMRF-COMMPASS - Multiple Myeloma CoMMpass Study (phs000748)
        *  Includes WGS, WXS, and RNA-Seq
    *   ORGANOID-PANCREATIC - Pancreas Cancer Organoid Profiling (phs001611)
        *  Includes WGS, WXS, and RNA-Seq
    *   TARGET-ALL-P1 - Acute Lymphoblastic Leukemia - Phase I (phs000218)
        *  Includes WGS
    *   TARGET-ALL-P2 - Acute Lymphoblastic Leukemia - Phase II (phs000218)
        *  Includes WGS
    *   CGCI-BLGSP - Burkitt Lymphoma Genome Sequencing Project (phs000235)
        *  Includes WGS and RNA-Seq
2.  New versions of RNA-Seq data for TARGET-ALL-P3     
3.  New RNA-Seq data for TARGET-CCSK
4.  New RNA-Seq data for TARGET-OS


A complete list of files for DR18.0 are listed for the GDC Data Portal and the GDC Legacy Archive are found below:

* [gdc_manifest_20190708_data_release_18.0_active.txt.gz](https://api.gdc.cancer.gov/data/0287c593-2450-4d80-8307-2f031c7b34d7)
* [gdc_manifest_20190708_data_release_18.0_legacy.txt.gz](https://api.gdc.cancer.gov/data/95a277cf-c0f8-4c61-82cc-8467ebb658e9)


### Bugs Fixed Since Last Release

*  New versions of RNA-Seq data for TARGET-ALL-P3 resolve issue with missing reads from BAM files.

### Known Issues and Workarounds

* Some tumor-only annotated VCFs (not raw VCFs) could have a small proportion of variants that appear twice.  Tumor-only annotated VCFs can be identified by searching for workflow "GATK4 MuTect2 Annotation" <!--SV-1425-->
* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release <!--DAT-1589-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
    * Tumor grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->
* TARGET projects
    * TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
        * TARGET-20-PASJGZ-04A-02D
        * TARGET-30-PAPTLY-01A-01D
        * TARGET-20-PAEIKD-09A-01D
        * TARGET-20-PASMYS-14A-02D
        * TARGET-20-PAMYAS-14A-02D
        * TARGET-10-PAPZST-09A-01D
    * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
    * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
    * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
    * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
    * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
    * Some TARGET files are not connected to all related aliquots <!--SV-929-->
    * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
    * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
        *  All microarray data and metadata
        *  All sequencing analyzed data and metadata
        *  1180 of 12063 sequencing runs of raw data
    * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
    * No data from TARGET-MDLS is available.
* Issues in the Legacy Archive
    * The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
    * SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
    * Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
    * SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
    * TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->


## Data Release 17.1

* __GDC Product__: Data
* __Release Date__: June 12, 2019

### New updates

1.  Rebuilt indices for NCICCR-DLBCL and CTSP-DLBCL1.  Fewer files viewable in GDC Data Portal or API.

A complete list of files for DR17.1 are listed for the GDC Data Portal and the GDC Legacy Archive are found below:

* [gdc_manifest_20190612_data_release_17.1_active.txt.gz](https://api.gdc.cancer.gov/data/7c890b5a-cc23-44b1-821d-7283b5604fdf)
* [gdc_manifest_20190612_data_release_17.1_legacy.txt.gz](https://api.gdc.cancer.gov/data/4d41c32c-90c3-4413-aa93-2dff5d8d41a8)

### Bugs Fixed Since Last Release

*  None

### Known Issues and Workarounds


* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release <!--DAT-1589-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
    * Tumor grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->
* TARGET projects
    * TARGET ALL-P3 RNA-Seq results from DR14 are missing ~18% of reads.  Downsampling appears to be completely random and count files have a very high correlation (>99.99%) with complete data. New versions of these files will be created that include the entire set of reads.
    * TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
        * TARGET-20-PASJGZ-04A-02D
        * TARGET-30-PAPTLY-01A-01D
        * TARGET-20-PAEIKD-09A-01D
        * TARGET-20-PASMYS-14A-02D
        * TARGET-20-PAMYAS-14A-02D
        * TARGET-10-PAPZST-09A-01D
    * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
    * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
    * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
    * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
    * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
    * Some TARGET files are not connected to all related aliquots <!--SV-929-->
    * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
    * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
        *  All microarray data and metadata
        *  All sequencing analyzed data and metadata
        *  1180 of 12063 sequencing runs of raw data
    * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
    * No data from TARGET-MDLS is available.
* Issues in the Legacy Archive
    * Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
    * SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
    * Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
    * SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
    * TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->



## Data Release 17.0 <!--REQ-383-->

* __GDC Product__: Data
* __Release Date__: June 5, 2019

### New updates

1.  New Projects released
    *   HCMI-CMDC - NCI Cancer Model Development for the Human Cancer Model Initiative (HCMI) (phs001486)
    *   BEATAML1.0-CRENOLANIB - Clinical Resistance to Crenolanib in Acute Myeloid Leukemia Due to Diverse Molecular Mechanisms (phs001628)
2.  RNA-Seq data for NCICCR-DLBCL and CTSP-DLBCL1 are released
3.  ATAC-Seq data for TCGA projects are released
4.  CPTAC-3 RNA-Seq data are released
5.  Clinical data updates for TCGA - to see parser code updates review [API v1.20 release notes](https://docs.gdc.cancer.gov/API/Release_Notes/API_Release_Notes/#v1200)
6.  Clinical data updates for other projects to accommodate migration of vital_status, days_to_birth, and days_to_death from the Diagnosis to the Demographic node

A complete list of files for DR17.0 are listed for the GDC Data Portal and the GDC Legacy Archive are found below:

* [gdc_manifest_20190605_data_release_17.0_active.txt.gz](https://api.gdc.cancer.gov/data/0493c694-d33a-4005-9461-a0332ce0b663)
* [gdc_manifest_20190605_data_release_17.0_legacy.txt.gz](https://api.gdc.cancer.gov/data/05681495-e346-4e51-ab16-d809bee8a0fa).

### Bugs Fixed Since Last Release

*  None

### Known Issues and Workarounds


* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release <!--DAT-1589-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* TCGA Projects
    * Incorrect information about treatment may be included for patients within TCGA-HNSC and TCGA-LGG.  Please refer to the clinical XML for accurate information on treatment <!--DAT-2264, DAT-2265-->
    * 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
    * Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
    * The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.<!--TT-602, DAT-1489-->
    * Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
    * Tumor grade property is not populated <!--SV-585-->
    * Progression_or_recurrence property is not populated <!--SV-584-->
* TARGET projects
    * TARGET ALL-P3 RNA-Seq results from DR14 are missing ~18% of reads.  Downsampling appears to be completely random and count files have a very high correlation (>99.99%) with complete data. New versions of these files will be created that include the entire set of reads.
    * TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
        * TARGET-20-PASJGZ-04A-02D
        * TARGET-30-PAPTLY-01A-01D
        * TARGET-20-PAEIKD-09A-01D
        * TARGET-20-PASMYS-14A-02D
        * TARGET-20-PAMYAS-14A-02D
        * TARGET-10-PAPZST-09A-01D
    * 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
    * There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
    * There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
    * Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
    * Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
    * Some TARGET files are not connected to all related aliquots <!--SV-929-->
    * Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
    * The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
        *  All microarray data and metadata
        *  All sequencing analyzed data and metadata
        *  1180 of 12063 sequencing runs of raw data
    * Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
    * No data from TARGET-MDLS is available.
* Issues in the Legacy Archive
    * Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
    * SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
    * Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
    * SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
    * TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->





## Data Release 16.0 <!--REQ-385-->

* __GDC Product__: Data
* __Release Date__: March 26, 2019

### New updates

1.  The CPTAC-3 project (phs001287) is released with WXS and WGS data.  RNA-Seq will be released at a later date.  Additional project details can be found at on the [CPTAC Data Source page](https://gdc.cancer.gov/about-gdc/contributed-genomic-data-cancer-research/clinical-proteomic-tumor-analysis-consortium-cptac).
2.  TARGET-ALL-P3 (phs000218) WGS BAM files are released.
3.  VAREPOP-APOLLO (phs001374) VCF files are released.

A complete list of files for DR16.0 are listed for the GDC Data Portal and the GDC Legacy Archive are found below:

* [gdc_manifest_20190326_data_release_16.0_active.txt.gz](https://api.gdc.cancer.gov/data/88cd6a25-b46d-435d-ade0-aae709f1f9e3)
* [gdc_manifest_20190326_data_release_16.0_legacy.txt.gz](https://api.gdc.cancer.gov/data/f8b6460c-eec1-4f94-8c2c-f17387549d70).

### Bugs Fixed Since Last Release

*  None

### Known Issues and Workarounds

* TARGET ALL-P3 RNA-Seq results from DR14 are missing ~18% of reads.  Downsampling appears to be completely random and count files have a very high correlation (>99.99%) with complete data. New versions of these files will be created that include the entire set of reads.
* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
    * TARGET-20-PASJGZ-04A-02D
    * TARGET-30-PAPTLY-01A-01D
    * TARGET-20-PAEIKD-09A-01D
    * TARGET-20-PASMYS-14A-02D
    * TARGET-20-PAMYAS-14A-02D
    * TARGET-10-PAPZST-09A-01D
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release <!--DAT-1589-->
* 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
* 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
* Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
* The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.
<!--TT-602, DAT-1489-->
* There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
* There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
* TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->
* Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
* Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
* Some TARGET files are not connected to all related aliquots <!--SV-929-->
* Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
    *  All microarray data and metadata
    *  All sequencing analyzed data and metadata
    *  1180 of 12063 sequencing runs of raw data
* Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
* Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* No data from TARGET-MDLS is available.
* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
* SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
* Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
* SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* Tumor grade property is not populated <!--SV-585-->
* Progression_or_recurrence property is not populated <!--SV-584-->




## Data Release 15.0 <!--REQ-381-->

* __GDC Product__: Data
* __Release Date__: February 20, 2019

### New updates

1.  TARGET-ALL-P3 is now available and includes RNA-Seq and WXS data.
2.  New RNA-Seq workflow is now being utilized for new projects.  More details can be found in the [RNA-Seq pipeline documentation](/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/#rna-seq-alignment-workflow).
3.  New tumor only variant calling pipeline is now being utilized for new projects.  More details can be found in the [Tumor only pipeline documentation](/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/#tumor-only-variant-calling-workflow).


A complete list of files for DR15.0 are listed for the GDC Data Portal and the GDC Legacy Archive are found below:

* [gdc_manifest_20190220_data_release_15.0_active.txt.gz](https://api.gdc.cancer.gov/data/5c44ddf9-2bab-4710-8012-dd677ff09547)
* [gdc_manifest_20190220_data_release_15.0_legacy.txt.gz](https://api.gdc.cancer.gov/data/46aaf192-3d31-4ced-a422-1cd8d3b5df21).

### Bugs Fixed Since Last Release

*  None

### Known Issues and Workarounds

* The read alignment end coordinates in the x.isoform.quantification.txt files produced by the miRNA pipeline are exclusive (i.e. offset by 1) for all TCGA miRNA legacy (GRCh37/hg19) and current harmonized (GRCh38/hg38) miRNA data.  This error has no impact on miRNA alignment or quantification - only the coordinates reported in the quantification file.
* TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
    * TARGET-20-PASJGZ-04A-02D
    * TARGET-30-PAPTLY-01A-01D
    * TARGET-20-PAEIKD-09A-01D
    * TARGET-20-PASMYS-14A-02D
    * TARGET-20-PAMYAS-14A-02D
    * TARGET-10-PAPZST-09A-01D
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release <!--DAT-1589-->
* 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
* 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
* Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
* The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.
<!--TT-602, DAT-1489-->
* There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
* There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
* TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->
* Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
* Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
* Some TARGET files are not connected to all related aliquots <!--SV-929-->
* Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
    *  All microarray data and metadata
    *  All sequencing analyzed data and metadata
    *  1180 of 12063 sequencing runs of raw data
* Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
* Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* No data from TARGET-MDLS is available.
* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
* SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
* Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
* SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* Tumor grade property is not populated <!--SV-585-->
* Progression_or_recurrence property is not populated <!--SV-584-->


## Data Release 14.0 <!--REQ-335-->

* __GDC Product__: Data
* __Release Date__: December 18, 2018

### New updates

1.  Copy Number Variation (CNV) data derived from GISTIC2 results are now available for download for TCGA projects <!--DAT-1681-->
2.  New miRNA data available for 181 aliquots for TARGET and TCGA <!--DAT-1823-->
3.  Released two SNP6 files (6cd4ef5e-324a-4ace-8779-7a33bd559c83, dfa89ee9-6ee5-460b-bd58-b5ca0e9cb7ac) <!--DAT-1805-->
4.  New versions of TCGA biospecimen supplements are available <!--DAT-1776-->
5.  Updated primary site for `TCGA-AG-3881` to `Unknown` <!--DAT-1853-->
6.  8 New Harmonized WGS BAM files for TARGET-WT, TARGET-NBL, TARGET-AML added to the portal <!--DAT-1839-->

A complete list of files for DR14.0 are listed for the GDC Data Portal and the GDC Legacy Archive are found below:

* [gdc_manifest_20181218_data_release_14.0_active.txt.gz](https://api.gdc.cancer.gov/data/2783d247-9784-4f53-a486-61e041520ab5)
* [gdc_manifest_20181218_data_release_14.0_legacy.txt.gz](https://api.gdc.cancer.gov/data/221b0d00-af70-4e2b-ac02-4b6c5d62d43a).


### Bugs Fixed Since Last Release

*  FM-AD clinial and biospecimen supplements are now correctly labeled as TSV rather than XLSX <!--DAT-1681-->

### Known Issues and Workarounds

* TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
    * TARGET-20-PASJGZ-04A-02D
    * TARGET-30-PAPTLY-01A-01D
    * TARGET-20-PAEIKD-09A-01D
    * TARGET-20-PASMYS-14A-02D
    * TARGET-20-PAMYAS-14A-02D
    * TARGET-10-PAPZST-09A-01D
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release <!--DAT-1589-->
* 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
* 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
* Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
* The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.
<!--TT-602, DAT-1489-->
* There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
* There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
* TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->
* Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
* Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
* Some TARGET files are not connected to all related aliquots <!--SV-929-->
* Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
    *  All microarray data and metadata
    *  All sequencing analyzed data and metadata
    *  1180 of 12063 sequencing runs of raw data
* Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
* Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* No data from TARGET-MDLS is available.
* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
* SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
* Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
* SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* Tumor grade property is not populated <!--SV-585-->
* Progression_or_recurrence property is not populated <!--SV-584-->



## Data Release 13.0

* __GDC Product__: Data
* __Release Date__: September 27, 2018

### New updates

1. Three new projects are released to the GDC (VAREPOP-APOLLO (phs001374), CTSP-DLBCL1 (phs001184), NCICCR-DLBCL (phs001444) <!--SPT-136,SPT-145,SPT-84-->
2. TARGET WGS alignments are released. VCFs will be provided in a later release <!--DAT-1699-->
3. Clinical data was harmonized with ICD-O-3 terminology for TCGA properties case.primary_site, case.disease_type, diagnosis.primary_diagnosis, diagnosis.site_of_resection_or_biopsy, diagnosis.tissue_or_organ_of_origin <!--DAT-1512-->
4. Redaction annotations applied to 11 aliquots in TCGA-DLBC <!--DAT-1169, Data-871-->
5. Redaction annotations applied to incorrectly trimmed miRNA file in the Legacy Achive <!--SV-653-->

A complete list of files for DR13.0 are listed for the GDC Data Portal and the GDC Legacy Archive are found below:

* [gdc_manifest_20180927_data_release_13.0_active.txt.gz](https://api.gdc.cancer.gov/data/d746f9ec-32d6-4d0a-aad3-a2481f5453a9)
* [gdc_manifest_20180927_data_release_13.0_legacy.txt.gz](https://api.gdc.cancer.gov/data/00c5fa11-dab8-4cdc-be3e-8337b9388295).


### Bugs Fixed Since Last Release

* 253 files Copy Number Segment and Masked Copy Number Segment files were released.  These were skipped in DR 12.0 <!--DAT-1404-->
* 36 Diagnostic TCGA slides were released.  They were skipped in DR 12.0  <!--SV-1109-->

### Known Issues and Workarounds

* 506 Copy Number Segment and 36 Slide Image files are designated as controlled-access on the GDC Data Portal.  These files are actually open-access and will be downloadable without a token using [this manifest](https://api.gdc.cancer.gov/data/d2da95c4-2576-4018-826f-9a4e1b6aabf5).  <!--DAT-1804-->
* 2 Copy Number Segment files from TCGA-TGCT do not appear on the GDC Portal. They can be downloaded using the Data Transfer Tool using the following UUIDs. <!--DAT-1805-->
    * 6cd4ef5e-324a-4ace-8779-7a33bd559c83 - RAMPS_p_TCGA_Batch_430_NSP_GenomeWideSNP_6_E07_1538238.nocnv_grch38.seg.v2.txt
    * dfa89ee9-6ee5-460b-bd58-b5ca0e9cb7ac - RAMPS_p_TCGA_Batch_430_NSP_GenomeWideSNP_6_E07_1538238.grch38.seg.v2.txt
* TARGET CGI BAMs in the Legacy Archive for the following aliquots should not be used because they were not repaired and concatenated into their original composite BAM files by CGHub.
    * TARGET-20-PASJGZ-04A-02D
    * TARGET-30-PAPTLY-01A-01D
    * TARGET-20-PAEIKD-09A-01D
    * TARGET-20-PASMYS-14A-02D
    * TARGET-20-PAMYAS-14A-02D
    * TARGET-10-PAPZST-09A-01D
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release <!--DAT-1589-->
* 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
* 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
* Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
* The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.
<!--TT-602, DAT-1489-->
* There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
* There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
* TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->
* Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
* Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
* Some TARGET files are not connected to all related aliquots <!--SV-929-->
* Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
* FM-AD clinical and biospecimen supplement files have incorrect data format.  They are listed as XLSX, but are in fact TSV files. <!--DAT-1123-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
    *  All microarray data and metadata
    *  All sequencing analyzed data and metadata
    *  1180 of 12063 sequencing runs of raw data
* Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
* Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* No data from TARGET-MDLS is available.
* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
* SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
* Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
* SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* Tumor grade property is not populated <!--SV-585-->
* Progression_or_recurrence property is not populated <!--SV-584-->




## Data Release 12.0

* __GDC Product__: Data
* __Release Date__: June 13, 2018

### New updates

1. Updated clinical and biospecimen XML files for TCGA cases are available in the GDC Data Portal. Equivalent Legacy Archive files may no longer be up to date. <!--TT-328-->
2. All biospecimen and clinical supplement files for TCGA projects formerly only found in the Legacy Archive have been updated and transferred to the GDC Data Portal. Equivalent Legacy Archive files and metadata retrieved from the API may no longer be up to date. <!--TT-327-->
3. Diagnostic slides from TCGA are now available in the GDC Data Portal and Slide Image Viewer.  They were formerly only available in the Legacy Archive. <!--DAT-1316-->
4. Updated Copy Number Segment and Masked Copy Number Segment files are now available.  These were generated using an improved mapping of hg38 coordinates for the Affymetrix SNP6.0 probe set. <!--DAT-1303-->
5.  VCF files containing SNVs produced from TARGET WGS CGI data are available.  The variant calls were initially produced by CGI and lifted over to hg38. <!--DAT-1281-->

Updated files for this release are listed [here](https://api.gdc.cancer.gov/data/780f8b48-895d-4bdd-b53a-fcead5adc3c6).
A complete list of files for DR12.0 are listed for the GDC Data Portal [here](https://api.gdc.cancer.gov/data/cb8fe4f2-8f40-4826-9db5-9a98d7a6c4b5) and the GDC Legacy Archive [here](https://api.gdc.cancer.gov/data/aa131b7d-45af-4dba-871a-9a6081bb39a3).


### Bugs Fixed Since Last Release

* TARGET NBL RNA-Seq data is now associated with the correct aliquot. <!--SV-1097-->

### Known Issues and Workarounds

* Some Copy Number Segment and Masked Copy Number Segment were not replaced in DR 12.0.  253 files remain to be swapped in a later release <!--DAT-1404-->
* Some miRNA files with QC failed reads were not swapped in DR11.0.  361 aliquots remain to be swapped in a later release <!--DAT-1589-->
* 74 Diagnostic TCGA slides are attached to a portion rather than a sample like the rest of the diagnostic slides. The reflects how these original samples were handled. <!--SV-1111-->
* 36 Diagnostic TCGA slides are not yet available in the active GDC Portal. They are still available in the GDC Legacy Archive.  <!--SV-1109-->
* 11 BAM files for TARGET-NBL RNA-Seq are not available in the GDC Data portal <!--DAT-1476-->
* Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
* The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` are not available. These VCFs files will be replaced in a later release.
<!--TT-602, DAT-1489-->
* There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
* There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
* TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->
* Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
* Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
* Some TARGET files are not connected to all related aliquots <!--SV-929-->
* Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
* FM-AD clinical and biospecimen supplement files have incorrect data format.  They are listed as XLSX, but are in fact TSV files. <!--DAT-1123-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
    *  All microarray data and metadata
    *  All sequencing analyzed data and metadata
    *  1180 of 12063 sequencing runs of raw data
* Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
* There are 11 cases in project TCGA-DLBC that are known to have incorrect WXS data in the GDC Data Portal.  Impacted cases are listed below.  This affects the BAMs and VCFs associated with these cases in the GDC Data Portal.  Corrected BAMs can be found in the GDC Legacy Archive.  Variants from affected aliquots appear in the protected MAFs with GDC_FILTER=ContEst to indicate a sample contamination problem, but are removed during the generation of the Somatic MAF file.  In a later release we will supply corrected BAM, VCF, and MAF files for these cases.  In the mean time, we advise you not to use any of the WXS files associated with these cases in the GDC Data Portal.  A list of these files can be found here. [Download list of affected files](https://api.gdc.cancer.gov/data/370678aa-432e-441b-8828-09c168cf1062). <!-- Data-871-->
    * TCGA-FF-8062
    * TCGA-FM-8000
    * TCGA-G8-6324
    * TCGA-G8-6325
    * TCGA-G8-6326
    * TCGA-G8-6906
    * TCGA-G8-6907
    * TCGA-G8-6909
    * TCGA-G8-6914
    * TCGA-GR-7351
    * TCGA-GR-7353
* Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* No data from TARGET-MDLS is available.
* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
* SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
* Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
* SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* Tumor grade property is not populated <!--SV-585-->
* Progression_or_recurrence property is not populated <!--SV-584-->




## Data Release 11.0

* __GDC Product__: Data
* __Release Date__: May 21, 2018

### New updates

1. Updated miRNA files to remove QCFail reads.  This included all BAM and downstream count files. <!--DAT-1246-->
2. TCGA Tissue slide images now available in GDC Data Portal.  Previously these were found only in the Legacy Archive <!--DAT-1251-->

Updated files for this release are listed [here](https://api.gdc.cancer.gov/data/0ca04712-221b-44d5-b1f0-d0e64e204f31).
A complete list of files for DR11.0 are listed for the GDC Data Portal [here](https://api.gdc.cancer.gov/data/a66a1432-928d-45b3-a07b-40c0ca8fa64e) and the GDC Legacy Archive [here](https://api.gdc.cancer.gov/data/8f21c0e9-65a4-4912-a528-fdcc2531788c).


### Bugs Fixed Since Last Release

* N/A

### Known Issues and Workarounds

* Two tissue slide images are unavailable for download from GDC Data Portal <!--DAT-1439-->
* RNA-Seq files for TARGET-NBL are attached to the incorrect aliquot.  The BAM files contain the correct information in their header but the connection in the GDC to read groups and aliquots is incorrect.  The linked file below contains a mapping between aliquots where file are currently associated and the aliquot where they should instead be associated [(mapping file)](https://api.gdc.cancer.gov/data/5f666089-8a49-4478-a126-5afa58575519). <!--SV-1097-->
* The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` were not replaced in DR10.0 and thus do not contain indels.  However, the indels from this aliquot can be found in the MAF files and are displayed in the Exploration section in the Data Portal.  These VCFs files will be replaced in a later release.
<!--SV-950-->
* There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
* There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
* TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->
* Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
* Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
* Some TARGET files are not connected to all related aliquots <!--SV-929-->
* miRNA alignments include QC failed reads.  
* Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
* FM-AD clinical and biospecimen supplement files have incorrect data format.  They are listed as XLSX, but are in fact TSV files. <!--DAT-1123-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
    *  All microarray data and metadata
    *  All sequencing analyzed data and metadata
    *  1180 of 12063 sequencing runs of raw data
* Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
* There are 11 cases in project TCGA-DLBC that are known to have incorrect WXS data in the GDC Data Portal.  Impacted cases are listed below.  This affects the BAMs and VCFs associated with these cases in the GDC Data Portal.  Corrected BAMs can be found in the GDC Legacy Archive.  Variants from affected aliquots appear in the protected MAFs with GDC_FILTER=ContEst to indicate a sample contamination problem, but are removed during the generation of the Somatic MAF file.  In a later release we will supply corrected BAM, VCF, and MAF files for these cases.  In the mean time, we advise you not to use any of the WXS files associated with these cases in the GDC Data Portal.  A list of these files can be found here. [Download list of affected files](https://api.gdc.cancer.gov/data/370678aa-432e-441b-8828-09c168cf1062). <!-- Data-871-->
    * TCGA-FF-8062
    * TCGA-FM-8000
    * TCGA-G8-6324
    * TCGA-G8-6325
    * TCGA-G8-6326
    * TCGA-G8-6906
    * TCGA-G8-6907
    * TCGA-G8-6909
    * TCGA-G8-6914
    * TCGA-GR-7351
    * TCGA-GR-7353
* Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* No data from TARGET-MDLS is available.
* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
* SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
* Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
* SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* Tumor grade property is not populated <!--SV-585-->
* Progression_or_recurrence property is not populated <!--SV-584-->



## Data Release 10.1

* __GDC Product__: Data
* __Release Date__: February 15, 2018

### New updates

1. Updated FM-AD clinical data to conform with Data Dictionary release v1.11

### Bugs Fixed Since Last Release

None

### Known Issues and Workarounds

* RNA-Seq files for TARGET-NBL are attached to the incorrect aliquot.  The BAM files contain the correct information in their header but the connection in the GDC to read groups and aliquots is incorrect.  The linked file below contains a mapping between aliquots where file are currently associated and the aliquot where they should instead be associated [(mapping file)](https://api.gdc.cancer.gov/data/5f666089-8a49-4478-a126-5afa58575519). <!--SV-1097-->
* The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` were not replaced in DR10.0 and thus do not contain indels.  However, the indels from this aliquot can be found in the MAF files and are displayed in the Exploration section in the Data Portal.  These VCFs files will be replaced in a later release.
<!--SV-950-->
* There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
* There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
* TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->
* Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
* Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
* Some TARGET files are not connected to all related aliquots <!--SV-929-->
* miRNA alignments include QC failed reads.  
* Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
* FM-AD clinical and biospecimen supplement files have incorrect data format.  They are listed as XLSX, but are in fact TSV files. <!--DAT-1123-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
    *  All microarray data and metadata
    *  All sequencing analyzed data and metadata
    *  1180 of 12063 sequencing runs of raw data
* Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
* There are 11 cases in project TCGA-DLBC that are known to have incorrect WXS data in the GDC Data Portal.  Impacted cases are listed below.  This affects the BAMs and VCFs associated with these cases in the GDC Data Portal.  Corrected BAMs can be found in the GDC Legacy Archive.  Variants from affected aliquots appear in the protected MAFs with GDC_FILTER=ContEst to indicate a sample contamination problem, but are removed during the generation of the Somatic MAF file.  In a later release we will supply corrected BAM, VCF, and MAF files for these cases.  In the mean time, we advise you not to use any of the WXS files associated with these cases in the GDC Data Portal.  A list of these files can be found here. [Download list of affected files](https://api.gdc.cancer.gov/data/370678aa-432e-441b-8828-09c168cf1062). <!-- Data-871-->
    * TCGA-FF-8062
    * TCGA-FM-8000
    * TCGA-G8-6324
    * TCGA-G8-6325
    * TCGA-G8-6326
    * TCGA-G8-6906
    * TCGA-G8-6907
    * TCGA-G8-6909
    * TCGA-G8-6914
    * TCGA-GR-7351
    * TCGA-GR-7353
* Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* No data from TARGET-MDLS is available.
* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
* SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
* Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
* SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* Tumor grade property is not populated <!--SV-585-->
* Progression_or_recurrence property is not populated <!--SV-584-->



## Data Release 10.0 <!--REQ-318-->

* __GDC Product__: Data
* __Release Date__: December 21, 2017

### New updates

1. New TARGET files for all projects <!--DAT-1135-->
2. TARGET updates for clinical and biospecimen data <!--DAT-1125, DAT-1153-->
3. Replace corrupted .bai files <!--TT-109-->
4. Update TCGA and TARGET MAF files to include VarScan2 indels and more information in all_effects column <!--DAT-987-->
5. Update VarScan VCF files <!--DAT-1020-->

Updated files for this release are listed [here](DR10.0_files_swap.txt.gz).
A complete list of files for DR10.0 are listed for the GDC Data Portal [here](https://api.gdc.cancer.gov/data/8483eced-0db2-4371-aa9e-d9061f5c1abf) and the GDC Legacy Archive [here](https://api.gdc.cancer.gov/data/0dd41708-d128-41bb-b78b-8c4f77d61afd).

### Bugs Fixed Since Last Release

None

### Known Issues and Workarounds

* The raw and annotated VarScan VCF files for aliquot `TCGA-VR-A8ET-01A-11D-A403-09` were not replaced in DR10.0 and thus do not contain indels.  However, the indels from this aliquot can be found in the MAF files and are displayed in the Exploration section in the Data Portal.  These VCFs files will be replaced in a later release.
<!--SV-950-->
* There are 5051 TARGET files for which `experimental_strategy`, `data_format`, `platform`, and `data_subtype` are blank <!--SV-944-->
* There are two cases with identical submitter_id `TARGET-10-PARUYU` <!--SV-940-->
* TARGET-MDLS cases do not have disease_type or primary_site populated <!--SV-939-->
* Some TARGET cases are missing `days_to_last_follow_up` <!--SV-934-->
* Some TARGET cases are missing `age_at_diagnosis` <!--SV-933-->
* Some TARGET files are not connected to all related aliquots <!--SV-929-->
* miRNA alignments include QC failed reads.  
* Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
* FM-AD clinical and biospecimen supplement files have incorrect data format.  They are listed as XLSX, but are in fact TSV files. <!--DAT-1123-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
    *  All microarray data and metadata
    *  All sequencing analyzed data and metadata
    *  1180 of 12063 sequencing runs of raw data
* Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
* There are 11 cases in project TCGA-DLBC that are known to have incorrect WXS data in the GDC Data Portal.  Impacted cases are listed below.  This affects the BAMs and VCFs associated with these cases in the GDC Data Portal.  Corrected BAMs can be found in the GDC Legacy Archive.  Variants from affected aliquots appear in the protected MAFs with GDC_FILTER=ContEst to indicate a sample contamination problem, but are removed during the generation of the Somatic MAF file.  In a later release we will supply corrected BAM, VCF, and MAF files for these cases.  In the mean time, we advise you not to use any of the WXS files associated with these cases in the GDC Data Portal.  A list of these files can be found here. [Download list of affected files](https://api.gdc.cancer.gov/data/370678aa-432e-441b-8828-09c168cf1062). <!-- Data-871-->
    * TCGA-FF-8062
    * TCGA-FM-8000
    * TCGA-G8-6324
    * TCGA-G8-6325
    * TCGA-G8-6326
    * TCGA-G8-6906
    * TCGA-G8-6907
    * TCGA-G8-6909
    * TCGA-G8-6914
    * TCGA-GR-7351
    * TCGA-GR-7353
* Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* No data from TARGET-MDLS is available.
* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
* SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
* Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
* SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* Tumor grade property is not populated <!--SV-585-->
* Progression_or_recurrence property is not populated <!--SV-584-->




## Data Release 9.0 <!--REQ-317-->

* __GDC Product__: Data
* __Release Date__: October 24, 2017

### New updates

1. Foundation Medicine Data Release
* This includes controlled-access VCF and MAF files as well as clinical and biospecimen supplements and metadata.
* Original Foundation Medicine supplied data can be found on the [Foundation Medicine Project Page](https://gdc.cancer.gov/about-gdc/contributed-genomic-data-cancer-research/foundation-medicine/foundation-medicine).
2.  Updated RNA-Seq data for TARGET NBL  
* Includes new BAM and count files

Updated files for this release are listed [here](https://api.gdc.cancer.gov/data/1acddc99-4912-4c8d-9477-585fad2d7910).
A complete list of files for DR9.0 are listed [here](https://api.gdc.cancer.gov/data/b6e22aaa-4d66-4395-8769-eb74efbcda5c).

### Bugs Fixed Since Last Release

None

### Known Issues and Workarounds

* miRNA alignments include QC failed reads.  
* Samples of TARGET sample_type `Recurrent Blood Derived Cancer - Bone Marrow` are mislabeled as `Recurrent Blood Derived Cancer - Peripheral Blood`.  A workaround is to look at the sample barcode, which is -04 for `Recurrent Blood Derived Cancer - Bone Marrow`. (e.g. `TARGET-20-PAMYAS-04A-03R`) <!--SV-918-->
* FM-AD clinical and biospecimen supplement files have incorrect data format.  They are listed as XLSX, but are in fact TSV files. <!--DAT-1123-->
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
    *  All microarray data and metadata
    *  All sequencing analyzed data and metadata
    *  1180 of 12063 sequencing runs of raw data
* Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
* There are 11 cases in project TCGA-DLBC that are known to have incorrect WXS data in the GDC Data Portal.  Impacted cases are listed below.  This affects the BAMs and VCFs associated with these cases in the GDC Data Portal.  Corrected BAMs can be found in the GDC Legacy Archive.  Variants from affected aliquots appear in the protected MAFs with GDC_FILTER=ContEst to indicate a sample contamination problem, but are removed during the generation of the Somatic MAF file.  In a later release we will supply corrected BAM, VCF, and MAF files for these cases.  In the mean time, we advise you not to use any of the WXS files associated with these cases in the GDC Data Portal.  A list of these files can be found here. [Download list of affected files](https://api.gdc.cancer.gov/data/370678aa-432e-441b-8828-09c168cf1062). <!-- Data-871-->
    * TCGA-FF-8062
    * TCGA-FM-8000
    * TCGA-G8-6324
    * TCGA-G8-6325
    * TCGA-G8-6326
    * TCGA-G8-6906
    * TCGA-G8-6907
    * TCGA-G8-6909
    * TCGA-G8-6914
    * TCGA-GR-7351
    * TCGA-GR-7353
* Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* No data from TARGET-MDLS is available.
* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
* SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
* Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
* SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* Tumor grade property is not populated <!--SV-585-->
* Progression_or_recurrence property is not populated <!--SV-584-->

## Data Release 8.0

* __GDC Product__: Data
* __Release Date__: August 22, 2017

### New updates

1. Released updated miRNA quantification files to address double counting of some normalized counts described in DR7.0 release notes. <!--DAT-1, DAT-988-->

Updated files for this release are listed [here](https://api.gdc.cancer.gov/data/8db48519-45d5-43a5-93ad-35585b38d229).
A Complete list of files for DR8.0 are listed [here](https://api.gdc.cancer.gov/data/435c3cd0-71d0-4696-8cc7-2f3f8db8f998).

### Bugs Fixed Since Last Release

None

### Known Issues and Workarounds

* TARGET-NBL RNA-Seq files were run as single ended even though they are derived from paired-end data.  These files will be rerun through the GDC RNA-Seq pipelines in a later release.  Impacted files can be found [here](https://portal.gdc.cancer.gov/repository?facetTab=files&files_offset=20&filters=~%28op~%27and~content~%28~%28op~%27in~content~%28field~%27cases.project.program.name~value~%28~%27TARGET%29%29%29~%28op~%27in~content~%28field~%27cases.project.project_id~value~%28~%27TARGET-NBL%29%29%29~%28op~%27in~content~%28field~%27files.data_format~value~%28~%27BAM%29%29%29~%28op~%27in~content~%28field~%27files.experimental_strategy~value~%28~%27RNA-Seq%29%29%29%29%29&searchTableTab=files).  Downstream count files are also affected.  Users may access original FASTQ files in the GDC Legacy Archive, which are not impacted by this issue.
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
    *  All microarray data and metadata
    *  All sequencing analyzed data and metadata
    *  1180 of 12063 sequencing runs of raw data
* Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
* There are 11 cases in project TCGA-DLBC that are known to have incorrect WXS data in the GDC Data Portal.  Impacted cases are listed below.  This affects the BAMs and VCFs associated with these cases in the GDC Data Portal.  Corrected BAMs can be found in the GDC Legacy Archive.  Variants from affected aliquots appear in the protected MAFs with GDC_FILTER=ContEst to indicate a sample contamination problem, but are removed during the generation of the Somatic MAF file.  In a later release we will supply corrected BAM, VCF, and MAF files for these cases.  In the mean time, we advise you not to use any of the WXS files associated with these cases in the GDC Data Portal.  A list of these files can be found here. [Download list of affected files](https://api.gdc.cancer.gov/data/370678aa-432e-441b-8828-09c168cf1062). <!-- Data-871-->
    * TCGA-FF-8062
    * TCGA-FM-8000
    * TCGA-G8-6324
    * TCGA-G8-6325
    * TCGA-G8-6326
    * TCGA-G8-6906
    * TCGA-G8-6907
    * TCGA-G8-6909
    * TCGA-G8-6914
    * TCGA-GR-7351
    * TCGA-GR-7353
* Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* No data from TARGET-MDLS is available.
* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
* SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
* Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
* SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* Tumor grade property is not populated <!--SV-585-->
* Progression_or_recurrence property is not populated <!--SV-584-->




## Data Release 7.0

* __GDC Product__: Data
* __Release Date__: June 29, 2017

### New updates

1.  Updated public Mutation Annotation Format (MAF) files are now available. Updates include filtering to remove variants impacted by OxoG artifacts and those impacted by strand bias. <!--AP-29-->
2.  Protected MAF files are updated to include flags for OxoG and strand bias. <!--AP-29-->
2.  Annotated VCFs are updated to include flags for OxoG artifacts and strand bias. <!--DAT-874-->

Updated files for this release are listed [here](https://api.gdc.cancer.gov/data/0e59555f-7688-4473-95f1-5ee2d369a274).
A Complete list of files for DR7.0 are listed [here](https://api.gdc.cancer.gov/data/1b7996f6-e3c6-447c-878f-a4a3ad83dce2)

### Bugs Fixed Since Last Release

None

### Known Issues and Workarounds

* TARGET-NBL RNA-Seq files were run as single ended even though they are derived from paired-end data.  These files will be rerun through the GDC RNA-Seq pipelines in a later release.  Impacted files can be found [here](https://portal.gdc.cancer.gov/repository?facetTab=files&files_offset=20&filters=~%28op~%27and~content~%28~%28op~%27in~content~%28field~%27cases.project.program.name~value~%28~%27TARGET%29%29%29~%28op~%27in~content~%28field~%27cases.project.project_id~value~%28~%27TARGET-NBL%29%29%29~%28op~%27in~content~%28field~%27files.data_format~value~%28~%27BAM%29%29%29~%28op~%27in~content~%28field~%27files.experimental_strategy~value~%28~%27RNA-Seq%29%29%29%29%29&searchTableTab=files).  Downstream count files are also affected.  Users may access original FASTQ files in the GDC Legacy Archive, which are not impacted by this issue.
* Reads that are mapped to multiple genomic locations are double counted in some of the GDC miRNA results.  The GDC will release updated files correcting the issue in an upcoming release.<!--SV-745-->  The specific impacts are described further below:
    *  Isoform Expression Quantification files
        *  Raw reads counts are accurate
        *  Normalized counts are proportionally skewed (r^2=1.0)
    *  miRNA Expression Quantification files
        *  A small proportion of miRNA counts are overestimated (mean r^2=0.9999)
        *  Normalized counts are proportionally skewed (mean r^2=0.9999)
    * miRNA BAM files
        * no impact
* Mutation frequency may be underestimated when using MAF files for genes that overlap other genes.  This is because MAF files only record one gene per variant.
* Most intronic mutations are removed for MAF generation.  However, validated variants may rescue these in some cases.  Therefore intronic mutations in MAF files are not representative of those called by mutation callers.
* The latest TARGET data is not yet available at the GDC.  For the complete and latest data, please see the [TARGET Data Matrix](https://ocg.cancer.gov/programs/target/data-matrix).  Data that is not present or is not the most up to date includes:
    *  All microarray data and metadata
    *  All sequencing analyzed data and metadata
    *  1180 of 12063 sequencing runs of raw data
* Demographic information for some TARGET patients is incorrect.  The correct information can be found in the associated clinical supplement file.  Impacted patients are TARGET-50-PAJNUS. <!--SV-710-->
* There are 11 cases in project TCGA-DLBC that are known to have incorrect WXS data in the GDC Data Portal.  Impacted cases are listed below.  This affects the BAMs and VCFs associated with these cases in the GDC Data Portal.  Corrected BAMs can be found in the GDC Legacy Archive.  Variants from affected aliquots appear in the protected MAFs with GDC_FILTER=ContEst to indicate a sample contamination problem, but are removed during the generation of the Somatic MAF file.  In a later release we will supply corrected BAM, VCF, and MAF files for these cases.  In the mean time, we advise you not to use any of the WXS files associated with these cases in the GDC Data Portal.  A list of these files can be found here. [Download list of affected files](https://api.gdc.cancer.gov/data/370678aa-432e-441b-8828-09c168cf1062). <!-- Data-871-->
    * TCGA-FF-8062
    * TCGA-FM-8000
    * TCGA-G8-6324
    * TCGA-G8-6325
    * TCGA-G8-6326
    * TCGA-G8-6906
    * TCGA-G8-6907
    * TCGA-G8-6909
    * TCGA-G8-6914
    * TCGA-GR-7351
    * TCGA-GR-7353
* Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* No data from TARGET-MLDS is available.
* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
* SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
* Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
* SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* Tumor grade property is not populated <!--SV-585-->
* Progression_or_recurrence property is not populated <!--SV-584-->




## Data Release 6.0

* __GDC Product__: Data
* __Release Date__: May 9, 2017

### New updates

1.  GDC updated public Mutation Annotation Format (MAF) files are now available. Updates include leveraging the MC3 variant filtering strategy, which results in more variants being recovered relative to the previous version. A detailed description of the new format can be found [here](/Data/File_Formats/MAF_Format/). <!--DAT-572-->
2.  Protected MAFs are updated to include additional variant annotation information <!--DAT-572-->
3.  Some MuTect2 VCFs updated to include dbSNP and COSMIC annotations found in other VCFs <!--TT-21-->

Updated files for this release are listed [here](https://api.gdc.cancer.gov/data/cf2c9b07-2683-4337-8987-9f50f12cd5d1).

### Bugs Fixed Since Last Release

None

### Known Issues and Workarounds

* There are 11 cases in project TCGA-DLBC that are known to have incorrect WXS data in the GDC Data Portal.  Impacted cases are listed below.  This affects the BAMs and VCFs associated with these cases in the GDC Data Portal.  Corrected BAMs can be found in the GDC Legacy Archive.  Variants from affected aliquots appear in the protected MAFs with GDC_FILTER=ContEst to indicate a sample contamination problem, but are removed during the generation of the Somatic MAF file.  In a later release we will supply corrected BAM, VCF, and MAF files for these cases.  In the mean time, we advise you not to use any of the WXS files associated with these cases in the GDC Data Portal.  A list of these files can be found here. [Download list of affected files](https://api.gdc.cancer.gov/data/370678aa-432e-441b-8828-09c168cf1062). <!-- Data-871-->
    * TCGA-FF-8062
    * TCGA-FM-8000
    * TCGA-G8-6324
    * TCGA-G8-6325
    * TCGA-G8-6326
    * TCGA-G8-6906
    * TCGA-G8-6907
    * TCGA-G8-6909
    * TCGA-G8-6914
    * TCGA-GR-7351
    * TCGA-GR-7353
* Variants found in VCF and MAF files may contain OxoG artifacts, which are produced during library preparation and may result in the apparent substitutions of C to A or G to T in certain sequence contexts.  In the future we will plan to label potential oxoG artifacts in the MAF files.
* Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
* Some validated somatic mutations may not be present in open-access MAF files.  Please review the protected MAF files in the GDC Data Portal if you are unable to find your mutation in the open-access files.
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* No data from TARGET-MLDS is available.
* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
* SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
* Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
* SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* Tumor grade property is not populated <!--SV-585-->
* Progression_or_recurrence property is not populated <!--SV-584-->


Details are provided in [Data Release Manifest](https://api.gdc.cancer.gov/data/bb0096b7-efbe-4335-92d1-22b0a64fc0a4)
<br>

## Data Release 5.0

* __GDC Product__: Data
* __Release Date__: March 16, 2017

### New updates

1.  Additional annotations from TCGA DCC are available <!--DAT-52-->
    * Complete list of updated TCGA files is found [here](https://api.gdc.cancer.gov/data/1d7ea03e-ed53-4fe5-90e5-0866c5ff0237)
2.  Clinical data added for TARGET ALL P1 and P2 <!--DAT-197-->
3.  Pathology reports now have submitter IDs as assigned by the BCR <!--DAT-81-->
4.  TARGET Data refresh
    * Most recent biospecimen and clinical information from the TARGET DCC. New imported files are listed [here](https://api.gdc.cancer.gov/data/8523d953-9fcb-41c5-ab83-7ca9eacdfba9)
    * Updated indexed biospecimen and clinical metadata
    * Updated SRA XMLs files
    * Does not include updates to TARGET NBL <!--SV-585-->

### Bugs Fixed Since Last Release

1. Missing cases from TCGA-LAML were added to Legacy Archive <!--DAT-272,DAT-324-->
2. Biotab files are now linked to Projects and Cases in Legacy Archive <!--SV-303,DAT-28-->

### Known Issues and Workarounds

* Some TCGA annotations are unavailable in the Legacy Archive or Data Portal<!--DAT-52-->. These annotations can be found [here](https://api.gdc.cancer.gov/data/62461eae-0cc7-4833-bcec-76c73f37858c).
* Some validated somatic mutations may not be present in open-access MAF files.  When creating open-access MAF files from the protected versions we are extremely conservative in removing potential germline variants.  Our approach is to remove all mutations that are present in dbSNP.  In a subsequent release we will provide updated open-access MAF files, which preserve variants found in MC3 or a TCGA validation study.  Please review the protected MAF files in the GDC Data Portal if you are unable to find your mutation in the open-access files.
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* MAF Column #109 "FILTER" entries are separated by both commas and semi-colons. <!-- PGDC-2589 -->
* TARGET-AML is undergoing reorganization.  Pending reorganization, cases from this projects may not contain many clinical, biospecimen, or genomic data files.
* No data from TARGET-MLDS is available.
* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
* SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
* Two biotab files are not linked to Project or Case in the Legacy Archive <!--SV-535, DAT-493-->
* SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->
* Tumor grade property is not populated <!--SV-585-->
* Progression_or_recurrence property is not populated <!--SV-584-->


Details are provided in [Data Release Manifest](https://api.gdc.cancer.gov/data/bb0096b7-efbe-4335-92d1-22b0a64fc0a4)
<br>


## Data Release 4.0

* __GDC Product__: Data
* __Release Date__: October 31, 2016

### New updates

1.  TARGET ALL P1 and P2 biospecimen and molecular data are now available in the Legacy Archive.  Clinical data will be available in a later release. <!-- Dat-185, Dat-194-->
2.  Methylation data from 27k/450k Arrays has been lifted over to hg38 and is now available in the GDC Data Portal <!-- Dat-109 -->
3.  Public MAF files are now available for VarScan2, MuSE, and SomaticSniper.  MuTect2 MAFs were made available in a previous release. <!--DAT-235-->
4.  Updated VCFs and MAF files are available for MuTect2 pipeline to compensate for WGA-related false positive indels.  See additional information on that change [here](https://gdc.cancer.gov/content/mutect2-insertion-artifacts). A listing of replaced files is provided [here](https://api.gdc.cancer.gov/data/ef4c8578-c8d2-4bcb-86af-901af1a5f0a4). <!-- Dat-145, Dat-260 -->
5.  Added submitter_id for Pathology Reports in Legacy Archive <!--DAT-81-->

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

* Some validated somatic mutations may not be present in open-access MAF files.  When creating open-access MAF files from the protected versions we are extremely conservative in removing potential germline variants.  Our approach is to remove all mutations that are present in dbSNP.  In a subsequent release we will provide updated open-access MAF files, which preserve variants found in COSMIC or a TCGA validation study.  Please review the protected MAF files in the GDC Data Portal if you are unable to find your mutation in the open-access files.
* Public MAF files for different variant calling pipelines but the same project may contain different numbers of samples.  Samples are omitted from the public MAF files if they have no PASS variants, which can lead to this apparent discrepancy.
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* MAF Column #109 "FILTER" entries are separated by both commas and semi-colons. <!-- PGDC-2589 -->
* TARGET-AML is undergoing reorganization.  Pending reorganization, cases from this projects may not contain many clinical, biospecimen, or genomic data files.
* No data from TARGET-MLDS is available.
* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
* SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
* There are 200 cases from TCGA-LAML that do not appear in the Legacy Archive <!--SV-327-->
* Biotab files are not linked to Project or Case in the Legacy Archive <!--SV-303-->
* SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->


Details are provided in [Data Release Manifest](https://api.gdc.cancer.gov/data/32d23ec5-143f-45c9-91ad-9949daf5538d)
<br>



## Data Release 3.0

* __GDC Product__: Data
* __Release Date__: September 16, 2016

### New updates

1.  CCLE data now available (in the Legacy Archive only)
2.  BMI calculation is corrected
3.  Slide is now categorized as a Biospecimen entity

### Bugs Fixed Since Last Release

* BMI calculation is corrected

### Known Issues and Workarounds

* Insertions called for tumor samples that underwent whole genome amplification may be of lower quality.  Whether a sample underwent this process can be found in the analyte_type property within analyte and aliquot. TCGA analyte type can be also identified in the 20th character of TCGA barcode, at which "W" corresponds to WGA.<!--BINF-6-->
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* Public MAFs (those with germline variants removed) are only available for MuTect2 pipeline.  MAFs for other pipelines are forthcoming.  
* MAF Column #109 "FILTER" entries are separated by both commas and semi-colons. <!-- PGDC-2589 -->
* TARGET-AML and TARGET-ALL projects are undergoing reorganization.  Pending reorganization, cases from these projects may not contain many clinical, biospecimen, or genomic data files.
* No data from TARGET-PPTP is available.
* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
* SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
* There are 200 cases from TCGA-LAML that do not appear in the Legacy Archive <!--SV-327-->
* Biotab files are not linked to Project or Case in the Legacy Archive <!--SV-303-->
* SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->


Details are provided in [Data Release Manifest](https://api.gdc.cancer.gov/data/bb0096b7-efbe-4335-92d1-22b0a64fc0a4)
<br>


## Data Release 2.0

* __GDC Product__: Data
* __Release Date__: August 9, 2016


### New updates

1.  Additional data, previously available via CGHub and the TCGA DCC, is now available in the GDC
2.  Better linking between files and their associated projects and cases in the Legacy Archive
3.  MAF files are now available in the GDC Data Portal

### Known Issues and Workarounds

* Insertions called for tumor samples that underwent whole genome amplification may be of lower quality.  These are present in VCF and MAF files produced by the MuTect2 variant calling pipeline.  This information can be found in the analyte_type property within analyte and aliquot. TCGA analyte type can be also identified in the 20th character of TCGA barcode, at which "W" corresponds to WGA.<!--BINF-6-->
* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* Public MAFs (those with germline variants removed) are only available for MuTect2 pipeline.  MAFs for other pipelines are forthcoming.  
* MAF Column #109 "FILTER" entries are separated by both commas and semi-colons. <!-- PGDC-2589 -->
* TARGET-AML and TARGET-ALL projects are undergoing reorganization.  Pending reorganization, cases from these projects may not contain many clinical, biospecimen, or genomic data files.
* No data from TARGET-PPTP is available.
* Slide barcodes (`submitter_id` values for Slide entities in the Legacy Archive) are not available <!-- DAT-10 -->
* SDF Files are not linked to Project or Case in the Legacy Archive <!--SV-332-->
* There are 200 cases from TCGA-LAML that do not appear in the Legacy Archive <!--SV-327-->
* Biotab files are not linked to Project or Case in the Legacy Archive <!--SV-303-->
* SDRF files are not linked to Project or Case in the Legacy Archive <!--SV-288-->
* Portion "weight" property is incorrectly described in the Data Dictionary as the weight of the patient in kg, should be described as the weight of the portion in mg <!--SV-391-->

Details are provided in [Data Release Manifest](https://api.gdc.cancer.gov/data/b767f710-9631-401e-b24d-aa90f85268fc)
<br>

## Initial Data Release (1.0)

* __GDC Product__: Data
* __Release Date__: June 6, 2016

### Available Program Data

* The Cancer Genome Atlas (TCGA)
* Therapeutically Applicable Research To Generate Effective Treatments (TARGET)

### Available Harmonized Data

* WXS
     * Co-cleaned BAM files aligned to GRCh38 using BWA
* mRNA-Seq
    * BAM files aligned to GRCh38 using STAR 2-pass strategy
    * Expression quantification using HTSeq
* miRNA-Seq
    * BAM files aligned to GRCh38 using BWA aln
    * Expression quantification using BCCA miRNA Profiling Pipeline*
* Genotyping Array
    * CNV segmentation data


### Known Issues and Workarounds

* BAM files produced by the GDC RNA-Seq Alignment workflow will currently fail validation using the Picard ValidateSamFiles tool.  This is caused by STAR2 not recording mate mapping information for unmapped reads, which are retained in our BAM files.  Importantly, all affected BAM files are known to behave normally in downstream workflows including expression quantification.
* All legacy files for TCGA are available in the GDC Legacy Archive, but not always linked back to cases depending on available metadata.
* Public MAFs (those with germline variants removed) are only available for MuTect2 pipeline.  MAFs for other pipelines are forthcoming.  
* TARGET-AML and TARGET-ALL projects are undergoing reorganization.  Pending reorganization, cases from these projects may not contain many clinical, biospecimen, or genomic data files.
* No data from TARGET-PPTP is available.
*	Legacy data not available in harmonized form:
    *	Annotated VCF files from TARGET, anticipated in future data release
    * TCGA data that failed harmonization or QC or have been newly updated in CGHub: ~1.0% of WXS aliquots, ~1.6% of RNA-Seq aliquots
    * TARGET data that failed harmonization or QC, have been newly updated in CGHub, or whose project names are undergoing reorganization: ~76% of WXS aliquots, ~49% of RNA-Seq aliquots, ~57% of miRNA-Seq.
* MAF Column #109 "FILTER" entries are separated by both commas and semi-colons. <!-- PGDC-2589 -->
*	MAFs are not yet available for query or search in the GDC Data Portal or API.  You may download these files using the following manifests, which can be passed directly to the Data Transfer Tool.  Links for the open-access TCGA MAFs are provided below for downloading individual files.
    * [Open-access MAFs manifest](https://api.gdc.cancer.gov/data/d645229f-f5ae-4f93-bd58-1fc7b2271aa5)
    * [Controlled-access MAFs manifest](https://api.gdc.cancer.gov/data/34789d4c-aeb6-43f2-88e6-1c44ed8e2d2a)

Details are provided in [Data Release Manifest](https://api.gdc.cancer.gov/data/1958147e-371d-41cf-b6bf-3dd1ed4dc237)

### Download Open-access MAF files

* Please note that these links no longer point to files and will be updated in the future.

<a href="https://api.gdc.cancer.gov/data/abbe72a5-cb39-48e4-8df5-5fd2349f2bb2">TCGA.ACC.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/cb97ef6e-7a13-4d42-81b4-1510c00d6373">TCGA.BLCA.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/91ae5ca9-55c2-4c9c-929e-8638444dc7b5">TCGA.BRCA.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/fedc0e90-5070-41e4-993d-603b92f4ecfd">TCGA.CESC.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/f33fd38b-287c-4978-a1bb-a95bbfd4351a">TCGA.CHOL.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/bf1c53dc-79bb-43ae-88e4-23758853e5c6">TCGA.COAD.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/4835f959-6ab4-4ee8-901f-c92aaad4592d">TCGA.DLBC.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/150c6a9f-71cd-4710-9617-cd150498202e">TCGA.ESCA.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/4b176a7b-a5c3-457e-af95-992018b6f3d7">TCGA.GBM.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/64683606-b957-4478-a7d5-673de68b0341">TCGA.HNSC.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/a8dc2dd2-74b3-4035-9551-c0ae3f76293e">TCGA.KICH.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/0cf0f121-4f10-436e-bfc3-8fcfd5f78d0d">TCGA.KIRC.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/b1c13b93-7ad6-4d09-b613-84a7c55e61d9">TCGA.KIRP.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/bac5a617-5b1d-4c33-b9ac-b48bd7e4947a">TCGA.LAML.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/42ff7a98-5a9a-48ad-ad9d-d3a23c245296">TCGA.LGG.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/dc3f239a-ffc6-4e60-b5f5-9f365daaf60a">TCGA.LIHC.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/76b0eec4-bbb6-4340-972d-05a5aace63a4">TCGA.LUAD.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/846c4788-cda0-4f11-b240-a7ad977e032f">TCGA.LUSC.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/34a32349-bf87-4e96-86a5-ca23f0db475e">TCGA.MESO.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/bc5ee1aa-969d-472d-920b-0e654cc585fa">TCGA.OV.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/6f72cd49-6c0e-4409-8e94-26ea5d421bc8">TCGA.PAAD.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/e087cf64-b514-4e92-af9d-2b18341098d5">TCGA.PCPG.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/0d708001-437e-4b78-9ffa-bfafdfc10a28">TCGA.PRAD.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/221f1dec-b539-4345-b687-435659fc21af">TCGA.READ.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/84fad8c0-ac06-4181-92d9-0562392325ba">TCGA.SARC.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/4aee3e32-2802-4e1e-8577-d74b414f30f7">TCGA.SKCM.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/98d14107-5fb5-49bd-ac38-a52178838d6c">TCGA.STAD.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/c3155e81-df07-49c6-8502-ef6ebc60812e">TCGA.TGCT.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/3207f158-6f79-4636-81a1-ce1b30157925">TCGA.THCA.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/ea93d4df-5e76-484a-b7b8-93900ea1d61c">TCGA.THYM.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/a142d8b8-7741-4869-9ca4-0025890eee18">TCGA.UCEC.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/cfc3f9e4-ac0d-4133-b616-a7d2caf28e54">TCGA.UCS.mutect.somatic.maf.gz</a><br>
<a href="https://api.gdc.cancer.gov/data/a96185ef-5b9b-4f0e-b437-f6a0f4f0892b">TCGA.UVM.mutect.somatic.maf.gz</a><br>


# Data Security

To protect the privacy of research participants and support data integrity, the GDC requires user authorization and authentication for:

 - downloading controlled-access data
 - submitting data to the GDC

To perform these functions, GDC users must first obtain appropriate authorization via dbGaP and then authenticate via eRA Commons. The GDC sets user permissions at the project level according to dbGaP authorizations.

*See [Data Access Processes and Tools](https://gdc.cancer.gov/access-data/data-access-processes-and-tools) to learn more about the difference between open-access and controlled-access data.*

## Authorization via dbGaP

Instructions for obtaining authorization via dbGaP are provided in [Obtaining Access to Controlled Data](https://gdc.cancer.gov/access-data/obtaining-access-controlled-data) and [Obtaining Access to Submit Data](https://gdc.cancer.gov/submit-data/obtaining-access-submit-data).

## Authentication via eRA Commons

The following authentication methods are supported by the GDC:

|GDC Tool|Authentication Method|
|----|----|
| GDC Data Portal | Log in using eRA Commons account|
| GDC Data Submission Portal | Log in using eRA Commons account |
| GDC Data Transfer Tool | Authentication Token |
| GDC API | Authentication Token |


### Authentication Tokens

The GDC Data Transfer Tool and the GDC API use tokens for authentication. GDC authentication tokens are alphanumeric strings of characters like this one:

	ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTOKEN-01234567890+AlPhAnUmErIcToKeN=0123456789-ALPHANUMERICTO

## Logging into the GDC

To login to the GDC, users must click on the `Login` button on the top right of the GDC Website.

![Login](images/v2_image_login.jpg)

After clicking Login, users authenticate themselves using their eRA Commons login and password.  If authentication is successful, the eRA Commons username will be displayed in the upper right corner of the screen, in place of the "Login" button.

Upon successful authentication, GDC Data Portal users can:

- See which controlled-access files they can access.
- Download controlled-access files directly from the GDC Data Portal.
- Download an authentication token for use with the GDC Data Transfer Tool or the GDC API.
- See controlled-access mutation data they can access.

Controlled-access files are identified via their status in the Access Column:

[![GDC Data Portal Main Page](images/gdc-file-status-controlled-v2_3.png)](images/gdc-file-status-controlled-v2_3.png "Click to see the full image.")

The rest of this section describes controlled data access features of the GDC Data Portal available to authorized users. For more information about open and controlled-access data, and about obtaining access to controlled data, see [Data Access Processes and Tools](https://gdc.cancer.gov/access-data/data-access-processes-and-tools).

## User Profile

After logging into the GDC Portal, users can view which projects they have access to by clicking the `User Profile` section in the dropdown menu in the top corner of the screen.

[![User Profile Drop Down](images/gdc-user-profile-v2_small.png)](images/gdc-user-profile-v2_small.png "Click to see the full image.")

Clicking this button shows the list of projects.

[![User Profile](images/Profile_Access_List.png)](images/Profile_Access_List.png "Click to see the full image.")

## GDC Authentication Tokens

The GDC Data Portal provides authentication tokens for use with the GDC Data Transfer Tool or the GDC API. To download a token:

1. Log into the GDC using your eRA Commons credentials.
2. Click the username in the top right corner of the screen.
3. Select the "Download token" option.

![Token Download Button](images/Token_Download_v2_small.jpg)

A new token is generated each time the `Download Token` button is clicked.

For more information about authentication tokens, see [Data Security](../../Data/Data_Security/Data_Security.md#authentication-tokens).

>__Note:__ The authentication token should be kept in a secure location, as it allows access to all data accessible by the associated user account.

## Logging Out

To log out of the GDC, click the username in the top right corner of the screen, and select the Logout option.

![Logout link](images/gdc-user-portal-logout-v2_3.jpg)





#### Obtaining A Token

Users can obtain authentication tokens from the [GDC Data Portal](https://portal.gdc.cancer.gov) and the [GDC Data Submission Portal](https://portal.gdc.cancer.gov/submission). See the [GDC Data Submission Portal User's Guide](../../Data_Submission_Portal/Users_Guide/Data_Submission_Process.md#authentication) for instructions.

#### Token Expiration

Tokens are valid for 30 days from the time of issue. Any request to the GDC API that uses an expired token will result in an error.

Tokens can be replaced at any time by downloading a new token, which will be valid for another 30 days.

## Checking User Permissions

Users can view the permissions granted to them by the GDC system as follows:

0. Log into the [GDC Data Portal](https://portal.gdc.cancer.gov) or the [GDC Data Submission Portal](https://portal.gdc.cancer.gov/submission) using your eRA Commons account.
0. Open the URL `https://portal.gdc.cancer.gov/auth/user` to see a JSON object that describes user permissions.


# Copy Number Variation Analysis Pipeline

## Introduction

The copy number variation (CNV) pipeline uses either NGS or Affymetrix SNP 6.0 (SNP6) array data to identify genomic regions that are repeated and infer the copy number of these repeats. Three sets of pipelines have been used for CNV inferences.

* ASCAT
* ABSOLUTE
* DNAcopy

The first set of CNV pipelines are built upon the ASCAT [[1]](https://www.pnas.org/content/107/39/16910) algorithm for both WGS and SNP6 data. ASCAT is able to generate Allele-specific Copy Number Segment data with integer copy number values, and the derived integer Gene-Level Copy Number. 1.) The WGS copy number analysis pipeline, [ascatNGS](https://github.com/cancerit/ascatNgs), is described in detail [here](https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/#whole-genome-sequencing-variant-calling). 2.) The SNP6 copy number analysis pipeline, ASCAT2, is adopted from the [example ASCAT analysis](https://github.com/VanLoo-lab/ascat/blob/v2.5/ExampleData/ASCAT_examplePipeline.R). 3.) The SNP6 copy number analysis pipeline, ASCAT3, is an updated version of ASCAT2. The ASCAT3 analysis in TCGA was done by the [Vanloo lab](https://github.com/VanLoo-lab/ascat/tree/master/ReleasedData/TCGA_SNP6_hg38), and the GDC released a reformatted version of these calls. Both ASCAT2 and ASCAT3 generates data similar to ascatNGS.

The second CNV pipeline, ABSOLUTE, also uses Affymetrix SNP 6.0 (SNP6) array data as input. The hg19 version of the segments were published as one of the [TCGA PanCancer analysis papers](https://doi.org/10.1016/j.ccell.2018.03.007) and the data is available in the [GDC publication page](https://gdc.cancer.gov/about-data/publications/pancanatlas). These calls have been manually curated and thus are considered of good quality. The GDC performed segment liftover and generated gene-level copy numbers. Note that the intermediate output of GRCh38 segments contain liftover artifacts and were not released in the GDC. Users can also obtain corresponding purity and ploidy measurements from the GDC publication page mentioned above.   

The third set of CNV pipelines are built onto the existing TCGA level 2 SNP6 data generated by [Birdsuite](https://www.broadinstitute.org/scientific-community/science/programs/medical-and-population-genetics/birdsuite/birdsuite) and uses the [DNAcopy](http://www.bioconductor.org/packages/release/bioc/html/DNAcopy.html) R-package to perform a circular binary segmentation (CBS) analysis [[2]](http://biostatistics.oxfordjournals.org/content/5/4/557.short). CBS translates noisy intensity measurements into chromosomal regions of equal copy number. The final output files are segmented into genomic regions with the estimated copy number for each region. The GDC further transforms these copy number values into segment mean values, which are equal to log2(copy-number/ 2). Diploid regions will have a segment mean of zero, amplified regions will have positive values, and deletions will have negative values.

## ASCAT Pipelines
### Data Processing Steps
#### Copy Number Segmentation

The [Somatic Copy Number Workflow](/Data_Dictionary/viewer/#?view=table-definition-view&id=somatic_copy_number_workflow) uses a tumor-normal pair of either SNP6 raw CEL data, or WGS data as input. The ASCAT algorithm derives allele-specific copy number segments while estimating and adjusting for tumor purity and ploidy [[1]](https://www.pnas.org/content/107/39/16910). Because there are two parental strands, the resulting Copy Number Segment or Allele-Specific Copy Number Segment files contain 3 different copy number integer values: Major_Copy_Number refers to the larger strand copy number, Minor_Copy_Number refers to the smaller strand copy number, Copy_Number is the sum of Major_Copy_Number and Minor_Copy_Number, and thus equals to the total copy number at the locus.


| I/O | Entity | Format |
|---|---|---|
| Input | [Submitted Genotype_Array](/Data_Dictionary/viewer/#?view=table-definition-view&id=submitted_genotyping_array) |  CEL |
| Output | [Copy Number Segment](/Data_Dictionary/viewer/#?view=table-definition-view&id=copy_number_segment) or [Allele-Specific Copy Number Segment](/Data_Dictionary/viewer/#?view=table-definition-view&id=copy_number_segment) | TXT  |

| I/O | Entity | Format |
|---|---|---|
| Input | [Aligned Reads](/Data_Dictionary/viewer/#?view=table-definition-view&id=aligned_reads) |  BAM |
| Output | [Copy Number Segment](/Data_Dictionary/viewer/#?view=table-definition-view&id=copy_number_segment) or [Allele-Specific Copy Number Segment](/Data_Dictionary/viewer/#?view=table-definition-view&id=copy_number_segment) | TXT  |

#### Gene-Level Copy Number

Gene-level Copy Number is generated by inheriting the Copy_Number value of the residing segment in the Copy Number Segment file generated from ASCAT2, ASCAT3, or ascatNGS workflows.

In some occasions, one gene may overlap with more than one segment. In this case, min_copy_number is the minimum value of all segments it overlaps, max_copy_number is the maximum value of all segments it overlaps, and copy_number is calculated as the weighted (on length of overlapped regions) median of copy number values from all overlapped segments. When there is a tie (very rare), the smaller number is used. If a gene overlaps with only one segment, copy_number = min_copy_number = max_copy_number. If a gene overlaps with no segments, the gene gets empty value "" in copy_number, min_copy_number and max_copy_number.


| I/O | Entity | Format |
|---|---|---|
| Input | [Copy Number Segment](/Data_Dictionary/viewer/#?view=table-definition-view&id=copy_number_segment) or Allele-Specific Copy Number Segment | TXT  |
| Output | [Copy Number Estimate](/Data_Dictionary/viewer/#?view=table-definition-view&id=copy_number_estimate) | TXT  |


### File Access and Availability

| Type | Description | Format |
|---|---|---|
| Copy Number Segment| A table that associates contiguous chromosomal segments with genomic coordinates, and integer copy numbers. |  TXT |
| Allele-Specific Copy Number Segment| A table that associates contiguous chromosomal segments with genomic coordinates, and integer copy numbers. |  TXT |
| Copy Number Estimate | A Gene-level Copy Number file that displays integer copy number on a gene level.  Generated from Copy Number Segment or Allele-Specific Copy Number Segment files. |  TXT |

## ABSOLUTE Copy Number
### Data Processing Steps

The source data were generated by external groups. Please check the [corresponding publication](https://doi.org/10.1016/j.ccell.2018.03.007) for details.

### File Access and Availability

File Access and Availability is similar to that from the ASCAT pipelines, except that only gene-level copy numbers are available, but not segmentation calls.


## DNACopy Pipeline
### Data Processing Steps

The GRCh38 SNP6 probe-set was produced by mapping probe sequences to the GRCh38 reference genome and can be downloaded at the [GDC Reference File Website](https://gdc.cancer.gov/about-data/data-harmonization-and-generation/gdc-reference-files).

#### Copy Number Segmentation

The [Copy Number Liftover Workflow](/Data_Dictionary/viewer/#?view=table-definition-view&id=copy_number_liftover_workflow) uses TCGA level 2 tangent.copynumber files. These files were generated by first normalizing array intensity values, estimating raw copy number, and performing tangent normalization, which subtracts variation that is found in a set of normal samples. 

The Copy Number Liftover Workflow performs CBS analysis using the DNACopy R-package to process tangent normalized data into [Copy Number Segment](/Data_Dictionary/viewer/#?view=table-definition-view&id=copy_number_segment) files, which associate contiguous chromosome regions with log2 ratio segment means in a tab-delimited format.  The number of probes with intensity values associated with each chromosome region is also reported (probes with no intensity values are not included in this count).  During copy number segmentation probe sets from Pseudo-Autosomal Regions (PARs) were removed from males and Y chromosome segments were removed from females.

Masked copy number segments are generated using the same method except that a filtering step is performed that removes the Y chromosome and probe sets that were previously indicated to be associated with frequent germline copy-number variation.   

| I/O | Entity | Format |
|---|---|---|
| Input | [Submitted Tangent Copy Number](/Data_Dictionary/viewer/#?view=table-definition-view&id=submitted_tangent_copy_number) |  TXT |
| Output | [Copy Number Segment](/Data_Dictionary/viewer/#?view=table-definition-view&id=copy_number_segment) or [Masked Copy Number Segment](/Data_Dictionary/viewer/#?view=table-definition-view&id=copy_number_segment) | TXT  |

[1] Van Loo, P., Nordgard, S. H., Lingjaerde, O. C., Russnes, H. G., Rye, I. H., Sun, W. et al. "Allele-specific copy number analysis of tumors." Proceedings of the National Academy of Sciences, 107.39 (2010): 16910-16915.

[2] Olshen, Adam B., E. S. Venkatraman, Robert Lucito, and Michael Wigler. "Circular binary segmentation for the analysis of array-based DNA copy number data." Biostatistics 5, no. 4 (2004): 557-572.


# Aligned Reads Summary Metrics

Various summary metrics are added to the aligned reads entity for query by the user.  These are generated by such tools as SAMtools, Picard, and GATK4.  These may be helpful to determine underlying quality or summary information regarding the submitted data. Examples are included below:

* average_base_quality
* average_insert_size
* average_read_length
* contamination
* contamination_error
* mean_coverage
* msi_score
* msi_status
* pairs_on_diff_chr
* proportion_base_mismatch
* proportion_coverage_10x
* proportion_coverage_30x
* proportion_reads_duplicated
* proportion_reads_mapped
* proportion_targets_no_coverage
* total_reads


For a complete list of the summary metrics as well as the tools used to generate them please visit the [Data Dictionary Viewer](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=aligned_reads).


# DNA-Seq Analysis Pipeline

## Introduction

The GDC DNA-Seq analysis pipeline identifies somatic variants within whole exome sequencing (WXS) and whole genome sequencing (WGS) data. Somatic variants are identified by comparing allele frequencies in normal and tumor sample alignments, annotating each mutation, and aggregating mutations from multiple cases into one project file.

The first pipeline starts with a reference alignment step followed by co-cleaning to increase the alignment quality. Four different variant calling pipelines are then implemented separately to identify somatic mutations. Somatic-caller-identified variants are then annotated. An aggregation pipeline incorporates variants from all cases in one project into a MAF file for each pipeline.


DNA-Seq analysis is implemented across six main procedures:

- Genome Alignment
- Alignment Co-Cleaning
- Somatic Variant Calling
- Variant Annotation
- Mutation Aggregation
- Aggregated Mutation Masking

## Data Processing Steps

### Pre-Alignment
Prior to alignment, BAM files that were submitted to the GDC are split by read groups and converted to FASTQ format. Reads that failed the Illumina chastity test are removed. Note that this filtering step is distinct from trimming reads using base quality scores.


### Alignment Workflow
DNA-Seq analysis begins with the [Alignment Workflow](/Data_Dictionary/viewer/#?view=table-definition-view&id=alignment_workflow). Read groups are aligned to the reference genome using one of two [BWA](http://bio-bwa.sourceforge.net) algorithms [[1]](http://www.ncbi.nlm.nih.gov/pubmed/19451168). BWA-MEM is used if mean read length is greater than or equal to 70 bp. Otherwise BWA-aln is used.
Each read group is aligned to the reference genome separately and all read group alignments that belong to a single aliquot are merged using [Picard Tools](http://broadinstitute.github.io/picard) [SortSam](https://broadinstitute.github.io/picard/command-line-overview.html#SortSam) and [MergeSamFiles](https://broadinstitute.github.io/picard/command-line-overview.html#MergeSamFiles). Duplicate reads, which may persist as PCR artifacts, are then flagged to prevent downstream variant call errors.


#### Reference Genome

All alignments are performed using the human reference genome GRCh38.d1.vd1. Decoy viral sequences are included in the reference genome to prevent reads from aligning erroneously and attract reads from viruses known to be present in human samples. Ten types of human viral genomes are included: human cytomegalovirus (CMV), Epstein-Barr virus (EBV), hepatitis B (HBV), hepatitis C (HCV), human immunodeficiency virus (HIV), human herpes virus 8 (HHV-8), human T-lymphotropic virus 1 (HTLV-1), Merkel cell polyomavirus (MCV), Simian vacuolating virus 40 (SV40), and human papillomavirus (HPV). Reference sequences used by the GDC can be downloaded [here](https://gdc.cancer.gov/download-gdc-reference-files).


| I/O | Entity | Format |
|---|---|---|
| Input | [Submitted Unaligned Reads](/Data_Dictionary/viewer/#?view=table-definition-view&id=submitted_unaligned_reads) or [Submitted Aligned Reads](/Data_Dictionary/viewer/#?view=table-definition-view&id=submitted_aligned_reads) | FASTQ or BAM |
| Output | [Aligned Reads](/Data_Dictionary/viewer/#?view=table-definition-view&id=aligned_reads) | BAM |

![DNA-Seq Alignment Pipeline](images/dna-alignment-pipeline_1.png)

### DNA-Seq Alignment Command Line Parameters

__Note that version numbers may vary in files downloaded from the GDC Portal due to ongoing pipeline development and improvement.__

#### Step 1: Converting BAMs to FASTQs with Biobambam - biobambam2

=== "Shell"

    ```shell
    bamtofastq \
    collate=1 \
    exclude=QCFAIL,SECONDARY,SUPPLEMENTARY \
    filename= <input.bam> \
    gz=1 \
    inputformat=bam \
    level=5 \
    outputdir= <output_path> \
    outputperreadgroup=1 \
    outputperreadgroupsuffixF=_1.fq.gz \
    outputperreadgroupsuffixF2=_2.fq.gz \
    outputperreadgroupsuffixO=_o1.fq.gz \
    outputperreadgroupsuffixO2=_o2.fq.gz \
    outputperreadgroupsuffixS=_s.fq.gz \
    tryoq=1 \
    ```

#### Step 2: BWA Alignment - bwa - samtools
If mean read length is greater than or equal to 70bp:

=== "Shell"

    ```shell
    bwa mem \
    -t 8 \
    -T 0 \
    -R <read_group> \
    <reference> \
    <fastq_1.fq.gz> \
    <fastq_2.fq.gz> |
    samtools view \
    -Shb
    -o <output.bam> -
    ```

If mean read length is less than 70bp:

=== "Shell"

    ```shell
    bwa aln -t 8 <reference> <fastq_1.fq.gz> > <sai_1.sai> &&
    bwa aln -t 8 <reference> <fastq_2.fq.gz> > <sai_2.sai> &&
    bwa sampe -r <read_group> <reference> <sai_1.sai> <sai_2.sai> <fastq_1.fq.gz> <fastq_2.fq.gz> | samtools view -Shb -o <output.bam> -
    ```

If the quality scores are encoded as Illumina 1.3 or 1.5, use BWA aln with the "-l" flag.  

#### Step 3: BAM Sort - picard 2

=== "Shell"

    ```shell
    java -jar picard.jar SortSam \
    CREATE_INDEX=true \
    INPUT=<input.bam> \
    OUTPUT=<output.bam> \
    SORT_ORDER=coordinate \
    VALIDATION_STRINGENCY=STRICT
    ```

#### Step 4: BAM Merge - picard 2

=== "Shell"

    ```Shell
    java -jar picard.jar MergeSamFiles \
    ASSUME_SORTED=false \
    CREATE_INDEX=true \                 
    [INPUT= <input.bam>]  \
    MERGE_SEQUENCE_DICTIONARIES=false \
    OUTPUT= <output_path> \
    SORT_ORDER=coordinate \
    USE_THREADING=true \
    VALIDATION_STRINGENCY=STRICT
    ```

#### Step 5: Mark Duplicates - picard 2

=== "Shell"

    ```Shell
    java -jar picard.jar MarkDuplicates \
    CREATE_INDEX=true \
    INPUT=<input.bam> \
    VALIDATION_STRINGENCY=STRICT
    ```

### Co-cleaning Workflow

The alignment quality is further improved by the [Co-cleaning workflow](/Data_Dictionary/viewer/#?view=table-definition-view&id=alignment_cocleaning_workflow). Co-cleaning is performed as a separate pipeline as it uses multiple BAM files (i.e. the tumor BAM and normal tissue BAM) associated with the same patient. Both steps of this process are implemented using [GATK](https://software.broadinstitute.org/gatk/).  

#### Indel Local Realignment

Local realignment of insertions and deletions is performed using [IndelRealigner](https://github.com/broadinstitute/gatk-docs/blob/master/gatk3-tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_indels_IndelRealigner.json). This step locates regions that contain misalignments across BAM files, which can often be caused by insertion-deletion (indel) mutations with respect to the reference genome. Misalignment of indel mutations, which can often be erroneously scored as substitutions, reduces the accuracy of downstream variant calling steps.

#### Base Quality Score Recalibration

A base quality score recalibration (BQSR) step is then performed using  [BaseRecalibrator](https://gatk.broadinstitute.org/hc/en-us/articles/360036898312-BaseRecalibrator). This step adjusts base quality scores based on detectable and systematic errors. This step also increases the accuracy of downstream variant calling algorithms. Note that the original quality scores are kept in the OQ field of co-cleaned BAM files. These scores should be used if conversion of BAM files to FASTQ format is desired.


| I/O | Entity | Format |
|---|---|---|
| Input | [Aligned Reads](/Data_Dictionary/viewer/#?view=table-definition-view&id=aligned_reads)  | BAM |
| Output | Harmonized Aligned Reads | BAM |

### DNA-Seq Co-Cleaning Command Line Parameters

#### __Step 1:__ RealignTargetCreator

=== "Shell"

    ```Shell
    java -jar GenomeAnalysisTK.jar \
    -T RealignerTargetCreator \
    -R <reference>
    -known <known_indels.vcf>
    [ -I <input.bam> ]
    -o <realign_target.intervals>
    ```

#### __Step 2:__ IndelRealigner

=== "Shell"

    ```Shell
    java -jar GenomeAnalysisTK.jar \
    -T IndelRealigner \
    -R <reference> \
    -known <known_indels.vcf> \
    -targetIntervals <realign_target.intervals> \
    --noOriginalAlignmentTags \
    [ -I <input.bam> ] \
    -nWayOut <output.map>
    ```

#### __Step 3:__ BaseRecalibrator; dbSNP v.144

=== "Shell"

    ```Shell
    java -jar GenomeAnalysisTK.jar \
    -T BaseRecalibrator \
    -R <reference> \
    -I <input.bam> \
    -knownSites <dbsnp.vcf>
    -o <bqsr.grp>
    ```

#### __Step 4:__ PrintReads

=== "Shell"

    ```Shell
    java -jar GenomeAnalysisTK.jar \
    -T PrintReads \
    -R <reference> \
    -I <input.bam> \
    --BQSR <bqsr.grp> \
    -o <output.bam>
    ```


### Somatic Variant Calling Workflow
Aligned and co-cleaned BAM files are processed through the [Somatic Mutation Calling Workflow](/Data_Dictionary/viewer/#?view=table-definition-view&id=somatic_mutation_calling_workflow) as tumor-normal pairs. Variant calling is performed using four separate pipelines:

- [MuSE](http://bioinformatics.mdanderson.org/main/MuSE) [[2]](http://www.biorxiv.org/content/early/2016/05/25/055467.abstract)
- [MuTect2](https://gatk.broadinstitute.org/hc/en-us/articles/360037593851-Mutect2) [[3]](http://www.nature.com/nbt/journal/v31/n3/abs/nbt.2514.html)  
- [VarScan2](http://dkoboldt.github.io/varscan/) [[4]](http://genome.cshlp.org/content/22/3/568.short)
- [Pindel](https://github.com/ucscCancer/pindel-tool)

Note that [SomaticSniper](http://gmt.genome.wustl.edu/packages/somatic-sniper/) [[5]](http://bioinformatics.oxfordjournals.org/content/28/3/311.short) was used and available on the GDC Data Portal prior to [GDC Data Release 35](https://docs.gdc.cancer.gov/Data/Release_Notes/Data_Release_Notes/#data-release-350).

Variant calls are reported by each pipeline in a VCF formatted file. See the GDC [VCF Format](/Data/File_Formats/VCF_Format/) documentation for details on each available field. At this point in the DNA-Seq pipeline, all downstream analyses are branched into four separate paths that correspond to their respective variant calling pipeline.

#### Pipeline Descriptions
Four separate variant calling pipelines are implemented for GDC data harmonization. There is currently no scientific consensus on the best variant calling pipeline so the investigator is responsible for choosing the pipeline(s) most appropriate for the data. Some details about the pipelines are indicated below.

The [MuTect2 pipeline](https://gdc.cancer.gov/files/public/image/Broad_MuTect_0.png) employs a "Panel of Normals" to identify additional germline mutations. This panel is generated using TCGA blood normal genomes from thousands of individuals that were curated and confidently assessed to be cancer-free. This method allows for a higher level of confidence to be assigned to somatic variants that were called by the MuTect2 pipeline.

Basic outlines for the other two pipelines can be found here:

- [VarScan2 pipeline](https://gdc.cancer.gov/files/public/image/varscan-somatic-variant-calling-pipeline.png)
- [MuSE pipeline](https://gdc.cancer.gov/files/public/image/muse-somatic-variant-calling-pipeline.png)  

#### Indels

Indel mutations that were generated with the MuTect2, Pindel, and VarScan pipelines are detected and reported in GDC VCF files.

#### Germline Variants
At this time, germline variants are deliberately excluded as harmonized data. The GDC does not recommend using germline variants that were previously detected and stored in the Legacy Archive as they do not meet the GDC criteria for high-quality data.  

| I/O | Entity | Format |
|---|---|---|
| Input | [Aligned Reads](/Data_Dictionary/viewer/#?view=table-definition-view&id=aligned_reads) |  BAM |
| Output | [Raw Simple Somatic Mutation](/Data_Dictionary/viewer/#?view=table-definition-view&id=simple_somatic_mutation) | VCF  |

### Variant Call Command-Line Parameters

#### MuSE

MuSEv1.0; dbSNP v.144

__Step 1:__ MuSE call

=== "Shell"

    ```Shell
    MuSE call \
    -f <reference> \
    -r <region> \                               	
    <tumor.bam> \
    <normal.bam> \
    -O <intermediate_muse_call.txt>
    ```

__Step 2:__ MuSE sump

=== "Shell"

    ```Shell
    MuSE sump \
    -I <intermediate_muse_call.txt> \                        	
    -E \                          		
    -D <dbsnp_known_snp_sites.vcf> \
    -O <muse_variants.vcf>  
    ```

__Note:__ -E is used for WXS data and -G can be used for WGS data.  


#### MuTect2

GATK; dbSNP v.144

=== "Shell"

    ```Shell
    java -jar GenomeAnalysisTK.jar \
    -T MuTect2 \
    -R <reference> \
    -L <region> \
    -I:tumor <tumor.bam> \
    -I:normal <normal.bam> \
    --normal_panel <pon.vcf> \                        
    --cosmic <cosmic.vcf> \
    --dbsnp <dbsnp.vcf> \
    --contamination_fraction_to_filter 0.02 \                   
    -o <mutect_variants.vcf> \
    --output_mode EMIT_VARIANTS_ONLY \
    --disable_auto_index_creation_and_locking_when_reading_rods
    ```

#### VarScan


__Step 1:__ Mpileup; Samtools

=== "Shell"

    ```Shell
    samtools mpileup \
    -f <reference> \
    -q 1 \
    -B \
    <normal.bam> \
    <tumor.bam> >
    <intermediate_mpileup.pileup>
    ```

__Step 2:__ Varscan Somatic; Varscan.v2

=== "Shell"

    ```Shell
    java -jar VarScan.jar somatic \
    <intermediate_mpileup.pileup> \
    <output_path> \
    --mpileup      1 \
    --min-coverage 8 \
    --min-coverage-normal 8 \
    --min-coverage-tumor 6 \
    --min-var-freq 0.10 \
    --min-freq-for-hom 0.75 \
    --normal-purity 1.0 \
    --tumor-purity 1.00 \
    --p-value 0.99 \
    --somatic-p-value 0.05 \
    --strand-filter 0 \
    --output-vcf
    ```

__Step 3:__ Varscan ProcessSomatic; Varscan.v2

=== "Shell"

    ```Shell
    java -jar VarScan.jar processSomatic \
    <intermediate_varscan_somatic.vcf> \
    --min-tumor-freq 0.10 \
    --max-normal-freq 0.05 \
    --p-value 0.07
    ```

#### Pindel

__Step 1:__ Filter Reads

Filter BAM reads that are not unmapped or duplicate or secondary_alignment or failed_quality_control or supplementary for both tumor and normal BAM files

Tool: sambamba

=== "Shell"

    ```Shell
    Sambamba view $(input.bam) --filter "not (unmapped or duplicate or secondary_alignment or failed_quality_control or supplementary)" --format bam --nthreads 1 --output-filename $(output.bam)
    ```

__Step 2:__ Pindel

[Pindel Repo](https://github.com/genome/pindel/releases/tag/v0.2.5b8)

__Step 2a.:__ Calculate mean insert size

=== "Python"

    ```Python
    cmd = "samtools view -f66 %s | head -n 1000000" % (bam)
    output = do_shell_command(cmd)
    lines = output.decode('utf-8').split('\n')
    b_sum = 0
    b_count = 0
    numlines = 0
    for line in lines:
        numlines += 1
        tmp = line.split("\t")
        if len(tmp) < 9:
            break
        if abs(int(tmp[8])) < 10000:
            b_sum += abs(int(tmp[8]))
            b_count += 1
    try:
        mean = b_sum / b_count
    ```

__Step 2b.:__ Write it to a config file

=== "Python"

    ```Python
    for inputBamFile, meanInsertSize, tag in zip(inputBamFiles, meanInsertSizes, tags):
            fil.write("%s\t%s\t%s\n" %(inputBamFile, meanInsertSize, tag))
        fil.close()
    ```

__Step 2c.:__ Run pindel

=== "Shell"

    ```Shell
    pindel \
    -f GRCh38.d1.vd1.fa \
    -i config_file \
    -o $(output_prefix) \
    --exclude GRCh38.d1.vd1.centromeres.telomeres.bed
    ```

__Step 2d.:__ Merge DI and SI OUTPUT

=== "Python"

    ```Python
    with open(os.path.join(args.workdir, "pindel_somatic"), "w") as handle:
            for p in pindel_files:
                if p.endswith("_D"):
                with open(p) as ihandle:
                    for line in ihandle:
                        if re.search("ChrID", line):
                            handle.write(line)
            for p in pindel_files:
                if p.endswith("_SI"):
                with open(p) as ihandle:
                    for line in ihandle:
                    if re.search("ChrID", line):
                        handle.write(line)
    ```

__Step 2e.:__ Create a config for pindel somatic filter

=== "Python"

    ```Python
    indel.filter.input = $(merged.pindel.output)
    indel.filter.vaf = 0.08
    indel.filter.cov = 20
    indel.filter.hom = 6
    indel.filter.pindel2vcf = "/path/to/pindel/pindel2vcf4tcga"
    indel.filter.reference =  "GRCh38.d1.vd1.fa"
    indel.filter.referencename = "GRCh38"
    indel.filter.referencedate = datetime.datetime.now().strftime("%Y%m%d")
    indel.filter.output = $(output.file.name.vcf)
    ```

__Step 2f.:__ Apply somatic filter on pindel output
Tool: pindel2vcf4tcga

=== "Perl"

    ```Perl
    perl pindel/somatic_filter/somatic_indelfilter.pl $(somatic.indel.filter.config)
    ```

__Step 3:__ Pindel
Tool: Picard.jar 2

=== "Shell"

    ```Shell
    java \
    -d64 \
    -XX: +UseSerialGC \
    -Xmx16G \
    -jar picard.jar \
    SortVcf \
    CREATE_INDEX=true \
    SEQUENCE_DICTIONARY=GRCh38.d1.vd1.dict \
    I=$(pindel.somatic.vcf) \
    OUTPUT=$(output.vcf.gz)
    ```

__Step 5:__ Vt Normalization
Tool: GenomeAnalysisTK.jar nightly-2016-02-25-gf39d340

=== "Shell"

    ```Shell
    java \
    -Xmx4G \
    -jar \
    /bin/GenomeAnalysisTK.jar \
    -T VariantFiltration \
    --disable_auto_index_creation_and_locking_when_reading_rods \
    --variant $(vt.normal.output.vcf.gz) \
    -R GRCh38.d1.vd1.fa \
    --filterExpression vc.isBiallelic() && vc.getGenotype(\"TUMOR\").getAD().1 < 3" \
    --filterName TALTDP \
    -o $(output.vcf.gz)
    ```

### Variant Call Annotation Workflow

Raw VCF files are then annotated in the [Somatic Annotation Workflow](/Data_Dictionary/viewer/#?view=table-definition-view&id=somatic_annotation_workflow) with the [Variant Effect Predictor (VEP)](https://www.ensembl.org/info/docs/tools/vep/index.html) v84 [[6]](http://dx.doi.org/10.1093/bioinformatics/btq330) along with VEP GDC plugins.   

The VEP uses the coordinates and alleles in the VCF file to infer biological context for each variant including the location of each mutation, its biological consequence (frameshift/ silent mutation), and the affected genes. See the documentation on the [GDC VCF Format](../File_Formats/VCF_Format/) for more details. Variants in the VCF files are also matched to known variants from external mutation databases. The following databases are used for VCF annotation:  

*   GENCODE v.22
*   sift v.5.2.2
*   ESP v.20141103
*   polyphen v.2.2.2
*   dbSNP v.146
*   Ensembl genebuild v.2014-07
*   Ensembl regbuild v.13.0
*   HGMD public v.20154
*   ClinVar v.201601

Due to licensing constraints COSMIC is not utilized for annotation in the GDC VEP workflow.

In addition to annotation, [False Positive Filter](https://github.com/ucscCancer/fpfilter-tool) is used to label low quality variants in VarScan.

| I/O | Entity | Format |
|---|---|---|
| Input | [Simple Somatic Mutation](/Data_Dictionary/viewer/#?view=table-definition-view&id=simple_somatic_mutation) | VCF  |
| Output | [Annotated Somatic Mutation](/Data_Dictionary/viewer/#?view=table-definition-view&id=annotated_somatic_mutation) | VCF  |

### Tumor-Only Variant Calling Workflow

Tumor only variant calling is performed on a tumor sample with no paired normal at the request of the research group. This method takes advantage of the normal cell contamination that is present in most tumor samples. These calls are made using the version of MuTect2 included in GATK4. Tumor-only variant call files can be found in the GDC Portal by filtering for "Workflow Type: GATK4 MuTect2".   

### Tumor-Only Variant Call Command-Line Parameters

```shell
GATK4 v4

## 1. Generate OXOG metrics:

java -d64 -XX:+UseSerialGC -Xmx3G -jar /gatk/gatk.jar \
CollectSequencingArtifactMetrics \
-I Tumor_Sample_Alignment.bam \
-O <job_identifier> \
--FILE_EXTENSION .txt \
-R GRCh38.d1.vd1.fa  ## Only chr1-22 + XYM

## 2. Generate pileup summaries on tumor sample:

java -d64 -XX:+UseSerialGC -Xmx3G -jar /gatk/gatk.jar \
GetPileupSummaries
-I Tumor_Sample_Alignment.bam \
-O <job_identifier>.targeted_sequencing.table \
-V af-only-gnomad-common-biallelic.grch38.main.vcf.gz \ # Germline reference from gnomad
-L intervals.bed \ ## Only chr1-22 + XYM
-R GRCh38.d1.vd1.fa

## 3. Calculate contamination on tumor sample

java -d64 -XX:+UseSerialGC -Xmx3G -jar /gatk/gatk.jar \
CalculateContamination \
-I <job_identifier>.targeted_sequencing.table \ # From step 2
-O <job_identifier>.targeted_sequencing.contamination.table

## 4. Find tumor sample name from BAM

java -d64 -XX:+UseSerialGC -Xmx3G -jar /gatk/gatk.jar \
GetSampleName \
-I Tumor_Sample_Alignment.bam \
-O <job_identifier>.targeted_sequencing.sample_name

## 5. Run MuTect2 using only tumor sample on chromosome level (25 commands with different intervals)

java -Djava.io.tmpdir=/tmp/job_tmp_3 -d64 -jar -Xmx3G -XX:+UseSerialGC \
/bin/gatk-4.2.4.0/gatk-package-4.2.4.0-local.jar \
Mutect2 \
-R GRCh38.d1.vd1.fa \
-L chr4:1-190214555 \ # Specify chromosome
-I Tumor_Sample_Alignment.bam \
-O 3.mt2.vcf \
-tumor <tumor_sample_name> \ # From step 4
--af-of-alleles-not-in-resource 2.5e-06 \
--germline-resource af-only-gnomad.hg38.vcf.gz \ # Germline reference from gnomad
-pon gatk4_mutect2_4136_pon.vcf.gz # New panel of normal created by 4136 TCGA curated normal samples, using GATK4

## After this step, all chromosome level VCFs are merged into one.

## 6. Sort VCF with Picard

java -d64 -XX:+UseSerialGC -Xmx16G -jar /usr/local/bin/picard.jar \
SortVcf \
SEQUENCE_DICTIONARY=GRCh38.d1.vd1.dict \
OUTPUT=<job_identifier>.targeted_sequencing.mutect2.tumor_only.sorted.vcf.gz \
I=merged_multi_gatk4_mutect2_tumor_only_calling.vcf \ # From step 5
CREATE_INDEX=true

## 7. Filter variant calls from MuTect
java -d64 -XX:+UseSerialGC -Xmx3G -jar /gatk/gatk.jar \
FilterMutectCalls \
-O <job_identifier>.targeted_sequencing.mutect2.tumor_only.contFiltered.vcf.gz \
-V <job_identifier>.targeted_sequencing.mutect2.tumor_only.sorted.vcf.gz \ # From step 6
--contamination-table <job_identifier>.targeted_sequencing.contamination.table \ # From step 3
-L intervals.bed

## 8. Filter variants by orientation bias
java -d64 -XX:+UseSerialGC -Xmx3G -jar /gatk/gatk.jar \
FilterByOrientationBias \
-O <job_identifier>.targeted_sequencing.tumor_only.gatk4_mutect2.raw_somatic_mutation.vcf.gz \ # final output
-P <job_identifier>.pre_adapter_detail_metrics.txt \ # From step 1
-V <job_identifier>.targeted_sequencing.mutect2.tumor_only.contFiltered.vcf.gz \ # From step 7
-L intervals.bed \
-R GRCh38.d1.vd1.fa \
-AM G/T \
-AM C/T
```

### Tumor-Only Variant Annotation Workflow

After single-tumor variant calling is performed with MuTect2, a series of filters are applied to minimize the release of germline variants in downloadable VCFs. In all cases, the GDC applies a set of custom filters based on allele frequency, mapping quality, somatic/germline probability, and copy number. In some cases an additional variant classification step is applied before the GDC filters.

The [PureCN](https://bioconductor.org/packages/devel/bioc/html/PureCN.html) R-package [[7]](https://doi.org/10.1186/s13029-016-0060-z) [[8]](https://doi.org/10.1101/552711) is used to classify the variants by somatic/germline status and clonality based on tumor purity, ploidy, contamination, copy number, and loss of heterozygosity. The following steps are performed with this package:

* __Interval Capture__ : Generates an interval file using a FASTA and BED file coordinates.
* __GC-Normalization__ : Calculates GC-normalized tumor/normal coverage data.
* __Normal DB Creation__ : Generates a normal database  using the normalized coverage file and panel-of-normals VCF
* __Somatic Variant Calling__ : Classifies each of the previously called variants

Note that PureCN will not be performed if there is insufficient data to produce a target capture kit specific normal database. In rare occasions, PureCN may not find a numeric solution. If PureCN is not performed or does not find a solution, this is indicated in the VCF header. VCF files that were annotated with these pipelines can be found in the GDC Portal by filtering for "Workflow Type: GATK4 MuTect2 Annotation".

### Somatic Aggregation Workflow

The Somatic Aggregation Workflow generates one MAF file from multiple VCF files; see the [GDC MAF Format](/Data/File_Formats/MAF_Format/) guide for details on file structure. In this step, one MAF file is generated per variant calling pipeline for each project and contains all available cases within this project.

| I/O | Entity | Format |
|---|---|---|
| Input | Multiple [Annotated Somatic Mutation](/Data_Dictionary/viewer/#?view=table-definition-view&id=annotated_somatic_mutation) | VCF  |
| Output | [Aggregated Somatic Mutation](/Data_Dictionary/viewer/#?view=table-definition-view&id=aggregated_somatic_mutation) | MAF  |

### Masked Somatic Aggregation Workflow

The MAF files generated by Somatic Aggregation Workflow are controlled-access due to the presence of germline mutations. Open-access MAF files are modified for public release by removing columns and variants that could potentially contain germline mutation information. See the GDC [MAF Format](/Data/File_Formats/MAF_Format/) for details about the criteria used to remove variants.

While these criteria cause the pipeline to over-filter some of the true positive somatic variants in open-access MAF files, they prevent personally identifiable germline mutation information from becoming publicly available. The GDC recommends that investigators explore both controlled and open-access MAF files if omission of certain somatic mutations is a concern.

| I/O | Entity | Format |
|---|---|---|
| Input | [Aggregated Somatic Mutation](/Data_Dictionary/viewer/#?view=table-definition-view&id=aggregated_somatic_mutation) | Protected MAF |
| Output | [Masked Somatic Mutation](/Data_Dictionary/viewer/#?view=table-definition-view&id=masked_somatic_mutation) | Somatic MAF |

### Whole Genome Sequencing Variant Calling

Variant calls are generated from WGS data using a different pipeline than WXS and Targeted Sequencing samples. This pipeline, based on a [workflow generated by the Sanger Institute](https://github.com/cancerit/dockstore-cgpwgs),  generates multiple downstream data types using the following software packages:

* __CaVEMan:__ Single nucleotide variants, which are available in [VCF format](https://docs.gdc.cancer.gov/Data/File_Formats/VCF_Format/).
* __Pindel:__ Small indel variants, which are available in [VCF format](https://docs.gdc.cancer.gov/Data/File_Formats/VCF_Format/).
* __BRASS:__ Structural variants, which are available in *BEDPE format*.
* __AscatNGS:__  Copy number variants, which are available as copy number estimates or copy number segment files, data may be available in *tab separated values (.TSV) or plain text file (.TXT)*

#### BEDPE File Format

[BEDPE file format](https://bedtools.readthedocs.io/en/latest/content/general-usage.html#bedpe-format), (**b**rowser **e**xtensible **d**ata **p**aired-**e**nd) is designed to concisely describe disjoint genome features, such as structural variations or paired-end sequence alignments. It's an enhanced version of the [BED format](http://genome.ucsc.edu/FAQ/FAQformat#format1), as BED does not allow inter-chromosomal feature definitions. In addition, BED only has one strand field, which is insufficient for paired-end sequence alignments, especially when studying structural variation. The BEDPE format is described below.


* __chr*x* (required):__ The name of the chromosome on which the *x*th end of the feature exists. (x is 1 or 2). Any string can be used. For example, "chr1", "III", "myChrom", "contig1112.23" (use "." for unknown).
* __start*x* (required):__ The zero-based starting position of the **first** end of the feature on chr*x*. The first base in a chromosome is numbered 0. The start position in each BEDPE feature is therefore interpreted to be 1 greater than the start position listed in the feature (use -1 for unknown).
* __end*x* (required):__ The one-based ending position of the first end of the feature on chr*x*. The end position in each BEDPE feature is one-based (use -1 for unknown).
* __name (optional):__ Defines the name of the BEDPE feature. Any string can be used.  
* __score (optional):__ A score between 0 and 1000. If the track line *useScore* attribute is set to 1 for this annotation data set, the score value will determine the level of gray in which this feature is displayed (higher numbers = darker gray). Any string can be used.
* __strand*x* (optional):__ Defines the strand for the *x*th end of the feature. Either "." (unknown),  "+", or "-".

In addition to the above fields, bedtools allows for the addition of user-defined fields to the normal, 10-column BEDPE format as necessary. These columns are merely "passed through" pairToBed and pairToPair and are not part of any analysis. One would use these additional columns to add extra information (e.g., edit distance for each end of an alignment, or "deletion", "inversion", etc.) to each BEDPE feature.

#### CNV from WGS File Format

AscatNGS, originally developed by [Raine *et al* (2016)]( https://doi.org/10.1002/cpbi.17) ([GitHub page](https://github.com/cancerit)), indicates the DNA copy number changes affecting a tumor genome when comparing to a matched normal sample. See below for a description of the copy number segment and copy number estimation files produced by AscatNGS:

* __GDC Aliquot:__  The GDC ID for the aliquot collected from the sample (copy number segment files only).
* __Gene ID:__  The gene ENSMBL ID (copy number variant only).
* __Gene Name:__  The gene symbol (copy number variant only).
* __Chromosome:__  The name of the chromosome on which the copy number change exists.
* __Start:__  The starting position of the copy.
* __End:__  The ending position of the copy.
* __Copy Number:__  The weighted median of the strand copy numbers [9].
* __Major Copy Number:__ The greater strand copy number of the two strands of the DNA (copy number segment files only).
* __Minor Copy number:__  The smaller strand copy number of the two strands of the DNA (copy number segment files only).
* __Max. Copy number:__ The highest copy number for overlapped segment (copy number variant only).
* __Min. Copy number:__ The lowest copy number for overlapped segment (copy number variant only).

## Microsatellite Instability Detection

The GDC adopts [MSIsensor2](https://github.com/niu-lab/msisensor2) to derive Microsatellite Instability (MSI) information from tumor DNA-Seq data. The MSIsensor2 software uses only the tumor BAM as input, and calculates the numeric MSI score (number of msi sites / all valid sites). The MSI status of MSI (Microsatellite Instable) or MSS (Microsatellite Stable) is then determined using a MSI score cutoff value of 20%.

The output `msi_score` and `msi_status` values are stored directly as properties of the `aligned_reads` (BAM files), and can be [accessible via API](https://api.gdc.cancer.gov/files/82488c57-9789-449c-a09d-594172381dc1?pretty=true&fields=msi_score,msi_status,file_id,file_name). In addition, the portal/API can be filtered using these proprties by choosing "Add a Custom Filter" in the [Repository Page](https://portal.gdc.cancer.gov/analysis_page?app=Downloads) and selecting `msi_score` or `msi_status`.

Please note:

1.  MSI status generated from DNA-Seq by the GDC is considered bioinformatics-derived information, and is not considered clinical data. If performed by the clinical lab, the clinical MSI test result would be stored as a `laboratory_test` in the molecular_test entity.
2.  MSIsensor2 can theoretically be applied to WGS, WXS, or Targeted Sequencing data. Given the number of MSI sites available in some Targeted Sequencing data, please consider the results carefully.
3.  It is possible that multiple MSI statuses exist within the same sample/case if more than one DNA-Seq BAM was generated. It is the users' responsibility to check for their consistency, especially when the MSI scores are close to 20%.

## File Access and Availability

Files from the GDC DNA-Seq analysis pipeline are available in the [GDC Data Portal](https://portal.gdc.cancer.gov) in BAM, VCF, and MAF formats. Descriptions are listed below for all available data types and their respective file formats.

| Data Type | Description | File Format |
|---|---|---|
| [Aligned Reads](/Data_Dictionary/viewer/#?view=table-definition-view&id=aligned_reads) | Reads that have been aligned to the GRCh38 reference and co-cleaned. Unaligned reads and reads that map to decoy sequences are also included in the BAM files.  | BAM |
| [Raw Simple Somatic Mutation](/Data_Dictionary/viewer/#?view=table-definition-view&id=simple_somatic_mutation)| A tab-delimited file with genotypic information related to genomic positions. Genomic variants are first identified here.  | VCF |
| [Annotated Somatic Mutation](/Data_Dictionary/viewer/#?view=table-definition-view&id=annotated_somatic_mutation)  | An annotated version of a raw simple somatic mutation file. Annotated files include biological context about each observed mutation. |  VCF |
| [Aggregated Somatic Mutation](/Data_Dictionary/viewer/#?view=table-definition-view&id=aggregated_somatic_mutation) | A tab-delimited file derived from multiple VCF files. Contains information from all available cases in a project.  | MAF |
| [Masked Somatic Mutation](/Data_Dictionary/viewer/#?view=table-definition-view&id=masked_somatic_mutation) | A modified version of the Aggregated Somatic Mutation MAF file with sensitive or potentially erroneous data removed. | MAF |

- - -

[1]. Li, Heng, and Richard Durbin. "Fast and accurate short read alignment with Burrows-Wheeler transform." Bioinformatics 25, no. 14 (2009): 1754-1760.

[2]. Fan, Yu, Liu Xi, Daniel ST Hughes, Jianjun Zhang, Jianhua Zhang, P. Andrew Futreal, David A. Wheeler, and Wenyi Wang. "Accounting for tumor heterogeneity using a sample-specific error model improves sensitivity and specificity in mutation calling for sequencing data." bioRxiv (2016): 055467.

[3]. Cibulskis, Kristian, Michael S. Lawrence, Scott L. Carter, Andrey Sivachenko, David Jaffe, Carrie Sougnez, Stacey Gabriel, Matthew Meyerson, Eric S. Lander, and Gad Getz. "Sensitive detection of somatic point mutations in impure and heterogeneous cancer samples." Nature biotechnology 31, no. 3 (2013): 213-219.

[4]. Koboldt, Daniel C., Qunyuan Zhang, David E. Larson, Dong Shen, Michael D. McLellan, Ling Lin, Christopher A. Miller, Elaine R. Mardis, Li Ding, and Richard K. Wilson. "VarScan 2: somatic mutation and copy number alteration discovery in cancer by exome sequencing." Genome research 22, no. 3 (2012): 568-576.

[5]. Larson, David E., Christopher C. Harris, Ken Chen, Daniel C. Koboldt, Travis E. Abbott, David J. Dooling, Timothy J. Ley, Elaine R. Mardis, Richard K. Wilson, and Li Ding. "SomaticSniper: identification of somatic point mutations in whole genome sequencing data." Bioinformatics 28, no. 3 (2012): 311-317.

[6]. McLaren, William, Bethan Pritchard, Daniel Rios, Yuan Chen, Paul Flicek, and Fiona Cunningham. "Deriving the consequences of genomic variants with the Ensembl API and SNP Effect Predictor." Bioinformatics 26, no. 16 (2010): 2069-2070.

[7]. Riester, Markus, Angad P. Singh, A. Rose Brannon, Kun Yu, Catarina D. Campbell, Derek Y. Chiang, and Michael P. Morrissey. "PureCN: copy number calling and SNV classification using targeted short read sequencing." Source code for biology and medicine 11, no. 1 (2016): 13.

[8]. Oh, Sehyun, Ludwig Geistlinger, Marcel Ramos, Martin Morgan, Levi Waldron, and Markus Riester. "Reliable analysis of clinical tumor-only whole exome sequencing data" bioRxiv 552711 (2019);

[9]. Gene-level copy number data is generated by intersection of copy number segment and gene ranges. It is possible for one gene to overlap with multiple segments, and in this case, copy_number, min_copy_number and max_copy_number could take different values. In particular, the copy_number value is calculated as the median, weighted on length of overlapped bases, of segment copy numbers from all overlapped segments.


# miRNA Analysis Pipeline

## Introduction
The GDC miRNA quantification analysis makes use of a modified version of the profiling pipeline that the British Columbia Genome Sciences Centre developed.  The pipeline generates TCGA-formatted miRNAseq data. The first step is read alignment. The tool then compares the individual reads to sequence feature annotations in miRBase v21 and UCSC. Of note, however, the tool only annotates those reads that have an exact match with known miRNAs in miRBase and should therefore not be considered for novel miRNA identification or mismatched alignments.

For more information see [BCGSC's GitHub](https://github.com/bcgsc/mirna) or the [original publication](http://nar.oxfordjournals.org/content/early/2015/08/13/nar.gkv808.full).


## Data Processing Steps

### Alignment Workflow
The miRNA pipeline begins with the [Alignment Workflow](/Data_Dictionary/viewer/#?view=table-definition-view&id=alignment_workflow), which in the case of miRNA uses BWA-aln. This outputs one BAM file for each read group in the input.

| I/O | Entity | Format |
|---|---|---|
| Input | [Submitted Unaligned Reads](/Data_Dictionary/viewer/#?view=table-definition-view&id=submitted_unaligned_reads) or [Submitted Aligned Reads](/Data_Dictionary/viewer/#?view=table-definition-view&id=submitted_aligned_reads) |  FASTQ or BAM |
| Output | [Aligned Reads](/Data_Dictionary/viewer/#?view=table-definition-view&id=aligned_reads) | BAM  |


### miRNA Expression Workflow
Following alignment, BAM files are processed through the [miRNA Expression Workflow](/Data_Dictionary/viewer/#?view=table-definition-view&id=mirna_expression_workflow).

The outputs of the miRNA profiling pipeline report raw read counts and counts normalized to reads per million mapped reads (RPM) in two separate files mirnas.quantification.txt and isoforms.quantification.txt. The former contains summed expression for all reads aligned to known miRNAs in the miRBase reference. If there are multiple alignments to different miRNAs or different regions of the same miRNA, the read is flagged as cross-mapped and every miRNA annotation is preserved. The latter contains observed isoforms.

| I/O | Entity | Format |
|---|---|---|
| Input | [Aligned Reads](/Data_Dictionary/viewer/#?view=table-definition-view&id=aligned_reads) |  BAM |
| Output | [miRNA Expression](/Data_Dictionary/viewer/#?view=table-definition-view&id=mirna_expression) | TXT  |

## File Access and Availability

| Type | Description | Format |
|---|---|---|
| Aligned Reads | miRNA-Seq reads that have been aligned to the GRCh38 build. Reads that were not aligned are included to facilitate the availability of raw read sets.|  BAM |
| miRNA Expression Quantification | A table that associates miRNA IDs with read count and a normalized count in reads-per-million-miRNA-mapped. |  TXT |
| Isoform Expression Quantification | A table with the same information as the miRNA Expression Quantification files with the addition of isoform information such as the coordinates of the isoform and the type of region it constitutes within the full miRNA transcript.  |  TXT |


# Methylation Analysis

## Methylation Array Harmonization Workflow

### Introduction

The [Methylation Array Harmonization Workflow](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=methylation_array_harmonization_workflow) uses raw methylation array data from multiple generations of Illumina Infinium DNA methylation arrays, namely Human Methylation 27 (HM27), HumanMethylation 450 (HM450) and EPIC platforms, to measure the level of methylation at known CpG sites as beta values, calculated from array intensities (Level 2 data) as Beta = M/(M+U). This differs from the [Methylation Liftover Pipeline](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=methylation_liftover_workflow) in that the raw methylation array data is used instead of submitted methylation beta values, and the data is processed through the software package [SeSAMe](https://github.com/zwdzwd/sesame)[1]. Additionally, the analysis results from the Methylation Array Harmonization Workflow are expected to be of higher quality than results from the Methylation Liftover Pipeline.

SeSAMe offers correction to detection failures that occur in other DNA methylation array software commonly due to germline and somatic deletions by utilizing a novel way to calculate the significance of detected signals in methylation arrays. By correcting for these artifacts as well as other improvements to DNA methylation data processing, SeSAMe improves upon detection calling and quality control of processed DNA methylation data. SeSAMe output files include: two Masked Methylation Array IDAT files, one for each color channel, that contains channel data from a raw methylation array after masking potential genotyping information; and a subsequent Methylation Beta Value TXT file derived from the two Masked Methylation Array IDAT files, that displays the calculated methylation beta value for CpG sites.

### SeSAMe Methylation Beta Values File Format

Descriptions for fields present in GDC Harmonized Methylation Beta Values File are detailed below:

| Field | Definition |
|---|---|
| Composite Element | A unique ID for the array probe associated with a CpG site |
| Beta Value | Represents the ratio between the methylated array intensity and total array intensity, falls between 0 (lower levels of methylation) and 1 (higher levels of methylation) |

---

| I/O | Entity | Format |
|---|---|---|
| Input | [Raw Methylation Array](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=raw_methylation_array) | IDAT |
| Output | [Masked Methylation Array](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=masked_methylation_array) | IDAT |
| Output | [Methylation Beta Values](/Data_Dictionary/viewer/#?view=table-definition-view&id=methylation_beta_value) | TXT |


## Methylation Liftover Pipeline

### Note: as of Data Release 32, Methylation Liftover files are no longer supported and do not appear in the GDC Data Portal.

### Introduction

The [DNA Methylation Liftover Pipeline](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=methylation_liftover_workflow) uses data from the Illumina Infinium Human Methylation 27 (HM27) and HumanMethylation450 (HM450) arrays to measure the level of methylation at known CpG sites as beta values, calculated from array intensities (Level 2 data) as Beta = M/(M+U).

Using probe sequence information provided in the manufacturer's manifest, HM27 and HM450 probes were remapped to the GRCh38 reference genome [2]. Type II probes with a mapping quality of <10, or Type I probes for which the methylated and unmethylated probes map to different locations in the genome, and/or had a mapping quality of <10, had an entry of '\*' for the 'chr' field, and '-1' for coordinates. These coordinates were then used to identify the associated transcripts from GENCODE v22, the associated CpG island (CGI), and the CpG sites' distance from each of these features. Multiple transcripts overlapping the target CpG were separated with semicolons. Beta values were inherited from existing TCGA Level 3 DNA methylation data (hg19-based) based on Probe IDs.

### Methylation Liftover Pipeline Table Format

| Field | Definition |
|---|---|
| Composite Element | A unique ID for the array probe associated with a CpG site |
| Beta Value | Represents the ratio between the methylated array intensity and total array intensity, falls between 0 (lower levels of methylation) and 1 (higher levels of methylation) |
| Chromosome | The chromosome in which the probe binding site is located |
| Start | The start of the CpG site on the chromosome |
| End | The end of the CpG site on the chromosome |
| Gene Symbol | The symbol for genes associated with the CpG site. Genes that fall within 1,500 bp upstream of the transcription start site (TSS) to the end of the gene body are used.    |
| Gene Type | A general classification for each gene (e.g. protein coding, miRNA, pseudogene) |
| Transcript ID |  Ensembl transcript IDs for each transcript associated with the genes detailed above |
| Position to TSS | Distance in base pairs from the CpG site to each associated transcript's start site  |
| CGI Coordinate | The start and end coordinates of the CpG island associated with the CpG site |
| Feature Type | The position of the CpG site in reference to the island: Island, N_Shore or S_Shore (0-2 kb upstream or downstream from CGI), or N_Shelf or S_Shelf (2-4 kbp upstream or downstream from CGI) |

---

| I/O | Entity | Format |
|---|---|---|
| Input | [Submitted Methylation Beta Values](/Data_Dictionary/viewer/#?view=table-definition-view&id=submitted_methylation_beta_value) |  TXT |
| Output | [Methylation Beta Values](/Data_Dictionary/viewer/#?view=table-definition-view&id=methylation_beta_value) or [Masked Methylation Array](/Data_Dictionary/viewer/#?view=table-definition-view&id=masked_methylation_array) | TXT/IDAT  |


## File Access and Availability

| Type | Description | Format |
|---|---|---|
| Methylation Beta Value | A table that associates array probes with CpG sites and associated metadata. | TXT |
| Masked Methylation Array | A data file that contains channel data from a raw methylation array after masking of potential genotyping information. | IDAT |


[1]. Zhou, Wanding, Triche Timothy J., Laird Peter W. and Shen Hui. "SeSAMe: Reducing artifactual detection of DNA methylation by Infinium BeadChips in genomic deletions." Nucleic Acids Research. (2018): doi: 10.1093/nar/gky691

[2]. Zhou, Wanding, Laird Peter L., and Hui Shen. "Comprehensive characterization, annotation and innovative use of Infinium DNA methylation BeadChip probes." Nucleic Acids Research. (2016): doi: 10.1093/nar/gkw967


# mRNA Analysis Pipeline

## Introduction
The GDC mRNA quantification analysis pipeline measures gene level expression with [STAR](https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf) as raw read counts. Subsequently the counts are augmented with several transformations including Fragments per Kilobase of transcript per Million mapped reads (FPKM), upper quartile normalized FPKM (FPKM-UQ), and Transcripts per Million (TPM). These values are additionally annotated with the gene symbol and gene bio-type. These data are generated through this pipeline by first aligning reads to the GRCh38 [reference genome](https://gdc.cancer.gov/download-gdc-reference-files) and then by quantifying the mapped reads. To facilitate harmonization across samples, all RNA-Seq reads are treated as unstranded during analyses.

## Data Processing Steps

### RNA-Seq Alignment Workflow
The mRNA Analysis pipeline begins with the [Alignment Workflow](/Data_Dictionary/viewer/#?view=table-definition-view&id=alignment_workflow), which is performed using a two-pass method with [STAR](https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf). STAR aligns each [read group](/Data_Dictionary/viewer/#?view=table-definition-view&id=read_group) separately and then merges the resulting alignments into one. Following the methods used by the International Cancer Genome Consortium [ICGC](https://icgc.org/) ([github](https://github.com/akahles/icgc_rnaseq_align)), the two-pass method includes a splice junction detection step, which is used to generate the final alignment. This workflow outputs a genomic BAM file, which contains both aligned and unaligned reads. Quality assessment is performed pre-alignment with [FASTQC](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/) and post-alignment with [Picard Tools](http://broadinstitute.github.io/picard/).

Files that were processed after Data Release 14 have associated transcriptomic and chimeric alignments in addition to the genomic alignment detailed above. This only applies to aliquots with at least one set of paired-end reads. The chimeric BAM file contains reads that were mapped to different chromosomes or strands (fusion alignments). The genomic alignment files contain chimeric and unaligned reads to facilitate the retrieval of all original reads. The transcriptomic alignment reports aligned reads with transcript coordinates rather than genomic coordinates. The transcriptomic alignment is also sorted differently to facilitate downstream analyses. BAM index file pairing is not supported by this method of sorting, which does not allow for BAM slicing on these alignments. The splice-junction file for these alignments are also available.

Files that were processed after Data Release 25 will have associated [gene fusion files](#fusion-pipelines).

As of Data Release 32 the reference annotation will be updated to GENCODE v36 and HT-Seq will no longer be used.

[![RNA Alignment Pipeline](images/RNA-Seq-DR32_Image.png)](images/RNA-Seq-DR32_Image.png "Click to see the full image.")

| I/O | Entity | Format |
|---|---|---|
| Input | [Submitted Unaligned Reads](/Data_Dictionary/viewer/#?view=table-definition-view&id=submitted_unaligned_reads) or [Submitted Aligned Reads](/Data_Dictionary/viewer/#?view=table-definition-view&id=submitted_aligned_reads) |  FASTQ or BAM |
| Output | [Aligned Reads](/Data_Dictionary/viewer/#?view=table-definition-view&id=aligned_reads) | BAM  |

### RNA-Seq Alignment Command Line Parameters

__Note that version numbers may vary in files downloaded from the GDC Data Portal due to ongoing pipeline development and improvement.__

=== "Original"

    ```shell
    # STAR-2

    ### For users with access to the ICGC pipeline:

    python star_align.py \
    --genomeDir <star_index_path> \
    --FastqFileIn <input_fastq_path> \
    --workDir <work_dir> \
    --out <output_bam> \
    --genomeFastaFiles <reference> \
    --runThreadN 8 \
    --outFilterMultimapScoreRange 1 \
    --outFilterMultimapNmax 20 \
    --outFilterMismatchNmax 10 \
    --alignIntronMax 500000 \
    --alignMatesGapMax 1000000 \
    --sjdbScore 2 \
    --limitBAMsortRAM 0 \
    --alignSJDBoverhangMin 1 \
    --genomeLoad NoSharedMemory \
    --outFilterMatchNminOverLread 0.33 \
    --outFilterScoreMinOverLread 0.33 \
    --twopass1readsN -1 \
    --sjdbOverhang 100 \
    --outSAMstrandField intronMotif \
    --outSAMunmapped Within

    ### For users without access to the ICGC pipeline:

    ### Step 1: Building the STAR index.*

    STAR
    --runMode genomeGenerate
    --genomeDir <star_index_path>
    --genomeFastaFiles <reference>
    --sjdbOverhang 100
    --sjdbGTFfile <gencode.v36.annotation.gtf>
    --runThreadN 8

    ### Step 2: Alignment 1st Pass.

    STAR
    --genomeDir <star_index_path>
    --readFilesIn <fastq_left_1>,<fastq_left2>,... <fastq_right_1>,<fastq_right_2>,...
    --runThreadN <runThreadN>
    --outFilterMultimapScoreRange 1
    --outFilterMultimapNmax 20
    --outFilterMismatchNmax 10
    --alignIntronMax 500000
    --alignMatesGapMax 1000000
    --sjdbScore 2
    --alignSJDBoverhangMin 1
    --genomeLoad NoSharedMemory
    --readFilesCommand <bzcat|cat|zcat>
    --outFilterMatchNminOverLread 0.33
    --outFilterScoreMinOverLread 0.33
    --sjdbOverhang 100
    --outSAMstrandField intronMotif
    --outSAMtype None
    --outSAMmode None

    ### Step 3: Intermediate Index Generation.

    STAR
    --runMode genomeGenerate
    --genomeDir <output_path>
    --genomeFastaFiles <reference>
    --sjdbOverhang 100
    --runThreadN <runThreadN>
    --sjdbFileChrStartEnd <SJ.out.tab from previous step>

    ### Step 4: Alignment 2nd Pass.

    STAR
    --genomeDir <output_path from previous step>
    --readFilesIn <fastq_left_1>,<fastq_left2>,... <fastq_right_1>,<fastq_right_2>,...
    --runThreadN <runThreadN>
    --outFilterMultimapScoreRange 1
    --outFilterMultimapNmax 20
    --outFilterMismatchNmax 10
    --alignIntronMax 500000
    --alignMatesGapMax 1000000
    --sjdbScore 2
    --alignSJDBoverhangMin 1
    --genomeLoad NoSharedMemory
    --limitBAMsortRAM 0
    --readFilesCommand <bzcat|cat|zcat>
    --outFilterMatchNminOverLread 0.33
    --outFilterScoreMinOverLread 0.33
    --sjdbOverhang 100
    --outSAMstrandField intronMotif
    --outSAMattributes NH HI NM MD AS XS
    --outSAMunmapped Within
    --outSAMtype BAM SortedByCoordinate
    --outSAMheaderHD @HD VN:1.4
    --outSAMattrRGline <formatted RG line provided by wrapper>
    ```

=== "DR15Plus"

    ```shell
    # STAR-2

    STAR \
    --readFilesIn <fastq_files> \
    --outSAMattrRGline <read_group_strings> \
    --alignIntronMax 1000000 \
    --alignIntronMin 20 \
    --alignMatesGapMax 1000000 \
    --alignSJDBoverhangMin 1 \
    --alignSJoverhangMin 8 \
    --alignSoftClipAtReferenceEnds Yes \
    --chimJunctionOverhangMin 15 \
    --chimMainSegmentMultNmax 1 \
    --chimOutType Junctions SeparateSAMold WithinBAM SoftClip \
    --chimSegmentMin 15 \
    --genomeDir <genome_dir> \
    --genomeLoad NoSharedMemory \
    --limitSjdbInsertNsj 1200000 \
    --outFileNamePrefix <output_prefix> \
    --outFilterIntronMotifs None \
    --outFilterMatchNminOverLread 0.33 \
    --outFilterMismatchNmax 999 \
    --outFilterMismatchNoverLmax 0.1 \
    --outFilterMultimapNmax 20 \
    --outFilterScoreMinOverLread 0.33 \
    --outFilterType BySJout \
    --outSAMattributes NH HI AS nM NM ch \
    --outSAMstrandField intronMotif \
    --outSAMtype BAM Unsorted \
    --outSAMunmapped Within \
    --quantMode TranscriptomeSAM GeneCounts \
    --readFilesCommand <zcat, etc> \
    --runThreadN <threads> \
    --twopassMode Basic
    ```

=== "DR32"

      ```shell
      # STAR Genome Index
      STAR
      --runMode genomeGenerate
      --genomeDir <star_index_path>
      --genomeFastaFiles <reference>
      --sjdbOverhang 100
      --sjdbGTFfile <gencode.v36.annotation.gtf>
      --runThreadN 8

      # STAR Alignment
      # STAR v2
      STAR
      --readFilesIn <fastq_files> \
      --outSAMattrRGline <read_group_strings> \
      --genomeDir <genome_dir> \
      --readFilesCommand <cat, zcat, etc> \
      --runThreadN <threads> \
      --twopassMode Basic \
      --outFilterMultimapNmax 20 \
      --alignSJoverhangMin 8 \
      --alignSJDBoverhangMin 1 \
      --outFilterMismatchNmax 999 \
      --outFilterMismatchNoverLmax 0.1 \
      --alignIntronMin 20 \
      --alignIntronMax 1000000 \
      --alignMatesGapMax 1000000 \
      --outFilterType BySJout \
      --outFilterScoreMinOverLread 0.33 \
      --outFilterMatchNminOverLread 0.33 \
      --limitSjdbInsertNsj 1200000 \
      --outFileNamePrefix <output prefix> \
      --outSAMstrandField intronMotif \
      --outFilterIntronMotifs None \
      --alignSoftClipAtReferenceEnds Yes \
      --quantMode TranscriptomeSAM GeneCounts \
      --outSAMtype BAM Unsorted \
      --outSAMunmapped Within \
      --genomeLoad NoSharedMemory \
      --chimSegmentMin 15 \
      --chimJunctionOverhangMin 15 \
      --chimOutType Junctions SeparateSAMold WithinBAM SoftClip \
      --chimOutJunctionFormat 1 \
      --chimMainSegmentMultNmax 1 \
      --outSAMattributes NH HI AS nM NM ch
      ```

\*These indices are available for download at the [GDC Website](https://gdc.cancer.gov/about-data/data-harmonization-and-generation/gdc-reference-files) and do not need to be built again.

### mRNA Expression Workflow

The primary counting data is generated by STAR and includes a gene ID, unstranded, and stranded counts data. Following alignment, the raw counts files produced by STAR are augmented with commonly used counts transformations (FPKM, FPKM-UQ, and TPM) along with basic annotations as part of the [RNA Expression Workflow](/Data_Dictionary/viewer/#?view=table-definition-view&id=rna_expression_workflow). These data are provided in a tab-delimited format. [GENCODE v36](https://www.gencodegenes.org/human/release_36.html) was used for gene annotation.

Note that the STAR counting results will not count reads that are mapped to more than one different gene. Below are two files that list genes that are completely encompassed by other genes and will likely display a value of zero.

* [Overlapped Genes (stranded)](/Data/Bioinformatics_Pipelines/overlap.gene.stranded.tsv)
* [Overlapped Genes (unstranded)](/Data/Bioinformatics_Pipelines/overlap.gene.strandless.tsv)

| I/O | Entity | Format |
|---|---|---|
| Input | [Aligned Reads](/Data_Dictionary/viewer/#?view=table-definition-view&id=aligned_reads) |  BAM |
| Output | [Gene Expression](/Data_Dictionary/viewer/#?view=table-definition-view&id=gene_expression) | TXT  |

### mRNA Quantification Command Line Parameters

HTSeq

=== "Current"

    ```
    Counts are produced by STAR concurrent with alignment.
    ```

=== "Original"

    ```shell
    htseq-count \
    -m intersection-nonempty \
    -i gene_id \
    -r pos \
    -s no \
    - gencode.v22.annotation.gtf
    ```

=== "DR15-31"

    ```shell
    htseq-count \
    -f bam \
    -r name \
    -s no \
    -a 10 \
    -t exon \
    -i gene_id \
    -m intersection-nonempty \
    <input_bam> \
    <gtf_file> > <counts_file>
    ```

## mRNA Expression Transformation

RNA-Seq expression level read counts produced by the workflow are normalized using three commonly used methods: FPKM, FPKM-UQ, and TPM. Normalized values should be used only within the context of the entire gene set. Users are encouraged to normalize raw read count values if a subset of genes is investigated.

### FPKM

The fragments per kilobase of transcript per million mapped reads (FPKM) calculation aims to control for transcript length and overall sequencing quantity.

### Upper Quartile FPKM

The upper quartile FPKM (FPKM-UQ) is a modified FPKM calculation in which the protein coding gene in the 75th percentile position is substituted for the sequencing quantity. This is thought to provide a more stable value than including the noisier genes at the extremes.

### TPM

The transcripts per million calculation is similar to FPKM, but the difference is that all transcripts are normalized for length first. Then, instead of using the total overall read count as a normalization for size, the sum of the length-normalized transcript values are used as an indicator of size.

### Calculations

[![FPKM Calculations](images/normalizations_calc.png)](images/normalizations_calc.png "Click to see the full image.")

__Note:__ The read count is multiplied by a scalar (10<sup>9</sup>) during normalization to account for the kilobase and 'million mapped reads' units.

### Examples

__Sample 1: Gene A__

* Gene length: 3,000 bp
* 1,000 reads mapped to Gene A
* 1,000,000 reads mapped to all protein-coding regions
* Read count in Sample 1 for 75th percentile gene: 2,000
* Number of protein coding genes on autosomes: 19,029
* Sum of length-normalized transcript counts: 9,000,000

__FPKM for Gene A__ = 1,000 \* 10^9 / (3,000 \* 50,000,000) = __6.67__

__FPKM-UQ for Gene A__ = 1,000) \* 10^9 / (3,000 \* 2,000 \* 19,029) = __8.76__

__TPM for Gene A__ = (1,000 * 1,000 / 3,000) * 1,000,000 / (9,000,000) = __37.04__

## Fusion Pipelines

The GDC uses two pipelines for the detection of gene fusions.

### STAR-Fusion Pipeline
The GDC gene fusion pipeline uses the STAR-Fusion v1.6 algorithm to generate gene fusion data.
STAR-Fusion pipeline processes the output generated by STAR aligner to map junction reads
and spanning reads to a junction annotation set. It utilizes a chimeric junction file from
running the STAR aligner and produces a tab-limited gene fusion prediction file.
The prediction file provides fused gene names, junction read count and breakpoint information.

### Arriba Fusion Pipeline

The [Arriba gene fusion pipeline](https://github.com/suhrig/arriba) uses Arriba v1.1.0 to detect gene fusions from the RNA-Seq data of tumor samples.


## scRNA-Seq Pipeline (single-nuclei)

The GDC processes single-cell RNA-Seq (scRNA-Seq) data using the [Cell Ranger pipeline](https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger) to calculate gene expression followed by [Seurat](https://satijalab.org/seurat/) for secondary expression analysis.

### scRNA Gene Expression Pipeline

The gene expression pipeline, which uses Cell Ranger, generates three files:

* Aligned reads file (BAM)
* Raw counts matrix - contains all barcodes in [Market Exchange Format](https://math.nist.gov/MatrixMarket/formats.html) (MEX)
* Filtered counts matrix - contains only detected cellular barcodes  (MEX)

### scRNA Analysis Pipeline

The analysis pipeline, which uses the Seurat software, generates three files from an input of Filtered counts matrix:

* Analysis - PCA, UMAP, tSNE values, and graph-based clustering results with associated metadata (TSV).
* Differential gene expression - DEG information comparing cells from one cluster to the rest of the cells (TSV).
* Full Seurat analysis log as a loom object in [HDF5](https://portal.hdfgroup.org/hdf5/v1_14/_u_g.html) format.

When the input RNA was extracted from nuclei instead of cytoplasm, a slightly modified quantification method is implemented to include introns. Currently, these single-nuclei RNA-Seq (snRNA-Seq) analyses share the same experimental strategy (scRNA-Seq) in the Data Portal, and can be filtered by querying for aliquot.analyte_type = "Nuclei RNA".

## File Access and Availability

To facilitate the use of harmonized data in user-created pipelines, RNA-Seq gene expression is accessible in the GDC Data Portal at several intermediate steps in the pipeline. Below is a description of each type of file available for download in the GDC Data Portal.

| Type | Description | Format |
|---|---|---|
| RNA-Seq Alignment | RNA-Seq reads that have been aligned to the GRCh38 build. Reads that were not aligned are included to facilitate the availability of raw read sets.  |  BAM |
| STAR Read Counts | The number of reads aligned to each gene, calculated by STAR, along with values using common normalization methods. |  TSV |


# Protein Expression - RPPA

## Introduction

**R**everse **P**hase **P**rotein **A**rray (RPPA) is a high-throughput antibody-based technique with a procedure similar to that of Western blots. In the procedure carried by [MD Anderson Cancer Center](https://www.mdanderson.org/), hundreds to thousands of different cell lysates are immobilized on a nitrocellulose-coated slide as many individual spots, followed by incubations with one protein-specific antibody, and detection. A group (often several hundreds) of antibodies form a *set*, which are used for each assay. Occasionally, antibodies may be added to or removed from the set depending on feasibility/functionality, which forms a new set.

To quantify protein expression, a "standard curve" is constructed from spots on each slide (one slide probed for one antibody). These spots include serial dilutions of each sample plus QC spots of standard lysates at different concentrations.

The technique is capable of the following types of analyses:

* Patient tumor classification
* DNA, RNA, and Protein correlation
* Prognosis
* Response prediction for targeted therapies
* Pharmacodynamics and biologically relevant dose
* Determination of appropriate handling procedures for clinical samples (based on antigen stability analysis)

## RPPA Data in the GDC

The antigens used for RPPA are available at ref [5], with the following information available:

* ```AGID```: The antigen unique ID
* ```peptide_target```: The unique ID for the target site that the antigen binds to
* ```gene_symbol```: The unique gene name abbreviation that codes the peptide
* ```antibody_origin```: The species that the antibody originated from
* ```source```: The antibody vendor company
* ```catalog_number```: Antibody vendor's catalog number
* ```validation_status```: Indicating how trustworthy those antibodies are, based on QC tests of antibody quality by the MD Anderson.

The GDC protein expression quantification data set is available in TSV format and contains ```AGID```,  ```catalog_number```, and ```peptide_target``` from the reference file, plus the following fields:

* ```lab_id```: The unique antibody ID
* ```set_id```: The ID for a set, ie list of antibodies (eg refs [3] & [4]).
* ```protein_expression```: Relative levels of protein expression - interpolation of each dilution curve to the "standard curve" (supercurve) of the slide (antibody).

## References

[1]. https://bioinformatics.mdanderson.org/public-software/tcpa/

[2]. https://www.mdanderson.org/research/research-resources/core-facilities/functional-proteomics-rppa-core/rppa-process.html

[3]. https://www.mdanderson.org/content/dam/mdanderson/documents/core-facilities/Functional%20Proteomics%20RPPA%20Core%20Facility/RPPA_Expanded_Ab_List_Updated.xlsx

[4]. https://www.mdanderson.org/content/dam/mdanderson/documents/core-facilities/Functional%20Proteomics%20RPPA%20Core%20Facility/RPPA_Standard_Ab_List_Updated.xlsx

[5]. https://gdc.cancer.gov/about-data/gdc-data-processing/gdc-reference-files


# GDC Data Model

## Introduction

The GDC Data Model is the central method of organization of all data artifacts in the GDC. An overview of the data model, including a visual representation of its components, is provided [on the GDC website](https://gdc.cancer.gov/developers/gdc-data-model). This section provides technical details about its implementation for data users, submitters, and developers.

## Entities, Properties, and Links

Although the GDC Data Model may contain some cyclic elements, it can be helpful to think of it as a [Directed Acyclic Graph (DAG)](https://en.wikipedia.org/wiki/Directed_acyclic_graph) composed of interconnected **entities**. Each entity in the GDC has a set of properties and links.

* **Properties** are key-value pairs associated with an entity. Properties cannot be nested, which means that the value must be numerical, boolean, or a string, and cannot be another key-value set. Properties can be either required or optional. The following properties are of particular importance in constructing the GDC Data Model:
    * **Type** is a required property for all entities. Entity types include `project`, `case`, `demographic`, `sample`, `read_group` and others.
    * **System properties** are properties used in GDC system operation and maintenance. They cannot be modified except under special circumstances.
    * **Unique keys** are properties, or combinations of properties, that can be used to uniquely identify the entity in the GDC. For example, the tuple (combination) of `[ project_id, submitter_id ]` is a unique key for most entities, which means that although `submitter_id` does not need to be unique in GDC, it must be unique within a project. See [GDC Identifiers](#gdc-identifiers) below for details.
* **Links** define relationships between entities, and the multiplicity of those relationships (e.g. one-to-one, one-to-many, many-to-many).

The [GDC Data Dictionary](/Data_Dictionary/) determines which properties and links an entity can have according to entity `type`.

Functionally similar entity types are grouped under the same **category**. For example, entity types `slide_image` and `submitted_unaligned_reads` belong to `data_file` category, which comprises entities that represent downloadable files.


## GDC Identifiers

### UUIDs

When an entity is created, it is assigned a unique identifier in the form of a [version 4 universally unique identifier (UUID)](https://en.wikipedia.org/wiki/Universally_unique_identifier). The UUID uniquely identifies the entity in the GDC, and is stored in the entity's `id` property.

### Program Name, Project Code, and Project ID

Programs are the highest level of organization of GDC datasets. Each program is assigned a unique `program.name` property. Datasets within a program are organized into projects, and each project is assigned a `project.code` property.

The `project_id` property is associated with most entities in the GDC data model and is generated by appending `project.code` to `program.name` as follows:

	program.name-project.code
	(e.g. TCGA-LAML)

Note that `program.name` never contains hyphens.

### Submitter ID

In addition to UUIDs stored in the `id` property, many entities also have a `submitter_id` property. This property can contain any string that the submitter wishes to use to identify the entity (e.g. a "barcode"). This can be used to identify a corresponding entry in the submitter's records. The GDC requires that `submitter_id` be unique for each entity within a project: the tuple (combination) of `[ project_id, submitter_id ]` is a unique key.

**Note:** The `submitter_id` of a `case` entity corresponds to the `submitted_subject_id` of the study participant in dbGaP records for the project.


## Working with the GDC Data Model

### Data Users

Users can access information stored in the GDC Data Model using the [GDC Data Portal](https://gdc.cancer.gov/access-data/gdc-data-portal), the [GDC API](https://gdc.cancer.gov/developers/gdc-application-programming-interface-api), and the [GDC Data Transfer Tool](https://gdc.cancer.gov/access-data/gdc-data-transfer-tool). For more information see [Data Access Processes and Tools](https://gdc.cancer.gov/access-data/data-access-processes-and-tools).

### Data Submitters

Data submitters can create and update submittable entities in the GDC Data Model and upload data files registered in the model using the [GDC Data Submission Portal](https://gdc.cancer.gov/submit-data/gdc-data-submission-portal), the [GDC API](https://gdc.cancer.gov/developers/gdc-application-programming-interface-api), and the [GDC Data Transfer Tool](https://gdc.cancer.gov/access-data/gdc-data-transfer-tool). For more information see [Data Submission Processes and Tools](https://gdc.cancer.gov/submit-data/data-submission-processes-and-tools).


# GDC Visualization Suite, User Acceptance Testing, April 2017

__URL: https://gdc-portal-staging.datacommons.io__

This guide describes the new data visualization features that can be accessed at the GDC Data Portal. Visualization features include detailed descriptions of mutations, genes, and their frequency; graphical representation of mutation positions, and dynamic survival analysis plots. Additionally, new API endpoints are available for users to programmatically retrieve the data used to generate the visualization features.

The mutation-based visualization features are derived from open-access MAF files that were produced by GDC variant-calling pipelines. The format of these MAF files was developed by and for the GDC and is outlined in the MAF Format documentation.

---

## GDC Portal Home Page

The GDC Portal home page is the entry way to accessing data in the Genomic Data Commons.

[![GDC Home Page](images/GDC-Home-Page.png)](images/GDC-Home-Page.png "Click to see the full image.")

The Genomic Data Commons can be queried by project or by primary site (i.e Brain). Data can be narrowed down in a few ways listed below.

* __Projects__: The projects link directs users to the Project List Page, which gives an overall summary of project-level information.

* __Repository__: The repository link allows users to see the data files available at the GDC and apply file/case filters to narrow down their search.

* __Human Outline__: The home page displays a human anatomical outline that can be used to refine their search. Choosing an associated organ will direct the user to a listing of all projects associated with that primary site. For example, clicking on the human brain will show only cases and projects associated with brain cancer (TCGA-GBM and TCGA-LGG).  The number of cases associated with each primary site is also displayed here and separated by project.

---

# Project List Page

The project list page displays statistics about the projects that are available at the GDC.

[![Project List Graphs](images/GDC-PL-Graphs.png)](images/GDC-PL-Graphs.png "Click to see the full image.")

## Top Mutated Genes in Selected Projects

This dynamically generated bar graph shows the ten genes with the most mutations across all projects. The bars represent the frequency of each mutation and is broken down into different colored segments by project and disease type. The graphic is updated as filters are applied for projects, programs, disease types, and data categories available in the project.

Hovering the cursor over each bar will display information about the number of cases affected by the disease type and clicking on each bar will bring the user to the Gene Summary Page page for the gene associated with the mutation.

## Case Distribution per Project

A pie graph displays the relative number of cases for each primary site (inner circle), which is further divided by project (outer circle). Hovering the cursor over each portion of the graph will display the primary site or project with the number of associated cases. Filtering projects at the left panel will update the pie chart.

## Project Table

[![Project Table](images/GDC-PL-Table.png)](images/GDC-PL-Table.png "Click to see the full image.")

General information about each project is displayed below the graphs. The numbers of available cases per data category for each project are displayed as links. Clicking on a specific project will bring a user to that project's Project Detail Page, which provides an overview of all cases, files and annotations available for the project.

---
# Project Detail Page

Each project has a detail page that provides an overview of all available cases, files, and annotations available. Clicking on the numbers in the summary table will display the corresponding data.

[![Project Detail Page](images/gdc-project-entity-page.png)](images/gdc-project-entity-page.png "Click to see the full image.")

Three download buttons in the top right corner of the screen allow the user to download the entire project dataset, along with the associated project metadata:

* __Download Manifest__: Downloads a manifest for all data files available in the project. The manifest can be used with the GDC Data Transfer Tool to download the files.
* __Download Clinical__: Downloads clinical metadata about all cases in the project.
* __Download Biospecimen__: Downloads biospecimen metadata associated with all cases in the project.

## Most Frequently Mutated Genes

The project detail page also reports the genes that have somatic mutations in the greatest numbers of cases in a graphical and tabular format.

[![Frequently Mutated Genes](images/gdc-frequently-mutated-genes.png)](images/gdc-frequently-mutated-genes.png "Click to see the full image.")

The top of this section contains a survival plot of all the cases within the specified project, in additiona to a bar graph of the most frequently mutated genes. Hovering over each bar in the plot will display information about the number of cases affected. Users may choose to download the underlying data in JSON or TSV format or an image of the graph in SVG or PNG format by clicking the `download` icon at the top of each graph.

Below these graphs is a tabular view of the genes affected, which includes the following information:

* __Symbol:__ The gene symbol, which links to the Gene Summary Page
* __Name:__ Full name of the gene
* __Cytoband:__ The location of the mutation on the chromosome in terms of Giemsa-stained samples.
* __# Affected Cases in Project:__ The number of cases within the project that contain a mutation on this gene
* __# Affected Cases Across all Projects:__ The number of cases within all the projects in the GDC that contain a mutation on this gene.  Clicking the red arrow will display the cases broken down by project
* __# Mutations:__ The number of SSMs (simple somatic mutations) detected in that gene
* __Survival Analysis:__ An icon that, when clicked, will plot the survival rate between cases in the project with mutated and non-mutated forms of the gene

### Survival Analysis

Survival analysis is used to analyze the occurrence of event data over time.  In the GDC, survival analysis is performed on the mortality of the cases. Survival analysis requires:

*  Data on the time to a particular event (days to death or last follow up)
    * Fields:  __diagnoses.days_to_death__ and __diagnoses.days_to_last_follow_up__
*  Information on whether the event has occurred (alive/deceased)
    * Fields:  __diagnoses.vital_status__
*  Data split into different categories or groups (i.e. gender, etc.)
    * Fields:  __demographic.gender__

The survival analysis in the GDC uses a Kaplan-Meier estimator:

[![Kaplan-Meier Estimator](images/gdc-kaplan-meier-estimator.png)](images/gdc-kaplan-meier-estimator "Click to see the full image.")

Where:

 * S(t<sub>i</sub>) is the estimated survival probability for any particular one of the t time periods
 * n<sub>i</sub> is the number of subjects at risk at the beginning of time period t<sub>i</sub>
 * and d<sub>i</sub> is the number of subjects who die during time period t<sub>i</sub>

The table below is an example data set to calculate survival for a set of seven cases:

[![Sample Survival Analysis Table](images/gdc-sample-survival-table.png)](images/gdc-sample-survival-table.png "Click to see the full image.")

The calculated cumulated survival probability can be plotted against the interval to obtain a survival plot like the one shown below.

[![Sample Survival Analysis Plot](images/gdc-survival-plot.png)](images/gdc-survival-plot.png "Click to see the full image.")


## OncoGrid

The project detail page includes an OncoGrid plot of the cases with the most mutations, for the top 50 mutated genes affected by high impact mutations. Genes displayed on the left of the grid (Y-axis) correspond to individual cases on the bottom of the grid (X-axis).

[![Oncogrid](images/gdc-oncogrid.png)](images/gdc-oncogrid.png "Click to see the full image.")

The grid is color-coded with a legend at the top left which describes what type of mutation consequence is observed for each gene/case combination. Clinical information and the available data for each case are available at the bottom of the grid.

The right side of the grid displays additional information about the genes:

* __Gene Sets:__ Describes whether a gene is part of the [Gene Census](http://cancer.sanger.ac.uk/census/).  (The cancer Gene Census is an ongoing effort to catalogue those genes for which mutations have been causally implicated in cancer)
* __GDC:__ Heat-map of all cases in the GDC affected with a mutation in this gene

### OncoGrid Options

To facilitate readability and comparisons, drag-and-drop can be used to reorder the gene rows.  Double clicking a row in the "# Cases Affected" bar at the right side of the graphic will bring the user to the respective Gene Summary Page page. Hovering over a cell will display information about the mutation such as its ID, affected case, and biological consequence. Clicking on the cell will bring the user to the respective Mutation Summary page.

A tool bar at the top right of the graphic allows the user to export the data as a JSON object, PNG image, or SVG image.  Seven buttons are available in this toolbar:

* __Download:__ Users can choose to export the contents either to a static image file (PNG or SVG format) or the underlying data in JSON format
* __Refresh:__ Sets all OncoGrid rows, columns, and zoom levels back to their initial positions
* __Cluster Data:__ Clusters the rows and columns to place mutated genes with the same cases and cases with the same mutated genes together
* __Toggle Heatmap:__ The view can be toggled between cells representing mutation consequences or number of mutations in each gene
* __Toggle Gridlines:__ Turn the gridlines on and off
* __Toggle Crosshairs:__ Turns crosshairs on, so that users can zoom into specific sections of the OncoGrid
* __Fullscreen:__ Turns Fullscreen mode on/off

## Most Frequent Mutations
The project detail page also displays the 20 most frequent mutations in the project as a bar graph that indicates the number of cases that are affected by each mutation. Hovering over each bar in the plot will show information about the number of cases affected.

[![Most Frequent Mutations](images/gdc-frequent-mutations.png)](images/gdc-frequent-mutations.png "Click to see the full image.")

A table is displayed below that lists information about each mutation:

* __ID:__ A UUID for the mutation assigned by the GDC, when clicked will bring a user to the Mutation Summary Page
* __DNA Change:__ The chromosome and starting coordinates of the mutation are displayed along with the nucleotide differences between the reference and tumor allele
* __Type:__ A general classification of the mutation
* __Consequences:__ The effects the mutation has on the gene coding for a protein (i.e. synonymous, missense, non-coding transcript).  A link to the Gene Summary Page for the gene affected by the mutation is included
* __# Affected Cases in Project:__ The number of affected cases in the project
* __# Affected Cases in Across all Projects:__ The number of affected cases, expressed as number across all projects. Choosing the arrow next to the percentage will display a breakdown of each affected project
* __Impact:__ A subjective classification of the severity of the variant consequence. The categories are:
    * __HIGH__: The variant is assumed to have high (disruptive) impact in the protein, probably causing protein truncation, loss of function, or triggering nonsense mediated decay
    * __MODERATE__: A non-disruptive variant that might change protein effectiveness
    * __LOW__: Assumed to be mostly harmless or unlikely to change protein behavior
* __Survival Analysis:__ An icon that when clicked, will plot the survival rate between the gene's mutated and non-mutated cases

## Most Affected Cases

The final section of the project detail page is a display of the top 20 cases in a specified project, with the greatest number of affected genes.

[![Most Affected Cases](images/gdc-most-affected-cases.png)](images/gdc-most-affected-cases.png "Click to see the full image.")

Below the bar graph is a table contains information about these cases:

* __UUID:__ The UUID of the case, which links to the Case Summary Page
* __Site:__  The anatomical location of the site affected
* __Gender:__ Text designations that identify gender. Gender is described as the assemblage of properties that distinguish people on the basis of their societal roles
* __Age at Diagnosis:__ Age at the time of diagnosis expressed in number of days since birth
* __Stage:__ The extent of a cancer in the body. Staging is usually based on the size of the tumor, whether lymph nodes contain cancer, and whether the cancer has spread from the original site to other parts of the body. The accepted values for tumor_stage depend on the tumor site, type, and accepted staging system
* __Survival:__ The number of days until death
* __Last Follow Up:__ Time interval from the date of last follow up to the date of initial pathologic diagnosis, represented as a calculated number of days
* __Available Files per Data Category:__ Five columns displaying the number of files available in each of the five data categories.  These link to the files for the specific case.
* __Genes:__ The number of genes affected by mutations for the case

---
# Case Summary Page

The Case Summary Page displays case details including the project and disease information, data files that are available for that case, and the experimental strategies employed. A button in the top-right corner of the page allows the user to add all files associated with the case to the file cart.

[![Case Page](images/gdc-case-entity-page.png)](images/gdc-case-entity-page.png "Click to see the full image.")

## Clinical and Biospecimen Information

The page also provides clinical and biospecimen information about that case. Links to export clinical and biospecimen information in JSON format are provided.

[![Case Page, Clinical and Biospecimen](images/gdc-case-clinical-biospecimen.png)](images/gdc-case-clinical-biospecimen.png "Click to see the full image.")

For clinical records that support multiple records of the same type (Diagnoses, Family Histories, or Exposures), a UUID of the record is provided on the left hand side of the corresponding tab, allowing the user to select the entry of interest.

### Biospecimen Search

A search filter just below the biospecimen section can be used to find and filter biospecimen data. The wildcard search will highlight entities in the tree that match the characters typed. This will search both the case submitter ID, as well as the additional metadata for each entity. For example, searching 'Primary Tumor' will highlight samples that match that type.

[![Biospecimen Search](images/gdc-case-biospecimen-search.png)](images/gdc-case-biospecimen-search.png "Click to see the full image.")

---
# Gene Summary Page

The Gene Summary Page describes each gene with mutation data featured at the GDC and provides results related to the analyses that are performed on these genes.

## Summary

The summary section of the gene page contains the following information:

[![Gene Summary](images/GDC-Gene-Summary.png)](images/GDC-Gene-Summary.png "Click to see the full image.")

* __Symbol:__ The gene symbol
* __Name:__ Full name of the gene
* __Synonyms:__ Synonyms of the gene name or symbol, if available
* __Type:__ A broad classification of the gene
* __Location:__ The chromosome on which the gene is located and its coordinates
* __Strand:__ If the gene is located on the forward (+) or reverse (-) strand
* __Description:__ A description of gene function and downstream consequences of gene alteration

## External References

A list with links that lead to external databases with additional information about each gene is displayed here. These external databases include: [Entrez](https://www.ncbi.nlm.nih.gov/gquery/), [Hugo Gene Nomenclature Committee](http://www.genenames.org/), [Online Mendelian Inheritance in Man](https://www.omim.org/), and [Uniprotkb SwissProt](http://www.uniprot.org/).

## Cancer Distribution

[![Cancer Distribution](images/GDC-Gene-CancerDist.png)](images/GDC-Gene-CancerDist.png "Click to see the full image.")

A table and bar graph displayed that shows how many cases are affected by mutations within the gene as a ratio and percentage. Each row/bar represents the number of cases for each project.

## Protein Viewer

[![Protein Plot](images/GDC-Gene-ProteinGraph.png)](images/GDC-Gene-ProteinGraph.png "Click to see the full image.")

Mutations and their frequency across cases are mapped to a graphical visualization of protein-coding regions with a lollipop plot. Pfam domains are highlighted along the x-axis to assign functionality to specific protein-coding regions. The bottom track represents a view of the full gene length. Different transcripts can be selected by using the drop-down menu above the plot.

The panel to the right of the plot allows the plot to be filtered by mutation consequences or impact.  The plot will dynamically change as filters are applied.  Mutation consequence and impact is denoted in the plot by color.

The plot can be viewed at different zoom levels by clicking and dragging across the x-axis, clicking and dragging across the bottom track, or double clicking the pfam domain IDs. The `Reset` button can be used to bring the zoom level back to its original position. The plot can also be exported as a PNG image, SVG image or as JSON formatted text by choosing the `Download` button above the plot.

## Most Frequent Mutations

The 20 most frequent mutations in the gene are displayed as a bar graph that indicates the number of cases that share each mutation.

[![Gene MFM](images/GDC-Gene-MFM.png)](images/GDC-Gene-MFM.png "Click to see the full image.")

A table is displayed below that lists information about each mutation including:

* __ID:__ A UUID Code for the mutation assigned by the GDC, when clicked will bring a user to the Mutation Summary Page
* __DNA Change:__ The chromosome and starting coordinates of the mutation are displayed along with the nucleotide differences between the reference and tumor allele
* __Type:__ A general classification of the mutation
* __Consequences:__ The effects the mutation has on the gene coding for a protein (i.e. synonymous, missense, non-coding transcript)
* __# Affected Cases:__ The number of affected cases, expressed as number across all projects. Choosing the arrow next to the percentage will expand the selection with a breakdown of each affected project
* __Impact:__ A subjective classification of the severity of the variant consequence. The categories are:
  - __HIGH__: The variant is assumed to have high (disruptive) impact in the protein, probably causing protein truncation, loss of function or triggering nonsense mediated decay
  - __MODERATE__: A non-disruptive variant that might change protein effectiveness
  - __LOW__: Assumed to be mostly harmless or unlikely to change protein behavior

---
# Mutation Summary Page

The Mutation Summary Page contains information about one somatic mutation and how it affects the associated gene. Each mutation is identified by its chromosomal position and nucleotide-level change.

## Summary

[![Mutation Summary](images/GDC-Mutation-Summary.png)](images/GDC-Mutation-Summary.png "Click to see the full image.")

- __ID:__ A unique identifier (UUID) for this mutation
- __DNA Change:__ Denotes the chromosome number, position, and nucleotide change of the mutation
- __Type:__ A broad categorization of the mutation
- __Reference Genome Assembly:__ The reference genome in which the chromosomal position refers to
- __Allele in the Reference Assembly:__ The nucleotide(s) that compose the site in the reference assembly
- __Functional Impact:__ A subjective classification of the severity of the variant consequence. The categories are:
  - __HIGH__: The variant is assumed to have high (disruptive) impact in the protein, probably causing protein truncation, loss of function or triggering nonsense mediated decay
  - __MODERATE__: A non-disruptive variant that might change protein effectiveness
  - __LOW__: Assumed to be mostly harmless or unlikely to change protein behavior

### External References

A separate panel contains links to information about the gene that the mutation effects in external databases. These databases include: [Entrez](https://www.ncbi.nlm.nih.gov/gquery/), [Hugo Gene Nomenclature Committee](http://www.genenames.org/), [Online Mendelian Inheritance in Man](https://www.omim.org/), and [Uniprotkb SwissProt](http://www.uniprot.org/).

## Consequences

[![Mutation Consequences](images/GDC-Mutation-Consequences.png)](images/GDC-Mutation-Consequences.png "Click to see the full image.")

The consequences of the mutation are displayed in a table. The fields that detail each mutation are listed below:

* __Gene:__ The symbol for the affected gene
* __AA Change:__ Details on the amino acid change, including compounds and position, if applicable
* __Consequence:__ The biological consequence of each mutation
* __Coding DNA Change:__ The specific nucleotide change and position of the mutation within the gene
* __Strand:__ If the gene is located on the forward (+) or reverse (-) strand
* __Transcript(s):__ The transcript(s) affected by the mutation. Each contains a link to the [Ensembl](https://www.ensembl.org) entry for the transcript

---
## Cancer Distribution

[![Mutation Distribution](images/GDC-Mutation-CancerDist.png)](images/GDC-Mutation-CancerDist.png "Click to see the full image.")

A bar graph detailing the number of cases affected by the mutation across all projects is generated dynamically.  Below the graph is a table with information about how the mutation affects each project, which can be exported as a JSON object. The table contains the following fields:

* __Project ID__: The ID for a specific project
* __Disease Type__: The disease associated with the project
* __Site__: The anatomical site affected by the disease
* __# Affected Cases__: The number of affected cases and total number of cases displayed as a fraction and percentage

## Protein Viewer

[![Mutation Protein Graph](images/GDC-Mutation-ProteinGraph.png)](images/GDC-Mutation-ProteinGraph.png "Click to see the full image.")

The protein viewer displays a plot representing the position of mutations along the polypeptide chain associated with the mutation. The y-axis represents the number of cases that exhibit each mutation, whereas the x-axis represents the polypeptide chain sequence. [Pfam domains](http://pfam.xfam.org/) that were identified along the polypeptide chain are identified with colored rectangles labeled with pfam IDs. See the Gene Summary Page for additional details about the protein viewer.

---
# Visualization API Endpoint Additions

The GDC Visualization Suite uses the same API as the rest of the Data Portal and takes advantage of several new endpoints:

* __ssms:__ The simple somatic mutation (`ssms`) endpoint allows users to access information about each somatic point mutation. For example, a `ssm` would represent the transition of C to T at position 52000 of chromosome 1.
* __ssm_occurrences:__ A SSM entity as applied to a single instance (case). An example of a `ssm occurrence` would be that the transition of C to T at position 52000 of chromosome 1 occurred in patient TCGA-XX-XXXX.
* __genes:__ The `genes` endpoint allows users to access in-depth information about each gene.

The methods for retrieving information from these endpoint are very similar to those used for the `cases` and `files` endpoints. These methods are explored in depth in the [API Search and Retrieval](https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/) documentation. The `_mapping` parameter can also be used with each of these endpoints to generate a list of potential fields.  For example:

`https://api.gdc.cancer.gov/ssms/_mapping`

While it is not an endpoint, the `observation` entity is featured in the visualization section of the API. The `observation` entity provides information from the MAF file, such as read depth and normal genotype, that supports the validity of the associated `ssm`.

## Analysis Endpoints

In addition the `ssms`, `ssm_occurrences`, and `genes` endpoints mentioned previously, several `analysis` endpoints were designed to quickly retrieve specific datasets used for visualization display.  

* __analysis/survival__
* __analysis/top_cases_counts_by_genes__
* __analysis/top_mutated_genes_by_project__
* __analysis/top_mutated_cases_by_gene__
* __analysis/top_mutated_cases_by_ssm__
* __analysis/mutated_cases_count_by_project__

For a set of API query examples, go to:  

https://github.com/NCIP/gdc-docs/blob/Visualization-Docs/docs/Visualization_Documentation/API_Endpoints.md

---
# GDC MAF Format v.1.0.0

## Introduction

Mutation Annotation Format (MAF) is a tab-delimited text file with aggregated mutation information from [VCF Files](VCF_Format.md) and are generated on a project-level. MAF files are produced through the [Somatic Aggregation Workflow](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=somatic_aggregation_workflow&_top=1)  The GDC produces MAF files at two permission levels: __protected__ and __somatic__ (or open-access). One MAF files is produced per variant calling pipeline per GDC project. MAFs are produced by aggregating the GDC annotated VCF files generated from one pipeline for one project.

Annotated VCF files often have variants reported on multiple transcripts whereas the MAF files generated from the VCFs (\*protected.maf) only report the most critically affected one. Somatic MAFs (\*somatic.maf), which are also known as [Masked Somatic Mutation](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/#?view=table-definition-view&id=masked_somatic_mutation) files, are further processed to remove lower quality and potential germline variants. For tumor samples that contain variants from multiple combinations of tumor-normal aliquot pairs, only one pair is selected in the Somatic MAF based on their sample type. Somatic MAFs are publicly available and can be freely distributed within the boundaries of the [GDC Data Access Policies](https://gdc.cancer.gov/access-data/data-access-policies).

The GDC MAF file format is based on the [TCGA Mutation Annotation Format](https://wiki.nci.nih.gov/display/TCGA/Mutation+Annotation+Format+(MAF)+Specification) specifications, with additional columns included.

__Note:__ The criteria for allowing mutations into open-access are purposefully implemented to overcompensate and filter out germline variants. If omission of true-positive somatic mutations is a concern, the GDC recommends using protected MAFs.  

## Somatic MAF File Generation

The process for modifying a protected MAF into a somatic MAF is as follows:

*  Aliquot Selection: only one tumor-normal pair are selected for each tumor sample based on the plate number, sample type, analyte type and other features extracted from tumor TCGA aliquot barcode.
*  Low quality variant filtering and germline masking:
    1. Variants with __Mutation_Status != 'Somatic'__ or __GDC_FILTER = 'Gapfiller', 'ContEst', 'multiallelic', 'nonselectedaliquot', 'BCR_Duplicate' or 'BadSeq'__ are __removed__.
    2. Remaining variants with __GDC_Valid_Somatic = True__ are __included__ in the Somatic MAF.
    3. Remaining variants with __FILTER != 'panel_of_normals' or PASS__ are __removed__. Note that the `FILTER != panel_of_normals` value is only relevant for the variants generated from the MuTect2 pipeline.
    4. Remaining variants with __MC3_Overlap = True__ are __included__ in the Somatic MAF.
    5. Remaining variants with __GDC_FILTER = 'ndp', 'NonExonic', 'bitgt', or 'gdc_pon'__ are __removed__.
    6. Remaining variants with __SOMATIC != null__ are __included__ in the Somatic MAF.
    7. Remaining variants with __dbSNP_RS = 'novel' or null__ are __included__ in the Somatic MAF.
    8. Remaining variants are __removed__.
* Removal of the following columns:
    * vcf_info
    * vcf_format
    * vcf_tumor_gt
    * vcf_normal_gt
    * GDC_Valid_Somatic
* Set values to be blank in the following columns that may contain information about germline genotypes:
    * Match_Norm_Seq_Allele1
    * Match_Norm_Seq_Allele2
    * Match_Norm_Validation_Allele1
    * Match_Norm_Validation_Allele2
    * n_ref_count
    * n_alt_count

---

![Somatic MAF Generation](images/ProtectedMAF4.png)

---
---

## Protected MAF File Structure

The table below describes the columns in a protected MAF and their definitions. Note that the somatic (open-access) MAF structure is the same except for having the last six columns removed.

| Column | Description |
|---|---|
| 1 - Hugo_Symbol | [HUGO](http://www.genenames.org/) symbol for the gene (HUGO symbols are always in all caps). "Unknown" is used for regions that do not correspond to a gene  |
| 2 - Entrez_Gene_Id | [Entrez gene](https://www.ncbi.nlm.nih.gov/gene) ID (an integer).  "0" is used for regions that do not correspond to a gene region or Ensembl ID |
| 3 - Center| One or more genome sequencing center reporting the variant |
| 4 - NCBI_Build| The reference genome used for the alignment (GRCh38) |
| 5 - Chromosome | The affected chromosome (chr1) |
| 6 - Start_Position | Lowest numeric position of the reported variant on the genomic reference sequence. Mutation start coordinate |
| 7 - End_Position | Highest numeric genomic position of the reported variant on the genomic reference sequence. Mutation end coordinate |
| 8 - Strand | Genomic strand of the reported allele. Currently, all variants will report the positive strand: '+' |
| 9 - Variant_Classification | Translational effect of variant allele  |
| 10 - Variant_Type | Type of mutation. TNP (tri-nucleotide polymorphism) is analogous to DNP (di-nucleotide polymorphism) but for three consecutive nucleotides. ONP (oligo-nucleotide polymorphism) is analogous to TNP but for consecutive runs of four or more (SNP, DNP, TNP, ONP, INS, DEL, or Consolidated)|
| 11 - Reference_Allele | The plus strand reference allele at this position. Includes the deleted sequence for a deletion or "-" for an insertion |
| 12 - Tumor_Seq_Allele1 | Primary data genotype for tumor sequencing (discovery) allele 1. A "-" symbol for a deletion represents a variant. A "-" symbol for an insertion represents wild-type allele. Novel inserted sequence for insertion does not include flanking reference bases  |
| 13 - Tumor_Seq_Allele2 | Tumor sequencing (discovery) allele 2  |
| 14 - dbSNP_RS| The rs-IDs from the [dbSNP](https://www.ncbi.nlm.nih.gov/projects/SNP/) database, "novel" if not found in any database used, or null if there is no dbSNP record, but it is found in other databases|
| 15 - dbSNP_Val_Status | The dbSNP validation status is reported as a semicolon-separated list of statuses. The union of all rs-IDs is taken when there are multiple |
| 16 - Tumor_Sample_Barcode| Aliquot barcode for the tumor sample |
| 17 - Matched_Norm_Sample_Barcode| Aliquot barcode for the matched normal sample |
| 18 - Match_Norm_Seq_Allele1 | Primary data genotype. Matched normal sequencing allele 1. A "-" symbol for a deletion represents a variant. A "-" symbol for an insertion represents wild-type allele. Novel inserted sequence for insertion does not include flanking reference bases (cleared in somatic MAF) |
| 19 - Match_Norm_Seq_Allele2| Matched normal sequencing allele 2 |
| 20 - Tumor_Validation_Allele1| Secondary data from orthogonal technology. Tumor genotyping (validation) for allele 1. A "-" symbol for a deletion represents a variant. A "-" symbol for an insertion represents wild-type allele. Novel inserted sequence for insertion does not include flanking reference bases  |
| 21 - Tumor_Validation_Allele2 | Secondary data from orthogonal technology. Tumor genotyping (validation) for allele 2 |
| 22 - Match_Norm_Validation_Allele1 | Secondary data from orthogonal technology. Matched normal genotyping (validation) for allele 1. A "-" symbol for a deletion represents a variant. A "-" symbol for an insertion represents wild-type allele. Novel inserted sequence for insertion does not include flanking reference bases (cleared in somatic MAF) |
| 23 - Match_Norm_Validation_Allele2 | Secondary data from orthogonal technology. Matched normal genotyping (validation) for allele 2 (cleared in somatic MAF) |
| 24 - Verification_Status| Second pass results from independent attempt using same methods as primary data source. Generally reserved for 3730 Sanger Sequencing |
| 25 - Validation_Status | Second pass results from orthogonal technology |
| 26 - Mutation_Status | An assessment of the mutation as somatic, germline, LOH, post transcriptional modification, unknown, or none. The values allowed in this field are constrained by the value in the Validation_Status field |
| 27 - Sequencing_Phase | TCGA sequencing phase (if applicable). Phase should change under any circumstance that the targets under consideration change |
| 28 - Sequence_Source | Molecular assay type used to produce the analytes used for sequencing. Allowed values are a subset of the SRA 1.5 library_strategy field values. This subset matches those used at CGHub |
| 29 - Validation_Method | The assay platforms used for the validation call |
| 30 - Score | Not in use |
| 31 - BAM_File | Not in use |
| 32 - Sequencer | Instrument used to produce primary sequence data |
| 33 - Tumor_Sample_UUID | GDC aliquot UUID for tumor sample |
| 34 - Matched_Norm_Sample_UUID | GDC aliquot UUID for matched normal sample |
| 35 - HGVSc | The coding sequence of the variant in HGVS recommended format |
| 36 - HGVSp | The protein sequence of the variant in HGVS recommended format. "p.=" signifies no change in the protein |
| 37 - HGVSp_Short | Same as the HGVSp column, but using 1-letter amino-acid codes |
| 38 - Transcript_ID| [Ensembl](http://useast.ensembl.org/index.html) ID of the transcript affected by the variant |
| 39 - Exon_Number | The exon number (out of total number) |
| 40 - t_depth | Read depth across this locus in tumor BAM |
| 41 - t_ref_count | Read depth supporting the reference allele in tumor BAM |
| 42 - t_alt_count | Read depth supporting the variant allele in tumor BAM |
| 43 - n_depth | Read depth across this locus in normal BAM |
| 44 - n_ref_count | Read depth supporting the reference allele in normal BAM (cleared in somatic MAF) |
| 45 - n_alt_count | Read depth supporting the variant allele in normal BAM (cleared in somatic MAF) |
| 46 - all_effects | A semicolon delimited list of all possible variant effects, sorted by priority ([SYMBOL,Consequence,HGVSp_Short,Transcript_ID,RefSeq]) |
|47 - Allele| The variant allele used to calculate the consequence |
|48 - Gene|Stable Ensembl ID of affected gene |
|49 - Feature|Stable Ensembl ID of feature (transcript, regulatory, motif) |
|50 - Feature_type| Type of feature. Currently one of Transcript, RegulatoryFeature, MotifFeature (or blank) |
|51 - Consequence | Consequence type of this variation; [sequence ontology](http://www.sequenceontology.org/) terms |
|52 - cDNA_position | Relative position of base pair in the cDNA sequence as a fraction. A "-" symbol is displayed as the numerator if the variant does not appear in cDNA |
|53 - CDS_position | Relative position of base pair in coding sequence. A "-" symbol is displayed as the numerator if the variant does not appear in coding sequence |
|54 - Protein_position | Relative position of affected amino acid in protein. A "-" symbol is displayed as the numerator if the variant does not appear in coding sequence |
|55 - Amino_acids | Only given if the variation affects the protein-coding sequence |
|56 - Codons | The alternative codons with the variant base in upper case |
|57 - Existing_variation | Known identifier of existing variation |
|58 - ALLELE_NUM | Allele number from input; 0 is reference, 1 is first alternate etc. |
|59 - DISTANCE | Shortest distance from the variant to transcript |
|60 - TRANSCRIPT_STRAND | The DNA strand (1 or -1) on which the transcript/feature lies |
|61 - SYMBOL | The gene symbol |
|62 - SYMBOL_SOURCE | The source of the gene symbol |
|63 - HGNC_ID | Gene identifier from the HUGO Gene Nomenclature Committee if applicable |
|64 - BIOTYPE | Biotype of transcript |
|65 - CANONICAL | A flag (YES) indicating that the VEP-based canonical transcript, the longest translation, was used for this gene. If not, the value is null |
|66 - CCDS | The [CCDS](https://www.ncbi.nlm.nih.gov/projects/CCDS/CcdsBrowse.cgi) identifier for this transcript, where applicable |
|67 - ENSP | The Ensembl protein identifier of the affected transcript |
|68 - SWISSPROT | [UniProtKB/Swiss-Prot](http://www.uniprot.org/) accession |
|69 - TREMBL | UniProtKB/TrEMBL identifier of protein product |
|70 - UNIPARC | UniParc identifier of protein product |
|71 - RefSeq | RefSeq identifier for this transcript |
|72 - SIFT | The [SIFT](http://sift.jcvi.org/) prediction and/or score, with both given as prediction (score) |
|73 - PolyPhen | The [PolyPhen](http://genetics.bwh.harvard.edu/pph2/) prediction and/or score |
|74 - EXON | The exon number (out of total number) |
|75 - INTRON | The intron number (out of total number) |
|76 - DOMAINS | The source and identifier of any overlapping protein domains |
|77 - GMAF | Non-reference allele and frequency of existing variant in [1000 Genomes](http://www.internationalgenome.org/) |
|78 - AFR_MAF | Non-reference allele and frequency of existing variant in 1000 Genomes combined African population |
|79 - AMR_MAF | Non-reference allele and frequency of existing variant in 1000 Genomes combined American population |
|80 - ASN_MAF | Non-reference allele and frequency of existing variant in 1000 Genomes combined Asian population |
|81 - EAS_MAF | Non-reference allele and frequency of existing variant in 1000 Genomes combined East Asian population |
|82 - EUR_MAF | Non-reference allele and frequency of existing variant in 1000 Genomes combined European population |
|83 - SAS_MAF | Non-reference allele and frequency of existing variant in 1000 Genomes combined South Asian population |
|84 - AA_MAF | Non-reference allele and frequency of existing variant in [NHLBI-ESP](http://evs.gs.washington.edu/EVS/) African American population |
|85 - EA_MAF | Non-reference allele and frequency of existing variant in NHLBI-ESP European American population|
|86 - CLIN_SIG | Clinical significance of variant from dbSNP |
|87 - SOMATIC |Somatic status of each ID reported under Existing_variation (0, 1, or null) |
|88 - PUBMED | Pubmed ID(s) of publications that cite existing variant |
|89 - MOTIF_NAME |The source and identifier of a transcription factor binding profile aligned at this position |
|90 - MOTIF_POS | The relative position of the variation in the aligned TFBP |
|91 - HIGH_INF_POS | A flag indicating if the variant falls in a high information position of a transcription factor binding profile (TFBP) (Y, N, or null) |
| 92 - MOTIF_SCORE_CHANGE | The difference in motif score of the reference and variant sequences for the TFBP |
| 93 - IMPACT | The impact modifier for the consequence type |
| 94 - PICK | Indicates if this block of consequence data was picked by VEP's [pick feature](http://useast.ensembl.org/info/docs/tools/vep/script/vep_options.html#opt_pick) (1 or null) |
| 95 - VARIANT_CLASS | Sequence Ontology variant class |
| 96 - TSL | [Transcript support level](http://useast.ensembl.org/Help/Glossary?id=492), which is based on independent RNA analyses |
| 97 - HGVS_OFFSET | Indicates by how many bases the HGVS notations for this variant have been shifted |
| 98 - PHENO|Indicates if existing variant is associated with a phenotype, disease or trait (0, 1, or null) |
| 99 - MINIMISED | Alleles in this variant have been converted to minimal representation before consequence calculation (1 or null) |
| 100 - ExAC_AF | Global Allele Frequency from [ExAC](http://exac.broadinstitute.org/) |
| 101 - ExAC_AF_Adj | Adjusted Global Allele Frequency from ExAC |
| 102 - ExAC_AF_AFR | African/African American Allele Frequency from ExAC |
| 103 - ExAC_AF_AMR | American Allele Frequency from ExAC |
| 104 - ExAC_AF_EAS | East Asian Allele Frequency from ExAC |
| 105 - ExAC_AF_FIN | Finnish Allele Frequency from ExAC |
| 106 - ExAC_AF_NFE | Non-Finnish European Allele Frequency from ExAC |
| 107 - ExAC_AF_OTH | Other Allele Frequency from ExAC |
| 108 - ExAC_AF_SAS | South Asian Allele Frequency from ExAC |
| 109 - GENE_PHENO | Indicates if gene that the variant maps to is associated with a phenotype, disease or trait (0, 1, or null) |
| 110 - FILTER | Copied from input VCF |
| 111 - CONTEXT | The reference allele per VCF specs, and its five flanking base pairs |
| 112 - src_vcf_id | GDC UUID for the input VCF file |
| 113 - tumor_bam_uuid | GDC UUID for the tumor bam file |
| 114 - normal_bam_uuid | GDC UUID for the normal bam file |
| 115 - case_id | GDC UUID for the case |
| 116 - GDC_FILTER | GDC filters applied universally across all MAFs |
| 117 - COSMIC | Overlapping COSMIC variants |
| 118 - MC3_Overlap| Indicates whether this region overlaps with an MC3 variant for the same sample pair |
| 119 - GDC_Validation_Status | GDC implementation of validation checks. See notes section (#5) below for details |
| 120 - GDC_Valid_Somatic | True or False (not in somatic MAF) |
| 121 - vcf_region | Colon separated string containing the CHROM, POS, ID, REF, and ALT columns from the VCF file (e.g., chrZ:20:rs1234:A:T) (not in somatic MAF) |
| 122 - vcf_info | INFO column from VCF (not in somatic MAF) |
| 123 - vcf_format | FORMAT column from VCF (not in somatic MAF) |
| 124 - vcf_tumor_gt | Tumor sample genotype column from VCF (not in somatic MAF) |
| 125 - vcf_normal_gt | Normal sample genotype column from VCF (not in somatic MAF) |  

### Notes About GDC MAF Implementation

1. Column #4 __NCBI_Build__ is GRCh38 by default
2. Column #32 __Sequencer__ includes the sequencers used. If different sequencers were used to generate normal and tumor data, the normal sequencer is listed first.
3. Column #60 VEP name "STRAND" is changed to __TRANSCRIPT_STRAND__ to avoid confusion with Column#8 "Strand"
4. Column #122-125 __vcf_info, vcf_format, vcf_tumor_gt, and vcf_normal_gt__ are the corresponding columns from the VCF files. Including them facilitates parsing specific variant information.
5. Column #119 __GDC_Validation_Status__: GDC also collects TCGA validation sequences. It compares these with variants derived from Next-Generation Sequencing data from the same sample and populates the comparison result in "GDC_Validation_Status".
    * "Valid", if the alternative allele(s) in the tumor validation sequence is(are) the same as GDC variant call
    * "Invalid", if none of the alternative allele(s) in the tumor validation sequence is the same as GDC variant call
    * "Inconclusive" if two alternative allele exists, and one matches while the other does not
    * "Unknown" if no validation sequence exists
6. Column #120 __GDC_Valid_Somatic__ is TRUE if GDC_Validation_Status is "Valid" and the variant is "Somatic" in validation calls.  It is FALSE if these criteria are not met


---
# DAVE Release Notes

## Data Release UAT

* __GDC Product__: GDC DAVE Tools
* __Release Date__: April 18, 2017

### New updates

* None

### Bugs Fixed Since Last Release

* None

### Known Issues and Workarounds

* Features requiring login are disabled
* Some standard GDC Data Portal features are unavailable: download manifest, download clinical/biospecimen metadata, export tables, bam slicing
* Only a single consequence for a mutation is shown on the case, project, and gene pages.  Multiple consequences can be viewed on the mutation entity page
* Some queries may exhibit longer than expected load times
* Mutation data is only available for TCGA projects
* The number of cases displayed/visible on the portal is less than what is in the downloadable MAF files <!--SV-624-->
* Cancer Distribution Plot displays incorrect static title <!--SV-614-->